diff -uNr poco/Data/MySQL/CMakeLists.txt poco_changed/Data/MySQL/CMakeLists.txt
--- poco/Data/MySQL/CMakeLists.txt	2021-06-28 12:16:01.000000000 +0200
+++ poco_changed/Data/MySQL/CMakeLists.txt	2021-10-14 14:53:52.605311071 +0200
@@ -21,7 +21,7 @@
 	DEFINE_SYMBOL MySQL_EXPORTS
 )
 
-target_link_libraries(DataMySQL PUBLIC Poco::Data MySQL::client)
+target_link_libraries(DataMySQL PUBLIC Poco::Data mariadb)
 target_include_directories(DataMySQL
 	PUBLIC
 		$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
