--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -835,9 +835,6 @@ if(NOT MSVC)
       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-noexcept-type") # Added for gtest
     endif()
   endif()
-  if(NOT WDDM_LINUX)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
-  endif()
   if(USE_SANITIZE_UB)
     check_cxx_compiler_flag(-fsanitize=undefined COMPILER_SUPPORTS_UNDEFINED_BEHAVIOR_SANITIZER)
     if(COMPILER_SUPPORTS_UNDEFINED_BEHAVIOR_SANITIZER)
