From 138600a5a7df32772db1a81fd2882886818c8076 Mon Sep 17 00:00:00 2001
From: Jeff Whitaker <jswhit@fastmail.fm>
Date: Thu, 24 May 2018 16:45:17 -0600
Subject: [PATCH] fix for failing test

---
 test/tst_cdf5.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/tst_cdf5.py b/test/tst_cdf5.py
index 1b46c91a..175f9b53 100644
--- a/test/tst_cdf5.py
+++ b/test/tst_cdf5.py
@@ -4,7 +4,7 @@
 from numpy.testing import assert_array_equal
 
 FILE_NAME = tempfile.NamedTemporaryFile(suffix='.nc', delete=False).name
-dimsize = np.iinfo(np.int64).max # max unsigned 64 bit integer
+dimsize = np.iinfo(np.int32).max*2 # only allowed in CDF5
 ndim = 100
 arrdata = np.random.randint(np.iinfo(np.uint8).min,np.iinfo(np.uint8).max,size=ndim)
 
