diff -rupN core-setup.0267ad09c6f2e2a37b23b7d230ffbf9e787dd388.orig/src/corehost/build.sh core-setup.0267ad09c6f2e2a37b23b7d230ffbf9e787dd388/src/corehost/build.sh
--- core-setup.0267ad09c6f2e2a37b23b7d230ffbf9e787dd388.orig/src/corehost/build.sh	2021-03-12 17:57:36.793772783 +0100
+++ core-setup.0267ad09c6f2e2a37b23b7d230ffbf9e787dd388/src/corehost/build.sh	2021-03-12 18:00:39.490798906 +0100
@@ -1,61 +1,5 @@
 #!/usr/bin/env bash
 
-init_rid_plat()
-{
-    # Detect Distro
-    if [ $__CrossBuild == 1 ]; then
-        if [ -z $ROOTFS_DIR ]; then
-            echo "ROOTFS_DIR is not defined."
-            exit -1
-        else
-            if [ -e $ROOTFS_DIR/etc/os-release ]; then
-                source $ROOTFS_DIR/etc/os-release
-                __rid_plat="$ID.$VERSION_ID"
-                if [[ "$ID" == "alpine" ]]; then
-                    __rid_plat="linux-musl"
-                fi
-            fi
-            echo "__rid_plat is $__rid_plat"
-        fi
-    else
-        __rid_plat=""
-        if [ -e /etc/os-release ]; then
-            source /etc/os-release
-            if [[ "$ID" == "rhel" ]]; then
-                # remove the last version number
-                VERSION_ID=${VERSION_ID%.*}
-            fi
-            __rid_plat="$ID.$VERSION_ID"
-            if [[ "$ID" == "alpine" ]]; then
-                __rid_plat="linux-musl"
-            fi
-        elif [ -e /etc/redhat-release ]; then
-            local redhatRelease=$(</etc/redhat-release)
-            if [[ $redhatRelease == "CentOS release 6."* || $redhatRelease == "Red Hat Enterprise Linux Server release 6."* ]]; then
-               __rid_plat="rhel.6"
-            fi
-        fi
-    fi
-
-    if [ "$(uname -s)" == "Darwin" ]; then
-        __rid_plat=osx.10.12
-    fi
-    if [ "$(uname -s)" == "FreeBSD" ]; then
-        major_ver=`uname -U | cut -b1-2`
-        __rid_plat=freebsd.$major_ver
-    fi
-
-    if [ $__linkPortable == 1 ]; then
-        if [ "$(uname -s)" == "Darwin" ]; then
-            __rid_plat="osx"
-        elif [ "$(uname -s)" == "FreeBSD" ]; then
-            __rid_plat="freebsd"
-        else
-            __rid_plat="linux"
-        fi
-    fi
-}
-
 usage()
 {
     echo "Usage: $0 --configuration <configuration> --arch <Architecture> --hostver <Dotnet exe version> --apphostver <app host exe version> --fxrver <HostFxr library version> --policyver <HostPolicy library version> --commithash <Git commit hash> [--xcompiler <Cross C++ Compiler>]"
@@ -186,8 +130,7 @@ fi
 
 # __base_rid is the base RID that corehost is shipped for, effectively, the name of the folder in "runtimes/{__base_rid}/native/" inside the nupkgs.
 # __rid_plat is the OS portion of the RID.
-__rid_plat=
-init_rid_plat
+__rid_plat="arch"
 
 if [ -z $__rid_plat ]; then
     echo "Unknown base rid (eg.: osx.10.12, ubuntu.14.04) being targeted"
