From e31986c6523253f3160753b202e22a1c321e2b1a Mon Sep 17 00:00:00 2001
From: Sergey Linev <S.Linev@gsi.de>
Date: Thu, 12 Sep 2019 14:06:13 +0200
Subject: [PATCH] [rbrowser] add missing include ROOT-10315

---
 gui/browserv7/inc/ROOT/RBrowserItem.hxx | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gui/browserv7/inc/ROOT/RBrowserItem.hxx b/gui/browserv7/inc/ROOT/RBrowserItem.hxx
index b21bc87b4bf..e9ae97e90c0 100644
--- a/gui/browserv7/inc/ROOT/RBrowserItem.hxx
+++ b/gui/browserv7/inc/ROOT/RBrowserItem.hxx
@@ -17,6 +17,9 @@
 #ifndef ROOT7_RBrowserItem
 #define ROOT7_RBrowserItem
 
+#include <string>
+#include <vector>
+
 namespace ROOT {
 namespace Experimental {
