diff -ru spring_0.82.7.1-old//AI/Interfaces/Java/bin/jna_wrappCommands.awk spring_0.82.7.1/AI/Interfaces/Java/bin/jna_wrappCommands.awk
--- spring_0.82.7.1-old//AI/Interfaces/Java/bin/jna_wrappCommands.awk	2011-07-26 17:29:16.623158278 +0200
+++ spring_0.82.7.1/AI/Interfaces/Java/bin/jna_wrappCommands.awk	2011-07-26 17:33:00.872921328 +0200
@@ -341,7 +341,7 @@
 }
 
 # beginn of struct S*Command
-/^\struct S.*Command( \{)?/ {
+/^struct S.*Command( \{)?/ {
 
 	isInsideCmdStruct = 1;
 	ind_cmdMember = 0;
diff -ru spring_0.82.7.1-old//AI/Interfaces/Java/bin/jna_wrappEvents.awk spring_0.82.7.1/AI/Interfaces/Java/bin/jna_wrappEvents.awk
--- spring_0.82.7.1-old//AI/Interfaces/Java/bin/jna_wrappEvents.awk	2011-07-26 17:29:16.623158278 +0200
+++ spring_0.82.7.1/AI/Interfaces/Java/bin/jna_wrappEvents.awk	2011-07-26 17:32:51.009745026 +0200
@@ -473,7 +473,7 @@
 }
 
 # beginn of struct S*Event
-/^\struct S.*Event( \{)?/ {
+/^struct S.*Event( \{)?/ {
 
 	isInsideEvtStruct = 1;
 	ind_evtMember = 0;
diff -ru spring_0.82.7.1-old//AI/Wrappers/Cpp/bin/wrappCommands.awk spring_0.82.7.1/AI/Wrappers/Cpp/bin/wrappCommands.awk
--- spring_0.82.7.1-old//AI/Wrappers/Cpp/bin/wrappCommands.awk	2011-07-26 17:29:16.626491558 +0200
+++ spring_0.82.7.1/AI/Wrappers/Cpp/bin/wrappCommands.awk	2011-07-26 17:30:25.695391913 +0200
@@ -387,7 +387,7 @@
 }
 
 # beginn of struct S*Command
-/^\struct S.*Command( \{)?/ {
+/^struct S.*Command( \{)?/ {
 
 	isInsideCmdStruct = 1;
 	ind_cmdMember = 0;
diff -ru spring_0.82.7.1-old//AI/Wrappers/Cpp/bin/wrappEvents.awk spring_0.82.7.1/AI/Wrappers/Cpp/bin/wrappEvents.awk
--- spring_0.82.7.1-old//AI/Wrappers/Cpp/bin/wrappEvents.awk	2011-07-26 17:29:16.626491558 +0200
+++ spring_0.82.7.1/AI/Wrappers/Cpp/bin/wrappEvents.awk	2011-07-26 17:30:13.942245701 +0200
@@ -366,7 +366,7 @@
 }
 
 # beginn of struct S*Event
-/^\struct S.*Event( \{)?/ {
+/^struct S.*Event( \{)?/ {
 
 	isInsideEvtStruct = 1;
 	ind_evtMember = 0;
