diff --git a/lib/src/facts/resolvers/az_resolver.cc b/lib/src/facts/resolvers/az_resolver.cc
index f789f4925..5f2d778b9 100644
--- a/lib/src/facts/resolvers/az_resolver.cc
+++ b/lib/src/facts/resolvers/az_resolver.cc
@@ -47,7 +47,7 @@ namespace facter { namespace facts { namespace resolvers {
     #endif
 #endif
 
-    struct json_event_handler
+    struct json_event_handler : public rapidjson::BaseReaderHandler<>
     {
         explicit json_event_handler(map_value& root) :
             _initialized(false),
