diff -rupN bamf.orig/configure.ac bamf/configure.ac
--- bamf.orig/configure.ac	2020-02-03 18:51:04.813711221 +0100
+++ bamf/configure.ac	2020-02-03 18:52:11.374267203 +0100
@@ -110,18 +110,6 @@ fi
 GTK_DOC_CHECK(1.0)
 
 ###########################
-# gtester2xunit checks    #
-###########################
-
-AC_PATH_PROG([PYTHON],[python])
-AC_MSG_CHECKING(for gtester2xunit dependencies)
-if !($PYTHON -c "import libxslt, libxml2" 2> /dev/null); then
-  AC_MSG_RESULT([no])
-  AC_MSG_ERROR([You need to install python-libxslt1 and python-libxml2]);
-fi
-AC_MSG_RESULT([yes])
-
-###########################
 # Headless tests
 ###########################
 AC_ARG_ENABLE([headless-tests],


