diff --git a/src/Web/Types.hs b/src/Web/Types.hs
index 8efe669..95e75f3 100644
--- a/src/Web/Types.hs
+++ b/src/Web/Types.hs
@@ -79,11 +79,6 @@ import           Yesod.Static
 import           Theory
 
 
--- | Derived Instances to fix things
-instance Bin.Binary ZonedTime where
-  get = liftM2 ZonedTime Bin.get Bin.get
-  put (ZonedTime d tod) = Bin.put d >> Bin.put tod
-
 ------------------------------------------------------------------------------
 -- Types
 ------------------------------------------------------------------------------
