diff --git a/src/debug.c b/src/debug.c
index 9306c763d..ddbca4019 100644
--- a/src/debug.c
+++ b/src/debug.c
@@ -150,7 +150,7 @@ static void debug_sigHandler( int sig, siginfo_t *info, void *unused )
          debug_sigCodeToStr(info->si_signo, info->si_code) );
    for (i=0; i<num; i++) {
       if (abfd != NULL)
-         debug_translateAddress(symbols[i], (bfd_vma) (bfd_hostptr_t) buf[i]);
+         debug_translateAddress(symbols[i], (bfd_vma) (uintptr_t) buf[i]);
       else
          DEBUG("   %s", symbols[i]);
    }
