diff -ru 0ad-r11863-alpha/source/graphics/TextureManager.cpp 0ad-r11863-alpha-patched/source/graphics/TextureManager.cpp
--- 0ad-r11863-alpha/source/graphics/TextureManager.cpp	2012-02-19 04:30:59.000000000 +0100
+++ 0ad-r11863-alpha-patched/source/graphics/TextureManager.cpp	2012-07-10 21:00:52.753251312 +0200
@@ -425,9 +425,9 @@
 			CTextureConverter::SettingsFile* f = GetSettingsFile(settingsPath);
 			if (f)
 				files.push_back(f);
-			p = p / *it;
+			p = p / it->wstring();
 		}
-		return m_TextureConverter.ComputeSettings(srcPath.leaf(), files);
+		return m_TextureConverter.ComputeSettings(srcPath.leaf().wstring(), files);
 	}
 
 	/**
diff -ru 0ad-r11863-alpha/source/lib/pch/pch_boost.h 0ad-r11863-alpha-patched/source/lib/pch/pch_boost.h
--- 0ad-r11863-alpha/source/lib/pch/pch_boost.h	2012-02-16 19:01:10.000000000 +0100
+++ 0ad-r11863-alpha-patched/source/lib/pch/pch_boost.h	2012-07-10 19:51:38.413009623 +0200
@@ -25,7 +25,7 @@
 
 // the following boost libraries have been included in TR1 and are
 // thus deemed usable:
-#define BOOST_FILESYSTEM_VERSION 2
+#define BOOST_FILESYSTEM_VERSION 3
 #include <boost/filesystem.hpp>
 namespace fs = boost::filesystem;
 #include <boost/shared_ptr.hpp>
diff -ru 0ad-r11863-alpha/source/simulation2/components/ICmpAIManager.cpp 0ad-r11863-alpha-patched/source/simulation2/components/ICmpAIManager.cpp
--- 0ad-r11863-alpha/source/simulation2/components/ICmpAIManager.cpp	2011-05-25 12:39:13.000000000 +0200
+++ 0ad-r11863-alpha-patched/source/simulation2/components/ICmpAIManager.cpp	2012-07-10 20:55:52.523233843 +0200
@@ -53,7 +53,7 @@
 		fs::wpath components = pathname.string();
 		fs::wpath::iterator it = components.begin();
 		std::advance(it, 2);
-		std::wstring dirname = *it;
+		std::wstring dirname = it->wstring();
 
 		CScriptValRooted ai;
 		self->m_ScriptInterface.Eval("({})", ai);
