commit 3fcc30f8af64b7eaadeed007301954bd69a23d38
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Mon Oct 15 07:06:14 2018 +0800

    Fix compatibility with Cabal 2.4
    
    Fixes build with GHC 8.6

diff --git a/Setup.hs b/Setup.hs
index 98dbe34..3fe4081 100644
--- a/Setup.hs
+++ b/Setup.hs
@@ -13,8 +13,13 @@ main :: IO ()
 main = defaultMainWithHooks hk
  where
  hk = simpleUserHooks { buildHook = \pd lbi uh bf -> do
+#if MIN_VERSION_Cabal(2,4,0)
+                                        let ccProg = Program "gcc" undefined undefined undefined undefined
+                                            hcProg = Program "ghc" undefined undefined undefined undefined
+#else
                                         let ccProg = Program "gcc" undefined undefined undefined
                                             hcProg = Program "ghc" undefined undefined undefined
+#endif
                                             mConf  = lookupProgram ccProg (withPrograms lbi)
                                             hcConf = lookupProgram hcProg (withPrograms lbi)
                                             err = error "Could not determine C compiler"
