diff -wbBur mg-20140306/Makefile mg-20140306.my/Makefile
--- mg-20140306/Makefile	2014-02-20 02:30:17.000000000 +0400
+++ mg-20140306.my/Makefile	2014-03-13 13:12:46.628940661 +0400
@@ -15,6 +15,9 @@
 #
 CFLAGS+=-Wall -DFKEYS -DREGEX -DXKEYS
 
+CFLAGS += -I/usr/include/clens -isystem /usr/include/bsd -DLIBBSD_OVERLAY -D__dead -O2
+LDADD+=<------>-lcurses -lutil -lclens -lbsd
+
 SRCS=	autoexec.c basic.c bell.c buffer.c cinfo.c dir.c display.c \
 	echo.c extend.c file.c fileio.c funmap.c help.c kbd.c keymap.c \
 	line.c macro.c main.c match.c modes.c paragraph.c random.c \
