--- etc/tinyproxy.conf.in	2010-03-03 04:37:24.000000000 -0500
+++ etc/tinyproxy.conf.in.arch	2014-04-11 15:43:53.340725405 -0400
@@ -12,8 +12,8 @@
 # as the root user. Either the user or group name or the UID or GID
 # number may be used.
 #
-User nobody
-Group nobody
+User tinyproxy
+Group tinyproxy
 
 #
 # Port: Specify the port which tinyproxy will listen on.  Please note
@@ -27,7 +27,7 @@
 # only one. If this is commented out, tinyproxy will bind to all
 # interfaces present.
 #
-#Listen 192.168.0.1
+Listen 127.0.0.1
 
 #
 # Bind: This allows you to specify which interface will be used for
@@ -99,7 +99,7 @@
 # option must not be enabled if the Logfile directive is being used.
 # These two directives are mutually exclusive.
 #
-#Syslog On
+Syslog On
 
 #
 # LogLevel: 
