diff --git a/plugins/kdecoration/chameleon.h b/plugins/kdecoration/chameleon.h
index 3d164ca..a716a7a 100644
--- a/plugins/kdecoration/chameleon.h
+++ b/plugins/kdecoration/chameleon.h
@@ -32,6 +32,7 @@
 #include <QDir>
 #include <QSettings>
 #include <QScreen>
+#include <QPainterPath>
 
 class Settings;
 class ChameleonWindowTheme;
diff --git a/plugins/kwineffects/scissor-window/scissorwindow.cpp b/plugins/kwineffects/scissor-window/scissorwindow.cpp
index eb17483..faa898e 100644
--- a/plugins/kwineffects/scissor-window/scissorwindow.cpp
+++ b/plugins/kwineffects/scissor-window/scissorwindow.cpp
@@ -28,6 +28,7 @@
 #include <kwingltexture.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QExplicitlySharedDataPointer>
 #include <QSignalBlocker>
 
