diff --git a/viewer/frame/toptoolbar.cpp b/viewer/frame/toptoolbar.cpp
index 545ab1fb..89648d2b 100644
--- a/viewer/frame/toptoolbar.cpp
+++ b/viewer/frame/toptoolbar.cpp
@@ -37,6 +37,7 @@
 #include <QImageReader>
 #include <QLabel>
 #include <QPainter>
+#include <QPainterPath>
 #include <QProcess>
 #include <QResizeEvent>
 #include <QShortcut>
diff --git a/viewer/module/view/contents/imageinfowidget.cpp b/viewer/module/view/contents/imageinfowidget.cpp
index ec6e8b91..623734b4 100644
--- a/viewer/module/view/contents/imageinfowidget.cpp
+++ b/viewer/module/view/contents/imageinfowidget.cpp
@@ -34,6 +34,7 @@
 #include <QScrollBar>
 #include <QString>
 #include <QtDebug>
+#include <QPainterPath>
 
 namespace {
 
diff --git a/viewer/module/view/contents/ttbcontent.cpp b/viewer/module/view/contents/ttbcontent.cpp
index 07d6b211..cc4718be 100644
--- a/viewer/module/view/contents/ttbcontent.cpp
+++ b/viewer/module/view/contents/ttbcontent.cpp
@@ -35,6 +35,7 @@
 #include <QHBoxLayout>
 #include <QTimer>
 #include <QtMath>
+#include <QPainterPath>
 
 #define TTB 0
 
diff --git a/viewer/widgets/popupmenustyle.cpp b/viewer/widgets/popupmenustyle.cpp
index ddb509cd..0fd91f73 100644
--- a/viewer/widgets/popupmenustyle.cpp
+++ b/viewer/widgets/popupmenustyle.cpp
@@ -19,6 +19,7 @@
 #include <QPixmap>
 #include <QPixmapCache>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStyleOption>
 #include <QStyleOptionMenuItem>
 
diff --git a/viewer/widgets/thumbnaildelegate.cpp b/viewer/widgets/thumbnaildelegate.cpp
index 64bd65ef..1a09702d 100644
--- a/viewer/widgets/thumbnaildelegate.cpp
+++ b/viewer/widgets/thumbnaildelegate.cpp
@@ -23,6 +23,7 @@
 #include <QHBoxLayout>
 #include <QLineEdit>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPixmapCache>
 #include <QStandardItemModel>
 #include <QThread>
diff --git a/viewer/widgets/thumbnaillistview.cpp b/viewer/widgets/thumbnaillistview.cpp
index 0681c9d9..1f2f3940 100644
--- a/viewer/widgets/thumbnaillistview.cpp
+++ b/viewer/widgets/thumbnaillistview.cpp
@@ -30,6 +30,7 @@
 #include <QMutex>
 #include <QPaintEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPen>
 #include <QScrollBar>
 #include <QStandardItemModel>
