diff --git a/CMakeLists.txt b/CMakeLists.txt
index d446e0b465..07fd13db85 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -831,8 +831,17 @@ if(NOT ENABLE_QT)
   set(USE_MGBA 0)
 endif()
 if(USE_MGBA)
-  message(STATUS "Using static libmgba from Externals")
-  add_subdirectory(Externals/mGBA)
+  if (SYSTEM_LIBMGBA)
+    find_library(MGBA_LIBRARY mgba)
+    add_library(mgba ${MGBA_LIBRARY})
+    add_library(mGBA::mgba ALIAS mgba)
+    set_target_properties(mgba PROPERTIES LINKER_LANGUAGE C)
+    message(STATUS "Found libmgba: ${MGBA_LIBRARY}")
+    message(STATUS "Using shared libmgba")
+  else()
+    message(STATUS "Using static libmgba from Externals")
+    add_subdirectory(Externals/mGBA)
+  endif()
 endif()

 find_package(SYSTEMD)
