--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -773,18 +773,10 @@ if(NOT DISABLE_WX AND NOT ANDROID)
 		execute_process(COMMAND lsb_release -c -s
 			OUTPUT_VARIABLE DIST_NAME
 			ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
-		if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION}
-				VERSION_EQUAL 2.8.2 OR "${DIST_NAME}" STREQUAL "natty")
-			check_lib(GTK2 gtk+-2.0 gtk+-2.0 gtk.h REQUIRED)
-		else()
-			include(FindGTK2)
-			if(GTK2_FOUND)
-				include_directories(${GTK2_INCLUDE_DIRS})
-				list(APPEND LIBS ${GTK2_LIBRARIES})
-			else()
-				message(FATAL_ERROR "GTK is required to build the WX UI. Please install the GTK development libraries.")
-			endif()
-		endif()
+
+		check_lib(GTK3 gtk+-3.0 gtk.h REQUIRED)
+		include_directories(${GTK3_INCLUDE_DIRS})
+		list(APPEND LIBS ${GTK3_LIBRARIES})
 	endif()
 
 	if(wxWidgets_FOUND)
