diff --git a/src/plugins/avcodec/avcodec.c b/src/plugins/avcodec/avcodec.c
index 23fd4615..506d7cd5 100644
--- a/src/plugins/avcodec/avcodec.c
+++ b/src/plugins/avcodec/avcodec.c
@@ -477,8 +477,12 @@ xmms_avcodec_internal_decode_some (xmms_avcodec_data_t *data)
 	/* clear buffers and reset fields to defaults */
 	av_frame_unref (data->read_out_frame);
 
-	bytes_read = avcodec_decode_audio4 (
-		data->codecctx, data->read_out_frame, &got_frame, &packet);
+        bytes_read = avcodec_send_packet(data->codecctx, &packet);
+        if (bytes_read >= 0 || bytes_read == AVERROR_EOF) {
+            bytes_read = avcodec_receive_frame(data->codecctx, data->read_out_frame);
+            if (bytes_read >= 0 || bytes_read == AVERROR(EAGAIN) || bytes_read == AVERROR_EOF)
+                got_frame = 1;
+        }
 
 	/* The DTS decoder of ffmpeg is buggy and always returns
 	 * the input buffer length, get frame length from header */
diff --git a/src/plugins/avcodec/wscript b/src/plugins/avcodec/wscript
index 00b182b2..f204fcf3 100644
--- a/src/plugins/avcodec/wscript
+++ b/src/plugins/avcodec/wscript
@@ -40,13 +40,6 @@ def plugin_configure(conf):
     conf.check_cc(header_name="avcodec.h", uselib="avcodec", type="cshlib", mandatory=False)
     conf.check_cc(header_name="libavcodec/avcodec.h", uselib="avcodec", type="cshlib", mandatory=False)
 
-    # mandatory function avcodec_decode_audio4 available since
-    # * ffmpeg: commit e4de716, lavc 53.40.0, release 0.9
-    # * libav: commit 0eea212, lavc 53.25.0, release 0.8
-    conf.check_cc(fragment=avcodec_decode_audio4_fragment, uselib="avcodec",
-                  uselib_store="avcodec_decode_audio4",
-                  msg="Checking for function avcodec_decode_audio4", mandatory=True)
-
     # non-mandatory function avcodec_free_frame since
     # * ffmpeg: commit 46a3595, lavc 54.59.100, release 1.0
     # * libav: commit a42aada, lavc 54.28.0, release 9
