diff --git a/cmake/FindQXlsx.cmake b/cmake/FindQXlsx.cmake
index f041ef537..1954b7d38 100644
--- a/cmake/FindQXlsx.cmake
+++ b/cmake/FindQXlsx.cmake
@@ -4,9 +4,9 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #=============================================================================
 
-find_library(QXLSX_LIBRARIES NAMES QXlsx)
+find_library(QXLSX_LIBRARIES NAMES QXlsx QXlsxQt5)
 
-find_path(QXLSX_INCLUDE_DIR xlsxdocument.h)
+find_path(QXLSX_INCLUDE_DIR xlsxdocument.h PATH_SUFFIXES QXlsx QXlsxQt5)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(QXlsx
