diff --git a/src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in b/src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in
index 6914f6de1..d13bd1f95 100644
--- a/src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in
+++ b/src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff --git a/src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in b/src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in
index 0c28330ff..4383a5abf 100644
--- a/src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in
+++ b/src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff --git a/src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in b/src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in
index 400bbaa07..9570d184b 100644
--- a/src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in
+++ b/src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff --git a/src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in b/src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in
index 88d2677f6..5b552b10a 100644
--- a/src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in
+++ b/src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff --git a/src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in b/src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in
index f63179bb3..daea9d134 100644
--- a/src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in
+++ b/src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff --git a/src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in b/src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in
index 14d0ef013..3f7798ec9 100644
--- a/src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in
+++ b/src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff --git a/src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in b/src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in
index b692f6912..dee79e7e5 100644
--- a/src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in
+++ b/src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff --git a/src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in b/src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in
index 3bf129cce..66dfaaf7f 100644
--- a/src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in
+++ b/src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff --git a/src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in b/src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in
index 22d6af4d8..cd0b048bb 100644
--- a/src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in
+++ b/src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff --git a/src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in b/src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in
index ac283c3c0..29c78ddbc 100644
--- a/src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in
+++ b/src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff --git a/src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in b/src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in
index 531f250b8..e687b4bfc 100644
--- a/src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in
+++ b/src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff --git a/src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in b/src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in
index 3581ead64..60d1d3e49 100644
--- a/src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in
+++ b/src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff --git a/src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in b/src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in
index 76f4683ea..f7d292439 100644
--- a/src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in
+++ b/src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff --git a/src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in b/src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in
index b9513629c..d65cafc22 100644
--- a/src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in
+++ b/src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
