diff --color -aur pytorch-1.10.0-old/torch/csrc/jit/frontend/ir_emitter.cpp pytorch-1.10.0-new/torch/csrc/jit/frontend/ir_emitter.cpp
--- pytorch-1.10.0-old/torch/csrc/jit/frontend/ir_emitter.cpp	2021-10-26 01:41:27.453059792 +0300
+++ pytorch-1.10.0-new/torch/csrc/jit/frontend/ir_emitter.cpp	2021-10-26 02:00:09.783068924 +0300
@@ -1678,7 +1678,7 @@
               << "Union type annotation `" << type_hint->repr_str()
               << "` can hold " << vector_repr.str() << ", but none of "
               << "those list types can hold the types of the given dict"
-              << " elements, which were unified to " << candidate->repr_str();
+              << " elements, which were unified to " << (*unified_value_type)->repr_str();
         } else {
           refined_type_hint = candidate;
         }
