From 7a6da26be6a830f4e1255091952cc0a1326a4520 Mon Sep 17 00:00:00 2001
From: Thomas Goyne <plorkyeran@aegisub.org>
Date: Fri, 8 Apr 2016 12:14:28 -0700
Subject: [PATCH] Fix compilation on X11-using platforms

---
 src/ffmpegsource_common.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/ffmpegsource_common.h b/src/ffmpegsource_common.h
index 5f6df7cfa..383d5baa2 100644
--- a/src/ffmpegsource_common.h
+++ b/src/ffmpegsource_common.h
@@ -51,6 +51,9 @@ class FFmpegSourceProvider {
 public:
 	FFmpegSourceProvider(agi::BackgroundRunner *br);
 
+	// X11 is wonderful
+#undef None
+
 	enum class TrackSelection : int {
 		None = -1,
 		All = -2
