diff -upr a/cmake/FindWHEREAMI.cmake b/cmake/FindWHEREAMI.cmake
--- a/cmake/FindWHEREAMI.cmake	2019-07-19 10:11:10.000000000 +0200
+++ b/cmake/FindWHEREAMI.cmake	2019-08-12 14:49:04.730902701 +0200
@@ -1,6 +1,6 @@
 include(FindDependency)
-find_dependency(WHEREAMI DISPLAY "whereami" HEADERS "whereami/whereami.hpp" LIBRARIES "libwhereami.a")
+find_dependency(WHEREAMI DISPLAY "whereami" HEADERS "whereami/whereami.hpp" LIBRARIES "libwhereami.so")
 
 include(FeatureSummary)
 set_package_properties(WHEREAMI PROPERTIES DESCRIPTION "A hypervisor detection library" URL "https://github.com/puppetlabs/libwhereami")
-set_package_properties(WHEREAMI PROPERTIES PURPOSE "Reports hypervisors in use.")
\ No newline at end of file
+set_package_properties(WHEREAMI PROPERTIES PURPOSE "Reports hypervisors in use.")
