diff --git a/frame/textwidget.cpp b/frame/textwidget.cpp
index 02ada9f8..932a3b04 100644
--- a/frame/textwidget.cpp
+++ b/frame/textwidget.cpp
@@ -63,7 +63,7 @@ void TextWidget::initUI()
     m_fontFamilyLabel->setFont(ft);
     m_fontComBox = new CFontComboBox(this);
     m_fontComBox->setFont(ft);
-    m_fontComBox->setFontFilters(DFontComboBox::AllFonts);
+    m_fontComBox->setFontFilters(QFontComboBox::AllFonts);
 
     m_fontComBox->setFixedSize(QSize(240, 36));
     m_fontComBox->setCurrentIndex(0);
