From 209ef11bb0e63681e54a9d74ea635bc6048450a3 Mon Sep 17 00:00:00 2001
From: Daniel Roethlisberger <daniel@roe.ch>
Date: Sat, 27 Oct 2018 00:24:01 +0200
Subject: [PATCH] Add session to testreqs and make server.pem order-only
 dependency

---
 extra/pki/GNUmakefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/extra/pki/GNUmakefile b/extra/pki/GNUmakefile
index 8053809..c23ae23 100644
--- a/extra/pki/GNUmakefile
+++ b/extra/pki/GNUmakefile
@@ -17,7 +17,7 @@ NO_SSL2:=	$(shell $(OPENSSL) s_server -h 2>&1|awk '/-no_ssl2/ {print $$1}')
 
 all: rsa dsa ec targets server pwd
 
-testreqs: rsa targets server
+testreqs: rsa targets server session
 
 session: session.pem
 
@@ -112,7 +112,7 @@ targets/wildcard.roe.ch.pem: rsa.crt
 		targets/wildcard.roe.ch.crt
 
 # localhost network connectivity is required
-session.pem: server.pem
+session.pem: | server.pem
 	$(OPENSSL) s_server -accept 46143 -cert server.pem -quiet $(NO_SSL2) & \
 		pid=$$! ; \
 		sleep 1 ; \
