diff -aur pythia8303-old/plugins/python/Makefile pythia8303-new/plugins/python/Makefile
--- pythia8303-old/plugins/python/Makefile	2020-09-01 14:37:05.000000000 +0300
+++ pythia8303-new/plugins/python/Makefile	2020-09-25 18:28:43.852513423 +0300
@@ -27,7 +27,7 @@
 LOCAL_INCLUDE=include
 LOCAL_MKDIRS:=$(shell mkdir -p $(LOCAL_TMP) $(TOP_LIB))
 OBJ_COMMON=-MD $(CXX_COMMON) -Iinclude $(PYTHON_INCLUDE) -w -fpermissive
-LIB_COMMON=-Wl,-rpath,$(TOP_LIB) -ldl $(GZIP_LIB)
+LIB_COMMON=-ldl $(GZIP_LIB)
 PYTHIA=$(TOP_LIB)/libpythia8$(LIB_SUFFIX)
 
 # Determine the headers.
