commit cbb5568c62304fe0b9818dea18dcb1e8f4572866
Author: Anatol Pomozov <anatol.pomozov@gmail.com>
Date:   Tue Sep 21 09:46:53 2021 -0700

    Arch Linux specific fixes

diff --git a/cmake/flags.cmake b/cmake/flags.cmake
index aad71a397..6dcb6d44c 100644
--- a/cmake/flags.cmake
+++ b/cmake/flags.cmake
@@ -84,11 +84,9 @@ function(setupBuildFlags)
       -Woverloaded-virtual
       -Wnon-virtual-dtor
       -Weffc++
-      -stdlib=libc++
     )
 
     set(posix_cxx_link_options
-      -stdlib=libc++
       -ldl
     )
 
@@ -179,7 +177,6 @@ function(setupBuildFlags)
       )
 
       set(linux_cxx_link_libraries
-        c++abi
         rt
         dl
       )
@@ -228,7 +225,6 @@ function(setupBuildFlags)
       )
 
       set(macos_cxx_link_options
-        -stdlib=libc++
         -lresolv
       )
 
@@ -237,7 +233,6 @@ function(setupBuildFlags)
         cups
         bsm
         xar
-        c++abi
         "-framework AppKit"
         "-framework Foundation"
         "-framework CoreServices"
diff --git a/libraries/cmake/source/augeas/generated/linux/x86_64/config/config.h b/libraries/cmake/source/augeas/generated/linux/x86_64/config/config.h
index 1c69ad5d1..3623cd58a 100644
--- a/libraries/cmake/source/augeas/generated/linux/x86_64/config/config.h
+++ b/libraries/cmake/source/augeas/generated/linux/x86_64/config/config.h
@@ -914,9 +914,6 @@
 /* Define to 1 if you have the <ws2tcpip.h> header file. */
 /* #undef HAVE_WS2TCPIP_H */
 
-/* Define to 1 if you have the <xlocale.h> header file. */
-#define HAVE_XLOCALE_H 1
-
 /* Define to 1 if the system has the type `_Bool'. */
 #define HAVE__BOOL 1
 
diff --git a/libraries/cmake/source/augeas/gnulib/generated/linux/x86_64/config/config.h b/libraries/cmake/source/augeas/gnulib/generated/linux/x86_64/config/config.h
index 1c69ad5d1..3623cd58a 100644
--- a/libraries/cmake/source/augeas/gnulib/generated/linux/x86_64/config/config.h
+++ b/libraries/cmake/source/augeas/gnulib/generated/linux/x86_64/config/config.h
@@ -914,9 +914,6 @@
 /* Define to 1 if you have the <ws2tcpip.h> header file. */
 /* #undef HAVE_WS2TCPIP_H */
 
-/* Define to 1 if you have the <xlocale.h> header file. */
-#define HAVE_XLOCALE_H 1
-
 /* Define to 1 if the system has the type `_Bool'. */
 #define HAVE__BOOL 1
 
diff --git a/libraries/cmake/source/augeas/gnulib/generated/linux/x86_64/lib/locale.h b/libraries/cmake/source/augeas/gnulib/generated/linux/x86_64/lib/locale.h
index 4f9baece2..0230ce625 100644
--- a/libraries/cmake/source/augeas/gnulib/generated/linux/x86_64/lib/locale.h
+++ b/libraries/cmake/source/augeas/gnulib/generated/linux/x86_64/lib/locale.h
@@ -48,11 +48,6 @@
 /* NetBSD 5.0 mis-defines NULL.  */
 #include <stddef.h>
 
-/* Mac OS X 10.5 defines the locale_t type in <xlocale.h>.  */
-#if 1
-# include <xlocale.h>
-#endif
-
 /* The definitions of _GL_FUNCDECL_RPL etc. are copied here.  */
 /* C++ compatible function declaration macros.
    Copyright (C) 2010-2019 Free Software Foundation, Inc.
diff --git a/libraries/cmake/source/libarchive/config/linux/aarch64/config.h b/libraries/cmake/source/libarchive/config/linux/aarch64/config.h
index ec8a7c01e..52465a38f 100644
--- a/libraries/cmake/source/libarchive/config/linux/aarch64/config.h
+++ b/libraries/cmake/source/libarchive/config/linux/aarch64/config.h
@@ -361,7 +361,7 @@ typedef uint64_t uintmax_t;
 /* #undef HAVE_ACL_SET_FILE */
 
 /* Define to 1 if you have the `arc4random_buf' function. */
-/* #undef HAVE_ARC4RANDOM_BUF */
+#define HAVE_ARC4RANDOM_BUF 1
 
 /* Define to 1 if you have the <attr/xattr.h> header file. */
 /* #undef HAVE_ATTR_XATTR_H */
diff --git a/libraries/cmake/source/libarchive/config/linux/x86_64/config.h b/libraries/cmake/source/libarchive/config/linux/x86_64/config.h
index ec8a7c01e..52465a38f 100644
--- a/libraries/cmake/source/libarchive/config/linux/x86_64/config.h
+++ b/libraries/cmake/source/libarchive/config/linux/x86_64/config.h
@@ -361,7 +361,7 @@ typedef uint64_t uintmax_t;
 /* #undef HAVE_ACL_SET_FILE */
 
 /* Define to 1 if you have the `arc4random_buf' function. */
-/* #undef HAVE_ARC4RANDOM_BUF */
+#define HAVE_ARC4RANDOM_BUF 1
 
 /* Define to 1 if you have the <attr/xattr.h> header file. */
 /* #undef HAVE_ATTR_XATTR_H */
diff --git a/libraries/cmake/source/libdpkg/config/x86_64/config.h b/libraries/cmake/source/libdpkg/config/x86_64/config.h
index eafef658b..51c4e6a91 100644
--- a/libraries/cmake/source/libdpkg/config/x86_64/config.h
+++ b/libraries/cmake/source/libdpkg/config/x86_64/config.h
@@ -296,9 +296,6 @@
 /* Define to 1 if 'WCOREDUMP' is declared in <sys/wait.h> */
 #define HAVE_WCOREDUMP 1
 
-/* Define to 1 if you have the <xlocale.h> header file. */
-#define HAVE_XLOCALE_H 1
-
 /* Define to 1 if you have the `__cxa_pure_virtual' function. */
 /* #undef HAVE___CXA_PURE_VIRTUAL */
 
diff --git a/libraries/cmake/source/libmagic/config/linux/x86_64/config.h b/libraries/cmake/source/libmagic/config/linux/x86_64/config.h
index 759ba47d2..4f8fbe63c 100644
--- a/libraries/cmake/source/libmagic/config/linux/x86_64/config.h
+++ b/libraries/cmake/source/libmagic/config/linux/x86_64/config.h
@@ -249,9 +249,6 @@
 /* Define to 1 if `vfork' works. */
 #define HAVE_WORKING_VFORK 1
 
-/* Define to 1 if you have the <xlocale.h> header file. */
-#define HAVE_XLOCALE_H 1
-
 /* Define to 1 if you have the <zlib.h> header file. */
 #define HAVE_ZLIB_H 1
 
diff --git a/libraries/cmake/source/thrift/CMakeLists.txt b/libraries/cmake/source/thrift/CMakeLists.txt
index 0b224a159..6f0d824dd 100644
--- a/libraries/cmake/source/thrift/CMakeLists.txt
+++ b/libraries/cmake/source/thrift/CMakeLists.txt
@@ -83,13 +83,6 @@ function(thriftMain)
     "${library_root}/src/thrift/transport/TWebSocketServer.h"
   )
 
-  if(PLATFORM_POSIX)
-    set(tsocket_pool_compile_options "-include;random_shuffle.h")
-
-  elseif(PLATFORM_WINDOWS)
-    set(tsocket_pool_compile_options "/FIrandom_shuffle.h")
-  endif()
-
   set_source_files_properties("${library_root}/src/thrift/transport/TSocketPool.cpp" PROPERTIES
     COMPILE_OPTIONS
       "${tsocket_pool_compile_options}"
diff --git a/osquery/core/shutdown.cpp b/osquery/core/shutdown.cpp
index 522bde887..9812ebf44 100644
--- a/osquery/core/shutdown.cpp
+++ b/osquery/core/shutdown.cpp
@@ -11,6 +11,7 @@
 #include <osquery/logger/data_logger.h>
 
 #include <atomic>
+#include <condition_variable>
 #include <mutex>
 #include <string>
 
diff --git a/osquery/logger/logger.cpp b/osquery/logger/logger.cpp
index ed8391f48..9702f203c 100644
--- a/osquery/logger/logger.cpp
+++ b/osquery/logger/logger.cpp
@@ -13,6 +13,7 @@
 
 #include <algorithm>
 #include <future>
+#include <iomanip>
 #include <optional>
 #include <queue>
 #include <thread>
diff --git a/osquery/tables/networking/CMakeLists.txt b/osquery/tables/networking/CMakeLists.txt
index 9f7f64578..35b44accb 100644
--- a/osquery/tables/networking/CMakeLists.txt
+++ b/osquery/tables/networking/CMakeLists.txt
@@ -30,6 +30,10 @@ function(generateOsqueryTablesNetworking)
       posix/interfaces.cpp
       posix/utils.cpp
     )
+
+    list(APPEND platform_deps
+      resolv
+    )
   endif()
 
   if(DEFINED PLATFORM_LINUX)
diff --git a/osquery/tables/system/linux/sysctl_utils.cpp b/osquery/tables/system/linux/sysctl_utils.cpp
index 1ff3e0b00..b66ecc79e 100644
--- a/osquery/tables/system/linux/sysctl_utils.cpp
+++ b/osquery/tables/system/linux/sysctl_utils.cpp
@@ -7,8 +7,6 @@
  * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
  */
 
-#include <sys/sysctl.h>
-
 #include <boost/algorithm/string/trim.hpp>
 
 #include <osquery/core/tables.h>
@@ -73,10 +71,8 @@ void genControlInfo(int* oid,
   // Get control size
   size_t response_size = CTL_MAX_VALUE;
   char response[CTL_MAX_VALUE + 1] = {0};
-  if (sysctl(oid, oid_size, response, &response_size, 0, 0) != 0) {
-    // Cannot request MIB data.
-    return;
-  }
+  // Cannot request MIB data.
+  return;
 
   // Data is output, but no way to determine type (long, int, string, struct).
   Row r;
diff --git a/osquery/tables/system/posix/augeas.cpp b/osquery/tables/system/posix/augeas.cpp
index fb09411d8..615f7adea 100644
--- a/osquery/tables/system/posix/augeas.cpp
+++ b/osquery/tables/system/posix/augeas.cpp
@@ -35,7 +35,7 @@ FLAG(string,
 #else
 FLAG(string,
      augeas_lenses,
-     "/opt/osquery/share/osquery/lenses",
+     "/usr/share/osquery/lenses",
      "Directory that contains augeas lenses files");
 #endif
 
diff --git a/osquery/tables/system/posix/openssl_utils.cpp b/osquery/tables/system/posix/openssl_utils.cpp
index 9d4734c3c..37bb4857f 100644
--- a/osquery/tables/system/posix/openssl_utils.cpp
+++ b/osquery/tables/system/posix/openssl_utils.cpp
@@ -12,7 +12,9 @@
 #include <array>
 #include <ctime>
 #include <iomanip>
+#include <memory>
 #include <sstream>
+#include <cstring>
 #include <string>
 
 #include <osquery/core/core.h>
diff --git a/osquery/tables/system/posix/sysctl_utils.h b/osquery/tables/system/posix/sysctl_utils.h
index e119f8a9e..0d4a399e4 100644
--- a/osquery/tables/system/posix/sysctl_utils.h
+++ b/osquery/tables/system/posix/sysctl_utils.h
@@ -9,8 +9,6 @@
 
 #pragma once
 
-#include <sys/sysctl.h>
-
 #include <osquery/core/tables.h>
 
 namespace osquery {
@@ -18,6 +16,8 @@ namespace tables {
 
 #define CTL_MAX_VALUE 128
 
+#define CTL_MAXNAME	12
+
 #ifndef CTL_DEBUG_MAXID
 #define CTL_DEBUG_MAXID (CTL_MAXNAME * 2)
 #endif
diff --git a/osquery/utils/config/default_paths.h b/osquery/utils/config/default_paths.h
index cda34298e..1c45718f3 100644
--- a/osquery/utils/config/default_paths.h
+++ b/osquery/utils/config/default_paths.h
@@ -26,7 +26,7 @@
 #define OSQUERY_SOCKET OSQUERY_DB_HOME
 #define OSQUERY_PIDFILE "/var/run/"
 #define OSQUERY_LOG_HOME "/var/log/osquery/"
-#define OSQUERY_CERTS_HOME "/opt/osquery/share/osquery/certs/"
+#define OSQUERY_CERTS_HOME "/usr/share/osquery/certs/"
 #elif defined(WIN32)
 #define OSQUERY_HOME "\\Program Files\\osquery\\"
 #define OSQUERY_DB_HOME OSQUERY_HOME
diff --git a/tools/deployment/linux_packaging/rpm/osqueryd.service b/tools/deployment/linux_packaging/rpm/osqueryd.service
index 6aa42752f..7bb3b3dc9 100644
--- a/tools/deployment/linux_packaging/rpm/osqueryd.service
+++ b/tools/deployment/linux_packaging/rpm/osqueryd.service
@@ -7,7 +7,7 @@ TimeoutStartSec=0
 EnvironmentFile=/etc/sysconfig/osqueryd
 ExecStartPre=/bin/sh -c "if [ ! -f $FLAG_FILE ]; then touch $FLAG_FILE; fi"
 ExecStartPre=/bin/sh -c "if [ -f $LOCAL_PIDFILE ]; then mv $LOCAL_PIDFILE $PIDFILE; fi"
-ExecStart=/opt/osquery/bin/osqueryd \
+ExecStart=/usr/bin/osqueryd \
   --flagfile $FLAG_FILE \
   --config_path $CONFIG_FILE
 Restart=on-failure
diff --git a/tools/deployment/osquery.example.conf b/tools/deployment/osquery.example.conf
index 6fd3594ea..1b2b888ca 100644
--- a/tools/deployment/osquery.example.conf
+++ b/tools/deployment/osquery.example.conf
@@ -41,19 +41,19 @@
   // There are several 'default' packs installed via
   // packages and/or Homebrew.
   //
-  // Linux:        /opt/osquery/share/osquery/packs
+  // Linux:        /usr/share/osquery/packs
   // OS X:         /var/osquery/packs
   // Homebrew:     /usr/local/share/osquery/packs
   // make install: {PREFIX}/share/osquery/packs
   //
   "packs": {
-    // "osquery-monitoring": "/opt/osquery/share/osquery/packs/osquery-monitoring.conf",
-    // "incident-response": "/opt/osquery/share/osquery/packs/incident-response.conf",
-    // "it-compliance": "/opt/osquery/share/osquery/packs/it-compliance.conf",
+    // "osquery-monitoring": "/usr/share/osquery/packs/osquery-monitoring.conf",
+    // "incident-response": "/usr/share/osquery/packs/incident-response.conf",
+    // "it-compliance": "/usr/share/osquery/packs/it-compliance.conf",
     // "osx-attacks": "/var/osquery/packs/osx-attacks.conf",
-    // "vuln-management": "/opt/osquery/share/osquery/packs/vuln-management.conf",
-    // "hardware-monitoring": "/opt/osquery/share/osquery/packs/hardware-monitoring.conf",
-    // "ossec-rootkit": "/opt/osquery/share/osquery/packs/ossec-rootkit.conf",
+    // "vuln-management": "/usr/share/osquery/packs/vuln-management.conf",
+    // "hardware-monitoring": "/usr/share/osquery/packs/hardware-monitoring.conf",
+    // "ossec-rootkit": "/usr/share/osquery/packs/ossec-rootkit.conf",
     // "windows-hardening": "C:\\Program Files\\osquery\\packs\\windows-hardening.conf",
     // "windows-attacks": "C:\\Program Files\\osquery\\packs\\windows-attacks.conf"
   },
