diff --git a/gb.pdf/src/CPdfDocument.cpp b/gb.pdf/src/CPdfDocument.cpp
index 64cc92796..6a0654bb8 100644
--- a/gb.pdf/src/CPdfDocument.cpp
+++ b/gb.pdf/src/CPdfDocument.cpp
@@ -120,7 +120,7 @@ static void return_unicode_string(const Unicode *unicode, int len)
 		gstr.append(buf, n);
 	}
 
-	GB.ReturnNewZeroString(gstr.getCString());
+	GB.ReturnNewZeroString(gstr.c_str());
 }
 
 
@@ -151,11 +151,11 @@ static void aux_return_string_info(void *_object, const char *key)
 
 		if (goo_value->hasUnicodeMarker())
 		{
-			GB.ConvString (&tmpstr,goo_value->getCString()+2,goo_value->getLength()-2,"UTF-16BE","UTF-8");
+			GB.ConvString (&tmpstr,goo_value->c_str()+2,goo_value->getLength()-2,"UTF-16BE","UTF-8");
 			GB.ReturnNewZeroString(tmpstr);		
 		}		
 		else
-			GB.ReturnNewString(goo_value->getCString(),goo_value->getLength());		
+			GB.ReturnNewString(goo_value->c_str(),goo_value->getLength());		
 	}
 	#if ! POPPLER_VERSION_0_58
 	dst.free();
@@ -194,10 +194,10 @@ static void aux_return_date_info(void *_object, const char *key)
 	{
 		goo = dst.getString();
 		if (goo->hasUnicodeMarker())
-			GB.ConvString (&datestr,goo->getCString()+2,goo->getLength()-2,"UTF-16BE","UTF-8");
+			GB.ConvString (&datestr,goo->c_str()+2,goo->getLength()-2,"UTF-16BE","UTF-8");
 		else
 		{
-			datestr = GB.NewString(goo->getCString(),goo->getLength());
+			datestr = GB.NewString(goo->c_str(),goo->getLength());
 			tofree=datestr;		
 		}
 
@@ -333,11 +333,11 @@ static char* aux_get_target_from_action(const_LinkAction *act)
 
 	if (tmp->hasUnicodeMarker())
 	{
-			GB.ConvString (&uni,tmp->getCString()+2,tmp->getLength()-2,"UTF-16BE","UTF-8");
+			GB.ConvString (&uni,tmp->c_str()+2,tmp->getLength()-2,"UTF-16BE","UTF-8");
 			vl = GB.AddString(vl, uni, 0);	
 	}	
 	else
-			vl = GB.AddString(vl,tmp->getCString(),tmp->getLength());
+			vl = GB.AddString(vl,tmp->c_str(),tmp->getLength());
 	
 
 	return vl;
@@ -1015,7 +1015,7 @@ BEGIN_METHOD(PDFPAGE_select, GB_INTEGER X; GB_INTEGER Y; GB_INTEGER W; GB_INTEGE
 		return;
 	}
 	
-	GB.ReturnNewString(str->getCString(),str->getLength());	
+	GB.ReturnNewString(str->c_str(),str->getLength());	
 	delete str;
 
 END_METHOD
@@ -1081,7 +1081,7 @@ BEGIN_PROPERTY (PDFPAGELINKDATA_parameters)
 		return;	
 	}
 
-	GB.ReturnNewZeroString(((LinkLaunch*)THIS->action)->getParams()->getCString());
+	GB.ReturnNewZeroString(((LinkLaunch*)THIS->action)->getParams()->c_str());
 
 END_PROPERTY
 
