diff -upr gambas-3.15.2.orig/gb.pdf/src/CPdfDocument.cpp gambas-3.15.2/gb.pdf/src/CPdfDocument.cpp
--- gambas-3.15.2.orig/gb.pdf/src/CPdfDocument.cpp	2020-09-17 14:12:42.000000000 +0300
+++ gambas-3.15.2/gb.pdf/src/CPdfDocument.cpp	2021-06-02 19:19:30.576503642 +0300
@@ -766,8 +766,6 @@ BEGIN_PROPERTY(PDFINDEX_is_open)
 
 	if (VPROP(GB_INTEGER))
 		item->open();
-	else
-		item->close();
 
 END_PROPERTY
 
