diff -upr facter-3.14.23.orig/lib/src/facts/glib/load_average_resolver.cc facter-3.14.23/lib/src/facts/glib/load_average_resolver.cc
--- facter-3.14.23.orig/lib/src/facts/glib/load_average_resolver.cc	2022-04-16 03:32:32.000000000 +0300
+++ facter-3.14.23/lib/src/facts/glib/load_average_resolver.cc	2022-06-13 14:38:41.793965119 +0300
@@ -1,5 +1,6 @@
 #include <internal/facts/glib/load_average_resolver.hpp>
 #include <leatherman/logging/logging.hpp>
+#include <array>
 #include <cstdlib>
 
 #ifdef __sun
