diff --git a/gdal/frmts/pdf/pdfobject.cpp b/gdal/frmts/pdf/pdfobject.cpp
index e8929aa181..9e9a76a0d2 100644
--- a/gdal/frmts/pdf/pdfobject.cpp
+++ b/gdal/frmts/pdf/pdfobject.cpp
@@ -1190,7 +1190,7 @@ GDALPDFObject* GDALPDFDictionaryPoppler::Get(const char* pszKey)
         return oIter->second;
 
 #ifdef POPPLER_0_58_OR_LATER
-    Object o = m_poDict->lookupNF(((char*)pszKey));
+    Object o = m_poDict->lookupNF(((char*)pszKey)).copy();
     if (!o.isNull())
     {
         int nRefNum = 0;
@@ -1324,7 +1324,7 @@ GDALPDFObject* GDALPDFArrayPoppler::Get(int nIndex)
         return m_v[nIndex];
 
 #ifdef POPPLER_0_58_OR_LATER
-    Object o = m_poArray->getNF(nIndex);
+    Object o = m_poArray->getNF(nIndex).copy();
     if( !o.isNull() )
     {
         int nRefNum = 0;
