diff --git a/library/ByteString/StrictBuilder/Population.hs.orig b/library/ByteString/StrictBuilder/Population.hs
index 4134009..e423daf 100644
--- a/library/ByteString/StrictBuilder/Population.hs.orig
+++ b/library/ByteString/StrictBuilder/Population.hs
@@ -16,11 +16,10 @@ instance Monoid Population where
   {-# INLINE mempty #-}
   mempty =
     Population return
-  {-# INLINE mappend #-}
-  mappend (Population leftPtrUpdate) (Population rightPtrUpdate) =
-    Population (leftPtrUpdate >=> rightPtrUpdate)
 
-instance Semigroup Population
+instance Semigroup Population where
+  Population leftPtrUpdate <> Population rightPtrUpdate =
+    Population (leftPtrUpdate >=> rightPtrUpdate)
 
 
 {-|
diff --git a/library/ByteString/StrictBuilder.hs.orig b/library/ByteString/StrictBuilder.hs
index b7f36b8..69921c0 100644
--- a/library/ByteString/StrictBuilder.hs.orig
+++ b/library/ByteString/StrictBuilder.hs
@@ -38,9 +38,6 @@ instance Monoid Builder where
   {-# INLINE mempty #-}
   mempty =
     Builder 0 mempty
-  {-# INLINE mappend #-}
-  mappend (Builder leftSize leftPopulation) (Builder rightSize rightPopulation) =
-    Builder (leftSize + rightSize) (leftPopulation <> rightPopulation)
   {-# INLINE mconcat #-}
   mconcat builders =
     Builder size population
@@ -50,7 +47,9 @@ instance Monoid Builder where
       population =
         foldMap (\(Builder _ x) -> x) builders
 
-instance Semigroup Builder
+instance Semigroup Builder where
+  Builder leftSize leftPopulation <> Builder rightSize rightPopulation =
+    Builder (leftSize + rightSize) (leftPopulation <> rightPopulation)
 
 instance IsString Builder where
   fromString =
