diff --git a/src/Network/MPD/Commands/Types.hs b/src/Network/MPD/Commands/Types.hs
index 699821a..b12e68e 100644
--- a/src/Network/MPD/Commands/Types.hs
+++ b/src/Network/MPD/Commands/Types.hs
@@ -44,6 +44,7 @@ import           Network.MPD.Commands.Arg (MPDArg(prep), Args(Args))
 import           Data.Default.Class
 
 import qualified Data.Map as M
+import           Data.Map.Strict (insertWith)
 import           Data.Time.Clock (UTCTime)
 import           Data.String
 
@@ -263,7 +264,7 @@ sgGetTag meta s = M.lookup meta $ sgTags s
 
 -- | Add metadata tag value.
 sgAddTag :: Metadata -> Value -> Song -> Song
-sgAddTag meta value s = s { sgTags = M.insertWith' (++) meta [value] (sgTags s) }
+sgAddTag meta value s = s { sgTags = insertWith (++) meta [value] (sgTags s) }
 
 defaultSong :: Path -> Song
 defaultSong path =
diff --git a/src/Network/MPD/Core.hs b/src/Network/MPD/Core.hs
index 29f7101..e4d4c1d 100644
--- a/src/Network/MPD/Core.hs
+++ b/src/Network/MPD/Core.hs
@@ -117,7 +117,8 @@ mpdOpen = MPD $ do
             (Just <$> connectTo host (PortNumber $ fromInteger port))
             `catchAny` const (return Nothing)
         checkConn = do
-            [msg] <- send ""
+            singleMsg <- send ""
+            let [msg] = singleMsg
             if "OK MPD" `isPrefixOf` msg
                 then MPD $ checkVersion $ parseVersion msg
                 else return False

