diff --git a/Codec/Encryption/OpenPGP/Types/Internal/Base.hs b/Codec/Encryption/OpenPGP/Types/Internal/Base.hs
index 861b7a5..f3dcc4a 100644
--- a/Codec/Encryption/OpenPGP/Types/Internal/Base.hs
+++ b/Codec/Encryption/OpenPGP/Types/Internal/Base.hs
@@ -1003,7 +1003,8 @@ instance Pretty SigSubPacket where
 
 instance Hashable SigSubPacket
 
-$(ATH.deriveJSON ATH.defaultOptions ''SigSubPacket)
+instance A.ToJSON SigSubPacket
+instance A.FromJSON SigSubPacket
 
 $(makeLenses ''SigSubPacket)
