From 13b358fa7899bb34e277b32a4c0d92833050f2c6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ladislav=20L=C3=A1ska?= <krakonos@krakonos.org>
Date: Tue, 14 Jan 2020 09:20:00 +0100
Subject: [PATCH 1/2] Added a condition to use the new timespec_t on gpsd APIs
 >= 9

---
 src/GPS/qgpsdevice.cpp | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/GPS/qgpsdevice.cpp b/src/GPS/qgpsdevice.cpp
index f8d1a0cf..cb727d2d 100644
--- a/src/GPS/qgpsdevice.cpp
+++ b/src/GPS/qgpsdevice.cpp
@@ -1052,8 +1052,13 @@ void QGPSDDevice::onDataAvailable()
     qreal Heading = gpsdata->fix.track;
     if (!__isnan(Heading))
         setHeading(Heading);
-    if (gpsdata->fix.time)
-        cur_datetime = QDateTime::fromTime_t(gpsdata->fix.time);
+#if GPSD_API_MAJOR_VERSION >= 9
+#define FIX_TIME fix.time.tv_sec
+#else
+#define FIX_TIME fix.time
+#endif
+    if (gpsdata->FIX_TIME)
+        cur_datetime = QDateTime::fromTime_t(gpsdata->FIX_TIME);
     emit updatePosition(gpsdata->fix.latitude,
                         gpsdata->fix.longitude,
                         cur_datetime,

From 1b76301ed7719f9ec137fd77f69b5702d8be94ab Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ladislav=20L=C3=A1ska?= <krakonos@krakonos.org>
Date: Tue, 14 Jan 2020 10:34:09 +0100
Subject: [PATCH 2/2] Added undef to FIX_TIME after end of its scope.

---
 src/GPS/qgpsdevice.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/GPS/qgpsdevice.cpp b/src/GPS/qgpsdevice.cpp
index cb727d2d..93605047 100644
--- a/src/GPS/qgpsdevice.cpp
+++ b/src/GPS/qgpsdevice.cpp
@@ -1059,6 +1059,7 @@ void QGPSDDevice::onDataAvailable()
 #endif
     if (gpsdata->FIX_TIME)
         cur_datetime = QDateTime::fromTime_t(gpsdata->FIX_TIME);
+#undef FIX_TIME
     emit updatePosition(gpsdata->fix.latitude,
                         gpsdata->fix.longitude,
                         cur_datetime,
