diff --git a/h5py/api_types_hdf5.pxd b/h5py/api_types_hdf5.pxd
index a282ecc..5bc76fa 100644
--- a/h5py/api_types_hdf5.pxd
+++ b/h5py/api_types_hdf5.pxd
@@ -297,7 +297,6 @@ cdef extern from "hdf5.h":
     H5I_DATASPACE,          # group ID for Dataspace objects
     H5I_DATASET,            # group ID for Dataset objects
     H5I_ATTR,               # group ID for Attribute objects
-    H5I_REFERENCE,          # group ID for Reference objects
     H5I_VFL,                # group ID for virtual file layer
     H5I_GENPROP_CLS,        # group ID for generic property list classes
     H5I_GENPROP_LST,        # group ID for generic property lists
diff --git a/h5py/h5i.pyx b/h5py/h5i.pyx
index 9033d50..bb5867c 100644
--- a/h5py/h5i.pyx
+++ b/h5py/h5i.pyx
@@ -23,6 +23,5 @@ DATASPACE   = H5I_DATASPACE
 DATASET     = H5I_DATASET
 ATTR        = H5I_ATTR
-REFERENCE   = H5I_REFERENCE
 GENPROP_CLS = H5I_GENPROP_CLS
 GENPROP_LST = H5I_GENPROP_LST
 DATATYPE    = H5I_DATATYPE
