diff -aur a/src/core/runtime/runtime.cpp b/src/core/runtime/runtime.cpp
--- a/src/core/runtime/runtime.cpp	2022-12-14 22:47:12.000000000 +0800
+++ b/src/core/runtime/runtime.cpp	2023-01-29 09:53:55.084690262 +0800
@@ -1592,7 +1592,7 @@
       }
       if (!ld(&hsa_api_table_.hsa_api,
         hsa_api_table_.hsa_api.version.major_id,
-        failed.size(), &failed[0])) {
+        failed.size(), failed.data())) {
           failed.push_back(lib.name_.c_str());
           os::CloseLib(tool);
           continue;
