diff -r 5abfb4de7961 CMakeLists.txt
--- a/CMakeLists.txt	Sun Jan 21 12:56:33 2018 +0100
+++ b/CMakeLists.txt	Sat Mar 23 11:56:35 2019 +0000
@@ -162,8 +162,7 @@
 ##
 #  This must come before FINDs for tiff, jpeg, png, zlib to enable
 #  finding the wxWidgets distributions of those packages (Win32 only).
-SET(wxWidgets_USE_LIBS base core aui xrc html xml adv gl net qa)
-FIND_PACKAGE(wxWidgets REQUIRED)
+FIND_PACKAGE(wxWidgets REQUIRED base core aui xrc html xml adv gl net qa)
 
 # hugin requires wxwidgets 3.0 or above
 IF (UNIX)
