diff -ru hugin-2020.0.0.orig/CMakeModules/FindOpenEXR.cmake hugin-2020.0.0/CMakeModules/FindOpenEXR.cmake
--- hugin-2020.0.0.orig/CMakeModules/FindOpenEXR.cmake	2021-04-24 13:32:14.405277312 +0200
+++ hugin-2020.0.0/CMakeModules/FindOpenEXR.cmake	2021-04-24 19:14:40.144215012 +0200
@@ -10,6 +10,12 @@
 # Redistribution and use is allowed according to the terms of the BSD license.
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
+FIND_PACKAGE(OpenEXR CONFIG REQUIRED)
+IF (TARGET OpenEXR::OpenEXR)
+    SET(OPENEXR_FOUND TRUE)
+    SET(OPENEXR_INCLUDE_DIR OpenEXR::OpenEXR)
+    SET(OPENEXR_LIBRARIES OpenEXR::OpenEXR)
+ENDIF()
 
 if (OPENEXR_INCLUDE_DIR AND OPENEXR_LIBRARIES)
   # in cache already
diff -ru hugin-2020.0.0.orig/CMakeModules/FindVIGRA.cmake hugin-2020.0.0/CMakeModules/FindVIGRA.cmake
--- hugin-2020.0.0.orig/CMakeModules/FindVIGRA.cmake	2021-04-24 13:32:14.405277312 +0200
+++ hugin-2020.0.0/CMakeModules/FindVIGRA.cmake	2021-04-24 18:54:50.461698528 +0200
@@ -74,7 +74,7 @@
         OUTPUT_STRIP_TRAILING_WHITESPACE
       )
       STRING(TOLOWER "${OUTPUT_LDD_VIGRA}" OUTPUT_LDD_VIGRA_LOWER)
-      IF(NOT "${OUTPUT_LDD_VIGRA_LOWER}" MATCHES "libilmimf")
+      IF(NOT "${OUTPUT_LDD_VIGRA_LOWER}" MATCHES "libopenexr")
         MESSAGE(FATAL_ERROR "Libvigraimpex found. But vigraimpex seems to compiled without OpenEXR support. OpenEXR support is required for Hugin.")
       ENDIF()
     ELSE()
