--- ibus-unikey-0.6.1/ukengine/data.cpp.old	2016-09-21 19:56:02.388382226 -0400
+++ ibus-unikey-0.6.1/ukengine/data.cpp	2016-09-21 20:18:07.943009680 -0400
@@ -96,149 +96,149 @@
 unsigned char SingleByteTables[][TOTAL_VNCHARS] = 
 
 // TCVN3
-{{'A','a','¸','¸','µ','µ','¶','¶','·','·','¹','¹',      // 0: a
-  '¢','©','Ê','Ê','Ç','Ç','È','È','É','É','Ë','Ë',		// 1: a^
-  '¡','¨','¾','¾','»','»','¼','¼','½','½','Æ','Æ',		// 2: a(
-  'B','b','C','c','D','d',
-  '§','®',
-  'E','e','Ğ','Ğ','Ì','Ì','Î','Î','Ï','Ï','Ñ','Ñ',		// 3: e
-  '£','ª','Õ','Õ','Ò','Ò','Ó','Ó','Ô','Ô','Ö','Ö',      // 4: e^
-  'F','f','G','g','H','h',
-  'I','i','İ','İ','×','×','Ø','Ø','Ü','Ü','Ş','Ş',		// 5: i
-  'J','j','K','k','L','l','M','m','N','n',
-  'O','o','ã','ã','ß','ß','á','á','â','â','ä','ä',		// 6: o
-  '¤','«','è','è','å','å','æ','æ','ç','ç','é','é',		// 7: o^
-  '¥','¬','í','í','ê','ê','ë','ë','ì','ì','î','î',		// 8: o+
-  'P','p','Q','q','R','r','S','s','T','t',
-  'U','u','ó','ó','ï','ï','ñ','ñ','ò','ò','ô','ô',		// 9: u
-  '¦','­','ø','ø','õ','õ','ö','ö','÷','÷','ù','ù',		//10: u+ 
-  'V','v','W','w','X','x',
-  'Y','y','ı','ı','ú','ú','û','û','ü','ü','ş','ş',		//11: y
-  'Z','z',
-  0x80, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88,
-  0x89, 0x8A, 0x8B, 0x8C, 0x8E, 0x91, 0x92, 0x93,
-  0x94, 0x95, 0x96, 0x97, 0x98, 0x99, 0x9A, 0x9B,
-  0x9C, 0x9E, 0x9F},
+{{'A',(unsigned char)'a',(unsigned char)'¸',(unsigned char)'¸',(unsigned char)'µ',(unsigned char)'µ',(unsigned char)'¶',(unsigned char)'¶',(unsigned char)'·',(unsigned char)'·',(unsigned char)'¹',(unsigned char)'¹',(unsigned char)      // 0: a
+  '¢',(unsigned char)'©',(unsigned char)'Ê',(unsigned char)'Ê',(unsigned char)'Ç',(unsigned char)'Ç',(unsigned char)'È',(unsigned char)'È',(unsigned char)'É',(unsigned char)'É',(unsigned char)'Ë',(unsigned char)'Ë',(unsigned char)		// 1: a^
+  '¡',(unsigned char)'¨',(unsigned char)'¾',(unsigned char)'¾',(unsigned char)'»',(unsigned char)'»',(unsigned char)'¼',(unsigned char)'¼',(unsigned char)'½',(unsigned char)'½',(unsigned char)'Æ',(unsigned char)'Æ',(unsigned char)		// 2: a(
+  'B',(unsigned char)'b',(unsigned char)'C',(unsigned char)'c',(unsigned char)'D',(unsigned char)'d',(unsigned char)
+  '§',(unsigned char)'®',(unsigned char)
+  'E',(unsigned char)'e',(unsigned char)'Ğ',(unsigned char)'Ğ',(unsigned char)'Ì',(unsigned char)'Ì',(unsigned char)'Î',(unsigned char)'Î',(unsigned char)'Ï',(unsigned char)'Ï',(unsigned char)'Ñ',(unsigned char)'Ñ',(unsigned char)		// 3: e
+  '£',(unsigned char)'ª',(unsigned char)'Õ',(unsigned char)'Õ',(unsigned char)'Ò',(unsigned char)'Ò',(unsigned char)'Ó',(unsigned char)'Ó',(unsigned char)'Ô',(unsigned char)'Ô',(unsigned char)'Ö',(unsigned char)'Ö',(unsigned char)      // 4: e^
+  'F',(unsigned char)'f',(unsigned char)'G',(unsigned char)'g',(unsigned char)'H',(unsigned char)'h',(unsigned char)
+  'I',(unsigned char)'i',(unsigned char)'İ',(unsigned char)'İ',(unsigned char)'×',(unsigned char)'×',(unsigned char)'Ø',(unsigned char)'Ø',(unsigned char)'Ü',(unsigned char)'Ü',(unsigned char)'Ş',(unsigned char)'Ş',(unsigned char)		// 5: i
+  'J',(unsigned char)'j',(unsigned char)'K',(unsigned char)'k',(unsigned char)'L',(unsigned char)'l',(unsigned char)'M',(unsigned char)'m',(unsigned char)'N',(unsigned char)'n',(unsigned char)
+  'O',(unsigned char)'o',(unsigned char)'ã',(unsigned char)'ã',(unsigned char)'ß',(unsigned char)'ß',(unsigned char)'á',(unsigned char)'á',(unsigned char)'â',(unsigned char)'â',(unsigned char)'ä',(unsigned char)'ä',(unsigned char)		// 6: o
+  '¤',(unsigned char)'«',(unsigned char)'è',(unsigned char)'è',(unsigned char)'å',(unsigned char)'å',(unsigned char)'æ',(unsigned char)'æ',(unsigned char)'ç',(unsigned char)'ç',(unsigned char)'é',(unsigned char)'é',(unsigned char)		// 7: o^
+  '¥',(unsigned char)'¬',(unsigned char)'í',(unsigned char)'í',(unsigned char)'ê',(unsigned char)'ê',(unsigned char)'ë',(unsigned char)'ë',(unsigned char)'ì',(unsigned char)'ì',(unsigned char)'î',(unsigned char)'î',(unsigned char)		// 8: o+
+  'P',(unsigned char)'p',(unsigned char)'Q',(unsigned char)'q',(unsigned char)'R',(unsigned char)'r',(unsigned char)'S',(unsigned char)'s',(unsigned char)'T',(unsigned char)'t',(unsigned char)
+  'U',(unsigned char)'u',(unsigned char)'ó',(unsigned char)'ó',(unsigned char)'ï',(unsigned char)'ï',(unsigned char)'ñ',(unsigned char)'ñ',(unsigned char)'ò',(unsigned char)'ò',(unsigned char)'ô',(unsigned char)'ô',(unsigned char)		// 9: u
+  '¦',(unsigned char)'­',(unsigned char)'ø',(unsigned char)'ø',(unsigned char)'õ',(unsigned char)'õ',(unsigned char)'ö',(unsigned char)'ö',(unsigned char)'÷',(unsigned char)'÷',(unsigned char)'ù',(unsigned char)'ù',(unsigned char)		//10: u+ 
+  'V',(unsigned char)'v',(unsigned char)'W',(unsigned char)'w',(unsigned char)'X',(unsigned char)'x',(unsigned char)
+  'Y',(unsigned char)'y',(unsigned char)'ı',(unsigned char)'ı',(unsigned char)'ú',(unsigned char)'ú',(unsigned char)'û',(unsigned char)'û',(unsigned char)'ü',(unsigned char)'ü',(unsigned char)'ş',(unsigned char)'ş',(unsigned char)		//11: y
+  'Z',(unsigned char)'z',(unsigned char)
+  0x80,(unsigned char) 0x82,(unsigned char) 0x83,(unsigned char) 0x84,(unsigned char) 0x85,(unsigned char) 0x86,(unsigned char) 0x87,(unsigned char) 0x88,(unsigned char)
+  0x89,(unsigned char) 0x8A,(unsigned char) 0x8B,(unsigned char) 0x8C,(unsigned char) 0x8E,(unsigned char) 0x91,(unsigned char) 0x92,(unsigned char) 0x93,(unsigned char)
+  0x94,(unsigned char) 0x95,(unsigned char) 0x96,(unsigned char) 0x97,(unsigned char) 0x98,(unsigned char) 0x99,(unsigned char) 0x9A,(unsigned char) 0x9B,(unsigned char)
+  0x9C,(unsigned char) 0x9E,(unsigned char) 0x9F},
 //VPS
-{ 'A','a','Á','á','€','à','','ä','‚','ã','å','å',
-  'Â','â','ƒ','Ã','„','À','…','Ä','Å','Å','Æ','Æ',
-  'ˆ','æ','','¡','','¢','','£','ğ','¤','¥','¥',
-  'B','b','C','c','D','d',
-  'ñ','Ç',
-  'E','e','É','é','×','è','Ş','È','ş','ë','Ë','Ë',
-  'Ê','ê','','‰','“','Š','”','‹','•','Í','Œ','Œ',
-  'F','f','G','g','H','h',
-  'I','i','´','í','µ','ì','·','Ì','¸','ï','Î','Î',
-  'J','j','K','k','L','l','M','m','N','n',
-  'O','o','¹','ó','¼','ò','½','Õ','¾','õ','†','†',
-  'Ô','ô','–','Ó','—','Ò','˜','°','™','‡','¶','¶',
-  '÷','Ö','','§','','©','Ÿ','ª','¦','«','®','®',
-  'P','p','Q','q','R','r','S','s','T','t',
-  'U','u','Ú','ú','¨','ù','Ñ','û','¬','Û','ø','ø',
-  'Ğ','Ü','­','Ù','¯','Ø','±','º','»','»','¿','¿',
-  'V','v','W','w','X','x',
-  'Y','y','İ','š','²','ÿ','ı','›','³','Ï','œ','œ',
-  'Z','z',
-  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-  0x00, 0x00, 0x00, 0x00, 0x8E, 0x91, 0x92, 0x00,
-  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-  0x00, 0x9E, 0x00},
+{ (unsigned char)'A',(unsigned char)'a',(unsigned char)'Á',(unsigned char)'á',(unsigned char)'€',(unsigned char)'à',(unsigned char)'',(unsigned char)'ä',(unsigned char)'‚',(unsigned char)'ã',(unsigned char)'å',(unsigned char)'å',(unsigned char)
+  'Â',(unsigned char)'â',(unsigned char)'ƒ',(unsigned char)'Ã',(unsigned char)'„',(unsigned char)'À',(unsigned char)'…',(unsigned char)'Ä',(unsigned char)'Å',(unsigned char)'Å',(unsigned char)'Æ',(unsigned char)'Æ',(unsigned char)
+  'ˆ',(unsigned char)'æ',(unsigned char)'',(unsigned char)'¡',(unsigned char)'',(unsigned char)'¢',(unsigned char)'',(unsigned char)'£',(unsigned char)'ğ',(unsigned char)'¤',(unsigned char)'¥',(unsigned char)'¥',(unsigned char)
+  'B',(unsigned char)'b',(unsigned char)'C',(unsigned char)'c',(unsigned char)'D',(unsigned char)'d',(unsigned char)
+  'ñ',(unsigned char)'Ç',(unsigned char)
+  'E',(unsigned char)'e',(unsigned char)'É',(unsigned char)'é',(unsigned char)'×',(unsigned char)'è',(unsigned char)'Ş',(unsigned char)'È',(unsigned char)'ş',(unsigned char)'ë',(unsigned char)'Ë',(unsigned char)'Ë',(unsigned char)
+  'Ê',(unsigned char)'ê',(unsigned char)'',(unsigned char)'‰',(unsigned char)'“',(unsigned char)'Š',(unsigned char)'”',(unsigned char)'‹',(unsigned char)'•',(unsigned char)'Í',(unsigned char)'Œ',(unsigned char)'Œ',(unsigned char)
+  'F',(unsigned char)'f',(unsigned char)'G',(unsigned char)'g',(unsigned char)'H',(unsigned char)'h',(unsigned char)
+  'I',(unsigned char)'i',(unsigned char)'´',(unsigned char)'í',(unsigned char)'µ',(unsigned char)'ì',(unsigned char)'·',(unsigned char)'Ì',(unsigned char)'¸',(unsigned char)'ï',(unsigned char)'Î',(unsigned char)'Î',(unsigned char)
+  'J',(unsigned char)'j',(unsigned char)'K',(unsigned char)'k',(unsigned char)'L',(unsigned char)'l',(unsigned char)'M',(unsigned char)'m',(unsigned char)'N',(unsigned char)'n',(unsigned char)
+  'O',(unsigned char)'o',(unsigned char)'¹',(unsigned char)'ó',(unsigned char)'¼',(unsigned char)'ò',(unsigned char)'½',(unsigned char)'Õ',(unsigned char)'¾',(unsigned char)'õ',(unsigned char)'†',(unsigned char)'†',(unsigned char)
+  'Ô',(unsigned char)'ô',(unsigned char)'–',(unsigned char)'Ó',(unsigned char)'—',(unsigned char)'Ò',(unsigned char)'˜',(unsigned char)'°',(unsigned char)'™',(unsigned char)'‡',(unsigned char)'¶',(unsigned char)'¶',(unsigned char)
+  '÷',(unsigned char)'Ö',(unsigned char)'',(unsigned char)'§',(unsigned char)'',(unsigned char)'©',(unsigned char)'Ÿ',(unsigned char)'ª',(unsigned char)'¦',(unsigned char)'«',(unsigned char)'®',(unsigned char)'®',(unsigned char)
+  'P',(unsigned char)'p',(unsigned char)'Q',(unsigned char)'q',(unsigned char)'R',(unsigned char)'r',(unsigned char)'S',(unsigned char)'s',(unsigned char)'T',(unsigned char)'t',(unsigned char)
+  'U',(unsigned char)'u',(unsigned char)'Ú',(unsigned char)'ú',(unsigned char)'¨',(unsigned char)'ù',(unsigned char)'Ñ',(unsigned char)'û',(unsigned char)'¬',(unsigned char)'Û',(unsigned char)'ø',(unsigned char)'ø',(unsigned char)
+  'Ğ',(unsigned char)'Ü',(unsigned char)'­',(unsigned char)'Ù',(unsigned char)'¯',(unsigned char)'Ø',(unsigned char)'±',(unsigned char)'º',(unsigned char)'»',(unsigned char)'»',(unsigned char)'¿',(unsigned char)'¿',(unsigned char)
+  'V',(unsigned char)'v',(unsigned char)'W',(unsigned char)'w',(unsigned char)'X',(unsigned char)'x',(unsigned char)
+  'Y',(unsigned char)'y',(unsigned char)'İ',(unsigned char)'š',(unsigned char)'²',(unsigned char)'ÿ',(unsigned char)'ı',(unsigned char)'›',(unsigned char)'³',(unsigned char)'Ï',(unsigned char)'œ',(unsigned char)'œ',(unsigned char)
+  'Z',(unsigned char)'z',(unsigned char)
+  0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x8E,(unsigned char) 0x91,(unsigned char) 0x92,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x9E,(unsigned char) 0x00},
 //VISCII
-{ 'A','a','Á','á','À','à','Ä','ä','Ã','ã','€','Õ',
-  'Â','â','„','¤','…','¥','†','¦','ç','ç','‡','§',
-  'Å','å','','¡','‚','¢','Æ','Æ','Ç','Ç','ƒ','£',
-  'B','b','C','c','D','d',
-  'Ğ','ğ',
-  'E','e','É','é','È','è','Ë','ë','ˆ','¨','‰','©',
-  'Ê','ê','Š','ª','‹','«','Œ','¬','','­','','®',
-  'F','f','G','g','H','h',
-  'I','i','Í','í','Ì','ì','›','ï','Î','î','˜','¸',
-  'J','j','K','k','L','l','M','m','N','n',
-  'O','o','Ó','ó','Ò','ò','™','ö','õ','õ','š','÷',
-  'Ô','ô','','¯','','°','‘','±','’','²','“','µ',
-  '´','½','•','¾','–','¶','—','·','³','Ş','”','ş',
-  'P','p','Q','q','R','r','S','s','T','t',
-  'U','u','Ú','ú','Ù','ù','œ','ü','','û','','ø',
-  '¿','ß','º','Ñ','»','×','¼','Ø','ÿ','æ','¹','ñ',
-  'V','v','W','w','X','x',
-  'Y','y','İ','ı','Ÿ','Ï','Ö','Ö','Û','Û','Ü','Ü',
-  'Z','z',
-  0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-  0x00, 0x00, 0x00, 0x00, 0x8E, 0x00, 0x00, 0x00,
-  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-  0x00, 0x9E, 0x00},
+{ (unsigned char)'A',(unsigned char)'a',(unsigned char)'Á',(unsigned char)'á',(unsigned char)'À',(unsigned char)'à',(unsigned char)'Ä',(unsigned char)'ä',(unsigned char)'Ã',(unsigned char)'ã',(unsigned char)'€',(unsigned char)'Õ',(unsigned char)
+  'Â',(unsigned char)'â',(unsigned char)'„',(unsigned char)'¤',(unsigned char)'…',(unsigned char)'¥',(unsigned char)'†',(unsigned char)'¦',(unsigned char)'ç',(unsigned char)'ç',(unsigned char)'‡',(unsigned char)'§',(unsigned char)
+  'Å',(unsigned char)'å',(unsigned char)'',(unsigned char)'¡',(unsigned char)'‚',(unsigned char)'¢',(unsigned char)'Æ',(unsigned char)'Æ',(unsigned char)'Ç',(unsigned char)'Ç',(unsigned char)'ƒ',(unsigned char)'£',(unsigned char)
+  'B',(unsigned char)'b',(unsigned char)'C',(unsigned char)'c',(unsigned char)'D',(unsigned char)'d',(unsigned char)
+  'Ğ',(unsigned char)'ğ',(unsigned char)
+  'E',(unsigned char)'e',(unsigned char)'É',(unsigned char)'é',(unsigned char)'È',(unsigned char)'è',(unsigned char)'Ë',(unsigned char)'ë',(unsigned char)'ˆ',(unsigned char)'¨',(unsigned char)'‰',(unsigned char)'©',(unsigned char)
+  'Ê',(unsigned char)'ê',(unsigned char)'Š',(unsigned char)'ª',(unsigned char)'‹',(unsigned char)'«',(unsigned char)'Œ',(unsigned char)'¬',(unsigned char)'',(unsigned char)'­',(unsigned char)'',(unsigned char)'®',(unsigned char)
+  'F',(unsigned char)'f',(unsigned char)'G',(unsigned char)'g',(unsigned char)'H',(unsigned char)'h',(unsigned char)
+  'I',(unsigned char)'i',(unsigned char)'Í',(unsigned char)'í',(unsigned char)'Ì',(unsigned char)'ì',(unsigned char)'›',(unsigned char)'ï',(unsigned char)'Î',(unsigned char)'î',(unsigned char)'˜',(unsigned char)'¸',(unsigned char)
+  'J',(unsigned char)'j',(unsigned char)'K',(unsigned char)'k',(unsigned char)'L',(unsigned char)'l',(unsigned char)'M',(unsigned char)'m',(unsigned char)'N',(unsigned char)'n',(unsigned char)
+  'O',(unsigned char)'o',(unsigned char)'Ó',(unsigned char)'ó',(unsigned char)'Ò',(unsigned char)'ò',(unsigned char)'™',(unsigned char)'ö',(unsigned char)'õ',(unsigned char)'õ',(unsigned char)'š',(unsigned char)'÷',(unsigned char)
+  'Ô',(unsigned char)'ô',(unsigned char)'',(unsigned char)'¯',(unsigned char)'',(unsigned char)'°',(unsigned char)'‘',(unsigned char)'±',(unsigned char)'’',(unsigned char)'²',(unsigned char)'“',(unsigned char)'µ',(unsigned char)
+  '´',(unsigned char)'½',(unsigned char)'•',(unsigned char)'¾',(unsigned char)'–',(unsigned char)'¶',(unsigned char)'—',(unsigned char)'·',(unsigned char)'³',(unsigned char)'Ş',(unsigned char)'”',(unsigned char)'ş',(unsigned char)
+  'P',(unsigned char)'p',(unsigned char)'Q',(unsigned char)'q',(unsigned char)'R',(unsigned char)'r',(unsigned char)'S',(unsigned char)'s',(unsigned char)'T',(unsigned char)'t',(unsigned char)
+  'U',(unsigned char)'u',(unsigned char)'Ú',(unsigned char)'ú',(unsigned char)'Ù',(unsigned char)'ù',(unsigned char)'œ',(unsigned char)'ü',(unsigned char)'',(unsigned char)'û',(unsigned char)'',(unsigned char)'ø',(unsigned char)
+  '¿',(unsigned char)'ß',(unsigned char)'º',(unsigned char)'Ñ',(unsigned char)'»',(unsigned char)'×',(unsigned char)'¼',(unsigned char)'Ø',(unsigned char)'ÿ',(unsigned char)'æ',(unsigned char)'¹',(unsigned char)'ñ',(unsigned char)
+  'V',(unsigned char)'v',(unsigned char)'W',(unsigned char)'w',(unsigned char)'X',(unsigned char)'x',(unsigned char)
+  'Y',(unsigned char)'y',(unsigned char)'İ',(unsigned char)'ı',(unsigned char)'Ÿ',(unsigned char)'Ï',(unsigned char)'Ö',(unsigned char)'Ö',(unsigned char)'Û',(unsigned char)'Û',(unsigned char)'Ü',(unsigned char)'Ü',(unsigned char)
+  'Z',(unsigned char)'z',(unsigned char)
+  0x80,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x8E,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x9E,(unsigned char) 0x00},
 // BKHCM1
-{'A','a','€','¾','','¿','‚','À','ƒ','Á','„','Â',
-  'Ÿ','İ','~','Ş','¡','ß','¢','à','£','á','¤','â',
-  '™','×','š','Ø','›','Ù','œ','Ú','','Û','˜','Ü',
-  'B','b','C','c','D','d',
-  '}','½',
-  'E','e','…','Ã','†','Ä','‡','Å','ˆ','Æ','‰','Ç',
-  '¥','ã','¦','ä','§','å','¨','æ','©','ç','ª','è',
-  'F','f','G','g','H','h',
-  'I','i','Š','È','‹','É','Œ','Ê','','Ë','','Ì',
-  'J','j','K','k','L','l','M','m','N','n',
-  'O','o','','Í','','Î','‘','Ï','’','Ğ','“','Ñ',
-  '«','é','¬','ê','­','ë','®','ì','¯','í','°','î',
-  '±','ï','²','ğ','³','ñ','´','ò','µ','ó','¶','ô',
-  'P','p','Q','q','R','r','S','s','T','t',
-  'U','u','”','Ò','•','Ó','–','Ô','—','Õ','˜','Ö',
-  '·','õ','¸','ö','¹','÷','º','ø','»','ù','¼','ú',
-  'V','v','W','w','X','x',
-  'Y','y','{','û','^','ü','`','ı','|','ş','','ÿ',
-  'Z','z',
-  0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-  0x00, 0x00, 0x00, 0x00, 0x8E, 0x00, 0x00, 0x00,
-  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-  0x00, 0x9E, 0x00},
+{(unsigned char)'A',(unsigned char)'a',(unsigned char)'€',(unsigned char)'¾',(unsigned char)'',(unsigned char)'¿',(unsigned char)'‚',(unsigned char)'À',(unsigned char)'ƒ',(unsigned char)'Á',(unsigned char)'„',(unsigned char)'Â',(unsigned char)
+  'Ÿ',(unsigned char)'İ',(unsigned char)'~',(unsigned char)'Ş',(unsigned char)'¡',(unsigned char)'ß',(unsigned char)'¢',(unsigned char)'à',(unsigned char)'£',(unsigned char)'á',(unsigned char)'¤',(unsigned char)'â',(unsigned char)
+  '™',(unsigned char)'×',(unsigned char)'š',(unsigned char)'Ø',(unsigned char)'›',(unsigned char)'Ù',(unsigned char)'œ',(unsigned char)'Ú',(unsigned char)'',(unsigned char)'Û',(unsigned char)'˜',(unsigned char)'Ü',(unsigned char)
+  'B',(unsigned char)'b',(unsigned char)'C',(unsigned char)'c',(unsigned char)'D',(unsigned char)'d',(unsigned char)
+  '}',(unsigned char)'½',(unsigned char)
+  'E',(unsigned char)'e',(unsigned char)'…',(unsigned char)'Ã',(unsigned char)'†',(unsigned char)'Ä',(unsigned char)'‡',(unsigned char)'Å',(unsigned char)'ˆ',(unsigned char)'Æ',(unsigned char)'‰',(unsigned char)'Ç',(unsigned char)
+  '¥',(unsigned char)'ã',(unsigned char)'¦',(unsigned char)'ä',(unsigned char)'§',(unsigned char)'å',(unsigned char)'¨',(unsigned char)'æ',(unsigned char)'©',(unsigned char)'ç',(unsigned char)'ª',(unsigned char)'è',(unsigned char)
+  'F',(unsigned char)'f',(unsigned char)'G',(unsigned char)'g',(unsigned char)'H',(unsigned char)'h',(unsigned char)
+  'I',(unsigned char)'i',(unsigned char)'Š',(unsigned char)'È',(unsigned char)'‹',(unsigned char)'É',(unsigned char)'Œ',(unsigned char)'Ê',(unsigned char)'',(unsigned char)'Ë',(unsigned char)'',(unsigned char)'Ì',(unsigned char)
+  'J',(unsigned char)'j',(unsigned char)'K',(unsigned char)'k',(unsigned char)'L',(unsigned char)'l',(unsigned char)'M',(unsigned char)'m',(unsigned char)'N',(unsigned char)'n',(unsigned char)
+  'O',(unsigned char)'o',(unsigned char)'',(unsigned char)'Í',(unsigned char)'',(unsigned char)'Î',(unsigned char)'‘',(unsigned char)'Ï',(unsigned char)'’',(unsigned char)'Ğ',(unsigned char)'“',(unsigned char)'Ñ',(unsigned char)
+  '«',(unsigned char)'é',(unsigned char)'¬',(unsigned char)'ê',(unsigned char)'­',(unsigned char)'ë',(unsigned char)'®',(unsigned char)'ì',(unsigned char)'¯',(unsigned char)'í',(unsigned char)'°',(unsigned char)'î',(unsigned char)
+  '±',(unsigned char)'ï',(unsigned char)'²',(unsigned char)'ğ',(unsigned char)'³',(unsigned char)'ñ',(unsigned char)'´',(unsigned char)'ò',(unsigned char)'µ',(unsigned char)'ó',(unsigned char)'¶',(unsigned char)'ô',(unsigned char)
+  'P',(unsigned char)'p',(unsigned char)'Q',(unsigned char)'q',(unsigned char)'R',(unsigned char)'r',(unsigned char)'S',(unsigned char)'s',(unsigned char)'T',(unsigned char)'t',(unsigned char)
+  'U',(unsigned char)'u',(unsigned char)'”',(unsigned char)'Ò',(unsigned char)'•',(unsigned char)'Ó',(unsigned char)'–',(unsigned char)'Ô',(unsigned char)'—',(unsigned char)'Õ',(unsigned char)'˜',(unsigned char)'Ö',(unsigned char)
+  '·',(unsigned char)'õ',(unsigned char)'¸',(unsigned char)'ö',(unsigned char)'¹',(unsigned char)'÷',(unsigned char)'º',(unsigned char)'ø',(unsigned char)'»',(unsigned char)'ù',(unsigned char)'¼',(unsigned char)'ú',(unsigned char)
+  'V',(unsigned char)'v',(unsigned char)'W',(unsigned char)'w',(unsigned char)'X',(unsigned char)'x',(unsigned char)
+  'Y',(unsigned char)'y',(unsigned char)'{',(unsigned char)'û',(unsigned char)'^',(unsigned char)'ü',(unsigned char)'`',(unsigned char)'ı',(unsigned char)'|',(unsigned char)'ş',(unsigned char)'',(unsigned char)'ÿ',(unsigned char)
+  'Z',(unsigned char)'z',(unsigned char)
+  0x80,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x8E,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x9E,(unsigned char) 0x00},
 //Vietware-F
-{ 'A','a','À','À','ª','ª','¶','¶','º','º','Á','Á',
-  '—','¡','Ê','Ê','Ç','Ç','È','È','É','É','Ë','Ë',
-  '–','Ÿ','Å','Å','Â','Â','Ã','Ã','Ä','Ä','Æ','Æ',
-  'B','b','C','c','D','d',
-  '˜','¢',
-  'E','e','Ï','Ï','Ì','Ì','Í','Í','Î','Î','Ñ','Ñ',
-  '™','£','Õ','Õ','Ò','Ò','Ó','Ó','Ô','Ô','Ö','Ö',
-  'F','f','G','g','H','h',
-  'I','i','Û','Û','Ø','Ø','Ù','Ù','Ú','Ú','Ü','Ü',
-  'J','j','K','k','L','l','M','m','N','n',
-  'O','o','â','â','ß','ß','à','à','á','á','ã','ã',
-  'š','¤','ç','ç','ä','ä','å','å','æ','æ','è','è',
-  '›','¥','ì','ì','é','é','ê','ê','ë','ë','í','í',
-  'P','p','Q','q','R','r','S','s','T','t',
-  'U','u','ò','ò','î','î','ï','ï','ñ','ñ','ó','ó',
-  'œ','§','÷','÷','ô','ô','õ','õ','ö','ö','ø','ø',
-  'V','v','W','w','X','x',
-  'Y','y','ü','ü','ù','ù','ú','ú','û','û','ÿ','ÿ',
-  'Z','z',
-  0x80, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88,
-  0x89, 0x8A, 0x8B, 0x8C, 0x8E, 0x91, 0x92, 0x93,
-  0x94, 0x95, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-  0x00, 0x9E, 0x00},
+{ (unsigned char)'A',(unsigned char)'a',(unsigned char)'À',(unsigned char)'À',(unsigned char)'ª',(unsigned char)'ª',(unsigned char)'¶',(unsigned char)'¶',(unsigned char)'º',(unsigned char)'º',(unsigned char)'Á',(unsigned char)'Á',(unsigned char)
+  '—',(unsigned char)'¡',(unsigned char)'Ê',(unsigned char)'Ê',(unsigned char)'Ç',(unsigned char)'Ç',(unsigned char)'È',(unsigned char)'È',(unsigned char)'É',(unsigned char)'É',(unsigned char)'Ë',(unsigned char)'Ë',(unsigned char)
+  '–',(unsigned char)'Ÿ',(unsigned char)'Å',(unsigned char)'Å',(unsigned char)'Â',(unsigned char)'Â',(unsigned char)'Ã',(unsigned char)'Ã',(unsigned char)'Ä',(unsigned char)'Ä',(unsigned char)'Æ',(unsigned char)'Æ',(unsigned char)
+  'B',(unsigned char)'b',(unsigned char)'C',(unsigned char)'c',(unsigned char)'D',(unsigned char)'d',(unsigned char)
+  '˜',(unsigned char)'¢',(unsigned char)
+  'E',(unsigned char)'e',(unsigned char)'Ï',(unsigned char)'Ï',(unsigned char)'Ì',(unsigned char)'Ì',(unsigned char)'Í',(unsigned char)'Í',(unsigned char)'Î',(unsigned char)'Î',(unsigned char)'Ñ',(unsigned char)'Ñ',(unsigned char)
+  '™',(unsigned char)'£',(unsigned char)'Õ',(unsigned char)'Õ',(unsigned char)'Ò',(unsigned char)'Ò',(unsigned char)'Ó',(unsigned char)'Ó',(unsigned char)'Ô',(unsigned char)'Ô',(unsigned char)'Ö',(unsigned char)'Ö',(unsigned char)
+  'F',(unsigned char)'f',(unsigned char)'G',(unsigned char)'g',(unsigned char)'H',(unsigned char)'h',(unsigned char)
+  'I',(unsigned char)'i',(unsigned char)'Û',(unsigned char)'Û',(unsigned char)'Ø',(unsigned char)'Ø',(unsigned char)'Ù',(unsigned char)'Ù',(unsigned char)'Ú',(unsigned char)'Ú',(unsigned char)'Ü',(unsigned char)'Ü',(unsigned char)
+  'J',(unsigned char)'j',(unsigned char)'K',(unsigned char)'k',(unsigned char)'L',(unsigned char)'l',(unsigned char)'M',(unsigned char)'m',(unsigned char)'N',(unsigned char)'n',(unsigned char)
+  'O',(unsigned char)'o',(unsigned char)'â',(unsigned char)'â',(unsigned char)'ß',(unsigned char)'ß',(unsigned char)'à',(unsigned char)'à',(unsigned char)'á',(unsigned char)'á',(unsigned char)'ã',(unsigned char)'ã',(unsigned char)
+  'š',(unsigned char)'¤',(unsigned char)'ç',(unsigned char)'ç',(unsigned char)'ä',(unsigned char)'ä',(unsigned char)'å',(unsigned char)'å',(unsigned char)'æ',(unsigned char)'æ',(unsigned char)'è',(unsigned char)'è',(unsigned char)
+  '›',(unsigned char)'¥',(unsigned char)'ì',(unsigned char)'ì',(unsigned char)'é',(unsigned char)'é',(unsigned char)'ê',(unsigned char)'ê',(unsigned char)'ë',(unsigned char)'ë',(unsigned char)'í',(unsigned char)'í',(unsigned char)
+  'P',(unsigned char)'p',(unsigned char)'Q',(unsigned char)'q',(unsigned char)'R',(unsigned char)'r',(unsigned char)'S',(unsigned char)'s',(unsigned char)'T',(unsigned char)'t',(unsigned char)
+  'U',(unsigned char)'u',(unsigned char)'ò',(unsigned char)'ò',(unsigned char)'î',(unsigned char)'î',(unsigned char)'ï',(unsigned char)'ï',(unsigned char)'ñ',(unsigned char)'ñ',(unsigned char)'ó',(unsigned char)'ó',(unsigned char)
+  'œ',(unsigned char)'§',(unsigned char)'÷',(unsigned char)'÷',(unsigned char)'ô',(unsigned char)'ô',(unsigned char)'õ',(unsigned char)'õ',(unsigned char)'ö',(unsigned char)'ö',(unsigned char)'ø',(unsigned char)'ø',(unsigned char)
+  'V',(unsigned char)'v',(unsigned char)'W',(unsigned char)'w',(unsigned char)'X',(unsigned char)'x',(unsigned char)
+  'Y',(unsigned char)'y',(unsigned char)'ü',(unsigned char)'ü',(unsigned char)'ù',(unsigned char)'ù',(unsigned char)'ú',(unsigned char)'ú',(unsigned char)'û',(unsigned char)'û',(unsigned char)'ÿ',(unsigned char)'ÿ',(unsigned char)
+  'Z',(unsigned char)'z',(unsigned char)
+  0x80,(unsigned char) 0x82,(unsigned char) 0x83,(unsigned char) 0x84,(unsigned char) 0x85,(unsigned char) 0x86,(unsigned char) 0x87,(unsigned char) 0x88,(unsigned char)
+  0x89,(unsigned char) 0x8A,(unsigned char) 0x8B,(unsigned char) 0x8C,(unsigned char) 0x8E,(unsigned char) 0x91,(unsigned char) 0x92,(unsigned char) 0x93,(unsigned char)
+  0x94,(unsigned char) 0x95,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x9E,(unsigned char) 0x00},
 // ISC
- {'A','a','ƒ','¸','€','µ','','¶','‚','·','„','¹', 
-  '¢','©','Ä',' ','Á','Ç','Â','È','Ã','É','†','Ë',
-  '¡','¨','À','¾','¯','»','º','¼','¿','½','…','Æ',
-  'B','b','C','c','D','d',
-  '§','®',
-  'E','e','Ğ','Š','‡','Ì','ˆ','Î','‰','Ï','Ñ','‹',
-  '£','ª','Ú','Õ','Å','Ò','Í','Ó','Ù','Ô','Œ','Ö',
-  'F','f','G','g','H','h',
-  'I','i','','İ','','×','','Ø','','Ü','‘','Ş',
-  'J','j','K','k','L','l','M','m','N','n',
-  'O','o','•','ã','’','ß','“','á','”','â','–','ä',
-  '¤','«','ÿ','è','Û','å','à','æ','ğ','ç','—','é',
-  '¥','¬','›','í','˜','ê','™','ë','š','ì','œ','î',
-  'P','p','Q','q','R','r','S','s','T','t',
-  'U','u','@','ó','','ï','','ñ','Ÿ','ò','|','ô',
-  '¦','­','`','ø',0x5C,'õ','^','ö','~','÷','#','ù',
-  'V','v','W','w','X','x',
-  'Y','y','³','ı','°','ú','±','û','²','ü','´','ş',
-  'Z','z',
-  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-  0x00, 0x00, 0x00, 0x00, 0x00, 0x11, 0x12, 0x13,
-  0x14, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-  0x00, 0x00, 0x00}
+ {(unsigned char)'A',(unsigned char)'a',(unsigned char)'ƒ',(unsigned char)'¸',(unsigned char)'€',(unsigned char)'µ',(unsigned char)'',(unsigned char)'¶',(unsigned char)'‚',(unsigned char)'·',(unsigned char)'„',(unsigned char)'¹',(unsigned char) 
+  '¢',(unsigned char)'©',(unsigned char)'Ä',(unsigned char)' ',(unsigned char)'Á',(unsigned char)'Ç',(unsigned char)'Â',(unsigned char)'È',(unsigned char)'Ã',(unsigned char)'É',(unsigned char)'†',(unsigned char)'Ë',(unsigned char)
+  '¡',(unsigned char)'¨',(unsigned char)'À',(unsigned char)'¾',(unsigned char)'¯',(unsigned char)'»',(unsigned char)'º',(unsigned char)'¼',(unsigned char)'¿',(unsigned char)'½',(unsigned char)'…',(unsigned char)'Æ',(unsigned char)
+  'B',(unsigned char)'b',(unsigned char)'C',(unsigned char)'c',(unsigned char)'D',(unsigned char)'d',(unsigned char)
+  '§',(unsigned char)'®',(unsigned char)
+  'E',(unsigned char)'e',(unsigned char)'Ğ',(unsigned char)'Š',(unsigned char)'‡',(unsigned char)'Ì',(unsigned char)'ˆ',(unsigned char)'Î',(unsigned char)'‰',(unsigned char)'Ï',(unsigned char)'Ñ',(unsigned char)'‹',(unsigned char)
+  '£',(unsigned char)'ª',(unsigned char)'Ú',(unsigned char)'Õ',(unsigned char)'Å',(unsigned char)'Ò',(unsigned char)'Í',(unsigned char)'Ó',(unsigned char)'Ù',(unsigned char)'Ô',(unsigned char)'Œ',(unsigned char)'Ö',(unsigned char)
+  'F',(unsigned char)'f',(unsigned char)'G',(unsigned char)'g',(unsigned char)'H',(unsigned char)'h',(unsigned char)
+  'I',(unsigned char)'i',(unsigned char)'',(unsigned char)'İ',(unsigned char)'',(unsigned char)'×',(unsigned char)'',(unsigned char)'Ø',(unsigned char)'',(unsigned char)'Ü',(unsigned char)'‘',(unsigned char)'Ş',(unsigned char)
+  'J',(unsigned char)'j',(unsigned char)'K',(unsigned char)'k',(unsigned char)'L',(unsigned char)'l',(unsigned char)'M',(unsigned char)'m',(unsigned char)'N',(unsigned char)'n',(unsigned char)
+  'O',(unsigned char)'o',(unsigned char)'•',(unsigned char)'ã',(unsigned char)'’',(unsigned char)'ß',(unsigned char)'“',(unsigned char)'á',(unsigned char)'”',(unsigned char)'â',(unsigned char)'–',(unsigned char)'ä',(unsigned char)
+  '¤',(unsigned char)'«',(unsigned char)'ÿ',(unsigned char)'è',(unsigned char)'Û',(unsigned char)'å',(unsigned char)'à',(unsigned char)'æ',(unsigned char)'ğ',(unsigned char)'ç',(unsigned char)'—',(unsigned char)'é',(unsigned char)
+  '¥',(unsigned char)'¬',(unsigned char)'›',(unsigned char)'í',(unsigned char)'˜',(unsigned char)'ê',(unsigned char)'™',(unsigned char)'ë',(unsigned char)'š',(unsigned char)'ì',(unsigned char)'œ',(unsigned char)'î',(unsigned char)
+  'P',(unsigned char)'p',(unsigned char)'Q',(unsigned char)'q',(unsigned char)'R',(unsigned char)'r',(unsigned char)'S',(unsigned char)'s',(unsigned char)'T',(unsigned char)'t',(unsigned char)
+  'U',(unsigned char)'u',(unsigned char)'@',(unsigned char)'ó',(unsigned char)'',(unsigned char)'ï',(unsigned char)'',(unsigned char)'ñ',(unsigned char)'Ÿ',(unsigned char)'ò',(unsigned char)'|',(unsigned char)'ô',(unsigned char)
+  '¦',(unsigned char)'­',(unsigned char)'`',(unsigned char)'ø',(unsigned char)0x5C,(unsigned char)'õ',(unsigned char)'^',(unsigned char)'ö',(unsigned char)'~',(unsigned char)'÷',(unsigned char)'#',(unsigned char)'ù',(unsigned char)
+  'V',(unsigned char)'v',(unsigned char)'W',(unsigned char)'w',(unsigned char)'X',(unsigned char)'x',(unsigned char)
+  'Y',(unsigned char)'y',(unsigned char)'³',(unsigned char)'ı',(unsigned char)'°',(unsigned char)'ú',(unsigned char)'±',(unsigned char)'û',(unsigned char)'²',(unsigned char)'ü',(unsigned char)'´',(unsigned char)'ş',(unsigned char)
+  'Z',(unsigned char)'z',(unsigned char)
+  0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x11,(unsigned char) 0x12,(unsigned char) 0x13,(unsigned char)
+  0x14,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char) 0x00,(unsigned char)
+  0x00,(unsigned char) 0x00, (unsigned char) 0x00}
 };
 
 UKWORD DoubleByteTables[][TOTAL_VNCHARS] = {
