diff -ruN a/support/mixer-qt4/ffado/configuration.py b/support/mixer-qt4/ffado/configuration.py
--- a/support/mixer-qt4/ffado/configuration.py	2017-03-28 13:38:34.000000000 +0200
+++ b/support/mixer-qt4/ffado/configuration.py	2019-11-24 20:55:05.284341627 +0100
@@ -20,7 +20,8 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-import re, os
+import os
+import dbus
 
 import shlex
 
@@ -53,6 +54,10 @@
                 self.addDevice( dev )
 
     def getDeviceById( self, vendor, model ):
+        if isinstance(vendor, dbus.Int32):
+            vendor = str(int(vendor))
+        if isinstance(model, dbus.Int32):
+            model = str(int(model))
         log.debug("DeviceList::getDeviceById( %s, %s )" % (vendor, model ))
         for dev in self.devices:
             if int("%s" % dev['vendorid'], 0) == int("%s" % vendor, 0) and \
