Index: libgomp/oacc-int.h
===================================================================
--- libgomp/oacc-int.h	(revision 228010)
+++ libgomp/oacc-int.h	(revision 228011)
@@ -97,6 +97,7 @@
 void goacc_save_and_set_bind (acc_device_t);
 void goacc_restore_bind (void);
 void goacc_lazy_initialize (void);
+void goacc_host_init (void);
 
 #ifdef HAVE_ATTRIBUTE_VISIBILITY
 # pragma GCC visibility pop
Index: libgomp/oacc-init.c
===================================================================
--- libgomp/oacc-init.c	(revision 228010)
+++ libgomp/oacc-init.c	(revision 228011)
@@ -580,6 +580,9 @@
 
   goacc_threads = NULL;
   gomp_mutex_init (&goacc_thread_lock);
+
+  /* Initialize and register the 'host' device type.  */
+  goacc_host_init ();
 }
 
 /* Compiler helper functions */
Index: libgomp/oacc-host.c
===================================================================
--- libgomp/oacc-host.c	(revision 228010)
+++ libgomp/oacc-host.c	(revision 228011)
@@ -82,8 +82,8 @@
   };
 
 /* Register this device type.  */
-static __attribute__ ((constructor))
-void goacc_host_init (void)
+void
+goacc_host_init (void)
 {
   gomp_mutex_init (&host_dispatch.lock);
   goacc_register (&host_dispatch);
