--- linbox-1.5.2/linbox/matrix/densematrix/blas-transposed-matrix.h.orig	2018-05-06 10:35:33.852181382 +0000
+++ linbox-1.5.2/linbox/matrix/densematrix/blas-transposed-matrix.h	2018-05-06 10:35:59.598847558 +0000
@@ -67,9 +67,6 @@
 	/*! TransposedBlasMatrix.
 	 * NO DOC
 	 */
-#if !defined(__INTEL_COMPILER) && !defined(__CUDACC__) & !defined(__clang__)
-	template <>
-#endif
 	template< class Matrix >
 	class TransposedBlasMatrix< TransposedBlasMatrix< Matrix > > : public Matrix {
 
