diff -u -r ndiswrapper-1.61/ndiswrapper/driver/ntoskernel.c ndiswrapper-1.61-5.0/ndiswrapper/driver/ntoskernel.c
--- ndiswrapper-1.61/ndiswrapper/driver/ntoskernel.c	2019-03-04 20:22:58.095332763 +0000
+++ ndiswrapper-1.61-5.0/ndiswrapper/driver/ntoskernel.c	2019-03-04 20:24:48.698777448 +0000
@@ -2498,7 +2498,7 @@
 
 int ntoskernel_init(void)
 {
-	struct timeval now;
+	struct timespec64 now;
 
 	spin_lock_init(&dispatcher_lock);
 	spin_lock_init(&ntoskernel_lock);
@@ -2518,10 +2518,10 @@
 	INIT_WORK(&ntos_work, ntos_work_worker);
 	wrap_timer_slist.next = NULL;
 
-	do_gettimeofday(&now);
+	ktime_get_real_ts64(&now);
 	wrap_ticks_to_boot = TICKS_1601_TO_1970;
 	wrap_ticks_to_boot += (u64)now.tv_sec * TICKSPERSEC;
-	wrap_ticks_to_boot += now.tv_usec * 10;
+	wrap_ticks_to_boot += now.tv_nsec / 100;
 	wrap_ticks_to_boot -= jiffies * TICKSPERJIFFY;
 	TRACE2("%llu", wrap_ticks_to_boot);
 
