diff --git a/meson.build b/meson.build
index 1540b19..b959ca5 100644
--- a/meson.build
+++ b/meson.build
@@ -306,7 +306,7 @@ futatabi_shader_srcs = bin2h_gen.process(shaders)
 futatabi_srcs += futatabi_shader_srcs
 
 executable('futatabi', futatabi_srcs,
-	dependencies: [shareddep, qt5deps, libjpegdep, movitdep, libmicrohttpddep, protobufdep, sqlite3dep, vax11dep, vadrmdep, x11dep, libavformatdep, libavcodecdep, libavutildep, libswscaledep],
+	dependencies: [shareddep, qt5deps, libjpegdep, movitdep, libmicrohttpddep, protobufdep, sqlite3dep, vax11dep, vadrmdep, x11dep, threaddep, libavformatdep, libavcodecdep, libavutildep, libswscaledep],
 	link_with: shared,
 	include_directories: [include_directories('futatabi')],
 	install: true)
