--- ./src/util/math.h.orig	2016-07-14 04:13:22.861482580 +0100
+++ ./src/util/math.h	2016-07-14 04:15:36.966067224 +0100
@@ -4,7 +4,13 @@
 // Causes MSVC to define M_PI and friends.
 // http://msdn.microsoft.com/en-us/library/4hwaceh6.aspx
 #define _USE_MATH_DEFINES
-#include <cmath>
+#include <math.h>
+#include <cmath> 
+// Note: Because of our fpclassify hack, we actualy need to inlude both, 
+// the c and the c++ version of the math header.  
+// From GCC 6.1.1 math.h depends on cmath, which failes to compile if included 
+// after our fpclassify hack 
+
 #include <algorithm>
 
 #include "util/assert.h"
