diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2022-10-13 20:54:07.000000000 +0200
+++ b/CMakeLists.txt	2022-10-29 16:40:01.350463440 +0200
@@ -82,14 +82,6 @@
 endif()
 message("Document root: ${DOC_ROOT}")
 
-#set debug define for source
-if(CMAKE_BUILD_TYPE MATCHES RELEASE)
-  set(DEBUG "OFF")
-  set(NDEBUG "true")
-else()
-  set(DEBUG "ON")
-endif()
-
 #required dependencies
 find_package(Threads REQUIRED)
 find_package(PCRE2 REQUIRED)
@@ -268,8 +260,7 @@
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -fPIE")
   set (CMAKE_EXE_LINKER_FLAGS "-Wl,-z,relro,-z,now,--gc-sections -pie -s")
 else()
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -ggdb -Og")
-  set (CMAKE_EXE_LINKER_FLAGS "")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIE")
 endif()
 
 #set source files
