diff --git a/graf3d/eve7/inc/ROOT/REveElement.hxx b/graf3d/eve7/inc/ROOT/REveElement.hxx
index 2a127888a2..9deb3af147 100644
--- a/graf3d/eve7/inc/ROOT/REveElement.hxx
+++ b/graf3d/eve7/inc/ROOT/REveElement.hxx
@@ -16,26 +16,13 @@
 #include <ROOT/REveVector.hxx>
 #include <ROOT/REveProjectionBases.hxx>
 
+#include <nlohmann/json_fwd.hpp>
+
 #include <map>
 #include <memory>
 
 class TGeoMatrix;
 
-namespace nlohmann {
-template<typename T, typename SFINAE>
-struct adl_serializer;
-
-template <template <typename U, typename V, typename... Args> class ObjectType,
-          template <typename U, typename... Args> class ArrayType, class StringType, class BooleanType,
-          class NumberIntegerType, class NumberUnsignedType, class NumberFloatType,
-          template <typename U> class AllocatorType, template <typename T, typename SFINAE = void> class JSONSerializer,
-          class BinaryType>
-class basic_json;
-
-using json = basic_json<std::map, std::vector, std::string, bool, std::int64_t, std::uint64_t, double, std::allocator,
-                        adl_serializer, std::vector<std::uint8_t>>;
-} // namespace nlohmann
-
 namespace ROOT {
 namespace Experimental {
