diff -wbBur libvirt-0.9.7/src/qemu/qemu_migration.c libvirt-0.9.7.my/src/qemu/qemu_migration.c
--- libvirt-0.9.7/src/qemu/qemu_migration.c	2011-10-31 07:46:04.000000000 +0400
+++ libvirt-0.9.7.my/src/qemu/qemu_migration.c	2011-11-08 15:46:10.000000000 +0400
@@ -1589,7 +1589,7 @@
                                            spec->dest.unix_socket.file);
         } else {
             const char *args[] = {
-                "nc", "-U", spec->dest.unix_socket.file, NULL
+                "nc.openbsd", "-U", spec->dest.unix_socket.file, NULL
             };
             ret = qemuMonitorMigrateToCommand(priv->mon, migrate_flags, args);
         }
diff -wbBur libvirt-0.9.7/src/remote/remote_driver.c libvirt-0.9.7.my/src/remote/remote_driver.c
--- libvirt-0.9.7/src/remote/remote_driver.c	2011-11-08 07:53:30.000000000 +0400
+++ libvirt-0.9.7.my/src/remote/remote_driver.c	2011-11-08 15:46:10.000000000 +0400
@@ -614,7 +614,7 @@
                                                 username,
                                                 !tty,
                                                 !verify,
-                                                netcat ? netcat : "nc",
+                                                netcat ? netcat : "nc.openbsd",
                                                 keyfile,
                                                 sockname)))
             goto failed;
diff -wbBur libvirt-0.9.7/src/rpc/virnetsocket.c libvirt-0.9.7.my/src/rpc/virnetsocket.c
--- libvirt-0.9.7/src/rpc/virnetsocket.c	2011-11-08 07:53:30.000000000 +0400
+++ libvirt-0.9.7.my/src/rpc/virnetsocket.c	2011-11-08 15:46:31.000000000 +0400
@@ -642,7 +642,7 @@
         virCommandAddArgList(cmd, "-o", "StrictHostKeyChecking=no", NULL);
 
     if (!netcat)
-        netcat = "nc";
+        netcat = "nc.openbsd";
 
     virCommandAddArgList(cmd, nodename, "sh", "-c", NULL);
 
