diff -upr opensips-3.1.2.orig/modules/ldap/ldap_api_fn.c opensips-3.1.2/modules/ldap/ldap_api_fn.c
--- opensips-3.1.2.orig/modules/ldap/ldap_api_fn.c	2021-05-06 17:40:16.000000000 +0300
+++ opensips-3.1.2/modules/ldap/ldap_api_fn.c	2021-11-13 22:51:36.709798951 +0200
@@ -118,7 +118,7 @@ struct ld_conn* get_ldap_connection(stru
 		return NULL;
 	}
 
-	if (ldap_connect(lds->name, NULL) < 0) {
+	if (opensips_ldap_connect(lds->name, NULL) < 0) {
 			LM_ERR("failed to create new ldap connection!\n");
 			return NULL;
 	}
diff -upr opensips-3.1.2.orig/modules/ldap/ldap_connect.c opensips-3.1.2/modules/ldap/ldap_connect.c
--- opensips-3.1.2.orig/modules/ldap/ldap_connect.c	2021-05-06 17:40:16.000000000 +0300
+++ opensips-3.1.2/modules/ldap/ldap_connect.c	2021-11-13 22:51:36.709798951 +0200
@@ -113,7 +113,7 @@ error:
 }
 
 
-int ldap_connect(char* _ld_name, struct ld_conn* conn)
+int opensips_ldap_connect(char* _ld_name, struct ld_conn* conn)
 {
 	int rc;
 	int ldap_proto_version;
@@ -411,7 +411,7 @@ int ldap_reconnect(char* _ld_name, struc
 		return -1;
 	}
 
-	if ((rc = ldap_connect(_ld_name, conn)) != 0)
+	if ((rc = opensips_ldap_connect(_ld_name, conn)) != 0)
 	{
 		LM_ERR("[%s]: reconnect failed\n",
 				_ld_name);
diff -upr opensips-3.1.2.orig/modules/ldap/ldap_connect.h opensips-3.1.2/modules/ldap/ldap_connect.h
--- opensips-3.1.2.orig/modules/ldap/ldap_connect.h	2021-05-06 17:40:16.000000000 +0300
+++ opensips-3.1.2/modules/ldap/ldap_connect.h	2021-11-13 22:51:38.336506198 +0200
@@ -43,7 +43,7 @@
 /* forward declaration for this structure */
 struct ld_conn;
 
-extern int ldap_connect(char* _ld_name, struct ld_conn* conn);
+extern int opensips_ldap_connect(char* _ld_name, struct ld_conn* conn);
 extern int ldap_disconnect(char* _ld_name, struct ld_conn* conn);
 extern int ldap_reconnect(char* _ld_name, struct ld_conn* conn);
 extern int ldap_get_vendor_version(char** _version);
diff -upr opensips-3.1.2.orig/modules/ldap/ldap_mod.c opensips-3.1.2/modules/ldap/ldap_mod.c
--- opensips-3.1.2.orig/modules/ldap/ldap_mod.c	2021-05-06 17:40:16.000000000 +0300
+++ opensips-3.1.2/modules/ldap/ldap_mod.c	2021-11-13 22:51:36.709798951 +0200
@@ -186,7 +186,7 @@ static int child_init(int rank)
 		}
 
 		/* won't check for null in get_ld_session since it's barely been initialized */
-		if (ldap_connect(ld_name, &get_ld_session(ld_name)->conn_s) != 0)
+		if (opensips_ldap_connect(ld_name, &get_ld_session(ld_name)->conn_s) != 0)
 		{
 			LM_ERR("[%s]: failed to connect to LDAP host(s)\n", ld_name);
 			ldap_disconnect(ld_name, NULL);
