diff -wbBur OpenSceneGraph-3.0.1/src/osgPlugins/xine/video_out_rgb.c OpenSceneGraph-3.0.1.my/src/osgPlugins/xine/video_out_rgb.c
--- OpenSceneGraph-3.0.1/src/osgPlugins/xine/video_out_rgb.c	2009-11-20 14:46:20.000000000 +0300
+++ OpenSceneGraph-3.0.1.my/src/osgPlugins/xine/video_out_rgb.c	2012-01-05 02:15:38.000000000 +0400
@@ -2769,8 +2769,8 @@
 	clear(rgb_class, sizeof(rgbout_class_t));
 
 	rgb_class->driver_class.open_plugin     = open_plugin;
-	rgb_class->driver_class.get_identifier  = get_identifier;
-	rgb_class->driver_class.get_description = get_description;
+	rgb_class->driver_class.identifier  = get_identifier(NULL);
+	rgb_class->driver_class.description = get_description(NULL);
 	rgb_class->driver_class.dispose         = dispose_class;
 
 	return(rgb_class);
