commit d86effe073d75a73e544a58b16371664b8fcaf73
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Tue Dec 4 16:21:48 2018 +0800

    Migrade to patience 0.2 (#1)

diff --git a/chell/Test/Chell.hs b/chell/Test/Chell.hs
index 5a28259..d02e24a 100644
--- a/chell/Test/Chell.hs
+++ b/chell/Test/Chell.hs
@@ -123,7 +123,6 @@ import qualified Control.Exception
 import           Control.Exception (Exception)
 import           Control.Monad (ap, liftM)
 import           Control.Monad.IO.Class (MonadIO, liftIO)
-import qualified Data.Algorithm.Patience as Patience
 import qualified Data.ByteString.Char8
 import qualified Data.ByteString.Lazy.Char8
 import           Data.Foldable (Foldable, foldMap)
@@ -136,6 +135,8 @@ import qualified Data.Text.Lazy
 
 import qualified Language.Haskell.TH as TH
 
+import qualified Patience
+
 import           Test.Chell.Main (defaultMain)
 import           Test.Chell.Types
 
diff --git a/chell/chell.cabal b/chell/chell.cabal
index b2df441..126975c 100644
--- a/chell/chell.cabal
+++ b/chell/chell.cabal
@@ -69,7 +69,7 @@ library
       base >= 4.1 && < 5.0
     , bytestring >= 0.9
     , options >= 1.0 && < 2.0
-    , patience >= 0.1 && < 0.2
+    , patience >= 0.2 && < 0.3
     , random >= 1.0
     , template-haskell >= 2.3
     , text
