diff --git a/game/engine.cc b/game/engine.cc
index 9f0f7ef..560ce71 100644
--- a/game/engine.cc
+++ b/game/engine.cc
@@ -8,6 +8,8 @@
 #include <iostream>
 #include <list>
 
+using namespace boost::placeholders;
+
 const double Engine::TIMESTEP = 0.01;
 
 Engine::Engine(Audio& audio, VocalTrackPtrs vocals, Database& database):
diff --git a/game/surface.cc b/game/surface.cc
index 75356fe..34951f9 100644
--- a/game/surface.cc
+++ b/game/surface.cc
@@ -16,6 +16,8 @@
 
 using std::uint32_t;
 
+using namespace boost::placeholders;
+
 Shader& getShader(std::string const& name) {
 	return Game::getSingletonPtr()->window().shader(name);  // FIXME
 }