diff --git a/src/calibre/utils/podofo/doc.cpp b/src/calibre/utils/podofo/doc.cpp
index b33647336c..3d25ac993f 100644
--- a/src/calibre/utils/podofo/doc.cpp
+++ b/src/calibre/utils/podofo/doc.cpp
@@ -39,7 +39,7 @@ PDFDoc_load(PDFDoc *self, PyObject *args) {
 
     if (PyArg_ParseTuple(args, "s#", &buffer, &size)) {
         try {
-            self->doc->Load(buffer, (long)size);
+            self->doc->LoadFromBuffer(buffer, (long)size);
         } catch(const PdfError & err) {
             podofo_set_exception(err);
             return NULL;
@@ -56,7 +56,7 @@ PDFDoc_open(PDFDoc *self, PyObject *args) {
 
     if (PyArg_ParseTuple(args, "s", &fname)) {
         try {
-            self->doc->Load(fname);
+            self->doc->LoadFromBuffer(fname, (long)strlen(fname));
         } catch(const PdfError & err) {
             podofo_set_exception(err);
             return NULL;
