diff -rupN projectM-complete-2.1.0-Source.old/src/projectM-test/CMakeLists.txt projectM-complete-2.1.0-Source/src/projectM-test/CMakeLists.txt
--- projectM-complete-2.1.0-Source.old/src/projectM-test/CMakeLists.txt	2012-05-20 00:43:26.000000000 -0300
+++ projectM-complete-2.1.0-Source/src/projectM-test/CMakeLists.txt	2012-06-30 01:01:55.947058502 -0300
@@ -35,11 +35,11 @@ elseif (${CMAKE_PROJECT_NAME} MATCHES ${
    set(PROJECTM_LINK ${LIBPROJECTM_LDFLAGS})
 endif (${CMAKE_PROJECT_NAME} MATCHES "projectM-complete")
 
-INCLUDE_DIRECTORIES(${PROJECTM_INCLUDE} ${SDL_INCLUDE_DIR} ${GTK_INCLUDE_DIR})
+INCLUDE_DIRECTORIES(${PROJECTM_INCLUDE} ${SDL_INCLUDE_DIR} ${GTK_INCLUDE_DIR} ${OPENGL_INCLUDE_DIR})
 LINK_DIRECTORIES(${PROJECTM_LINK})
 
-TARGET_LINK_LIBRARIES(projectM-test projectM  ${SDL_LIBRARY})
-TARGET_LINK_LIBRARIES(projectM-test-memleak projectM  ${SDL_LIBRARY} )
-TARGET_LINK_LIBRARIES(projectM-test-texture projectM  ${SDL_LIBRARY} )
+TARGET_LINK_LIBRARIES(projectM-test projectM  ${SDL_LIBRARY} ${OPENGL_gl_LIBRARY} )
+TARGET_LINK_LIBRARIES(projectM-test-memleak projectM  ${SDL_LIBRARY} ${OPENGL_gl_LIBRARY} )
+TARGET_LINK_LIBRARIES(projectM-test-texture projectM  ${SDL_LIBRARY} ${OPENGL_gl_LIBRARY} )
 
 INSTALL(TARGETS projectM-test projectM-test-texture DESTINATION ${CMAKE_INSTALL_PREFIX}/bin )
