diff --git a/weakmethod.py.orig b/weakmethod.py
index 4638abf..54bdc1b 100644
--- a/weakmethod.py.orig
+++ b/weakmethod.py
@@ -53,7 +53,10 @@ if sys.version > '3':
             '''Returns True if the referenced callable was a bound method and
             the instance no longer exists. Otherwise, return False.
             '''
-            return self.proxy is not None and not bool(dir(self.proxy))
+            try:
+                return self.proxy is not None and not bool(dir(self.proxy))
+            except ReferenceError:
+                return True
 
         def __eq__(self, other):
             try:
