diff -ruN qmapshack-V_1.15.0-orig/src/qmapshack/gis/wpt/CGisItemWpt.cpp qmapshack-V_1.15.0/src/qmapshack/gis/wpt/CGisItemWpt.cpp
--- qmapshack-V_1.15.0-orig/src/qmapshack/gis/wpt/CGisItemWpt.cpp	2020-06-14 12:22:49.000000000 +0200
+++ qmapshack-V_1.15.0/src/qmapshack/gis/wpt/CGisItemWpt.cpp	2020-07-10 17:05:41.238391674 +0200
@@ -38,6 +38,7 @@
 
 #include <QtWidgets>
 #include <QtXml>
+#include <QPainterPath>
 
 IGisItem::key_t CGisItemWpt::keyUserFocus;
 QMap<searchProperty_e, CGisItemWpt::fSearch> CGisItemWpt::keywordLambdaMap;
diff -ruN qmapshack-V_1.15.0-orig/src/qmapshack/helpers/CDraw.cpp qmapshack-V_1.15.0/src/qmapshack/helpers/CDraw.cpp
--- qmapshack-V_1.15.0-orig/src/qmapshack/helpers/CDraw.cpp	2020-06-14 12:22:49.000000000 +0200
+++ qmapshack-V_1.15.0/src/qmapshack/helpers/CDraw.cpp	2020-07-10 17:06:06.955057192 +0200
@@ -24,6 +24,7 @@
 #include <QImage>
 #include <QPointF>
 #include <QtMath>
+#include <QPainterPath>
 
 QPen CDraw::penBorderBlue(QColor(10, 10, 150, 220), 2);
 QPen CDraw::penBorderGray(Qt::lightGray, 2);
diff -ruN qmapshack-V_1.15.0-orig/src/qmapshack/map/CMapIMG.cpp qmapshack-V_1.15.0/src/qmapshack/map/CMapIMG.cpp
--- qmapshack-V_1.15.0-orig/src/qmapshack/map/CMapIMG.cpp	2020-06-14 12:22:49.000000000 +0200
+++ qmapshack-V_1.15.0/src/qmapshack/map/CMapIMG.cpp	2020-07-10 17:06:37.215055842 +0200
@@ -33,6 +33,7 @@
 #include "units/IUnit.h"
 
 #include <QtWidgets>
+#include <QPainterPath>
 
 #undef DEBUG_SHOW_SECT_DESC
 #undef DEBUG_SHOW_TRE_DATA
diff -ruN qmapshack-V_1.15.0-orig/src/qmapshack/mouse/IMouseSelect.cpp qmapshack-V_1.15.0/src/qmapshack/mouse/IMouseSelect.cpp
--- qmapshack-V_1.15.0-orig/src/qmapshack/mouse/IMouseSelect.cpp	2020-06-14 12:22:49.000000000 +0200
+++ qmapshack-V_1.15.0/src/qmapshack/mouse/IMouseSelect.cpp	2020-07-10 17:07:03.928387982 +0200
@@ -23,6 +23,7 @@
 #include "mouse/IScrOpt.h"
 
 #include <QtWidgets>
+#include <QPainterPath>
 
 IMouseSelect::IMouseSelect(CGisDraw *gis, CCanvas *canvas, CMouseAdapter *mouse)
     : IMouse(gis, canvas, mouse)
diff -ruN qmapshack-V_1.15.0-orig/src/qmaptool/helpers/CDraw.cpp qmapshack-V_1.15.0/src/qmaptool/helpers/CDraw.cpp
--- qmapshack-V_1.15.0-orig/src/qmaptool/helpers/CDraw.cpp	2020-06-14 12:22:49.000000000 +0200
+++ qmapshack-V_1.15.0/src/qmaptool/helpers/CDraw.cpp	2020-07-10 17:07:57.688385582 +0200
@@ -24,6 +24,7 @@
 #include <QImage>
 #include <QPointF>
 #include <QtMath>
+#include <QPainterPath>
 
 QPen CDraw::penBorderBlue(QColor(10,10,150,220),2);
 QPen CDraw::penBorderGray(Qt::lightGray,2);
diff -ruN qmapshack-V_1.15.0-orig/src/qmaptool/overlay/COverlayCutMap.cpp qmapshack-V_1.15.0/src/qmaptool/overlay/COverlayCutMap.cpp
--- qmapshack-V_1.15.0-orig/src/qmaptool/overlay/COverlayCutMap.cpp	2020-06-14 12:22:49.000000000 +0200
+++ qmapshack-V_1.15.0/src/qmaptool/overlay/COverlayCutMap.cpp	2020-07-10 17:08:27.798384237 +0200
@@ -24,6 +24,8 @@
 
 #include <functional>
 #include <QtWidgets>
+#include <QPainterPath>
+
 using std::bind;
 
 static inline qreal sqr(qreal a)
