From 9509dff47c44ef321b5133051d949a066310f4bf Mon Sep 17 00:00:00 2001
From: Maxime Gauduin <alucryd@gmail.com>
Date: Mon, 4 Jan 2021 10:46:05 +0100
Subject: [PATCH] use system fribidi

---
 module/renpybidicore.c |  4 ----
 module/setup.py        | 16 ++--------------
 2 files changed, 2 insertions(+), 18 deletions(-)

diff --git a/module/renpybidicore.c b/module/renpybidicore.c
index 18182946b..a85124f28 100644
--- a/module/renpybidicore.c
+++ b/module/renpybidicore.c
@@ -1,10 +1,6 @@
 #include <Python.h>
 
-#ifdef RENPY_BUILD
 #include <fribidi.h>
-#else
-#include <fribidi-src/lib/fribidi.h>
-#endif
 
 #include <stdlib.h>
 
diff --git a/module/setup.py b/module/setup.py
index e3a3dc784..7b1de7cbc 100755
--- a/module/setup.py
+++ b/module/setup.py
@@ -125,29 +125,17 @@ cython(
     sdl + [ png, 'z', 'm' ])
 
 FRIBIDI_SOURCES = """
-fribidi-src/lib/fribidi.c
-fribidi-src/lib/fribidi-arabic.c
-fribidi-src/lib/fribidi-bidi.c
-fribidi-src/lib/fribidi-bidi-types.c
-fribidi-src/lib/fribidi-deprecated.c
-fribidi-src/lib/fribidi-joining.c
-fribidi-src/lib/fribidi-joining-types.c
-fribidi-src/lib/fribidi-mem.c
-fribidi-src/lib/fribidi-mirroring.c
-fribidi-src/lib/fribidi-run.c
-fribidi-src/lib/fribidi-shape.c
 renpybidicore.c
 """.split()
 cython(
     "_renpybidi",
     FRIBIDI_SOURCES,
+    ["fribidi"],
     includes=[
-        BASE + "/fribidi-src/",
-        BASE + "/fribidi-src/lib/",
+        "/usr/include/fribidi/",
         ],
     define_macros=[
         ("FRIBIDI_ENTRY", ""),
-        ("HAVE_CONFIG_H", "1"),
         ])
 
 cython("_renpysteam", language="c++", compile_if=steam_sdk, libs=["steam_api"])
-- 
2.29.2

