diff -rupN renpy-7.1.0-source.old/module/renpybidicore.c renpy-7.1.0-source/module/renpybidicore.c
--- renpy-7.1.0-source.old/module/renpybidicore.c	2018-09-13 14:10:03.102145102 +0200
+++ renpy-7.1.0-source/module/renpybidicore.c	2018-09-13 14:11:07.782646087 +0200
@@ -1,5 +1,5 @@
 #include <Python.h>
-#include <fribidi-src/lib/fribidi.h>
+#include <fribidi.h>
 #include <stdlib.h>
 
 #ifndef alloca
diff -rupN renpy-7.1.0-source.old/module/setup.py renpy-7.1.0-source/module/setup.py
--- renpy-7.1.0-source.old/module/setup.py	2018-09-13 14:10:03.102145102 +0200
+++ renpy-7.1.0-source/module/setup.py	2018-09-13 14:11:51.122981804 +0200
@@ -119,32 +119,19 @@ cython(
     sdl + [ png, 'z', 'm' ])
 
 FRIBIDI_SOURCES = """
-fribidi-src/lib/fribidi.c
-fribidi-src/lib/fribidi-arabic.c
-fribidi-src/lib/fribidi-bidi.c
-fribidi-src/lib/fribidi-bidi-types.c
-fribidi-src/lib/fribidi-deprecated.c
-fribidi-src/lib/fribidi-joining.c
-fribidi-src/lib/fribidi-joining-types.c
-fribidi-src/lib/fribidi-mem.c
-fribidi-src/lib/fribidi-mirroring.c
-fribidi-src/lib/fribidi-run.c
-fribidi-src/lib/fribidi-shape.c
 renpybidicore.c
 """.split()
 cython(
     "_renpybidi",
     FRIBIDI_SOURCES,
+    ['fribidi'],
     includes=[
-        BASE + "/fribidi-src/",
-        BASE + "/fribidi-src/lib/",
+        "/usr/include/fribidi/"
         ],
     define_macros=[
         ("FRIBIDI_ENTRY", ""),
-        ("HAVE_CONFIG_H", "1"),
         ])
 
-
 cython("_renpysteam", language="c++", compile_if=steam_sdk, libs=["steam_api"])
 
 # Sound.
