diff --git a/build/pkgs/sagelib/src/setup.py b/build/pkgs/sagelib/src/setup.py
index 0d7c29d746..fb34cbfd5e 100755
--- a/build/pkgs/sagelib/src/setup.py
+++ b/build/pkgs/sagelib/src/setup.py
@@ -69,10 +69,9 @@ if sdist:
 else:
     distributions = ['']
     optional_packages_with_extensions = ['mcqd', 'bliss', 'tdlib', 'primecount',
-                                         'coxeter3', 'fes', 'sirocco', 'meataxe']
+                                         'coxeter3', 'sirocco', 'meataxe']
     distributions += ['sage-{}'.format(pkg)
-                      for pkg in optional_packages_with_extensions
-                      if is_package_installed_and_updated(pkg)]
+                      for pkg in optional_packages_with_extensions]
     log.warn('distributions = {0}'.format(distributions))
 
 from sage_setup.find import find_python_sources
