--- sage-8.0/src/sage/libs/polybori/pb_wrap.h.0	2017-10-24 07:49:41.623327120 +0000
+++ sage-8.0/src/sage/libs/polybori/pb_wrap.h	2017-10-24 07:50:02.720034688 +0000
@@ -178,9 +178,9 @@
 
 template <class Type>
 class PBWrappedPtr:
-  public boost::shared_ptr<Type> {
+  public std::shared_ptr<Type> {
     typedef PBWrappedPtr self;
-    typedef boost::shared_ptr<Type> base;
+    typedef std::shared_ptr<Type> base;
 
  public:
     PBWrappedPtr(): base() {}
