diff --git a/src/sage/interfaces/gap.py b/src/sage/interfaces/gap.py
index 177fc1b6eb..51c67bdd09 100644
--- a/src/sage/interfaces/gap.py
+++ b/src/sage/interfaces/gap.py
@@ -197,11 +197,11 @@ import string
 
 WORKSPACE = gap_workspace_file()
 
-GAP_BINARY = os.path.join(SAGE_LOCAL, 'bin', 'gap')
+GAP_BINARY = os.path.join(SAGE_LOCAL, 'bin', 'gap-4.8')
 
 first_try = True
 
-gap_cmd = "gap -r"
+gap_cmd = "gap-4.8 -r"
 if platform.processor() == 'ia64' and os.path.exists('/usr/bin/prctl'):
     # suppress unaligned access to 0x..., ip=0x... warnings
     gap_cmd = 'prctl --unaligned=silent ' + gap_cmd
diff --git a/src/sage/libs/gap/util.pyx b/src/sage/libs/gap/util.pyx
index 7bffcfc00d..51dad178d5 100644
--- a/src/sage/libs/gap/util.pyx
+++ b/src/sage/libs/gap/util.pyx
@@ -170,7 +170,7 @@ def gap_root():
     if os.path.exists(GAP_ROOT_DIR):
         return GAP_ROOT_DIR
     print('The gap-4.5.5.spkg (or later) seems to be not installed!')
-    gap_sh = open(os.path.join(SAGE_LOCAL, 'bin', 'gap')).read().splitlines()
+    gap_sh = open(os.path.join(SAGE_LOCAL, 'bin', 'gap-4.8')).read().splitlines()
     gapdir = filter(lambda dir:dir.strip().startswith('GAP_DIR'), gap_sh)[0]
     gapdir = gapdir.split('"')[1]
     gapdir = gapdir.replace('$SAGE_LOCAL', SAGE_LOCAL)
