--- mixxx-2.2.1/build/depends.py.orig	2019-07-26 23:29:29.253916199 +0000
+++ mixxx-2.2.1/build/depends.py	2019-07-26 23:30:03.631040632 +0000
@@ -615,7 +615,7 @@
             env.Append(CPPPATH=[self.SOUNDTOUCH_INTERNAL_PATH])
 
             # Prevents circular import.
-            from features import Optimize
+            from .features import Optimize
 
             # If we do not want optimizations then disable them.
             optimize = (build.flags['optimize'] if 'optimize' in build.flags
@@ -1304,7 +1305,7 @@
             'preferences/dialog/dlgprefvinyldlg.ui',
             'preferences/dialog/dlgprefwaveformdlg.ui',
         ]
-        map(Qt.uic(build), ui_files)
+        list(map(Qt.uic(build), ui_files))

         if build.platform_is_windows:
             # Add Windows resource file with icons and such
