diff -ruN a/bqfft/build/Makefile.inc b/bqfft/build/Makefile.inc
--- a/bqfft/build/Makefile.inc	2019-12-10 10:29:35.000000000 +0100
+++ b/bqfft/build/Makefile.inc	2019-12-12 15:14:52.369303949 +0100
@@ -14,7 +14,7 @@
 
 OPTFLAGS ?= -O3 -ffast-math
 
-CXXFLAGS ?= -std=c++98 -Wall -Wextra -Werror $(FFT_DEFINES) $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) $(OPTFLAGS) -I. $(THIRD_PARTY_INCLUDES) -I../bqvec -fpic
+CXXFLAGS ?= -std=c++98 -Wall -Wextra $(FFT_DEFINES) $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) $(OPTFLAGS) -I. $(THIRD_PARTY_INCLUDES) -I../bqvec -fpic
 
 LIBRARY	:= libbqfft.a
 
diff -ruN a/bqresample/build/Makefile.inc b/bqresample/build/Makefile.inc
--- a/bqresample/build/Makefile.inc	2019-12-10 10:29:37.000000000 +0100
+++ b/bqresample/build/Makefile.inc	2019-12-12 15:15:19.425731836 +0100
@@ -15,7 +15,7 @@
 
 OPTFLAGS ?= -O3 -ffast-math
 
-CXXFLAGS ?= -std=c++98 -fpic -Wall -Wextra -Werror $(RESAMPLE_DEFINES) $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) $(OPTFLAGS) -I. $(THIRD_PARTY_INCLUDES) -I../bqvec
+CXXFLAGS ?= -std=c++98 -fpic -Wall -Wextra $(RESAMPLE_DEFINES) $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) $(OPTFLAGS) -I. $(THIRD_PARTY_INCLUDES) -I../bqvec
 
 CFLAGS	:= -fpic -Wall -Wextra $(RESAMPLE_DEFINES) $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) $(OPTFLAGS) -I. $(THIRD_PARTY_INCLUDES)
 
diff -ruN a/checker/checker.pri b/checker/checker.pri
--- a/checker/checker.pri	2019-12-10 10:29:25.000000000 +0100
+++ b/checker/checker.pri	2019-12-12 15:12:42.627107945 +0100
@@ -2,10 +2,6 @@
 CONFIG += qt stl c++11 exceptions console warn_on 
 QT -= xml network gui widgets
 
-!win32 {
-    QMAKE_CXXFLAGS += -Werror
-}
-
 OBJECTS_DIR = o
 MOC_DIR = o
 
diff -ruN a/checker/helper.pro b/checker/helper.pro
--- a/checker/helper.pro	2019-12-10 10:29:25.000000000 +0100
+++ b/checker/helper.pro	2019-12-12 15:13:00.130288658 +0100
@@ -10,10 +10,6 @@
 
 macx*: CONFIG -= app_bundle
 
-!win32* {
-    QMAKE_CXXFLAGS += -Werror
-}
-
 linux* {
     LIBS += -ldl
 }
diff -ruN a/configure.ac b/configure.ac
--- a/configure.ac	2019-12-10 10:29:11.000000000 +0100
+++ b/configure.ac	2019-12-12 15:11:29.904405648 +0100
@@ -57,7 +57,7 @@
 
 if test "x$GCC" = "xyes"; then
    	CXXFLAGS_ANY="-fpic -Wall -Wextra -Woverloaded-virtual -Wformat-nonliteral -Wformat-security -Winit-self -Wswitch-enum -Wconversion -pipe"
-        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -g -Werror"
+        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -g"
         OPTFLAGS_DEBUG="-O2"
    	CXXFLAGS_RELEASE="$CXXFLAGS_ANY -g0"
         OPTFLAGS_RELEASE="-O3 -ffast-math"
diff -ruN a/dataquay/config.pri b/dataquay/config.pri
--- a/dataquay/config.pri	2019-12-10 10:29:32.000000000 +0100
+++ b/dataquay/config.pri	2019-12-12 15:15:30.962296540 +0100
@@ -8,6 +8,6 @@
 
 # Define this to use the Sord datastore (http://drobilla.net/software/sord/)
 #DEFINES += USE_SORD
-#QMAKE_CXXFLAGS += -I/usr/include/sord-0 -I/usr/include/serd-0 -Werror
+#QMAKE_CXXFLAGS += -I/usr/include/sord-0 -I/usr/include/serd-0
 #EXTRALIBS += -lsord-0 -lserd-0
 
diff -ruN a/piper-vamp-cpp/Makefile b/piper-vamp-cpp/Makefile
--- a/piper-vamp-cpp/Makefile	2019-12-10 10:29:29.000000000 +0100
+++ b/piper-vamp-cpp/Makefile	2019-12-12 15:13:45.643223896 +0100
@@ -7,7 +7,7 @@
 OPTFLAGS	:= -O3 -fPIC
 
 CFLAGS		:= -Wall -std=c99 $(OPTFLAGS) $(INCFLAGS)
-CXXFLAGS	:= -Wall -Wextra -Werror -Wno-error=unused-parameter -std=c++1y $(OPTFLAGS) $(INCFLAGS)
+CXXFLAGS	:= -Wall -Wextra -Wno-error=unused-parameter -std=c++1y $(OPTFLAGS) $(INCFLAGS)
 
 CXX		?= c++
 
diff -ruN a/piper-vamp-cpp/ext/json11/CMakeLists.txt b/piper-vamp-cpp/ext/json11/CMakeLists.txt
--- a/piper-vamp-cpp/ext/json11/CMakeLists.txt	2019-12-10 10:29:29.000000000 +0100
+++ b/piper-vamp-cpp/ext/json11/CMakeLists.txt	2019-12-12 15:14:32.436146202 +0100
@@ -29,7 +29,7 @@
 
 # Set warning flags, which may vary per platform
 include(CheckCXXCompilerFlag)
-set(_possible_warnings_flags /W4 /WX -Wextra -Werror)
+set(_possible_warnings_flags /W4 /WX -Wextra)
 foreach(_warning_flag in ${_possible_warnings_flags})
   CHECK_CXX_COMPILER_FLAG(_warning_flag _flag_supported)
   if(${_flag_supported})
diff -ruN a/piper-vamp-cpp/vamp-client/qt/test.pro b/piper-vamp-cpp/vamp-client/qt/test.pro
--- a/piper-vamp-cpp/vamp-client/qt/test.pro	2019-12-10 10:29:29.000000000 +0100
+++ b/piper-vamp-cpp/vamp-client/qt/test.pro	2019-12-12 15:14:09.346349232 +0100
@@ -4,10 +4,6 @@
 CONFIG += qt stl c++11 c++14 exceptions console warn_on
 QT -= xml network gui widgets
 
-!win32 {
-    QMAKE_CXXFLAGS += -Werror
-}
-
 OBJECTS_DIR = ../o
 MOC_DIR = ../o
 
diff -ruN a/svapp/configure.ac b/svapp/configure.ac
--- a/svapp/configure.ac	2019-12-10 10:29:23.000000000 +0100
+++ b/svapp/configure.ac	2019-12-12 15:12:14.114022634 +0100
@@ -54,7 +54,7 @@
 
 if test "x$GCC" = "xyes"; then
    	CXXFLAGS_ANY="-Wall -Wextra -Woverloaded-virtual -Wformat-nonliteral -Wformat-security -Winit-self -Wswitch-enum -Wconversion -pipe"
-        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -Werror -g"
+        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -g"
    	CXXFLAGS_RELEASE="$CXXFLAGS_ANY -g0 -O2"
    	CXXFLAGS_MINIMAL="$CXXFLAGS_ANY -g0 -O0"
 fi
diff -ruN a/svcore/configure.ac b/svcore/configure.ac
--- a/svcore/configure.ac	2019-12-10 10:29:18.000000000 +0100
+++ b/svcore/configure.ac	2019-12-12 15:12:00.407474914 +0100
@@ -54,7 +54,7 @@
 
 if test "x$GCC" = "xyes"; then
    	CXXFLAGS_ANY="-Wall -Wextra -Woverloaded-virtual -Wformat-nonliteral -Wformat-security -Winit-self -Wswitch-enum -Wconversion -pipe"
-        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -Werror -g"
+        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -g"
    	CXXFLAGS_RELEASE="$CXXFLAGS_ANY -g0 -O2"
    	CXXFLAGS_MINIMAL="$CXXFLAGS_ANY -g0 -O0"
 fi
diff -ruN a/svgui/configure.ac b/svgui/configure.ac
--- a/svgui/configure.ac	2019-12-10 10:29:21.000000000 +0100
+++ b/svgui/configure.ac	2019-12-12 15:11:44.720944164 +0100
@@ -54,7 +54,7 @@
 
 if test "x$GCC" = "xyes"; then
    	CXXFLAGS_ANY="-Wall -Wextra -Woverloaded-virtual -Wformat-nonliteral -Wformat-security -Winit-self -Wswitch-enum -Wconversion -pipe"
-        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -Werror -g"
+        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -g"
    	CXXFLAGS_RELEASE="$CXXFLAGS_ANY -g0 -O2"
    	CXXFLAGS_MINIMAL="$CXXFLAGS_ANY -g0 -O0"
 fi
