--- springlobby-0.267/src/springsettings/tab_abstract.cpp.orig	2019-03-26 07:45:27.981903360 +0000
+++ springlobby-0.267/src/springsettings/tab_abstract.cpp	2019-03-26 07:45:42.628735154 +0000
@@ -95,7 +95,7 @@
 {
 	try {
 		//special treatment for resolution that'll set proper defaults for res
-		wxDisplay display(0);
+		wxDisplay display(0U);
 		//		wxRect display_rect ( display.GetGeometry() );
 		//		const int current_x_res = LSL::susynclib().GetSpringConfigInt(RC_TEXT[0].key,display_rect.width);
 		//		const int current_y_res = LSL::susynclib().GetSpringConfigInt(RC_TEXT[1].key,display_rect.height);
