diff --git a/include/srsgui/common/WaterfallData.h b/include/srsgui/common/WaterfallData.h
index a289c6d..5cde564 100755
--- a/include/srsgui/common/WaterfallData.h
+++ b/include/srsgui/common/WaterfallData.h
@@ -2,6 +2,7 @@
 #define WATERFALLDATA_H
 
 #include <qwt_raster_data.h>
+#include <qwt_interval.h>
 #ifndef Q_MOC_RUN
 #include <boost/circular_buffer.hpp>
 #include <boost/shared_ptr.hpp>
@@ -65,6 +66,16 @@ public:
     return *(std::min_element(minVec.begin(),minVec.end()));
   }
 
+  QwtInterval interval( Qt::Axis axis) const
+  {
+    return d_intervals[axis];
+  }
+
+  void setInterval( Qt::Axis axis, const QwtInterval &interval )
+  {
+    d_intervals[axis] = interval;
+  }
+
   double value(double x, double y) const
   {
     double bottom = interval(Qt::YAxis).minValue();
@@ -88,6 +99,7 @@ private:
   VecPtrBuf data_;
   int nData_;
   int nRows_;
+  QwtInterval d_intervals[3];
 };
 
 #endif // WATERFALLDATA_H
diff --git a/src/common/Spectrogramplot.cpp b/src/common/Spectrogramplot.cpp
index ca9fad9..92a25d5 100755
--- a/src/common/Spectrogramplot.cpp
+++ b/src/common/Spectrogramplot.cpp
@@ -8,6 +8,8 @@
 #include <qwt_plot_canvas.h>
 #include "srsgui/common/Spectrogramplot.h"
 
+#include <QPen>
+
 class MyZoomer: public QwtPlotZoomer
 {
 public:
@@ -101,8 +103,8 @@ Spectrogramplot::Spectrogramplot(int numDataPoints, int numRows, QWidget *parent
   sd->setMinimumExtent( fm.width("100.00") );
 
   const QColor c(Qt::darkBlue);
-  zoomer_->setRubberBandPen(c);
-  zoomer_->setTrackerPen(c);
+  zoomer_->setRubberBandPen(QPen(c));
+  zoomer_->setTrackerPen(QPen(c));
 }
 
 void Spectrogramplot::appendData(double* data, int n)
