From 782dd242ddb50a77f999c93c96e98ae028bf6ab7 Mon Sep 17 00:00:00 2001
From: Antonio Rojas <arojas@archlinux.org>
Date: Sat, 24 Dec 2022 11:14:31 +0100
Subject: [PATCH] Restore libgcc_s symlinkin in !macOS

Commit c8b72e2bf49046e8daca64214765694377277947 completely removed libgcc_s symlinking (I assume unintentionally) in !macOS.
---
 base/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/base/Makefile b/base/Makefile
index bb79549aeea2..c320465a1049 100644
--- a/base/Makefile
+++ b/base/Makefile
@@ -227,6 +227,8 @@ else
 $(eval $(call symlink_system_library,CSL,libgcc_s,1))
 endif
 endif
+else
+$(eval $(call symlink_system_library,CSL,libgcc_s,1))
 endif
 ifneq (,$(LIBGFORTRAN_VERSION))
 $(eval $(call symlink_system_library,CSL,libgfortran,$(LIBGFORTRAN_VERSION)))
