--- transcode-1.1.7/filter/filter_resample.c.orig	2018-12-15 20:00:16.111657609 +0000
+++ transcode-1.1.7/filter/filter_resample.c	2018-12-15 20:01:13.141800445 +0000
@@ -37,7 +37,7 @@
 #include "libtc/optstr.h"
 #include "libtc/tcavcodec.h"
 #include "libtc/tcmodule-plugin.h"
-#include <libavresample/avresample.h>
+#include <libswresample/swresample.h>
 
 
 typedef struct {
@@ -46,7 +46,7 @@
 
     int bytes_per_sample;
 
-    AVAudioResampleContext *resample_ctx;
+    SwrContext *resample_ctx;
 } ResamplePrivateData;
 
 static const char resample_help[] = ""
