From b63f3afa42a43e2a261a989f87e9ccd1e18caaa5 Mon Sep 17 00:00:00 2001
From: GitLab Bot <gitlab-bot@gitlab.com>
Date: Sat, 2 May 2020 18:09:45 +0000
Subject: [PATCH] Add latest changes from gitlab-org/gitlab@master

---
 Gemfile                     | 2 +-
 Gemfile.lock                | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Gemfile b/Gemfile
index 6bd15a9a190..18d8e4586bd 100644
--- a/Gemfile
+++ b/Gemfile
@@ -204,7 +204,7 @@ gem 'ruby-progressbar'
 gem 'settingslogic', '~> 2.0.9'
 
 # Linear-time regex library for untrusted regular expressions
-gem 're2', '~> 1.1.1'
+gem 're2', '~> 1.2.0'
 
 # Misc
 
diff --git a/Gemfile.lock b/Gemfile.lock
index e796144f62a..eb1a0eb5a27 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -854,7 +854,7 @@ GEM
       msgpack (>= 0.4.3)
       optimist (>= 3.0.0)
     rdoc (6.1.2)
-    re2 (1.1.1)
+    re2 (1.2.0)
     recaptcha (4.13.1)
       json
     recursive-open-struct (1.1.0)
@@ -1346,7 +1346,7 @@ DEPENDENCIES
   rblineprof (~> 0.3.6)
   rbtrace (~> 0.4)
   rdoc (~> 6.1.2)
-  re2 (~> 1.1.1)
+  re2 (~> 1.2.0)
   recaptcha (~> 4.11)
   redis (~> 4.0)
   redis-namespace (~> 1.6.0)
