diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 023bbe9e8d..e25aa0aa7f 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1308,7 +1308,7 @@ if(USE_DISTRIBUTED AND USE_TENSORPIPE)
   if(MSVC)
     message(WARNING "Tensorpipe cannot be used on Windows.")
   else()
-    set(TP_BUILD_LIBUV ON CACHE BOOL "" FORCE)
+    set(TP_BUILD_LIBUV OFF CACHE BOOL "" FORCE)
     set(TP_ENABLE_SHM OFF CACHE BOOL "" FORCE)
     set(TP_ENABLE_CMA OFF CACHE BOOL "" FORCE)
     set(TP_STATIC_OR_SHARED STATIC CACHE STRING "" FORCE)
