diff --git i/valabindwriter.vala w/valabindwriter.vala
index 165cdf5..d3911b3 100644
--- i/valabindwriter.vala
+++ w/valabindwriter.vala
@@ -27,13 +27,17 @@ public class ValabindWriter : CodeVisitor {
 		this.vapidir = vapidir;
 		this.glibmode = glibmode;
 		context.vapi_directories = { vapidir };
+#if VALA_0_50
+		context.set_target_profile (Profile.GOBJECT);
+#else
 		add_package (context, "glib-2.0");
 		add_package (context, "gobject-2.0");
 		/* vala 0.17 only support gobject profile */
 		if (glibmode)
 			context.add_define ("GOBJECT");
 		// required to avoid ugly runtime errors
 		context.profile = Profile.GOBJECT;
+#endif
 	}
 
 	public void parse () {
