diff -rup xautomation-1.05.orig/Makefile.am xautomation-1.05/Makefile.am
--- xautomation-1.05.orig/Makefile.am	2009-04-14 18:06:23.000000000 +0200
+++ xautomation-1.05/Makefile.am	2012-01-27 02:06:44.016492587 +0100
@@ -1,7 +1,7 @@
 bin_PROGRAMS = xte rgb2pat png2pat visgrep pat2ppm patextract xmousepos
 man_MANS = pat2ppm.1 patextract.1 png2pat.1 rgb2pat.1 visgrep.1 xautomation.7 xmousepos.1 xte.1
 xte_SOURCES = xte.c debug.c kbd.h keysym_map.h multikey_map.h
-xte_LDADD = $(X_LIBS) $(X_EXTRALIBS) -lXtst
+xte_LDADD = $(X_LIBS) $(X_EXTRALIBS) -lXtst -lX11
 rgb2pat_SOURCES = rgb2pat.c image.c image.h debug.c debug.h
 rgb2pat_LDADD = -lpng
 png2pat_SOURCES = png2pat.c image.c image.h debug.c debug.h
diff -rup xautomation-1.05.orig/Makefile.in xautomation-1.05/Makefile.in
--- xautomation-1.05.orig/Makefile.in	2011-09-07 00:48:19.000000000 +0200
+++ xautomation-1.05/Makefile.in	2012-01-27 02:14:02.010831581 +0100
@@ -220,7 +220,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 man_MANS = pat2ppm.1 patextract.1 png2pat.1 rgb2pat.1 visgrep.1 xautomation.7 xmousepos.1 xte.1
 xte_SOURCES = xte.c debug.c kbd.h keysym_map.h multikey_map.h
-xte_LDADD = $(X_LIBS) $(X_EXTRALIBS) -lXtst
+xte_LDADD = $(X_LIBS) $(X_EXTRALIBS) -lXtst -lX11
 rgb2pat_SOURCES = rgb2pat.c image.c image.h debug.c debug.h
 rgb2pat_LDADD = -lpng
 png2pat_SOURCES = png2pat.c image.c image.h debug.c debug.h
