diff --git a/Makefile.am b/Makefile.am
index e59a6b30..111ac662 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -12,7 +12,6 @@ full:
 EXTRA_DIST = \
 	README.md \
 	HACKING \
-	gnome-doc-utils.make \
 	ChangeLog.0 \
 	viking.spec \
 	mingw-viking.spec \
@@ -29,9 +28,6 @@ generate-changelog:
 dist-hook: viking.spec generate-changelog
 	cp $(top_builddir)/viking.spec $(distdir)
 
-MAINTAINERCLEANFILES = \
-	gnome-doc-utils.make
-
 DISTCLEANFILES = \
 	intltool-extract \
 	intltool-merge \
@@ -39,8 +35,5 @@ DISTCLEANFILES = \
 
 ACLOCAL_AMFLAGS = -I m4
 
-DISTCHECK_CONFIGURE_FLAGS = \
-	--disable-scrollkeeper
-
 # Ignore gtk theme cache files on distcheck
 distuninstallcheck_listfiles = find . -type f -print | grep -v 'icon-theme.cache'
diff --git a/configure.ac b/configure.ac
index 94b7bff8..4feaa7d8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -111,7 +111,7 @@ AC_SUBST(PACKAGE_LIBS)
 dnl  -------------
 dnl | User Manual |---------------------------------------
 dnl  -------------
-GNOME_DOC_INIT
+YELP_HELP_INIT
 
 dnl ---------------------------------------------------------------------------
 dnl - Use deprecated options (default enabled for devs, disabled in releases) 
@@ -553,23 +553,7 @@ AC_DEFINE_UNQUOTED(VIK_CONFIG_DEFAULT_TILE_AGE, ${VIK_CONFIG_DEFAULT_TILE_AGE},
 AC_DEFINE(HAVE_VIKING, 1, [Enable Viking specifics in otherwise reusable code])
 
 dnl man pages processing
-dnl Different distributions have differing locations for the docbook.xsl
-dnl Debian like
-if test -r /usr/share/xml/docbook/stylesheet/docbook-xsl/manpages/docbook.xsl; then
-  DB2MAN_XSL=/usr/share/xml/docbook/stylesheet/docbook-xsl/manpages/docbook.xsl
-dnl RedHat like
-elif test -r /usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl; then
-  DB2MAN_XSL=/usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl
-else
-dnl Old default
-  DB2MAN_XSL=/usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl
-fi
-AC_SUBST(DB2MAN_XSL)
-AC_PATH_PROG(XP,xsltproc)
-AM_CONDITIONAL([HAVE_XSLTPROC],[test "x$XP" != "x"])
-AC_CHECK_PROG([HAVE_SCROLLKEEPER],scrollkeeper-config,"yes")
-AM_CONDITIONAL([GEN_MANPAGES],[test "x$XP" != "x" && test -r "$DB2MAN_XSL"] && test "${HAVE_SCROLLKEEPER}" = "yes" )
-AM_COND_IF([GEN_MANPAGES], [ac_cv_enable_man=yes], [ac_cv_enable_man=no])
+AC_PATH_PROG(XSLTPROC,xsltproc)
 
 ISODATE="$(date --utc --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}" +%Y-%m-%d)"
 AC_SUBST(ISODATE)
@@ -627,7 +611,6 @@ OAuth                            : $ac_cv_enable_oauth
 Size of map cache (in memory)    : ${VIK_CONFIG_MAPCACHE_SIZE}
 Age of tiles (in seconds)        : ${VIK_CONFIG_DEFAULT_TILE_AGE}
 GeoNames user                    : ${VIK_CONFIG_GEONAMES_USERNAME}
-Man page generation              : $ac_cv_enable_man
 Documentation (+HTML)            : ${enable_gtk_doc} (HTML: ${enable_gtk_doc_html})
 -------------------------------------------
 
diff --git a/help/Makefile.am b/help/Makefile.am
index 9ef48691..f7e5b38d 100644
--- a/help/Makefile.am
+++ b/help/Makefile.am
@@ -1,6 +1,4 @@
-if GEN_MANPAGES
-include $(top_srcdir)/gnome-doc-utils.make
-endif
+@YELP_HELP_RULES@
 
 # Annoyingly in the XML, yelp refuses to load relative images fileref, i.e. ../../src/icons/file.png
 #  whereas the PDF generation inserts these references OK
@@ -11,17 +9,18 @@ get-icons:
 # '-' ^^^^^ means it doesn't care about the exit status
 #  this copy statement is not critical (especially when run under make distcheck)
 
-dist-hook: get-icons doc-dist-hook
+dist-hook: get-icons
 
-DOC_MODULE = viking
-DOC_ENTITIES = legal.xml \
+HELP_ID = viking
+HELP_FILES = index.docbook
+HELP_EXTRA = legal.xml \
     attribution.xml \
     commandline.xml \
+    geoclue_layer.xml \
     georef_layer.xml \
     mapnik_rendering_layer.xml \
     recommends.xml
-DOC_INCLUDES =
-DOC_FIGURES = \
+HELP_MEDIA = \
     figures/Aggregate_statistics.png \
     figures/Aggregate_statistics_years.png \
     figures/addtr_18.png \
@@ -57,20 +56,17 @@ DOC_FIGURES = \
     figures/Viking-TAC.png \
     figures/DEM_file_info_dialog.png \
     figures/customize_toolbar_dialog.png
-DOC_LINGUAS =
+HELP_LINGUAS =
 
 EXTRA_DIST = viking.xml.in
 
 CLEANFILES=
 
 # man pages processing
-if GEN_MANPAGES
 MANPAGES=viking.1
 man_MANS = $(MANPAGES)
 CLEANFILES+=$(MANPAGES)
-endif
-DB2MAN=@DB2MAN_XSL@
-XP=@XP@ -''-nonet
+XSLTPROC=@XSLTPROC@ -''-nonet
 
 %.1: %.xml
-	$(XP) -o $@ $(DB2MAN) $<
+	$(XSLTPROC) -o $@ http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl $<
diff --git a/src/Makefile.am b/src/Makefile.am
index 37528972..78bda6b8 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -16,16 +16,8 @@ authors.h: $(top_srcdir)/AUTHORS
 	sed -e "s/ \*\*.*//" -e 's/^/"/' -e 's/$$/",\\/' -e '$$s/\\$$//' $< >> $@
 	echo "NULL};" >> $@
 
-if HAVE_XSLTPROC
-documenters.h: $(top_srcdir)/help/C/viking.xml
-	$(XP) $(srcdir)/docbook2documenters.xsl $(top_srcdir)/help/C/viking.xml > $@
-else
-documenters.h:
-	rm -f $@
-	echo "/* Generated file. */" >> $@
-	echo "const gchar *DOCUMENTERS[] = {\\" >> $@
-	echo "NULL};" >> $@
-endif
+documenters.h: $(top_srcdir)/help/C/index.docbook
+	$(XSLTPROC) $(srcdir)/docbook2documenters.xsl $(top_srcdir)/help/C/index.docbook > $@
 
 BUILT_SOURCES = authors.h documenters.h
 
