Index: cherokee/header.c
===================================================================
--- cherokee/header.c	(revision 3965)
+++ cherokee/header.c	(working copy)
@@ -648,12 +648,9 @@
 		return ret;
 	}
 
-	ret = cherokee_buffer_unescape_uri (request);
-	if (unlikely (ret != ret_ok)) {
-		return ret_error;
-	}
+	sanitize_buffer (request);
 
-	return sanitize_buffer (request);
+	return cherokee_buffer_unescape_uri (request);
 }
 
 
