commit 3d1d3cb6ac53a15996a8fc25df25c91e451f200b
Author: Etienne Laurin <atnnn@rethinkdb.com>
Date:   Thu Oct 2 18:55:36 2014 -0700

    allow building with protobuf 2.6.0
    
    OTS review by mlucy
    Closes #3137

diff --git a/src/rdb_protocol/ql2_extensions.proto b/src/rdb_protocol/ql2_extensions.proto
index 7cdfb2d..a2016b2 100644
--- a/src/rdb_protocol/ql2_extensions.proto
+++ b/src/rdb_protocol/ql2_extensions.proto
@@ -3,5 +3,5 @@ package ql2.extension;
 import "rdb_protocol/ql2.proto";
 
 extend Term {
-    required Backtrace backtrace = 10000;
+    optional Backtrace backtrace = 10000;
 };
