--- a/Makefile
+++ b/Makefile
@@ -48,10 +48,11 @@
 
 # END system dependant block
 
-SSL_LIBS := $(shell pkg-config --libs libssl 2>/dev/null)
-ifeq ($(SSL_LIBS),)
+# We're not using libssl, so comment out the following line
+# SSL_LIBS := $(shell pkg-config --libs libssl 2>/dev/null)
+# ifeq ($(SSL_LIBS),)
 SSL_LIBS := $(shell pkg-config --libs openssl 2>/dev/null)
-endif
+# endif
 ifeq ($(SSL_LIBS),)
 SSL_LIBS := -lssl -lcrypto
 endif
