--- a/core/base/file.cpp
+++ b/core/base/file.cpp
@@ -20,6 +20,7 @@
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <string.h>
 
 #include <string>
 
--- a/core/liblog/log_read.c
+++ b/core/liblog/log_read.c
@@ -26,6 +26,7 @@
 #include <string.h>
 #include <sys/cdefs.h>
 #include <unistd.h>
+#include <sys/user.h>
 
 #include <cutils/list.h>
 #include <cutils/sockets.h>

--- a/libselinux/src/label_internal.h
+++ b/libselinux/src/label_internal.h
@@ -10,6 +10,7 @@
 
 #include <stdlib.h>
 #include <stdarg.h>
+#include <stdbool.h>
 #include <selinux/selinux.h>
 #include <selinux/label.h>
 #include "dso.h"
