=== modified file 'duplicity/backends/u1backend.py'
--- duplicity/backends/u1backend.py	2012-10-27 12:16:19 +0000
+++ duplicity/backends/u1backend.py	2012-11-03 21:39:23 +0000
@@ -207,7 +207,7 @@
                    "Content-Type": content_type}
         resp, content = self.client.request(remote_full,
                                             method="PUT",
-                                            body=str(data),
+                                            body=bytearray(data),
                                             headers=headers)
 
     def get(self, filename, local_path):

