diff -rup pion-net-4.0.12/configure pion-net-4.0.12.new/configure
--- pion-net-4.0.12/configure	2012-03-24 08:00:10.000000000 +0100
+++ pion-net-4.0.12.new/configure	2012-07-07 16:37:06.805347251 +0200
@@ -22197,7 +22197,7 @@ LIBS="$LIBS_SAVED"
 BOOST_TRY_LIB=thread
 BOOST_TRY_LINK="boost_${BOOST_TRY_LIB}${BOOST_LIB_EXTENSION}"
 LIBS_SAVED="$LIBS"
-LIBS="$LIBS_SAVED -l${BOOST_TRY_LINK} ${BOOST_DATE_TIME_LIB}"
+LIBS="$LIBS_SAVED -lboost_system -l${BOOST_TRY_LINK} ${BOOST_DATE_TIME_LIB}"
 echo "$as_me:$LINENO: checking for boost::${BOOST_TRY_LIB} library" >&5
 echo $ECHO_N "checking for boost::${BOOST_TRY_LIB} library... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
