diff -aur pd-0.45-3/configure.ac pd-0.45-3.fftw3/configure.ac
--- pd-0.45-3/configure.ac	2013-10-03 18:22:29.000000000 +0200
+++ pd-0.45-3.fftw3/configure.ac	2013-10-24 02:14:09.555763309 +0200
@@ -212,13 +212,13 @@
 AM_CONDITIONAL(PORTMIDI, test x$portmidi = xyes)
 if test x$portmidi = xyes; then echo Using included portmidi; fi
 
-dnl fftw v2
+dnl fftw v3
 AC_ARG_ENABLE([fftw],
 	[AS_HELP_STRING([--enable-fftw],
 		[use FFTW package])],
     [fftw=$enableval])
 if test x$fftw = xyes; then
-    AC_CHECK_LIB(fftw, fftw_one, [LIBS="$LIBS -lfftw"],
+    AC_CHECK_LIB(fftw3f, fftwf_execute, [LIBS="$LIBS -lfftw3f"],
         [AC_MSG_NOTICE([fftw package not found - using built-in FFT]); fftw=no])
 fi
 AM_CONDITIONAL(FFTW, test x$fftw = xyes)
