diff --git c/pyproject.toml w/pyproject.toml
index ce03cd2..9468b8c 100644
--- c/pyproject.toml
+++ w/pyproject.toml
@@ -1,9 +1,7 @@
 [build-system]
 requires = [
-  "setuptools >= 42.0.0",  # required by pyproject+setuptools_scm integration
-  "setuptools_scm[toml] >= 3.5.0",  # required for "no-local-version" scheme
-  "setuptools_scm_git_archive >= 1.0",
-  "wheel",
+  "setuptools >= 45.0.0",  # required by pyproject+setuptools_scm integration
+  "setuptools_scm[toml] >= 7.0.0",  # required for "no-local-version" scheme
 ]
 build-backend = "setuptools.build_meta"
 
