--- a/main/main.js
+++ b/main/main.js
@@ -161,7 +161,7 @@
     minWidth: (process.platform === 'win32' ? 400 : 320), // controls take up more horizontal space on Windows
     minHeight: 350,
     titleBarStyle: 'hiddenInset',
-    icon: __dirname + '/icons/icon256.png',
+    icon: '/usr/share/icons/hicolor/256x256/apps/min.png',
     frame: process.platform === 'darwin' || settings.get('useSeparateTitlebar') === true,
     alwaysOnTop: settings.get('windowAlwaysOnTop'),
     backgroundColor: '#fff', // the value of this is ignored, but setting it seems to work around https://github.com/electron/electron/issues/10559
