--- a/main/main.js
+++ b/main/main.js
@@ -181,7 +181,7 @@
     minHeight: 350,
     titleBarStyle: settings.get('useSeparateTitlebar') ? 'default' : 'hidden',
     trafficLightPosition: { x: 12, y: 10 },
-    icon: __dirname + '/icons/icon256.png',
+    icon: '/usr/share/icons/hicolor/256x256/apps/min.png',
     frame: settings.get('useSeparateTitlebar'),
     alwaysOnTop: settings.get('windowAlwaysOnTop'),
     backgroundColor: '#fff', // the value of this is ignored, but setting it seems to work around https://github.com/electron/electron/issues/10559
