--- gpick-gpick-0.2.6rc1/tools/gpick.py.orig	2019-07-26 18:58:37.277863800 +0000
+++ gpick-gpick-0.2.6rc1/tools/gpick.py	2019-07-26 18:58:51.344706165 +0000
@@ -6,12 +6,12 @@
 import string
 import glob
 import subprocess
-from lemon import *
-from flex import *
-from gettext import *
-from resource_template import *
-from ragel import *
-from template import *
+from .lemon import * 
+from .flex import * 
+from .gettext import *
+from .resource_template import *
+from .ragel import *
+from .template import *
 from SCons.Script import *
 from SCons.Util import *
 from SCons.Script.SConscript import SConsEnvironment
@@ -100,9 +100,9 @@
 	def ConfirmPrograms(self, conf, programs):
 		conf.AddTests({'CheckProgram': CheckProgram})
 		
-		for evar, args in programs.iteritems():
+		for evar, args in programs.items():
 			found = False
-			for name, member_name in args['checks'].iteritems():
+			for name, member_name in args['checks'].items():
 				if conf.CheckProgram(self, name, member_name):
 					found = True;
 					break
@@ -116,9 +116,9 @@
 	def ConfirmLibs(self, conf, libs):
 		conf.AddTests({'CheckPKG': CheckPKG})
 		
-		for evar, args in libs.iteritems():
+		for evar, args in libs.items():
 			found = False
-			for name, version in args['checks'].iteritems():
+			for name, version in args['checks'].items():
 				if conf.CheckPKG(name + ' ' + version):
 					self[evar]=name
 					found = True;
@@ -155,9 +155,9 @@
 				self.AddPostAction(i, Chmod(i, perm))
 		return dir
 
-	InstallProgram = lambda self, dir, source: GpickEnvironment.InstallPerm(self, dir, source, 0755)
-	InstallData = lambda self, dir, source: GpickEnvironment.InstallPerm(self, dir, source, 0644)
-	InstallDataAutoDir = lambda self, dir, relative_dir, source: GpickEnvironment.InstallPermAutoDir(self, dir, relative_dir, source, 0644)
+	InstallProgram = lambda self, dir, source: GpickEnvironment.InstallPerm(self, dir, source, '0755')
+	InstallData = lambda self, dir, source: GpickEnvironment.InstallPerm(self, dir, source, '0644')
+	InstallDataAutoDir = lambda self, dir, relative_dir, source: GpickEnvironment.InstallPermAutoDir(self, dir, relative_dir, source, '0644')
 
 	def GetSourceFiles(self, dir_exclude_pattern, file_exclude_pattern):
 		dir_exclude_prog = re.compile(dir_exclude_pattern)
--- gpick-gpick-0.2.6rc1/SConscript.orig	2019-07-26 22:26:26.884324858 +0000
+++ gpick-gpick-0.2.6rc1/SConscript	2019-07-26 22:27:07.534897504 +0000
@@ -49,19 +49,19 @@
 env.GetVersionInfo()
 
 try:
-	umask = os.umask(022)
+	umask = os.umask(0o22)
 except OSError:     # ignore on systems that don't support umask
 	pass
 
-if os.environ.has_key('CC'):
+if os.environ.get('CC'):
 	env['CC'] = os.environ['CC']
-if os.environ.has_key('CFLAGS'):
+if os.environ.get('CFLAGS'):
 	env['CCFLAGS'] += SCons.Util.CLVar(os.environ['CFLAGS'])
-if os.environ.has_key('CXX'):
+if os.environ.get('CXX'):
 	env['CXX'] = os.environ['CXX']
-if os.environ.has_key('CXXFLAGS'):
+if os.environ.get('CXXFLAGS'):
 	env['CXXFLAGS'] += SCons.Util.CLVar(os.environ['CXXFLAGS'])
-if os.environ.has_key('LDFLAGS'):
+if os.environ.get('LDFLAGS'):
 	env['LINKFLAGS'] += SCons.Util.CLVar(os.environ['LDFLAGS'])
 
 if not env.GetOption('clean'):
@@ -97,7 +97,7 @@
 Decider('MD5-timestamp')
 
 if not env['TOOLCHAIN'] == 'msvc':
-	if not (os.environ.has_key('CFLAGS') or os.environ.has_key('CXXFLAGS') or os.environ.has_key('LDFLAGS')):
+	if not (os.environ.get('CFLAGS') or os.environ.get('CXXFLAGS') or os.environ.get('LDFLAGS')):
 		if env['DEBUG']:
 			env.Append(
 				CPPFLAGS = ['-Wall', '-g3', '-O0'],
--- gpick-gpick-0.2.6rc1/source/dbus/SConscript.orig	2019-07-26 22:38:12.841066003 +0000
+++ gpick-gpick-0.2.6rc1/source/dbus/SConscript	2019-07-26 22:39:19.508817931 +0000
@@ -9,7 +9,7 @@
 if not local_env.GetOption('clean') and not env['TOOLCHAIN'] == 'msvc':
 	if not env['USE_GTK3']:
 		local_env.ParseConfig('pkg-config --cflags $GIO_PC')
-        else:
+	else:
 		local_env.ParseConfig('pkg-config --cflags $GTK_PC')
 
 if not env['BUILD_TARGET'] == 'win32':
