diff --git a/plugins/datetime/Makefile.am b/plugins/datetime/Makefile.am
index 7c3b76a..161ed5a 100644
--- a/plugins/datetime/Makefile.am
+++ b/plugins/datetime/Makefile.am
@@ -1,5 +1,5 @@
 dbus_servicesdir = $(datadir)/dbus-1/system-services
-dbus_confdir = $(sysconfdir)/dbus-1/system.d
+dbus_confdir = $(DBUS_SYS_DIR)
 polkitdir = $(datadir)/polkit-1/actions
 
 dbus_services_in_files = org.mate.SettingsDaemon.DateTimeMechanism.service.in
