diff --git a/plugins/media-keys/msd-media-keys-manager.c b/plugins/media-keys/msd-media-keys-manager.c
index ad7665d..cc819c5 100644
--- a/plugins/media-keys/msd-media-keys-manager.c
+++ b/plugins/media-keys/msd-media-keys-manager.c
@@ -999,7 +999,9 @@ do_action (MsdMediaKeysManager *manager,
                 do_media_action (manager);
                 break;
         case CALCULATOR_KEY:
-                if ((cmd = g_find_program_in_path ("mate-calc"))) {
+                if ((cmd = g_find_program_in_path ("galculator"))) {
+                        execute (manager, "galculator", FALSE, FALSE);
+                } else if ((cmd = g_find_program_in_path ("mate-calc"))) {
                         execute (manager, "mate-calc", FALSE, FALSE);
                 } else {
                         execute (manager, "gcalctool", FALSE, FALSE);
