diff -ruN QScintilla-gpl-2.9.orig/designer-Qt4Qt5/qscintillaplugin.h QScintilla-gpl-2.9/designer-Qt4Qt5/qscintillaplugin.h
--- QScintilla-gpl-2.9.orig/designer-Qt4Qt5/qscintillaplugin.h	2015-09-19 14:41:37.557741350 +0300
+++ QScintilla-gpl-2.9/designer-Qt4Qt5/qscintillaplugin.h	2015-09-19 14:41:58.393662846 +0300
@@ -4,7 +4,7 @@
 #ifndef _QSCINTILLAPLUGIN_H
 #define _QSCINTILLAPLUGIN_H
 
-#include <QDesignerCustomWidgetInterface>
+#include <QtUiPlugin/QDesignerCustomWidgetInterface>
 
 
 class QScintillaPlugin : public QObject, public QDesignerCustomWidgetInterface
