diff -u -r giggle-0.7/configure.ac giggle-0.7-vte2.91/configure.ac
--- giggle-0.7/configure.ac	2012-11-01 06:29:01.000000000 +0100
+++ giggle-0.7-vte2.91/configure.ac	2014-09-28 23:31:40.389429740 +0200
@@ -56,7 +56,7 @@
 GTK_REQUIRED_VERSION=3.3.12
 GTKSOURCE_REQUIRED_VERSION=3.0
 EDS_REQUIRED_VERSION=3.2
-VTE_REQUIRED_VERSION=0.28
+VTE_REQUIRED_VERSION=0.38
 
 
 PKG_CHECK_MODULES(GIGGLE, [
@@ -109,7 +109,7 @@
 AS_IF([test "x$enable_terminal" = "xyes"],
       [
         PKG_CHECK_MODULES([VTE],
-                          [vte-2.90 >= $VTE_REQUIRED_VERSION],
+                          [vte-2.91 >= $VTE_REQUIRED_VERSION],
                           [have_vte=yes
                            AC_DEFINE([HAVE_VTE], [1],
                                      [Use features from vte])
diff -u -r giggle-0.7/plugins/giggle-view-terminal.c giggle-0.7-vte2.91/plugins/giggle-view-terminal.c
--- giggle-0.7/plugins/giggle-view-terminal.c	2012-11-01 06:29:01.000000000 +0100
+++ giggle-0.7-vte2.91/plugins/giggle-view-terminal.c	2014-09-28 23:46:07.965686517 +0200
@@ -186,15 +186,16 @@
 	real_argv[1] = NULL;
 	spawn_flags = G_SPAWN_CHILD_INHERITS_STDIN | G_SPAWN_SEARCH_PATH | G_SPAWN_FILE_AND_ARGV_ZERO;
 
-	succes = vte_terminal_fork_command_full (VTE_TERMINAL (terminal),
-	                                         pty_flags,
-	                                         directory,
-	                                         real_argv,
-	                                         NULL,
-	                                         spawn_flags,
-	                                         NULL, NULL,
-	                                         NULL,
-	                                         &error);
+	succes = vte_terminal_spawn_sync (VTE_TERMINAL (terminal),
+	                                  pty_flags,
+	                                  directory,
+	                                  real_argv,
+	                                  NULL,
+	                                  spawn_flags,
+	                                  NULL, NULL,
+	                                  NULL,
+                                          NULL,
+	                                  &error);
 	g_strfreev (real_argv);
 
 	if (succes == FALSE) {
