diff --git a/src/livestreamer/utils.py b/src/livestreamer/utils.py
index 34e881b..16a4495 100644
--- a/src/livestreamer/utils.py
+++ b/src/livestreamer/utils.py
@@ -30,10 +30,10 @@ def urlget(url, *args, **kwargs):
     try:
         if session:
             res = session.request(method, url, timeout=timeout, data=data,
-                                  *args, **kwargs)
+                                   verify=False, *args, **kwargs)
         else:
             res = requests.request(method, url, timeout=timeout, data=data,
-                                   *args, **kwargs)
+                                   verify=False, *args, **kwargs)
 
         res.raise_for_status()
     except (requests.exceptions.RequestException, IOError) as rerr:
