diff --git i/apps/desktop/desktop_native/index.js w/apps/desktop/desktop_native/index.js
index 8b48ac994..db9e13eb8 100644
--- i/apps/desktop/desktop_native/index.js
+++ w/apps/desktop/desktop_native/index.js
@@ -151,13 +151,13 @@ switch (platform) {
     switch (arch) {
       case 'x64':
         localFileExisted = existsSync(
-          join(__dirname, 'desktop_native.linux-x64-musl.node')
+          join(__dirname, 'desktop_native.linux-x64-gnu.node')
         )
         try {
           if (localFileExisted) {
-            nativeBinding = require('./desktop_native.linux-x64-musl.node')
+            nativeBinding = require('./desktop_native.linux-x64-gnu.node')
           } else {
-            nativeBinding = require('@bitwarden/desktop-native-linux-x64-musl')
+            nativeBinding = require('@bitwarden/desktop-native-linux-x64-gnu')
           }
         } catch (e) {
           loadError = e
