From 26f274011096b9eacfd2753d8f7afcb63e796ab9 Mon Sep 17 00:00:00 2001
From: nicm <nicm>
Date: Thu, 1 Aug 2019 11:45:34 +0000
Subject: [PATCH] xterm 348 now disables margins when resized, so send DECLRMM
 again.

---
 tty.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tty.c b/tty.c
index ab1da9fd3..024aef271 100644
--- a/tty.c
+++ b/tty.c
@@ -1879,6 +1879,8 @@ tty_invalidate(struct tty *tty)
 	tty->rlower = tty->rright = UINT_MAX;
 
 	if (tty->flags & TTY_STARTED) {
+		if (tty_use_margin(tty))
+			tty_puts(tty, "\033[?69h"); /* DECLRMM */
 		tty_putcode(tty, TTYC_SGR0);
 
 		tty->mode = ALL_MODES;
