diff -Naur wxcam.orig/include/wxcam.h wxcam/include/wxcam.h
--- wxcam.orig/include/wxcam.h	2010-05-23 16:00:56.000000000 +0200
+++ wxcam/include/wxcam.h	2017-07-03 07:25:23.521791275 +0200
@@ -51,7 +51,6 @@
         void OnSettingVideoStd( wxCommandEvent& event );
         void OnRecording( wxCommandEvent& event );
         void OnTakeSnapshot( wxCommandEvent& event );        
-        void OnPaintBackground( wxPaintEvent& event );
         void OnResize( wxSizeEvent& event );
         void enableRec(bool enable);
         void setRecordingTime(const wxTimeSpan &ts);
diff -Naur wxcam.orig/src/wxcam.cpp wxcam/src/wxcam.cpp
--- wxcam.orig/src/wxcam.cpp	2016-12-07 12:11:12.000000000 +0100
+++ wxcam/src/wxcam.cpp	2017-07-03 07:25:36.508822088 +0200
@@ -86,7 +86,6 @@
 EVT_MENU( DateTime_Filter, wxCam::OnSettingDateTime )
 EVT_MENU( Menu_MovementDetection, wxCam::OnMovementDetection )
 EVT_MENU( Menu_ShowMDAreas, wxCam::OnShowMDAreas )
-EVT_PAINT( wxCam::OnPaintBackground )
 EVT_SIZE( wxCam::OnResize )
 END_EVENT_TABLE()
 
@@ -286,16 +285,6 @@
     frame->SetSize( GetClientSize() );
 }
 
-void wxCam::OnPaintBackground( wxPaintEvent& event ) 
-{
-    int w, h;
-    wxPaintDC dc( panel );
-    dc.GetSize( &w, &h );
-    dc.SetBrush( wxBrush( *wxWHITE ) );
-    dc.SetPen( wxPen( *wxWHITE, 1, wxSOLID ) );
-    dc.DrawRectangle( 0, 0, w, h );
-}
-
 void wxCam::OnQuit( wxCommandEvent& WXUNUSED( event ) ) 
 {
     Close( TRUE );
