--- a/src/config/torrc.sample 2020-10-06 18:11:24.173821006 +0200
+++ b/src/config/torrc.sample 2020-10-06 18:11:06.000000000 +0200
@@ -11,6 +11,7 @@
 ##
 ## Tor will look for this file in various places based on your platform:
 ## https://www.torproject.org/docs/faq#torrc
+User tor

 ## Tor opens a SOCKS proxy on port 9050 by default -- even if you don't
 ## configure one below. Set "SOCKSPort 0" if you plan to run Tor only
@@ -39,7 +40,7 @@
 ## Send every possible message to /var/log/tor/debug.log
 #Log debug file /var/log/tor/debug.log
 ## Use the system log instead of Tor's logfiles
-#Log notice syslog
+Log notice syslog
 ## To send all messages to stderr:
 #Log debug stderr

@@ -50,7 +51,7 @@
 
 ## The directory for keeping all the keys/etc. By default, we store
 ## things in $HOME/.tor on Unix, and in Application Data\tor on Windows.
-#DataDirectory /var/lib/tor
+DataDirectory /var/lib/tor
 
 ## The port on which Tor will listen for local connections from Tor
 ## controller applications, as documented in control-spec.txt.
