diff -wbBur tomoyo-tools/usr_lib_tomoyo/tomoyo-editpolicy-agent.c tomoyo-tools.my/usr_lib_tomoyo/tomoyo-editpolicy-agent.c
--- tomoyo-tools/usr_lib_tomoyo/tomoyo-editpolicy-agent.c	2011-08-06 04:00:00.000000000 +0400
+++ tomoyo-tools.my/usr_lib_tomoyo/tomoyo-editpolicy-agent.c	2011-08-11 21:47:05.000000000 +0400
@@ -35,6 +35,7 @@
 #include <dirent.h>
 #define _GNU_SOURCE
 #include <sched.h>
+#include <linux/sched.h>
 #include <sys/mount.h>
 
 static _Bool wait_data(const int fd)
