diff -up krusader-2.4.0-beta1/krusader/UserAction/tstring.h.fix-for-g++47 krusader-2.4.0-beta1/krusader/UserAction/tstring.h
--- krusader-2.4.0-beta1/krusader/UserAction/tstring.h.fix-for-g++47	2012-01-16 08:44:16.064963041 +0100
+++ krusader-2.4.0-beta1/krusader/UserAction/tstring.h	2012-01-16 08:44:33.988425196 +0100
@@ -125,7 +125,7 @@ TagString_t<T>& TagString_t<T>::operator
     str += s.str;
     const unsigned disp = length();
     for (typename taglist::const_iterator it = s.tags.begin(), end = s.tags.end();it != end;++it) {
-        tags.push_back(make_pair((*it).first + disp, (*it).second));
+        tags.push_back(std::make_pair((*it).first + disp, (*it).second));
     }
     return *this;
 }
@@ -134,7 +134,7 @@ template <class T>
 void TagString_t<T>::insertTag(uint pos, const T& t)
 {
     assert(pos <= length());
-    tags.push_back(make_pair(pos, t));
+    tags.push_back(std::make_pair(pos, t));
 }
 
 #endif
