diff --git a/metricbeat/Makefile b/metricbeat/Makefile
index c0ff55b..5f7a241 100644
--- a/metricbeat/Makefile
+++ b/metricbeat/Makefile
@@ -76,7 +76,8 @@ test-module: python-env update metricbeat.test
 assets:
 	go run ${ES_BEATS}/metricbeat/scripts/assets/assets.go ${ES_BEATS}/metricbeat/module
 	mkdir -p include/fields
-	go run  ${ES_BEATS}/libbeat/scripts/cmd/global_fields/main.go -es_beats_path ${ES_BEATS} -beat_path ${PWD} | go run  ${ES_BEATS}/dev-tools/cmd/asset/asset.go -license ${LICENSE} -out ./include/fields/fields.go -pkg include -priority asset.LibbeatFieldsPri ${ES_BEATS}/libbeat/fields.yml $(BEAT_NAME)
+	go run ${ES_BEATS}/libbeat/scripts/cmd/global_fields/main.go -es_beats_path ${ES_BEATS} -beat_path ${PWD} > tmp_global_fields.bin
+	go run ${ES_BEATS}/dev-tools/cmd/asset/asset.go -license ${LICENSE} -out ./include/fields/fields.go -pkg include -priority asset.LibbeatFieldsPri ${ES_BEATS}/libbeat/fields.yml $(BEAT_NAME) < tmp_global_fields.bin
 
 .PHONY: integration-tests
 integration-tests: ## @testing Run golang integration tests.
