diff -aur old/src/notify_desktop.c new/src/notify_desktop.c
--- old/src/notify_desktop.c	2011-03-07 23:50:24.096691200 +0100
+++ new/src/notify_desktop.c	2011-03-07 23:52:35.486691201 +0100
@@ -95,7 +95,7 @@
 				: (NOTIF_WARNING == n->severity) ? "WARNING"
 				: (NOTIF_OKAY == n->severity) ? "OKAY" : "UNKNOWN");
 
-	notification = notify_notification_new (summary, n->message, NULL, NULL);
+	notification = notify_notification_new (summary, n->message, NULL);
 	if (NULL == notification) {
 		log_err ("Failed to create a new notification.");
 		return -1;
