diff -ruN eric6-19.03-orig/eric/install.py eric6-19.03/eric/install.py
--- eric6-19.03-orig/eric/install.py	2019-03-30 03:14:11.236099494 +0300
+++ eric6-19.03/eric/install.py	2019-03-30 03:16:36.238216137 +0300
@@ -1905,13 +1905,13 @@
     try:
         if sys.platform.startswith(("win", "cygwin")):
             optlist, args = getopt.getopt(
-                argv[1:], "chxyza:b:d:f:", ["help", "pyqt=", "noapis"])
+                argv[1:], "chxyza:b:d:f:", ["help", "pyqt=", "no-apis"])
         elif sys.platform == "darwin":
             optlist, args = getopt.getopt(
-                argv[1:], "chxyza:b:d:f:i:m:n:p:", ["help", "pyqt=", "noapis"])
+                argv[1:], "chxyza:b:d:f:i:m:n:p:", ["help", "pyqt=", "no-apis"])
         else:
             optlist, args = getopt.getopt(
-                argv[1:], "chxyza:b:d:f:i:", ["help", "pyqt=", "noapis"])
+                argv[1:], "chxyza:b:d:f:i:", ["help", "pyqt=", "no-apis"])
     except getopt.GetoptError as err:
         print(err)
         usage()
diff -ruN eric6-19.03-orig/install.py eric6-19.03/install.py
--- eric6-19.03-orig/install.py	2019-03-30 03:14:11.039434611 +0300
+++ eric6-19.03/install.py	2019-03-30 03:16:36.238216137 +0300
@@ -1905,13 +1905,13 @@
     try:
         if sys.platform.startswith(("win", "cygwin")):
             optlist, args = getopt.getopt(
-                argv[1:], "chxyza:b:d:f:", ["help", "pyqt=", "noapis"])
+                argv[1:], "chxyza:b:d:f:", ["help", "pyqt=", "no-apis"])
         elif sys.platform == "darwin":
             optlist, args = getopt.getopt(
-                argv[1:], "chxyza:b:d:f:i:m:n:p:", ["help", "pyqt=", "noapis"])
+                argv[1:], "chxyza:b:d:f:i:m:n:p:", ["help", "pyqt=", "no-apis"])
         else:
             optlist, args = getopt.getopt(
-                argv[1:], "chxyza:b:d:f:i:", ["help", "pyqt=", "noapis"])
+                argv[1:], "chxyza:b:d:f:i:", ["help", "pyqt=", "no-apis"])
     except getopt.GetoptError as err:
         print(err)
         usage()
