--- aaa/configure.ac	2022-12-13 22:37:13.517250622 +0100
+++ bbb/configure.ac	2022-12-13 22:37:29.767271256 +0100
@@ -32,9 +32,9 @@
 DBLIBS=""
 # the only way to find out which is compileable is to look into db.h:
 
-AC_CHECK_HEADER(db.h,,[AC_MSG_ERROR(["no db.h found"])])
+AC_CHECK_HEADER(db5.3/db.h,,[AC_MSG_ERROR(["no db5.3/db.h found"])])
 
-AC_CHECK_LIB(db, db_create, [DBLIBS="-ldb $DBLIBS"
+AC_CHECK_LIB(db-5.3, db_create, [DBLIBS="-ldb-5.3 $DBLIBS"
 	],[AC_MSG_ERROR(["no libdb found"])],[$DBLIBS])
 AC_SUBST([DBLIBS])
 
