diff -rup luxrender-luxrays-ed424ec77571.orig/include/luxrays/luxrays.h luxrender-luxrays-ed424ec77571/include/luxrays/luxrays.h
--- luxrender-luxrays-ed424ec77571.orig/include/luxrays/luxrays.h	2011-04-24 17:43:08.000000000 +0200
+++ luxrender-luxrays-ed424ec77571/include/luxrays/luxrays.h	2011-11-26 11:38:00.336459874 +0100
@@ -30,6 +30,8 @@
 
 #define __CL_ENABLE_EXCEPTIONS
 
+#define CL_USE_DEPRECATED_OPENCL_1_1_APIS
+
 #if defined(__APPLE__)
 #include <OpenCL/cl.hpp>
 #else
