diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-12-31 09:56:25.000000000 +0100
+++ b/CMakeLists.txt	2021-01-07 17:29:04.152477922 +0100
@@ -18,15 +18,16 @@
 
 include(Options)
 include(Tools)
+include(GNUInstallDirs)
 
 find_package(Libconfig REQUIRED)
 find_package(ProtobufC REQUIRED)
 
 add_subdirectory(src)
 
-find_path(OLD_CONFIG_FILE NAMES "umurmur.conf" PATHS ${CMAKE_INSTALL_PREFIX} PATH_SUFFIXES "etc")
-
-if(NOT OLD_CONFIG_FILE)
-  install(FILES "umurmur.conf.example" DESTINATION "etc" RENAME "umurmur.conf")
-endif()
-
+install(
+  FILES "umurmur.conf.example"
+  DESTINATION "/${CMAKE_INSTALL_SYSCONFDIR}/umurmur"
+  PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ
+  RENAME "umurmur.conf"
+)
