--- nemo-extensions-2.4.x.o/nemo-seahorse/configure.ac	2014-11-01 05:42:22.000000000 -0500
+++ nemo-extensions-2.4.x/nemo-seahorse/configure.ac	2014-11-16 12:38:22.529985197 -0500
@@ -58,7 +58,7 @@
 	DO_CHECK=$enableval, DO_CHECK=yes)
 
 if test	"$DO_CHECK" = "yes"; then
-	accepted_versions="1.2 1.4 2.0"
+	accepted_versions="1.2 1.4 2.0 2.1"
 	AC_PATH_PROGS(GNUPG, [gpg gpg2], no)
 	ok="no"
 	if test "$GNUPG" != "no"; then
