diff --git a/gcc/d/Make-lang.in b/gcc/d/Make-lang.in
index 5bdab1b..aa1122e 100644
--- a/gcc/d/Make-lang.in
+++ b/gcc/d/Make-lang.in
@@ -25,9 +25,9 @@ D_TARGET_INSTALL_NAME = $(target_alias)-$(shell echo gdc|sed '$(program_transfor
 
 # This should be configured
 ifeq ($(host), $(target))
-  gcc_d_include_dir = $(prefix)/include/d/$(version)
+  gcc_d_include_dir = $(prefix)/include/dlang/gdc/$(version)
 else
-  gcc_d_include_dir = $(libsubdir)/include/d
+  gcc_d_include_dir = $(libsubdir)/include/dlang/gdc
 endif
 
 
diff --git a/libphobos/configure.in b/libphobos/configure.in
index f6b3660..7e88b23 100644
--- a/libphobos/configure.in
+++ b/libphobos/configure.in
@@ -232,9 +232,9 @@ dnl (# Default case for install directory for include files.) and on
 # will have to modify gcc/configure.ac ..
 # For now, basic workaround for cross compilers ..
 if test "${host}" != "${build}"; then
-    gdc_include_dir='${libdir}/gcc/${host_alias}'/${d_gcc_ver}/include/d
+    gdc_include_dir='${libdir}/gcc/${host_alias}'/${d_gcc_ver}/include/dlang/gdc
 else
-    gdc_include_dir='${prefix}'/include/d/${d_gcc_ver}
+    gdc_include_dir='${prefix}'/include/dlang/gdc/${d_gcc_ver}
 fi
 AC_SUBST(gdc_include_dir)
 AC_ARG_WITH([cross-host],
