From 930df975df6e955f1791bbb40dab5da32eca729f Mon Sep 17 00:00:00 2001
From: Guido Scholz <gscholz@users.sourceforge.net>
Date: Wed, 7 Nov 2018 21:31:50 +0100
Subject: [PATCH 10/11] Remove warnings about not find translation files

---
 src/main.cpp | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/main.cpp b/src/main.cpp
index d485efc..7b322b8 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -117,18 +117,12 @@ int main(int argc, char *argv[])
   if (qtTr.load(QString("qt_") + loc.name(),
               QLibraryInfo::location(QLibraryInfo::TranslationsPath)))
       app.installTranslator(&qtTr);
-  else
-      qWarning("No Qt translation for locale '%s' found.",
-              loc.name().toLatin1().constData());
 
   // translator for ams strings
   QTranslator amsTr;
 
   if (amsTr.load(QString("ams_") + loc.name(), TRANSLATIONSDIR))
       app.installTranslator(&amsTr);
-  else
-      qWarning("No " AMS_LONGNAME " translation for locale '%s' found.",
-              loc.name().toLatin1().constData());
 
   int getopt_return;
   int option_index;
-- 
2.21.0

