diff -upr a/cheroot.egg-info/requires.txt b/cheroot.egg-info/requires.txt
--- a/cheroot.egg-info/requires.txt	2019-10-18 02:59:50.000000000 +0200
+++ b/cheroot.egg-info/requires.txt	2019-11-21 15:12:09.409022012 +0100
@@ -18,7 +18,6 @@ collective.checkdocs
 pytest>=2.8
 pytest-mock>=1.11.0
 pytest-sugar>=0.9.1
-pytest-testmon>=0.9.7
 pytest-watch==4.2.0
 pytest-xdist>=1.28.0
 coverage==4.5.3
diff -upr a/pytest.ini b/pytest.ini
--- a/pytest.ini	2019-10-18 02:59:30.000000000 +0200
+++ b/pytest.ini	2019-11-21 15:11:39.575862172 +0100
@@ -1,6 +1,6 @@
 [pytest]
 norecursedirs = dist docs build .tox .eggs
-addopts = --durations=10 -v -rxXs -l -n auto --testmon --doctest-modules --ignore cheroot/ssl/pyopenssl.py --junitxml=junit-test-results.xml --cov=cheroot --cov-report term-missing:skip-covered --cov-report xml
+addopts = --durations=10 -v -rxXs -l -n auto --doctest-modules --ignore cheroot/ssl/pyopenssl.py --junitxml=junit-test-results.xml --cov=cheroot --cov-report term-missing:skip-covered --cov-report xml
 filterwarnings =
     error
 doctest_optionflags = ALLOW_UNICODE ELLIPSIS
diff -upr a/setup.cfg b/setup.cfg
--- a/setup.cfg	2019-10-18 02:59:51.000000000 +0200
+++ b/setup.cfg	2019-11-21 15:11:55.979100130 +0100
@@ -78,7 +78,6 @@ testing =
 	pytest>=2.8
 	pytest-mock>=1.11.0
 	pytest-sugar>=0.9.1
-	pytest-testmon>=0.9.7
 	pytest-watch==4.2.0
 	pytest-xdist>=1.28.0
 	
diff -upr a/tox.ini b/tox.ini
--- a/tox.ini	2019-10-18 02:59:30.000000000 +0200
+++ b/tox.ini	2019-11-21 15:07:15.714056061 +0100
@@ -6,7 +6,7 @@ minversion = 3.13.2
 deps =
     setuptools>=31.0.1
 commands =
-    pytest --testmon-off {posargs}
+    pytest {posargs}
     codecov -f coverage.xml -X gcov
 usedevelop = True
 extras = testing
