diff -wbBur dbmail-3.2.3/src/dm_debug.c dbmail-3.2.3.my/src/dm_debug.c
--- dbmail-3.2.3/src/dm_debug.c	2015-01-25 13:20:50.000000000 +0300
+++ dbmail-3.2.3.my/src/dm_debug.c	2016-04-08 19:51:11.837835259 +0300
@@ -98,7 +98,8 @@
 		"Notice",
 		"Info",
 		"Debug",
-		"Database"
+		"Database",
+		"Network"
 	};
 	return Trace_Text[ilogb((double) level)];
 }
diff -wbBur dbmail-3.2.3/src/dm_debug.h dbmail-3.2.3.my/src/dm_debug.h
--- dbmail-3.2.3/src/dm_debug.h	2015-01-25 13:20:50.000000000 +0300
+++ dbmail-3.2.3.my/src/dm_debug.h	2016-04-08 18:54:33.797752209 +0300
@@ -36,7 +36,8 @@
 	TRACE_NOTICE = 32,
 	TRACE_INFO = 64,
 	TRACE_DEBUG = 128,
-	TRACE_DATABASE = 256 // Logs at Debug Level
+	TRACE_DATABASE = 256, // Logs at Debug Level
+	TRACE_NETWORK = 512 // Logs at Debug Level
 } Trace_T;
 
 
diff -wbBur dbmail-3.2.3/src/imap4.c dbmail-3.2.3.my/src/imap4.c
--- dbmail-3.2.3/src/imap4.c	2015-01-25 13:20:50.000000000 +0300
+++ dbmail-3.2.3.my/src/imap4.c	2016-04-08 20:01:05.247849764 +0300
@@ -784,6 +784,16 @@
 
 	imap_unescape_args(session);
 
-	TRACE(TRACE_INFO, "dispatch [%s]...\n", IMAP_COMMANDS[session->command_type]);
+	char caddr_buf[64];
+	TRACE(TRACE_NETWORK, "Dispatch: client:%s, uid:%d, cmd:%s, rxed: %u, txed: %u\n",
+		inet_ntop(session->ci->sock->caddr.sa_family,
+			(const void*)&((struct sockaddr_in*)&session->ci->sock->caddr)->sin_addr,
+			caddr_buf,
+			session->ci->sock->caddr_len),
+		(int)session->userid,
+		IMAP_COMMANDS[session->command_type],
+		(unsigned)session->ci->bytes_rx,
+		(unsigned)session->ci->bytes_tx);
+
 	return (*imap_handler_functions[session->command_type]) (session);
 }
