Index: xnoise-0.2.21/libxnoise/MainWindow/xnoise-main-window.vala
===================================================================
--- xnoise-0.2.21.orig/libxnoise/MainWindow/xnoise-main-window.vala
+++ xnoise-0.2.21/libxnoise/MainWindow/xnoise-main-window.vala
@@ -1239,7 +1239,7 @@ public class Xnoise.MainWindow : Gtk.Win
 
     private void on_remove_all_button_clicked() {
         global.position_reference = null;
-        var store = (ListStore)trackList.get_model();
+        var store = (Gtk.ListStore)trackList.get_model();
         store.clear();
     }
 
@@ -1821,7 +1821,7 @@ public class Xnoise.MainWindow : Gtk.Win
             tbx.pack_start(removeAllButton, false, false, 0);
             removeAllButton.clicked.connect( () => {
                 global.position_reference = null;
-                var store = (ListStore)tlm;
+                var store = (Gtk.ListStore)tlm;
                 store.clear();
             });
             var posjumper = new Gtk.Button();
@@ -1848,7 +1848,7 @@ public class Xnoise.MainWindow : Gtk.Win
                 if(global.position_reference == null || !global.position_reference.valid())
                     return;
                 TreePath path = global.position_reference.get_path();
-                var store = (ListStore)tlm;
+                var store = (Gtk.ListStore)tlm;
                 TreeIter iter;
                 store.get_iter(out iter, path);
                 tl.set_focus_on_iter(ref iter);
@@ -2285,7 +2285,7 @@ public class Xnoise.MainWindow : Gtk.Win
 }
 
 private class Xnoise.SideBarHeadline : Gtk.TreeView {
-    private ListStore store;
+    private Gtk.ListStore store;
     private string headline = "";
     
     public SideBarHeadline(string headline = "") {
@@ -2334,7 +2334,7 @@ private class Xnoise.SideBarHeadline : G
     }
     
     private void setup_widgets() {
-        store = new ListStore(Column.N_COUNT, typeof(string), typeof(int));
+        store = new Gtk.ListStore(Column.N_COUNT, typeof(string), typeof(int));
         
         var renderer = new CustomCellRendererHeadline();
         var column = new TreeViewColumn();
Index: xnoise-0.2.21/libxnoise/ExtraWidgets/AddMedia/xnoise-add-media-widget.vala
===================================================================
--- xnoise-0.2.21.orig/libxnoise/ExtraWidgets/AddMedia/xnoise-add-media-widget.vala
+++ xnoise-0.2.21/libxnoise/ExtraWidgets/AddMedia/xnoise-add-media-widget.vala
@@ -47,7 +47,7 @@ private class Xnoise.AddMediaWidget : Gt
     }
     
     private const string XNOISEICON = "xnoise";
-    private ListStore listmodel;
+    private Gtk.ListStore listmodel;
     private TreeView tv;
     
     public Gtk.Builder builder;
@@ -108,7 +108,7 @@ private class Xnoise.AddMediaWidget : Gt
         tv.headers_visible = false;
         tv.get_selection().set_mode(SelectionMode.MULTIPLE);
         
-        listmodel = new ListStore(Column.COL_COUNT, 
+        listmodel = new Gtk.ListStore(Column.COL_COUNT, 
                                   typeof(Gdk.Pixbuf), 
                                   typeof(string), 
                                   typeof(Item?), 
Index: xnoise-0.2.21/libxnoise/ExtraWidgets/AlbumArtView/xnoise-albumart-view.vala
===================================================================
--- xnoise-0.2.21.orig/libxnoise/ExtraWidgets/AlbumArtView/xnoise-albumart-view.vala
+++ xnoise-0.2.21/libxnoise/ExtraWidgets/AlbumArtView/xnoise-albumart-view.vala
@@ -187,7 +187,7 @@ private class Xnoise.AlbumArtView : Gtk.
                 if(global.position_reference == null || !global.position_reference.valid())
                     return false;
                 TreePath p = global.position_reference.get_path();
-                var store = (ListStore)tlm;
+                var store = (Gtk.ListStore)tlm;
                 TreeIter it;
                 store.get_iter(out it, p);
                 tl.set_focus_on_iter(ref it);
Index: xnoise-0.2.21/libxnoise/ExtraWidgets/xnoise-first-start-widget.vala
===================================================================
--- xnoise-0.2.21.orig/libxnoise/ExtraWidgets/xnoise-first-start-widget.vala
+++ xnoise-0.2.21/libxnoise/ExtraWidgets/xnoise-first-start-widget.vala
@@ -102,7 +102,7 @@ private class Xnoise.FirstStartWidget :
     }
     
     private TreeView tv;
-    private ListStore listmodel;
+    private Gtk.ListStore listmodel;
     private Box infobox;
     private Box waitbox;
     private Box bigbox;
@@ -176,7 +176,7 @@ private class Xnoise.FirstStartWidget :
             }
 
             imported_box.pack_start(tv, false, false, 0);
-            listmodel = new ListStore(Column.COL_COUNT, typeof(Gdk.Pixbuf), typeof(string));
+            listmodel = new Gtk.ListStore(Column.COL_COUNT, typeof(Gdk.Pixbuf), typeof(string));
             
             var column = new TreeViewColumn();
             var rendererpb = new CellRendererPixbuf();
Index: xnoise-0.2.21/libxnoise/ExtraWidgets/xnoise-tree-media-selector.vala
===================================================================
--- xnoise-0.2.21.orig/libxnoise/ExtraWidgets/xnoise-tree-media-selector.vala
+++ xnoise-0.2.21/libxnoise/ExtraWidgets/xnoise-tree-media-selector.vala
@@ -36,7 +36,7 @@ using Gtk;
 private class Xnoise.TreeMediaSelector : TreeView, MediaSelector {
     
     private unowned MediaSoureWidget msw;
-    private ListStore store;
+    private Gtk.ListStore store;
     private bool mouse_over = false;
     private int row_height = 24;
     
@@ -63,7 +63,7 @@ private class Xnoise.TreeMediaSelector :
         this.headers_visible = false;
         this.set_enable_search(false);
         this.get_selection().set_mode(SelectionMode.SINGLE);
-        this.store = new ListStore(Column.N_COLUMNS, 
+        this.store = new Gtk.ListStore(Column.N_COLUMNS, 
                                    typeof(string),           //icon
                                    typeof(string),               //vis_text
                                    typeof(int),                  //weight
@@ -216,7 +216,7 @@ private class Xnoise.TreeMediaSelector :
         });
         if(path != null) {
             this.model.foreach( (mo,px,iy) => {
-                ListStore mx = (ListStore)mo;
+                Gtk.ListStore mx = (Gtk.ListStore)mo;
                 mx.set(iy, 
                        Column.SELECTION_STATE, false,
                        Column.SELECTION_ICON, null
@@ -225,7 +225,7 @@ private class Xnoise.TreeMediaSelector :
             });
             TreeIter it;
             this.model.get_iter(out it, path);
-            ((ListStore)this.model).set(it,
+            ((Gtk.ListStore)this.model).set(it,
                            Column.SELECTION_STATE, true,
                            Column.SELECTION_ICON, icon_repo.selected_collection_icon
             );
@@ -251,9 +251,9 @@ private class Xnoise.TreeMediaSelector :
             return true;
         
         TreeIter it;
-        ListStore m = (ListStore)this.get_model();
+        Gtk.ListStore m = (Gtk.ListStore)this.get_model();
             m.foreach( (mo,p,iy) => {
-                ListStore mx = (ListStore)mo;
+                Gtk.ListStore mx = (Gtk.ListStore)mo;
                 mx.set(iy, 
                        Column.SELECTION_STATE, false,
                        Column.SELECTION_ICON, null
@@ -289,7 +289,7 @@ private class Xnoise.TreeMediaSelector :
                     TreeIter iter;
                     this.model.get_iter(out iter, treepath);
                     m.foreach( (mo,p,iy) => {
-                        ListStore mx = (ListStore)mo;
+                        Gtk.ListStore mx = (Gtk.ListStore)mo;
                         mx.set(iy, 
                                Column.SELECTION_STATE, false,
                                Column.SELECTION_ICON, null
@@ -297,7 +297,7 @@ private class Xnoise.TreeMediaSelector :
                         return false;
                     });
                     this.set_cursor(treepath, null,false);
-                    ListStore mx = (ListStore)this.model;
+                    Gtk.ListStore mx = (Gtk.ListStore)this.model;
                     string? name;
                     mx.get(iter, Column.NAME, out name);
                     mx.set(iter, 
Index: xnoise-0.2.21/libxnoise/TrackList/xnoise-tracklist-model.vala
===================================================================
--- xnoise-0.2.21.orig/libxnoise/TrackList/xnoise-tracklist-model.vala
+++ xnoise-0.2.21/libxnoise/TrackList/xnoise-tracklist-model.vala
@@ -37,7 +37,7 @@ using Xnoise.Resources;
 using Xnoise.Utilities;
 using Xnoise.TagAccess;
 
-public class Xnoise.TrackListModel : ListStore, TreeModel {
+public class Xnoise.TrackListModel : Gtk.ListStore, TreeModel {
 
     private Main xn;
     private IconTheme icon_theme;
Index: xnoise-0.2.21/libxnoise/DockableMedia/VideoListing/xnoise-treeview-videos-model.vala
===================================================================
--- xnoise-0.2.21.orig/libxnoise/DockableMedia/VideoListing/xnoise-treeview-videos-model.vala
+++ xnoise-0.2.21/libxnoise/DockableMedia/VideoListing/xnoise-treeview-videos-model.vala
@@ -108,7 +108,7 @@ private class Xnoise.TreeViewVideosModel
 
     public void remove_all() {
         view.set_model(null);
-        assert(this is ListStore);
+        assert(this is Gtk.ListStore);
         this.clear();
         view.set_model(this);
     }
@@ -116,7 +116,7 @@ private class Xnoise.TreeViewVideosModel
     public void filter() {
         //print("filter\n");
         view.set_model(null);
-        assert(this is ListStore);
+        assert(this is Gtk.ListStore);
         this.clear();
         this.populate();
     }
