--- system/cdemu-daemon-system.sh.orig	2012-02-27 21:38:05.000000000 +0100
+++ system/cdemu-daemon-system.sh	2012-03-10 20:53:06.787503891 +0100
@@ -6,10 +6,10 @@
 # Default settings
 NUM_DEVICES=2
 AUDIO_DRIVER=null
-LOG_FILE=/tmp/cdemu-daemon.log
+LOG_FILE=/var/log/cdemu-daemon.log
 
 # Read the settings
-CONFIG_FILE=/etc/sysconfig/cdemu-daemon
+CONFIG_FILE=/etc/conf.d/cdemud
 
 if [ -f ${CONFIG_FILE} ]; then
     . ${CONFIG_FILE};
