diff --git a/flowblade-trunk/Flowblade/app.py b/flowblade-trunk/Flowblade/app.py
index fc9b8149..1c23080c 100644
--- a/flowblade-trunk/Flowblade/app.py
+++ b/flowblade-trunk/Flowblade/app.py
@@ -40,7 +40,7 @@ from gi.repository import Gtk
 from gi.repository import Gdk
 
 import locale
-import mlt
+import mlt7 as mlt
 import hashlib
 import os
 import sys
diff --git a/flowblade-trunk/Flowblade/audiomonitoring.py b/flowblade-trunk/Flowblade/audiomonitoring.py
index d55cf01a..a0de59f7 100644
--- a/flowblade-trunk/Flowblade/audiomonitoring.py
+++ b/flowblade-trunk/Flowblade/audiomonitoring.py
@@ -29,7 +29,7 @@ except ImportError:
 import gi
 
 import cairo
-import mlt
+import mlt7 as mlt
 import time
 
 from gi.repository import Gtk, GObject
diff --git a/flowblade-trunk/Flowblade/audiosync.py b/flowblade-trunk/Flowblade/audiosync.py
index 8de23c03..17b26073 100644
--- a/flowblade-trunk/Flowblade/audiosync.py
+++ b/flowblade-trunk/Flowblade/audiosync.py
@@ -23,7 +23,7 @@ Handles syncing clips based on their audio data.
 """
 import datetime
 import hashlib
-import mlt
+import mlt7 as mlt
 import os
 import subprocess
 import sys
diff --git a/flowblade-trunk/Flowblade/audiowaveform.py b/flowblade-trunk/Flowblade/audiowaveform.py
index 3d1d5e0e..3de5a8ef 100644
--- a/flowblade-trunk/Flowblade/audiowaveform.py
+++ b/flowblade-trunk/Flowblade/audiowaveform.py
@@ -22,7 +22,7 @@
 Modules handles creating and caching audio waveform images for clips.
 """
 
-import mlt
+import mlt7 as mlt
 import os
 import pickle
 import threading
diff --git a/flowblade-trunk/Flowblade/audiowaveformrenderer.py b/flowblade-trunk/Flowblade/audiowaveformrenderer.py
index 67bc5779..e4a67b4a 100644
--- a/flowblade-trunk/Flowblade/audiowaveformrenderer.py
+++ b/flowblade-trunk/Flowblade/audiowaveformrenderer.py
@@ -23,7 +23,7 @@ Modules handles creating and caching audio waveform images for clips.
 """
 
 import locale
-import mlt
+import mlt7 as mlt
 import os
 import pickle
 import subprocess
diff --git a/flowblade-trunk/Flowblade/clipmenuaction.py b/flowblade-trunk/Flowblade/clipmenuaction.py
index 1b8f357a..6dae9ca5 100644
--- a/flowblade-trunk/Flowblade/clipmenuaction.py
+++ b/flowblade-trunk/Flowblade/clipmenuaction.py
@@ -27,7 +27,7 @@ from PIL import Image
 from gi.repository import GLib
 from gi.repository import Gtk
 
-import mlt
+import mlt7 as mlt
 from operator import itemgetter
 import os
 import shutil
diff --git a/flowblade-trunk/Flowblade/containeractions.py b/flowblade-trunk/Flowblade/containeractions.py
index b12d9db9..148a588b 100644
--- a/flowblade-trunk/Flowblade/containeractions.py
+++ b/flowblade-trunk/Flowblade/containeractions.py
@@ -24,7 +24,7 @@ from gi.repository import GLib
 import cairo
 import copy
 import hashlib
-import mlt
+import mlt7 as mlt
 import os
 from os import listdir
 from os.path import isfile, join
diff --git a/flowblade-trunk/Flowblade/exporting.py b/flowblade-trunk/Flowblade/exporting.py
index 0f5dcf63..224dee9c 100644
--- a/flowblade-trunk/Flowblade/exporting.py
+++ b/flowblade-trunk/Flowblade/exporting.py
@@ -22,7 +22,7 @@ from gi.repository import Gtk
 import os
 from xml.dom import minidom
 from math import floor
-import mlt
+import mlt7 as mlt
 import time
 import hashlib
 import re
diff --git a/flowblade-trunk/Flowblade/medialinker.py b/flowblade-trunk/Flowblade/medialinker.py
index a8e963e8..895a5142 100644
--- a/flowblade-trunk/Flowblade/medialinker.py
+++ b/flowblade-trunk/Flowblade/medialinker.py
@@ -19,7 +19,7 @@
 """
 
 import glob
-import mlt
+import mlt7 as mlt
 import locale
 import os
 import subprocess
diff --git a/flowblade-trunk/Flowblade/mltenv.py b/flowblade-trunk/Flowblade/mltenv.py
index d97406c1..3da17092 100644
--- a/flowblade-trunk/Flowblade/mltenv.py
+++ b/flowblade-trunk/Flowblade/mltenv.py
@@ -22,7 +22,7 @@
 Module checks environment for available codecs and formats.
 """
 from gi.repository import GObject
-import mlt
+import mlt7 as mlt
 
 import dialogutils
 import editorstate
diff --git a/flowblade-trunk/Flowblade/mltfilters.py b/flowblade-trunk/Flowblade/mltfilters.py
index 644fd9bc..8ec082c6 100644
--- a/flowblade-trunk/Flowblade/mltfilters.py
+++ b/flowblade-trunk/Flowblade/mltfilters.py
@@ -27,7 +27,7 @@ import copy
 
 from gi.repository import GdkPixbuf
 
-import mlt
+import mlt7 as mlt
 import xml.dom.minidom
 
 import appconsts
diff --git a/flowblade-trunk/Flowblade/mltplayer.py b/flowblade-trunk/Flowblade/mltplayer.py
index fddd4c97..84d79070 100644
--- a/flowblade-trunk/Flowblade/mltplayer.py
+++ b/flowblade-trunk/Flowblade/mltplayer.py
@@ -24,7 +24,7 @@ a Xwindow of a GTK+ widget and os audiosystem using a SDL consumer.
 """
 from gi.repository import Gdk
 
-import mlt
+import mlt7 as mlt
 import os
 import time
 
diff --git a/flowblade-trunk/Flowblade/mltprofiles.py b/flowblade-trunk/Flowblade/mltprofiles.py
index 4a891a52..293ed226 100644
--- a/flowblade-trunk/Flowblade/mltprofiles.py
+++ b/flowblade-trunk/Flowblade/mltprofiles.py
@@ -22,7 +22,7 @@
 MLT framework profiles.
 """
 import os
-import mlt
+import mlt7 as mlt
 import xml.dom.minidom
 
 import appconsts
diff --git a/flowblade-trunk/Flowblade/mlttransitions.py b/flowblade-trunk/Flowblade/mlttransitions.py
index 0b77938a..6af1a581 100644
--- a/flowblade-trunk/Flowblade/mlttransitions.py
+++ b/flowblade-trunk/Flowblade/mlttransitions.py
@@ -23,7 +23,7 @@ Module contains objects that wrap mlt.Transition objects used to mix video betwe
 two tracks.
 """
 import copy
-import mlt
+import mlt7 as mlt
 import os
 import xml.dom.minidom
 
diff --git a/flowblade-trunk/Flowblade/monitorwidget.py b/flowblade-trunk/Flowblade/monitorwidget.py
index 48a01ae8..88b0b56a 100644
--- a/flowblade-trunk/Flowblade/monitorwidget.py
+++ b/flowblade-trunk/Flowblade/monitorwidget.py
@@ -21,7 +21,7 @@ from gi.repository import Gdk
 from gi.repository import Gtk, GLib
 
 import cairo
-import mlt
+import mlt7 as mlt
 import numpy as np
 import os
 import threading
diff --git a/flowblade-trunk/Flowblade/patternproducer.py b/flowblade-trunk/Flowblade/patternproducer.py
index afa1271a..35751273 100644
--- a/flowblade-trunk/Flowblade/patternproducer.py
+++ b/flowblade-trunk/Flowblade/patternproducer.py
@@ -27,7 +27,7 @@ import hashlib
 
 from gi.repository import Gtk, Gdk
 
-import mlt
+import mlt7 as mlt
 
 import appconsts
 import dialogutils
diff --git a/flowblade-trunk/Flowblade/projectaction.py b/flowblade-trunk/Flowblade/projectaction.py
index a684f514..c44a6590 100644
--- a/flowblade-trunk/Flowblade/projectaction.py
+++ b/flowblade-trunk/Flowblade/projectaction.py
@@ -28,7 +28,7 @@ import datetime
 import fnmatch
 import glob
 import hashlib
-import mlt
+import mlt7 as mlt
 import os
 from os import listdir
 from os.path import isfile, join, expanduser
diff --git a/flowblade-trunk/Flowblade/projectdata.py b/flowblade-trunk/Flowblade/projectdata.py
index f063275a..a401ade3 100644
--- a/flowblade-trunk/Flowblade/projectdata.py
+++ b/flowblade-trunk/Flowblade/projectdata.py
@@ -23,7 +23,7 @@ Module contains objects used to capture project data.
 """
 import cairo
 import datetime
-import mlt
+import mlt7 as mlt
 import hashlib
 import os
 
diff --git a/flowblade-trunk/Flowblade/projectmediaimport.py b/flowblade-trunk/Flowblade/projectmediaimport.py
index 52c294df..9c03aae0 100644
--- a/flowblade-trunk/Flowblade/projectmediaimport.py
+++ b/flowblade-trunk/Flowblade/projectmediaimport.py
@@ -18,7 +18,7 @@
     along with Flowblade Movie Editor.  If not, see <http://www.gnu.org/licenses/>.
 """
 
-import mlt
+import mlt7 as mlt
 import locale
 import os
 import subprocess
diff --git a/flowblade-trunk/Flowblade/proxyediting.py b/flowblade-trunk/Flowblade/proxyediting.py
index b0eae3a3..a9a5fb24 100644
--- a/flowblade-trunk/Flowblade/proxyediting.py
+++ b/flowblade-trunk/Flowblade/proxyediting.py
@@ -20,7 +20,7 @@
 
 import glob
 import hashlib
-import mlt
+import mlt7 as mlt
 import os
 import shutil
 import threading
diff --git a/flowblade-trunk/Flowblade/render.py b/flowblade-trunk/Flowblade/render.py
index ae22d939..34f5527f 100644
--- a/flowblade-trunk/Flowblade/render.py
+++ b/flowblade-trunk/Flowblade/render.py
@@ -28,7 +28,7 @@ Rendering is done in app.player object of class mltplayer.Player
 
 from gi.repository import Gtk
 
-import mlt
+import mlt7 as mlt
 import hashlib
 import os
 import subprocess
diff --git a/flowblade-trunk/Flowblade/renderconsumer.py b/flowblade-trunk/Flowblade/renderconsumer.py
index 71dad22e..6c454556 100644
--- a/flowblade-trunk/Flowblade/renderconsumer.py
+++ b/flowblade-trunk/Flowblade/renderconsumer.py
@@ -25,7 +25,7 @@ Module contains objects and methods needed to create render consumers.
 
 from gi.repository import Gdk
 
-import mlt
+import mlt7 as mlt
 import time
 import threading
 import xml.dom.minidom
diff --git a/flowblade-trunk/Flowblade/sequence.py b/flowblade-trunk/Flowblade/sequence.py
index 76b4ccaa..ad909866 100644
--- a/flowblade-trunk/Flowblade/sequence.py
+++ b/flowblade-trunk/Flowblade/sequence.py
@@ -23,7 +23,7 @@ Module contains class Sequence that is the multitrack media object being edited
 by the application. A project has 1-n of these.
 """
 import time
-import mlt
+import mlt7 as mlt
 import os
 
 import appconsts
diff --git a/flowblade-trunk/Flowblade/tlinerenderserver.py b/flowblade-trunk/Flowblade/tlinerenderserver.py
index 8228da5f..9a9a6a05 100644
--- a/flowblade-trunk/Flowblade/tlinerenderserver.py
+++ b/flowblade-trunk/Flowblade/tlinerenderserver.py
@@ -32,7 +32,7 @@ import dbus
 import dbus.service
 from dbus.mainloop.glib import DBusGMainLoop
 import locale
-import mlt
+import mlt7 as mlt
 import os
 import subprocess
 import sys
diff --git a/flowblade-trunk/Flowblade/tools/batchrendering.py b/flowblade-trunk/Flowblade/tools/batchrendering.py
index b136cb59..9c534150 100644
--- a/flowblade-trunk/Flowblade/tools/batchrendering.py
+++ b/flowblade-trunk/Flowblade/tools/batchrendering.py
@@ -29,7 +29,7 @@ from gi.repository import Gtk, Gdk, GdkPixbuf
 import dbus
 import dbus.service
 from dbus.mainloop.glib import DBusGMainLoop
-import mlt
+import mlt7 as mlt
 import hashlib
 import locale
 import os
diff --git a/flowblade-trunk/Flowblade/tools/blenderheadless.py b/flowblade-trunk/Flowblade/tools/blenderheadless.py
index a0df37b9..ad2645c6 100644
--- a/flowblade-trunk/Flowblade/tools/blenderheadless.py
+++ b/flowblade-trunk/Flowblade/tools/blenderheadless.py
@@ -30,7 +30,7 @@ gi.require_version('Gtk', '3.0')
 from gi.repository import GLib
 
 import locale
-import mlt
+import mlt7 as mlt
 import os
 from os import listdir
 from os.path import isfile, join
diff --git a/flowblade-trunk/Flowblade/tools/gmic.py b/flowblade-trunk/Flowblade/tools/gmic.py
index 1053b5e8..d6539f41 100644
--- a/flowblade-trunk/Flowblade/tools/gmic.py
+++ b/flowblade-trunk/Flowblade/tools/gmic.py
@@ -29,7 +29,7 @@ from gi.repository import Pango
 
 import cairo
 import locale
-import mlt
+import mlt7 as mlt
 import os
 import re
 import shutil
diff --git a/flowblade-trunk/Flowblade/tools/gmicheadless.py b/flowblade-trunk/Flowblade/tools/gmicheadless.py
index 433c7722..424fd977 100644
--- a/flowblade-trunk/Flowblade/tools/gmicheadless.py
+++ b/flowblade-trunk/Flowblade/tools/gmicheadless.py
@@ -30,7 +30,7 @@ gi.require_version('Gtk', '3.0')
 from gi.repository import GLib
 
 import locale
-import mlt
+import mlt7 as mlt
 import os
 import pickle
 import subprocess
diff --git a/flowblade-trunk/Flowblade/tools/gmicplayer.py b/flowblade-trunk/Flowblade/tools/gmicplayer.py
index 4fd0aac5..61037abb 100644
--- a/flowblade-trunk/Flowblade/tools/gmicplayer.py
+++ b/flowblade-trunk/Flowblade/tools/gmicplayer.py
@@ -23,7 +23,7 @@ Clip player used to select frames for preview and range selection.
 """
 
 
-import mlt
+import mlt7 as mlt
 import os
 from os import listdir
 from os.path import isfile, join
diff --git a/flowblade-trunk/Flowblade/tools/mltheadlessutils.py b/flowblade-trunk/Flowblade/tools/mltheadlessutils.py
index ed0b64d9..4b8437df 100644
--- a/flowblade-trunk/Flowblade/tools/mltheadlessutils.py
+++ b/flowblade-trunk/Flowblade/tools/mltheadlessutils.py
@@ -24,7 +24,7 @@ in initialized Flowblade/MLT enviroment.
 """
 
 import locale
-import mlt
+import mlt7 as mlt
 import os
 
 import ccrutils
diff --git a/flowblade-trunk/Flowblade/tools/mltxmlheadless.py b/flowblade-trunk/Flowblade/tools/mltxmlheadless.py
index cc24e216..194be4f3 100644
--- a/flowblade-trunk/Flowblade/tools/mltxmlheadless.py
+++ b/flowblade-trunk/Flowblade/tools/mltxmlheadless.py
@@ -30,7 +30,7 @@ gi.require_version('Gtk', '3.0')
 from gi.repository import GLib
 
 import locale
-import mlt
+import mlt7 as mlt
 import os
 import pickle
 import subprocess
diff --git a/flowblade-trunk/Flowblade/tools/motionheadless.py b/flowblade-trunk/Flowblade/tools/motionheadless.py
index f0dc350b..516ea16a 100644
--- a/flowblade-trunk/Flowblade/tools/motionheadless.py
+++ b/flowblade-trunk/Flowblade/tools/motionheadless.py
@@ -18,7 +18,7 @@
     along with Flowblade Movie Editor. If not, see <http://www.gnu.org/licenses/>.
 """
 
-import mlt
+import mlt7 as mlt
 import os
 import threading
 import time
diff --git a/flowblade-trunk/Flowblade/tools/proxyheadless.py b/flowblade-trunk/Flowblade/tools/proxyheadless.py
index 1a34f437..c3f090f8 100644
--- a/flowblade-trunk/Flowblade/tools/proxyheadless.py
+++ b/flowblade-trunk/Flowblade/tools/proxyheadless.py
@@ -19,7 +19,7 @@
 """
 
 import glob
-import mlt
+import mlt7 as mlt
 import os
 from PIL import Image
 import threading
diff --git a/flowblade-trunk/flowblade b/flowblade-trunk/flowblade
index 01f6d092..7c8bc14d 100755
--- a/flowblade-trunk/flowblade
+++ b/flowblade-trunk/flowblade
@@ -57,7 +57,7 @@ sys.path.insert(0, modules_path)
 
 # Check that we have MLT, missing is fatal.
 try:
-    import mlt
+    import mlt7 as mlt
     try:
         mlt_version = mlt.LIBMLT_VERSION
         print ("MLT found, version:", mlt_version)
