diff --git a/ginac/function.cpp b/ginac/function.cpp
index c158723..689e2b8 100644
--- a/ginac/function.cpp
+++ b/ginac/function.cpp
@@ -21,6 +21,7 @@
  */
 
 #define register
+#define PY_SSIZE_T_CLEAN
 #include <Python.h>
 #include "py_funcs.h"
 #include "function.h"
diff --git a/ginac/numeric.cpp b/ginac/numeric.cpp
index 276d86c..b463806 100644
--- a/ginac/numeric.cpp
+++ b/ginac/numeric.cpp
@@ -50,6 +50,7 @@
  */
 
 #define register
+#define PY_SSIZE_T_CLEAN
 #include <Python.h>
 #include <longintrepr.h>
 #include "flint/fmpz.h"
