diff -urN goldendict-1.0.1-src.orig/processwrapper.cc goldendict-1.0.1-src/processwrapper.cc
--- goldendict-1.0.1-src.orig/processwrapper.cc	2010-12-04 02:42:46.000000000 +0530
+++ goldendict-1.0.1-src/processwrapper.cc	2012-08-04 20:51:45.835055609 +0530
@@ -94,6 +94,8 @@
 
 #else
 
+#include <unistd.h>
+
 unsigned int ProcessWrapper::currentProcessId()
 {
     return getpid();
diff -urN goldendict-1.0.1-src.orig/qtsingleapplication/src/qtlocalpeer.cpp goldendict-1.0.1-src/qtsingleapplication/src/qtlocalpeer.cpp
--- goldendict-1.0.1-src.orig/qtsingleapplication/src/qtlocalpeer.cpp	2010-12-04 02:42:46.000000000 +0530
+++ goldendict-1.0.1-src/qtsingleapplication/src/qtlocalpeer.cpp	2012-08-04 20:48:47.761593825 +0530
@@ -50,6 +50,7 @@
 #endif
 #if defined(Q_OS_UNIX)
 #include <time.h>
+#include <unistd.h>
 #endif
 
 namespace QtLP_Private {
