diff -aur old/src/oping.c new/src/oping.c
--- old/src/oping.c	2011-01-26 14:18:42.000000000 +0100
+++ new/src/oping.c	2011-02-27 16:48:51.046668361 +0100
@@ -672,7 +672,7 @@
 			context->window = NULL;
 		}
 		context->window = newwin (/* height = */ 4,
-				/* width = */ 0,
+				/* width = */ width,
 				/* y = */ main_win_height + (4 * context->index),
 				/* x = */ 0);
 	}
@@ -725,7 +725,7 @@
 
 	main_win_height = height - (4 * host_num);
 	main_win = newwin (/* height = */ main_win_height,
-			/* width = */ 0,
+			/* width = */ width,
 			/* y = */ 0, /* x = */ 0);
 	/* Allow scrolling */
 	scrollok (main_win, TRUE);
@@ -751,7 +751,7 @@
 			context->window = NULL;
 		}
 		context->window = newwin (/* height = */ 4,
-				/* width = */ 0,
+				/* width = */ width,
 				/* y = */ main_win_height + (4 * context->index),
 				/* x = */ 0);
 	}
