diff --color -aur j2cli-0.3.12b-old/j2cli/__init__.py j2cli-0.3.12b-new/j2cli/__init__.py
--- j2cli-0.3.12b-old/j2cli/__init__.py	2022-11-23 00:43:32.691081564 +0200
+++ j2cli-0.3.12b-new/j2cli/__init__.py	2022-11-23 00:47:19.301073735 +0200
@@ -5,7 +5,11 @@
 
 __author__ = "Mark Vartanyan"
 __email__ = "kolypto@gmail.com"
-__version__ = pkg_resources.get_distribution('j2cli').version
+
+try:
+  __version__ = pkg_resources.get_distribution('j2cli').version
+except pkg_resources.DistributionNotFound:
+  __version__ = '0.3.12b'
 
 from j2cli.cli import main
 
