diff -rup luxrender-luxrays-475fbf15f0ca/samples/luxmark/luxmarkapp.h luxrender-luxrays-475fbf15f0ca.new/samples/luxmark/luxmarkapp.h
--- luxrender-luxrays-475fbf15f0ca/samples/luxmark/luxmarkapp.h	2013-02-22 22:29:42.000000000 +0100
+++ luxrender-luxrays-475fbf15f0ca.new/samples/luxmark/luxmarkapp.h	2013-06-24 14:54:12.946992718 +0200
@@ -36,11 +36,11 @@
 //------------------------------------------------------------------------------
 
 // List of supported scenes
-#define SCENE_ROOM "scenes/room/render.cfg"
-#define SCENE_SALA "scenes/sala/render.cfg"
-#define SCENE_LUXBALL_HDR "scenes/luxball/render-hdr.cfg"
-#define SCENE_LUXBALL "scenes/luxball/render.cfg"
-#define SCENE_LUXBALL_SKY "scenes/luxball/render-sunset.cfg"
+#define SCENE_ROOM "/usr/share/luxmark/room/render.cfg"
+#define SCENE_SALA "/usr/share/luxmark/sala/render.cfg"
+#define SCENE_LUXBALL_HDR "/usr/share/luxmark/luxball/render-hdr.cfg"
+#define SCENE_LUXBALL "/usr/share/luxmark/luxball/render.cfg"
+#define SCENE_LUXBALL_SKY "/usr/share/luxmark/luxball/render-sunset.cfg"
 
 class LuxMarkApp : public QApplication {
 	Q_OBJECT
