--- a/inter.c	2014-04-12 18:48:39.020103780 +0200
+++ b/inter.c	2014-04-12 18:48:33.323365100 +0200
@@ -556,7 +556,7 @@
 static char *interactive_text_readline(char *prompt, char *initval, int *is_blank, int *error)/*{{{*/
 {
   char *line;
-  Function *old_rl_pre_input_hook = NULL;
+  rl_command_func_t *old_rl_pre_input_hook = NULL;
   
   *error = 0;
   old_rl_pre_input_hook = rl_pre_input_hook;
@@ -633,9 +633,9 @@
 #ifdef USE_READLINE
   if (isatty(0)) {
     char *result;
-    rl_attempted_completion_function = (CPPFunction *) null_tdl_completion;
+    rl_attempted_completion_function = null_tdl_completion;
     result = interactive_text_readline(prompt, initval, is_blank, error);
-    rl_attempted_completion_function = (CPPFunction *) tdl_completion;
+    rl_attempted_completion_function = tdl_completion;
     return result;
   } else {
     /* In case someone wants to drive tdl from a script, by redirecting stdin to it. */
@@ -653,7 +653,7 @@
 #ifdef USE_READLINE
   if (isatty(0)) {
     rl_completion_entry_function = NULL;
-    rl_attempted_completion_function = (CPPFunction *) tdl_completion;
+    rl_attempted_completion_function = tdl_completion;
     interactive_readline();
   } else {
     /* In case someone wants to drive tdl from a script, by redirecting stdin to it. */
