=== modified file 'Onboard/Config.py'
--- Onboard/Config.py	2012-10-18 16:17:12 +0000
+++ Onboard/Config.py	2012-10-18 17:19:00 +0000
@@ -73,7 +73,7 @@
 DEFAULT_RESIZE_HANDLES     = list(Handle.RESIZERS)
 
 SCHEMA_VERSION_0_97         = Version(1, 0)   # Onboard 0.97
-SCHEMA_VERSION_0_98       = Version(2, 0)   # Onboard 0.97.1
+SCHEMA_VERSION_0_98         = Version(2, 0)   # Onboard 0.97.1
 SCHEMA_VERSION              = SCHEMA_VERSION_0_98
 
 
@@ -145,7 +145,7 @@
         Singleton magic.
         """
         if not hasattr(cls, "self"):
-            cls.self = object.__new__(cls, args, kwargs)
+            cls.self = object.__new__(cls, *args, **kwargs)
             cls.self.construct()
         return cls.self
 

