=== modified file 'Onboard/Appearance.py'
--- Onboard/Appearance.py	2012-02-14 08:16:34 +0000
+++ Onboard/Appearance.py	2012-02-15 23:27:20 +0000
@@ -11,7 +11,6 @@
 _logger = logging.getLogger("Appearance")
 ###############
 
-from gettext import gettext as _
 from xml.dom import minidom
 import os
 import re

=== modified file 'Onboard/Config.py'
--- Onboard/Config.py	2012-02-15 10:58:02 +0000
+++ Onboard/Config.py	2012-02-15 23:27:20 +0000
@@ -9,7 +9,6 @@
 import sys
 from shutil import copytree
 from optparse import OptionParser
-from gettext import gettext as _
 
 from gi.repository import GLib, Gtk
 

=== modified file 'Onboard/ConfigUtils.py'
--- Onboard/ConfigUtils.py	2012-02-08 14:56:38 +0000
+++ Onboard/ConfigUtils.py	2012-02-15 23:27:20 +0000
@@ -13,7 +13,6 @@
 import os
 import sys
 from ast import literal_eval
-from gettext import gettext as _
 try:
     import configparser
 except ImportError:
@@ -418,8 +417,8 @@
             # convert ini file strings to property values
             sysdef_gskeys = dict((k.sysdef, k) for k in list(self.gskeys.values()))
             for sysdef, value in items:
-                _logger.info(_("Found system default '{}={}'") \
-                              .format(sysdef, value))
+                _logger.info(_("Found system default '[{}] {}={}'") \
+                              .format(self.sysdef_section, sysdef, value))
 
                 gskey = sysdef_gskeys.get(sysdef, None)
                 value = self._convert_sysdef_key(gskey, sysdef, value)

=== modified file 'Onboard/IconPalette.py'
--- Onboard/IconPalette.py	2012-02-15 01:34:13 +0000
+++ Onboard/IconPalette.py	2012-02-15 23:27:20 +0000
@@ -43,7 +43,6 @@
 config = Config()
 ########################
 
-from gettext import gettext as _
 
 class IconPalette(Gtk.Window, WindowRectTracker, WindowManipulator):
     """

=== modified file 'Onboard/Indicator.py'
--- Onboard/Indicator.py	2012-02-11 14:07:56 +0000
+++ Onboard/Indicator.py	2012-02-15 23:27:20 +0000
@@ -18,7 +18,6 @@
 config = Config()
 ########################
 
-from gettext import gettext as _
 
 class Indicator(GObject.GObject):
 

=== modified file 'Onboard/KbdWindow.py'
--- Onboard/KbdWindow.py	2012-02-15 10:18:48 +0000
+++ Onboard/KbdWindow.py	2012-02-15 23:27:20 +0000
@@ -11,7 +11,6 @@
 from Onboard.WindowUtils import Orientation, WindowRectTracker
 from Onboard.IconPalette import IconPalette
 
-from gettext import gettext as _
 
 ### Logging ###
 import logging

=== modified file 'Onboard/Keyboard.py'
--- Onboard/Keyboard.py	2012-02-14 20:21:02 +0000
+++ Onboard/Keyboard.py	2012-02-15 23:27:20 +0000
@@ -3,7 +3,6 @@
 from __future__ import division, print_function, unicode_literals
 
 import sys
-from gettext import gettext as _
 
 from gi.repository import GObject, Gtk, Gdk
 

=== modified file 'Onboard/KeyboardGTK.py'
--- Onboard/KeyboardGTK.py	2012-02-15 01:33:34 +0000
+++ Onboard/KeyboardGTK.py	2012-02-15 23:27:20 +0000
@@ -5,7 +5,6 @@
 
 import os
 import time
-from gettext import gettext as _
 
 import cairo
 from gi.repository import GObject, Gdk, Gtk

=== modified file 'Onboard/KeyboardSVG.py'
--- Onboard/KeyboardSVG.py	2012-02-09 19:18:36 +0000
+++ Onboard/KeyboardSVG.py	2012-02-15 23:27:20 +0000
@@ -11,7 +11,6 @@
 import re
 import sys
 import shutil
-from gettext import gettext as _
 from xml.dom import minidom
 
 from Onboard             import Exceptions

=== modified file 'Onboard/MouseControl.py'
--- Onboard/MouseControl.py	2012-01-19 08:50:34 +0000
+++ Onboard/MouseControl.py	2012-02-15 23:27:20 +0000
@@ -5,7 +5,6 @@
 
 from __future__ import division, print_function, unicode_literals
 
-from gettext import gettext as _
 try:
     import dbus
     from dbus.mainloop.glib import DBusGMainLoop

=== modified file 'Onboard/OnboardGtk.py'
--- Onboard/OnboardGtk.py	2012-02-15 10:58:02 +0000
+++ Onboard/OnboardGtk.py	2012-02-15 23:27:20 +0000
@@ -13,14 +13,14 @@
 import signal
 import gettext
 import os.path
-from gettext import gettext as _
 import virtkey
-
 from gi.repository import GObject, Gio, Gdk, Gtk
 
+# setup gettext, install _() function for all modules
+app = "onboard"
+gettext.install(app, unicode=True)
 
 from Onboard.Indicator import Indicator
-
 from Onboard.Keyboard import Keyboard
 from Onboard.Scanner import Scanner
 from Onboard.KeyGtk import *
@@ -37,11 +37,6 @@
 import Onboard.KeyCommon
 import Onboard.utils as utils
 
-#setup gettext
-app="onboard"
-gettext.textdomain(app)
-gettext.bindtextdomain(app)
-
 DEFAULT_FONTSIZE = 10
 
 class OnboardGtk(Gtk.Application):

=== modified file 'Onboard/SnippetView.py'
--- Onboard/SnippetView.py	2012-01-09 22:25:10 +0000
+++ Onboard/SnippetView.py	2012-02-15 23:27:20 +0000
@@ -4,7 +4,6 @@
 
 from gi.repository import GObject, Gtk
 
-from gettext import gettext as _
 
 from Onboard.utils import show_error_dialog
 

=== modified file 'Onboard/TouchHandles.py'
--- Onboard/TouchHandles.py	2012-02-14 08:32:08 +0000
+++ Onboard/TouchHandles.py	2012-02-15 23:27:20 +0000
@@ -5,7 +5,6 @@
 
 from math import pi, sqrt, sin, log
 import cairo
-from gettext import gettext as _
 
 from Onboard.utils       import Rect
 from Onboard.WindowUtils import Handle

=== modified file 'Onboard/WindowUtils.py'
--- Onboard/WindowUtils.py	2012-02-15 10:23:36 +0000
+++ Onboard/WindowUtils.py	2012-02-15 23:27:20 +0000
@@ -5,7 +5,6 @@
 
 import time
 from math import sqrt
-from gettext import gettext as _
 
 from gi.repository import GObject, Gtk, Gdk
 

=== modified file 'Onboard/settings.py'
--- Onboard/settings.py	2012-02-15 10:58:02 +0000
+++ Onboard/settings.py	2012-02-15 23:27:20 +0000
@@ -11,9 +11,14 @@
 from subprocess import Popen
 from xml.parsers.expat import ExpatError
 from xml.dom import minidom
+import gettext
 
 from gi.repository import GObject, Pango, Gdk, Gtk
 
+# setup gettext, install _() function for all modules
+app = "onboard"
+gettext.install(app, unicode=True)
+
 from Onboard.KeyboardSVG import KeyboardSVG
 from Onboard.SnippetView import SnippetView
 from Onboard.Appearance  import Theme, ColorScheme
@@ -36,14 +41,6 @@
 config = Config()
 ########################
 
-#setup gettext
-import gettext
-from gettext import gettext as _
-app = "onboard"
-gettext.textdomain(app)
-gettext.bindtextdomain(app)
-
-
 
 def LoadUI(filebase):
     builder = Gtk.Builder()

=== modified file 'Onboard/utils.py'
--- Onboard/utils.py	2012-02-15 01:33:34 +0000
+++ Onboard/utils.py	2012-02-15 23:27:20 +0000
@@ -11,7 +11,6 @@
 from subprocess import Popen
 from math import pi, sqrt, sin
 from contextlib import contextmanager
-from gettext import gettext as _
 
 from gi.repository import GObject, Gtk
 

