diff --git a/presets/Makefile.am b/presets/Makefile.am
--- a/presets/Makefile.am
+++ b/presets/Makefile.am
@@ -1,5 +1,5 @@
 
-presetdir = $(datadir)/transmageddon/presets
+presetdir = $(datadir)/gstreamer-0.10/presets
 preset_DATA = ffenc_ac3.prs GstFaac.prs GstVorbisEnc.prs \
               ffenc_flv.prs GstLameMP3Enc.prs \
               ffenc_h263p.prs GstXvidEnc.prs \
diff --git a/src/transcoder_engine.py b/src/transcoder_engine.py
--- a/src/transcoder_engine.py
+++ b/src/transcoder_engine.py
@@ -56,9 +57,6 @@ class Transcoder(gobject.GObject):
                self.containercaps=gst.Caps("application/x-id3")
                self.container=gst.Caps("application/x-id3")
 
-       # set preset directory 
-       gst.preset_set_app_dir("/usr/share/transmageddon/presets/")
-   
        # Choose plugin based on Codec Name
        # or switch to remuxing mode if any of the values are set to 'pastr'
        self.stoptoggle=False
