From f10c60d254bce7b73738415b4f7a87f8c74d01a2 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Thu, 26 Jul 2018 04:47:04 +0000
Subject: [PATCH] fix build on python 3.7

---
 Qt/Python/pqPythonSyntaxHighlighter.cxx | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Qt/Python/pqPythonSyntaxHighlighter.cxx b/Qt/Python/pqPythonSyntaxHighlighter.cxx
index 36e704e7d5..021be2d666 100644
--- a/Qt/Python/pqPythonSyntaxHighlighter.cxx
+++ b/Qt/Python/pqPythonSyntaxHighlighter.cxx
@@ -194,6 +194,8 @@ void pqPythonSyntaxHighlighter::rehighlightSyntax()
 #if PY_MAJOR_VERSION == 2
     vtkSmartPyObject resultingTextBytes(PyUnicode_AsUTF8String(resultingText));
     char* resultingTextAsCString = PyString_AsString(resultingTextBytes);
+#elif PY_VERSION_HEX >= 0x03070000
+    char* resultingTextAsCString = const_cast<char *>(PyUnicode_AsUTF8(resultingText));
 #else
     char* resultingTextAsCString = PyUnicode_AsUTF8(resultingText);
 #endif
-- 
2.18.0

