From 783679b3b315905932882f09013ebc9f7131f763 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 20 Feb 2016 11:46:55 +0100
Subject: [PATCH] batt plugin: Fix with GTK3

---
 plugins/batt/batt.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugins/batt/batt.c b/plugins/batt/batt.c
index 4c081fd..a9275f5 100644
--- a/plugins/batt/batt.c
+++ b/plugins/batt/batt.c
@@ -394,7 +394,11 @@ static gint configureEvent(GtkWidget *widget, GdkEventConfigure *event,
 }
 
 
+#if GTK_CHECK_VERSION(3, 0, 0)
+static gint draw(GtkWidget *widget, cairo_t *cr, lx_battery *lx_b) {
+#else
 static gint exposeEvent(GtkWidget *widget, GdkEventExpose *event, lx_battery *lx_b) {
+#endif
 
     ENTER;
 
-- 
2.7.1

