From b8cf05aeb294ddcd06d79c6e838de84cf32f581b Mon Sep 17 00:00:00 2001
From: Guido Scholz <gscholz@users.sourceforge.net>
Date: Tue, 6 Nov 2018 21:35:48 +0100
Subject: [PATCH 01/11] Use iterator instead of typeof

---
 src/m_pcmout.cpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/m_pcmout.cpp b/src/m_pcmout.cpp
index 47624b2..1efb4fe 100644
--- a/src/m_pcmout.cpp
+++ b/src/m_pcmout.cpp
@@ -54,9 +54,12 @@ void M_pcmout::mcAbleChanged(MidiControllableBase *mcAble)
   else
     QTextStream(&i3Name) << "K. " << ag;
 
-  for (typeof(mcAble->mcws.constBegin()) mcw = mcAble->mcws.constBegin();
-       mcw != mcAble->mcws.constEnd();  mcw++) {
-    MidiComboBox *b = dynamic_cast<MidiComboBox *>(*mcw);
+  QList<MCed*>::const_iterator it;
+  for (it = mcAble->mcws.constBegin(); it != mcAble->mcws.constEnd(); ++it) {
+  //for (typeof(mcAble->mcws.constBegin()) mcw = mcAble->mcws.constBegin();
+  //     mcw != mcAble->mcws.constEnd();  mcw++) {
+    //MidiComboBox *b = dynamic_cast<MidiComboBox *>(*mcw);
+    MidiComboBox *b = dynamic_cast<MidiComboBox *>(*it);
     if (b) {
       b->comboBox->setItemText(2, i3Name);
       b->comboBox->update();
-- 
2.21.0

