From 08b3f11cb5000540733549288878a6c0ada554e6 Mon Sep 17 00:00:00 2001
From: Laurent Carlier <lordheavym@gmail.com>
Date: Fri, 22 Nov 2019 17:51:06 +0100
Subject: [PATCH] Fix building with libglvnd 1.2.0

Signed-off-by: Laurent Carlier <lordheavym@gmail.com>
---
 configure.ac | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 213399f8..988fda67 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1212,7 +1212,7 @@ AS_IF([test "x$NEED_EGL" = "xyes" && test "x$EGL_CHECKED" != "xyes"],
               [],
               [AC_MSG_ERROR([Unable to locate required EGL headers])])
             AC_CHECK_HEADERS(
-              [EGL/eglext.h],
+              [EGL/eglext.h EGL/eglmesaext.h],
               [],
               [AC_MSG_ERROR([Unable to locate required EGL headers])],
               [#include <EGL/egl.h>])
@@ -1226,7 +1226,8 @@ AS_IF([test "x$NEED_EGL" = "xyes" && test "x$EGL_CHECKED" != "xyes"],
           )
 
         COGL_EGL_INCLUDES="#include <EGL/egl.h>
-#include <EGL/eglext.h>"
+#include <EGL/eglext.h>
+#include <EGL/eglmesaext.h>"
         AC_SUBST([COGL_EGL_INCLUDES])
       ])
 
-- 
2.24.0

