diff --git a/src/modules/rlm_python/configure b/src/modules/rlm_python/configure
index 636acb13c8..9240afc015 100755
--- a/src/modules/rlm_python/configure
+++ b/src/modules/rlm_python/configure
@@ -2994,7 +2994,7 @@ $as_echo "$as_me: Python sys.prefix \"${PY_PREFIX}\"" >&6;}
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: Python sys.exec_prefix \"${PY_EXEC_PREFIX}\"" >&5
 $as_echo "$as_me: Python sys.exec_prefix \"${PY_EXEC_PREFIX}\"" >&6;}
 
-	PY_SYS_VERSION=`${PYTHON_BIN} -c 'import sys ; print(sys.version[0:3])'`
+	PY_SYS_VERSION=`${PYTHON_BIN} -c 'import sys ; print(".".join(map(str, sys.version_info[:2])))'`
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: Python sys.version \"${PY_SYS_VERSION}\"" >&5
 $as_echo "$as_me: Python sys.version \"${PY_SYS_VERSION}\"" >&6;}
 
