From 4f4c8e873682f10eeddb0bc86e08f3b64b5dcb68 Mon Sep 17 00:00:00 2001
From: Guido Scholz <gscholz@users.sourceforge.net>
Date: Tue, 6 Nov 2018 21:38:35 +0100
Subject: [PATCH 02/11] Fix obsolete white spaces

---
 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 15cc3af..94ccbc3 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -278,7 +278,7 @@ SUFFIXES = _moc.cpp
 
 # rule for translation file updates
 $(translations): $(ams_SOURCES)
-	        $(LUPDATE) -verbose $(ams_SOURCES) -ts $@
+	$(LUPDATE) -verbose $(ams_SOURCES) -ts $@
 
 # rule to generate binary translation files
 .ts.qm: $(translations)
-- 
2.21.0

