Do not force custom fontconfig configuration.
Index: tdesktop-1.9.4-full/Telegram/lib_ui/fonts/fonts.qrc
===================================================================
--- tdesktop-1.9.4-full.orig/Telegram/lib_ui/fonts/fonts.qrc
+++ tdesktop-1.9.4-full/Telegram/lib_ui/fonts/fonts.qrc
@@ -1,10 +1,4 @@
 <RCC>
   <qresource prefix="/gui/fonts">
-    <file>DAOpenSansRegular.ttf</file>
-    <file>DAOpenSansRegularItalic.ttf</file>
-    <file>DAOpenSansBold.ttf</file>
-    <file>DAOpenSansBoldItalic.ttf</file>
-    <file>DAOpenSansSemibold.ttf</file>
-    <file>DAOpenSansSemiboldItalic.ttf</file>
   </qresource>
 </RCC>
Index: tdesktop-1.9.4-full/Telegram/lib_ui/ui/style/style_core_font.cpp
===================================================================
--- tdesktop-1.9.4-full.orig/Telegram/lib_ui/ui/style/style_core_font.cpp
+++ tdesktop-1.9.4-full/Telegram/lib_ui/ui/style/style_core_font.cpp
@@ -94,12 +94,12 @@ enum {
 	FontTypesCount,
 };
 QString FontTypeNames[FontTypesCount] = {
-	"DAOpenSansRegular",
-	"DAOpenSansRegularItalic",
-	"DAOpenSansBold",
-	"DAOpenSansBoldItalic",
-	"DAOpenSansSemibold",
-	"DAOpenSansSemiboldItalic",
+	"Open Sans",
+	"Open Sans",
+	"Open Sans",
+	"Open Sans",
+	"Open Sans Semibold",
+	"Open Sans Semibold",
 };
 int32 FontTypeFlags[FontTypesCount] = {
 	0,
@@ -135,7 +135,7 @@ void StartFonts() {
 	for (auto i = 0; i != FontTypesCount; ++i) {
 		const auto name = FontTypeNames[i];
 		const auto flags = FontTypeFlags[i];
-		areGood[i] = LoadCustomFont(":/gui/fonts/" + name + ".ttf", name, flags);
+		areGood[i] = ValidateFont(name, flags);
 		Overrides[i] = name;
 #ifdef Q_OS_WIN
 		// Attempt to workaround a strange font bug with Open Sans Semibold not loading.
Index: tdesktop-1.9.4-full/Telegram/lib_ui/qt_conf/linux.qrc
===================================================================
--- tdesktop-1.9.4-full.orig/Telegram/lib_ui/qt_conf/linux.qrc
+++ tdesktop-1.9.4-full/Telegram/lib_ui/qt_conf/linux.qrc
@@ -3,6 +3,5 @@
     <file alias="qt.conf">qt_linux.conf</file>
   </qresource>
   <qresource prefix="/fc">
-    <file>fc-custom.conf</file>
   </qresource>
 </RCC>
