From e766434fe01f4e0d9a43b18c65908adf43f8a80e Mon Sep 17 00:00:00 2001
From: Guido Scholz <gscholz@users.sourceforge.net>
Date: Wed, 7 Nov 2018 20:33:14 +0100
Subject: [PATCH 09/11] Select Qt5 as default instead of Qt4

---
 configure.ac | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index b84b352..784ca67 100644
--- a/configure.ac
+++ b/configure.ac
@@ -110,10 +110,10 @@ LDFLAGS="$saved_libs"
 AC_LANG_POP(C++)
 fi
 
-dnl option to enable Qt5 instead of default Qt4 library 
-AC_ARG_ENABLE(qt5, AC_HELP_STRING([--enable-qt5],
-                          [enable Qt5 instead of Qt4 library]),,
-                          enable_qt4=yes)
+dnl option to enable Qt4 instead of default Qt5 library 
+AC_ARG_ENABLE(qt4, AC_HELP_STRING([--enable-qt4],
+                          [enable Qt4 instead of Qt5 library]),,
+                          enable_qt5=yes)
 AS_IF([test x$enable_qt4 = xyes],
 dnl Check for Qt4 library files
 [
-- 
2.21.0

