From 51dedd859764ed2063de43b89c21586640adff60 Mon Sep 17 00:00:00 2001
From: "Carsten Haitzler (Rasterman)" <raster@rasterman.com>
Date: Wed, 26 May 2021 01:28:38 +0100
Subject: [PATCH 14/14] alpm - fix db downloading to work

---
 backends/alpm/pk-backend-alpm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backends/alpm/pk-backend-alpm.c b/backends/alpm/pk-backend-alpm.c
index 5571aaafb..f66633e11 100644
--- a/backends/alpm/pk-backend-alpm.c
+++ b/backends/alpm/pk-backend-alpm.c
@@ -80,7 +80,7 @@ pk_alpm_initialize (PkBackend *backend, GError **error)
 {
 	PkBackendAlpmPrivate *priv = pk_backend_get_user_data (backend);
 
-	priv->alpm = pk_alpm_configure (backend, PK_BACKEND_CONFIG_FILE, FALSE, error);
+	priv->alpm = pk_alpm_configure (backend, PK_BACKEND_CONFIG_FILE, TRUE, error);
 	if (priv->alpm == NULL) {
 		g_prefix_error (error, "using %s: ", PK_BACKEND_CONFIG_FILE);
 		return FALSE;
-- 
2.31.1

