# https://bugs.archlinux.org/task/35420
--- a/src/perl/tabbed	2012-09-12 17:32:04.000000000 +0900
+++ b/src/perl/tabbed	2013-05-24 13:04:37.000000000 +0900
@@ -402,7 +402,7 @@
 # simply proxies all interesting calls back to the tabbed class.
 
 {
-   for my $hook qw(start destroy key_press property_notify) {
+   for my $hook (qw(start destroy key_press property_notify)) {
       eval qq{
          sub on_$hook {
             my \$parent = \$_[0]{term}{parent}
