diff -Naurp a/contrib/i2pd.conf b/contrib/i2pd.conf
--- a/contrib/i2pd.conf	2019-08-27 14:17:32.000000000 +0000
+++ b/contrib/i2pd.conf	2019-10-03 01:26:29.274686807 +0000
@@ -7,13 +7,13 @@
 ## by removing the "#" symbol.
 
 ## Tunnels config file
-## Default: ~/.i2pd/tunnels.conf or /var/lib/i2pd/tunnels.conf
+## Default: /var/lib/i2pd/tunnels.conf (symlink to /etc/i2pd/tunnels.conf)
 # tunconf = /var/lib/i2pd/tunnels.conf
 
 ## Tunnels config files path
 ## Use that path to store separated tunnels in different config files.
-## Default: ~/.i2pd/tunnels.d or /var/lib/i2pd/tunnels.d
-# tunnelsdir = /var/lib/i2pd/tunnels.conf.d
+## Default: /var/lib/i2pd/tunnels.d (symlink to /etc/i2pd/tunnels.d)
+# tunnelsdir = /var/lib/i2pd/tunnels.d
 
 ## Where to write pidfile (don't write by default)
 # pidfile = /var/run/i2pd.pid
@@ -27,7 +27,7 @@
 ##  * syslog - use syslog, see man 3 syslog
 # log = file
 ## Path to logfile (default - autodetect)
-# logfile = /var/log/i2pd.log
+logfile = /var/log/i2pd/i2pd.log
 ## Log messages above this level (debug, *info, warn, error, none)
 ## If you set it to none, logging will be disabled
 # loglevel = info
