From 427f770374ec2a583426aa371105f8c0780a729b Mon Sep 17 00:00:00 2001
From: Pete Gadomski <pete.gadomski@gmail.com>
Date: Tue, 25 Apr 2017 16:08:00 +0000
Subject: [PATCH 1/2] Add cmath and limits headers to utils

---
 include/hexer/Utils.hpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/hexer/Utils.hpp b/include/hexer/Utils.hpp
index 8c26fd3..fd7ed30 100644
--- a/include/hexer/Utils.hpp
+++ b/include/hexer/Utils.hpp
@@ -15,6 +15,9 @@
 
 #pragma once
 
+#include <cmath>
+#include <limits>
+
 namespace hexer
 {
 

From e06d361eec613ed046592d378a01bf0a0694a6f8 Mon Sep 17 00:00:00 2001
From: Pete Gadomski <pete.gadomski@gmail.com>
Date: Tue, 25 Apr 2017 16:08:24 +0000
Subject: [PATCH 2/2] Link with dl for curse on non-win32

---
 apps/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/apps/CMakeLists.txt b/apps/CMakeLists.txt
index 3ce66d7..11f5d02 100644
--- a/apps/CMakeLists.txt
+++ b/apps/CMakeLists.txt
@@ -17,6 +17,9 @@ endif()
 if(CURSE)
     add_executable(${CURSE} curse.cpp lasfile.hpp las.hpp las.cpp OGR.hpp OGR.cpp mmaplib.hpp pdal_util_export.hpp ProgramArgs.hpp Utils.cpp Utils.hpp )
     target_link_libraries(${CURSE} ${HEXER_LIB_NAME} ${HEXERBOOST_LIB_NAME})
+    if(NOT WIN32)
+        target_link_libraries(${CURSE} dl)
+    endif()
 endif()
 
 install(TARGETS ${HEXER_UTILITIES}
