commit b63da2b6ab2a623539eb1bc8ed3850cb88ef6f26
Author: Walt Steverson <waltsteverson@users.noreply.github.com>
Date:   Fri Jun 9 11:12:42 2017 -0500

    Add missing header file option6_pdexclude.h

diff --git a/src/lib/dhcp/Makefile.am b/src/lib/dhcp/Makefile.am
index 461429ba8..12c3ebf84 100644
--- a/src/lib/dhcp/Makefile.am
+++ b/src/lib/dhcp/Makefile.am
@@ -99,6 +99,7 @@ libkea_dhcp___include_HEADERS = \
     option6_ia.h \
     option6_iaaddr.h \
     option6_iaprefix.h \
+    option6_pdexclude.h \
     option6_status_code.h \
     option.h \
     option_custom.h \
