Only in performous-1.1: build
diff -upr performous-1.1.orig/game/audio.cc performous-1.1/game/audio.cc
--- performous-1.1.orig/game/audio.cc	2016-01-23 12:40:01.000000000 +0200
+++ performous-1.1/game/audio.cc	2018-06-03 03:29:35.251425495 +0300
@@ -77,7 +77,7 @@ class AudioClock {
 	static ptime getTime() { return microsec_clock::universal_time(); }
 	// Conversion helpers
 	static double getSeconds(time_duration t) { return 1e-6 * t.total_microseconds(); }
-	static time_duration getDuration(double seconds) { return microseconds(1e6 * seconds); }
+	static time_duration getDuration(double seconds) { return microseconds(static_cast<long long>(1e6 * seconds)); }
 
 	mutable boost::mutex m_mutex;
 	ptime m_baseTime; ///< A reference time (corresponds to m_basePos)
@@ -149,7 +149,7 @@ class Music {
 	int64_t m_pos; ///< Current sample position
 	bool m_preview;
 	AudioClock m_clock;
-	time_duration durationOf(int64_t samples) const { return microseconds(1e6 * samples / srate / 2.0); }
+	time_duration durationOf(int64_t samples) const { return microseconds(static_cast<long long>(1e6 * samples / srate / 2.0)); }
 public:
 	bool suppressCenterChannel;
 	double fadeLevel;
