diff -upr SymPol-0.1.9.orig/sympol/raycomputationcdd.cpp SymPol-0.1.9/sympol/raycomputationcdd.cpp
--- SymPol-0.1.9.orig/sympol/raycomputationcdd.cpp	2016-05-08 22:52:35.000000000 +0300
+++ SymPol-0.1.9/sympol/raycomputationcdd.cpp	2020-12-07 01:19:11.840719385 +0200
@@ -25,8 +25,8 @@
 #include "polyhedron.h"
 
 extern "C" {
-	#include <setoper.h>
-	#include <cdd.h>
+	#include <cddlib/setoper.h>
+	#include <cddlib/cdd.h>
 }
 
 #include <ctime>
