diff --git a/src/libpcp/src/check-statics b/src/libpcp/src/check-statics
index 8db3aac..9a7e5c1 100755
--- a/src/libpcp/src/check-statics
+++ b/src/libpcp/src/check-statics
@@ -680,6 +680,7 @@ do
 		    -e 's/b \([_a-zA-Z][_a-zA-Z0-9]*\)\.[0-9]*.[0b]$/b \1/' \
 		    -e 's/\([bds] \).*\.\([_a-zA-Z]\)/\1\2/' \
 		    -e 's/s _glib_relative_date\./s /' \
+		    -e '/ B __gnu_lto_slim/d' \
 		    -e '/ s EH_/d' \
 		    -e '/ b \.bss/d' \
 		    -e '/ d \.data/d' \
