--- a/chrome/browser/ui/libgtk2ui/gtk2_ui.cc
+++ b/chrome/browser/ui/libgtk2ui/gtk2_ui.cc
@@ -488,6 +488,10 @@
       default_font_weight_(gfx::Font::Weight::NORMAL),
       middle_click_action_(GetDefaultMiddleClickAction()),
       device_scale_factor_(1.0) {
+#if GTK_MAJOR_VERSION > 2
+  g_setenv("GDK_SCALE", "1", FALSE);
+  gdk_set_allowed_backends("x11");
+#endif
   GtkInitFromCommandLine(*base::CommandLine::ForCurrentProcess());
 }
 
