diff -rupN Clementine-1.2.3.orig/src/CMakeLists.txt Clementine-1.2.3/src/CMakeLists.txt
--- Clementine-1.2.3.orig/src/CMakeLists.txt	2015-05-30 11:15:10.903093300 +0200
+++ Clementine-1.2.3/src/CMakeLists.txt	2015-05-30 11:15:34.801955100 +0200
@@ -1,6 +1,6 @@
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Woverloaded-virtual -Wall -Wno-sign-compare -Wno-deprecated-declarations -Wno-unused-local-typedefs -Wno-unused-private-field -Wno-unknown-warning-option --std=c++0x -U__STRICT_ANSI__")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Woverloaded-virtual -Wall -Wno-sign-compare -Wno-deprecated-declarations -Wno-unused-local-typedefs --std=c++0x -U__STRICT_ANSI__")
 
 option(BUILD_WERROR "Build with -Werror" ON)
 
diff -rupN Clementine-1.2.3.orig/src/ui/mainwindow.cpp Clementine-1.2.3/src/ui/mainwindow.cpp
--- Clementine-1.2.3.orig/src/ui/mainwindow.cpp	2015-05-30 11:15:10.867894300 +0200
+++ Clementine-1.2.3/src/ui/mainwindow.cpp	2015-05-30 11:15:54.578517000 +0200
@@ -911,7 +911,7 @@ void MainWindow::ScrobblingEnabledChange
   if (ui_->action_toggle_scrobbling->isVisible())
     SetToggleScrobblingIcon(value);
 
-  if (!app_->player()->GetState() == Engine::Idle) {
+  if (app_->player()->GetState() != Engine::Idle) {
     return;
   }
   else {
