diff -Naur clementine-0.7.1.ori/src/config.h.in clementine-0.7.1/src/config.h.in
--- clementine-0.7.1.ori/src/config.h.in	2011-03-29 12:16:15.000000000 -0700
+++ clementine-0.7.1/src/config.h.in	2011-05-09 16:00:26.505080057 -0700
@@ -29,6 +29,7 @@
 #cmakedefine HAVE_STATIC_SQLITE
 
 #cmakedefine HAVE_DBUS
+#cmakedefine HAVE_DEVICEKIT
 #cmakedefine HAVE_GIO
 #cmakedefine HAVE_IMOBILEDEVICE
 #cmakedefine HAVE_LIBARCHIVE
