commit 124058dd0d314e5259e96bf04444e702e80ca5c7
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Mon Dec 10 05:18:13 2018 +0800

    Update cmark-gfm to 0.28.3.gfm.19

diff --git a/generated/unix/cmark-gfm_version.h b/generated/unix/cmark-gfm_version.h
index f98bf7a..88c1ec2 100644
--- a/generated/unix/cmark-gfm_version.h
+++ b/generated/unix/cmark-gfm_version.h
@@ -1,7 +1,7 @@
 #ifndef CMARK_GFM_VERSION_H
 #define CMARK_GFM_VERSION_H
 
-#define CMARK_GFM_VERSION ((0 << 24) | (28 << 16) | (3 << 8) | 17)
-#define CMARK_GFM_VERSION_STRING "0.28.3.gfm.17"
+#define CMARK_GFM_VERSION ((0 << 24) | (28 << 16) | (3 << 8) | 19)
+#define CMARK_GFM_VERSION_STRING "0.28.3.gfm.19"
 
 #endif
diff --git a/src/cmarkgfm/cmark.cffi.h b/src/cmarkgfm/cmark.cffi.h
index c073824..97a620f 100644
--- a/src/cmarkgfm/cmark.cffi.h
+++ b/src/cmarkgfm/cmark.cffi.h
@@ -44,7 +44,7 @@ cmark_node *cmark_parser_finish(cmark_parser *parser);
 #define CMARK_OPT_DEFAULT 0
 #define CMARK_OPT_SOURCEPOS ...
 #define CMARK_OPT_HARDBREAKS ...
-#define CMARK_OPT_SAFE ...
+#define CMARK_OPT_UNSAFE ...
 #define CMARK_OPT_NOBREAKS ...
 #define CMARK_OPT_NORMALIZE ...
 #define CMARK_OPT_VALIDATE_UTF8 ...
diff --git a/src/cmarkgfm/cmark.py b/src/cmarkgfm/cmark.py
index 76e3b1b..f5a5b42 100644
--- a/src/cmarkgfm/cmark.py
+++ b/src/cmarkgfm/cmark.py
@@ -9,7 +9,7 @@ class Options(object):
     CMARK_OPT_DEFAULT = _cmark.lib.CMARK_OPT_DEFAULT
     CMARK_OPT_SOURCEPOS = _cmark.lib.CMARK_OPT_SOURCEPOS
     CMARK_OPT_HARDBREAKS = _cmark.lib.CMARK_OPT_HARDBREAKS
-    CMARK_OPT_SAFE = _cmark.lib.CMARK_OPT_SAFE
+    CMARK_OPT_UNSAFE = _cmark.lib.CMARK_OPT_UNSAFE
     CMARK_OPT_NOBREAKS = _cmark.lib.CMARK_OPT_NOBREAKS
     CMARK_OPT_NORMALIZE = _cmark.lib.CMARK_OPT_NORMALIZE
     CMARK_OPT_VALIDATE_UTF8 = _cmark.lib.CMARK_OPT_VALIDATE_UTF8
