diff --git a/styleplugins/dstyleplugin/painterhelper.cpp b/styleplugins/dstyleplugin/painterhelper.cpp
index 3db216d..66d548f 100644
--- a/styleplugins/dstyleplugin/painterhelper.cpp
+++ b/styleplugins/dstyleplugin/painterhelper.cpp
@@ -18,6 +18,7 @@
 #include "painterhelper.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 
 namespace dstyle {
diff --git a/styleplugins/dstyleplugin/pushbuttonhelper.cpp b/styleplugins/dstyleplugin/pushbuttonhelper.cpp
index a903a7d..31026db 100644
--- a/styleplugins/dstyleplugin/pushbuttonhelper.cpp
+++ b/styleplugins/dstyleplugin/pushbuttonhelper.cpp
@@ -30,6 +30,7 @@
 
 #include <QDebug>
 #include <QStyleOptionButton>
+#include <QPainterPath>
 
 DWIDGET_USE_NAMESPACE
 
diff --git a/styleplugins/dstyleplugin/tabbarhelper.cpp b/styleplugins/dstyleplugin/tabbarhelper.cpp
index 51f677b..e0ef36c 100644
--- a/styleplugins/dstyleplugin/tabbarhelper.cpp
+++ b/styleplugins/dstyleplugin/tabbarhelper.cpp
@@ -33,6 +33,7 @@ DWIDGET_USE_NAMESPACE
 #include <QStyleOptionTabBarBase>
 #include <QStyleOptionToolButton>
 #include <QPainter>
+#include <QPainterPath>
 #include <QToolButton>
 #include <QLinearGradient>
 #include <QDebug>
diff --git a/styleplugins/dstyleplugin/tabwidgethelper.cpp b/styleplugins/dstyleplugin/tabwidgethelper.cpp
index a2b6756..f51d43d 100644
--- a/styleplugins/dstyleplugin/tabwidgethelper.cpp
+++ b/styleplugins/dstyleplugin/tabwidgethelper.cpp
@@ -20,6 +20,7 @@
 #include "painterhelper.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QTabWidget>
 #include <QDebug>
 
