diff --git a/frame/toptoolbar.cpp b/frame/toptoolbar.cpp
index f6f8073..a1dc84e 100644
--- a/frame/toptoolbar.cpp
+++ b/frame/toptoolbar.cpp
@@ -35,6 +35,7 @@
 #include <QDebug>
 #include <QHBoxLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QProcess>
 #include <QResizeEvent>
 #include <QShortcut>
diff --git a/module/view/contents/imageinfowidget.cpp b/module/view/contents/imageinfowidget.cpp
index 534bcbf..f9d5440 100644
--- a/module/view/contents/imageinfowidget.cpp
+++ b/module/view/contents/imageinfowidget.cpp
@@ -33,6 +33,7 @@
 #include <QPushButton>
 #include <QScrollBar>
 #include <QtDebug>
+#include <QPainterPath>
 #include <DPushButton>
 #include <DScrollBar>
 #include <DFontSizeManager>
diff --git a/module/view/contents/ttbcontent.cpp b/module/view/contents/ttbcontent.cpp
index 1f95a63..7d03cdc 100644
--- a/module/view/contents/ttbcontent.cpp
+++ b/module/view/contents/ttbcontent.cpp
@@ -38,6 +38,7 @@
 #include <DApplicationHelper>
 #include <DSpinner>
 #include <QtMath>
+#include <QPainterPath>
 
 DWIDGET_USE_NAMESPACE
 namespace {
diff --git a/thumbnail/thumbnaildelegate.cpp b/thumbnail/thumbnaildelegate.cpp
index a789f41..3b2b50a 100644
--- a/thumbnail/thumbnaildelegate.cpp
+++ b/thumbnail/thumbnaildelegate.cpp
@@ -24,6 +24,7 @@
 #include <QHBoxLayout>
 #include <QLineEdit>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPixmapCache>
 #include <QStandardItemModel>
 #include <QThread>
diff --git a/widgets/dialogs/imginfodialog.cpp b/widgets/dialogs/imginfodialog.cpp
index 9b12d06..cb463eb 100644
--- a/widgets/dialogs/imginfodialog.cpp
+++ b/widgets/dialogs/imginfodialog.cpp
@@ -35,6 +35,7 @@
 #include <DApplicationHelper>
 #include <DDialogCloseButton>
 #include <QGraphicsOpacityEffect>
+#include <QPainterPath>
 
 namespace {
 
diff --git a/widgets/dlmenuarrow.h b/widgets/dlmenuarrow.h
index fe01781..201a6d2 100644
--- a/widgets/dlmenuarrow.h
+++ b/widgets/dlmenuarrow.h
@@ -7,6 +7,7 @@
 #include <QEvent>
 #include <QPaintEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 
 DWIDGET_USE_NAMESPACE
diff --git a/widgets/popupmenustyle.cpp b/widgets/popupmenustyle.cpp
index ddb509c..0fd91f7 100644
--- a/widgets/popupmenustyle.cpp
+++ b/widgets/popupmenustyle.cpp
@@ -19,6 +19,7 @@
 #include <QPixmap>
 #include <QPixmapCache>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStyleOption>
 #include <QStyleOptionMenuItem>
 
