--- def/defines.h.orig	2011-03-12 22:44:51.000000000 +0100
+++ def/defines.h	2011-03-12 22:45:02.000000000 +0100
@@ -27,9 +27,9 @@
 #define DEFINES_H
 
 // OS constants
-#define APP_WIN						true
+#define APP_WIN						false
 #define APP_MAC						false
-#define APP_X11						false
+#define APP_X11						true
 #define APP_PORTABLE				false //at least one of the 3 OS must be true too!
 
 // Languages
