diff --git a/src/wm/20_solus-project.budgie.wm.gschema.override b/src/wm/20_solus-project.budgie.wm.gschema.override
index f568a2c7..8b91bf65 100644
--- a/src/wm/20_solus-project.budgie.wm.gschema.override
+++ b/src/wm/20_solus-project.budgie.wm.gschema.override
@@ -1,7 +1,10 @@
-[org.gnome.mutter]
+[org.gnome.mutter:Budgie]
 edge-tiling = true
 attach-modal-dialogs = true
 
-[org.gnome.desktop.wm.preferences]
+[org.gnome.desktop.background:Budgie]
+picture-uri='file:///usr/share/backgrounds/gnome/adwaita-day.jpg'
+
+[org.gnome.desktop.wm.preferences:Budgie]
 button-layout = 'appmenu:minimize,maximize,close'
 
