diff -wbBur open-vm-tools-2013.09.16-1328054/modules/linux/dkms.conf open-vm-tools-2013.09.16-1328054.q/modules/linux/dkms.conf
--- open-vm-tools-2013.09.16-1328054/modules/linux/dkms.conf	2013-09-23 09:57:48.000000000 +0400
+++ open-vm-tools-2013.09.16-1328054.q/modules/linux/dkms.conf	2013-12-02 14:38:38.385365105 +0400
@@ -15,9 +15,9 @@
 CLEAN[0]="$MAKE_CMD_TMPL -C vmblock clean;    \
           $MAKE_CMD_TMPL -C vmci clean;       \
           $MAKE_CMD_TMPL -C vmhgfs clean;     \
-          $MAKE_CMD_TMPL -C vmsync clean";    \
+          $MAKE_CMD_TMPL -C vmsync clean; \
           $MAKE_CMD_TMPL -C vmxnet clean;     \
-          $MAKE_CMD_TMPL -C vsock clean
+          $MAKE_CMD_TMPL -C vsock clean"
 BUILT_MODULE_NAME[0]="vmblock"
 BUILT_MODULE_NAME[1]="vmci"
 BUILT_MODULE_NAME[2]="vmhgfs"
