diff -Naur gpsbabel-gpsbabel_1_6_0.orig/gui/advdlg.cc gpsbabel-gpsbabel_1_6_0/gui/advdlg.cc
--- gpsbabel-gpsbabel_1_6_0.orig/gui/advdlg.cc	2019-05-02 10:26:47.000000000 +0200
+++ gpsbabel-gpsbabel_1_6_0/gui/advdlg.cc	2019-06-19 13:08:12.030299296 +0200
@@ -37,7 +37,6 @@
 {
   ui_.setupUi(this);
   ui_.synthShortNames->setChecked(synthShortNames);
-  ui_.previewGmap->setChecked(previewGmap);
   ui_.debugCombo->setCurrentIndex(debugLevel+1);
 #if defined (Q_OS_WIN)
   ui_.buttonBox->button(QDialogButtonBox::Ok)->setIcon(QIcon(":images/ok"));
@@ -50,7 +49,6 @@
 void AdvDlg::acceptClicked()
 {
   synthShortNames_ = ui_.synthShortNames->isChecked();
-  previewGmap_ = ui_.previewGmap->isChecked();
   debugLevel_ = ui_.debugCombo->currentIndex()-1;
   accept();
 }
diff -Naur gpsbabel-gpsbabel_1_6_0.orig/gui/advui.ui gpsbabel-gpsbabel_1_6_0/gui/advui.ui
--- gpsbabel-gpsbabel_1_6_0.orig/gui/advui.ui	2019-05-02 10:26:47.000000000 +0200
+++ gpsbabel-gpsbabel_1_6_0/gui/advui.ui	2019-06-19 13:08:12.030299296 +0200
@@ -38,13 +38,6 @@
     </widget>
    </item>
    <item>
-    <widget class="QCheckBox" name="previewGmap">
-     <property name="text">
-      <string>Preview in Google Maps</string>
-     </property>
-    </widget>
-   </item>
-   <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QComboBox" name="debugCombo">
diff -Naur gpsbabel-gpsbabel_1_6_0.orig/gui/app.pro gpsbabel-gpsbabel_1_6_0/gui/app.pro
--- gpsbabel-gpsbabel_1_6_0.orig/gui/app.pro	2019-05-02 10:26:47.000000000 +0200
+++ gpsbabel-gpsbabel_1_6_0/gui/app.pro	2019-06-19 13:10:54.800233673 +0200
@@ -11,14 +11,8 @@
 QT += core \
       gui \
       network \
-      xml
-
-qtHaveModule(webenginewidgets) {
-  QT += webenginewidgets webchannel
-  DEFINES += HAVE_WEBENGINE
-} else {
-  QT += webkit webkitwidgets 
-}
+      xml \
+      widgets
 
 unix:DESTDIR = objects
 unix:MOC_DIR = objects
@@ -72,13 +66,11 @@
 SOURCES += filterwidgets.cc
 SOURCES += format.cc
 SOURCES += formatload.cc
-SOURCES += gmapdlg.cc
 SOURCES += gpx.cc
 SOURCES += help.cc
 SOURCES += latlng.cc
 SOURCES += main.cc
 SOURCES += mainwindow.cc
-SOURCES += map.cc
 SOURCES += optionsdlg.cc
 SOURCES += preferences.cc
 SOURCES += processwait.cc
@@ -98,11 +90,9 @@
 HEADERS += filterwidgets.h
 HEADERS += format.h
 HEADERS += formatload.h
-HEADERS += gmapdlg.h
 HEADERS += gpx.h
 HEADERS += help.h
 HEADERS += mainwindow.h
-HEADERS += map.h
 HEADERS += optionsdlg.h
 HEADERS += preferences.h
 HEADERS += processwait.h
diff -Naur gpsbabel-gpsbabel_1_6_0.orig/gui/mainwindow.cc gpsbabel-gpsbabel_1_6_0/gui/mainwindow.cc
--- gpsbabel-gpsbabel_1_6_0.orig/gui/mainwindow.cc	2019-05-02 10:26:47.000000000 +0200
+++ gpsbabel-gpsbabel_1_6_0/gui/mainwindow.cc	2019-06-19 13:12:30.790606265 +0200
@@ -62,7 +62,6 @@
 #include "donate.h"                    // for Donate
 #include "filterdlg.h"                 // for FilterDialog
 #include "formatload.h"                // for FormatLoad
-#include "gmapdlg.h"                   // for GMapDialog
 #include "help.h"                      // for ShowHelp
 #include "optionsdlg.h"                // for OptionsDlg
 #include "preferences.h"               // for Preferences
@@ -1027,24 +1026,6 @@
     formatList_[fidx].bumpWriteUseCount(1);
   }
 
-  // Now output for preview in google maps
-  QString tempName;
-  if (babelData_.previewGmap_) {
-    QTemporaryFile ftemp;
-    ftemp.open();
-    tempName = ftemp.fileName();
-    ftemp.close();
-
-    // Ideally, expost this in the UI.  For now, just split the track
-    // if we've no recorded fixes for > 5 mins and we've moved > 300 meters.
-    //args << "-x";
-    //args << "track,pack,sdistance=0.3k,split=5m";
-
-    args << "-o";
-    args << "gpx";
-    args << "-F" << tempName;
-  }
-
   ui_.outputWindow->clear();
   ui_.outputWindow->appendPlainText("gpsbabel " + args.join(" "));
 
@@ -1056,14 +1037,6 @@
   ui_.outputWindow->appendPlainText(outputString);
   if (x) {
     ui_.outputWindow->appendPlainText(tr("Translation successful"));
-    if (babelData_.previewGmap_) {
-      this->hide();
-      GMapDialog dlg(nullptr, tempName, babelData_.debugLevel_ >=1 ? ui_.outputWindow : nullptr);
-      dlg.show();
-      dlg.exec();
-      QFile(tempName).remove();
-      this->show();
-    }
   } else {
     ui_.outputWindow->appendPlainText(tr("Error running gpsbabel: %1\n").arg(errorString));
   }
