diff --git a/ebpf/src/tracepointdescriptor.cpp b/ebpf/src/tracepointdescriptor.cpp
index 29c3219..2758cc5 100644
--- a/ebpf/src/tracepointdescriptor.cpp
+++ b/ebpf/src/tracepointdescriptor.cpp
@@ -7,6 +7,7 @@
 */
 
 #include <algorithm>
+#include <array>
 #include <cassert>
 #include <fstream>
 #include <iostream>
diff --git a/error/include/tob/error/erroror.h b/error/include/tob/error/erroror.h
index 1d5a13d..f627a09 100644
--- a/error/include/tob/error/erroror.h
+++ b/error/include/tob/error/erroror.h
@@ -10,6 +10,7 @@
 
 #include <stdexcept>
 #include <variant>
+#include <utility>
 
 namespace tob {
 template <typename ValueType, typename ErrorType> class ErrorOr final {
diff --git a/error/include/tob/error/successor.h b/error/include/tob/error/successor.h
index 6d8260e..4f19845 100644
--- a/error/include/tob/error/successor.h
+++ b/error/include/tob/error/successor.h
@@ -10,6 +10,7 @@
 
 #include <optional>
 #include <stdexcept>
+#include <utility>
 
 namespace tob {
 template <typename ErrorType> class SuccessOr final {
