diff -ru ext3grep-0.10.2.orig/src/ext3.h ext3grep-0.10.2/src/ext3.h
--- ext3grep-0.10.2.orig/src/ext3.h	2018-06-04 22:13:06.563391984 +0000
+++ ext3grep-0.10.2/src/ext3.h	2018-06-04 22:13:44.486724597 +0000
@@ -117,7 +117,7 @@
     __u32 const* block(void) const { return i_block; }
     __u32 generation(void) const { return i_generation; }
     __u32 file_acl(void) const { return i_file_acl; }
-    __u32 dir_acl(void) const { return i_dir_acl; }
+    __u32 dir_acl(void) const { return i_size_high; }
     __u32 faddr(void) const { return i_faddr; }
     __u16 uid_high(void) const { return i_uid_high; }
     __u16 gid_high(void) const { return i_gid_high; }
