diff -ru apitrace-5.0/cli/cli_diff.cpp apitrace-5.0a/cli/cli_diff.cpp
--- apitrace-5.0/cli/cli_diff.cpp	2014-05-03 00:22:02.000000000 +0200
+++ apitrace-5.0a/cli/cli_diff.cpp	2014-08-03 02:50:33.635426106 +0200
@@ -50,7 +50,7 @@
     }
 
     char *args[4];
-    args[0] = (char *) "python";
+    args[0] = (char *) APITRACE_PYTHON_EXECUTABLE;
     args[1] = (char *) command.str();
     args[2] = (char *) "--help";
     args[3] = NULL;
@@ -71,7 +71,7 @@
     os::String apitracePath = os::getProcessName();
 
     std::vector<const char *> args;
-    args.push_back("python");
+    args.push_back(APITRACE_PYTHON_EXECUTABLE);
     args.push_back(command.str());
     args.push_back("--apitrace");
     args.push_back(apitracePath.str());
diff -ru apitrace-5.0/cli/cli_diff_images.cpp apitrace-5.0a/cli/cli_diff_images.cpp
--- apitrace-5.0/cli/cli_diff_images.cpp	2014-05-03 00:22:02.000000000 +0200
+++ apitrace-5.0a/cli/cli_diff_images.cpp	2014-08-03 02:50:02.439320408 +0200
@@ -50,7 +50,7 @@
     }
 
     char *args[4];
-    args[0] = (char *) "python";
+    args[0] = (char *) APITRACE_PYTHON_EXECUTABLE;
     args[1] = (char *) command.str();
     args[2] = (char *) "--help";
     args[3] = NULL;
@@ -69,7 +69,7 @@
     }
 
     std::vector<const char *> args;
-    args.push_back("python");
+    args.push_back(APITRACE_PYTHON_EXECUTABLE);
     args.push_back(command.str());
     for (i = 1; i < argc; i++) {
         args.push_back(argv[i]);
diff -ru apitrace-5.0/cli/cli_diff_state.cpp apitrace-5.0a/cli/cli_diff_state.cpp
--- apitrace-5.0/cli/cli_diff_state.cpp	2014-05-03 00:22:02.000000000 +0200
+++ apitrace-5.0a/cli/cli_diff_state.cpp	2014-08-03 02:49:04.987020260 +0200
@@ -87,7 +87,7 @@
 
     char *args[5];
 
-    args[0] = const_cast<char *>("python");
+    args[0] = const_cast<char *>(APITRACE_PYTHON_EXECUTABLE);
     args[1] = const_cast<char *>(command.str());
     args[2] = file1;
     args[3] = file2;
diff -ru apitrace-5.0/cli/CMakeLists.txt apitrace-5.0a/cli/CMakeLists.txt
--- apitrace-5.0/cli/CMakeLists.txt	2014-05-03 00:22:02.000000000 +0200
+++ apitrace-5.0a/cli/CMakeLists.txt	2014-08-03 02:46:05.803576581 +0200
@@ -7,6 +7,7 @@
     -DAPITRACE_PROGRAMS_INSTALL_DIR="${CMAKE_INSTALL_PREFIX}/bin"
     -DAPITRACE_SCRIPTS_INSTALL_DIR="${CMAKE_INSTALL_PREFIX}/${SCRIPTS_INSTALL_DIR}"
     -DAPITRACE_WRAPPERS_INSTALL_DIR="${CMAKE_INSTALL_PREFIX}/${WRAPPER_INSTALL_DIR}"
+    -DAPITRACE_PYTHON_EXECUTABLE="${PYTHON_EXECUTABLE}"
 )
 
 add_executable (apitrace
