diff --git a/dde-dock-plugins/disk-mount/disk-mount.pro b/dde-dock-plugins/disk-mount/disk-mount.pro
index c5d336e8..c9575b15 100644
--- a/dde-dock-plugins/disk-mount/disk-mount.pro
+++ b/dde-dock-plugins/disk-mount/disk-mount.pro
@@ -15,7 +15,7 @@ DISTFILES       += disk-mount.json
 
 DEFINES += QT_MESSAGELOGCONTEXT DFM_NO_FILE_WATCHER
 
-#unix: LIBS += -L$$OUT_PWD/../../dde-file-manager-lib -ldde-file-manager
+LIBS += -L$$OUT_PWD/../../dde-file-manager-lib -ldde-file-manager
 
 HEADERS += \
     diskmountplugin.h \
