--- a/bitmaps_png/CMakeLists.txt
+++ b/bitmaps_png/CMakeLists.txt
@@ -782,7 +782,7 @@ else()
 endif()
 
 add_custom_target( bitmap_archive ALL
-    COMMAND ${TAR} cfz "${TAR_OUTPUT_FILE}" .
+    COMMAND ${TAR} --create --file "${TAR_OUTPUT_FILE}" --gzip --numeric-owner --owner=0 --group=0 .
     DEPENDS ${ALL_PNGS}
     BYPRODUCTS ${BITMAP_ARCHIVE_PATH}
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/png
