From 6a71bf81b1b663387c0f0ca9e41ab24601c678c5 Mon Sep 17 00:00:00 2001
From: Ed Hartnett <edwardjameshartnett@gmail.com>
Date: Mon, 23 Oct 2017 09:49:38 -0600
Subject: [PATCH] fixed memory problem in tst_h_dimscales

---
 h5_test/tst_h_dimscales.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/h5_test/tst_h_dimscales.c b/h5_test/tst_h_dimscales.c
index 73c817d58..349b1a836 100644
--- a/h5_test/tst_h_dimscales.c
+++ b/h5_test/tst_h_dimscales.c
@@ -85,7 +85,7 @@ rec_scan_group(hid_t grpid)
 	    }
 	    else
 	    {
-	       int visitor_data = 0;
+		hid_t visitor_data = 0;
 
 	       /* Here's how to get the number of scales attached
 		* to the dataset's dimension 0. */
@@ -379,7 +379,7 @@ main()
       if ((grpid = H5Gopen(fileid, GRP_NAME)) < 0) ERR;
 
       /* Loop through datasets to find variables. */
-      if (H5Gget_num_objs(grpid, &num_obj) < 0) ERR;
+      if (H5Gget_num_objs(grpid, (hsize_t *)&num_obj) < 0) ERR;
       for (i=0; i<num_obj; i++)
       {
 	 /* Get the type (i.e. group, dataset, etc.), and the name of
@@ -426,7 +426,7 @@ main()
 	       else
 	       {
 		  char label[STR_LEN+1];
-		  int visitor_data = 0;
+		  hid_t visitor_data = 0;
 
 		  /* Here's how to get the number of scales attached
 		   * to the dataset's dimension 0. */
@@ -617,7 +617,7 @@ main()
 	       else
 	       {
 		  char label[STR_LEN+1];
-		  int visitor_data = 0;
+		  hid_t visitor_data = 0;
 
 		  /* SHould have these dimensions... */
 		  if (dims[TIME_DIM] != 0 || dims[LAT_DIM] != LAT_LEN ||
