diff -aur LHAPDF-6.2.3-old/bin/lhapdf LHAPDF-6.2.3-new/bin/lhapdf
--- LHAPDF-6.2.3-old/bin/lhapdf	2019-05-11 23:06:38.000000000 +0300
+++ LHAPDF-6.2.3-new/bin/lhapdf	2019-05-27 13:38:55.737898779 +0300
@@ -287,7 +287,7 @@
         (by default lists all sets available for download; use --installed or --outdated to explore those installed on the current system)"""
         ap = argparse.ArgumentParser(description=__doc__, usage="%(prog)s list [options] [pattern...]")
         ap.add_argument("PATTERNS", nargs="*", help="patterns to match PDF sets against")
-        ag = ap.add_add_mutually_exclusive_group()
+        ag = ap.add_mutually_exclusive_group()
         ag.add_argument("--installed", dest="INSTALLED", action="store_true", help="list installed PDF sets")
         ag.add_argument("--outdated", dest="OUTDATED", action="store_true", help="list installed, but outdated, PDF sets")
         ap.add_argument("--codes", dest="CODES", action="store_true", help="additionally show ID codes")
