diff -upr gambas-3.15.2.orig/gb.poppler/configure.ac gambas-3.15.2/gb.poppler/configure.ac
--- gambas-3.15.2.orig/gb.poppler/configure.ac	2020-09-17 14:12:42.000000000 +0300
+++ gambas-3.15.2/gb.poppler/configure.ac	2020-12-02 10:35:34.428168215 +0200
@@ -9,6 +9,6 @@ GB_COMPONENT_PKG_CONFIG(
   POPPLER,
   gb.poppler,
   [src],
-  'poppler >= 0.20' poppler-splash poppler-glib)
+  'poppler >= 0.20' poppler-glib)
 AC_OUTPUT( Makefile src/Makefile )
 GB_PRINT_MESSAGES
