diff -ru gap4r8.old/pkg/PolymakeInterface/src/polymake_cone.cpp gap4r8/pkg/PolymakeInterface/src/polymake_cone.cpp
--- gap4r8.old/pkg/PolymakeInterface/src/polymake_cone.cpp	2016-11-12 15:25:17.000000000 +0100
+++ gap4r8/pkg/PolymakeInterface/src/polymake_cone.cpp	2017-03-29 13:37:29.000000000 +0200
@@ -321,7 +321,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr.cols());
     SET_LEN_PLIST( LIZeil , matr_cols );
     for(int j = 0;j<matr.cols();j++){
-      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (matr(i,j)).to_int() ));
+      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (matr(i,j)) ));
     }
     SET_ELM_PLIST(RETLI,i+1,LIZeil);
     CHANGED_BAG(RETLI);
@@ -333,7 +333,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr2.cols());
     SET_LEN_PLIST( LIZeil ,matr2_cols );
     for(int j = 0;j<matr2.cols();j++){
-      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (matr2(i,j)).to_int() ));
+      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (matr2(i,j)) ));
     }
     SET_ELM_PLIST(RETLI,matr.rows() + i +1,LIZeil);
     CHANGED_BAG(RETLI);
@@ -343,7 +343,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr2.cols());
     SET_LEN_PLIST( LIZeil ,matr2_cols);
     for(int j = 0;j<matr2.cols();j++){
-      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (-matr2(i,j)).to_int() ));
+      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (-matr2(i,j)) ));
     }
     SET_ELM_PLIST(RETLI,matr.rows() + matr2.rows() + i +1,LIZeil);
     CHANGED_BAG(RETLI);
@@ -384,7 +384,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr.cols());
     SET_LEN_PLIST( LIZeil , matr_cols );
     for(int j = 0;j<matr.cols();j++){
-      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (matr(i,j)).to_int() ));
+      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (matr(i,j)) ));
     }
     SET_ELM_PLIST(RETLI,i+1,LIZeil);
     CHANGED_BAG(RETLI);
@@ -422,7 +422,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr.cols());
     SET_LEN_PLIST( LIZeil ,matr_cols);
     for(int j = 0;j<matr.cols();j++){
-      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT((matr(i,j)).to_int()));
+      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT((matr(i,j))));
     }
     SET_ELM_PLIST(RETLI,i+1,LIZeil);
   }
@@ -500,7 +500,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr.cols());
     SET_LEN_PLIST( LIZeil , matr_cols );
     for(int j = 0;j<matr.cols();j++){
-      SET_ELM_PLIST(LIZeil, j+1, INTOBJ_INT( matr(i,j).to_int() ) );
+      SET_ELM_PLIST(LIZeil, j+1, INTOBJ_INT( matr(i,j) ) );
       CHANGED_BAG( LIZeil );
     }
     SET_ELM_PLIST(RETLI,i+1,LIZeil);
@@ -517,9 +517,9 @@
     SET_LEN_PLIST( LIZeil , matr2_cols );
     SET_LEN_PLIST( LIZeil2, matr2_cols );
     for(int j = 0;j<matr2.cols();j++){
-      SET_ELM_PLIST( LIZeil,  j+1, INTOBJ_INT( matr2(i,j).to_int() ));
+      SET_ELM_PLIST( LIZeil,  j+1, INTOBJ_INT( matr2(i,j) ));
       CHANGED_BAG( LIZeil );
-      SET_ELM_PLIST( LIZeil2, j+1, INTOBJ_INT( -matr2(i,j).to_int() ) );
+      SET_ELM_PLIST( LIZeil2, j+1, INTOBJ_INT( -matr2(i,j) ) );
       CHANGED_BAG( LIZeil2 );
       
     }
@@ -581,7 +581,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr.cols());
     SET_LEN_PLIST( LIZeil , matr_cols );
     for(int j = 0;j<matr.cols();j++){
-      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (matr(i,j)).to_int() ));
+      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (matr(i,j)) ));
     }
     SET_ELM_PLIST(RETLI,i+1,LIZeil);
     CHANGED_BAG(RETLI);
diff -ru gap4r8.old/pkg/PolymakeInterface/src/polymake_fan.cpp gap4r8/pkg/PolymakeInterface/src/polymake_fan.cpp
--- gap4r8.old/pkg/PolymakeInterface/src/polymake_fan.cpp	2016-11-12 15:25:17.000000000 +0100
+++ gap4r8/pkg/PolymakeInterface/src/polymake_fan.cpp	2017-03-29 13:37:58.000000000 +0200
@@ -407,7 +407,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr.cols());
     SET_LEN_PLIST( LIZeil , matr_cols );
     for(int j = 0;j<matr.cols();j++){
-      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (matr(i,j)).to_int() ));
+      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT( (matr(i,j)) ));
     }
     SET_ELM_PLIST(RETLI,i+1,LIZeil);
     CHANGED_BAG(RETLI);
@@ -439,8 +439,8 @@
   Obj RETLI = NEW_PLIST( T_PLIST , matr.size() );
   SET_LEN_PLIST( RETLI , matr_rows );
   for(int i = 0;i<matr.size(); i++){
-      SET_ELM_PLIST(RETLI,i+1,INTOBJ_INT( (matr[i]).to_int() ));
+      SET_ELM_PLIST(RETLI,i+1,INTOBJ_INT( (matr[i]) ));
       CHANGED_BAG(RETLI);
   }
   return RETLI;
-}
\ No hay ningún carácter de nueva línea al final del fichero
+}
diff -ru gap4r8.old/pkg/PolymakeInterface/src/polymake_fktn.cpp gap4r8/pkg/PolymakeInterface/src/polymake_fktn.cpp
--- gap4r8.old/pkg/PolymakeInterface/src/polymake_fktn.cpp	2016-11-12 15:25:17.000000000 +0100
+++ gap4r8/pkg/PolymakeInterface/src/polymake_fktn.cpp	2017-03-29 13:15:26.000000000 +0200
@@ -120,7 +120,7 @@
 }
 
 
-Obj REAL_OBJECT_HAS_PROPERTY( Polymake_Data* data, Obj cone, const char* prop ){
+Obj REAL_OBJECT_HAS_PROPERTY( Polymake_Data* data, Obj cone, polymake::AnyString& prop ){
 
 #ifdef MORE_TESTS
   if(! IS_POLYMAKE_OBJECT(cone) ){
@@ -143,7 +143,7 @@
 
 }
 
-Obj REAL_OBJECT_HAS_INT_PROPERTY( Polymake_Data* data, Obj cone, const char* prop ){
+Obj REAL_OBJECT_HAS_INT_PROPERTY( Polymake_Data* data, Obj cone, polymake::AnyString& prop ){
 
 #ifdef MORE_TESTS
   if(! IS_POLYMAKE_OBJECT(cone) ){
@@ -188,7 +188,7 @@
 }
 
 
-void REAL_SET_PROPERTY_TRUE( Polymake_Data* data, Obj conv, const char* prop){
+void REAL_SET_PROPERTY_TRUE( Polymake_Data* data, Obj conv, polymake::AnyString& prop){
 
 #ifdef MORE_TESTS
   if(! IS_POLYMAKE_OBJECT(conv) ){
@@ -252,8 +252,10 @@
           return NULL;
       }
       Obj content = ELM_PLIST( current_option, 2 );
+      std::string d(CSTR_STRING(description));
       if( IS_STRING( content ) ){
-        visual_options[ CSTR_STRING( description ) ] << CSTR_STRING( content );
+        
+        visual_options[ d ] << CSTR_STRING( content );
       }else{
         if( IS_PLIST( content ) ){
           pm::Integer* cont = new pm::Integer[ LEN_PLIST( content ) ];
@@ -265,7 +267,7 @@
             }
             cont[ j - 1 ] = INT_INTOBJ( elem );
           }
-          visual_options[ CSTR_STRING( description ) ] << cont;
+          visual_options[ d ] << cont;
         }
       }
     }
diff -ru gap4r8.old/pkg/PolymakeInterface/src/polymake_polytopes.cpp gap4r8/pkg/PolymakeInterface/src/polymake_polytopes.cpp
--- gap4r8.old/pkg/PolymakeInterface/src/polymake_polytopes.cpp	2016-11-12 15:25:17.000000000 +0100
+++ gap4r8/pkg/PolymakeInterface/src/polymake_polytopes.cpp	2017-03-29 13:37:00.000000000 +0200
@@ -105,7 +105,7 @@
       LIZeil = NEW_PLIST( T_PLIST, matr.cols()-1);
       SET_LEN_PLIST( LIZeil , matr_cols );
       for(int j = 1;j<matr.cols();j++){
-        SET_ELM_PLIST(LIZeil,j,INTOBJ_INT((matr(i,j)).to_int()));
+        SET_ELM_PLIST(LIZeil,j,INTOBJ_INT((matr(i,j))));
       }
       SET_ELM_PLIST(RETLI,k,LIZeil);
       CHANGED_BAG(RETLI);
@@ -147,7 +147,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr.cols()-1);
     SET_LEN_PLIST( LIZeil , matr_cols );
     for(int j = 1;j<matr.cols();j++){
-      SET_ELM_PLIST(LIZeil,j,INTOBJ_INT((matr(i,j)).to_int()));
+      SET_ELM_PLIST(LIZeil,j,INTOBJ_INT((matr(i,j))));
     }
     SET_ELM_PLIST(RETLI,i+1,LIZeil);
     CHANGED_BAG(RETLI);
@@ -252,7 +252,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr.cols() );
     SET_LEN_PLIST( LIZeil , matr_cols );
     for(int j = 0;j<matr.cols();j++){
-      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT((matr(i,j)).to_int()));
+      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT((matr(i,j))));
     }
     SET_ELM_PLIST(RETLI,i+1,LIZeil);
     CHANGED_BAG(RETLI);
@@ -290,7 +290,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr.cols() );
     SET_LEN_PLIST( LIZeil , matr_cols );
     for(int j = 0;j<matr.cols();j++){
-      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT((matr(i,j)).to_int()));
+      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT((matr(i,j))));
     }
     SET_ELM_PLIST(RETLI,i+1,LIZeil);
     CHANGED_BAG(RETLI);
@@ -328,7 +328,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr.cols()-1);
     SET_LEN_PLIST( LIZeil , matr_cols );
     for(int j = 1;j<matr.cols();j++){
-      SET_ELM_PLIST(LIZeil,j,INTOBJ_INT((matr(i,j)).to_int()));
+      SET_ELM_PLIST(LIZeil,j,INTOBJ_INT((matr(i,j))));
     }
     SET_ELM_PLIST(RETLI,i+1,LIZeil);
     CHANGED_BAG(RETLI);
@@ -434,7 +434,7 @@
     LIZeil = NEW_PLIST( T_PLIST, matr.cols());
     SET_LEN_PLIST( LIZeil , matr_cols );
     for(int j = 0;j<matr.cols();j++){
-      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT((matr(i,j)).to_int()));
+      SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT((matr(i,j))));
     }
     SET_ELM_PLIST(RETLI,i+1,LIZeil);
     CHANGED_BAG(RETLI);
@@ -479,7 +479,7 @@
       LIZeil = NEW_PLIST( T_PLIST, matr.cols()-1);
       SET_LEN_PLIST( LIZeil , matr_cols );
       for(int j = 1;j<matr.cols();j++){
-        SET_ELM_PLIST(LIZeil,j, INTOBJ_INT( (matr(i,j)).to_int() ) );
+        SET_ELM_PLIST(LIZeil,j, INTOBJ_INT( (matr(i,j)) ) );
       }
       SET_ELM_PLIST(RETLI,k,LIZeil);
       CHANGED_BAG(RETLI);
@@ -591,7 +591,7 @@
         LIZeil = NEW_PLIST( T_PLIST, matr.cols()-1);
         SET_LEN_PLIST( LIZeil , matr_cols );
         for(int j = 1;j<matr.cols();j++){
-          SET_ELM_PLIST(LIZeil,j,INTOBJ_INT((matr(i,j)).to_int()));
+          SET_ELM_PLIST(LIZeil,j,INTOBJ_INT((matr(i,j))));
         }
         SET_ELM_PLIST(RETLI,i+1,LIZeil);
         CHANGED_BAG(RETLI);
@@ -630,4 +630,4 @@
   
   return elem;
   
-}
\ No hay ningún carácter de nueva línea al final del fichero
+}
diff -ru gap4r8.old/pkg/PolymakeInterface/src/polymake_tropical.cpp gap4r8/pkg/PolymakeInterface/src/polymake_tropical.cpp
--- gap4r8.old/pkg/PolymakeInterface/src/polymake_tropical.cpp	2016-11-12 15:25:17.000000000 +0100
+++ gap4r8/pkg/PolymakeInterface/src/polymake_tropical.cpp	2017-03-29 13:38:24.000000000 +0200
@@ -134,7 +134,7 @@
       LIZeil = NEW_PLIST( T_PLIST, matr.cols());
       SET_LEN_PLIST( LIZeil , matr_cols );
       for(int j = 0;j<matr.cols();j++){
-        SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT((matr(i,j)).to_int()));
+        SET_ELM_PLIST(LIZeil,j+1,INTOBJ_INT((matr(i,j))));
       }
       SET_ELM_PLIST(RETLI,k,LIZeil);
       CHANGED_BAG(RETLI);
