diff --git a/frame/frame.pro b/frame/frame.pro
index bdc5f2d..953bea1 100644
--- a/frame/frame.pro
+++ b/frame/frame.pro
@@ -8,7 +8,7 @@ DESTDIR         = $$_PRO_FILE_PWD_/../
 TEMPLATE        = app
 CONFIG         += c++11 link_pkgconfig
 
-PKGCONFIG += xcb-ewmh gtk+-2.0 dtkwidget dtkbase dtkutil
+PKGCONFIG += xcb-ewmh gtk+-2.0 dtkwidget dtkbase dtkutil gio-unix-2.0
 
 SOURCES += main.cpp \
     window/mainwindow.cpp \
