diff -upr scribus-1.5.7.orig/scribus/fonts/sfnt.cpp scribus-1.5.7/scribus/fonts/sfnt.cpp
--- scribus-1.5.7.orig/scribus/fonts/sfnt.cpp	2021-09-18 23:22:51.537708802 +0300
+++ scribus-1.5.7/scribus/fonts/sfnt.cpp	2021-09-18 23:23:53.602675236 +0300
@@ -1271,7 +1271,7 @@ namespace sfnt {
 #endif
 #endif
 
-		QScopedPointer<hb_face_t, HbFaceDeleter> hbSubsetFace(hb_subset(hbFullFace.get(), hbSubsetInput.get()));
+		QScopedPointer<hb_face_t, HbFaceDeleter> hbSubsetFace(hb_subset_or_fail(hbFullFace.get(), hbSubsetInput.get()));
 		if (hbSubsetFace.isNull())
 			return QByteArray();
 
