diff -rupN intellij-community-pycharm-182.1177.orig/build/dependencies/setupJbre.gradle intellij-community-pycharm-182.1177/build/dependencies/setupJbre.gradle
--- intellij-community-pycharm-182.1177.orig/build/dependencies/setupJbre.gradle	2018-03-29 19:26:20.265970726 +0200
+++ intellij-community-pycharm-182.1177/build/dependencies/setupJbre.gradle	2018-03-29 19:26:52.690160085 +0200
@@ -1,7 +1,7 @@
 task setupJbre {}
 
 def jdkRepo = inJetBrainsNetwork ? 'http://repo.labs.intellij.net/intellij-jdk' : 'https://dl.bintray.com/jetbrains/intellij-jdk'
-def targetOs = "${System.getProperty("intellij.build.target.os", "all")}"
+def targetOs = "${System.getProperty("intellij.build.target.os", "linux")}"
 
 jrePlatformsToDownload(targetOs).each { platform ->
   archToDownload(platform).each { arch ->
diff -rupN intellij-community-pycharm-182.1177.orig/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy intellij-community-pycharm-182.1177/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy
--- intellij-community-pycharm-182.1177.orig/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy	2018-03-29 19:26:25.415841858 +0200
+++ intellij-community-pycharm-182.1177/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy	2018-03-29 19:31:46.883868914 +0200
@@ -33,7 +33,7 @@ class BuildOptions {
   /**
    * Specifies for which operating systems distributions should be built.
    */
-  String targetOS = System.getProperty("intellij.build.target.os", OS_ALL)
+  String targetOS = System.getProperty("intellij.build.target.os", OS_LINUX)
   static final String OS_LINUX = "linux"
   static final String OS_WINDOWS = "windows"
   static final String OS_MAC = "mac"
@@ -100,7 +100,7 @@ class BuildOptions {
    * Build number without product code (e.g. '162.500.10'), if {@code null} '&lt;baseline&gt;.SNAPSHOT' will be used. Use {@link BuildContext#buildNumber} to
    * get the actual build number in build scripts.
    */
-  String buildNumber = System.getProperty("build.number")
+  String buildNumber = System.getProperty("idea.build.number")
 
   /**
    * By default build process produces temporary and resulting files under projectHome/out/productName directory, use this property to
@@ -115,4 +115,4 @@ class BuildOptions {
    */
   boolean isInDevelopmentMode = SystemProperties.getBooleanProperty("intellij.build.dev.mode",
                                                                     System.getProperty("teamcity.buildType.id") == null)
-}
\ No newline at end of file
+}
