diff --git a/include/spdlog/details/circular_q.h b/include/spdlog/details/circular_q.h
index fee3a3bf..d147c5e0 100644
--- a/include/spdlog/details/circular_q.h
+++ b/include/spdlog/details/circular_q.h
@@ -5,6 +5,7 @@
 #pragma once
 
 #include <vector>
+#include <cassert>
 
 namespace spdlog {
 namespace details {
diff --git a/include/spdlog/details/thread_pool-inl.h b/include/spdlog/details/thread_pool-inl.h
index 77d0b1fb..6b740ca2 100644
--- a/include/spdlog/details/thread_pool-inl.h
+++ b/include/spdlog/details/thread_pool-inl.h
@@ -8,6 +8,7 @@
 #endif
 
 #include "spdlog/common.h"
+#include <cassert>
 
 namespace spdlog {
 namespace details {
