diff -u -r mypaint-1.2.1/brushlib/mypaint-brush.c mypaint-1.2.1-json-c/brushlib/mypaint-brush.c
--- mypaint-1.2.1/brushlib/mypaint-brush.c	2017-01-21 17:51:52.000000000 +0000
+++ mypaint-1.2.1-json-c/brushlib/mypaint-brush.c	2020-04-26 00:23:04.098737838 +0000
@@ -28,9 +28,6 @@
 #include "rng-double.h"
 
 #ifdef HAVE_JSON_C
-// Allow the C99 define from json.h
-#undef TRUE
-#undef FALSE
 #include <json.h>
 #endif // HAVE_JSON_C
 
