--- julia-0.6.0/src/julia_internal.h.0	2017-06-25 16:10:29.688382943 +0000
+++ julia-0.6.0/src/julia_internal.h	2017-06-25 16:11:15.328315342 +0000
@@ -643,11 +643,8 @@
 typedef unw_context_t bt_context_t;
 typedef unw_cursor_t bt_cursor_t;
-#  if (!defined(SYSTEM_LIBUNWIND) || UNW_VERSION_MAJOR > 1 ||   \
-       (UNW_VERSION_MAJOR == 1 && UNW_VERSION_MINOR > 1))
 // Enable our memory manager only for libunwind with our patch or
 // on a newer release
 #    define JL_UNW_HAS_FORMAT_IP 1
-#  endif
 #endif
 size_t rec_backtrace(uintptr_t *data, size_t maxsize);
 size_t rec_backtrace_ctx(uintptr_t *data, size_t maxsize, bt_context_t *ctx);

