diff --git a/deps/checksums/libgit2 b/deps/checksums/libgit2
index 6d836e4a4501c..9b360b711ceb5 100644
--- a/deps/checksums/libgit2
+++ b/deps/checksums/libgit2
@@ -1,34 +1,34 @@
-libgit2-7f4fa178629d559c037a1f72f79f79af9c1ef8ce.tar.gz/md5/c6a819fb0bf924df61e1595624a0988a
-libgit2-7f4fa178629d559c037a1f72f79f79af9c1ef8ce.tar.gz/sha512/3de9c042115b309dae3b8e0008edf2e762addd90a7bdb54b3cf634811271ab9dbfea35656650eb53a3faec73caf33ed199fb885ec21f611c79d909d9f4fe48c5
-LibGit2.v1.2.3+0.aarch64-apple-darwin.tar.gz/md5/0eb4d35fa6078de8da3dc79a420d5bc5
-LibGit2.v1.2.3+0.aarch64-apple-darwin.tar.gz/sha512/aff23e9d56827be54d8f515489eed92d9cb4d9a1b69c12bbee44fa1343ebc906b38b082c11e6ecb227d256de5cba19b2f3a03c022b88809a33c225c34db08f7c
-LibGit2.v1.2.3+0.aarch64-linux-gnu.tar.gz/md5/f2f31aa978e43aa68a0e73107067c7cd
-LibGit2.v1.2.3+0.aarch64-linux-gnu.tar.gz/sha512/46f1ed9158cc00fc0ca71768e9a475d99597e4e63d806b061f08c2b5de40ecde61020f60ad4ff5a7377b52dfcd910c6610022fd95c1a08aa9af305489712786f
-LibGit2.v1.2.3+0.aarch64-linux-musl.tar.gz/md5/aa1429e946185092bacda87fb5ff4ed2
-LibGit2.v1.2.3+0.aarch64-linux-musl.tar.gz/sha512/2ab81db22c4ca9855a84835174cee249bc95c3a3158b72fe0c8d18da074ef7a7790bbd90ffb4ec5847ff23ef6098e7d9410ab765c27bdba8bafed5ab3820452a
-LibGit2.v1.2.3+0.armv6l-linux-gnueabihf.tar.gz/md5/07bef4e0e69f1b1cfe870f554262bd3c
-LibGit2.v1.2.3+0.armv6l-linux-gnueabihf.tar.gz/sha512/ccd424e5722e3319aa6dea25e088b7ab4ef30afbab57ba243c92db2cd9906d2b6110b866d4aee61bcd8a29e75fc6399b59109a1457d0fe4dd0e1d2c77aef3808
-LibGit2.v1.2.3+0.armv6l-linux-musleabihf.tar.gz/md5/cb56cabe9e705e8ca81ea8f890498bc5
-LibGit2.v1.2.3+0.armv6l-linux-musleabihf.tar.gz/sha512/45823b4f8fbd472fb508c0186baddb480f02356d193e493a541b064c92a919bbfad83df7ffa8b1631446c099e29cb109f4e1519246f9511ab81973ada9e2e22b
-LibGit2.v1.2.3+0.armv7l-linux-gnueabihf.tar.gz/md5/ea80b5ca2fa469f5056d3716d173e15d
-LibGit2.v1.2.3+0.armv7l-linux-gnueabihf.tar.gz/sha512/9040d8841dfcc26af42b409f776ce19bc232b7bf9679cd291c802d8e8186dd52d55bf085ca5bafdf1090a224c6fdf5bb42722434a8d771809d0bb083a768df90
-LibGit2.v1.2.3+0.armv7l-linux-musleabihf.tar.gz/md5/be089f9e5986deca632f170b137879e3
-LibGit2.v1.2.3+0.armv7l-linux-musleabihf.tar.gz/sha512/3266a3d40278324f4def5ff0823fce4859bc4eb065919500265ff962d01c7ebbcdfb8ba43b09d00a05ea2bb53a82984166442dfbc33ed6870a48b8af6cd26feb
-LibGit2.v1.2.3+0.i686-linux-gnu.tar.gz/md5/27b9d4e956ce649d0c3c412e65662e4d
-LibGit2.v1.2.3+0.i686-linux-gnu.tar.gz/sha512/85160f6950affbc8e161bf054ecdcf5cc09ecfac410765dae5f7ba35b3f2f5eb1a94f5b0299c1500ea6972be8dc2b9f0802a2f587c74aac4236c2c9c325fd573
-LibGit2.v1.2.3+0.i686-linux-musl.tar.gz/md5/5f8f66ace7a0d564d36a93f6c60f2e9a
-LibGit2.v1.2.3+0.i686-linux-musl.tar.gz/sha512/0438682a560ebdd13db1786748e5f08af327a6b9b9b8a4c70606696db4d311b45129ea67b960e14a4db948cbd741039afc3f4e9284adb209de168265b146d8eb
-LibGit2.v1.2.3+0.i686-w64-mingw32.tar.gz/md5/b36dd8bffdc106c2f9cc6c8a694eed01
-LibGit2.v1.2.3+0.i686-w64-mingw32.tar.gz/sha512/72d9230281b51f9530c3a33a139633b0e7199de3e001e1c265ac7d393ba5c6042b8fdc822b8a3bf661e113166459b2d61b8251dfdb68231bc3b7a3f10efc3bb6
-LibGit2.v1.2.3+0.powerpc64le-linux-gnu.tar.gz/md5/6060a6bf226eae115b1af93937523104
-LibGit2.v1.2.3+0.powerpc64le-linux-gnu.tar.gz/sha512/424e6b174c1cf19602fe0c2f01c432bc31e810a533093d990ef5e3bac46e23ff509437151f2932cf21b3bdd626c877689049fbe8891991256d8672dcc3339efc
-LibGit2.v1.2.3+0.x86_64-apple-darwin.tar.gz/md5/10431767db6173db70cfd34a14223f3a
-LibGit2.v1.2.3+0.x86_64-apple-darwin.tar.gz/sha512/8217b832ec98986465d89d3262cb2a9e409595978e39a5e21cc2c808576354515398a597c608258e74c7dc80cebb2563134ee6f903fd2ce371b4fdb953b07323
-LibGit2.v1.2.3+0.x86_64-linux-gnu.tar.gz/md5/2e1d21253669ef438840662d661c0b4f
-LibGit2.v1.2.3+0.x86_64-linux-gnu.tar.gz/sha512/c739d47165dff39800d4f3915c31d06dbe430f26730e00e04314fe813c4170d253f74cda6d8707c9240f3f0774b7e2ee9986c553a2892074756698b9eea6b7f1
-LibGit2.v1.2.3+0.x86_64-linux-musl.tar.gz/md5/cf61d8502d04e9c561da341182c94aae
-LibGit2.v1.2.3+0.x86_64-linux-musl.tar.gz/sha512/74aa67ab5a43fafcf6cb0a6389031e41d996f63b2a9b1fd6b0ff72159a10d199216fe1c4b882c6e7e052178db7d5213d7073ffc14ec648af19bea5eb55733e15
-LibGit2.v1.2.3+0.x86_64-unknown-freebsd.tar.gz/md5/e1623fd3f8f564085d47ec650a40e724
-LibGit2.v1.2.3+0.x86_64-unknown-freebsd.tar.gz/sha512/295d55b78b21ef1c2ba471c8b5618b168dd633e986db9e1ec3e9630e352446ab18e8fd0992010b6afdd922463bb285bc45885a8b35a502d574553fe61c1f7b9f
-LibGit2.v1.2.3+0.x86_64-w64-mingw32.tar.gz/md5/c845901c4d9dc145f76469d45abad934
-LibGit2.v1.2.3+0.x86_64-w64-mingw32.tar.gz/sha512/21951f3bc902f30b8cc75c3af233aa7fe8457e412e7758d556bf71de149c7f2325a5c4c204a7a462cc6a61b3dcb90f0d25e684ffd8617b0a1505a1d31cf2f69a
+LibGit2.v1.3.0+0.aarch64-apple-darwin.tar.gz/md5/af98f6fafe5678873b7f974c481c4238
+LibGit2.v1.3.0+0.aarch64-apple-darwin.tar.gz/sha512/1c3509d6b2e00ddfb282e4081a3994302b86cef2ff54c74e63000ec5319cf9f37b7685a14cad85f48a90e37afa507efa97881c27a4f4926fae1b74e96a4aed5a
+LibGit2.v1.3.0+0.aarch64-linux-gnu.tar.gz/md5/4884296753929a70f6f01b36bfec1f61
+LibGit2.v1.3.0+0.aarch64-linux-gnu.tar.gz/sha512/42babda48f23b672ac382780b450e314ee16929c523125246f7d66e11fd27208354fd4d4c7e663e2a6091de78612be0e825f8d1cd4eba595a056838df12fd213
+LibGit2.v1.3.0+0.aarch64-linux-musl.tar.gz/md5/c0b53bfdfa9d4b9e653a5470eccb40c6
+LibGit2.v1.3.0+0.aarch64-linux-musl.tar.gz/sha512/b3d79bbaeb26869066d6b7e228bc2712b67c5dc45badd317c3023eda86d82ac2b712a2126d4049f1074d0ed86ec9f80f9a2e7d6458d47c1d3c953a37a4b3ac0e
+LibGit2.v1.3.0+0.armv6l-linux-gnueabihf.tar.gz/md5/ac91abf4ce2ef1f25729d352c2bd3630
+LibGit2.v1.3.0+0.armv6l-linux-gnueabihf.tar.gz/sha512/c8297da990ead579e285e4347a99a824a81c147965a8394e034690b63e3e84f55f21d37f2754725dacc7836812698a06fb6101fe05b222cbe11d558742986e91
+LibGit2.v1.3.0+0.armv6l-linux-musleabihf.tar.gz/md5/eaf893aabde1ec021bbeda5150df6212
+LibGit2.v1.3.0+0.armv6l-linux-musleabihf.tar.gz/sha512/3bca29dd9bb724db74eae72a9bbeae777e315a616938d76144c2183c9bb9b1656c83de3e7c743c3cba8f2492f74a1cf4254c8942859211f1d6beb514e3532b18
+LibGit2.v1.3.0+0.armv7l-linux-gnueabihf.tar.gz/md5/fd6b13ae9129c6f082ab194782e33c01
+LibGit2.v1.3.0+0.armv7l-linux-gnueabihf.tar.gz/sha512/e236667bc0e3a929b625a410c55fffa0be3f71aef090c3c18c9806ff3c866a0f2dfc1afdcba34d7f0b81b0fd38fa3441b56f4343fedf66c5eead64f059c095d3
+LibGit2.v1.3.0+0.armv7l-linux-musleabihf.tar.gz/md5/9370e574abf25984dda0521b9b3d2df9
+LibGit2.v1.3.0+0.armv7l-linux-musleabihf.tar.gz/sha512/281c7f04d8560f5ef4c9902ace66f41b255e6868d83bbe37e61923371b0752a076f93f249abbd64e6d3849460c2c36dee9207303f0e0eb0ef5d37990aa7337df
+LibGit2.v1.3.0+0.i686-linux-gnu.tar.gz/md5/e0ecd37c7dd9709ddedf3eab8a4f2d47
+LibGit2.v1.3.0+0.i686-linux-gnu.tar.gz/sha512/04ff40e26df0f66413564f4189a031abf538dbea2cd41192164ab5e5361527b18d2a448ca7dacd9754a419d198dd816dd25bfecde4b2cfb1f497aa437a1784a3
+LibGit2.v1.3.0+0.i686-linux-musl.tar.gz/md5/4d98934e46f16eaa3afb597474639de6
+LibGit2.v1.3.0+0.i686-linux-musl.tar.gz/sha512/6d2fea7fe42d12642004f41d31d8ecf5213d0ce7cef3489c96583c9096d8b044b70dd3466bfc2ad901a7302fb8d320713a2ee4bb03702dd7487b0eb0b34966dc
+LibGit2.v1.3.0+0.i686-w64-mingw32.tar.gz/md5/29fc6f7d785aebaed7a916106efaf690
+LibGit2.v1.3.0+0.i686-w64-mingw32.tar.gz/sha512/ac29ef9b8d2d2b44a8d37bba4a40103fb174a55c3210b2b5c6d02baf7cf7f0bbea5acefc5c465ce931fc21275045ad0b39ada32b96795a74e4a46fb405c76398
+LibGit2.v1.3.0+0.powerpc64le-linux-gnu.tar.gz/md5/8b412e41808ff41418b35feab033c445
+LibGit2.v1.3.0+0.powerpc64le-linux-gnu.tar.gz/sha512/fecbcbe322e53a311aa28a6734ca6331438f14e030342efcab7d3f9f2b582c1c23b9d40ce63e1decb92d4d8620685d2e9ae7fadc55bf5db9169590f83327e2de
+LibGit2.v1.3.0+0.x86_64-apple-darwin.tar.gz/md5/033cb0d768a322dc6fc8f1fc58963f62
+LibGit2.v1.3.0+0.x86_64-apple-darwin.tar.gz/sha512/69d6c9e4b38257c89311cdf0a219a8497482c85a7a39c2c6d5819c5d9443d93f4978fbe08291313492f673e02ff4ae709fa4047c3ce015a806b4f22284c1c2cf
+LibGit2.v1.3.0+0.x86_64-linux-gnu.tar.gz/md5/bd7a36813ec371d2a39a98eced238294
+LibGit2.v1.3.0+0.x86_64-linux-gnu.tar.gz/sha512/b3a605a9cb7057f4c9652cd551922b4e11f65d74383a9c5b7a0234b7cf599e0a23cf79695e1db6c760dea194f76b59a347e38e14729b0bcc9b120812b63bb2c3
+LibGit2.v1.3.0+0.x86_64-linux-musl.tar.gz/md5/856ac3655c979881c6c0537f49907d18
+LibGit2.v1.3.0+0.x86_64-linux-musl.tar.gz/sha512/8cae2271bfa4a9dc6168fdd1d6133997756f56c60cd94525357f907109b41cb7d6594f7916557de03f1450b4bc694705f2f21d8426cb909c5678cee7f1477b88
+LibGit2.v1.3.0+0.x86_64-unknown-freebsd.tar.gz/md5/5c864ad058dd9c30340bb1c196d97b66
+LibGit2.v1.3.0+0.x86_64-unknown-freebsd.tar.gz/sha512/e2d3953ebe99743b0f6e62901b953e7c57030cfd5b1cc40d2bda85ed26573212c2e4748f2fdf46d88b75f584471a8b0b287c0a65ef83f502164e92624aaa091e
+LibGit2.v1.3.0+0.x86_64-w64-mingw32.tar.gz/md5/45d92f092eb1a319396dd7ebb36a2273
+LibGit2.v1.3.0+0.x86_64-w64-mingw32.tar.gz/sha512/0f219513972259f131fbc509eb035103a986bf1483f7cf9d0bfbca2802f5bbf23296a36a9f9d72e4d89a70f86acf781aec3d49df6aaf408da4f025f65d559cfa
+libgit2-b7bad55e4bb0a285b073ba5e02b01d3f522fc95d.tar.gz/md5/02582c680d006890def088ffaccea7d8
+libgit2-b7bad55e4bb0a285b073ba5e02b01d3f522fc95d.tar.gz/sha512/ee51c06c012503d66ba28d9c2fc9ad42af69f22fd1ae1be54642820ccd80c74e24d78eeec7fe5222daf2432930bcce163800502db1224571da852238c1970e36
diff --git a/deps/libgit2.mk b/deps/libgit2.mk
index 8f6ab58bc00da..5902cc68960ae 100644
--- a/deps/libgit2.mk
+++ b/deps/libgit2.mk
@@ -40,28 +40,14 @@ $(LIBGIT2_SRC_PATH)/libgit2-agent-nonfatal.patch-applied: $(LIBGIT2_SRC_PATH)/so
 		patch -p1 -f < $(SRCDIR)/patches/libgit2-agent-nonfatal.patch
 	echo 1 > $@
 
-# This can be removed once a release with https://github.com/libgit2/libgit2/pull/5685 lands
-$(LIBGIT2_SRC_PATH)/libgit2-mbedtls-incdir.patch-applied: $(LIBGIT2_SRC_PATH)/libgit2-agent-nonfatal.patch-applied
-	cd $(LIBGIT2_SRC_PATH) && \
-		patch -p1 -f < $(SRCDIR)/patches/libgit2-mbedtls-incdir.patch
-	echo 1 > $@
-
-$(LIBGIT2_SRC_PATH)/libgit2-hostkey.patch-applied: $(LIBGIT2_SRC_PATH)/libgit2-mbedtls-incdir.patch-applied
+$(LIBGIT2_SRC_PATH)/libgit2-hostkey.patch-applied: $(LIBGIT2_SRC_PATH)/libgit2-agent-nonfatal.patch-applied
 	cd $(LIBGIT2_SRC_PATH) && \
 		patch -p1 -f < $(SRCDIR)/patches/libgit2-hostkey.patch
 	echo 1 > $@
 
-# This can be removed once a release with https://github.com/libgit2/libgit2/pull/5740 lands
-$(LIBGIT2_SRC_PATH)/libgit2-continue-zlib.patch-applied: $(LIBGIT2_SRC_PATH)/libgit2-hostkey.patch-applied
-	cd $(LIBGIT2_SRC_PATH) && \
-		patch -p1 -f < $(SRCDIR)/patches/libgit2-continue-zlib.patch
-	echo 1 > $@
-
 $(BUILDDIR)/$(LIBGIT2_SRC_DIR)/build-configured: \
 	$(LIBGIT2_SRC_PATH)/libgit2-agent-nonfatal.patch-applied \
-	$(LIBGIT2_SRC_PATH)/libgit2-mbedtls-incdir.patch-applied \
-	$(LIBGIT2_SRC_PATH)/libgit2-hostkey.patch-applied \
-	$(LIBGIT2_SRC_PATH)/libgit2-continue-zlib.patch-applied
+	$(LIBGIT2_SRC_PATH)/libgit2-hostkey.patch-applied
 
 $(BUILDDIR)/$(LIBGIT2_SRC_DIR)/build-configured: $(LIBGIT2_SRC_PATH)/source-extracted
 	mkdir -p $(dir $@)
diff --git a/deps/libgit2.version b/deps/libgit2.version
index 7c7c34a4909e0..042f76bba673e 100644
--- a/deps/libgit2.version
+++ b/deps/libgit2.version
@@ -1,2 +1,2 @@
-LIBGIT2_BRANCH=v1.1.0
-LIBGIT2_SHA1=7f4fa178629d559c037a1f72f79f79af9c1ef8ce
+LIBGIT2_BRANCH=v1.3.0
+LIBGIT2_SHA1=b7bad55e4bb0a285b073ba5e02b01d3f522fc95d
diff --git a/stdlib/LibGit2/src/callbacks.jl b/stdlib/LibGit2/src/callbacks.jl
index 18de45a994420..5da032d3143e2 100644
--- a/stdlib/LibGit2/src/callbacks.jl
+++ b/stdlib/LibGit2/src/callbacks.jl
@@ -366,8 +366,8 @@ struct CertHostKey
     sha1    :: NTuple{20,UInt8}
     sha256  :: NTuple{32,UInt8}
     type    :: Cint
+    hostkey :: Ptr{Cchar}
     len     :: Csize_t
-    data    :: NTuple{1024,UInt8}
 end
 
 function verify_host_error(message::AbstractString)
@@ -433,14 +433,14 @@ function ssh_knownhost_check(
     host  :: AbstractString,
     cert  :: CertHostKey,
 )
-    key = collect(cert.data)[1:cert.len]
+    key = unsafe_wrap(Array, cert.hostkey, cert.len)
     return ssh_knownhost_check(files, host, key)
 end
 
 function ssh_knownhost_check(
     files :: AbstractVector{<:AbstractString},
     host  :: AbstractString,
-    key   :: Vector{UInt8},
+    key   :: Vector{Cchar},
 )
     if (m = match(r"^(.+):(\d+)$", host)) !== nothing
         host = m.captures[1]
@@ -476,7 +476,7 @@ function ssh_knownhost_check(
             hosts  :: Ptr{Cvoid},
             host   :: Cstring,
             port   :: Cint,
-            key    :: Ptr{UInt8},
+            key    :: Ptr{Cchar},
             len    :: Csize_t,
             mask   :: Cint,
             C_NULL :: Ptr{Ptr{KnownHost}},
diff --git a/stdlib/LibGit2/src/types.jl b/stdlib/LibGit2/src/types.jl
index b68dbb7c0bf02..2d95596cb276d 100644
--- a/stdlib/LibGit2/src/types.jl
+++ b/stdlib/LibGit2/src/types.jl
@@ -230,6 +230,9 @@ Matches the [`git_remote_callbacks`](https://libgit2.org/libgit2/#HEAD/type/git_
     push_update_reference::Ptr{Cvoid}  = C_NULL
     push_negotiation::Ptr{Cvoid}       = C_NULL
     transport::Ptr{Cvoid}              = C_NULL
+    @static if LibGit2.VERSION >= v"1.2.0"
+        remote_ready::Ptr{Cvoid}       = C_NULL
+    end
     payload::Any                       = nothing
     @static if LibGit2.VERSION >= v"0.99.0"
         resolve_url::Ptr{Cvoid}        = C_NULL
diff --git a/stdlib/LibGit2/test/libgit2.jl b/stdlib/LibGit2/test/libgit2.jl
index 892fb8bb6f3fe..2357536e19b5c 100644
--- a/stdlib/LibGit2/test/libgit2.jl
+++ b/stdlib/LibGit2/test/libgit2.jl
@@ -2473,11 +2473,11 @@ mktempdir() do dir
             CHECK_FAILURE  = LibGit2.Consts.LIBSSH2_KNOWNHOST_CHECK_FAILURE
 
             # randomly generated hashes matching no hosts
-            random_key = "\0\0\0\assh-rsa\0\0\0\x01#\0\0\0\x81\0¿\x95\xbe9\xfc9g\n:\xcf&\x06YA\xb5`\x97\xc13A\xbf;T+C\xc9Ut J>\xc5ҍ\xc4_S\x8a \xc1S\xeb\x15FH\xd2a\x04.D\xeeb\xac\x8f\xdb\xcc\xef\xc4l G\x9bR\xafp\x17s<=\x12\xab\x04ڳif\\A\x9ba0\xde%\xdei\x04\xc3\r\xb3\x81w\x88\xec\xc0f\x15A;AÝ\xc0r\xa1\u5fe\xd3\xf6)8\x8e\xa3\xcbc\xee\xdd\$\x04\x0f\xc1\xb4\x1f\xcc\xecK\xe0\x99" |> codeunits |> collect
+            random_key = collect(reinterpret(Cchar, codeunits("\0\0\0\assh-rsa\0\0\0\x01#\0\0\0\x81\0¿\x95\xbe9\xfc9g\n:\xcf&\x06YA\xb5`\x97\xc13A\xbf;T+C\xc9Ut J>\xc5ҍ\xc4_S\x8a \xc1S\xeb\x15FH\xd2a\x04.D\xeeb\xac\x8f\xdb\xcc\xef\xc4l G\x9bR\xafp\x17s<=\x12\xab\x04ڳif\\A\x9ba0\xde%\xdei\x04\xc3\r\xb3\x81w\x88\xec\xc0f\x15A;AÝ\xc0r\xa1\u5fe\xd3\xf6)8\x8e\xa3\xcbc\xee\xdd\$\x04\x0f\xc1\xb4\x1f\xcc\xecK\xe0\x99")))
             # hashes of the unique github.com fingerprint
-            github_key = "\0\0\0\assh-rsa\0\0\0\x01#\0\0\x01\x01\0\xab`;\x85\x11\xa6vy\xbd\xb5@\xdb;\xd2\x03K\0J\xe96\xd0k\xe3\xd7`\xf0\x8f˪\xdbN\xb4\xedóǑ\xc7\n\xae\x9at\xc9Xi\xe4wD!«\xea\x92\xe5T0_8\xb5\xfdAK2\b\xe5t\xc37\xe3 \x93e\x18F,vRɋ1\xe1n}\xa6R;\xd2\0t*dD\xd8?\xcd^\x172\xd06sǷ\x81\x15UH{U\xf0\xc4IO8)\xec\xe6\x0f\x94%Z\x95˚\xf57\xd7\xfc\x8c\x7f\xe4\x9e\xf3\x18GN\xf2\x92\t\x92\x05\"e\xb0\xa0n\xa6mJ\x16\x7f\xd9\xf3\xa4\x8a\x1aJ0~\xc1\xea\xaaQI\xa9i\xa6\xac]V\xa5\xefb~Q}\x81\xfbdO[t\\OG\x8e\xcd\b*\x94\x92\xf7D\xaa\xd3&\xf7l\x8cM\xc9\x10\vƫyF\x1d&W\xcbo\x06\xde\xc9.kd\xa6V/\xf0\xe3 \x84\xea\x06\xce\x0e\xa9\xd3ZX;\xfb\0\xbaӌ\x9d\x19p<T\x98\x92\xe5\xaaxܕ\xe2PQ@i" |> codeunits |> collect
+            github_key = collect(reinterpret(Cchar, codeunits("\0\0\0\assh-rsa\0\0\0\x01#\0\0\x01\x01\0\xab`;\x85\x11\xa6vy\xbd\xb5@\xdb;\xd2\x03K\0J\xe96\xd0k\xe3\xd7`\xf0\x8f˪\xdbN\xb4\xedóǑ\xc7\n\xae\x9at\xc9Xi\xe4wD!«\xea\x92\xe5T0_8\xb5\xfdAK2\b\xe5t\xc37\xe3 \x93e\x18F,vRɋ1\xe1n}\xa6R;\xd2\0t*dD\xd8?\xcd^\x172\xd06sǷ\x81\x15UH{U\xf0\xc4IO8)\xec\xe6\x0f\x94%Z\x95˚\xf57\xd7\xfc\x8c\x7f\xe4\x9e\xf3\x18GN\xf2\x92\t\x92\x05\"e\xb0\xa0n\xa6mJ\x16\x7f\xd9\xf3\xa4\x8a\x1aJ0~\xc1\xea\xaaQI\xa9i\xa6\xac]V\xa5\xefb~Q}\x81\xfbdO[t\\OG\x8e\xcd\b*\x94\x92\xf7D\xaa\xd3&\xf7l\x8cM\xc9\x10\vƫyF\x1d&W\xcbo\x06\xde\xc9.kd\xa6V/\xf0\xe3 \x84\xea\x06\xce\x0e\xa9\xd3ZX;\xfb\0\xbaӌ\x9d\x19p<T\x98\x92\xe5\xaaxܕ\xe2PQ@i")))
             # hashes of the middle github.com fingerprint
-            gitlab_key = "\0\0\0\vssh-ed25519\0\0\0 \a\xee\br\x95N:\xae\xc6\xfbz\bέtn\x12.\x9dA\xb6\x7f\xe79\xe1\xc7\x13\x95\x0e\xcd\x17_" |> codeunits |> collect
+            gitlab_key = collect(reinterpret(Cchar, codeunits("\0\0\0\vssh-ed25519\0\0\0 \a\xee\br\x95N:\xae\xc6\xfbz\bέtn\x12.\x9dA\xb6\x7f\xe79\xe1\xc7\x13\x95\x0e\xcd\x17_")))
 
             # various known hosts files
             no_file = tempname()
diff --git a/stdlib/LibGit2_jll/Project.toml b/stdlib/LibGit2_jll/Project.toml
index d38676c874025..5c4c42945a2a9 100644
--- a/stdlib/LibGit2_jll/Project.toml
+++ b/stdlib/LibGit2_jll/Project.toml
@@ -1,6 +1,6 @@
 name = "LibGit2_jll"
 uuid = "e37daf67-58a4-590a-8e99-b0245dd2ffc5"
-version = "1.2.3+0"
+version = "1.3.0+0"
 
 [deps]
 MbedTLS_jll = "c8ffd9c3-330d-5841-b78e-0817d7145fa1"
diff --git a/stdlib/LibGit2_jll/test/runtests.jl b/stdlib/LibGit2_jll/test/runtests.jl
index 9baea29d44be5..3a26e26d87ebd 100644
--- a/stdlib/LibGit2_jll/test/runtests.jl
+++ b/stdlib/LibGit2_jll/test/runtests.jl
@@ -7,5 +7,5 @@ using Test, Libdl, LibGit2_jll
     minor = Ref{Cint}(0)
     patch = Ref{Cint}(0)
     @test ccall((:git_libgit2_version, libgit2), Cint, (Ref{Cint}, Ref{Cint}, Ref{Cint}), major, minor, patch) == 0
-    @test VersionNumber(major[], minor[], patch[]) == v"1.1.0"
+    @test VersionNumber(major[], minor[], patch[]) == v"1.3.0"
 end
