diff --git a/src/jitlayers.h b/src/jitlayers.h
index c4a89f882bea..54a76630330f 100644
--- a/src/jitlayers.h
+++ b/src/jitlayers.h
@@ -13,8 +13,14 @@
 #include <llvm/ExecutionEngine/Orc/IRTransformLayer.h>
 #include <llvm/ExecutionEngine/JITEventListener.h>
 
+#include <llvm/Passes/PassBuilder.h>
+#include <llvm/Passes/PassPlugin.h>
+#include <llvm/Passes/StandardInstrumentations.h>
+
 #include <llvm/Target/TargetMachine.h>
 #include "julia_assert.h"
+#include "julia.h"
+#include "platform.h"
 #include "debug-registry.h"
 
 // As of LLVM 13, there are two runtime JIT linker implementations, the older
diff --git a/src/llvm-simdloop.cpp b/src/llvm-simdloop.cpp
index 5d7b86f067..7ca8767cc2 100644
--- a/src/llvm-simdloop.cpp
+++ b/src/llvm-simdloop.cpp
@@ -20,6 +20,7 @@
 #include <llvm-c/Core.h>
 #include <llvm-c/Types.h>
 
+#include <llvm/Analysis/LoopInfo.h>
 #include <llvm/Analysis/LoopPass.h>
 #include <llvm/IR/LegacyPassManager.h>
 #include <llvm/IR/Instructions.h>
