diff -ruN a/hack/lib/golang.sh b/hack/lib/golang.sh
--- a/hack/lib/golang.sh	2020-12-04 22:45:25.000000000 +0100
+++ b/hack/lib/golang.sh	2020-12-12 19:15:53.971460223 +0100
@@ -793,7 +793,7 @@
 
     # extract tags if any specified in GOFLAGS
     # shellcheck disable=SC2001
-    gotags="selinux,notest,$(echo "${GOFLAGS:-}" | sed -e 's|.*-tags=\([^-]*\).*|\1|')"
+    gotags="selinux notest $(echo "${GOFLAGS:-}" | sed -e 's|.*-tags=\([^-]*\).*|\1|')"
 
     local -a targets=()
     local arg
