commit fabc394472e1b8b8dea11497f56990bc9a1432bf
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Wed Jun 10 23:10:29 2020 +0800

    fix: compatibility with kwin 5.19

diff --git a/plugins/platforms/lib/kwinutils.cpp b/plugins/platforms/lib/kwinutils.cpp
index ed68ec8..f01a91e 100644
--- a/plugins/platforms/lib/kwinutils.cpp
+++ b/plugins/platforms/lib/kwinutils.cpp
@@ -589,7 +589,7 @@ QObject *KWinUtils::tabBox()

 QObject *KWinUtils::cursor()
 {
-    return KWin::Cursor::s_self;
+    return nullptr; //KWin::Cursor;
 }

 QObject *KWinUtils::virtualDesktop()
diff --git a/plugins/kwineffects/blur/blur.cpp b/plugins/kwineffects/blur/blur.cpp
index b5cae9e..0354bc0 100644
--- a/plugins/kwineffects/blur/blur.cpp
+++ b/plugins/kwineffects/blur/blur.cpp
@@ -32,10 +32,10 @@
 #include <cmath> // for ceil()
 #include <QTimer>
 
-#include <KWayland/Server/surface_interface.h>
-#include <KWayland/Server/blur_interface.h>
-#include <KWayland/Server/shadow_interface.h>
-#include <KWayland/Server/display.h>
+#include <KWaylandServer/surface_interface.h>
+#include <KWaylandServer/blur_interface.h>
+#include <KWaylandServer/shadow_interface.h>
+#include <KWaylandServer/display.h>
 
 static const QByteArray s_blurAtomName = QByteArrayLiteral("_KDE_NET_WM_BLUR_BEHIND_REGION");
 
@@ -61,10 +61,9 @@ BlurEffect::BlurEffect(QObject *, const QVariantList &)
     //     Should be included in _NET_SUPPORTED instead.
     if (m_shader && m_shader->isValid() && m_renderTargetsValid) {
         net_wm_blur_region = effects->announceSupportProperty(s_blurAtomName, this);
-        KWayland::Server::Display *display = effects->waylandDisplay();
+        KWaylandServer::Display *display = effects->waylandDisplay();
         if (display) {
             m_blurManager = display->createBlurManager(this);
-            m_blurManager->create();
         }
     } else {
         net_wm_blur_region = 0;
@@ -310,7 +309,7 @@ void BlurEffect::updateBlurRegion(EffectWindow *w) const
         }
     }
 
-    KWayland::Server::SurfaceInterface *surf = w->surface();
+    KWaylandServer::SurfaceInterface *surf = w->surface();
 
     if (surf && surf->blur()) {
         region = surf->blur()->region();
@@ -338,10 +337,10 @@ void BlurEffect::updateBlurRegion(EffectWindow *w) const
 
 void BlurEffect::slotWindowAdded(EffectWindow *w)
 {
-    KWayland::Server::SurfaceInterface *surf = w->surface();
+    KWaylandServer::SurfaceInterface *surf = w->surface();
 
     if (surf) {
-        windowBlurChangedConnections[w] = connect(surf, &KWayland::Server::SurfaceInterface::blurChanged, this, [this, w] () {
+        windowBlurChangedConnections[w] = connect(surf, &KWaylandServer::SurfaceInterface::blurChanged, this, [this, w] () {
             if (w) {
                 updateBlurRegion(w);
             }
@@ -445,7 +444,7 @@ QRegion BlurEffect::blurRegion(const EffectWindow *w) const
         const QRegion appRegion = qvariant_cast<QRegion>(value);
         if (!appRegion.isEmpty()) {
             if (w->decorationHasAlpha() && effects->decorationSupportsBlurBehind()) {
-                region = w->shape();
+                region = w->shape() & w->rect();
                 region -= w->decorationInnerRect();
             }
             region |= appRegion.translated(w->contentsRect().topLeft()) &
@@ -453,12 +452,12 @@ QRegion BlurEffect::blurRegion(const EffectWindow *w) const
         } else {
             // An empty region means that the blur effect should be enabled
             // for the whole window.
-            region = w->shape();
+            region = w->shape() & w->rect();
         }
     } else if (w->decorationHasAlpha() && effects->decorationSupportsBlurBehind()) {
         // If the client hasn't specified a blur region, we'll only enable
         // the effect behind the decoration.
-        region = w->shape();
+        region = w->shape() & w->rect();
         region -= w->decorationInnerRect();
     }
 
diff --git a/plugins/kwineffects/blur/blur.h b/plugins/kwineffects/blur/blur.h
index 7c76a82..53740ef 100644
--- a/plugins/kwineffects/blur/blur.h
+++ b/plugins/kwineffects/blur/blur.h
@@ -31,13 +31,10 @@
 
 using namespace KWin;
 
-namespace KWayland
-{
-namespace Server
+namespace KWaylandServer
 {
 class BlurManagerInterface;
 }
-}
 
 static const int borderSize = 5;
 
@@ -131,7 +128,7 @@ private:
     QVector <BlurValuesStruct> blurStrengthValues;
 
     QMap <EffectWindow*, QMetaObject::Connection> windowBlurChangedConnections;
-    KWayland::Server::BlurManagerInterface *m_blurManager = nullptr;
+    KWaylandServer::BlurManagerInterface *m_blurManager = nullptr;
 };
 
 inline
