Description: fix for c++14/GCC 6 compatibility
Author: Andreas Stührk <andy@hammerhartes.de>
Bug-Debian: https://bugs.debian.org/811627
Last-Update: <2016-09-24>

--- kyotocabinet-1.2.76.orig/kcdbext.h
+++ kyotocabinet-1.2.76/kcdbext.h
@@ -1278,7 +1278,7 @@ class IndexDB {
     if (omode_ == 0) {
       set_error(_KCCODELINE_, BasicDB::Error::INVALID, "not opened");
       *sp = 0;
-      return false;
+      return nullptr;
     }
     if (!cache_) return db_.get(kbuf, ksiz, sp);
     size_t dvsiz = 0;

