--- ./macros/lapack-check.m4.orig	2012-06-07 02:30:26.000000000 -0600
+++ ./macros/lapack-check.m4	2012-10-02 15:05:47.051001770 -0600
@@ -38,7 +38,7 @@ LIBS="${BACKUP_LIBS} ${BLAS_LIBS}"
 
 AC_TRY_RUN(dnl ICC ?
 [   #include "fflas-ffpack/fflas-ffpack-config.h"
-	#ifdef __FFLAS_FFPACK_HAVE_LAPACK
+	#ifdef __FFLASFFPACK_HAVE_LAPACK
 	   int main() { return 0 ; }
    #else
    a pas lapack
--- ./configure.orig	2012-06-07 15:19:31.000000000 -0600
+++ ./configure	2012-10-02 15:06:07.629005714 -0600
@@ -18077,7 +18077,7 @@ else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
    #include "fflas-ffpack/fflas-ffpack-config.h"
-	#ifdef __FFLAS_FFPACK_HAVE_LAPACK
+	#ifdef __FFLASFFPACK_HAVE_LAPACK
 	   int main() { return 0 ; }
    #else
    a pas lapack
