--- libffado-2.3.0/src/libieee1394/configrom.cpp.derefptr	2017-02-16 12:03:33.894937836 +0000
+++ libffado-2.3.0/src/libieee1394/configrom.cpp	2017-02-16 12:03:44.000947451 +0000
@@ -176,7 +176,7 @@
                 ( void* )CSR1212_TEXTUAL_DESCRIPTOR_LEAF_DATA( m_vendorNameKv ),
                 len );
 
-    while ((buf + len - 1) == '\0') {
+    while (*(buf + len - 1) == '\0') {
             len--;
         }
         // \todo XXX seems a bit strage to do this but the nodemgr.c code does
@@ -195,7 +195,7 @@
         memcpy( buf,
                 ( void* )CSR1212_TEXTUAL_DESCRIPTOR_LEAF_DATA( m_modelNameKv ),
                 len );
-    while ((buf + len - 1) == '\0') {
+    while (*(buf + len - 1) == '\0') {
             len--;
         }
         // \todo XXX for edirol fa-66 it seems somehow broken. see above
