diff -ruN a/generator/actions_core.ml b/generator/actions_core.ml
--- a/generator/actions_core.ml	2022-03-09 10:28:02.397896774 +0100
+++ b/generator/actions_core.ml	2022-04-27 18:45:42.691484853 +0200
@@ -6829,7 +6829,7 @@
     tests = [
       InitNone, Always, TestResult (
         [["isoinfo_device"; "/dev/sdd"]],
-          "STREQ (ret->iso_volume_id, \"ISOIMAGE\") && "^
+          "STREQ (ret->iso_volume_id, \"CDROM\") && "^
           "STREQ (ret->iso_volume_set_id, \"\") && "^
           "ret->iso_volume_set_size == 1 && "^
           "ret->iso_volume_sequence_number == 1 && "^
diff -ruN a/tests/c-api/tests.c b/tests/c-api/tests.c
--- a/tests/c-api/tests.c	2022-04-14 16:31:43.202811561 +0200
+++ b/tests/c-api/tests.c	2022-04-27 15:03:59.821806995 +0200
@@ -17739,9 +17739,9 @@
   ret = guestfs_isoinfo_device (g, arg1943);
   if (ret == NULL)
       return -1;
-  if (! (STREQ (ret->iso_volume_id, "ISOIMAGE") && STREQ (ret->iso_volume_set_id, "") && ret->iso_volume_set_size == 1 && ret->iso_volume_sequence_number == 1 && ret->iso_logical_block_size == 2048)) {
+  if (! (STREQ (ret->iso_volume_id, "CDROM") && STREQ (ret->iso_volume_set_id, "") && ret->iso_volume_set_size == 1 && ret->iso_volume_sequence_number == 1 && ret->iso_logical_block_size == 2048)) {
     fprintf (stderr, "%s: test failed: expression false: %s\n",
-             "test_isoinfo_device_0", "STREQ (ret->iso_volume_id, \"ISOIMAGE\") && STREQ (ret->iso_volume_set_id, \"\") && ret->iso_volume_set_size == 1 && ret->iso_volume_sequence_number == 1 && ret->iso_logical_block_size == 2048");
+             "test_isoinfo_device_0", "STREQ (ret->iso_volume_id, \"CDROM\") && STREQ (ret->iso_volume_set_id, \"\") && ret->iso_volume_set_size == 1 && ret->iso_volume_sequence_number == 1 && ret->iso_logical_block_size == 2048");
     if (!guestfs_get_trace (g))
       fprintf (stderr, "Set LIBGUESTFS_TRACE=1 to see values returned from API calls.\n");
     return -1;
