diff --git a/Makefile b/Makefile
index 84b4527f..01fd2b21 100644
--- a/Makefile
+++ b/Makefile
@@ -193,7 +193,7 @@ $(warning Your system does not have gtk3/gtk2, disabling test-inspect)
 endif
 
 # Can we use LLVM (needed for ... sparse-llvm)?
-LLVM_CONFIG:=llvm-config
+LLVM_CONFIG:=llvm-config-14
 HAVE_LLVM:=$(shell $(LLVM_CONFIG) --version >/dev/null 2>&1 && echo 'yes')
 ifeq ($(HAVE_LLVM),yes)
 arch := $(shell uname -m)
diff --git a/sparse-llvm-dis b/sparse-llvm-dis
index 2958217b..453fcf85 100755
--- a/sparse-llvm-dis
+++ b/sparse-llvm-dis
@@ -4,7 +4,7 @@
 
 set +e
 
-DIS=$("${LLVM_CONFIG:-llvm-config}" --bindir)/llvm-dis
+DIS=$("${LLVM_CONFIG:-llvm-config-14}" --bindir)/llvm-dis
 
 if [ $# -eq 0 ]; then
 	echo "$(basename $0): no input files"
diff --git a/sparsec b/sparsec
index 3f9b60c6..0d24ffe8 100755
--- a/sparsec
+++ b/sparsec
@@ -32,7 +32,7 @@ done
 TMPFILE=`mktemp -t tmp.XXXXXX`
 
 
-LLC=`"${LLVM_CONFIG:-llvm-config}" --bindir`/llc
+LLC=`"${LLVM_CONFIG:-llvm-config-14}" --bindir`/llc
 
 LLC_ARCH_OPTS=
 case "$(uname -s)" in
diff --git a/sparsei b/sparsei
index 3232200e..bfd33f7e 100755
--- a/sparsei
+++ b/sparsei
@@ -6,7 +6,7 @@ SPARSEOPTS=
 JIT_OPT=
 
 DIRNAME=`dirname $0`
-LLI=`"${LLVM_CONFIG:-llvm-config}" --bindir`/lli
+LLI=`"${LLVM_CONFIG:-llvm-config-14}" --bindir`/lli
 
 if [ $# -eq 0 ]; then
   echo "`basename $0`: no input files"
