diff --git a/source/shared_lib/include/graphics/math_util.h b/source/shared_lib/include/graphics/math_util.h
index 8fcfb27..3cb7a41 100644
--- a/source/shared_lib/include/graphics/math_util.h
+++ b/source/shared_lib/include/graphics/math_util.h
@@ -184,10 +184,10 @@ public:
 
 	Rect2<T> computeBoundingRect() const{
 		return Rect2i(
-			min(p[0].x, p[1].x), 
-			min(p[0].y, p[2].y), 
-			max(p[2].x, p[3].x), 
-			max(p[1].y, p[3].y));
+			std::min(p[0].x, p[1].x),
+			std::min(p[0].y, p[2].y),
+			std::max(p[2].x, p[3].x),
+			std::max(p[1].y, p[3].y));
 	}
 
 	bool isInside(const Vec2<T> &pt) const{
diff --git a/source/shared_lib/sources/platform/posix/ircclient.cpp b/source/shared_lib/sources/platform/posix/ircclient.cpp
index b09c972..ddebd03 100644
--- a/source/shared_lib/sources/platform/posix/ircclient.cpp
+++ b/source/shared_lib/sources/platform/posix/ircclient.cpp
@@ -13,6 +13,7 @@
 #include "ircclient.h"
 #include "util.h"
 #include "platform_common.h"
+#include "libirc_rfcnumeric.h"
 #include "libircclient.h"
 
 #include <stdio.h>
diff --git a/source/shared_lib/sources/util/util.cpp b/source/shared_lib/sources/util/util.cpp
index 05478fc..c27aa62 100644
--- a/source/shared_lib/sources/util/util.cpp
+++ b/source/shared_lib/sources/util/util.cpp
@@ -19,6 +19,7 @@
 #include <stdarg.h>
 #include <time.h>
 #include <fcntl.h> // for open()
+#include <unistd.h>
 
 #ifdef WIN32
   #include <io.h> // for open()
