diff -wbBur luasocket-2.0.2/src/auxiliar.c luasocket-2.0.2.my/src/auxiliar.c
--- luasocket-2.0.2/src/auxiliar.c	2007-10-15 08:21:05.000000000 +0400
+++ luasocket-2.0.2.my/src/auxiliar.c	2012-11-14 18:23:14.000000000 +0400
@@ -24,7 +24,7 @@
 * Creates a new class with given methods
 * Methods whose names start with __ are passed directly to the metatable.
 \*-------------------------------------------------------------------------*/
-void auxiliar_newclass(lua_State *L, const char *classname, luaL_reg *func) {
+void auxiliar_newclass(lua_State *L, const char *classname, luaL_Reg *func) {
     luaL_newmetatable(L, classname); /* mt */
     /* create __index table to place methods */
     lua_pushstring(L, "__index");    /* mt,"__index" */
diff -wbBur luasocket-2.0.2/src/auxiliar.h luasocket-2.0.2.my/src/auxiliar.h
--- luasocket-2.0.2/src/auxiliar.h	2007-10-15 08:21:05.000000000 +0400
+++ luasocket-2.0.2.my/src/auxiliar.h	2012-11-14 18:23:42.000000000 +0400
@@ -35,7 +35,7 @@
 #include "lauxlib.h"
 
 int auxiliar_open(lua_State *L);
-void auxiliar_newclass(lua_State *L, const char *classname, luaL_reg *func);
+void auxiliar_newclass(lua_State *L, const char *classname, luaL_Reg *func);
 void auxiliar_add2group(lua_State *L, const char *classname, const char *group);
 void auxiliar_setclass(lua_State *L, const char *classname, int objidx);
 void *auxiliar_checkclass(lua_State *L, const char *classname, int objidx);
diff -wbBur luasocket-2.0.2/src/except.c luasocket-2.0.2.my/src/except.c
--- luasocket-2.0.2/src/except.c	2007-10-15 08:21:05.000000000 +0400
+++ luasocket-2.0.2.my/src/except.c	2012-11-14 18:23:23.000000000 +0400
@@ -21,7 +21,7 @@
 static int do_nothing(lua_State *L);
 
 /* except functions */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"newtry",    global_newtry},
     {"protect",   global_protect},
     {NULL,        NULL}
diff -wbBur luasocket-2.0.2/src/inet.c luasocket-2.0.2.my/src/inet.c
--- luasocket-2.0.2/src/inet.c	2007-10-15 08:21:05.000000000 +0400
+++ luasocket-2.0.2.my/src/inet.c	2012-11-14 18:23:47.000000000 +0400
@@ -21,7 +21,7 @@
 static int inet_global_gethostname(lua_State *L);
 
 /* DNS functions */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     { "toip", inet_global_toip },
     { "tohostname", inet_global_tohostname },
     { "gethostname", inet_global_gethostname},
diff -wbBur luasocket-2.0.2/src/luasocket.c luasocket-2.0.2.my/src/luasocket.c
--- luasocket-2.0.2/src/luasocket.c	2007-10-15 08:21:05.000000000 +0400
+++ luasocket-2.0.2.my/src/luasocket.c	2012-11-14 18:23:26.000000000 +0400
@@ -47,7 +47,7 @@
 /*-------------------------------------------------------------------------*\
 * Modules and functions
 \*-------------------------------------------------------------------------*/
-static const luaL_reg mod[] = {
+static const luaL_Reg mod[] = {
     {"auxiliar", auxiliar_open},
     {"except", except_open},
     {"timeout", timeout_open},
@@ -59,7 +59,7 @@
     {NULL, NULL}
 };
 
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"skip",      global_skip},
     {"__unload",  global_unload},
     {NULL,        NULL}
diff -wbBur luasocket-2.0.2/src/mime.c luasocket-2.0.2.my/src/mime.c
--- luasocket-2.0.2/src/mime.c	2007-10-15 08:21:05.000000000 +0400
+++ luasocket-2.0.2.my/src/mime.c	2012-11-14 18:23:31.000000000 +0400
@@ -48,7 +48,7 @@
 static size_t qppad(UC *input, size_t size, luaL_Buffer *buffer);
 
 /* code support functions */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     { "dot", mime_global_dot },
     { "b64", mime_global_b64 },
     { "eol", mime_global_eol },
diff -wbBur luasocket-2.0.2/src/select.c luasocket-2.0.2.my/src/select.c
--- luasocket-2.0.2/src/select.c	2007-10-15 08:21:05.000000000 +0400
+++ luasocket-2.0.2.my/src/select.c	2012-11-14 18:23:35.000000000 +0400
@@ -27,7 +27,7 @@
 static int global_select(lua_State *L);
 
 /* functions in library namespace */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"select", global_select},
     {NULL,     NULL}
 };
diff -wbBur luasocket-2.0.2/src/tcp.c luasocket-2.0.2.my/src/tcp.c
--- luasocket-2.0.2/src/tcp.c	2007-10-15 08:21:05.000000000 +0400
+++ luasocket-2.0.2.my/src/tcp.c	2012-11-14 18:23:38.000000000 +0400
@@ -38,7 +38,7 @@
 static int meth_dirty(lua_State *L);
 
 /* tcp object methods */
-static luaL_reg tcp[] = {
+static luaL_Reg tcp[] = {
     {"__gc",        meth_close},
     {"__tostring",  auxiliar_tostring},
     {"accept",      meth_accept},
@@ -73,7 +73,7 @@
 };
 
 /* functions in library namespace */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"tcp", global_create},
     {NULL, NULL}
 };
diff -wbBur luasocket-2.0.2/src/timeout.c luasocket-2.0.2.my/src/timeout.c
--- luasocket-2.0.2/src/timeout.c	2007-10-15 08:21:05.000000000 +0400
+++ luasocket-2.0.2.my/src/timeout.c	2012-11-14 18:23:21.000000000 +0400
@@ -33,7 +33,7 @@
 static int timeout_lua_gettime(lua_State *L);
 static int timeout_lua_sleep(lua_State *L);
 
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     { "gettime", timeout_lua_gettime },
     { "sleep", timeout_lua_sleep },
     { NULL, NULL }
diff -wbBur luasocket-2.0.2/src/udp.c luasocket-2.0.2.my/src/udp.c
--- luasocket-2.0.2/src/udp.c	2007-10-15 08:21:05.000000000 +0400
+++ luasocket-2.0.2.my/src/udp.c	2012-11-14 18:23:44.000000000 +0400
@@ -43,7 +43,7 @@
 static int meth_dirty(lua_State *L);
 
 /* udp object methods */
-static luaL_reg udp[] = {
+static luaL_Reg udp[] = {
     {"__gc",        meth_close},
     {"__tostring",  auxiliar_tostring},
     {"close",       meth_close},
@@ -76,7 +76,7 @@
 };
 
 /* functions in library namespace */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"udp", global_create},
     {NULL, NULL}
 };
diff -wbBur luasocket-2.0.2/src/unix.c luasocket-2.0.2.my/src/unix.c
--- luasocket-2.0.2/src/unix.c	2007-10-15 08:21:05.000000000 +0400
+++ luasocket-2.0.2.my/src/unix.c	2012-11-14 18:23:17.000000000 +0400
@@ -39,7 +39,7 @@
 static const char *unix_trybind(p_unix un, const char *path);
 
 /* unix object methods */
-static luaL_reg un[] = {
+static luaL_Reg un[] = {
     {"__gc",        meth_close},
     {"__tostring",  auxiliar_tostring},
     {"accept",      meth_accept},
@@ -71,7 +71,7 @@
 };
 
 /* our socket creation function */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"unix", global_create},
     {NULL,          NULL}
 };
