diff --git a/cypari2/pari_instance.pyx b/cypari2/pari_instance.pyx
index a0dfb07..9f2bdb4 100644
--- a/cypari2/pari_instance.pyx
+++ b/cypari2/pari_instance.pyx
@@ -491,7 +491,7 @@ cdef class Pari(Pari_auto):
 
         # Take 1MB as minimal stack. Use maxprime=0, which PARI will
         # internally increase to some small value like 65537.
-        pari_init_opts(1000000, 0, INIT_DFTm)
+        pari_init_opts(1000000, 500000, INIT_DFTm)
         after_resize()
 
         # Disable PARI's stack overflow checking which is incompatible
