diff --git a/Makefile b/Makefile
index a369415b0..7ae5064ec 100644
--- a/Makefile
+++ b/Makefile
@@ -66,7 +66,7 @@ NEKO_VERSION_TAG=v$(shell echo "$(NEKO_VERSION)" | sed "s/\./-/g")
 ifneq ($(STATICLINK),0)
 	LIB_PARAMS= -cclib '-Wl,-Bstatic -lpcre2-8 -lz -lmbedtls -lmbedx509 -lmbedcrypto -Wl,-Bdynamic '
 else
-	LIB_PARAMS?= -cclib -lpcre2-8 -cclib -lz -cclib -lmbedtls -cclib -lmbedx509 -cclib -lmbedcrypto
+	LIB_PARAMS?= -ccopt -L/usr/lib/mbedtls2 -cclib -lpcre2-8 -cclib -lz -cclib -lmbedtls -cclib -lmbedx509 -cclib -lmbedcrypto
 endif
 ifeq ($(SYSTEM_NAME),Mac)
 	LIB_PARAMS+= -cclib '-framework Security -framework CoreFoundation'
diff --git a/libs/mbedtls/dune b/libs/mbedtls/dune
index ca0780ddd..1fc9a5704 100644
--- a/libs/mbedtls/dune
+++ b/libs/mbedtls/dune
@@ -4,6 +4,7 @@
 	(name mbedtls)
 	(foreign_stubs
 		(language c)
-		(names mbedtls_stubs))
+		(names mbedtls_stubs)
+		(flags (:standard -I/usr/include/mbedtls2)))
 	(wrapped false)
 )
