diff -up megaglest-3.12.0/source/shared_lib/sources/platform/common/platform_common.cpp.orig megaglest-3.12.0/source/shared_lib/sources/platform/common/platform_common.cpp
--- megaglest-3.12.0/source/shared_lib/sources/platform/common/platform_common.cpp.orig	2016-06-24 07:56:09.496569134 -0300
+++ megaglest-3.12.0/source/shared_lib/sources/platform/common/platform_common.cpp	2016-06-24 07:56:11.962569290 -0300
@@ -124,8 +124,7 @@ tm threadsafe_localtime(const time_t &ti
 // extracting std::time_t from std:chrono for "now"
 time_t systemtime_now() {
 #if __cplusplus > 199711L
-	system_time_point system_now = std::chrono::system_clock::now();
-	return std::chrono::system_clock::to_time_t(system_now);
+	return std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
 #else
 	return time(NULL);
 #endif
