diff -upr proxytunnel-1.9.0.orig/ptstream.c proxytunnel-1.9.0/ptstream.c
--- proxytunnel-1.9.0.orig/ptstream.c	2008-02-27 01:27:27.000000000 +0200
+++ proxytunnel-1.9.0/ptstream.c	2016-03-02 20:43:32.909153946 +0200
@@ -151,7 +151,7 @@ int stream_enable_ssl(PTSTREAM *pts) {
 	
 	/* Initialise the connection */
 	SSLeay_add_ssl_algorithms();
-	meth = SSLv3_client_method();
+	meth = SSLv23_client_method();
 	SSL_load_error_strings();
 
 	ctx = SSL_CTX_new (meth);
