diff --git a/build/features.py b/build/features.py
index 7d5c76249f..2275567e4b 100644
--- a/build/features.py
+++ b/build/features.py
@@ -502,7 +502,7 @@ class FAAD(Feature):
         if not self.enabled(build):
             return
 
-        have_mp4v2_h = conf.CheckHeader('mp4v2/mp4v2.h')
+        have_mp4v2_h = conf.CheckHeader('mp4v2/mp4v2.h', language='C++')
         have_mp4v2 = conf.CheckLib(['mp4v2', 'libmp4v2'], autoadd=False)
         have_mp4_h = conf.CheckHeader('mp4.h')
         have_mp4 = conf.CheckLib('mp4', autoadd=False)
diff --git a/plugins/soundsourcem4a/SConscript b/plugins/soundsourcem4a/SConscript
index 9f9a74cbca..9f8006908a 100644
--- a/plugins/soundsourcem4a/SConscript
+++ b/plugins/soundsourcem4a/SConscript
@@ -38,7 +38,7 @@ if int(build.flags['faad']):
 
     conf = Configure(env)
 
-    have_mp4v2_h = conf.CheckHeader('mp4v2/mp4v2.h')
+    have_mp4v2_h = conf.CheckHeader('mp4v2/mp4v2.h', language='C++')
     if have_mp4v2_h:
         env.Append(CPPDEFINES = '__MP4V2__')
 
