diff -Naur mysql++-3.1.0.old//lib/refcounted.h mysql++-3.1.0/lib/refcounted.h
--- mysql++-3.1.0.old//lib/refcounted.h	2010-06-03 19:59:23.000000000 +0200
+++ mysql++-3.1.0/lib/refcounted.h	2011-04-23 12:35:39.943366970 +0200
@@ -101,7 +101,7 @@
 	{
 		std::auto_ptr<T> exception_guard(counted_);
 		if (counted_) {
-			refs_ = new size_t(1);
+			refs_ = new std::size_t(1);
 		}
 		exception_guard.release();	// previous new didn't throw
 	}
@@ -255,7 +255,7 @@
 	/// We can't keep this as a plain integer because this object
 	/// allows itself to be copied.  All copies need to share this
 	/// reference count, not just the pointer to the counted object.
-	size_t* refs_;
+	std::size_t* refs_;
 };
 
 
