diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6f4b631..dd5d925 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,12 +34,6 @@ add_subdirectory (po)
 # now let's actually check for the required dependencies
 pkg_check_modules(DEPS REQUIRED granite)
 
-pkg_check_modules (UNITY REQUIRED unity>=4.0.0)
-if (UNITY_FOUND)
-    add_definitions(${UNITY_CFLAGS})
-    link_directories(${UNITY_LIBRARY_DIRS})
-endif ()
-
 add_definitions(${DEPS_CFLAGS})
 link_directories(${DEPS_LIBRARY_DIRS})
 include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/transmission)
@@ -71,7 +65,6 @@ CUSTOM_VAPIS
 # tell what libraries to use when compiling
 PACKAGES
     granite
-    unity
     posix
 OPTIONS
     --fatal-warnings
@@ -87,7 +80,7 @@ add_schema (data/com.github.davidmhewitt.torrential.gschema.xml)
 # tell cmake what to call the executable we just made
 add_executable(${EXEC_NAME} ${VALA_C} ${GLIB_RESOURCES_ICONS})
 add_dependencies(${EXEC_NAME} transmission)
-target_link_libraries(${EXEC_NAME} ${DEPS_LIBRARIES} ${UNITY_LIBRARIES} transmission)
+target_link_libraries(${EXEC_NAME} ${DEPS_LIBRARIES} transmission)
 
 # install the binaries we just made
 install (TARGETS ${EXEC_NAME} RUNTIME DESTINATION bin)
diff --git a/src/MainWindow.vala b/src/MainWindow.vala
index 5b40124..5829902 100644
--- a/src/MainWindow.vala
+++ b/src/MainWindow.vala
@@ -35,7 +35,6 @@ public class Torrential.MainWindow : Gtk.Window {
     private Widgets.MultiInfoBar infobar;
     private Widgets.TorrentListBox list_box;
     private Gtk.ScrolledWindow list_box_scroll;
-    private Unity.LauncherEntry launcher_entry;
 
     private Gtk.SearchEntry search_entry;
 
@@ -147,8 +146,6 @@ public class Torrential.MainWindow : Gtk.Window {
         set_titlebar (headerbar);
         show_all ();
 
-        launcher_entry = Unity.LauncherEntry.get_for_desktop_id ("com.github.davidmhewitt.torrential.desktop");
-
         var torrents = torrent_manager.get_torrents ();
         if (torrents.size > 0) {
             enable_main_view ();
@@ -173,13 +170,6 @@ public class Torrential.MainWindow : Gtk.Window {
         refresh_timer = Timeout.add_seconds (1, () => {
             list_box.update ();
             update_category_totals (torrent_manager.get_torrents ());
-            launcher_entry.progress = torrent_manager.get_overall_progress ();
-            var focused = (get_window ().get_state () & Gdk.WindowState.FOCUSED) != 0;
-            if (!focused && list_box.has_visible_children ()) {
-                launcher_entry.progress_visible = true;
-            } else {
-                launcher_entry.progress_visible = false;
-            }
             return true;
         });
 
@@ -395,7 +385,6 @@ public class Torrential.MainWindow : Gtk.Window {
         Source.remove (int_sig);
         Source.remove (term_sig);
 
-        launcher_entry.progress_visible = false;
         quitting_for_real = true;
         close ();
     }
