diff --git a/iconengineplugins/builtinengine/builtinengine.pro b/iconengineplugins/builtinengine/builtinengine.pro
index 7d7c448..317b7a8 100644
--- a/iconengineplugins/builtinengine/builtinengine.pro
+++ b/iconengineplugins/builtinengine/builtinengine.pro
@@ -7,7 +7,7 @@ SOURCES += main.cpp \
            dbuiltiniconengine.cpp
 OTHER_FILES += builtinengine.json
 
-QT += gui-private dtkgui$${DTK_VERSION}
+QT += gui-private dtkgui
 DESTDIR = $$_PRO_FILE_PWD_/../../bin/plugins/iconengines/$${BASED_DTK_DIR}
 
 unix {
diff --git a/iconengineplugins/svgiconengine/svgiconengine.pro b/iconengineplugins/svgiconengine/svgiconengine.pro
index 113cfa2..e3608f5 100644
--- a/iconengineplugins/svgiconengine/svgiconengine.pro
+++ b/iconengineplugins/svgiconengine/svgiconengine.pro
@@ -9,7 +9,7 @@ OTHER_FILES += qsvgiconengine.json
 
 CONFIG += c++11
 
-QT += gui-private dtkgui$${DTK_VERSION} concurrent
+QT += gui-private dtkgui concurrent
 DESTDIR = $$_PRO_FILE_PWD_/../../bin/plugins/iconengines/$${BASED_DTK_DIR}
 
 unix {
diff --git a/imageformatplugins/svg/svg.pro b/imageformatplugins/svg/svg.pro
index d2be96d..4a972dd 100644
--- a/imageformatplugins/svg/svg.pro
+++ b/imageformatplugins/svg/svg.pro
@@ -8,7 +8,7 @@ SOURCES += main.cpp \
 OTHER_FILES += svg.json
 
 CONFIG += c++11 link_pkgconfig
-PKGCONFIG += dtkgui$${DTK_VERSION}
+PKGCONFIG += dtkgui
 
 DESTDIR = $$_PRO_FILE_PWD_/../../bin/plugins/imageformats/$${BASED_DTK_DIR}
 
diff --git a/platformthemeplugin/qt5deepintheme-plugin.pro b/platformthemeplugin/qt5deepintheme-plugin.pro
index 6a2c75c..bc536ee 100644
--- a/platformthemeplugin/qt5deepintheme-plugin.pro
+++ b/platformthemeplugin/qt5deepintheme-plugin.pro
@@ -1,6 +1,6 @@
 # load(configure) will clear DTK_VERSION
 include(../.qmake.conf)
-QT       += dbus x11extras dtkgui$${DTK_VERSION}
+QT       += dbus x11extras dtkgui
 QT       += core-private gui-private
 greaterThan(QT_MAJOR_VERSION, 4) {
   QT += widgets widgets-private
@@ -17,7 +17,7 @@ CONFIG += plugin link_pkgconfig c++11
 load(configure)
 qtCompileTest(dthemed_icon_lookup) {
     DEFINES += DTHEMED_ICON_LOOKUP
-    PKGCONFIG += dtkwidget$${DTK_VERSION} til
+    PKGCONFIG += dtkwidget til
 
     SOURCES += diconengine.cpp
     HEADERS += diconengine.h
diff --git a/styleplugins/chameleon/chameleon.pro b/styleplugins/chameleon/chameleon.pro
index fe05fd5..661e291 100644
--- a/styleplugins/chameleon/chameleon.pro
+++ b/styleplugins/chameleon/chameleon.pro
@@ -1,4 +1,4 @@
-QT        += core gui widgets widgets-private core-private gui-private dtkgui$${DTK_VERSION} dtkwidget$${DTK_VERSION}
+QT        += core gui widgets widgets-private core-private gui-private dtkgui dtkwidget
 
 TARGET = chameleon$${DTK_VERSION}
 TEMPLATE = lib
diff --git a/styleplugins/dstyleplugin/dstyleplugin.pro b/styleplugins/dstyleplugin/dstyleplugin.pro
index 9124039..e82e86c 100644
--- a/styleplugins/dstyleplugin/dstyleplugin.pro
+++ b/styleplugins/dstyleplugin/dstyleplugin.pro
@@ -6,7 +6,7 @@
 
 QT        += core gui widgets widgets-private core-private gui-private
 CONFIG    += link_pkgconfig c++11
-PKGCONFIG += dtkwidget$${DTK_VERSION}
+PKGCONFIG += dtkwidget
 
 TARGET = dstyleplugin$${DTK_VERSION}
 TEMPLATE = lib
@@ -66,9 +66,9 @@ RESOURCES += \
     assets.qrc
 
 load(configure)
-qtCompileTest(dtkwidget$${DTK_VERSION}) {
+qtCompileTest(dtkwidget) {
     DEFINES += DTK_SUPPORT_BLUR_WINDOW
-    message(The libdtkwidget$${DTK_VERSION} support blur window)
+    message(The libdtkwidget support blur window)
 } else {
-    warning(The libdtkwidget$${DTK_VERSION} can not support blur window)
+    warning(The libdtkwidget can not support blur window)
 }
diff --git a/styles/styles.pro b/styles/styles.pro
index d152b48..0ae8b80 100644
--- a/styles/styles.pro
+++ b/styles/styles.pro
@@ -1,4 +1,4 @@
-QT += widgets dtkwidget$${DTK_VERSION}
+QT += widgets dtkwidget
 
 HEADERS       = norwegianwoodstyle.h \
                 widgetgallery.h
diff --git a/tests/tests.pro b/tests/tests.pro
index 0f223fe..ddfa2f8 100644
--- a/tests/tests.pro
+++ b/tests/tests.pro
@@ -1,5 +1,5 @@
 TEMPLATE = app
-QT += dtkgui$${DTK_VERSION} concurrent dtkwidget$${DTK_VERSION}
+QT += dtkgui concurrent dtkwidget
 CONFIG -= app_bundle
 
 unix:LIBS += -lgtest
