diff --git a/data/xsession/openbox-kde-session.in b/data/xsession/openbox-kde-session.in
index 35722796..ee91c16f 100644
--- a/data/xsession/openbox-kde-session.in
+++ b/data/xsession/openbox-kde-session.in
@@ -17,4 +17,4 @@ xprop -root -remove _NET_NUMBER_OF_DESKTOPS \
 
 # Run KDE with Openbox as its window manager
 export KDEWM="@bindir@/openbox"
-exec startkde "$@"
+exec /usr/bin/startplasma-x11 "$@"
diff --git a/data/xsession/openbox-kde.desktop.in b/data/xsession/openbox-kde.desktop.in
index bc78a97c..a9bcc219 100644
--- a/data/xsession/openbox-kde.desktop.in
+++ b/data/xsession/openbox-kde.desktop.in
@@ -2,6 +2,6 @@
 Name=KDE/Openbox
 Comment=Use the Openbox window manager inside of the K Desktop Environment
 Exec=@bindir@/openbox-kde-session
-TryExec=startkde
+TryExec=/usr/bin/startplasma-x11
 Icon=openbox
 Type=Application
