diff -rupN open-zwave.orig/cpp/build/Makefile open-zwave/cpp/build/Makefile
--- open-zwave.orig/cpp/build/Makefile	2019-07-18 09:49:01.185154576 +0200
+++ open-zwave/cpp/build/Makefile	2019-07-18 09:58:36.413855845 +0200
@@ -23,7 +23,7 @@ DEBUG_LDFLAGS	:= -g
 
 #default Libraries we should link with
 ifneq ($(UNAME),FreeBSD)
-LIBS += -lresolv
+LIBS += -lresolv -ltinyxml
 endif
 
 top_srcdir := $(abspath $(dir $(lastword $(MAKEFILE_LIST)))../../)
@@ -84,47 +84,19 @@ endif
 #where to put the temporary library
 LIBDIR	?= $(top_builddir)
 
-INCLUDES	:= -I $(top_srcdir)/cpp/src -I $(top_srcdir)/cpp/tinyxml/
+INCLUDES	:= -I $(top_srcdir)/cpp/src
 
 
 ifeq ($(USE_HID),1)
-INCLUDES += -I $(top_srcdir)/cpp/hidapi/hidapi/
+INCLUDES += -I /usr/include/hidapi/
 CFLAGS += -DUSE_HID
-
-ifeq ($(UNAME),Darwin)
-SOURCES_HIDAPI =$(top_srcdir)/cpp/hidapi/mac
-else ifeq ($(UNAME),FreeBSD)
-SOURCES_HIDAPI =$(top_srcdir)/cpp/hidapi/libusb
-else ifeq ($(UNAME),NetBSD)
-SOURCES_HIDAPI =$(top_srcdir)/cpp/hidapi/libusb
-else ifeq ($(UNAME),SunOS)
-SOURCES_HIDAPI =$(top_srcdir)/cpp/hidapi/libusb
-else
-SOURCES_HIDAPI =$(top_srcdir)/cpp/hidapi/linux
-LIBS += -ludev
-endif
+LIBS += -ludev -lhidapi-libusb
 endif # USE_HID
 
-SOURCES		:= $(top_srcdir)/cpp/src $(top_srcdir)/cpp/src/command_classes $(top_srcdir)/cpp/tinyxml \
-	$(top_srcdir)/cpp/src/value_classes $(top_srcdir)/cpp/src/platform $(top_srcdir)/cpp/src/platform/unix $(SOURCES_HIDAPI) $(top_srcdir)/cpp/src/aes/
-VPATH = $(top_srcdir)/cpp/src:$(top_srcdir)/cpp/src/command_classes:$(top_srcdir)/cpp/tinyxml:\
-	$(top_srcdir)/cpp/src/value_classes:$(top_srcdir)/cpp/src/platform:$(top_srcdir)/cpp/src/platform/unix:$(SOURCES_HIDAPI):$(top_srcdir)/cpp/src/aes/
-
-tinyxml := $(notdir $(wildcard $(top_srcdir)/cpp/tinyxml/*.cpp))
-
-ifeq ($(USE_HID),1)
-ifeq ($(UNAME),Darwin)
-hidapi := $(notdir $(wildcard $(top_srcdir)/cpp/hidapi/mac/*.c))
-else ifeq ($(UNAME),FreeBSD)
-hidapi := $(notdir $(wildcard $(top_srcdir)/cpp/hidapi/libusb/*.c))
-else ifeq ($(UNAME),NetBSD)
-hidapi := $(notdir $(wildcard $(top_srcdir)/cpp/hidapi/libusb/*.c))
-else ifeq ($(UNAME),SunOS)
-hidapi := $(notdir $(wildcard $(top_srcdir)/cpp/hidapi/libusb/*.c))
-else
-hidapi := $(notdir $(wildcard $(top_srcdir)/cpp/hidapi/linux/*.c)) # we do not want the libusb version
-endif
-endif # USE_HID
+SOURCES		:= $(top_srcdir)/cpp/src $(top_srcdir)/cpp/src/command_classes \
+	$(top_srcdir)/cpp/src/value_classes $(top_srcdir)/cpp/src/platform $(top_srcdir)/cpp/src/platform/unix $(top_srcdir)/cpp/src/aes/
+VPATH = $(top_srcdir)/cpp/src:$(top_srcdir)/cpp/src/command_classes:\
+	$(top_srcdir)/cpp/src/value_classes:$(top_srcdir)/cpp/src/platform:$(top_srcdir)/cpp/src/platform/unix:$(top_srcdir)/cpp/src/aes/
 
 ifeq ($(BITBAKE_ENV),1)
 ar_option := "rc"
@@ -147,8 +119,6 @@ printversion:
 	@echo "Building OpenZWave Version $(GITVERSION)"	
 
 
--include $(patsubst %.cpp,$(DEPDIR)/%.d,$(tinyxml))
--include $(patsubst %.c,$(DEPDIR)/%.d,$(hidapi))
 -include $(patsubst %.cpp,$(DEPDIR)/%.d,$(cclasses))
 -include $(patsubst %.cpp,$(DEPDIR)/%.d,$(vclasses))
 -include $(patsubst %.cpp,$(DEPDIR)/%.d,$(pform))
@@ -167,9 +137,7 @@ $(top_srcdir)/cpp/src/vers.cpp:
 
 #$(OBJDIR)/vers.o:	$(top_builddir)/vers.cpp
 
-$(LIBDIR)/libopenzwave.a:	$(patsubst %.cpp,$(OBJDIR)/%.o,$(tinyxml)) \
-			$(patsubst %.c,$(OBJDIR)/%.o,$(hidapi)) \
-			$(patsubst %.c,$(OBJDIR)/%.o,$(aes)) \
+$(LIBDIR)/libopenzwave.a:	$(patsubst %.c,$(OBJDIR)/%.o,$(aes)) \
 			$(patsubst %.cpp,$(OBJDIR)/%.o,$(cclasses)) \
 			$(patsubst %.cpp,$(OBJDIR)/%.o,$(vclasses)) \
 			$(patsubst %.cpp,$(OBJDIR)/%.o,$(pform)) \
@@ -179,9 +147,7 @@ $(LIBDIR)/libopenzwave.a:	$(patsubst %.c
 	@$(AR) $(ar_option) $@ $+
 	@$(RANLIB) $@
 
-$(LIBDIR)/$(SHARED_LIB_NAME):	$(patsubst %.cpp,$(OBJDIR)/%.o,$(tinyxml)) \
-			$(patsubst %.c,$(OBJDIR)/%.o,$(hidapi)) \
-			$(patsubst %.c,$(OBJDIR)/%.o,$(aes)) \
+$(LIBDIR)/$(SHARED_LIB_NAME):	$(patsubst %.c,$(OBJDIR)/%.o,$(aes)) \
 			$(patsubst %.cpp,$(OBJDIR)/%.o,$(cclasses)) \
 			$(patsubst %.cpp,$(OBJDIR)/%.o,$(vclasses)) \
 			$(patsubst %.cpp,$(OBJDIR)/%.o,$(pform)) \
