--- performous-1.0/game/ffmpeg.cc.orig	2016-02-17 20:37:00.157658695 +0000
+++ performous-1.0/game/ffmpeg.cc	2016-02-17 20:38:20.287291460 +0000
@@ -116,7 +116,7 @@
 		height = cc->height;
 		m_swsContext = sws_getContext(
 		  cc->width, cc->height, cc->pix_fmt,
-		  width, height, PIX_FMT_RGB24,
+		  width, height, AV_PIX_FMT_RGB24,
 		  SWS_POINT, nullptr, nullptr, nullptr);
 		break;
 	default:  // Should never be reached but avoids compile warnings
@@ -187,7 +187,7 @@
 		if (packetSize < 0) throw std::logic_error("negative packet size?!");
 		if (m_quit || m_seekTarget == m_seekTarget) return;
 		if (packet.stream_index != m_streamId) return;
-		boost::shared_ptr<AVFrame> frame(avcodec_alloc_frame(), &av_free);
+		boost::shared_ptr<AVFrame> frame(av_frame_alloc(), &av_free);
 		int frameFinished = 0;
 		int decodeSize = (m_mediaType == AVMEDIA_TYPE_VIDEO ?
 		  avcodec_decode_video2(m_codecContext, frame.get(), &frameFinished, &packet) :
