diff -ru performous-1.1.orig/game/backgrounds.hh performous-1.1/game/backgrounds.hh
--- performous-1.1.orig/game/backgrounds.hh	2019-01-15 10:39:54.147584168 +0000
+++ performous-1.1/game/backgrounds.hh	2019-01-15 10:40:41.284376151 +0000
@@ -7,6 +7,7 @@
 #include <boost/scoped_ptr.hpp>
 #include <boost/thread/mutex.hpp>
 #include <boost/thread/thread.hpp>
+#include <boost/noncopyable.hpp>
 #include <vector>
 
 /// songs class for songs screen
diff -ru performous-1.1.orig/game/songs.hh performous-1.1/game/songs.hh
--- performous-1.1.orig/game/songs.hh	2019-01-15 10:39:54.147584168 +0000
+++ performous-1.1/game/songs.hh	2019-01-15 10:41:03.901102679 +0000
@@ -6,6 +6,7 @@
 #include <boost/scoped_ptr.hpp>
 #include <boost/thread/mutex.hpp>
 #include <boost/thread/thread.hpp>
+#include <boost/noncopyable.hpp>
 #include <set>
 #include <sstream>
 #include <vector>
