--- PokerTH-1.1.1-src/pokerth_db.pro.orig	2016-05-12 20:20:57.336018308 +0000
+++ PokerTH-1.1.1-src/pokerth_db.pro	2016-05-12 20:21:14.335959204 +0000
@@ -49,7 +49,6 @@
 !win32{
 	##### My release static build options
 	#QMAKE_CXXFLAGS += -ffunction-sections -fdata-sections
-	INCLUDEPATH += $${PREFIX}/include
 }
 
 mac{
--- PokerTH-1.1.1-src/pokerth_lib.pro.orig	2016-05-12 20:24:53.581864921 +0000
+++ PokerTH-1.1.1-src/pokerth_lib.pro	2016-05-12 20:25:20.441771862 +0000
@@ -243,7 +243,7 @@
 !win32{
 	##### My release static build options
 	#QMAKE_CXXFLAGS += -ffunction-sections -fdata-sections
-	INCLUDEPATH += $${PREFIX}/include /opt/gsasl/include
+	INCLUDEPATH += /opt/gsasl/include
 }
 
 mac{
--- PokerTH-1.1.1-src/pokerth_game.pro.orig	2016-05-12 21:02:08.527791266 +0000
+++ PokerTH-1.1.1-src/pokerth_game.pro	2016-05-12 21:02:19.027758386 +0000
@@ -402,7 +402,6 @@
 	# #### My release static build options
 	# QMAKE_CXXFLAGS += -ffunction-sections -fdata-sections
 	# QMAKE_LFLAGS += -Wl,--gc-sections
-	INCLUDEPATH += $${PREFIX}/include
 	QMAKE_LIBDIR += lib
 	!android{
 		LIBPATH += $${PREFIX}/lib /opt/gsasl/lib
--- PokerTH-1.1.1-src/pokerth_server.pro.orig	2016-05-12 21:10:02.469679181 +0000
+++ PokerTH-1.1.1-src/pokerth_server.pro	2016-05-12 21:10:34.106251260 +0000
@@ -174,7 +174,6 @@
 	#QMAKE_LFLAGS += -Wl,--gc-sections
 
 	LIBPATH += lib $${PREFIX}/lib /opt/gsasl/lib
-	INCLUDEPATH += $${PREFIX}/include
 	LIB_DIRS = $${PREFIX}/lib $${PREFIX}/lib64 $$system(qmake -query QT_INSTALL_LIBS)
 	BOOST_FS = boost_filesystem boost_filesystem-mt
 	BOOST_THREAD = boost_thread boost_thread-mt
