diff -upr tensorflow-1.9.0.orig/tensorflow/workspace.bzl tensorflow-1.9.0/tensorflow/workspace.bzl
--- tensorflow-1.9.0.orig/tensorflow/workspace.bzl	2018-07-30 13:22:06.387368887 +0300
+++ tensorflow-1.9.0/tensorflow/workspace.bzl	2018-07-30 13:26:15.734924242 +0300
@@ -335,6 +335,7 @@ def tf_workspace(path_prefix="", tf_repo
       ],
       sha256 = "50a5753995b3142627ac55cfd496cebc418a2e575ca0236e29033c67bd5665f4",
       strip_prefix = "protobuf-3.6.0",
+      patch_file = clean_dep("//third_party:protobuf-python37.patch"),
   )
 
   # We need to import the protobuf library under the names com_google_protobuf
@@ -348,6 +349,7 @@ def tf_workspace(path_prefix="", tf_repo
       ],
       sha256 = "50a5753995b3142627ac55cfd496cebc418a2e575ca0236e29033c67bd5665f4",
       strip_prefix = "protobuf-3.6.0",
+      patch_file = clean_dep("//third_party:protobuf-python37.patch"),
   )
 
   tf_http_archive(
@@ -358,6 +360,7 @@ def tf_workspace(path_prefix="", tf_repo
       ],
       sha256 = "50a5753995b3142627ac55cfd496cebc418a2e575ca0236e29033c67bd5665f4",
       strip_prefix = "protobuf-3.6.0",
+      patch_file = clean_dep("//third_party:protobuf-python37.patch"),
   )
 
   tf_http_archive(
