diff --git a/tensorflow/workspace.bzl b/tensorflow/workspace.bzl
index dff151246a..d3fe95ec04 100755
--- a/tensorflow/workspace.bzl
+++ b/tensorflow/workspace.bzl
@@ -349,11 +349,11 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
     )
 
     PROTOBUF_URLS = [
-        "https://mirror.bazel.build/github.com/protocolbuffers/protobuf/archive/v3.6.1.2.tar.gz",
-        "https://github.com/protocolbuffers/protobuf/archive/v3.6.1.2.tar.gz",
+        "https://mirror.bazel.build/github.com/protocolbuffers/protobuf/archive/v3.6.1.3.tar.gz",
+        "https://github.com/protocolbuffers/protobuf/archive/v3.6.1.3.tar.gz",
     ]
-    PROTOBUF_SHA256 = "2244b0308846bb22b4ff0bcc675e99290ff9f1115553ae9671eba1030af31bc0"
-    PROTOBUF_STRIP_PREFIX = "protobuf-3.6.1.2"
+    PROTOBUF_SHA256 = "73fdad358857e120fd0fa19e071a96e15c0f23bb25f85d3f7009abfd4f264a2a"
+    PROTOBUF_STRIP_PREFIX = "protobuf-3.6.1.3"
 
     tf_http_archive(
         name = "protobuf_archive",
@@ -364,6 +364,7 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
             "//third_party/systemlibs:protobuf.bzl": "protobuf.bzl",
         },
         urls = PROTOBUF_URLS,
+        patch_file = clean_dep("//third_party:protobuf_temp_fix_cuda10.1.patch"),
     )
 
     # We need to import the protobuf library under the names com_google_protobuf
@@ -378,6 +379,7 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
             "//third_party/systemlibs:protobuf.bzl": "protobuf.bzl",
         },
         urls = PROTOBUF_URLS,
+        patch_file = clean_dep("//third_party:protobuf_temp_fix_cuda10.1.patch"),
     )
 
     tf_http_archive(
@@ -389,6 +391,7 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
             "//third_party/systemlibs:protobuf.bzl": "protobuf.bzl",
         },
         urls = PROTOBUF_URLS,
+        patch_file = clean_dep("//third_party:protobuf_temp_fix_cuda10.1.patch"),
     )
 
     tf_http_archive(
