diff --git a/meshroom/ui/__main__.py b/meshroom/ui/__main__.py
index 5914f1ad..aaff2737 100644
--- a/meshroom/ui/__main__.py
+++ b/meshroom/ui/__main__.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python
+
 import signal
 import sys
 import meshroom
diff --git a/meshroom/ui/reconstruction.py b/meshroom/ui/reconstruction.py
index c37ffafc..9d5373a2 100755
--- a/meshroom/ui/reconstruction.py
+++ b/meshroom/ui/reconstruction.py
@@ -3,7 +3,7 @@ import logging
 import math
 import os
 from threading import Thread
-from collections import Iterable
+from collections.abc import Iterable
 
 from PySide2.QtCore import QObject, Slot, Property, Signal, QUrl, QSizeF
 from PySide2.QtGui import QMatrix4x4, QMatrix3x3, QQuaternion, QVector3D, QVector2D
