commit ae539894500a744ec87d9a7f2d6aca5306fc26a1
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Fri Jun 2 15:03:59 2017 +0800

    Add support for Qt 5.9+
    
    Change-Id: I5596a429ed3e0f4e7522a479cd420c57f0a88dc9

diff --git a/platformplugin/windoweventhook.cpp b/platformplugin/windoweventhook.cpp
index 2bf3e36..92ae529 100644
--- a/platformplugin/windoweventhook.cpp
+++ b/platformplugin/windoweventhook.cpp
@@ -204,7 +204,12 @@ void WindowEventHook::handleFocusInEvent(const xcb_focus_in_event_t *event)
     if (relayFocusToModalWindow(w, xcbWindow->connection()))
         return;
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 9, 0)
+    xcbWindow->connection()->setFocusWindow(w);
+#else
     xcbWindow->connection()->setFocusWindow(static_cast<QXcbWindow *>(w->handle()));
+#endif
+
     QWindowSystemInterface::handleWindowActivated(w, Qt::ActiveWindowFocusReason);
 }
 
