commit 71884b1d65b6cf5835948a35f6083f1f85d9bbed
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Mon Apr 4 04:07:51 2022 +0300

    fix: remove unused gio dependencies

diff --git a/README.md b/README.md
index e6aad45a..4b1e23bb 100644
--- a/README.md
+++ b/README.md
@@ -24,7 +24,7 @@ Deepin Image Viewer is an image viewing tool with fashion interface and smooth p
 
 ## Installation
 
-sudo apt install cmake qtbase5-dev pkg-config libexif-dev libqt5svg5-dev libqt5x11extras5-dev libsqlite3-dev qttools5-dev-tools qttools5-dev libxcb-util0-dev libstartup-notification0-dev libraw-dev libfreeimage-dev libqt5opengl5-dev qtbase5-private-dev qtmultimedia5-dev x11proto-xext-dev libmtdev-dev libegl1-mesa-dev libudev-dev libfontconfig1-dev libfreetype6-dev libglib2.0-dev libxrender-dev libdtkwidget-dev libdtkwidget5-bin libdtkcore5-bin libgio-qt-dev libudisks2-qt5-dev libimageeditor-dev
+sudo apt install cmake qtbase5-dev pkg-config libexif-dev libqt5svg5-dev libqt5x11extras5-dev libsqlite3-dev qttools5-dev-tools qttools5-dev libxcb-util0-dev libstartup-notification0-dev libraw-dev libfreeimage-dev libqt5opengl5-dev qtbase5-private-dev qtmultimedia5-dev x11proto-xext-dev libmtdev-dev libegl1-mesa-dev libudev-dev libfontconfig1-dev libfreetype6-dev libglib2.0-dev libxrender-dev libdtkwidget-dev libdtkwidget5-bin libdtkcore5-bin libudisks2-qt5-dev libimageeditor-dev
 
 ### Build from source code
 
diff --git a/debian/control b/debian/control
index 51b2130e..68c32d92 100644
--- a/debian/control
+++ b/debian/control
@@ -6,7 +6,7 @@ Build-Depends: debhelper (>= 11), cmake, qtbase5-dev, pkg-config,libexif-dev, li
  libraw-dev,libfreeimage-dev, libqt5opengl5-dev, qtbase5-private-dev,
  qtmultimedia5-dev, x11proto-xext-dev, libmtdev-dev, libegl1-mesa-dev,
  libudev-dev, libfontconfig1-dev, libfreetype6-dev, libglib2.0-dev,
- libxrender-dev, libdtkwidget-dev, libdtkwidget5-bin,libdtkcore5-bin,libgio-qt-dev,libudisks2-qt5-dev,libimageeditor-dev
+ libxrender-dev, libdtkwidget-dev, libdtkwidget5-bin,libdtkcore5-bin,libudisks2-qt5-dev,libimageeditor-dev
 Standards-Version: 3.9.8
 Homepage: http://www.deepin.org
 
diff --git a/rpm/deepin-image-viewer.spec b/rpm/deepin-image-viewer.spec
index e14694d1..b0e20b7a 100644
--- a/rpm/deepin-image-viewer.spec
+++ b/rpm/deepin-image-viewer.spec
@@ -19,7 +19,6 @@ BuildRequires:  pkgconfig(Qt5Sql)
 BuildRequires:  pkgconfig(Qt5Svg)
 BuildRequires:  pkgconfig(Qt5X11Extras)
 BuildRequires:  pkgconfig(dtkwidget) >= 2.0.6
-BuildRequires:  pkgconfig(gio-unix-2.0)
 BuildRequires:  pkgconfig(libraw)
 BuildRequires:  pkgconfig(libexif)
 BuildRequires:  pkgconfig(libstartup-notification-1.0)
@@ -27,7 +26,6 @@ BuildRequires:  pkgconfig(xcb-util)
 BuildRequires:  pkgconfig(xext)
 BuildRequires:  pkgconfig(Qt5Multimedia) 
 BuildRequires:  udisks2-qt5-devel
-BuildRequires:  libgio-qt libgio-qt-devel
 BuildRequires:  desktop-file-utils
 BuildRequires:  libappstream-glib
 Requires:       hicolor-icon-theme
@@ -146,4 +144,4 @@ export PATH=%{_qt5_bindir}:$PATH
 - Fixed build dependecies
  
 * Sat Dec 10 2016 Jaroslav <cz.guardian@gmail.com> Stepanek 1.1.3-1
-- Initial package build
\ No newline at end of file
+- Initial package build
diff --git a/src/src.pro b/src/src.pro
index 979471a0..1a4de930 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -10,8 +10,8 @@ QT += core gui dbus concurrent svg  printsupport sql
 greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
 CONFIG -= app_bundle
 CONFIG += c++11 link_pkgconfig
-PKGCONFIG +=   libexif dtkwidget  gio-qt
-# PKGCONFIG += xext x11 gio-unix-2.0
+PKGCONFIG +=   libexif dtkwidget
+# PKGCONFIG += xext x11
  QT += dtkwidget
  QT += dbus
 #CONFIG += object_parallel_to_source
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index b14a9b2b..05115cf2 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -72,8 +72,6 @@ find_package(Qt5LinguistTools)
 pkg_check_modules(3rd_lib REQUIRED
     dtkcore
     dtkwidget
-    gio-qt
-    gio-unix-2.0
 #    freeimage
         )
 
@@ -118,8 +116,6 @@ target_link_libraries(
     X11
     Xext
     exif
-#    gio-unix-2.0
-    gio-qt
     udisks2-qt5
     freeimage
     pthread
commit 93fe7f6825d34f3cdcdbef6969fd285d82d01aa6
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Mon Apr 4 04:09:58 2022 +0300

    fix: remove unused udisks dependencies

diff --git a/README.md b/README.md
index 4b1e23bb..c7d66c26 100644
--- a/README.md
+++ b/README.md
@@ -24,7 +24,7 @@ Deepin Image Viewer is an image viewing tool with fashion interface and smooth p
 
 ## Installation
 
-sudo apt install cmake qtbase5-dev pkg-config libexif-dev libqt5svg5-dev libqt5x11extras5-dev libsqlite3-dev qttools5-dev-tools qttools5-dev libxcb-util0-dev libstartup-notification0-dev libraw-dev libfreeimage-dev libqt5opengl5-dev qtbase5-private-dev qtmultimedia5-dev x11proto-xext-dev libmtdev-dev libegl1-mesa-dev libudev-dev libfontconfig1-dev libfreetype6-dev libglib2.0-dev libxrender-dev libdtkwidget-dev libdtkwidget5-bin libdtkcore5-bin libudisks2-qt5-dev libimageeditor-dev
+sudo apt install cmake qtbase5-dev pkg-config libexif-dev libqt5svg5-dev libqt5x11extras5-dev libsqlite3-dev qttools5-dev-tools qttools5-dev libxcb-util0-dev libstartup-notification0-dev libraw-dev libfreeimage-dev libqt5opengl5-dev qtbase5-private-dev qtmultimedia5-dev x11proto-xext-dev libmtdev-dev libegl1-mesa-dev libudev-dev libfontconfig1-dev libfreetype6-dev libglib2.0-dev libxrender-dev libdtkwidget-dev libdtkwidget5-bin libdtkcore5-bin libimageeditor-dev
 
 ### Build from source code
 
diff --git a/debian/control b/debian/control
index 68c32d92..7bd9569a 100644
--- a/debian/control
+++ b/debian/control
@@ -6,7 +6,7 @@ Build-Depends: debhelper (>= 11), cmake, qtbase5-dev, pkg-config,libexif-dev, li
  libraw-dev,libfreeimage-dev, libqt5opengl5-dev, qtbase5-private-dev,
  qtmultimedia5-dev, x11proto-xext-dev, libmtdev-dev, libegl1-mesa-dev,
  libudev-dev, libfontconfig1-dev, libfreetype6-dev, libglib2.0-dev,
- libxrender-dev, libdtkwidget-dev, libdtkwidget5-bin,libdtkcore5-bin,libudisks2-qt5-dev,libimageeditor-dev
+ libxrender-dev, libdtkwidget-dev, libdtkwidget5-bin,libdtkcore5-bin,libimageeditor-dev
 Standards-Version: 3.9.8
 Homepage: http://www.deepin.org
 
diff --git a/rpm/deepin-image-viewer.spec b/rpm/deepin-image-viewer.spec
index b0e20b7a..c4cec029 100644
--- a/rpm/deepin-image-viewer.spec
+++ b/rpm/deepin-image-viewer.spec
@@ -25,7 +25,6 @@ BuildRequires:  pkgconfig(libstartup-notification-1.0)
 BuildRequires:  pkgconfig(xcb-util)
 BuildRequires:  pkgconfig(xext)
 BuildRequires:  pkgconfig(Qt5Multimedia) 
-BuildRequires:  udisks2-qt5-devel
 BuildRequires:  desktop-file-utils
 BuildRequires:  libappstream-glib
 Requires:       hicolor-icon-theme
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 05115cf2..f7478eab 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -116,7 +116,6 @@ target_link_libraries(
     X11
     Xext
     exif
-    udisks2-qt5
     freeimage
     pthread
     imageviewer
