diff -aur xmonad-contrib-0.11.2/XMonad/Layout/MultiToggle.hs xmonad-contrib-new/XMonad/Layout/MultiToggle.hs
--- xmonad-contrib-0.11.2/XMonad/Layout/MultiToggle.hs	2013-07-16 04:52:11.000000000 +0200
+++ xmonad-contrib-new/XMonad/Layout/MultiToggle.hs	2014-04-27 21:26:20.746206511 +0200
@@ -99,7 +99,7 @@
     transform :: (LayoutClass l a) => t -> l a ->
         (forall l'. (LayoutClass l' a) => l' a -> (l' a -> l a) -> b) -> b
 
-data (LayoutClass l a) => EL l a = forall l'. (LayoutClass l' a) => EL (l' a) (l' a -> l a)
+data  EL l a = forall l'. (LayoutClass l' a) => EL (l' a) (l' a -> l a)
 
 unEL :: (LayoutClass l a) => EL l a -> (forall l'. (LayoutClass l' a) => l' a -> b) -> b
 unEL (EL x _) k = k x
@@ -159,7 +159,7 @@
 infixr 0 ??
 -- | Prepend an element to a heterogeneous list. Used to build transformer
 -- tables for 'mkToggle'.
-(??) :: (HList b w) => a -> b -> HCons a b
+(??) ::  a -> b -> HCons a b
 (??) = HCons
 
 -- | Construct a singleton transformer table.
