diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0f1c0e9..a862fac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,22 +24,7 @@ pkg_check_modules(RSVG REQUIRED librsvg-2.0)
 pkg_check_modules(WNCK REQUIRED libwnck-3.0)
 pkg_check_modules(FONTCONFIG REQUIRED fontconfig)
 pkg_check_modules(GLIB REQUIRED glib-2.0)
-
-ExternalProject_Add(Vte-0.29
-  SOURCE_DIR ${CMAKE_SOURCE_DIR}/3rdparty/vte-0.50.1
-  CONFIGURE_COMMAND ./configure --enable-static
-  BUILD_COMMAND make
-  INSTALL_COMMAND ""
-  BUILD_IN_SOURCE 1
-  )
-
-ExternalProject_Add(zssh-1.5c
-  SOURCE_DIR ${CMAKE_SOURCE_DIR}/3rdparty/zssh-1.5c
-  CONFIGURE_COMMAND ./configure
-  BUILD_COMMAND make
-  INSTALL_COMMAND ""
-  BUILD_IN_SOURCE 1
-  )
+pkg_check_modules(VTE REQUIRED vte-2.91)
 
 set(CFLAGS
 	${GIO_CFLAGS}
@@ -60,9 +45,8 @@ set(INCLUDE_PATHS
 	${RSVG_INCLUDE_DIRS}
 	${JsonGlib_INCLUDE_DIRS}
 	${LIBSECRET_INCLUDE_DIRS}
-	${CMAKE_SOURCE_DIR}/3rdparty/vte-0.50.1/src
+	${VTE_INCLUDE_DIRS}
 )
-find_library(VTE_LIB NAMES libvte-2.91.a tcmalloc_minimal)
 include_directories(${INCLUDE_PATHS})
 set(LIBS
 	${GIO_LIBRARIES}
@@ -72,17 +56,12 @@ set(LIBS
 	${RSVG_LIBRARIES}
 	${JsonGlib_LIBRARIES}
 	${LIBSECRET_LIBRARIES}
-	${ZLIB_LIBRARIES}
+	${VTE_LIBRARIES}
 	${X11_LIBRARIES}
 	${XTST_LIBRARIES}
 	${FONTCONFIG_LIBRARY}
 	${M_LIBRARIES}
-	stdc++
 	m
-	z
-	pthread
-	gnutls
-	pcre2-8
 )
 vala_precompile(VALA_C
 	./lib/animation.vala
@@ -98,7 +77,7 @@ vala_precompile(VALA_C
 	./widget/add_button.vala
 	./widget/appbar.vala
 	./widget/check_button.vala
-	./widget/click_event_box.vala 
+	./widget/click_event_box.vala
 	./widget/command_button.vala
 	./widget/command_dialog.vala
 	./widget/command_panel.vala
@@ -109,11 +88,11 @@ vala_precompile(VALA_C
 	./widget/cursor_toggle_button.vala
 	./widget/dialog.vala
 	./widget/dialog_button.vala
-	./widget/encoding_button.vala 
-	./widget/encoding_panel.vala 
-	./widget/entry.vala 
-	./widget/entry_menu.vala 
-	./widget/file_button.vala 
+	./widget/encoding_button.vala
+	./widget/encoding_panel.vala
+	./widget/entry.vala
+	./widget/entry_menu.vala
+	./widget/file_button.vala
 	./widget/image_button.vala
 	./widget/window_button.vala
 	./widget/panel_button.vala
@@ -142,7 +121,7 @@ vala_precompile(VALA_C
 	./widget/window.vala
 	./widget/window_event_area.vala
 	./widget/workspace.vala
-	./widget/workspace_manager.vala 
+	./widget/workspace_manager.vala
 	main.vala
 
 OPTIONS
@@ -164,8 +143,7 @@ PACKAGES
 	librsvg-2.0
 )
 add_executable(${target} ${VALA_C} ${CMAKE_SOURCE_DIR}/project_path.c ${CMAKE_SOURCE_DIR}/lib/font.c)
-target_link_libraries(${target} ${LIBS} ${CMAKE_SOURCE_DIR}/3rdparty/vte-0.50.1/src/.libs/libvte-2.91.a)
-add_dependencies(${target} Vte-0.29)
+target_link_libraries(${target} ${LIBS})
 
 FILE(GLOB children LIST_DIRECTORIES true po/*)
 SET(dirlist "")
@@ -182,7 +160,7 @@ foreach(lang ${dirlist})
 endforeach()
 
 INSTALL (TARGETS ${target} DESTINATION bin)
-INSTALL (PROGRAMS ssh_login.sh ${CMAKE_SOURCE_DIR}/3rdparty/zssh-1.5c/zssh DESTINATION lib/${target})
+INSTALL (PROGRAMS ssh_login.sh DESTINATION lib/${target})
 INSTALL (FILES hicolor/32x32/deepin-terminal.png DESTINATION share/icons/hicolor/32x32/apps)
 INSTALL (FILES hicolor/48x48/deepin-terminal.png DESTINATION share/icons/hicolor/48x48/apps)
 INSTALL (FILES hicolor/96x96/deepin-terminal.png DESTINATION share/icons/hicolor/96x96/apps)
