From 0b5406554e2304e012d975ee746e4ec27896701d Mon Sep 17 00:00:00 2001
From: Sebastian Ramacher <sebastian+dev@ramacher.at>
Date: Sat, 3 Sep 2016 12:00:49 +0200
Subject: [PATCH] Revert unintended changes

Signed-off-by: Sebastian Ramacher <sebastian+dev@ramacher.at>
---
 data/girara-post-3.20.css_t | 6 ------
 1 file changed, 0 insertions(+), 6 deletions(-)

diff --git a/data/girara-post-3.20.css_t b/data/girara-post-3.20.css_t
index 29758de..1fdc4d2 100644
--- a/data/girara-post-3.20.css_t
+++ b/data/girara-post-3.20.css_t
@@ -5,12 +5,6 @@
   font-family: @font-family@;
   font-size: @font-size@;
   font-weight: @font-weight@;
-  box-shadow: none;
-  border-style: none;
-  margin: 0px;
-  padding: @bottombox-padding1@px @bottombox-padding2@px @bottombox-padding3@px
-    @bottombox-padding4@px;
-
 }
 
 /* Scrollbar */
--
libgit2 0.24.0

