diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2019-04-16 01:18:21.000000000 +0200
+++ b/CMakeLists.txt	2019-04-16 21:10:10.934681045 +0200
@@ -251,14 +251,24 @@
             DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig/")
     endif()
     install(FILES include/rtosc/rtosc.h
-        include/rtosc/thread-link.h
-        include/rtosc/ports.h
+        include/rtosc/arg-val-cmp.h
+        include/rtosc/arg-val-math.h
+        include/rtosc/automations.h
+        include/rtosc/bundle-foreach.h
+        include/rtosc/default-value.h
         include/rtosc/miditable.h
         include/rtosc/port-sugar.h
-        include/rtosc/undo-history.h
+        include/rtosc/ports-runtime.h
+        include/rtosc/ports.h
+        include/rtosc/pretty-format.h
+        include/rtosc/rtosc-time.h
+        include/rtosc/rtosc-version.h
+        include/rtosc/savefile.h
         include/rtosc/subtree-serialize.h
+        include/rtosc/thread-link.h
         include/rtosc/typed-message.h
-        include/rtosc/rtosc-version.h
+        include/rtosc/typestring.hh
+        include/rtosc/undo-history.h
         DESTINATION include/rtosc)
     install(TARGETS rtosc rtosc-cpp
         DESTINATION ${CMAKE_INSTALL_LIBDIR})
