diff --git a/stringio.gemspec b/stringio.gemspec
index 1015d26..80bad5d 100644
--- a/stringio.gemspec
+++ b/stringio.gemspec
@@ -1,18 +1,9 @@
 # -*- coding: utf-8 -*-
 # frozen_string_literal: true
 
-source_version = ["", "ext/stringio/"].find do |dir|
-  begin
-    break File.open(File.join(__dir__, "#{dir}stringio.c")) {|f|
-      f.gets("\n#define STRINGIO_VERSION ")
-      f.gets[/\s*"(.+)"/, 1]
-    }
-  rescue Errno::ENOENT
-  end
-end
 Gem::Specification.new do |s|
   s.name = "stringio"
-  s.version = source_version
+  s.version = "3.0.2"
 
   s.required_rubygems_version = Gem::Requirement.new(">= 2.6")
   s.require_paths = ["lib"]
