diff --git a/test/lib/core_assertions.rb b/test/lib/core_assertions.rb
index bac3856..e668ac2 100644
--- a/test/lib/core_assertions.rb
+++ b/test/lib/core_assertions.rb
@@ -45,7 +45,6 @@ module Test
       def assert_in_out_err(args, test_stdin = "", test_stdout = [], test_stderr = [], message = nil,
                             success: nil, **opt)
         args = Array(args).dup
-        args.insert((Hash === args[0] ? 1 : 0), '--disable=gems')
         stdout, stderr, status = EnvUtil.invoke_ruby(args, test_stdin, true, true, **opt)
         desc = FailDesc[status, message, stderr]
         if block_given?
@@ -110,7 +109,6 @@ module Test
         token_re = Regexp.quote(token)
         envs = args.shift if Array === args and Hash === args.first
         args = [
-          "--disable=gems",
           "-r", File.expand_path("../memory_status", __FILE__),
           *args,
           "-v", "-",
@@ -275,7 +273,7 @@ BEGIN {
 #{line -= __LINE__; src}
 eom
         args = args.dup
-        args.insert((Hash === args.first ? 1 : 0), "-w", "--disable=gems", *$:.map {|l| "-I#{l}"})
+        args.insert((Hash === args.first ? 1 : 0), "-w", *$:.map {|l| "-I#{l}"})
         stdout, stderr, status = EnvUtil.invoke_ruby(args, src, capture_stdout, true, **opt)
       ensure
         if res_c
