diff --git a/etc.gemspec b/etc.gemspec
index 7d687e3..9d68939 100644
--- a/etc.gemspec
+++ b/etc.gemspec
@@ -1,19 +1,8 @@
 # frozen_string_literal: true
 
-version = ["", "ext/etc/"].find do |dir|
-  begin
-    break File.open(File.expand_path("../#{dir}/etc.c", __FILE__)) do |f|
-      f.gets "\n#define RUBY_ETC_VERSION "
-      f.gets[/"(.+)"/, 1]
-    end
-  rescue
-    next
-  end
-end
-
 Gem::Specification.new do |spec|
   spec.name          = "etc"
-  spec.version       = version
+  spec.version       = "1.3.0"
   spec.authors       = ["Yukihiro Matsumoto"]
   spec.email         = ["matz@ruby-lang.org"]
 
