diff --git a/psych.gemspec b/psych.gemspec
index 65dd715..c178af2 100644
--- a/psych.gemspec
+++ b/psych.gemspec
@@ -1,14 +1,9 @@
 # -*- encoding: utf-8 -*-
 # frozen_string_literal: true
 
-version_module = Module.new do
-  version_rb = File.join(__dir__, "lib/psych/versions.rb")
-  module_eval(File.read(version_rb), version_rb)
-end
-
 Gem::Specification.new do |s|
   s.name = "psych"
-  s.version = version_module::Psych::VERSION
+  s.version = "4.0.4"
   s.authors = ["Aaron Patterson", "SHIBATA Hiroshi", "Charles Oliver Nutter"]
   s.email = ["aaron@tenderlovemaking.com", "hsbt@ruby-lang.org", "headius@headius.com"]
   s.summary = "Psych is a YAML parser and emitter"
@@ -59,7 +54,7 @@ DESCRIPTION
       "lib/psych_jars.rb",
       "lib/psych.jar"
     ]
-    s.requirements = "jar org.yaml:snakeyaml, #{version_module::Psych::DEFAULT_SNAKEYAML_VERSION}"
+    s.requirements = "jar org.yaml:snakeyaml, 1.28"
     s.add_dependency 'jar-dependencies', '>= 0.1.7'
   else
     s.extensions = ["ext/psych/extconf.rb"]
