diff --git a/src/setup.py b/src/setup.py
index f7e04e61e9..3ebc34245f 100755
--- a/src/setup.py
+++ b/src/setup.py
@@ -59,14 +59,11 @@ t = time.time()
 
 distributions = ['']
 
-from sage_setup.optional_extension import is_package_installed_and_updated
-
 optional_packages_with_extensions = ['mcqd', 'bliss', 'tdlib', 'primecount',
-                                     'coxeter3', 'fes', 'sirocco', 'meataxe']
+                                     'coxeter3', 'sirocco', 'meataxe']
 
 distributions += ['sage-{}'.format(pkg)
-                  for pkg in optional_packages_with_extensions
-                  if is_package_installed_and_updated(pkg)]
+                  for pkg in optional_packages_with_extensions]
 
 log.warn('distributions = {0}'.format(distributions))
 
 
