--- a/src/sage/libs/pynac/pynac.pxd
+++ b/src/sage/libs/pynac/pynac.pxd
@@ -302,7 +302,6 @@ cdef extern from "sage/libs/pynac/wrap.h":
     bint is_a_fderivative "is_a<GiNaC::fderivative>" (GEx e)
     bint is_a_function "is_a<GiNaC::function>" (GEx e)
     bint is_exactly_a_function "is_exactly_a<GiNaC::function>" (GEx e)
-    bint is_a_ncmul "is_a<GiNaC::ncmul>" (GEx e)
 
     # Arithmetic
     int ginac_error()
--- a/src/sage/symbolic/expression.pyx
+++ b/src/sage/symbolic/expression.pyx
@@ -5536,7 +5536,7 @@ cdef class Expression(CommutativeRingElement):
         cdef unsigned serial
         if is_a_add(self._gobj):
             return add_vararg
-        elif is_a_mul(self._gobj) or is_a_ncmul(self._gobj):
+        elif is_a_mul(self._gobj):
             return mul_vararg
         elif is_a_power(self._gobj):
             return operator.pow
