--- src/sage/libs/singular/decl.pxd.orig	2017-02-21 07:58:27.088118954 +0000
+++ src/sage/libs/singular/decl.pxd	2017-02-21 07:59:01.794846500 +0000
@@ -484,8 +484,8 @@
 
     ring *rDefault(int char , int nvars, char **names)
     ring *rDefault(const n_Procs_s* cf, int nvars, char **names)
-    ring *rDefault(int ch             , int nvars, char **names,int ord_size, int *ord, int *block0, int *block1, int **wvhdl)
-    ring *rDefault(const n_Procs_s* cf, int nvars, char **names,int ord_size, int *ord, int *block0, int *block1, int **wvhdl)
+    ring *rDefault(int ch             , int nvars, char **names,int ord_size, rRingOrder_t *ord, int *block0, int *block1, int **wvhdl)
+    ring *rDefault(const n_Procs_s* cf, int nvars, char **names,int ord_size, rRingOrder_t *ord, int *block0, int *block1, int **wvhdl)
 
 
 
--- src/sage/libs/singular/ring.pyx.orig	2017-02-21 07:24:23.197906897 +0000
+++ src/sage/libs/singular/ring.pyx	2017-02-21 07:29:23.731763098 +0000
@@ -21,7 +21,7 @@
 from sage.libs.singular.decl cimport number, poly, ring, currRing
 from sage.libs.singular.decl cimport rChangeCurrRing, rCopy0, rComplete, rDelete, idInit
 from sage.libs.singular.decl cimport omAlloc0, omStrDup, omAlloc, omAlloc0Bin,  sip_sring_bin, rnumber_bin
-from sage.libs.singular.decl cimport ringorder_dp, ringorder_Dp, ringorder_lp, ringorder_rp, ringorder_ds, ringorder_Ds, ringorder_ls, ringorder_M, ringorder_C, ringorder_wp, ringorder_Wp, ringorder_ws, ringorder_Ws, ringorder_a
+from sage.libs.singular.decl cimport ringorder_dp, ringorder_Dp, ringorder_lp, ringorder_rp, ringorder_ds, ringorder_Ds, ringorder_ls, ringorder_M, ringorder_C, ringorder_wp, ringorder_Wp, ringorder_ws, ringorder_Ws, ringorder_a, rRingOrder_t
 from sage.libs.singular.decl cimport p_Copy, prCopyR
 from sage.libs.singular.decl cimport n_unknown,  n_Zp,  n_Q,   n_R,   n_GF,  n_long_R,  n_algExt,n_transExt,n_long_C,   n_Z,   n_Zn,  n_Znm,  n_Z2m,  n_CF
 from sage.libs.singular.decl cimport n_coeffType, cfInitCharProc
@@ -165,7 +165,7 @@
     ##         q    q : GF(q=p^n)       *names         TRUE              (todo)
 
     _wvhdl  = <int **>omAlloc0((nblcks + 2) * sizeof(int *))
-    _order  = <int *>omAlloc0((nblcks + 2) * sizeof(int))
+    _order  = <rRingOrder_t *>omAlloc0((nblcks + 2) * sizeof(int))
     _block0 = <int *>omAlloc0((nblcks + 2) * sizeof(int))
     _block1 = <int *>omAlloc0((nblcks + 2) * sizeof(int))
 
