diff --git a/src/sage/numerical/optimize.py b/src/sage/numerical/optimize.py
index 17b5ebb84b..7ce9055cba 100644
--- a/src/sage/numerical/optimize.py
+++ b/src/sage/numerical/optimize.py
@@ -486,9 +486,9 @@ def minimize_constrained(func,cons,x0,gradient=None,algorithm='default', **args)
                 else:
                     min = optimize.fmin_tnc(f, x0, approx_grad=True, bounds=cons, messages=0, **args)[0]
         elif isinstance(cons[0], function_type) or isinstance(cons[0], Expression):
-            min = optimize.fmin_cobyla(f, x0, cons, iprint=0, **args)
+            min = optimize.fmin_cobyla(f, x0, cons, **args)
     elif isinstance(cons, function_type) or isinstance(cons, Expression):
-        min = optimize.fmin_cobyla(f, x0, cons, iprint=0, **args)
+        min = optimize.fmin_cobyla(f, x0, cons, **args)
     return vector(RDF, min)
 
 
