--- src/bin/sage-env.orig	2015-01-06 10:46:04.469687605 +0100
+++ src/bin/sage-env	2015-01-06 10:49:59.547762019 +0100
@@ -111,41 +111,7 @@
 }
 
 
-# New value for SAGE_ROOT: either SAGE_ROOT (if given)
-# or a guessed value based on pwd.
-if [ -n "$SAGE_ROOT" ]; then
-    NEW_SAGE_ROOT="$SAGE_ROOT"
-elif [ -f sage -a -d build ]; then
-    NEW_SAGE_ROOT="."
-elif [ -f ../../sage -a -d ../../build ]; then
-    NEW_SAGE_ROOT="../.."
-else
-    # No idea what SAGE_ROOT should be...
-    echo >&2 "Error: You must set the SAGE_ROOT environment variable or run this"
-    echo >&2 "script from the SAGE_ROOT or SAGE_ROOT/local/bin/ directory."
-    return 1
-fi
-
-# Make NEW_SAGE_ROOT absolute
-NEW_SAGE_ROOT=`cd "$NEW_SAGE_ROOT" && pwd -P`
-
-# Sanity check NEW_SAGE_ROOT
-if [ -f "$NEW_SAGE_ROOT/sage" -a -d "$NEW_SAGE_ROOT/build" ]; then
-    :
-else
-    echo >&2 "Error: SAGE_ROOT is set to a bad value:"
-    echo >&2 "SAGE_ROOT=$SAGE_ROOT"
-    echo >&2 "You must correct it or erase it and run this script from the SAGE_ROOT"
-    echo >&2 "or SAGE_ROOT/local/bin/ directory."
-    return 1
-fi
-
-# Warn if NEW_SAGE_ROOT does not equal the old SAGE_ROOT
-if [ "$SAGE_ROOT" != "$NEW_SAGE_ROOT" -a -n "$SAGE_ROOT" ]; then
-    echo >&2 "Warning: overwriting SAGE_ROOT environment variable:"
-    echo >&2 "Old SAGE_ROOT=$SAGE_ROOT"
-    echo >&2 "New SAGE_ROOT=$NEW_SAGE_ROOT"
-fi
+NEW_SAGE_ROOT="/usr"
 
 
 # Don't execute the commands more than once for the same version of
@@ -174,33 +174,9 @@
 export SAGE_ENV_SOURCED=$SAGE_ENV_VERSION
 
 export SAGE_ROOT="$NEW_SAGE_ROOT"
+export SAGE_LOCAL='/usr'
+export SAGE_SCRIPTS_DIR='/usr/bin'
 
-
-# sage-env must know where the Sage's script files are.
-# Note that SAGE_SCRIPTS_DIR is only used here, so it does not need to
-# be exported.
-if [ -z "$SAGE_SCRIPTS_DIR" ]; then
-    if [ -n "$SAGE_LOCAL" ] && [ -f "$SAGE_LOCAL/bin/sage-env-config" ]; then
-        SAGE_SCRIPTS_DIR="$SAGE_LOCAL/bin"
-    elif [ -f "$SAGE_ROOT/src/bin/sage-env-config" ]; then
-        SAGE_SCRIPTS_DIR="$SAGE_ROOT/src/bin"
-    else
-        echo >&2 "Error: You must set either the SAGE_LOCAL or SAGE_SCRIPTS_DIR environment variable to run this"
-        return 1
-    fi
-elif [ ! -f "$SAGE_SCRIPTS_DIR/sage-env-config" ]; then
-    echo >&2 "Error: SAGE_SCRIPTS_DIR is set to a bad value:"
-    echo >&2 "SAGE_SCRIPTS_DIR=$SAGE_SCRIPTS_DIR"
-    echo >&2 "You must correct it or erase it and rerun this script"
-    return 1
-fi
-
-# Set environment variables (like SAGE_LOCAL) depending on ./configure
-. "$SAGE_SCRIPTS_DIR/sage-env-config"
-if [ $? -ne 0 ]; then
-    echo >&2 "Error: failed to source $SAGE_SCRIPTS_DIR/sage-env-config"
-    return 1
-fi
 
 # The compilers are set in order of priority by
 # 1) environment variables
@@ -296,8 +296,8 @@
 export SAGE_SPKG_INST="$SAGE_LOCAL/var/lib/sage/installed"
 export SAGE_SPKG_SCRIPTS="$SAGE_LOCAL/var/lib/sage/scripts"
 export SAGE_LOGS="$SAGE_ROOT/logs/pkgs"
-export SAGE_SRC="$SAGE_ROOT/src"
-export SAGE_DOC_SRC="$SAGE_SRC/doc"
+export SAGE_SRC="$SAGE_LOCAL/lib/python2.7/site-packages"
+export SAGE_DOC_SRC="$SAGE_SHARE/doc/sage"
 export SAGE_DOC="$SAGE_SHARE/doc/sage"
 
 if [ -z "${SAGE_ORIG_PATH_SET}" ]; then
@@ -403,29 +403,8 @@
     export SAGE_STARTUP_FILE
 fi
 
+export LIBMTX="$DOT_SAGE/meataxe"
+[[ -d $LIBMTX ]] || mkdir -p $LIBMTX # Create meataxe data dir
-if [ "$PYTHON_EGG_CACHE" = "" ]; then
-    PYTHON_EGG_CACHE="$DOT_SAGE/.python-eggs"
-    export PYTHON_EGG_CACHE
-fi
-
-# Set PYTHONUSERBASE to avoid picking up non-Sage versions of
-# Matplotlib, numpy, etc. See http://trac.sagemath.org/ticket/19612.
-#
-# For more history (it used to be PYTHONNOUSERSITE=yes which killed
-# the ability to do "sage -pip install PACKAGE --user"), see
-# http://trac.sagemath.org/ticket/14243 and
-# http://trac.sagemath.org/ticket/18955.
-
-if [ "$PYTHONUSERBASE" = "" ]; then
-    PYTHONUSERBASE="$DOT_SAGE/local"
-    export PYTHONUSERBASE
-fi
-
-if [ -n "$PYTHONHOME" ]; then
-    >&2 echo "Warning: PYTHONHOME must not be set when running Sage, clearing env..."
-    unset PYTHONHOME
-fi
-
 LDFLAGS="-L$SAGE_LOCAL/lib -Wl,-rpath,$SAGE_LOCAL/lib $LDFLAGS"
 export LDFLAGS
 
@@ -489,8 +489,6 @@
     fi
 fi

-export MAXIMA_PREFIX="$SAGE_LOCAL"
-export MAXIMA_USERDIR="$DOT_SAGE/maxima"
 
 PERL5LIB="$SAGE_LOCAL/lib/perl5:$PERL5LIB" && export PERL5LIB
 
@@ -601,9 +601,6 @@
     LD_LIBRARY_PATH="$SAGE_LOCAL/bin:$LD_LIBRARY_PATH" && export LD_LIBRARY_PATH
 fi
 
-# See trac 7186 -- this is needed if ecl is moved
-ECLDIR="$SAGE_LOCAL/lib/ecl/" && export ECLDIR
-
 # Handle parallel building/testing/...
 # See Trac Ticket #12016
 # First, figure out the right values for SAGE_NUM_THREADS (default
--- src/sage/env.py.orig       2017-12-07 19:51:25.554281539 +0000
+++ src/sage/env.py    2017-12-07 19:51:42.787654650 +0000
@@ -117,13 +117,13 @@
 _add_variable_or_fallback('LOCAL_IDENTIFIER','$HOSTNAME.%s'%os.getpid())
 
 # bunch of sage directories and files
-_add_variable_or_fallback('SAGE_ROOT',       None)
-_add_variable_or_fallback('SAGE_LOCAL',      None)
+_add_variable_or_fallback('SAGE_ROOT',       '')
+_add_variable_or_fallback('SAGE_LOCAL',      '/usr')
 _add_variable_or_fallback('SAGE_ETC',        opj('$SAGE_LOCAL', 'etc'))
 _add_variable_or_fallback('SAGE_INC',        opj('$SAGE_LOCAL', 'include'))
 _add_variable_or_fallback('SAGE_SHARE',      opj('$SAGE_LOCAL', 'share'))
 
-_add_variable_or_fallback('SAGE_SRC',        opj('$SAGE_ROOT', 'src'))
+_add_variable_or_fallback('SAGE_SRC',        opj('$SAGE_LOCAL', 'lib', 'python2.7', 'site-packages'))
 
 try:
     sitepackages_dirs = site.getsitepackages()
