diff --git a/src/bin/sage-notebook b/src/bin/sage-notebook
index 83109e6..b954c22 100755
--- a/src/bin/sage-notebook
+++ b/src/bin/sage-notebook
@@ -91,12 +91,8 @@ class NotebookJupyter(object):
 
     def __init__(self, argv):
         self.print_banner()
-        from sage.repl.ipython_kernel.install import have_prerequisites
-        if not have_prerequisites():
-            print(self.PREREQUISITE_ERROR)
-            raise SystemExit(1)
-        from notebook.notebookapp import main
-        main(argv)
+        from sys import prefix
+        os.execv(os.path.join(prefix, 'bin', 'jupyter-notebook'), ['jupyter-notebook'] + argv)
 
 
 class NotebookJupyterlab(object):
@@ -109,15 +105,15 @@ class NotebookJupyterlab(object):
         cls(['help'])
 
     def __init__(self, argv):
+        self.print_banner()
+        from sys import prefix
         try:
-            from jupyterlab.labapp import main
-        except ImportError:
+            os.execv(os.path.join(prefix, 'bin', 'jupyter-lab'), ['jupyter-lab'] + argv)
+        except OSError:
             print("Jupyterlab is not installed (at least not in this Sage installation).")
             print("You can install it by running")
             print("  sage --pip install jupyterlab")
             raise SystemExit(1)
-        self.print_banner()
-        main(argv)
 
 
 class SageNBExport(NotebookJupyter):
diff --git a/src/sage/repl/ipython_kernel/install.py b/src/sage/repl/ipython_kernel/install.py
index 073fae5..81378f4 100644
--- a/src/sage/repl/ipython_kernel/install.py
+++ b/src/sage/repl/ipython_kernel/install.py
@@ -279,35 +279,3 @@ class SageKernelSpec(object):
         instance.use_local_threejs()
         instance._install_spec()
         instance._symlink_resources()
-
-
-def have_prerequisites(debug=True):
-    """
-    Check that we have all prerequisites to run the Jupyter notebook.
-
-    In particular, the Jupyter notebook requires OpenSSL whether or
-    not you are using https. See :trac:`17318`.
-
-    INPUT:
-
-    ``debug`` -- boolean (default: ``True``). Whether to print debug
-    information in case that prerequisites are missing.
-
-    OUTPUT:
-
-    Boolean.
-
-    EXAMPLES::
-
-        sage: from sage.repl.ipython_kernel.install import have_prerequisites
-        sage: have_prerequisites(debug=False) in [True, False]
-        True
-    """
-    try:
-        from notebook.notebookapp import NotebookApp
-        return True
-    except ImportError:
-        if debug:
-            import traceback
-            traceback.print_exc()
-        return False
