--- network.orig	2011-11-09 18:42:28.000000000 +0400
+++ network	2011-11-09 18:46:22.000000000 +0400
@@ -76,7 +76,7 @@
          else
             echo `date` "[rescue_nic] activating $nic ..."
 
-            ifup $nic
+            ifconfig $nic up
             exitCode=`expr $exitCode \| $?`
          fi
       done < $activeList
@@ -105,7 +105,7 @@
    # `which' may be a bit noisy, so we'll shush it.
    dbusSend=`which dbus-send 2>/dev/null`
    rc=$?
-   if [ $rc = 0 ]; then
+   if [ $rc = 0 ] && [ `pidof dbus-daemon` ]; then
       # NetworkManager 0.8.0
       $dbusSend --system --print-reply          \
          --dest=org.freedesktop.NetworkManager  \
@@ -154,7 +154,7 @@
    # `which' may be a bit noisy, so we'll shush it.
    dbusSend=`which dbus-send 2>/dev/null`
    rc=$?
-   if [ $rc = 0 ]; then
+   if [ $rc = 0 ] && [ `pidof dbus-daemon` ]; then
       # NetworkManager 0.8.0
       $dbusSend --system --print-reply          \
          --dest=org.freedesktop.NetworkManager  \
@@ -203,7 +203,7 @@
 
    # XXX Are these really necessary?  If so, we should have seen customer
    # complaints by now.
-   which ifup >/dev/null 2>&1      || Panic "ifup not in search path."
+#   which ifup >/dev/null 2>&1     || Panic "ifup not in search path."
    which ifconfig >/dev/null 2>&1  || Panic "ifconfig not in search path."
 
    case "$1" in
