--- network.org	2012-03-17 22:20:11.000000000 +0400
+++ network	2012-03-17 22:18:47.000000000 +0400
@@ -34,51 +34,6 @@
 
 
 #
-# find_networking_script --
-#
-#    Searches common Linux distro init/rc paths to find a singular network
-#    services script.
-#
-# Result:
-#    Returns a valid networking script path on success or "error" on failure.
-#
-# Side effects:
-#    None.
-#
-
-find_networking_script() {
-   local script="error"
-   for dir in "/etc/init.d" "/sbin/init.d" "/etc" "/etc/rc.d" ; do
-      if [ -d "$dir/rc0.d" ] &&
-         [ -d "$dir/rc1.d" ] &&
-         [ -d "$dir/rc2.d" ] &&
-         [ -d "$dir/rc3.d" ] &&
-         [ -d "$dir/rc4.d" ] &&
-         [ -d "$dir/rc5.d" ] &&
-         [ -d "$dir/rc6.d" ]; then
-
-         # Now find the appropriate networking script.
-         if [ -d "$dir/init.d" ]; then
-            if [ -x "$dir/init.d/network" ]; then
-               script="$dir/init.d/network"
-            elif [ -x "$dir/init.d/networking" ]; then
-               script="$dir/init.d/networking"
-            fi
-         else
-            if [ -x "$dir/network" ]; then
-               script="$dir/network"
-            elif [ -x "$dir/networking" ]; then
-               script="$dir/networking"
-            fi
-         fi
-      fi
-   done
-
-   echo "$script"
-}
-
-
-#
 # run_network_script --
 #
 # Finds out how to run the system's script used to control networking, and
@@ -87,17 +42,7 @@
 #
 run_network_script()
 {
-   script=`find_networking_script`
-   [ "$script" != "error" ] || Panic "Cannot find system networking script."
-
-   # Using SysV "service" if it exists, otherwise fall back to run the script directly
-   service=`which service 2>/dev/null`
-   if [ $? = 0 -a -n "$service" ]; then
-      serviceName=`basename "$script"`
-      "$service" "$serviceName" "$1"
-   else
-      "$script" "$1"
-   fi
+  /etc/rc.d/network "$1"
 }
 
 
