diff -Naur gnome-panel-3.8.0.orig/data/gnome-flashback.session.desktop.in gnome-panel-3.8.0/data/gnome-flashback.session.desktop.in
--- gnome-panel-3.8.0.orig/data/gnome-flashback.session.desktop.in	2013-09-27 03:12:58.291413318 +0000
+++ gnome-panel-3.8.0/data/gnome-flashback.session.desktop.in	2013-09-27 03:15:34.183583841 +0000
@@ -1,3 +1,3 @@
 [GNOME Session]
 _Name=GNOME Flashback
-RequiredComponents=gnome-panel;gnome-settings-daemon;gnome-screensaver;metacity;
+RequiredComponents=gnome-panel;gnome-settings-daemon;gnome-screensaver;metacity;notification-daemon;polkit-gnome-authentication-agent-1;gnome-fallback-background-helper;gnome-fallback-media-keys-helper;gnome-fallback-mount-helper;
