--- build/groovy/org/jetbrains/intellij/build/IdeaCommunityBuilder.groovy	2021-09-09 23:57:31.252582123 +0300
+++ build/groovy/org/jetbrains/intellij/build/IdeaCommunityBuilder.groovy	2021-09-09 23:58:18.939737294 +0300
@@ -41,15 +41,15 @@
     buildContext.options.incrementalCompilation = true
     def tasks = BuildTasks.create(buildContext)
     tasks.buildDistributions()
-    buildContext.messages.block("Build standalone JPS") {
+    /*buildContext.messages.block("Build standalone JPS") {
       String jpsArtifactDir = "$buildContext.paths.artifacts/jps"
       new CommunityStandaloneJpsBuilder(buildContext).processJpsLayout(jpsArtifactDir, buildContext.fullBuildNumber, new ProjectStructureMapping(),
                                                                        true, {})
-    }
+    }*/
     tasks.buildUpdaterJar()
   }
 
   void buildUnpackedDistribution(String targetDirectory) {
     BuildTasks.create(buildContext).buildUnpackedDistribution(Paths.get(targetDirectory), false)
   }
-}
\ No newline at end of file
+}
