From ed442747240d4f14d275890f4a327a1bb6675af7 Mon Sep 17 00:00:00 2001
From: Stephane MANKOWSKI <stephane@mankowski.fr>
Date: Thu, 7 Mar 2019 19:00:20 +0100
Subject: Migration failure with sqlite >= 3.25.0 due to ALTER TABLE behavior
 modification

---
 skgbasemodeler/skgdocument.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/skgbasemodeler/skgdocument.cpp b/skgbasemodeler/skgdocument.cpp
index 01cdcf6..fe8fa03 100644
--- a/skgbasemodeler/skgdocument.cpp
+++ b/skgbasemodeler/skgdocument.cpp
@@ -1524,6 +1524,7 @@ SKGError SKGDocument::load(const QString& iName, const QString& iPassword, bool
                      << QStringLiteral("PRAGMA temp_store=MEMORY")
                      // << QStringLiteral("PRAGMA locking_mode=EXCLUSIVE")
                      << QStringLiteral("PRAGMA synchronous = OFF")
+                     << QStringLiteral("PRAGMA legacy_alter_table=ON")  // For migration on sqlite >=3.25 (see https://sqlite.org/lang_altertable.html)
                      << QStringLiteral("PRAGMA recursive_triggers=true");
         IFOKDO(err, executeSqliteOrders(optimization))
 
-- 
cgit v1.1

