diff --git a/rts/Map/SMF/ROAM/Patch.h b/rts/Map/SMF/ROAM/Patch.h
index c7f311beee..f3a6fba96f 100644
--- a/rts/Map/SMF/ROAM/Patch.h
+++ b/rts/Map/SMF/ROAM/Patch.h
@@ -10,6 +10,7 @@
 
 #include <array>
 #include <vector>
+#include <limits>
 
 
 class CSMFGroundDrawer;
diff --git a/rts/Sim/Weapons/WeaponLoader.cpp b/rts/Sim/Weapons/WeaponLoader.cpp
index fd98590971..771d830583 100644
--- a/rts/Sim/Weapons/WeaponLoader.cpp
+++ b/rts/Sim/Weapons/WeaponLoader.cpp
@@ -26,6 +26,8 @@
 #include "Sim/Units/UnitDef.h"
 #include "System/Log/ILog.h"
 
+#include <limits>
+
 static std::array<uint8_t, 2048> udWeaponCounts;
 
 WeaponMemPool weaponMemPool;
