--- sympol/CMakeLists.txt.orig	2016-05-08 13:52:35.000000000 -0600
+++ sympol/CMakeLists.txt	2020-07-06 08:35:36.253421935 -0600
@@ -73,7 +73,7 @@ set_target_properties(sympol PROPERTIES
 add_executable(sympolbin sympol.cpp)
 set_target_properties(sympolbin PROPERTIES OUTPUT_NAME sympol)
 
-set(TARGET_LIBS ${TARGET_LIBS} lrsgmp cddgmp ${GMP_LIBRARIES} ${Boost_LIBRARIES})
+set(TARGET_LIBS ${TARGET_LIBS} lrs cddgmp ${GMP_LIBRARIES} ${Boost_LIBRARIES})
 target_link_libraries(sympol ${TARGET_LIBS})
 target_link_libraries(sympolbin sympol)
 
--- sympol/raycomputationlrs.cpp.orig	2016-05-08 13:52:35.000000000 -0600
+++ sympol/raycomputationlrs.cpp	2020-07-06 08:37:15.240288168 -0600
@@ -25,6 +25,7 @@
 #include "polyhedron.h"
 
 extern "C" {
+  #include <lrsrestart.h>
   #include <lrslib.h>
 }
 
@@ -66,7 +67,6 @@ bool RayComputationLRS::finish() {
         return true;
     }
 
-    lrs_mp_close();
     
     if (RayComputationLRS::ms_fIn != NULL) {
         if (std::fclose(RayComputationLRS::ms_fIn)) {
