--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -191,10 +191,10 @@
 # where to put generated binaries
 set(EXECUTABLE_OUTPUT_PATH "${BUILD_OUTPUT_ROOT_DIRECTORY}")
 
-if(PYARROW_USE_TENSORFLOW)
-  # TensorFlow uses the old GLIBCXX ABI, so we have to use it too
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
-endif()
+#if(PYARROW_USE_TENSORFLOW)
+  ## TensorFlow uses the old GLIBCXX ABI, so we have to use it too
+  #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
+#endif()
 
 # Python and Numpy libraries
 find_package(Python3Alt REQUIRED)
