diff -ruN a/interface/Makefile.am b/interface/Makefile.am
--- a/interface/Makefile.am	2013-12-12 03:04:52.000000000 +0100
+++ b/interface/Makefile.am	2020-09-04 14:22:48.890913333 +0200
@@ -394,37 +394,37 @@
 	$(COMPILE) $(SHCFLAGS) -o $@ -c $<
 
 if_ncurses.$(so): ncurs_c.$(so) support.$(so) bitset.$(so) timer.$(so)
-	$(SHLD) -o $@ ncurs_c.$(so) support.$(so) bitset.$(so) timer.$(so) $(n_so_libs)
+	$(SHLD) $(LDFLAGS) -o $@ ncurs_c.$(so) support.$(so) bitset.$(so) timer.$(so) $(n_so_libs)
 
 if_slang.$(so): slang_c.$(so) support.$(so) timer.$(so)
-	$(SHLD) -o $@ slang_c.$(so) support.$(so) timer.$(so) $(s_so_libs)
+	$(SHLD) $(LDFLAGS) -o $@ slang_c.$(so) support.$(so) timer.$(so) $(s_so_libs)
 
 if_motif.$(so): motif_c.$(so) support.$(so) motif_i.$(so) motif_p.$(so)
-	$(SHLD) -o $@ motif_c.$(so) support.$(so) motif_i.$(so) motif_p.$(so) $(m_so_libs)
+	$(SHLD) $(LDFLAGS) -o $@ motif_c.$(so) support.$(so) motif_i.$(so) motif_p.$(so) $(m_so_libs)
 
 if_tcltk.$(so): tk_c.$(so) support.$(so)
-	$(SHLD) -o $@ tk_c.$(so) support.$(so) $(k_so_libs)
+	$(SHLD) $(LDFLAGS) -o $@ tk_c.$(so) support.$(so) $(k_so_libs)
 
 if_emacs.$(so): emacs_c.$(so) support.$(so)
-	$(SHLD) -o $@ emacs_c.$(so) support.$(so) $(e_so_libs)
+	$(SHLD) $(LDFLAGS) -o $@ emacs_c.$(so) support.$(so) $(e_so_libs)
 
 if_vt100.$(so): vt100_c.$(so) support.$(so) vt100.$(so) bitset.$(so) timer.$(so)
-	$(SHLD) -o $@ vt100_c.$(so) support.$(so) vt100.$(so) bitset.$(so) timer.$(so) $(T_so_libs)
+	$(SHLD) $(LDFLAGS) -o $@ vt100_c.$(so) support.$(so) vt100.$(so) bitset.$(so) timer.$(so) $(T_so_libs)
 
 if_xaw.$(so): xaw_c.$(so) support.$(so) xaw_i.$(so) x_trace.${so} timer.$(so) ${XDND_SO}
-	$(SHLD) -o $@ xaw_c.$(so) support.$(so) xaw_i.$(so) x_trace.${so} timer.$(so) ${XDND_SO} $(a_so_libs)
+	$(SHLD) $(LDFLAGS) -o $@ xaw_c.$(so) support.$(so) xaw_i.$(so) x_trace.${so} timer.$(so) ${XDND_SO} $(a_so_libs)
 
 if_xskin.$(so): xskin_c.$(so) support.$(so) xskin_i.$(so) xskin_loadBMP.$(so) xskin_spectrum.$(so) url_buff.$(so)
-	$(SHLD) -o $@ xskin_c.$(so) support.$(so) xskin_i.$(so) xskin_loadBMP.$(so) xskin_spectrum.$(so) url_buff.$(so) $(i_so_libs)
+	$(SHLD) $(LDFLAGS) -o $@ xskin_c.$(so) support.$(so) xskin_i.$(so) xskin_loadBMP.$(so) xskin_spectrum.$(so) url_buff.$(so) $(i_so_libs)
 
 if_gtk.$(so): gtk_c.$(so) support.$(so) gtk_i.$(so) gtk_p.$(so)
-	$(SHLD) -o $@ gtk_c.$(so) support.$(so) gtk_i.$(so) gtk_p.$(so) $(g_so_libs)
+	$(SHLD) $(LDFLAGS) -o $@ gtk_c.$(so) support.$(so) gtk_i.$(so) gtk_p.$(so) $(g_so_libs)
 
 if_server.$(so): server_c.$(so) support.$(so)
-	$(SHLD) -o $@ server_c.$(so) support.$(so)
+	$(SHLD) $(LDFLAGS) -o $@ server_c.$(so) support.$(so)
 
 if_alsaseq.$(so): alsaseq_c.$(so) support.$(so)
-	$(SHLD) -o $@ alsaseq_c.$(so) support.$(so)
+	$(SHLD) $(LDFLAGS) -o $@ alsaseq_c.$(so) support.$(so)
 
 endif
 
diff -ruN a/interface/Makefile.in b/interface/Makefile.in
--- a/interface/Makefile.in	2018-08-29 01:33:39.000000000 +0200
+++ b/interface/Makefile.in	2020-09-04 14:19:07.612876650 +0200
@@ -65,7 +65,7 @@
 #CPPFLAGS = @CPPFLAGS@
 
 #DEFS = @DEFS@
-#LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@
 #LIBS = @LIBS@
 #SHLD = @SHLD@
 #SHCFLAGS = @SHCFLAGS@
@@ -1175,37 +1175,37 @@
 @NEEDDLOPEN_TRUE@	$(COMPILE) $(SHCFLAGS) -o $@ -c $<
 
 @NEEDDLOPEN_TRUE@if_ncurses.$(so): ncurs_c.$(so) support.$(so) bitset.$(so) timer.$(so)
-@NEEDDLOPEN_TRUE@	$(SHLD) -o $@ ncurs_c.$(so) support.$(so) bitset.$(so) timer.$(so) $(n_so_libs)
+@NEEDDLOPEN_TRUE@	$(SHLD) $(LDFLAGS) -o $@ ncurs_c.$(so) support.$(so) bitset.$(so) timer.$(so) $(n_so_libs)
 
 @NEEDDLOPEN_TRUE@if_slang.$(so): slang_c.$(so) support.$(so) timer.$(so)
-@NEEDDLOPEN_TRUE@	$(SHLD) -o $@ slang_c.$(so) support.$(so) timer.$(so) $(s_so_libs)
+@NEEDDLOPEN_TRUE@	$(SHLD) $(LDFLAGS) -o $@ slang_c.$(so) support.$(so) timer.$(so) $(s_so_libs)
 
 @NEEDDLOPEN_TRUE@if_motif.$(so): motif_c.$(so) support.$(so) motif_i.$(so) motif_p.$(so)
-@NEEDDLOPEN_TRUE@	$(SHLD) -o $@ motif_c.$(so) support.$(so) motif_i.$(so) motif_p.$(so) $(m_so_libs)
+@NEEDDLOPEN_TRUE@	$(SHLD) $(LDFLAGS) -o $@ motif_c.$(so) support.$(so) motif_i.$(so) motif_p.$(so) $(m_so_libs)
 
 @NEEDDLOPEN_TRUE@if_tcltk.$(so): tk_c.$(so) support.$(so)
-@NEEDDLOPEN_TRUE@	$(SHLD) -o $@ tk_c.$(so) support.$(so) $(k_so_libs)
+@NEEDDLOPEN_TRUE@	$(SHLD) $(LDFLAGS) -o $@ tk_c.$(so) support.$(so) $(k_so_libs)
 
 @NEEDDLOPEN_TRUE@if_emacs.$(so): emacs_c.$(so) support.$(so)
-@NEEDDLOPEN_TRUE@	$(SHLD) -o $@ emacs_c.$(so) support.$(so) $(e_so_libs)
+@NEEDDLOPEN_TRUE@	$(SHLD) $(LDFLAGS) -o $@ emacs_c.$(so) support.$(so) $(e_so_libs)
 
 @NEEDDLOPEN_TRUE@if_vt100.$(so): vt100_c.$(so) support.$(so) vt100.$(so) bitset.$(so) timer.$(so)
-@NEEDDLOPEN_TRUE@	$(SHLD) -o $@ vt100_c.$(so) support.$(so) vt100.$(so) bitset.$(so) timer.$(so) $(T_so_libs)
+@NEEDDLOPEN_TRUE@	$(SHLD) $(LDFLAGS) -o $@ vt100_c.$(so) support.$(so) vt100.$(so) bitset.$(so) timer.$(so) $(T_so_libs)
 
 @NEEDDLOPEN_TRUE@if_xaw.$(so): xaw_c.$(so) support.$(so) xaw_i.$(so) x_trace.${so} timer.$(so) ${XDND_SO}
-@NEEDDLOPEN_TRUE@	$(SHLD) -o $@ xaw_c.$(so) support.$(so) xaw_i.$(so) x_trace.${so} timer.$(so) ${XDND_SO} $(a_so_libs)
+@NEEDDLOPEN_TRUE@	$(SHLD) $(LDFLAGS) -o $@ xaw_c.$(so) support.$(so) xaw_i.$(so) x_trace.${so} timer.$(so) ${XDND_SO} $(a_so_libs)
 
 @NEEDDLOPEN_TRUE@if_xskin.$(so): xskin_c.$(so) support.$(so) xskin_i.$(so) xskin_loadBMP.$(so) xskin_spectrum.$(so) url_buff.$(so)
-@NEEDDLOPEN_TRUE@	$(SHLD) -o $@ xskin_c.$(so) support.$(so) xskin_i.$(so) xskin_loadBMP.$(so) xskin_spectrum.$(so) url_buff.$(so) $(i_so_libs)
+@NEEDDLOPEN_TRUE@	$(SHLD) $(LDFLAGS) -o $@ xskin_c.$(so) support.$(so) xskin_i.$(so) xskin_loadBMP.$(so) xskin_spectrum.$(so) url_buff.$(so) $(i_so_libs)
 
 @NEEDDLOPEN_TRUE@if_gtk.$(so): gtk_c.$(so) support.$(so) gtk_i.$(so) gtk_p.$(so)
-@NEEDDLOPEN_TRUE@	$(SHLD) -o $@ gtk_c.$(so) support.$(so) gtk_i.$(so) gtk_p.$(so) $(g_so_libs)
+@NEEDDLOPEN_TRUE@	$(SHLD) $(LDFLAGS) -o $@ gtk_c.$(so) support.$(so) gtk_i.$(so) gtk_p.$(so) $(g_so_libs)
 
 @NEEDDLOPEN_TRUE@if_server.$(so): server_c.$(so) support.$(so)
-@NEEDDLOPEN_TRUE@	$(SHLD) -o $@ server_c.$(so) support.$(so)
+@NEEDDLOPEN_TRUE@	$(SHLD) $(LDFLAGS) -o $@ server_c.$(so) support.$(so)
 
 @NEEDDLOPEN_TRUE@if_alsaseq.$(so): alsaseq_c.$(so) support.$(so)
-@NEEDDLOPEN_TRUE@	$(SHLD) -o $@ alsaseq_c.$(so) support.$(so)
+@NEEDDLOPEN_TRUE@	$(SHLD) $(LDFLAGS) -o $@ alsaseq_c.$(so) support.$(so)
 
 clean-compile:
 	-rm -f *.$(so)
