diff -ENwbur ucblogo-6.0/config.h ucblogo-6.0-no-wxWidgets/config.h
--- src/ucblogo-6.0/config.h	2007-11-27 00:42:40.000000000 +0530
+++ src/ucblogo-6.0-b/config.h	2013-03-19 07:19:15.000000000 +0530
@@ -74,4 +74,4 @@
 /* #undef HAVE_LIBTERMLIB */
 
 /* Define if you have the wx library.  */
-#define HAVE_WX 1
+/* #undef HAVE_WX */
Binary files ucblogo-6.0/logo and ucblogo-6.0-no-wxWidgets/logo differ
diff -ENwbur ucblogo-6.0/makefile ucblogo-6.0-no-wxWidgets/makefile
--- src/ucblogo-6.0/makefile	2008-09-14 02:51:18.000000000 +0530
+++ src/ucblogo-6.0-b/makefile	2013-03-19 07:25:44.000000000 +0530
@@ -1,23 +1,23 @@
 CC	= gcc
-CFLAGS	= -g -O -DHAVE_WX    -O0
+CFLAGS	= -g -O2
 CXX     = g++
-CXXFLAGS = -g  -DHAVE_WX -I/usr/local/lib/wx/include/gtk2-ansi-release-static-2.8 -I/usr/local/include/wx-2.8 -D_FILE_OFFSET_BITS=64 -D_LARGE_FILES -D__WXGTK__ -pthread
 LDFLAGS	= 
-LIBS  =   -lbsd -lm  -L/usr/local/lib -pthread   /usr/local/lib/libwx_gtk2_richtext-2.8.a /usr/local/lib/libwx_gtk2_aui-2.8.a /usr/local/lib/libwx_gtk2_xrc-2.8.a /usr/local/lib/libwx_gtk2_qa-2.8.a /usr/local/lib/libwx_gtk2_html-2.8.a /usr/local/lib/libwx_gtk2_adv-2.8.a /usr/local/lib/libwx_gtk2_core-2.8.a /usr/local/lib/libwx_base_xml-2.8.a /usr/local/lib/libwx_base_net-2.8.a /usr/local/lib/libwx_base-2.8.a -pthread -L/lib -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgdk_pixbuf-2.0 -lpango-1.0 -lgobject-2.0 -lgmodule-2.0 -lgthread-2.0 -lrt -lglib-2.0 -lXinerama -lXxf86vm -lSM -lpng -ljpeg -ltiff -lexpat -lz -ldl -lm  -ltermcap -lX11 
+LIBS  =  -lSM -lICE  -lm  -lcurses -lX11 
 prefix = /usr/local
 BINDIR        = $(prefix)/bin
 LIBLOC        = $(prefix)/lib/logo
-LINKER = $(CXX)
+LINKER = $(CC)
+
 
 # LIBLOC      = `pwd`
 
 OBJS 	= coms.o error.o eval.o files.o graphics.o init.o intern.o \
 	  libloc.o lists.o logodata.o main.o math.o mem.o paren.o parse.o \
-	  print.o wrksp.o nographics.o svn.o wxMain.o wxTerminal.o wxTurtleGraphics.o  TextEditor.o wxterm.o 
+	  print.o wrksp.o nographics.o svn.o term.o xgraphics.o 
 
 SRCS	= coms.c error.c eval.c files.c graphics.c init.c intern.c \
 	  libloc.c lists.c logodata.c main.c math.c mem.c paren.c parse.c \
-	  print.c wrksp.c nographics.c wxMain.cpp wxTerminal.cpp wxTurtleGraphics.cpp  TextEditor.cpp wxterm.c 
+	  print.c wrksp.c nographics.c term.c xgraphics.c 
 
 HDRS	= globals.h logo.h xgraphics.h
 
@@ -70,7 +70,7 @@
 #	cd emacs; $(MAKE) ship
 	cd docs; $(MAKE) ship
 
-install: all
+install:
 	for d in $(BINDIR) $(LIBLOC) $(LIBLOC)/logolib $(LIBLOC)/helpfiles $(LIBLOC)/csls; do [ -d $$d ] || mkdir -p $$d || exit 1; done
 	cp logo $(BINDIR)/.
 	cp -f logolib/* $(LIBLOC)/logolib/.
