diff -rupN vimb.orig/config.mk vimb/config.mk
--- vimb.orig/config.mk	2019-07-23 11:18:15.460717705 +0200
+++ vimb/config.mk	2019-07-23 11:19:04.091175665 +0200
@@ -36,7 +36,7 @@ endif
 EXTTARGET   = webext_main.so
 EXTCFLAGS   = ${CFLAGS} $(shell pkg-config --cflags webkit2gtk-web-extension-4.0)
 EXTCPPFLAGS = $(CPPFLAGS)
-EXTLDFLAGS  = $(shell pkg-config --libs webkit2gtk-web-extension-4.0) -shared
+EXTLDFLAGS  = $(LDFLAGS) $(shell pkg-config --libs webkit2gtk-web-extension-4.0) -shared
 
 # flags used for the main application
 CFLAGS     += $(shell pkg-config --cflags $(LIBS))
