--- a/lib/tests/facts/double_value.cc	2019-09-30 09:13:02.000000000 +0200
+++ b/lib/tests/facts/double_value.cc	2019-10-12 19:54:49.184377937 +0200
@@ -23,13 +23,6 @@ SCENARIO("using a double fact value") {
             REQUIRE(json.GetDouble() == Approx(42.4242));
         }
     }
-    WHEN("serialized to YAML") {
-        THEN("it should have the same value") {
-            Emitter emitter;
-            value.write(emitter);
-            REQUIRE(string(emitter.c_str()) == "42.4242");
-        }
-    }
     WHEN("serialized to text") {
         THEN("it should have the same value") {
             ostringstream stream;
