--- src/WORKSPACE	1980-01-01 00:00:00.000000000 +0000
+++ src/WORKSPACE	2019-03-02 22:23:29.544116897 +0000
@@ -72,8 +72,8 @@
 
 new_local_repository(
     name = "com_google_protobuf",
-    build_file = "./third_party/protobuf/3.6.1/BUILD",
-    path = "./third_party/protobuf/3.6.1/",
+    build_file = "./third_party/protobuf/3.7.0/BUILD",
+    path = "./third_party/protobuf/3.7.0/",
 )
 
 local_repository(
