diff -aur xmonad-contrib-0.12/xmonad-contrib.cabal xmonad-contrib-0.12.new/xmonad-contrib.cabal
--- xmonad-contrib-0.12/xmonad-contrib.cabal	2015-12-18 21:11:11.000000000 +0100
+++ xmonad-contrib-0.12.new/xmonad-contrib.cabal	2017-01-08 12:13:09.828927857 +0100
@@ -64,7 +64,7 @@
                     random,
                     mtl >= 1 && < 3,
                     unix,
-                    X11>=1.6.1 && < 1.7,
+                    X11>=1.6.1 && < 1.8,
                     xmonad>=0.12   && < 0.13,
                     utf8-string
 
