--- a/linux/hydrogen.desktop	2016-11-05 09:22:50.000000000 +0100
+++ b/linux/hydrogen.desktop	2017-07-28 15:25:19.412346723 +0200
@@ -19,19 +13,17 @@
 GenericName[tr]=Davul Makinası
 GenericName[fr]=Boîte à rythme
 
-Encoding=UTF-8
 Version=1.0
 
 Type=Application
-Categories=Application;AudioVideo;Sound;Audio;Qt;
+Categories=AudioVideo;Audio;Qt;
 Keywords=audio;sound;jackd;sampler;drum;
 
-MimeType=text/xml
+MimeType=text/xml;
 
-FilePattern=hydrogen
 Exec=hydrogen %F
 TryExec=hydrogen
 Terminal=false
 StartupNotify=true
 
-Icon=h2-icon
+Icon=hydrogen
