diff -ruN a/data/i18n/CMakeLists.txt b/data/i18n/CMakeLists.txt
--- a/data/i18n/CMakeLists.txt	2020-08-03 21:05:33.000000000 +0200
+++ b/data/i18n/CMakeLists.txt	2020-08-03 21:37:38.195846093 +0200
@@ -5,4 +5,4 @@
 
 QT5_ADD_TRANSLATION(QM_FILES ${TS_FILES})
 ADD_CUSTOM_TARGET(translations ALL DEPENDS ${QM_FILES})
-INSTALL(FILES ${QM_FILES} DESTINATION "data/i18n")
+INSTALL(FILES ${QM_FILES} DESTINATION "${H2_SYS_PATH}/data/i18n")
