Index: mp3lib/dct64_sse.c
===================================================================
--- mp3lib/dct64_sse.c	(revision 33468)
+++ mp3lib/dct64_sse.c	(revision 33469)
@@ -112,7 +112,6 @@
     }
 
     {
-        real *costab = costab_mmx + 24;
         int i;
 
         __asm__(
@@ -121,7 +120,7 @@
             "movaps    %1, %%xmm5\n\t"
             "movaps    %%xmm5, %%xmm6\n\t"
             :
-            :"m"(*costab), "m"(*nnnn)
+            :"m"(costab_mmx[24]), "m"(*nnnn)
            );
 
         for (i = 0; i < 0x20; i += 8)
