--- a/slime.el
+++ b/slime.el
@@ -179,7 +179,7 @@ This applies to the *inferior-lisp* buffer and the network connections."
   :prefix "slime-"
   :group 'slime)
 
-(defcustom slime-backend "swank-loader.lisp"
+(defcustom slime-backend "/usr/share/common-lisp/source/swank/swank-loader.lisp"
   "The name of the Lisp file that loads the Swank server.
 This name is interpreted relative to the directory containing
 slime.el, but could also be set to an absolute filename."
--- a/swank-loader.lisp
+++ b/swank-loader.lisp
@@ -16,6 +16,7 @@
 ;;   (load ".../swank-loader.lisp")
 ;;   (setq swank-loader::*fasl-directory* "/tmp/fasl/")
 ;;   (swank-loader:init)
+(require "asdf")
 
 (cl:defpackage :swank-loader
   (:use :cl)
@@ -145,14 +146,10 @@ operating system, and hardware architecture."
 (defun slime-version-string ()
   "Return a string identifying the SLIME version.
 Return nil if nothing appropriate is available."
-  (with-open-file (s (merge-pathnames "slime.el" *source-directory*)
+  (with-open-file (s (merge-pathnames "slime-version" *source-directory*)
                      :if-does-not-exist nil)
     (when s
-      (loop with prefix = ";; Version: "
-            for line = (read-line s nil :eof)
-            until (eq line :eof)
-            when (string-starts-with line prefix)
-              return (subseq line (length prefix))))))
+      (read-line s))))
 
 (defun default-fasl-dir ()
   (merge-pathnames
