diff --git a/lib/libaudit.h b/lib/libaudit.h
index 05ee91e..2ee3842 100644
--- a/lib/libaudit.h
+++ b/lib/libaudit.h
@@ -260,7 +260,6 @@ extern "C" {
 #define AUDIT_KEY_SEPARATOR 0x01
 
 /* These are used in filter control */
-#define AUDIT_FILTER_EXCLUDE	AUDIT_FILTER_TYPE
 #define AUDIT_FILTER_MASK	0x07	/* Mask to get actual filter */
 #define AUDIT_FILTER_UNSET	0x80	/* This value means filter is unset */
 
