diff --git a/src/asgen/zarchive.d b/src/asgen/zarchive.d
index 17a40d6..c7a1181 100644
--- a/src/asgen/zarchive.d
+++ b/src/asgen/zarchive.d
@@ -572,7 +572,8 @@ unittest
 
     writeln ("TEST: ", "Extracting a tarball");
 
-    import std.file : buildPath, tempDir;
+    import std.file : tempDir;
+    import std.path: buildPath;
     import asgen.utils : getTestSamplesDir;
 
     auto archive = buildPath (getTestSamplesDir (), "test.tar.xz");
