diff --git a/src/asgen/config.d b/src/asgen/config.d
index 7ed01d1..a5daaff 100644
--- a/src/asgen/config.d
+++ b/src/asgen/config.d
@@ -147,7 +147,7 @@ private:
         import glib.Util : Util;
 
         // our default export format version
-        formatVersion = FormatVersion.V0_12;
+        formatVersion = FormatVersion._12;
 
         // find all the external binaries we (may) need
         // we search for them unconditionally, because the unittests may rely on their absolute
@@ -370,19 +370,19 @@ public:
 
             switch (versionStr) {
             case "0.8":
-                formatVersion = FormatVersion.V0_8;
+                formatVersion = FormatVersion._8;
                 break;
             case "0.9":
-                formatVersion = FormatVersion.V0_9;
+                formatVersion = FormatVersion._9;
                 break;
             case "0.10":
-                formatVersion = FormatVersion.V0_10;
+                formatVersion = FormatVersion._10;
                 break;
             case "0.11":
-                formatVersion = FormatVersion.V0_11;
+                formatVersion = FormatVersion._11;
                 break;
             case "0.12":
-                formatVersion = FormatVersion.V0_12;
+                formatVersion = FormatVersion._12;
                 break;
             default:
                 logWarning ("Configuration tried to set unknown AppStream format version '%s'. Falling back to default version.", versionStr);
diff --git a/src/asgen/handlers/iconhandler.d b/src/asgen/handlers/iconhandler.d
index 7f0746c..a2465c2 100644
--- a/src/asgen/handlers/iconhandler.d
+++ b/src/asgen/handlers/iconhandler.d
@@ -407,7 +407,7 @@ public:
         // quickly identify it here.
         auto icon = componentGetRawIcon (cpt);
         if (!icon.isNull) {
-            if (icon.getKind == IconKind.LOCAL)
+            if (icon.get.getKind == IconKind.LOCAL)
                 name = icon.get.getFilename ();
             else
                 name = icon.get.getName ();
