diff --git a/src/core/checkedint.d b/src/core/checkedint.d
index 18e19bf..ccaeec3 100644
--- a/src/core/checkedint.d
+++ b/src/core/checkedint.d
@@ -447,7 +447,7 @@ unittest
 
 int muls(int x, int y, ref bool overflow)
 {
-    version(LDC)
+    version(none)
     {
         auto res = llvm_smul_with_overflow(x, y);
         overflow |= res.overflow;
@@ -485,7 +485,7 @@ unittest
 /// ditto
 long muls(long x, long y, ref bool overflow)
 {
-    version(LDC)
+    version(none)
     {
         auto res = llvm_smul_with_overflow(x, y);
         overflow |= res.overflow;
@@ -536,7 +536,7 @@ unittest
 
 uint mulu(uint x, uint y, ref bool overflow)
 {
-    version(LDC)
+    version(none)
     {
         auto res = llvm_umul_with_overflow(x, y);
         overflow |= res.overflow;
@@ -574,7 +574,7 @@ unittest
 /// ditto
 ulong mulu(ulong x, ulong y, ref bool overflow)
 {
-    version(LDC)
+    version(none)
     {
         auto res = llvm_umul_with_overflow(x, y);
         overflow |= res.overflow;
