diff --git a/hphp/runtime/server/fastcgi/fastcgi-transport.cpp b/hphp/runtime/server/fastcgi/fastcgi-transport.cpp
--- a/hphp/runtime/server/fastcgi/fastcgi-transport.cpp
+++ b/hphp/runtime/server/fastcgi/fastcgi-transport.cpp
@@ -298,7 +298,7 @@
   }

   // IIS sets this value but sets it to off when SSL is off.
-  if (m_requestParams.count("HTTPS") &&
+  if (m_requestParams.count("HTTPS") && !m_requestParams["HTTPS"].empty() &&
       strcasecmp(m_requestParams["HTTPS"].c_str(), "OFF")) {
     setSSL();
   }
diff --git a/hphp/test/server/fastcgi/server_root/test_https.php b/hphp/test/server/fastcgi/server_root/test_https.php
new file mode 100644
--- /dev/null
+++ b/hphp/test/server/fastcgi/server_root/test_https.php
@@ -0,0 +1,2 @@
+<?php
+var_dump($_SERVER['HTTPS']);
diff --git a/hphp/test/server/fastcgi/tests/httpsTest.php b/hphp/test/server/fastcgi/tests/httpsTest.php
new file mode 100644
--- /dev/null
+++ b/hphp/test/server/fastcgi/tests/httpsTest.php
@@ -0,0 +1,11 @@
+<?php
+
+require_once('test_base.inc');
+
+function httpsTestController($serverPort) {
+  $args = array('HTTPS' => '');
+  var_dump(request(php_uname('n'), $serverPort, "test_https.php",
+                  [], [], $args));
+}
+
+runTest("httpsTestController");
diff --git a/hphp/test/server/fastcgi/tests/httpsTest.php.expect b/hphp/test/server/fastcgi/tests/httpsTest.php.expect
new file mode 100644
--- /dev/null
+++ b/hphp/test/server/fastcgi/tests/httpsTest.php.expect
@@ -0,0 +1 @@
+string(12) "string(0) """
