--- modules/readline/manifests/init.pp
+++ modules/readline/manifests/init.pp
@@ -61,6 +61,7 @@ class readline(
     cwd              => $source_dir_path,
     environment      => $real_autotools_environment,
     install_sentinel => "${prefix}/lib/libreadline.a",
+    make_command     => "make SHLIB_LIBS=-lncurses",
     make_notify      => $make_notify,
     make_sentinel    => "${source_dir_path}/libreadline.a",
     require          => Exec["untar-readline"]
