Index: libegg/pixbufthumbnail/Makefile.am
===================================================================
--- libegg/pixbufthumbnail/Makefile.am	(revision 938)
+++ libegg/pixbufthumbnail/Makefile.am	(working copy)
@@ -26,7 +26,7 @@
 
 testeggpixbufthumbnail_LDADD = \
 	$(EGG_LIBS) \
-	$(top_builddir)/libegg/pixbufthumbnail/libeggpixbufthumbnail.la
+	libeggpixbufthumbnail.la
 
 testeggpixbufthumbnail_SOURCES = \
 	testeggpixbufthumbnail.c
