diff --git a/libraries/cmake/system/modules/Findlibdevmapper.cmake b/libraries/cmake/system/modules/Findlibdevmapper.cmake
index 34eb119959..47b5f566a2 100644
--- a/libraries/cmake/system/modules/Findlibdevmapper.cmake
+++ b/libraries/cmake/system/modules/Findlibdevmapper.cmake
@@ -4,4 +4,12 @@
 # This source code is licensed in accordance with the terms specified in
 # the LICENSE file found in the root directory of this source tree.
 
-include("${CMAKE_SOURCE_DIR}/libraries/cmake/source/modules/Findlibdevmapper.cmake")
+cmake_minimum_required(VERSION 3.15)
+include("${CMAKE_CURRENT_LIST_DIR}/api.cmake")
+
+locateSystemLibrary(
+  NAME libdevmapper
+  REQUIRED ${libdevmapper_REQUIRED}
+  MAIN_HEADER "libdevmapper.h"
+  LIBRARIES "devmapper" "lvm2app"
+)
