diff -urN boinc-6.12.26.old//configure.ac boinc-6.12.26//configure.ac
--- boinc-6.12.26.old//configure.ac	2011-05-22 09:40:36.392051762 +0200
+++ boinc-6.12.26//configure.ac	2011-05-22 11:03:28.687889510 +0200
@@ -887,7 +887,6 @@
                  apps/Makefile
                  clientgui/Makefile
                  clientgui/res/Makefile
-                 clientscr/Makefile
                  client/Makefile
                  client/win/boinc_path_config.py:py/boinc_path_config.py.in
                  client/scripts/Makefile
diff -urN boinc-6.12.26.old//Makefile.am boinc-6.12.26//Makefile.am
--- boinc-6.12.26.old//Makefile.am	2011-05-22 09:40:36.376051962 +0200
+++ boinc-6.12.26//Makefile.am	2011-05-22 11:03:16.277044666 +0200
@@ -21,10 +21,7 @@
 endif
 
 if ENABLE_MANAGER
-	CLIENTGUI_SUBDIRS = clientgui clientgui/res locale
-	if BUILD_X11_SCREENSAVER
-		CLIENTGUI_SUBDIRS += clientscr
-	endif 
+   CLIENTGUI_SUBDIRS = clientgui clientgui/res locale
 endif
 
 # ORDER MATTERS below.  One must build dependencies FIRST, then things
