--- ffmpeg2theora-0.30/SConstruct.orig	2019-07-26 18:13:38.624612595 +0000
+++ ffmpeg2theora-0.30/SConstruct	2019-07-26 18:15:11.695727967 +0000
@@ -21,7 +21,7 @@
 try:
     EnsureSConsVersion(*scons_version)
 except TypeError:
-    print 'SCons %d.%d.%d or greater is required, but you have an older version' % scons_version
+    print ('SCons %d.%d.%d or greater is required, but you have an older version' % scons_version)
     Exit(2)
 
 opts = Variables()
@@ -119,29 +119,29 @@
 if not env.GetOption('clean'):
   pkgconfig_version='0.15.0'
   if not conf.CheckPKGConfig(pkgconfig_version): 
-     print 'pkg-config >= %s not found.' % pkgconfig_version 
+     print ('pkg-config >= %s not found.' % pkgconfig_version) 
      Exit(1)
 
   if not conf.CheckPKG("ogg >= 1.1"): 
-    print 'ogg >= 1.1 missing'
+    print ('ogg >= 1.1 missing')
     Exit(1) 
 
   if not conf.CheckPKG("vorbis"): 
-    print 'vorbis missing'
+    print ('vorbis missing')
     Exit(1) 
 
   if not conf.CheckPKG("vorbisenc"): 
-    print 'vorbisenc missing'
+    print ('vorbisenc missing')
     Exit(1) 
 
   if not conf.CheckPKG("theoraenc >= 1.1.0"): 
-    print 'theoraenc >= 1.1.0 missing'
+    print ('theoraenc >= 1.1.0 missing')
     Exit(1) 
 
   XIPH_LIBS="ogg >= 1.1 vorbis vorbisenc theoraenc >= 1.1.0"
 
   if not conf.CheckPKG(XIPH_LIBS): 
-    print 'some xiph libs are missing, ffmpeg2theora depends on %s' % XIPH_LIBS
+    print ('some xiph libs are missing, ffmpeg2theora depends on %s' % XIPH_LIBS)
     Exit(1) 
   ParsePKGConfig(env, XIPH_LIBS)
 
@@ -177,13 +177,13 @@
     FFMPEG_LIBS.append('libavresample')
 
   if not conf.CheckPKG(' '.join(FFMPEG_LIBS)): 
-    print """
+    print ("""
         Could not find %s.
         You can install it via
          sudo apt-get install %s
         or update PKG_CONFIG_PATH to point to ffmpeg's source folder
         or run ./get_ffmpeg.sh (for more information see INSTALL)
-    """ %(" ".join(FFMPEG_LIBS), " ".join(["%s-dev"%l.split()[0] for l in FFMPEG_LIBS]))
+    """ %(" ".join(FFMPEG_LIBS), " ".join(["%s-dev"%l.split()[0] for l in FFMPEG_LIBS])))
     Exit(1) 
 
   for lib in FFMPEG_LIBS:
@@ -204,11 +204,11 @@
       ParsePKGConfig(env, KATE_LIBS)
       env.Append(CCFLAGS=['-DHAVE_KATE', '-DHAVE_OGGKATE'])
     else:
-      print """
+      print ("""
           Could not find libkate. Subtitles support will be disabled.
           You can also run ./get_libkate.sh (for more information see INSTALL)
           or update PKG_CONFIG_PATH to point to libkate's source folder
-      """
+      """)
 
   if conf.CheckCHeader('iconv.h'):
       env.Append(CCFLAGS=[
