diff -ruN a/Makefile.in b/Makefile.in
--- a/Makefile.in	2021-05-09 14:39:59.000000000 +0200
+++ b/Makefile.in	2021-06-03 21:10:29.658438256 +0200
@@ -71,8 +71,8 @@
 am__make_keepgoing = (target_option=k; $(am__make_running_with_option))
 pkgdatadir = $(datadir)/@PACKAGE@
 pkgincludedir = $(includedir)/@PACKAGE@
-pkglibdir = $(libdir)/@PACKAGE@
-pkglibexecdir = $(libexecdir)/@PACKAGE@
+pkglibdir = $(libdir)/
+pkglibexecdir = $(libexecdir)/
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
 install_sh_DATA = $(install_sh) -c -m 644
 install_sh_PROGRAM = $(install_sh) -c
diff -ruN a/doc/Makefile.in b/doc/Makefile.in
--- a/doc/Makefile.in	2021-05-09 14:39:59.000000000 +0200
+++ b/doc/Makefile.in	2021-06-03 21:02:44.757057009 +0200
@@ -71,8 +71,8 @@
 am__make_keepgoing = (target_option=k; $(am__make_running_with_option))
 pkgdatadir = $(datadir)/@PACKAGE@
 pkgincludedir = $(includedir)/@PACKAGE@
-pkglibdir = $(libdir)/@PACKAGE@
-pkglibexecdir = $(libexecdir)/@PACKAGE@
+pkglibdir = $(libdir)/
+pkglibexecdir = $(libexecdir)/
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
 install_sh_DATA = $(install_sh) -c -m 644
 install_sh_PROGRAM = $(install_sh) -c
diff -ruN a/doc/quickstart/Makefile.in b/doc/quickstart/Makefile.in
--- a/doc/quickstart/Makefile.in	2021-05-09 14:39:59.000000000 +0200
+++ b/doc/quickstart/Makefile.in	2021-06-03 21:10:42.911811038 +0200
@@ -72,8 +72,8 @@
 am__make_keepgoing = (target_option=k; $(am__make_running_with_option))
 pkgdatadir = $(datadir)/@PACKAGE@
 pkgincludedir = $(includedir)/@PACKAGE@
-pkglibdir = $(libdir)/@PACKAGE@
-pkglibexecdir = $(libexecdir)/@PACKAGE@
+pkglibdir = $(libdir)/
+pkglibexecdir = $(libexecdir)/
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
 install_sh_DATA = $(install_sh) -c -m 644
 install_sh_PROGRAM = $(install_sh) -c
diff -ruN a/gfx/Makefile.in b/gfx/Makefile.in
--- a/gfx/Makefile.in	2021-05-09 14:39:59.000000000 +0200
+++ b/gfx/Makefile.in	2021-06-03 21:10:53.145174831 +0200
@@ -72,8 +72,8 @@
 am__make_keepgoing = (target_option=k; $(am__make_running_with_option))
 pkgdatadir = $(datadir)/@PACKAGE@
 pkgincludedir = $(includedir)/@PACKAGE@
-pkglibdir = $(libdir)/@PACKAGE@
-pkglibexecdir = $(libexecdir)/@PACKAGE@
+pkglibdir = $(libdir)/
+pkglibexecdir = $(libexecdir)/
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
 install_sh_DATA = $(install_sh) -c -m 644
 install_sh_PROGRAM = $(install_sh) -c
diff -ruN a/src/Makefile.in b/src/Makefile.in
--- a/src/Makefile.in	2021-05-09 14:39:59.000000000 +0200
+++ b/src/Makefile.in	2021-06-03 21:11:06.048546576 +0200
@@ -71,8 +71,8 @@
 am__make_keepgoing = (target_option=k; $(am__make_running_with_option))
 pkgdatadir = $(datadir)/@PACKAGE@
 pkgincludedir = $(includedir)/@PACKAGE@
-pkglibdir = $(libdir)/@PACKAGE@
-pkglibexecdir = $(libexecdir)/@PACKAGE@
+pkglibdir = $(libdir)/
+pkglibexecdir = $(libexecdir)/
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
 install_sh_DATA = $(install_sh) -c -m 644
 install_sh_PROGRAM = $(install_sh) -c
diff -ruN a/src/gigedit/Makefile.in b/src/gigedit/Makefile.in
--- a/src/gigedit/Makefile.in	2021-05-09 14:40:00.000000000 +0200
+++ b/src/gigedit/Makefile.in	2021-06-03 21:11:19.035251903 +0200
@@ -73,8 +73,8 @@
 am__make_keepgoing = (target_option=k; $(am__make_running_with_option))
 pkgdatadir = $(datadir)/@PACKAGE@
 pkgincludedir = $(includedir)/@PACKAGE@
-pkglibdir = $(libdir)/@PACKAGE@
-pkglibexecdir = $(libexecdir)/@PACKAGE@
+pkglibdir = $(libdir)/
+pkglibexecdir = $(libexecdir)/
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
 install_sh_DATA = $(install_sh) -c -m 644
 install_sh_PROGRAM = $(install_sh) -c
diff -ruN a/src/gigedit/gfx/Makefile.in b/src/gigedit/gfx/Makefile.in
--- a/src/gigedit/gfx/Makefile.in	2021-05-09 14:40:00.000000000 +0200
+++ b/src/gigedit/gfx/Makefile.in	2021-06-03 21:11:38.128642081 +0200
@@ -72,8 +72,8 @@
 am__make_keepgoing = (target_option=k; $(am__make_running_with_option))
 pkgdatadir = $(datadir)/@PACKAGE@
 pkgincludedir = $(includedir)/@PACKAGE@
-pkglibdir = $(libdir)/@PACKAGE@
-pkglibexecdir = $(libexecdir)/@PACKAGE@
+pkglibdir = $(libdir)/
+pkglibexecdir = $(libexecdir)/
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
 install_sh_DATA = $(install_sh) -c -m 644
 install_sh_PROGRAM = $(install_sh) -c
