diff -rupN a/gif.c b/gif.c
--- a/gif.c	2003-08-24 22:23:02.000000000 +0200
+++ b/gif.c	2013-07-30 10:27:55.851099159 +0200
@@ -81,7 +81,7 @@ int fh_gif_load(char *name,unsigned char
     ColorMapObject *cmap;
     int cmaps;
 
-    gft=DGifOpenFileName(name);
+    gft=DGifOpenFileName(name, NULL);
     if(gft==NULL){printf("err5\n"); gflush;} //////////
     do
     {
@@ -184,7 +184,7 @@ int fh_gif_getsize(char *name,int *x,int
     int extcode;
     GifRecordType rt;
 
-    gft=DGifOpenFileName(name);
+    gft=DGifOpenFileName(name, NULL);
     if(gft==NULL) gflush;
     do
     {
