--- libsanitizer/tsan/tsan_platform_linux.cc.orig	2018-02-03 18:26:53.153703109 +0100
+++ libsanitizer/tsan/tsan_platform_linux.cc	2018-02-03 18:27:30.326068569 +0100
@@ -377,7 +377,7 @@
 int ExtractResolvFDs(void *state, int *fds, int nfd) {
 #if SANITIZER_LINUX
   int cnt = 0;
-  __res_state *statp = (__res_state*)state;
+  res_state statp = (res_state)state;
   for (int i = 0; i < MAXNS && cnt < nfd; i++) {
     if (statp->_u._ext.nsaddrs[i] && statp->_u._ext.nssocks[i] != -1)
       fds[cnt++] = statp->_u._ext.nssocks[i];
