diff --git a/base/binaryplatforms.jl b/base/binaryplatforms.jl
index d70cca9c48..1d72853845 100644
--- a/base/binaryplatforms.jl
+++ b/base/binaryplatforms.jl
@@ -903,7 +903,7 @@ function detect_cxxstring_abi()
     end
 
     function open_libllvm(f::Function)
-        for lib_name in ("libLLVM-13jl", "libLLVM", "LLVM", "libLLVMSupport")
+        for lib_name in ("libLLVM-13jl", "libLLVM", "LLVM", "libLLVMSupport", "libLLVM-14")
             hdl = Libdl.dlopen_e(lib_name)
             if hdl != C_NULL
                 try
