From f5d99185002284440947f8eac7b216ac77b4f2e4 Mon Sep 17 00:00:00 2001
From: Steven Roberts <strobert@strobe.net>
Date: Sun, 7 Jul 2013 19:07:14 -0700
Subject: [PATCH] added missing =back.

---
 rsnapshot-program.pl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rsnapshot-program.pl b/rsnapshot-program.pl
index 85972fd..95b7031 100755
--- a/rsnapshot-program.pl
+++ b/rsnapshot-program.pl
@@ -6609,6 +6609,8 @@ additional disk space will be taken up.
 
 =back
 
+=back
+
 Remember that tabs must separate all elements, and that
 there must be a trailing slash on the end of every directory.
 
-- 
1.9.0

