diff -Naur tasque-0.1.9.orig/configure.ac tasque-0.1.9/configure.ac
--- tasque-0.1.9.orig/configure.ac	2012-01-11 22:29:05.377456060 +0100
+++ tasque-0.1.9/configure.ac	2012-01-11 22:34:31.094387156 +0100
@@ -30,12 +30,6 @@
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Gettext package])
 AM_GLIB_GNU_GETTEXT
 
-#
-# Find gconftool-2
-#
-AC_PATH_PROG(GCONFTOOL, gconftool-2)
-AM_GCONF_SOURCE_2
-
 PKG_CHECK_MODULES([GLIB_SHARP_20], [glib-sharp-2.0])
 PKG_CHECK_MODULES([GTK_DOTNET_20], [gtk-dotnet-2.0])
 PKG_CHECK_MODULES([GTK_SHARP_20], [gtk-sharp-2.0 >= 2.10.0])
