diff --git a/configure.ac b/configure.ac
index 59d1838..35c0ba8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -341,21 +341,6 @@ else
 fi
 
 SADC_DIR=$AuxPrefix/lib
-if test -d $AuxPrefix/lib64; then
-    # Verify that this OS is really 64 bit
-    BITS=$(getconf LONG_BIT 2>/dev/null)
-    if test $? = 0; then
-        if test $BITS = 64; then
-            SADC_DIR=$AuxPrefix/lib64
-        fi
-    else
-        # Fallback: look for lm (long mode) flag to know if CPU is 64 bit
-        grep " lm " /proc/cpuinfo >/dev/null 2>&1
-        if test $? = 0; then
-            SADC_DIR=$AuxPrefix/lib64
-        fi
-    fi
-fi
 
 AC_MSG_CHECKING(sadc directory)
 AC_ARG_VAR([sa_lib_dir],[sadc directory])
