--- Makefile.orig	2022-06-26 00:45:01.000000000 +0300
+++ Makefile	2022-07-02 00:51:03.816042128 +0300
@@ -40,7 +40,7 @@
 
 include $(GLKINCLUDEDIR)/$(GLKMAKEFILE)
 
-CFLAGS = $(OPTIONS) -I$(GLKINCLUDEDIR) $(XMLLIBINCLUDEDIR)
+CFLAGS += $(OPTIONS) -I$(GLKINCLUDEDIR) $(XMLLIBINCLUDEDIR)
 LIBS = -L$(GLKLIBDIR) $(GLKLIB) $(LINKLIBS) -lm $(XMLLIB)
 
 OBJS = main.o files.o vm.o exec.o funcs.o operand.o string.o glkop.o \
@@ -50,7 +50,7 @@
 all: glulxe
 
 glulxe: $(OBJS) unixstrt.o unixautosave.o
-	$(CC) $(OPTIONS) -o glulxe $(OBJS) unixstrt.o unixautosave.o $(LIBS)
+	$(CC) ${CFLAGS} ${LDFLAGS} -o glulxe $(OBJS) unixstrt.o unixautosave.o $(LIBS)
 
 glulxdump: glulxdump.o
 	$(CC) -o glulxdump glulxdump.o
@@ -62,4 +62,3 @@
 
 clean:
 	rm -f *~ *.o glulxe glulxdump profile-raw
-
