diff --git a/factory/utils.py b/factory/utils.py
index 75f352b..b0407e8 100644
--- a/factory/utils.py
+++ b/factory/utils.py
@@ -133,7 +133,10 @@ class ResetableIterator(object):
             if self.next_elements:
                 yield self.next_elements.popleft()
             else:
-                value = next(self.iterator)
+                try:
+                    value = next(self.iterator)
+                except StopIteration:
+                    return
                 self.past_elements.append(value)
                 yield value
 
