From e191553d1dc80cd6d65d05f0cb29f8967fab6983 Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay.sievers@vrfy.org>
Date: Wed, 17 Aug 2011 19:38:07 +0200
Subject: [PATCH] convert int to boolean for dbus_bool_t

---
 src/dbus-manager.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/dbus-manager.c b/src/dbus-manager.c
index ae88895..cfc2afc 100644
--- a/src/dbus-manager.c
+++ b/src/dbus-manager.c
@@ -444,7 +444,7 @@ static DBusMessage *message_from_file_changes(
         if (carries_install_info >= 0) {
                 dbus_bool_t b;
 
-                b = carries_install_info;
+                b = !!carries_install_info;
                 if (!dbus_message_iter_append_basic(&iter, DBUS_TYPE_BOOLEAN, &b))
                         goto oom;
         }
-- 
1.7.6

