diff -aur antlr4-4.8-old/runtime/Python3/test/ctest.py antlr4-4.8-new/runtime/Python3/test/ctest.py
--- antlr4-4.8-old/runtime/Python3/test/ctest.py	2020-01-19 18:18:55.000000000 +0200
+++ antlr4-4.8-new/runtime/Python3/test/ctest.py	2020-06-30 16:32:13.862026103 +0300
@@ -7,8 +7,8 @@
 import sys
 sys.setrecursionlimit(4000)
 import antlr4
-from parser.cparser import CParser
-from parser.clexer import CLexer
+from .parser.cparser import CParser
+from .parser.clexer import CLexer
 from datetime import datetime
 import cProfile
 
diff -aur antlr4-4.8-old/runtime/Python3/test/run.py antlr4-4.8-new/runtime/Python3/test/run.py
--- antlr4-4.8-old/runtime/Python3/test/run.py	2020-01-19 18:18:55.000000000 +0200
+++ antlr4-4.8-new/runtime/Python3/test/run.py	2020-06-30 16:32:26.675359328 +0300
@@ -3,6 +3,6 @@
 src_path = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), 'src')
 sys.path.insert(0,src_path)
 from TestTokenStreamRewriter import TestTokenStreamRewriter
-from xpathtest import XPathTest
+from .xpathtest import XPathTest
 import unittest
-unittest.main()
\ No newline at end of file
+unittest.main()
diff -aur antlr4-4.8-old/runtime/Python3/test/TestTokenStreamRewriter.py antlr4-4.8-new/runtime/Python3/test/TestTokenStreamRewriter.py
--- antlr4-4.8-old/runtime/Python3/test/TestTokenStreamRewriter.py	2020-01-19 18:18:55.000000000 +0200
+++ antlr4-4.8-new/runtime/Python3/test/TestTokenStreamRewriter.py	2020-06-30 16:32:36.462025917 +0300
@@ -5,7 +5,7 @@
 import unittest
 
 
-from mocks.TestLexer import TestLexer, TestLexer2
+from .mocks.TestLexer import TestLexer, TestLexer2
 from antlr4.TokenStreamRewriter import TokenStreamRewriter
 from antlr4.InputStream import InputStream
 from antlr4.CommonTokenStream import CommonTokenStream
diff -aur antlr4-4.8-old/runtime/Python3/test/xpathtest.py antlr4-4.8-new/runtime/Python3/test/xpathtest.py
--- antlr4-4.8-old/runtime/Python3/test/xpathtest.py	2020-01-19 18:18:55.000000000 +0200
+++ antlr4-4.8-new/runtime/Python3/test/xpathtest.py	2020-06-30 16:32:44.398692519 +0300
@@ -2,8 +2,8 @@
 from antlr4 import InputStream, CommonTokenStream, TerminalNode
 from antlr4.xpath.XPath import XPath
 import unittest
-from expr.ExprParser import ExprParser
-from expr.ExprLexer  import ExprLexer
+from .expr.ExprParser import ExprParser
+from .expr.ExprLexer  import ExprLexer
 
 def tokenToString(token, ruleNames):
     if isinstance(token, TerminalNode):
