--- ggadget/gtk/main_loop.cc
+++ ggadget/gtk/main_loop.cc
@@ -16,9 +16,8 @@
 
 #include <fcntl.h>
 #include <stdint.h>
-#include <glib/ghash.h>
+#include <glib.h>
 #include <gtk/gtk.h>
-#include <glib/gthread.h>
 #include <ggadget/common.h>
 #include "main_loop.h"
 #include "utilities.h"
--- hosts/gtk/main.cc
+++ hosts/gtk/main.cc
@@ -14,8 +14,8 @@
   limitations under the License.
 */
 
+#include <glib.h>
 #include <gtk/gtk.h>
-#include <glib/gthread.h>
 #include <locale.h>
 #include <signal.h>
 #include <unistd.h>
