diff --git a/gcc/d/Make-lang.in b/gcc/d/Make-lang.in
index 9570bf3..52addc0 100644
--- a/gcc/d/Make-lang.in
+++ b/gcc/d/Make-lang.in
@@ -25,10 +25,10 @@ D_TARGET_INSTALL_NAME = $(target_alias)-$(shell echo gdc|sed '$(program_transfor
 
 # This should be configured
 ifeq ($(host), $(target))
-  D_include_dir = `echo $(exec_prefix) | sed -e 's|^$(prefix)||' -e 's|/[^/]*|/..|g'`/include/d/$(version)
+  D_include_dir = `echo $(exec_prefix) | sed -e 's|^$(prefix)||' -e 's|/[^/]*|/..|g'`/include/dlang/gdc/$(version)
   gcc_d_include_dir = $(libsubdir)/$(unlibsubdir)/..$(D_include_dir)
 else
-  gcc_d_include_dir = $(libsubdir)/include/d
+  gcc_d_include_dir = $(libsubdir)/include/dlang/gdc
 endif
 
 
diff --git a/libphobos/configure.ac b/libphobos/configure.ac
index dfb8fec..0b88ff4 100644
--- a/libphobos/configure.ac
+++ b/libphobos/configure.ac
@@ -220,9 +220,9 @@ dnl (# Default case for install directory for include files.) and on
 # will have to modify gcc/configure.ac ..
 # For now, basic workaround for cross compilers ..
 if test "${host}" != "${build}"; then
-    gdc_include_dir='${libdir}/gcc/${host_alias}'/${d_gcc_ver}/include/d
+    gdc_include_dir='${libdir}/gcc/${host_alias}'/${d_gcc_ver}/include/dlang/
 else
-    gdc_include_dir='${prefix}'/include/d/${d_gcc_ver}
+    gdc_include_dir='${prefix}'/include/dlang/gdc/${d_gcc_ver}
 fi
 AC_SUBST(gdc_include_dir)
 AC_ARG_WITH([cross-host],
