--- a/src/avbin.c	2008-09-20 23:45:33.000000000 -0700
+++ b/src/avbin.c	2010-04-09 09:11:12.000000000 -0700
@@ -28,6 +28,7 @@
 #include <libavformat/avformat.h>
 #include <libavcodec/avcodec.h>
 #include <libavutil/avutil.h>
+#include <libavutil/log.h>
 
 struct _AVbinFile {
     AVFormatContext *context;
@@ -56,7 +56,7 @@
     static char message[8192];
     const char *module = NULL;
 
-    if (level > av_log_level || !user_log_callback)
+    if (level > av_log_get_level() || !user_log_callback)
         return;
 
     if (ptr)
@@ -98,7 +98,7 @@
 
 AVbinResult avbin_set_log_level(AVbinLogLevel level)
 {
-    av_log_level = level;
+    av_log_set_level(level);
     return AVBIN_RESULT_OK;
 }
 
