diff --git a/src/interface.c b/src/interface.c
index 52c6727..62e89ff 100644
--- a/src/interface.c
+++ b/src/interface.c
@@ -545,10 +545,10 @@ gint compare_size_list_item(GtkTreeModel *model, GtkTreeIter *iter1, GtkTreeIter
     gtk_tree_model_get(model, iter1, column, &s1, -1);
     gtk_tree_model_get(model, iter2, column, &s2, -1);
 
-    gint64 i1 = string_to_size(s1);
-    gint64 i2 = string_to_size(s2);
+    guint64 i1 = string_to_size(s1);
+    guint64 i2 = string_to_size(s2);
 
-    ret = i2 - i1;
+    ret = (i2 > i1) - (i2 < i1);
 
     g_free(s1);
     g_free(s2);
