diff -wbBur calibre/src/calibre/ebooks/pdf/reflow.cpp calibre.my/src/calibre/ebooks/pdf/reflow.cpp
--- calibre/src/calibre/ebooks/pdf/reflow.cpp	2012-05-11 08:10:22.000000000 +0400
+++ calibre.my/src/calibre/ebooks/pdf/reflow.cpp	2012-05-18 15:31:10.000000000 +0400
@@ -647,7 +647,7 @@
 
 
 void XMLOutputDev::endPage() {
-    Links *slinks = catalog->getPage(current_page->number())->getLinks(catalog);
+    Links *slinks = catalog->getPage(current_page->number())->getLinks();
     for (int i = 0; i < slinks->getNumLinks(); i++)
     {
         this->process_link(slinks->getLink(i));
@@ -878,7 +878,7 @@
         throw ReflowException("Failed to allocate SplashOutputDev");
     }
     try {
-        out->startDoc(doc->getXRef());
+        out->startDoc((PDFDoc*)doc->getXRef());
         out->startPage(1, NULL);
 
         double pg_w, pg_h;
