From 91264d8b8b746f7eee8a3e1ee9b5e57758441cfc Mon Sep 17 00:00:00 2001
From: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>
Date: Mon, 20 Jan 2014 15:56:21 +0000
Subject: build: Ensure pkg-config file provides the right informations

It's not -lnftables anymore but -lnftnl (among other details)

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>
Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
---
diff --git a/libnftnl.pc.in b/libnftnl.pc.in
index 0c6d57e..fd5cc6a 100644
--- a/libnftnl.pc.in
+++ b/libnftnl.pc.in
@@ -1,15 +1,15 @@
-# libnftables pkg-config file
+# libnftnl pkg-config file
 
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
 
-Name: libnftables
+Name: libnftnl
 Description: Netfilter nf_tables infrastructure library
-URL: http://netfilter.org/projects/libnftables/
+URL: http://netfilter.org/projects/libnftnl/
 Version: @VERSION@
 Requires:
 Conflicts:
-Libs: -L${libdir} -lnftables
+Libs: -L${libdir} -lnftnl
 Cflags: -I${includedir}
--
cgit v0.9.2
