diff --git a/cmake/SundialsTPLOptions.cmake b/cmake/SundialsTPLOptions.cmake
index 6445b367..e992035a 100644
--- a/cmake/SundialsTPLOptions.cmake
+++ b/cmake/SundialsTPLOptions.cmake
@@ -275,8 +275,7 @@ sundials_option(ONEMKL_WORKS BOOL "Set to ON to force CMake to accept a given on
 # Enable Caliper support?
 # ---------------------------------------------------------------
 
-sundials_option(ENABLE_CALIPER BOOL "Enable CALIPER support" OFF
-                DEPENDS_ON SUNDIALS_BUILD_WITH_PROFILING)
+sundials_option(ENABLE_CALIPER BOOL "Enable CALIPER support" OFF)
 
 sundials_option(CALIPER_DIR PATH "Path to the root of an CALIPER installation" "${CALIPER_DIR}"
                 DEPENDS_ON ENABLE_CALIPER)
