--- cfn-python-lint-0.16.0/src/cfnlint/config.py.orig	2019-03-14 22:41:20.817312234 +0800
+++ cfn-python-lint-0.16.0/src/cfnlint/config.py	2019-03-14 22:41:40.973955815 +0800
@@ -456,7 +456,7 @@
                     if add_filename not in ignore_templates:
                         all_filenames.append(add_filename)
 
-        return all_filenames
+        return sorted(all_filenames)
 
     def _ignore_templates(self):
         """ templates """
