diff --git a/src/cfnlint/config.py b/src/cfnlint/config.py
index a130acb..2a5f4c9 100644
--- a/src/cfnlint/config.py
+++ b/src/cfnlint/config.py
@@ -448,7 +448,7 @@ class ConfigMixIn(TemplateArgs, CliArgs, ConfigFileArgs, object):
             else:
                 all_filenames.extend(add_filenames)
 
-        return all_filenames
+        return sorted(all_filenames)
 
     @property
     def append_rules(self):
