commit 1bf6c98f3f7513b18b8eed5c666122fe1d56b9bd
Author: Anatol Pomozov <anatol.pomozov@gmail.com>
Date:   Sat Mar 21 09:13:31 2020 -0700

    ArchLinux fixes

diff --git a/Gemfile b/Gemfile
index 90abfd30900..3a14b21f9b1 100644
--- a/Gemfile
+++ b/Gemfile
@@ -317,6 +317,12 @@ gem 'gettext_i18n_rails', '~> 1.8.0'
 gem 'gettext_i18n_rails_js', '~> 1.3'
 gem 'gettext', '~> 3.3', require: false, group: :development
 
+# Ping the gems that are needed for the old gettext
+# https://github.com/ruby-gettext/gettext/issues/64 the fix is available in gettext 3.3.0
+gem 'irb', '~> 1.1.0'  # use irb version that matches ruby 2.6. Later irb versions heavily modified its API and it breaks gettext
+gem 'e2mmap' # 'irb' 1.1.1 requires it
+
+
 gem 'batch-loader', '~> 1.4.0'
 
 # Perf bar
