diff -r 8c7296ca3f1c bpython/cli.py
--- a/bpython/cli.py	Fri Jul 13 21:18:28 2012 +0200
+++ b/bpython/cli.py	Sat Oct 20 21:20:03 2012 -0400
@@ -1714,7 +1714,8 @@
         caller.scr.nodelay(True)
         key = caller.scr.getch()
         caller.scr.nodelay(False)
-        curses.ungetch(key)
+        if key != -1:
+            curses.ungetch(key)
     caller.statusbar.check()
     caller.check()
