From 5f4f1ed45747343f26f280401bc4915d92b71d89 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Mon, 30 Sep 2013 12:06:22 +0000
Subject: updates: Raise the dep on PackageKit to avoid complexity

---
diff --git a/configure.ac b/configure.ac
index 87eb473..199542c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -306,7 +306,7 @@ AC_ARG_ENABLE(packagekit,
        [WANT_PACKAGEKIT=yes]) dnl Default value
 
 if test x$WANT_PACKAGEKIT = xyes ; then
-       PK_REQUIRED_VERSION=0.7.4
+       PK_REQUIRED_VERSION=0.8.1
        PKG_CHECK_MODULES(PACKAGEKIT, glib-2.0 packagekit-glib2 >= $PK_REQUIRED_VERSION upower-glib >= $UPOWER_REQUIRED_VERSION gudev-1.0 libnotify >= $LIBNOTIFY_REQUIRED_VERSION,
              [have_packagekit=true
               AC_DEFINE(HAVE_PACKAGEKIT, 1, [Define if PackageKit should be used])],
diff --git a/plugins/updates/gsd-updates-manager.c b/plugins/updates/gsd-updates-manager.c
index 3aff29a..0f73a42 100644
--- a/plugins/updates/gsd-updates-manager.c
+++ b/plugins/updates/gsd-updates-manager.c
@@ -742,7 +742,6 @@ auto_download_updates (GsdUpdatesManager *manager)
                 package_ids[i] = g_strdup (pk_package_get_id (pkg));
         }
 
-#if PK_CHECK_VERSION(0,8,1)
         /* we've set only-download in PkTask */
         pk_task_update_packages_async (manager->priv->task,
                                        package_ids,
@@ -750,16 +749,6 @@ auto_download_updates (GsdUpdatesManager *manager)
                                        NULL, NULL,
                                        (GAsyncReadyCallback) package_download_finished_cb,
                                        manager);
-#else
-        /* download them all */
-        pk_client_download_packages_async (PK_CLIENT(manager->priv->task),
-                                           package_ids,
-                                           NULL, /* this means system cache */
-                                           manager->priv->cancellable,
-                                           NULL, NULL,
-                                           (GAsyncReadyCallback) package_download_finished_cb,
-                                           manager);
-#endif
         g_strfreev (package_ids);
 }
 
@@ -1506,9 +1495,7 @@ gsd_updates_manager_start (GsdUpdatesManager *manager,
         g_object_set (manager->priv->task,
                       "background", TRUE,
                       "interactive", FALSE,
-#if PK_CHECK_VERSION(0,8,1)
                       "only-download", TRUE,
-#endif
                       NULL);
 
         /* watch UDev for missing firmware */
--
cgit v0.9.2
