--- proc_linux.cpp	2011-07-17 08:03:52.000000000 -0400
+++ proc_linux.cpp.new	2011-08-09 11:09:47.278589772 -0400
@@ -1962,7 +1962,10 @@
 	    int result;
 
         result = sscanf(p, "%d.%d.%d", &major, &minor, &patch);
-        if(result < 3) {
+        if(result == 2) {
+		patch = 0;
+	}
+        else if(result < 3) {
 			fprintf(stderr, "Qps: can't determine version, read %s \n",	p);
 			fprintf(stderr, "please report this bug.\n");
 			exit(1);
