From ea6fee0fbf8662a32a5cd2b5c96c1f75a9233813 Mon Sep 17 00:00:00 2001
From: Sebastien Bacher <seb128@ubuntu.com>
Date: Thu, 07 Jun 2012 16:26:26 +0000
Subject: grecord: Should call "gnome-control-center sound" not the old g-v-c

Signed-off-by: Oleksij Rempel <bug-track@fisher-privat.net>

https://bugzilla.gnome.org/show_bug.cgi?id=667402
---
diff --git a/grecord/src/gsr-window.c b/grecord/src/gsr-window.c
index bebb991..bd802d6 100644
--- a/grecord/src/gsr-window.c
+++ b/grecord/src/gsr-window.c
@@ -805,16 +805,16 @@ run_mixer_cb (GtkAction *action,
 	       GSRWindow *window)
 {
 	char *mixer_path;
-	char *argv[4] = {NULL, "--page", "recording",  NULL};
+	char *argv[4] = {NULL, "sound", "input",  NULL};
 	GError *error = NULL;
 	gboolean ret;
 
 	/* Open the mixer */
-	mixer_path = g_find_program_in_path ("gnome-volume-control");
+	mixer_path = g_find_program_in_path ("gnome-control-center");
 	if (mixer_path == NULL) {
 		show_error_dialog (GTK_WINDOW (window), NULL,
 		                   _("%s is not installed in the path."),
-		                   "gnome-volume-control");
+		                   "gnome-control-center");
 		return;
 	}
 
--
cgit v0.9.1
