diff -Naur gnome-do-plugins-0.8.4.orig/Flickr/src/AccountConfig.cs gnome-do-plugins-0.8.4/Flickr/src/AccountConfig.cs
--- gnome-do-plugins-0.8.4.orig/Flickr/src/AccountConfig.cs	2011-05-29 03:45:52.000000000 +0200
+++ gnome-do-plugins-0.8.4/Flickr/src/AccountConfig.cs	2011-11-30 12:14:02.239286506 +0100
@@ -99,7 +99,7 @@
 			try {
 				Auth auth = flickr.AuthGetToken(Frob);
 		  		AuthToken = auth.Token;
-		  		Username = auth.User.Username;
+		  		Username = auth.User.UserName;
 		  		flickr = new FlickrNet.Flickr (ApiKey, ApiSecret, AuthToken);
 		  		SetBtnStateComplete ();
 		  	} catch (FlickrNet.FlickrException ex) {
