diff -Naur gnome-do-plugins-0.8.4.orig/Banshee/Makefile.am gnome-do-plugins-0.8.4/Banshee/Makefile.am
--- gnome-do-plugins-0.8.4.orig/Banshee/Makefile.am	2011-06-12 03:07:01.000000000 +0200
+++ gnome-do-plugins-0.8.4/Banshee/Makefile.am	2012-02-27 23:09:54.051270529 +0100
@@ -31,14 +31,6 @@
 	System.Core \
 	$(BANSHEE_INDEXER_LIBS) \
 	$(DO_PLATFORM_LIBS) \
-	$(DO_UNIVERSE_LIBS)
-
-if BANSHEE_DBUS_SHARP
-REFERENCES += \
-	$(DBUS_SHARP_LIBS) \
-	$(DBUS_SHARP_GLIB_LIBS)
-else
-REFERENCES += \
-	$(NDESK_DBUS_10_LIBS) \
-	$(NDESK_DBUS_GLIB_10_LIBS)
-endif
+	$(DO_UNIVERSE_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
+	$(DBUS_SHARP_GLIB_10_LIBS)
diff -Naur gnome-do-plugins-0.8.4.orig/Banshee/src/BansheeDBus.cs gnome-do-plugins-0.8.4/Banshee/src/BansheeDBus.cs
--- gnome-do-plugins-0.8.4.orig/Banshee/src/BansheeDBus.cs	2011-06-12 13:34:28.000000000 +0200
+++ gnome-do-plugins-0.8.4/Banshee/src/BansheeDBus.cs	2012-02-27 23:08:20.973727937 +0100
@@ -23,11 +23,7 @@
 using System.Threading;
 using System.Collections.Generic;
 
-#if USE_DBUS_SHARP
 using DBus;
-#else
-using NDesk.DBus;
-#endif
 
 using org.freedesktop.DBus;
 
diff -Naur gnome-do-plugins-0.8.4.orig/BundledLibraries/Makefile.am gnome-do-plugins-0.8.4/BundledLibraries/Makefile.am
--- gnome-do-plugins-0.8.4.orig/BundledLibraries/Makefile.am	2011-02-06 08:50:04.000000000 +0100
+++ gnome-do-plugins-0.8.4/BundledLibraries/Makefile.am	2012-02-27 23:08:20.973727937 +0100
@@ -1,3 +1 @@
-if HAVE_EMPATHY
 SUBDIRS=telepathy-sharp
-endif
diff -Naur gnome-do-plugins-0.8.4.orig/BundledLibraries/telepathy-sharp/Makefile.am gnome-do-plugins-0.8.4/BundledLibraries/telepathy-sharp/Makefile.am
--- gnome-do-plugins-0.8.4.orig/BundledLibraries/telepathy-sharp/Makefile.am	2011-02-06 08:49:28.000000000 +0100
+++ gnome-do-plugins-0.8.4/BundledLibraries/telepathy-sharp/Makefile.am	2012-02-27 23:09:23.929811183 +0100
@@ -9,8 +9,8 @@
 	telepathy-sharp/Telepathy.cs
 
 REFERENCES = \
-	$(DBUS_SHARP_LIBS) \
-	$(DBUS_SHARP_GLIB_LIBS)
+	$(DBUS_SHARP_10_LIBS) \
+	$(DBUS_SHARP_GLIB_10_LIBS)
 
 # Include all of telepathy-sharp files, even though we don't
 # use them
@@ -28,4 +28,4 @@
 	tests/McTest.cs \
 	tests/MiscTest.cs \
 	tests.csproj \
-	)
\ Nincs újsor a fájl végén
+	)
diff -Naur gnome-do-plugins-0.8.4.orig/configure.ac gnome-do-plugins-0.8.4/configure.ac
--- gnome-do-plugins-0.8.4.orig/configure.ac	2011-06-12 11:34:02.000000000 +0200
+++ gnome-do-plugins-0.8.4/configure.ac	2012-02-27 23:08:20.983727333 +0100
@@ -38,10 +38,6 @@
 PKG_CHECK_EXISTS([banshee-collection-indexer >= 2.1], BANSHEE_USE_DBUS_SHARP=yes, BANSHEE_USE_DBUS_SHARP=no)
 AM_CONDITIONAL(BANSHEE_DBUS_SHARP, test "x$BANSHEE_USE_DBUS_SHARP" = "xyes")
 
-dnl -- DBus# required for Banshee and Telepathy support
-PKG_CHECK_MODULES([DBUS_SHARP], [dbus-sharp-1.0], HAVE_DBUS_SHARP=yes, HAVE_DBUS_SHARP=no)
-PKG_CHECK_MODULES([DBUS_SHARP_GLIB], [dbus-sharp-glib-1.0], HAVE_DBUS_SHARP_GLIB=yes, HAVE_DBUS_SNARP_GLIB=no)
-
 dnl {{{ check for Banshee
 AC_ARG_ENABLE(banshee, 
 	AC_HELP_STRING([--enable-banshee], [Build plugin to interact with Banshee (default: auto)]),
@@ -62,9 +58,6 @@
 	if test "x$enable_banshee" = "xyes" -a "x$HAVE_BANSHEE" = "xno" ; then
 		AC_MSG_ERROR([Banshee support explicitly requested but dependencies not found.])
 	fi
-	if test "x$enable_banshee" = "xyes" -a "x$BANSHEE_USE_DBUS_SHARP" = "xyes" -a "x$HAVE_DBUS_SHARP" = "xno" ; then
-	   	AC_MSG_ERROR([Banshee support requires DBus♯, not found])
-	fi
 fi
 
 AM_CONDITIONAL(HAVE_BANSHEE, test "x$HAVE_BANSHEE" = "xyes")
@@ -187,30 +180,6 @@
 dnl}}}
 
 dnl {{{
-# Temporarily check for dbus-sharp only in Empathy build
-# The DBus-Sharp fork is relatively new, and probably not ubiquitous yet.
-AC_ARG_ENABLE(empathy,
-	AC_HELP_STRING([--enable-empathy], [Build plugin to Telepathy IM framework (default: auto]),
-	enable_empathy=$enableval,
-	enable_empathy=auto)
-
-if test "x$enable_empathy" != "xno"; then
-   	if test "x$HAVE_DBUS_SHARP" = "xno" ; then
-	   	if test "x$enable_empathy" = "xyes" ; then
-		   	AC_MSG_ERROR([Empathy plugin explicitly requested but dbus-sharp dependencies not found])
-		fi
-		HAVE_EMPATHY=no
-	else
-		HAVE_EMPATHY=yes
-	fi
-else
-	HAVE_EMPATHY=no
-fi
-AC_SUBST([DBUS_SHARP_LIBS])
-AC_SUBST([DBUS_SHARP_GLIB_LIBS])
-
-AM_CONDITIONAL(HAVE_EMPATHY, test "x$HAVE_EMPATHY" = "xyes")
-
 # We need a bundled telepathy-sharp until there's an actual upstream for it
 TELEPATHY_SHARP_LIBS=-r:$(pwd)/build/telepathy-sharp.dll
 AC_SUBST([TELEPATHY_SHARP_LIBS])
@@ -221,14 +190,13 @@
 PKG_CHECK_MODULES([GLIB_SHARP_20], [glib-sharp-2.0])
 PKG_CHECK_MODULES([GNOME_DESKTOP_SHARP_20], [gnome-desktop-sharp-2.0])
 PKG_CHECK_MODULES([GNOME_KEYRING_SHARP], [gnome-keyring-sharp-1.0])
-PKG_CHECK_MODULES([GNOME_SHARP_20], [gnome-sharp-2.0])
 PKG_CHECK_MODULES([GNOME_VFS_SHARP_20], [gnome-vfs-sharp-2.0])
 PKG_CHECK_MODULES([GTK_SHARP_20], [gtk-sharp-2.0])
 PKG_CHECK_MODULES([MONO_ADDINS], [mono-addins])
 PKG_CHECK_MODULES([MONO_ADDINS_GUI], [mono-addins-gui])
 PKG_CHECK_MODULES([MONO_ADDINS_SETUP], [mono-addins-setup])
-PKG_CHECK_MODULES([NDESK_DBUS_10], [ndesk-dbus-1.0])
-PKG_CHECK_MODULES([NDESK_DBUS_GLIB_10], [ndesk-dbus-glib-1.0])
+PKG_CHECK_MODULES([DBUS_SHARP_10], [dbus-sharp-1.0])
+PKG_CHECK_MODULES([DBUS_SHARP_GLIB_10], [dbus-sharp-glib-1.0])
 PKG_CHECK_MODULES([NOTIFY_SHARP], [notify-sharp])
 
 
@@ -237,14 +205,13 @@
 AC_SUBST([GLIB_SHARP_20_LIBS])
 AC_SUBST([GNOME_DESKTOP_SHARP_20_LIBS])
 AC_SUBST([GNOME_KEYRING_SHARP_LIBS])
-AC_SUBST([GNOME_SHARP_20_LIBS])
 AC_SUBST([GNOME_VFS_SHARP_20_LIBS])
 AC_SUBST([GTK_SHARP_20_LIBS])
 AC_SUBST([MONO_ADDINS_LIBS])
 AC_SUBST([MONO_ADDINS_GUI_LIBS])
 AC_SUBST([MONO_ADDINS_SETUP_LIBS])
-AC_SUBST([NDESK_DBUS_10_LIBS])
-AC_SUBST([NDESK_DBUS_GLIB_10_LIBS])
+AC_SUBST([DBUS_SHARP_10_LIBS])
+AC_SUBST([DBUS_SHARP_GLIB_10_LIBS])
 AC_SUBST([NOTIFY_SHARP_LIBS])
 
 dnl Do core references
@@ -455,7 +422,6 @@
 
   Optional plugins:
     Banshee:           ${HAVE_BANSHEE}
-    Empathy:	       ${HAVE_EMPATHY}
     Flickr:            ${HAVE_FLICKR}
     Google Calendar:   ${HAVE_GCAL}
     Google Contacts:   ${HAVE_GMAIL}
diff -Naur gnome-do-plugins-0.8.4.orig/Emesene/Makefile.am gnome-do-plugins-0.8.4/Emesene/Makefile.am
--- gnome-do-plugins-0.8.4.orig/Emesene/Makefile.am	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Emesene/Makefile.am	2012-02-27 23:08:20.987060461 +0100
@@ -34,8 +34,8 @@
 	System \
 	System.Core \
 	System.Xml \
-	$(NDESK_DBUS_10_LIBS) \
-	$(NDESK_DBUS_GLIB_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
+	$(DBUS_SHARP_GLIB_LIBS) \
 	$(DO_UNIVERSE_LIBS) \
 	$(DO_PLATFORM_LIBS)
 
diff -Naur gnome-do-plugins-0.8.4.orig/Emesene/src/EmeseneChatAction.cs gnome-do-plugins-0.8.4/Emesene/src/EmeseneChatAction.cs
--- gnome-do-plugins-0.8.4.orig/Emesene/src/EmeseneChatAction.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Emesene/src/EmeseneChatAction.cs	2012-02-27 23:08:20.993726714 +0100
@@ -23,7 +23,7 @@
 using System.Collections.Generic;
 using System.Threading;
 using Do.Universe;
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 
 namespace Emesene
diff -Naur gnome-do-plugins-0.8.4.orig/Emesene/src/Emesene.cs gnome-do-plugins-0.8.4/Emesene/src/Emesene.cs
--- gnome-do-plugins-0.8.4.orig/Emesene/src/Emesene.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Emesene/src/Emesene.cs	2012-02-27 23:08:20.997059839 +0100
@@ -19,7 +19,7 @@
  */
 
 using System;
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 using System.Collections.Generic;
 using Do.Universe;
diff -Naur gnome-do-plugins-0.8.4.orig/Empathy/Makefile.am gnome-do-plugins-0.8.4/Empathy/Makefile.am
--- gnome-do-plugins-0.8.4.orig/Empathy/Makefile.am	2011-03-07 01:43:05.000000000 +0100
+++ gnome-do-plugins-0.8.4/Empathy/Makefile.am	2012-02-27 23:08:54.818289005 +0100
@@ -1,6 +1,4 @@
-if HAVE_EMPATHY
 include $(top_srcdir)/build.rules.mk
-endif
 
 ASSEMBLY=Empathy
 
@@ -27,8 +25,8 @@
 	Resources/Empathy.addin.xml
 
 REFERENCES = \
-	$(DBUS_SHARP_LIBS) \
-	$(DBUS_SHARP_GLIB_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
+	$(DBUS_SHARP_GLIB_10_LIBS) \
 	$(DO_PLATFORM_LIBS) \
 	$(DO_UNIVERSE_LIBS) \
 	$(MONO_ADDINS_LIBS) \
diff -Naur gnome-do-plugins-0.8.4.orig/GNOME-Session/Makefile.am gnome-do-plugins-0.8.4/GNOME-Session/Makefile.am
--- gnome-do-plugins-0.8.4.orig/GNOME-Session/Makefile.am	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/GNOME-Session/Makefile.am	2012-02-27 23:08:21.003726090 +0100
@@ -15,7 +15,7 @@
 REFERENCES = \
 	System \
 	System.Core \
-	$(NDESK_DBUS_10_LIBS) \
-	$(NDESK_DBUS_GLIB_10_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
+	$(DBUS_SHARP_GLIB_10_LIBS) \
 	$(DO_PLATFORM_LIBS) \
 	$(DO_UNIVERSE_LIBS)
diff -Naur gnome-do-plugins-0.8.4.orig/GNOME-Session/src/PowerManagement.cs gnome-do-plugins-0.8.4/GNOME-Session/src/PowerManagement.cs
--- gnome-do-plugins-0.8.4.orig/GNOME-Session/src/PowerManagement.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/GNOME-Session/src/PowerManagement.cs	2012-02-27 23:08:21.007059215 +0100
@@ -20,7 +20,7 @@
 using System;
 using System.Diagnostics;
 
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 
 using Do.Platform;
diff -Naur gnome-do-plugins-0.8.4.orig/GNOME-Session/src/ScreenSaver.cs gnome-do-plugins-0.8.4/GNOME-Session/src/ScreenSaver.cs
--- gnome-do-plugins-0.8.4.orig/GNOME-Session/src/ScreenSaver.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/GNOME-Session/src/ScreenSaver.cs	2012-02-27 23:08:21.013725468 +0100
@@ -21,7 +21,7 @@
 using System;
 using System.Diagnostics;
 
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 
 using Do.Platform;
diff -Naur gnome-do-plugins-0.8.4.orig/GNOME-Session/src/SystemManagement.cs gnome-do-plugins-0.8.4/GNOME-Session/src/SystemManagement.cs
--- gnome-do-plugins-0.8.4.orig/GNOME-Session/src/SystemManagement.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/GNOME-Session/src/SystemManagement.cs	2012-02-27 23:08:21.017058594 +0100
@@ -20,7 +20,7 @@
 using System;
 using System.Diagnostics;
 
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 
 using Do.Platform;
diff -Naur gnome-do-plugins-0.8.4.orig/Pidgin/Makefile.am gnome-do-plugins-0.8.4/Pidgin/Makefile.am
--- gnome-do-plugins-0.8.4.orig/Pidgin/Makefile.am	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Pidgin/Makefile.am	2012-02-27 23:08:21.020391720 +0100
@@ -26,7 +26,7 @@
 	System \
 	System.Xml \
 	System.Core \
-	$(NDESK_DBUS_10_LIBS) \
-	$(NDESK_DBUS_GLIB_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
+	$(DBUS_SHARP_GLIB_LIBS) \
 	$(DO_PLATFORM_LIBS) \
 	$(DO_UNIVERSE_LIBS) 
diff -Naur gnome-do-plugins-0.8.4.orig/Pidgin/src/Pidgin.cs gnome-do-plugins-0.8.4/Pidgin/src/Pidgin.cs
--- gnome-do-plugins-0.8.4.orig/Pidgin/src/Pidgin.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Pidgin/src/Pidgin.cs	2012-02-27 23:08:21.023724846 +0100
@@ -23,7 +23,7 @@
 using System.Diagnostics;
 using System.Collections.Generic;
 
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 
 using Do.Platform;
diff -Naur gnome-do-plugins-0.8.4.orig/Skype/Makefile.am gnome-do-plugins-0.8.4/Skype/Makefile.am
--- gnome-do-plugins-0.8.4.orig/Skype/Makefile.am	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Skype/Makefile.am	2012-02-27 23:08:21.027057972 +0100
@@ -31,8 +31,8 @@
 REFERENCES = \
 	System \
 	System.Core \
-	$(NDESK_DBUS_10_LIBS) \
-	$(NDESK_DBUS_GLIB_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
+	$(DBUS_SHARP_GLIB_LIBS) \
 	$(DO_PLATFORM_LIBS) \
 	$(DO_UNIVERSE_LIBS)
 
diff -Naur gnome-do-plugins-0.8.4.orig/Skype/src/Skype.cs gnome-do-plugins-0.8.4/Skype/src/Skype.cs
--- gnome-do-plugins-0.8.4.orig/Skype/src/Skype.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Skype/src/Skype.cs	2012-02-27 23:08:21.030391098 +0100
@@ -23,7 +23,7 @@
 using System.Collections.Generic;
 using System.Text.RegularExpressions;
 
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 
 using Mono.Addins;
@@ -310,4 +310,4 @@
 			return input;
 		}
 	}
-}
\ Nincs újsor a fájl végén
+}
diff -Naur gnome-do-plugins-0.8.4.orig/Tasque/Makefile.am gnome-do-plugins-0.8.4/Tasque/Makefile.am
--- gnome-do-plugins-0.8.4.orig/Tasque/Makefile.am	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Tasque/Makefile.am	2012-02-27 23:08:21.033724224 +0100
@@ -14,8 +14,8 @@
 REFERENCES = \
 	System \
 	System.Core \
-	$(NDESK_DBUS_10_LIBS) \
-	$(NDESK_DBUS_GLIB_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
+	$(DBUS_SHARP_GLIB_LIBS) \
 	$(GCONF_SHARP_20_LIBS) \
 	$(DO_PLATFORM_LIBS) \
 	$(DO_UNIVERSE_LIBS)
diff -Naur gnome-do-plugins-0.8.4.orig/Tasque/src/TasqueDBus.cs gnome-do-plugins-0.8.4/Tasque/src/TasqueDBus.cs
--- gnome-do-plugins-0.8.4.orig/Tasque/src/TasqueDBus.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Tasque/src/TasqueDBus.cs	2012-02-27 23:08:21.033724224 +0100
@@ -21,7 +21,7 @@
 using System.Linq;
 using System.Collections.Generic;
 
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 
 using Do.Platform;
diff -Naur gnome-do-plugins-0.8.4.orig/Tomboy/Makefile.am gnome-do-plugins-0.8.4/Tomboy/Makefile.am
--- gnome-do-plugins-0.8.4.orig/Tomboy/Makefile.am	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Tomboy/Makefile.am	2012-02-27 23:08:21.037057350 +0100
@@ -19,8 +19,8 @@
 REFERENCES = \
 	System \
 	System.Core \
-	$(NDESK_DBUS_10_LIBS) \
-	$(NDESK_DBUS_GLIB_10_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
+	$(DBUS_SHARP_GLIB_10_LIBS) \
 	$(GTK_SHARP_20_LIBS) \
 	$(DO_PLATFORM_LIBS) \
 	$(DO_PLATFORM_LINUX_LIBS) \
diff -Naur gnome-do-plugins-0.8.4.orig/Tomboy/src/TomboyDBus.cs gnome-do-plugins-0.8.4/Tomboy/src/TomboyDBus.cs
--- gnome-do-plugins-0.8.4.orig/Tomboy/src/TomboyDBus.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Tomboy/src/TomboyDBus.cs	2012-02-27 23:08:21.037057350 +0100
@@ -20,7 +20,7 @@
 using System;
 using System.Collections;
 
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 
 namespace Tomboy
diff -Naur gnome-do-plugins-0.8.4.orig/Tracker/Makefile.am gnome-do-plugins-0.8.4/Tracker/Makefile.am
--- gnome-do-plugins-0.8.4.orig/Tracker/Makefile.am	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Tracker/Makefile.am	2012-02-27 23:08:21.040390476 +0100
@@ -13,6 +13,6 @@
 REFERENCES = \
 	System \
 	System.Core \
-	$(NDESK_DBUS_10_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
 	$(DO_PLATFORM_LIBS) \
 	$(DO_UNIVERSE_LIBS)
diff -Naur gnome-do-plugins-0.8.4.orig/Tracker/src/ITrackerSearch.cs gnome-do-plugins-0.8.4/Tracker/src/ITrackerSearch.cs
--- gnome-do-plugins-0.8.4.orig/Tracker/src/ITrackerSearch.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Tracker/src/ITrackerSearch.cs	2012-02-27 23:08:21.040390476 +0100
@@ -1,6 +1,6 @@
 
 using System;
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 
 namespace Tracker.Dbus
diff -Naur gnome-do-plugins-0.8.4.orig/Tracker/src/Tracker.cs gnome-do-plugins-0.8.4/Tracker/src/Tracker.cs
--- gnome-do-plugins-0.8.4.orig/Tracker/src/Tracker.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Tracker/src/Tracker.cs	2012-02-27 23:08:21.043723602 +0100
@@ -1,7 +1,7 @@
 
 using System;
 
-using NDesk.DBus;
+using DBus;
 
 using Do.Platform;
 
diff -Naur gnome-do-plugins-0.8.4.orig/Woof/Makefile.am gnome-do-plugins-0.8.4/Woof/Makefile.am
--- gnome-do-plugins-0.8.4.orig/Woof/Makefile.am	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Woof/Makefile.am	2012-02-27 23:08:21.047056727 +0100
@@ -12,6 +12,6 @@
 REFERENCES = \
 	System \
 	System.Core \
-	$(NDESK_DBUS_10_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
 	$(DO_PLATFORM_LIBS) \
 	$(DO_UNIVERSE_LIBS)
diff -Naur gnome-do-plugins-0.8.4.orig/Woof/src/Woof.cs gnome-do-plugins-0.8.4/Woof/src/Woof.cs
--- gnome-do-plugins-0.8.4.orig/Woof/src/Woof.cs	2011-02-05 08:45:18.000000000 +0100
+++ gnome-do-plugins-0.8.4/Woof/src/Woof.cs	2012-02-27 23:08:21.047056727 +0100
@@ -22,7 +22,7 @@
 using System.Collections.Generic;
 using System.Reflection;
 using System.Threading;
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 
 using Do.Universe;
