diff -rupN wxWidgets-3.0.1.orig/Makefile.in wxWidgets-3.0.1/Makefile.in
--- wxWidgets-3.0.1.orig/Makefile.in	2014-07-30 00:40:09.604520200 +0200
+++ wxWidgets-3.0.1/Makefile.in	2014-07-30 00:43:08.131765800 +0200
@@ -16140,7 +16140,8 @@ install-wxconfig:
 	$(INSTALL_DIR) $(DESTDIR)$(bindir)
 	$(INSTALL_DIR) $(DESTDIR)$(libdir)/wx/config
 	$(INSTALL_PROGRAM) lib/wx/config/$(TOOLCHAIN_FULLNAME) $(DESTDIR)$(libdir)/wx/config
-	(cd $(DESTDIR)$(bindir) && rm -f wx-config && $(LN_S) $(libdir)/wx/config/$(TOOLCHAIN_FULLNAME) wx-config || cp -p $(DESTDIR)$(libdir)/wx/config/$(TOOLCHAIN_FULLNAME) wx-config)
+	(cd $(DESTDIR)$(bindir) && rm -f wx-config && $(LN_S) $(libdir)/wx/config/$(TOOLCHAIN_FULLNAME) wx-config32-$(WX_RELEASE) || cp -p $(DESTDIR)$(libdir)/wx/config/$(TOOLCHAIN_FULLNAME) wx-config32-$(WX_RELEASE))
+	(cd $(DESTDIR)$(bindir) && $(LN_S) wx-config32-$(WX_RELEASE) wx-config32$(EXEEXT))
 
 locale_install: 
 	$(INSTALL_DIR) $(DESTDIR)$(datadir)/locale
diff -rupN wxWidgets-3.0.1.orig/utils/wxrc/Makefile.in wxWidgets-3.0.1/utils/wxrc/Makefile.in
--- wxWidgets-3.0.1.orig/utils/wxrc/Makefile.in	2014-07-30 00:40:10.712676300 +0200
+++ wxWidgets-3.0.1/utils/wxrc/Makefile.in	2014-07-30 00:47:04.729183400 +0200
@@ -124,8 +124,8 @@ distclean: clean
 @COND_USE_XRC_1@	$(INSTALL_PROGRAM) wxrc$(EXEEXT) $(DESTDIR)$(bindir)
 @COND_USE_XRC_1@	rm -f $(DESTDIR)$(bindir)/wxrc$(EXEEXT) $(DESTDIR)$(bindir)/wxrc-$(WX_RELEASE)
 @COND_USE_XRC_1@	$(INSTALL_PROGRAM) wxrc$(EXEEXT) $(DESTDIR)$(bindir)
-@COND_USE_XRC_1@	mv -f $(DESTDIR)$(bindir)/wxrc$(EXEEXT) $(DESTDIR)$(bindir)/wxrc-$(WX_RELEASE)
-@COND_USE_XRC_1@	(cd $(DESTDIR)$(bindir) && $(LN_S) wxrc-$(WX_RELEASE) wxrc$(EXEEXT))
+@COND_USE_XRC_1@	mv -f $(DESTDIR)$(bindir)/wxrc$(EXEEXT) $(DESTDIR)$(bindir)/wxrc32-$(WX_RELEASE)
+@COND_USE_XRC_1@	(cd $(DESTDIR)$(bindir) && $(LN_S) wxrc32-$(WX_RELEASE) wxrc32$(EXEEXT))
 
 @COND_USE_XRC_1@uninstall_wxrc: 
 @COND_USE_XRC_1@	rm -f $(DESTDIR)$(bindir)/wxrc$(EXEEXT)
