--- a/test/meson.build	2020-10-21 17:30:15.000000000 -0700
+++ b/test/meson.build	2020-10-22 08:38:03.794099893 -0700
@@ -47,6 +47,3 @@
     include_directories: [gtk_layer_shell_inc],
     link_with: [gtk_layer_shell_lib])
 test('test-get-version', test_get_version, args: [meson.project_version()])
-
-check_licenses_script = files(meson.current_source_dir() + '/check-licenses.py')
-test('check-licenses', py, args: [check_licenses_script])
