diff -ruN a/lib/wptoml/array.cpp b/lib/wptoml/array.cpp
--- a/lib/wptoml/array.cpp	2020-06-29 18:36:49.000000000 +0200
+++ b/lib/wptoml/array.cpp	2020-07-19 09:50:14.930311451 +0200
@@ -10,7 +10,7 @@
 #include <functional>
 
 /* CPPTOML */
-#include <include/cpptoml.h>
+#include <cpptoml.h>
 
 /* TOML */
 #include "private.h"
diff -ruN a/lib/wptoml/file.cpp b/lib/wptoml/file.cpp
--- a/lib/wptoml/file.cpp	2020-06-29 18:36:49.000000000 +0200
+++ b/lib/wptoml/file.cpp	2020-07-19 09:50:28.923522152 +0200
@@ -7,7 +7,7 @@
  */
 
 /* CPPTOML */
-#include <include/cpptoml.h>
+#include <cpptoml.h>
 
 /* TOML */
 #include "private.h"
diff -ruN a/lib/wptoml/table.cpp b/lib/wptoml/table.cpp
--- a/lib/wptoml/table.cpp	2020-06-29 18:36:49.000000000 +0200
+++ b/lib/wptoml/table.cpp	2020-07-19 09:50:46.786698703 +0200
@@ -10,7 +10,7 @@
 #include <functional>
 
 /* CPPTOML */
-#include <include/cpptoml.h>
+#include <cpptoml.h>
 
 /* TOML */
 #include "private.h"
diff -ruN a/meson.build b/meson.build
--- a/meson.build	2020-06-29 18:36:49.000000000 +0200
+++ b/meson.build	2020-07-19 10:17:13.288978511 +0200
@@ -26,9 +26,13 @@
   wireplumber_config_dir = join_paths(get_option('prefix'), get_option('sysconfdir'), 'wireplumber')
 endif
 
-cmake = import('cmake')
-cpptoml = cmake.subproject('cpptoml')
-cpptoml_dep = cpptoml.dependency('cpptoml')
+if get_option('wrap_mode') == 'nodownload'
+  cpptoml_dep = dependency('cpptoml')
+else
+  cmake = import('cmake')
+  cpptoml = cmake.subproject('cpptoml')
+  cpptoml_dep = cpptoml.dependency('cpptoml')
+endif
 
 gobject_dep = dependency('gobject-2.0', version : '>= 2.58')
 gmodule_dep = dependency('gmodule-2.0', version : '== ' + gobject_dep.version())
diff -ruN a/subprojects/cpptoml.wrap b/subprojects/cpptoml.wrap
--- a/subprojects/cpptoml.wrap	2020-06-29 18:36:49.000000000 +0200
+++ b/subprojects/cpptoml.wrap	2020-07-19 10:16:27.232734437 +0200
@@ -1,4 +1,4 @@
 [wrap-git]
 directory=cpptoml
 url=https://github.com/skystrife/cpptoml.git
-revision=master
+revision=v0.1.1
