diff -ruN a/Makefile b/Makefile
--- a/Makefile	2020-08-22 16:54:34.000000000 +0200
+++ b/Makefile	2022-04-18 12:53:36.830425742 +0200
@@ -97,7 +97,7 @@
 	install -d $(DESTDIR)$(INCDIR)
 	install -m644 $(HFILES) $(DESTDIR)$(INCDIR)
 	install -d $(DESTDIR)$(LIBDIR)/pkgconfig
-	sed -e "s,@PREFIX@,$(PREFIX)," -e "s,@LIBDIR@,$(LIBDIR)," -e "s,@VERSION@,$(VERSION)," <vterm.pc.in >$(DESTDIR)$(LIBDIR)/pkgconfig/vterm.pc
+	sed -e "s,@PREFIX@,$(PREFIX)," -e "s,@INCDIR@,$(INCDIR)," -e "s,@LIBDIR@,$(LIBDIR)," -e "s,@VERSION@,$(VERSION)," <vterm.pc.in >$(DESTDIR)$(LIBDIR)/pkgconfig/vterm.pc
 
 install-lib: $(LIBRARY)
 	install -d $(DESTDIR)$(LIBDIR)
diff -ruN a/vterm.pc.in b/vterm.pc.in
--- a/vterm.pc.in	2020-08-22 16:54:34.000000000 +0200
+++ b/vterm.pc.in	2022-04-18 12:52:28.900128681 +0200
@@ -1,6 +1,6 @@
 prefix=@PREFIX@
 libdir=@LIBDIR@
-includedir=${prefix}/include
+includedir=@INCDIR@
 
 Name: vterm
 Description: Abstract VT220/Xterm/ECMA-48 emulation library
