--- setup.py.old	2012-11-08 20:33:26.224458028 +0100
+++ setup.py	2012-11-08 20:33:41.656633881 +0100
@@ -365,10 +365,7 @@ class Install(install):
             raise OSError("not found: %s" % ", ".join(mis_stys))
 
     def run(self):
-        if self.install_layout == "deb":
-            db = DebianInstaller(self)
-        else:
-            db = None
+        db = None
 
         if not(db) and not(self.nodeps):
             try:
