diff -aur newsbeuter-r2.9/src/pb_controller.cpp newsbeuter-r2.9.new/src/pb_controller.cpp
--- newsbeuter-r2.9/src/pb_controller.cpp	2015-02-19 11:56:59.000000000 +0100
+++ newsbeuter-r2.9.new/src/pb_controller.cpp	2017-09-16 20:43:29.180683163 +0200
@@ -306,9 +306,9 @@
 	if (player == "")
 		return;
 	cmdline.append(player);
-	cmdline.append(" \"");
-	cmdline.append(utils::replace_all(file,"\"", "\\\""));
-	cmdline.append("\"");
+	cmdline.append(" \'");
+	cmdline.append(utils::replace_all(file,"'", "%27"));
+	cmdline.append(" \'");
 	stfl::reset();
 	LOG(LOG_DEBUG, "pb_controller::play_file: running `%s'", cmdline.c_str());
 	::system(cmdline.c_str());
diff -aur newsbeuter-r2.9/src/queueloader.cpp newsbeuter-r2.9.new/src/queueloader.cpp
--- newsbeuter-r2.9/src/queueloader.cpp	2015-02-19 11:56:59.000000000 +0100
+++ newsbeuter-r2.9.new/src/queueloader.cpp	2017-09-16 20:47:01.884411680 +0200
@@ -130,7 +130,7 @@
 		strftime(lbuf, sizeof(lbuf), "%Y-%b-%d-%H%M%S.unknown", localtime(&t));
 		fn.append(lbuf);
 	} else {
-		fn.append(base);
+		fn.append(utils::replace_all(base, "'", "%27"));
 	}
 	return fn;
 }
