--- src/common/weapons/AccessoryStore.cpp.orig	2016-02-13 22:16:08.997822186 -0600
+++ src/common/weapons/AccessoryStore.cpp	2016-02-13 22:16:47.949819533 -0600
@@ -156,7 +156,7 @@
 	Accessory *parent, XMLNode *currentNode)
 {
 	XMLNode *typeNode = 0;
-	if (!currentNode->getNamedParameter("type", typeNode)) return false;
+	if (!currentNode->getNamedParameter("type", typeNode)) return 0;
 
 	AccessoryPart *accessoryPart = 
 		AccessoryMetaRegistration::getNewAccessory(typeNode->getContent(), this);
