diff --git a/meson.build b/meson.build
index b19e5e0..c72af31 100644
--- a/meson.build
+++ b/meson.build
@@ -43,6 +43,9 @@ subdir('meson/suppressions')
 suil_abs_module_dir = get_option('prefix') / suil_module_dir
 platform_defines = ['-DSUIL_MODULE_DIR="@0@"'.format(suil_abs_module_dir)]
 
+nodelete_c_link_args = cc.get_supported_link_arguments(['-Wl,-z,nodelete'])
+nodelete_cpp_link_args = cpp.get_supported_link_arguments(['-Wl,-z,nodelete'])
+
 ################
 # Dependencies #
 ################
@@ -220,6 +223,7 @@ if gtk2_dep.found() and qt5_dep.found()
     include_directories: include_dirs,
     install: true,
     install_dir: suil_module_dir,
+    link_args: nodelete_cpp_link_args,
   )
 endif
 
@@ -233,6 +237,7 @@ if gtk2_dep.found() and gtk2_x11_dep.found() and x11_dep.found()
     include_directories: include_dirs,
     install: true,
     install_dir: suil_module_dir,
+    link_args: nodelete_c_link_args,
   )
 endif
 
@@ -246,6 +251,7 @@ if gtk3_dep.found() and gtk3_x11_dep.found() and x11_dep.found()
     include_directories: include_dirs,
     install: true,
     install_dir: suil_module_dir,
+    link_args: nodelete_c_link_args,
   )
 endif
 
@@ -259,6 +265,7 @@ if gtk3_dep.found() and gtk3_x11_dep.found() and qt5_dep.found()
     include_directories: include_dirs,
     install: true,
     install_dir: suil_module_dir,
+    link_args: nodelete_cpp_link_args,
   )
 endif
 
@@ -285,6 +292,7 @@ if gtk2_dep.found() and host_machine.system() == 'windows'
     include_directories: include_dirs,
     install: true,
     install_dir: suil_module_dir,
+    link_args: nodelete_cpp_link_args,
   )
 endif
 
-- 
GitLab

