From f8f2ffca1928aeca2fa9771093436dba49baa538 Mon Sep 17 00:00:00 2001
From: Felix Yan <felixonmars@gmail.com>
Date: Fri, 12 Dec 2014 23:10:15 +0800
Subject: [PATCH] Don't depend on backports.ssl-match-hostname with python
 >=2.7.9, <3.0

---
 setup.py               | 2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/setup.py b/setup.py
index f09169f..f795807 100644
--- a/setup.py
+++ b/setup.py
@@ -121,7 +121,7 @@ def build_extension(self, ext):
     if sys.version_info < (2, 7):
         # Only needed indirectly, for singledispatch.
         install_requires.append('ordereddict')
-    if sys.version_info < (3, 2):
+    if sys.version_info < (2, 7, 9) or (3, 0) <= sys.version_info < (3, 2):
         install_requires.append('backports.ssl_match_hostname')
     if sys.version_info < (3, 4):
         install_requires.append('singledispatch')