diff --git a/ffmpegaudio.cc b/ffmpegaudio.cc
index 77585119..e3b90b5d 100644
--- a/ffmpegaudio.cc
+++ b/ffmpegaudio.cc
@@ -51,7 +51,6 @@ AudioPlayer & AudioPlayer::instance()
 
 AudioPlayer::AudioPlayer()
 {
-  av_register_all();
   ao_initialize();
 }
 
@@ -91,7 +90,7 @@ struct DecoderContext
   QByteArray audioData_;
   QDataStream audioDataStream_;
   AVFormatContext * formatContext_;
-  AVCodec * codec_;
+  const AVCodec * codec_;
   AVCodecContext * codecContext_;
   AVIOContext * avioContext_;
   AVStream * audioStream_;
