diff -Naur a/kwin/qtcurvehandler.cpp b/kwin/qtcurvehandler.cpp
--- a/kwin/qtcurvehandler.cpp	2012-08-02 13:36:58.811347708 +0000
+++ b/kwin/qtcurvehandler.cpp	2012-08-02 13:38:03.240996792 +0000
@@ -489,6 +489,11 @@
 
 extern "C"
 {
+    KWIN_EXPORT int decoration_version()
+    {
+        return KWIN_DECORATION_API_VERSION;
+    }
+
     KDE_EXPORT KDecorationFactory *create_factory()
     {
         KWinQtCurve::handler = new KWinQtCurve::QtCurveHandler();
