diff -ruN a/test/commit_test.rb b/test/commit_test.rb
--- a/test/commit_test.rb	2019-12-11 12:44:48.000000000 +0100
+++ b/test/commit_test.rb	2020-02-26 13:49:04.120935531 +0100
@@ -281,74 +281,6 @@
     assert_raises(Rugged::OdbError) { Rugged::Commit.extract_signature(@repo, "181037049a54a1eb5fab404658a3a250b44335d8") }
   end
 
-  def test_create_with_signature
-    signed_commit = <<-COMMIT
-tree 6b79e22d69bf46e289df0345a14ca059dfc9bdf6
-parent 34734e478d6cf50c27c9d69026d93974d052c454
-author Ben Burkert <ben@benburkert.com> 1358451456 -0800
-committer Ben Burkert <ben@benburkert.com> 1358451456 -0800
-gpgsig -----BEGIN PGP SIGNATURE-----
- Version: GnuPG v1.4.12 (Darwin)
- 
- iQIcBAABAgAGBQJQ+FMIAAoJEH+LfPdZDSs1e3EQAJMjhqjWF+WkGLHju7pTw2al
- o6IoMAhv0Z/LHlWhzBd9e7JeCnanRt12bAU7yvYp9+Z+z+dbwqLwDoFp8LVuigl8
- JGLcnwiUW3rSvhjdCp9irdb4+bhKUnKUzSdsR2CK4/hC0N2i/HOvMYX+BRsvqweq
- AsAkA6dAWh+gAfedrBUkCTGhlNYoetjdakWqlGL1TiKAefEZrtA1TpPkGn92vbLq
- SphFRUY9hVn1ZBWrT3hEpvAIcZag3rTOiRVT1X1flj8B2vGCEr3RrcwOIZikpdaW
- who/X3xh/DGbI2RbuxmmJpxxP/8dsVchRJJzBwG+yhwU/iN3MlV2c5D69tls/Dok
- 6VbyU4lm/ae0y3yR83D9dUlkycOnmmlBAHKIZ9qUts9X7mWJf0+yy2QxJVpjaTGG
- cmnQKKPeNIhGJk2ENnnnzjEve7L7YJQF6itbx5VCOcsGh3Ocb3YR7DMdWjt7f8pu
- c6j+q1rP7EpE2afUN/geSlp5i3x8aXZPDj67jImbVCE/Q1X9voCtyzGJH7MXR0N9
- ZpRF8yzveRfMH8bwAJjSOGAFF5XkcR/RNY95o+J+QcgBLdX48h+ZdNmUf6jqlu3J
- 7KmTXXQcOVpN6dD3CmRFsbjq+x6RHwa8u1iGn+oIkX908r97ckfB/kHKH7ZdXIJc
- cpxtDQQMGYFpXK/71stq
- =ozeK
- -----END PGP SIGNATURE-----
-
-a simple commit which works
-COMMIT
-
-    signature = <<-SIGNATURE.strip
------BEGIN PGP SIGNATURE-----
-Version: GnuPG v1.4.12 (Darwin)
-
-iQIcBAABAgAGBQJQ+FMIAAoJEH+LfPdZDSs1e3EQAJMjhqjWF+WkGLHju7pTw2al
-o6IoMAhv0Z/LHlWhzBd9e7JeCnanRt12bAU7yvYp9+Z+z+dbwqLwDoFp8LVuigl8
-JGLcnwiUW3rSvhjdCp9irdb4+bhKUnKUzSdsR2CK4/hC0N2i/HOvMYX+BRsvqweq
-AsAkA6dAWh+gAfedrBUkCTGhlNYoetjdakWqlGL1TiKAefEZrtA1TpPkGn92vbLq
-SphFRUY9hVn1ZBWrT3hEpvAIcZag3rTOiRVT1X1flj8B2vGCEr3RrcwOIZikpdaW
-who/X3xh/DGbI2RbuxmmJpxxP/8dsVchRJJzBwG+yhwU/iN3MlV2c5D69tls/Dok
-6VbyU4lm/ae0y3yR83D9dUlkycOnmmlBAHKIZ9qUts9X7mWJf0+yy2QxJVpjaTGG
-cmnQKKPeNIhGJk2ENnnnzjEve7L7YJQF6itbx5VCOcsGh3Ocb3YR7DMdWjt7f8pu
-c6j+q1rP7EpE2afUN/geSlp5i3x8aXZPDj67jImbVCE/Q1X9voCtyzGJH7MXR0N9
-ZpRF8yzveRfMH8bwAJjSOGAFF5XkcR/RNY95o+J+QcgBLdX48h+ZdNmUf6jqlu3J
-7KmTXXQcOVpN6dD3CmRFsbjq+x6RHwa8u1iGn+oIkX908r97ckfB/kHKH7ZdXIJc
-cpxtDQQMGYFpXK/71stq
-=ozeK
------END PGP SIGNATURE-----
-SIGNATURE
-
-    base_data = <<-SIGNEDDATA
-tree 6b79e22d69bf46e289df0345a14ca059dfc9bdf6
-parent 34734e478d6cf50c27c9d69026d93974d052c454
-author Ben Burkert <ben@benburkert.com> 1358451456 -0800
-committer Ben Burkert <ben@benburkert.com> 1358451456 -0800
-
-a simple commit which works
-SIGNEDDATA
-
-    id1 = Rugged::Commit::create_with_signature(@repo, base_data, signature, "gpgsig")
-    id2 = Rugged::Commit::create_with_signature(@repo, base_data, signature)
-    sig, data = Rugged::Commit::extract_signature(@repo, id1)
-
-    assert_equal id1, id2
-    assert_equal base_data, data
-    assert_equal signature, sig
-
-    raw_commit = Rugged::Commit::lookup(@repo, id1).read_raw.data
-    assert_equal signed_commit, raw_commit
-  end
-
   def test_commit_summary
     person = {:name => 'Scott', :email => 'schacon@gmail.com', :time => Time.now }
 
diff -ruN a/test/online/fetch_test.rb b/test/online/fetch_test.rb
--- a/test/online/fetch_test.rb	2019-12-11 12:44:48.000000000 +0100
+++ b/test/online/fetch_test.rb	2020-02-26 13:49:28.654033711 +0100
@@ -45,18 +45,6 @@
       assert_equal({ valid: true, host: "github.com" }, args)
     end
 
-    def test_fetch_over_https_with_certificate_callback_fail
-      @repo.remotes.create("origin", "https://github.com/libgit2/TestGitRepository.git")
-
-      exception = assert_raises Rugged::NetworkError do
-        @repo.fetch("origin", {
-          certificate_check: lambda { |valid, host| false }
-        })
-      end
-
-      assert_equal "user rejected certificate for github.com", exception.message
-    end
-
     def test_fetch_over_https_with_certificate_callback_exception
       @repo.remotes.create("origin", "https://github.com/libgit2/TestGitRepository.git")
 
