--- a/custom/conf/app.ini.sample
+++ b/custom/conf/app.ini.sample
@@ -628,13 +628,6 @@ sr-SP = sr
 sv-SE = sv
 ko-KR = ko
 
-[U2F]
-; Two Factor authentication with security keys
-; https://developers.yubico.com/U2F/App_ID.html
-APP_ID = %(PROTOCOL)s://%(DOMAIN)s:%(HTTP_PORT)s/
-; Comma seperated list of truisted facets
-TRUSTED_FACETS = %(PROTOCOL)s://%(DOMAIN)s:%(HTTP_PORT)s/
-
 ; Extension mapping to highlight class
 ; e.g. .toml=ini
 [highlight.mapping]
