diff -wbBur gtypist-2.9.1/configure gtypist-2.9.1.my/configure
--- gtypist-2.9.1/configure	2011-11-29 00:14:11.000000000 +0400
+++ gtypist-2.9.1.my/configure	2011-11-29 13:03:51.000000000 +0400
@@ -4011,13 +4011,13 @@
 ALL_LINGUAS="cs de es eu fi fr nl pl ru tr zh_CN zh_TW"
 
 # Checks for libraries.
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for add_wch in -lncursesw" >&5
-$as_echo_n "checking for add_wch in -lncursesw... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for add_wch in -lncurses" >&5
+$as_echo_n "checking for add_wch in -lncurses... " >&6; }
 if ${ac_cv_lib_ncursesw_add_wch+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lncursesw  $LIBS"
+LIBS="-lncurses  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -4464,7 +4464,7 @@
 done
 
 
-ac_fn_c_check_header_mongrel "$LINENO" "ncursesw/ncurses.h" "ac_cv_header_ncursesw_ncurses_h" "$ac_includes_default"
+ac_fn_c_check_header_mongrel "$LINENO" "ncurses.h" "ac_cv_header_ncursesw_ncurses_h" "$ac_includes_default"
 if test "x$ac_cv_header_ncursesw_ncurses_h" = xyes; then :
   HAVE_NCURSESW_H=1
 fi
@@ -4473,7 +4473,7 @@
 
 # sanity check for libncursesw:
 if test -n "$HAVE_NCURSESW_H" -a -n "$HAVE_LIBNCURSESW";  then
-   LIBS="-lncursesw $LIBS"
+   LIBS="-lncurses $LIBS"
 else
    echo -e "Error:  both library and header files for the ncursesw library\n"\
    	"are required to build this package.  See INSTALL file for"\
diff -wbBur gtypist-2.9.1/configure.ac gtypist-2.9.1.my/configure.ac
--- gtypist-2.9.1/configure.ac	2011-11-21 03:27:02.000000000 +0400
+++ gtypist-2.9.1.my/configure.ac	2011-11-29 13:02:53.000000000 +0400
@@ -21,13 +21,13 @@
 ALL_LINGUAS="cs de es eu fi fr nl pl ru tr zh_CN zh_TW"
 
 # Checks for libraries.
-AC_CHECK_LIB(ncursesw, add_wch, HAVE_LIBNCURSESW=1)
+AC_CHECK_LIB(ncurses, add_wch, HAVE_LIBNCURSESW=1)
 
 # Checks for header files.
 AC_HEADER_STDC
 AC_CHECK_HEADERS([unistd.h alloca.h argz.h errno.h fcntl.h langinfo.h libintl.h limits.h locale.h malloc.h stddef.h stdio_ext.h stdlib.h string.h strings.h sys/param.h unistd.h])
 
-AC_CHECK_HEADER(ncursesw/ncurses.h, HAVE_NCURSESW_H=1)
+AC_CHECK_HEADER(ncurses.h, HAVE_NCURSESW_H=1)
 
 # sanity check for libncursesw: 
 if test -n "$HAVE_NCURSESW_H" -a -n "$HAVE_LIBNCURSESW";  then
diff -wbBur gtypist-2.9.1/src/cursmenu.c gtypist-2.9.1.my/src/cursmenu.c
--- gtypist-2.9.1/src/cursmenu.c	2011-11-01 04:13:02.000000000 +0400
+++ gtypist-2.9.1.my/src/cursmenu.c	2011-11-29 13:04:15.000000000 +0400
@@ -24,7 +24,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif
 
 #include "error.h"
diff -wbBur gtypist-2.9.1/src/error.c gtypist-2.9.1.my/src/error.c
--- gtypist-2.9.1/src/error.c	2011-11-01 04:13:02.000000000 +0400
+++ gtypist-2.9.1.my/src/error.c	2011-11-29 13:04:15.000000000 +0400
@@ -25,7 +25,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif
 
 #include <stdlib.h>
diff -wbBur gtypist-2.9.1/src/gtypist.c gtypist-2.9.1.my/src/gtypist.c
--- gtypist-2.9.1/src/gtypist.c	2011-11-22 17:31:25.000000000 +0400
+++ gtypist-2.9.1.my/src/gtypist.c	2011-11-29 13:04:15.000000000 +0400
@@ -31,7 +31,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif
 
 #include <time.h>
diff -wbBur gtypist-2.9.1/src/script.c gtypist-2.9.1.my/src/script.c
--- gtypist-2.9.1/src/script.c	2011-11-28 19:39:48.000000000 +0400
+++ gtypist-2.9.1.my/src/script.c	2011-11-29 13:04:15.000000000 +0400
@@ -24,7 +24,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif
 
 #include "error.h"
diff -wbBur gtypist-2.9.1/src/utf8.c gtypist-2.9.1.my/src/utf8.c
--- gtypist-2.9.1/src/utf8.c	2011-11-28 19:39:19.000000000 +0400
+++ gtypist-2.9.1.my/src/utf8.c	2011-11-29 13:04:15.000000000 +0400
@@ -23,7 +23,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif
 
 #include <stdlib.h>
