--- a/include/afpfs-ng/dsi.h	2012-02-23 10:02:07.062734160 +0100
+++ b/include/afpfs-ng/dsi.h	2012-02-23 10:09:46.297111254 +0100
@@ -2,7 +2,7 @@
 #ifndef __DSI_H_
 #define __DSI_H_
 
-#include "afpfs-ng/afp.h"
+#include "afp.h"
 
 struct dsi_request
 {
--- a/include/afpfs-ng/map_def.h	2012-02-23 10:02:07.062734160 +0100
+++ b/include/afpfs-ng/map_def.h	2012-02-23 10:11:32.769953053 +0100
@@ -1,7 +1,7 @@
 #ifndef __MAP_H_
 #define __MAP_H_
 
-#include "afpfs-ng/afp.h"
+#include "afp.h"
 
 #define AFP_MAPPING_UNKNOWN 0
 #define AFP_MAPPING_COMMON 1
--- a/include/afpfs-ng/midlevel.h	2012-02-23 10:02:07.062734160 +0100
+++ b/include/afpfs-ng/midlevel.h	2012-02-23 10:12:40.519943675 +0100
@@ -2,7 +2,7 @@
 #define __MIDLEVEL_H_ 
 
 #include <utime.h>
-#include "afpfs-ng/afp.h"
+#include "afp.h"
 
 int ml_open(struct afp_volume * volume, const char *path, int flags, 
         struct afp_file_info **newfp);
--- a/include/afpfs-ng/utils.h	2012-02-23 10:14:42.212332892 +0100
+++ b/include/afpfs-ng/utils.h	2012-02-23 10:14:51.116069525 +0100
@@ -2,7 +2,7 @@
 #define __UTILS_H_
 #include <stdio.h>
 
-#include "afpfs-ng/afp.h"
+#include "afp.h"
 
 #if BYTE_ORDER == BIG_ENDIAN
 #define hton64(x)       (x)
