diff --git a/aiobotocore/client.py b/aiobotocore/client.py
index 5dee865..13cf364 100644
--- a/aiobotocore/client.py
+++ b/aiobotocore/client.py
@@ -28,6 +28,8 @@ class AioClientCreator(ClientCreator):
         service_name = first_non_none_response(responses, default=service_name)
         service_model = self._load_service_model(service_name, api_version)
         cls = await self._create_client_class(service_name, service_model)
+        region_name, client_config = self._normalize_fips_region(
+            region_name, client_config)
         endpoint_bridge = ClientEndpointBridge(
             self._endpoint_resolver, scoped_config, client_config,
             service_signing_name=service_model.metadata.get('signingName'))
@@ -42,7 +44,6 @@ class AioClientCreator(ClientCreator):
         self._register_endpoint_discovery(
             service_client, endpoint_url, client_config
         )
-        self._register_lazy_block_unknown_fips_pseudo_regions(service_client)
         return service_client
 
     async def _create_client_class(self, service_name, service_model):
