diff --git a/Makefile b/Makefile
index 38104ee..afb4f9b 100755
--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,6 @@ DOC2MAN ?= docbook2x-man
 
 PREFIX ?= $(DESTDIR)/usr
 BINDIR = $(PREFIX)/bin
-ETCDIR = $(DESTDIR)/etc
 DATADIR = $(PREFIX)/share/griffith
 LIBDIR = $(DATADIR)/lib
 IMAGESDIR = $(DATADIR)/images
@@ -34,9 +33,9 @@ IMPORTPLUGINSDIR = $(PLUGINSDIR)/imp
 EXTENSIONSPLUGINSDIR = $(PLUGINSDIR)/extensions
 TPLDIR = $(DATADIR)/export_templates
 APPLICATIONSDIR = $(PREFIX)/share/applications
-ICONDIR = $(PREFIX)/share/pixmaps
+ICONDIR = $(PREFIX)/share/icons/hicolor/scalable/apps
 LOCALEDIR = $(PREFIX)/share/locale
-BASHCOMPDIR = $(ETCDIR)/bash_completion.d
+BASHCOMPDIR = $(PREFIX)/share/bash-completion/completions
 
 help:
 	@echo Usage:
@@ -56,7 +55,7 @@ install:
 	$(INSTALL) -m 755 -d $(BINDIR) $(DATADIR) $(LIBDIR) $(LIBDIR)/db \
 		$(PLUGINSDIR) $(MOVIEPLUGINSDIR) $(EXPORTPLUGINSDIR) $(IMPORTPLUGINSDIR) $(EXTENSIONSPLUGINSDIR) \
 		$(FONTSDIR) $(APPLICATIONSDIR) $(ICONDIR) $(TPLDIR) \
-		$(IMAGESDIR) $(GLADEDIR) $(ETCDIR) $(BASHCOMPDIR)
+		$(IMAGESDIR) $(GLADEDIR) $(BASHCOMPDIR)
 	$(INSTALL) -m 755 griffith $(LIBDIR)
 	$(INSTALL) -m 644 lib/*.py $(LIBDIR)
 	$(INSTALL) -m 644 lib/db/*.py $(LIBDIR)/db/
@@ -68,8 +67,7 @@ install:
 	$(INSTALL) -m 644 glade/*.ui $(GLADEDIR)
 	$(INSTALL) -m 644 glade/*.png $(GLADEDIR)
 	$(INSTALL) -m 644 images/*.png $(IMAGESDIR)
-	$(INSTALL) -m 644 images/griffith.png $(ICONDIR)
-	$(INSTALL) -m 644 images/griffith.xpm $(ICONDIR)
+	$(INSTALL) -m 644 images/griffith.svg $(ICONDIR)
 	$(INSTALL) -m 644 data/griffith.desktop $(APPLICATIONSDIR)
 	$(INSTALL) -m 644 data/bash_completion $(BASHCOMPDIR)/griffith
 	
@@ -88,7 +86,7 @@ install:
 	
 	if test -f $(PREFIX)/bin/griffith; then ${RM} $(PREFIX)/bin/griffith; fi	
 	
-	ln -s $(LIBDIR)/griffith $(BINDIR)/griffith
+	ln -s ../share/griffith/lib/griffith $(BINDIR)/griffith
 	chmod +x $(BINDIR)/griffith
 	$(MAKE) -C docs install
 
@@ -106,8 +104,7 @@ uninstall:
 	${RM} -r $(IMAGESDIR)
 	${RM} -r $(GLADEDIR)
 	${RM} -r $(DATADIR)
-	${RM} -r $(ICONDIR)/griffith.png
-	${RM} -r $(ICONDIR)/griffith.xpm
+	${RM} -r $(ICONDIR)/griffith.svg
 	${RM} -r $(APPLICATIONSDIR)/griffith.desktop
 	for lang in $(LANGUAGES); do \
 		${RM} -r $(LOCALEDIR)/$$lang/LC_MESSAGES/griffith.mo; \
