--- a/src/gldit/CMakeLists.txt	2013-03-23 16:55:12.000000000 +0100
+++ b/src/gldit/CMakeLists.txt	2013-03-26 11:57:14.159200918 +0100
@@ -82,9 +82,7 @@ link_directories(
 	${GTK_LIBRARY_DIRS}
 	${XEXTEND_LIBRARY_DIRS}
 	${XINERAMA_LIBRARY_DIRS}
-	${CMAKE_SOURCE_DIR}/src/gldit
-	${CMAKE_SOURCE_DIR}/src/icon-factory
-	${CMAKE_SOURCE_DIR}/src/implementations)
+	)
 
 # Define the library
 add_library ("gldi" SHARED ${core_lib_SRCS})
