diff -upr ceph-12.2.5.orig/src/rbd_replay/Replayer.cc ceph-12.2.5/src/rbd_replay/Replayer.cc
--- ceph-12.2.5.orig/src/rbd_replay/Replayer.cc	2018-04-23 19:18:33.000000000 +0300
+++ ceph-12.2.5/src/rbd_replay/Replayer.cc	2018-06-03 01:55:46.476396645 +0300
@@ -350,7 +350,7 @@ void Replayer::wait_for_actions(const ac
     dout(DEPGRAPH_LEVEL) << "Finished waiting for " << dep.id << " after " << micros << " microseconds" << dendl;
     // Apparently the nanoseconds constructor is optional:
     // http://www.boost.org/doc/libs/1_46_0/doc/html/date_time/details.html#compile_options
-    boost::system_time sub_release_time(action_completed_time + boost::posix_time::microseconds(dep.time_delta * m_latency_multiplier / 1000));
+    boost::system_time sub_release_time(action_completed_time + boost::posix_time::microseconds((static_cast<long long>(dep.time_delta * m_latency_multiplier / 1000)));
     if (sub_release_time > release_time) {
       release_time = sub_release_time;
     }
diff -upr ceph-12.2.5.orig/src/test/librbd/test_mock_Journal.cc ceph-12.2.5/src/test/librbd/test_mock_Journal.cc
--- ceph-12.2.5.orig/src/test/librbd/test_mock_Journal.cc	2018-06-03 01:56:35.407273176 +0300
+++ ceph-12.2.5/src/test/librbd/test_mock_Journal.cc	2018-06-03 01:56:29.800506081 +0300
@@ -1,6 +1,8 @@
 // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab
 
+#define BOOST_BIND_NO_PLACEHOLDERS
+
 #include "test/librbd/test_mock_fixture.h"
 #include "test/journal/mock/MockJournaler.h"
 #include "test/librbd/test_support.h"
