diff -u -r ceph-13.2.1/src/CMakeLists.txt ceph-13.2.1-w/src/CMakeLists.txt
--- ceph-13.2.1/src/CMakeLists.txt	2018-07-26 17:39:56.000000000 +0000
+++ ceph-13.2.1-w/src/CMakeLists.txt	2019-09-13 14:12:40.391981716 +0000
@@ -34,6 +34,7 @@
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ftemplate-depth-1024")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-pragmas")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-address-of-packed-member")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unknown-pragmas")
 if(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -rdynamic")
@@ -47,7 +48,6 @@
   set(CMAKE_LINKER_FLAGS "${CMAKE_LINKER_FLAGS} -rdynamic -export-dynamic ${CMAKE_EXE_EXPORTS_C_FLAG}")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-inconsistent-missing-override -Wno-mismatched-tags")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-private-field")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-address-of-packed-member")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-function")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-local-typedef")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-varargs")
diff -u -r ceph-13.2.1/src/spdk/dpdk/mk/toolchain/gcc/rte.vars.mk ceph-13.2.1-w/src/spdk/dpdk/mk/toolchain/gcc/rte.vars.mk
--- ceph-13.2.1/src/spdk/dpdk/mk/toolchain/gcc/rte.vars.mk	2017-11-17 20:30:40.000000000 +0000
+++ ceph-13.2.1-w/src/spdk/dpdk/mk/toolchain/gcc/rte.vars.mk	2019-09-13 14:26:09.267136251 +0000
@@ -106,5 +106,10 @@
 WERROR_FLAGS += -Wno-format-truncation
 endif
 
+# disable packed member unalign warnings
+ifeq ($(shell test $(GCC_VERSION) -gt 90 && echo 1), 1)
+WERROR_FLAGS += -Wno-address-of-packed-member
+endif
+
 export CC AS AR LD OBJCOPY OBJDUMP STRIP READELF
 export TOOLCHAIN_CFLAGS TOOLCHAIN_LDFLAGS TOOLCHAIN_ASFLAGS
