diff --git a/dcraw.cc b/dcraw.cc
index 88c8068..d78c66b 100644
--- a/dcraw.cc
+++ b/dcraw.cc
@@ -8766,8 +8766,8 @@ void CLASS identify()
       "Nikon", "Nokia", "Olympus", "Pentax", "Phase One", "Ricoh",
       "Samsung", "Sigma", "Sinar", "Sony" };
   char head[32], *cp;
-  unsigned fsize, i, c;
-  int hlen, flen, zero_fsize=1;
+  unsigned fsize;
+  int hlen, flen, zero_fsize=1, i, c;
   struct jhead jh;
 
   tiff_flip = flip = filters = UINT_MAX;	/* unknown */
