From 136add59744446a13a276f50cb0285e116d19fa8 Mon Sep 17 00:00:00 2001
From: Takeshi KOMIYA <i.tkomiya@gmail.com>
Date: Tue, 6 Mar 2018 22:10:14 +0900
Subject: [PATCH] Skip checking cleanup() is called

---
 tests/test_util.py | 32 +++++++++++++-------------------
 1 file changed, 13 insertions(+), 19 deletions(-)

diff --git a/tests/test_util.py b/tests/test_util.py
index f432325..1a5ce86 100644
--- a/tests/test_util.py
+++ b/tests/test_util.py
@@ -96,30 +96,24 @@ class TestSphinxTesting(unittest.TestCase):
         app = TestApp(create_new_srcdir=True)
         self.assertTrue(app.builddir.exists())
 
-        with patch("sphinx.theming.Theme") as Theme:
-            with patch("sphinx.ext.autodoc.AutoDirective") as AutoDirective:
-                app.cleanup()
-                self.assertEqual(1, Theme.themes.clear.call_count)
-                self.assertEqual(1, AutoDirective._registry.clear.call_count)
-                self.assertFalse(app.builddir.exists())
+        with patch("sphinx.ext.autodoc.AutoDirective") as AutoDirective:
+            app.cleanup()
+            self.assertEqual(1, AutoDirective._registry.clear.call_count)
+            self.assertFalse(app.builddir.exists())
 
     def test_TestApp_cleanup_when_cleanup_on_errors(self):
         app = TestApp(create_new_srcdir=True, cleanup_on_errors=False)
         self.assertTrue(app.builddir.exists())
 
-        with patch("sphinx.theming.Theme") as Theme:
-            with patch("sphinx.ext.autodoc.AutoDirective") as AutoDirective:
-                app.cleanup(error=True)
-                self.assertEqual(0, Theme.themes.clear.call_count)
-                self.assertEqual(0, AutoDirective._registry.clear.call_count)
-                self.assertTrue(app.builddir.exists())
-
-        with patch("sphinx.theming.Theme") as Theme:
-            with patch("sphinx.ext.autodoc.AutoDirective") as AutoDirective:
-                app.cleanup(error=None)
-                self.assertEqual(1, Theme.themes.clear.call_count)
-                self.assertEqual(1, AutoDirective._registry.clear.call_count)
-                self.assertFalse(app.builddir.exists())
+        with patch("sphinx.ext.autodoc.AutoDirective") as AutoDirective:
+            app.cleanup(error=True)
+            self.assertEqual(0, AutoDirective._registry.clear.call_count)
+            self.assertTrue(app.builddir.exists())
+
+        with patch("sphinx.ext.autodoc.AutoDirective") as AutoDirective:
+            app.cleanup(error=None)
+            self.assertEqual(1, AutoDirective._registry.clear.call_count)
+            self.assertFalse(app.builddir.exists())
 
     def test_with_app(self):
         srcdir = path(__file__).dirname() / 'examples'
-- 
2.17.0

