--- Makefile	2009-09-24 22:42:25.837690792 +0200
+++ Makefile	2009-09-24 22:49:20.071815818 +0200
@@ -25,8 +25,8 @@ LD     := g++
 AR     := ar rc
 RANLIB := ranlib
 
-DEBUG_CFLAGS     := -Wall -Wno-format -g -DDEBUG
-RELEASE_CFLAGS   := -Wall -Wno-unknown-pragmas -Wno-format -O3
+DEBUG_CFLAGS     := -Wall -fPIC -Wno-format -g -DDEBUG
+RELEASE_CFLAGS   := -Wall -fPIC -Wno-unknown-pragmas -Wno-format -O3
 
 LIBS		 :=
 
@@ -102,7 +102,7 @@ OBJS := $(addsuffix .o,$(basename ${SRCS
 #****************************************************************************
 
 ${OUTPUT}: ${OBJS}
-	${LD} -o $@ ${LDFLAGS} ${OBJS} ${LIBS} ${EXTRA_LIBS}
+	${LD} -o $@ ${LDFLAGS} -fPIC ${OBJS} ${LIBS} ${EXTRA_LIBS}
 
 #****************************************************************************
 # common rules
