diff -rupN cairo-dock-plug-ins-3.4.1.orig/CMakeLists.txt cairo-dock-plug-ins-3.4.1/CMakeLists.txt
--- cairo-dock-plug-ins-3.4.1.orig/CMakeLists.txt	2015-03-12 09:43:57.171067365 +0100
+++ cairo-dock-plug-ins-3.4.1/CMakeLists.txt	2015-03-12 09:46:20.717120675 +0100
@@ -573,11 +573,11 @@ if (enable-mono-interface)
 	if (NOT GMCS_EXECUTABLE OR NOT EXISTS ${GMCS_EXECUTABLE})
 		message (STATUS "Could not find Mono compiler gmcs, won't build Mono interface.")
 	else()
-		pkg_check_modules (MONO_PACKAGE glib-sharp-2.0 ndesk-dbus-1.0 ndesk-dbus-glib-1.0)
+		pkg_check_modules (MONO_PACKAGE dbus-sharp-2.0 dbus-sharp-glib-2.0 glib-sharp-3.0)
 		if (NOT MONO_PACKAGE_FOUND)
-			message (STATUS "Could not find glib-sharp-2.0, ndesk-dbus-1.0 or ndesk-dbus-glib-1.0; won't be built Mono interface.")
-			message (WARNING "These modules are required to compile DBus applet with Mono interface: glib-sharp-2.0, ndesk-dbus-1.0 and ndesk-dbus-glib-1.0")
-			set (MODULES_MISSING "${MODULES_MISSING} glib-sharp-2.0 ndesk-dbus-1.0 ndesk-dbus-glib-1.0")
+			message (STATUS "Could not find dbus-sharp-2.0, dbus-sharp-glib-2.0 or glib-sharp-3.0; won't be built Mono interface.")
+			message (WARNING "These modules are required to compile DBus applet with Mono interface: dbus-sharp-2.0, dbus-sharp-glib-2.0 and glib-sharp-3.0")
+			set (MODULES_MISSING "${MODULES_MISSING} dbus-sharp-2.0 dbus-sharp-glib-2.0 glib-sharp-3.0")
 		else()
 			set (MONO_FOUND TRUE)
 			set (with_mono yes)
diff -rupN cairo-dock-plug-ins-3.4.1.orig/Dbus/interfaces/mono/CDApplet.cs cairo-dock-plug-ins-3.4.1/Dbus/interfaces/mono/CDApplet.cs
--- cairo-dock-plug-ins-3.4.1.orig/Dbus/interfaces/mono/CDApplet.cs	2015-03-12 09:52:56.491934965 +0100
+++ cairo-dock-plug-ins-3.4.1/Dbus/interfaces/mono/CDApplet.cs	2015-03-12 09:56:50.191652099 +0100
@@ -21,7 +21,7 @@
 // Make your own class derive from a CDApplet, and override the functions you need (the ones which don't start with an underscore).
 
 // Compile it with the following command, then rename 'demo_mono.exe' to 'demo_mono'.
-// gmcs /target:library CDApplet.cs ISubApplet.cs IApplet.cs -pkg:glib-sharp-2.0 -pkg:ndesk-dbus-1.0 -pkg:ndesk-dbus-glib-1.0
+// gmcs /target:library CDApplet.cs ISubApplet.cs IApplet.cs -pkg:dbus-sharp-2.0 -pkg:dbus-sharp-glib-2.0 -pkg:glib-sharp-3.0
   //////////////////////////
  ////// dependancies //////
 //////////////////////////
@@ -29,7 +29,7 @@ using System;  // Environment
 using System.IO;  // Path, Directory
 using System.Reflection;
 using GLib;
-using NDesk.DBus;
+using DBus;
 using CairoDock.Applet;
 
 //namespace CairoDock.Applet
@@ -258,8 +258,8 @@ public class CDApplet
 	
 	private void _connect_to_dock ()
 	{
-		NDesk.DBus.BusG.Init();
-		NDesk.DBus.Bus bus = NDesk.DBus.Bus.Session;
+		DBus.BusG.Init();
+		DBus.Bus bus = DBus.Bus.Session;
 		this.icon = bus.GetObject<IApplet> ("org.cairodock.CairoDock", new ObjectPath (this.cBusPath));
 		this.icon.on_click 			+= new OnClickEvent (on_click);
 		this.icon.on_middle_click 	+= new OnMiddleClickEvent (on_middle_click);
diff -rupN cairo-dock-plug-ins-3.4.1.orig/Dbus/interfaces/mono/CMakeLists.txt cairo-dock-plug-ins-3.4.1/Dbus/interfaces/mono/CMakeLists.txt
--- cairo-dock-plug-ins-3.4.1.orig/Dbus/interfaces/mono/CMakeLists.txt	2015-03-12 09:52:56.491934965 +0100
+++ cairo-dock-plug-ins-3.4.1/Dbus/interfaces/mono/CMakeLists.txt	2015-03-12 09:54:07.418777441 +0100
@@ -3,7 +3,7 @@
 
 execute_process(COMMAND ${GMCS_EXECUTABLE}
 	-target:library
-	-pkg:glib-sharp-2.0 -pkg:ndesk-dbus-1.0 -pkg:ndesk-dbus-glib-1.0
+	-pkg:dbus-sharp-2.0 -pkg:dbus-sharp-glib-2.0 -pkg:glib-sharp-3.0
 	-out:${CMAKE_CURRENT_BINARY_DIR}/CDApplet.dll
 	${CMAKE_CURRENT_SOURCE_DIR}/CDApplet.cs ${CMAKE_CURRENT_SOURCE_DIR}/ISubApplet.cs ${CMAKE_CURRENT_SOURCE_DIR}/IApplet.cs)
 ### find how to register to GAC ...
diff -rupN cairo-dock-plug-ins-3.4.1.orig/Dbus/demos/demo_mono/demo_mono.cs cairo-dock-plug-ins-3.4.1/Dbus/demos/demo_mono/demo_mono.cs
--- cairo-dock-plug-ins-3.4.1.orig/Dbus/demos/demo_mono/demo_mono.cs	2015-03-12 09:59:41.064312097 +0100
+++ cairo-dock-plug-ins-3.4.1/Dbus/demos/demo_mono/demo_mono.cs	2015-03-12 10:02:39.473424651 +0100
@@ -138,7 +138,7 @@ public class Applet : CDApplet
 	public override void on_build_menu ()
 	{
 		Console.WriteLine("*** build menu");
-		/// Warning : teh AddMenuItems fails with NDesk-DBus 0.6.0; until this is fixed, use the PopulateMenu method.
+		/// Warning : the AddMenuItems fails with DBus Sharp 0.6.0; until this is fixed, use the PopulateMenu method.
 		/**Dictionary<string, object>[] pItems = new Dictionary<string, object>[] {
 			new Dictionary<string, object>()
 			{
diff -rupN cairo-dock-plug-ins-3.4.1.orig/Dbus/interfaces/mono/IApplet.cs cairo-dock-plug-ins-3.4.1/Dbus/interfaces/mono/IApplet.cs
--- cairo-dock-plug-ins-3.4.1.orig/Dbus/interfaces/mono/IApplet.cs	2015-03-12 09:59:41.067645289 +0100
+++ cairo-dock-plug-ins-3.4.1/Dbus/interfaces/mono/IApplet.cs	2015-03-12 10:00:35.395334804 +0100
@@ -1,6 +1,6 @@
 using System;
 using System.Collections.Generic;  // Dictionnary
-using NDesk.DBus;
+using DBus;
 
 namespace CairoDock.Applet
 {
@@ -27,7 +27,7 @@ namespace CairoDock.Applet
 		Left
 	}
 
-	[NDesk.DBus.Interface("org.cairodock.CairoDock.applet")]
+	[DBus.Interface("org.cairodock.CairoDock.applet")]
 	public interface IApplet
 	{
 		object Get(string cProperty);
diff -rupN cairo-dock-plug-ins-3.4.1.orig/Dbus/interfaces/mono/ISubApplet.cs cairo-dock-plug-ins-3.4.1/Dbus/interfaces/mono/ISubApplet.cs
--- cairo-dock-plug-ins-3.4.1.orig/Dbus/interfaces/mono/ISubApplet.cs	2015-03-12 09:59:41.067645289 +0100
+++ cairo-dock-plug-ins-3.4.1/Dbus/interfaces/mono/ISubApplet.cs	2015-03-12 10:00:16.522802989 +0100
@@ -1,6 +1,6 @@
 using System;
 using System.Collections.Generic;  // Dictionnary
-using NDesk.DBus;
+using DBus;
 
 namespace CairoDock.Applet
 {
