--- cairo-dock-plugins-3.4.1/dock-rendering/src/rendering-config.c.gcc10	2015-02-20 01:13:47.000000000 +0900
+++ cairo-dock-plugins-3.4.1/dock-rendering/src/rendering-config.c	2020-01-23 18:10:47.301788756 +0900
@@ -25,8 +25,8 @@
 #include "rendering-config.h"
 #include "rendering-rainbow.h"
 
-extern int iVanishingPointY;double my_fRainbowColor[4];
-double my_fRainbowLineColor[4];
+extern int iVanishingPointY;extern double my_fRainbowColor[4];
+extern double my_fRainbowLineColor[4];
 
 extern double my_fInclinationOnHorizon;
 extern double my_fForegroundRatio;
