From 55ec4f789df5836f21154a2d6e25c0e9cb8cf814 Mon Sep 17 00:00:00 2001
From: Hans Schoenemann <hannes@mathematik.uni-kl.de>
Date: Wed, 1 Aug 2018 15:35:44 +0200
Subject: [PATCH] fix: tr. #834 (gcd in Z[x])

---

diff --git a/kernel/polys.cc b/kernel/polys.cc
index 3c66ccd5d2..a7b2f47ec6 100644
--- a/kernel/polys.cc
+++ b/kernel/polys.cc
@@ -180,8 +180,9 @@ poly singclap_gcd ( poly f, poly g, const ring r )
   }
   else         return f; // g==0 => gcd=f (but do a p_Cleardenom/pNorm)
   if (f==NULL) return g; // f==0 => gcd=g (but do a p_Cleardenom/pNorm)
-  if(p_IsConstant(f,currRing)
-  ||p_IsConstant(g,currRing))
+  if(!rField_is_Ring(currRing)
+  && (p_IsConstant(f,currRing)
+  ||p_IsConstant(g,currRing)))
   {
     res=p_One(currRing);
   }
