diff --git a/Documentation/Makefile b/Documentation/Makefile
index 067f698..0532c31 100644
--- a/Documentation/Makefile
+++ b/Documentation/Makefile
@@ -1,12 +1,5 @@
 PANDOC:=$(shell \
-	if pandoc </dev/null 2>/dev/null; then \
-		echo pandoc; \
-		touch .docs-available; \
-	else \
-		echo "Warning: pandoc not installed; can't generate manpages." >&2; \
-		echo '@echo Skipping: pandoc'; \
-		rm -f .docs-available; \
-	fi)
+		touch .docs-available)
 BUP_VERSION:=$(shell ../bup version --tag)
 BUP_DATE:=$(shell ../bup version --date)
 
@@ -19,10 +12,10 @@ man: $(patsubst %.md,%.1,$(wildcard *.md))
 html: $(patsubst %.md,%.html,$(wildcard *.md))
 
 %.1: %.md.tmp Makefile
-	$(PANDOC) -s -r markdown -w man -o $@ $<
+	ronn -r --pipe $< > $@
 	
 %.html: %.md.tmp Makefile
-	$(PANDOC) -s -r markdown -w html -o $@ $<
+	 rdiscount $< > $@
 	
 .PRECIOUS: %.md.tmp
 %.md.tmp: %.md Makefile
