diff -u -r gnash-0.8.9/libbase/log.h gnash-0.8.9-nodebug/libbase/log.h
--- gnash-0.8.9/libbase/log.h	2011-02-26 19:11:08.000000000 +0100
+++ gnash-0.8.9-nodebug/libbase/log.h	2012-01-18 09:39:02.004103494 +0100
@@ -331,27 +331,27 @@
 
 // Define to 0 to completely remove parse debugging at compile-time
 #ifndef VERBOSE_PARSE
-#define VERBOSE_PARSE 1
+#define VERBOSE_PARSE 0
 #endif
 
 // Define to 0 to completely remove action debugging at compile-time
 #ifndef VERBOSE_ACTION
-#define VERBOSE_ACTION 1
+#define VERBOSE_ACTION 0
 #endif
 
 // Define to 0 to remove ActionScript errors verbosity at compile-time
 #ifndef VERBOSE_ASCODING_ERRORS
-#define VERBOSE_ASCODING_ERRORS  1
+#define VERBOSE_ASCODING_ERRORS  0
 #endif
 
 // Define to 0 this to remove invalid SWF verbosity at compile-time
 #ifndef VERBOSE_MALFORMED_SWF
-#define VERBOSE_MALFORMED_SWF 1
+#define VERBOSE_MALFORMED_SWF 0
 #endif
 
 // Define to 0 this to remove Networking verbosity at compile-time
 #ifndef VERBOSE_NETWORKING
-#define VERBOSE_NETWORKING 1
+#define VERBOSE_NETWORKING 0
 #endif
 
 #if VERBOSE_PARSE
diff -u -r gnash-0.8.9/libcore/vm/ActionExec.cpp gnash-0.8.9-nodebug/libcore/vm/ActionExec.cpp
--- gnash-0.8.9/libcore/vm/ActionExec.cpp	2011-03-13 17:47:36.000000000 +0100
+++ gnash-0.8.9-nodebug/libcore/vm/ActionExec.cpp	2012-01-18 09:40:25.653458508 +0100
@@ -46,7 +46,7 @@
 // too much information for my tastes. I really want just
 // to see how stack changes while executing actions...
 // --strk Fri Jun 30 02:28:46 CEST 2006
-# define DEBUG_STACK 1
+//# define DEBUG_STACK 1
 
 // Max number of stack item to dump. 0 for unlimited.
 # define STACK_DUMP_LIMIT 32
