diff -u -r gnash-0.8.9/plugin/npapi/mozilla-sdk/np_entry.cpp gnash-0.8.9-xul/plugin/npapi/mozilla-sdk/np_entry.cpp
--- gnash-0.8.9/plugin/npapi/mozilla-sdk/np_entry.cpp	2011-03-13 17:47:36.000000000 +0100
+++ gnash-0.8.9-xul/plugin/npapi/mozilla-sdk/np_entry.cpp	2011-11-18 16:25:55.198007371 +0100
@@ -238,7 +238,7 @@
     return NS_PluginInitialize();
 }
 
-char *
+const char *
 NP_GetMIMEDescription(void)
 {
     return NPP_GetMIMEDescription();
diff -u -r gnash-0.8.9/plugin/npapi/plugin.cpp gnash-0.8.9-xul/plugin/npapi/plugin.cpp
--- gnash-0.8.9/plugin/npapi/plugin.cpp	2011-03-18 14:44:30.000000000 +0100
+++ gnash-0.8.9-xul/plugin/npapi/plugin.cpp	2011-11-18 16:35:42.496794816 +0100
@@ -128,10 +128,10 @@
 }
 
 /// \brief Return the MIME Type description for this plugin.
-char*
+const char*
 NPP_GetMIMEDescription(void)
 {
-    return const_cast<char *>(MIME_TYPES_DESCRIPTION);
+    return MIME_TYPES_DESCRIPTION;
 }
 
 static bool waitforgdb = false;
diff -u -r gnash-0.8.9/plugin/npapi/test.cpp gnash-0.8.9-xul/plugin/npapi/test.cpp
--- gnash-0.8.9/plugin/npapi/test.cpp	2011-03-13 17:47:36.000000000 +0100
+++ gnash-0.8.9-xul/plugin/npapi/test.cpp	2011-11-18 16:37:06.997539299 +0100
@@ -408,10 +408,10 @@
 {
 }
 
-char*
+const char*
 NPP_GetMIMEDescription(void)
 {
-    char *x = 0;
+    const char *x = 0;
     return x;
 }
 
