diff -ruN mednafen.orig//src/pcfx/soundbox.cpp mednafen/src/pcfx/soundbox.cpp
--- mednafen.orig//src/pcfx/soundbox.cpp	2010-06-05 09:35:40.000000000 -0400
+++ mednafen/src/pcfx/soundbox.cpp	2010-06-05 09:35:55.000000000 -0400
@@ -416,7 +416,7 @@
   int rolloff = (int)((double)0.90 * 21477272 * 2 / 1365 / (1 << SampleFreq) ) / 2;
   for(int ch = 0; ch < 2; ch++)
    for(int lr = 0; lr < 2; lr++)
-    ADPCMSynth[ch][lr].treble_eq(blip_eq_t::blip_eq_t(-1000, rolloff, FSettings.SndRate));
+    ADPCMSynth[ch][lr].treble_eq(blip_eq_t(-1000, rolloff, FSettings.SndRate));
   LSampleFreq = SampleFreq;
  }
 }
