diff --git c/src/core/init.cpp w/src/core/init.cpp
index ef551592..ca5ec023 100644
--- c/src/core/init.cpp
+++ w/src/core/init.cpp
@@ -27,6 +27,7 @@
 
 #include <thread>
 #include <ctime>
+#include <atomic>
 #ifdef __APPLE__
 	#include <pwd.h>
 #endif
@@ -245,4 +246,4 @@ void shutdown()
 	gu_log("[init] Giada " G_VERSION_STR " closed\n\n");
 	gu_logClose();
 }
-}}} // giada::m::init
\ No newline at end of file
+}}} // giada::m::init
