diff -ruN a/src/core/conf.cpp b/src/core/conf.cpp
--- a/src/core/conf.cpp	2021-03-31 18:44:48.000000000 +0200
+++ b/src/core/conf.cpp	2021-04-24 16:18:22.530287443 +0200
@@ -27,7 +27,7 @@
 #include "conf.h"
 #include "core/const.h"
 #include "core/types.h"
-#include "deps/json/single_include/nlohmann/json.hpp"
+#include <nlohmann/json.hpp>
 #include "utils/fs.h"
 #include "utils/log.h"
 #include <FL/Fl.H>
@@ -315,4 +315,4 @@
 	ofs << j;
 	return true;
 }
-} // namespace giada::m::conf
\ No newline at end of file
+} // namespace giada::m::conf
diff -ruN a/src/core/init.cpp b/src/core/init.cpp
--- a/src/core/init.cpp	2021-03-31 18:44:48.000000000 +0200
+++ b/src/core/init.cpp	2021-04-24 16:19:01.257069455 +0200
@@ -54,7 +54,7 @@
 #include "core/sequencer.h"
 #include "core/wave.h"
 #include "core/waveManager.h"
-#include "deps/json/single_include/nlohmann/json.hpp"
+#include <nlohmann/json.hpp>
 #include "glue/main.h"
 #include "gui/dialogs/mainWindow.h"
 #include "gui/dialogs/warnings.h"
diff -ruN a/src/core/midiMapConf.cpp b/src/core/midiMapConf.cpp
--- a/src/core/midiMapConf.cpp	2021-03-31 18:44:48.000000000 +0200
+++ b/src/core/midiMapConf.cpp	2021-04-24 16:17:56.890211073 +0200
@@ -26,7 +26,7 @@
 
 #include "midiMapConf.h"
 #include "const.h"
-#include "deps/json/single_include/nlohmann/json.hpp"
+#include <nlohmann/json.hpp>
 #include "utils/fs.h"
 #include "utils/log.h"
 #include "utils/string.h"
diff -ruN a/src/core/patch.cpp b/src/core/patch.cpp
--- a/src/core/patch.cpp	2021-03-31 18:44:48.000000000 +0200
+++ b/src/core/patch.cpp	2021-04-24 16:19:23.867136797 +0200
@@ -26,7 +26,7 @@
 
 #include "patch.h"
 #include "core/mixer.h"
-#include "deps/json/single_include/nlohmann/json.hpp"
+#include <nlohmann/json.hpp>
 #include "utils/log.h"
 #include "utils/math.h"
 #include <fstream>
