diff -ruN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2020-11-15 21:26:40.000000000 +0100
+++ b/Makefile.am	2020-11-20 18:53:06.031868664 +0100
@@ -402,7 +402,7 @@
 if LINUX
 
 # Add preprocessor flags to enable ALSA, Pulse and JACK in RtAudio.
-cppFlags += -D__LINUX_ALSA__ -D__LINUX_PULSE__ -D__UNIX_JACK__
+cppFlags += -D__LINUX_ALSA__ -D__LINUX_PULSE__ -D__UNIX_JACK__ $(RTMIDI_CFLAGS)
 
 ldAdd += -lsndfile -lfltk -lXext -lX11 -lXft -lXpm -lm -ljack -lasound \
 	-lpthread -ldl -lpulse-simple -lpulse -lsamplerate -lrtmidi \
diff -ruN a/configure.ac b/configure.ac
--- a/configure.ac	2020-11-15 21:26:40.000000000 +0100
+++ b/configure.ac	2020-11-20 18:52:11.715602516 +0100
@@ -138,6 +138,7 @@
 		[],
 		[AC_MSG_ERROR([library 'rtMidi' not found!])]
 	)
+	PKG_CHECK_MODULES([RTMIDI], [rtmidi])
 	AC_LANG_POP
 fi
 
