--- sage-brial/brial/gbrefs.py.orig	2019-09-01 22:09:29.585226434 +0000
+++ sage-brial/brial/gbrefs.py	2019-09-01 22:10:49.416270516 +0000
@@ -6,7 +6,7 @@
     from io import StringIO
 import uu
 import re
-import imp
+import types
 from .PyPolyBoRi import *
 AUTO = "auto"
 SINGLE = "single"
@@ -118,7 +118,7 @@
 
 
 def dyn_generate(content, name):
-    module = imp.new_module(name)
+    module = types.ModuleType(name)
     import_header = """from .PyPolyBoRi import Variable,Monomial, Polynomial, Ring, OrderCode
 from itertools import chain
 from .blocks import AlternatingBlock,Block,AdderBlock,if_then,HigherOrderBlock,declare_ring as orig_declare_ring,declare_block_scheme,MacroBlock\n
