diff -ruN zita-bls1-0.1.0/source/shuffler.cc zita-bls1-0.1.0-b/source/shuffler.cc
--- zita-bls1-0.1.0/source/shuffler.cc	2012-01-10 18:51:09.000000000 +0100
+++ zita-bls1-0.1.0-b/source/shuffler.cc	2018-08-23 16:27:12.523304581 +0200
@@ -75,7 +75,7 @@
     memset (_fft_time, 0, _iplen * sizeof (float));
     _fft_time [_iplen / 2] = 1.0f;
 
-    _convproc.configure (1, 1, _iplen, _quant, _minpt, _minpt);
+    _convproc.configure (1, 1, _iplen, _quant, _minpt, _minpt, 1.0f);
     _convproc.impdata_create (0, 0, 1, _fft_time, 0, _iplen);
     _convproc.start_process (abspri, policy);
 }
