diff -wbBur libmatchbox-1.9/libmb/mbpixbuf.c libmatchbox-1.9.org/libmb/mbpixbuf.c
--- libmatchbox-1.9/libmb/mbpixbuf.c	2012-01-19 21:57:56.000000000 +0400
+++ libmatchbox-1.9.org/libmb/mbpixbuf.c	2006-07-26 23:12:25.000000000 +0400
@@ -247,7 +247,7 @@
     return NULL;
   }
 
-  if ( setjmp( png_jmpbuf(png_ptr) ) ) {
+  if ( setjmp( png_ptr->jmpbuf ) ) {
     png_destroy_read_struct( &png_ptr, &info_ptr, NULL);
     fclose(fd);
     return NULL;
@@ -269,8 +269,8 @@
       ( color_type == PNG_COLOR_TYPE_GRAY_ALPHA ))
     png_set_gray_to_rgb(png_ptr);
  
-  if ( png_get_color_type(png_ptr, info_ptr) == PNG_COLOR_TYPE_RGB_ALPHA 
-       || png_get_color_type(png_ptr, info_ptr) == PNG_COLOR_TYPE_GRAY_ALPHA
+  if ( info_ptr->color_type == PNG_COLOR_TYPE_RGB_ALPHA 
+       || info_ptr->color_type == PNG_COLOR_TYPE_GRAY_ALPHA
        )
     *has_alpha = 1;
   else
