diff --git a/library/ByteString/TreeBuilder.hs.orig b/library/ByteString/TreeBuilder.hs
index 18c7f05..5f8ffbb 100644
--- a/library/ByteString/TreeBuilder.hs.orig
+++ b/library/ByteString/TreeBuilder.hs
@@ -43,9 +43,6 @@ instance Monoid Builder where
   {-# INLINE mempty #-}
   mempty =
     Builder 0 A.Empty
-  {-# INLINABLE mappend #-}
-  mappend (Builder length1 tree1) (Builder length2 tree2) =
-    Builder (length1 + length2) (A.Branch tree1 tree2)
   {-# INLINE mconcat #-}
   mconcat =
     foldl' mappend mempty
@@ -54,6 +51,8 @@ instance Semigroup Builder where
   {-# INLINE sconcat #-}
   sconcat =
     foldl' mappend mempty
+  Builder length1 tree1 <> Builder length2 tree2 =
+    Builder (length1 + length2) (A.Branch tree1 tree2)
 
 instance IsString Builder where
   {-# INLINE fromString #-}
