Index: src/qcommon/q_shared.h
===================================================================
--- src/qcommon/q_shared.h	(revision 1112)
+++ src/qcommon/q_shared.h	(working copy)
@@ -953,6 +953,7 @@
 // if none of the catchers are active, bound key strings will be executed
 #define KEYCATCH_CONSOLE		0x0001
 #define	KEYCATCH_UI					0x0002
+#define	KEYCATCH_MESSAGE		0x0004
 #define	KEYCATCH_CGAME			0x0008
 
 
@@ -1070,8 +1071,6 @@
 	int			torsoTimer;		// don't change low priority animations until this runs out
 	int			torsoAnim;		// mask off ANIM_TOGGLEBIT
 
-	int			weaponAnim;		// mask off ANIM_TOGGLEBIT
-
 	int			movementDir;	// a number 0 to 7 that represents the reletive angle
 								// of movement to the view angle (axial and diagonals)
 								// when at rest, the value will remain unchanged
@@ -1107,6 +1106,7 @@
 	int			misc[MAX_MISC];	// misc data
 	int			ammo;			// ammo held
 	int			clips;			// clips held
+	int			padding[14];
 
 	int			generic1;
 	int			loopSound;
@@ -1228,7 +1228,6 @@
 	int		weapon;			// determines weapon and flash model, etc
 	int		legsAnim;		// mask off ANIM_TOGGLEBIT
 	int		torsoAnim;		// mask off ANIM_TOGGLEBIT
-	int		weaponAnim;		// mask off ANIM_TOGGLEBIT
 
 	int		generic1;
 } entityState_t;
@@ -1296,9 +1295,9 @@
 
 // server browser sources
 // AS_MPLAYER is no longer used
-#define AS_GLOBAL			0
+#define AS_GLOBAL			2
 #define AS_MPLAYER		1
-#define AS_LOCAL			2
+#define AS_LOCAL			0
 #define AS_FAVORITES	3
 
 
Index: src/qcommon/qcommon.h
===================================================================
--- src/qcommon/qcommon.h	(revision 1112)
+++ src/qcommon/qcommon.h	(working copy)
@@ -229,7 +229,7 @@
 ==============================================================
 */
 
-#define	PROTOCOL_VERSION	70
+#define	PROTOCOL_VERSION	69
 
 // maintain a list of compatible protocols for demo playing
 // NOTE: that stuff only works with two digits protocols
Index: src/qcommon/msg.c
===================================================================
--- src/qcommon/msg.c	(revision 1112)
+++ src/qcommon/msg.c	(working copy)
@@ -819,7 +819,6 @@
 { NETF(angles2[1]), 0 },
 { NETF(eType), 8 },
 { NETF(torsoAnim), 8 },
-{ NETF(weaponAnim), 8 },
 { NETF(eventParm), 8 },
 { NETF(legsAnim), 8 },
 { NETF(groundEntityNum), GENTITYNUM_BITS },
@@ -839,7 +838,7 @@
 { NETF(modelindex), 8 },
 { NETF(otherEntityNum2), GENTITYNUM_BITS },
 { NETF(loopSound), 8 },
-{ NETF(generic1), 16 },
+{ NETF(generic1), 8 },
 { NETF(origin2[2]), 0 },
 { NETF(origin2[0]), 0 },
 { NETF(origin2[1]), 0 },
@@ -1135,7 +1134,6 @@
 { PSF(pm_time), -16 },
 { PSF(eventSequence), 16 },
 { PSF(torsoAnim), 8 },
-{ PSF(weaponAnim), 8 },
 { PSF(movementDir), 4 },
 { PSF(events[0]), 8 },
 { PSF(legsAnim), 8 },
@@ -1154,9 +1152,7 @@
 { PSF(damageYaw), 8 },
 { PSF(damagePitch), 8 },
 { PSF(damageCount), 8 },
-{ PSF(ammo), 12 },
-{ PSF(clips), 4 },
-{ PSF(generic1), 16 },
+{ PSF(generic1), 8 },
 { PSF(pm_type), 8 },					
 { PSF(delta_angles[0]), 16 },
 { PSF(delta_angles[2]), 16 },
@@ -1184,6 +1180,7 @@
 	playerState_t	dummy;
 	int				statsbits;
 	int				persistantbits;
+	int				ammobits;
 	int				miscbits;
 	int				numFields;
 	int				c;
@@ -1264,6 +1261,12 @@
 			persistantbits |= 1<<i;
 		}
 	}
+	ammobits = (to->ammo != from->ammo) | (to->clips != from->clips) << 1;
+	for (i=0 ; i<14 ; i++) {
+		if (to->padding[i] != from->padding[i]) {
+			ammobits |= 4<<i;
+		}
+	}
 	miscbits = 0;
 	for (i=0 ; i<MAX_MISC ; i++) {
 		if (to->misc[i] != from->misc[i]) {
@@ -1271,7 +1274,7 @@
 		}
 	}
 
-	if (!statsbits && !persistantbits && !miscbits) {
+	if (!statsbits && !persistantbits && !ammobits && !miscbits) {
 		MSG_WriteBits( msg, 0, 1 );	// no change
 		oldsize += 4;
 		return;
@@ -1300,6 +1303,21 @@
 	}
 
 
+	if ( ammobits ) {
+		MSG_WriteBits( msg, 1, 1 );	// changed
+		MSG_WriteBits( msg, ammobits, MAX_WEAPONS );
+		if(ammobits & 1)
+			MSG_WriteShort(msg, to->ammo);
+		if(ammobits & 2)
+			MSG_WriteShort(msg, to->clips);
+		for (i=0 ; i<14 ; i++)
+			if (ammobits & (4<<i) )
+				MSG_WriteShort (msg, to->padding[i]);
+	} else {
+		MSG_WriteBits( msg, 0, 1 );	// no change
+	}
+
+
 	if ( miscbits ) {
 		MSG_WriteBits( msg, 1, 1 );	// changed
 		MSG_WriteBits( msg, miscbits, MAX_MISC );
@@ -1419,6 +1437,21 @@
 			}
 		}
 
+		// parse ammo
+		if ( MSG_ReadBits( msg, 1 ) ) {
+			LOG("PS_AMMO");
+			bits = MSG_ReadBits (msg, MAX_WEAPONS);
+			if(bits & 1)
+				to->ammo = MSG_ReadShort(msg);
+			if(bits & 2)
+				to->clips = MSG_ReadShort(msg);
+			for (i=0 ; i<14 ; i++) {
+				if (bits & (4<<i) ) {
+					to->padding[i] = MSG_ReadShort(msg);
+				}
+			}
+		}
+
 		// parse misc data
 		if ( MSG_ReadBits( msg, 1 ) ) {
 			LOG("PS_MISC");
Index: src/cgame/cg_public.h
===================================================================
--- src/cgame/cg_public.h	(revision 1112)
+++ src/cgame/cg_public.h	(working copy)
@@ -138,6 +138,11 @@
   CG_KEY_GETCATCHER,
   CG_KEY_SETCATCHER,
   CG_KEY_GETKEY,
+  CG_PARSE_ADD_GLOBAL_DEFINE,
+  CG_PARSE_LOAD_SOURCE,
+  CG_PARSE_FREE_SOURCE,
+  CG_PARSE_READ_TOKEN,
+  CG_PARSE_SOURCE_FILE_AND_LINE,
   CG_S_STOPBACKGROUNDTRACK,
   CG_REAL_TIME,
   CG_SNAPVECTOR,
@@ -172,12 +177,6 @@
   CG_KEY_GETBINDINGBUF,
   CG_KEY_SETBINDING,
 
-  CG_PARSE_ADD_GLOBAL_DEFINE,
-  CG_PARSE_LOAD_SOURCE,
-  CG_PARSE_FREE_SOURCE,
-  CG_PARSE_READ_TOKEN,
-  CG_PARSE_SOURCE_FILE_AND_LINE,
-
   CG_KEY_SETOVERSTRIKEMODE,
   CG_KEY_GETOVERSTRIKEMODE,
 
Index: src/ui/ui_public.h
===================================================================
--- src/ui/ui_public.h	(revision 1112)
+++ src/ui/ui_public.h	(working copy)
@@ -94,6 +94,11 @@
   UI_MEMORY_REMAINING,
   UI_R_REGISTERFONT,
   UI_R_MODELBOUNDS,
+  UI_PARSE_ADD_GLOBAL_DEFINE,
+  UI_PARSE_LOAD_SOURCE,
+  UI_PARSE_FREE_SOURCE,
+  UI_PARSE_READ_TOKEN,
+  UI_PARSE_SOURCE_FILE_AND_LINE,
   UI_S_STOPBACKGROUNDTRACK,
   UI_S_STARTBACKGROUNDTRACK,
   UI_REAL_TIME,
@@ -121,12 +126,6 @@
   UI_FS_SEEK,
   UI_SET_PBCLSTATUS,
 
-  UI_PARSE_ADD_GLOBAL_DEFINE,
-  UI_PARSE_LOAD_SOURCE,
-  UI_PARSE_FREE_SOURCE,
-  UI_PARSE_READ_TOKEN,
-  UI_PARSE_SOURCE_FILE_AND_LINE,
-
   UI_MEMSET = 100,
   UI_MEMCPY,
   UI_STRNCPY,
Index: src/client/cl_console.c
===================================================================
--- src/client/cl_console.c	(revision 1112)
+++ src/client/cl_console.c	(working copy)
@@ -82,6 +82,67 @@
 
 /*
 ================
+Con_MessageMode_f
+================
+*/
+void Con_MessageMode_f (void) {
+	chat_playerNum = -1;
+	chat_team = qfalse;
+	Field_Clear( &chatField );
+	chatField.widthInChars = 30;
+
+	Key_SetCatcher( Key_GetCatcher( ) ^ KEYCATCH_MESSAGE );
+}
+
+/*
+================
+Con_MessageMode2_f
+================
+*/
+void Con_MessageMode2_f (void) {
+	chat_playerNum = -1;
+	chat_team = qtrue;
+	Field_Clear( &chatField );
+	chatField.widthInChars = 25;
+	Key_SetCatcher( Key_GetCatcher( ) ^ KEYCATCH_MESSAGE );
+}
+
+/*
+================
+Con_MessageMode3_f
+================
+*/
+void Con_MessageMode3_f (void) {
+	chat_playerNum = VM_Call( cgvm, CG_CROSSHAIR_PLAYER );
+	if ( chat_playerNum < 0 || chat_playerNum >= MAX_CLIENTS ) {
+		chat_playerNum = -1;
+		return;
+	}
+	chat_team = qfalse;
+	Field_Clear( &chatField );
+	chatField.widthInChars = 30;
+	Key_SetCatcher( Key_GetCatcher( ) ^ KEYCATCH_MESSAGE );
+}
+
+/*
+================
+Con_MessageMode4_f
+================
+*/
+void Con_MessageMode4_f (void) {
+	chat_playerNum = VM_Call( cgvm, CG_LAST_ATTACKER );
+	if ( chat_playerNum < 0 || chat_playerNum >= MAX_CLIENTS ) {
+		chat_playerNum = -1;
+		return;
+	}
+	chat_team = qfalse;
+	Field_Clear( &chatField );
+	chatField.widthInChars = 30;
+	Key_SetCatcher( Key_GetCatcher( ) ^ KEYCATCH_MESSAGE );
+}
+
+/*
+================
 Con_Clear_f
 ================
 */
@@ -253,6 +314,10 @@
 	CL_LoadConsoleHistory( );
 
 	Cmd_AddCommand ("toggleconsole", Con_ToggleConsole_f);
+	Cmd_AddCommand ("messagemode", Con_MessageMode_f);
+	Cmd_AddCommand ("messagemode2", Con_MessageMode2_f);
+	Cmd_AddCommand ("messagemode3", Con_MessageMode3_f);
+	Cmd_AddCommand ("messagemode4", Con_MessageMode4_f);
 	Cmd_AddCommand ("clear", Con_Clear_f);
 	Cmd_AddCommand ("condump", Con_Dump_f);
 }
@@ -541,6 +606,26 @@
 
 	if( Key_GetCatcher( ) & ( KEYCATCH_UI | KEYCATCH_CGAME ) )
 		return;
+
+	// draw the chat line
+	if( Key_GetCatcher( ) & KEYCATCH_MESSAGE )
+	{
+		int skip;
+
+		if( chat_team )
+		{
+			SCR_DrawBigString( 8, 232, "Team Say:", 1.0f, qfalse );
+			skip = 11;
+		}
+		else
+		{ 
+			SCR_DrawBigString( 8, 232, "Say:", 1.0f, qfalse );
+			skip = 5;
+		}
+
+		Field_BigDraw( &chatField, skip * BIGCHAR_WIDTH, 232,
+				SCREEN_WIDTH - ( skip + 1 ) * BIGCHAR_WIDTH, qtrue, qtrue );
+	}
 }
 
 //================================================================
Index: src/client/cl_keys.c
===================================================================
--- src/client/cl_keys.c	(revision 1112)
+++ src/client/cl_keys.c	(working copy)
@@ -35,7 +35,10 @@
 							// will be <= nextHistoryLine
 
 field_t		g_consoleField;
+field_t		chatField;
+qboolean	chat_team;
 
+int			chat_playerNum;
 
 qboolean	key_overstrikeMode;
 
@@ -701,7 +704,52 @@
 	Field_KeyDownEvent( &g_consoleField, key );
 }
 
+//============================================================================
 
+
+/*
+================
+Message_Key
+
+In game talk message
+================
+*/
+void Message_Key( int key ) {
+
+	char	buffer[MAX_STRING_CHARS];
+
+
+	if (key == K_ESCAPE) {
+		Key_SetCatcher( Key_GetCatcher( ) & ~KEYCATCH_MESSAGE );
+		Field_Clear( &chatField );
+		return;
+	}
+
+	if ( key == K_ENTER || key == K_KP_ENTER )
+	{
+		if ( chatField.buffer[0] && cls.state == CA_ACTIVE ) {
+			if (chat_playerNum != -1 )
+
+				Com_sprintf( buffer, sizeof( buffer ), "tell %i \"%s\"\n", chat_playerNum, chatField.buffer );
+
+			else if (chat_team)
+
+				Com_sprintf( buffer, sizeof( buffer ), "say_team \"%s\"\n", chatField.buffer );
+			else
+				Com_sprintf( buffer, sizeof( buffer ), "say \"%s\"\n", chatField.buffer );
+
+
+
+			CL_AddReliableCommand( buffer );
+		}
+		Key_SetCatcher( Key_GetCatcher( ) & ~KEYCATCH_MESSAGE );
+		Field_Clear( &chatField );
+		return;
+	}
+
+	Field_KeyDownEvent( &chatField, key );
+}
+
 //============================================================================
 
 
@@ -1148,6 +1196,12 @@
 
 	// escape is always handled special
 	if ( key == K_ESCAPE && down ) {
+		if ( Key_GetCatcher( ) & KEYCATCH_MESSAGE ) {
+			// clear message mode
+			Message_Key( key );
+			return;
+		}
+
 		// escape always gets out of CGAME stuff
 		if (Key_GetCatcher( ) & KEYCATCH_CGAME) {
 			Key_SetCatcher( Key_GetCatcher( ) & ~KEYCATCH_CGAME );
@@ -1205,6 +1259,8 @@
 		if ( cgvm ) {
 			VM_Call( cgvm, CG_KEY_EVENT, key, down );
 		} 
+	} else if ( Key_GetCatcher( ) & KEYCATCH_MESSAGE ) {
+		Message_Key( key );
 	} else if ( cls.state == CA_DISCONNECTED ) {
 		Console_Key( key );
 	} else {
@@ -1279,6 +1335,10 @@
 	{
 		VM_Call( uivm, UI_KEY_EVENT, key | K_CHAR_FLAG, qtrue );
 	}
+	else if ( Key_GetCatcher( ) & KEYCATCH_MESSAGE ) 
+	{
+		Field_CharEvent( &chatField, key );
+	}
 	else if ( cls.state == CA_DISCONNECTED )
 	{
 		Field_CharEvent( &g_consoleField, key );
Index: Makefile
===================================================================
--- Makefile	(revision 1112)
+++ Makefile	(working copy)
@@ -52,6 +52,8 @@
 #
 #############################################################################
 -include Makefile.local
+BUILD_GAME_SO=0
+BUILD_GAME_QVM=0
 
 ifndef PLATFORM
 PLATFORM=$(COMPILE_PLATFORM)
