Index: src/client/cl_ui.c
===================================================================
--- src/client/cl_ui.c	(revision 9)
+++ src/client/cl_ui.c	(working copy)
@@ -364,7 +364,7 @@
 		Info_SetValueForKey( info, "game", server->game);
 		Info_SetValueForKey( info, "gametype", va("%i",server->gameType));
 		Info_SetValueForKey( info, "nettype", va("%i",server->netType));
-		Info_SetValueForKey( info, "addr", NET_AdrToString(server->adr));
+		Info_SetValueForKey( info, "addr", NET_AdrToStringwPort(server->adr));
 		Q_strncpyz(buf, info, buflen);
 	} else {
 		if (buf) {
