From 4ea9d5b78540c972e3fe1bf44db9f7b3f87c0ad0 Mon Sep 17 00:00:00 2001
From: Folkert van Heusden <folkert@vanheusden.com>
Date: Sun, 3 Oct 2021 20:33:21 +0200
Subject: theoretical security problem fix

---
 nc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nc.c b/nc.c
index ffb6d17..e60fbe8 100644
--- a/nc.c
+++ b/nc.c
@@ -235,7 +235,7 @@ void myprint(WINDOW *w, const char *what)
 	}
 	else
 	{
-		wprintw(w, what);
+		wprintw(w, "%s", what);
 	}
 }
 
-- 
2.20.1

