diff --git a/Codec/Encryption/OpenPGP/Types/Internal/Base.hs b/Codec/Encryption/OpenPGP/Types/Internal/Base.hs
index f3dcc4a..e72b5a2 100644
--- a/Codec/Encryption/OpenPGP/Types/Internal/Base.hs
+++ b/Codec/Encryption/OpenPGP/Types/Internal/Base.hs
@@ -2,6 +2,7 @@
 -- Copyright © 2012-2018  Clint Adams
 -- This software is released under the terms of the Expat license.
 -- (See the LICENSE file).
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE DeriveGeneric #-}
@@ -671,8 +672,10 @@ instance FutureFlag FeatureFlag where
 
 instance Hashable FeatureFlag
 
+#if !MIN_VERSION_hashable(1,3,4)
 instance Hashable a => Hashable (Set a) where
   hashWithSalt salt = hashWithSalt salt . Set.toList
+#endif
 
 instance Pretty FeatureFlag where
   pretty ModificationDetection = pretty "modification-detection"
