diff --git a/ui-gtk/UgtkApp.c b/ui-gtk/UgtkApp.c
index c62bd51..361adbb 100644
--- a/ui-gtk/UgtkApp.c
+++ b/ui-gtk/UgtkApp.c
@@ -62,7 +62,6 @@ void  ugtk_app_init (UgtkApp* app, UgetRpc* rpc)
 	uget_app_set_config_dir ((UgetApp*) app, dir);
 	g_free (dir);
 
-	app->rss_builtin = uget_rss_new ();
 	ugtk_app_load (app);
 	ugtk_app_init_ui (app);
 	ugtk_app_init_callback (app);
@@ -88,11 +87,6 @@ void  ugtk_app_init (UgtkApp* app, UgetRpc* rpc)
 
 	app->recent.category_index = 0;
 	app->recent.info = ug_info_new(8, 0);
-	// RSS
-	uget_rss_add_builtin (app->rss_builtin, UGET_RSS_STABLE);
-	uget_rss_add_builtin (app->rss_builtin, UGET_RSS_NEWS);
-	uget_rss_add_builtin (app->rss_builtin, UGET_RSS_TUTORIALS);
-	uget_rss_update (app->rss_builtin, FALSE);
 	gtk_widget_hide (app->banner.self);
 
 	uget_app_use_uri_hash ((UgetApp*) app);
@@ -118,7 +112,6 @@ void  ugtk_app_final (UgtkApp* app)
 	else
 		shutdown_now = FALSE;
 	ug_info_unref(app->recent.info);
-	uget_rss_unref (app->rss_builtin);
 	uget_app_final ((UgetApp*) app);
 	// plug-in finalize
 	uget_plugin_global_set(UgetPluginAria2Info, UGET_PLUGIN_ARIA2_GLOBAL_SHUTDOWN_NOW,
@@ -140,11 +133,6 @@ void  ugtk_app_save (UgtkApp* app)
 	ugtk_setting_save (&app->setting, file);
 	g_free (file);
 
-	// RSS
-	file = g_build_filename (app->config_dir, "RSS-built-in.json", NULL);
-	uget_rss_save_feeds (app->rss_builtin, file);
-	g_free (file);
-
 //	uget_app_save_categories ((UgetApp*) app, ugtk_get_config_dir ());
 	uget_app_save_categories ((UgetApp*) app, NULL);
 }
@@ -166,11 +154,6 @@ void  ugtk_app_load (UgtkApp* app)
 		                7*24 - app->setting.scheduler.state.length);
 	}
 
-	// RSS
-	file = g_build_filename (app->config_dir, "RSS-built-in.json", NULL);
-	uget_rss_load_feeds (app->rss_builtin, file);
-	g_free (file);
-
 //	uget_app_load_categories ((UgetApp*) app, ugtk_get_config_dir ());
 	counts = uget_app_load_categories ((UgetApp*) app, NULL);
 	if (counts == 0)
