diff --git i/py/__init__.py w/py/__init__.py
index b892ce1a..725991e8 100644
--- i/py/__init__.py
+++ w/py/__init__.py
@@ -10,14 +10,8 @@ dictionary or an import path.
 """
 from py._error import error
 
-try:
-    from py._vendored_packages import apipkg
-    lib_not_mangled_by_packagers = True
-    vendor_prefix = '._vendored_packages.'
-except ImportError:
-    import apipkg
-    lib_not_mangled_by_packagers = False
-    vendor_prefix = ''
+import apipkg
+vendor_prefix = ''
 
 try:
     from ._version import version as __version__
