--- a/static/index.js
+++ b/static/index.js
@@ -66,6 +66,10 @@
         snapshotResult.setGlobals(global, process, window, document, console, require)
       }
 
+      if (!devMode) {
+        process.env.NODE_ENV = 'production'
+      }
+
       const FileSystemBlobStore = useSnapshot ? snapshotResult.customRequire('../src/file-system-blob-store.js') : require('../src/file-system-blob-store')
       blobStore = FileSystemBlobStore.load(path.join(process.env.ATOM_HOME, 'blob-store'))
 
