commit 96420ab204b6079eef1331faecf80fa73476c206
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Fri May 17 03:19:47 2019 +0800

    Fix compatibility with generics-sop 0.5

diff --git a/src/Data/Swagger/Internal/AesonUtils.hs b/src/Data/Swagger/Internal/AesonUtils.hs
index c1280f0..c785e76 100644
--- a/src/Data/Swagger/Internal/AesonUtils.hs
+++ b/src/Data/Swagger/Internal/AesonUtils.hs
@@ -144,7 +144,11 @@ sopSwaggerGenericToJSON'
     -> DatatypeInfo '[xs]
     -> POP Maybe '[xs]
     -> [Pair]
+#if MIN_VERSION_generics_sop(0,5,0)
+sopSwaggerGenericToJSON' opts (SOP (Z fields)) (ADT _ _ (Record _ fieldsInfo :* Nil) _) (POP (defs :* Nil)) =
+#else
 sopSwaggerGenericToJSON' opts (SOP (Z fields)) (ADT _ _ (Record _ fieldsInfo :* Nil)) (POP (defs :* Nil)) =
+#endif
     sopSwaggerGenericToJSON'' opts fields fieldsInfo defs
 sopSwaggerGenericToJSON' _ _ _ _ = error "sopSwaggerGenericToJSON: unsupported type"
 
@@ -220,7 +224,11 @@ sopSwaggerGenericParseJSON'
     -> DatatypeInfo '[xs]
     -> POP Maybe '[xs]
     -> Parser (SOP I '[xs])
+#if MIN_VERSION_generics_sop(0,5,0)
+sopSwaggerGenericParseJSON' opts obj (ADT _ _ (Record _ fieldsInfo :* Nil) _) (POP (defs :* Nil)) =
+#else
 sopSwaggerGenericParseJSON' opts obj (ADT _ _ (Record _ fieldsInfo :* Nil)) (POP (defs :* Nil)) =
+#endif
     SOP . Z <$> sopSwaggerGenericParseJSON'' opts obj fieldsInfo defs
 sopSwaggerGenericParseJSON' _ _ _ _ = error "sopSwaggerGenericParseJSON: unsupported type"
 
@@ -292,7 +300,11 @@ sopSwaggerGenericToEncoding'
     -> DatatypeInfo '[xs]
     -> POP Maybe '[xs]
     -> Series
+#if MIN_VERSION_generics_sop(0,5,0)
+sopSwaggerGenericToEncoding' opts (SOP (Z fields)) (ADT _ _ (Record _ fieldsInfo :* Nil) _) (POP (defs :* Nil)) =
+#else
 sopSwaggerGenericToEncoding' opts (SOP (Z fields)) (ADT _ _ (Record _ fieldsInfo :* Nil)) (POP (defs :* Nil)) =
+#endif
     sopSwaggerGenericToEncoding'' opts fields fieldsInfo defs
 sopSwaggerGenericToEncoding' _ _ _ _ = error "sopSwaggerGenericToEncoding: unsupported type"
 
diff --git a/swagger2.cabal b/swagger2.cabal
index a338669..2154705 100644
--- a/swagger2.cabal
+++ b/swagger2.cabal
@@ -77,7 +77,7 @@ library
     , aeson                     >=1.4.2.0  && <1.5
     -- cookie 0.4.3 is needed by GHC 7.8 due to time>=1.4 constraint
     , cookie                    >=0.4.3    && <0.5
-    , generics-sop              >=0.3.2.0  && <0.5
+    , generics-sop              >=0.3.2.0  && <0.6
     , hashable                  >=1.2.7.0  && <1.3
     , http-media                >=0.7.1.2  && <0.9
     , insert-ordered-containers >=0.2.1.0  && <0.3
