diff --git a/src/luarocks/build.lua b/src/luarocks/build.lua
index af098f3..3c7f440 100644
--- a/src/luarocks/build.lua
+++ b/src/luarocks/build.lua
@@ -3,6 +3,7 @@
 -- Builds a rock, compiling its C parts if any.
 module("luarocks.build", package.seeall)
 
+local pack = require("luarocks.pack")
 local path = require("luarocks.path")
 local util = require("luarocks.util")
 local rep = require("luarocks.rep")
