From 226b5c7b0aafc3cb744eb79a666d9a1bd0803a47 Mon Sep 17 00:00:00 2001
From: Massimiliano Torromeo <massimiliano.torromeo@gmail.com>
Date: Wed, 3 Sep 2014 09:40:27 +0200
Subject: [PATCH] Open README and HISTORY as utf-8 encoded files (fixes #2196)

---
 setup.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 9540a47..e39d13b 100755
--- a/setup.py
+++ b/setup.py
@@ -5,6 +5,8 @@
 
 import requests
 
+from codecs import open
+
 try:
     from setuptools import setup
 except ImportError:
@@ -27,9 +29,9 @@
 
 requires = ['certifi']
 
-with open('README.rst') as f:
+with open('README.rst', 'r', 'utf-8') as f:
     readme = f.read()
-with open('HISTORY.rst') as f:
+with open('HISTORY.rst', 'r', 'utf-8') as f:
     history = f.read()
 
 setup(
