From 671531892b78eeb12fbdeb025547d3b1201409bf Mon Sep 17 00:00:00 2001
From: jlh <jlh@gmx.ch>
Date: Wed, 10 Jul 2013 13:45:41 +0200
Subject: [PATCH] Fix linking issue with ogg/vorbis libraries

---
 CMakeLists.txt                   |  2 +-
 CMakeModules/Findvorbisenc.cmake | 10 ++++++----
 2 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index abcf2e3..33b79cb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,7 +81,7 @@ SET(LIBRARIES ${LIBRARIES} ${ID3_LIBRARY})
 
 FIND_PACKAGE(vorbisenc REQUIRED)
 INCLUDE_DIRECTORIES(${VORBISENC_INCLUDE_DIR})
-SET(LIBRARIES ${LIBRARIES} ${VORBISENC_LIBRARY})
+SET(LIBRARIES ${LIBRARIES} ${OGG_LIBRARY} ${VORBIS_LIBRARY} ${VORBISENC_LIBRARY})
 
 # Qt
 
diff --git a/CMakeModules/Findvorbisenc.cmake b/CMakeModules/Findvorbisenc.cmake
index d519b39..53210f0 100644
--- a/CMakeModules/Findvorbisenc.cmake
+++ b/CMakeModules/Findvorbisenc.cmake
@@ -1,18 +1,20 @@
 
 FIND_PATH(VORBISENC_INCLUDE_DIR vorbis/vorbisenc.h /usr/include /usr/local/include)
+FIND_LIBRARY(OGG_LIBRARY NAMES ogg PATH /usr/lib /usr/local/lib)
+FIND_LIBRARY(VORBIS_LIBRARY NAMES vorbis PATH /usr/lib /usr/local/lib)
 FIND_LIBRARY(VORBISENC_LIBRARY NAMES vorbisenc PATH /usr/lib /usr/local/lib)
 
-IF (VORBISENC_INCLUDE_DIR AND VORBISENC_LIBRARY)
+IF (VORBISENC_INCLUDE_DIR AND OGG_LIBRARY AND VORBIS_LIBRARY AND VORBISENC_LIBRARY)
 	SET(VORBISENC_FOUND TRUE)
-ENDIF (VORBISENC_INCLUDE_DIR AND VORBISENC_LIBRARY)
+ENDIF (VORBISENC_INCLUDE_DIR AND OGG_LIBRARY AND VORBIS_LIBRARY AND VORBISENC_LIBRARY)
 
 IF (VORBISENC_FOUND)
 	IF (NOT vorbisenc_FIND_QUIETLY)
-		MESSAGE(STATUS "Found vorbisenc: ${VORBISENC_INCLUDE_DIR}/vorbis/vorbisenc.h ${VORBISENC_LIBRARY}")
+		MESSAGE(STATUS "Found vorbisenc: ${VORBISENC_INCLUDE_DIR}/vorbis/vorbisenc.h ${OGG_LIBRARY} ${VORBIS_LIBRARY} ${VORBISENC_LIBRARY}")
 	ENDIF (NOT vorbisenc_FIND_QUIETLY)
 ELSE (VORBISENC_FOUND)
 	IF (vorbisenc_FIND_REQUIRED)
-		MESSAGE(FATAL_ERROR "Could not find vorbisenc")
+		MESSAGE(FATAL_ERROR "Could not find ogg, vorbis or vorbisenc")
 	ENDIF (vorbisenc_FIND_REQUIRED)
 ENDIF (VORBISENC_FOUND)
 
-- 
1.8.4

