--- a/Makefile
+++ b/Makefile
@@ -69,7 +69,7 @@ build: Makefile
 	$(Q)echo '#ifndef KLYSTRON_VERSION_H' > ./src/version.h
 	$(Q)echo '#define KLYSTRON_VERSION_H' >> ./src/version.h
 	$(Q)echo '#define KLYSTRON_REVISION "' | tr -d '\n' >> ./src/version.h
-	$(Q)date +"%Y%m%d" | tr -d '\n' >> ./src/version.h
+	$(Q)git show --no-patch --format=%cd --date=format:%Y%m%d | tr -d '\n' >> ./src/version.h
 	$(Q)echo '"' >> ./src/version.h
 	$(Q)echo '#define KLYSTRON_VERSION_STRING "klystron " KLYSTRON_REVISION' >> ./src/version.h
 	$(Q)echo '#endif' >> ./src/version.h
