diff -Naur gnome-applets-3.5.92.orig/configure.in gnome-applets-3.5.92/configure.in
--- gnome-applets-3.5.92.orig/configure.in	2012-09-20 03:07:34.000000000 +0200
+++ gnome-applets-3.5.92/configure.in	2013-04-09 09:48:48.427334056 +0200
@@ -4,7 +4,7 @@
 AC_INIT(gnome-applets, 3.5.92)
 AC_PREREQ(2.59)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE([1.11 no-dist-gzip dist-xz tar-ustar])
 AC_CONFIG_MACRO_DIR(m4)
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
