From fe7e338e20f5f533d460f9151eba20250857c350 Mon Sep 17 00:00:00 2001
From: Sebastien GODARD <sysstat@users.noreply.github.com>
Date: Mon, 24 Aug 2020 10:53:24 +0200
Subject: [PATCH] Fix #272: --disable-compress-manpg option does not work

Signed-off-by: Sebastien GODARD <sysstat@users.noreply.github.com>
---
 Makefile.in  | 2 +-
 configure    | 9 ++++-----
 configure.in | 5 ++---
 3 files changed, 7 insertions(+), 9 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 4741ae4b..096e7eaf 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -383,7 +383,7 @@ ifeq ($(INSTALL_DOC),y)
 	$(INSTALL_DATA) $(MANGRPARG) man/pidstat.1 $(DESTDIR)$(MAN1_DIR)
 	rm -f $(DESTDIR)$(MAN1_DIR)/cifsiostat.1*
 	$(INSTALL_DATA) $(MANGRPARG) man/cifsiostat.1 $(DESTDIR)$(MAN1_DIR)
-ifeq ($(COMPRESS_MANPG),n)
+ifeq ($(COMPRESS_MANPG),y)
 	$(ZIP) $(DESTDIR)$(MAN8_DIR)/sa1.8
 	$(ZIP) $(DESTDIR)$(MAN8_DIR)/sa2.8
 	$(ZIP) $(DESTDIR)$(MAN8_DIR)/sadc.8
diff --git a/configure b/configure
index 51ceee60..c0e6b6b7 100755
--- a/configure
+++ b/configure
@@ -5346,18 +5346,17 @@ $as_echo "$AUX_IMG" >&6; }
 
 
 # Compress manual pages?
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether manual pages should not be compressed" >&5
-$as_echo_n "checking whether manual pages should not be compressed... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether manual pages should be compressed" >&5
+$as_echo_n "checking whether manual pages should be compressed... " >&6; }
 # Check whether --enable-compress-manpg was given.
 if test "${enable_compress_manpg+set}" = set; then :
-  enableval=$enable_compress_manpg; AUX_MPG=$enableval
+  enableval=$enable_compress_manpg; AUX_MPG=no
 else
-  AUX_MPG=no
+  AUX_MPG=yes
 fi
 
 if test $AUX_MPG != "yes"; then
    COMPRESS_MANPG=n
-   AUX_MPG=no
 else
    COMPRESS_MANPG=y
 fi
diff --git a/configure.in b/configure.in
index b3d9b11d..b6ab705f 100644
--- a/configure.in
+++ b/configure.in
@@ -492,14 +492,13 @@ AC_MSG_RESULT($AUX_IMG)
 AC_SUBST(IGNORE_FILE_ATTRIBUTES)
 
 # Compress manual pages?
-AC_MSG_CHECKING(whether manual pages should not be compressed)
+AC_MSG_CHECKING(whether manual pages should be compressed)
 AC_ARG_ENABLE(compress-manpg,
 	      AC_HELP_STRING([--disable-compress-manpg],
 			     [do not compress sysstat manual pages]),
-			     AUX_MPG=$enableval,AUX_MPG=no)
+			     AUX_MPG=no,AUX_MPG=yes)
 if test $AUX_MPG != "yes"; then
    COMPRESS_MANPG=n
-   AUX_MPG=no
 else
    COMPRESS_MANPG=y
 fi
