--- vamp-aubio-plugins-0.5.1/wscript.0	2017-08-29 18:28:53.262861416 +0000
+++ vamp-aubio-plugins-0.5.1/wscript	2017-08-29 18:29:05.932890054 +0000
@@ -45,7 +45,7 @@
             local_vamp_lib = local_vamp_lib_mingw
         elif platform.machine() == 'x86_64':
             local_vamp_lib = local_vamp_lib_amd64
-        elif platform.machine() == 'x86_64':
+        elif platform.machine() == 'i686':
             local_vamp_lib = local_vamp_lib_i686
     elif sys.platform == 'darwin':
         local_vamp_lib = local_vamp_lib_osx
