--- a/include/conntrackd.h	2012-05-26 15:29:15.630027223 +0200
+++ b/include/conntrackd.h	2012-05-27 02:26:59.426811589 +0200
@@ -50,7 +50,7 @@
 #define EXP_DUMP_INT_XML	47	/* dump internal cache in XML	*/
 #define EXP_DUMP_EXT_XML	48	/* dump external cache in XML	*/
 
-#define DEFAULT_CONFIGFILE	"/etc/conntrackd/conntrackd.conf"
+#define DEFAULT_CONFIGFILE	"/etc/conntrackd.conf"
-#define DEFAULT_LOCKFILE	"/var/lock/conntrackd.lock"
+#define DEFAULT_LOCKFILE	"/run/lock/conntrackd.lock"
 #define DEFAULT_LOGFILE		"/var/log/conntrackd.log"
 #define DEFAULT_STATS_LOGFILE	"/var/log/conntrackd-stats.log"
