From 72acb76a3ba65ed982f13ac12297e8d1411f7e43 Mon Sep 17 00:00:00 2001
From: lxz <justforlxz@gmail.com>
Date: Wed, 25 Nov 2020 20:23:17 +0800
Subject: [PATCH] fix: unload kwin blur

disable buildin blur effect

Log:
Change-Id: I98c8b2fc654a8fbc93e958a807928e492538120c
---

diff --git a/plugins/kwineffects/blur/blur.cpp b/plugins/kwineffects/blur/blur.cpp
index 9a92921..6f5040c 100644
--- a/plugins/kwineffects/blur/blur.cpp
+++ b/plugins/kwineffects/blur/blur.cpp
@@ -48,17 +48,6 @@
 
 BlurEffect::BlurEffect(QObject *, const QVariantList &)
 {
-    // 禁用kwin自己的模糊特效，防止两个特效之间出现冲突
-    {
-        bool blur_loaded = false;
-        QMetaObject::invokeMethod(effects, "isEffectLoaded", Qt::DirectConnection, Q_RETURN_ARG(bool, blur_loaded), Q_ARG(QString, "blur"));
-
-        if (blur_loaded) {
-            // qWarning() << "will try unload builtIn blur effect of kwin";
-            QMetaObject::invokeMethod(effects, "unloadEffect", Qt::DirectConnection, Q_ARG(QString, "blur"));
-        }
-    }
-
     m_shader = new BlurShader(this);
 
     initBlurStrengthValues();
diff --git a/plugins/platforms/plugin/libkwinpreload.cpp b/plugins/platforms/plugin/libkwinpreload.cpp
index 4f5f89b..8d23fa2 100644
--- a/plugins/platforms/plugin/libkwinpreload.cpp
+++ b/plugins/platforms/plugin/libkwinpreload.cpp
@@ -36,6 +36,7 @@
 #include <QStyleFactory>
 #include <QStyle>
 #include <QTimer>
+#include <QLibrary>
 
 // deepin dbus menu
 #define MenuDBusService "com.deepin.menu"
@@ -275,6 +276,18 @@
     }
 }
 
+namespace BuiltInEffects {
+bool supported(BuiltInEffect effect) {
+    if (effect == BuiltInEffect::Blur) {
+        return false;
+    }
+
+    typedef bool (*ClientBuiltInEffect)(KWin::BuiltInEffect);
+    static ClientBuiltInEffect clientBuildInEffect = (ClientBuiltInEffect)QLibrary::resolve("kwin.so", qApp->applicationVersion(), "_ZN4KWin14BuiltInEffects9supportedENS_13BuiltInEffectE");
+    return clientBuildInEffect ? clientBuildInEffect(effect) : false;
+}
+}
+
 #endif // USE_DBUS_MENU
 } // namespace KWin
 
diff --git a/plugins/platforms/plugin/libkwinpreload.h b/plugins/platforms/plugin/libkwinpreload.h
index 03cd475..9b60ba6 100644
--- a/plugins/platforms/plugin/libkwinpreload.h
+++ b/plugins/platforms/plugin/libkwinpreload.h
@@ -62,6 +62,15 @@
 };
 
 #endif // USE_DBUS_MENU
+
+enum class BuiltInEffect {
+    InValid,
+    Blur,
+};
+
+namespace BuiltInEffects {
+    bool supported(BuiltInEffect effect);
+}
 }
 
 #if !defined(KWIN_VERSION) || KWIN_VERSION <= KWIN_VERSION_CHECK(5, 8, 6, 0)
