From c3dcab8b237d42961c0dc12ab2105f3df9073116 Mon Sep 17 00:00:00 2001
From: xhanulik <vhanulik@redhat.com>
Date: Mon, 25 Oct 2021 15:47:44 +0200
Subject: [PATCH] pkcs11-tool: load legacy provider for RIPEMD160 in test

---
 src/tools/pkcs11-tool.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/src/tools/pkcs11-tool.c b/src/tools/pkcs11-tool.c
index 66f1acaf4e..53e7c7d75f 100644
--- a/src/tools/pkcs11-tool.c
+++ b/src/tools/pkcs11-tool.c
@@ -52,6 +52,7 @@
 #include <openssl/asn1t.h>
 #include <openssl/rsa.h>
 #include <openssl/pem.h>
+#include <openssl/provider.h>
 #if !defined(OPENSSL_NO_EC) && !defined(OPENSSL_NO_ECDSA)
 #include <openssl/ec.h>
 #include <openssl/ecdsa.h>
@@ -71,6 +72,10 @@
 #include "util.h"
 #include "libopensc/sc-ossl-compat.h"
 
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+OSSL_PROVIDER *legacy_provider = NULL;
+#endif
+
 #ifdef _WIN32
 #ifndef STDOUT_FILENO
 #define STDOUT_FILENO 1
@@ -5766,7 +5771,6 @@ static int test_digest(CK_SESSION_HANDLE session)
 	CK_ULONG        hashLen1, hashLen2;
 	CK_MECHANISM_TYPE firstMechType;
 	CK_SESSION_INFO sessionInfo;
-
 	CK_MECHANISM_TYPE mechTypes[] = {
 		CKM_MD5,
 		CKM_RIPEMD160,
@@ -5868,6 +5872,11 @@ static int test_digest(CK_SESSION_HANDLE session)
 #endif
 	for (; mechTypes[i] != 0xffffff; i++) {
 		ck_mech.mechanism = mechTypes[i];
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+		if (!legacy_provider) {
+			legacy_provider = OSSL_PROVIDER_load(NULL, "legacy");
+		}
+#endif
 
 		rv = p11->C_DigestInit(session, &ck_mech);
 		if (rv == CKR_MECHANISM_INVALID)
@@ -6065,6 +6074,11 @@ static int sign_verify_openssl(CK_SESSION_HANDLE session,
 		EVP_sha256(),
 	};
 #endif
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+	if (!legacy_provider) {
+		legacy_provider = OSSL_PROVIDER_load(NULL, "legacy");
+	}
+#endif
 
 	rv = p11->C_SignInit(session, ck_mech, privKeyObject);
 	/* mechanism not implemented, don't test */
