--- configure.in	2013-09-20 16:23:41.000000000 +0200
+++ /home/mschmoelzer/Projects/newlib/libgloss/arm/configure.in	2014-01-11 00:04:01.000000000 +0100
@@ -59,7 +59,7 @@
 
 AC_SUBST(objtype)
 
-host_makefile_frag=${srcdir}/../config/default.mh
+host_makefile_frag=`cd $srcdir/../config;pwd`/default.mh
 
 dnl We have to assign the same value to other variables because autoconf
 dnl doesn't provide a mechanism to substitute a replacement keyword with
