From 74cc6eb7881be7b9cb5d9fdd90e6d7e30d4639fb Mon Sep 17 00:00:00 2001
From: _sam <3804518+aahung@users.noreply.github.com>
Date: Wed, 16 Nov 2022 10:53:02 -0800
Subject: [PATCH] chore: Prevent bin/ from installed as a package (#2597)

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 7c4fcd876..f99184d19 100755
--- a/setup.py
+++ b/setup.py
@@ -59,7 +59,7 @@ def read_requirements(req="base.txt"):
     license="Apache License 2.0",
     # Exclude all but the code folders
     packages=find_packages(
-        exclude=("tests", "tests.*", "integration", "integration.*", "docs", "examples", "versions")
+        exclude=("bin", "bin.*", "tests", "tests.*", "integration", "integration.*", "docs", "examples", "versions")
     ),
     license_files=(
         "LICENSE",
