diff -baur libffado-2.4.0.orig/admin/dbus.py libffado-2.4.0/admin/dbus.py
--- libffado-2.4.0.orig/admin/dbus.py	2010-09-10 23:50:58.000000000 +0200
+++ libffado-2.4.0/admin/dbus.py	2017-12-23 19:53:45.298791443 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 # Copyright (C) 2007-2008 Pieter Palmers
diff -baur libffado-2.4.0.orig/admin/doxygen.py libffado-2.4.0/admin/doxygen.py
--- libffado-2.4.0.orig/admin/doxygen.py	2017-11-06 10:54:30.000000000 +0100
+++ libffado-2.4.0/admin/doxygen.py	2017-12-23 19:53:55.772030266 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 #
diff -baur libffado-2.4.0.orig/admin/pkgconfig.py libffado-2.4.0/admin/pkgconfig.py
--- libffado-2.4.0.orig/admin/pkgconfig.py	2010-09-12 22:15:27.000000000 +0200
+++ libffado-2.4.0/admin/pkgconfig.py	2017-12-23 19:54:07.335259008 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 #
diff -baur libffado-2.4.0.orig/admin/pyuic4.py libffado-2.4.0/admin/pyuic4.py
--- libffado-2.4.0.orig/admin/pyuic4.py	2008-09-23 12:42:04.000000000 +0200
+++ libffado-2.4.0/admin/pyuic4.py	2017-12-23 19:55:20.601257321 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 #
@@ -31,7 +31,7 @@
 	return "building '%s' from '%s'" % ( str(target[0]), str( source[0] ) )
 
 def PyQt4Check( context ):
-	context.Message( "Checking for pyuic4 (by checking for the python module pyqtconfig) " )
+	context.Message( "Checking for pyuic4 (by checking for the python2 module pyqtconfig) " )
 	ret = True
 	try:
 		imp.find_module( "pyqtconfig" )
diff -baur libffado-2.4.0.orig/admin/pyuic5.py libffado-2.4.0/admin/pyuic5.py
--- libffado-2.4.0.orig/admin/pyuic5.py	2017-06-03 13:17:13.000000000 +0200
+++ libffado-2.4.0/admin/pyuic5.py	2017-12-23 19:55:34.657794715 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 # Copyright (C) 2017 Jonathan Woithe
@@ -32,7 +32,7 @@
 	return "building '%s' from '%s'" % ( str(target[0]), str( source[0] ) )
 
 def PyQt5Check( context ):
-	context.Message( "Checking for pyuic5 (by checking for the python module pyqtconfig) " )
+	context.Message( "Checking for pyuic5 (by checking for the python2 module pyqtconfig) " )
 	ret = True
 	try:
 		imp.find_module( "pyqtconfig" )
diff -baur libffado-2.4.0.orig/admin/pyuic.py libffado-2.4.0/admin/pyuic.py
--- libffado-2.4.0.orig/admin/pyuic.py	2008-01-20 11:29:17.000000000 +0100
+++ libffado-2.4.0/admin/pyuic.py	2017-12-23 19:55:07.654709433 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 #
@@ -31,7 +31,7 @@
 	return "building '%s' from '%s'" % ( str(target[0]), str( source[0] ) )
 
 def PyQtCheck( context ):
-	context.Message( "Checking for pyuic (by checking for the python module pyqtconfig) " )
+	context.Message( "Checking for pyuic (by checking for the python2 module pyqtconfig) " )
 	ret = True
 	try:
 		imp.find_module( "pyqtconfig" )
diff -baur libffado-2.4.0.orig/admin/scanreplace.py libffado-2.4.0/admin/scanreplace.py
--- libffado-2.4.0.orig/admin/scanreplace.py	2008-01-20 11:29:17.000000000 +0100
+++ libffado-2.4.0/admin/scanreplace.py	2017-12-23 19:55:45.581027372 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 #
diff -baur libffado-2.4.0.orig/doc/SConscript libffado-2.4.0/doc/SConscript
--- libffado-2.4.0.orig/doc/SConscript	2008-05-18 21:13:16.000000000 +0200
+++ libffado-2.4.0/doc/SConscript	2017-12-23 20:06:55.697237554 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 
 Import( 'env' )
 
diff -baur libffado-2.4.0.orig/README libffado-2.4.0/README
--- libffado-2.4.0.orig/README	2017-12-23 11:42:59.000000000 +0100
+++ libffado-2.4.0/README	2017-12-23 19:57:00.253624504 +0100
@@ -301,7 +301,7 @@
 It is also possible to run it from the source tree:
 
     $ cd support/tools
-    $ python ffado-diag.py > ffado-diag.log
+    $ python2 ffado-diag.py > ffado-diag.log
 
 It will check your system for basic problems and gather some information
 regarding your hardware configuration. This will allow us to diagnose
diff -baur libffado-2.4.0.orig/SConstruct libffado-2.4.0/SConstruct
--- libffado-2.4.0.orig/SConstruct	2017-12-23 11:43:57.000000000 +0100
+++ libffado-2.4.0/SConstruct	2017-12-23 20:08:33.212882126 +0100
@@ -50,7 +50,7 @@
     PathVariable( "INCLUDEDIR", "Overwrite the directory where headers are installed to.", "$PREFIX/include", PathVariable.PathAccept ),
     PathVariable( "SHAREDIR", "Overwrite the directory where misc shared files are installed to.", "$PREFIX/share/libffado", PathVariable.PathAccept ),
     PathVariable( "MANDIR", "Overwrite the directory where manpages are installed", "$PREFIX/man", PathVariable.PathAccept ),
-    PathVariable( "PYPKGDIR", "The directory where the python modules get installed.",
+    PathVariable( "PYPKGDIR", "The directory where the python2 modules get installed.",
         distutils.sysconfig.get_python_lib( prefix="$PREFIX" ), PathVariable.PathAccept ),
     PathVariable( "UDEVDIR", "Overwrite the directory where udev rules are installed to.", "/lib/udev/rules.d/", PathVariable.PathAccept ),
     BoolVariable( "ENABLE_BEBOB", "Enable/Disable support for the BeBoB platform.", True ),
@@ -163,8 +163,8 @@
     return ret[0]
 
 def CheckForPyModule( context, module ):
-    context.Message( "Checking for the python module '" + module + "' " )
-    ret = context.TryAction( "python $SOURCE", "import %s" % module, ".py" )
+    context.Message( "Checking for the python2 module '" + module + "' " )
+    ret = context.TryAction( "python2 $SOURCE", "import %s" % module, ".py" )
     context.Result( ret[0] )
     return ret[0]
 
@@ -395,9 +395,9 @@
 
 # PyQT checks
 if env['BUILD_MIXER'] != 'false':
-    have_dbus = (conf.CheckForApp( 'which pyuic4' ) and conf.CheckForPyModule( 'dbus.mainloop.qt' ))
-    have_pyqt4 = (conf.CheckForApp( 'which pyuic4' ) and conf.CheckForPyModule( 'PyQt4' ))
-    have_pyqt5 = (conf.CheckForApp( 'which pyuic5' ) and conf.CheckForPyModule( 'PyQt5' ))
+    have_dbus = (conf.CheckForApp( 'which python2-pyuic4' ) and conf.CheckForPyModule( 'dbus.mainloop.qt' ))
+    have_pyqt4 = (conf.CheckForApp( 'which python2-pyuic4' ) and conf.CheckForPyModule( 'PyQt4' ))
+    have_pyqt5 = (conf.CheckForApp( 'which python2-pyuic5' ) and conf.CheckForPyModule( 'PyQt5' ))
     if ((have_pyqt4 or have_pyqt5) and have_dbus):
         env['BUILD_MIXER'] = 'true'
     elif not env.GetOption('clean'):
diff -baur libffado-2.4.0.orig/support/dbus/SConscript libffado-2.4.0/support/dbus/SConscript
--- libffado-2.4.0.orig/support/dbus/SConscript	2017-11-08 11:45:42.000000000 +0100
+++ libffado-2.4.0/support/dbus/SConscript	2017-12-23 19:57:34.466597349 +0100
@@ -1,4 +1,4 @@
-#!/bin/env python
+#!/bin/env python2
 #
 # Copyright (C) 2007,2008,2011 Arnold Krille
 # Copyright (C) 2007-2008 Pieter Palmers
diff -baur libffado-2.4.0.orig/support/mixer-qt4/ffado/ffadowindow.py libffado-2.4.0/support/mixer-qt4/ffado/ffadowindow.py
--- libffado-2.4.0.orig/support/mixer-qt4/ffado/ffadowindow.py	2017-06-04 09:23:13.000000000 +0200
+++ libffado-2.4.0/support/mixer-qt4/ffado/ffadowindow.py	2017-12-23 20:03:44.372640789 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2005-2008 by Pieter Palmers
 #               2007-2009 by Arnold Krille
diff -baur libffado-2.4.0.orig/support/mixer-qt4/ffado-mixer.in libffado-2.4.0/support/mixer-qt4/ffado-mixer.in
--- libffado-2.4.0.orig/support/mixer-qt4/ffado-mixer.in	2012-04-19 01:16:02.000000000 +0200
+++ libffado-2.4.0/support/mixer-qt4/ffado-mixer.in	2017-12-23 20:00:57.884474310 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2005-2008 by Pieter Palmers
 #               2007-2009 by Arnold Krille
diff -baur libffado-2.4.0.orig/support/mixer-qt4/ffado-mixer-profiler.in libffado-2.4.0/support/mixer-qt4/ffado-mixer-profiler.in
--- libffado-2.4.0.orig/support/mixer-qt4/ffado-mixer-profiler.in	2012-04-19 01:16:02.000000000 +0200
+++ libffado-2.4.0/support/mixer-qt4/ffado-mixer-profiler.in	2017-12-23 20:02:11.477004530 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2005-2009 by Pieter Palmers
 #               2007-2009 by Arnold Krille
diff -baur libffado-2.4.0.orig/support/mixer-qt4/SConscript libffado-2.4.0/support/mixer-qt4/SConscript
--- libffado-2.4.0.orig/support/mixer-qt4/SConscript	2015-04-12 13:18:15.000000000 +0200
+++ libffado-2.4.0/support/mixer-qt4/SConscript	2017-12-23 20:01:21.157566614 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2009 Arnold Krille
 # Copyright (C) 2007-2008 Pieter Palmers
diff -baur libffado-2.4.0.orig/support/tools/ffado-diag.in libffado-2.4.0/support/tools/ffado-diag.in
--- libffado-2.4.0.orig/support/tools/ffado-diag.in	2017-06-03 13:17:13.000000000 +0200
+++ libffado-2.4.0/support/tools/ffado-diag.in	2017-12-23 19:58:17.006150642 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 
 #
@@ -120,8 +120,8 @@
     print " Prerequisites (dynamic at run-time)..."
     print "   gcc ............... %s" % get_version_first_line('gcc --version')
     print "   g++ ............... %s" % get_version_first_line('g++ --version')
-    print "   PyQt4 (by pyuic4) . %s" % get_version_first_line('pyuic4 --version')
-    print "   PyQt5 (by pyuic5) . %s" % get_version_first_line('pyuic5 --version')
+    print "   PyQt4 (by pyuic4) . %s" % get_version_first_line('python2-pyuic4 --version')
+    print "   PyQt5 (by pyuic5) . %s" % get_version_first_line('python2-pyuic5 --version')
     print "   jackd ............. %s" % get_version_first_line('jackd --version')
     print "     path ............ %s" % get_command_path('jackd')
     print "     flags ........... %s" % get_package_flags("jack")
Only in libffado-2.4.0/support/tools: ffado-diag.in.orig
diff -baur libffado-2.4.0.orig/support/tools/ffado-diag-static libffado-2.4.0/support/tools/ffado-diag-static
--- libffado-2.4.0.orig/support/tools/ffado-diag-static	2017-06-03 13:17:13.000000000 +0200
+++ libffado-2.4.0/support/tools/ffado-diag-static	2017-12-23 19:27:53.417959063 +0100
@@ -74,8 +74,8 @@
     # check libraries
     print "   gcc ............... %s" % get_version_first_line('gcc --version')
     print "   g++ ............... %s" % get_version_first_line('g++ --version')
-    print "   PyQt4 (by pyuic4) . %s" % get_version_first_line('pyuic4 --version')
-    print "   PyQt5 (by pyuic5) . %s" % get_version_first_line('pyuic5 --version')
+    print "   PyQt4 (by pyuic4) . %s" % get_version_first_line('python2-pyuic4 --version')
+    print "   PyQt5 (by pyuic5) . %s" % get_version_first_line('python2-pyuic5 --version')
     print "   jackd ............. %s" % get_version_first_line('jackd --version')
     print "     path ............ %s" % get_command_path('jackd')
     print "     flags ........... %s" % get_package_flags("jack")
Only in libffado-2.4.0/support/tools: ffado-diag-static.orig
diff -baur libffado-2.4.0.orig/support/tools/ffado-sandbox-install.py libffado-2.4.0/support/tools/ffado-sandbox-install.py
--- libffado-2.4.0.orig/support/tools/ffado-sandbox-install.py	2017-03-27 11:52:30.000000000 +0200
+++ libffado-2.4.0/support/tools/ffado-sandbox-install.py	2017-12-23 20:00:16.424904540 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 
 #
diff -baur libffado-2.4.0.orig/support/tools/listirqinfo.py libffado-2.4.0/support/tools/listirqinfo.py
--- libffado-2.4.0.orig/support/tools/listirqinfo.py	2017-03-27 12:39:34.000000000 +0200
+++ libffado-2.4.0/support/tools/listirqinfo.py	2017-12-23 19:59:20.962148592 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 
 #
diff -baur libffado-2.4.0.orig/support/tools/SConscript libffado-2.4.0/support/tools/SConscript
--- libffado-2.4.0.orig/support/tools/SConscript	2017-11-08 11:45:42.000000000 +0100
+++ libffado-2.4.0/support/tools/SConscript	2017-12-23 19:59:56.855108017 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2007-2008 Arnold Krille
 # Copyright (C) 2007-2008 Pieter Palmers
@@ -42,11 +42,11 @@
 
 #
 # For the the ffado-diag tools
-e['PYTHONDIR'] = Template( os.path.join( e['SHAREDIR'], 'python' ) ).safe_substitute( e )
+e['PYTHONDIR'] = Template( os.path.join( e['SHAREDIR'], 'python2' ) ).safe_substitute( e )
 # For the installation of the stuff
-e['pythondir'] = Template( os.path.join( e['sharedir'], 'python' ) ).safe_substitute( e )
+e['pythondir'] = Template( os.path.join( e['sharedir'], 'python2' ) ).safe_substitute( e )
 
-e.Command( "static_info.txt", "#/SConstruct", "python support/tools/ffado-diag-static > $TARGET" )
+e.Command( "static_info.txt", "#/SConstruct", "python2 support/tools/ffado-diag-static > $TARGET" )
 
 e.ScanReplace( "ffado-diag.in" )
 
diff -baur libffado-2.4.0.orig/tests/dbus_test.py libffado-2.4.0/tests/dbus_test.py
--- libffado-2.4.0.orig/tests/dbus_test.py	2017-03-27 11:52:30.000000000 +0200
+++ libffado-2.4.0/tests/dbus_test.py	2017-12-23 20:09:02.045914530 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2005-2007 by Pieter Palmers
 #               2007-2008 by Arnold Krille
diff -baur libffado-2.4.0.orig/tests/python/test-eap-ctrl.py libffado-2.4.0/tests/python/test-eap-ctrl.py
--- libffado-2.4.0.orig/tests/python/test-eap-ctrl.py	2017-03-27 11:52:30.000000000 +0200
+++ libffado-2.4.0/tests/python/test-eap-ctrl.py	2017-12-23 20:09:54.382036296 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright (C) 2005-2009 by Pieter Palmers
 #
diff -baur libffado-2.4.0.orig/tests/python/test-echo-digital.py libffado-2.4.0/tests/python/test-echo-digital.py
--- libffado-2.4.0.orig/tests/python/test-echo-digital.py	2017-03-27 11:52:30.000000000 +0200
+++ libffado-2.4.0/tests/python/test-echo-digital.py	2017-12-23 20:10:10.005207278 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 '''
 This test is for checking a function to switch digital interface on devices
diff -baur libffado-2.4.0.orig/tests/python/test-echo-routing.py libffado-2.4.0/tests/python/test-echo-routing.py
--- libffado-2.4.0.orig/tests/python/test-echo-routing.py	2017-03-27 11:52:30.000000000 +0200
+++ libffado-2.4.0/tests/python/test-echo-routing.py	2017-12-23 20:10:41.498213755 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 '''
 This test is for checking playback-routing of devices based on Echo Fireworks.
