diff -u -r libevent-2.1.12-stable/CMakeLists.txt libevent-2.1.12-stable-lib32/CMakeLists.txt
--- libevent-2.1.12-stable/CMakeLists.txt	2020-07-05 12:01:34.000000000 +0000
+++ libevent-2.1.12-stable-lib32/CMakeLists.txt	2022-08-03 19:36:07.559379519 +0000
@@ -162,10 +162,10 @@
     set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 endif()
 if (NOT DEFINED CMAKE_LIBRARY_OUTPUT_DIRECTORY)
-    set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+    set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib32)
 endif()
 if (NOT DEFINED CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
-    set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+    set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib32)
 endif()
 
 if (EVENT__ENABLE_VERBOSE_DEBUG)
@@ -1453,7 +1453,7 @@
 #
 
 set(EVENT_INSTALL_CMAKE_DIR
-    "${CMAKE_INSTALL_PREFIX}/lib/cmake/libevent")
+    "${CMAKE_INSTALL_PREFIX}/lib32/cmake/libevent")
 
 export(PACKAGE libevent)
 
diff -u -r libevent-2.1.12-stable/cmake/AddEventLibrary.cmake libevent-2.1.12-stable-lib32/cmake/AddEventLibrary.cmake
--- libevent-2.1.12-stable/cmake/AddEventLibrary.cmake	2020-07-05 12:01:34.000000000 +0000
+++ libevent-2.1.12-stable-lib32/cmake/AddEventLibrary.cmake	2022-08-03 19:49:24.253528837 +0000
@@ -13,7 +13,7 @@
 macro(generate_pkgconfig LIB_NAME)
     set(prefix      ${CMAKE_INSTALL_PREFIX})
     set(exec_prefix ${CMAKE_INSTALL_PREFIX})
-    set(libdir      ${CMAKE_INSTALL_PREFIX}/lib)
+    set(libdir      ${CMAKE_INSTALL_PREFIX}/lib32)
     set(includedir  ${CMAKE_INSTALL_PREFIX}/include)
 
     set(VERSION ${EVENT_ABI_LIBVERSION})
@@ -31,7 +31,7 @@
     configure_file("lib${LIB_NAME}.pc.in" "lib${LIB_NAME}.pc" @ONLY)
     install(
         FILES "${CMAKE_CURRENT_BINARY_DIR}/lib${LIB_NAME}.pc"
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/lib32/pkgconfig"
     )
 endmacro()
 
@@ -40,9 +40,9 @@
 macro(export_install_target TYPE LIB_NAME OUTER_INCLUDES)
     if("${LIB_NAME}" STREQUAL "event")
         install(TARGETS "${LIB_NAME}_${TYPE}"
-            LIBRARY DESTINATION "lib" COMPONENT lib
-            ARCHIVE DESTINATION "lib" COMPONENT lib
-            RUNTIME DESTINATION "lib" COMPONENT lib
+            LIBRARY DESTINATION "lib32" COMPONENT lib
+            ARCHIVE DESTINATION "lib32" COMPONENT lib
+            RUNTIME DESTINATION "lib32" COMPONENT lib
             COMPONENT dev
         )
     else()
@@ -67,9 +67,9 @@
         )
         install(TARGETS "${LIB_NAME}_${TYPE}"
             EXPORT LibeventTargets-${TYPE}
-            LIBRARY DESTINATION "lib" COMPONENT lib
-            ARCHIVE DESTINATION "lib" COMPONENT lib
-            RUNTIME DESTINATION "lib" COMPONENT lib
+            LIBRARY DESTINATION "lib32" COMPONENT lib
+            ARCHIVE DESTINATION "lib32" COMPONENT lib
+            RUNTIME DESTINATION "lib32" COMPONENT lib
             COMPONENT dev
         )
     endif()
@@ -150,7 +150,7 @@
             set_target_properties(
                 "${LIB_NAME}_shared" PROPERTIES
                 OUTPUT_NAME "${LIB_NAME}-${EVENT_PACKAGE_RELEASE}.${CURRENT_MINUS_AGE}"
-                INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib"
+                INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib32"
                 LINK_FLAGS "-compatibility_version ${COMPATIBILITY_VERSION} -current_version ${COMPATIBILITY_VERSION}.${EVENT_ABI_LIBVERSION_REVISION}")
         else()
             math(EXPR CURRENT_MINUS_AGE "${EVENT_ABI_LIBVERSION_CURRENT}-${EVENT_ABI_LIBVERSION_AGE}")
@@ -159,7 +159,7 @@
                 OUTPUT_NAME "${LIB_NAME}-${EVENT_PACKAGE_RELEASE}"
                 VERSION "${CURRENT_MINUS_AGE}.${EVENT_ABI_LIBVERSION_AGE}.${EVENT_ABI_LIBVERSION_REVISION}"
                 SOVERSION "${CURRENT_MINUS_AGE}"
-                INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+                INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib32")
         endif()
 
         if (NOT WIN32)
@@ -181,7 +181,7 @@
         if (NOT WIN32)
             install(FILES
                 "$<TARGET_FILE_DIR:${LIB_NAME}_shared>/${LIB_LINK_NAME}"
-                DESTINATION "lib"
+                DESTINATION "lib32"
                 COMPONENT lib)
         endif()
     endif()
diff -u -r libevent-2.1.12-stable/cmake/LibeventConfig.cmake.in libevent-2.1.12-stable-lib32/cmake/LibeventConfig.cmake.in
--- libevent-2.1.12-stable/cmake/LibeventConfig.cmake.in	2020-07-05 12:01:34.000000000 +0000
+++ libevent-2.1.12-stable-lib32/cmake/LibeventConfig.cmake.in	2022-08-03 19:35:46.016109621 +0000
@@ -130,7 +130,7 @@
         unset(_event_lib CACHE)
         find_library(_event_lib
                     NAMES "event_${_comp}"
-                    PATHS "${_INSTALL_PREFIX}/lib"
+                    PATHS "${_INSTALL_PREFIX}/lib32"
                     NO_DEFAULT_PATH)
         if(_event_lib)
             list(APPEND LIBEVENT_LIBRARIES "libevent::${_comp}")
