diff -ruN a/gui/bassup_ui.cpp b/gui/bassup_ui.cpp
--- a/gui/bassup_ui.cpp	2016-12-11 13:07:52.000000000 +0100
+++ b/gui/bassup_ui.cpp	2021-08-01 13:22:38.061730413 +0200
@@ -38,7 +38,7 @@
 #define BASSUP_GUI_URI "http://eq10q.sourceforge.net/bassup/gui"
 
 
-static LV2UI_Handle instantiateBassUp_gui(const _LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature *const *features)
+static LV2UI_Handle instantiateBassUp_gui(const LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature *const *features)
 {
   #ifdef TESTING_EQ10Q
   cout<<"instantiateEq10q_gui Entring... ";
@@ -116,4 +116,4 @@
     }
     
 
-}
\ No newline at end of file
+}
diff -ruN a/gui/dyn_ui.cpp b/gui/dyn_ui.cpp
--- a/gui/dyn_ui.cpp	2016-12-11 13:07:50.000000000 +0100
+++ b/gui/dyn_ui.cpp	2021-08-01 13:23:05.758472847 +0200
@@ -37,7 +37,7 @@
 using namespace std;
 #endif
 
-static LV2UI_Handle instantiateDyn_gui(const _LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature *const *features)
+static LV2UI_Handle instantiateDyn_gui(const LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature *const *features)
 {
   #ifdef TESTING_EQ10Q
   cout<<"instantiateDyn_gui Entring... ";
diff -ruN a/gui/eq10q_ui.cpp b/gui/eq10q_ui.cpp
--- a/gui/eq10q_ui.cpp	2016-12-11 13:07:50.000000000 +0100
+++ b/gui/eq10q_ui.cpp	2021-08-01 13:23:19.615177359 +0200
@@ -38,7 +38,7 @@
 #endif
 
 
-static LV2UI_Handle instantiateEq10q_gui(const _LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature *const *features)
+static LV2UI_Handle instantiateEq10q_gui(const LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature *const *features)
 {
   #ifdef TESTING_EQ10Q
   cout<<"instantiateEq10q_gui Entring... ";
diff -ruN a/gui/midside_ui.cpp b/gui/midside_ui.cpp
--- a/gui/midside_ui.cpp	2016-12-11 13:07:53.000000000 +0100
+++ b/gui/midside_ui.cpp	2021-08-01 13:22:16.605004934 +0200
@@ -37,7 +37,7 @@
 using namespace std;
 #endif
 
-static LV2UI_Handle instantiateMidSide_gui(const _LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature *const *features)
+static LV2UI_Handle instantiateMidSide_gui(const LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature *const *features)
 {
   #ifdef TESTING_EQ10Q
   cout<<"instantiateEq10q_gui Entring... ";
