From 5c8fc4e9a6afe0e0b5bcfa4f3e7dbb4bfe167ba1 Mon Sep 17 00:00:00 2001
From: Baptiste Jonglez <git@bitsofnetworks.org>
Date: Thu, 3 Jan 2019 00:47:39 +0100
Subject: [PATCH] build: Only run tests if catch2 is installed

---
 CMakeLists.txt        | 14 ++++++++------
 cmake/Findcatch.cmake | 11 +++--------
 2 files changed, 11 insertions(+), 14 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c618244..95889e1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,7 +42,14 @@ endif ( )
 set( CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake" )
 
 find_package( asio REQUIRED )
-find_package( catch REQUIRED )
+find_package( catch )
+if ( CATCH_FOUND )
+    enable_testing( )
+    add_subdirectory( "${PROJECT_SOURCE_DIR}/test/unit" )
+    add_subdirectory( "${PROJECT_SOURCE_DIR}/test/feature" )
+    add_subdirectory( "${PROJECT_SOURCE_DIR}/test/regression" )
+    add_subdirectory( "${PROJECT_SOURCE_DIR}/test/integration" )
+endif ( )
 if ( BUILD_SSL )
     find_package( openssl REQUIRED )
 endif ( )
@@ -77,11 +84,6 @@ else ( )
     target_link_libraries( ${SHARED_LIBRARY_NAME} )
 endif ( )
 
-enable_testing( )
-add_subdirectory( "${PROJECT_SOURCE_DIR}/test/unit" )
-add_subdirectory( "${PROJECT_SOURCE_DIR}/test/feature" )
-add_subdirectory( "${PROJECT_SOURCE_DIR}/test/regression" )
-add_subdirectory( "${PROJECT_SOURCE_DIR}/test/integration" )
 
 #
 # Install
diff --git a/cmake/Findcatch.cmake b/cmake/Findcatch.cmake
index d070243..6c5be19 100644
--- a/cmake/Findcatch.cmake
+++ b/cmake/Findcatch.cmake
@@ -1,9 +1,4 @@
-find_path( catch_INCLUDE catch.hpp HINTS "${PROJECT_SOURCE_DIR}/dependency/catch/single_include" "/usr/include/catch2/single_include" "/usr/local/include/catch2/single_include" "/opt/local/include/catch2/single_include" )
+find_path( catch_INCLUDE catch.hpp HINTS "${PROJECT_SOURCE_DIR}/dependency/catch/single_include" "/usr/include/catch2" "/usr/local/include/catch2/single_include" "/opt/local/include/catch2/single_include" )
 
-if ( catch_INCLUDE )
-    set( CATCH_FOUND TRUE )
-
-    message( STATUS "Found Catch include at: ${catch_INCLUDE}" )
-else ( )
-    message( FATAL_ERROR "Failed to locate Catch dependency." )
-endif ( )
+include(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(catch DEFAULT_MSG catch_INCLUDE)
-- 
2.20.0

