diff --git a/cmake/modules/Findcatch.cmake b/cmake/modules/Findcatch.cmake
index eff91fe..15c7f87 100644
--- a/cmake/modules/Findcatch.cmake
+++ b/cmake/modules/Findcatch.cmake
@@ -1,6 +1,6 @@
 # Copyright 2013-2017, Corvusoft Ltd, All Rights Reserved.
 
-find_path( catch_INCLUDE catch.hpp HINTS "${PROJECT_SOURCE_DIR}/dependency/catch/include" "/usr/include" "/usr/local/include" "/opt/local/include" )
+find_path( catch_INCLUDE catch.hpp HINTS "${PROJECT_SOURCE_DIR}/dependency/catch/include" "/usr/include" "/usr/include/catch" "/usr/local/include" "/opt/local/include" )
 
 if ( catch_INCLUDE )
     set( CATCH_FOUND TRUE )
