diff --git a/Makefile.am b/Makefile.am
index f35b315..6bf871f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -34,7 +34,6 @@ clean-local:
 
 EXTRA_DIST = \
 	lib/poppler-sharp/poppler-sharp/poppler-sharp.dll.config \
-	gnome-doc-utils.make \
 	icon-theme-installer \
 	intltool-extract.in \
 	intltool-merge.in \
@@ -48,7 +47,6 @@ CLEANFILES = \
 	intltool-merge
 
 DISTCLEANFILES = \
-	gnome-doc-utils.make \
 	*.bak \
 	*~ \
 	*.pidb
diff --git a/configure.ac b/configure.ac
index e3b4570..19d517d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -12,7 +12,7 @@ if test "x$PKG_CONFIG" = "xno"; then
         AC_MSG_ERROR([You need to install pkg-config])
 fi
 
-GNOME_DOC_INIT
+YELP_HELP_INIT
 
 AC_DEFUN([SHAMROCK_EXPAND_LIBDIR],
 [	
diff --git a/docs/C/index.docbook b/docs/C/index.docbook
index 6b22ce1..0a29e45 100644
--- a/docs/C/index.docbook
+++ b/docs/C/index.docbook
@@ -2,7 +2,7 @@
 <?db.chunk.max_depth 1?>
 <!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook XML 4.1.2//EN" "http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd">
 <book id="pdfmod" lang="C">
-	<bookinfo>
+	<bookinfo id="index">
 		<title>The PDF Mod Manual</title>
 		<abstract role="description">
 			<para>This is the user's manual for PDF Mod, a simple tool for manipulating PDF documents.</para>
diff --git a/docs/Makefile.am b/docs/Makefile.am
index d9fba88..66bbf4a 100644
--- a/docs/Makefile.am
+++ b/docs/Makefile.am
@@ -1,9 +1,7 @@
-include $(top_srcdir)/gnome-doc-utils.make
-dist-hook: doc-dist-hook
+@YELP_HELP_RULES@
 
-DOC_MODULE = pdfmod
-DOC_ENTITIES = 
-DOC_INCLUDES = 
-DOC_FIGURES =
+HELP_ID = pdfmod
 
-DOC_LINGUAS = ca cs da de es fr hr pl sl sv zh_CN
+HELP_FILES = index.docbook
+
+HELP_LINGUAS = ca cs da de es fr hr pl sl sv zh_CN
diff --git a/src/PdfMod/Gui/Actions.cs b/src/PdfMod/Gui/Actions.cs
index 3fc452b..db16fe6 100644
--- a/src/PdfMod/Gui/Actions.cs
+++ b/src/PdfMod/Gui/Actions.cs
@@ -485,15 +485,15 @@ namespace PdfMod.Gui
                 var langs = GLib.Marshaller.NullTermPtrToStringArray (lang_ptr, false);
 
                 string help_dir = null;
-                foreach (var dir in new string [] { Core.Defines.PREFIX + "/share/gnome/help/", "/usr/local/share/gnome/help/", "docs/" }) {
+                foreach (var dir in new string [] { Core.Defines.PREFIX + "/share/help/", "/usr/local/share/help/", "docs/" }) {
                     help_dir = dir;
-                    if (System.IO.Directory.Exists (dir + "pdfmod/")) {
+                    if (System.IO.Directory.Exists (dir + "C/pdfmod/")) {
                         break;
                     }
                 }
 
                 foreach (var lang in langs) {
-                    var help_path = CombinePaths (help_dir, "pdfmod", lang, "pdfmod.xml");
+                    var help_path = CombinePaths (help_dir, lang, "pdfmod", "index.docbook");
                     if (System.IO.File.Exists (help_path)) {
                         System.Diagnostics.Process.Start (String.Format ("ghelp://{0}", help_path));
                         shown = true;
