diff -rupN vbam.orig/src/gtk/CMakeLists.txt vbam/src/gtk/CMakeLists.txt
--- vbam.orig/src/gtk/CMakeLists.txt	2016-10-23 19:14:22.423451961 +0200
+++ vbam/src/gtk/CMakeLists.txt	2016-10-23 19:14:33.346550202 +0200
@@ -47,7 +47,7 @@ SET(SRC_GTK
     soundconfig.cpp
     screenarea.cpp
     screenarea-cairo.cpp
-    #screenarea-opengl.cpp
+    screenarea-opengl.cpp
     tools.cpp
     window.cpp
     ../sdl/inputSDL.cpp
diff -rupN vbam.orig/src/gtk/screenarea.cpp vbam/src/gtk/screenarea.cpp
--- vbam.orig/src/gtk/screenarea.cpp	2016-10-23 19:14:22.423451961 +0200
+++ vbam/src/gtk/screenarea.cpp	2016-10-23 19:14:33.346550202 +0200
@@ -49,7 +49,7 @@ ScreenArea::ScreenArea(int _iWidth, int
     pixbuf->fill(0);
 
 #if !GTK_CHECK_VERSION(3, 0, 0)
-    m_poEmptyCursor = new Gdk::Cursor(get_display, pixbuf, 0, 0);
+    m_poEmptyCursor = new Gdk::Cursor(get_display(), pixbuf, 0, 0);
 #else
     m_poEmptyCursor = Gdk::Cursor::create(get_display(), pixbuf, 0, 0);
 #endif
diff -rupN vbam.orig/src/sdl/expr.cpp vbam/src/sdl/expr.cpp
--- vbam.orig/src/sdl/expr.cpp	2016-10-23 19:14:22.423451961 +0200
+++ vbam/src/sdl/expr.cpp	2016-10-23 19:14:33.346550202 +0200
@@ -85,12 +85,10 @@ enum yytokentype {
 /* Copy the first part of user declarations.  */
 #line 1 "expr.ypp"
 
-namespace std {
 #include <memory.h>
 #include <stdio.h>
-#include <stdlib.h>
+#include <cstdlib>
 #include <string.h>
-}
 
 using namespace std;
 
diff -rupN vbam.orig/src/sdl/expr.ypp vbam/src/sdl/expr.ypp
--- vbam.orig/src/sdl/expr.ypp	2016-10-23 19:14:22.423451961 +0200
+++ vbam/src/sdl/expr.ypp	2016-10-23 19:15:20.352228711 +0200
@@ -1,10 +1,8 @@
 %{
-namespace std {  
 #include <stdio.h>
 #include <memory.h>
-#include <stdlib.h>
+#include <cstdlib>
 #include <string.h>
-}
 
 using namespace std;
  
