diff -aur curlftpfs-0.9.2.orig/ftpfs.c curlftpfs-0.9.2.new/ftpfs.c
--- curlftpfs-0.9.2.orig/ftpfs.c	2016-01-25 17:01:32.150581272 +0100
+++ curlftpfs-0.9.2.new/ftpfs.c	2016-01-25 17:02:08.566538053 +0100
@@ -1625,9 +1625,7 @@
   }
 
   if (ftpfs.no_verify_hostname) {
-    /* The default is 2 which verifies even the host string. This sets to 1
-     * which means verify the host but not the string. */
-    curl_easy_setopt_or_die(easy, CURLOPT_SSL_VERIFYHOST, 1);
+    curl_easy_setopt_or_die(easy, CURLOPT_SSL_VERIFYHOST, 0);
   }
 
   curl_easy_setopt_or_die(easy, CURLOPT_INTERFACE, ftpfs.interface);
