diff --git a/ruby/src/api.cc b/ruby/src/api.cc
index 72a6144..3f8ad5d 100644
--- a/ruby/src/api.cc
+++ b/ruby/src/api.cc
@@ -505,7 +505,7 @@ namespace leatherman { namespace ruby {
         }
 
         // Search the path for ruby.exe and query it for the location of its library.
-        string ruby = execution::which("ruby");
+        string ruby = execution::which("ruby-2.7");
         if (ruby.empty()) {
             LOG_DEBUG("ruby could not be found on the PATH.");
             return library;
