From e2659914cad36f73c4944e018498af30ee71591b Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Mon, 25 May 2015 02:21:07 +0200
Subject: [PATCH] Fix a __GNUC__ check to work with GCC 5

---
 vm/object_utils.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vm/object_utils.hpp b/vm/object_utils.hpp
index 551b3f5..66c3203 100644
--- a/vm/object_utils.hpp
+++ b/vm/object_utils.hpp
@@ -13,7 +13,7 @@
 
 // A stupid work around for g++ changing it's behavior on 4.3
 
-#if __clang__ || (__GNUC__ >= 4 && __GNUC_MINOR__ >= 3)
+#if __clang__ || __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 3)
   #define SPECIALIZATION_STORAGE
 #else
   #define SPECIALIZATION_STORAGE static inline
-- 
2.4.2

