diff --git i/gio/tests/meson.build w/gio/tests/meson.build
index 41d4e3e6c..9f8110884 100644
--- i/gio/tests/meson.build
+++ w/gio/tests/meson.build
@@ -838,6 +838,7 @@ if meson.can_run_host_binaries()
                  '-r',
                  '-b','binary',
                  '@INPUT@',
+                 '--oformat', 'elf32-i386',
                  '-o','@OUTPUT@'])
 
     # Rename symbol to match the one in the C file
diff --git i/glib/tests/meson.build w/glib/tests/meson.build
index db01b54b4..55f9c2e1f 100644
--- i/glib/tests/meson.build
+++ w/glib/tests/meson.build
@@ -126,7 +126,7 @@ glib_tests = {
   'thread-pool' : {},
   'thread-pool-slow' : {'suite' : ['slow']},
   'timeout' : {},
-  'timer' : {},
+  'timer' : {'suite' : ['flaky']},
   'tree' : {},
   'types' : {},
   'utf8-performance' : {},
