diff --git a/src/main.py b/src/main.py
index bfb7005..3ef77ee 100755
--- a/src/main.py
+++ b/src/main.py
@@ -57,7 +57,7 @@ import subprocess
 import sys
 import traceback
 import urllib
-import vte
+import deepinvte
 import shlex
 
 PROJECT_NAME = "deepin-terminal"
@@ -1324,9 +1324,9 @@ class Terminal(object):
 
         gtk.main()
 
-class TerminalWrapper(vte.Terminal):
+class TerminalWrapper(deepinvte.Terminal):
     """
-    Wrapper class for vte.Terminal. Propagate keys. Make some customize as well.
+    Wrapper class for deepinvte.Terminal. Propagate keys. Make some customize as well.
     """
 
     def __init__(self,
@@ -1340,7 +1340,7 @@ class TerminalWrapper(vte.Terminal):
         Initial values.
         :param parent_widget: which grid this widget belongs to.
         """
-        vte.Terminal.__init__(self)
+        deepinvte.Terminal.__init__(self)
         self.parent_widget = parent_widget
         self.press_q_quit = press_q_quit
         self.set_word_chars("-A-Za-z0-9,./?%&#:_")
@@ -1582,19 +1582,19 @@ class TerminalWrapper(vte.Terminal):
 
     def change_cursor_shape(self, cursor_shape):
         if cursor_shape == "block":
-            self.set_cursor_shape(vte.CURSOR_SHAPE_BLOCK)
+            self.set_cursor_shape(deepinvte.CURSOR_SHAPE_BLOCK)
         elif cursor_shape == "ibeam":
-            self.set_cursor_shape(vte.CURSOR_SHAPE_IBEAM)
+            self.set_cursor_shape(deepinvte.CURSOR_SHAPE_IBEAM)
         elif cursor_shape == "underline":
-            self.set_cursor_shape(vte.CURSOR_SHAPE_UNDERLINE)
+            self.set_cursor_shape(deepinvte.CURSOR_SHAPE_UNDERLINE)
 
     def change_cursor_blink_mode(self, cursor_blink_mode):
         if cursor_blink_mode == "system":
-            self.set_cursor_blink_mode(vte.CURSOR_BLINK_SYSTEM)
+            self.set_cursor_blink_mode(deepinvte.CURSOR_BLINK_SYSTEM)
         elif cursor_blink_mode == "on":
-            self.set_cursor_blink_mode(vte.CURSOR_BLINK_ON)
+            self.set_cursor_blink_mode(deepinvte.CURSOR_BLINK_ON)
         elif cursor_blink_mode == "off":
-            self.set_cursor_blink_mode(vte.CURSOR_BLINK_OFF)
+            self.set_cursor_blink_mode(deepinvte.CURSOR_BLINK_OFF)
 
     def change_encoding(self, encoding):
         self.set_encoding(encoding)
