commit 06c987c7564b2ee7359462587a55c11fdb3d4f84
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Sat Jul 1 17:06:02 2017 +0800

    Fix build with path 0.6.1

diff --git a/src/Network/HTTP/Download/Verified.hs b/src/Network/HTTP/Download/Verified.hs
index fad8236b..67f19f2b 100644
--- a/src/Network/HTTP/Download/Verified.hs
+++ b/src/Network/HTTP/Download/Verified.hs
@@ -55,7 +55,7 @@ import              Network.HTTP.Types.Header (hContentLength, hContentMD5)
 import              Path
 import              Prelude -- Fix AMP warning
 import              System.Directory
-import              System.FilePath ((<.>))
+import qualified    System.FilePath as FP ((<.>))
 import              System.IO
 
 -- | A request together with some checks to perform.
@@ -243,7 +243,7 @@ verifiedDownload DownloadRequest{..} destpath progressSink = do
         if p then m >> return True else return False
 
     fp = toFilePath destpath
-    fptmp = fp <.> "tmp"
+    fptmp = fp FP.<.> "tmp"
     dir = toFilePath $ parent destpath
 
     getShouldDownload = do
diff --git a/src/Stack/Fetch.hs b/src/Stack/Fetch.hs
index 1439d114..746b695d 100644
--- a/src/Stack/Fetch.hs
+++ b/src/Stack/Fetch.hs
@@ -75,7 +75,6 @@ import              Stack.Types.PackageIdentifier
 import              Stack.Types.PackageIndex
 import              Stack.Types.PackageName
 import              Stack.Types.Version
-import              System.FilePath ((<.>))
 import qualified    System.FilePath as FP
 import              System.IO
 import              System.PosixCompat (setFileMode)
@@ -561,7 +560,7 @@ fetchPackages' mdistDir toFetchAll = do
                 let cabalFP =
                         innerDest FP.</>
                         packageNameString (packageIdentifierName ident)
-                        <.> "cabal"
+                        FP.<.> "cabal"
                 S.writeFile cabalFP $ tfCabal toFetch
 
                 atomically $ modifyTVar outputVar $ Map.insert ident destDir
