--- configure.orig	2017-03-22 10:11:04.082473251 +0000
+++ configure	2017-03-22 10:14:19.083251343 +0000
@@ -10,10 +10,14 @@
 chomp($xmlcflags);
 my $xmlldflags = `xml2-config --libs`;
 chomp($xmlldflags);
+my $cryptocflags = `pkg-config libcrypto --cflags`;
+chomp($cryptocflags);
+my $cryptoldflags = `pkg-config libcrypto --libs`;
+chomp($cryptoldflags);
 
 my $prefix = "/usr/local";
-my $cflags = "-Wall -Wno-format-y2k -O2 -DLOCALEPATH=\"\\\"\$(LOCALEPATH)\\\"\" -DOS=\\\"$os\\\" $xmlcflags \$(EXTRA_CFLAGS) ";
-my $ldflags = "-lncurses -lcrypto $xmlldflags \$(EXTRA_LDFLAGS) ";
+my $cflags = "-Wall -Wno-format-y2k -O2 -DLOCALEPATH=\"\\\"\$(LOCALEPATH)\\\"\" -DOS=\\\"$os\\\" $cryptocflags $xmlcflags \$(EXTRA_CFLAGS) ";
+my $ldflags = "-lncurses $cryptoldflags $xmlldflags \$(EXTRA_LDFLAGS) ";
 
 my $use_nls = 1;
 parse_cmdl_line();
