diff -u -r WireGuard-0.0.20191212/src/compat/compat.h WireGuard-0.0.20191212-linux-5.4.5/src/compat/compat.h
--- WireGuard-0.0.20191212/src/compat/compat.h	2019-12-12 11:24:51.000000000 +0000
+++ WireGuard-0.0.20191212-linux-5.4.5/src/compat/compat.h	2019-12-18 20:40:30.831102200 +0000
@@ -861,7 +861,7 @@
 })
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 5, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 5)
 #define ipv6_dst_lookup_flow(a, b, c, d) ipv6_dst_lookup(a, b, &dst, c) + (void *)0 ?: dst
 #endif
 
