diff -r d1b44eea341b gtk/SciTEGTK.cxx
--- a/scite/gtk/SciTEGTK.cxx	Sat Jun 06 20:55:49 2020 +1000
+++ b/scite/gtk/SciTEGTK.cxx	Mon Jun 08 12:02:44 2020 +1000
@@ -5463,7 +5463,6 @@
 		executableDirectory = FilePath(selfExe).Directory();
 	}
 #endif
-	LexillaSetDefaultDirectory(executableDirectory.AsInternal());
 
 	// Get this now because gtk_init() clears it
 	const gchar *startup_id = g_getenv("DESKTOP_STARTUP_ID");
diff -r d1b44eea341b gtk/makefile
--- a/scite/gtk/makefile	Sat Jun 06 20:55:49 2020 +1000
+++ b/scite/gtk/makefile	Mon Jun 08 12:02:44 2020 +1000
@@ -45,6 +45,7 @@
 datadir=$(prefix)/share
 pixmapdir=$(datadir)/pixmaps
 bindir=$(prefix)/bin
+libdir=$(prefix)/lib/scite
 SYSCONF_PATH=$(prefix)/share/scite
 
 INSTALL=install
@@ -149,7 +150,7 @@
 	Utf8_16.o
 
 $(PROG): SciTEGTK.o GUIGTK.o Widget.o DirectorExtension.o $(SRC_OBJS) $(LUA_OBJS)
-	$(CXX) $(BASE_FLAGS) -rdynamic -Wl,--as-needed -Wl,-rpath,'$${ORIGIN}' -Wl,--version-script $(srcdir)/lua.vers $^ -o $@ $(CONFIGLIB) $(LIBS) -L ../../scintilla/bin -lscintilla $(LDLIBS)
+	$(CXX) $(BASE_FLAGS) -rdynamic -Wl,--as-needed -Wl,-rpath,'$${ORIGIN}' -Wl,--version-script $(srcdir)/lua.vers -Wl,-rpath,$(libdir) $^ -o $@ $(CONFIGLIB) $(LIBS) -L ../../scintilla/bin -lscintilla $(LDLIBS)
 
 # Automatically generate header dependencies with "make deps"
 include deps.mak
@@ -159,10 +160,11 @@
 # Dead:	install -D SciTEGTK.properties $(SYSCONF_PATH)/SciTEGlobal.properties
 install:
 	$(INSTALL) -m 755 -d $(DESTDIR)$(bindir) $(DESTDIR)$(SYSCONF_PATH)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(libdir)
 
 	$(INSTALL) -m 755 $(PROG) $(DESTDIR)$(bindir)
-	$(INSTALL) -m 755 $(COMPONENT) $(DESTDIR)$(bindir)
-	$(INSTALL) -m 755 $(LEXILLA) $(DESTDIR)$(bindir)
+	$(INSTALL) -m 755 $(COMPONENT) $(DESTDIR)$(libdir)
+	$(INSTALL) -m 755 $(LEXILLA) $(DESTDIR)$(libdir)
 
 	for files in $(srcdir)/../src/*.properties $(srcdir)/../doc/*.html $(srcdir)/../doc/SciTEIco.png $(srcdir)/../doc/PrintHi.png $(srcdir)/../doc/SciTEIndicators.png; \
 	do \
@@ -177,8 +179,8 @@
 
 uninstall:
 	rm -f $(DESTDIR)$(bindir)/SciTE
-	rm -f $(DESTDIR)$(bindir)/libscintilla.so
-	rm -f $(DESTDIR)$(bindir)/liblexilla.so
+	rm -f $(DESTDIR)$(libdir)/libscintilla.so
+	rm -f $(DESTDIR)$(libdir)/liblexilla.so
 	rm -rf $(DESTDIR)$(SYSCONF_PATH)
 ifdef gnomeprefix
 	rm -f $(DESTDIR)$(datadir)/applications/SciTE.desktop
diff -r d1b44eea341b src/LexillaLibrary.cxx
--- a/scite/src/LexillaLibrary.cxx	Sat Jun 06 20:55:49 2020 +1000
+++ b/scite/src/LexillaLibrary.cxx	Mon Jun 08 12:02:44 2020 +1000
@@ -122,8 +122,12 @@
 			paths.remove_prefix(separator + 1);
 		}
 		if (path == ".") {
-			path = directoryLoadDefault;
-			path += pathSeparator;
+			if (directoryLoadDefault.empty()) {
+				path = "";
+			} else {
+				path = directoryLoadDefault;
+				path += pathSeparator;
+			}
 			path += defaultName;
 		}
 		if (!NameContainsDot(path)) {
