--- configure_2012-11-21_104449.ac	2012-11-21 10:44:49.577069798 +0100
+++ configure.ac	2012-11-21 10:51:47.620933574 +0100
@@ -185,11 +185,7 @@
 
 dnl ==========================================================================
 
-PKG_CHECK_MODULES(GRAPHITE2, graphite2, have_graphite=true, have_graphite=false)
-if $have_graphite; then
-    AC_DEFINE(HAVE_GRAPHITE2, 1, [Have Graphite library])
-fi
-AM_CONDITIONAL(HAVE_GRAPHITE2, $have_graphite)
+AM_CONDITIONAL(HAVE_GRAPHITE2, 0)
 
 dnl ==========================================================================
 
