From:  	Antonio Rojas <arojas@archlinux.org>
Subject: handle configuration 'NoProgressBar'

diff --git a/backends/alpm/pk-alpm-config.c b/backends/alpm/pk-alpm-config.c
index 20da84e45..cfe916aa0 100644
--- a/backends/alpm/pk-alpm-config.c
+++ b/backends/alpm/pk-alpm-config.c
@@ -40,7 +40,7 @@ static gchar *xfercmd = NULL;
 typedef struct
 {
 	 gboolean		 checkspace, color, disabledownloadtimeout, ilovecandy,
-				totaldl, usesyslog, verbosepkglists, is_check;
+				noprogressbar, totaldl, usesyslog, verbosepkglists, is_check;
 
 	 gchar			*arch, *cleanmethod, *dbpath, *gpgdir, *logfile,
 				*root, *xfercmd;
@@ -146,6 +146,14 @@ pk_alpm_config_set_ilovecandy (PkAlpmConfig *config)
 	config->ilovecandy = TRUE;
 }
 
+static void
+pk_alpm_config_set_noprogressbar (PkAlpmConfig *config)
+{
+	g_return_if_fail (config != NULL);
+
+	config->noprogressbar = TRUE;
+}
+
 static void
 pk_alpm_config_set_totaldl (PkAlpmConfig *config)
 {
@@ -182,6 +190,7 @@ static const PkAlpmConfigBoolean pk_alpm_config_boolean_options[] = {
 	{ "Color", pk_alpm_config_set_color },
 	{ "DisableDownloadTimeout", pk_alpm_config_set_disabledownloadtimeout },
 	{ "ILoveCandy", pk_alpm_config_set_ilovecandy },
+	{ "NoProgressBar", pk_alpm_config_set_noprogressbar },
 	{ "TotalDownload", pk_alpm_config_set_totaldl },
 	{ "UseSyslog", pk_alpm_config_set_usesyslog },
 	{ "VerbosePkgLists", pk_alpm_config_set_verbosepkglists },
@@ -599,6 +608,11 @@ pk_alpm_config_parse (PkAlpmConfig *config, const gchar *filename,
 			continue;
 		}
 
+		if (g_strcmp0 (key, "ParallelDownloads") == 0 && str != NULL) {
+			/* Ignore "ParallelDownloads" key instead of crashing */
+			continue;
+		}
+
 		/* report errors from above */
 		g_set_error (&e, PK_ALPM_ERROR, PK_ALPM_ERR_CONFIG_INVALID,
 			     "unrecognised directive '%s'", key);
