diff -u -r -U5 newsbeuter-2.5/src/controller.cpp newsbeuter-2.5.includes/src/controller.cpp
--- newsbeuter-2.5/src/controller.cpp	2012-01-06 09:41:13.000000000 -0500
+++ newsbeuter-2.5.includes/src/controller.cpp	2012-02-18 15:57:32.525947811 -0500
@@ -36,10 +36,11 @@
 
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <pwd.h>
+#include <unistd.h>
 
 #include <ncurses.h>
 
 #include <libxml/xmlversion.h>
 #include <libxml/parser.h>
diff -u -r -U5 newsbeuter-2.5/src/google_api.cpp newsbeuter-2.5.includes/src/google_api.cpp
--- newsbeuter-2.5/src/google_api.cpp	2012-01-06 09:41:13.000000000 -0500
+++ newsbeuter-2.5.includes/src/google_api.cpp	2012-02-18 15:28:32.287929217 -0500
@@ -7,10 +7,12 @@
 #include <config.h>
 #include <utils.h>
 
 #include <curl/curl.h>
 
+#include <unistd.h>
+
 #define GREADER_LOGIN					"https://www.google.com/accounts/ClientLogin"
 #define GREADER_API_PREFIX				"http://www.google.com/reader/api/0/"
 #define GREADER_FEED_PREFIX				"http://www.google.com/reader/atom/"
 
 #define GREADER_SUBSCRIPTION_LIST		GREADER_API_PREFIX "subscription/list"
diff -u -r -U5 newsbeuter-2.5/src/pb_controller.cpp newsbeuter-2.5.includes/src/pb_controller.cpp
--- newsbeuter-2.5/src/pb_controller.cpp	2012-01-06 09:41:13.000000000 -0500
+++ newsbeuter-2.5.includes/src/pb_controller.cpp	2012-02-18 15:57:48.241947978 -0500
@@ -10,10 +10,11 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <pwd.h>
 #include <cstdlib>
 #include <signal.h>
+#include <unistd.h>
 
 #include <keymap.h>
 #include <configcontainer.h>
 #include <colormanager.h>
 #include <exceptions.h>
diff -u -r -U5 newsbeuter-2.5/src/queueloader.cpp newsbeuter-2.5.includes/src/queueloader.cpp
--- newsbeuter-2.5/src/queueloader.cpp	2012-01-06 09:41:13.000000000 -0500
+++ newsbeuter-2.5.includes/src/queueloader.cpp	2012-02-18 15:57:59.472948099 -0500
@@ -5,10 +5,11 @@
 #include <logger.h>
 #include <fstream>
 #include <cstring>
 #include <config.h>
 #include <libgen.h>
+#include <unistd.h>
 
 using namespace newsbeuter;
 
 namespace podbeuter {
 
diff -u -r -U5 newsbeuter-2.5/src/reloadthread.cpp newsbeuter-2.5.includes/src/reloadthread.cpp
--- newsbeuter-2.5/src/reloadthread.cpp	2012-01-06 09:41:13.000000000 -0500
+++ newsbeuter-2.5.includes/src/reloadthread.cpp	2012-02-18 15:21:21.137924605 -0500
@@ -1,8 +1,10 @@
 #include <reloadthread.h>
 #include <logger.h>
 
+#include <unistd.h>
+
 namespace newsbeuter {
 
 reloadthread::reloadthread(controller * c, configcontainer * cf) : ctrl(c), oldtime(0), waittime_sec(0), suppressed_first(false), cfg(cf) {
 	LOG(LOG_INFO,"reloadthread: waiting %u seconds between reloads",waittime_sec);
 }
diff -u -r -U5 newsbeuter-2.5/src/utils.cpp newsbeuter-2.5.includes/src/utils.cpp
--- newsbeuter-2.5/src/utils.cpp	2012-01-06 09:41:13.000000000 -0500
+++ newsbeuter-2.5.includes/src/utils.cpp	2012-02-18 15:01:55.076912171 -0500
@@ -8,10 +8,11 @@
 #include <iconv.h>
 #include <errno.h>
 #include <pwd.h>
 #include <libgen.h>
 #include <sys/utsname.h>
+#include <unistd.h>
 
 #include <sstream>
 #include <locale>
 #include <cwchar>
 #include <cstring>
