--- gcc-4.8.2/libgcc/Makefile.in.orig	2013-02-04 20:06:20.000000000 +0100
+++ gcc-4.8.2/libgcc/Makefile.in	2013-10-17 08:05:38.314091932 +0200
@@ -492,7 +492,7 @@
 ifeq ($(LIB2_DIVMOD_EXCEPTION_FLAGS),)
 # Provide default flags for compiling divmod functions, if they haven't been
 # set already by a target-specific Makefile fragment.
-LIB2_DIVMOD_EXCEPTION_FLAGS := -fexceptions -fnon-call-exceptions
+LIB2_DIVMOD_EXCEPTION_FLAGS := -fno-exceptions -fnon-call-exceptions
 endif
 
 # Build LIB2_DIVMOD_FUNCS.
@@ -813,7 +813,7 @@
 # libgcc_eh.a, only LIB2ADDEH matters.  If we do, only LIB2ADDEHSTATIC and
 # LIB2ADDEHSHARED matter.  (Usually all three are identical.)
 
-c_flags := -fexceptions
+c_flags := -fno-exceptions
 
 ifeq ($(enable_shared),yes)
 
