diff -aur Amaya11.3.1/Amaya/thotlib/base/AmayaApp.cpp Amaya11.3.1.new/Amaya/thotlib/base/AmayaApp.cpp
--- Amaya11.3.1/Amaya/thotlib/base/AmayaApp.cpp	2009-10-29 09:59:22.000000000 +0100
+++ Amaya11.3.1.new/Amaya/thotlib/base/AmayaApp.cpp	2010-08-17 21:01:23.823806630 +0200
@@ -326,7 +326,7 @@
 
   // Log window.
 //  new wxLogWindow(NULL, wxT("Amaya traces"));
-
+  wxWakeUpIdle();
   return true;
 }
