--- mysql-gui-tools-5.0r14/administrator/configure.in.BAD	2009-08-31 11:55:53.000000000 -0500
+++ mysql-gui-tools-5.0r14/administrator/configure.in	2009-08-31 11:56:57.000000000 -0500
@@ -19,7 +19,7 @@
 #
 # internal libraries
 #
-AC_CHECK_FILE(../mysql-gui-common/library/base-library, [],
+AC_CHECK_FILE(../common/library/base-library, [],
 	        AC_MSG_ERROR([mysql-gui-common source tree must be present in the same directory as this program]))
 
 #AC_CHECK_FILE(../mysql-gui-common/source/linux/libmacommongui.a, [],
--- mysql-gui-tools-5.0r14/query-browser/configure.in.BAD	2009-08-31 11:57:25.000000000 -0500
+++ mysql-gui-tools-5.0r14/query-browser/configure.in	2009-08-31 16:11:28.000000000 -0500
@@ -17,8 +17,8 @@
 #
 # internal libraries
 #
-AC_CHECK_FILE(../mysql-gui-common/library/base-library/include, 
-        [MYSQLA_INCLUDE='-I$(top_srcdir)/../mysql-gui-common/library/base-library/include -I$(top_srcdir)/../mysql-gui-common/source/linux'],
+AC_CHECK_FILE(../common/library/base-library/include, 
+        [MYSQLA_INCLUDE='-I$(top_srcdir)/../common/library/base-library/include -I$(top_srcdir)/../common/source/linux'],
 	        AC_MSG_ERROR([mysql-gui-common source tree must be present in the same directory as this program]))
 
 #AC_CHECK_FILE(../mysql-gui-common/source/linux/libqbcommongui.a, [],
@@ -73,6 +73,7 @@
 PKG_CHECK_MODULES(GNOME, [libglade-2.0
 			gthread-2.0
 			libxml-2.0 >= 2.6.2
+                        libgnome-2.0
                         $gtkhtml
                         gtkmm-2.4])
 AC_SUBST(GNOME_CFLAGS)
