diff -rupN a/image.c b/image.c
--- a/image.c	2014-04-24 20:40:26.000000000 +0200
+++ b/image.c	2014-05-27 11:39:41.786715632 +0200
@@ -278,7 +278,11 @@ bool img_load_gif(img_t *img, const file
 		}
 	} while (rec != TERMINATE_RECORD_TYPE);
 
+#if defined(GIFLIB_MAJOR) && GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+	DGifCloseFile(gif, NULL);
+#else
 	DGifCloseFile(gif);
+#endif
 
 	if (err && !file->loaded)
 		warn("corrupted gif file: %s", file->name);
