diff -rupN clementine-0.7.1.old/data/schema/schema.sql clementine-0.7.1/data/schema/schema.sql
--- clementine-0.7.1.old/data/schema/schema.sql	2011-03-29 15:16:12.000000000 -0400
+++ clementine-0.7.1/data/schema/schema.sql	2012-01-12 19:15:39.889374607 -0500
@@ -35,4 +35,5 @@ CREATE TABLE songs (
 CREATE TABLE schema_version (
   version INTEGER NOT NULL
 );
+
 INSERT INTO schema_version (version) VALUES (0);
