diff -aur awn-extras-0.4.0/applets/maintained/notification-daemon/daemon.c awn-extras-0.4.0-new//applets/maintained/notification-daemon/daemon.c
--- awn-extras-0.4.0/applets/maintained/notification-daemon/daemon.c	2010-04-09 16:26:20.000000000 -0700
+++ awn-extras-0.4.0-new//applets/maintained/notification-daemon/daemon.c	2011-03-06 09:33:33.603336393 -0800
@@ -1487,7 +1487,15 @@
   if (fork() == 0)
   {
     notify_init("notify-send");
-    notify = notify_notification_new(summary, body, icon_str, NULL);
+    #ifdef NOTIFY_CHECK_VERSION
+    #if NOTIFY_CHECK_VERSION (0, 7, 0)
+        notify = notify_notification_new(summary, body, icon_str);
+    #else
+        notify = notify_notification_new(summary, body, icon_str, NULL);
+    #endif
+    #else
+        notify = notify_notification_new(summary, body, icon_str, NULL);
+    #endif
     notify_notification_set_category(notify, type);
     notify_notification_set_urgency(notify, urgency);
     notify_notification_set_timeout(notify, expire_timeout);
