diff -Naur gtranslator-2.91.5.orig/src/gtr-tab.c gtranslator-2.91.5/src/gtr-tab.c
--- gtranslator-2.91.5.orig/src/gtr-tab.c	2012-06-07 13:48:56.000000000 +0200
+++ gtranslator-2.91.5/src/gtr-tab.c	2012-11-05 19:18:01.899014670 +0100
@@ -149,9 +149,6 @@
   g_return_if_fail (GTR_IS_TAB (tab));
   g_return_if_fail (filename != NULL);
 
-  gdl_dock_layout_save_layout (tab->priv->layout_manager, name);
-  if (!gdl_dock_layout_save_to_file (tab->priv->layout_manager, filename))
-    g_warning ("Saving dock layout to '%s' failed!", filename);
 }
 
 static void
@@ -160,23 +157,6 @@
                      const gchar *name)
 {
   g_return_if_fail (GTR_IS_TAB (tab));
-
-  if (!layout_filename ||
-      !gdl_dock_layout_load_from_file (tab->priv->layout_manager,
-                                       layout_filename))
-    {
-      gchar *path;
-
-      path = gtr_dirs_get_ui_file ("layout.xml");
-
-      if (!gdl_dock_layout_load_from_file (tab->priv->layout_manager,
-                                           path))
-        g_warning ("Loading layout from '%s' failed!!", path);
-      g_free (path);
-    }
-
-  if (!gdl_dock_layout_load_layout (tab->priv->layout_manager, name))
-    g_warning ("Loading layout failed!!");
 }
 
 static void
@@ -702,7 +682,6 @@
   GtkWidget *vertical_box;
   GtkWidget *msgid_label;
   GtkWidget *scroll;
-  GtkWidget *dockbar;
   GtrTabPrivate *priv;
 
   tab->priv = GTR_TAB_GET_PRIVATE (tab);
@@ -727,22 +706,6 @@
   gtk_widget_show (priv->dock);
   gtk_box_pack_end (GTK_BOX (hbox), priv->dock, TRUE, TRUE, 0);
 
-  dockbar = gdl_dock_bar_new (GDL_DOCK (priv->dock));
-  gtk_widget_show (dockbar);
-  gtk_box_pack_start (GTK_BOX (hbox), dockbar, FALSE, FALSE, 0);
-
-  priv->layout_manager = gdl_dock_layout_new (GDL_DOCK (priv->dock));
-  g_signal_connect (priv->layout_manager->master,
-                    "layout-changed",
-                    G_CALLBACK (on_layout_changed),
-                    tab);
-
-  g_settings_bind (priv->ui_settings,
-                   GTR_SETTINGS_PANEL_SWITCHER_STYLE,
-                   priv->layout_manager->master,
-                   "switcher-style",
-                   G_SETTINGS_BIND_GET | G_SETTINGS_BIND_SET);
-
   /* Message table */
   priv->message_table = gtr_message_table_new (GTK_WIDGET (tab));
   gtk_widget_show (priv->message_table);
@@ -904,7 +867,6 @@
   g_clear_object (&priv->files_settings);
   g_clear_object (&priv->editor_settings);
   g_clear_object (&priv->state_settings);
-  g_clear_object (&priv->layout_manager);
 
   G_OBJECT_CLASS (gtr_tab_parent_class)->dispose (object);
 }
