From 8a123f9f68e93452dc2c044f8a2ffa07a6038364 Mon Sep 17 00:00:00 2001
From: Adrien Beraud <adrien.beraud@savoirfairelinux.com>
Date: Sun, 9 Jul 2017 07:51:05 -0400
Subject: [PATCH 1/2] network: prevent sending packets larger than 1280 bytes

---
 include/opendht/network_engine.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/opendht/network_engine.h b/include/opendht/network_engine.h
index 53f9044..a037cde 100644
--- a/include/opendht/network_engine.h
+++ b/include/opendht/network_engine.h
@@ -464,7 +464,7 @@ private:
     static constexpr unsigned BLACKLISTED_MAX {10};
 
     static constexpr size_t MTU {1280};
-    static constexpr size_t MAX_PACKET_VALUE_SIZE {8 * 1024};
+    static constexpr size_t MAX_PACKET_VALUE_SIZE {600};
 
     static const std::string my_v;
     static std::mt19937 rd_device;
-- 
2.13.2

