--- a/tests/test_libmat2.py
+++ b/tests/test_libmat2.py
@@ -449,9 +449,10 @@ class TestCleaning(unittest.TestCase):
                 'HandlerType': 'Metadata',
                 'HandlerVendorID': 'Apple',
                 'MajorBrand': 'MP4  Base Media v1 [IS0 14496-12:2003]',
+                'MediaDataOffset': 48,
+                'MediaDataSize': 379872,
                 'MediaHeaderVersion': 0,
                 'MinorVersion': '0.2.0',
-                'MovieDataOffset': 48,
                 'MovieHeaderVersion': 0,
                 'NextTrackID': 3,
                 'PreferredRate': 1,
