diff -u -r jedi-0.13.1/jedi/api/environment.py jedi-0.13.1-fix/jedi/api/environment.py
--- jedi-0.13.1/jedi/api/environment.py	2018-10-02 17:07:35.000000000 +0000
+++ jedi-0.13.1-fix/jedi/api/environment.py	2018-11-23 15:03:25.989461415 +0000
@@ -190,15 +190,11 @@
 
 
 def get_cached_default_environment():
-    environment = _get_cached_default_environment()
-    if environment.path != os.environ.get('VIRTUAL_ENV'):
-        _get_cached_default_environment.clear_cache()
-        return _get_cached_default_environment()
-    return environment
+    return _get_cached_default_environment(os.environ.get('VIRTUAL_ENV'))
 
 
 @time_cache(seconds=10 * 60)  # 10 Minutes
-def _get_cached_default_environment():
+def _get_cached_default_environment(_venv):
     return get_default_environment()
 
 
