diff -wbBur drivel-3.0.5/src/soup-xmlrpc-message.c drivel-3.0.5.my/src/soup-xmlrpc-message.c
--- drivel-3.0.5/src/soup-xmlrpc-message.c	2011-04-15 21:54:16.000000000 +0400
+++ drivel-3.0.5.my/src/soup-xmlrpc-message.c	2016-07-26 15:54:32.676453746 +0300
@@ -60,7 +60,7 @@
 
 
 SoupXmlrpcMessage *
-soup_xmlrpc_message_new (const char *uri_string)
+_soup_xmlrpc_message_new (const char *uri_string)
 {
 	SoupXmlrpcMessage *msg;
 	SoupURI *uri;
diff -wbBur drivel-3.0.5/src/soup-xmlrpc-message.h drivel-3.0.5.my/src/soup-xmlrpc-message.h
--- drivel-3.0.5/src/soup-xmlrpc-message.h	2011-04-15 21:54:16.000000000 +0400
+++ drivel-3.0.5.my/src/soup-xmlrpc-message.h	2016-07-26 15:54:21.019787214 +0300
@@ -33,7 +33,7 @@
 
 GType soup_xmlrpc_message_get_type (void);
 
-SoupXmlrpcMessage *soup_xmlrpc_message_new          (const char *uri_string);
+SoupXmlrpcMessage *_soup_xmlrpc_message_new          (const char *uri_string);
 SoupXmlrpcMessage *soup_xmlrpc_message_new_from_uri (SoupURI *uri);
 
 void soup_xmlrpc_message_start_call     (SoupXmlrpcMessage *msg,
diff -wbBur drivel-3.0.5/src/xmlrpc.c drivel-3.0.5.my/src/xmlrpc.c
--- drivel-3.0.5/src/xmlrpc.c	2011-04-15 21:54:16.000000000 +0400
+++ drivel-3.0.5.my/src/xmlrpc.c	2016-07-26 15:54:45.196453603 +0300
@@ -37,7 +37,7 @@
 {
 	SoupXmlrpcMessage *msg;
 
-	msg = soup_xmlrpc_message_new (uri);
+	msg = _soup_xmlrpc_message_new (uri);
 	soup_xmlrpc_message_start_call (msg, name);
 
 	return (msg);
@@ -48,7 +48,7 @@
 {
 	SoupXmlrpcMessage *msg;
 
-	msg = soup_xmlrpc_message_new (uri);
+	msg = _soup_xmlrpc_message_new (uri);
 	soup_xmlrpc_message_start_call (msg, name);
 	soup_xmlrpc_message_start_param (msg);
 	soup_xmlrpc_message_start_struct (msg);
