diff -upr gambas3-3.3.4.orig/gb.jit/src/jit_codegen.cpp gambas3-3.3.4/gb.jit/src/jit_codegen.cpp
--- gambas3-3.3.4.orig/gb.jit/src/jit_codegen.cpp	2012-09-25 17:04:41.000000000 +0300
+++ gambas3-3.3.4/gb.jit/src/jit_codegen.cpp	2012-12-25 03:55:54.000000000 +0200
@@ -3614,7 +3614,7 @@ void TryExpression::codegen(){
 		create_gep(temp_errcontext1, TARGET_BITS, 0, TARGET_BITS, 0));
 	
 	llvm::Function* f = llvm::cast<llvm::Function>(get_global_function(_setjmp, 'i', "p"));
-	f->addFnAttr(llvm::Attribute::ReturnsTwice);
+	f->addFnAttr(llvm::Attributes::ReturnsTwice);
 	
 	llvm::Value* setjmp_return = builder->CreateCall(f, jmpbuf);
 	
@@ -3665,7 +3665,7 @@ void LargeTryExpression::codegen(){
 		create_gep(temp_errcontext2, TARGET_BITS, 0, TARGET_BITS, 0));
 	
 	llvm::Function* f = llvm::cast<llvm::Function>(get_global_function(_setjmp, 'i', "p"));
-	f->addFnAttr(llvm::Attribute::ReturnsTwice);
+	f->addFnAttr(llvm::Attributes::ReturnsTwice);
 	
 	llvm::Value* setjmp_return = builder->CreateCall(f, jmpbuf);
 	
diff -upr gambas3-3.3.4.orig/gb.jit/src/jit_gambas_pass.cpp gambas3-3.3.4/gb.jit/src/jit_gambas_pass.cpp
--- gambas3-3.3.4.orig/gb.jit/src/jit_gambas_pass.cpp	2012-09-25 17:04:41.000000000 +0300
+++ gambas3-3.3.4/gb.jit/src/jit_gambas_pass.cpp	2012-12-25 03:38:36.000000000 +0200
@@ -25,7 +25,7 @@
 
 #include "llvm/Pass.h"
 #include "llvm/Function.h"
-#include "llvm/Support/IRBuilder.h"
+#include "llvm/IRBuilder.h"
 #include "llvm/Support/raw_ostream.h"
 
 #include "main.h"
diff -upr gambas3-3.3.4.orig/gb.jit/src/jit.h gambas3-3.3.4/gb.jit/src/jit.h
--- gambas3-3.3.4.orig/gb.jit/src/jit.h	2012-09-25 17:04:41.000000000 +0300
+++ gambas3-3.3.4/gb.jit/src/jit.h	2012-12-25 03:38:23.000000000 +0200
@@ -39,7 +39,7 @@
 #include "llvm/Support/TargetSelect.h"
 #include "llvm/Support/ManagedStatic.h"
 #include "llvm/Support/raw_ostream.h"
-#include "llvm/Support/IRBuilder.h"
+#include "llvm/IRBuilder.h"
 #include "llvm/Support/DynamicLibrary.h"
 #include "llvm/Support/CFG.h"
 #include "llvm/PassManager.h"
