diff -Nur HandBrake-0.9.5.orig/gtk/src/callbacks.c HandBrake-0.9.5/gtk/src/callbacks.c
--- HandBrake-0.9.5.orig/gtk/src/callbacks.c	2011-03-07 13:06:20.333338164 -0800
+++ HandBrake-0.9.5/gtk/src/callbacks.c	2011-03-07 13:21:32.973338933 -0800
@@ -39,6 +39,10 @@
 #endif
 
 #include <libnotify/notify.h>
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
+
 #include <gdk/gdkx.h>
 #else
 #define WINVER 0x0500
@@ -4889,9 +4893,13 @@
 	notification = notify_notification_new(
 		"Encode Complete",
 		"Put down that cocktail, Your HandBrake queue is done!",
-		"hb-icon",
-		NULL);
+		"hb-icon"
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+                );
+#else
+		,NULL);
 	notify_notification_attach_to_status_icon(notification, si);
+#endif
 	g_signal_connect(notification, "closed", (GCallback)notify_closed_cb, ud);
 	notify_notification_show(notification, NULL);
 #endif
