--- a/script/create-dist
+++ b/script/create-dist
@@ -93,7 +93,6 @@
     'libsecurity_state.a',
     'libcookie_config.a',
     'libos_crypt.a',
-    os.path.join('lib', 'libffmpeg.so'),
   ],
   'win32': [
     'chromedriver.exe',
@@ -314,7 +313,6 @@
       copy_generated_sources(target_arch, component, output_dir)
       copy_locales(target_arch, component, output_dir)
 
-  copy_ffmpeg(target_arch)
   copy_sources()
   generate_licenses()
   if not args.no_zip:
