--- requests/certs.py.orig	2013-04-03 10:25:52.618680901 +0200
+++ requests/certs.py	2013-04-03 10:26:29.291120949 +0200
@@ -12,13 +12,11 @@
 packaged CA bundle.
 """
 
-import os.path
-
 
 def where():
     """Return the preferred certificate bundle."""
     # vendored bundle inside Requests
-    return os.path.join(os.path.dirname(__file__), 'cacert.pem')
+    return "/etc/ssl/certs/ca-certificates.crt"
 
 if __name__ == '__main__':
     print(where())
