From 973549cd49d0867367a100269dd4ec37c50735cc Mon Sep 17 00:00:00 2001
From: Basil Eric Rabi <ericbasil.rabi@gmail.com>
Date: Fri, 31 Mar 2023 20:04:12 +0800
Subject: [PATCH] add missing headers

fixes #52462
---
diff --git a/external/untwine/api/QgisUntwine.cpp b/external/untwine/api/QgisUntwine.cpp
index c25fe99365d8..f79e99348d14 100644
--- a/external/untwine/api/QgisUntwine.cpp
+++ b/external/untwine/api/QgisUntwine.cpp
@@ -1,3 +1,4 @@
+#include <cstdint>
 #include <iostream>
 
 #include "QgisUntwine.hpp"
diff --git a/external/untwine/untwine/VoxelKey.hpp b/external/untwine/untwine/VoxelKey.hpp
index a29a8024b9c0..839e1ec746b8 100644
--- a/external/untwine/untwine/VoxelKey.hpp
+++ b/external/untwine/untwine/VoxelKey.hpp
@@ -11,6 +11,7 @@
  ****************************************************************************/
 
 #include <string>
+#include <cstdint>
 #include <algorithm>
 
 #pragma once
