diff -u gtk-0.12.2/Graphics/UI/Gtk/General/hsgthread.c\~ gtk-0.12.2/Graphics/UI/Gtk/General/hsgthread.c
--- gtk-0.12.2/Graphics/UI/Gtk/General/hsgthread.c~	2011-11-14 07:15:41.000000000 +0900
+++ gtk-0.12.2/Graphics/UI/Gtk/General/hsgthread.c	2012-01-08 14:58:45.849209318 +0900
@@ -26,7 +26,8 @@
  */
 
 #include <glib.h>
-#include <glib/gthread.h>
+/* f17 gives Error "Only <glib.h> can be included directly."
+   #include <glib/gthread.h> */
 #include <gdk/gdk.h>
 #include "hsgthread.h"
 

Diff finished.  Sun Jan  8 14:58:54 2012
