From fd139c5cf48f1a805b479899c2bc64156e733209 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Mon, 25 Jan 2016 15:52:46 +0000
Subject: [PATCH] Update the AppData to spec version 0.7+

---
 appdata/transmageddon.appdata.xml.in | 75 ++++++++++++++++------------
 1 file changed, 44 insertions(+), 31 deletions(-)

diff --git a/appdata/transmageddon.appdata.xml.in b/appdata/transmageddon.appdata.xml.in
index bb1fd65..ccf5556 100644
--- a/appdata/transmageddon.appdata.xml.in
+++ b/appdata/transmageddon.appdata.xml.in
@@ -1,33 +1,46 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2013 Christian F.K. Schaller <uraeus@linuxrising.org>  -->
-<application>
- <id type="desktop">transmageddon.desktop</id>
- <metadata_license>CC-BY-3.0</metadata_license>
- <project_license>LGPL-2.1+</project_license>
- <_name>Transmageddon Media Transcoder</_name>
- <_summary>Easy to use audio and video transcoding application</_summary>
- <description>
-  <_p>
-Transmageddon is an application for converting audio and video files between different formats. It allows you to either manually choose the formats you want to convert between or alternatively you can choose a predefined target and the selections will be made for you. Transmageddon supports a very wide range of formats and can also optionally function as a DVD ripping tool.
-  </_p>
-  <_p>Feature list:</_p>
-  <ul>
-   <_li>Wide range of media codecs</_li>
-   <_li>Allows extracting audio or video streams separately</_li>
-   <_li>Alows ripping of DVD if needed software libraries are available</_li>
-   <_li>Supports named device presets</_li>
-   <_li>Lets you add language metadata if missing</_li>
-   <_li>Can change orientation of video if necessary</_li>
-   <_li>Supports files with multiple audio streams</_li>
-  </ul>
-  <_p>
-You might want to install this application if your need to convert media files between different formats due to for example having hardware that only supports a limited range of formats.
-  </_p>
- </description>
- <screenshots>
-  <screenshot type="default">https://git.gnome.org/browse/transmageddon/plain/appdata/transmageddon.png</screenshot>
- </screenshots>
- <url type="homepage">http://www.linuxrising.org/</url>
- <updatecontact>uraeus@linuxrising.org</updatecontact>
- <project_group>GStreamer</project_group>
-</application>
+<component type="desktop">
+  <id>transmageddon.desktop</id>
+  <metadata_license>CC-BY-3.0</metadata_license>
+  <project_license>LGPL-2.1+</project_license>
+  <_name>Transmageddon Media Transcoder</_name>
+  <_summary>Easy to use audio and video transcoding application</_summary>
+  <description>
+    <_p>
+      Transmageddon is an application for converting audio and video files
+      between different formats.
+      It allows you to either manually choose the formats you want to convert
+      between or alternatively you can choose a predefined target and the
+      selections will be made for you.
+      Transmageddon supports a very wide range of formats and can also
+      optionally function as a DVD ripping tool.
+    </_p>
+    <_p>Feature list:</_p>
+    <ul>
+      <_li>Wide range of media codecs</_li>
+      <_li>Allows extracting audio or video streams separately</_li>
+      <_li>Alows ripping of DVD if needed software libraries are available</_li>
+      <_li>Supports named device presets</_li>
+      <_li>Lets you add language metadata if missing</_li>
+      <_li>Can change orientation of video if necessary</_li>
+      <_li>Supports files with multiple audio streams</_li>
+    </ul>
+    <_p>
+      You might want to install this application if your need to convert media
+      files between different formats due to for example having hardware that
+      only supports a limited range of formats.
+    </_p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://git.gnome.org/browse/transmageddon/plain/appdata/transmageddon.png</image>
+    </screenshot>
+  </screenshots>
+  <url type="homepage">http://www.linuxrising.org/</url>
+  <url type="bugtracker">https://bugzilla.gnome.org/enter_bug.cgi?product=transmageddon</url>
+  <url type="donation">http://www.gnome.org/friends/</url>
+  <update_contact>uraeus@linuxrising.org</update_contact>
+  <project_group>GNOME</project_group>
+  <translation type="gettext">transmageddon</translation>
+</component>
-- 
2.18.1

