From 5fdc432976ea725f880645cca7fce023eeab54af Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 20 Jun 2019 20:49:11 +0200
Subject: [PATCH] Fix message dialog about ImageMagick Policies

---
 src/Widgets/MergePDF.vala | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/Widgets/MergePDF.vala b/src/Widgets/MergePDF.vala
index 389b0a9..9437ddc 100644
--- a/src/Widgets/MergePDF.vala
+++ b/src/Widgets/MergePDF.vala
@@ -241,6 +241,10 @@ namespace pdftricks {
                 var file_pdf = (string) cell1;
                 if(!file_pdf.contains(".pdf")){
                     file_pdf = convert_to_pdf(file_pdf);
+                    if (file_pdf == ""){
+                        files_pdf = "";
+                        return true;
+                    }
                 }
                 files_pdf = files_pdf + " " + file_pdf.replace(" ", "\\ ");
                 return false;
@@ -283,7 +287,7 @@ namespace pdftricks {
                 if(output.contains("Error")){
                     return "";
                 }
-                if(stderr.contains("not authorized")){
+                if(stderr.contains("not allowed")){
                     var message_dialog = new Granite.MessageDialog.with_image_from_icon_name (_("ImageMagick Policies"), _("Change the ImageMagick security policies that prevent this operation and try again."), "process-stop", Gtk.ButtonsType.CLOSE);
                     message_dialog.set_transient_for(window);
                     message_dialog.show_all ();
-- 
2.22.0

