diff -Naur ubuntu-sso-client-4.2.0.orig/ubuntu_sso/utils/webclient/common.py ubuntu-sso-client-4.2.0/ubuntu_sso/utils/webclient/common.py
--- ubuntu-sso-client-4.2.0.orig/ubuntu_sso/utils/webclient/common.py	2013-04-02 00:10:25.870644798 +0000
+++ ubuntu-sso-client-4.2.0/ubuntu_sso/utils/webclient/common.py	2013-04-02 00:12:54.226165669 +0000
@@ -158,8 +158,7 @@
                               timestamp=str(timestamp))
 
         url, signed_headers, body = oauth_client.sign(
-            uri, method, parameters if parameters is not None else {},
-            {'Content-Type': 'application/x-www-form-urlencoded'})
+            uri, method)
 
         return url, signed_headers, body
 
