diff -Naur xarchiver-0.5.4.orig/autogen.sh xarchiver-0.5.4/autogen.sh
--- xarchiver-0.5.4.orig/autogen.sh	2008-10-13 16:03:03.000000000 +0200
+++ xarchiver-0.5.4/autogen.sh	2014-12-23 13:45:00.959300133 +0100
@@ -1,39 +1,9 @@
 #!/bin/sh
-#
-# $Id: autogen.sh 22391 2006-07-10 12:54:50Z benny $
-#
-# Copyright (c) 2002-2006
-#         The Xfce development team. All rights reserved.
-#
-# Written for Xfce by Benedikt Meurer <benny@xfce.org>.
-#
-
-(type xdt-autogen) >/dev/null 2>&1 || {
-  cat >&2 <<EOF
-autogen.sh: You don't seem to have the Xfce development tools installed on
-            your system, which are required to build this software.
-            Please install the xfce4-dev-tools package first, it is available
-            from http://www.xfce.org/.
-EOF
-  exit 1
-}
-
-# verify that po/LINGUAS is present
-(test -f po/LINGUAS) >/dev/null 2>&1 || {
-  cat >&2 <<EOF
-autogen.sh: The file po/LINGUAS could not be found. Please check your snapshot
-            or try to checkout again.
-EOF
-  exit 1
-}
-
-# substitute revision and linguas
-linguas=`sed -e '/^#/d' po/LINGUAS`
-revision=`LC_ALL=C svn info $0 | awk '/^Revision: / {printf "%05d\n", $2}'`
-sed -e "s/@LINGUAS@/${linguas}/g" \
-    -e "s/@REVISION@/${revision}/g" \
-    < "configure.in.in" > "configure.in"
-
-exec xdt-autogen $@
-
-# vi:set ts=2 sw=2 et ai:
+# use automake v1.9
+aclocal \
+&& libtoolize --force --copy \
+&& intltoolize --force --copy --automake \
+&& autoheader \
+&& automake --add-missing --gnu --copy \
+&& autoconf \
+&& ./configure $@
diff -Naur xarchiver-0.5.4.orig/configure.in xarchiver-0.5.4/configure.in
--- xarchiver-0.5.4.orig/configure.in	2014-12-23 12:30:10.265150000 +0100
+++ xarchiver-0.5.4/configure.in	2014-12-23 13:50:45.394385149 +0100
@@ -27,7 +27,6 @@
 AC_INIT([xarchiver], [xarchiver_version], [http://bugzilla.xfce.org/])
 AC_PREREQ([2.50])
 AC_CANONICAL_TARGET()
-AC_REVISION([$Id: configure.in.in 28647 2008-11-07 13:53:00Z colossus $])
 
 dnl ***************************
 dnl *** Initialize automake ***
@@ -75,7 +74,11 @@
 dnl ******************************
 dnl *** Check for i18n support ***
 dnl ******************************
-XDT_I18N([bg ca cs da de el en_GB es eu fi fr hu id ja nb_NO nl pl pt_BR pt_PT ru sq sv tr uk])
+GETTEXT_PACKAGE=AC_PACKAGE_TARNAME()
+AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE], ["$GETTEXT_PACKAGE"], [Name of default gettext domain])
+AC_SUBST([GETTEXT_PACKAGE])
+
+AM_GLIB_GNU_GETTEXT()
 
 dnl ***********************************
 dnl *** Check for required packages ***
@@ -93,15 +96,10 @@
     GTK_API_VERSION="2.0"
 fi
 
-XDT_CHECK_PACKAGE([GTK], [gtk+-$GTK_API_VERSION], [2.10.0])
-XDT_CHECK_PACKAGE([GLIB], [glib-2.0], [2.10.0])
+PKG_CHECK_MODULES([GTK], [gtk+-$GTK_API_VERSION >= 2.10.0])
+PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.10.0])
 GTK_VERSION=`$PKG_CONFIG --modversion gtk+-$GTK_API_VERSION`
 
-dnl ***********************************
-dnl *** Check for debugging support ***
-dnl ***********************************
-XDT_FEATURE_DEBUG()
-
 AC_OUTPUT([
 Makefile
 po/Makefile.in
