--- xarchiver/trunk/src/interface.c	2009/03/19 18:03:52	29676
+++ xarchiver/trunk/src/interface.c	2009/03/19 20:48:19	29677
@@ -310,22 +310,14 @@
 	gtk_widget_show (tmp_image);
 	gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (help1),tmp_image);
 
+	donate = gtk_image_menu_item_new_with_mnemonic (_("_Donate"));
+	gtk_widget_show (donate);
+	gtk_container_add (GTK_CONTAINER (menuitem4_menu),donate);
+
 	about1 = gtk_image_menu_item_new_from_stock ("gtk-about",accel_group);
 	gtk_widget_show (about1);
 	gtk_container_add (GTK_CONTAINER (menuitem4_menu),about1);
 
-	thanks_to = gtk_image_menu_item_new_with_mnemonic (_("_Thanks to"));
-	gtk_widget_show (thanks_to);
-	gtk_container_add (GTK_CONTAINER (menuitem4_menu),thanks_to);
-
-	thanks_to_submenu = gtk_menu_new();
-	gtk_widget_show (thanks_to_submenu);
-	gtk_menu_item_set_submenu (GTK_MENU_ITEM (thanks_to),thanks_to_submenu);
-
-	donators = gtk_image_menu_item_new_with_mnemonic ("Your name and website here");
-	gtk_widget_show (donators);
-	gtk_container_add (GTK_CONTAINER (thanks_to_submenu),donators);
-
 	/* Create the toolbar */
 	toolbar1 = gtk_toolbar_new ();
 	gtk_widget_show (toolbar1);
@@ -543,7 +535,7 @@
 	g_signal_connect ((gpointer) multi_extract_menu,"activate",G_CALLBACK (xa_show_multi_extract_dialog),NULL);
 	g_signal_connect ((gpointer) help1,"activate",G_CALLBACK (xa_show_help),NULL);
 	g_signal_connect ((gpointer) about1,"activate",G_CALLBACK (xa_about),NULL);
-	g_signal_connect ((gpointer) donators,"activate",G_CALLBACK (xa_show_donate_page_on_the_web),NULL);
+	g_signal_connect ((gpointer) donate,"activate",G_CALLBACK (xa_show_donate_page_on_the_web),NULL);
 
 	g_signal_connect ((gpointer) New_button,	"clicked",G_CALLBACK (xa_new_archive),NULL);
 	g_signal_connect ((gpointer) Open_button,	"clicked",G_CALLBACK (xa_open_archive),NULL);
--- xarchiver/trunk/src/interface.h	2008/11/04 10:06:17	28603
+++ xarchiver/trunk/src/interface.h	2009/03/19 20:48:19	29677
@@ -48,8 +48,8 @@
 *name_label, *type_label, *compression_data, *encrypted_data, *encrypted_label,*number_of_files_data, *content_data, *comment_data, *size_data, *modified_data, *path_data, *type_data,
 *name_data, *separatormenuitem1, *separatormenuitem2, *separatormenuitem3, *separatormenuitem4, *separatormenuitem5, *separatormenuitem6, *quit1,
 *close1, *check_menu, *properties, *menuitem2, *menuitem2_menu, *addfile, *extract_menu, *delete_menu, *comment_menu, *multi_extract_menu,*view_shell_output1,
-*prefs_menu,*password_entry_menu, *image1, *image2, *menuitem4, *thanks_to,*thanks_to_submenu,*select_all,*deselect_all,*select_pattern, *exe_menu, *menuitem4_menu,
-*about1, *help1, *donators,*toolbar1, *toolbar2, *hbox1, *tmp_image, *pad_image, *New_button, *Open_button, *back_button, *home_button, *forward_button,
+*prefs_menu,*password_entry_menu, *image1, *image2, *menuitem4, *donate,*select_all,*deselect_all,*select_pattern, *exe_menu, *menuitem4_menu,
+*about1, *help1,*toolbar1, *toolbar2, *hbox1, *tmp_image, *pad_image, *New_button, *Open_button, *back_button, *home_button, *forward_button,
 *up_button,*separatortoolitem1,*separatortoolitem2,*separatortoolitem3,*AddFile_button,*Extract_button,*Stop_button,*toolitem1,*location_label,
 *location_entry,*hpaned1,*archive_dir_treeview,*scrolledwindow2,*ddelete,*rename_menu,*rrename,*cut,*copy,*paste,*view,*open_popupmenu;
