--- src/sage/repl/interpreter.py.orig	2015-08-16 14:57:55.571128473 +0200
+++ src/sage/repl/interpreter.py	2015-08-16 14:58:06.647849803 +0200
@@ -104,8 +104,8 @@
 import sys
 from sage.repl.preparse import preparse
 
-from IPython import Config
-from IPython.utils.traitlets import Bool, Type
+from traitlets.config.loader import Config
+from traitlets import Bool, Type
 
 from sage.env import SAGE_LOCAL
 
--- src/sage/repl/ipython_kernel/__main__.py.orig  2015-08-22 12:32:58.595605714 +0200
+++ src/sage/repl/ipython_kernel/__main__.py       2015-08-22 12:33:36.065900491 +0200
@@ -1,3 +1,3 @@
-from IPython.kernel.zmq.kernelapp import IPKernelApp
+from ipykernel.kernelapp import IPKernelApp
 from sage.repl.ipython_kernel.kernel import SageKernel
 IPKernelApp.launch_instance(kernel_class=SageKernel)
--- src/sage/repl/ipython_kernel/kernel.py.orig    2015-08-22 12:35:08.709928354 +0200
+++ src/sage/repl/ipython_kernel/kernel.py 2015-08-22 12:35:31.983429208 +0200
@@ -15,9 +15,9 @@
 #*****************************************************************************
 
 import sys
-from IPython.kernel.zmq.ipkernel import IPythonKernel
-from IPython.kernel.zmq.zmqshell import ZMQInteractiveShell
-from IPython.utils.traitlets import Type
+from ipykernel.ipkernel import IPythonKernel
+from ipykernel.zmqshell import ZMQInteractiveShell
+from traitlets import Type
 
 from sage.env import SAGE_VERSION, SAGE_EXTCODE, SAGE_DOC
 from sage.repl.interpreter import SageNotebookInteractiveShell
