From 57bd66cabf6e6b9ecf622cdbf350804897a8df58 Mon Sep 17 00:00:00 2001
From: Roland McGrath <roland@hack.frob.com>
Date: Tue, 11 Dec 2012 17:42:07 +0000
Subject: nm: Fix size passed to snprintf for invalid sh_name case.

Signed-off-by: Roland McGrath <roland@hack.frob.com>
---
(limited to 'src/nm.c')

diff --git a/src/nm.c b/src/nm.c
index f50da0b..8a1c57a 100644
--- a/src/nm.c
+++ b/src/nm.c
@@ -769,8 +769,9 @@ show_symbols_sysv (Ebl *ebl, GElf_Word strndx, const char *fullname,
 			       gelf_getshdr (scn, &shdr_mem)->sh_name);
       if (unlikely (name == NULL))
 	{
-	  name = alloca (sizeof "[invalid sh_name 0x12345678]");
-	  snprintf (name, sizeof name, "[invalid sh_name %#" PRIx32 "]",
+          const size_t bufsz = sizeof "[invalid sh_name 0x12345678]"
+	  name = alloca (bufsz);
+	  snprintf (name, bufsz, "[invalid sh_name %#" PRIx32 "]",
 		    gelf_getshdr (scn, &shdr_mem)->sh_name);
 	}
       scnnames[elf_ndxscn (scn)] = name;
--
cgit v0.9.1
From 7df3d2cd70932cd70515dbeb75e4db66fd27f192 Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mjw@redhat.com>
Date: Tue, 11 Dec 2012 21:27:05 +0000
Subject: Add missing semicolon in show_symbols_sysv

Signed-off-by: Mark Wielaard <mjw@redhat.com>
---
(limited to 'src/nm.c')

diff --git a/src/nm.c b/src/nm.c
index 8a1c57a..7aae84b 100644
--- a/src/nm.c
+++ b/src/nm.c
@@ -769,7 +769,7 @@ show_symbols_sysv (Ebl *ebl, GElf_Word strndx, const char *fullname,
 			       gelf_getshdr (scn, &shdr_mem)->sh_name);
       if (unlikely (name == NULL))
 	{
-          const size_t bufsz = sizeof "[invalid sh_name 0x12345678]"
+	  const size_t bufsz = sizeof "[invalid sh_name 0x12345678]";
 	  name = alloca (bufsz);
 	  snprintf (name, bufsz, "[invalid sh_name %#" PRIx32 "]",
 		    gelf_getshdr (scn, &shdr_mem)->sh_name);
--
cgit v0.9.1
From 1a4d0668d18bf1090c5c08cdb5cb3ba2b8eb5410 Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <David.Abdurachmanov@cern.ch>
Date: Sun, 13 Jan 2013 15:44:21 +0000
Subject: ar.c (do_oper_delete): Fix num passed to memset.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>
---
(limited to 'src/ar.c')

diff --git a/src/ar.c b/src/ar.c
index 03da1b7..2d6ad60 100644
--- a/src/ar.c
+++ b/src/ar.c
@@ -919,7 +919,7 @@ do_oper_delete (const char *arfname, char **argv, int argc,
 		long int instance)
 {
   bool *found = alloca (sizeof (bool) * argc);
-  memset (found, '\0', sizeof (found));
+  memset (found, '\0', sizeof (bool) * argc);
 
   /* List of the files we keep.  */
   struct armem *to_copy = NULL;
--
cgit v0.9.1
