diff -Naur viking-1.6.2.orig/src/map_ids.h viking-1.6.2/src/map_ids.h
--- viking-1.6.2.orig/src/map_ids.h	2015-11-10 22:01:35.000000000 +0100
+++ viking-1.6.2/src/map_ids.h	2017-07-01 11:25:32.496994499 +0200
@@ -35,7 +35,6 @@
 #define MAP_ID_OSM_MAPNIK 13
 #define MAP_ID_BLUE_MARBLE 15
 #define MAP_ID_OSM_CYCLE 17
-#define MAP_ID_MAPQUEST_OSM 19
 #define MAP_ID_OSM_TRANSPORT 20
 #define MAP_ID_OSM_ON_DISK 21
 #define MAP_ID_OSM_HUMANITARIAN 22
diff -Naur viking-1.6.2.orig/src/osm.c viking-1.6.2/src/osm.c
--- viking-1.6.2.orig/src/osm.c	2015-11-10 22:01:35.000000000 +0100
+++ viking-1.6.2/src/osm.c	2017-07-01 11:26:25.051880940 +0200
@@ -89,21 +89,6 @@
                                 "license", "CC-BY-SA",
                                 "license-url", "http://www.openstreetmap.org/copyright",
                                 NULL));
-  VikMapSource *mapquest_type =
-    VIK_MAP_SOURCE(g_object_new(VIK_TYPE_SLIPPY_MAP_SOURCE,
-                                "id", MAP_ID_MAPQUEST_OSM,
-                                "name", "OSM-MapQuest",
-                                "label", "OpenStreetMap (MapQuest)",
-                                "hostname", "otile1.mqcdn.com",
-                                "url", "/tiles/1.0.0/osm/%d/%d/%d.png",
-                                "check-file-server-time", TRUE,
-                                "use-etag", FALSE,
-                                "zoom-min", 0,
-                                "zoom-max", 19,
-                                "copyright", "Tiles Courtesy of MapQuest © OpenStreetMap contributors",
-                                "license", "MapQuest Specific",
-                                "license-url", "http://developer.mapquest.com/web/info/terms-of-use",
-                                NULL));
   VikMapSource *hot_type =
     VIK_MAP_SOURCE(g_object_new(VIK_TYPE_SLIPPY_MAP_SOURCE,
                                 "id", MAP_ID_OSM_HUMANITARIAN,
@@ -152,7 +137,6 @@
                                 "is-osm-meta-tiles", TRUE,
                                 NULL));
 
-  maps_layer_register_map_source (mapquest_type);
   maps_layer_register_map_source (mapnik_type);
   maps_layer_register_map_source (cycle_type);
   maps_layer_register_map_source (transport_type);
diff -Naur viking-1.6.2.orig/src/vikmapslayer.c viking-1.6.2/src/vikmapslayer.c
--- viking-1.6.2.orig/src/vikmapslayer.c	2015-12-21 12:44:57.000000000 +0100
+++ viking-1.6.2/src/vikmapslayer.c	2017-07-01 11:25:52.947597509 +0200
@@ -127,7 +127,7 @@
  { 0, 255, 3, 0 }, /* alpha */
 };
 
-static VikLayerParamData id_default ( void ) { return VIK_LPD_UINT ( MAP_ID_MAPQUEST_OSM ); }
+static VikLayerParamData id_default ( void ) { return VIK_LPD_UINT ( MAP_ID_OSM_MAPNIK ); }
 static VikLayerParamData directory_default ( void )
 {
   VikLayerParamData data;
