diff -wbBur flightgear-2016.2.1.org/src/Navaids/PositionedOctree.hxx flightgear-2016.2.1/src/Navaids/PositionedOctree.hxx
--- flightgear-2016.2.1.org/src/Navaids/PositionedOctree.hxx	2016-05-17 11:36:25.000000000 +0300
+++ flightgear-2016.2.1/src/Navaids/PositionedOctree.hxx	2016-05-17 21:20:17.788837738 +0300
@@ -31,6 +31,7 @@
 #include <cassert>
 #include <map>
 #include <functional>
+#include <cmath>
 
 // SimGear
 #include <simgear/math/SGGeometry.hxx>
@@ -60,7 +61,7 @@
     _order(x),
     _inner(v)
     {
-      assert(!isnan(x));
+      assert(!std::isnan(x));
     }
     
     Ordered(const Ordered<T>& a) :
