diff -rub r8168-8.027.00.orig/src/r8168_n.c r8168-8.027.00/src/r8168_n.c
--- r8168-8.027.00.orig/src/r8168_n.c	2011-11-16 13:25:12.000000000 +0100
+++ r8168-8.027.00/src/r8168_n.c	2012-01-19 00:49:11.897863375 +0100
@@ -14282,7 +14282,7 @@
 
 		txd = tp->TxDescArray + entry;
 		len = frag->size;
-		addr = ((void *) page_address(frag->page)) + frag->page_offset;
+		addr = ((void *) page_address(frag->page.p)) + frag->page_offset;
 		mapping = pci_map_single(tp->pci_dev, addr, len, PCI_DMA_TODEVICE);
 
 		/* anti gcc 2.95.3 bugware (sic) */
