diff -wbBur qc-usb-messenger-1.8/qc-driver.c qc-usb-messenger-1.8.my/qc-driver.c
--- qc-usb-messenger-1.8/qc-driver.c	2008-05-26 11:29:16.000000000 +0000
+++ qc-usb-messenger-1.8.my/qc-driver.c	2010-12-16 14:08:06.000000000 +0000
@@ -3476,7 +3494,7 @@
 	release:	qc_v4l_close,
 	read:		qc_v4l_read,
 //	write:		qc_v4l_write,
-	ioctl:		qc_v4l_ioctl,
+	compat_ioctl:		qc_v4l_ioctl,
 	mmap:		qc_v4l_mmap,
 	poll:		qc_v4l_poll,
 };
