diff -rup luxrender-luxblend25-ad6536b0dda8/src/luxrender/__init__.py luxrender-luxblend25-ad6536b0dda8.new/src/luxrender/__init__.py
--- luxrender-luxblend25-ad6536b0dda8/src/luxrender/__init__.py	2013-11-04 22:30:25.000000000 +0100
+++ luxrender-luxblend25-ad6536b0dda8.new/src/luxrender/__init__.py	2013-11-15 11:06:42.435244653 +0100
@@ -50,14 +50,7 @@ else:
 	from nodeitems_utils import NodeCategory, NodeItem, NodeItemCustom
 	
 	def find_luxrender_path():
-		from os import getenv
-		from extensions_framework import util as efutil
-		return getenv(
-			# Use the env var path, if set ...
-			'LUXRENDER_ROOT',
-			# .. or load the last path from CFG file
-			efutil.find_config_value('luxrender', 'defaults', 'install_path', '')
-		)
+		return "/usr/bin"
 		
 	class LuxRenderAddonPreferences(AddonPreferences):
 		# this must match the addon name
