diff --git a/fiddle.gemspec b/fiddle.gemspec
index a9c0ec4..8490764 100644
--- a/fiddle.gemspec
+++ b/fiddle.gemspec
@@ -1,13 +1,8 @@
 # frozen_string_literal: true
 
-version_module = Module.new do
-  version_rb = File.join(__dir__, "lib/fiddle/version.rb")
-  module_eval(File.read(version_rb), version_rb, __LINE__)
-end
-
 Gem::Specification.new do |spec|
   spec.name          = "fiddle"
-  spec.version       = version_module::Fiddle::VERSION
+  spec.version       = "1.1.0"
   spec.authors       = ["Aaron Patterson", "SHIBATA Hiroshi"]
   spec.email         = ["aaron@tenderlovemaking.com", "hsbt@ruby-lang.org"]
 
