--- a/main/main.js
+++ b/main/main.js
@@ -174,7 +174,7 @@
     minHeight: 350,
     titleBarStyle: 'hidden',
     trafficLightPosition: { x: 12, y: 19 },
-    icon: __dirname + '/icons/icon256.png',
+    icon: '/usr/share/icons/hicolor/256x256/apps/min.png',
     frame: process.platform === 'darwin' || settings.get('useSeparateTitlebar') === true,
     alwaysOnTop: settings.get('windowAlwaysOnTop'),
     backgroundColor: '#fff', // the value of this is ignored, but setting it seems to work around https://github.com/electron/electron/issues/10559
