diff -rup luxrender-luxrays-d16b0f132b4a/samples/luxmark/slgdefs.h luxrender-luxrays-d16b0f132b4a.new/samples/luxmark/slgdefs.h
--- luxrender-luxrays-d16b0f132b4a/samples/luxmark/slgdefs.h	2013-10-16 11:16:00.000000000 +0200
+++ luxrender-luxrays-d16b0f132b4a.new/samples/luxmark/slgdefs.h	2013-10-30 20:29:36.768650444 +0100
@@ -41,11 +41,13 @@
 #include "luxrays/core/utils.h"
 #include "luxrays/utils/atomic.h"
 
+#ifndef Q_MOC_RUN
 #include "slg/slg.h"
 #include "slg/rendersession.h"
 #include "slg/renderconfig.h"
 #include "slg/sdl/scene.h"
 #include "slg/film/film.h"
+#endif
 
 #include "mainwindow.h"
 
