From 8afaa090a5f52a9dbde057b79ccf9739bcef8812 Mon Sep 17 00:00:00 2001
From: Oleg Grenrus <oleg.grenrus@iki.fi>
Date: Sat, 6 Feb 2021 11:55:15 +0200
Subject: [PATCH] Allow Cabal-3.4

---
 spdx.cabal          | 6 +++---
 tests/Generators.hs | 4 +++-
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/spdx.cabal b/spdx.cabal
index a9ad636..4174b21 100644
--- a/spdx.cabal
+++ b/spdx.cabal
@@ -49,7 +49,7 @@ library
   ghc-options:      -Wall
   build-depends:
     , base          >=4.3.0.0 && <4.15
-    , Cabal         ^>=2.4.0.1 || ^>=3.0.0.0 || ^>=3.2.0.0
+    , Cabal         ^>=2.4.0.1 || ^>=3.0.0.0 || ^>=3.2.0.0 || ^>=3.4.0.0
     , containers    >=0.5.0.0 && <0.7
     , transformers  >=0.3     && <0.6
 
diff --git a/tests/Generators.hs b/tests/Generators.hs
index 90ef406..1a048d3 100644
--- a/tests/Generators.hs
+++ b/tests/Generators.hs
@@ -15,7 +15,9 @@ import           Distribution.SPDX
 import           Distribution.SPDX.Extra.Internal (LatticeSyntax(..))
 
 latestLicenseListVersion :: LicenseListVersion
-#if MIN_VERSION_Cabal(3,0,0)
+#if MIN_VERSION_Cabal(3,4,0)
+latestLicenseListVersion = LicenseListVersion_3_9
+#elif MIN_VERSION_Cabal(3,0,0)
 latestLicenseListVersion = LicenseListVersion_3_6
 #else
 latestLicenseListVersion = LicenseListVersion_3_2
