--- a/src/installer/corehost/cli/apphost/static/CMakeLists.txt
+++ b/src/installer/corehost/cli/apphost/static/CMakeLists.txt
@@ -204,11 +204,12 @@ target_link_libraries(singlefilehost
     libhostcommon
     ${CORECLR_LIBRARIES}
 
-    ${ZLIB_LIBRARIES}
-    ${LIBGSS}
-    ${NATIVE_LIBS_EXTRA}
-
     ${START_WHOLE_ARCHIVE}
     ${NATIVE_LIBS}
     ${END_WHOLE_ARCHIVE}
+
+    ${ZLIB_LIBRARIES}
+    ${LIBGSS}
+    ${NATIVE_LIBS_EXTRA}
+
 )

