diff -rupN sdk.51369266643769f9f0c1184e89715cd1045126d0.orig/src/Cli/dotnet/Program.cs sdk.51369266643769f9f0c1184e89715cd1045126d0/src/Cli/dotnet/Program.cs
--- sdk.51369266643769f9f0c1184e89715cd1045126d0.orig/src/Cli/dotnet/Program.cs	2021-04-15 07:25:31.290095535 +0000
+++ sdk.51369266643769f9f0c1184e89715cd1045126d0/src/Cli/dotnet/Program.cs	2021-04-15 07:27:01.742066297 +0000
@@ -187,7 +187,7 @@ namespace Microsoft.DotNet.Cli
                         bool generateAspNetCertificate =
                             environmentProvider.GetEnvironmentVariableAsBool("DOTNET_GENERATE_ASPNET_CERTIFICATE", defaultValue: true);
                         bool telemetryOptout =
-                          environmentProvider.GetEnvironmentVariableAsBool("DOTNET_CLI_TELEMETRY_OPTOUT", defaultValue: false);
+                          environmentProvider.GetEnvironmentVariableAsBool("DOTNET_CLI_TELEMETRY_OPTOUT", defaultValue: true);
                         bool addGlobalToolsToPath =
                             environmentProvider.GetEnvironmentVariableAsBool("DOTNET_ADD_GLOBAL_TOOLS_TO_PATH", defaultValue: true);
                         bool nologo =
