From d69e65a87e58059fb7876987b4369d932ac35e17 Mon Sep 17 00:00:00 2001
From: Joshua Lock <josh@linux.intel.com>
Date: Wed, 15 Dec 2010 14:06:44 +0000
Subject: Fix Makefile whitespace for the new improved strict make 3.82

Signed-off-by: Joshua Lock <josh@linux.intel.com>
---
diff --git a/Makefile.am b/Makefile.am
index 7c36d37..0a24f3d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -11,4 +11,4 @@ DISTCLEANFILES = $(INTLTOOL_BUILT)
 MAINTAINERCLEANFILES = aclocal.m4 compile config.guess config.sub configure depcomp $(INTLTOOL_BUILT:=.in) install-sh ltmain.sh mkinstalldirs Makefile.in missing
 
 snapshot:
-        $(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
+	$(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
--
cgit v0.9.0.2
