diff --git a/lily/spacing-spanner.cc b/lily/spacing-spanner.cc
index 99fe982..6949458 100644
--- a/lily/spacing-spanner.cc
+++ b/lily/spacing-spanner.cc
@@ -38,8 +38,8 @@ Spacing_spanner::get_columns (Grob *me_grob)
   vsize end = binary_search (all, (Grob*) me->get_bound (RIGHT),
 			     &Paper_column::less_than);  
   
-  all = vector<Grob*>::vector<Grob*> (all.begin () + start,
-				      all.begin () + end + 1);
+  all = vector<Grob*> (all.begin () + start,
+		       all.begin () + end + 1);
   return all;
 }
 
-- 
