diff --git a/delegate.gemspec b/delegate.gemspec
index 1cfacfe..d26d0fd 100644
--- a/delegate.gemspec
+++ b/delegate.gemspec
@@ -1,15 +1,8 @@
 # frozen_string_literal: true
 
-name = File.basename(__FILE__, ".gemspec")
-version = ["lib", Array.new(name.count("-")+1, "..").join("/")].find do |dir|
-  break File.foreach(File.join(__dir__, dir, "#{name.tr('-', '/')}.rb")) do |line|
-    /^\s*VERSION\s*=\s*"(.*)"/ =~ line and break $1
-  end rescue nil
-end
-
 Gem::Specification.new do |spec|
-  spec.name          = name
-  spec.version       = version
+  spec.name          = "delegate"
+  spec.version       = "0.2.0"
   spec.authors       = ["Yukihiro Matsumoto"]
   spec.email         = ["matz@ruby-lang.org"]
 
