diff --git a/toml.gemspec b/toml.gemspec
index 67f5493..e70c14d 100644
--- a/toml.gemspec
+++ b/toml.gemspec
@@ -30,8 +30,6 @@ Gem::Specification.new do |s|

   s.add_development_dependency "rake"

-  all_files       = `git ls-files -z`.split("\x0")
-  s.files         = all_files.grep(%r{^(bin|lib)/})
-  s.executables   = all_files.grep(%r{^bin/}) { |f| File.basename(f) }
+  s.files         = `find . -print0`.split("\u0000").grep(/.\/lib/)
   s.require_paths = ["lib"]
 end
