--- runtime.f27d33729518f5aa478aa818b7b4f54a4d50bef1.orig/src/installer/corehost/cli/apphost/static/CMakeLists.txt	2021-03-16 15:05:11.692543567 +0000
+++ runtime.f27d33729518f5aa478aa818b7b4f54a4d50bef1/src/installer/corehost/cli/apphost/static/CMakeLists.txt	2021-03-16 15:06:51.581738910 +0000
@@ -207,12 +207,12 @@ target_link_libraries(
     libhostcommon
     ${CORECLR_LIBRARIES}

-    ${ZLIB_LIBRARIES}
-    ${LIBGSS}
-    ${NATIVE_LIBS_EXTRA}
-
     ${START_WHOLE_ARCHIVE}
     ${RUNTIMEINFO_LIB}
     ${NATIVE_LIBS}
     ${END_WHOLE_ARCHIVE}
+
+    ${ZLIB_LIBRARIES}
+    ${LIBGSS}
+    ${NATIVE_LIBS_EXTRA}
 )

