diff --git a/src/external/pac_responder.m4 b/src/external/pac_responder.m4
index dc986a1..114d847 100644
--- a/src/external/pac_responder.m4
+++ b/src/external/pac_responder.m4
@@ -19,7 +19,8 @@ then
         Kerberos\ 5\ release\ 1.14* | \
         Kerberos\ 5\ release\ 1.15* | \
         Kerberos\ 5\ release\ 1.16* | \
-        Kerberos\ 5\ release\ 1.17*)
+        Kerberos\ 5\ release\ 1.17* | \
+        Kerberos\ 5\ release\ 1.18*)
             krb5_version_ok=yes
             AC_MSG_RESULT([yes])
             ;;
