diff --git a/test/snapshot_test.rb b/test/snapshot_test.rb
index a5885dc..6b9f3be 100644
--- a/test/snapshot_test.rb
+++ b/test/snapshot_test.rb
@@ -1,5 +1,4 @@
 require "test_helper"
-require "blankslate"

 class SnapshotTest < Minitest::Test

@@ -47,10 +46,4 @@ class SnapshotTest < Minitest::Test
     instance = Class.new.new
     assert_snapshot_unchanged(instance) {}
   end
-
-  def test_should_cope_with_blankslate_object
-    # Should not raise anything
-    Snapshot.new(BlankSlate.new)
-  end
-
 end
