diff --git a/src/Language/Haskell/TH/Env.hs b/src/Language/Haskell/TH/Env.hs
index 38859e2..62c88fe 100644
--- a/src/Language/Haskell/TH/Env.hs
+++ b/src/Language/Haskell/TH/Env.hs
@@ -8,6 +8,7 @@ where
 
 import Data.String
 import Language.Haskell.TH
+import Language.Haskell.TH.Syntax.Compat
 import System.Environment
 
 -- | Produce a typed expression with the current value of an
@@ -15,8 +16,8 @@ import System.Environment
 envQ :: IsString a
      => String
      -- ^ Environment variable name.
-     -> TExpQ (Maybe a)
-envQ name =
+     -> SpliceQ (Maybe a)
+envQ name = liftSplice $
   runIO (lookupEnv name) >>= \case
-    Just v  -> [|| Just (fromString v) ||]
-    Nothing -> [|| Nothing ||]
+    Just v  -> fromCode $ toCode [|| Just (fromString v) ||]
+    Nothing -> fromCode $ toCode [|| Nothing ||]
