From 99602f63ecabc3533439b7c8af5b09d7f793c4fb Mon Sep 17 00:00:00 2001
From: Ben Darnell <ben@bendarnell.com>
Date: Sat, 13 Feb 2016 19:06:05 -0500
Subject: [PATCH] Disable flaky threaded test

---
 plop/test/collector_test.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/plop/test/collector_test.py b/plop/test/collector_test.py
index d5e22c7..12270cb 100644
--- a/plop/test/collector_test.py
+++ b/plop/test/collector_test.py
@@ -76,7 +76,8 @@ def c(end):
         time_per_sample = float(collector.sample_time) / collector.samples_taken
         self.assertTrue(time_per_sample < 0.000100, time_per_sample)
 
-    def test_collect_threads(self):
+    # TODO: any way to make this test not flaky?
+    def disabled_test_collect_threads(self):
         start = time.time()
         def a(end):
             while time.time() < end: pass
