diff -rupN higan_v094-source.orig/ananke/Makefile higan_v094-source/ananke/Makefile
--- higan_v094-source.orig/ananke/Makefile	2015-01-20 14:56:18.634660183 +0100
+++ higan_v094-source/ananke/Makefile	2015-01-20 15:00:39.164662554 +0100
@@ -2,7 +2,8 @@ include ../nall/Makefile
 include ../phoenix/Makefile
 
 path := /usr/local/lib
-flags := $(flags) -O3 -fomit-frame-pointer -I..
+flags := $(flags) $(CXXFLAGS) -I..
+link := $(link) $(LDFLAGS)
 
 all:
 	$(compiler) $(cppflags) $(flags) -fPIC -o obj/ananke.o -c ananke.cpp
diff -rupN higan_v094-source.orig/Makefile higan_v094-source/Makefile
--- higan_v094-source.orig/Makefile	2015-01-20 14:56:18.634660183 +0100
+++ higan_v094-source/Makefile	2015-01-20 15:06:30.531332416 +0100
@@ -13,8 +13,8 @@ target  := ethos
 # console := true
 
 # compiler
-flags   += -I. -O3 -fomit-frame-pointer
-link    +=
+flags   += $(CXXFLAGS) -I.
+link    += $(LDFLAGS)
 objects := libco
 
 # profile-guided optimization mode
@@ -45,7 +45,7 @@ else ifeq ($(platform),macosx)
   flags += -march=native
 else ifeq ($(platform),linux)
-  flags += -march=native
-  link += -s -Wl,-export-dynamic -lX11 -lXext -ldl
+  flags +=
+  link += -Wl,-export-dynamic -lX11 -lXext -ldl
 else ifeq ($(platform),bsd)
   flags += -march=native
   link += -s -Wl,-export-dynamic -lX11 -lXext
diff -rupN higan_v094-source.orig/nall/Makefile higan_v094-source/nall/Makefile
--- higan_v094-source.orig/nall/Makefile	2015-01-20 14:56:18.641326850 +0100
+++ higan_v094-source/nall/Makefile	2015-01-20 15:04:30.104664654 +0100
@@ -52,13 +52,13 @@ ifeq ($(compiler),)
     flags :=
     link :=
   endif
-
-  cflags := -x c -std=c99
-  objcflags := -x objective-c -std=c99
-  cppflags := -x c++ -std=c++11
-  objcppflags := -x objective-c++ -std=c++11
 endif
 
+cflags := -x c -std=c99
+objcflags := -x objective-c -std=c99
+cppflags := -x c++ -std=c++11
+objcppflags := -x objective-c++ -std=c++11
+
 # cross-compilation support
 ifeq ($(arch),x86)
   flags := -m32 $(flags)
