diff -wbBur sysstat-10.1.6.q/configure.in sysstat-10.1.6/configure.in
--- sysstat-10.1.6.q/configure.in	2013-04-21 18:20:32.000000000 +0400
+++ sysstat-10.1.6/configure.in	2013-06-17 14:45:41.733915298 +0400
@@ -207,13 +207,6 @@
 fi
 
 SADC_DIR=$AuxPrefix/lib
-if test -d $AuxPrefix/lib64; then
-   # Look for lm (long mode) flag to know if CPU is 64 bit
-   grep " lm " /proc/cpuinfo >/dev/null 2>&1
-   if test $? = 0; then
-      SADC_DIR=$AuxPrefix/lib64
-   fi
-fi
 
 AC_MSG_CHECKING(sadc directory)
 AC_ARG_VAR([sa_lib_dir],[sadc directory])
