diff -wbBur ibam-0.5.2/ibam.inl ibam-0.5.2.my/ibam.inl
--- ibam-0.5.2/ibam.inl	2009-03-09 16:53:20.000000000 +0000
+++ ibam-0.5.2.my/ibam.inl	2011-03-24 19:41:32.000000000 +0000
@@ -959,7 +959,13 @@
       acpi.close();
       apm = new acpi_status();
    } else {
+      sysfs.open((sysfs_path+"/BAT0/charge_full").c_str());
+      if(!sysfs.is_open()) {
       sysfs.open((sysfs_path+"/BAT1/charge_full").c_str());
+      }
+      if(!sysfs.is_open()) {
+          sysfs.open((sysfs_path+"/BAT2/charge_full").c_str());
+      }
       if(sysfs.is_open()) {
 #ifdef IBAM_DEBUG
       cout << "using sysfs" << endl;
