diff --git a/Rakefile b/Rakefile
index 24425f3..165c995 100644
--- a/Rakefile
+++ b/Rakefile
@@ -40,31 +40,6 @@ task :test do
   Dir['./test/**/test_*.rb'].each {|f| require f }
 end

-desc "Generate RCov test coverage and open in your browser"
-task :coverage do
-  if RUBY_VERSION =~ /^1\./
-    require 'rubygems'
-    require 'bundler'
-    Bundler.setup(:test)
-    require 'simplecov'
-    require 'simplecov-gem-adapter'
-
-    sh "rm -fr coverage"
-    SimpleCov.command_name 'Unit Tests'
-    SimpleCov.start 'gem'
-    Rake::Task[:test].invoke
-    SimpleCov.at_exit do
-      SimpleCov.result.format!
-      sh "open coverage/index.html"
-    end
-  else
-    require 'rcov'
-    sh "rm -fr coverage"
-    sh "rcov test/test_*.rb"
-    sh "open coverage/index.html"
-  end
-end
-
 require 'rdoc/task'
 Rake::RDocTask.new do |rdoc|
   rdoc.rdoc_dir = 'rdoc'
