diff --git a/toml.gemspec b/toml.gemspec
index ffe2492..292c3ce 100644
--- a/toml.gemspec
+++ b/toml.gemspec
@@ -26,12 +26,10 @@ Gem::Specification.new do |s|
   s.rdoc_options = ["--charset=UTF-8"]
   s.extra_rdoc_files = %w[README.md LICENSE CHANGELOG.md]

-  s.add_dependency "parslet", "~> 1.8.0"
+  s.add_dependency "parslet", ">= 1.8.0"

   s.add_development_dependency "rake"

-  all_files       = `git ls-files -z`.split("\x0")
-  s.files         = all_files.grep(%r{^(bin|lib)/})
-  s.executables   = all_files.grep(%r{^bin/}) { |f| File.basename(f) }
+  s.files         = `find . -print0`.split("\u0000").grep(/.\/lib/)
   s.require_paths = ["lib"]
 end
