diff -wbBur apache-couchdb-1.2.2/src/couchdb/couch_app.erl apache-couchdb-1.2.2.q/src/couchdb/couch_app.erl
--- apache-couchdb-1.2.2/src/couchdb/couch_app.erl	2013-03-24 00:33:33.000000000 +0400
+++ apache-couchdb-1.2.2.q/src/couchdb/couch_app.erl	2013-06-21 19:30:30.083579366 +0400
@@ -20,7 +20,7 @@
 
 start(_Type, DefaultIniFiles) ->
     IniFiles = get_ini_files(DefaultIniFiles),
-    case start_apps([crypto, public_key, sasl, inets, oauth, ssl, ibrowse, mochiweb, os_mon]) of
+    case start_apps([crypto, asn1, public_key, sasl, inets, oauth, ssl, ibrowse, mochiweb, os_mon]) of
     ok ->
         couch_server_sup:start_link(IniFiles);
     {error, Reason} ->
