--- a/chromiumcontent/args/native_mksnapshot.gn
+++ b/chromiumcontent/args/native_mksnapshot.gn
@@ -12,3 +12,8 @@
 if (target_cpu == "arm64") {
   v8_snapshot_toolchain="//build/toolchain/linux:clang_arm64"
 }
+clang_base_path = "/usr"
+clang_use_chrome_plugins = false
+treat_warnings_as_errors = false
+use_custom_libcxx = false
+use_sysroot = false
--- a/chromiumcontent/args/static_library.gn
+++ b/chromiumcontent/args/static_library.gn
@@ -1,13 +1,24 @@
 root_extra_deps = [ "//chromiumcontent:chromiumcontent" ]
 is_electron_build = true
 is_component_build = false
-is_official_build = true
-symbol_level = 2
+is_debug = false
+symbol_level = 0
 enable_nacl = false
 enable_widevine = true
 proprietary_codecs = true
 is_component_ffmpeg = true
 ffmpeg_branding = "Chrome"
+clang_base_path = "/usr"
+clang_use_chrome_plugins = false
+fatal_linker_warnings = false
+treat_warnings_as_errors = false
+fieldtrial_testing_like_official_build = true
+remove_webcore_debug_symbols = true
+link_pulseaudio = true
+linux_use_bundled_binutils = false
+use_cups = true
+use_gconf = false
+use_sysroot = false
 
 # CFI is disabled for the time being, as Electron is not a monolithic binary
 # with at least one shared library component (Node) and CFI is tricky in that
