# Copyright 2015 Séastien Luttringer
--- a/configure.in	2015-09-18 02:55:59.825544232 +0200
+++ b/configure.in	2015-09-18 03:23:33.544191972 +0200
@@ -19,15 +19,7 @@
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
 
-dnl Figure out where to get res_query from. First, see if it's in the
-dnl C library (Linux, and probably a few others). Then, check libbind
-dnl (generally, if you've installed bind 8, you're going to want to use
-dnl the new library). Finally, check for libresolv (which should exist
-dnl on most systems).
-AC_CHECK_FUNC(res_query,,
-	AC_CHECK_LIB(resolv,res_query,,
-		AC_CHECK_LIB(bind,res_query,,
-			AC_MSG_ERROR(cannot locate res_query function))))
+AC_CHECK_LIB(resolv, __res_query)
 
 dnl All done.
 AC_OUTPUT(Makefile docs/Makefile config/rblcheck.spec config/pkginfo)
