--- autoconvert-0.3.16/Makefile	2001-04-28 09:45:46.000000000 +0800
+++ autoconvert-0.3.16-fix/Makefile	2011-05-01 11:25:23.432934576 +0800
@@ -5,7 +5,7 @@
 
 .PHONY: all hzlib clean install-home install
 
-all: autob5 autogb hzlib  xchat-plugins
+all: autob5 autogb hzlib
 
 autob5: autogb
 	rm -f autob5;ln -s autogb autob5
@@ -15,8 +15,6 @@
 	$(CC) $(CFLAG) autogb.c lib/libhz.a -o autogb
 hzlib:
 	cd hzconvert;make
-xchat-plugins: 
-	cd contrib/xchat-plugins;make
 clean:
 	rm -f autob5 autogb *.o *~
 	rm -f include/*~
@@ -30,12 +28,13 @@
 	ln -s $(HOME)/bin/autogb $(HOME)/bin/autob5
 	chmod 755 $(HOME)/bin/autogb 
 install:
+	mkdir -p $(DESTDIR)/usr/bin
+	mkdir -p $(DESTDIR)/usr/lib
+	mkdir -p $(DESTDIR)/usr/include
 	install -m 755 autogb $(DESTDIR)/usr/bin
 	install -m 755 lib/libhz.so.0.0 $(DESTDIR)/usr/lib
 	cd $(DESTDIR)/usr/lib;ln -s  libhz.so.0.0 libhz.so.0
 	cd $(DESTDIR)/usr/lib;ln -s  libhz.so.0 libhz.so
-	install -m 755 lib/libhz.a $(DESTDIR)/usr/lib
-	install -m 755 contrib/xchat-plugins/xchat-autogb.so $(DESTDIR)/usr/share/zh-autoconvert
-	install -m 755 contrib/xchat-plugins/xchat-autob5.so $(DESTDIR)/usr/share/zh-autoconvert
+	install -m 644 lib/libhz.a $(DESTDIR)/usr/lib
 	install -m 644 include/*.h  $(DESTDIR)/usr/include
 	ln -s autogb $(DESTDIR)/usr/bin/autob5
