diff --git a/Data/Configurator/Parser/Implementation.hs.orig b/Data/Configurator/Parser/Implementation.hs
index 3b3c82a..be9080f 100644
--- a/Data/Configurator/Parser/Implementation.hs.orig
+++ b/Data/Configurator/Parser/Implementation.hs
@@ -152,9 +152,8 @@ newtype ConfigTransform = ConfigTransform (ConfigPlan ())
 
 -- | 'mempty' is the identity 'ConfigTransform',  'mappend' is the composition
 --   of two 'ConfigTransform's.
-instance Monoid ConfigTransform where
-   mempty = ConfigTransform (ConfigPlan ())
-   (ConfigTransform x) `mappend` (ConfigTransform y) = (ConfigTransform (go x))
+instance Semigroup ConfigTransform where
+  ConfigTransform x <> ConfigTransform y = ConfigTransform (go x)
      where
        go (ConfigPlan _)      = y
        go (Union a b)         = Union (go a) (go b)
@@ -162,6 +161,9 @@ instance Monoid ConfigTransform where
        go (Subconfig pre a)   = Subconfig pre (go a)
        go Empty               = Empty
 
+instance Monoid ConfigTransform where
+   mempty = ConfigTransform (ConfigPlan ())
+
 -- | Conceptually,  @'union' f g = \\config -> union\' (f config) (g config)@,
 -- where @union\'@ is the left-biased union of two 'Config's.
 union :: ConfigTransform -> ConfigTransform -> ConfigTransform
