diff -ruN a/src/headers/gx_system.h b/src/headers/gx_system.h
--- a/src/headers/gx_system.h	2021-12-10 18:53:33.000000000 +0100
+++ b/src/headers/gx_system.h	2022-04-28 11:23:16.176694581 +0200
@@ -112,7 +112,11 @@
 }
 
 inline bool atomic_compare_and_exchange(volatile int *p, int oldv, int newv) {
+#if (GLIB_MAJOR_VERSION == 2 && GLIB_MINOR_VERSION < 68 && GLIB_MICRO_VERSION < 1)
     return g_atomic_int_compare_and_exchange(p, oldv, newv);
+#else
+    return g_atomic_int_compare_and_exchange(const_cast<int*>(p), oldv, newv);
+#endif
 }
 
 template <class T>
