Index: src/mod_archive_webview.erl
===================================================================
--- src/mod_archive_webview.erl	(revision 1109)
+++ src/mod_archive_webview.erl	(working copy)
@@ -102,7 +102,7 @@
     make_xhtml(?T("Chat with ") ++ Jid, contact_config(Jid,US,Lang) ++
                            [?XE("ul", lists:map( fun({Id, Node, Server, Resource, Utc, Subject }) -> 
                                                     With = jlib:jid_to_string({Node,Server,Resource}),
-                                                    ?LI([?AC(?LINK("show/" ++ integer_to_list(Id)), "On " ++ Utc ++ " with " ++ With ++ " -> " ++ escape_str(Subject)  )] ) end,
+                                                    ?LI([?AC(?LINK("show/" ++ Id), "On " ++ Utc ++ " with " ++ With ++ " -> " ++ escape_str(Subject)  )] ) end,
                                                 get_collection_list(jlib:string_to_jid(Jid), US)))
                ], Lang);
 
@@ -209,7 +209,9 @@
 format_message({ Utc, Dir, Body } ,{WithU,WithS,WithR}, {LUser,LServer} ) ->
     {From, Class} = case Dir of 
         0 -> { jlib:jid_to_string({WithU,WithS,WithR}) , "message_from" } ;
-        1 -> { jlib:jid_to_string({LUser,LServer,""}) , "message_to" } 
+        1 -> { jlib:jid_to_string({LUser,LServer,""}) , "message_to" };
+        "0" -> { jlib:jid_to_string({WithU,WithS,WithR}) , "message_from" } ;
+        "1" -> { jlib:jid_to_string({LUser,LServer,""}) , "message_to" } 
     end,
     [_Date, Time] = string:tokens(Utc, " "),
     ?XAE("p", [{"class", Class}] , [ ?XAE("span", [{"class","time"}], [?C("["++Time++"]")]), ?C(" "),
@@ -422,7 +424,7 @@
 links_previous_next_aux(Class, Text, Id) ->
     case Id of
         -1 -> [];
-        _ -> [?XAE("a",[{"href",?LINK("show/" ++ integer_to_list(Id))},{"class",Class}], [?C(Text)])]
+        _ -> [?XAE("a",[{"href",?LINK("show/" ++ Id)},{"class",Class}], [?C(Text)])]
     end.
 
 %------------------------
