diff -Naur gnome-schedule-2.2.0.orig/src/atEditor.py gnome-schedule-2.2.0/src/atEditor.py
--- gnome-schedule-2.2.0.orig/src/atEditor.py	2011-11-08 16:18:03.000000000 +0100
+++ gnome-schedule-2.2.0/src/atEditor.py	2012-05-09 14:37:31.042776449 +0200
@@ -20,7 +20,6 @@
 #pygtk modules
 import gtk
 import gobject
-import gnome
 
 #python modules
 import os
diff -Naur gnome-schedule-2.2.0.orig/src/crontabEditor.py gnome-schedule-2.2.0/src/crontabEditor.py
--- gnome-schedule-2.2.0.orig/src/crontabEditor.py	2010-11-03 11:39:09.000000000 +0100
+++ gnome-schedule-2.2.0/src/crontabEditor.py	2012-05-09 14:37:31.049442720 +0200
@@ -20,7 +20,6 @@
 #pygtk modules
 import gtk
 import gobject
-import gnome
 
 #python modules
 import re
diff -Naur gnome-schedule-2.2.0.orig/src/gnome-schedule.py gnome-schedule-2.2.0/src/gnome-schedule.py
--- gnome-schedule-2.2.0.orig/src/gnome-schedule.py	2011-06-11 14:28:00.000000000 +0200
+++ gnome-schedule-2.2.0/src/gnome-schedule.py	2012-05-09 14:40:51.947536161 +0200
@@ -58,9 +58,6 @@
 try:
   import gtk
   import gtk.glade
-  # TODO: Gnome specific
-  import gnome
-  import gnome.ui
 
 except:
   print _("You need to install pyGTK or GTKv2,\n"
@@ -68,8 +65,5 @@
           "try: export PYTHONPATH= ")
   sys.exit(1)
 
-props = { gnome.PARAM_APP_DATADIR : config.getPrefix() + "/share" }
-pr = gnome.program_init ("gnome-schedule", config.getVersion(), properties=props)
-
-mainWindow = mainWindow.main(debug_flag, False, pr, manual_poscorrect)
+mainWindow = mainWindow.main(debug_flag, False, manual_poscorrect)
 
diff -Naur gnome-schedule-2.2.0.orig/src/mainWindow.py gnome-schedule-2.2.0/src/mainWindow.py
--- gnome-schedule-2.2.0.orig/src/mainWindow.py	2011-06-10 10:14:48.000000000 +0200
+++ gnome-schedule-2.2.0/src/mainWindow.py	2012-05-09 14:44:42.297281138 +0200
@@ -22,15 +22,13 @@
 import gtk.glade
 import gobject
 
-# gnome specific
-import gnome
-from gnome import url_show
 
 #python modules
 import os
 import pwd
 import tempfile
 import stat
+import subprocess
 
 #custom modules
 import config
@@ -52,10 +50,9 @@
 ## The MainWindow class
 ##
 class main:
-    def __init__(self, debug_flag=None, inapplet=False, gprogram = None, manual_poscorrect=False):
+    def __init__(self, debug_flag=None, inapplet=False, manual_poscorrect=False):
         self.debug_flag = debug_flag
         self.inapplet = inapplet
-        self.gprogram = gprogram
         self.manual_poscorrect = manual_poscorrect
 
         self.__loadIcon__()
@@ -646,7 +643,7 @@
 
     #about box
     def open_url (self, *args):
-        url_show("http://gnome-schedule.sourceforge.net")
+        gtk.show_uri(None, "http://gnome-schedule.sourceforge.net", 0)
 
 
 
@@ -717,7 +714,7 @@
             # unset POSIXLY_CORRECT if manually set, bug 612459
             if self.manual_poscorrect: os.unsetenv ('POSIXLY_CORRECT')
 
-            gnome.execute_terminal_shell (self.user_home_dir, execute)
+            subprocess.Popen('gnome-terminal --working-directory="' + self.user_home_dir + '" -e "' + execute + '"', shell=True)
 
             if self.manual_poscorrect: os.putenv ('POSIXLY_CORRECT', 'enabled')
 
@@ -761,9 +758,7 @@
     #open help
     def on_manual_menu_activate (self, *args):
         try:
-            gnome.help_display (
-                    'gnome-schedule',
-                    None)
+            gtk.show_uri(None, 'ghelp:gnome-schedule', 0)
         except gobject.GError, error:
             dialog = gtk.MessageDialog (
                     self.widget,
diff -Naur gnome-schedule-2.2.0.orig/src/scheduleapplet.py gnome-schedule-2.2.0/src/scheduleapplet.py
--- gnome-schedule-2.2.0.orig/src/scheduleapplet.py	2011-11-08 16:27:57.000000000 +0100
+++ gnome-schedule-2.2.0/src/scheduleapplet.py	2012-05-09 14:46:15.591795456 +0200
@@ -58,8 +58,6 @@
   import gtk
   import gtk.glade
   # TODO: Gnome specific
-  import gnome
-  import gnome.ui
   import gnomeapplet
   import gobject
 
@@ -69,13 +67,10 @@
           "try: export PYTHONPATH= ")
   sys.exit(1)
 
-props = { gnome.PARAM_APP_DATADIR : config.getPrefix() + "/share"}
-pr = gnome.program_init ("gnome-schedule", config.getVersion(), properties=props)
-
 
 
 class ScheduleApplet(gnomeapplet.Applet):
-    def __init__(self, applet, iid, gprogram, debug_flag, manual_poscorrect):
+    def __init__(self, applet, iid, debug_flag, manual_poscorrect):
         self.__gobject_init__()
         self.debug_flag = debug_flag
         self.manual_poscorrect = manual_poscorrect
@@ -88,7 +83,6 @@
 
 
         self.applet = applet
-        self.gprogram = gprogram
         self.__loadIcon__()
 
 
@@ -150,7 +144,7 @@
     def show_main_window(self, *args):
         if self.main_loaded == False:
             self.main_loaded = True
-            self.main_window = mainWindow.main(None, True, self.gprogram, self.manual_poscorrect)
+            self.main_window = mainWindow.main(None, True, self.manual_poscorrect)
         else:
             self.main_window.widget.show ()
             self.main_window.schedule_reload()
@@ -179,7 +173,7 @@
 
 #factory
 def schedule_applet_factory(applet, iid):
-    ScheduleApplet(applet, iid, pr, debug_flag, manual_poscorrect)
+    ScheduleApplet(applet, iid, debug_flag, manual_poscorrect)
     return True
 
 gnomeapplet.bonobo_factory("OAFIID:GNOME_GnomeSchedule_Factory",
