Index: gui/com/frostwire/UpdateManager.java
===================================================================
--- gui/com/frostwire/UpdateManager.java	(revision 1177)
+++ gui/com/frostwire/UpdateManager.java	(working copy)
@@ -28,6 +28,7 @@
 
 import javax.swing.JOptionPane;
 
+import org.limewire.util.CommonUtils;
 import org.limewire.util.OSUtils;
 
 import org.xml.sax.Attributes;
@@ -350,7 +351,7 @@
 
     private void loadSeenMessages() {
         //deserializes seen messages if available
-        File f = new File("seenMessages.dat");
+        File f = new File(CommonUtils.getUserSettingsDir(), "seenMessages.dat");
         _seenMessages = new HashSet<UpdateMessage>();
 
         if (!f.exists()) {
@@ -380,7 +381,7 @@
         if (_seenMessages == null || _seenMessages.size() < 1)
             return;
 
-        File f = new File("seenMessages.dat");
+        File f = new File(CommonUtils.getUserSettingsDir(), "seenMessages.dat");
         if (!f.exists()) {
             try { 
                 f.createNewFile(); 
