From 9217f054b7eccc120f84e01995479125e07de59a Mon Sep 17 00:00:00 2001
From: meejah <meejah@meejah.ca>
Date: Fri, 22 Feb 2019 12:01:46 -0700
Subject: [PATCH] make pytest happy

---
 test/conftest.py | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/test/conftest.py b/test/conftest.py
index 4a225c8..a3ca2d6 100644
--- a/test/conftest.py
+++ b/test/conftest.py
@@ -25,9 +25,9 @@ def framework(request):
 
     try:
         if request.param == 'twisted':
-            return framework_tx()
+            return _notfixture_framework_tx()
         elif request.param == 'asyncio':
-            return framework_aio()
+            return _notfixture_framework_aio()
     except ImportError:
         pytest.skip()
 
@@ -43,6 +43,10 @@ def framework_uninitialized():
 
 @pytest.fixture
 def framework_tx():
+    return _notfixture_framework_tx()
+
+
+def _notfixture_framework_tx():
     try:
         import txaio
         from txaio import tx
@@ -56,6 +60,10 @@ def framework_tx():
 
 @pytest.fixture
 def framework_aio():
+    return _notfixture_framework_aio()
+
+
+def _notfixture_framework_aio():
     try:
         import txaio
         from txaio import aio
