diff -upr mixxx-2.3.1.orig/lib/benchmark/src/benchmark_register.h mixxx-2.3.1/lib/benchmark/src/benchmark_register.h
--- mixxx-2.3.1.orig/lib/benchmark/src/benchmark_register.h	2021-09-29 16:23:02.000000000 +0300
+++ mixxx-2.3.1/lib/benchmark/src/benchmark_register.h	2021-12-22 23:15:18.935803347 +0200
@@ -1,6 +1,7 @@
 #ifndef BENCHMARK_REGISTER_H
 #define BENCHMARK_REGISTER_H
 
+#include <limits>
 #include <vector>
 
 #include "check.h"
