--- mixxx-2.0.0/src/musicbrainz/chromaprinter.cpp.old	2017-01-13 14:24:15.105139156 +0100
+++ mixxx-2.0.0/src/musicbrainz/chromaprinter.cpp	2017-01-13 14:24:06.568450904 +0100
@@ -57,12 +57,12 @@ QString ChromaPrinter::calcFingerPrint(c
     }
     chromaprint_finish(ctx);
 
-    void* fprint = NULL;
+    uint32_t* fprint = NULL;
     int size = 0;
     int ret = chromaprint_get_raw_fingerprint(ctx, &fprint, &size);
     QByteArray fingerprint;
     if (ret == 1) {
-        void* encoded = NULL;
+        char* encoded = NULL;
         int encoded_size = 0;
         chromaprint_encode_fingerprint(fprint, size,
                                        CHROMAPRINT_ALGORITHM_DEFAULT,
