--- src/Text/Pandoc/Readers/Haddock.hs.orig	2021-03-22 06:09:57.716035128 +0800
+++ src/Text/Pandoc/Readers/Haddock.hs	2021-03-22 06:11:17.853110456 +0800
@@ -128,7 +128,7 @@
           DocIdentifier s -> B.codeWith ("",["haskell","identifier"],[]) $ T.pack s
           _               -> mempty
     DocIdentifierUnchecked s -> B.codeWith ("",["haskell","identifier"],[]) $ T.pack s
-    DocModule s -> B.codeWith ("",["haskell","module"],[]) $ T.pack s
+    DocModule (H.ModLink s _) -> B.codeWith ("",["haskell","module"],[]) $ T.pack s
     DocWarning _ -> mempty -- TODO
     DocEmphasis d -> B.emph (docHToInlines isCode d)
     DocMonospaced (DocString s) -> B.code $ T.pack s
