diff --git a/src/dbus-common.c b/src/dbus-common.c
index fe7f84b..9bf0dab 100644
--- a/src/dbus-common.c
+++ b/src/dbus-common.c
@@ -569,15 +569,15 @@ int bus_property_append_ul(DBusMessageIter *i, const char *property, void *data)
 }
 
 int bus_property_append_long(DBusMessageIter *i, const char *property, void *data) {
-        uint64_t u;
+        int64_t l;
 
         assert(i);
         assert(property);
         assert(data);
 
-        u = (int64_t) *(long*) data;
+        l = (int64_t) *(long*) data;
 
-        if (!dbus_message_iter_append_basic(i, DBUS_TYPE_INT64, &u))
+        if (!dbus_message_iter_append_basic(i, DBUS_TYPE_INT64, &l))
                 return -ENOMEM;
 
         return 0;
diff --git a/src/dbus-socket.c b/src/dbus-socket.c
index b81569d..7194611 100644
--- a/src/dbus-socket.c
+++ b/src/dbus-socket.c
@@ -115,8 +115,8 @@ DBusHandlerResult bus_socket_message_handler(Unit *u, DBusConnection *c, DBusMes
                 { "org.freedesktop.systemd1.Socket", "MaxConnections", bus_property_append_unsigned,     "u", &u->socket.max_connections },
                 { "org.freedesktop.systemd1.Socket", "NConnections",   bus_property_append_unsigned,     "u", &u->socket.n_connections   },
                 { "org.freedesktop.systemd1.Socket", "NAccepted",      bus_property_append_unsigned,     "u", &u->socket.n_accepted      },
-                { "org.freedesktop.systemd1.Socket", "MessageQueueMaxMessages", bus_property_append_long,"t", &u->socket.mq_maxmsg       },
-                { "org.freedesktop.systemd1.Socket", "MessageQueueMessageSize", bus_property_append_long,"t", &u->socket.mq_msgsize      },
+                { "org.freedesktop.systemd1.Socket", "MessageQueueMaxMessages", bus_property_append_long,"x", &u->socket.mq_maxmsg       },
+                { "org.freedesktop.systemd1.Socket", "MessageQueueMessageSize", bus_property_append_long,"x", &u->socket.mq_msgsize      },
                 { NULL, NULL, NULL, NULL, NULL }
         };
 
