diff -urN playonlinux.orig/playonlinux playonlinux/playonlinux
--- playonlinux.orig/playonlinux	2012-08-06 01:19:22.021641671 +0300
+++ playonlinux/playonlinux	2012-08-06 01:20:13.255750423 +0300
@@ -69,7 +69,7 @@
 
 [ "$(uname -s)" = "Linux" ] || die "This script must be run on a linux system"
 
-export PYTHON="python"
+export PYTHON="python2"
 export POL_OS="Linux"
 export UBUNTU_MENUPROXY=0 
 export MACHTYPE
@@ -141,4 +141,4 @@
 fi
 
 [ $exitcode -eq 63 ] && exec "$POLDIR/playonlinux"
-exit $exitcode
\ No newline at end of file
+exit $exitcode
diff -urN playonlinux.orig/playonlinux-bash playonlinux/playonlinux-bash
--- playonlinux.orig/playonlinux-bash	2012-08-06 01:19:22.021641671 +0300
+++ playonlinux/playonlinux-bash	2012-08-06 01:20:31.839363354 +0300
@@ -37,7 +37,7 @@
 	CURDIR="$(pwd)"
 	export MACHTYPE
 	cd "$WorkingDirectory"
-	python "$CURDIR/python/wrapper.py" "$SCRIPTNAME" "$@"
+	python2 "$CURDIR/python/wrapper.py" "$SCRIPTNAME" "$@"
 else
 	bash "$SCRIPTNAME" "$@"
 fi
diff -urN playonlinux.orig/playonlinux-pkg playonlinux/playonlinux-pkg
--- playonlinux.orig/playonlinux-pkg	2012-08-06 01:19:22.021641671 +0300
+++ playonlinux/playonlinux-pkg	2012-08-06 01:20:39.892817581 +0300
@@ -29,7 +29,7 @@
 	CURDIR="$(pwd)"
 	export MACHTYPE
 	cd "$WorkingDirectory"
-	python "$CURDIR/python/wrapper.py" "$CURDIR/bash/$(basename "$0")" "$@"
+	python2 "$CURDIR/python/wrapper.py" "$CURDIR/bash/$(basename "$0")" "$@"
 else
 	bash "$PLAYONLINUX/bash/$(basename "$0")" "$@"
 fi
diff -urN playonlinux.orig/playonlinux-url_handler playonlinux/playonlinux-url_handler
--- playonlinux.orig/playonlinux-url_handler	2012-08-06 01:19:22.021641671 +0300
+++ playonlinux/playonlinux-url_handler	2012-08-06 01:20:49.886300702 +0300
@@ -29,7 +29,7 @@
 	CURDIR="$(pwd)"
 	export MACHTYPE
 	cd "$WorkingDirectory"
-	python "$CURDIR/python/wrapper.py" "$CURDIR/bash/$(basename "$0")" "$@"
+	python2 "$CURDIR/python/wrapper.py" "$CURDIR/bash/$(basename "$0")" "$@"
 else
 	bash "$PLAYONLINUX/bash/$(basename "$0")" "$@"
 fi