From 9201b4708a90dc1f841285c6ec47814f6ed17495 Mon Sep 17 00:00:00 2001
From: Emin Mastizada <emin@linux.com>
Date: Wed, 16 Oct 2019 17:47:27 +0300
Subject: [PATCH] Support pytest5 syntax for tests

diff --git a/tests/test_clean.py b/tests/test_clean.py
index e306cc5..d041ffa 100644
--- a/tests/test_clean.py
+++ b/tests/test_clean.py
@@ -24,11 +24,11 @@ def test_only_text_is_cleaned():
 
     with pytest.raises(TypeError) as e:
         clean(some_type)
-    assert "argument cannot be of 'type' type" in str(e)
+    assert "argument cannot be of 'type' type" in str(e.value)
 
     with pytest.raises(TypeError) as e:
         clean(no_type)
-    assert "NoneType" in str(e)
+    assert "NoneType" in str(e.value)
 
 
 def test_empty():
