diff -rupN scratch-text-editor-2.3.orig/plugins/browser-preview/CMakeLists.txt scratch-text-editor-2.3/plugins/browser-preview/CMakeLists.txt
--- scratch-text-editor-2.3.orig/plugins/browser-preview/CMakeLists.txt	2017-01-19 23:08:49.613757781 +0100
+++ scratch-text-editor-2.3/plugins/browser-preview/CMakeLists.txt	2017-01-19 23:11:17.183287592 +0100
@@ -1,4 +1,4 @@
-pkg_check_modules (WEBKIT REQUIRED "webkitgtk-3.0")
+pkg_check_modules (WEBKIT webkitgtk-3.0)
 if(${WEBKIT_FOUND})
 add_definitions(${NORMAL_CFLAGS})
 include_directories(${CMAKE_BINARY_DIR} ${CMAKE_BINARY_DIR}/scratchcore/)
