From 838b600fead202416013db5c1b57f7031f06bed6 Mon Sep 17 00:00:00 2001
From: Mark Guptill <compguy284@yahoo.com>
Date: Tue, 30 Jun 2020 04:34:32 -0400
Subject: [PATCH] kick tray clients before destroying the bar

---
 i3bar/src/xcb.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/i3bar/src/xcb.c b/i3bar/src/xcb.c
index 94212630..5f01c1d8 100644
--- a/i3bar/src/xcb.c
+++ b/i3bar/src/xcb.c
@@ -1613,6 +1613,8 @@ void destroy_window(i3_output *output) {
         return;
     }
 
+    kick_tray_clients(output);
+
     draw_util_surface_free(xcb_connection, &(output->bar));
     draw_util_surface_free(xcb_connection, &(output->buffer));
     draw_util_surface_free(xcb_connection, &(output->statusline_buffer));
@@ -1620,8 +1622,6 @@ void destroy_window(i3_output *output) {
     xcb_free_pixmap(xcb_connection, output->buffer.id);
     xcb_free_pixmap(xcb_connection, output->statusline_buffer.id);
     output->bar.id = XCB_NONE;
-
-    kick_tray_clients(output);
 }
 
 /* Strut partial tells i3 where to reserve space for i3bar. This is determined
-- 
2.28.0

