From b095db986c7a22c03e2766180e053817563ef244 Mon Sep 17 00:00:00 2001
From: Dave Reisner <d@falconindy.com>
Date: Mon, 25 Apr 2011 17:06:51 -0400
Subject: [PATCH] ipc: fixup for yajl 2.0 API changes

---
 src/ipc.c |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/ipc.c b/src/ipc.c
index 1937d55..ab0a265 100644
--- a/src/ipc.c
+++ b/src/ipc.c
@@ -182,7 +182,7 @@ IPC_HANDLER(get_workspaces) {
         if (last_focused == SLIST_END(&(c_ws->focus_stack)))
                 last_focused = NULL;
 
-        yajl_gen gen = yajl_gen_alloc(NULL, NULL);
+        yajl_gen gen = yajl_gen_alloc(NULL);
         y(array_open);
 
         TAILQ_FOREACH(ws, workspaces, workspaces) {
@@ -226,7 +226,7 @@ IPC_HANDLER(get_workspaces) {
         y(array_close);
 
         const unsigned char *payload;
-        unsigned int length;
+        size_t length;
         y(get_buf, &payload, &length);
 
         ipc_send_message(fd, payload, I3_IPC_REPLY_TYPE_WORKSPACES, length);
@@ -241,7 +241,7 @@ IPC_HANDLER(get_workspaces) {
 IPC_HANDLER(get_outputs) {
         Output *output;
 
-        yajl_gen gen = yajl_gen_alloc(NULL, NULL);
+        yajl_gen gen = yajl_gen_alloc(NULL);
         y(array_open);
 
         TAILQ_FOREACH(output, &outputs, outputs) {
@@ -276,7 +276,7 @@ IPC_HANDLER(get_outputs) {
         y(array_close);
 
         const unsigned char *payload;
-        unsigned int length;
+        size_t length;
         y(get_buf, &payload, &length);
 
         ipc_send_message(fd, payload, I3_IPC_REPLY_TYPE_OUTPUTS, length);
@@ -338,7 +338,7 @@ IPC_HANDLER(subscribe) {
         memset(&callbacks, 0, sizeof(yajl_callbacks));
         callbacks.yajl_string = add_subscription;
 
-        p = yajl_alloc(&callbacks, NULL, NULL, (void*)client);
+        p = yajl_alloc(&callbacks, NULL, (void*)client);
         stat = yajl_parse(p, (const unsigned char*)message, message_size);
         if (stat != yajl_status_ok) {
                 unsigned char *err;
-- 
1.7.4.4

