--- buildbot/master/buildbot/test/__init__.py	2019-10-10 20:56:41.191340974 +0800
+++ buildbot/master/buildbot/test/__init__.py	2019-10-10 20:57:49.008283272 +0800
@@ -119,3 +119,7 @@
 
 # sqlalchemy-migrate uses deprecated api from sqlalchemy https://review.openstack.org/#/c/648072/
 warnings.filterwarnings('ignore', ".*Engine.contextual_connect.*", DeprecationWarning)
+
+# automat not fully compatible with attrs 19.2 yet
+# https://github.com/glyph/automat/issues/117
+warnings.filterwarnings('ignore', ".*The usage of `cmp` is deprecated.*", DeprecationWarning)
