diff -ruN a/dnscrypt-proxy/example-dnscrypt-proxy.toml b/dnscrypt-proxy/example-dnscrypt-proxy.toml
--- a/dnscrypt-proxy/example-dnscrypt-proxy.toml	2019-11-18 13:00:34.000000000 +0100
+++ b/dnscrypt-proxy/example-dnscrypt-proxy.toml	2019-11-19 22:11:15.890730311 +0100
@@ -138,12 +138,12 @@
 
 ## log file for the application
 
-# log_file = 'dnscrypt-proxy.log'
+# log_file = '/var/log/dnscrypt-proxy/dnscrypt-proxy.log'
 
 
 ## Use the system logger (syslog on Unix, Event Log on Windows)
 
-# use_syslog = true
+use_syslog = true
 
 
 ## Delay, in minutes, after which certificates are reloaded
@@ -280,7 +280,7 @@
 ## example.com 9.9.9.9
 ## example.net 9.9.9.9,8.8.8.8,1.1.1.1
 
-# forwarding_rules = 'forwarding-rules.txt'
+# forwarding_rules = '/etc/dnscrypt-proxy/forwarding-rules.txt'
 
 
 
@@ -296,7 +296,7 @@
 ## example.com     10.1.1.1
 ## www.google.com  forcesafesearch.google.com
 
-# cloaking_rules = 'cloaking-rules.txt'
+# cloaking_rules = '/etc/dnscrypt-proxy/cloaking-rules.txt'
 
 ## TTL used when serving entries in cloaking-rules.txt
 
@@ -349,7 +349,7 @@
   ## Path to the query log file (absolute, or relative to the same directory as the executable file)
   ## Can be /dev/stdout to log to the standard output (and set log_files_max_size to 0)
 
-  # file = 'query.log'
+  # file = '/var/log/dnscrypt-proxy/query.log'
 
 
   ## Query log format (currently supported: tsv and ltsv)
@@ -375,7 +375,7 @@
 
   ## Path to the query log file (absolute, or relative to the same directory as the executable file)
 
-  # file = 'nx.log'
+  # file = '/var/log/dnscrypt-proxy/nx.log'
 
 
   ## Query log format (currently supported: tsv and ltsv)
@@ -405,12 +405,12 @@
 
   ## Path to the file of blocking rules (absolute, or relative to the same directory as the executable file)
 
-  # blacklist_file = 'blacklist.txt'
+  # blacklist_file = '/etc/dnscrypt-proxy/blacklist.txt'
 
 
   ## Optional path to a file logging blocked queries
 
-  # log_file = 'blocked.log'
+  # log_file = '/var/log/dnscrypt-proxy/blocked.log'
 
 
   ## Optional log format: tsv or ltsv (default: tsv)
@@ -433,12 +433,12 @@
 
   ## Path to the file of blocking rules (absolute, or relative to the same directory as the executable file)
 
-  # blacklist_file = 'ip-blacklist.txt'
+  # blacklist_file = '/etc/dnscrypt-proxy/ip-blacklist.txt'
 
 
   ## Optional path to a file logging blocked queries
 
-  # log_file = 'ip-blocked.log'
+  # log_file = '/var/log/dnscrypt-proxy/ip-blocked.log'
 
 
   ## Optional log format: tsv or ltsv (default: tsv)
@@ -461,12 +461,12 @@
 
   ## Path to the file of whitelisting rules (absolute, or relative to the same directory as the executable file)
 
-  # whitelist_file = 'whitelist.txt'
+  # whitelist_file = '/etc/dnscrypt-proxy/whitelist.txt'
 
 
   ## Optional path to a file logging whitelisted queries
 
-  # log_file = 'whitelisted.log'
+  # log_file = '/var/log/dnscrypt-proxy/whitelisted.log'
 
 
   ## Optional log format: tsv or ltsv (default: tsv)
@@ -536,7 +536,7 @@
 
   [sources.'public-resolvers']
   urls = ['https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v2/public-resolvers.md', 'https://download.dnscrypt.info/resolvers-list/v2/public-resolvers.md']
-  cache_file = 'public-resolvers.md'
+  cache_file = '/var/cache/dnscrypt-proxy/public-resolvers.md'
   minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'
   prefix = ''
 
@@ -544,7 +544,7 @@
 
   [sources.'relays']
   urls = ['https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v2/relays.md', 'https://download.dnscrypt.info/resolvers-list/v2/relays.md']
-  cache_file = 'relays.md'
+  cache_file = '/var/cache/dnscrypt-proxy/relays.md'
   minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'
   refresh_delay = 72
   prefix = ''
@@ -554,7 +554,7 @@
   # [sources.quad9-resolvers]
   # urls = ['https://www.quad9.net/quad9-resolvers.md']
   # minisign_key = 'RWQBphd2+f6eiAqBsvDZEBXBGHQBJfeG6G+wJPPKxCZMoEQYpmoysKUN'
-  # cache_file = 'quad9-resolvers.md'
+  # cache_file = '/var/cache/dnscrypt-proxy/quad9-resolvers.md'
   # prefix = 'quad9-'
 
   ## Another example source, with resolvers censoring some websites not appropriate for children
@@ -562,7 +562,7 @@
 
   #  [sources.'parental-control']
   #  urls = ['https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v2/parental-control.md', 'https://download.dnscrypt.info/resolvers-list/v2/parental-control.md']
-  #  cache_file = 'parental-control.md'
+  #  cache_file = '/var/cache/dnscrypt-proxy/parental-control.md'
   #  minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'
 
 
