diff --git a/example_project/settings.py b/example_project/settings.py
index 5aca0186..aad3e4f5 100644
--- a/example_project/settings.py
+++ b/example_project/settings.py
@@ -157,6 +157,9 @@ DATABASES = {
     #}
 }
 
+# Set default type of primary key (feature introduced with django 3.2)
+# https://docs.djangoproject.com/en/3.2/releases/3.2/#customizing-type-of-auto-created-primary-keys
+DEFAULT_AUTO_FIELD = 'django.db.models.AutoField'
 
 # If you're behind a proxy, use the X-Forwarded-Host header
 # See https://docs.djangoproject.com/en/1.8/ref/settings/#use-x-forwarded-host
diff --git a/hyperkitty/tests/settings_test.py b/hyperkitty/tests/settings_test.py
index 71da81c3..ab271a3b 100644
--- a/hyperkitty/tests/settings_test.py
+++ b/hyperkitty/tests/settings_test.py
@@ -149,6 +149,9 @@ DATABASES = {
     # }
 }
 
+# Set default type of primary key (feature introduced with django 3.2)
+# https://docs.djangoproject.com/en/3.2/releases/3.2/#customizing-type-of-auto-created-primary-keys
+DEFAULT_AUTO_FIELD = 'django.db.models.AutoField'
 
 # If you're behind a proxy, use the X-Forwarded-Host header
 # See https://docs.djangoproject.com/en/1.8/ref/settings/#use-x-forwarded-host
-- 
GitLab

