diff -ruN a/cloudinit/distros/parsers/sys_conf.py b/cloudinit/distros/parsers/sys_conf.py
--- a/cloudinit/distros/parsers/sys_conf.py	2021-08-23 19:43:37.000000000 +0200
+++ b/cloudinit/distros/parsers/sys_conf.py	2021-08-24 13:29:30.646813586 +0200
@@ -101,7 +101,7 @@
         cmnt = self._decode_element(comment)
         return '%s%s%s%s%s' % (indent_string,
                                key,
-                               self._a_to_u('='),
+                               "=",
                                val,
                                cmnt)
 
