diff --git c/pyproject.toml w/pyproject.toml
index 2f2aa27..8e28a5f 100644
--- c/pyproject.toml
+++ w/pyproject.toml
@@ -1,9 +1,7 @@
 [build-system]
 requires = [
   "setuptools >= 45.0.0", # required by pyproject+setuptools_scm integration
-  "setuptools_scm >= 6.3.1", # required for "no-local-version" scheme
-  "setuptools_scm_git_archive >= 1.0",
-  "wheel",
+  "setuptools_scm[toml] >= 7.0.0", # required for "no-local-version" scheme
 ]
 build-backend = "setuptools.build_meta"
 
