=== modified file 'src/lightdm-gtk-greeter.c'
--- src/lightdm-gtk-greeter.c	2014-02-16 18:37:06 +0000
+++ src/lightdm-gtk-greeter.c	2014-02-22 11:31:35 +0000
@@ -1241,11 +1241,10 @@
     set_user_image (username);
     user = lightdm_user_list_get_user_by_name (lightdm_user_list_get_instance (), username);
     if (user)
-        if (lightdm_user_get_logged_in (user))
-        {
-            set_language (lightdm_user_get_language (user));
-            set_session (lightdm_user_get_session (user));
-        }
+    {
+        set_language (lightdm_user_get_language (user));
+        set_session (lightdm_user_get_session (user));
+    }
     gtk_widget_set_tooltip_text (GTK_WIDGET (user_combo), user_tooltip);
     start_authentication (username);
     g_free (user_tooltip);
