diff --git i/Makefile.am w/Makefile.am
index a2ec6f42..236d01e3 100644
--- i/Makefile.am
+++ w/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = . gdk-pixbuf-loader tests doc win32
+SUBDIRS = . gdk-pixbuf-loader tests win32
 
 NULL =
 
@@ -137,9 +137,7 @@ else
 CARGO_RELEASE_ARGS=--release
 endif
 
-if CROSS_COMPILING
 CARGO_TARGET_ARGS=--target=$(RUST_TARGET)
-endif
 
 CARGO_VERBOSE = $(cargo_verbose_$(V))
 cargo_verbose_ = $(cargo_verbose_$(AM_DEFAULT_VERBOSITY))
@@ -228,13 +226,6 @@ $(RSVG_CONVERT_BIN): $(RSVG_CONVERT_SRC)
 rsvg-convert$(EXEEXT): $(RSVG_CONVERT_BIN)
 	cd $(LIBRSVG_BUILD_DIR) && mv $(RSVG_CONVERT_BIN) rsvg-convert$(EXEEXT)
 
-rsvg-convert.1: rsvg-convert.rst
-	$(RST2MAN) $(top_srcdir)/rsvg-convert.rst rsvg-convert.1
-
-man1_MANS = rsvg-convert.1
-
-CLEANFILES += rsvg-convert.1
-
 dist_doc_DATA =				\
 	COMPILING.md			\
 	CONTRIBUTING.md			\
diff --git i/configure.ac w/configure.ac
index 0bc55612..07b0cea4 100644
--- i/configure.ac
+++ w/configure.ac
@@ -113,20 +113,6 @@ AX_COMPARE_VERSION([$rust_version],[lt],[$MINIMUM_RUST_VER], [
     AC_MSG_ERROR([librsvg needs at least rustc version $MINIMUM_RUST_VER])
 ])
 
-dnl Man page generator
-
-AC_CHECK_PROGS(RST2MAN, [rst2man rst2man.py], [no])
-AS_IF(test x$RST2MAN = xno,
-    AC_MSG_ERROR([rst2man is required.  Please install python3-docutils.])
-)
-
-dnl Gi-docgen
-
-AC_CHECK_TOOL(GI_DOCGEN, [gi-docgen], [no])
-AS_IF(test x$GI_DOCGEN = xno,
-    AC_MSG_ERROR([gi-docgen is required.  Please install it.])
-)
-
 dnl ===========================================================================
 
 #AC_CANONICAL_HOST
@@ -276,9 +262,7 @@ AC_ARG_VAR(RUST_TARGET, [Set rust target (default=$host)])
 : ${RUST_TARGET:=$host}
 
 AM_CONDITIONAL([CROSS_COMPILING], [test "x$cross_compiling" = xyes])
-AS_IF([test "x$cross_compiling" = "xyes"], [
-	RUST_TARGET_SUBDIR="$RUST_TARGET/$RUST_TARGET_SUBDIR"
-])
+RUST_TARGET_SUBDIR="$RUST_TARGET/$RUST_TARGET_SUBDIR"
 AC_SUBST([RUST_TARGET_SUBDIR])
 
 dnl ===========================================================================
