 Makefile.am  | 2 --
 configure.ac | 4 +---
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git c/Makefile.am i/Makefile.am
index e93714e4..effe5d3d 100644
--- c/Makefile.am
+++ i/Makefile.am
@@ -137,9 +137,7 @@ else
 CARGO_RELEASE_ARGS=--release
 endif
 
-if CROSS_COMPILING
 CARGO_TARGET_ARGS=--target=$(RUST_TARGET)
-endif
 
 CARGO_VERBOSE = $(cargo_verbose_$(V))
 cargo_verbose_ = $(cargo_verbose_$(AM_DEFAULT_VERBOSITY))
diff --git c/configure.ac i/configure.ac
index 41590ca9..84cd56bd 100644
--- c/configure.ac
+++ i/configure.ac
@@ -255,9 +255,7 @@ AC_ARG_VAR(RUST_TARGET, [Set rust target (default=$host)])
 : ${RUST_TARGET:=$host}
 
 AM_CONDITIONAL([CROSS_COMPILING], [test "x$cross_compiling" = xyes])
-AS_IF([test "x$cross_compiling" = "xyes"], [
-	RUST_TARGET_SUBDIR="$RUST_TARGET/$RUST_TARGET_SUBDIR"
-])
+RUST_TARGET_SUBDIR="$RUST_TARGET/$RUST_TARGET_SUBDIR"
 AC_SUBST([RUST_TARGET_SUBDIR])
 
 dnl ===========================================================================
