--- L-1.23/include/Lcommon.h.orig	2011-04-09 04:09:25.984121006 -0300
+++ L-1.23/include/Lcommon.h	2011-04-09 04:09:58.750265003 -0300
@@ -25,6 +25,8 @@ inline double lcalc_to_double(const Doub
 #ifdef USE_MPFR
 inline double lcalc_to_double(const double& x) { return x; }
 #endif
+
+#if 0
 //inline double lcalc_to_double(const long double& x) { return x; }
 inline double lcalc_to_double(const int& x) { return x; }
 inline double lcalc_to_double(const long long& x) { return x; }
@@ -33,6 +35,10 @@ inline double lcalc_to_double(const char
 inline double lcalc_to_double(const long int& x) { return x; }
 inline double lcalc_to_double(const unsigned int& x) { return x; }
 inline double lcalc_to_double(const long unsigned int& x) { return x; }
+#else
+#  define lcalc_to_double(x)	((double)(x))
+#endif
+
 #define Int(x) (int)(lcalc_to_double(x))
 #define Long(x) (Long)(lcalc_to_double(x))
 #define double(x) (double)(lcalc_to_double(x))
 