diff -rup luxrender-luxblend25-a249c861c2f3/src/luxrender/extensions_framework/util.py luxrender-luxblend25-a249c861c2f3.new/src/luxrender/extensions_framework/util.py
--- luxrender-luxblend25-a249c861c2f3/src/luxrender/extensions_framework/util.py	2015-01-27 20:48:23.000000000 +0100
+++ luxrender-luxblend25-a249c861c2f3.new/src/luxrender/extensions_framework/util.py	2015-02-16 10:35:51.456810705 +0100
@@ -99,7 +99,7 @@ def find_config_value(module, section, k
     fc = []
 
     for p in config_paths:
-        if os.path.exists(p) and os.path.isdir(p) and os.access(p, os.W_OK):
+        if os.path.exists(p) and os.path.isdir(p):
             fc.append('/'.join([p, '%s.cfg' % module]))
 
     if len(fc) < 1:
