diff -wbBur freewrl-1.22.6/src/bin/Makefile.in freewrl-1.22.6.my/src/bin/Makefile.in
--- freewrl-1.22.6/src/bin/Makefile.in	2009-12-31 18:36:32.000000000 +0000
+++ freewrl-1.22.6.my/src/bin/Makefile.in	2010-01-25 13:55:13.000000000 +0000
@@ -266,7 +266,7 @@
 
 # Compiler warnings: 
 # http://gcc.gnu.org/onlinedocs/gcc-4.3.2/gcc/Preprocessor-Options.html#Preprocessor-Options
-CC_WARNINGS = -Wall -Wundef -Wunused-macros
+CC_WARNINGS = -Wundef -Wunused-macros
 
 # Compiler options / general flags
 
diff -wbBur freewrl-1.22.6/src/lib/scenegraph/readpng.c freewrl-1.22.6.my/src/lib/scenegraph/readpng.c
--- freewrl-1.22.6/src/lib/scenegraph/readpng.c	2009-10-05 15:07:24.000000000 +0000
+++ freewrl-1.22.6.my/src/lib/scenegraph/readpng.c	2010-01-25 13:48:42.000000000 +0000
@@ -106,7 +106,7 @@
      * have used slightly more general png_sig_cmp() function instead */
 
     rv=fread(sig, 1, 8, infile);
-    if (!png_check_sig(sig, 8))
+    if (!png_sig_cmp(sig, 0, 8))
         return 1;   /* bad signature */
 
 
diff -wbBur freewrl-1.22.6/src/lib/world_script/jsVRMLBrowser.c freewrl-1.22.6.my/src/lib/world_script/jsVRMLBrowser.c
--- freewrl-1.22.6/src/lib/world_script/jsVRMLBrowser.c	2009-12-10 20:51:55.000000000 +0000
+++ freewrl-1.22.6.my/src/lib/world_script/jsVRMLBrowser.c	2010-01-25 13:50:59.000000000 +0000
@@ -107,7 +107,7 @@
 
 /* for setting field values to the output of a CreateVrml style of call */
 /* it is kept at zero, unless it has been used. Then it is reset to zero */
-jsval JSCreate_global_return_val = INT_TO_JSVAL(0);
+jsval JSCreate_global_return_val = 0;
 
 /* we add/remove routes with this call */
 void jsRegisterRoute(
diff -wbBur freewrl-1.22.6/src/lib/x3d_parser/X3DParser.c freewrl-1.22.6.my/src/lib/x3d_parser/X3DParser.c
--- freewrl-1.22.6/src/lib/x3d_parser/X3DParser.c	2009-12-09 18:28:51.000000000 +0000
+++ freewrl-1.22.6.my/src/lib/x3d_parser/X3DParser.c	2010-01-25 13:50:11.000000000 +0000
@@ -54,7 +54,7 @@
 #include "X3DProtoScript.h"
 
 #if HAVE_EXPAT_H
-# include <expat.h>
+# include </usr/include/expat.h>
 #endif
 
 #define PROTO_MARKER 567000
diff -wbBur freewrl-1.22.6/src/plugin/npunix.c freewrl-1.22.6.my/src/plugin/npunix.c
--- freewrl-1.22.6/src/plugin/npunix.c	2009-10-05 15:07:24.000000000 +0000
+++ freewrl-1.22.6.my/src/plugin/npunix.c	2010-01-25 13:57:47.000000000 +0000
@@ -45,6 +45,12 @@
 
 #define XP_UNIX 1
 
+#include <stdint.h>
+#define int32 int32_t
+#define uint32 uint32_t
+#define int16 int16_t
+#define uint16 uint16_t
+
 #include <stdio.h>
 #include "npapi.h"
 #if (((NP_VERSION_MAJOR << 8) + NP_VERSION_MINOR) < 20)
@@ -377,7 +383,7 @@
        NPP_URLNotify(instance, url, reason, notifyData);
 }
 
-JRIGlobalRef
+void*
 Private_GetJavaClass(void)
 {
 /*JAS 
diff -wbBur freewrl-1.22.6/src/plugin/plugin_main.c freewrl-1.22.6.my/src/plugin/plugin_main.c
--- freewrl-1.22.6/src/plugin/plugin_main.c	2009-10-05 15:07:24.000000000 +0000
+++ freewrl-1.22.6.my/src/plugin/plugin_main.c	2010-01-25 13:54:14.000000000 +0000
@@ -42,6 +42,12 @@
 #include <config.h>
 #include <system.h>
 
+#include <stdint.h>
+#define int32 int32_t
+#define uint32 uint32_t
+#define int16 int16_t
+#define uint16 uint16_t
+
 #include <plugin_utils.h>
 #include <npapi.h>
 
