diff -ruN pkg-orig//usr/bin/gufw pkg/usr/bin/gufw
--- pkg-orig//usr/bin/gufw	2010-05-15 23:21:29.000000000 +0200
+++ pkg/usr/bin/gufw	2010-06-26 22:45:03.000000000 +0200
@@ -1,4 +1,4 @@
 #!/bin/bash
 # This script starts Gufw from the console
-su-to-root -X -c /usr/share/gufw/gufw.py
+gksudo --preserve-env /usr/share/gufw/gufw.py
 
diff -ruN pkg-orig//usr/share/applications/gufw.desktop pkg/usr/share/applications/gufw.desktop
--- pkg-orig//usr/share/applications/gufw.desktop	2010-05-15 23:21:29.000000000 +0200
+++ pkg/usr/share/applications/gufw.desktop	2010-06-26 22:45:29.000000000 +0200
@@ -2,7 +2,7 @@
 Version=1.0
 Type=Application
 Terminal=false
-Exec=su-to-root -X -c /usr/share/gufw/gufw.py
+Exec=gksudo --preserve-env /usr/share/gufw/gufw.py
 Name=Firewall configuration
 Name[zh_CN]=防火墙配置
 Name[uk]=Налаштування фаєрволу
