--- anki-2.1.8/aqt/qt.py.orig	2019-02-12 21:32:36.515462520 +0000
+++ anki-2.1.8/aqt/qt.py	2019-02-12 21:32:45.032117131 +0000
@@ -13,7 +13,7 @@
 from PyQt5.Qt import *
 # trigger explicit message in case of missing libraries
 # instead of silently failing to import
-from PyQt5.QtWebEngineWidgets import QWebEnginePage
+from PyQt5.QtWebEngineWidgets import *
 try:
     from PyQt5 import sip
 except ImportError:
