diff -U 3 -d -r -N -- bbpager-0.4.7/configure.ac bbpager-0.4.7b/configure.ac
--- bbpager-0.4.7/configure.ac	2008-01-14 18:03:50.000000000 +0100
+++ bbpager-0.4.7b/configure.ac	2010-07-17 10:36:04.042040059 +0200
@@ -68,7 +68,7 @@
 dnl pkgconfig checks for blackbox library
 PKG_CHECK_MODULES(LIBBT, libbt)
 CXXFLAGS="$CXXFLAGS $LIBBT_CFLAGS"
-LIBS="$LIBS $LIBBT_LIBS"
+LIBS="$LIBS $LIBBT_LIBS -lXext -lX11"
 
 dnl generate the config header
 AM_CONFIG_HEADER(config.h) 
