diff -Naur variety-0.6.3.orig/variety/indicator.py variety-0.6.3/variety/indicator.py
--- variety-0.6.3.orig/variety/indicator.py	2016-11-01 19:59:39.000000000 +0100
+++ variety-0.6.3/variety/indicator.py	2017-01-27 17:09:16.537550484 +0100
@@ -316,8 +316,8 @@
         self.status_icon = None
         self.visible = True
 
-        def pos(menu, icon):
-            return Gtk.StatusIcon.position_menu(self.menu, icon)
+        def pos(menu, x, y, icon):
+            return Gtk.StatusIcon.position_menu(self.menu, x, y, icon)
 
         def right_click_event(icon, button, time):
             self.menu.popup(None, None, pos, self.status_icon, 0, time)
diff -Naur variety-0.6.3.orig/variety/PreferencesVarietyDialog.py variety-0.6.3/variety/PreferencesVarietyDialog.py
--- variety-0.6.3.orig/variety/PreferencesVarietyDialog.py	2016-11-01 19:59:39.000000000 +0100
+++ variety-0.6.3/variety/PreferencesVarietyDialog.py	2017-01-27 17:18:14.865800288 +0100
@@ -357,7 +357,7 @@
             timer.start()
 
     def on_add_button_clicked(self, widget=None):
-        def position(x, y):
+        def position(menu, x, y, icon):
             button_alloc = self.ui.add_button.get_allocation()
             window_pos = self.ui.add_button.get_window().get_position()
             return button_alloc.x + window_pos[0], button_alloc.y + button_alloc.height + window_pos[1], True
@@ -365,7 +365,7 @@
         self.add_menu.popup(None, self.ui.add_button, position, None, 0, Gtk.get_current_event_time())
 
     def on_remove_sources_clicked(self, widget=None):
-        def position(x, y):
+        def position(menu, x, y, icon):
             button_alloc = self.ui.remove_sources.get_allocation()
             window_pos = self.ui.remove_sources.get_window().get_position()
             return button_alloc.x + window_pos[0], button_alloc.y + button_alloc.height + window_pos[1], True
diff -Naur variety-0.6.3.orig/variety/ThumbsManager.py variety-0.6.3/variety/ThumbsManager.py
--- variety-0.6.3.orig/variety/ThumbsManager.py	2016-11-01 19:59:39.000000000 +0100
+++ variety-0.6.3/variety/ThumbsManager.py	2017-01-27 17:24:33.175557472 +0100
@@ -268,7 +268,7 @@
             _resume_scrolling()
         else:
             menu = self.create_menu(file)
-            def _compute_position(a, b, event=event):
+            def _compute_position(menu, x, y, icon):
                 x, y = event.get_root_coords()[0], event.get_root_coords()[1]
                 h = menu.get_preferred_height()[1]
                 return x, y - h if y - h >= 40 else y, True
