diff -Naur ocrfeeder-0.7.11.orig/src/ocrfeeder/feeder/imageManipulation.py ocrfeeder-0.7.11/src/ocrfeeder/feeder/imageManipulation.py
--- ocrfeeder-0.7.11.orig/src/ocrfeeder/feeder/imageManipulation.py	2012-10-02 20:00:33.000000000 +0200
+++ ocrfeeder-0.7.11/src/ocrfeeder/feeder/imageManipulation.py	2013-09-15 15:28:11.779102147 +0200
@@ -21,7 +21,7 @@
 from ocrfeeder.util.lib import debug
 
 import gettext
-import Image, ImageDraw
+from PIL import Image, ImageDraw
 import os.path
 from ocrfeeder.util import graphics
 import sys
diff -Naur ocrfeeder-0.7.11.orig/src/ocrfeeder/feeder/layoutAnalysis.py ocrfeeder-0.7.11/src/ocrfeeder/feeder/layoutAnalysis.py
--- ocrfeeder-0.7.11.orig/src/ocrfeeder/feeder/layoutAnalysis.py	2013-02-08 08:57:12.000000000 +0100
+++ ocrfeeder-0.7.11/src/ocrfeeder/feeder/layoutAnalysis.py	2013-09-15 15:28:11.779102147 +0200
@@ -23,7 +23,7 @@
 from ocrfeeder.util.constants import OCRFEEDER_DEBUG, DTP
 from ocrfeeder.studio.dataHolder import DataBox
 from imageManipulation import ImageProcessor
-import Image
+from PIL import Image
 import re
 import math
 
diff -Naur ocrfeeder-0.7.11.orig/src/ocrfeeder/studio/dataHolder.py ocrfeeder-0.7.11/src/ocrfeeder/studio/dataHolder.py
--- ocrfeeder-0.7.11.orig/src/ocrfeeder/studio/dataHolder.py	2013-02-08 08:57:12.000000000 +0100
+++ ocrfeeder-0.7.11/src/ocrfeeder/studio/dataHolder.py	2013-09-15 15:27:20.787882011 +0200
@@ -20,7 +20,7 @@
 
 import os.path
 from ocrfeeder.util import graphics
-import Image
+from PIL import Image
 import gobject
 from ocrfeeder.util.graphics import getImagePrintSize, getImageResolution
 from ocrfeeder.util import TEXT_TYPE, IMAGE_TYPE, ALIGN_LEFT, lib
diff -Naur ocrfeeder-0.7.11.orig/src/ocrfeeder/studio/widgetPresenter.py ocrfeeder-0.7.11/src/ocrfeeder/studio/widgetPresenter.py
--- ocrfeeder-0.7.11.orig/src/ocrfeeder/studio/widgetPresenter.py	2013-02-08 08:57:12.000000000 +0100
+++ ocrfeeder-0.7.11/src/ocrfeeder/studio/widgetPresenter.py	2013-09-15 15:28:11.775768733 +0200
@@ -26,7 +26,7 @@
 from ocrfeeder.util.constants import *
 from ocrfeeder.util.graphics import convertPixbufToImage
 from enchant.checker import SpellChecker
-import Image
+from PIL import Image
 import gettext
 import gobject
 import goocanvas
diff -Naur ocrfeeder-0.7.11.orig/src/ocrfeeder/util/graphics.py ocrfeeder-0.7.11/src/ocrfeeder/util/graphics.py
--- ocrfeeder-0.7.11.orig/src/ocrfeeder/util/graphics.py	2012-10-02 20:00:33.000000000 +0200
+++ ocrfeeder-0.7.11/src/ocrfeeder/util/graphics.py	2013-09-15 15:28:11.775768733 +0200
@@ -19,7 +19,7 @@
 ###########################################################################
 
 from lib import debug
-import Image
+from PIL import Image
 import gtk
 import math
 
diff -Naur ocrfeeder-0.7.11.orig/src/ocrfeeder/util/lib.py ocrfeeder-0.7.11/src/ocrfeeder/util/lib.py
--- ocrfeeder-0.7.11.orig/src/ocrfeeder/util/lib.py	2013-02-08 08:57:12.000000000 +0100
+++ ocrfeeder-0.7.11/src/ocrfeeder/util/lib.py	2013-09-15 15:28:11.775768733 +0200
@@ -20,7 +20,7 @@
 
 import os
 import mimetypes
-import Image
+from PIL import Image
 import tempfile
 import gtk
 import math
