diff -wbBur Pantomime/Framework/Pantomime/NSString+Extensions.m Pantomime.my/Framework/Pantomime/NSString+Extensions.m
--- Pantomime/Framework/Pantomime/NSString+Extensions.m	2006-09-04 20:04:02.000000000 +0000
+++ Pantomime.my/Framework/Pantomime/NSString+Extensions.m	2011-05-03 09:49:54.000000000 +0000
@@ -41,7 +41,7 @@
 #include <CoreFoundation/CFString.h>
 #include <CoreFoundation/CFStringEncodingExt.h>
 #else
-#include <GNUstepBase/GSCategories.h>
+//#include <GNUstepBase/GSCategories.h>
 #endif
 
 #include <ctype.h>
