diff -rup pion-net-4.0.12/common/include/pion/PluginManager.hpp pion-net-4.0.12-new/common/include/pion/PluginManager.hpp
--- pion-net-4.0.12/common/include/pion/PluginManager.hpp	2011-02-12 01:46:50.000000000 +0100
+++ pion-net-4.0.12-new/common/include/pion/PluginManager.hpp	2012-07-07 16:54:22.470284487 +0200
@@ -420,7 +420,7 @@ inline void PluginManager<PLUGIN_TYPE>::
 				delete i->second.first;
 			}
 		}
-		erase(std::map<std::string, std::pair<PLUGIN_TYPE *, PionPluginPtr<PLUGIN_TYPE> > >::begin(),
+		this->erase(std::map<std::string, std::pair<PLUGIN_TYPE *, PionPluginPtr<PLUGIN_TYPE> > >::begin(),
 			  std::map<std::string, std::pair<PLUGIN_TYPE *, PionPluginPtr<PLUGIN_TYPE> > >::end());
 	}
 }
