diff -ru pion-net-4.0.5/net/include/pion/net/TCPConnection.hpp pion-net-4.0.5.fixed/net/include/pion/net/TCPConnection.hpp
--- pion-net-4.0.5/net/include/pion/net/TCPConnection.hpp	2010-12-21 03:07:13.000000000 +0100
+++ pion-net-4.0.5.fixed/net/include/pion/net/TCPConnection.hpp	2011-07-31 09:01:38.445559768 +0200
@@ -639,7 +639,7 @@
 	
 	/// returns reference to the io_service used for async operations
 	inline boost::asio::io_service& getIOService(void) {
-		return m_ssl_socket.lowest_layer().io_service();
+		return m_ssl_socket.lowest_layer().get_io_service();
 	}
 
 	/// returns non-const reference to underlying TCP socket object
