diff --git i/libs/qm-dsp/wscript w/libs/qm-dsp/wscript
index 775098e06b..13f6b9410a 100644
--- i/libs/qm-dsp/wscript
+++ w/libs/qm-dsp/wscript
@@ -23,17 +23,10 @@ def options(opt):
     autowaf.set_options(opt)
 
 def configure(conf):
-    if conf.is_defined('USE_EXTERNAL_LIBS'):
-        conf.check_cxx(header_name="base/Pitch.h", mandatory=True)
-        conf.check_cxx(lib="qm-dsp", uselib_store="QMDSP", mandatory=True)
-    else:
-        conf.load('compiler_cxx')
-        autowaf.configure(conf)
+    conf.load('compiler_cxx')
+    autowaf.configure(conf)
 
 def build(bld):
-    if bld.is_defined('USE_EXTERNAL_LIBS'):
-        return
-
     # Host Library
     obj = bld(features = 'c cxx cxxstlib')
     obj.source = '''
diff --git i/wscript w/wscript
index 1c9bb1ff5d..95d377d8a7 100644
--- i/wscript
+++ w/wscript
@@ -1041,8 +1041,6 @@ def configure(conf):
 
     if Options.options.use_external_libs:
         conf.define('USE_EXTERNAL_LIBS', 1)
-        conf.env.append_value(
-            'CXXFLAGS', '-I' + Options.options.qm_dsp_include)
 
     if Options.options.boost_include != '':
         conf.env.append_value('CXXFLAGS', '-I' + Options.options.boost_include)
