diff -Naur gnome-panel-3.6.2.orig/configure.ac gnome-panel-3.6.2/configure.ac
--- gnome-panel-3.6.2.orig/configure.ac	2012-11-12 18:51:14.000000000 +0100
+++ gnome-panel-3.6.2/configure.ac	2013-04-10 14:50:14.132509059 +0200
@@ -1,6 +1,6 @@
 AC_INIT([gnome-panel], [3.6.2],
         [http://bugzilla.gnome.org/enter_bug.cgi?product=gnome-panel])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE([1.11 no-dist-gzip dist-xz tar-ustar -Wno-portability])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
