diff -upr proxytunnel-1.9.0.orig/ptstream.c proxytunnel-1.9.0/ptstream.c
--- proxytunnel-1.9.0.orig/ptstream.c	2008-02-27 01:27:27.000000000 +0200
+++ proxytunnel-1.9.0/ptstream.c	2016-03-02 20:43:32.909153946 +0200
@@ -167,7 +167,7 @@
 	SSLeay_add_ssl_algorithms();
 	SSL_load_error_strings();
 
-	ctx = SSL_CTX_new (SSLv3_client_method());
+	ctx = SSL_CTX_new (SSLv23_client_method());
 	ssl = SSL_new (ctx);
 
 	if (args_info.verbose_flag) {
