Subject: [PATCH] skip-checks-for-git-repo
---
Index: solr/distribution/source-release.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/solr/distribution/source-release.gradle b/solr/distribution/source-release.gradle
--- a/solr/distribution/source-release.gradle	(revision 92c5515e2918c22513f7aa527d6c6db943150fea)
+++ b/solr/distribution/source-release.gradle	(date 1681294983703)
@@ -23,7 +23,6 @@
     description "Assemble source Solr artifact as a .tgz file."
 
     // Make sure no unstaged/ dirty files are present.
-    dependsOn ":gitStatus", ":checkWorkingCopyClean"
 
     ext {
       destination = file("${buildDir}/packages/solr-${version}-src.tgz")
Index: gradle/validation/precommit.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gradle/validation/precommit.gradle b/gradle/validation/precommit.gradle
--- a/gradle/validation/precommit.gradle	(revision 92c5515e2918c22513f7aa527d6c6db943150fea)
+++ b/gradle/validation/precommit.gradle	(date 1681294983698)
@@ -48,6 +48,5 @@
 
   // Each validation task should be attached to check but make sure
   // precommit() as a whole is a dependency on rootProject.check
-  check.dependsOn precommit
 }
 
Index: solr/distribution/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/solr/distribution/build.gradle b/solr/distribution/build.gradle
--- a/solr/distribution/build.gradle	(revision 92c5515e2918c22513f7aa527d6c6db943150fea)
+++ b/solr/distribution/build.gradle	(date 1681294983701)
@@ -70,7 +70,6 @@
 
   files = objects.fileCollection()
   [
-      tasks.assembleSourceTgz,
       distTarTask,
   ].each { dep ->
     dependsOn dep
@@ -129,7 +128,6 @@
   })
 
   from tasks.prepareGitRev
-  from tasks.assembleSourceTgz
   from distTarTask
 
   from tasks.computeChecksums
