Subject: [PATCH] allow system gradle
---
Index: gradle/globals.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gradle/globals.gradle b/gradle/globals.gradle
--- a/gradle/globals.gradle	(revision 92c5515e2918c22513f7aa527d6c6db943150fea)
+++ b/gradle/globals.gradle	(date 1681294916077)
@@ -170,4 +170,8 @@
     // anything starting with "JENKINS_" or "HUDSON_": Jenkins/Hudson (https://jenkins.thetaphi.de/env-vars.html/)
     isCIBuild = System.getenv().keySet().find { it ==~ /(?i)((JENKINS|HUDSON)(_\w+)?|CI)/ } != null
   }
+  tasks.withType(AbstractArchiveTask).configureEach {
+    preserveFileTimestamps = false
+    reproducibleFileOrder = true
+  }
 }
Index: gradle/java/jar-manifest.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gradle/java/jar-manifest.gradle b/gradle/java/jar-manifest.gradle
--- a/gradle/java/jar-manifest.gradle	(revision 92c5515e2918c22513f7aa527d6c6db943150fea)
+++ b/gradle/java/jar-manifest.gradle	(date 1681294916075)
@@ -46,7 +46,7 @@
                     } else {
                       def sysProps = System.properties
                       def manifestUsername = sysProps.getOrDefault('manifest.username', sysProps['user.name'])
-                      return "${project.version} ${gitRev} - ${manifestUsername} - ${buildDate} ${buildTime}"
+                      return "${project.version} ${gitRev} - ${manifestUsername}"
                     }
                 }
 
