diff --git a/src/Text/Parse.hs.orig b/src/Text/Parse.hs
index 8e50ecd..36cec28 100644
--- a/src/Text/Parse.hs.orig
+++ b/src/Text/Parse.hs
@@ -1,3 +1,5 @@
+{-# LANGUAGE NoMonadFailDesugaring #-}
+
 module Text.Parse
   ( -- * The Parse class is a replacement for the standard Read class. 
     -- $parser
diff --git a/src/Text/Parse/ByteString.hs.orig b/src/Text/Parse/ByteString.hs
index a8a8a9b..5483fe8 100644
--- a/src/Text/Parse/ByteString.hs.orig
+++ b/src/Text/Parse/ByteString.hs
@@ -1,3 +1,5 @@
+{-# LANGUAGE NoMonadFailDesugaring #-}
+
 module Text.Parse.ByteString
   ( -- * The Parse class is a replacement for the standard Read class. 
     --   This particular instance reads from ByteString rather than String.
