diff -ru kadu-2.1.orig/kadu-core/os/single-application/single-application.cpp kadu-2.1/kadu-core/os/single-application/single-application.cpp
--- kadu-2.1.orig/kadu-core/os/single-application/single-application.cpp	2015-07-11 11:49:28.020701056 +0200
+++ kadu-2.1/kadu-core/os/single-application/single-application.cpp	2015-07-11 11:50:30.834040618 +0200
@@ -26,6 +26,7 @@
 #include "long-lived-lock-file.h"
 
 #include <QtCore/QCoreApplication>
+#include <QtCore/QDataStream>
 #include <QtCore/QDir>
 #include <QtCore/QRegExp>
 #include <QtNetwork/QLocalServer>
