From 9cbd09b81028679a507f751c206e8f6769fd450c Mon Sep 17 00:00:00 2001
From: Igor Murzov <e-mail@date.by>
Date: Thu, 7 Jul 2011 22:31:44 +0400
Subject: [PATCH 2/2] ffmpeg: version check fix

---
 plugins/ffmpeg/ffmpeg.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/plugins/ffmpeg/ffmpeg.c b/plugins/ffmpeg/ffmpeg.c
index 0cb9955..37c7e80 100644
--- a/plugins/ffmpeg/ffmpeg.c
+++ b/plugins/ffmpeg/ffmpeg.c
@@ -718,7 +718,7 @@ ffmpeg_start (void) {
     ffmpeg_init_exts ();
     avcodec_init ();
     av_register_all ();
-#if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(52, 64, 0)
+#if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(52, 69, 0)
     av_register_protocol2 (&vfswrapper, sizeof(vfswrapper));
 #else
     av_register_protocol (&vfswrapper);
-- 
1.7.8.4

