diff -bur gourmet-0.16.0.orig/gourmet/exporters/exporter.py gourmet-0.16.0/gourmet/exporters/exporter.py
--- gourmet-0.16.0.orig/gourmet/exporters/exporter.py	2013-09-15 08:40:09.090420987 +0000
+++ gourmet-0.16.0/gourmet/exporters/exporter.py	2013-09-15 09:00:15.000000000 +0000
@@ -1,10 +1,15 @@
-import re, Image, os.path, os, xml.sax.saxutils, time, shutil, urllib, textwrap, types
+import re, os.path, os, xml.sax.saxutils, time, shutil, urllib, textwrap, types
 from gourmet import gglobals, convert
 from gourmet.gdebug import *
 from gettext import gettext as _
 from gourmet.plugin_loader import Pluggable, pluggable_method
 from gourmet.plugin import BaseExporterPlugin, BaseExporterMultiRecPlugin
 from gourmet.threadManager import SuspendableThread
+# Require PIL module.
+try:
+    from PIL import Image
+except ImportError:
+    import Image
 
 REC_ATTR_DIC = gglobals.REC_ATTR_DIC
 DEFAULT_ATTR_ORDER = gglobals.DEFAULT_ATTR_ORDER
diff -bur gourmet-0.16.0.orig/gourmet/ImageExtras.py gourmet-0.16.0/gourmet/ImageExtras.py
--- gourmet-0.16.0.orig/gourmet/ImageExtras.py	2013-09-15 08:40:09.097087657 +0000
+++ gourmet-0.16.0/gourmet/ImageExtras.py	2013-09-15 07:12:30.000000000 +0000
@@ -1,5 +1,10 @@
-import os, os.path, tempfile, gtk, Image, StringIO
+import os, os.path, tempfile, gtk, StringIO
 from gdebug import *
+# Require PIL module.
+try:
+    from PIL import Image
+except ImportError:
+    import Image
 
 TMPFILE = tempfile.mktemp(prefix='gourmet_tempfile_')
 
diff -bur gourmet-0.16.0.orig/gourmet/plugins/import_export/html_plugin/html_exporter.py gourmet-0.16.0/gourmet/plugins/import_export/html_plugin/html_exporter.py
--- gourmet-0.16.0.orig/gourmet/plugins/import_export/html_plugin/html_exporter.py	2013-09-15 08:40:09.137087303 +0000
+++ gourmet-0.16.0/gourmet/plugins/import_export/html_plugin/html_exporter.py	2013-09-15 07:13:00.000000000 +0000
@@ -1,8 +1,13 @@
-import re, Image, os.path, os, xml.sax.saxutils, time, shutil, urllib, textwrap
+import re, os.path, os, xml.sax.saxutils, time, shutil, urllib, textwrap
 from gettext import gettext as _
 from gourmet import convert,gglobals
 from gourmet.gdebug import *
 from gourmet.exporters.exporter import *
+# Require PIL module.
+try:
+    from PIL import Image
+except ImportError:
+    import Image
 
 
 HTML_HEADER_START = """<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
diff -bur gourmet-0.16.0.orig/gourmet/plugins/import_export/mealmaster_plugin/mealmaster_exporter.py gourmet-0.16.0/gourmet/plugins/import_export/mealmaster_plugin/mealmaster_exporter.py
--- gourmet-0.16.0.orig/gourmet/plugins/import_export/mealmaster_plugin/mealmaster_exporter.py	2013-09-15 08:40:09.137087303 +0000
+++ gourmet-0.16.0/gourmet/plugins/import_export/mealmaster_plugin/mealmaster_exporter.py	2013-09-15 07:13:21.000000000 +0000
@@ -1,8 +1,13 @@
-import re, Image, os.path, os, xml.sax.saxutils, time, shutil, urllib, textwrap
+import re, os.path, os, xml.sax.saxutils, time, shutil, urllib, textwrap
 from gourmet import gglobals,  convert
 from gourmet.exporters.exporter import *
 from gourmet.gdebug import *
 from gettext import gettext as _
+# Require PIL module.
+try:
+    from PIL import Image
+except ImportError:
+    import Image
 
 class mealmaster_exporter (exporter_mult):
     def __init__ (self, rd, r, out, conv=None, change_units=True, mult=1):
diff -bur gourmet-0.16.0.orig/gourmet/reccard.py gourmet-0.16.0/gourmet/reccard.py
--- gourmet-0.16.0.orig/gourmet/reccard.py	2013-09-15 08:40:09.137087303 +0000
+++ gourmet-0.16.0/gourmet/reccard.py	2013-09-15 07:13:54.000000000 +0000
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 import gc
-import gtk, gobject, os.path, time, os, sys, re, threading, gtk.gdk, Image, StringIO, pango, string
+import gtk, gobject, os.path, time, os, sys, re, threading, gtk.gdk, StringIO, pango, string
 import types
 import xml.sax.saxutils, pango
 import exporters.exportManager
@@ -30,6 +30,11 @@
 import plugin_loader
 import timeScanner
 import defaults
+# Require PIL module.
+try:
+    from PIL import Image
+except ImportError:
+    import Image
 
 # TODO
 #
