diff --git a/src/module_list.py b/src/module_list.py
index 96e8081..4728e9a 100644
--- a/src/module_list.py
+++ b/src/module_list.py
@@ -45,6 +45,7 @@ givaro_library_dirs = givaro_pc['library_dirs']
 gsl_pc = pkgconfig.parse('gsl')
 gsl_libs = gsl_pc['libraries']
 gsl_libs.remove('gslcblas')
+gsl_libs.extend(cblas_libs)
 gsl_library_dirs = gsl_pc['library_dirs']
 gsl_include_dirs = gsl_pc['include_dirs']
 

