diff --git a/src/sage/combinat/cluster_algebra_quiver/interact.py b/src/sage/combinat/cluster_algebra_quiver/interact.py
index 4e54c64..1c88e61 100644
--- a/src/sage/combinat/cluster_algebra_quiver/interact.py
+++ b/src/sage/combinat/cluster_algebra_quiver/interact.py
@@ -107,7 +107,7 @@ def cluster_interact(self, fig_size=1, circular=True, kind='seed'):
     show_lastmutation.observe(refresh, 'value')
     which_plot.observe(refresh, 'value')
 
-    mut_buttons.on_displayed(refresh)
+    mut_buttons.on_widget_constructed(refresh)
 
     if kind == 'seed':
         top = widgets.HBox([show_seq, show_vars])
diff --git a/src/sage/repl/ipython_kernel/interact.py b/src/sage/repl/ipython_kernel/interact.py
index 9b94bc4..a2665a4 100644
--- a/src/sage/repl/ipython_kernel/interact.py
+++ b/src/sage/repl/ipython_kernel/interact.py
@@ -126,7 +126,7 @@ class sage_interactive(interactive):
         super(sage_interactive, self).__init__(f, options, **kwds)
         if self.manual:
             # In Sage, manual interacts are always run once
-            self.on_displayed(self.update)
+            self.on_widget_constructed(self.update)
         else:
             # In automatic mode, clicking on a ToggleButtons button
             # should also run the interact
diff --git a/src/sage/repl/ipython_kernel/widgets_sagenb.py b/src/sage/repl/ipython_kernel/widgets_sagenb.py
index 2ce59d7..76f4f52 100644
--- a/src/sage/repl/ipython_kernel/widgets_sagenb.py
+++ b/src/sage/repl/ipython_kernel/widgets_sagenb.py
@@ -470,15 +470,6 @@ def selector(values, label=None, default=None, nrows=None, ncols=None, width=Non
         sage: selector([(1,"one"), (2,"two"), (3,"three")], buttons=True)
         ToggleButtons(options=(('one', 1), ('two', 2), ('three', 3)), value=1)
 
-    A dict of ``label:value`` pairs is also allowed. Since a ``dict``
-    is not ordered, it is better to use an :class:`OrderedDict`::
-
-        sage: from collections import OrderedDict
-        sage: selector(OrderedDict(one=1, two=2, three=3))
-        Dropdown(options=OrderedDict([('one', 1), ('two', 2), ('three', 3)]), value=1)
-        sage: selector(OrderedDict(one=1, two=2, three=3), buttons=True)
-        ToggleButtons(options=OrderedDict([('one', 1), ('two', 2), ('three', 3)]), value=1)
-
     The values can be any kind of object:
 
         sage: selector([sin(x^2), GF(29), EllipticCurve('37a1')])
