diff --git a/src/sage/interfaces/sage0.py b/src/sage/interfaces/sage0.py
index 4ba76ab0b8..e7740177f4 100644
--- a/src/sage/interfaces/sage0.py
+++ b/src/sage/interfaces/sage0.py
@@ -161,7 +161,7 @@ class Sage(ExtraTabCompletion, Expect):
                 '--no-term-title',
                 '--simple-prompt',
             ])
-            prompt = re.compile('In \[\d+\]: ')
+            prompt = re.compile('sage: ')
             if init_code is None:
                 init_code = ['from six.moves import cPickle']
 
