diff -aur afl-utils-v1.34a/setup.py afl-utils-v1.34a.old/setup.py
--- afl-utils-v1.34a/setup.py	2018-11-02 10:57:33.000000000 +0100
+++ afl-utils-v1.34a.old/setup.py	2023-04-07 14:58:58.504037764 +0200
@@ -20,7 +20,7 @@
 import afl_utils
 
 dependencies = [
-    'exploitable==1.32-rcor',       # needed for gdb script execution
+    'exploitable',       # needed for gdb script execution
     'simplejson',                   # needed for config files
     'twitter',                      # needed for afl-stats (twitter access)
 ]
