--- a/plugins/amdgpu/Makefile
+++ b/plugins/amdgpu/Makefile
@@ -50,16 +50,16 @@ clean: amdgpu_plugin_clean amdgpu_plugin_test_clean
 mrproper: clean
 
 install:
-	$(Q) mkdir -p $(PLUGINDIR)
+	$(Q) mkdir -p $(DESTDIR)$(PLUGINDIR)
 ifeq ($(CONFIG_AMDGPU),y)
 	$(E) "  INSTALL " $(PLUGIN_NAME)
-	$(Q) install -m 644 $(PLUGIN_SOBJ) $(PLUGINDIR)
+	$(Q) install -m 644 $(PLUGIN_SOBJ) $(DESTDIR)$(PLUGINDIR)
 endif
 .PHONY: install
 
 uninstall:
 ifeq ($(CONFIG_AMDGPU),y)
 	$(E) " UNINSTALL" $(PLUGIN_NAME)
-	$(Q) $(RM) $(PLUGINDIR)/$(PLUGIN_SOBJ)
+	$(Q) $(RM) $(DESTDIR)$(PLUGINDIR)/$(PLUGIN_SOBJ)
 endif
 .PHONY: uninstall
