--- a/Makefile.install
+++ b/Makefile.install
@@ -33,15 +33,15 @@ install-man:
 	$(Q) $(MAKE) -C Documentation install
 .PHONY: install-man
 
-install-lib: lib
+install-lib:
 	$(Q) $(MAKE) $(build)=lib install
 .PHONY: install-lib
 
-install-criu: criu
+install-criu:
 	$(Q) $(MAKE) $(build)=criu install
 .PHONY: install-criu
 
-install-amdgpu_plugin: amdgpu_plugin
+install-amdgpu_plugin:
 	$(Q) $(MAKE) -C plugins/amdgpu install
 .PHONY: install-amdgpu_plugin
 
