diff -upr frogatto-1.3.1.orig/src/water.cpp frogatto-1.3.1/src/water.cpp
--- frogatto-1.3.1.orig/src/water.cpp	2020-12-26 10:32:22.419994864 +0300
+++ frogatto-1.3.1/src/water.cpp	2020-12-26 10:33:18.116224965 +0300
@@ -165,8 +165,6 @@
 	if (glBlendEquationOES) {
 		glBlendEquationOES(GL_FUNC_REVERSE_SUBTRACT_OES);
 	}
-#elif defined(GL_OES_blend_subtract)
-	glBlendEquationOES(GL_FUNC_REVERSE_SUBTRACT_OES);
 #elif defined(USE_GLES2)
 	glBlendEquation(GL_FUNC_REVERSE_SUBTRACT);
 #else
@@ -212,8 +210,6 @@
 	if (glBlendEquationOES) {
 		glBlendEquationOES(GL_FUNC_ADD_OES);
 	}
-#elif defined(GL_OES_blend_subtract)
-	glBlendEquationOES(GL_FUNC_ADD_OES);
 #elif defined(USE_GLES2)
 	glBlendEquation(GL_FUNC_ADD);
 #else
