diff -rupN a/plugins/jabber_protocol/3rdparty/CMakeLists.txt b/plugins/jabber_protocol/3rdparty/CMakeLists.txt
--- a/plugins/jabber_protocol/3rdparty/CMakeLists.txt	2012-09-14 21:54:35.000000000 +0000
+++ b/plugins/jabber_protocol/3rdparty/CMakeLists.txt	2014-02-24 17:43:59.696495517 +0000
@@ -26,7 +26,7 @@ macro (kadu_jabber_automoc)
 					get_filename_component (_basename ${_current_MOC} NAME_WE)
 					set (_header ${_abs_FILE})
 					set (_moc ${_abs_PATH}/${_current_MOC})
-					QT4_CREATE_MOC_COMMAND (${_header} ${_moc} "${_moc_INCS}" "")
+					QT4_CREATE_MOC_COMMAND (${_header} ${_moc} "${_moc_INCS}" "" "${moc_target}")
 					MACRO_ADD_FILE_DEPENDENCIES (${_abs_FILE} ${_moc})
 				endforeach (_current_MOC_INC)
 			endif (_match)
