diff --git a/kadu-2.1/plugins/qt4_docking/qt4_docking.cpp b/kadu-2.1/plugins/qt4_docking/qt4_docking.cpp
index 5d228c8..1b576e6 100644
--- a/kadu-2.1/plugins/qt4_docking/qt4_docking.cpp
+++ b/kadu-2.1/plugins/qt4_docking/qt4_docking.cpp 
@@ -26,6 +26,7 @@
 #include <QtCore/QEvent>
 #include <QtGui/QMouseEvent>
 #include <QtGui/QMovie>
+#include <QtGui/QIcon>
 
 #include "plugins/docking/docking.h"
 
@@ -68,7 +69,8 @@ Qt4TrayIcon::Qt4TrayIcon(QWidget *parent) :
 {
 	kdebugf();
 
-	setIcon(DockingManager::instance()->defaultIcon().icon());
+	QIcon icon = QIcon::fromTheme(DockingManager::instance()->defaultIcon().fullPath());
+	setIcon(icon);
 
 	connect(this, SIGNAL(activated(QSystemTrayIcon::ActivationReason)), this, SLOT(trayActivated(QSystemTrayIcon::ActivationReason)));
 
@@ -102,7 +104,7 @@ void Qt4TrayIcon::changeTrayIcon(const KaduIcon &icon)
 		Movie->deleteLater();
 		Movie = 0;
 	}
-	setIcon(icon.icon());
+	setIcon(QIcon::fromTheme(icon.fullPath()));
 }
 
 void Qt4TrayIcon::changeTrayMovie(const QString &moviePath)
