diff --git a/sympol/CMakeLists.txt b/sympol/CMakeLists.txt
index de9b4a8..da759d6 100644
--- a/sympol/CMakeLists.txt
+++ b/sympol/CMakeLists.txt
@@ -99,11 +99,6 @@ install(FILES
   configuration.h
   facesuptosymmetrylist.h
   facewithdata.h
-  matrix/algorithm.h
-  matrix/invert.h
-  matrix/matrix.h
-  matrix/rank.h
-  matrix/zmatrix.h
   polyhedrondatastorage.h
   polyhedron.h
   polyhedronio.h
@@ -122,6 +117,16 @@ install(FILES
   symmetrycomputationidm.h
   symmetrycomputationidmmemento.h
   symmetrycomputationmemento.h
+  types.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/sympol)
+install(FILES
+  matrix/algorithm.h
+  matrix/invert.h
+  matrix/matrix.h
+  matrix/rank.h
+  matrix/zmatrix.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/sympol/matrix)
+install(FILES
   symmetrygroupconstruction/computesymmetries.h
   symmetrygroupconstruction/graphconstructionbliss.h
   symmetrygroupconstruction/graphconstructiondefault.h
@@ -129,10 +134,11 @@ install(FILES
   symmetrygroupconstruction/matrixconstructiondefault.h
   symmetrygroupconstruction/matrixconstructioneigen.h
   symmetrygroupconstruction/matrixconstruction.h
-  types.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/sympol/symmetrygroupconstruction)
+install(FILES
   yal/logger.h
   yal/reportlevel.h
   yal/usagestats.h
-  DESTINATION ${INCLUDE_INSTALL_DIR}/sympol)
+  DESTINATION ${INCLUDE_INSTALL_DIR}/sympol/yal)
 install(FILES ../man/sympol.1
         DESTINATION "share/man/man1")
