diff -wbBur sk1-0.9.3/src/sk1/managers/dialogmanager.py sk1-0.9.3.my/src/sk1/managers/dialogmanager.py
--- sk1-0.9.3/src/sk1/managers/dialogmanager.py	2014-10-28 20:54:25.000000000 +0300
+++ sk1-0.9.3.my/src/sk1/managers/dialogmanager.py	2015-02-20 19:13:54.588252445 +0300
@@ -199,7 +199,7 @@
 	tmpfile = NamedTemporaryFile()
 	execline = ''
 	if sk1.LANG: execline += 'export LANG=' + sk1.LANG + ';'
-	execline += 'python %s/gtk/open_file_dialog.py ' % (PATH,)
+	execline += 'python2 %s/gtk/open_file_dialog.py ' % (PATH,)
 	execline += ' caption="' + name + '" start_dir="' + initialdir + '"'
 	execline += ' filetypes="' + filetypes + '" window-icon="' + icon + '"'
 	os.system(execline + '>' + tmpfile.name)
@@ -229,7 +229,7 @@
 	execline = ''
 	if sk1.LANG: execline += 'export LANG=' + sk1.LANG + ';'
 	path = os.path.join(initialdir, initialfile)
-	execline += 'python %s/gtk/save_file_dialog.py ' % (PATH,)
+	execline += 'python2 %s/gtk/save_file_dialog.py ' % (PATH,)
 	execline += ' caption="' + name + '" path="' + path + '" '
 	execline += ' filetypes="' + filetypes + '" window-icon="' + icon + '"'
 	os.system(execline + '>' + tmpfile.name)
diff -wbBur sk1-0.9.3/src/sk1/managers/docmanager.py sk1-0.9.3.my/src/sk1/managers/docmanager.py
--- sk1-0.9.3/src/sk1/managers/docmanager.py	2014-11-02 20:51:38.000000000 +0300
+++ sk1-0.9.3.my/src/sk1/managers/docmanager.py	2015-02-20 19:13:59.778162450 +0300
@@ -223,7 +223,7 @@
 
 		execline = ''
 		if sk1.LANG: execline += 'export LANG=' + sk1.LANG + ';'
-		execline += 'python %s/gtk/print_dialog.py ' % (PATH,)
+		execline += 'python2 %s/gtk/print_dialog.py ' % (PATH,)
 		execline += ' filepath="' + pdffile.name + '"'
 		execline += ' window-icon="' + icon + '"'
 
diff -wbBur sk1-0.9.3/src/sk1sdk/tkstyle/colors.py sk1-0.9.3.my/src/sk1sdk/tkstyle/colors.py
--- sk1-0.9.3/src/sk1sdk/tkstyle/colors.py	2014-09-03 20:42:00.000000000 +0400
+++ sk1-0.9.3.my/src/sk1sdk/tkstyle/colors.py	2015-02-20 19:14:21.267789842 +0300
@@ -146,7 +146,7 @@
 			" style.bg[gtk.STATE_SELECTED].to_string(),"+ \
 			" style.bg[gtk.STATE_INSENSITIVE].to_string();"
 	
-		os.system('python -c "%s" >%s 2>/dev/null'%(command, tmpfile.name))	
+		os.system('python2 -c "%s" >%s 2>/dev/null'%(command, tmpfile.name))	
 
 		for type in ["base","text","fg","bg"]:
 			line=tmpfile.readline().strip().split()
diff -wbBur sk1-0.9.3/src/sk1sdk/tkstyle/fonts.py sk1-0.9.3.my/src/sk1sdk/tkstyle/fonts.py
--- sk1-0.9.3/src/sk1sdk/tkstyle/fonts.py	2014-09-10 13:06:18.000000000 +0400
+++ sk1-0.9.3.my/src/sk1sdk/tkstyle/fonts.py	2015-02-20 19:14:26.454366585 +0300
@@ -50,7 +50,7 @@
 
 	tmpfile = NamedTemporaryFile()
 	command = "import gtk;w = gtk.Window();w.realize();style=w.get_style(); print style.font_desc.to_string();"
-	os.system('python -c "%s" >%s 2>/dev/null' % (command, tmpfile.name))
+	os.system('python2 -c "%s" >%s 2>/dev/null' % (command, tmpfile.name))
 
 	font = tmpfile.readline().strip()
 
