From 2c29bf2b3d68f4897d708a9fa215277e1c40d2e8 Mon Sep 17 00:00:00 2001
From: =?utf8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 18 Dec 2014 04:29:17 +0100
Subject: [PATCH] Fix GtkBuilder UI file for GTK+ 3, add mouse icon to mouse
 tab

---
 data/lxinput.ui | 27 +++++++++++++--------------
 1 file changed, 13 insertions(+), 14 deletions(-)

diff --git a/data/lxinput.ui b/data/lxinput.ui
index a61cb8e..a3a4175 100644
--- a/data/lxinput.ui
+++ b/data/lxinput.ui
@@ -107,7 +107,6 @@
                                 <property name="width_request">100</property>
                                 <property name="visible">True</property>
                                 <property name="can_focus">True</property>
-                                <property name="update_policy">discontinuous</property>
                                 <property name="adjustment">mouse_accel_adj</property>
                                 <property name="digits">1</property>
                                 <property name="value_pos">right</property>
@@ -123,7 +122,6 @@
                                 <property name="width_request">100</property>
                                 <property name="visible">True</property>
                                 <property name="can_focus">True</property>
-                                <property name="update_policy">discontinuous</property>
                                 <property name="adjustment">mouse_threshold_adj</property>
                                 <property name="digits">0</property>
                                 <property name="value_pos">right</property>
@@ -247,6 +245,19 @@
                 <property name="can_focus">False</property>
                 <property name="spacing">6</property>
                 <child>
+                  <object class="GtkImage" id="image1">
+                    <property name="visible">True</property>
+                    <property name="can_focus">False</property>
+                    <property name="pixel_size">16</property>
+                    <property name="icon_name">input-mouse</property>
+                  </object>
+                  <packing>
+                    <property name="expand">True</property>
+                    <property name="fill">True</property>
+                    <property name="position">0</property>
+                  </packing>
+                </child>
+                <child>
                   <object class="GtkLabel" id="label1">
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
@@ -312,7 +323,6 @@
                                     <property name="width_request">100</property>
                                     <property name="visible">True</property>
                                     <property name="can_focus">True</property>
-                                    <property name="update_policy">discontinuous</property>
                                     <property name="adjustment">kb_daley_adj</property>
                                     <property name="digits">0</property>
                                     <property name="value_pos">right</property>
@@ -329,7 +339,6 @@
                                     <property name="visible">True</property>
                                     <property name="can_focus">True</property>
                                     <property name="tooltip_text" translatable="yes">Interval between each key repeat</property>
-                                    <property name="update_policy">discontinuous</property>
                                     <property name="adjustment">kb_interval_adj</property>
                                     <property name="digits">0</property>
                                     <property name="value_pos">right</property>
@@ -624,16 +633,6 @@
       <action-widget response="-5">ok</action-widget>
     </action-widgets>
   </object>
-  <object class="GtkImage" id="image1">
-    <property name="visible">True</property>
-    <property name="can_focus">False</property>
-    <property name="icon_name">input-keyboard</property>
-  </object>
-  <object class="GtkImage" id="image10">
-    <property name="visible">True</property>
-    <property name="can_focus">False</property>
-    <property name="icon_name">input-keyboard</property>
-  </object>
   <object class="GtkAdjustment" id="kb_daley_adj">
     <property name="lower">100</property>
     <property name="upper">1100</property>
-- 
2.1.4

From 2b52f014c63cca1581612ea7777a6d552187bdee Mon Sep 17 00:00:00 2001
From: =?utf8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 2 Aug 2015 00:45:47 +0200
Subject: [PATCH] Add MATE to NotShowIn in desktop entries

---
 data/lxinput.desktop.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/lxinput.desktop.in b/data/lxinput.desktop.in
index 0587227..83d29b6 100644
--- a/data/lxinput.desktop.in
+++ b/data/lxinput.desktop.in
@@ -7,4 +7,4 @@ _Comment=Configure keyboard, mouse, and other input devices
 StartupNotify=true
 Terminal=false
 Categories=GTK;Settings;HardwareSettings;X-LXDE-Settings;
-NotShowIn=GNOME;KDE;XFCE;
+NotShowIn=GNOME;KDE;XFCE;MATE;
-- 
2.1.4

