--- processing/app/src/processing/app/contrib/ContributionManager.java	2018-07-26 23:59:08.000000000 +0200
+++ processing/app/src/processing/app/contrib/ContributionManager.java	2018-11-20 12:53:07.229171545 +0100
@@ -206,7 +206,7 @@
                   .interpolate("contrib.errors.contrib_download.timeout",
                                ad.getName()));
               } else {
-                status.setErrorMessage(Language
+                status.setErrorMessage("Could not download " + url + " to " + contribZip + " : " + Language
                   .interpolate("contrib.errors.download_and_install",
                                ad.getName()));
               }
