diff -Naur gnome-settings-daemon-3.6.4.orig/plugins/automount/gnome-fallback-mount-helper.desktop.in.in gnome-settings-daemon-3.6.4/plugins/automount/gnome-fallback-mount-helper.desktop.in.in
--- gnome-settings-daemon-3.6.4.orig/plugins/automount/gnome-fallback-mount-helper.desktop.in.in	2013-01-10 14:37:58.000000000 +0100
+++ gnome-settings-daemon-3.6.4/plugins/automount/gnome-fallback-mount-helper.desktop.in.in	2013-04-24 14:34:43.743878119 +0200
@@ -9,4 +9,4 @@
 NoDisplay=true
 OnlyShowIn=GNOME;Unity;
 X-GNOME-Autostart-Notify=true
-AutostartCondition=GNOME3 if-session gnome-fallback
+AutostartCondition=GNOME3 if-session gnome-flashback
diff -Naur gnome-settings-daemon-3.6.4.orig/plugins/background/gnome-fallback-background-helper.desktop.in.in gnome-settings-daemon-3.6.4/plugins/background/gnome-fallback-background-helper.desktop.in.in
--- gnome-settings-daemon-3.6.4.orig/plugins/background/gnome-fallback-background-helper.desktop.in.in	2013-04-24 14:32:47.290988000 +0200
+++ gnome-settings-daemon-3.6.4/plugins/background/gnome-fallback-background-helper.desktop.in.in	2013-04-24 14:34:43.743878119 +0200
@@ -9,4 +9,4 @@
 NoDisplay=true
 OnlyShowIn=GNOME;Unity;
 X-GNOME-Autostart-Notify=true
-AutostartCondition=GNOME3 if-session gnome-fallback
+AutostartCondition=GNOME3 if-session gnome-flashback
diff -Naur gnome-settings-daemon-3.6.4.orig/plugins/media-keys/gnome-fallback-media-keys-helper.desktop.in.in gnome-settings-daemon-3.6.4/plugins/media-keys/gnome-fallback-media-keys-helper.desktop.in.in
--- gnome-settings-daemon-3.6.4.orig/plugins/media-keys/gnome-fallback-media-keys-helper.desktop.in.in	2013-04-24 14:32:52.807791000 +0200
+++ gnome-settings-daemon-3.6.4/plugins/media-keys/gnome-fallback-media-keys-helper.desktop.in.in	2013-04-24 14:35:16.714696636 +0200
@@ -9,4 +9,4 @@
 NoDisplay=true
 OnlyShowIn=GNOME;Unity;
 X-GNOME-Autostart-Notify=true
-AutostartCondition=GNOME3 if-session gnome-fallback
+AutostartCondition=GNOME3 if-session gnome-flashback
