=== modified file 'src/xauthority.c'
--- src/xauthority.c	2013-09-10 23:14:12 +0000
+++ src/xauthority.c	2013-09-30 02:59:01 +0000
@@ -237,7 +237,7 @@
 gboolean
 x_authority_write (XAuthority *auth, XAuthWriteMode mode, const gchar *filename, GError **error)
 {
-    gchar *input;
+    gchar *input = NULL;
     gsize input_length = 0, input_offset = 0;
     GList *link, *records = NULL;
     XAuthority *a;
@@ -328,6 +328,7 @@
     }
 
     errno = 0;
+    result = TRUE;
     for (link = records; link && result; link = link->next)
     {
         XAuthority *a = link->data;

