diff --git a/test_prctl.py b/test_prctl.py
index df84986..ea53fb0 100644
--- a/test_prctl.py
+++ b/test_prctl.py
@@ -20,7 +20,7 @@ except ImportError:
     pass
 
 curdir = os.path.dirname(__file__)
-builddir = os.path.join(curdir, 'build', 'lib.%s-%s' % (distutils.util.get_platform(), sys.version[0:3]))
+builddir = os.path.join(curdir, 'build', 'lib.%s-%d.%d' % (distutils.util.get_platform(), *sys.version_info[0:2]))
 
 # Always run from the builddir
 if not os.path.exists(builddir) or \
