From fd8da8994f7006b9fcefc90d2fe5c2b484c7871e Mon Sep 17 00:00:00 2001
From: Luis Menina <liberforce@freeside.fr>
Date: Mon, 04 Mar 2013 13:42:22 +0000
Subject: Remove deprecated libgnomeui.

https://bugzilla.gnome.org/show_bug.cgi?id=589045
---
diff --git a/README b/README
index e6d5c31..b2489ef 100644
--- a/README
+++ b/README
@@ -59,7 +59,6 @@ Dependencies:
 GLIB 2.6.0
 GTK 2.14.0
 LIBGNOMECANVAS 2.10.0
-LIBGNOMEUI 2.10.0
 LIBGLADE 2.3.0
 GNOME_VFS 2.10.0
 GCONF 2.10.0
diff --git a/configure.in b/configure.in
index 9f3d112..06568fd 100644
--- a/configure.in
+++ b/configure.in
@@ -67,7 +67,6 @@ dnl it too, or it will never make it into the spec file!
 GLIB_REQUIRED=2.6.0
 GTK_REQUIRED=2.14.0
 LIBGNOMECANVAS_REQUIRED=2.10.0
-LIBGNOMEUI_REQUIRED=2.10.0
 LIBGLADE_REQUIRED=2.3.0
 GNOME_VFS_REQUIRED=2.10.0
 GCONF_REQUIRED=2.10.0
@@ -116,7 +115,6 @@ PKG_CHECK_MODULES(PLANNER,
 	gobject-2.0 gmodule-2.0
 	gtk+-2.0 >= $GTK_REQUIRED
 	libgnomecanvas-2.0 >= $LIBGNOMECANVAS_REQUIRED
-	libgnomeui-2.0 >= $LIBGNOMEUI_REQUIRED
 	libglade-2.0 >= $LIBGLADE_REQUIRED
 	gconf-2.0 >= $GCONF_REQUIRED
 ])
--
cgit v0.9.2
