From d1e3ca1144b664421facf30acd5524cd28961a3c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Johannes=20L=C3=B6thberg?= <johannes@kyriasis.com>
Date: Mon, 4 May 2020 18:46:38 +0200
Subject: [PATCH 3/4] Compile .py's before building wheel
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Johannes Löthberg <johannes@kyriasis.com>
---
 pylib/Makefile | 2 ++
 qt/Makefile    | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/pylib/Makefile b/pylib/Makefile
index feae25d8..9d60bfe2 100644
--- a/pylib/Makefile
+++ b/pylib/Makefile
@@ -103,6 +103,8 @@ build: .build/build
 
 .build/build: $(BUILD_STEPS) $(CHECKDEPS)
 	rm -rf dist build
+	python setup.py build
+	python -O -m compileall .
 	python setup.py -q bdist_wheel
 	rsync -a dist/*.whl ../dist/
 	touch $@
diff --git a/qt/Makefile b/qt/Makefile
index c0ba93e3..364fa890 100644
--- a/qt/Makefile
+++ b/qt/Makefile
@@ -131,6 +131,8 @@ build: .build/build
 
 .build/build: $(BUILD_STEPS) $(CHECKDEPS)
 	rm -rf dist build
+	python setup.py build
+	python -O -m compileall .
 	python setup.py -q bdist_wheel
 	rsync -a dist/*.whl ../dist/
 	touch $@
-- 
2.26.2

