From 894a3e04560ca11651016ede480fb372e8a2d45b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Johannes=20L=C3=B6thberg?= <johannes@kyriasis.com>
Date: Sat, 7 Nov 2020 17:01:04 +0100
Subject: [PATCH 5/5] Make pyenv target just create venv
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Johannes Löthberg <johannes@kyriasis.com>
---
 Makefile | 18 +-----------------
 1 file changed, 1 insertion(+), 17 deletions(-)

diff --git a/Makefile b/Makefile
index cffa2e30..d7233235 100644
--- a/Makefile
+++ b/Makefile
@@ -61,23 +61,7 @@ all: run
 # - modern pip required for wheel
 # - add qt if missing
 pyenv:
-# 	https://github.com/PyO3/maturin/issues/283 - Expected `python` to be a python interpreter inside a virtualenv
-	set -eu -o pipefail ${SHELLFLAGS}; \
-	"${PYTHON_BIN}" -m venv pyenv; \
-	case "$$(uname -s)" in CYGWIN*|MINGW*|MSYS*) \
-		dos2unix "${ACTIVATE_SCRIPT}"; \
-		VIRTUAL_ENV="$$(pwd)"; \
-		VIRTUAL_ENV="$$(cygpath -m "$${VIRTUAL_ENV}")"; \
-		sed -i -- "s@VIRTUAL_ENV=\".*\"@VIRTUAL_ENV=\"$$(pwd)/pyenv\"@g" "${ACTIVATE_SCRIPT}"; \
-		sed -i -- "s@export PATH@export PATH; VIRTUAL_ENV=\"$${VIRTUAL_ENV}/pyenv\";@g" "${ACTIVATE_SCRIPT}"; \
-		;; esac; \
-	. "${ACTIVATE_SCRIPT}"; \
-	python --version; \
-	python -m pip install --upgrade pip setuptools; \
-	${ANKI_EXTRA_PIP}; \
-	if ! python -c 'import PyQt5' 2>/dev/null; then \
-		python -m pip install -r qt/requirements.qt; \
-	fi;
+	python -m venv pyenv --system-site-packages
 
 # update build hash
 .PHONY: buildhash
-- 
2.29.2

