diff -wbBur rdanbrook-nestopia-53aa38a/Makefile rdanbrook-nestopia-53aa38a.my/Makefile
--- rdanbrook-nestopia-53aa38a/Makefile	2012-09-26 04:21:32.000000000 +0400
+++ rdanbrook-nestopia-53aa38a.my/Makefile	2012-10-01 21:12:08.000000000 +0400
@@ -186,12 +186,12 @@
 	install -m 0644 NstDatabase.xml $(DATADIR)
 	install -m 0644 source/linux/icons/*.png $(DATADIR)/icons
 	install -m 0644 source/linux/icons/*.svg $(DATADIR)/icons
-	install -m 0644 source/linux/icons/nestopia.svg /usr/share/pixmaps
-	xdg-desktop-menu install --novendor $(DATADIR)/nestopia.desktop
+	install -Dm0644 source/linux/icons/nestopia.svg $(PREFIX)/share/pixmaps/nestopia.svg
+	install -Dm0644 $(DATADIR)/nestopia.desktop $(PREFIX)/share/applications/nestopia.desktop
 
 uninstall:
 	xdg-desktop-menu uninstall $(DATADIR)/nestopia.desktop
-	rm /usr/share/pixmaps/nestopia.svg
+	rm $(PREFIX)/share/pixmaps/nestopia.svg
 	rm $(BINDIR)/$(EXE)
 	rm -rf $(DATADIR)
 
