diff --git a/Rakefile b/Rakefile
index db82161..f2a1362 100644
--- a/Rakefile
+++ b/Rakefile
@@ -1,8 +1,5 @@
 desc 'Run all tests'
-task :test => %i[rubocop spec test:acceptance]
-
-require 'rubocop/rake_task'
-RuboCop::RakeTask.new
+task :test => %i[spec test:acceptance]
 
 namespace :test do
   desc 'Acceptance suite under test/ which runs metadata-json-lint against sample files with expected output'
