diff -up nautilus-open-terminal-0.9/configure.in.gnome-vfs nautilus-open-terminal-0.9/configure.in
--- nautilus-open-terminal-0.9/configure.in.gnome-vfs	2008-11-13 13:57:33.000000000 -0500
+++ nautilus-open-terminal-0.9/configure.in	2008-11-13 13:57:53.000000000 -0500
@@ -28,7 +28,7 @@ fi
 
 PKG_CHECK_MODULES(NAUTILUS, libnautilus-extension >= $NAUTILUS_REQUIRED glib-2.0 >= $GLIB_REQUIRED)
 PKG_CHECK_MODULES(GCONF, gconf-2.0)
-PKG_CHECK_MODULES(GNOMEDESKTOP, gnome-desktop-2.0 >= $LIBGNOME_DESKTOP_REQUIRED)
+PKG_CHECK_MODULES(GNOMEDESKTOP, gnome-desktop-2.0 >= $LIBGNOME_DESKTOP_REQUIRED gnome-vfs-2.0)
 
 PKG_CHECK_MODULES(GLIB_RECENT, glib-2.0 >= 2.13.3,
                   AC_DEFINE(HAVE_GLIB_DESKTOP_DIR_API, 1, [g_get_user_special_dir() support]),
