diff -rup luxrender-lux-3066b2161439/core/film.cpp luxrender-lux-3066b2161439.new/core/film.cpp
--- luxrender-lux-3066b2161439/core/film.cpp	2013-03-16 19:32:24.000000000 +0000
+++ luxrender-lux-3066b2161439.new/core/film.cpp	2013-03-26 17:23:21.359679166 +0000
@@ -890,8 +890,8 @@ Film::Film(u_int xres, u_int yres, Filte
 	filename(filename1),
 	colorSpace(0.63f, 0.34f, 0.31f, 0.595f, 0.155f, 0.07f, 0.314275f, 0.329411f), // default is SMPTE
 	convTest(NULL), varianceBuffer(NULL),
-	noiseAwareMap(NULL), noiseAwareMapVersion(0),
-	userSamplingMapFileName(samplingmapfilename), userSamplingMap(NULL), userSamplingMapVersion(0),
+	noiseAwareMap(), noiseAwareMapVersion(0),
+	userSamplingMapFileName(samplingmapfilename), userSamplingMap(), userSamplingMapVersion(0),
 	ZBuffer(NULL), use_Zbuf(useZbuffer),
 	debug_mode(debugmode), premultiplyAlpha(premult),
 	writeResumeFlm(w_resume_FLM), restartResumeFlm(restart_resume_FLM), writeFlmDirect(write_FLM_direct),
