diff --git a/components/sceneutil/mwshadowtechnique.cpp b/components/sceneutil/mwshadowtechnique.cpp
index cb3a1b278..653e4d3a1 100644
--- a/components/sceneutil/mwshadowtechnique.cpp
+++ b/components/sceneutil/mwshadowtechnique.cpp
@@ -23,6 +23,7 @@
 #include <osg/Geometry>
 #include <osg/io_utils>
 
+#include <limits>
 #include <sstream>
 
 namespace {
diff --git a/components/sceneutil/shadow.cpp b/components/sceneutil/shadow.cpp
index 035c19a5f..1ec2b1c5e 100644
--- a/components/sceneutil/shadow.cpp
+++ b/components/sceneutil/shadow.cpp
@@ -1,5 +1,7 @@
 #include "shadow.hpp"
 
+#include <limits>
+
 #include <osgShadow/ShadowedScene>
 
 #include <components/sceneutil/vismask.hpp>
diff --git a/components/terrain/quadtreeworld.cpp b/components/terrain/quadtreeworld.cpp
index f998b7877..cadf5eaa4 100644
--- a/components/terrain/quadtreeworld.cpp
+++ b/components/terrain/quadtreeworld.cpp
@@ -2,6 +2,7 @@
 
 #include <osgUtil/CullVisitor>
 
+#include <limits>
 #include <sstream>
 
 #include <components/misc/constants.hpp>
