diff -Naur gnome-do-0.95.2.orig/Do/src/Do.cs gnome-do-0.95.2/Do/src/Do.cs
--- gnome-do-0.95.2.orig/Do/src/Do.cs	2014-10-20 02:44:11.403239022 +0000
+++ gnome-do-0.95.2/Do/src/Do.cs	2014-10-20 02:44:23.873198312 +0000
@@ -73,9 +73,7 @@
 			if (!Preferences.QuietStart)
 				Controller.Summon ();
 
-			Gdk.Threads.Enter ();
 			Gtk.Application.Run ();
-			Gdk.Threads.Leave ();
 
 			RelevanceProvider.Serialize (RelevanceProvider.DefaultProvider);
 		}
