diff --git a/spec/spec_helper.rb b/spec/spec_helper.rb
index 5fcddd5..ae1a04e 100644
--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -16,19 +16,9 @@
 #
 # See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
 
-require 'simplecov'
-require 'coveralls'
 require 'webmock/rspec'
 WebMock.disable_net_connect!(allow_localhost: true)
 
-SimpleCov.formatters = [SimpleCov::Formatter::HTMLFormatter, Coveralls::SimpleCov::Formatter]
-
-SimpleCov.start do
-  add_filter '/spec/'
-  minimum_coverage 84
-  minimum_coverage_by_file 26
-end
-
 # Ensure all /lib files are loaded
 # so they will be included in the test coverage report.
 Dir['./lib/**/*.rb'].sort.each { |file| require file }
