From 729257077e328e0b2fa1bba5a5509c09a743d5d0 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 13 Jul 2010 11:49:47 +1000
Subject: [PATCH 2/3] Add w8001 flag for serial Wacom devices.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 inputattach.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/inputattach.c b/inputattach.c
index a7de302..0382954 100644
--- a/inputattach.c
+++ b/inputattach.c
@@ -495,6 +495,9 @@ static struct input_types input_types[] = {
 { "--dump",		"-dump",	"Just enable device",
 	B2400, CS8,
 	0,			0x00,	0x00,	0,	dump_init },
+{ "--w8001",		"-w8001",	"Wacom W8001",
+	B38400, CS8,
+	SERIO_W8001,		0x00,	0x00,	0,	NULL },
 { NULL, NULL, NULL, 0, 0, 0, 0, 0, 0, NULL }
 };
 
-- 
1.7.2.3

