diff -wbBur hex-a-hop/gfx.cpp hex-a-hop.my/gfx.cpp
--- hex-a-hop/gfx.cpp	2007-07-06 14:18:54.000000000 +0000
+++ hex-a-hop.my/gfx.cpp	2010-05-24 09:49:35.000000000 +0000
@@ -274,12 +274,11 @@
 			if ((SDL_focus & 6)==6)
 			{
 				videoExposed = 1;
-
 				x = TickTimer();
 
-				while (x<10)
+				while (x<50)
 				{
-					SDL_Delay(10-x);
+					SDL_Delay(50-x);
 					x += TickTimer();
 				}
 			
@@ -288,7 +287,7 @@
 			else
 			{
 				// Not focussed. Try not to eat too much CPU!
-				SDL_Delay(150);
+				SDL_Delay(300);
 			}
 
 			// experimental...
Only in hex-a-hop.my: gfx.o
Only in hex-a-hop.my: Hex-a-hop
diff -wbBur hex-a-hop/hex_puzzzle.cpp hex-a-hop.my/hex_puzzzle.cpp
--- hex-a-hop/hex_puzzzle.cpp	2007-07-06 14:18:54.000000000 +0000
+++ hex-a-hop.my/hex_puzzzle.cpp	2010-05-24 09:49:35.000000000 +0000
@@ -76,8 +76,10 @@
 //	printf("file_open( \"%s\", \"%s\" )\n", file, flags );
 	extern String base_path;
 	static String filename; // static to reduce memory alloc/free calls.
-	if (file[0]=='\0' || file[1]!=':') //If a full path is specified, don't prepend base_path
+	if (file[0] != '/') //If a full path is specified, don't prepend base_path
 		filename = base_path;
+	else
+		filename = "";
 	filename += file;
 //	printf("   -> \"%s\"\n", filename );
 	
@@ -1513,7 +1515,7 @@
 	double time;
 	double undoTime;
 
-	#define MAX_UNDO 6
+	#define MAX_UNDO 64
 	Undo undo[MAX_UNDO];
 	int numUndo;
 	LevelInfo* currentLevelInfo;
Only in hex-a-hop.my: hex_puzzzle.o
diff -wbBur hex-a-hop/menus.h hex-a-hop.my/menus.h
--- hex-a-hop/menus.h	2007-07-06 14:18:54.000000000 +0000
+++ hex-a-hop.my/menus.h	2010-05-24 09:51:05.000000000 +0000
@@ -170,7 +170,7 @@
 		numLines = 1;
 		state = 0; time = 0;
 		memset(title, 0, sizeof(title));
-		char * x = strstr(m, "|");
+		char * x = strstr((char*)m, "|");
 		if (!x)
 		{
 			msg = m;
@@ -360,7 +360,8 @@
 int freeSlot = -1;
 char* GetSlotName(int i, char * t)
 {
-	 sprintf(t, "save%d.dat", i+1);
+	char *home = getenv("HOME");
+	 sprintf(t, "%s/.hex-a-hop/save%d.dat", home != NULL ? home : "", i+1);
 	 return t;
 }
 
diff -wbBur hex-a-hop/savestate.h hex-a-hop.my/savestate.h
--- hex-a-hop/savestate.h	2007-07-06 14:18:54.000000000 +0000
+++ hex-a-hop.my/savestate.h	2010-05-24 09:51:14.000000000 +0000
@@ -238,7 +238,7 @@
 
 	LevelSave* GetLevel(const char * name, bool create)
 	{
-		char * l = strstr(name, "Levels");
+		char * l = strstr((char*)name, "Levels");
 		if (l)
 			name = l;
 
