Description: fix lxqt-config FTBFS after qtxdg bump
Author: Alf Gaida <agaida@siduction.org>
Last-Update: 2016-09-19

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,7 @@
 add_subdirectory(liblxqt-config-cursor)
 add_subdirectory(lxqt-config-input)
 add_subdirectory(lxqt-config-file-associations)
-add_subdirectory(lxqt-config-appearance)
+# add_subdirectory(lxqt-config-appearance)
 add_subdirectory(lxqt-config-monitor)
 add_subdirectory(lxqt-config-locale)
 
