From 86057e813b61a20ee8ffc414d842625db5221123 Mon Sep 17 00:00:00 2001
From: portix <portix@gmx.net>
Date: Tue, 15 Apr 2014 16:49:08 +0200
Subject: [PATCH] Set G_APPLICATION_HANDLES_COMMANDLINE for glib2 > 2.40

Conflicts:
	src/application.c
---
 src/application.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/application.c b/src/application.c
index 77e7412..51b5c36 100644
--- a/src/application.c
+++ b/src/application.c
@@ -227,6 +227,12 @@ dwb_application_local_command_line(GApplication *app, gchar ***argv, gint *exit_
     single_instance = GET_BOOL("single-instance");
     if (s_opt_single || !single_instance) 
         g_application_set_flags(app, G_APPLICATION_NON_UNIQUE);
+#if GLIB_CHECK_VERSION(2, 40, 0)
+	else {
+		int flags = g_application_get_flags(app);
+		g_application_set_flags(app, flags | G_APPLICATION_HANDLES_COMMAND_LINE);
+	}
+#endif
 
     bus = g_bus_get_sync(G_BUS_TYPE_SESSION, NULL, NULL);
     if (!s_opt_fallback && bus != NULL && g_application_register(app, NULL, &error)) 
-- 
2.0.0

