From 0a11643f7ca595dc03a945155921cfdb68004f1e Mon Sep 17 00:00:00 2001
From: Mirco Bauer <meebey@meebey.net>
Date: Thu, 22 Aug 2013 22:27:29 +0200
Subject: [PATCH] Don't require a specific log4net version

---
 SmartIrc4net.csproj | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SmartIrc4net.csproj b/SmartIrc4net.csproj
index 58f7ab7..9633b03 100644
--- a/SmartIrc4net.csproj
+++ b/SmartIrc4net.csproj
@@ -77,6 +77,7 @@
     <Reference Include="System" />
     <Reference Include="log4net, Version=1.2.10.0, Culture=neutral, PublicKeyToken=a5715cc6d5c3540b">
       <Package>log4net</Package>
+      <SpecificVersion>False</SpecificVersion>
     </Reference>
     <Reference Include="System.Core" />
   </ItemGroup>
-- 
1.8.4

