diff -rupN pygame-sdl2-for-renpy-6.99.8.orig/setuplib.py pygame-sdl2-for-renpy-6.99.8/setuplib.py
--- pygame-sdl2-for-renpy-6.99.8.orig/setuplib.py	2015-12-26 17:20:24.573648895 +0100
+++ pygame-sdl2-for-renpy-6.99.8/setuplib.py	2015-12-26 17:20:42.733819376 +0100
@@ -58,7 +58,7 @@ windows = platform.win32_ver()[0]
 if windows:
     cython_command = os.path.join(os.path.dirname(sys.executable), "Scripts", "cython.exe")
 else:
-    cython_command = "cython"
+    cython_command = "cython2"
 
 
 if sys.version_info[0] >= 3:
