--- src/kvpncconfig.cpp.orig	2013-05-19 14:37:11.867517045 -0400
+++ src/kvpncconfig.cpp	2013-05-19 14:37:31.702107829 -0400
@@ -865,7 +865,7 @@
     enableDebugXl2tpdtunnel = QVariant(configgroup.readEntry("EnableDebugXl2tpdtunnel", false)).toBool();
 
 
-    pathToVpnc = configgroup.readEntry("Path to vpnc-bin", "/usr/sbin/vpnc");
+    pathToVpnc = configgroup.readEntry("Path to vpnc-bin", "/usr/bin/vpnc");
     pathToCiscoVpnc = configgroup.readEntry("Path to vpnclient", "/usr/local/bin/vpnclient");
     pathToSetkey = configgroup.readEntry("Path to setkey", "/usr/sbin/setkey");
     pathToRacoon = configgroup.readEntry("Path to racoon", "/usr/sbin/racoon");
@@ -1489,7 +1489,7 @@
 void KVpncConfig::checkStandardPathValues()
 {
     if (pathToVpnc.isEmpty())
-        pathToVpnc = "/usr/sbin/vpnc";
+        pathToVpnc = "/usr/bin/vpnc";
 
     if (pathToCiscoVpnc.isEmpty())
         pathToCiscoVpnc = "/usr/local/bin/vpnclient";
