diff -wbBur Unreal3.2/doc/example.conf Unreal3.2.my/doc/example.conf
--- Unreal3.2/doc/example.conf	2009-04-13 15:03:55.000000000 +0400
+++ Unreal3.2.my/doc/example.conf	2011-08-23 15:48:46.000000000 +0400
@@ -33,8 +33,8 @@
  */
 
 /* FOR *NIX, uncomment the following 2lines: */
-//loadmodule "src/modules/commands.so";
-//loadmodule "src/modules/cloak.so";
+loadmodule "modules/commands.so";
+loadmodule "modules/cloak.so";
 
 /* FOR Windows, uncomment the following 2 lines: */
 //loadmodule "modules/commands.dll";
diff -wbBur Unreal3.2/src/ircd.c Unreal3.2.my/src/ircd.c
--- Unreal3.2/src/ircd.c	2009-03-01 21:37:58.000000000 +0300
+++ Unreal3.2.my/src/ircd.c	2011-08-23 15:44:30.000000000 +0400
@@ -844,7 +844,7 @@
 const char *compiledfor, *runtime;
 int error = 0;
 
-#ifdef USE_SSL
+#if 0
 	compiledfor = OPENSSL_VERSION_TEXT;
 	runtime = SSLeay_version(SSLEAY_VERSION);
 	if (strcasecmp(compiledfor, runtime))
@@ -854,7 +854,7 @@
 		error=1;
 	}
 #endif
-#ifdef ZIP_LINKS
+#if 0
 	runtime = zlibVersion();
 	compiledfor = ZLIB_VERSION;
 	if (*compiledfor != *runtime)
@@ -864,7 +864,7 @@
 		error = 1;
 	}
 #endif
-#ifdef USE_LIBCURL
+#if 0
 	/* Perhaps someone should tell them to do this a bit more easy ;)
 	 * problem is runtime output is like: 'libcurl/7.11.1 zlib/1.2.1 c-ares/1.2.0'
 	 * while header output is like: '7.11.1'.
diff -wbBur Unreal3.2/unreal.in Unreal3.2.my/unreal.in
--- Unreal3.2/unreal.in	2009-03-01 21:37:55.000000000 +0300
+++ Unreal3.2.my/unreal.in	2011-08-23 15:49:20.000000000 +0400
@@ -7,7 +7,7 @@
 	if [ -r $PID_FILE ] ; then
 		mv -f $PID_FILE $PID_BACKUP
 	fi
-	@BINDIR@
+	@BINDIR@/ircd
 	sleep 1
 	if [ ! -r $PID_FILE ] ; then
 		echo "Possible error encountered (IRCd seemingly not started)"	
