diff -wbBur phpldapadmin-1.2.3/lib/functions.php phpldapadmin-1.2.3.my/lib/functions.php
--- phpldapadmin-1.2.3/lib/functions.php	2012-10-01 10:54:14.000000000 +0400
+++ phpldapadmin-1.2.3.my/lib/functions.php	2017-02-02 20:02:14.424245233 +0300
@@ -745,6 +745,7 @@
 	if (! trim($secret))
 		return $data;
 
+/*
 	if (function_exists('mcrypt_module_open') && ! empty($data)) {
 		$td = mcrypt_module_open(MCRYPT_BLOWFISH,'',MCRYPT_MODE_ECB,'');
 		$iv = mcrypt_create_iv(mcrypt_enc_get_iv_size($td),MCRYPT_DEV_URANDOM);
@@ -754,7 +755,7 @@
 
 		return $encrypted_data;
 	}
-
+*/
 	if (file_exists(LIBDIR.'blowfish.php'))
 		require_once LIBDIR.'blowfish.php';
 	else
@@ -801,6 +802,7 @@
 	if (! trim($secret))
 		return $encdata;
 
+/*
 	if (function_exists('mcrypt_module_open') && ! empty($encdata)) {
 		$td = mcrypt_module_open(MCRYPT_BLOWFISH,'',MCRYPT_MODE_ECB,'');
 		$iv = mcrypt_create_iv(mcrypt_enc_get_iv_size($td),MCRYPT_DEV_URANDOM);
@@ -810,6 +812,7 @@
 
 		return $decrypted_data;
 	}
+*/
 
 	if (file_exists(LIBDIR.'blowfish.php'))
 		require_once LIBDIR.'blowfish.php';
