diff -aur xmonad-contrib-0.11/XMonad/Hooks/DebugEvents.hs xmonad-contrib.new/XMonad/Hooks/DebugEvents.hs
--- xmonad-contrib-0.11/XMonad/Hooks/DebugEvents.hs	2013-01-01 01:32:44.000000000 +0000
+++ xmonad-contrib.new/XMonad/Hooks/DebugEvents.hs	2013-03-18 07:54:52.082723862 +0000
@@ -50,6 +50,7 @@
 import           System.Process
 
 -- this should move into X11
+{-
 foreign import ccall unsafe "XGetAtomName"
   xGetAtomName :: Display -> Atom -> IO CString
 
@@ -63,6 +64,7 @@
       xFree n
       return $ Just n'
 
+-}
 -- | Event hook to dump all received events.  You should probably not use this
 --   unconditionally; it will produce massive amounts of output.
 debugEventsHook   :: Event -> X All
