Index: haveged-1.7/src/Makefile.am
===================================================================
--- haveged-1.7.orig/src/Makefile.am
+++ haveged-1.7/src/Makefile.am
@@ -27,7 +27,7 @@ pkginclude_HEADERS = havege.h
 
 haveged_SOURCES = haveged.c haveged.h
 
-haveged_LDADD = @HA_LDFLAGS@ $(top_builddir)/src/libhavege.la
+haveged_LDADD = @HA_LDFLAGS@ libhavege.la
 ##libtool_end##
 
 MAINTAINERCLEANFILES = Makefile.in
