From 4b838c04fa90ab56974bc4c57aaf7d4e80b8fff8 Mon Sep 17 00:00:00 2001
From: Vitozz <thetvg@gmail.com>
Date: Mon, 28 May 2018 20:45:04 +0300
Subject: [PATCH] Try to fix build with qt-5.11

---
 src/accountmanagedlg.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/accountmanagedlg.cpp b/src/accountmanagedlg.cpp
index 3770906d..6150349f 100644
--- a/src/accountmanagedlg.cpp
+++ b/src/accountmanagedlg.cpp
@@ -28,6 +28,7 @@
 #include <QTimer>
 #include <QHeaderView>
 #include <QDropEvent>
+#include <QButtonGroup>
 
 #include "psicon.h"
 #include "psiaccount.h"
