diff --git a/clang/test/Driver/SOURCE_DATE_EPOCH.c b/clang/test/Driver/SOURCE_DATE_EPOCH.c
--- a/clang/test/Driver/SOURCE_DATE_EPOCH.c
+++ b/clang/test/Driver/SOURCE_DATE_EPOCH.c
@@ -1,5 +1,2 @@
-// RUN: %clang -E %s -### 2>&1 | FileCheck %s -check-prefix=NO_EPOCH
-// NO_EPOCH-NOT: "-source-date-epoch"
-
 // RUN: env SOURCE_DATE_EPOCH=123 %clang -E %s -### 2>&1 | FileCheck %s
 // CHECK: "-source-date-epoch" "123"

