Index: src/VBox/Runtime/r0drv/linux/mpnotification-r0drv-linux.c
===================================================================
--- src/VBox/Runtime/r0drv/linux/mpnotification-r0drv-linux.c	(revision 73165)
+++ src/VBox/Runtime/r0drv/linux/mpnotification-r0drv-linux.c	(revision 73166)
@@ -32,6 +32,7 @@
 #include "internal/iprt.h"
 
 #include <iprt/mp.h>
+#include <iprt/asm-amd64-x86.h>
 #include <iprt/err.h>
 #include <iprt/cpuset.h>
 #include <iprt/thread.h>
@@ -82,7 +83,8 @@
     NOREF(pvUser1);
 
     AssertRelease(!RTThreadPreemptIsEnabled(NIL_RTTHREAD));
-    AssertRelease(idCpu == RTMpCpuId());   /* ASSUMES iCpu == RTCPUID */
+    AssertReleaseMsg(idCpu == RTMpCpuId(),  /* ASSUMES iCpu == RTCPUID */
+                     ("idCpu=%u RTMpCpuId=%d ApicId=%d\n", idCpu, RTMpCpuId(), ASMGetApicId() ));
 
     switch (ulNativeEvent)
     {