diff --git a/hphp/compiler/analysis/emitter.cpp b/hphp/compiler/analysis/emitter.cpp
index ba33bbd..38fe9ed 100644
--- a/hphp/compiler/analysis/emitter.cpp
+++ b/hphp/compiler/analysis/emitter.cpp
@@ -124,6 +124,10 @@ namespace HPHP {
 namespace Compiler {
 ///////////////////////////////////////////////////////////////////////////////
 
+#ifndef MAX
+#define MAX(a,b) ((a)<(b)?(b):(a))
+#endif
+
 TRACE_SET_MOD(emitter)
 
 using uchar = unsigned char;
diff --git a/hphp/runtime/base/zend-pack.cpp b/hphp/runtime/base/zend-pack.cpp
index d878ea4..01034c2 100644
--- a/hphp/runtime/base/zend-pack.cpp
+++ b/hphp/runtime/base/zend-pack.cpp
@@ -23,6 +23,10 @@
 
 namespace HPHP {
 
+#ifndef MAX
+#define MAX(a,b) ((a)<(b)?(b):(a))
+#endif
+
 #define INC_OUTPUTPOS(a,b)                                              \
   if ((a) < 0 || ((INT_MAX - outputpos)/((int)b)) < (a)) {              \
     throw_invalid_argument                                              \
