diff --git a/CMake/HPHPSetup.cmake b/CMake/HPHPSetup.cmake
index 48e7da7..7585b24 100644
--- a/CMake/HPHPSetup.cmake
+++ b/CMake/HPHPSetup.cmake
@@ -230,15 +230,16 @@ if (NOT FASTLZ_LIBRARY)
   include_directories("${TP_DIR}/fastlz")
 endif()
 
-if (ENABLE_MCROUTER)
-  include_directories("${TP_DIR}/mcrouter")
-endif()
-
 include_directories("${TP_DIR}/timelib")
 include_directories("${TP_DIR}/libafdt/src")
 include_directories("${TP_DIR}/libmbfl")
 include_directories("${TP_DIR}/libmbfl/mbfl")
 include_directories("${TP_DIR}/libmbfl/filter")
+
+if (ENABLE_MCROUTER)
+  include_directories("${TP_DIR}/mcrouter")
+endif()
+
 add_definitions(-DNO_LIB_GFLAGS)
 include_directories("${TP_DIR}/folly")
 include_directories("${TP_DIR}/thrift")
