diff --git i/meson_post_install.py w/meson_post_install.py
index e742310..b8cd360 100644
--- i/meson_post_install.py
+++ w/meson_post_install.py
@@ -29,7 +29,9 @@ except KeyError:
 
 dst = os.path.join(bindir, 'pkexec')
 
-if os.geteuid() == 0:
+if not os.path.exists(dst):
+    pass
+elif os.geteuid() == 0:
     os.chown(dst, 0, -1)
     os.chmod(dst, 0o4755)
 else:
@@ -62,7 +64,9 @@ for dst in dst_dirs:
 # and/or other users.
 dst = os.path.join(pkglibdir, 'polkit-agent-helper-1')
 
-if os.geteuid() == 0:
+if not os.path.exists(dst):
+    pass
+elif os.geteuid() == 0:
     os.chown(dst, 0, -1)
     os.chmod(dst, 0o4755)
 else:
