--- astrometry.net-0.64/Makefile.orig	2015-11-21 20:36:45.915734989 +0100
+++ astrometry.net-0.64/Makefile	2015-11-21 20:37:06.019071691 +0100
@@ -67,7 +67,7 @@
 
 # Detect GSL -- this minimum version was chosen to match the version in gsl-an.
 # Earlier versions would probably work fine.
-SYSTEM_GSL ?= $(shell (pkg-config --atleast-version=1.14 gsl && echo "yes") || echo "no")
+SYSTEM_GSL := yes
 
 ifneq ($(SYSTEM_GSL),yes)
 thirdparty: gsl-an
