diff -wbBur ejabberd-13.10/src/ejabberd_c2s.erl ejabberd-13.10.q/src/ejabberd_c2s.erl
--- ejabberd-13.10/src/ejabberd_c2s.erl	2013-09-27 13:19:22.000000000 +0400
+++ ejabberd-13.10.q/src/ejabberd_c2s.erl	2013-11-16 03:37:15.525501597 +0400
@@ -536,8 +536,8 @@
 				[StateData#state.socket,
 				 jlib:jid_to_string(JID), AuthModule]),
 			SID = {now(), self()},
-			Conn = (StateData#state.sockmod):get_conn_type(
-				    StateData#state.socket),
+			Conn = get_conn_type(
+				    StateData),
 			Info = [{ip, StateData#state.ip}, {conn, Conn},
 				    {auth_module, AuthModule}],
                         Res = jlib:make_result_iq_reply(
