--- knemo.orig/src/common/utils.cpp	2016-05-07 14:30:45.888889978 +0000
+++ knemo/src/common/utils.cpp	2016-05-07 14:32:06.985290816 +0000
@@ -41,6 +41,7 @@
 #include <KSharedConfig>
 #include <Plasma/Theme>
 #include "data.h"
+#include "math.h"
 #include "utils.h"
 
 #ifdef __linux__
--- knemo.orig/src/knemod/trayicon.cpp	2015-10-03 17:22:07.000000000 +0000
+++ knemo/src/knemo/src/knemod/trayicon.cpp	2016-05-07 14:42:30.796566437 +0000
@@ -22,6 +22,7 @@
 #include "global.h"
 #include "interface.h"
 #include "knemodaemon.h"
+#include "math.h"
 #include "utils.h"
 
 #ifdef __linux__
