diff -ruN a/AbletonLinkConfig.cmake b/AbletonLinkConfig.cmake
--- a/AbletonLinkConfig.cmake	2018-05-29 12:37:09.000000000 +0200
+++ b/AbletonLinkConfig.cmake	2020-03-21 20:19:34.040771008 +0100
@@ -5,7 +5,7 @@
 add_library(Ableton::Link IMPORTED INTERFACE)
 set_property(TARGET Ableton::Link APPEND PROPERTY
   INTERFACE_INCLUDE_DIRECTORIES
-  ${CMAKE_CURRENT_LIST_DIR}/include
+  ${CMAKE_CURRENT_LIST_DIR}/../../../include
 )
 
 # Force C++11 support for consuming targets
@@ -50,5 +50,5 @@
 
 set_property(TARGET Ableton::Link APPEND PROPERTY
   INTERFACE_SOURCES
-  ${CMAKE_CURRENT_LIST_DIR}/include/ableton/Link.hpp
+  ${CMAKE_CURRENT_LIST_DIR}/../../../include/ableton/Link.hpp
 )
