--- a/mscore/CMakeLists.txt
+++ b/mscore/CMakeLists.txt
@@ -32,7 +32,7 @@ set(GCC_VISIBILITY TRUE)
 set(HAS_AUDIOFILE TRUE)                   # requires libsndfile
 set(OSC           TRUE)                   # osc remote control
 
-set(BUILD_SCRIPTGEN TRUE)                 #Generate Qt script bindings. (collides with qtscriptgenerator project on several distrib)
+set(BUILD_SCRIPTGEN FALSE)                 #Generate Qt script bindings. (collides with qtscriptgenerator project on several distrib)
 set(STATIC_SCRIPT_BINDINGS FALSE)
 set(USE_SSE FALSE)
 
@@ -421,14 +421,14 @@ include (CPack)
 
 add_custom_target(lupdate
    COMMAND ${PROJECT_SOURCE_DIR}/gen-qt-projectfile ${PROJECT_SOURCE_DIR} > mscore.pro
-   COMMAND lupdate -noobsolete ${PROJECT_BINARY_DIR}/mscore.pro
+   COMMAND lupdate-qt4 -noobsolete ${PROJECT_BINARY_DIR}/mscore.pro
    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
    )
 
 add_custom_target(lrelease
    COMMAND ${PROJECT_SOURCE_DIR}/gen-qt-projectfile ${PROJECT_SOURCE_DIR} > mscore.pro
-   COMMAND lrelease ${PROJECT_SOURCE_DIR}/share/locale/*.ts
-   COMMAND lrelease ${PROJECT_SOURCE_DIR}/share/plugins/musescore.com/translations/*.ts
+   COMMAND lrelease-qt4 ${PROJECT_SOURCE_DIR}/share/locale/*.ts
+   COMMAND lrelease-qt4 ${PROJECT_SOURCE_DIR}/share/plugins/musescore.com/translations/*.ts
    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
    )
 
--- a/mscore/mscore/genft.cpp
+++ b/mscore/mscore/genft.cpp
@@ -27,7 +27,7 @@
 
 #include <ft2build.h>
 #include FT_FREETYPE_H
-#include <freetype/tttables.h>
+#include <freetype2/tttables.h>
 #include "al/xml.h"
 
 QMap<int, int> codemap;
--- a/mscore/mscore/mscore.cpp
+++ b/mscore/mscore/mscore.cpp
@@ -919,15 +919,15 @@
       if (debugMode)
             printf("open handbook for language <%s>\n", qPrintable(lang));

-      QFileInfo mscoreHelp(mscoreGlobalShare + QString("man/MuseScore-") + lang + QString(".pdf"));
+      QFileInfo mscoreHelp(("/usr/share/doc/musescore/MuseScore-") + lang + QString(".pdf"));
       if (!mscoreHelp.isReadable()) {
             if (debugMode) {
                   printf("cannot open doc <%s>\n", qPrintable(mscoreHelp.filePath()));
                   }
             lang = lang.left(2);
-            mscoreHelp.setFile(mscoreGlobalShare + QString("man/MuseScore-") + lang + QString(".pdf"));
+            mscoreHelp.setFile(("/usr/share/doc/musescore/MuseScore-") + lang + QString(".pdf"));
             if(!mscoreHelp.isReadable()){
-                mscoreHelp.setFile(mscoreGlobalShare + QString("man/MuseScore-en.pdf"));
+                mscoreHelp.setFile(("/usr/share/doc/musescore/MuseScore-en.pdf"));
                 if (!mscoreHelp.isReadable()) {
                       QString info(tr("MuseScore handbook not found at: \n"));
                       info += mscoreHelp.filePath();
