diff -up rssguard-3.6.3/src/librssguard/network-web/googlesuggest.h.orig rssguard-3.6.3/src/librssguard/network-web/googlesuggest.h
--- rssguard-3.6.3/src/librssguard/network-web/googlesuggest.h.orig	2020-06-03 08:35:36.000000000 +0200
+++ rssguard-3.6.3/src/librssguard/network-web/googlesuggest.h	2020-06-19 22:18:15.929983840 +0200
@@ -33,8 +33,11 @@
 
 #include <QObject>
 
+#include <QListWidget>
 #include <QNetworkReply>
 
+#include "network-web/downloader.h"
+
 class LocationLineEdit;
 
 class QTimer;
