From 51294ef3e472aee3008583750f6e2dbe4c8c5adb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Johannes=20L=C3=B6thberg?= <johannes@kyriasis.com>
Date: Sat, 21 Jan 2017 20:08:12 +0100
Subject: [PATCH] Make bleach allow for vendored html5lib version
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Johannes Löthberg <johannes@kyriasis.com>
---
 bleach/__init__.py | 4 ++++
 setup.py           | 3 ---
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/bleach/__init__.py b/bleach/__init__.py
index 09dad63..c6f5929 100644
--- a/bleach/__init__.py
+++ b/bleach/__init__.py
@@ -3,6 +3,10 @@
 from __future__ import unicode_literals
 import logging
 import re
+import os
+import sys
+
+sys.path.insert(0, os.path.dirname(__file__))
 
 import html5lib
 from html5lib.sanitizer import HTMLSanitizer
diff --git a/setup.py b/setup.py
index 908928e..bdcecf6 100644
--- a/setup.py
+++ b/setup.py
@@ -15,9 +15,6 @@ tests_require = [
     
 install_requires = [
     'six',
-    # 3 9s up to but not including 8 9s, but not 4 9s or 5 9s because they're
-    # busted
-    'html5lib>=0.999,!=0.9999,!=0.99999,<0.99999999',
 ]
 
 
-- 
2.11.0

