diff --git a/keepalived/Makefile.am b/keepalived/Makefile.am
index 86c3ed3a..9a47eb61 100644
--- a/keepalived/Makefile.am
+++ b/keepalived/Makefile.am
@@ -75,26 +75,38 @@ init_DATA		= etc/openrc/keepalived
 endif
 
 if WITH_IPVS
+# checks for realpath, and also not busybox version which does not support --relative-to
 install-exec-hook:
+	$(MKDIR_P) $(DESTDIR)/$(bindir)
 	@( \
-	d=`echo $(bindir) | sed -e "s:^/::"`; \
-	s=`echo $(sbindir) | sed -e "s:^/::"`; \
-	\
-	while [ 1 ]; do \
-		d1=`echo $$d | sed -e "s:/.*::"`; \
-		s1=`echo $$s | sed -e "s:/.*::"`; \
-		if [ $$d1 != $$s1 ]; then \
-			break; \
+	realpath --relative-to=/ / >/dev/null 2>&1; \
+	if [ $$? -eq 0 ]; then \
+		$(LN_S) `realpath --relative-to="$(DESTDIR)/$(bindir)" "$(DESTDIR)/$(sbindir)/keepalived"` $(DESTDIR)/$(bindir)/genhash; \
+	else \
+		if [ $(bindir) = $(sbindir) ]; then \
+			d= ; \
+			s= ; \
+		else \
+			d=`echo $(bindir) | sed -e "s:^/::"`; \
+			s=`echo $(sbindir) | sed -e "s:^/::"`; \
+			\
+			while [ 1 ]; do \
+				d1=`echo $$d | sed -e "s:/.*::"`; \
+				s1=`echo $$s | sed -e "s:/.*::"`; \
+				if [ $$d1 != $$s1 ]; then \
+					break; \
+				fi; \
+				d=`echo $$d | sed -e "s:^[^/]*/::"`; \
+				s=`echo $$s | sed -e "s:^[^/]*/::"`; \
+				if [ -z $$d ]; then break; fi; \
+				if [ -z $$s ]; then break; fi; \
+			done; \
+			\
+			d=`echo /$$d/ | sed -e "s:/[^/.]*/:/../:g" -e "s:/[^/.]*/:/../:g" -e "s:^/::"`; \
+			s=$$s/ ; \
 		fi; \
-		d=`echo $$d | sed -e "s:^[^/]*/::"`; \
-		s=`echo $$s | sed -e "s:^[^/]*/::"`; \
-		if [ -z $$d ]; then break; fi; \
-		if [ -z $$s ]; then break; fi; \
-	done; \
-	\
-	d=`echo /$$d/ | sed -e "s:/[^/.]*/:/../:g" -e "s:/[^/.]*/:/../:g" -e "s:^/::"`; \
-	\
-	$(MKDIR_P) $(DESTDIR)/$(bindir); \
-	$(LN_S) $$d$$s/keepalived $(DESTDIR)/$(bindir)/genhash; \
+		\
+		$(LN_S) $$d$${s}keepalived $(DESTDIR)/$(bindir)/genhash; \
+	fi; \
 	)
 endif
