From 200121f93d7c6ac9aadb53c58f45906ecd1e3b7a Mon Sep 17 00:00:00 2001
From: Julien Lavergne <julien.lavergne@gmail.com>
Date: Sun, 24 Jul 2011 15:43:24 +0200
Subject: [PATCH] Fix usage of GtkRequisition

---
 src/plugins/menu.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/plugins/menu.c b/src/plugins/menu.c
index 59a8dad..6d0ad9c 100644
--- a/src/plugins/menu.c
+++ b/src/plugins/menu.c
@@ -142,10 +142,10 @@ menu_pos(GtkMenu *menu, gint *x, gint *y, gboolean *push_in, GtkWidget *widget)
     gdk_window_get_origin(widget->window, &ox, &oy);
 #endif
 #if GTK_CHECK_VERSION(2,20,0)
-    GtkRequisition *requisition;
-    gtk_widget_get_requisition(GTK_WIDGET(menu),requisition);
-    w = requisition->width;
-    h = requisition->height;
+    GtkRequisition requisition;
+    gtk_widget_get_requisition(GTK_WIDGET(menu), &requisition);
+    w = requisition.width;
+    h = requisition.height;
 
 #else
     w = GTK_WIDGET(menu)->requisition.width;
-- 
1.7.0.1
