diff -ruN rtaudio-5.0.0-a/configure.ac rtaudio-5.0.0-b/configure.ac
--- rtaudio-5.0.0-a/configure.ac	2017-08-30 23:30:36.000000000 +0200
+++ rtaudio-5.0.0-b/configure.ac	2018-01-31 22:04:03.896801359 +0100
@@ -127,7 +127,9 @@
 AS_IF([test "x$with_jack" == "xyes"], [
   api="$api -D__UNIX_JACK__"
   AC_MSG_RESULT(using JACK)
-  AC_CHECK_LIB(jack, jack_client_open, , AC_MSG_ERROR(JACK support requires the jack library!))])
+  AC_CHECK_LIB(jack, jack_client_open, , AC_MSG_ERROR(JACK support requires the jack library!))
+  req="$req jack"
+])
 
 case $host in
   *-*-netbsd*)
