=== modified file 'terminatorlib/terminator.py'
--- terminatorlib/terminator.py	2017-02-26 16:15:11 +0000
+++ terminatorlib/terminator.py	2017-02-28 16:04:37 +0000
@@ -401,7 +401,7 @@
             window.grab_focus()
             try:
                 t = GdkX11.x11_get_server_time(window.get_window())
-            except AttributeError:
+            except (TypeError, AttributeError):
                 t = 0
             window.get_window().focus(t)
 
@@ -417,7 +417,7 @@
                 window.grab_focus()
                 try:
                     t = GdkX11.x11_get_server_time(window.get_window())
-                except AttributeError:
+                except (TypeError, AttributeError):
                     t = 0
                 window.get_window().focus(t)
 

=== modified file 'terminatorlib/window.py'
--- terminatorlib/window.py	2017-02-13 16:30:37 +0000
+++ terminatorlib/window.py	2017-02-28 16:07:28 +0000
@@ -311,7 +311,7 @@
             self.grab_focus()
             try:
                 t = GdkX11.x11_get_server_time(self.get_window())
-            except AttributeError:
+            except (TypeError, AttributeError):
                 t = 0
             self.get_window().focus(t)
         else:

