From a65eb174adaa08cce0029ac38c5bc58ff8adaa7b Mon Sep 17 00:00:00 2001
From: Ken <ken@kenh.fr>
Date: Mon, 21 Dec 2020 19:43:06 +0100
Subject: [PATCH] fixed 'validate' module import for newer version of configobj

---
 terminatorlib/config.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/terminatorlib/config.py b/terminatorlib/config.py
index ec7a2a2e..ff37597d 100644
--- a/terminatorlib/config.py
+++ b/terminatorlib/config.py
@@ -74,7 +74,10 @@ import os
 import shutil
 from copy import copy
 from configobj import ConfigObj, flatten_errors
-from validate import Validator
+try:
+    from validate import Validator
+except ModuleNotFoundError:
+    from configobj.validate import Validator
 from .borg import Borg
 from .util import dbg, err, DEBUG, get_system_config_dir, get_config_dir, dict_diff
 
-- 
2.29.2

