From 1073a282e797efb28695aaec69204b2dd59288ab Mon Sep 17 00:00:00 2001
From: Johnny Jazeix <jazeix@gmail.com>
Date: Sat, 18 Feb 2017 22:38:08 +0100
Subject: [PATCH 3/3] chess, allow to move again the same piece after undo

---
 src/activities/chess/chess.js | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/activities/chess/chess.js b/src/activities/chess/chess.js
index 3b9b7b41..1fde004b 100644
--- a/src/activities/chess/chess.js
+++ b/src/activities/chess/chess.js
@@ -238,6 +238,8 @@ function undo() {
         redo_stack.push(state.history[state.moveno - 1])
         state.jump_to_moveno(state.moveno - 1)
     }
+    // without it, you can't move again the same piece
+    Engine.p4_prepare(state)
     items.redo_stack = redo_stack
     refresh()
     items.positions = [] // Force a model reload
@@ -311,8 +313,9 @@ function clearAcceptMove() {
 function showPossibleMoves(from) {
     var result = Engine.p4_parse(state, state.to_play, 0, 0)
     clearAcceptMove()
+    var fromEngine = viewPosToEngine(from)
     for(var i=0; i < result.length; ++i) {
-        if(viewPosToEngine(from) == result[i][1]) {
+        if(fromEngine == result[i][1]) {
             var pos = engineToViewPos(result[i][2])
             items.squares.getSquareAt(pos)['acceptMove'] = true
         }
-- 
2.12.2

