From d53989d7c8051ae0be95a91725646bd4f5fe38b9 Mon Sep 17 00:00:00 2001
From: Daniel Nicoletti <dantti12@gmail.com>
Date: Wed, 7 May 2014 16:16:25 -0300
Subject: [PATCH 04/12] Remove RoleUpgradeSystem as it's not supported in 0.9.x
 anymore

---
 src/transaction.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/transaction.h b/src/transaction.h
index 2b03f41..7b17fc5 100644
--- a/src/transaction.h
+++ b/src/transaction.h
@@ -140,7 +140,6 @@ public:
         RoleGetDistroUpgrades,
         RoleGetCategories,
         RoleGetOldTransactions,
-        RoleUpgradeSystem,      // Since 0.6.11
         RoleRepairSystem,       // Since 0.7.2
         RoleGetDetailsLocal,    // Since 0.8.17
         RoleGetFilesLocal,      // Since 0.9.1
-- 
2.0.1

