From 00fec992f5c760f546a1db933fb08325fc9cccbf Mon Sep 17 00:00:00 2001
From: Peng Wu <alexepico@gmail.com>
Date: Sat, 23 Mar 2013 09:52:46 +0800
Subject: [PATCH] fixes import dict

---
 src/storage/phrase_large_table2.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/storage/phrase_large_table2.h b/src/storage/phrase_large_table2.h
index 368d03d..34bb12d 100644
--- a/src/storage/phrase_large_table2.h
+++ b/src/storage/phrase_large_table2.h
@@ -133,7 +133,7 @@ static inline int reduce_tokens(const PhraseTokens tokens,
     }
 
     /* the following line will be removed in future after code are verified. */
-    assert(0 == num || 1 == num);
+    assert(0 <= num && num <= 3);
 
     return num;
 }
-- 
1.8.1.5

