diff -rupN corefx.707cf4f62ea1fdbca730c910c4c18c0d3217412e.orig/src/Native/Unix/CMakeLists.txt corefx.707cf4f62ea1fdbca730c910c4c18c0d3217412e/src/Native/Unix/CMakeLists.txt
--- corefx.707cf4f62ea1fdbca730c910c4c18c0d3217412e.orig/src/Native/Unix/CMakeLists.txt	2021-07-27 14:26:14.020698478 +0000
+++ corefx.707cf4f62ea1fdbca730c910c4c18c0d3217412e/src/Native/Unix/CMakeLists.txt	2021-07-27 14:26:52.279672576 +0000
@@ -26,6 +26,8 @@ add_compile_options(-Wno-empty-translati
 add_compile_options(-Wno-cast-align)
 add_compile_options(-Wno-typedef-redefinition)
 add_compile_options(-Wno-c11-extensions)
+add_compile_options(-Wno-sign-conversion)
+add_compile_options(-Wno-unused-result)
 add_compile_options(-I${CMAKE_CURRENT_SOURCE_DIR}/Common)
 add_compile_options(-I${CMAKE_CURRENT_BINARY_DIR}/Common)
 add_compile_options(-g)

