diff --git a/src/actions/action.vala b/src/actions/action.vala
index ee13b39..bf64b06 100644
--- a/src/actions/action.vala
+++ b/src/actions/action.vala
@@ -69,7 +69,7 @@ public abstract class Action : GLib.Object {
     /// C'tor, initializes all members.
     /////////////////////////////////////////////////////////////////////
 
-    public Action(string name, string icon, bool is_quickaction) {
+    Action(string name, string icon, bool is_quickaction) {
         GLib.Object(name : name, icon : icon, is_quickaction : is_quickaction);
     }
 
