--- appliance/packagelist.in	2019-10-14 22:47:20.974379583 +0000
+++ appliance/packagelist.in	2019-10-14 22:51:28.587897142 +0000
@@ -118,11 +118,14 @@
   iputils
   jansson
   libcap
+  libldm
   libtirpc
   linux
   lrzip
+  lvm2
   dnl syslinux has mtools as optional dependency, but in reality it's
   dnl a hard one:
+  mdadm
   mtools
   multipath-tools  dnl for kpartx
   nilfs-utils
@@ -130,6 +133,7 @@
   pcre
   reiserfsprogs
   systemd
+  systemd-sysvcompat
   vim
   xz
   yara
