diff -Naur notify-sharp-3.0.0.orig/configure.ac notify-sharp-3.0.0/configure.ac
--- notify-sharp-3.0.0.orig/configure.ac	2013-11-03 15:59:42.000000000 +0100
+++ notify-sharp-3.0.0/configure.ac	2014-04-08 11:20:22.477915534 +0200
@@ -9,6 +9,9 @@
 AC_SUBST(API_VERSION)
 AC_SUBST(VERSION)
 
+PACKAGE_VERSION=notify-sharp-3.0
+AC_SUBST(PACKAGE_VERSION)
+
 AM_INIT_AUTOMAKE(notify-sharp, $VERSION)
 
 AM_MAINTAINER_MODE
diff -Naur notify-sharp-3.0.0.orig/notify-sharp-3.0.pc.in notify-sharp-3.0.0/notify-sharp-3.0.pc.in
--- notify-sharp-3.0.0.orig/notify-sharp-3.0.pc.in	2013-11-03 15:57:49.000000000 +0100
+++ notify-sharp-3.0.0/notify-sharp-3.0.pc.in	2014-04-08 11:20:39.985024993 +0200
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
-pkglibdir=${libdir}/mono/notify-sharp
+pkglibdir=${libdir}/mono/@PACKAGE_VERSION@
 
 Name: notify-sharp
 Description: C# client library for notification-daemon
diff -Naur notify-sharp-3.0.0.orig/src/Makefile.am notify-sharp-3.0.0/src/Makefile.am
--- notify-sharp-3.0.0.orig/src/Makefile.am	2013-01-31 06:10:36.000000000 +0100
+++ notify-sharp-3.0.0/src/Makefile.am	2014-04-08 14:08:31.835246749 +0200
@@ -21,13 +21,13 @@
 install-data-local:
 	@if test -n '$(TARGET)'; then                       \
           echo "$(GACUTIL) /i $(TARGET) /f /gacdir $(DESTDIR)$(libdir)";    \
-          $(GACUTIL) /i $(TARGET) /package $(ASSEMBLY) /f /gacdir $(libdir) /root $(DESTDIR)$(libdir) || exit 1; \
+          $(GACUTIL) /i $(TARGET) /package $(PACKAGE_VERSION) /f /gacdir $(libdir) /root $(DESTDIR)$(libdir) || exit 1; \
         fi
 
 uninstall-local:
 	@if test -n '$(TARGET)'; then                       \
           echo "$(GACUTIL) /u $(ASSEMBLY) /gacdir $(DESTDIR)$(libdir)";  \
-          $(GACUTIL) /u $(ASSEMBLY) /package $(ASSEMBLY) /gacdir $(libdir) /root $(DESTDIR)$(libdir) || exit 1;   \
+          $(GACUTIL) /u $(ASSEMBLY) /package $(PACKAGE_VERSION) /gacdir $(libdir) /root $(DESTDIR)$(libdir) || exit 1;   \
         fi
 
 EXTRA_DIST = \
