diff -Naurd bless-0.6.0.orig/src/tools/Preferences.cs bless-0.6.0/src/tools/Preferences.cs
--- bless-0.6.0.orig/src/tools/Preferences.cs	2008-06-07 14:18:14.000000000 +0200
+++ bless-0.6.0/src/tools/Preferences.cs	2016-10-28 22:47:50.000000000 +0200
@@ -127,10 +127,10 @@
 	///</summary>
 	public void Save(string path)
 	{
-		XmlTextWriter xml = new XmlTextWriter(path, null);
-		xml.Formatting = Formatting.Indented;
-		xml.Indentation = 1;
-		xml.IndentChar = '\t';
+		XmlWriterSettings settings = new XmlWriterSettings();
+		settings.Indent = true;
+		settings.IndentChars = ("\t");
+		XmlWriter xml = XmlWriter.Create(path, settings);
 
 		xml.WriteStartElement(null, "preferences", null);
 
diff -Naurd bless-0.6.0.orig/src/tools/Session.cs bless-0.6.0/src/tools/Session.cs
--- bless-0.6.0.orig/src/tools/Session.cs	2008-06-07 14:18:14.000000000 +0200
+++ bless-0.6.0/src/tools/Session.cs	2017-01-08 15:21:29.000000000 +0100
@@ -72,10 +72,10 @@
 
 	public void Save(string path)
 	{
-		XmlTextWriter xml = new XmlTextWriter(path, null);
-		xml.Formatting = Formatting.Indented;
-		xml.Indentation = 1;
-		xml.IndentChar = '\t';
+		XmlWriterSettings settings = new XmlWriterSettings();
+		settings.Indent = true;
+		settings.IndentChars = ("\t");
+		XmlWriter xml = XmlWriter.Create(path, settings);
 
 		xml.WriteStartElement(null, "session", null);
 
