--- logging/src/logging.cc.orig	2016-10-14 08:37:12.829303659 +0100
+++ logging/src/logging.cc	2016-10-14 08:38:21.799662293 +0100
@@ -82,7 +82,7 @@
         core->remove_all_sinks();
 
         using sink_t = sinks::synchronous_sink<color_writer>;
-        boost::shared_ptr<sink_t> sink(new sink_t(&dst));
+        boost::shared_ptr<sink_t> sink = boost::make_shared<sink_t>(boost::make_shared<color_writer>(&dst));
         core->add_sink(sink);
 
 
