diff -Naur spyder-3.2.7.orig/scripts/spyder3.desktop spyder-3.2.7/scripts/spyder3.desktop
--- spyder-3.2.7.orig/scripts/spyder3.desktop	2018-02-25 17:17:21.000000000 +0100
+++ spyder-3.2.7/scripts/spyder3.desktop	2018-03-03 20:15:01.330490153 +0100
@@ -1,9 +1,8 @@
 [Desktop Entry]
 Type=Application
 Version=1.0
-Name=Spyder3
-GenericName=Spyder3
-Comment=Scientific PYthon Development EnviRonment - Python3
+Name=Spyder for Python 3.x
+Comment=Scientific PYthon Development EnviRonment - Python 3.x
 Icon=spyder3
 TryExec=spyder3
 Exec=spyder3 %F
diff -Naur spyder-3.2.7.orig/scripts/spyder.desktop spyder-3.2.7/scripts/spyder.desktop
--- spyder-3.2.7.orig/scripts/spyder.desktop	2018-02-25 17:17:21.000000000 +0100
+++ spyder-3.2.7/scripts/spyder.desktop	2018-03-03 20:16:20.206886722 +0100
@@ -1,12 +1,12 @@
 [Desktop Entry]
 Type=Application
 Version=1.0
-Name=Spyder
+Name=Spyder for Python 2.x
 GenericName=Spyder
-Comment=Scientific PYthon Development EnviRonment
-Icon=spyder
-TryExec=spyder
-Exec=spyder %F
+Comment=Scientific PYthon Development EnviRonment - Python 2.x
+Icon=spyder2
+TryExec=spyder2
+Exec=spyder2 %F
 Terminal=false
 MimeType=text/x-python;
 Categories=Development;Science;IDE;Qt;
diff -Naur spyder-3.2.7.orig/setup.py spyder-3.2.7/setup.py
--- spyder-3.2.7.orig/setup.py	2018-02-25 17:17:21.000000000 +0100
+++ spyder-3.2.7/setup.py	2018-03-03 20:16:54.951582263 +0100
@@ -82,8 +82,8 @@
                           ('share/icons', ['img_src/spyder3.png']),
                           ('share/metainfo', ['scripts/spyder3.appdata.xml'])]
         else:
-            data_files = [('share/applications', ['scripts/spyder.desktop']),
-                          ('share/icons', ['img_src/spyder.png'])]
+            data_files = [('share/applications', ['scripts/spyder2.desktop']),
+                          ('share/icons', ['img_src/spyder2.png'])]
     elif os.name == 'nt':
         data_files = [('scripts', ['img_src/spyder.ico',
                                    'img_src/spyder_reset.ico'])]
