diff --git a/Gemfile b/Gemfile
index b6f57297c07..810ef2b1b54 100644
--- a/Gemfile
+++ b/Gemfile
@@ -375,7 +375,7 @@ group :development, :test do
   gem 'scss_lint', '~> 0.56.0', require: false
   gem 'haml_lint', '~> 0.34.0', require: false
   gem 'simplecov', '~> 0.16.1', require: false
-  gem 'bundler-audit', '~> 0.5.0', require: false
+  gem 'bundler-audit', '>= 0', require: false
 
   gem 'benchmark-ips', '~> 2.3.0', require: false
 
@@ -477,3 +477,6 @@ gem 'gitlab-net-dns', '~> 0.9.1'
 gem 'countries', '~> 3.0'
 
 gem 'retriable', '~> 3.1.2'
+
+gem 'irb', '~> 1.1.0'  # use irb version that matches ruby 2.6. Later irb versions heavily modified its API and it breaks gettext
+gem 'e2mmap' # 'irb' 1.1.1 requires it
