From a1cd313f48c9dce60016a1cb48de8ed7ec2179f0 Mon Sep 17 00:00:00 2001
From: Maxime Gauduin <alucryd@archlinux.org>
Date: Wed, 18 Mar 2020 20:05:50 +0100
Subject: [PATCH] use more sensible paths

---
 higan/target-byuu/byuu.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/higan/target-byuu/byuu.cpp b/higan/target-byuu/byuu.cpp
index 026df2f..e73f00d 100644
--- a/higan/target-byuu/byuu.cpp
+++ b/higan/target-byuu/byuu.cpp
@@ -10,7 +10,7 @@ auto locate(const string& name) -> string {
   string location = {Path::program(), name};
   if(inode::exists(location)) return location;
 
-  location = {Path::userData(), "byuu/", name};
+  location = {Path::sharedData(), "byuu/", name};
   if(inode::exists(location)) return location;
 
   directory::create({Path::userSettings(), "byuu/"});
