From a185a9bd8de1e5b860dcce7bccc5c71b1caec53a Mon Sep 17 00:00:00 2001
From: Maxime Gauduin <alucryd@archlinux.org>
Date: Wed, 18 Mar 2020 20:07:03 +0100
Subject: [PATCH] use system flags

---
 nall/GNUmakefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/nall/GNUmakefile b/nall/GNUmakefile
index 67fb3a1..fddec93 100755
--- a/nall/GNUmakefile
+++ b/nall/GNUmakefile
@@ -127,7 +127,8 @@ endif
 
 # linux settings
 ifeq ($(platform),linux)
-  options += -ldl
+  flags += ${CXXFLAGS}
+  options += ${LDFLAGS} -ldl
 endif
 
 # bsd settings
