--- carla-2.0.0/source/bridges-ui/CarlaBridgeToolkitQt.cpp.orig	2019-07-02 19:08:36.959433872 +0000
+++ carla-2.0.0/source/bridges-ui/CarlaBridgeToolkitQt.cpp	2019-07-02 19:09:24.813406097 +0000
@@ -256,12 +256,6 @@
 #endif
 };
 
-#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
-# include "CarlaBridgeToolkitQt5.moc"
-#else
-# include "CarlaBridgeToolkitQt4.moc"
-#endif
-
 // -------------------------------------------------------------------------
 
 CarlaBridgeToolkit* CarlaBridgeToolkit::createNew(CarlaBridgeFormat* const format)
@@ -285,4 +279,10 @@
 # include "resources.qt4.cpp"
 #endif
 
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+# include "CarlaBridgeToolkitQt5.moc"
+#else
+# include "CarlaBridgeToolkitQt4.moc"
+#endif
+
 // -------------------------------------------------------------------------
