diff -upr qmmp-0.6.5.orig/src/plugins/Input/cdaudio/decoder_cdaudio.cpp qmmp-0.6.5/src/plugins/Input/cdaudio/decoder_cdaudio.cpp
--- qmmp-0.6.5.orig/src/plugins/Input/cdaudio/decoder_cdaudio.cpp	2013-01-10 21:55:55.000000000 +0200
+++ qmmp-0.6.5/src/plugins/Input/cdaudio/decoder_cdaudio.cpp	2013-01-10 21:56:26.000000000 +0200
@@ -25,7 +25,7 @@
 #include <QFileInfo>
 #include <QDir>
 #include <cdio/cdio.h>
-#include <cdio/cdda.h>
+#include <cdio/paranoia/cdda.h>
 #include <cdio/audio.h>
 #include <cdio/cd_types.h>
 #include <cdio/logging.h>
