diff -Naur cinnamon-screensaver-1.7.1.orig/configure.ac cinnamon-screensaver-1.7.1/configure.ac
--- ./cinnamon-screensaver-1.7.1.orig/configure.ac	2013-02-19 17:35:35.000000000 +0100
+++ ./cinnamon-screensaver-1.7.1/configure.ac	2013-02-20 12:40:34.936078052 +0100
@@ -10,7 +10,7 @@
 AM_INIT_AUTOMAKE([1.10 no-dist-gzip dist-xz tar-ustar])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AM_MAINTAINER_MODE([enable])
 
