diff -ruN a/pkg/config/containers.conf b/pkg/config/containers.conf
--- a/pkg/config/containers.conf	2020-11-20 21:33:14.000000000 +0100
+++ b/pkg/config/containers.conf	2020-11-25 19:27:56.224267486 +0100
@@ -59,28 +59,28 @@
 # List of default capabilities for containers. If it is empty or commented out,
 # the default capabilities defined in the container engine will be added.
 #
-default_capabilities = [
-    "CHOWN",
-    "DAC_OVERRIDE",
-    "FOWNER",
-    "FSETID",
-    "KILL",
-    "NET_BIND_SERVICE",
-    "SETFCAP",
-    "SETGID",
-    "SETPCAP",
-    "SETUID",
-    "SYS_CHROOT"
-]
+# default_capabilities = [
+#     "CHOWN",
+#     "DAC_OVERRIDE",
+#     "FOWNER",
+#     "FSETID",
+#     "KILL",
+#     "NET_BIND_SERVICE",
+#     "SETFCAP",
+#     "SETGID",
+#     "SETPCAP",
+#     "SETUID",
+#     "SYS_CHROOT"
+# ]
 
 
 # A list of sysctls to be set in containers by default,
 # specified as "name=value",
 # for example:"net.ipv4.ping_group_range = 0 0".
 #
-default_sysctls = [
- "net.ipv4.ping_group_range=0 0",
-]
+# default_sysctls = [
+#  "net.ipv4.ping_group_range=0 0",
+# ]
 
 # A list of ulimits to be set in containers by default, specified as
 # "<ulimit name>=<soft limit>:<hard limit>", for example:
