diff --git a/src/hd/kbd.c b/src/hd/kbd.c
index fa08817..184ac08 100644
--- a/src/hd/kbd.c
+++ b/src/hd/kbd.c
@@ -134,6 +134,7 @@ void add_serial_console(hd_data_t *hd_data)
   }
 
   if(!dev && (fd = open(DEV_CONSOLE, O_RDWR | O_NONBLOCK | O_NOCTTY)) >= 0) {
+    /* Removing since TIOCGDEV is not supported by the mainline kernel
     if(ioctl(fd, TIOCGDEV, &u) != -1) {
       tty_major = (u >> 8) & 0xfff;
       tty_minor = (u & 0xff) | ((u >> 12) & 0xfff00);
@@ -152,6 +153,7 @@ void add_serial_console(hd_data_t *hd_data)
       free_mem(dev_link);
       free_mem(dev_name);
     }
+    */
 
     if (dev)
 	    ;
