diff -wbBurN dbmail.git/configure.in dbmail-3.1.5/configure.in
--- dbmail.git/configure.in	2013-09-05 18:07:48.661707165 +0400
+++ dbmail-3.1.5/configure.in	2013-09-09 15:05:17.639600103 +0400
@@ -44,10 +44,6 @@
     AS_HELP_STRING([--enable-manpages], [Enable building and installation of man pages]))
 AM_CONDITIONAL(MANPAGES, [ test "$enable_manpages" = "yes" ])
 
-AC_ARG_ENABLE([systemd],
-    AS_HELP_STRING([--enable-systemd], [Enable systemd support]))
-AM_CONDITIONAL(SYSTEMD, [ test "$enable_systemd" = "yes" ])
-
 AC_PROG_CC
 AC_COMPILE_WARNINGS
 AC_C_CONST
@@ -82,8 +78,9 @@
 AC_SUBST(DM_STATEDIR)
 AC_SUBST(DM_PKGLIBDIR)
 AC_SUBST(DM_PWD)
+
 AC_PROG_LIBTOOL
 
-AC_OUTPUT(src/dbmail.h Makefile src/Makefile src/modules/Makefile man/Makefile test/Makefile systemd/Makefile)
+AC_OUTPUT(src/dbmail.h Makefile src/Makefile src/modules/Makefile man/Makefile test/Makefile)
 
 DM_MSG_CONFIGURE_RESULTS
diff -wbBurN dbmail.git/systemd/dbmail-imapd.service.in dbmail-3.1.5/systemd/dbmail-imapd.service.in
--- dbmail.git/systemd/dbmail-imapd.service.in	2013-09-05 18:07:48.711707166 +0400
+++ dbmail-3.1.5/systemd/dbmail-imapd.service.in	1970-01-01 03:00:00.000000000 +0300
@@ -1,11 +0,0 @@
-[Unit]
-Description=DBMail Imap Server
-After=syslog.target network.target mysqld.service postgresql.service
-
-[Service]
-Type=forking
-PIDFile=@piddir@/dbmail-imapd.pid
-ExecStart=@sbindir@/dbmail-imapd -p @piddir@/dbmail-imapd.pid
-
-[Install]
-WantedBy=multi-user.target
diff -wbBurN dbmail.git/systemd/dbmail-lmtpd.service.in dbmail-3.1.5/systemd/dbmail-lmtpd.service.in
--- dbmail.git/systemd/dbmail-lmtpd.service.in	2013-09-05 18:07:48.711707166 +0400
+++ dbmail-3.1.5/systemd/dbmail-lmtpd.service.in	1970-01-01 03:00:00.000000000 +0300
@@ -1,11 +0,0 @@
-[Unit]
-Description=DBMail LMTP Server
-After=syslog.target network.target mysqld.service postgresql.service
-
-[Service]
-Type=forking
-PIDFile=@piddir@/dbmail-lmtpd.pid
-ExecStart=@sbindir@/dbmail-lmtpd -p @piddir@/dbmail-lmtpd.pid
-
-[Install]
-WantedBy=multi-user.target
diff -wbBurN dbmail.git/systemd/dbmail-pop3d.service.in dbmail-3.1.5/systemd/dbmail-pop3d.service.in
--- dbmail.git/systemd/dbmail-pop3d.service.in	2013-09-05 18:07:48.711707166 +0400
+++ dbmail-3.1.5/systemd/dbmail-pop3d.service.in	1970-01-01 03:00:00.000000000 +0300
@@ -1,11 +0,0 @@
-[Unit]
-Description=DBMail pop3 Server
-After=syslog.target network.target mysqld.service postgresql.service
-
-[Service]
-Type=forking
-PIDFile=@piddir@/dbmail-pop3d.pid
-ExecStart=@sbindir@/dbmail-pop3d -p @piddir@/dbmail-pop3d.pid
-
-[Install]
-WantedBy=multi-user.target
diff -wbBurN dbmail.git/systemd/dbmail-timsieved.service.in dbmail-3.1.5/systemd/dbmail-timsieved.service.in
--- dbmail.git/systemd/dbmail-timsieved.service.in	2013-09-05 18:07:48.711707166 +0400
+++ dbmail-3.1.5/systemd/dbmail-timsieved.service.in	1970-01-01 03:00:00.000000000 +0300
@@ -1,11 +0,0 @@
-[Unit]
-Description=DBMail Sieve Server
-After=syslog.target network.target mysqld.service postgresql.service
-
-[Service]
-Type=forking
-PIDFile=@piddir@/dbmail-timsieved.pid
-ExecStart=@sbindir@/dbmail-timsieved -p @piddir@/dbmail-timsieved.pid
-
-[Install]
-WantedBy=multi-user.target
diff -wbBurN dbmail.git/systemd/dbmail.tmpfiles.in dbmail-3.1.5/systemd/dbmail.tmpfiles.in
--- dbmail.git/systemd/dbmail.tmpfiles.in	2013-09-05 18:07:48.711707166 +0400
+++ dbmail-3.1.5/systemd/dbmail.tmpfiles.in	1970-01-01 03:00:00.000000000 +0300
@@ -1 +0,0 @@
-d @piddir@ 0755 nobody nobody -
diff -wbBurN dbmail.git/systemd/Makefile.am dbmail-3.1.5/systemd/Makefile.am
--- dbmail.git/systemd/Makefile.am	2013-09-07 16:12:34.840475834 +0400
+++ dbmail-3.1.5/systemd/Makefile.am	1970-01-01 03:00:00.000000000 +0300
@@ -1,68 +0,0 @@
-# Copyright (C) 1999-2004 IC & S  dbmail@ic-s.nl
-# Copyright (c) 2004-2011 NFG Net Facilities Group BV support@nfg.nl
-#
-# This program is free software; you can redistribute it and/or 
-# modify it under the terms of the GNU General Public License 
-# as published by the Free Software Foundation; either 
-# version 2 of the License, or (at your option) any later 
-# version.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
-
-
-EXTRA_DIST = dbmail-imapd.service.in  dbmail-lmtpd.service.in  dbmail-pop3d.service.in  dbmail-timsieved.service.in
-SYSTEMD_UNIT_DIR = /usr/lib/systemd/system
-SYSTEMD_TMPFILES_DIR = /usr/lib/tmpfiles.d
-PID_DIR=/run/dbmail
-
-if SYSTEMD
-install-systemd: dbmail-imapd.service dbmail-lmtpd.service dbmail-pop3d.service dbmail-timsieved.service dbmail.tmpfiles
-	$(MKDIR_P) $(DESTDIR)$(SYSTEMD_UNIT_DIR)
-	$(INSTALL_DATA) dbmail-imapd.service \
-	  $(DESTDIR)$(SYSTEMD_UNIT_DIR)/dbmail-imapd.service
-	$(INSTALL_DATA) dbmail-lmtpd.service \
-	  $(DESTDIR)$(SYSTEMD_UNIT_DIR)/dbmail-lmtpd.service
-	$(INSTALL_DATA) dbmail-pop3d.service \
-	  $(DESTDIR)$(SYSTEMD_UNIT_DIR)/dbmail-pop3d.service
-	$(INSTALL_DATA) dbmail-timsieved.service \
-	  $(DESTDIR)$(SYSTEMD_UNIT_DIR)/dbmail-timsieved.service
-	$(MKDIR_P) $(DESTDIR)$(SYSTEMD_TMPFILES_DIR)
-	$(INSTALL_DATA) dbmail.tmpfiles \
-	  $(DESTDIR)$(SYSTEMD_TMPFILES_DIR)/dbmail.conf
-
-uninstall-systemd:
-	rm -f $(DESTDIR)$(SYSTEMD_UNIT_DIR)/dbmail-imapd.service
-	rm -f $(DESTDIR)$(SYSTEMD_UNIT_DIR)/dbmail-lmtpd.service
-	rm -f $(DESTDIR)$(SYSTEMD_UNIT_DIR)/dbmail-pop3d.service
-	rm -f $(DESTDIR)$(SYSTEMD_UNIT_DIR)/dbmail-timsieved.service
-	rm -f $(DESTDIR)$(SYSTEMD_TMPFILES_DIR)/dbmail.tmpfiles
-
-SUFFIXES=.service.in .service .tmpfiles.in .tmpfiles
-
-.service.in.service:
-	$(AM_V_GEN)sed						\
-	    -e 's|[@]sbindir[@]|$(sbindir)|g'			\
-	    -e 's|[@]piddir[@]|$(PID_DIR)|g'		\
-	    < $< > $@-t &&					\
-	    mv $@-t $@
-
-.tmpfiles.in.tmpfiles:
-	$(AM_V_GEN)sed						\
-	    -e 's|[@]piddir[@]|$(PID_DIR)|g'		\
-	    < $< > $@-t &&					\
-	    mv $@-t $@
-
-else
-install-systemd:
-uninstall-systemd:
-endif
-
-install-data-local: install-systemd
-uninstall-local: uninstall-systemd
diff -wbBurN dbmail.git/Makefile.am dbmail-3.1.5/Makefile.am
--- dbmail.git/Makefile.am	2013-09-04 19:00:55.000000000 +0400
+++ dbmail-3.1.5/Makefile.am	2013-09-09 15:14:22.549612297 +0400
@@ -6,10 +6,6 @@
 SUBDIRS += man
 endif
 
-if SYSTEMD
-SUBDIRS += systemd
-endif
-
 testall:
 	CK_FORK=no make check
 	imaptest user=testuser1 pass=test port=10143 host=localhost test=test-scripts/imap
