From 97129644e1fdd56a9254083e22430308ff0c9db6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 1 Aug 2020 13:54:20 +0200
Subject: [PATCH] Install the application icon

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 863aa2f..946ba53 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,8 +102,8 @@ install(TARGETS dde-calendar DESTINATION bin)
 
 file(GLOB APP_QM_FILES "translations/*.qm")
 
-#install(FILES /usr/share/icons/bloom/apps/96/dde-calendar.svg
-#        DESTINATION /share/icons/hicolor/scalable/apps/)
+install(FILES assets/resources/icon/dde-calendar.svg
+        DESTINATION share/icons/hicolor/scalable/apps)
 install(FILES ${APP_QM_FILES} DESTINATION share/dde-calendar/translations)
 install(FILES ${APP_DESKTOP} DESTINATION share/applications)
 install(FILES ${APP_SERVICE} DESTINATION share/dbus-1/services)
