diff --git a/modules/simplelistdelegate.cpp b/modules/simplelistdelegate.cpp
index 40e30b3..ae1b028 100644
--- a/modules/simplelistdelegate.cpp
+++ b/modules/simplelistdelegate.cpp
@@ -22,6 +22,7 @@
 #include <QEvent>
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <DGuiApplicationHelper>
 #include "dthememanager.h"
 #include "simplelistmodel.h"
diff --git a/widgets/memorybutton.cpp b/widgets/memorybutton.cpp
index b316632..19ee0de 100644
--- a/widgets/memorybutton.cpp
+++ b/widgets/memorybutton.cpp
@@ -2,6 +2,7 @@
 
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QTimer>
 #include "dthememanager.h"
 
diff --git a/widgets/textbutton.cpp b/widgets/textbutton.cpp
index 80a039d..0c8e37a 100644
--- a/widgets/textbutton.cpp
+++ b/widgets/textbutton.cpp
@@ -20,6 +20,7 @@
 #include "textbutton.h"
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QTimer>
 #include "dthememanager.h"
 
