From bccfca9fb8ffd2b73a0286308c68c3c54f6cda50 Mon Sep 17 00:00:00 2001
From: Oliver Haessler <oliver@redhat.com>
Date: Tue, 2 Jun 2015 17:02:02 +0200
Subject: [PATCH] set default value for /apps/guake/general/debug_mode

---
 data/guake.schemas | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/guake.schemas b/data/guake.schemas
index fedff64..2254f8f 100644
--- a/data/guake.schemas
+++ b/data/guake.schemas
@@ -5,7 +5,7 @@
             <applyto>/apps/guake/general/debug_mode</applyto>
             <owner>guake</owner>
             <type>bool</type>
-            <default></default>
+            <default>false</default>
             <locale name="C">
                 <short>Enable debug mode</short>
                 <long>When debug mode is enabled, logs are printed in the
