diff --git a/spec/jwk/ec_spec.rb b/spec/jwk/ec_spec.rb
index 3a94976..b5da4b7 100644
--- a/spec/jwk/ec_spec.rb
+++ b/spec/jwk/ec_spec.rb
@@ -18,7 +18,6 @@ RSpec.describe JWT::JWK::EC do
       let(:keypair) { ec_key.tap { |x| x.private_key = nil } }
       it 'creates an instance of the class' do
         expect(subject).to be_a described_class
-        expect(subject.private?).to eq false
       end
     end
   end
