diff -Naur sugar-0.117.orig/src/jarabe/main.py sugar-0.117/src/jarabe/main.py
--- sugar-0.117.orig/src/jarabe/main.py	2020-03-17 01:02:30.000000000 +0100
+++ sugar-0.117/src/jarabe/main.py	2020-05-27 21:38:18.454003200 +0200
@@ -188,7 +188,7 @@
     global _metacity_process, _metacity_sid
 
     _metacity_process = subprocess.Popen(
-        ['metacity', '--no-force-fullscreen', '--no-composite'],
+        ['metacity', '--no-force-fullscreen'],
         stdout=subprocess.PIPE)
     _metacity_sid = GLib.io_add_watch(_metacity_process.stdout, GLib.IO_HUP,
                                       __window_manager_failed_cb)
@@ -196,12 +196,6 @@
 
 
 def _start_window_manager():
-    global _cursor_theme_settings, _cursor_theme
-
-    _cursor_theme_settings = Gio.Settings.new('org.gnome.desktop.interface')
-    _cursor_theme = _cursor_theme_settings.get_string('cursor-theme')
-    _cursor_theme_settings.set_string('cursor-theme', 'sugar')
-
     _restart_window_manager()
 
     screen = Wnck.Screen.get_default()
@@ -211,7 +205,6 @@
 
 
 def _stop_window_manager():
-    _cursor_theme_settings.set_string('cursor-theme', _cursor_theme)
     _metacity_process.terminate()
 
 
