diff --git a/tests/test.sh b/tests/test.sh
index aeecb34..eccaab5 100755
--- a/tests/test.sh
+++ b/tests/test.sh
@@ -31,7 +31,7 @@ test_bin() {
   local expect=$1; shift
   local RESULT=-1
   cd $name;
-  bundle exec metadata-json-lint $* metadata.json >last_output 2>&1
+   ../../bin/metadata-json-lint $* metadata.json >last_output 2>&1
   RESULT=$?
   if [ $RESULT -ne $expect ]; then
     fail "Failing Test '${name}' (unexpected exit code '${RESULT}' instead of '${expect}') (bin)"
@@ -66,7 +66,7 @@ test_rake() {
   local RESULT=-1;
 
   cd $name;
-  bundle exec rake $rake_task >last_rake_output 2>&1
+  rake $rake_task >last_rake_output 2>&1
   RESULT=$?
   if [ $RESULT -ne $expect ]; then
     fail "Failing Test '${name}' (rake: ${rake_task})"
@@ -152,7 +152,7 @@ test "duplicate-requirement" $FAILURE
 
 # Test running without specifying file to parse
 cd perfect
-bundle exec metadata-json-lint
+../../bin/metadata-json-lint
 if [ $? -ne 0 ]; then
     fail "Failing Test 'running without specifying metadata.json' (bin)"
 else
