diff -upr jupyterlab_widgets-1.0.2.orig/setup.py jupyterlab_widgets-1.0.2/setup.py
--- jupyterlab_widgets-1.0.2.orig/setup.py	2021-09-14 02:30:14.000000000 +0300
+++ jupyterlab_widgets-1.0.2/setup.py	2021-12-03 19:44:10.702567429 +0200
@@ -5,7 +5,7 @@ import os
 
 from jupyter_packaging import (
     create_cmdclass, install_npm, ensure_targets,
-    combine_commands, ensure_python, get_version,
+    combine_commands, get_version,
 )
 import setuptools
 
@@ -14,9 +14,6 @@ HERE = os.path.abspath(os.path.dirname(_
 # The name of the project
 name = "jupyterlab_widgets"
 
-# Ensure a valid python version
-ensure_python(">=3.6")
-
 # Get our version
 version = get_version(os.path.join(name, "_version.py"))
 
