diff -Naur ario-1.5.1.orig/configure.ac ario-1.5.1/configure.ac
--- ario-1.5.1.orig/configure.ac	2011-06-13 18:11:30.000000000 +0200
+++ ario-1.5.1/configure.ac	2017-01-26 14:20:51.574818457 +0100
@@ -36,7 +36,6 @@
                          gthread-2.0
                          gio-2.0
                          libxml-2.0
-                         unique-1.0
                          libcurl
                          gnutls])
 
diff -Naur ario-1.5.1.orig/src/ario-main.c ario-1.5.1/src/ario-main.c
--- ario-1.5.1.orig/src/ario-main.c	2011-06-13 11:58:37.000000000 +0200
+++ ario-1.5.1/src/ario-main.c	2017-01-26 15:15:24.790597138 +0100
@@ -36,20 +36,14 @@
 
 #ifdef WIN32
 #include <windows.h>
-#else
-#include <unique/unique.h>
 #endif
 
 #ifndef WIN32
-static UniqueResponse
-ario_main_on_message_received (G_GNUC_UNUSED UniqueApp *app,
-                               G_GNUC_UNUSED UniqueCommand command,
-                               G_GNUC_UNUSED UniqueMessageData *message,
-                               G_GNUC_UNUSED guint time_,
-                               ArioShell *shell)
+static void
+ario_main_on_activated (G_GNUC_UNUSED GApplication *app,
+                        ArioShell *shell)
 {
         ario_shell_present (shell);
-        return UNIQUE_RESPONSE_OK;
 }
 #endif
 
@@ -89,11 +83,12 @@
                 return 0;
         }
 #else
-        UniqueApp *app;
-        app = unique_app_new ("org.Ario", NULL);
-        if (unique_app_is_running (app)) {
+        GApplication *app;
+        app = g_application_new ("org.Ario", G_APPLICATION_FLAGS_NONE);
+        g_application_register (app, NULL, NULL);
+        if (g_application_get_is_remote (app)) {
                 if (ario_conf_get_boolean (PREF_ONE_INSTANCE, PREF_ONE_INSTANCE_DEFAULT)) {
-                        unique_app_send_message (app, UNIQUE_ACTIVATE, NULL);
+                        g_application_activate (app);
                         g_object_unref (app);
                         return 0;
                 }
@@ -134,8 +129,7 @@
         ario_shell_construct (shell, minimized);
 
 #ifndef WIN32
-        unique_app_watch_window (app, GTK_WINDOW (shell));
-        g_signal_connect (app, "message-received", G_CALLBACK (ario_main_on_message_received), shell);
+        g_signal_connect (app, "activate", G_CALLBACK (ario_main_on_activated), shell);
 #endif
         /* Initialisation of plugins engine */
         ario_plugins_engine_init (shell);
