diff --git a/.gitignore b/.gitignore
index e4955720e5a00e9868aad96ce3068112394a60c7..ff56adac0f6d353f87c79eccbe6a2f0c82589c37 100644
--- a/.gitignore
+++ b/.gitignore
@@ -39,3 +39,6 @@ CMakeLists.txt.user*
 
 *.qm
 com.deepin.pkexec.deepin-clone.policy
+
+build/
+
diff --git a/app/src/main.cpp b/app/src/main.cpp
index 5e6afa7c9906a066a28da0eaa1e9d8d0dd618fd6..dbb0fc35e6f3b1d6d78fae1f43199cc352f421c2 100644
--- a/app/src/main.cpp
+++ b/app/src/main.cpp
@@ -233,7 +233,7 @@ int main(int argc, char *argv[])
         MainWindow *window = new MainWindow;
 
         window->setFixedSize(860, 660);
-        window->setStyleSheet(DThemeManager::instance()->getQssForWidget("main", window));
+        window->setStyleSheet(DThemeManager::instance()->getQssForWidget("main"));
         window->setWindowIcon(QIcon::fromTheme("deepin-clone"));
         window->setWindowFlags(Qt::WindowCloseButtonHint | Qt::WindowMinimizeButtonHint | Qt::WindowSystemMenuHint);
         window->titlebar()->setIcon(window->windowIcon());
diff --git a/app/src/widgets/selectfilepage.cpp b/app/src/widgets/selectfilepage.cpp
index 1efe0babc3c23d5e99ed6b40ab555daff8e8b131..5f8c4649db90ab184cdc25617c057168eafa2167 100644
--- a/app/src/widgets/selectfilepage.cpp
+++ b/app/src/widgets/selectfilepage.cpp
@@ -28,7 +28,7 @@
 #include "iconlabel.h"
 #include "helper.h"
 
-#include <dlinkbutton.h>
+#include <DCommandLinkButton>
 
 #include <QVBoxLayout>
 #include <QFileDialog>
@@ -77,7 +77,7 @@ private:
     QString m_defaultFileName;
     QLabel *m_dragDropLabel = 0;
     IconLabel *m_label;
-    DLinkButton *m_button;
+    DCommandLinkButton *m_button;
 
     friend class SelectFilePage;
 };
@@ -98,7 +98,7 @@ SelectFileWidget::SelectFileWidget(Mode mode, QWidget *parent)
         setAcceptDrops(true);
     }
 
-    m_button = new DLinkButton(mode == GetSaveName ? tr("Select storage location") : tr("Select image file"), this);
+    m_button = new DCommandLinkButton(mode == GetSaveName ? tr("Select storage location") : tr("Select image file"), this);
 
     layout->addStretch();
     layout->addWidget(m_label, 0, Qt::AlignHCenter);
@@ -121,7 +121,7 @@ SelectFileWidget::SelectFileWidget(Mode mode, QWidget *parent)
     layout->addWidget(m_button, 0, Qt::AlignHCenter);
     layout->addStretch();
 
-    connect(m_button, &DLinkButton::clicked, this, &SelectFileWidget::openFileDialog);
+    connect(m_button, &DCommandLinkButton::clicked, this, &SelectFileWidget::openFileDialog);
 
     setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
 }
diff --git a/app/src/widgets/utilitylistitem.cpp b/app/src/widgets/utilitylistitem.cpp
index eb898bd49c6ee0c205d2c9f1d6da7216775477e2..aadb466ca53572348c54927f44fb79fecff13c49 100644
--- a/app/src/widgets/utilitylistitem.cpp
+++ b/app/src/widgets/utilitylistitem.cpp
@@ -21,7 +21,7 @@
 
 #include "utilitylistitem.h"
 
-#include <dseparatorhorizontal.h>
+#include <DHorizontalLine>
 
 #include <QHBoxLayout>
 #include <QLabel>
@@ -107,7 +107,7 @@ UtilityListItem::UtilityListItem(QWidget *parent)
     m_layout->addWidget(m_label, 0, Qt::AlignVCenter);
 
     m_spaceSeparator = new QWidget(this);
-    m_bottomSeparator = new DSeparatorHorizontal(this);
+    m_bottomSeparator = new DHorizontalLine(this);
 
     m_spaceSeparator->setFixedHeight(2);
 
