diff --git a/examples/dwidget-examples/collections/graphicseffecttab.cpp b/examples/dwidget-examples/collections/graphicseffecttab.cpp
index 2497f47..22483bf 100644
--- a/examples/dwidget-examples/collections/graphicseffecttab.cpp
+++ b/examples/dwidget-examples/collections/graphicseffecttab.cpp
@@ -21,6 +21,7 @@
 
 #include <QLabel>
 #include <QDebug>
+#include <QPainterPath>
 DWIDGET_USE_NAMESPACE
 
 GraphicsEffectTab::GraphicsEffectTab(QWidget *parent) : QWidget(parent)
diff --git a/examples/dwidget-examples/collections/multilistitem.cpp b/examples/dwidget-examples/collections/multilistitem.cpp
index 6c55abf..0cdda21 100644
--- a/examples/dwidget-examples/collections/multilistitem.cpp
+++ b/examples/dwidget-examples/collections/multilistitem.cpp
@@ -17,6 +17,7 @@
 
 #include "multilistitem.h"
 #include <QColor>
+#include <QPainterPath>
 
 DWIDGET_USE_NAMESPACE
 
diff --git a/examples/dwidget-examples/collections/singlelistitem.cpp b/examples/dwidget-examples/collections/singlelistitem.cpp
index 9e1cb15..27f77df 100644
--- a/examples/dwidget-examples/collections/singlelistitem.cpp
+++ b/examples/dwidget-examples/collections/singlelistitem.cpp
@@ -17,6 +17,7 @@
 
 #include "singlelistitem.h"
 #include <QColor>
+#include <QPainterPath>
 
 DWIDGET_USE_NAMESPACE
 
diff --git a/src/widgets/dclipeffectwidget.cpp b/src/widgets/dclipeffectwidget.cpp
index b9e9190..79011e2 100644
--- a/src/widgets/dclipeffectwidget.cpp
+++ b/src/widgets/dclipeffectwidget.cpp
@@ -22,6 +22,7 @@
 #include <QBackingStore>
 #include <QPainter>
 #include <QPaintEvent>
+#include <QPainterPath>
 #include <QDebug>
 
 #include <qpa/qplatformbackingstore.h>
diff --git a/src/widgets/dclipeffectwidget.h b/src/widgets/dclipeffectwidget.h
index f845ba5..71d1432 100644
--- a/src/widgets/dclipeffectwidget.h
+++ b/src/widgets/dclipeffectwidget.h
@@ -22,6 +22,7 @@
 #include "dobject.h"
 
 #include <QWidget>
+#include <QPainterPath>
 
 DWIDGET_BEGIN_NAMESPACE
 
diff --git a/src/widgets/dcrumbedit.cpp b/src/widgets/dcrumbedit.cpp
index 31dfc8e..44a0224 100644
--- a/src/widgets/dcrumbedit.cpp
+++ b/src/widgets/dcrumbedit.cpp
@@ -24,6 +24,7 @@
 
 #include <QAbstractTextDocumentLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QTextBlock>
 #include <QStyleOptionFrame>
 #include <QMouseEvent>
diff --git a/src/widgets/ddialog.h b/src/widgets/ddialog.h
index 9a706b9..04f6bcd 100644
--- a/src/widgets/ddialog.h
+++ b/src/widgets/ddialog.h
@@ -94,8 +94,8 @@ public Q_SLOTS:
     bool setDefaultButton(int index);
     bool setDefaultButton(const QString &str);
     void setDefaultButton(QAbstractButton *button);
-    void addContent(QWidget *widget, Qt::Alignment alignment = 0);
-    void insertContent(int index, QWidget *widget, Qt::Alignment alignment = 0);
+    void addContent(QWidget *widget, Qt::Alignment alignment = {});
+    void insertContent(int index, QWidget *widget, Qt::Alignment alignment = {});
     void removeContent(QWidget *widget, bool isDelete = true);
     void clearContents(bool isDelete = true);
     void setSpacing(int spacing);
diff --git a/src/widgets/dgraphicsclipeffect.cpp b/src/widgets/dgraphicsclipeffect.cpp
index 1624784..81b5487 100644
--- a/src/widgets/dgraphicsclipeffect.cpp
+++ b/src/widgets/dgraphicsclipeffect.cpp
@@ -19,6 +19,7 @@
 #include <DObjectPrivate>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 
 DWIDGET_BEGIN_NAMESPACE
diff --git a/src/widgets/dgraphicsclipeffect.h b/src/widgets/dgraphicsclipeffect.h
index 8ee8944..cd65749 100644
--- a/src/widgets/dgraphicsclipeffect.h
+++ b/src/widgets/dgraphicsclipeffect.h
@@ -22,6 +22,7 @@
 #include "dobject.h"
 
 #include <QGraphicsEffect>
+#include <QPainterPath>
 
 DWIDGET_BEGIN_NAMESPACE
 
diff --git a/src/widgets/dmainwindow.h b/src/widgets/dmainwindow.h
index daaa620..4fa20f3 100644
--- a/src/widgets/dmainwindow.h
+++ b/src/widgets/dmainwindow.h
@@ -23,6 +23,7 @@
 #include <DObject>
 
 #include <QMainWindow>
+#include <QPainterPath>
 
 DWIDGET_BEGIN_NAMESPACE
 
diff --git a/src/widgets/dsegmentedcontrol.cpp b/src/widgets/dsegmentedcontrol.cpp
index b39fe0b..6bd38ef 100644
--- a/src/widgets/dsegmentedcontrol.cpp
+++ b/src/widgets/dsegmentedcontrol.cpp
@@ -18,6 +18,7 @@
 #include <QDebug>
 #include <QPainter>
 #include <QPaintEvent>
+#include <QPainterPath>
 #include <QApplication>
 
 #include "dsegmentedcontrol.h"
diff --git a/src/widgets/dsimplelistview.cpp b/src/widgets/dsimplelistview.cpp
index 01d5ff7..872b84d 100644
--- a/src/widgets/dsimplelistview.cpp
+++ b/src/widgets/dsimplelistview.cpp
@@ -31,6 +31,7 @@
 #include <QWheelEvent>
 #include <QtMath>
 #include <QPointer>
+#include <QPainterPath>
 
 #include "dhidpihelper.h"
 
diff --git a/src/widgets/dspinner.cpp b/src/widgets/dspinner.cpp
index 50acddc..b6d6e7e 100644
--- a/src/widgets/dspinner.cpp
+++ b/src/widgets/dspinner.cpp
@@ -2,6 +2,7 @@
 
 #include <QtMath>
 #include <QPainter>
+#include <QPainterPath>
 #include <QTimer>
 #include <QEvent>
 
diff --git a/src/widgets/dstyle.cpp b/src/widgets/dstyle.cpp
index 039467d..c76cce2 100644
--- a/src/widgets/dstyle.cpp
+++ b/src/widgets/dstyle.cpp
@@ -29,6 +29,7 @@
 #include <QPixmapCache>
 #include <QGuiApplication>
 #include <QAbstractItemView>
+#include <QPainterPath>
 
 #include <qmath.h>
 #include <private/qfixed_p.h>
diff --git a/src/widgets/dvideowidget.cpp b/src/widgets/dvideowidget.cpp
index f0a0d71..53e0973 100644
--- a/src/widgets/dvideowidget.cpp
+++ b/src/widgets/dvideowidget.cpp
@@ -24,6 +24,7 @@
 #include <QMediaPlayer>
 #include <QMediaPlaylist>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPointer>
 
 DWIDGET_BEGIN_NAMESPACE
diff --git a/src/widgets/dwaterprogress.cpp b/src/widgets/dwaterprogress.cpp
index 79e0228..8c537f1 100644
--- a/src/widgets/dwaterprogress.cpp
+++ b/src/widgets/dwaterprogress.cpp
@@ -20,6 +20,7 @@
 #include <QtMath>
 #include <QTimer>
 #include <QPainter>
+#include <QPainterPath>
 #include <QGraphicsDropShadowEffect>
 #include <QEvent>
 #include <QIcon>
diff --git a/src/widgets/private/dblureffectwidget_p.h b/src/widgets/private/dblureffectwidget_p.h
index ecb8377..d3822d8 100644
--- a/src/widgets/private/dblureffectwidget_p.h
+++ b/src/widgets/private/dblureffectwidget_p.h
@@ -18,6 +18,7 @@
 #ifndef DBLUREFFECTWIDGET_P_H
 #define DBLUREFFECTWIDGET_P_H
 
+#include <QPainterPath>
 #include <DObjectPrivate>
 #include "dblureffectwidget.h"
 
diff --git a/src/widgets/private/settings/shortcutedit.cpp b/src/widgets/private/settings/shortcutedit.cpp
index d0d6d45..c502c4c 100644
--- a/src/widgets/private/settings/shortcutedit.cpp
+++ b/src/widgets/private/settings/shortcutedit.cpp
@@ -19,6 +19,7 @@
 
 #include <QDebug>
 #include <QPainter>
+#include <QPainterPath>
 #include <QKeySequence>
 #include <QKeyEvent>
 
