From eabc679bd4f564a6b5294d0558b579898d66e2fa Mon Sep 17 00:00:00 2001
From: Kjetil Matheussen <k.s.matheussen@notam02.no>
Date: Sat, 15 Jul 2017 13:21:36 +0200
Subject: [PATCH] Example clients: Fix compilation of tw.c

---
 example-clients/tw.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/example-clients/tw.c b/example-clients/tw.c
index 623ff240..edc7ef31 100644
--- a/example-clients/tw.c
+++ b/example-clients/tw.c
@@ -9,6 +9,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
+#include <signal.h>
 
 #include <jack/jack.h>
 
-- 
2.15.0

