--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -171,9 +171,7 @@ module.exports = grunt => {
   });
 
   grunt.registerTask('getExpireTime', () => {
-    grunt.task.requires('gitinfo');
-    const gitinfo = grunt.config.get('gitinfo');
-    const committed = gitinfo.local.branch.current.lastCommitTime;
+    const committed = parseInt(process.env.SOURCE_DATE_EPOCH, 10) * 1000;
     const buildCreation = Date.parse(committed);
     const buildExpiration = buildCreation + 1000 * 60 * 60 * 24 * 90;
     grunt.file.write(
