From e193e67718b77a1603a91079d7ab4d2307570a8e Mon Sep 17 00:00:00 2001
From: Steven Hiscocks <steven@hiscocks.me.uk>
Date: Sun, 16 Mar 2014 18:25:16 +0000
Subject: [PATCH] BF: fail2ban-regex mix of tabs and spaces

---
 bin/fail2ban-regex | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/bin/fail2ban-regex b/bin/fail2ban-regex
index 270b70d..4317d06 100755
--- a/bin/fail2ban-regex
+++ b/bin/fail2ban-regex
@@ -395,7 +395,7 @@ class Fail2banRegex(object):
 				if lines < self._maxlines or getattr(self, '_print_all_' + ltype):
 					ans = [[]]
 					for arg in [l, regexlist]:
-					    ans = [ x + [y] for x in ans for y in arg ]
+						ans = [ x + [y] for x in ans for y in arg ]
 					b = map(lambda a: a[0] +  ' | ' + a[1].getFailRegex() + ' |  ' + debuggexURL(a[0], a[1].getFailRegex()), ans)
 					pprint_list([x.rstrip() for x in b], header)
 				else:
@@ -466,11 +466,11 @@ if __name__ == "__main__":
 	if opts.print_no_missed and opts.print_all_missed:
 		sys.stderr.write("ERROR: --print-no-missed and --print-all-missed are mutually exclusive.\n\n")
 		parser.print_help()
-	        sys.exit(-1)
+		sys.exit(-1)
 	if opts.print_no_ignored and opts.print_all_ignored:
 		sys.stderr.write("ERROR: --print-no-ignored and --print-all-ignored are mutually exclusive.\n\n")
 		parser.print_help()
-	        sys.exit(-1)
+		sys.exit(-1)
 
 	print
 	print "Running tests"
-- 
1.8.5.5

