diff --git a/src/player.c b/src/player.c
index 875f473..5379d4d 100644
--- a/src/player.c
+++ b/src/player.c
@@ -235,7 +235,7 @@ static bool openStream (player_t * const player) {
 		softfail ("avcodec_parameters_to_context");
 	}
 
-	AVCodec * const decoder = avcodec_find_decoder (cp->codec_id);
+	const AVCodec *decoder = avcodec_find_decoder (cp->codec_id);
 	if (decoder == NULL) {
 		softfail ("find_decoder");
 	}
diff --git a/src/player.h b/src/player.h
index 3179785..0cff9dc 100644
--- a/src/player.h
+++ b/src/player.h
@@ -32,6 +32,7 @@ THE SOFTWARE.
 #include <signal.h>
 
 #include <ao/ao.h>
+#include <libavcodec/avcodec.h>
 #include <libavformat/avformat.h>
 #include <libavfilter/avfilter.h>
 #include <piano.h>
