diff -upr pork-0.99.8.1.orig/src/pork_command.c pork-0.99.8.1/src/pork_command.c
--- pork-0.99.8.1.orig/src/pork_command.c	2005-04-28 20:35:19.000000000 +0300
+++ pork-0.99.8.1/src/pork_command.c	2022-05-30 18:37:37.732536640 +0300
@@ -2124,7 +2124,7 @@ USER_COMMAND(cmd_ctcp) {
 
 USER_COMMAND(cmd_echo) {
 	if (args != NULL)
-		screen_win_msg(cur_window(), 0, 0, 1, MSG_TYPE_CMD_OUTPUT, args);
+		screen_win_msg(cur_window(), 0, 0, 1, MSG_TYPE_CMD_OUTPUT, "%s", args);
 }
 
 USER_COMMAND(cmd_disconnect) {
diff -upr pork-0.99.8.1.orig/src/pork_perl_xs.c pork-0.99.8.1/src/pork_perl_xs.c
--- pork-0.99.8.1.orig/src/pork_perl_xs.c	2005-04-29 03:43:53.000000000 +0300
+++ pork-0.99.8.1/src/pork_perl_xs.c	2022-05-30 19:39:23.523379048 +0300
@@ -289,7 +289,7 @@ XS(PORK_echo) {
 		XSRETURN_IV(-1);
 
 	msg = SvPV(ST(0), notused);
-	screen_win_msg(cur_window(), 0, 0, 1, MSG_TYPE_CMD_OUTPUT, msg);
+	screen_win_msg(cur_window(), 0, 0, 1, MSG_TYPE_CMD_OUTPUT, "%s", msg);
 	XSRETURN_IV(0);
 }
 
