diff --git a/tests/conftest.py b/tests/conftest.py
index 2af2ff7..14c2974 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -71,18 +71,20 @@ async def assert_num_uploads_found(
 @pytest.fixture
 def aa_fail_proxy_config(monkeypatch):
     # NOTE: name of this fixture must be alphabetically first to run first
-    monkeypatch.setenv('HTTP_PROXY', 'http://localhost:54321')
-    monkeypatch.setenv('HTTPS_PROXY', 'http://localhost:54321')
+    monkeypatch.setenv('http_proxy', 'http://localhost:54321')
+    monkeypatch.setenv('https_proxy', 'http://localhost:54321')
+
+    monkeypatch.delenv('no_proxy', raising=False)
 
 
 @pytest.fixture
 def aa_succeed_proxy_config(monkeypatch):
     # NOTE: name of this fixture must be alphabetically first to run first
-    monkeypatch.setenv('HTTP_PROXY', 'http://localhost:54321')
-    monkeypatch.setenv('HTTPS_PROXY', 'http://localhost:54321')
+    monkeypatch.setenv('http_proxy', 'http://localhost:54321')
+    monkeypatch.setenv('https_proxy', 'http://localhost:54321')
 
     # this will cause us to skip proxying
-    monkeypatch.setenv('NO_PROXY', 'amazonaws.com')
+    monkeypatch.setenv('no_proxy', 'amazonaws.com')
 
 
 @pytest.fixture
