From 236a03dcddbd21b276818c4016991ca137e7a330 Mon Sep 17 00:00:00 2001
From: hillwoodroc <hillwoodroc@gmail.com>
Date: Mon, 21 Feb 2022 16:49:55 +0800
Subject: [PATCH] Fix build on x86_64: add fPIC option

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 86970347..2c4dca9c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,8 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTORCC ON)
 set(CMAKE_AUTOUIC ON)
+set(CMAKE_C_FLAGS "-fPIC")
+set(CMAKE_CXX_FLAGS "-fPIC")
 
 option(USE_DXCB "integration with dxcb platform plugin" OFF)
 option(DMR_DEBUG "turn on debug output" off)
