--- cinnamon-original/files/usr/lib/cinnamon-settings/cinnamon-settings.py	2012-11-14 15:45:14.000000000 +0100
+++ cinnamon-new/files/usr/lib/cinnamon-settings/cinnamon-settings.py	2013-01-21 12:49:11.332556000 +0100
 try:
     import os
@@ -915,7 +915,8 @@
         valid.sort(lambda a,b: cmp(a.lower(), b.lower()))
         res = []
         for i in valid:
-            res.append((i, i))
+            if i != 'default':
+                res.append((i, i))
         return res
         
     def _load_keybinding_themes(self):
@@ -933,7 +934,8 @@
         valid.sort(lambda a,b: cmp(a.lower(), b.lower()))
         res = []
         for i in valid:
-            res.append((i, i))
+            if i != 'default':
+                res.append((i, i))
         return res
         
     def _load_window_themes(self):
@@ -1638,7 +1640,7 @@
     def onCategoryChanged(self, tree):
         self.kb_store.clear()
         if tree.get_selection() is not None:
-            categories, iter = tree.get_selection().get_selected()
+            categories, iter = tree.get_selection().get_selected() 
             if iter:
                 category = categories[iter][1]
                 if category.int_name is not "custom":
@@ -2005,7 +2007,7 @@
 
     def on_my_setting_changed(self, settings, key):
         self.content_widget.set_text(self.settings.get_string(self.key))
-
+    
     def on_my_value_changed(self, event, widget):        
         self.settings.set_string(self.key, self.content_widget.get_text())
 
@@ -2260,8 +2262,14 @@
     def on_my_value_changed(self, widget):
         tree_iter = widget.get_active_iter()
         if tree_iter != None:            
-            value = self.model[tree_iter][0]            
+            value = self.model[tree_iter][0]          
             self.settings.set_string(self.key, value)
+            if self.key == 'cursor-theme':
+                if os.path.exists("/usr/share/icons/%s" % value):
+                    commands.getoutput('gksudo "ln -snf /usr/share/icons/%s /usr/share/icons/default"' % value.strip())
+                elif os.path.exists("%s/.icons/%s" % (os.path.join(os.path.expanduser("~")), value)):
+                    commands.getoutput('gksudo "ln -snf %s/.icons/%s /usr/share/icons/default"' % (os.path.join(os.path.expanduser("~")), value))
+                   
 
     def on_dependency_setting_changed(self, settings, dep_key):
         if not self.dependency_invert:
