diff --git a/lib/models/note.js b/lib/models/note.js
index 9fe02359..49b7ce84 100644
--- a/lib/models/note.js
+++ b/lib/models/note.js
@@ -96,7 +96,7 @@ module.exports = function (sequelize, DataTypes) {
             if (!note.alias) {
               filePath = config.defaultNotePath
             } else {
-              filePath = path.join(config.docsPath, note.alias + '.md')
+              filePath = path.join(config.docsPath, path.basename(note.alias) + '.md')
             }
             if (Note.checkFileExist(filePath)) {
               var fsCreatedTime = moment(fs.statSync(filePath).ctime)
@@ -195,7 +195,7 @@ module.exports = function (sequelize, DataTypes) {
           }
         }).then(function (note) {
           if (note) {
-            let filePath = path.join(config.docsPath, noteId + '.md')
+            let filePath = path.join(config.docsPath, path.basename(noteId) + '.md')
             if (Note.checkFileExist(filePath)) {
               // if doc in filesystem have newer modified time than last change time
               // then will update the doc in db
@@ -237,7 +237,7 @@ module.exports = function (sequelize, DataTypes) {
               return callback(null, note.id)
             }
           } else {
-            var filePath = path.join(config.docsPath, noteId + '.md')
+            var filePath = path.join(config.docsPath, path.basename(noteId) + '.md')
             if (Note.checkFileExist(filePath)) {
               Note.create({
                 alias: noteId,
