From 55f136eb5b160a5dcaf149988f28d2f02b707fa3 Mon Sep 17 00:00:00 2001
From: Sara Dickinson <sara@sinodun.com>
Date: Wed, 8 Nov 2017 11:44:41 +0000
Subject: [PATCH] Add DESTDIR to runstatedir creation path

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 1c8186f..94d05d0 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -16,4 +16,4 @@ stubby.yml.windows: ${srcdir}/stubby.yml.example
 install-data-hook: $(CONFIG_FILE)
 	test -f $(DESTDIR)${stubbyconfdir}/stubby.yml || \
 		${INSTALL_DATA} $(CONFIG_FILE) $(DESTDIR)${stubbyconfdir}/stubby.yml
-	$(INSTALL) -m 755 -d $(runstatedir)
\ No newline at end of file
+	$(INSTALL) -m 755 -d $(DESTDIR)$(runstatedir)
\ No newline at end of file
