commit 02fb78fb4f1e46b5176f45e2c221da6e52e9d014
Author: dgod <dgod.osa@gmail.com>
Date:   Wed Oct 31 20:31:48 2012 +0800

    fix sort tasks by clicking the User Name

diff --git a/src/interface.c b/src/interface.c
index 45cf256..1f98351 100644
--- a/src/interface.c
+++ b/src/interface.c
@@ -195,7 +195,7 @@ void create_list_store(void)
     column = gtk_tree_view_column_new_with_attributes(_("User"), cell_renderer, "text", COLUMN_UNAME, NULL);
     gtk_tree_view_column_set_resizable(column, TRUE);
     gtk_tree_view_column_set_sort_column_id(column, COLUMN_UNAME);
-    gtk_tree_sortable_set_sort_func(GTK_TREE_SORTABLE(list_store), COLUMN_UNAME, compare_int_list_item, (void *)COLUMN_UNAME, NULL);
+    gtk_tree_sortable_set_sort_func(GTK_TREE_SORTABLE(list_store), COLUMN_UNAME, compare_string_list_item, (void *)COLUMN_UNAME, NULL);
     gtk_tree_view_append_column(GTK_TREE_VIEW(treeview), column);
 
     cell_renderer = gtk_cell_renderer_text_new();
