diff -ruN Radicale-2.1.8-a/radicale/__init__.py Radicale-2.1.8-b/radicale/__init__.py
--- Radicale-2.1.8-a/radicale/__init__.py	2017-09-24 02:01:14.000000000 +0200
+++ Radicale-2.1.8-b/radicale/__init__.py	2018-02-04 14:08:39.257626655 +0100
@@ -120,7 +120,7 @@
             try:
                 self.server_bind()
                 self.server_activate()
-            except:
+            except Exception:
                 self.server_close()
                 raise
 
diff -ruN Radicale-2.1.8-a/radicale/storage.py Radicale-2.1.8-b/radicale/storage.py
--- Radicale-2.1.8-a/radicale/storage.py	2017-09-24 02:01:14.000000000 +0200
+++ Radicale-2.1.8-b/radicale/storage.py	2018-02-04 14:09:06.850699075 +0100
@@ -779,7 +779,7 @@
                                    (path, e)) from e
             tmp.close()
             os.replace(tmp.name, path)
-        except:
+        except Exception:
             tmp.close()
             os.remove(tmp.name)
             raise
