diff -Naur activity-log-manager-0.9.7.orig/src/unified-privacy-applications.vala activity-log-manager-0.9.7/src/unified-privacy-applications.vala
--- activity-log-manager-0.9.7.orig/src/unified-privacy-applications.vala	2018-01-15 21:41:36.620167000 +0100
+++ activity-log-manager-0.9.7/src/unified-privacy-applications.vala	2018-01-15 21:44:26.360957457 +0100
@@ -263,7 +263,7 @@
 				string can_app = can_app_value.get_string();
 				if(app == can_app)
 				{
-					this.store.remove(iter);
+					this.store.remove(ref iter);
 					break;
 				}
 				bool more_entires = model.iter_next(ref iter);
@@ -282,7 +282,7 @@
 			if(selection.get_selected(out model, out iter))
 			{
 				model.get(iter, 2, out app);
-				this.liststore.remove(iter);
+				this.liststore.remove(ref iter);
 			}
 
 			return app;
diff -Naur activity-log-manager-0.9.7.orig/src/unified-privacy.vala activity-log-manager-0.9.7/src/unified-privacy.vala
--- activity-log-manager-0.9.7.orig/src/unified-privacy.vala	2018-01-15 21:41:32.020056000 +0100
+++ activity-log-manager-0.9.7/src/unified-privacy.vala	2018-01-15 21:41:59.804064208 +0100
@@ -376,7 +376,7 @@
 					model.get(iter, 0, out name, 3, out type);
 					if(name != null)
 					{
-						exception_list_store.remove(iter);
+						exception_list_store.remove(ref iter);
 						if (type == ItemType.FILE)
 							this.path_blacklist.unblock(name);
 						else
@@ -437,7 +437,7 @@
 				sel.get_selected(out model, out iter);
 				string app;
 				model.get(iter, 2, out app);
-				this.app_dialog.liststore.remove (iter);
+				this.app_dialog.liststore.remove (ref iter);
 
 				// Add the App to View
 				app_change_recursive = true;
@@ -505,7 +505,7 @@
 				string cand_folder = can_app_value.get_string();
 				if(folder == cand_folder)
 				{
-					this.exception_list_store.remove(iter);
+					this.exception_list_store.remove(ref iter);
 					return true;
 				}
 				bool more_entires = model.iter_next(ref iter);
@@ -550,7 +550,7 @@
 				string can_app = can_app_value.get_string();
 				if(app == can_app)
 				{
-					this.exception_list_store.remove(iter);
+					this.exception_list_store.remove(ref iter);
 					break;
 				}
 				bool more_entires = model.iter_next(ref iter);
