From 47b5addb4609607bddebbaa707f7a08576cb6efb Mon Sep 17 00:00:00 2001
From: Jordi Mas <jmas@softcatala.org>
Date: Thu, 13 Jan 2011 21:19:15 +0000
Subject: Fixes CS0188 compiler error in Mono 2.9

---
diff --git a/src/Games/Logic/PuzzleExtraCircle.cs b/src/Games/Logic/PuzzleExtraCircle.cs
index 58797eb..8fce87c 100644
--- a/src/Games/Logic/PuzzleExtraCircle.cs
+++ b/src/Games/Logic/PuzzleExtraCircle.cs
@@ -43,7 +43,7 @@ namespace gbrainy.Games.Logic
 		{
 			public Color [] Colors {get; set; }
 
-			public CircleParameters (Color [] colors)
+			public CircleParameters (Color [] colors) : this ()
 			{
 				Colors = colors;
 			}
--
cgit v0.8.3.1
