From e22f6a1ad4f83942a7ea2240337b99a2ee9a3e89 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Mon, 2 Dec 2019 13:02:49 +0000
Subject: [PATCH] fix: crash resizing notmuch mailbox

Fixes: #2014
---
 mx.c                   | 2 +-
 notmuch/mutt_notmuch.c | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/mx.c b/mx.c
index 4a5c1c55b..1e13fd78a 100644
--- a/mx.c
+++ b/mx.c
@@ -1181,7 +1181,7 @@ void mx_alloc_memory(struct Mailbox *m)
     m->emails = mutt_mem_calloc(m->email_max, sizeof(struct Email *));
     m->v2r = mutt_mem_calloc(m->email_max, sizeof(int));
   }
-  for (int i = m->msg_count; i < m->email_max; i++)
+  for (int i = m->email_max - 25; i < m->email_max; i++)
   {
     m->emails[i] = NULL;
     m->v2r[i] = -1;
diff --git a/notmuch/mutt_notmuch.c b/notmuch/mutt_notmuch.c
index 1078d4432..2c98d07c4 100644
--- a/notmuch/mutt_notmuch.c
+++ b/notmuch/mutt_notmuch.c
@@ -1980,8 +1980,8 @@ static int nm_mbox_check_stats(struct Mailbox *m, int flags)
 
   /* all emails */
   m->msg_count = count_query(db, db_query, limit);
-  m->email_max = MAX(m->email_max, m->msg_count);
-  mx_alloc_memory(m);
+  while (m->email_max < m->msg_count)
+    mx_alloc_memory(m);
 
   // holder variable for extending query to unread/flagged
   char *qstr = NULL;
