--- guake-0.5.0.orig/src/guake	2014-10-13 23:11:31.803523945 +0200
+++ guake-0.5.0.orig/src/guake	2014-10-13 23:11:43.130502724 +0200
@@ -1094,7 +1094,6 @@
         """Hides the main window of the terminal and sets the visible
         flag to False.
         """
-        self.window.set_keep_below(True)
         self.window.hide()  # Don't use hide_all here!
 
     def get_final_window_monitor(self):
