From f40fe4232da38d74353920755413996b9577c6cb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 17 Apr 2019 21:55:42 +0200
Subject: [PATCH] Use correct syntax for array parameter

---
 aggregate-avatars.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aggregate-avatars.vala b/aggregate-avatars.vala
index d73ee32..2d0e33d 100644
--- a/aggregate-avatars.vala
+++ b/aggregate-avatars.vala
@@ -25,7 +25,7 @@
  
 using Folks;
 
-static int main(string args[])
+static int main(string[] args)
 {
 	MainLoop mainloop = new MainLoop();
 	var aggregator = IndividualAggregator.dup();
