diff --git a/tests/testformatter.rb b/tests/testformatter.rb
index 673c8dd..4ff70a1 100644
--- a/tests/testformatter.rb
+++ b/tests/testformatter.rb
@@ -12,7 +12,7 @@ class TestFormatter < TestCase
     sf = SimpleFormatter.new
     f = Logger.new('simple formatter')
     event = LogEvent.new(0, f, nil, "some data")
-    assert_match(sf.format(event), /simple formatter/)
+    assert_match(/simple formatter/, sf.format(event))
   end
   def test_basic_formatter
     b = BasicFormatter.new
@@ -20,12 +20,12 @@ class TestFormatter < TestCase
     event = LogEvent.new(0, f, caller, "fake formatter")
     event2 = LogEvent.new(0, f, nil, "fake formatter")
     # this checks for tracing
-    assert_match(b.format(event), /in/)
-    assert_not_match(b.format(event2), /in/)
+    assert_match(/in/, b.format(event))
+    assert_not_match(/in/, b.format(event2))
     e = ArgumentError.new("argerror")
     e.set_backtrace ['backtrace']
     event3 = LogEvent.new(0, f, nil, e)
-    assert_match(b.format(event3), /ArgumentError/)
-    assert_match(b.format(LogEvent.new(0,f,nil,[1,2,3])), /Array/)
+    assert_match(/ArgumentError/, b.format(event3))
+    assert_match(/Array/, b.format(LogEvent.new(0,f,nil,[1,2,3])))
   end
 end
