From 0ada751ca26dfcdd3b4cd9a1a95acdb6e6301881 Mon Sep 17 00:00:00 2001
From: Sara Dickinson <sara@sinodun.com>
Date: Wed, 1 Nov 2017 10:24:24 +0000
Subject: [PATCH] Use DESTDIR when installing stubby.yml file

---
 Makefile.am | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index aab3b3b..1f86472 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -14,5 +14,6 @@ stubby.yml.windows: ${srcdir}/stubby.yml.example
 	awk "{sub(/$$/,\"\r\")}1" $< > $@
 
 install-data-hook: $(CONFIG_FILE)
-	test -f ${stubbyconfdir}/stubby.yml || \
-		${INSTALL_DATA} $(CONFIG_FILE) ${stubbyconfdir}/stubby.yml
+	$(INSTALL) -m 755 -d $(DESTDIR)$(stubbyconfdir)
+	test -f $(DESTDIR)${stubbyconfdir}/stubby.yml || \
+		${INSTALL_DATA} $(CONFIG_FILE) $(DESTDIR)${stubbyconfdir}/stubby.yml
