Index: /trunk/src/VBox/Runtime/r0drv/linux/mpnotification-r0drv-linux.c
===================================================================
--- /trunk/src/VBox/Runtime/r0drv/linux/mpnotification-r0drv-linux.c (revision 38199)
+++ /trunk/src/VBox/Runtime/r0drv/linux/mpnotification-r0drv-linux.c (revision 38232)
@@ -78,5 +78,5 @@
  * @remarks This can be invoked in interrupt context.
  */
-static void rtMpNotificationLinuxOnCurrentCpu(RTCPUID idCpu, void *pvUser1, void *pvUser2)
+static DECLCALLBACK(void) rtMpNotificationLinuxOnCurrentCpu(RTCPUID idCpu, void *pvUser1, void *pvUser2)
 {
     unsigned long ulNativeEvent = *(unsigned long *)pvUser2;
