diff -u -r l/images/image-harddisk/libmirage-apple.xml.in l2/images/image-harddisk/libmirage-apple.xml.in
--- l/images/image-harddisk/libmirage-apple.xml.in	2021-04-17 11:19:54.000000000 +0000
+++ l2/images/image-harddisk/libmirage-apple.xml.in	2021-12-16 15:52:00.461808945 +0000
@@ -13,14 +13,5 @@
         <magic priority="50">
             <match value="koly" type="string" offset="0"/>
         </magic>
-
-        <magic priority="20">
-            <match value="ER" type="string" offset="0"/>
-            <match value="PM" type="string" offset="512"/>
-
-            <match value="BD" type="string" offset="1024"/>
-            <match value="H+" type="string" offset="1024"/>
-            <match value="HX" type="string" offset="1024"/>
-        </magic>
     </mime-type>
 </mime-info>
