--- pythia8301-old/Makefile	2019-11-18 16:03:32.498781658 +0200
+++ pythia8301-new/Makefile	2019-11-18 15:57:35.342991437 +0200
@@ -106,13 +106,13 @@
 # PYTHIA.
 $(LOCAL_TMP)/Pythia.o: $(LOCAL_SRC)/Pythia.cc Makefile.inc
 	$(CXX) $< -o $@ -c $(OBJ_COMMON) -DXMLDIR=\"$(PREFIX_SHARE)/xmldoc\"
 $(LOCAL_TMP)/%.o: $(LOCAL_SRC)/%.cc Makefile.inc
 	$(CXX) $< -o $@ -c $(OBJ_COMMON)
 $(LOCAL_LIB)/libpythia8.a: $(OBJECTS) $(OBJECTS_ME)
-	ar cr $@ $^
+	ar rcs $@ $^
 $(LOCAL_LIB)/libpythia8$(LIB_SUFFIX): $(OBJECTS) $(OBJECTS_ME)
 	$(CXX) $^ -o $@ $(CXX_COMMON) $(CXX_SHARED) $(CXX_SONAME)$(notdir $@)\
 	  $(LIB_COMMON)
 
 # LHAPDF (turn off all warnings for readability).
 $(LOCAL_TMP)/LHAPDF%Plugin.o: $(LOCAL_INCLUDE)/Pythia8Plugins/LHAPDF%.h
