diff -au6r pythia8301-old/plugins/python/Makefile pythia8301-new/plugins/python/Makefile
--- pythia8301-old/plugins/python/Makefile	2019-10-29 18:08:54.000000000 +0200
+++ pythia8301-new/plugins/python/Makefile	2019-11-18 17:08:21.959441069 +0200
@@ -24,13 +24,13 @@
 TOP_INCLUDE=../../include
 LOCAL_SRC=src
 LOCAL_TMP=tmp
 LOCAL_INCLUDE=include
 LOCAL_MKDIRS:=$(shell mkdir -p $(LOCAL_TMP) $(TOP_LIB))
 OBJ_COMMON:=-MD $(CXX_COMMON) -Iinclude $(PYTHON_INCLUDE) -w -fpermissive
-LIB_COMMON=-Wl,-rpath,$(TOP_LIB) -ldl $(GZIP_LIB)
+LIB_COMMON=-ldl $(GZIP_LIB)
 PYTHIA=$(TOP_LIB)/libpythia8$(LIB_SUFFIX)
 
 # Determine the headers.
 HEADERS=$(patsubst ../../include/Pythia8%.h,$(LOCAL_INCLUDE)/Pythia8%.h,\
 	$(sort $(wildcard ../../include/Pythia8*/*.h)))
 
