diff --git a/src/mainwindow.cpp.orig b/src/mainwindow.cpp
index f1e0a59..8350862 100644
--- a/src/mainwindow.cpp.orig
+++ b/src/mainwindow.cpp
@@ -1395,7 +1395,16 @@ bool MainWindow::saveAction(const QPixmap &pix)
     if (copyToClipboard) {
         Q_ASSERT(!screenShotPix.isNull());
         QClipboard* cb = qApp->clipboard();
-        cb->setPixmap(screenShotPix, QClipboard::Clipboard);
+        // On X11 clipboard content is owned by the application,
+        // Data copied to clipboard remains in the clipboard only
+        // as long as the parent application is running and is discarded when the application quits.
+        // Added "x-kde-force-image-copy" mime type to screenshots being copied to clipboard,
+        // which forces klipper clipboard manager to remember this clipboard entry
+        // even if user has set "ignore images" option to true in klipper.
+        QMimeData *data = new QMimeData();
+        data->setImageData(screenShotPix.toImage());
+        data->setData(QStringLiteral("x-kde-force-image-copy"), QByteArray());
+        cb->setMimeData(data, QClipboard::Clipboard);
     }
 
     return true;
@@ -1455,7 +1464,7 @@ void MainWindow::sendNotify(SaveAction saveAction, QString saveFilePath, const b
                                     summary, actions, hints, 0);
     }
 
-    QTimer::singleShot(2, [=]{
+    QTimer::singleShot(500, [=]{
         exitApp();
     });
 
