diff --git a/Opcodes/fluidOpcodes/fluidOpcodes.cpp b/Opcodes/fluidOpcodes/fluidOpcodes.cpp
index d6f4da0fd..735c1021c 100644
--- a/Opcodes/fluidOpcodes/fluidOpcodes.cpp
+++ b/Opcodes/fluidOpcodes/fluidOpcodes.cpp
@@ -208,16 +208,31 @@ public:
     } else if (listPresets) {
       fluid_sfont_t *fluidSoundfont =
           fluid_synth_get_sfont_by_id(fluidSynth, soundFontId);
+#if FLUIDSYNTH_VERSION_MAJOR < 2
       fluid_preset_t fluidPreset;
       fluidSoundfont->iteration_start(fluidSoundfont);
       OPARMS oparms;
       csound->GetOParms(csound, &oparms);
       if (oparms.msglevel & 0x7)
-        while (fluidSoundfont->iteration_next(fluidSoundfont, &fluidPreset)) {
+        while (fluidSoundfont->iteration_next(fluidSoundfont, &fluidPreset))
+        {
           log(csound, "SoundFont: %3d  Bank: %3d  Preset: %3d  %s\n",
               soundFontId, fluidPreset.get_banknum(&fluidPreset),
               fluidPreset.get_num(&fluidPreset),
               fluidPreset.get_name(&fluidPreset));
+#else
+      fluid_preset_t *fluidPreset;
+      fluid_sfont_iteration_start(fluidSoundfont);
+      OPARMS oparms;
+      csound->GetOParms(csound, &oparms);
+      if (oparms.msglevel & 0x7)
+        while (fluidPreset = fluid_sfont_iteration_next(fluidSoundfont))
+        {
+          log(csound, "SoundFont: %3d  Bank: %3d  Preset: %3d  %s\n",
+              soundFontId, fluid_preset_get_banknum(fluidPreset),
+              fluid_preset_get_num(fluidPreset),
+              fluid_preset_get_name(fluidPreset));
+#endif
         }
     }
     return result;
@@ -277,19 +292,32 @@ public:
       LockGuard guard(csound, mutex);
       int32_t result = OK;
       toa(iFluidSynth, fluidSynth);
-      fluid_sfont_t *fluidSoundfont =
-        fluid_synth_get_sfont(fluidSynth, 0);
+      fluid_sfont_t *fluidSoundfont = fluid_synth_get_sfont(fluidSynth, 0);
+#if FLUIDSYNTH_VERSION_MAJOR < 2
       fluid_preset_t fluidPreset;
       fluidSoundfont->iteration_start(fluidSoundfont);
       OPARMS oparms;
       csound->GetOParms(csound, &oparms);
       if (oparms.msglevel & 0x7)
         while (fluidSoundfont->iteration_next(fluidSoundfont, &fluidPreset))
-          {
+        {
             std::stringstream ss;
             ss << "Bank: " << fluidPreset.get_banknum(&fluidPreset) <<
               " Preset: " << fluidPreset.get_num(&fluidPreset) <<
                 " Name: " << fluidPreset.get_name(&fluidPreset);
+#else
+      fluid_preset_t *fluidPreset;
+      fluid_sfont_iteration_start(fluidSoundfont);
+      OPARMS oparms;
+      csound->GetOParms(csound, &oparms);
+      if (oparms.msglevel & 0x7)
+        while (fluidPreset = fluid_sfont_iteration_next(fluidSoundfont))
+        {
+            std::stringstream ss;
+            ss << "Bank: " << fluid_preset_get_banknum(fluidPreset) <<
+              " Preset: " << fluid_preset_get_num(fluidPreset) <<
+                " Name: " << fluid_preset_get_name(fluidPreset);
+#endif
           programs.push_back(ss.str());
         }
 
