diff -ru netwatch-1.3.1-2.orig/dispdata.c netwatch-1.3.1-2/dispdata.c
--- netwatch-1.3.1-2.orig/dispdata.c	2022-03-11 21:56:12.752986724 +0100
+++ netwatch-1.3.1-2/dispdata.c	2022-03-11 21:57:36.180151654 +0100
@@ -1488,7 +1488,7 @@
 	      fprintf (fpspeclog, "%s\n", ttt);
 	    else
 	      //!!mvprintw (yact, xleft, "%s",ttt);
-	      mvprintw (yact, xleft, ttt);
+	      mvprintw (yact, xleft, "%s", ttt);
 	    attron (col4);
 	    if (current->update)
 	    {
@@ -1736,7 +1736,7 @@
 	      fprintf (fpspeclog, "%s\n", ttt);
 	    else
 	      //!!mvprintw (yact, xright,"%s", ttt);
-	      mvprintw (yact, xright, ttt);
+	      mvprintw (yact, xright, "%s", ttt);
 	    attron (col4);
 	    if (current->update)
 	    {
