diff -Nur mysql-workbench-gpl-5.2.34-src.orig/ext/ctemplate/ctemplate-src/src/ctemplate/template_dictionary.h.in mysql-workbench-gpl-5.2.34-src/ext/ctemplate/ctemplate-src/src/ctemplate/template_dictionary.h.in
--- mysql-workbench-gpl-5.2.34-src.orig/ext/ctemplate/ctemplate-src/src/ctemplate/template_dictionary.h.in	2011-05-26 09:33:54.128803186 +0000
+++ mysql-workbench-gpl-5.2.34-src/ext/ctemplate/ctemplate-src/src/ctemplate/template_dictionary.h.in	2011-05-26 09:34:45.312611839 +0000
@@ -54,6 +54,7 @@
 #include <map>
 #include <string>
 #include <vector>
+#include <cstddef>
 #include <ctemplate/template_dictionary_interface.h>
 #include <ctemplate/template_string.h>
 #include <ctemplate/template_modifiers.h>
