diff -urN rtorrent-0.8.4.orig/src/command_download.cc rtorrent-0.8.4/src/command_download.cc
--- rtorrent-0.8.4.orig/src/command_download.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/command_download.cc	2009-02-15 00:03:15.000000000 -0800
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstdio>
 #include <functional>
 #include <unistd.h>
 #include <rak/file_stat.h>
diff -urN rtorrent-0.8.4.orig/src/command_events.cc rtorrent-0.8.4/src/command_events.cc
--- rtorrent-0.8.4.orig/src/command_events.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/command_events.cc	2009-02-14 23:53:18.000000000 -0800
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <cstdio>
+
 #include <functional>
 #include <rak/file_stat.h>
 #include <rak/path.h>
diff -urN rtorrent-0.8.4.orig/src/command_network.cc rtorrent-0.8.4/src/command_network.cc
--- rtorrent-0.8.4.orig/src/command_network.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/command_network.cc	2009-02-15 00:11:13.000000000 -0800
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstdio>
 #include <functional>
 #include <rak/address_info.h>
 #include <rak/path.h>
diff -urN rtorrent-0.8.4.orig/src/command_ui.cc rtorrent-0.8.4/src/command_ui.cc
--- rtorrent-0.8.4.orig/src/command_ui.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/command_ui.cc	2009-02-15 00:12:06.000000000 -0800
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstdio>
 #include <ctime>
 #include <rak/functional.h>
 #include <rak/functional_fun.h>
diff -urN rtorrent-0.8.4.orig/src/core/dht_manager.cc rtorrent-0.8.4/src/core/dht_manager.cc
--- rtorrent-0.8.4.orig/src/core/dht_manager.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/core/dht_manager.cc	2009-02-15 00:12:51.000000000 -0800
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstdio>
 #include <fstream>
 #include <sstream>
 #include <torrent/object.h>
diff -urN rtorrent-0.8.4.orig/src/display/text_element_string.cc rtorrent-0.8.4/src/display/text_element_string.cc
--- rtorrent-0.8.4.orig/src/display/text_element_string.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/display/text_element_string.cc	2009-02-15 00:13:03.000000000 -0800
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstdio>
 #include <rak/string_manip.h>
 
 #include "rpc/parse_commands.h"
diff -urN rtorrent-0.8.4.orig/src/display/text_element_value.cc rtorrent-0.8.4/src/display/text_element_value.cc
--- rtorrent-0.8.4.orig/src/display/text_element_value.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/display/text_element_value.cc	2009-02-15 00:13:13.000000000 -0800
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstdio>
 #include <ctime>
 
 #include "globals.h"
diff -urN rtorrent-0.8.4.orig/src/display/utils.h rtorrent-0.8.4/src/display/utils.h
--- rtorrent-0.8.4.orig/src/display/utils.h	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/display/utils.h	2009-02-13 20:37:03.000000000 -0800
@@ -39,6 +39,7 @@
 
 #include <ctime>
 #include <string>
+#include <cstdio>
 
 namespace core {
   class Download;
diff -urN rtorrent-0.8.4.orig/src/display/window_download_statusbar.cc rtorrent-0.8.4/src/display/window_download_statusbar.cc
--- rtorrent-0.8.4.orig/src/display/window_download_statusbar.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/display/window_download_statusbar.cc	2009-02-15 00:13:22.000000000 -0800
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstdio>
 #include <torrent/rate.h>
 #include <torrent/tracker_list.h>
 #include <torrent/peer/connection_list.h>
diff -urN rtorrent-0.8.4.orig/src/rpc/parse.cc rtorrent-0.8.4/src/rpc/parse.cc
--- rtorrent-0.8.4.orig/src/rpc/parse.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/rpc/parse.cc	2009-02-13 20:37:29.000000000 -0800
@@ -37,6 +37,7 @@
 #include "config.h"
 
 #include <cstring>
+#include <cstdio>
 #include <locale>
 #include <rak/path.h>
 #include <torrent/exceptions.h>
diff -urN rtorrent-0.8.4.orig/src/rpc/parse_commands.cc rtorrent-0.8.4/src/rpc/parse_commands.cc
--- rtorrent-0.8.4.orig/src/rpc/parse_commands.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/rpc/parse_commands.cc	2009-02-15 00:13:36.000000000 -0800
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstdio>
 #include <algorithm>
 #include <fstream>
 #include <string>
diff -urN rtorrent-0.8.4.orig/src/rpc/scgi_task.cc rtorrent-0.8.4/src/rpc/scgi_task.cc
--- rtorrent-0.8.4.orig/src/rpc/scgi_task.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/rpc/scgi_task.cc	2009-02-13 20:46:56.000000000 -0800
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <cstdio>
+
 #include <rak/error_number.h>
 #include <sys/types.h>
 #include <sys/socket.h>
diff -urN rtorrent-0.8.4.orig/src/rpc/xmlrpc.cc rtorrent-0.8.4/src/rpc/xmlrpc.cc
--- rtorrent-0.8.4.orig/src/rpc/xmlrpc.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/rpc/xmlrpc.cc	2009-02-13 21:08:39.000000000 -0800
@@ -98,7 +98,7 @@
 
 #ifdef XMLRPC_HAVE_I8
   case XMLRPC_TYPE_I8:
-    long long v2;
+    xmlrpc_int64 v2;
     xmlrpc_read_i8(env, tmp, &v2);
     xmlrpc_DECREF(tmp);
     return v2;
@@ -244,7 +244,7 @@
 
 #ifdef XMLRPC_HAVE_I8
   case XMLRPC_TYPE_I8:
-    long long v2;
+    xmlrpc_int64 v2;
     xmlrpc_read_i8(env, value, &v2);
       
     return torrent::Object((int64_t)v2);
diff -urN rtorrent-0.8.4.orig/src/utils/lockfile.cc rtorrent-0.8.4/src/utils/lockfile.cc
--- rtorrent-0.8.4.orig/src/utils/lockfile.cc	2009-02-13 20:36:44.000000000 -0800
+++ rtorrent-0.8.4/src/utils/lockfile.cc	2009-02-14 23:43:56.000000000 -0800
@@ -40,6 +40,7 @@
 #include <cctype>
 #include <cerrno>
 #include <cstring>
+#include <cstdio>
 #include <sstream>
 #include <fcntl.h>
 #include <signal.h>
