diff -up mysql-workbench-gpl-5.2.33-src/backend/wbprivate/workbench/metaclasses.cpp.gcc46 mysql-workbench-gpl-5.2.33-src/backend/wbprivate/workbench/metaclasses.cpp
--- mysql-workbench-gpl-5.2.33-src/backend/wbprivate/workbench/metaclasses.cpp.gcc46	2011-03-23 17:05:33.000000000 +0100
+++ mysql-workbench-gpl-5.2.33-src/backend/wbprivate/workbench/metaclasses.cpp	2011-03-23 17:06:02.000000000 +0100
@@ -23,7 +23,7 @@
 #include "grts/structs.workbench.model.reporting.h"
 #include "grts/structs.workbench.physical.h"
 
-
+namespace wb {
 void register_all_metaclasses()
 {
   register_structs_app_xml();
@@ -46,3 +46,6 @@ void register_all_metaclasses()
   register_structs_workbench_model_reporting_xml();
   register_structs_workbench_physical_xml();  
 }
+// end of namespace wb
+}
+
diff -up mysql-workbench-gpl-5.2.33-src/frontend/linux/linux_utilities/toolbar_manager.cpp.gcc46 mysql-workbench-gpl-5.2.33-src/frontend/linux/linux_utilities/toolbar_manager.cpp
--- mysql-workbench-gpl-5.2.33-src/frontend/linux/linux_utilities/toolbar_manager.cpp.gcc46	2011-03-23 16:56:16.000000000 +0100
+++ mysql-workbench-gpl-5.2.33-src/frontend/linux/linux_utilities/toolbar_manager.cpp	2011-03-23 17:07:36.000000000 +0100
@@ -31,7 +31,7 @@ static Gtk::ComboBox *create_color_combo
   }
 
   Glib::RefPtr<Gtk::ListStore> model= Gtk::ListStore::create(*color_combo_columns);
-  Gtk::ComboBox *combo= new Gtk::ComboBox(model);
+  Gtk::ComboBox *combo= new Gtk::ComboBox((Glib::RefPtr<Gtk::TreeModel> &)model);
 
   combo->pack_start(color_combo_columns->image);
 
diff -up mysql-workbench-gpl-5.2.33-src/library/forms/utilities.cpp.gcc46 mysql-workbench-gpl-5.2.33-src/library/forms/utilities.cpp
--- mysql-workbench-gpl-5.2.33-src/library/forms/utilities.cpp.gcc46	2011-03-23 17:04:17.000000000 +0100
+++ mysql-workbench-gpl-5.2.33-src/library/forms/utilities.cpp	2011-03-23 17:04:40.000000000 +0100
@@ -32,10 +32,11 @@
 #include <math.h>
 #include "base/log.h"
 ENABLE_LOG("mforms.utils")
-
 using namespace mforms;
 
+namespace mforms {
 GThread *_mforms_main_thread=0;
+}
 
 static std::map<std::string, int> remembered_message_answers;
 static std::string remembered_message_answer_file;
diff -up mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_lua_help.cpp.gcc46 mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_lua_help.cpp
--- mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_lua_help.cpp.gcc46	2011-03-17 13:48:45.000000000 +0100
+++ mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_lua_help.cpp	2011-03-23 16:44:22.000000000 +0100
@@ -1909,6 +1909,8 @@ static MYX_GRT_SHELL_COMMAND_HELP_GROUP 
 
 //============================================================================
 
+namespace grt {
+
 static void myx_grt_shell_show_command_help_print(grt::GRT *grt, const char *group_name, const char *cmd)
 {
   MYX_GRT_SHELL_COMMAND_HELP_GROUP help_group;
@@ -2097,3 +2099,6 @@ void myx_grt_shell_show_help(grt::GRT *g
   }
 }
 
+// End of namspace grt
+}
+
diff -up mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_python_help.cpp.gcc46 mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_python_help.cpp
--- mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_python_help.cpp.gcc46	2011-03-17 13:48:41.000000000 +0100
+++ mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_python_help.cpp	2011-03-23 16:44:22.000000000 +0100
@@ -187,7 +187,7 @@ NL
 };
 
 
-void grt_shell_show_python_help(grt::GRT *grt, const char *command)
+void grt::grt_shell_show_python_help(grt::GRT *grt, const char *command)
 {
   if (!command || !*command)
     grt->send_output(
diff -up mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_python_help.h.gcc46 mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_python_help.h
--- mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_python_help.h.gcc46	2011-03-17 13:48:39.000000000 +0100
+++ mysql-workbench-gpl-5.2.33-src/library/grt/src/grtpp_shell_python_help.h	2011-03-23 16:44:22.000000000 +0100
@@ -21,6 +21,8 @@
 #ifndef __grtpp_shell_python_help_h__
 #define __grtpp_shell_python_help_h__
 
+namespace grt {
 void grt_shell_show_python_help(grt::GRT *grt, const char *command);
+}
 
-#endif /* __grtpp_shell_python_help_h__ */
\ Pas de fin de ligne à la fin du fichier.
+#endif /* __grtpp_shell_python_help_h__ */
diff -up mysql-workbench-gpl-5.2.33-src/modules/db.mysql/src/db_mysql_catalog_report.h.gcc46 mysql-workbench-gpl-5.2.33-src/modules/db.mysql/src/db_mysql_catalog_report.h
--- mysql-workbench-gpl-5.2.33-src/modules/db.mysql/src/db_mysql_catalog_report.h.gcc46	2011-03-23 16:51:30.000000000 +0100
+++ mysql-workbench-gpl-5.2.33-src/modules/db.mysql/src/db_mysql_catalog_report.h	2011-03-23 16:52:33.000000000 +0100
@@ -10,6 +10,7 @@
 #include "config.h"
 #endif
 
+#include <cstddef>
 #include "ctemplate/template.h"
 
 #include "grt/common.h"
