--- rtorrent-0.8.2.orig/src/command_ui.cc
+++ rtorrent-0.8.2/src/command_ui.cc
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <ctime>
+
 #include <rak/functional.h>
 #include <rak/functional_fun.h>
 #include <sigc++/adaptors/bind.h>
--- rtorrent-0.8.2.orig/src/rpc/parse.cc
+++ rtorrent-0.8.2/src/rpc/parse.cc
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstring>
 #include <locale>
 #include <rak/path.h>
 #include <torrent/exceptions.h>
--- rtorrent-0.8.2.orig/src/rpc/exec_file.cc
+++ rtorrent-0.8.2/src/rpc/exec_file.cc
@@ -36,6 +36,7 @@
 
 #include "config.h"
 
+#include <cstring>
 #include <string>
 #include <unistd.h>
 #include <rak/error_number.h>
--- rtorrent-0.8.2.orig/src/display/text_element_value.cc
+++ rtorrent-0.8.2/src/display/text_element_value.cc
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <ctime>
+
 #include "globals.h"
 #include "text_element_value.h"
 
--- rtorrent-0.8.2.orig/rak/functional.h
+++ rtorrent-0.8.2/rak/functional.h
@@ -37,6 +37,7 @@
 #ifndef RAK_FUNCTIONAL_H
 #define RAK_FUNCTIONAL_H
 
+#include <cstddef>
 #include <functional>
 
 namespace rak {
