diff --git a/src/libponyc/codegen/genexe.c b/src/libponyc/codegen/genexe.c
index 304a6f675..a063818d8 100644
--- a/src/libponyc/codegen/genexe.c
+++ b/src/libponyc/codegen/genexe.c
@@ -277,7 +277,7 @@ static bool link_exe(compile_t* c, ast_t* program,
 #ifdef PONY_USE_LTO
     "-flto -fuse-linker-plugin "
 #endif
-    "%s %s %s %s -lpthread %s -lm %s",
+    "%s %s %s %s -lpthread -latomic %s -lm %s",
     linker, file_exe, arch, mcx16_arg, fuseld, file_o, lib_args, ponyrt, ldl,
     export
     );
diff --git a/Makefile b/Makefile
index a795d8f57..dd8923b97 100644
--- a/Makefile
+++ b/Makefile
@@ -437,11 +437,11 @@ libponyc.benchmarks.links = libgbenchmark libponyc libponyrt llvm
 libponyrt.benchmarks.links = libgbenchmark libponyrt
 
 ifeq ($(OSTYPE),linux)
-  ponyc.links += libpthread libdl
-  libponyc.tests.links += libpthread libdl
-  libponyrt.tests.links += libpthread libdl
-  libponyc.benchmarks.links += libpthread libdl
-  libponyrt.benchmarks.links += libpthread libdl
+  ponyc.links += libpthread libdl libatomic
+  libponyc.tests.links += libpthread libdl libatomic
+  libponyrt.tests.links += libpthread libdl libatomic
+  libponyc.benchmarks.links += libpthread libdl libatomic
+  libponyrt.benchmarks.links += libpthread libdl libatomic
 endif

 