diff -Naur activity-log-manager-0.9.7.orig/configure.ac activity-log-manager-0.9.7/configure.ac
--- activity-log-manager-0.9.7.orig/configure.ac	2013-07-10 08:28:23.000000000 +0200
+++ activity-log-manager-0.9.7/configure.ac	2014-08-11 15:16:37.573942945 +0200
@@ -43,7 +43,7 @@
 
 PKG_CHECK_MODULES(ZEITGEIST, zeitgeist-2.0, [HAVE_LIBZG=yes])
 PKG_CHECK_MODULES(GTK, gtk+-3.0, [HAVE_GTK=yes])
-PKG_CHECK_MODULES(GEE, gee-1.0, [HAVE_GEE=yes])
+PKG_CHECK_MODULES(GEE, gee-0.8, [HAVE_GEE=yes])
 PKG_CHECK_MODULES(GIO_UNIX, gio-unix-2.0, [HAVE_GIO_UNIX=yes])
 PKG_CHECK_MODULES(POLKIT, polkit-gobject-1, [HAVE_POLKIT=yes], [HAVE_POLKIT=no])
 PKG_CHECK_MODULES(LIBWHOOPSIE, libwhoopsie, [HAVE_LIBWHOOPSIE=yes], [HAVE_LIBWHOOPSIE=no])
diff -Naur activity-log-manager-0.9.7.orig/src/Makefile.am activity-log-manager-0.9.7/src/Makefile.am
--- activity-log-manager-0.9.7.orig/src/Makefile.am	2013-07-10 08:28:23.000000000 +0200
+++ activity-log-manager-0.9.7/src/Makefile.am	2014-08-11 15:35:14.909210489 +0200
@@ -2,7 +2,7 @@
 	--pkg zeitgeist-2.0 \
 	--pkg glib-2.0 \
 	--pkg gtk+-3.0 \
-	--pkg gee-1.0  \
+	--pkg gee-0.8  \
 	--target-glib=2.26 \
 	--pkg gio-unix-2.0 \
 	config.vapi
diff -Naur activity-log-manager-0.9.7.orig/src/unified-privacy.vala activity-log-manager-0.9.7/src/unified-privacy.vala
--- activity-log-manager-0.9.7.orig/src/unified-privacy.vala	2014-08-11 15:15:53.586144000 +0200
+++ activity-log-manager-0.9.7/src/unified-privacy.vala	2014-08-12 01:27:54.292926227 +0200
@@ -114,7 +114,7 @@
 				remove_folder_from_view(folder);
 			});
 			
-			mime_dict = new HashMap<string, string>(str_hash, str_equal);
+			mime_dict = new HashMap<string, string>((Gee.HashDataFunc<string>) str_hash, (Gee.EqualDataFunc<string>) str_equal);
 			mime_dict.set(_("Music"), NFO.AUDIO);
 			mime_dict.set(_("Videos"), NFO.VIDEO);
 			mime_dict.set(_("Pictures"), NFO.IMAGE);
@@ -213,7 +213,7 @@
 			// Checkbox Box
 			var checkbox_hbox = new Box(Orientation.VERTICAL, 0);
 
-			CompareFunc<string> mime_compare = (s1, s2) => {
+			GLib.CompareDataFunc<string> mime_compare = (s1, s2) => {
 				return s1.collate(s2);
 			};
 			Gee.ArrayList<string> keys = new Gee.ArrayList<string>();
