diff -Naur gloobus-sushi.orig/src/gloobus-sushi.py gloobus-sushi/src/gloobus-sushi.py
--- gloobus-sushi.orig/src/gloobus-sushi.py	2014-01-22 21:54:41.297590000 +0100
+++ gloobus-sushi/src/gloobus-sushi.py	2014-01-22 21:45:21.244184455 +0100
@@ -16,7 +16,7 @@
 
 import dbus
 import dbus.service
-import gobject
+from gi.repository import GObject
 import os
 
 from dbus.mainloop.glib import DBusGMainLoop
@@ -28,7 +28,7 @@
     def __init__(self):
         bus_name = dbus.service.BusName(DBUS_NAME, bus=dbus.SessionBus())
         super(DBusWrapper, self).__init__(bus_name, DBUS_PATH)
-        self.loop = gobject.MainLoop()
+        self.loop = GObject.MainLoop()
         self.loop.run()
 
     @dbus.service.method(DBUS_NAME)
