diff -ruN a/Makefile b/Makefile
--- a/Makefile	2022-10-31 16:07:16.000000000 +0100
+++ b/Makefile	2022-11-02 22:36:55.426618672 +0100
@@ -101,7 +101,7 @@
 else
 override CFLAGS += -DPTW32_STATIC_LIB
 endif
-override LOADLIBES += `$(PKG_CONFIG) $(PKG_UI_FLAGS) --libs glib-2.0`
+override LOADLIBES += `$(PKG_CONFIG) $(PKG_UI_FLAGS) --libs glib-2.0 fluidsynth`
 
 
 ###############################################################################
@@ -126,38 +126,8 @@
 $(BUILDDIR)%.sf2:
 	cp -v sf2/$(*F).sf2 $@
 
-FLUID_SRC = \
-            fluidsynth/src/fluid_adsr_env.c \
-            fluidsynth/src/fluid_chan.c \
-            fluidsynth/src/fluid_chorus.c \
-            fluidsynth/src/fluid_conv.c \
-            fluidsynth/src/fluid_defsfont.c \
-            fluidsynth/src/fluid_event.c \
-            fluidsynth/src/fluid_gen.c \
-            fluidsynth/src/fluid_hash.c \
-            fluidsynth/src/fluid_iir_filter.c \
-            fluidsynth/src/fluid_lfo.c \
-            fluidsynth/src/fluid_list.c \
-            fluidsynth/src/fluid_midi.c \
-            fluidsynth/src/fluid_mod.c \
-            fluidsynth/src/fluid_rev.c \
-            fluidsynth/src/fluid_ringbuffer.c \
-            fluidsynth/src/fluid_rvoice.c \
-            fluidsynth/src/fluid_rvoice_dsp.c \
-            fluidsynth/src/fluid_rvoice_event.c \
-            fluidsynth/src/fluid_rvoice_mixer.c \
-            fluidsynth/src/fluid_samplecache.c \
-            fluidsynth/src/fluid_settings.c \
-            fluidsynth/src/fluid_sffile.c \
-            fluidsynth/src/fluid_sfont.c \
-            fluidsynth/src/fluid_synth.c \
-            fluidsynth/src/fluid_synth_monopoly.c \
-            fluidsynth/src/fluid_sys.c \
-            fluidsynth/src/fluid_tuning.c \
-            fluidsynth/src/fluid_voice.c
-
-CPPFLAGS += -Ifluidsynth -I fluidsynth/fluidsynth -DHAVE_CONFIG_H -D DEFAULT_SOUNDFONT=\"\"
-DSP_SRC  = src/$(LV2NAME).c $(FLUID_SRC)
+CPPFLAGS += -DHAVE_CONFIG_H -D DEFAULT_SOUNDFONT=\"\"
+DSP_SRC  = src/$(LV2NAME).c
 DSP_DEPS = $(DSP_SRC)
 
 $(BUILDDIR)$(LV2NAME)$(LIB_EXT): $(DSP_DEPS) Makefile
