diff -Naur pychess-0.10beta3-old/glade/newInOut.glade pychess-0.10beta3/glade/newInOut.glade
--- pychess-0.10beta3-old/glade/newInOut.glade	2009-10-20 00:01:26.000000000 +1000
+++ pychess-0.10beta3/glade/newInOut.glade	2010-05-04 21:23:36.321983001 +1000
@@ -169,7 +169,7 @@
 			  <property name="spacing">0</property>
 
 			  <child>
-			    <widget class="GtkImage" id="image341">
+			    <widget class="GtkImage" id="playersIcon">
 			      <property name="width_request">58</property>
 			      <property name="visible">True</property>
 			      <property name="icon_size">6</property>
@@ -466,7 +466,7 @@
 			  <property name="spacing">0</property>
 
 			  <child>
-			    <widget class="GtkImage" id="image343">
+			    <widget class="GtkImage" id="timeIcon">
 			      <property name="width_request">58</property>
 			      <property name="visible">True</property>
 			      <property name="icon_size">6</property>
diff -Naur pychess-0.10beta3-old/glade/taskers.glade pychess-0.10beta3/glade/taskers.glade
--- pychess-0.10beta3-old/glade/taskers.glade	2009-10-20 00:01:26.000000000 +1000
+++ pychess-0.10beta3/glade/taskers.glade	2010-05-04 21:23:36.325315855 +1000
@@ -439,7 +439,7 @@
 		      </child>
 
 		      <child>
-			<widget class="GtkImage" id="image996">
+			<widget class="GtkImage" id="startIcon">
 			  <property name="visible">True</property>
 			  <property name="icon_name">stock_init</property>
 			  <property name="pixel_size">48</property>
diff -Naur pychess-0.10beta3-old/lib/pychess/ic/ICLounge.py pychess-0.10beta3/lib/pychess/ic/ICLounge.py
--- pychess-0.10beta3-old/lib/pychess/ic/ICLounge.py	2010-02-26 03:23:03.000000000 +1000
+++ pychess-0.10beta3/lib/pychess/ic/ICLounge.py	2010-05-04 21:23:36.415315895 +1000
@@ -22,6 +22,7 @@
 from pychess.widgets.ChatWindow import ChatWindow
 from pychess.widgets.SpotGraph import SpotGraph
 from pychess.Utils.const import *
+from pychess.Utils.IconLoader import load_icon
 from pychess.Utils.TimeModel import TimeModel
 from pychess.Utils.GameModel import GameModel
 from pychess.Players.ICPlayer import ICPlayer
@@ -616,14 +617,12 @@
 
 class PlayerTabSection (ParrentListSection):
 
-    icons = gtk.icon_theme_get_default()
-    l = gtk.ICON_LOOKUP_USE_BUILTIN
-    peoplepix = icons.load_icon("stock_people", 15, l)
-    bookpix = icons.load_icon("stock_book_blue", 15, l)
-    easypix = icons.load_icon("weather-few-clouds", 15, l)
-    advpix = icons.load_icon("weather-overcast", 15, l)
-    exppix = icons.load_icon("weather-storm", 15, l)
-    cmppix = icons.load_icon("stock_notebook", 15, l)
+    peoplepix = load_icon(15, "stock_people", "system-users")
+    bookpix = load_icon(15, "stock_book_blue", "accessories-dictionary")
+    easypix = load_icon(15, "weather-few-clouds")
+    advpix = load_icon(15, "weather-overcast")
+    exppix = load_icon(15, "weather-storm")
+    cmppix = load_icon(15, "stock_notebook", "computer")
 
     def __init__ (self, widgets, connection):
         ParrentListSection.__init__(self)
@@ -718,8 +717,7 @@
 
         self.games = {}
 
-        icons = gtk.icon_theme_get_default()
-        self.recpix = icons.load_icon("media-record", 16, gtk.ICON_LOOKUP_USE_BUILTIN)
+        self.recpix = load_icon(16, "media-record")
         self.clearpix = pixbuf_new_from_file(addDataPrefix("glade/board.png"))
 
         self.tv = self.widgets["gametreeview"]
@@ -846,9 +844,8 @@
 
         # Set up the treeview
 
-        icons = gtk.icon_theme_get_default()
-        self.wpix = icons.load_icon("stock_draw-rounded-square-unfilled", 16, gtk.ICON_LOOKUP_USE_BUILTIN)
-        self.bpix = icons.load_icon("stock_draw-rounded-square", 16, gtk.ICON_LOOKUP_USE_BUILTIN)
+        self.wpix = load_icon(16, "stock_draw-rounded-square-unfilled", "computer")
+        self.bpix = load_icon(16, "stock_draw-rounded-square", "computer")
 
         self.tv = widgets["adjournedtreeview"]
         self.store = gtk.ListStore(gtk.gdk.Pixbuf, str, str, str, str)
diff -Naur pychess-0.10beta3-old/lib/pychess/Utils/IconLoader.py pychess-0.10beta3/lib/pychess/Utils/IconLoader.py
--- pychess-0.10beta3-old/lib/pychess/Utils/IconLoader.py	1970-01-01 10:00:00.000000000 +1000
+++ pychess-0.10beta3/lib/pychess/Utils/IconLoader.py	2010-05-04 21:23:36.388649469 +1000
@@ -0,0 +1,14 @@
+from gtk import icon_theme_get_default, ICON_LOOKUP_USE_BUILTIN
+
+it = icon_theme_get_default()
+def load_icon(size, name, alternative=None):
+    try:
+        return it.load_icon(name, size, ICON_LOOKUP_USE_BUILTIN)
+    except:
+        if alternative is not None:
+            try:
+                return it.load_icon(alternative, size, ICON_LOOKUP_USE_BUILTIN)
+            except:
+                print "no %s icon in icon-theme-gnome" % alternative
+        else:
+            print "no %s icon in icon-theme-gnome" % name
diff -Naur pychess-0.10beta3-old/lib/pychess/widgets/ChatWindow.py pychess-0.10beta3/lib/pychess/widgets/ChatWindow.py
--- pychess-0.10beta3-old/lib/pychess/widgets/ChatWindow.py	2009-10-20 00:00:31.000000000 +1000
+++ pychess-0.10beta3/lib/pychess/widgets/ChatWindow.py	2010-05-04 21:23:36.431982442 +1000
@@ -4,6 +4,7 @@
 import gobject
 import pango
 
+from pychess.Utils.IconLoader import load_icon
 from pychess.System import uistuff
 from pychess.System.glock import glock_connect
 from pychess.widgets.ChatView import ChatView
@@ -90,8 +91,7 @@
         self.append_column(self.leftcol)
         
         # Second column
-        icons = gtk.icon_theme_get_default()
-        pixbuf = icons.load_icon(icon_name, 16, gtk.ICON_LOOKUP_USE_BUILTIN)
+        pixbuf = load_icon(16, icon_name)
         crp = gtk.CellRendererPixbuf()
         crp.props.pixbuf = pixbuf
         self.rightcol = gtk.TreeViewColumn("", crp)
diff -Naur pychess-0.10beta3-old/lib/pychess/widgets/gamewidget.py pychess-0.10beta3/lib/pychess/widgets/gamewidget.py
--- pychess-0.10beta3-old/lib/pychess/widgets/gamewidget.py	2010-02-19 02:45:44.000000000 +1000
+++ pychess-0.10beta3/lib/pychess/widgets/gamewidget.py	2010-05-04 21:23:36.441982681 +1000
@@ -6,8 +6,8 @@
 import cStringIO
 
 import gtk, gobject
-from gtk import ICON_LOOKUP_USE_BUILTIN
 
+from pychess.Utils.IconLoader import load_icon
 from pychess.System.Log import log
 from pychess.System import glock, conf, prefix
 from ChessClock import ChessClock
@@ -35,26 +35,19 @@
     notebook.set_show_border(False)
     return notebook
 
-icons = gtk.icon_theme_get_default()
-def lookup16 (name, alternative=None):
-    try:
-        return icons.load_icon(name, 16, ICON_LOOKUP_USE_BUILTIN)
-    except:
-        return icons.load_icon(alternative, 16, ICON_LOOKUP_USE_BUILTIN)
-
 def createImage (pixbuf):
     image = gtk.Image()
     image.set_from_pixbuf(pixbuf)
     return image
 
-light_on = lookup16("stock_3d-light-on", "weather-clear")
-light_off = lookup16("stock_3d-light-off", "weather-clear-night")
-gtk_close = lookup16("gtk-close")
-
-media_previous = lookup16("gtk-media-previous-ltr")
-media_rewind = lookup16("gtk-media-rewind-ltr")
-media_forward = lookup16("gtk-media-forward-ltr")
-media_next = lookup16("gtk-media-next-ltr")
+light_on = load_icon(16, "stock_3d-light-on", "weather-clear")
+light_off = load_icon(16, "stock_3d-light-off", "weather-clear-night")
+gtk_close = load_icon(16, "gtk-close")
+
+media_previous = load_icon(16, "gtk-media-previous-ltr")
+media_rewind = load_icon(16, "gtk-media-rewind-ltr")
+media_forward = load_icon(16, "gtk-media-forward-ltr")
+media_next = load_icon(16, "gtk-media-next-ltr")
 
 GAME_MENU_ITEMS = ("save_game1", "save_game_as1", "properties1", "close1")
 ACTION_MENU_ITEMS = ("draw", "pause1", "resume1", "undo1", 
diff -Naur pychess-0.10beta3-old/lib/pychess/widgets/newGameDialog.py pychess-0.10beta3/lib/pychess/widgets/newGameDialog.py
--- pychess-0.10beta3-old/lib/pychess/widgets/newGameDialog.py	2010-03-02 06:40:27.000000000 +1000
+++ pychess-0.10beta3/lib/pychess/widgets/newGameDialog.py	2010-05-04 21:23:36.445316055 +1000
@@ -20,6 +20,7 @@
     from gtksourceview2 import View as SourceView
     from gtksourceview2 import LanguageManager
 
+from pychess.Utils.IconLoader import load_icon
 from pychess.Utils.GameModel import GameModel
 from pychess.Utils.TimeModel import TimeModel
 from pychess.Utils.const import *
@@ -41,35 +42,35 @@
 # Background.Taskers so they have a similar look.
 #===============================================================================
 
-it = gtk.icon_theme_get_default()
-
-iwheels = it.load_icon("gtk-execute", 24, gtk.ICON_LOOKUP_USE_BUILTIN)
-ipeople = it.load_icon("stock_people", 24, gtk.ICON_LOOKUP_USE_BUILTIN)
-inotebook = it.load_icon("stock_notebook", 24, gtk.ICON_LOOKUP_USE_BUILTIN)
-speople = it.load_icon("stock_people", 16, gtk.ICON_LOOKUP_USE_BUILTIN)
-snotebook = it.load_icon("stock_notebook", 16, gtk.ICON_LOOKUP_USE_BUILTIN)
+big_time = load_icon(48, "stock_alarm", "appointment-soon")
+big_people = load_icon(48, "stock_people", "system-users")
+iwheels = load_icon(24, "gtk-execute")
+ipeople = load_icon(24, "stock_people", "system-users")
+inotebook = load_icon(24, "stock_notebook", "computer")
+speople = load_icon(16, "stock_people", "system-users")
+snotebook = load_icon(16, "stock_notebook", "computer")
 
 skillToIcon = {
-    1: it.load_icon("weather-clear", 16, gtk.ICON_LOOKUP_USE_BUILTIN),
-    2: it.load_icon("weather-few-clouds", 16, gtk.ICON_LOOKUP_USE_BUILTIN),
-    3: it.load_icon("weather-few-clouds", 16, gtk.ICON_LOOKUP_USE_BUILTIN),
-    4: it.load_icon("weather-overcast", 16, gtk.ICON_LOOKUP_USE_BUILTIN),
-    5: it.load_icon("weather-overcast", 16, gtk.ICON_LOOKUP_USE_BUILTIN),
-    6: it.load_icon("weather-showers", 16, gtk.ICON_LOOKUP_USE_BUILTIN),
-    7: it.load_icon("weather-showers", 16, gtk.ICON_LOOKUP_USE_BUILTIN),
-    8: it.load_icon("weather-storm", 16, gtk.ICON_LOOKUP_USE_BUILTIN),
+    1: load_icon(16, "weather-clear"),
+    2: load_icon(16, "weather-few-clouds"),
+    3: load_icon(16, "weather-few-clouds"),
+    4: load_icon(16, "weather-overcast"),
+    5: load_icon(16, "weather-overcast"),
+    6: load_icon(16, "weather-showers"),
+    7: load_icon(16, "weather-showers"),
+    8: load_icon(16, "weather-storm"),
 }
 
 # Used by TaskerManager. Put here to help synchronization
 skillToIconLarge = {
-    1: it.load_icon("weather-clear", 48, gtk.ICON_LOOKUP_USE_BUILTIN),
-    2: it.load_icon("weather-few-clouds", 48, gtk.ICON_LOOKUP_USE_BUILTIN),
-    3: it.load_icon("weather-few-clouds", 48, gtk.ICON_LOOKUP_USE_BUILTIN),
-    4: it.load_icon("weather-overcast", 48, gtk.ICON_LOOKUP_USE_BUILTIN),
-    5: it.load_icon("weather-overcast", 48, gtk.ICON_LOOKUP_USE_BUILTIN),
-    6: it.load_icon("weather-showers", 48, gtk.ICON_LOOKUP_USE_BUILTIN),
-    7: it.load_icon("weather-showers", 48, gtk.ICON_LOOKUP_USE_BUILTIN),
-    8: it.load_icon("weather-storm", 48, gtk.ICON_LOOKUP_USE_BUILTIN),
+    1: load_icon(48, "weather-clear"),
+    2: load_icon(48, "weather-few-clouds"),
+    3: load_icon(48, "weather-few-clouds"),
+    4: load_icon(48, "weather-overcast"),
+    5: load_icon(48, "weather-overcast"),
+    6: load_icon(48, "weather-showers"),
+    7: load_icon(48, "weather-showers"),
+    8: load_icon(48, "weather-storm"),
 }
 
 playerItems = []
@@ -119,6 +120,9 @@
         uistuff.createCombo(cls.widgets["blackPlayerCombobox"],
                             (i[:2] for i in playerItems[0]))
 
+        cls.widgets["playersIcon"].set_from_pixbuf(big_people)
+        cls.widgets["timeIcon"].set_from_pixbuf(big_time)
+
 
         def on_playerCombobox_changed (widget, skillHbox):
             skillHbox.props.visible = widget.get_active() > 0
diff -Naur pychess-0.10beta3-old/lib/pychess/widgets/preferencesDialog.py pychess-0.10beta3/lib/pychess/widgets/preferencesDialog.py
--- pychess-0.10beta3-old/lib/pychess/widgets/preferencesDialog.py	2010-02-26 03:23:03.000000000 +1000
+++ pychess-0.10beta3/lib/pychess/widgets/preferencesDialog.py	2010-05-04 21:23:36.445316055 +1000
@@ -5,6 +5,7 @@
 from pychess.System import conf, gstreamer, uistuff
 from pychess.Players.engineNest import discoverer
 from pychess.Utils.const import *
+from pychess.Utils.IconLoader import load_icon
 
 firstRun = True
 def run(widgets):
@@ -315,17 +316,12 @@
                  (_("Beep"), "stock_bell", "audio-x-generic"), 
                  (_("Select sound file..."), "gtk-open", "document-open"))
         
-        it = gtk.icon_theme_get_default()
         items = []
         for level, stock, altstock in icons:
-            try:
-                image = it.load_icon(stock, 16, gtk.ICON_LOOKUP_USE_BUILTIN)
-            except gobject.GError:
-                image = it.load_icon(altstock, 16, gtk.ICON_LOOKUP_USE_BUILTIN)
+            image = load_icon(16, stock, altstock)
             items += [(image, level)]
         
-        audioIco = it.load_icon("audio-x-generic", 16,
-                                gtk.ICON_LOOKUP_USE_BUILTIN)
+        audioIco = load_icon(16, "audio-x-generic")
         
         # Set-up combos
         
diff -Naur pychess-0.10beta3-old/lib/pychess/widgets/TaskerManager.py pychess-0.10beta3/lib/pychess/widgets/TaskerManager.py
--- pychess-0.10beta3-old/lib/pychess/widgets/TaskerManager.py	2009-10-20 00:00:31.000000000 +1000
+++ pychess-0.10beta3/lib/pychess/widgets/TaskerManager.py	2010-05-04 21:23:36.435315735 +1000
@@ -13,6 +13,7 @@
 from pychess.widgets import newGameDialog
 from pychess.ic import ICLogon
 
+from pychess.Utils.IconLoader import load_icon
 from pychess.Utils.GameModel import GameModel
 from pychess.Utils.TimeModel import TimeModel
 from pychess.Players.Human import Human
@@ -207,6 +208,9 @@
             ionest.generalStart(gamemodel, player0tup, player1tup)
         else: ionest.generalStart(gamemodel, player1tup, player0tup)
 
+
+big_start = load_icon(48, "stock_init", "gnome-globe")
+
 class InternetGameTasker (gtk.Alignment):
     
     def __init__ (self):
@@ -228,6 +232,8 @@
         
         self.widgets["connectButton"].connect("clicked", self.connectClicked)
         self.widgets["opendialog2"].connect("clicked", self.openDialogClicked)
+
+        self.widgets["startIcon"].set_from_pixbuf(big_start)
     
     def openDialogClicked (self, button):
         ICLogon.run()
diff -Naur pychess-0.10beta3-old/lib/pychess/widgets/ToggleComboBox.py pychess-0.10beta3/lib/pychess/widgets/ToggleComboBox.py
--- pychess-0.10beta3-old/lib/pychess/widgets/ToggleComboBox.py	2009-10-20 00:00:31.000000000 +1000
+++ pychess-0.10beta3/lib/pychess/widgets/ToggleComboBox.py	2010-05-04 21:23:36.435315735 +1000
@@ -2,7 +2,10 @@
 pygtk.require("2.0")
 import gtk
 from gobject import *
+
 from pychess.System.Log import log
+from pychess.Utils.IconLoader import load_icon
+
 
 class ToggleComboBox (gtk.ToggleButton):
 
@@ -78,8 +81,7 @@
             label = gtk.Label(text)
             label.props.xalign = 0
             if type(stock) == str:
-                stock = gtk.icon_theme_get_default().load_icon(
-                        stock, 12, gtk.ICON_LOOKUP_USE_BUILTIN)            
+                stock = load_icon(12, stock)
             image = gtk.Image()
             image.set_from_pixbuf(stock)            
             hbox = gtk.HBox()
