diff -Nur gambas-3.17.0/gb.gtk3/configure.ac gambas-3.17.0a/gb.gtk3/configure.ac
--- gambas-3.17.0/gb.gtk3/configure.ac	2022-03-09 14:28:18.000000000 +0100
+++ gambas-3.17.0a/gb.gtk3/configure.ac	2022-03-13 12:57:36.551478184 +0100
@@ -36,7 +36,7 @@
 
 GB_COMPONENT_PKG_CONFIG(
   gtk3opengl, GTK3OPENGL, gb.gtk3.opengl, [opengl],
-  'gtk+-3.0 >= 3.16'
+  'gtk+-3.0 >= 3.16' gl
 )
 
 AC_OUTPUT( \
diff -Nur gambas-3.17.0/gb.gtk3/src/opengl/Makefile.am gambas-3.17.0a/gb.gtk3/src/opengl/Makefile.am
--- gambas-3.17.0/gb.gtk3/src/opengl/Makefile.am	2022-03-09 14:28:18.000000000 +0100
+++ gambas-3.17.0a/gb.gtk3/src/opengl/Makefile.am	2022-03-13 12:58:11.352346960 +0100
@@ -3,9 +3,9 @@
 
 gblib_LTLIBRARIES = gb.gtk3.opengl.la
 
-gb_gtk3_opengl_la_LIBADD = @GTK3_LIB@
-gb_gtk3_opengl_la_LDFLAGS = -module @LD_FLAGS@ @GTK3_LDFLAGS@
-gb_gtk3_opengl_la_CPPFLAGS = @GTK3_INC@
+gb_gtk3_opengl_la_LIBADD = @GTK3OPENGL_LIB@
+gb_gtk3_opengl_la_LDFLAGS = -module @LD_FLAGS@ @GTK3OPENGL_LDFLAGS@
+gb_gtk3_opengl_la_CPPFLAGS = @GTK3OPENGL_INC@
 
 gb_gtk3_opengl_la_SOURCES = \
  main.h main.c \
