From 70893dc7a32fb983f44c8e9be07ffe65baf6bbd3 Mon Sep 17 00:00:00 2001
From: Michael Webster <miketwebster@gmail.com>
Date: Mon, 6 Apr 2015 21:17:28 -0400
Subject: [PATCH] main.c: Add CLUTTER_BACKEND="x11" for forward-compatibility.

---
 cinnamon-session/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cinnamon-session/main.c b/cinnamon-session/main.c
index 011cd6e..1f94e12 100644
--- a/cinnamon-session/main.c
+++ b/cinnamon-session/main.c
@@ -348,6 +348,7 @@ main (int argc, char **argv)
          * detect if GNOME is running. We keep this for compatibility reasons.
          */
         csm_util_setenv ("GNOME_DESKTOP_SESSION_ID", "this-is-deprecated");
+        csm_util_setenv ("CLUTTER_BACKEND", "x11");
 
         client_store = csm_store_new ();
 
