diff -Naur cinnamon-control-center-3.2.0.orig/autogen.sh cinnamon-control-center-3.2.0.changed/autogen.sh
--- cinnamon-control-center-3.2.0.orig/autogen.sh	2016-11-07 07:18:37.000000000 -0500
+++ cinnamon-control-center-3.2.0.changed/autogen.sh	2016-11-07 17:41:38.407293799 -0500
@@ -24,7 +24,6 @@
 
 aclocal --install || exit 1
 glib-gettextize --force --copy || exit 1
-gtkdocize --copy || exit 1
 intltoolize --force --copy --automake || exit 1
 autoreconf --verbose --force --install || exit 1
 
