diff -rupN guayadeque-0.3.7.orig/src/CMakeLists.txt guayadeque-0.3.7/src/CMakeLists.txt
--- guayadeque-0.3.7.orig/src/CMakeLists.txt	2015-05-30 11:46:08.985762500 +0200
+++ guayadeque-0.3.7/src/CMakeLists.txt	2015-05-30 11:47:40.436389200 +0200
@@ -319,6 +319,7 @@ TARGET_LINK_LIBRARIES( guayadeque
 	${LIBINDICATE07_LIBRARIES}
 	${LIBAPPINDICATOR_LIBRARIES}
 	${LIBWXSQLITE3_LIBRARIES}
+	-lgdk_pixbuf-2.0
 	)
 
 INSTALL( TARGETS guayadeque
