diff --git a/clash-prelude/clash-prelude.cabal b/clash-prelude/clash-prelude.cabal
index 43dfaa6fc..2e2c1e619 100644
--- a/clash-prelude/clash-prelude.cabal
+++ b/clash-prelude/clash-prelude.cabal
@@ -326,7 +326,7 @@ Library
                       ghc-typelits-extra        >= 0.4     && < 0.5,
                       ghc-typelits-knownnat     >= 0.7.2   && < 0.8,
                       ghc-typelits-natnormalise >= 0.7.2   && < 0.8,
-                      hashable                  >= 1.2.1.0  && < 1.4,
+                      hashable                  >= 1.2.1.0  && < 1.5,
                       half                      >= 0.2.2.3 && < 1.0,
                       interpolate               >= 0.2     && < 0.3,
                       lens                      >= 4.10    && < 5.1.0,
diff --git a/clash-prelude/src/Clash/Annotations/Primitive.hs b/clash-prelude/src/Clash/Annotations/Primitive.hs
index 475b0977a..13db3aece 100644
--- a/clash-prelude/src/Clash/Annotations/Primitive.hs
+++ b/clash-prelude/src/Clash/Annotations/Primitive.hs
@@ -272,7 +272,7 @@ data Primitive
   -- ^ Description of a primitive for a given 'HDL's in a file at 'FilePath'
   | InlinePrimitive [HDL] String
   -- ^ Description of a primitive for a given 'HDL's as an inline 'String'
-  deriving (Show, Read, Data, Generic, NFData, Hashable)
+  deriving (Show, Read, Data, Generic, NFData, Hashable, Eq)
 
 -- | Primitive guard to mark a value as either not translatable or as having a
 -- blackbox with an optional extra warning. Helps Clash generate better error
@@ -287,7 +287,9 @@ data PrimitiveGuard a
   | HasBlackBox [PrimitiveWarning] a
   -- ^ Marks a value as having a blackbox. Clash will error if it hasn't found
   -- a blackbox.
-    deriving (Show, Read, Data, Generic, NFData, Hashable, Functor, Foldable, Traversable, Binary)
+  deriving
+    ( Show, Read, Data, Generic, NFData, Hashable, Functor, Foldable
+    , Traversable, Binary, Eq )
 
 -- | Warning that will be emitted on instantiating a guarded value.
 data PrimitiveWarning
@@ -296,7 +298,7 @@ data PrimitiveWarning
   -- instantiated in a non-testbench context.
   | WarnAlways String
   -- ^ Always emit warning upon primitive instantiation.
-    deriving (Show, Read, Data, Generic, NFData, Hashable, Binary)
+    deriving (Show, Read, Data, Generic, NFData, Hashable, Binary, Eq)
 
 -- | Extract primitive definition from a PrimitiveGuard. Will yield Nothing
 -- for guards of value 'DontTranslate'.
