diff --color -aur lesspipe-1.91-old/lesspipe.sh.in lesspipe-1.91-new/lesspipe.sh.in
--- lesspipe-1.91-old/lesspipe.sh.in	2021-11-10 20:11:39.000000000 +0200
+++ lesspipe-1.91-new/lesspipe.sh.in	2021-11-26 14:43:55.134403225 +0200
@@ -600,9 +600,9 @@
     return
 #ifdef html2text
   elif cmd_exist html2text; then
-    if [[ "$1" = - ]]; then html2text -utf8 2>/dev/null ||
+    if [[ "$1" = - ]]; then html2text - utf8 2>/dev/null ||
       html2text -from_encoding utf-8
-    else html2text -utf8 "$1" 2>/dev/null ||
+    else html2text "$1" utf8 2>/dev/null ||
       html2text -from_encoding utf-8 "$1"; fi
 #ifdef lynx
   elif cmd_exist lynx; then
