diff --git a/authenticate/OpenId2/Discovery.hs b/authenticate/OpenId2/Discovery.hs
index 8b3f63f..0db3b17 100644
--- a/authenticate/OpenId2/Discovery.hs
+++ b/authenticate/OpenId2/Discovery.hs
@@ -81,7 +81,9 @@ discoverYADIS ident mb_loc redirects manager = do
     let uri = fromMaybe (unpack $ identifier ident) mb_loc
     req <- liftIO $ parseUrl uri
     res <- httpLbs req
-#if MIN_VERSION_http_conduit(1, 9, 0)
+#if MIN_VERSION_http_conduit(2, 2, 0)
+        { checkResponse = \_ _ -> return ()
+#elif MIN_VERSION_http_conduit(1, 9, 0)
         { checkStatus = \_ _ _ -> Nothing
 #else
         { checkStatus = \_ _ -> Nothing
