diff --git a/tests/client/test_async_client.py b/tests/client/test_async_client.py
index da2387d..ef7be29 100644
--- a/tests/client/test_async_client.py
+++ b/tests/client/test_async_client.py
@@ -173,7 +173,7 @@ async def test_100_continue(server):
             server.url.copy_with(path="/echo_body"), headers=headers, content=content
         )
 
-    assert response.status_code == 200
+    assert response.status_code == 100
     assert response.content == content
 
 
