diff --git a/patchwork/files.py b/patchwork/files.py
index 74b5a74..4dd1c3d 100644
--- a/patchwork/files.py
+++ b/patchwork/files.py
@@ -3,8 +3,7 @@ Tools for file and directory management.
 """
 
 import re
-
-from invoke.vendor import six
+import six
 
 from .util import set_runner
 
diff --git a/patchwork/transfers.py b/patchwork/transfers.py
index e7292da..a1d97bd 100644
--- a/patchwork/transfers.py
+++ b/patchwork/transfers.py
@@ -2,7 +2,7 @@
 File transfer functionality above and beyond basic ``put``/``get``.
 """
 
-from invoke.vendor import six
+import six
 
 
 def rsync(
