diff -wbBur psi-plus-snapshots.org/src/groupchatdlg.cpp psi-plus-snapshots/src/groupchatdlg.cpp
--- psi-plus-snapshots.org/src/groupchatdlg.cpp	2018-01-05 19:34:53.698040879 +0300
+++ psi-plus-snapshots/src/groupchatdlg.cpp	2018-01-05 19:35:35.934708491 +0300
@@ -1518,7 +1518,7 @@
         QString nick = d->self;
 
         if(!account()->groupChatJoin(host, room, nick, d->password)) {
-            appendSysMsg(tr("Error: You are in or joining this room already!"), true);
+//            appendSysMsg(tr("Error: You are in or joining this room already!"), true);
             d->state = Private::Idle;
         }
     }
diff -wbBur psi-plus-snapshots.org/src/mucjoindlg.cpp psi-plus-snapshots/src/mucjoindlg.cpp
--- psi-plus-snapshots.org/src/mucjoindlg.cpp	2018-01-05 19:34:53.711374212 +0300
+++ psi-plus-snapshots/src/mucjoindlg.cpp	2018-01-05 19:35:41.211375275 +0300
@@ -262,7 +262,7 @@
 
 
     if (!account_->groupChatJoin(host, room, nick, pass, !ui_.ck_history->isChecked())) {
-        QMessageBox::information(this, tr("Error"), tr("You are in or joining this room already!"));
+//        QMessageBox::information(this, tr("Error"), tr("You are in or joining this room already!"));
         return;
     }
 
