diff --git a/configure.ac b/configure.ac
index 839d012..eead71d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -194,9 +194,7 @@ fi
 
 PKG_CHECK_MODULES(DBUS, dbus-1)
 PKG_CHECK_MODULES(XML2, libxml-2.0)
-if test "x$with_zypp" = "xyes"; then
    PKG_CHECK_MODULES(JSONC, json-c, [], [AC_MSG_WARN([Cannot find json-c. Please install libjson-c-devel])])
-fi
 
 AC_CHECK_HEADER(acl/libacl.h,[],[AC_MSG_ERROR([Cannout find libacl headers. Please install libacl-devel])])
