diff -ruN a/extras/Projucer/Source/Settings/jucer_StoredSettings.cpp b/extras/Projucer/Source/Settings/jucer_StoredSettings.cpp
--- a/extras/Projucer/Source/Settings/jucer_StoredSettings.cpp	2019-10-16 11:09:54.000000000 +0200
+++ b/extras/Projucer/Source/Settings/jucer_StoredSettings.cpp	2020-01-07 15:11:25.562748860 +0100
@@ -381,15 +381,15 @@
 {
     if (key == Ids::jucePath)
     {
-        return (os == TargetOS::windows ? "C:\\JUCE" : "~/JUCE");
+        return (os == TargetOS::windows ? "C:\\JUCE" : "/usr/share/juce");
     }
     else if (key == Ids::defaultJuceModulePath)
     {
-        return (os == TargetOS::windows ? "C:\\JUCE\\modules" : "~/JUCE/modules");
+        return (os == TargetOS::windows ? "C:\\JUCE\\modules" : "/usr/share/juce/modules");
     }
     else if (key == Ids::defaultUserModulePath)
     {
-        return (os == TargetOS::windows ? "C:\\modules" : "~/modules");
+        return (os == TargetOS::windows ? "C:\\modules" : "~/.local/share/JUCE/modules");
     }
     else if (key == Ids::vstLegacyPath || key == Ids::vst3Path)
     {
