diff --git a/stdlib/LinearAlgebra/src/LinearAlgebra.jl b/stdlib/LinearAlgebra/src/LinearAlgebra.jl
index 624cca69b8..daa52bf300 100644
--- a/stdlib/LinearAlgebra/src/LinearAlgebra.jl
+++ b/stdlib/LinearAlgebra/src/LinearAlgebra.jl
@@ -631,6 +631,8 @@ end
 function __init__()
     try
         BLAS.lbt_forward(OpenBLAS_jll.libopenblas_path; clear=true)
+        BLAS.lbt_forward("/usr/lib/liblapack64.so")
+        BLAS.lbt_forward("/usr/lib/libcblas64.so")
         BLAS.check()
     catch ex
         Base.showerror_nostdio(ex, "WARNING: Error during initialization of module LinearAlgebra")
