diff --git a/stdlib/LinearAlgebra/src/lbt.jl b/stdlib/LinearAlgebra/src/lbt.jl
index b1a2dc24b3..b0d6242738 100644
--- a/stdlib/LinearAlgebra/src/lbt.jl
+++ b/stdlib/LinearAlgebra/src/lbt.jl
@@ -155,7 +155,7 @@ function lbt_set_num_threads(nthreads)
 end
 
 function lbt_forward(path; clear::Bool = false, verbose::Bool = false)
-    ccall((:lbt_forward, libblastrampoline), Int32, (Cstring, Int32, Int32), path, clear ? 1 : 0, verbose ? 1 : 0)
+    ccall((:lbt_forward, libblastrampoline), Int32, (Cstring, Int32, Int32, Cstring), path, clear ? 1 : 0, verbose ? 1 : 0, C_NULL)
 end
 
 function lbt_set_default_func(addr)
