diff -ruN a/hack/lib/golang.sh b/hack/lib/golang.sh
--- a/hack/lib/golang.sh	2021-12-07 19:08:39.000000000 +0100
+++ b/hack/lib/golang.sh	2021-12-09 18:53:23.588259078 +0100
@@ -718,8 +718,8 @@
       -ldflags "${goldflags:-}"
       -tags "${gotags:-}"
     )
-    V=1 kube::log::info "> static build CGO_ENABLED=0: ${statics[*]}"
-    CGO_ENABLED=0 kube::golang::build_some_binaries "${statics[@]}"
+    V=1 kube::log::info "> static build CGO_ENABLED=1: ${statics[*]}"
+    CGO_ENABLED=1 kube::golang::build_some_binaries "${statics[@]}"
   fi
 
   if [[ "${#nonstatics[@]}" != 0 ]]; then
