Index: Makefile.fpc
===================================================================
--- Makefile.fpc	(wersja 37097)
+++ Makefile.fpc	(kopia robocza)
@@ -51,6 +51,6 @@
         -$(DEL) $(COMPILER_UNITTARGETDIR)/lazcontrols$(PPUEXT)
 
 compiled:
-        $(COPY) Makefile.compiled $(COMPILER_UNITTARGETDIR)/LazControls.compiled
+        $(CPPROG) -f Makefile.compiled $(COMPILER_UNITTARGETDIR)/LazControls.compiled
 
 all: cleartarget $(COMPILER_UNITTARGETDIR) lazcontrols$(PPUEXT) compiled
Index: Makefile.compiled
===================================================================
--- Makefile.compiled	(wersja 37097)
+++ Makefile.compiled	(kopia robocza)
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
 <CONFIG>
   <Makefile Value="True"/>
-  <Params Value=" -MObjFPC -Scghi -O1 -gl -vewnhi -l -Fu../../lcl/units/%(CPU_TARGET)-%(OS_TARGET);../../lcl/units/%(CPU_TARGET)-%(OS_TARGET)/%(LCL_PLATFORM);../../packager/units/%(CPU_TARGET)-%(OS_TARGET);. -Fi. -dLCL -dLCL%(LCL_PLATFORM)"/>
+  <Params Value=" -MObjFPC -Scghi -O1 -gl -vewnhi -l -Fu../../lcl/units/%(CPU_TARGET)-%(OS_TARGET);../../lcl/units/%(CPU_TARGET)-%(OS_TARGET)/%(LCL_PLATFORM);../../packager/units/%(CPU_TARGET)-%(OS_TARGET);.;.;lib/%(CPU_TARGET)-%(OS_TARGET) -Fi. -dLCL -dLCL%(LCL_PLATFORM) lazcontrols.pas"/>
 </CONFIG>
Index: Makefile
===================================================================
--- Makefile	(wersja 37097)
+++ Makefile	(kopia robocza)
@@ -2961,5 +2961,5 @@
 cleartarget:
 	-$(DEL) $(COMPILER_UNITTARGETDIR)/lazcontrols$(PPUEXT)
 compiled:
-	$(COPY) Makefile.compiled $(COMPILER_UNITTARGETDIR)/LazControls.compiled
+	$(CPPROG) -f Makefile.compiled $(COMPILER_UNITTARGETDIR)/LazControls.compiled
 all: cleartarget $(COMPILER_UNITTARGETDIR) lazcontrols$(PPUEXT) compiled
Index: lazcontrols.lpk
===================================================================
--- lazcontrols.lpk	(wersja 37097)
+++ lazcontrols.lpk	(kopia robocza)
@@ -10,6 +10,7 @@
         <UnitOutputDirectory Value="lib\$(TargetCPU)-$(TargetOS)"/>
       </SearchPaths>
       <Other>
+        <CustomOptions Value="$(IDEBuildOptions)"/>
         <CompilerPath Value="$(CompPath)"/>
       </Other>
     </CompilerOptions>
