--- a/Makefile
+++ b/Makefile
@@ -1,10 +1,11 @@
 PREFIX ?= /usr/local
 WFLAGS ?= -Wall -Wextra -Wmissing-prototypes -Wdiv-by-zero -Wbad-function-cast -Wcast-align -Wcast-qual -Wfloat-equal -Wmissing-declarations -Wnested-externs -Wno-unknown-pragmas -Wpointer-arith -Wredundant-decls -Wstrict-prototypes -Wswitch-enum -Wno-type-limits
 CFLAGS ?= -Os -march=native -fno-exceptions $(WFLAGS)
-CFLAGS += -I. -Iext/libhydrogen
-OBJ = ext/libhydrogen/hydrogen.o src/encpipe.o src/safe_rw.o
+CPPFLAGS += -I. -Iext/libhydrogen
 STRIP ?= strip
 
+OBJ = ext/libhydrogen/hydrogen.o src/encpipe.o src/safe_rw.o
+
 SRC = \
 	ext/libhydrogen/hydrogen.c \
 	src/common.h \
@@ -20,10 +21,10 @@ bin: encpipe
 $(OBJ): $(SRC)
 
 .c.o:
-	$(CC) $(CFLAGS) -o $@ -c $<
+	$(CC) $(CFLAGS) $(CPPFLAGS) -o $@ -c $<
 
 encpipe: $(OBJ)
-	$(CC) $(CFLAGS) -o encpipe $(OBJ)
+	$(CC) $(CFLAGS) -o encpipe $(LDFLAGS) $(OBJ)
 
 ext/libhydrogen/hydrogen.c:
 	git submodule update --init || echo "** Make sure you cloned the repository **" >&2
