From 211d72396a55b63696ed39d97c54e382d139ee15 Mon Sep 17 00:00:00 2001
From: Jonas Witschel <diabonas@archlinux.org>
Date: Sat, 2 Nov 2019 19:02:49 +0100
Subject: [PATCH] Use system gtest instead of building from submodule

---
 UnitTests/CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/UnitTests/CMakeLists.txt b/UnitTests/CMakeLists.txt
index 59ae773..ffad014 100644
--- a/UnitTests/CMakeLists.txt
+++ b/UnitTests/CMakeLists.txt
@@ -1,7 +1,5 @@
-ADD_SUBDIRECTORY (googletest EXCLUDE_FROM_ALL)
 enable_testing()
 	
-SET(gtest_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/googletest/googletest")
 add_definitions( -DTESTFILESPATH="${CMAKE_CURRENT_SOURCE_DIR}/../TestFiles")
 add_definitions( -DLTESTFILESPATH=L"${CMAKE_CURRENT_SOURCE_DIR}/../TestFiles")
 add_definitions( -DLOUTFILESPATH=L"${CMAKE_BINARY_DIR}/")
-- 
2.23.0

