--- a/usr/lib/emscripten/tools/system_libs.py
+++ b/usr/lib/emscripten/tools/system_libs.py
@@ -758,6 +758,9 @@ class libcxxabi(CXXLibrary, NoExceptLibrary, MTLibrary):
   name = 'libc++abi'
   depends = ['libc']
   cflags = ['-std=c++11', '-Oz', '-D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS']
+  includes = [
+    ['system', 'lib', 'libunwind', 'include'],
+  ]
 
   def get_cflags(self):
     cflags = super(libcxxabi, self).get_cflags()
