--- a/script/build
+++ b/script/build
@@ -13,7 +13,7 @@ from lib.config import MIPS64EL_GCC, IS_MIPS64EL_HOST, set_mips64el_env, \
 SOURCE_ROOT = os.path.abspath(os.path.dirname(os.path.dirname(__file__)))
 VENDOR_DIR = os.path.join(SOURCE_ROOT, 'vendor')
 TARGETS = ['chromiumcontent_all']
-COMPONENTS = ['static_library', 'shared_library', 'ffmpeg']
+COMPONENTS = ['static_library']
 
 NINJA = 'ninja'
 if sys.platform == 'win32':
--- a/script/create-dist
+++ b/script/create-dist
@@ -76,7 +76,7 @@ BINARIES = {
     'libffmpeg.dylib',
   ],
   'linux': [
-    'libffmpeg.so',
+    # 'libffmpeg.so',
   ],
   'win32': [
     'd3dcompiler_47.dll',
@@ -390,7 +390,7 @@ def generate_ninja(args, ninja):
       copy_generated_sources(target_arch, component, ninja)
       copy_locales(target_arch, component, ninja)
 
-  copy_ffmpeg(target_arch, ninja)
+  # copy_ffmpeg(target_arch, ninja)
   copy_sources(ninja)
   generate_licenses(ninja)
 
@@ -513,9 +513,9 @@ def copy_binaries(target_arch, component, create_debug_archive,
       else:
         binaries = [ 'chromedriver', 'clang_x64_v8_mips64el/mksnapshot' ]
     else:
-      binaries = [ 'chromedriver', 'mksnapshot' ]
+      binaries = [ 'mksnapshot' ]
 
-    ffmpeg_output_dir = get_output_dir(SOURCE_ROOT, target_arch, 'ffmpeg')
+    ffmpeg_output_dir = get_output_dir(SOURCE_ROOT, target_arch, 'static_library')
     for binary in binaries:
       ninja.copy(os.path.join(ffmpeg_output_dir, binary), target_dir)
 
--- a/script/update
+++ b/script/update
@@ -19,7 +19,7 @@ VENDOR_DIR = os.path.join(SOURCE_ROOT, 'vendor')
 SRC_DIR = os.path.join(SOURCE_ROOT, 'src')
 CHROMIUMCONTENT_SOURCE_DIR = os.path.join(SOURCE_ROOT, 'chromiumcontent')
 CHROMIUMCONTENT_DESTINATION_DIR = os.path.join(SRC_DIR, 'chromiumcontent')
-COMPONENTS = ['static_library', 'shared_library', 'ffmpeg']
+COMPONENTS = ['static_library']
 DEPOT_TOOLS = os.path.join(VENDOR_DIR, 'depot_tools')
 
 NINJA = os.path.join(DEPOT_TOOLS, 'ninja')
