From b9bbe17c08e5dcbe3ce841e6bed52ce8d8b10f9e Mon Sep 17 00:00:00 2001
From: John Lindgren <john.lindgren@aol.com>
Date: Thu, 29 May 2014 22:42:53 -0400
Subject: Remove "Requires: glib-2.0" since libmms no longer depends on GLib.

Signed-off-by: Hans de Goede <j.w.r.degoede@gmail.com>

diff --git a/pkgconfig/libmms.pc.in b/pkgconfig/libmms.pc.in
index 602c967..a86d15a 100644
--- a/pkgconfig/libmms.pc.in
+++ b/pkgconfig/libmms.pc.in
@@ -4,8 +4,7 @@ libdir=@libdir@
 includedir=@includedir@/
 
 Name: libmms
-Description: Library implementing the MMS protocol 
-Requires: glib-2.0
+Description: Library implementing the MMS protocol
 Version: @VERSION@
 Libs: -L${libdir} -lmms -lm
 Cflags: -I${includedir}
