--- boinc_core_release_6_12_34/clientgui/gtk/taskbarex.cpp	2012-02-16 14:01:10.116369636 +1100
+++ boinc_core_release_6_12_34-build/clientgui/gtk/taskbarex.cpp	2012-02-16 15:17:26.001649020 +1100
@@ -227,11 +227,10 @@
     if (!g_pNotification)
     {
         g_pNotification = 
-            notify_notification_new_with_status_icon(
+            notify_notification_new(
                 title.mb_str(),
                 message.mb_str(),
-                desired_icon,
-                g_pStatusIcon
+                gtk_status_icon_get_icon_name(g_pStatusIcon)
         );
 
         g_signal_connect(
