diff -rupN libretro-reicast.orig/Makefile libretro-reicast/Makefile
--- libretro-reicast.orig/Makefile	2018-10-08 09:46:38.219688904 +0200
+++ libretro-reicast/Makefile	2018-10-08 09:48:19.840450287 +0200
@@ -26,12 +26,12 @@ CC_AS    = ${CC_PREFIX}as
 
 MFLAGS   := 
 ASFLAGS  := 
-LDFLAGS  :=
+LDFLAGS  := ${LDFLAGS}
 LDFLAGS_END :=
 INCFLAGS :=
 LIBS     :=
-CFLAGS   := 
-CXXFLAGS :=
+CFLAGS   := ${CFLAGS}
+CXXFLAGS := ${CXXFLAGS}
 
 GIT_VERSION := " $(shell git rev-parse --short HEAD || echo unknown)"
 ifneq ($(GIT_VERSION)," unknown")
