diff -wbBur lilypond-2.18.2/lily/pango-font.cc lilypond-2.18.2.fix/lily/pango-font.cc
--- lilypond-2.18.2/lily/pango-font.cc	2014-03-17 19:29:16.000000000 +0400
+++ lilypond-2.18.2.fix/lily/pango-font.cc	2016-12-02 19:36:55.634555707 +0300
@@ -315,7 +315,8 @@
   pango_fc_font_unlock_face (fcfont);
   pango_glyph_string_free (pgs);
   pgs = 0;
-  PangoFontDescription *descr = pango_font_describe (pa->font);
+//  PangoFontDescription *descr = pango_font_describe (pa->font);
+  PangoFontDescription *descr = pango_context_get_font_description (context_);
   Real size = pango_font_description_get_size (descr)
               / (Real (PANGO_SCALE));
 
