diff --git a/CMakeLists.txt b/CMakeLists.txt
index c81da8b5..967e7cb7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -416,8 +416,7 @@ if (WITH_LLVM)
     endforeach()
     set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -DNDEBUG")
 
-    llvm_map_components_to_libnames(llvm_libs_direct ${SYMENGINE_LLVM_COMPONENTS})
-    llvm_expand_dependencies(llvm_libs ${llvm_libs_direct})
+    set (llvm_libs LLVM)
 
     if (BUILD_SHARED_LIBS)
         set(SYMENGINE_LLVM_LINK_DOWNSTREAM_DEFAULT False)
