From e0caf5391989534a684845999f567bab16832cd3 Mon Sep 17 00:00:00 2001
From: Maxime Gauduin <alucryd@archlinux.org>
Date: Fri, 14 Jul 2017 14:59:34 +0200
Subject: [PATCH] Customize libexecdir

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 0eee8e9b..017bbd2b 100644
--- a/meson.build
+++ b/meson.build
@@ -18,7 +18,7 @@ dependency('gobject-introspection-1.0', version: '>=1.35.9')
 dependency('gtk+-3.0', version :'>=3.14')
 
 python_dir = join_paths(get_option('prefix'), python.sysconfig_path('purelib'))
-LIBEXEC_DIR = join_paths(get_option('prefix'), 'libexec')
+LIBEXEC_DIR = join_paths(get_option('prefix'), get_option('libexecdir'))
 DATA_DIR = join_paths(get_option('prefix'), get_option('datadir'), meson.project_name())
 
 conf = configuration_data()
