diff -ru lrcalc-2.0.orig/src/hashtab.tpl.h lrcalc-2.0/src/hashtab.tpl.h
--- lrcalc-2.0.orig/src/hashtab.tpl.h	2021-01-31 03:10:04.000000000 +0100
+++ lrcalc-2.0/src/hashtab.tpl.h	2021-02-01 17:26:50.827256281 +0100
@@ -1,7 +1,7 @@
 #include <stdio.h>
 
 #include "alloc.h"
-
+#include "ivlincomb.h"
 
 /* This structure allows fast lookup, insertion, and reset.
    Remove is less optimized.  Iterating through the table will be
