diff -rupN music-0.4.1.orig/data/org.pantheon.noise.appdata.xml.in music-0.4.1/data/org.pantheon.noise.appdata.xml.in
--- music-0.4.1.orig/data/org.pantheon.noise.appdata.xml.in	2017-08-08 23:41:12.570798209 +0200
+++ music-0.4.1/data/org.pantheon.noise.appdata.xml.in	2017-08-08 23:42:54.621177616 +0200
@@ -56,7 +56,7 @@
         </release>
     </releases>
     <provides>
-        <binary>noise</binary>
+        <binary>pantheon-music</binary>
         <library>libnoise-core.so.0</library>
     </provides>
     <developer_name>elementary LLC.</developer_name>
diff -rupN music-0.4.1.orig/data/org.pantheon.noise.desktop.in.in music-0.4.1/data/org.pantheon.noise.desktop.in.in
--- music-0.4.1.orig/data/org.pantheon.noise.desktop.in.in	2017-08-08 23:41:12.570798209 +0200
+++ music-0.4.1/data/org.pantheon.noise.desktop.in.in	2017-08-08 23:43:25.620407132 +0200
@@ -3,7 +3,7 @@ Type=Application
 _Name=@APP_NAME@
 _GenericName=Music Player
 _Comment=Listen to music
-Exec=noise %U
+Exec=pantheon-music %U
 Icon=multimedia-audio-player
 Terminal=false
 Categories=Audio;Music;Player;AudioVideo;GNOME;GTK;
diff -rupN music-0.4.1.orig/src/CMakeLists.txt music-0.4.1/src/CMakeLists.txt
--- music-0.4.1.orig/src/CMakeLists.txt	2017-08-08 23:41:12.580797885 +0200
+++ music-0.4.1/src/CMakeLists.txt	2017-08-08 23:45:06.061592605 +0200
@@ -88,7 +88,7 @@ vala_precompile (CLIENT_VALA_C ${SRC_TAR
   ${CLIENT_VAPI}
 )
 
-set (EXEC_NAME ${SRC_TARGET})
+set (EXEC_NAME pantheon-music)
 
 add_definitions (${DEPS_CFLAGS})
 link_directories (${DEPS_LIBRARY_DIRS})
