*** mkmf-gnome2.rb.orig	2011-02-21 12:41:16.563334382 -0600
--- mkmf-gnome2.rb	2011-02-21 12:41:37.610001048 -0600
***************
*** 364,370 ****
      rescue LoadError
      end
    end
!   return false if rcairo_source_dir.nil?
  
    if /mingw|cygwin|mswin32/ =~ RUBY_PLATFORM
      options = {}
--- 364,370 ----
      rescue LoadError
      end
    end
!   return false if rcairo_source_dir.nil? && !(PKGConfig.have_package('cairo') && have_header('rb_cairo.h'))  
  
    if /mingw|cygwin|mswin32/ =~ RUBY_PLATFORM
      options = {}
