--- mypaint-1.2.1/SConstruct.orig	2019-09-07 08:59:20.675785294 +0000
+++ mypaint-1.2.1/SConstruct	2019-09-07 08:59:33.795961507 +0000
@@ -63,7 +63,7 @@
     # patch with a suitable options switch.
     tools.append("mingw")
 
-env = Environment(ENV=os.environ, options=opts, tools=tools)
+env = Environment(ENV=os.environ, options=opts, tools=tools, SWIG='swig-3')
 
 Help(opts.GenerateHelpText(env))
 
