--- a/backend/wbpublic/sqlide/recordset_text_storage.cpp	2014-03-26 16:47:23.000000000 +0100
+++ b/backend/wbpublic/sqlide/recordset_text_storage.cpp	2014-05-09 01:31:36.099430631 +0200
@@ -205,7 +205,7 @@
       if (!pre_tpl)
         g_warning("Failed to open template file: `%s`", pre_tpl_path.c_str()); 
       else
-        pre_tpl->ReloadIfChanged();
+        ctemplate::Template::ReloadAllIfChanged();
     }      
     if (g_file_test((name+".post.tpl").c_str(), G_FILE_TEST_EXISTS))
     {
@@ -214,7 +214,7 @@
       if (!post_tpl)
         g_warning("Failed to open template file: `%s`", post_tpl_path.c_str());
       else
-        post_tpl->ReloadIfChanged();
+        ctemplate::Template::ReloadAllIfChanged();
     }
   }    
   
@@ -223,7 +223,7 @@
       throw std::runtime_error(strfmt("Failed to open output file: `%s`", _file_path.c_str()));
   }
 
-  tpl->ReloadIfChanged();
+  ctemplate::Template::ReloadAllIfChanged();
 
   std::auto_ptr<TemplateDictionary> dict(new TemplateDictionary("/"));
   BOOST_FOREACH (const Parameters::value_type &param, _parameters)