diff -ru ncrack-0.5/ncrack_resume.cc ncrack-0.5.new/ncrack_resume.cc
--- ncrack-0.5/ncrack_resume.cc	2016-04-10 18:56:02.000000000 +0200
+++ ncrack-0.5.new/ncrack_resume.cc	2017-04-13 17:53:09.369522756 +0200
@@ -527,7 +527,7 @@
 
       j = 0;
       buf[j++] = *q;
-      while (*q != '\0' && j < sizeof(buf)) {
+      while ((q != NULL) && (j < sizeof(buf))) {
         q++;
         if (q - filestr >= filelen)
           fatal("Corrupted file! Error 5\n");
