diff --git a/spec/nio/selectables/ssl_socket_spec.rb b/spec/nio/selectables/ssl_socket_spec.rb
index dec7f7d..90c2d5e 100644
--- a/spec/nio/selectables/ssl_socket_spec.rb
+++ b/spec/nio/selectables/ssl_socket_spec.rb
@@ -1,11 +1,9 @@
 # frozen_string_literal: true
 
 require "spec_helper"
+require "openssl"
 
 RSpec.describe OpenSSL::SSL::SSLSocket do
-
-  require "openssl"
-
   before(:all) do
     @tls = []
   end
