diff -upr flask-paranoid-0.2.orig/tests/test_paranoid.py flask-paranoid-0.2/tests/test_paranoid.py
--- flask-paranoid-0.2.orig/tests/test_paranoid.py	2017-11-17 21:44:46.000000000 +0200
+++ flask-paranoid-0.2/tests/test_paranoid.py	2021-12-02 21:05:34.088953533 +0200
@@ -7,7 +7,7 @@ from flask_paranoid import Paranoid
 
 class ParanoidTests(unittest.TestCase):
     def _delete_cookie(self, name):
-        return (name + '=; Expires=Thu, 01-Jan-1970 00:00:00 GMT; '
+        return (name + '=; Expires=Thu, 01 Jan 1970 00:00:00 GMT; '
                 'Max-Age=0; Path=/')
 
     def test_401(self):
