diff --git a/tools/gen_struct_info.py b/tools/gen_struct_info.py
index 404d576ac..fec750809 100755
--- a/tools/gen_struct_info.py
+++ b/tools/gen_struct_info.py
@@ -396,7 +396,7 @@ def inspect_code(headers, cpp_opts, structs, defines):
   # -Oz optimizes enough to avoid warnings on code size/num locals
   cmd = [shared.EMCC] + cpp_opts + ['-o', js_file[1], src_file[1],
                                     '-O0', '--js-opts', '0', '--memory-init-file', '0',
-                                    '-Werror', '-Wno-format',
+                                    '-Werror', '-Wno-format', '-Wno-error=version-check',
                                     '-s', 'BOOTSTRAPPING_STRUCT_INFO=1',
                                     '-s', 'WARN_ON_UNDEFINED_SYMBOLS=0',
                                     '-s', 'STRICT=1',
