diff --git a/libraries/cmake/system/modules/Findlibdevmapper.cmake b/libraries/cmake/system/modules/Findlibdevmapper.cmake
index 47b5f56..c464f5c 100644
--- a/libraries/cmake/system/modules/Findlibdevmapper.cmake
+++ b/libraries/cmake/system/modules/Findlibdevmapper.cmake
@@ -11,5 +11,5 @@ locateSystemLibrary(
   NAME libdevmapper
   REQUIRED ${libdevmapper_REQUIRED}
   MAIN_HEADER "libdevmapper.h"
-  LIBRARIES "devmapper" "lvm2app"
+  LIBRARIES "devmapper"
 )
diff --git a/osquery/tables/system/linux/block_devices.cpp b/osquery/tables/system/linux/block_devices.cpp
index 2eab290..01a4bf3 100644
--- a/osquery/tables/system/linux/block_devices.cpp
+++ b/osquery/tables/system/linux/block_devices.cpp
@@ -20,13 +20,13 @@
 #include <osquery/logger/logger.h>
 
 extern "C" {
-#include <lvm2app.h>
 #include <sys/sysmacros.h>
 }
 
 namespace osquery {
 namespace tables {
 
+#if 0
 void populatePVChildren(lvm_t lvm,
                         const std::string& devname,
                         const std::string& pvid,
@@ -61,6 +61,7 @@ void populatePVChildren(lvm_t lvm,
   }
   lvm_vg_close(vg);
 }
+#endif
 
 static void getBlockDevice(struct udev_device* dev,
                            QueryData& results,
@@ -125,11 +126,13 @@ static void getBlockDevice(struct udev_device* dev,
         r["label"] = blk_value;
       }
       if (boost::algorithm::starts_with(r["type"], "LVM")) {
+#if 0
         lvm_t lvm = lvm_init(nullptr);
         if (lvm != nullptr) {
           populatePVChildren(lvm, name, r["uuid"], lvm_lv2pv);
           lvm_quit(lvm);
         }
+#endif
       }
     }
     blkid_free_probe(pr);
