diff -wbBur virt-manager-0.9.2/src/virtManager/console.py virt-manager-0.9.2.my/src/virtManager/console.py
--- virt-manager-0.9.2/src/virtManager/console.py	2012-07-09 16:09:02.000000000 +0400
+++ virt-manager-0.9.2.my/src/virtManager/console.py	2012-07-12 12:52:26.000000000 +0400
@@ -140,11 +140,11 @@
             nc_params = "%s %s" % (ginfo.gaddr, ginfo.gport)
 
         nc_cmd = (
-            """nc -q 2>&1 | grep "requires an argument" >/dev/null;"""
+            """nc.openbsd -q 2>&1 | grep "requires an argument" >/dev/null;"""
             """if [ $? -eq 0 ] ; then"""
-            """   CMD="nc -q 0 %(nc_params)s";"""
+            """   CMD="nc.openbsd -q 0 %(nc_params)s";"""
             """else"""
-            """   CMD="nc %(nc_params)s";"""
+            """   CMD="nc.openbsd %(nc_params)s";"""
             """fi;"""
             """eval "$CMD";""" %
             {'nc_params': nc_params})
diff -wbBur virt-manager-0.9.2/src/virtManager/manager.py virt-manager-0.9.2.my/src/virtManager/manager.py
--- virt-manager-0.9.2/src/virtManager/manager.py	2012-07-09 16:09:02.000000000 +0400
+++ virt-manager-0.9.2.my/src/virtManager/manager.py	2012-07-12 12:52:17.000000000 +0400
@@ -653,7 +653,7 @@
 
         if conn.is_remote():
             logging.debug(conn.get_transport())
-            if re.search(r"nc: .* -- 'U'", tb):
+            if re.search(r"nc.openbsd: .* -- 'U'", tb):
                 hint += _("The remote host requires a version of netcat/nc\n"
                           "which supports the -U option.")
                 show_errmsg = False
