diff --git a/adb/adb_auth_host.cpp b/adb/adb_auth_host.cpp
index 7b6671d73..d9fc00f86 100644
--- a/adb/adb_auth_host.cpp
+++ b/adb/adb_auth_host.cpp
@@ -71,6 +71,8 @@ static int RSA_to_RSAPublicKey(RSA *rsa, RSAPublicKey *pkey)
     BIGNUM* rem = BN_new();
     BIGNUM* n = BN_new();
     BIGNUM* n0inv = BN_new();
+    const BIGNUM* rsa_n;
+    const BIGNUM* rsa_e;
 
     if (RSA_size(rsa) != RSANUMBYTES) {
         ret = 0;
@@ -78,7 +80,8 @@ static int RSA_to_RSAPublicKey(RSA *rsa, RSAPublicKey *pkey)
     }
 
     BN_set_bit(r32, 32);
-    BN_copy(n, rsa->n);
+    RSA_get0_key (rsa, &rsa_n, &rsa_e, NULL);
+    BN_copy(n, rsa_n);
     BN_set_bit(r, RSANUMWORDS * 32);
     BN_mod_sqr(rr, r, n, ctx);
     BN_div(NULL, rem, n, r32, ctx);
@@ -92,7 +95,7 @@ static int RSA_to_RSAPublicKey(RSA *rsa, RSAPublicKey *pkey)
         BN_div(n, rem, n, r32, ctx);
         pkey->n[i] = BN_get_word(rem);
     }
-    pkey->exponent = BN_get_word(rsa->e);
+    pkey->exponent = BN_get_word(rsa_e);
 
 out:
     BN_free(n0inv);
