From e95fa133303026833c2606dc8270ed0d79bd58bf Mon Sep 17 00:00:00 2001
From: Erik de Castro Lopo <erikd@mega-nerd.com>
Date: Mon, 18 Apr 2016 11:43:54 +1000
Subject: [PATCH] yesod-auth: Fixes for persistent 2.5

---
 yesod-auth/Yesod/Auth.hs | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/yesod-auth/Yesod/Auth.hs b/yesod-auth/Yesod/Auth.hs
index 86de266..e020c18 100644
--- a/yesod-auth/Yesod/Auth.hs
+++ b/yesod-auth/Yesod/Auth.hs
@@ -500,10 +500,10 @@ class (YesodAuth master, YesodPersist master) => YesodAuthPersist master where
     getAuthEntity :: AuthId master -> HandlerT master IO (Maybe (AuthEntity master))
 
     default getAuthEntity
-        :: ( YesodPersistBackend master
-               ~ PersistEntityBackend (AuthEntity master)
-           , Key (AuthEntity master) ~ AuthId master
-           , PersistStore (YesodPersistBackend master)
+        :: ( YesodPersistBackend master ~ PersistEntityBackend (AuthEntity master)
+           , BaseBackend (PersistEntityBackend (AuthEntity master)) ~ PersistEntityBackend (AuthEntity master)
+           , PersistStoreRead (PersistEntityBackend (AuthEntity master))
+           , AuthId master ~ Key (AuthEntity master)
            , PersistEntity (AuthEntity master)
            )
         => AuthId master -> HandlerT master IO (Maybe (AuthEntity master))
From e2cc081ac108159f22cc9aa57395ea1960731c5e Mon Sep 17 00:00:00 2001
From: Erik de Castro Lopo <erikd@mega-nerd.com>
Date: Sat, 16 Apr 2016 09:05:47 +1000
Subject: [PATCH] yesod-auth.cabal: Bump persistent dependency

---
 yesod-auth/yesod-auth.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/yesod-auth/yesod-auth.cabal b/yesod-auth/yesod-auth.cabal
index a388396..e774ade 100644
--- a/yesod-auth/yesod-auth.cabal
+++ b/yesod-auth/yesod-auth.cabal
@@ -37,8 +37,8 @@ library
                    , unordered-containers
                    , yesod-form              >= 1.4       && < 1.5
                    , transformers            >= 0.2.2
-                   , persistent              >= 2.1       && < 2.3
-                   , persistent-template     >= 2.1       && < 2.2
+                   , persistent              >= 2.5       && < 2.6
+                   , persistent-template     >= 2.5       && < 2.6
                    , http-client
                    , http-conduit            >= 2.1
                    , aeson                   >= 0.7
