diff --git a/build.zig b/build.zig
index 1d33de7..f19c880 100644
--- a/build.zig
+++ b/build.zig
@@ -17,6 +17,7 @@ pub fn build(b: *std.build.Builder) !void {
     const exe_options = b.addOptions();
     exe.addOptions("build_options", exe_options);
 
+    const pie = b.option(bool, "pie", "Build a Position Independent Executable") orelse false;
     const enable_tracy = b.option(bool, "enable_tracy", "Whether tracy should be enabled.") orelse false;
     const coverage = b.option(bool, "generate_coverage", "Generate coverage data with kcov") orelse false;
     const coverage_output_dir = b.option([]const u8, "coverage_output_dir", "Output directory for coverage data") orelse b.pathJoin(&.{b.install_prefix, "kcov"});
@@ -33,6 +34,12 @@ pub fn build(b: *std.build.Builder) !void {
         b.option(std.log.Level, "log_level", "The Log Level to be used.") orelse .info,
     );
 
+    exe_options.addOption(
+        bool,
+        "pie",
+        pie,
+    );
+
     exe_options.addOption(
         bool,
         "enable_tracy",
@@ -115,6 +122,7 @@ pub fn build(b: *std.build.Builder) !void {
 
     exe.setTarget(target);
     exe.setBuildMode(mode);
+    exe.pie = pie;
     exe.install();
 
     const test_step = b.step("test", "Run all the tests");
