--- configure.ac.orig	2021-04-15 01:29:13.000000000 +0200
+++ configure.ac	2021-05-15 18:21:16.651840936 +0200
@@ -3,6 +3,7 @@
 
 AC_PREREQ([2.69])
 AC_INIT([powertop], [2.14], [powertop@lists.01.org], [], [https://01.org/powertop])
+AC_CANONICAL_TARGET
 AM_INIT_AUTOMAKE([
 	-Wall
 	1.12.2
@@ -37,6 +38,7 @@
 AC_PROG_INSTALL
 AM_PROG_CC_C_O
 PKG_PROG_PKG_CONFIG
+AX_PTHREAD
 
 # Checks for libraries.
 
