--- src/third_party/protobuf/3.7.0/examples/BUILD	2019-02-28 20:55:14.000000000 +0000
+++ src/third_party/protobuf/3.7.0/examples/BUILD	1980-01-01 00:00:00.000000000 +0000
@@ -10,6 +10,14 @@
 # of .proto source files to the protocol compiler.
 #
 # Remote repository "com_google_protobuf" must be defined to use this rule.
+licenses(["notice"])
+
+filegroup(
+    name = "srcs",
+    srcs = glob(["**"]),
+    visibility = ["//visibility:public"],
+)
+
 proto_library(
     name = "addressbook_proto",
     srcs = ["addressbook.proto"],
