From a120278ad6a28abcc632673b47ef3af3c2c64d70 Mon Sep 17 00:00:00 2001
From: Gasper Zejn <zelo.zejn@gmail.com>
Date: Fri, 12 Apr 2019 10:30:37 +0200
Subject: [PATCH] Fix incorrect use of pytest.raises(message=...)

diff --git a/tests/test_jwt.py b/tests/test_jwt.py
index 9ceb239..75ed3fc 100644
--- a/tests/test_jwt.py
+++ b/tests/test_jwt.py
@@ -62,7 +62,7 @@ def return_invalid_json(token, key, algorithms, verify=True):
 
             jws.verify = return_invalid_json
 
-            with pytest.raises(JWTError, message='Invalid payload string: ["a", "b"}'):
+            with pytest.raises(JWTError, match='Invalid payload string: '):
                 jwt.decode(token, 'secret', ['HS256'])
         finally:
             jws.verify = old_jws_verify
@@ -77,7 +77,7 @@ def return_encoded_array(token, key, algorithms, verify=True):
 
             jws.verify = return_encoded_array
 
-            with pytest.raises(JWTError, message='Invalid payload string: must be a json object'):
+            with pytest.raises(JWTError, match='Invalid payload string: must be a json object'):
                 jwt.decode(token, 'secret', ['HS256'])
         finally:
             jws.verify = old_jws_verify
