diff --git a/tests/conftest.py b/tests/conftest.py
index bddf44a..daae169 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -54,7 +54,7 @@ async def get_version():
 
   return __call__
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="session")
 def event_loop(request):
   """Override pytest-asyncio's event_loop fixture,
      Don't create an instance of the default event loop for each test case.
