diff --git c/pyproject.toml w/pyproject.toml
index c95e6d1..a53c5d8 100644
--- c/pyproject.toml
+++ w/pyproject.toml
@@ -1,9 +1,6 @@
 [build-system]
 requires = [
-  "pip>=19.1.1",
-  "setuptools >= 41.0.0",
-  "setuptools_scm >= 1.15.0",
-  "setuptools_scm_git_archive >= 1.0",
-  "wheel",
+  "setuptools >= 45.0.0",
+  "setuptools_scm[toml] >= 7.0.0",
 ]
 build-backend = "setuptools.build_meta"
diff --git c/setup.cfg w/setup.cfg
index adbc4b2..fee0bce 100644
--- c/setup.cfg
+++ w/setup.cfg
@@ -60,11 +60,9 @@ include_package_data = True
 zip_safe = True
 install_requires =
     distro>=1.3.0
-    setuptools>=39.0
 
 # These are required during `setup.py` run:
 setup_requires =
-    setuptools_scm >= 1.15.0
-    setuptools_scm_git_archive >= 1.0
+    setuptools_scm >= 7.0.0
 [options.packages.find]
 where = .
