--- a/fix_python_shebang.sh
+++ b/fix_python_shebang.sh
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-sed -e "s@^#\!/usr/bin/python\$@#\!/usr/bin/python -Es@" -i $@
+sed -e "s@^#\!/usr/bin/python2\$@#\!/usr/bin/python2 -Es@" -i $@
--- a/src/firewall-applet
+++ b/src/firewall-applet
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2010-2012 Red Hat, Inc.
--- a/src/firewall-cmd
+++ b/src/firewall-cmd
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2009-2012 Red Hat, Inc.
--- a/src/firewall-config
+++ b/src/firewall-config
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2011-2012 Red Hat, Inc.
--- a/src/firewall-offline-cmd
+++ b/src/firewall-offline-cmd
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2009-2012 Red Hat, Inc.
--- a/src/firewalld
+++ b/src/firewalld
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2010-2012 Red Hat, Inc.
--- a/src/gtk3_chooserbutton.py
+++ b/src/gtk3_chooserbutton.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python -Es
+#!/usr/bin/python2 -Es
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2008,2012 Red Hat, Inc.
--- a/src/tests/firewalld_config.py
+++ b/src/tests/firewalld_config.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2010-2012 Red Hat, Inc.
--- a/src/tests/firewalld_direct.py
+++ b/src/tests/firewalld_direct.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2012 Red Hat, Inc.
--- a/src/tests/firewalld_test.py
+++ b/src/tests/firewalld_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2010-2012 Red Hat, Inc.
