diff -upr playitslowly-1.5.1.orig/setup.py playitslowly-1.5.1/setup.py
--- playitslowly-1.5.1.orig/setup.py	2016-05-21 23:57:51.000000000 +0300
+++ playitslowly-1.5.1/setup.py	2021-12-11 08:18:39.611049106 +0200
@@ -22,7 +22,7 @@ class new_install(install):
         # write install-info
         basename = "%s-py%s.install-info" % (
             to_filename(safe_name(self.distribution.get_name())),
-            sys.version[:3]
+            ".".join(map(str, sys.version_info[:2]))
         )
         install_info = os.path.join(self.install_libbase, basename)
         outputs = self.get_outputs()
