diff -upr harparser-0.4.orig/harparser.py harparser-0.4/harparser.py
--- harparser-0.4.orig/harparser.py	2018-03-15 17:14:35.000000000 +0200
+++ harparser-0.4/harparser.py	2021-12-03 16:11:54.218798617 +0200
@@ -1,7 +1,7 @@
 """
     This module implements HAR as per specification version 1.2.
 """
-from collections import MutableMapping
+from collections.abc import MutableMapping
 from json import dumps, loads
 from zlib import compress, decompress
 
