diff -ruN a/include/fastnumbers/numbers.h b/include/fastnumbers/numbers.h
--- a/include/fastnumbers/numbers.h	2020-01-07 05:01:44.000000000 +0100
+++ b/include/fastnumbers/numbers.h	2020-11-19 20:59:35.905868338 +0100
@@ -6,6 +6,9 @@
  */
 
 #include <Python.h>
+#define Py_BUILD_CORE
+#include <internal/pycore_dtoa.h>
+#undef Py_BUILD_CORE
 #include <limits.h>
 #include "fastnumbers/fn_bool.h"
 #include "fastnumbers/options.h"
