--- pdfslicer/src/backend/pdfsaver.cpp	2022-09-13 16:49:50.000000000 +0200
+++ pdfslicer/src/backend/pdfsaver.cpp.new	2022-09-13 16:51:52.299074613 +0200
@@ -1,5 +1,8 @@
 #include "pdfsaver.hpp"
 #include "tempfile.hpp"
+
+#define POINTERHOLDER_TRANSITION 0
+
 #include <qpdf/QPDFWriter.hh>
 #include <range/v3/view/iota.hpp>
 #include <range/v3/view/set_algorithm.hpp>
--- pdfslicer/src/backend/pdfsaver.hpp	2022-09-13 16:49:50.000000000 +0200
+++ pdfslicer/src/backend/pdfsaver.hpp.new	2022-09-13 16:55:11.194063160 +0200
@@ -3,6 +3,9 @@
 
 #include <vector>
 #include <giomm/file.h>
+
+#define POINTERHOLDER_TRANSITION 0
+
 #include <qpdf/QPDF.hh>
 #include <qpdf/QPDFPageDocumentHelper.hh>
 
