From b6c9fddd0dbd0554d2c6ec096f42d0557101178a Mon Sep 17 00:00:00 2001
From: Sergey Ilinykh <rion4ik@gmail.com>
Date: Fri, 29 May 2020 14:50:56 +0300
Subject: [PATCH] Added missing QPainterPath header

---
 src/avatars.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/avatars.cpp b/src/avatars.cpp
index bc60ad14..b24f0799 100644
--- a/src/avatars.cpp
+++ b/src/avatars.cpp
@@ -50,6 +50,7 @@
 #include <QFile>
 #include <QBuffer>
 #include <QPainter>
+#include <QPainterPath>
 
 #include "xmpp_xmlcommon.h"
 #include "xmpp_vcard.h"
From 1b0421f01b9c1cd5f27e0888469a53be34e92a40 Mon Sep 17 00:00:00 2001
From: Sergey Ilinykh <rion4ik@gmail.com>
Date: Fri, 29 May 2020 15:07:35 +0300
Subject: [PATCH] Add include to QPainterPath wherever it's not added

---
 src/contactlistdragview.cpp       | 1 +
 src/multifiletransferdelegate.cpp | 1 +
 src/rosteravatarframe.cpp         | 2 ++
 src/whiteboarding/wbnewpath.cpp   | 1 +
 4 files changed, 5 insertions(+)

diff --git a/src/contactlistdragview.cpp b/src/contactlistdragview.cpp
index a3885a99..091984fe 100644
--- a/src/contactlistdragview.cpp
+++ b/src/contactlistdragview.cpp
@@ -43,6 +43,7 @@
 #include <QMimeData>
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QScrollBar>
 
 ContactListDragView::ContactListDragView(QWidget *parent) :
diff --git a/src/rosteravatarframe.cpp b/src/rosteravatarframe.cpp
index e3945aff..5a15db24 100644
--- a/src/rosteravatarframe.cpp
+++ b/src/rosteravatarframe.cpp
@@ -23,6 +23,8 @@
 #include "iconset.h"
 #include "qpainter.h"
 
+#include <QPainterPath>
+
 RosterAvatarFrame::RosterAvatarFrame(QWidget *parent)
         : QFrame(parent)
         , statusMessage_("")
diff --git a/src/whiteboarding/wbnewpath.cpp b/src/whiteboarding/wbnewpath.cpp
index fbd20c17..cbb4fbf4 100644
--- a/src/whiteboarding/wbnewpath.cpp
+++ b/src/whiteboarding/wbnewpath.cpp
@@ -23,6 +23,7 @@
 #include "../sxe/sxesession.h"
 
 #include <QGraphicsScene>
+#include <QPainterPath>
 
 WbNewPath::WbNewPath(QGraphicsScene *s, QPointF startPos, int strokeWidth, const QColor &strokeColor,
                      const QColor &fillColor) :
