diff --git a/src/gui/process_table_view.cpp b/src/gui/process_table_view.cpp
index 3a18329..ae0f4b5 100644
--- a/src/gui/process_table_view.cpp
+++ b/src/gui/process_table_view.cpp
@@ -20,6 +20,7 @@
 #include <QMessageBox>
 #include <QProcess>
 #include <QTimer>
+#include <QKeyEvent>
 
 #include "kill_process_confirm_dialog.h"
 #include "main_window.h"
diff --git a/src/gui/system_service_table_view.cpp b/src/gui/system_service_table_view.cpp
index 15d18b4..d1b1f8c 100644
--- a/src/gui/system_service_table_view.cpp
+++ b/src/gui/system_service_table_view.cpp
@@ -10,6 +10,7 @@
 #include <QFutureWatcher>
 #include <QScrollBar>
 #include <QtConcurrent>
+#include <QKeyEvent>
 
 #include "common/error_context.h"
 #include "main_window.h"
