commit bab1bd78ff47da389e7f35153eb6ba93e4f5e685
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Mon Jan 21 01:15:45 2019 +0800

    Fix compatibility with QuickCheck 2.12

diff --git a/chell-quickcheck/Test/Chell/QuickCheck.hs b/chell-quickcheck/Test/Chell/QuickCheck.hs
index 0ec25b0..aaf35fb 100644
--- a/chell-quickcheck/Test/Chell/QuickCheck.hs
+++ b/chell-quickcheck/Test/Chell/QuickCheck.hs
@@ -53,8 +53,16 @@ property name prop = Chell.test name $ \opts -> do
 		, State.computeSize = computeSize (QuickCheck.maxSize args) (QuickCheck.maxSuccess args)
 		, State.numSuccessTests = 0
 		, State.numDiscardedTests = 0
+#if MIN_VERSION_QuickCheck(2,12,0)
+		, State.classes = mempty
+		, State.tables = mempty
+		, State.requiredCoverage = mempty
+		, State.expected = True
+		, State.coverageConfidence = Nothing
+#else
 		, State.collected = []
 		, State.expectedFailure = False
+#endif
 
 #if MIN_VERSION_QuickCheck(2,7,0)
 		, State.randomSeed = QCRandom.mkQCGen seed
@@ -77,12 +85,16 @@ property name prop = Chell.test name $ \opts -> do
 #endif
 		}
 	
+#if MIN_VERSION_QuickCheck(2,12,0)
+	result <- Test.test state (QuickCheck.property prop)
+#else
 #if MIN_VERSION_QuickCheck(2,7,0)
 	let genProp = unProperty (QuickCheck.property prop)
 #else
 	let genProp = QuickCheck.property prop
 #endif
 	result <- Test.test state (Gen.unGen genProp)
+#endif
 	let output = Test.output result
 	let notes = [("seed", show seed)]
 	let failure = Chell.failure { Chell.failureMessage = output }
diff --git a/chell-quickcheck/chell-quickcheck.cabal b/chell-quickcheck/chell-quickcheck.cabal
index 5b5482c..673363e 100644
--- a/chell-quickcheck/chell-quickcheck.cabal
+++ b/chell-quickcheck/chell-quickcheck.cabal
@@ -27,7 +27,7 @@ library
   build-depends:
       base >= 4.0 && < 5.0
     , chell >= 0.3 && < 0.5
-    , QuickCheck >= 2.3 && < 2.11
+    , QuickCheck >= 2.3 && < 2.13
     , random
 
   exposed-modules:
