diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 74ba18b38..bebc39c6c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -582,9 +582,14 @@ ENDIF ()
 IF (ENABLE_EXCEL)
         # build before use
         IF (BUILD_EXCEL)
+                target_include_directories( labplot2lib PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/QXlsx/header" )
                 add_dependencies( labplot2lib QXlsx )
         ENDIF ()
 
+        IF (QXLSX_FOUND)
+                target_include_directories( labplot2lib PRIVATE ${QXLSX_INCLUDE_DIR} )
+        ENDIF ()
+
 	IF (QXLSX_FOUND OR BUILD_EXCEL)
 		target_link_libraries( labplot2lib QXlsx )
         ENDIF ()
diff --git a/src/backend/datasources/filters/ExcelFilter.h b/src/backend/datasources/filters/ExcelFilter.h
index 8d93a0733..ac26a3854 100644
--- a/src/backend/datasources/filters/ExcelFilter.h
+++ b/src/backend/datasources/filters/ExcelFilter.h
@@ -16,9 +16,9 @@
 #include <QObject>
 
 #ifdef HAVE_EXCEL
-#include "3rdparty/QXlsx/header/xlsxcellrange.h"
-#include "3rdparty/QXlsx/header/xlsxcellreference.h"
-#include "3rdparty/QXlsx/header/xlsxdocument.h"
+#include "xlsxcellrange.h"
+#include "xlsxcellreference.h"
+#include "xlsxdocument.h"
 #endif
 
 class ExcelFilterPrivate;
diff --git a/src/kdefrontend/datasources/ExcelOptionsWidget.cpp b/src/kdefrontend/datasources/ExcelOptionsWidget.cpp
index c6542e456..2afac3a8d 100644
--- a/src/kdefrontend/datasources/ExcelOptionsWidget.cpp
+++ b/src/kdefrontend/datasources/ExcelOptionsWidget.cpp
@@ -14,7 +14,7 @@
 #include "src/kdefrontend/datasources/ImportFileWidget.h"
 
 #ifdef HAVE_EXCEL
-#include "3rdparty/QXlsx/header/xlsxcellrange.h"
+#include "xlsxcellrange.h"
 #endif
 
 #include <QAbstractItemModel>
