diff --git a/plugins/navigation/navitemwidget.h b/plugins/navigation/navitemwidget.h
index 571f1153..462373a2 100644
--- a/plugins/navigation/navitemwidget.h
+++ b/plugins/navigation/navitemwidget.h
@@ -39,9 +39,6 @@ signals:
     void itemClicked(const QString &id) const;
     void itemEntered(const QString &id) const;
 
-public slots:
-    void recheckHoverState();
-
 protected:
     void paintEvent(QPaintEvent *e);
     void mouseReleaseEvent(QMouseEvent *e);
