diff -ruN vdirsyncer-0.16.3-a/tests/storage/__init__.py vdirsyncer-0.16.3-b/tests/storage/__init__.py
--- vdirsyncer-0.16.3-a/tests/storage/__init__.py	2017-10-03 10:50:44.000000000 +0200
+++ vdirsyncer-0.16.3-b/tests/storage/__init__.py	2018-02-04 20:48:29.051827826 +0100
@@ -297,19 +297,6 @@
             assert rv == x
             assert isinstance(rv, str)
 
-    @given(value=st.one_of(
-        st.none(),
-        printable_characters_strategy.filter(lambda x: x.strip() != x)
-    ))
-    def test_metadata_normalization(self, requires_metadata, s, value):
-        x = s.get_meta('displayname')
-        assert x == normalize_meta_value(x)
-
-        if not getattr(self, 'dav_server', None):
-            # ownCloud replaces "" with "unnamed"
-            s.set_meta('displayname', value)
-            assert s.get_meta('displayname') == normalize_meta_value(value)
-
     def test_recurring_events(self, s, item_type):
         if item_type != 'VEVENT':
             pytest.skip('This storage instance doesn\'t support iCalendar.')
