--- a/pyproject.toml
+++ b/pyproject.toml
@@ -40,7 +40,7 @@ keywords = ["sphinx", "shell", "prompt"]
 packages = [{ include = "sphinx_prompt" }, { include = "sphinx_prompt/py.typed" }]
 
 [tool.poetry.dependencies]
-python = ">=3.8,<3.11"
+python = ">=3.8"
 Sphinx = "6.1.3"
 pygments = "2.14.0"
 docutils = "0.19"
@@ -51,14 +51,5 @@ pytest = "7.2.1"
 types-docutils = "0.19.1.2"
 
 [build-system]
-requires = ["poetry-core>=1.0.0", "poetry-plugin-tweak-dependencies-version", "poetry-dynamic-versioning"]
+requires = ["poetry-core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
-
-[tool.poetry-plugin-tweak-dependencies-version]
-default = "present"
-sphinx = "major"
-
-[tool.poetry-dynamic-versioning]
-enable = true
-vcs = "git"
-pattern = "^((?P<epoch>\\d+)!)?(?P<base>\\d+(\\.\\d+)*)"
