diff --git a/Makefile b/Makefile
index f7add11..840a160 100644
--- a/Makefile
+++ b/Makefile
@@ -67,7 +67,8 @@ GO_LINK_FLAGS := -s -w
 TAG_VERSION := $(shell git describe --tags)
 VERSION_FLAG := -X "main.version=$(if $(TAG_VERSION),$(TAG_VERSION),$(VERSION))"
 # Flag to embed the date and time of the build into the binary.
-DATE_FLAG := -X "main.buildTime=$(shell date)"
+BUILDDATE ?= $(shell date)
+DATE_FLAG := -X "main.buildTime='"$(BUILDDATE)"'"
 
 override GO_LINK_FLAGS += $(VERSION_FLAG) $(DATE_FLAG) -extldflags "$(LDFLAGS)"
 override GO_FLAGS += --ldflags '$(GO_LINK_FLAGS)'
