diff -ruN a/source/Makefile b/source/Makefile
--- a/source/Makefile	2022-06-28 12:39:06.000000000 +0200
+++ b/source/Makefile	2022-06-29 13:03:28.693568568 +0200
@@ -21,19 +21,20 @@
 
 CPPFLAGS += -I. -fPIC -D_REENTRANT
 CXXFLAGS += -Wall -O2
+LDFLAGS += -shared
 
 all:	zita-reverbs.so
 
 REVERBS_O = revplugins.o reverbs.o zreverb.o pareq.o
 zita-reverbs.so:	$(REVERBS_O)	
-	$(CXX) -shared  -o $@  $(REVERBS_O)
+	$(CXX) $(LDFLAGS) -o $@  $(REVERBS_O)
 
 reverbs.o:	ladspaplugin.h reverbs.h zreverb.h pareq.h
 revplugins.o:	ladspaplugin.h reverbs.h zreverb.h pareq.h
 
 
 install:	zita-reverbs.so
-	cp  *.so /usr/lib/ladspa
+	install -vDm 755 *.so -t $(DESTDIR)/usr/lib/ladspa
 
 clean:
 	rm -rf *~ *.o *.so
