diff --git a/llvm/include/llvm/OffloadArch/OffloadArch.h b/llvm/include/llvm/OffloadArch/OffloadArch.h
index a3accff9f448..85831fb108df 100644
--- a/llvm/include/llvm/OffloadArch/OffloadArch.h
+++ b/llvm/include/llvm/OffloadArch/OffloadArch.h
@@ -11,6 +11,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 #define MAXPATHSIZE 512
 
diff --git a/openmp/libomptarget/include/Debug.h b/openmp/libomptarget/include/Debug.h
index 4e7da8b7d53f..2ce303946a2a 100644
--- a/openmp/libomptarget/include/Debug.h
+++ b/openmp/libomptarget/include/Debug.h
@@ -39,6 +39,9 @@
 
 #include <atomic>
 #include <mutex>
+#include <string>
+#include <cstdint>
+#include <cstdlib>
 
 /// 32-Bit field data attributes controlling information presented to the user.
 enum OpenMPInfoType : uint32_t {
