diff --git a/slim.gemspec b/slim.gemspec
index 0bbc7ae..2edffb1 100644
--- a/slim.gemspec
+++ b/slim.gemspec
@@ -19,6 +19,6 @@ Gem::Specification.new do |s|
 
   s.required_ruby_version = '>=2.0.0'
 
-  s.add_runtime_dependency('temple', ['>= 0.7.6', '< 0.9'])
-  s.add_runtime_dependency('tilt', ['>= 2.0.6', '< 2.1'])
+  s.add_runtime_dependency('temple', '>= 0.7.6')
+  s.add_runtime_dependency('tilt', '>= 2.0.6')
 end
