diff --git a/src/sage_setup/docbuild/ext/sage_autodoc.py b/src/sage_setup/docbuild/ext/sage_autodoc.py
index 2c399bad51..96abec3382 100644
--- a/src/sage_setup/docbuild/ext/sage_autodoc.py
+++ b/src/sage_setup/docbuild/ext/sage_autodoc.py
@@ -39,8 +39,9 @@ from docutils.statemachine import ViewList
 import sphinx
 from sphinx.util import rpartition, force_decode
 from sphinx.locale import _
-from sphinx.pycode import ModuleAnalyzer, PycodeError
-from sphinx.application import ExtensionError
+from sphinx.pycode import ModuleAnalyzer
+from sphinx.errors import ExtensionError, PycodeError
+from sphinx.util.logging import getLogger
 from sphinx.util.nodes import nested_parse_with_titles
 from docutils.parsers.rst import Directive
 from sphinx.util.inspect import getargspec, isdescriptor, safe_getmembers, \
