--- ./sbuild/sbuild-util.cc.orig	2018-05-05 12:39:09.706894741 -0400
+++ ./sbuild/sbuild-util.cc	2018-05-05 12:41:46.963886076 -0400
@@ -630,7 +630,7 @@
   ::passwd *pwd_result;
 
   while ((error = getpwuid_r(uid, this,
-                             &buffer[0], buffer.capacity(),
+                             buffer.data(), buffer.capacity(),
                              &pwd_result)) == ERANGE)
     {
       size <<= 1;
@@ -653,7 +653,7 @@
   ::passwd *pwd_result;
 
   while ((error = getpwnam_r(name, this,
-                             &buffer[0], buffer.capacity(),
+                             buffer.data(), buffer.capacity(),
                              &pwd_result)) == ERANGE)
     {
       size <<= 1;
@@ -739,7 +739,7 @@
   ::group *grp_result;
 
   while ((error = getgrgid_r(gid, this,
-                             &buffer[0], buffer.capacity(),
+                             buffer.data(), buffer.capacity(),
                              &grp_result)) == ERANGE)
     {
       size <<= 1;
@@ -762,7 +762,7 @@
   ::group *grp_result;
 
   while ((error = getgrnam_r(name, this,
-                             &buffer[0], buffer.capacity(),
+                             buffer.data(), buffer.capacity(),
                              &grp_result)) == ERANGE)
     {
       size <<= 1;
