--- libffado-2.4.1/SConstruct.orig	2019-07-26 22:54:47.409490023 +0000
+++ libffado-2.4.1/SConstruct	2019-07-26 22:55:33.906830766 +0000
@@ -719,8 +719,8 @@
         # /bin/mount:     file format elf64-x86-64
         # or like this:
         # /bin/mount:     file format elf32-powerpc
-        for line in x.split(b'\n'):
-            line = line.strip().decode()
+        for line in x.split('\n'):
+            line = line.strip()
             if line.startswith(real_exe):
                 x, fmt = line.rsplit(None, 1)
                 answer = 'elf32' in fmt
--- libffado-2.4.1/support/mixer-qt4/SConscript.orig	2019-07-26 23:02:25.349419059 +0000
+++ libffado-2.4.1/support/mixer-qt4/SConscript	2019-07-26 23:02:43.363010354 +0000
@@ -38,7 +38,7 @@
                 arg.append( os.path.join( dirname, name ) )
 
     pythonfiles = [ 'ffado/config.py' ]
-    os.path.walk( "ffado", findfiles, pythonfiles )
+    os.walk( "ffado", findfiles, pythonfiles )
 
     e.ScanReplace( "ffado/config.py.in" )
     e.Depends( "ffado/config.py", "#/SConstruct" )
