--- pingus-0.7.6/SConscript.orig	2019-07-27 08:28:23.143619494 +0000
+++ pingus-0.7.6/SConscript	2019-07-27 08:29:35.844531921 +0000
@@ -126,12 +126,12 @@
     def configure_end(self):
         self.env = self.conf.Finish()
 
-        print "Reports:"
-        print self.reports
+        print ("Reports:")
+        print (self.reports)
 
         if not self.fatal_error == "":
-            print "Fatal Errors:"
-            print self.fatal_error
+            print ("Fatal Errors:")
+            print (self.fatal_error)
             Exit(1)
 
     def configure_gxx(self): 
