Subject: [PATCH] configure-install-path
---
Index: bin/kafka-run-class.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bin/kafka-run-class.sh b/bin/kafka-run-class.sh
--- a/bin/kafka-run-class.sh	(revision b66af662e61082cb8def576ded1fe5cee37e155f)
+++ b/bin/kafka-run-class.sh	(date 1675290857583)
@@ -45,7 +45,7 @@
   fi
 }
 
-base_dir=$(dirname $0)/..
+base_dir=/usr/share/kafka
 
 if [ -z "$SCALA_VERSION" ]; then
   SCALA_VERSION=2.13.6
