--- a/tests/tools.bats
+++ b/tests/tools.bats
@@ -128,22 +128,3 @@ Binary" ]
   exp=$(cat Icon/firefox.16.0.2.0.ico.exe | sha1sum)
   [ "$output" = "$exp" ]
 }
-
-@test "msibuild - update _SummaryInformation table" {
-  run "$msibuild" -i out.msi _SummaryInformation.idt
-  run "$msiinfo" suminfo out.msi
-  [ "$output" = "Title: Installation Database
-Subject: Acme's Foobar 1.0 Installer
-Author: Acme Ltd.
-Keywords: Installer
-Comments: Foobar is a registered trademark of Acme Ltd.
-Template: Intel;1033
-Revision number (UUID): {D045A303-F114-4A3B-A01D-24FC2D2A67D7}
-Created: Tue Dec 18 15:12:50 2012
-Last saved: Tue Dec 18 15:12:50 2012
-Version: 100 (64)
-Source: 2 (2)
-Restrict: 0 (0)
-Application: Windows Installer XML (3.7.1119.0)
-Security: 2 (2)" ]
-}
