diff -wbBur phpldapadmin-1.2.3.org/lib/TemplateRender.php phpldapadmin-1.2.3/lib/TemplateRender.php
--- phpldapadmin-1.2.3.org/lib/TemplateRender.php	2012-10-01 10:54:14.000000000 +0400
+++ phpldapadmin-1.2.3/lib/TemplateRender.php	2018-02-02 19:06:45.674760057 +0300
@@ -321,6 +321,8 @@
 
 				$vals = array();
 
+				asort($picklistvalues);
+
 				foreach ($picklistvalues as $key => $values) {
 					$display = $args[3];
 
