--- sqlite-src-3370000/configure.ac.orig	2021-11-27 16:05:17.000000000 +0100
+++ sqlite-src-3370000/configure.ac	2021-12-03 21:21:47.506868981 +0100
@@ -116,6 +116,13 @@
 USE_AMALGAMATION=1
 
 #########
+# By default, amalgamation sqlite3.c will have #line directives.
+# This is a build option not shown by ./configure --help
+# To control it, use configure option: amalgamation_line_macros=?
+# where ? is no to suppress #line directives or yes to create them.
+AMALGAMATION_LINE_MACROS=--linemacros=0
+
+#########
 # See whether we can run specific tclsh versions known to work well;
 # if not, then we fall back to plain tclsh.
 # TODO: try other versions before falling back?
@@ -574,6 +581,20 @@
 fi
 AC_SUBST(USE_AMALGAMATION)
 
+########
+# See whether --disable
+if test "${amalgamation_line_macros+set}" = set; then :
+  enableval=$amalgamation_line_macros;
+fi
+
+if test "${amalgamation_line_macros}" = "yes" ; then
+  AMALGAMATION_LINE_MACROS=--linemacros=1
+fi
+if test "${amalgamation_line_macros}" = "no" ; then
+  AMALGAMATION_LINE_MACROS=--linemacros=0
+fi
+AC_SUBST(AMALGAMATION_LINE_MACROS)
+
 #########
 # Look for zlib.  Only needed by extensions and by the sqlite3.exe shell
 AC_CHECK_HEADERS(zlib.h)
