diff -aur aaa/pyproject.toml bbb/pyproject.toml
--- aaa/pyproject.toml	2022-12-14 21:51:47.000000000 +0100
+++ bbb/pyproject.toml	2022-12-24 14:19:41.960029538 +0100
@@ -248,3 +248,9 @@
 [tool.setuptools_scm]
 local_scheme = "no-local-version"
 write_to = "src/ansiblelint/_version.py"
+
+[tool.setuptools.packages.find]
+where = ["src"]
+
+[tool.setuptools.package-data]
+"*" = ["py.typed", "**/*.json", "**/*.yml", "**/*.yaml", "**/*.md"]
