diff --git a/gradle/globals.gradle b/gradle/globals.gradle
--- a/gradle/globals.gradle	(revision a4eb7aa123dc53f8dac74d80b66a490f2d6b4a26)
+++ b/gradle/globals.gradle	(date 1668129963200)
@@ -152,4 +152,8 @@
       return taskList
     }
   }
+  tasks.withType(AbstractArchiveTask).configureEach {
+    preserveFileTimestamps = false
+    reproducibleFileOrder = true
+  }
 }
diff --git a/gradle/java/jar-manifest.gradle b/gradle/java/jar-manifest.gradle
--- a/gradle/java/jar-manifest.gradle	(revision a4eb7aa123dc53f8dac74d80b66a490f2d6b4a26)
+++ b/gradle/java/jar-manifest.gradle	(date 1668116039242)
@@ -44,7 +44,7 @@
                     if (snapshotBuild) {
                       return "${project.version} ${gitRev} [snapshot build, details omitted]"
                     } else {
-                      return "${project.version} ${gitRev} - ${System.properties['user.name']} - ${buildDate} ${buildTime}"
+                      return "${project.version} ${gitRev} - ${System.properties['user.name']}"
                     }
                 }
 
