diff --git a/raven/conf/defaults.py b/raven/conf/defaults.py
index 3505a098..0568fe0f 100644
--- a/raven/conf/defaults.py
+++ b/raven/conf/defaults.py
@@ -42,9 +42,4 @@ PROCESSORS = (
     'raven.processors.SanitizePasswordsProcessor',
 )
 
-try:
-    # Try for certifi first since they likely keep their bundle more up to date
-    import certifi
-    CA_BUNDLE = certifi.where()
-except ImportError:
-    CA_BUNDLE = os.path.join(ROOT, 'data', 'cacert.pem')
+CA_BUNDLE = "/etc/ssl/certs/ca-certificates.crt"
