diff --git a/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp b/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp
index 5d0e50926..7d89692f8 100644
--- a/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp
+++ b/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp
@@ -355,6 +355,12 @@ QImage CreateFrameStorage(QSize size) {
 }
 
 void UnPremultiply(QImage &to, const QImage &from) {
+	to = from.convertToFormat(QImage::Format_ARGB32);
+        return;
+        //// below is original tdesktop code, we shortcut them by using Qt public API
+        //// see https://github.com/telegramdesktop/tdesktop/issues/6219
+        /*
+
 	// This creates QImage::Format_ARGB32_Premultiplied, but we use it
 	// as an image in QImage::Format_ARGB32 format.
 	if (!GoodStorageForFrame(to, from.size())) {
@@ -387,9 +393,16 @@ void UnPremultiply(QImage &to, const QImage &from) {
 			layout,
 			nullptr);
 	}
+	*/
 }
 
 void PremultiplyInplace(QImage &image) {
+	image = image.convertToFormat(QImage::Format_ARGB32_Premultiplied);
+	return;
+        //// below is original tdesktop code, we shortcut them by using Qt public API
+        //// see https://github.com/telegramdesktop/tdesktop/issues/6219
+        /*
+
 	const auto layout = &qPixelLayouts[QImage::Format_ARGB32];
 	const auto convert = layout->convertToARGB32PM;
 	const auto perLine = image.bytesPerLine();
@@ -413,6 +426,7 @@ void PremultiplyInplace(QImage &image) {
 			layout,
 			nullptr);
 	}
+	*/
 }
 
 } // namespace FFmpeg
