--- a/daemon/core/version.go
+++ b/daemon/core/version.go
@@ -3,7 +3,7 @@ package core
 // version related consts
 const (
 	Name    = "opensnitch-daemon"
-	Version = "1.5.3"
+	Version = "@VERSION@"
 	Author  = "Simone 'evilsocket' Margaritelli"
 	Website = "https://github.com/evilsocket/opensnitch"
 )
--- a/ui/opensnitch/version.py
+++ b/ui/opensnitch/version.py
@@ -1 +1 @@
-version = '1.5.3'
+version = '@VERSION@'
