diff -ruN a/scripts/utils.mk b/scripts/utils.mk
--- a/scripts/utils.mk	2021-07-27 00:58:04.000000000 +0200
+++ b/scripts/utils.mk	2021-07-29 19:34:44.741532160 +0200
@@ -72,7 +72,7 @@
 
 do_compile_shared_library =			\
 	($(print_shared_lib_compile)		\
-	$(CC) --shared $^ '-Wl,-soname,$(@F),-rpath=$$ORIGIN' -o $@ $(LIBS))
+	$(CC) --shared $^ '-Wl,-soname,$(@F),-rpath=$$ORIGIN' -o $@ $(LDFLAGS) $(LIBS))
 
 do_compile_plugin_obj =				\
 	($(print_plugin_obj_compile)		\
