diff -ENwbur ucblogo-6.0-no-wxWidgets/docs/makefile ucblogo-6.0-no-texinfo/docs/makefile
--- src/ucblogo-6.0/docs/makefile	2013-03-19 07:18:52.000000000 +0530
+++ src/ucblogo-6.0-b/docs/makefile	2013-03-21 06:21:03.000000000 +0530
@@ -30,7 +30,7 @@
 ship:
 	-rm -f *.{aux,cp,cps,dvi,fn,ky,log,pg,toc,tp,vr}
 
-install: all
+install:
 	for d in $(INFODIR) $(DOCSDIR) $(HTMLDIR); do [ -d $$d ] || mkdir -p $$d || exit 1; done
 	-cp -f *.info* $(INFODIR)/.
 	-cp -f html/*.html $(HTMLDIR)/.
