diff --git a/tests/botocore/test_credentials.py b/tests/botocore/test_credentials.py
index ecbe17e..1b2172e 100644
--- a/tests/botocore/test_credentials.py
+++ b/tests/botocore/test_credentials.py
@@ -8,7 +8,7 @@ import datetime
 import json
 import subprocess
 
-import mock
+from unittest import mock
 from typing import Optional
 
 import pytest
diff --git a/tests/botocore/test_signers.py b/tests/botocore/test_signers.py
index 0e4035e..7529760 100644
--- a/tests/botocore/test_signers.py
+++ b/tests/botocore/test_signers.py
@@ -1,5 +1,5 @@
 import pytest
-import mock
+from unittest import mock
 
 import aiobotocore
 import aiobotocore.credentials
diff --git a/tests/botocore/test_utils.py b/tests/botocore/test_utils.py
index 3d17b87..4089cf1 100644
--- a/tests/botocore/test_utils.py
+++ b/tests/botocore/test_utils.py
@@ -1,7 +1,7 @@
 import asyncio
 import pytest
 import json
-import mock
+from unittest import mock
 import itertools
 from typing import Union, List, Tuple
 
