diff --git a/tests/test_vcs.py b/tests/test_vcs.py
index 4e8061e..c9cf71e 100644
--- a/tests/test_vcs.py
+++ b/tests/test_vcs.py
@@ -13,12 +13,6 @@ async def test_git(get_version):
     os.path.exists("example") or os.mkdir("example")
     assert await get_version("example", {"vcs": "git+https://github.com/harry-sanabria/ReleaseTestRepo.git"}) == "1.1.2b3cdf6134b07ae6ac77f11b586dc1ae6d1521db"
 
-@pytest.mark.skipif(shutil.which("hg") is None,
-                    reason="requires hg command")
-async def test_mercurial(get_version):
-    os.path.exists("example") or os.mkdir("example")
-    assert await get_version("example", {"vcs": "hg+https://bitbucket.org/pil0t/testrepo"}) == "1.1.84679e29c7d9"
-
 @pytest.mark.skipif(shutil.which("git") is None,
                     reason="requires git command")
 async def test_git_max_tag(get_version):
