--- a/setup.py	2015-06-03 08:44:39.000000000 +0800
+++ b/setup.py	2015-10-26 11:52:15.798195748 +0800
@@ -41,7 +41,6 @@
     packages=["websocket", "websocket.tests"],
     package_data={
         'websocket.tests': ['data/*.txt'],
-        'websocket': ["cacert.pem"]
     },
     tests_require=tests_require,
     test_suite="websocket.tests.test_websocket",
--- a/websocket/_http.py	2015-06-03 08:44:39.000000000 +0800
+++ b/websocket/_http.py	2015-10-26 11:52:50.398576537 +0800
@@ -143,8 +143,7 @@
 
 def _ssl_socket(sock, user_sslopt, hostname):
     sslopt = dict(cert_reqs=ssl.CERT_REQUIRED)
-    certPath = os.path.join(
-        os.path.dirname(__file__), "cacert.pem")
+    certPath = "/etc/ssl/certs/ca-certificates.crt"
     if os.path.isfile(certPath):
         sslopt['ca_certs'] = certPath
     sslopt.update(user_sslopt)
