--- a/configure.ac
+++ b/configure.ac
@@ -672,17 +672,13 @@
     [],
     [AC_MSG_WARN([Proton headers not found, disabling Proton support])
     use_proton=no])
- AC_CHECK_LIB([qpid-proton-cpp-static], [main],
-    [PROTON_LIBS="-lqpid-proton-cpp-static"],
-    [AC_MSG_WARN([Proton qpid-proton-cpp-static library not found, disabling Proton support])
+ AC_CHECK_LIB([qpid-proton-cpp], [main],
+    [PROTON_LIBS="-lqpid-proton-cpp"],
+    [AC_MSG_WARN([Proton qpid-proton-cpp library not found, disabling Proton support])
     use_proton=no])
- AC_CHECK_LIB([qpid-proton-core-static], [main],
-    [PROTON_LIBS+=" -lqpid-proton-core-static"],
-    [AC_MSG_WARN([Proton qpid-proton-core-static library not found, disabling Proton support])
-    use_proton=no])
- AC_CHECK_LIB([qpid-proton-static], [main],
-    [PROTON_LIBS+=" -lqpid-proton-static"],
-    [AC_MSG_WARN([Proton qpid-proton-static library not found, disabling Proton support])
+ AC_CHECK_LIB([qpid-proton-core], [main],
+    [PROTON_LIBS+=" -lqpid-proton-core"],
+    [AC_MSG_WARN([Proton qpid-proton-core library not found, disabling Proton support])
     use_proton=no])
 fi
 if test x$use_proton = xyes; then
