Index: src/SFML/Graphics/Makefile
===================================================================
--- src/SFML/Graphics/Makefile	(revision 1495)
+++ src/SFML/Graphics/Makefile	(working copy)
@@ -1,15 +1,5 @@
 SRC     = $(wildcard *.cpp)
-SRCGLEW = $(wildcard ./GLEW/*.c)
-SRCJPEG = $(wildcard ./libjpeg/*.c)
-SRCPNG  = $(wildcard ./libpng/*.c)
-SRCSOIL = $(wildcard ./SOIL/*.c)
-SRCZLIB = $(wildcard ./zlib/*.c)
 OBJ     = $(SRC:.cpp=.o)
-OBJGLEW = $(SRCGLEW:.c=.o)
-OBJJPEG = $(SRCJPEG:.c=.o)
-OBJPNG  = $(SRCPNG:.c=.o)
-OBJSOIL = $(SRCSOIL:.c=.o)
-OBJZLIB = $(SRCZLIB:.c=.o)
 
 ifeq ($(STATIC), yes)
     LIB     = libsfml-graphics-s.a
@@ -23,22 +13,19 @@
 
 all: $(LIB)
 
-libsfml-graphics-s.a: $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
-	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
+libsfml-graphics-s.a: $(OBJ) 
+	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ) -static -lGLEW -ljpeg -lpng -lsoil -lz -lSOIL
 
-libsfml-graphics.so: $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
-	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB) -lfreetype -lX11 -lGL
+libsfml-graphics.so: $(OBJ) 
+	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) -lGLEW -ljpeg -lpng -lSOIL -lz -lfreetype -lX11 -lGL
 
 $(OBJ): %.o: %.cpp
 	$(CPP) -o $@ -c $< $(CFLAGS) -I/usr/include/freetype2
 
-$(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB): %.o: %.c
-	$(CC) -o $@ -c $< $(CFLAGSEXT) -DSTBI_FAILURE_USERMSG
-
 .PHONY: clean mrproper
 
 clean:
-	@rm -rf $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
+	@rm -rf $(OBJ)
 
 mrproper: clean
 	@rm -rf $(LIBNAME)
