--- a/makefile
+++ b/makefile
@@ -315,9 +315,10 @@ MKDIR	= mkdir -p
 # Add WARN to COPTS for more checking
 WARN	= -Wpointer-arith -Wclobbered -Wempty-body -Wmissing-parameter-type -Wmissing-field-initializers -Wold-style-declaration -Wtype-limits -Wuninitialized -Winit-self -Wlogical-op -Wmissing-prototypes -Wmissing-declarations -Wnested-externs -Wbad-function-cast
 
-COPTS	= -ansi -pedantic -Wall -O3 -pipe
+CFLAGS ?= -ansi -pedantic -Wall -O3 -pipe
 
-CFLAGS	= -DOS_UNIX=$(OSUNIX)					\
+
+CFLAGS	+= -DOS_UNIX=$(OSUNIX)					\
 	  -DOS_DOS=$(OSDOS)					\
 	  -DOS_MAC=$(OSMAC)					\
 	  -DDB_FIX=$(DBFIX)					\
@@ -334,7 +335,7 @@ CFLAGS	= -DOS_UNIX=$(OSUNIX)					\
 	  -DCHAR_IN=$(CHARIN)					\
 	  -DCHAR_OUT=$(CHAROUT)					\
 	  -DLOCALE_ON=$(USELOC)					\
-	  -DASSERT_ON=1 $(COPTS)				\
+	  -DASSERT_ON=1 					\
 	  -DDEBUG_ON=$(DEBUGGING)				\
 	  $(TRACING)						\
 	  -DPDF_COMPRESSION=$(PDF_COMPRESSION)			\
@@ -349,16 +350,14 @@ OBJS	= z01.o z02.o z03.o z04.o z05.o z06.o z07.o z08.o	\
 	  z49.o z50.o z51.o z52.o
 
 lout:	$(OBJS)
-	$(CC) $(CFLAGS) -o lout $(OBJS) $(ZLIB) -lm
-	chmod a+x lout
+	$(CC) $(LDFLAGS) $(CFLAGS) -o lout $(OBJS) $(ZLIB) -lm
 
 $(OBJS): externs.h
 
 externs.h:
 
 prg2lout:	prg2lout.c
-	$(CC) $(COPTS) -o prg2lout prg2lout.c
-	chmod a+x prg2lout
+	$(CC) $(LDFLAGS) $(CFLAGS) -o prg2lout prg2lout.c
 
 all:	lout prg2lout
 
