diff --git a/dialogs/reboot-reminder-dialog/reminderdialog.cpp b/dialogs/reboot-reminder-dialog/reminderdialog.cpp
index b11130b2..efb302de 100644
--- a/dialogs/reboot-reminder-dialog/reminderdialog.cpp
+++ b/dialogs/reboot-reminder-dialog/reminderdialog.cpp
@@ -27,6 +27,7 @@
 
 #include <QApplication>
 #include <QDesktopWidget>
+#include <QScreen>
 
 #include <QDebug>
 
@@ -80,9 +81,14 @@ Manager::Manager() :
 
 void Manager::setupDialog()
 {
-    const QDesktopWidget *desktop = QApplication::desktop();
     const QPoint cursorPoint = QCursor::pos();
+#if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
+    const QDesktopWidget *desktop = QApplication::desktop();
     const QRect geom = desktop->screenGeometry(cursorPoint);
+#else
+    const QScreen *screen = qApp->screenAt(cursorPoint);
+    const QRect geom = screen->geometry();
+#endif
 
     if (!m_dialog)
         m_dialog = new ReminderDialog(geom);
