--- a/depends/packages/librustzcash.mk
+++ b/depends/packages/librustzcash.mk
@@ -5,9 +5,8 @@
 $(package)_download_file=$($(package)_git_commit).tar.gz
 $(package)_sha256_hash=9909ec59fa7a411c2071d6237b3363a0bc6e5e42358505cf64b7da0f58a7ff5a
 $(package)_git_commit=06da3b9ac8f278e5d4ae13088cf0a4c03d2c13f5
-$(package)_dependencies=rust
 ifeq ($(LIBRUSTZCASH_OVERRIDE),)
-$(package)_dependencies+=$(rust_crates)
+$(package)_dependencies=$(rust_crates)
 $(package)_patches=cargo.config 0001-Start-using-cargo-clippy-for-CI.patch remove-dev-dependencies.diff
 endif
 
@@ -61,7 +60,7 @@
 endef
 
 define $(package)_build_cmds
-  $(host_prefix)/native/bin/cargo build --package librustzcash $($(package)_build_opts)
+  cargo build --package librustzcash $($(package)_build_opts)
 endef
 
 define $(package)_stage_cmds
