--- a/configure.ac	2013-04-23 10:59:21.284914117 +0100
+++ b/configure.ac	2013-04-23 10:58:59.891664217 +0100
@@ -528,6 +528,13 @@
 m4_ifdef([LT_OUTPUT], [LT_OUTPUT])
 
 
+if test "x$GCC" = "xyes"; then
+  case " $CFLAGS " in
+  *[[\ \	]]-Wall[[\ \	]]*) ;;
+  *) CFLAGS="$CFLAGS -Wall" ;;
+  esac
+fi
+
 AS_IF([test "$glib_native_win32" = "yes"], [
   if test x$enable_static = xyes -a x$enable_shared = xyes; then
     AC_MSG_ERROR([Can not build both shared and static at the same time on Windows.])
