--- wlc-0.0.11/src/platform/render/gles2.c.orig	2020-05-30 20:59:49.759731716 +0000
+++ wlc-0.0.11/src/platform/render/gles2.c	2020-05-30 21:00:29.300322929 +0000
@@ -4,6 +4,9 @@
 #include <dlfcn.h>
 #include <GLES2/gl2.h>
 #include <GLES2/gl2ext.h>
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <EGL/eglmesaext.h>
 #include <wayland-server.h>
 #include <chck/string/string.h>
 #include "internal.h"
--- wlc-0.0.11/src/platform/context/egl.c.orig	2020-05-30 21:02:22.835353572 +0000
+++ wlc-0.0.11/src/platform/context/egl.c	2020-05-30 20:57:57.000000000 +0000
@@ -4,6 +4,7 @@
 #include <assert.h>
 #include <EGL/egl.h>
 #include <EGL/eglext.h>
+#include <EGL/eglmesaext.h>
 #include <wayland-server.h>
 #include <chck/string/string.h>
 #include "internal.h"
