From ca8036d425bf59020b476259e7857e8c67ebb6ba Mon Sep 17 00:00:00 2001
From: Maxime Gauduin <alucryd@gmail.com>
Date: Mon, 12 Aug 2013 05:20:41 +0200
Subject: [PATCH] wxStandardPaths is protected as of wxgtk 2.9.5

---
 aegisub/src/utils.cpp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/aegisub/src/utils.cpp b/aegisub/src/utils.cpp
index 70c9ea1..f72593d 100644
--- a/aegisub/src/utils.cpp
+++ b/aegisub/src/utils.cpp
@@ -121,8 +121,7 @@ void RestartAegisub() {
 	config::opt->Flush();
 
 #if defined(__WXMSW__)
-	wxStandardPaths stand;
-	wxExecute("\"" + stand.GetExecutablePath() + "\"");
+	wxExecute("\"" + wxStandardPaths::Get().GetExecutablePath() + "\"");
 #elif defined(__WXMAC__)
 	std::string bundle_path = agi::util::OSX_GetBundlePath();
 	std::string helper_path = agi::util::OSX_GetBundleAuxillaryExecutablePath("restart-helper");
@@ -132,8 +131,7 @@ void RestartAegisub() {
 	LOG_I("util/restart/exec") << exec;
 	wxExecute(exec);
 #else
-	wxStandardPaths stand;
-	wxExecute(stand.GetExecutablePath());
+	wxExecute(wxStandardPaths::Get().GetExecutablePath());
 #endif
 }
 
-- 
1.8.3.4

