diff -rupN a/common/edaappl.cpp b/common/edaappl.cpp
--- a/common/edaappl.cpp	2013-03-09 21:07:56.000000000 +0000
+++ b/common/edaappl.cpp	2014-01-07 07:49:38.984938912 +0000
@@ -451,7 +451,7 @@ bool EDA_APP::SetBinDir()
 
     // Linux and Unix
 #elif defined(__UNIX__)
-    m_BinDir = wxStandardPaths().GetExecutablePath();
+    m_BinDir = wxStandardPaths::Get().GetExecutablePath();
 #else
     m_BinDir = argv[0];
 #endif // __UNIX__
