--- src/main/display.cc	2012/02/19 14:16:05	4372
+++ src/main/display.cc	2012/02/20 10:56:39	4374
@@ -29,9 +29,12 @@
 #include <thmlhtmlhref.h>
 #include <gbfhtmlhref.h>
 #include <teihtmlhref.h>
+#ifndef NO_SWORD_SET_RENDER_NOTE_NUMBERS
+/* these files are erroneously uninstalled as of previous release */
 #include <osisxhtml.h>
 #include <thmlxhtml.h>
 #include <gbfxhtml.h>
+#endif
 
 #include <osisvariants.h>
 #include <thmlvariants.h>
--- src/main/display.cc (revision 4374)
+++ src/main/display.cc (revision 4375)
@@ -762,7 +762,10 @@
       }
 }

-#ifndef NO_SWORD_SET_RENDER_NOTE_NUMBERS
+#ifdef NO_SWORD_SET_RENDER_NOTE_NUMBERS
+// placeholder for older Sword.
+#define        set_render_numbers(x,y) /* nothing */
+#else
 void
 set_render_numbers(SWModule& imodule, GLOBAL_OPS *ops)
 { 
