diff -ur tigervnc-1.2.0.orig/unix/xserver/hw/vnc/xorg-version.h tigervnc-1.2.0/unix/xserver/hw/vnc/xorg-version.h
--- tigervnc-1.2.0.orig/unix/xserver/hw/vnc/xorg-version.h	2013-03-19 16:22:42.240358422 +0100
+++ tigervnc-1.2.0/unix/xserver/hw/vnc/xorg-version.h	2013-03-19 16:23:33.846355724 +0100
@@ -42,8 +42,10 @@
 #define XORG 112
 #elif XORG_VERSION_CURRENT < ((1 * 10000000) + (13 * 100000) + (99 * 1000))
 #define XORG 113
+#elif XORG_VERSION_CURRENT < ((1 * 10000000) + (14 * 100000) + (99 * 1000))
+#define XORG 114
 #else
-#error "X.Org newer than 1.13 is not supported"
+#error "X.Org newer than 1.14 is not supported"
 #endif
 
 #endif
diff -ur tigervnc-1.2.0.orig/unix/xserver/hw/vnc/xvnc.cc tigervnc-1.2.0/unix/xserver/hw/vnc/xvnc.cc
--- tigervnc-1.2.0.orig/unix/xserver/hw/vnc/xvnc.cc	2013-03-19 16:22:51.327357948 +0100
+++ tigervnc-1.2.0/unix/xserver/hw/vnc/xvnc.cc	2013-03-19 16:23:33.847355727 +0100
@@ -54,7 +54,9 @@
 #include "servermd.h"
 #include "fb.h"
 #include "mi.h"
+#if XORG < 114
 #include "mibstore.h"
+#endif
 #include "colormapst.h"
 #include "gcstruct.h"
 #include "input.h"
