diff --git a/Radius.pm b/Radius.pm
index db56f23..b88db32 100644
--- a/Radius.pm
+++ b/Radius.pm
@@ -436,7 +436,8 @@ sub vendorID ($) {
 	return defined $dict_vendor_name{ $attr->{'Vendor'} }{'id'} ? $dict_vendor_name{ $attr->{'Vendor'} }{'id'} : int($attr->{'Vendor'});
     } else {
 	# look up vendor by attribute name
-	my $vendor_name = $dict_name{$attr->{'Name'}}{'vendor'};
+	my $vendor_name = defined ($dict_name{$attr->{'Name'}}{'vendor'}) ?
+	    $dict_name{$attr->{'Name'}}{'vendor'} : 'not defined';
 	my $vendor_id = defined ($dict_vendor_name{$vendor_name}{'id'}) ?
 	    $dict_vendor_name{$vendor_name}{'id'} : 'not defined';
 	return $vendor_id;
