# https://bugzilla.netfilter.org/show_bug.cgi?id=893

diff -r -u nftables-0.099/configure.ac nftables-0.099.new/configure.ac
--- nftables-0.099/configure.ac	2014-01-20 08:15:16.000000000 -0500
+++ nftables-0.099.new/configure.ac	2014-02-02 13:01:39.355368732 -0500
@@ -24,11 +24,19 @@
 AC_PROG_MKDIR_P
 AC_PROG_INSTALL
 AC_PROG_SED
+AC_PROG_GREP
 
-AC_CHECK_PROG(CONFIG_MAN, docbook2x-man, y, n)
-if test "$CONFIG_MAN" != "y"
+AC_PATH_PROGS_FEATURE_CHECK([DOCBOOK2MAN], [docbook2x-man docbook2man], 
+        [$ac_path_DOCBOOK2MAN --version | grep -qm1 'part of docbook2X' \
+         && ac_cv_path_DOCBOOK2MAN=$ac_path_DOCBOOK2MAN ac_path_DOCBOOK2MAN_found=:])
+
+if test -n "$ac_cv_path_DOCBOOK2MAN"
 then
-	AC_MSG_WARN([docbookx2-man not found, no manpages will be built])
+    AC_SUBST([CONFIG_MAN], "y")
+    AC_SUBST([DOCBOOK2MAN], [$ac_cv_path_DOCBOOK2MAN])
+else
+    AC_SUBST([CONFIG_MAN], "n")
+    AC_MSG_WARN([docbookx2-man not found, no manpages will be built])
 fi
 
 AC_CHECK_PROG(CONFIG_PDF, db2pdf, y, n)
diff -r -u nftables-0.099/Makefile.rules.in nftables-0.099.new/Makefile.rules.in
--- nftables-0.099/Makefile.rules.in	2014-01-20 08:15:16.000000000 -0500
+++ nftables-0.099.new/Makefile.rules.in	2014-02-02 13:02:53.660639570 -0500
@@ -37,7 +37,7 @@
 
 %.8:			%.xml	$(makedeps)
 			@echo -e "  MAN\t\t$@"
-			(cd $(SUBDIR); docbook2x-man ../$<)
+			(cd $(SUBDIR); @DOCBOOK2MAN@ ../$<)
 
 %.pdf:			%.xml	$(makedeps)
 			@echo -e "  PDF\t\t$@"
