diff -upr python-ldap-python-ldap-3.3.1.orig/setup.cfg python-ldap-python-ldap-3.3.1/setup.cfg
--- python-ldap-python-ldap-3.3.1.orig/setup.cfg	2020-06-29 17:43:42.000000000 +0300
+++ python-ldap-python-ldap-3.3.1/setup.cfg	2021-11-13 20:13:55.037043241 +0200
@@ -23,7 +23,7 @@ extra_objects =
 
 # Example for full-featured build:
 # Support for StartTLS/LDAPS, SASL bind and reentrant libldap_r.
-libs = ldap_r lber
+libs = ldap lber
 
 # Installation options
 [install]
