diff -aur oolite-source-1.84.pristine/src/SDL/main.m oolite-source-1.84.new/src/SDL/main.m
--- oolite-source-1.84.pristine/src/SDL/main.m	2015-09-07 05:52:51.000000000 +0200
+++ oolite-source-1.84.new/src/SDL/main.m	2016-08-22 17:30:28.997554264 +0200
@@ -25,6 +25,9 @@
 
 #ifdef GNUSTEP
 #import <Foundation/NSAutoreleasePool.h>
+#if (GNUSTEP_BASE_MAJOR_VERSION == 1 && (GNUSTEP_BASE_MINOR_VERSION == 24 && GNUSTEP_BASE_SUBMINOR_VERSION >= 9) || (GNUSTEP_BASE_MINOR_VERSION > 24)) || (GNUSTEP_BASE_MAJOR_VERSION > 1)
+#import <Foundation/NSDate.h>
+#endif
 #import <Foundation/NSString.h>
 
 #import "GameController.h"
@@ -47,6 +50,10 @@
 {
 #ifdef GNUSTEP
 	int i;
+
+#if (GNUSTEP_BASE_MAJOR_VERSION == 1 && (GNUSTEP_BASE_MINOR_VERSION == 24 && GNUSTEP_BASE_SUBMINOR_VERSION >= 9) || (GNUSTEP_BASE_MINOR_VERSION > 24)) || (GNUSTEP_BASE_MAJOR_VERSION > 1)
+	[NSDate class]; // See github issue #202
+#endif
 	
 #if OOLITE_WINDOWS
 
