diff --git a/Gemfile b/Gemfile
index bc5f7c5..3101ce2 100644
--- a/Gemfile
+++ b/Gemfile
@@ -9,11 +9,9 @@ end
 
 group :test do
   gem "childlabor"
-  gem "coveralls", ">= 0.8.19"
   gem "rspec", ">= 3.2"
   gem "rspec-mocks", ">= 3"
   gem "rubocop", "~> 0.50.0"
-  gem "simplecov", ">= 0.13"
   gem "webmock"
 end
 
diff --git a/spec/helper.rb b/spec/helper.rb
index c19dd6f..6d32258 100644
--- a/spec/helper.rb
+++ b/spec/helper.rb
@@ -1,15 +1,5 @@
 $TESTING = true
 
-require "simplecov"
-require "coveralls"
-
-SimpleCov.formatters = [SimpleCov::Formatter::HTMLFormatter, Coveralls::SimpleCov::Formatter]
-
-SimpleCov.start do
-  add_filter "/spec"
-  minimum_coverage(90)
-end
-
 $LOAD_PATH.unshift(File.join(File.dirname(__FILE__), "..", "lib"))
 require "thor"
 require "thor/group"
@@ -20,8 +10,6 @@ require "rspec"
 require "diff/lcs" # You need diff/lcs installed to run specs (but not to run Thor).
 require "webmock/rspec"
 
-WebMock.disable_net_connect!(:allow => "coveralls.io")
-
 # Set shell to basic
 ENV["THOR_COLUMNS"] = "10000"
 $0 = "thor"
diff --git a/spec/quality_spec.rb b/spec/quality_spec.rb
index a6ceb3e..da0c500 100644
--- a/spec/quality_spec.rb
+++ b/spec/quality_spec.rb
@@ -40,7 +40,7 @@ describe "The library itself" do
     exempt = /\.gitmodules|\.marshal|fixtures|vendor|spec|ssl_certs|LICENSE|.devcontainer/
     error_messages = []
     Dir.chdir(File.expand_path("../..", __FILE__)) do
-      `git ls-files`.split("\n").each do |filename|
+      `find -type f -not -path '*/\.git/*' -not -path '*.gem'`.split("\n").each do |filename|
         next if filename =~ exempt
         error_messages << check_for_tab_characters(filename)
         error_messages << check_for_extra_spaces(filename)
@@ -53,7 +53,7 @@ describe "The library itself" do
     included = /spec/
     error_messages = []
     Dir.chdir(File.expand_path("../", __FILE__)) do
-      `git ls-files`.split("\n").each do |filename|
+      `find -type f -not -path '*/\.git/* -not -path '*.gem''`.split("\n").each do |filename|
         next unless filename =~ included
         error_messages << check_for_spec_defs_with_single_quotes(filename)
       end
