From cc31d235cd3c4f0adf35614e09965c28175c7c99 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9rgio=20Basto?= <sergio@serjux.com>
Date: Tue, 27 Jan 2015 13:30:27 +0000
Subject: [PATCH] Compile with Lensfun 0.3 version where CCI correction have
 been removed

---
 plugins/lensfun/lensfun.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/lensfun/lensfun.c b/plugins/lensfun/lensfun.c
index af1f189..4f7fffb 100644
--- a/plugins/lensfun/lensfun.c
+++ b/plugins/lensfun/lensfun.c
@@ -668,7 +668,7 @@ get_image(RSFilter *filter, const RSFilterRequest *request)
 			}
 
 			/* Start threads to apply phase 2, Vignetting and CA Correction */
-			if (effective_flags & (LF_MODIFY_VIGNETTING | LF_MODIFY_CCI)) 
+			if (effective_flags & LF_MODIFY_VIGNETTING)
 			{
 				/* Phase 2 is corrected inplace, so copy input first */
 				guint y_offset, y_per_thread, threaded_h;
