--- a/supermin/src/ph_pacman.ml	2019-10-14 21:28:25.138769513 +0200
+++ b/supermin/src/ph_pacman.ml	2019-10-14 21:33:31.479364225 +0200
@@ -171,24 +171,24 @@
    * only downloads each package once, even though each call to pacman will
    * download dependent packages as well.
    *)
-  List.iter (
-    fun name ->
-      let cmd = sprintf "\
-        set -e
-        umask 0000
-        cd %s
-        mkdir -p var/lib/pacman
-        %s %s%s -Syw --noconfirm --cachedir=$(pwd) --root=$(pwd) %s
-      "
-        (quote tdir)
-        Config.fakeroot Config.pacman
-        (match !settings.packager_config with
-         | None -> ""
-         | Some filename -> " --config " ^ (quote filename))
-        (quoted_list names) in
-      if !settings.debug >= 2 then printf "%s" cmd;
-      if Sys.command cmd <> 0 then (
-        (* The package may not be in the main repos, check the AUR. *)
+  let cmd = sprintf "\
+    set -e
+    umask 0000
+    cd %s
+    mkdir -p var/lib/pacman
+    %s %s%s -Syw --noconfirm --cachedir=$(pwd) --root=$(pwd) %s
+  "
+    (quote tdir)
+    Config.fakeroot Config.pacman
+    (match !settings.packager_config with
+     | None -> ""
+     | Some filename -> " --config " ^ (quote filename))
+    (quoted_list names) in
+  if !settings.debug >= 2 then printf "%s" cmd;
+  if Sys.command cmd <> 0 then (
+    (* The package may not be in the main repos, check the AUR. *)
+    List.iter (
+      fun name ->
         let cmd = sprintf "\
           set -e
           umask 0000
@@ -201,16 +201,16 @@
        "
           (quote tdir)
           (quote ("https://aur.archlinux.org/packages/" ^
-	          (String.sub name 0 2) ^
-	          "/" ^ name ^ "/" ^ name ^ ".tar.gz"))
+            (String.sub name 0 2) ^
+            "/" ^ name ^ "/" ^ name ^ ".tar.gz"))
           (quote (name ^ ".tar.gz"))
           (quote name) (* cd *)
           Config.makepkg
           (quote name) (quote tdir) (* mv *) in
         if !settings.debug >= 2 then printf "%s" cmd;
         run_command cmd
-      );
-  ) names;
+    ) names;
+  );
 
   (* Unpack the downloaded packages. *)
   let cmd =
