diff -upr supermin-5.2.1.orig/src/ph_pacman.ml supermin-5.2.1/src/ph_pacman.ml
--- supermin-5.2.1.orig/src/ph_pacman.ml	2021-02-01 11:30:45.000000000 +0200
+++ supermin-5.2.1/src/ph_pacman.ml	2021-05-24 18:11:28.002222316 +0300
@@ -219,7 +219,10 @@ let pacman_download_all_packages pkgs di
   let cmd =
     sprintf "
       umask 0000
-      for f in %s/*.pkg.tar.*; do tar -xf \"$f\" -C %s; done
+      for f in %s/*.pkg.tar.*; do
+        [[ $f == *.sig ]] && continue
+        tar -xf \"$f\" -C %s
+      done
     "
       (quote tdir) (quote dir) in
   if !settings.debug >= 2 then printf "%s" cmd;
