diff -wbBur gnustep-gui-0.24.0/Source/NSBitmapImageRep+GIF.m gnustep-gui-0.24.0.my/Source/NSBitmapImageRep+GIF.m
--- gnustep-gui-0.24.0/Source/NSBitmapImageRep+GIF.m	2013-11-24 15:56:33.000000000 +0400
+++ gnustep-gui-0.24.0.my/Source/NSBitmapImageRep+GIF.m	2014-05-29 18:48:13.333993653 +0400
@@ -161,7 +161,7 @@
       return NO;
     }
 
-  DGifCloseFile(file);
+  DGifCloseFile(file, NULL);
   return YES;
 }
 
@@ -180,7 +180,7 @@
    SET_ERROR_MSG(msg); \
    if (file != NULL) \
      {\
-       DGifCloseFile(file); \
+       DGifCloseFile(file, NULL); \
      }\
    if (imgBuffer != NULL) \
      {\
@@ -392,7 +392,7 @@
           withValue: [NSNumber numberWithInt: 0]];
 
   /* don't forget to close the gif */
-  DGifCloseFile(file);
+  DGifCloseFile(file, NULL);
 
   return self;
 }
