From b78be80d138076471cdfdc2c4edab4a3af6241b0 Mon Sep 17 00:00:00 2001
From: Petr Šabata <contyk@redhat.com>
Date: Mon, 21 Dec 2015 16:57:59 +0100
Subject: [PATCH] Build even without SyncTeX

---
 zathura/main.c | 2 --
 1 file changed, 0 insertions(+), 2 deletions(-)

diff --git a/zathura/main.c b/zathura/main.c
index 060382a..1c5b73e 100644
--- a/zathura/main.c
+++ b/zathura/main.c
@@ -135,10 +135,8 @@ main(int argc, char* argv[])
   gchar* plugin_path    = NULL;
   gchar* loglevel       = NULL;
   gchar* password       = NULL;
-#ifdef WITH_SYNCTEX
   gchar* synctex_editor = NULL;
   gchar* synctex_fwd    = NULL;
-#endif
   gchar* mode           = NULL;
   bool forkback         = false;
   bool print_version    = false;
--
libgit2 0.23.3

