diff --git a/Makefile b/Makefile
index 1838c13..6c28506 100644
--- a/Makefile
+++ b/Makefile
@@ -27,7 +27,7 @@ SRC_BENCH = src/bench.c
 SRC_GENKAT = src/genkat.c
 OBJ = $(SRC:.c=.o)
 
-CFLAGS += -std=c89 -pthread -O3 -Wall -g -Iinclude -Isrc
+CFLAGS += -std=c89 -pthread -Iinclude -Isrc
 CI_CFLAGS := $(CFLAGS) -Werror=declaration-after-statement -D_FORTIFY_SOURCE=2 \
 				-Wextra -Wno-type-limits -Werror -coverage -DTEST_LARGE_RAM
 
@@ -40,7 +40,6 @@ $(info Building without optimizations)
 	SRC += src/ref.c
 else
 $(info Building with optimizations for $(OPTTARGET))
-	CFLAGS += -march=$(OPTTARGET)
 	SRC += src/opt.c
 endif
 
