diff --git a/Makefile b/Makefile
index 6c28506..1c1b028 100644
--- a/Makefile
+++ b/Makefile
@@ -151,10 +151,10 @@ format:
 		clang-format -style="{BasedOnStyle: llvm, IndentWidth: 4}" \
 			-i include/*.h src/*.c src/*.h src/blake2/*.c src/blake2/*.h
 
-install: $(RUN) libs
+install:
 	$(INSTALL) -d $(INST_INCLUDE)
 	$(INSTALL) $(HEADERS) $(INST_INCLUDE)
 	$(INSTALL) -d $(INST_LIBRARY)
-	$(INSTALL) $(LIBRARIES) $(INST_LIBRARY)
+	$(INSTALL) $(LIB_SH) $(INST_LIBRARY)
 	$(INSTALL) -d $(INST_BINARY)
 	$(INSTALL) $(RUN) $(INST_BINARY)
