diff --git a/lib/std/net/test.zig b/lib/std/net/test.zig
index 9f40bb5a3..15dd8c920 100644
--- a/lib/std/net/test.zig
+++ b/lib/std/net/test.zig
@@ -106,18 +106,6 @@ test "resolve DNS" {
         }
     }
 
-    // Resolve localhost, this should not fail.
-    {
-        const localhost_v4 = try net.Address.parseIp("127.0.0.1", 80);
-        const localhost_v6 = try net.Address.parseIp("::2", 80);
-
-        const result = try net.getAddressList(testing.allocator, "localhost", 80);
-        defer result.deinit();
-        for (result.addrs) |addr| {
-            if (addr.eql(localhost_v4) or addr.eql(localhost_v6)) break;
-        } else @panic("unexpected address for localhost");
-    }
-
     {
         // The tests are required to work even when there is no Internet connection,
         // so some of these errors we must accept and skip the test.
