diff --git a/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/parser/CustomExpression.java b/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/parser/CustomExpression.java
index 4be4e00a96..1e0cc8d840 100644
--- a/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/parser/CustomExpression.java
+++ b/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/parser/CustomExpression.java
@@ -19,6 +19,7 @@
 import net.sf.jsqlparser.expression.Expression;
 import net.sf.jsqlparser.expression.ExpressionVisitor;
 import net.sf.jsqlparser.expression.StringValue;
+import net.sf.jsqlparser.parser.SimpleNode;
 
 /**
  * CustomExpression
@@ -26,6 +27,7 @@
 public class CustomExpression implements Expression {
 
     private final String expression;
+    private SimpleNode simpleNode;
 
     public CustomExpression(String expression) {
         this.expression = expression;
@@ -40,4 +42,14 @@ public void accept(ExpressionVisitor expressionVisitor) {
     public String toString() {
         return expression;
     }
+
+    @Override
+    public SimpleNode getASTNode() {
+        return simpleNode;
+    }
+
+    @Override
+    public void setASTNode(SimpleNode simpleNode) {
+        this.simpleNode = simpleNode;
+    }
 }
