diff --git a/gb.db.postgresql/src/main.c b/gb.db.postgresql/src/main.c
index 5401768d4..e0cca37d5 100644
--- a/gb.db.postgresql/src/main.c
+++ b/gb.db.postgresql/src/main.c
@@ -24,15 +24,18 @@
 
 #define __MAIN_C
 
+#include <libpq-fe.h>
+#include <postgres.h>
+#include <pg_type.h>
+
+#undef fprintf
+#undef snprintf
+#undef sprintf
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdarg.h>
 #include <string.h>
 
-#include <libpq-fe.h>
-#include <postgres.h>
-#include <pg_type.h>
-
 #ifdef PACKAGE_NAME
 	#undef PACKAGE_NAME
 	#undef PACKAGE_BUGREPORT
