--- a/gb.sdl2/src/audio/main.c	2017-07-18 17:48:44.000000000 +0200
+++ b/gb.sdl2/src/audio/main.c.new	2017-11-09 21:33:11.442013948 +0100
@@ -58,7 +58,7 @@
 	init_mixer(MIX_INIT_OGG, "OGG");
 	init_mixer(MIX_INIT_MOD, "MOD");
 	init_mixer(MIX_INIT_FLAC, "FLAC");
-	init_mixer(MIX_INIT_FLUIDSYNTH, "FLUIDSYNTH");
+	init_mixer(MIX_INIT_MID, "MID");
 
 	if (Mix_OpenAudio(AUDIO_frequency, MIX_DEFAULT_FORMAT, 2, AUDIO_buffer_size))
 	{
