diff -Naur activity-log-manager-0.9.4.orig/src/alm.vala activity-log-manager-0.9.4/src/alm.vala
--- activity-log-manager-0.9.4.orig/src/alm.vala	2012-02-22 20:24:08.000000000 +0100
+++ activity-log-manager-0.9.4/src/alm.vala	2012-11-01 09:33:15.466534392 +0100
@@ -41,10 +41,7 @@
 		static int main (string[] args) 
 		{
 			Gtk.init (ref args);
-			var alm = new Gtk.Application ("org.zeitgeist.Alm", 
-										ApplicationFlags.FLAGS_NONE);
-			var window = new Main ();
-			alm.add_window (window);
+			new Main ();
 			
 			Gtk.main ();
 			return 0;
