diff --git i/cdemu-client/CMakeLists.txt w/cdemu-client/CMakeLists.txt
index cdcff25232886440..33507a7a5db4332e 100644
--- i/cdemu-client/CMakeLists.txt
+++ w/cdemu-client/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required (VERSION 2.8.5)
 
 # Project name
-project (cdemu NONE)
+project (cdemu-client NONE)
 
 # Additional CMake modules.
 list (APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
@@ -38,7 +38,8 @@ install (
 
 install (
     FILES data/cdemu-bash-completion.sh
-    DESTINATION ${CMAKE_INSTALL_SYSCONFDIR}/bash_completion.d
+    DESTINATION ${CMAKE_INSTALL_DATADIR}/bash-completion/completions
+    RENAME cdemu
 )
 
 install (
