diff --git i/cdemu-client/CMakeLists.txt w/cdemu-client/CMakeLists.txt
index cdcff25232886440..5296229538cd64bf 100644
--- i/cdemu-client/CMakeLists.txt
+++ w/cdemu-client/CMakeLists.txt
@@ -38,7 +38,8 @@ install (
 
 install (
     FILES data/cdemu-bash-completion.sh
-    DESTINATION ${CMAKE_INSTALL_SYSCONFDIR}/bash_completion.d
+    DESTINATION ${CMAKE_INSTALL_DATADIR}/bash-completion/completions
+    RENAME cdemu
 )
 
 install (
