--- HIPIFY-rocm-5.5.0/CMakeLists.txt.bak	2023-05-14 15:40:43.155264590 +0200
+++ HIPIFY-rocm-5.5.0/CMakeLists.txt	2023-05-14 15:41:03.815224712 +0200
@@ -126,25 +126,13 @@
 
 set(HIPIFY_INSTALL_PATH ${CMAKE_INSTALL_PREFIX})
 
-install(TARGETS hipify-clang DESTINATION ${CMAKE_INSTALL_PREFIX})
+install(TARGETS hipify-clang DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 install(
     DIRECTORY ${CMAKE_SOURCE_DIR}/bin/
-    DESTINATION ${CMAKE_INSTALL_PREFIX}
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
     USE_SOURCE_PERMISSIONS
     PATTERN "hipify-perl"
     PATTERN "*.sh")
-install(
-    DIRECTORY ${LLVM_DIR}/../../clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/
-    DESTINATION ${CMAKE_INSTALL_PREFIX}
-    COMPONENT clang-resource-headers
-    FILES_MATCHING
-    PATTERN "*.h"
-    PATTERN "*.modulemap"
-    PATTERN "algorithm"
-    PATTERN "complex"
-    PATTERN "new"
-    PATTERN "ppc_wrappers" EXCLUDE
-    PATTERN "openmp_wrappers" EXCLUDE)
 
 option(FILE_REORG_BACKWARD_COMPATIBILITY "Enable File Reorg with backward compatibility" ON)
 
