diff -Naur Marker-2020.04.04.orig/src/marker-markdown.c Marker-2020.04.04/src/marker-markdown.c
--- Marker-2020.04.04.orig/src/marker-markdown.c	2020-04-05 00:31:59.000000000 +0200
+++ Marker-2020.04.04/src/marker-markdown.c	2020-07-22 12:48:35.666097636 +0200
@@ -87,8 +87,7 @@
       } else
       {
         mathjs_css = g_strdup(" ");
-        mathjs_script = g_strdup_printf("<script src=\"file://%smathjax/MathJax.js?config=TeX-AMS_HTML\"></script>",
-                                        SCRIPTS_DIR);
+        mathjs_script = g_strdup_printf("<script src=\"file:///usr/share/mathjax2/MathJax.js?config=TeX-AMS_HTML\"></script>");
         mathjs_auto = g_strdup(" ");
       }
       break;
