diff -wbBur units-2.00.org/Makefile.in units-2.00/Makefile.in
--- units-2.00.org/Makefile.in	2012-05-03 00:37:55.000000000 +0400
+++ units-2.00/Makefile.in	2012-07-04 12:54:20.000000000 +0400
@@ -84,10 +84,10 @@
 	sed -e "s:outfile = 'currency.units':outfile='@UDAT@currency.units':" units_cur > units_cur_inst
 
 install-support: definitions.units currency.units units_cur_inst
-	$(srcdir)/mkinstalldirs @UDAT@  $(DESTDIR)$(bindir) 
-	$(INSTALL_DATA) $(srcdir)/definitions.units @UDAT@definitions.units
-	$(INSTALL_DATA) $(srcdir)/currency.units @UDAT@currency.units
-	$(INSTALL_DATA) $(srcdir)/locale.map @UDAT@locale.map
+	$(srcdir)/mkinstalldirs $(DESTDIR)/@UDAT@  $(DESTDIR)$(bindir) 
+	$(INSTALL_DATA) $(srcdir)/definitions.units $(DESTDIR)/@UDAT@definitions.units
+	$(INSTALL_DATA) $(srcdir)/currency.units $(DESTDIR)/@UDAT@currency.units
+	$(INSTALL_DATA) $(srcdir)/locale.map $(DESTDIR)/@UDAT@locale.map
 	$(INSTALL_PROGRAM) units_cur_inst $(DESTDIR)$(bindir)/`echo units_cur|sed '$(transform)'`
 
 install: units@EXEEXT@ install-doc install-support
diff -wbBur units-2.00.org/units_cur units-2.00/units_cur
--- units-2.00.org/units_cur	2012-05-01 04:51:35.000000000 +0400
+++ units-2.00/units_cur	2012-07-04 12:53:47.000000000 +0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 import urllib
 import datetime
