--- a/brightray/brightray.gyp
+++ b/brightray/brightray.gyp
@@ -149,8 +149,6 @@
             }, {
               'link_settings': {
                 'libraries': [
-                  # Link with ffmpeg.
-                  '<(libchromiumcontent_dir)/libffmpeg.so',
                   # Following libraries are required by libchromiumcontent:
                   '-lasound',
                   '-lcap',
--- a/electron.gyp
+++ b/electron.gyp
@@ -215,7 +215,6 @@
                   }, {
                     'copied_libraries': [
                       '<(PRODUCT_DIR)/lib/libnode.so',
-                      '<(libchromiumcontent_dir)/libffmpeg.so',
                     ],
                   }],
                 ],
