diff --git a/date.gemspec b/date.gemspec
index cf07696..01a91d3 100644
--- a/date.gemspec
+++ b/date.gemspec
@@ -1,12 +1,8 @@
 # frozen_string_literal: true
 
-version = File.foreach(File.expand_path("../lib/date.rb", __FILE__)).find do |line|
-  /^\s*VERSION\s*=\s*["'](.*)["']/ =~ line and break $1
-end
-
 Gem::Specification.new do |s|
   s.name = "date"
-  s.version = version
+  s.version = "3.2.2"
   s.summary = "A subclass of Object includes Comparable module for handling dates."
   s.description = "A subclass of Object includes Comparable module for handling dates."
 
