diff -Naur clinica-0.2.1.orig/libclinica/DoctorContextMenu.vala clinica-0.2.1/libclinica/DoctorContextMenu.vala
--- clinica-0.2.1.orig/libclinica/DoctorContextMenu.vala	2011-11-30 13:42:30.000000000 +0100
+++ clinica-0.2.1/libclinica/DoctorContextMenu.vala	2012-10-16 16:57:35.000000000 +0200
@@ -22,7 +22,7 @@
 
 namespace Clinica {
 
-	public class DoctorContextMenu : Menu {
+	public class DoctorContextMenu : Gtk.Menu {
 	
 		/**
 		 * @brief The doctor associated to this popup menu 
@@ -38,13 +38,13 @@
 		/**
 		 * @brief MenuItem associated to doctor deletion
 		 */
-		private MenuItem delete_menuitem;
+		private Gtk.MenuItem delete_menuitem;
 		
 		/**
 		 * @brief MenuItem that triggers doctor edit
 		 * dialog.
 		 */
-		private MenuItem edit_menuitem;
+		private Gtk.MenuItem edit_menuitem;
 		
 		private ResourceManager resource_manager;
 	
@@ -55,8 +55,8 @@
 			
 			/* Create menu items, connect them to their callback
 			 * and add it to the menu */
-			delete_menuitem = new MenuItem.with_label (_("Delete"));
-			edit_menuitem   = new MenuItem.with_label (_("Edit"));
+			delete_menuitem = new Gtk.MenuItem.with_label (_("Delete"));
+			edit_menuitem   = new Gtk.MenuItem.with_label (_("Edit"));
 			
 			delete_menuitem.activate.connect (on_delete_menuitem_activated);
 			edit_menuitem.activate.connect (on_edit_menuitem_activated);
@@ -67,11 +67,11 @@
 			show_all ();
 		}
 		
-		private void on_delete_menuitem_activated (MenuItem item) {
+		private void on_delete_menuitem_activated (Gtk.MenuItem item) {
 			view.remove_selected_doctor ();
 		}
 		
-		private void on_edit_menuitem_activated (MenuItem item) {
+		private void on_edit_menuitem_activated (Gtk.MenuItem item) {
 			var doctor_editor = new DoctorEditor.with_doctor (resource_manager, doctor);
 			doctor_editor.dialog.set_transient_for (resource_manager.user_interface.window);
 			doctor_editor.run ();
diff -Naur clinica-0.2.1.orig/libclinica/EventDetail.vala clinica-0.2.1/libclinica/EventDetail.vala
--- clinica-0.2.1.orig/libclinica/EventDetail.vala	2011-11-30 13:42:30.000000000 +0100
+++ clinica-0.2.1/libclinica/EventDetail.vala	2012-10-16 16:55:51.000000000 +0200
@@ -104,9 +104,9 @@
          */
         private bool on_button_press_event (Widget widget, EventButton event) {
             if (event.button == 3 && event.type == EventType.BUTTON_PRESS) {
-                var popup_menu = new Menu ();
-                var edit_menu_item = new MenuItem.with_label (_("Edit"));
-                var remove_menu_item = new MenuItem.with_label (_("Remove"));
+                var popup_menu = new Gtk.Menu ();
+                var edit_menu_item = new Gtk.MenuItem.with_label (_("Edit"));
+                var remove_menu_item = new Gtk.MenuItem.with_label (_("Remove"));
                 
                 /* Connect callbacks for the item */
                 edit_menu_item.activate.connect (on_edit_menu_item_activate);
@@ -125,7 +125,7 @@
             return false;
         }
         
-        private void on_edit_menu_item_activate (MenuItem? item = null) {
+        private void on_edit_menu_item_activate (Gtk.MenuItem? item = null) {
             var event_editor = new EventEditor (resource_manager, associated_event);
             event_editor.set_transient_for (resource_manager.user_interface.calendar_window);            
             if (event_editor.run () == EventEditor.Response.SAVE) {
@@ -137,7 +137,7 @@
             event_editor.destroy ();
         }
         
-        private void on_remove_menu_item_activate (MenuItem? item = null) {
+        private void on_remove_menu_item_activate (Gtk.MenuItem? item = null) {
         	var question = new MessageDialog (null, 
         				DialogFlags.MODAL | DialogFlags.DESTROY_WITH_PARENT, 
         				MessageType.QUESTION, ButtonsType.YES_NO, "%s", 
diff -Naur clinica-0.2.1.orig/libclinica/PatientContextMenu.vala clinica-0.2.1/libclinica/PatientContextMenu.vala
--- clinica-0.2.1.orig/libclinica/PatientContextMenu.vala	2011-12-09 18:14:42.000000000 +0100
+++ clinica-0.2.1/libclinica/PatientContextMenu.vala	2012-10-16 16:54:59.000000000 +0200
@@ -26,7 +26,7 @@
      * @brief Menu to be displayed when a user right click on a patient
      * in the TreeView
      */
-    public class PatientContextMenu : Menu {
+    public class PatientContextMenu : Gtk.Menu {
     
     	/* Patient where that user has right clicked on */
     	Patient patient;
@@ -35,9 +35,9 @@
     	PatientListView view;
     	
     	/* MENU ITEMS */
-    	MenuItem edit_menuitem;
-    	MenuItem delete_menuitem;
-    	MenuItem view_visits_menuitem;
+    	Gtk.MenuItem edit_menuitem;
+    	Gtk.MenuItem delete_menuitem;
+    	Gtk.MenuItem view_visits_menuitem;
     	
 		private ResourceManager resource_manager;
     
@@ -48,13 +48,13 @@
     		this.view = view;
     		
     		/* Instantiate menu items */
-    		edit_menuitem = new MenuItem.with_label(_("Edit"));
+    		edit_menuitem = new Gtk.MenuItem.with_label(_("Edit"));
     		edit_menuitem.activate.connect (on_edit_menuitem_activated);
     		
-    		delete_menuitem = new MenuItem.with_label(_("Delete"));
+    		delete_menuitem = new Gtk.MenuItem.with_label(_("Delete"));
     		delete_menuitem.activate.connect(on_delete_menuitem_activated);
     		
-    		view_visits_menuitem = new MenuItem.with_label(_("Edit visits"));
+    		view_visits_menuitem = new Gtk.MenuItem.with_label(_("Edit visits"));
     		view_visits_menuitem.activate.connect(on_view_visits_menuitem_activated);
     		
     		/* Append menuitems */
@@ -65,18 +65,18 @@
     		show_all ();
     	}
     	
-    	private void on_edit_menuitem_activated (MenuItem item) {
+    	private void on_edit_menuitem_activated (Gtk.MenuItem item) {
     		var patient_editor = new PatientEditor.with_patient (resource_manager, patient);
     		patient_editor.dialog.set_transient_for (resource_manager.user_interface.window);
     		patient_editor.run();
     		patient_editor.destroy ();
     	}
     	
-    	private void on_delete_menuitem_activated (MenuItem item) {
+    	private void on_delete_menuitem_activated (Gtk.MenuItem item) {
     		view.remove_selected_patient ();
     	}
     	
-    	private void on_view_visits_menuitem_activated (MenuItem item) {
+    	private void on_view_visits_menuitem_activated (Gtk.MenuItem item) {
             resource_manager.user_interface.show_visit_window (patient);
     	}
     }
diff -Naur clinica-0.2.1.orig/libclinica/UserInterface.vala clinica-0.2.1/libclinica/UserInterface.vala
--- clinica-0.2.1.orig/libclinica/UserInterface.vala	2011-12-09 19:15:27.000000000 +0100
+++ clinica-0.2.1/libclinica/UserInterface.vala	2012-10-16 16:56:48.000000000 +0200
@@ -216,7 +216,7 @@
 		/* Menu item */
 		
 		[CCode (instance_pos = -1)]
-		public void on_new_patient_menu_item_activate (MenuItem item) {
+		public void on_new_patient_menu_item_activate (Gtk.MenuItem item) {
 			var new_patient_dialog = new PatientEditor (resource_manager);
 			new_patient_dialog.dialog.set_transient_for (window);
 			new_patient_dialog.run ();
@@ -224,7 +224,7 @@
 		}
 		
 		[CCode (instance_pos = -1)]
-        	public void on_new_doctor_menu_item_activate (MenuItem item) {
+        	public void on_new_doctor_menu_item_activate (Gtk.MenuItem item) {
         		var new_doc_dialog = new DoctorEditor (resource_manager);
         		new_doc_dialog.dialog.set_transient_for (window);
 	        	new_doc_dialog.run ();
@@ -232,22 +232,22 @@
         	}
 		
 		[CCode (instance_pos = -1)]
-		public void on_patient_menu_item_activate (MenuItem item) {
+		public void on_patient_menu_item_activate (Gtk.MenuItem item) {
 		    show_patient_list_page ();
 		}
 		
 		[CCode (instance_pos = -1)]
-		public void on_doctor_menu_item_activate (MenuItem item) {
+		public void on_doctor_menu_item_activate (Gtk.MenuItem item) {
 		    show_doctor_list_page ();
 		}
 		
 		[CCode (instance_pos = -1)]
-		public void on_home_menu_item_activate (MenuItem item) {
+		public void on_home_menu_item_activate (Gtk.MenuItem item) {
 		    show_start_page();
 		}
 		
 		[CCode (instance_pos = -1)]
-		public void on_search_medicines_menu_item_activate (MenuItem item) {
+		public void on_search_medicines_menu_item_activate (Gtk.MenuItem item) {
 		    set_page (medicine_search_page);
         }
 		
@@ -260,7 +260,7 @@
 		}
 		
 		[CCode (instance_pos = -1)]
-		public void on_settings_menu_item_activate (MenuItem item) {
+		public void on_settings_menu_item_activate (Gtk.MenuItem item) {
 		    if (settings_manager == null){
     		    settings_manager = new SettingsManager (resource_manager);
     		    /* Dereference the window on release to make it be freed and
@@ -275,7 +275,7 @@
 		}
 		
 		[CCode (instance_pos = -1)]
-		public void on_help_menu_item_activate (MenuItem item) {
+		public void on_help_menu_item_activate (Gtk.MenuItem item) {
 		    try {
     		    Gtk.show_uri (this.window.get_screen (), "ghelp:clinica", Gdk.CURRENT_TIME);
     		} catch (GLib.Error e) {
diff -Naur clinica-0.2.1.orig/libclinica/VisitDetail.vala clinica-0.2.1/libclinica/VisitDetail.vala
--- clinica-0.2.1.orig/libclinica/VisitDetail.vala	2011-12-09 18:29:19.000000000 +0100
+++ clinica-0.2.1/libclinica/VisitDetail.vala	2012-10-16 16:53:08.000000000 +0200
@@ -87,10 +87,10 @@
          */
         private bool on_button_press_event (Widget widget, EventButton event) {
             if (event.button == 3 && event.type == EventType.BUTTON_PRESS) {
-                var popup_menu = new Menu ();
+                var popup_menu = new Gtk.Menu ();
                 popup_menu.attach_to_widget (this, null);
                 
-                var open_visit_menu_item = new MenuItem.with_label (_("Show details"));
+                var open_visit_menu_item = new Gtk.MenuItem.with_label (_("Show details"));
                 open_visit_menu_item.activate.connect (on_open_visit_menu_item_activate);
                 popup_menu.append (open_visit_menu_item);
                 
@@ -102,7 +102,7 @@
             return false;
         }
         
-        private void on_open_visit_menu_item_activate (MenuItem? item = null) {
+        private void on_open_visit_menu_item_activate (Gtk.MenuItem? item = null) {
             var visit_window = resource_manager.user_interface.show_visit_window (associated_visit.patient);
             visit_window.select_visit (associated_visit);
         }
