diff -Naur Cinnamon-1.8.8.orig/files/usr/share/cinnamon/applets/sound@cinnamon.org/applet.js Cinnamon-1.8.8/files/usr/share/cinnamon/applets/sound@cinnamon.org/applet.js
--- Cinnamon-1.8.8.orig/files/usr/share/cinnamon/applets/sound@cinnamon.org/applet.js	2013-08-14 16:44:19.955273614 +0000
+++ Cinnamon-1.8.8/files/usr/share/cinnamon/applets/sound@cinnamon.org/applet.js	2013-08-14 16:46:34.171029992 +0000
@@ -600,13 +600,6 @@
 
             // menu not showed by default
             this._players = {};
-            // watch players
-            for (var p=0; p<compatible_players.length; p++) {
-                Gio.DBus.session.watch_name('org.mpris.MediaPlayer2.'+compatible_players[p], Gio.BusNameWatcherFlags.NONE,
-                    Lang.bind(this, this._addPlayer),
-                    Lang.bind(this, this._removePlayer)
-                );
-            }
 
             this._control = new Gvc.MixerControl({ name: 'Cinnamon Volume Control' });
             this._control.connect('state-changed', Lang.bind(this, this._onControlStateChanged));
