diff -wbBur gcc-4.9.0/gcc/graphite-clast-to-gimple.c gcc-4.9.0.my/gcc/graphite-clast-to-gimple.c
--- gcc-4.9.0/gcc/graphite-clast-to-gimple.c	2014-03-03 15:39:22.000000000 +0400
+++ gcc-4.9.0.my/gcc/graphite-clast-to-gimple.c	2014-06-09 15:23:14.858543000 +0400
@@ -28,6 +28,8 @@
 #include <isl/constraint.h>
 #include <isl/ilp.h>
 #include <isl/aff.h>
+#include <isl/deprecated/int.h>
+#include <isl/deprecated/ilp_int.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
 #endif
diff -wbBur gcc-4.9.0/gcc/graphite-interchange.c gcc-4.9.0.my/gcc/graphite-interchange.c
--- gcc-4.9.0/gcc/graphite-interchange.c	2014-01-03 02:23:26.000000000 +0400
+++ gcc-4.9.0.my/gcc/graphite-interchange.c	2014-06-09 15:24:02.275209122 +0400
@@ -29,6 +29,9 @@
 #include <isl/map.h>
 #include <isl/union_map.h>
 #include <isl/ilp.h>
+#include <isl/deprecated/int.h>
+#include <isl/deprecated/ilp_int.h>
+#include <isl/deprecated/constraint_int.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
 #endif
diff -wbBur gcc-4.9.0/gcc/graphite-optimize-isl.c gcc-4.9.0.my/gcc/graphite-optimize-isl.c
--- gcc-4.9.0/gcc/graphite-optimize-isl.c	2014-01-03 02:23:26.000000000 +0400
+++ gcc-4.9.0.my/gcc/graphite-optimize-isl.c	2014-06-09 15:28:20.325206155 +0400
@@ -28,6 +28,10 @@
 #include <isl/band.h>
 #include <isl/aff.h>
 #include <isl/options.h>
+#include <isl/deprecated/int.h>
+#include <isl/deprecated/ilp_int.h>
+#include <isl/deprecated/aff_int.h>
+#include <isl/deprecated/constraint_int.h>
 #endif
 
 #include "system.h"
@@ -373,7 +377,7 @@
 	{
 	  for (i = ScheduleDimensions - 1 ;  i >= 0 ; i--)
 	    {
-	      if (isl_band_member_is_zero_distance (Band, i))
+	      if (isl_band_member_is_coincident (Band, i))
 		{
 		  isl_map *TileMap;
 		  isl_union_map *TileUMap;
diff -wbBur gcc-4.9.0/gcc/graphite-poly.c gcc-4.9.0.my/gcc/graphite-poly.c
--- gcc-4.9.0/gcc/graphite-poly.c	2014-01-03 02:23:26.000000000 +0400
+++ gcc-4.9.0.my/gcc/graphite-poly.c	2014-06-09 15:31:04.541870933 +0400
@@ -28,6 +28,10 @@
 #include <isl/constraint.h>
 #include <isl/ilp.h>
 #include <isl/aff.h>
+#include <isl/deprecated/int.h>
+#include <isl/deprecated/ilp_int.h>
+#include <isl/deprecated/aff_int.h>
+#include <isl/deprecated/constraint_int.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
 #endif
diff -wbBur gcc-4.9.0/gcc/graphite-sese-to-poly.c gcc-4.9.0.my/gcc/graphite-sese-to-poly.c
--- gcc-4.9.0/gcc/graphite-sese-to-poly.c	2014-04-08 14:59:40.000000000 +0400
+++ gcc-4.9.0.my/gcc/graphite-sese-to-poly.c	2014-06-09 15:31:21.731870661 +0400
@@ -26,6 +26,10 @@
 #include <isl/union_map.h>
 #include <isl/constraint.h>
 #include <isl/aff.h>
+#include <isl/deprecated/int.h>
+#include <isl/deprecated/ilp_int.h>
+#include <isl/deprecated/aff_int.h>
+#include <isl/deprecated/constraint_int.h>
 #include <cloog/cloog.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
