diff -wbBur gq-1.3.4/src/glib-helpers.h gq-1.3.4.my/src/glib-helpers.h
--- gq-1.3.4/src/glib-helpers.h	2008-01-09 19:35:12.000000000 +0300
+++ gq-1.3.4.my/src/glib-helpers.h	2013-12-25 18:53:46.695364792 +0400
@@ -24,7 +24,7 @@
 #ifndef GLIB_HELPERS_H
 #define GLIB_HELPERS_H
 
-#include <glib/glist.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
diff -wbBur gq-1.3.4/src/gq-hash.h gq-1.3.4.my/src/gq-hash.h
--- gq-1.3.4/src/gq-hash.h	2008-01-09 19:35:12.000000000 +0300
+++ gq-1.3.4.my/src/gq-hash.h	2013-12-25 18:54:12.485364495 +0400
@@ -24,7 +24,7 @@
 #ifndef GQ_ENCRYPTION_H
 #define GQ_ENCRYPTION_H
 
-#include <glib/garray.h>
+#include <glib.h>
 #include "gq-utilities.h"
 
 G_BEGIN_DECLS
diff -wbBur gq-1.3.4/src/gq-ldap.h gq-1.3.4.my/src/gq-ldap.h
--- gq-1.3.4/src/gq-ldap.h	2008-02-14 00:15:05.000000000 +0300
+++ gq-1.3.4.my/src/gq-ldap.h	2013-12-25 16:28:25.672131705 +0400
@@ -24,7 +24,7 @@
 #ifndef GQ_LDAP_H
 #define GQ_LDAP_H
 
-
+#define LDAP_DEPRECATED 1
 
 #include <ldap.h>
 
diff -wbBur gq-1.3.4/src/gq-main-loop.h gq-1.3.4.my/src/gq-main-loop.h
--- gq-1.3.4/src/gq-main-loop.h	2008-01-09 19:35:12.000000000 +0300
+++ gq-1.3.4.my/src/gq-main-loop.h	2013-12-25 18:54:29.135364303 +0400
@@ -24,7 +24,7 @@
 #ifndef GQ_MAIN_LOOP_H
 #define GQ_MAIN_LOOP_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
diff -wbBur gq-1.3.4/src/gq-marshallers.c gq-1.3.4.my/src/gq-marshallers.c
--- gq-1.3.4/src/gq-marshallers.c	2008-01-09 20:44:57.000000000 +0300
+++ gq-1.3.4.my/src/gq-marshallers.c	2013-12-25 16:28:26.202131699 +0400
@@ -5,7 +5,7 @@
 
 #ifdef G_ENABLE_DEBUG
 #define g_marshal_value_peek_boolean(v)  g_value_get_boolean (v)
-#define g_marshal_value_peek_char(v)     g_value_get_char (v)
+#define g_marshal_value_peek_char(v)     g_value_get_schar (v)
 #define g_marshal_value_peek_uchar(v)    g_value_get_uchar (v)
 #define g_marshal_value_peek_int(v)      g_value_get_int (v)
 #define g_marshal_value_peek_uint(v)     g_value_get_uint (v)
@@ -22,6 +22,7 @@
 #define g_marshal_value_peek_boxed(v)    g_value_get_boxed (v)
 #define g_marshal_value_peek_pointer(v)  g_value_get_pointer (v)
 #define g_marshal_value_peek_object(v)   g_value_get_object (v)
+#define g_marshal_value_peek_variant(v)  g_value_get_variant (v)
 #else /* !G_ENABLE_DEBUG */
 /* WARNING: This code accesses GValues directly, which is UNSUPPORTED API.
  *          Do not access GValues directly in your code. Instead, use the
@@ -45,16 +46,17 @@
 #define g_marshal_value_peek_boxed(v)    (v)->data[0].v_pointer
 #define g_marshal_value_peek_pointer(v)  (v)->data[0].v_pointer
 #define g_marshal_value_peek_object(v)   (v)->data[0].v_pointer
+#define g_marshal_value_peek_variant(v)  (v)->data[0].v_pointer
 #endif /* !G_ENABLE_DEBUG */
 
 
 /* VOID:OBJECT,INT (gq-marshallers.list:1) */
 void
 gq_marshal_VOID__OBJECT_INT (GClosure     *closure,
-                             GValue       *return_value,
+                             GValue       *return_value G_GNUC_UNUSED,
                              guint         n_param_values,
                              const GValue *param_values,
-                             gpointer      invocation_hint,
+                             gpointer      invocation_hint G_GNUC_UNUSED,
                              gpointer      marshal_data)
 {
   typedef void (*GMarshalFunc_VOID__OBJECT_INT) (gpointer     data1,
diff -wbBur gq-1.3.4/src/gq-stack.h gq-1.3.4.my/src/gq-stack.h
--- gq-1.3.4/src/gq-stack.h	2008-01-09 19:35:12.000000000 +0300
+++ gq-1.3.4.my/src/gq-stack.h	2013-12-25 18:54:46.318697439 +0400
@@ -24,7 +24,7 @@
 #ifndef GQ_STACK_H
 #define GQ_STACK_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
diff -wbBur gq-1.3.4/src/gq-tab.h gq-1.3.4.my/src/gq-tab.h
--- gq-1.3.4/src/gq-tab.h	2008-01-09 19:35:12.000000000 +0300
+++ gq-1.3.4.my/src/gq-tab.h	2013-12-25 18:55:10.392030496 +0400
@@ -24,7 +24,7 @@
 #ifndef GQ_TAB_H
 #define GQ_TAB_H
 
-#include <glib/gmacros.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
diff -wbBur gq-1.3.4/src/syntax.h gq-1.3.4.my/src/syntax.h
--- gq-1.3.4/src/syntax.h	2008-01-09 19:35:13.000000000 +0300
+++ gq-1.3.4.my/src/syntax.h	2013-12-25 18:53:19.428698438 +0400
@@ -27,7 +27,7 @@
 #ifndef GQ_SYNTAX_H_INCLUDED
 #define GQ_SYNTAX_H_INCLUDED
 
-#include <glib/gmacros.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
