From 74d97c59d381253cca74c862689b4e22a2708f9f Mon Sep 17 00:00:00 2001
From: Maxime Gauduin <alucryd@archlinux.org>
Date: Sat, 4 Apr 2020 07:20:40 +0200
Subject: [PATCH] use system flags

---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index e4767d84..0d0fd5a3 100644
--- a/Makefile
+++ b/Makefile
@@ -5,8 +5,8 @@ MANDIR ?= $(PREFIX)/share/man
 CC ?= gcc
 CXX ?= g++
 
-CFLAGS ?= -O2
-CXXFLAGS ?= $(CFLAGS)
+CFLAGS ?= ${CFLAGS}
+CXXFLAGS ?= ${CXXFLAGS}
 
 SRC_CFLAGS += -W -Wall -Wextra -Wno-implicit-function-declaration -DNDEBUG=1
 SRC_CXXFLAGS += -W -Wall -Wextra -std=c++11 -Izopfli/src -I7zip -DNDEBUG=1 \
@@ -31,7 +31,7 @@ ZOPFLI_C_OBJ = $(ZOPFLI_C_SRC:.c=.o)
 	$(CC) -c $(SRC_CFLAGS) $(CFLAGS) -o $@ $<
 
 maxcso: $(SRC_CXX_OBJ) $(CLI_CXX_OBJ) $(ZOPFLI_C_OBJ) 7zip/7zip.a
-	$(CXX) -o $@ $(SRC_CXXFLAGS) $(CXXFLAGS) $^ -luv -llz4 -lz
+	$(CXX) -o $@ $(SRC_CXXFLAGS) $(CXXFLAGS) $^ ${LDFLAGS} -luv -llz4 -lz
 
 7zip/7zip.a:
 	$(MAKE) -C 7zip 7zip.a
