diff --git a/source/Lib/TLibCommon/ContextModel.h b/source/Lib/TLibCommon/ContextModel.h
index 5b9d7629..1ed41c55 100644
--- a/source/Lib/TLibCommon/ContextModel.h
+++ b/source/Lib/TLibCommon/ContextModel.h
@@ -53,12 +53,16 @@
 // Class definition
 // ====================================================================================================================
 
+#define GCC_9_1_0_WARNING_FIX 1
+
 /// context model class
 class ContextModel
 {
 public:
+#if !GCC_9_1_0_WARNING_FIX
   ContextModel  ()                        { m_ucState = 0; m_binsCoded = 0; }
   ~ContextModel ()                        {}
+#endif
 
   UChar getState  ()                { return ( m_ucState >> 1 ); }                    ///< get current state
   UChar getMps    ()                { return ( m_ucState  & 1 ); }                    ///< get curret MPS
@@ -90,7 +94,11 @@ public:
   UInt getBinsCoded()           { return m_binsCoded;   }
 
 private:
+#if GCC_9_1_0_WARNING_FIX
+  UChar         m_ucState = 0;                                                              ///< internal state variable
+#else
   UChar         m_ucState;                                                                  ///< internal state variable
+#endif
 
   static const  UInt  m_totalStates = (1 << CONTEXT_STATE_BITS) * 2; //*2 for MPS = [0|1]
   static const  UChar m_aucNextStateMPS[m_totalStates];
@@ -99,7 +107,11 @@ private:
 #if FAST_BIT_EST
   static UChar m_nextState[m_totalStates][2 /*MPS = [0|1]*/];
 #endif
+#if GCC_9_1_0_WARNING_FIX
+  UInt          m_binsCoded = 0;
+#else
   UInt          m_binsCoded;
+#endif
 };
 
 //! \}

