--- daemon/mainwin.cpp.orig	2010-05-04 11:17:40.000000000 +0200
+++ daemon/mainwin.cpp	2010-05-18 01:24:05.560974294 +0200
@@ -647,7 +647,7 @@
 			fs+=" "+sett->value("FontStyle").toString();
 			fs+=" "+sett->value("FontSize").toString();
 			fs=fs.simplified();
-			out<<"gtk-font-name=\""<<sett->value("FontFamily").toString()<<" "<<" "<<sett->value("FontSize").toString();
+			out<<"gtk-font-name=\""<<sett->value("FontFamily").toString()<<" "<<" "<<sett->value("FontSize").toString()<<"\""<<endl;
 		}
 		else
 		{	fs=QApplication::font().family();
