From 25da5747d55adceb7ccebba4e82af295eb397cb7 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Thu, 1 Jun 2017 16:15:59 +0200
Subject: Add new status "Running hooks"

Signed-off-by: Christian Hesse <mail@eworm.de>
Signed-off-by: Richard Hughes <richard@hughsie.com>
---
 src/gpk-enum.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/gpk-enum.c b/src/gpk-enum.c
index 09a1118..2bf8af5 100644
--- a/src/gpk-enum.c
+++ b/src/gpk-enum.c
@@ -95,6 +95,9 @@ static const PkEnumMatch enum_status_icon_name[] = {
 	{PK_STATUS_ENUM_CHECK_EXECUTABLE_FILES,	"pk-package-info"},
 	{PK_STATUS_ENUM_CHECK_LIBRARIES,	"pk-package-info"},
 	{PK_STATUS_ENUM_COPY_FILES,		"pk-package-info"},
+#if PK_CHECK_VERSION(1,1,6)
+	{PK_STATUS_ENUM_RUN_HOOK,		"pk-setup"},
+#endif
 	{0, NULL}
 };
 
@@ -888,6 +891,12 @@ gpk_status_enum_to_localised_text (PkStatusEnum status)
 		/* TRANSLATORS: we are copying package files to prepare to install */
 		text = _("Copying files");
 		break;
+#if PK_CHECK_VERSION(1,1,6)
+	case PK_STATUS_ENUM_RUN_HOOK:
+		/* TRANSLATORS: we are running hooks pre or post transaction */
+		text = _("Running hooks");
+		break;
+#endif
 	default:
 		g_warning ("status unrecognized: %s", pk_status_enum_to_string (status));
 	}
