--- a/src/liblwqq/http.c
+++ b/src/liblwqq/http.c
@@ -515,7 +515,9 @@ static int multi_timer_cb(CURLM *multi, long timeout_ms, void *userp)
     }
     if(g->still_running){
         lwqq_log(LOG_NOTICE,"still running:%d,timeout:%ld\n",g->still_running,timeout_ms);
+	#if USE_DEBUG
         if(g->still_running>1)lwqq_gdb_whats_running();
+	#endif
     }
     //close time clock
     //this should always return 0 this is curl!!
