From 984207e348fe9ef1444cbec897cb8c429859a9c0 Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Tue, 25 Dec 2012 04:55:30 +0200
Subject: [PATCH] Fix llvm/IRBuilder.h include when using LLVM 3.2

---
 gen/passes/GarbageCollect2Stack.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gen/passes/GarbageCollect2Stack.cpp b/gen/passes/GarbageCollect2Stack.cpp
index f20bdd4..9927615 100644
--- a/gen/passes/GarbageCollect2Stack.cpp
+++ b/gen/passes/GarbageCollect2Stack.cpp
@@ -27,7 +27,7 @@
 #include "llvm/Intrinsics.h"
 #include "llvm/Support/CallSite.h"
 #include "llvm/Support/CommandLine.h"
-#if LDC_LLVM_VER >= 303
+#if LDC_LLVM_VER >= 302
 #include "llvm/IRBuilder.h"
 #else
 #include "llvm/Support/IRBuilder.h"
-- 
1.8.0.2

