diff --git a/futatabi/export.cpp b/futatabi/export.cpp
index 7833f91..ca6a558 100644
--- a/futatabi/export.cpp
+++ b/futatabi/export.cpp
@@ -17,6 +17,7 @@
 
 extern "C" {
 #include <libavformat/avformat.h>
+#include <libavutil/channel_layout.h>
 }
 
 using namespace std;
diff --git a/futatabi/video_stream.cpp b/futatabi/video_stream.cpp
index 9647836..6ea3bf1 100644
--- a/futatabi/video_stream.cpp
+++ b/futatabi/video_stream.cpp
@@ -3,6 +3,7 @@
 extern "C" {
 #include <libavformat/avformat.h>
 #include <libavformat/avio.h>
+#include <libavutil/channel_layout.h>
 }
 
 #include "chroma_subsampler.h"
diff --git a/nageru/mjpeg_encoder.cpp b/nageru/mjpeg_encoder.cpp
index 033f67a..a6ae011 100644
--- a/nageru/mjpeg_encoder.cpp
+++ b/nageru/mjpeg_encoder.cpp
@@ -9,6 +9,7 @@
 
 extern "C" {
 #include <libavformat/avformat.h>
+#include <libavutil/channel_layout.h>
 }
 
 #include "defs.h"
