diff -wbBur dbmail-3.2.3/src/dm_imapsession.c dbmail-3.2.3.q/src/dm_imapsession.c
--- dbmail-3.2.3/src/dm_imapsession.c	2015-01-25 13:20:50.000000000 +0300
+++ dbmail-3.2.3.q/src/dm_imapsession.c	2015-08-05 16:43:36.489596671 +0300
@@ -988,7 +988,7 @@
 
 	snprintf(query, DEF_QUERYSIZE-1, "SELECT message_idnr,envelope "
 			"FROM %senvelope e "
-			"LEFT JOIN %smessages m USING (physmessage_id) "
+			"LEFT JOIN %smessages m "
 			"WHERE m.mailbox_idnr = %" PRIu64 " "
 			"AND message_idnr %s",
 			DBPFX, DBPFX,  
diff -wbBur dbmail-3.2.3/src/dm_mailbox.c dbmail-3.2.3.q/src/dm_mailbox.c
--- dbmail-3.2.3/src/dm_mailbox.c	2015-01-25 13:20:50.000000000 +0300
+++ dbmail-3.2.3.q/src/dm_mailbox.c	2015-08-05 16:41:58.912931125 +0300
@@ -317,7 +317,7 @@
 		stmt = db_stmt_prepare(c,
 				"SELECT min(m.message_idnr),v.sortfield "
 				"FROM %smessages m "
-				"LEFT JOIN %sheader h USING (physmessage_id) "
+				"LEFT JOIN %sheader h "
 				"LEFT JOIN %sheadername n ON h.headername_id = n.id "
 				"LEFT JOIN %sheadervalue v ON h.headervalue_id = v.id "
 				"WHERE m.mailbox_idnr=? "
@@ -355,7 +355,7 @@
 		stmt = db_stmt_prepare(c, 
 				"SELECT m.message_idnr,v.sortfield "
 				"FROM %smessages m "
-				"LEFT JOIN %sheader h USING (physmessage_id) "
+				"LEFT JOIN %sheader h "
 				"LEFT JOIN %sheadername n ON h.headername_id = n.id "
 				"LEFT JOIN %sheadervalue v ON h.headervalue_id = v.id "
 				"WHERE m.mailbox_idnr = ? "
@@ -1268,7 +1268,7 @@
 					op = "=";
 
 				p_string_printf(q,"SELECT message_idnr FROM %smessages m "
-						"LEFT JOIN %sheader h USING (physmessage_id) "
+						"LEFT JOIN %sheader h "
 						"LEFT JOIN %sheadername n ON h.headername_id = n.id "
 						"LEFT JOIN %sheadervalue v ON h.headervalue_id = v.id "
 						"WHERE m.mailbox_idnr=? AND m.status IN (?,?) "
@@ -1290,7 +1290,7 @@
 			case IST_HDR:
 			
 			p_string_printf(q, "SELECT message_idnr FROM %smessages m "
-					"LEFT JOIN %sheader h USING (physmessage_id) "
+					"LEFT JOIN %sheader h "
 					"LEFT JOIN %sheadername n ON h.headername_id = n.id "
 					"LEFT JOIN %sheadervalue v ON h.headervalue_id = v.id "
 					"WHERE mailbox_idnr=? AND status IN (?,?) "
