--- a/main/main.js
+++ b/main/main.js
@@ -112,7 +112,7 @@
     minWidth: 320,
     minHeight: 350,
     titleBarStyle: 'hiddenInset',
-    icon: __dirname + '/icons/icon256.png',
+    icon: '/usr/share/icons/hicolor/256x256/apps/min.png',
     frame: process.platform !== 'win32',
     backgroundColor: '#fff', // the value of this is ignored, but setting it seems to work around https://github.com/electron/electron/issues/10559
     webPreferences: {
