diff -upr ponyc-0.21.3.orig/src/libponyc/codegen/host.cc ponyc-0.21.3/src/libponyc/codegen/host.cc
--- ponyc-0.21.3.orig/src/libponyc/codegen/host.cc	2018-01-14 18:46:16.000000000 +0200
+++ ponyc-0.21.3/src/libponyc/codegen/host.cc	2018-03-19 02:30:12.452611178 +0200
@@ -79,7 +79,11 @@ char* LLVMGetHostCPUFeatures()
 
 void LLVMSetUnsafeAlgebra(LLVMValueRef inst)
 {
+#if PONY_LLVM >= 600
+  unwrap<Instruction>(inst)->setFast(true);
+#else
   unwrap<Instruction>(inst)->setHasUnsafeAlgebra(true);
+#endif
 }
 
 void LLVMSetNoUnsignedWrap(LLVMValueRef inst)
