diff -Naur orig/dosemu-1.4.0/src/base/init/config.c modified/dosemu-1.4.0/src/base/init/config.c
--- orig/src/base/init/config.c	2007-05-04 22:19:41.000000000 +0800
+++ modified/src/base/init/config.c	2012-12-06 10:05:31.780346032 +0800
@@ -480,6 +480,8 @@
       case 5: case 586:
       case 6: case 686:
       case 15:
+      case 16:
+      case 17:
         config.realcpu = CPU_586;
         cpuflags = get_proc_string_by_key("features");
         if (!cpuflags) {
