--- zsnes_1_51/src/linux/sdllink.c.orig   2008-07-27 22:13:02.000000000 +0200
+++ zsnes_1_51/src/linux/sdllink.c   2009-02-10 00:50:17.000000000 +0100
@@ -307,6 +307,8 @@
           case SDL_HAT_UP:
             pressed[offset + 3] = 1;
             pressed[offset + 2] = 0;
+            pressed[offset + 1] = 0;
+            pressed[offset + 0] = 0;
             break;
           case SDL_HAT_RIGHTUP:
             pressed[offset] = 1;
@@ -317,6 +317,8 @@
           case SDL_HAT_RIGHT:
             pressed[offset] = 1;
             pressed[offset + 1] = 0;
+            pressed[offset + 2] = 0;
+            pressed[offset + 3] = 0;
             break;
           case SDL_HAT_RIGHTDOWN:
             pressed[offset] = 1;
@@ -326,6 +326,8 @@
             break;
           case SDL_HAT_DOWN:
             pressed[offset + 2] = 1;
+            pressed[offset + 0] = 0;
+            pressed[offset + 1] = 0;
             pressed[offset + 3] = 0;
             break;
           case SDL_HAT_LEFTDOWN:
@@ -337,6 +337,8 @@
           case SDL_HAT_LEFT:
             pressed[offset + 1] = 1;
             pressed[offset] = 0;
+            pressed[offset + 2] = 0;
+            pressed[offset + 3] = 0;
             break;
           case SDL_HAT_LEFTUP:
             pressed[offset + 1] = 1;
