https://bugs.gentoo.org/show_bug.cgi?id=361359

Index: gpac/modules/img_in/jp2_dec.c
===================================================================
--- gpac.orig/modules/img_in/jp2_dec.c
+++ gpac/modules/img_in/jp2_dec.c
@@ -340,9 +340,16 @@ static GF_Err JP2_ProcessData(GF_MediaDe
 	return GF_OK;
 }
 
+char opj_version_buffer[500];
+
 static const char *JP2_GetCodecName(GF_BaseDecoder *dec)
 {
+#ifndef OPENJPEG_VERSION
+	snprintf(opj_version_buffer, sizeof(opj_version_buffer), "OpenJPEG %s", opj_version());
+	return opj_version_buffer;
+#else
 	return "OpenJPEG "OPENJPEG_VERSION ;
+#endif
 }
 
 
