diff -Naur shadowsocks-qt5-2.8.0.orig/src/ss-qt5.pri shadowsocks-qt5-2.8.0/src/ss-qt5.pri
--- shadowsocks-qt5-2.8.0.orig/src/ss-qt5.pri	2016-12-14 22:56:30.000000000 +0100
+++ shadowsocks-qt5-2.8.0/src/ss-qt5.pri	2017-01-25 04:42:25.227070273 +0100
@@ -68,8 +68,4 @@
 unix: {
     CONFIG    += link_pkgconfig
     PKGCONFIG += libqrencode QtShadowsocks botan-$$BOTAN_VER zbar
-    !macx: {
-        PKGCONFIG += gtk+-2.0 appindicator-0.1
-        DEFINES   += USE_APP_INDICATOR
-    }
 }
