--- qt5dxcb-plugin-1.1.5/platformplugin/windoweventhook.cpp.orig	2017-12-07 16:01:23.609087845 +0000
+++ qt5dxcb-plugin-1.1.5/platformplugin/windoweventhook.cpp	2017-12-07 16:02:33.895912926 +0000
@@ -125,7 +125,7 @@
             dropData = drag->currentDrag()->mimeData();
             supported_drop_actions = Qt::DropActions(l[4]);
         } else {
-            dropData = drag->platformDropData();
+            dropData = drag->dragMimeData();
             supported_drop_actions = drag->accepted_drop_action;
 
             // Drop coming from another app? Update keyboard modifiers.
