diff -wbBur buildbot-0.8.10.org/buildbot/db/enginestrategy.py buildbot-0.8.10/buildbot/db/enginestrategy.py
--- buildbot-0.8.10.org/buildbot/db/enginestrategy.py	2014-12-02 19:53:49.000000000 +0300
+++ buildbot-0.8.10/buildbot/db/enginestrategy.py	2015-01-06 00:25:47.462298489 +0300
@@ -56,7 +56,7 @@
                 version = "0.6"
         except:
             version = "0.0"
-    return tuple(map(int, version.split('.')))
+    return tuple(map(int, version.split('.')[0:3]))
 
 
 class BuildbotEngineStrategy(strategies.ThreadLocalEngineStrategy):
diff -wbBur buildbot-0.8.10.org/buildbot/db/model.py buildbot-0.8.10/buildbot/db/model.py
--- buildbot-0.8.10.org/buildbot/db/model.py	2014-12-02 19:53:49.000000000 +0300
+++ buildbot-0.8.10/buildbot/db/model.py	2015-01-06 00:25:59.228965020 +0300
@@ -499,7 +499,7 @@
                         version = "0.6"
                 except:
                     version = "0.0"
-            version_tup = tuple(map(int, version.split('.')))
+            version_tup = tuple(map(int, version.split('.')[0:3]))
             log.msg("using SQLAlchemy-Migrate version %s" % (version,))
             if version_tup < (0, 6, 1):
                 raise RuntimeError("You are using SQLAlchemy-Migrate %s. "
