diff -ur cups-1.5_2012-05-20_223115.3/config-scripts/cups-common.m4 cups-1.5.3/config-scripts/cups-common.m4
--- cups-1.5_2012-05-20_223115.3/config-scripts/cups-common.m4	2012-05-20 22:31:15.702071734 +0200
+++ cups-1.5.3/config-scripts/cups-common.m4	2012-05-20 22:31:29.395622693 +0200
@@ -216,7 +216,7 @@
 AC_SUBST(LIBUSB)
 
 if test "x$PKGCONFIG" != x; then
-	if test x$enable_libusb = xyes -o $uname != Darwin; then
+	if test x$enable_libusb = xyes -a $uname != Darwin; then
 		AC_MSG_CHECKING(for libusb-1.0)
 		if $PKGCONFIG --exists libusb-1.0; then
 			AC_MSG_RESULT(yes)
