--- stk-4.4.2/configure.ac	2010-02-04 15:52:39.000000000 -0500
+++ configure.ac	2010-03-11 23:55:44.000000000 -0500
@@ -51,7 +51,7 @@
 AC_ARG_ENABLE(debug,
   [  --enable-debug = enable various debug output],
   [AC_SUBST( cppflag, ["-D_STK_DEBUG_ -D__RTAUDIO_DEBUG__ -D__RTMIDI_DEBUG__"] ) AC_SUBST( cxxflag, ["-g"] ) AC_SUBST( object_path, [Debug] ) AC_MSG_RESULT(yes)],
-  [AC_SUBST( cppflag, [] ) AC_SUBST( cxxflag, [-O3] ) AC_SUBST( object_path, [Release] ) AC_MSG_RESULT(no)])
+  [AC_SUBST( cppflag, [] ) AC_SUBST( cxxflag, [] ) AC_SUBST( object_path, [Release] ) AC_MSG_RESULT(no)])
 
 # Checks for functions
 if test $realtime = yes; then
@@ -63,12 +63,7 @@
 CPPFLAGS="$CPPFLAGS $cppflag"
 
 # For debugging and optimization ... overwrite default because it has both -g and -O2
-CXXFLAGS="$cxxflag"
-
-# Check compiler and use -Wall if gnu.
-if [test $GXX = "yes" ;] then
-  AC_SUBST( cxxflag, [-Wall] )
-fi
+CXXFLAGS="$cxxflag $CXXFLAGS"
 
 CXXFLAGS="$CXXFLAGS $cxxflag"
 
