diff -Naur supertux2-0.3.3.orig/external/squirrel/squirrel/sqmem.h supertux2-0.3.3/external/squirrel/squirrel/sqmem.h
--- supertux2-0.3.3.orig/external/squirrel/squirrel/sqmem.h	1970-01-01 01:00:00.000000000 +0100
+++ supertux2-0.3.3/external/squirrel/squirrel/sqmem.h	2012-05-01 11:59:37.000000000 +0200
@@ -0,0 +1,8 @@
+/*	see copyright notice in squirrel.h */
+#ifndef _SQMEM_H_
+#define _SQMEM_H_
+
+void *sq_vm_malloc(SQUnsignedInteger size);
+void *sq_vm_realloc(void *p,SQUnsignedInteger oldsize,SQUnsignedInteger size);
+void sq_vm_free(void *p,SQUnsignedInteger size);
+#endif //_SQMEM_H_
diff -Naur supertux2-0.3.3.orig/external/squirrel/squirrel/sqstate.h supertux2-0.3.3/external/squirrel/squirrel/sqstate.h
--- supertux2-0.3.3.orig/external/squirrel/squirrel/sqstate.h	2010-03-01 13:40:58.000000000 +0100
+++ supertux2-0.3.3/external/squirrel/squirrel/sqstate.h	2012-05-01 11:59:37.000000000 +0200
@@ -136,8 +136,4 @@
 extern SQObjectPtr _minusone_;
 
 bool CompileTypemask(SQIntVec &res,const SQChar *typemask);
-
-void *sq_vm_malloc(SQUnsignedInteger size);
-void *sq_vm_realloc(void *p,SQUnsignedInteger oldsize,SQUnsignedInteger size);
-void sq_vm_free(void *p,SQUnsignedInteger size);
 #endif //_SQSTATE_H_
diff -Naur supertux2-0.3.3.orig/external/squirrel/squirrel/squtils.h supertux2-0.3.3/external/squirrel/squirrel/squtils.h
--- supertux2-0.3.3.orig/external/squirrel/squirrel/squtils.h	2010-03-01 13:40:58.000000000 +0100
+++ supertux2-0.3.3/external/squirrel/squirrel/squtils.h	2012-05-01 11:59:37.000000000 +0200
@@ -2,6 +2,8 @@
 #ifndef _SQUTILS_H_
 #define _SQUTILS_H_
 
+#include "sqmem.h"
+
 #define sq_new(__ptr,__type) {__ptr=(__type *)sq_vm_malloc(sizeof(__type));new (__ptr) __type;}
 #define sq_delete(__ptr,__type) {__ptr->~__type();sq_vm_free(__ptr,sizeof(__type));}
 #define SQ_MALLOC(__size) sq_vm_malloc((__size));
