diff --git a/platform/build-scripts/src/org/jetbrains/intellij/build/LinuxDistributionCustomizer.kt b/platform/build-scripts/src/org/jetbrains/intellij/build/LinuxDistributionCustomizer.kt
--- a/platform/build-scripts/src/org/jetbrains/intellij/build/LinuxDistributionCustomizer.kt	(revision d2f6e301b539928d66c37f71ee199e9b738e5bb3)
+++ b/platform/build-scripts/src/org/jetbrains/intellij/build/LinuxDistributionCustomizer.kt	(date 1670184559625)
@@ -24,7 +24,7 @@
   /**
    * If {@code true} a separate *-no-jbr.tar.gz artifact without runtime will be produced.
    */
-  var buildTarGzWithoutBundledRuntime = false
+  var buildTarGzWithoutBundledRuntime = true
 
   /**
    * If {@code true}, the only *-no-jbr.tar.gz will be produced, no other binaries for Linux will be built.
