From b041e381e6f2a952383bafe7812df7287754ff27 Mon Sep 17 00:00:00 2001
From: Laurent Carlier <lordheavym@gmail.com>
Date: Tue, 5 Jun 2018 17:40:28 +0200
Subject: [PATCH] [GB.SDL2] fix detection with at least pkgconf 2.4.2

Signed-off-by: Laurent Carlier <lordheavym@gmail.com>
---
 gb.sdl2/configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gb.sdl2/configure.ac b/gb.sdl2/configure.ac
index c22addebc..2406d49f1 100644
--- a/gb.sdl2/configure.ac
+++ b/gb.sdl2/configure.ac
@@ -9,12 +9,12 @@ AC_PROG_LIBTOOL
 
 GB_COMPONENT_PKG_CONFIG(
    sdl2, SDL2, gb.sdl2, [src],
-   sdl2 ">= 2.0.2" SDL2_image ">= 2.0.0" SDL2_ttf ">= 2.0.12"
+   'sdl2 >= 2.0.2' 'SDL2_image >= 2.0.0' 'SDL2_ttf >= 2.0.12'
 )
 
 GB_COMPONENT_PKG_CONFIG(
    sdl2audio, SDL2AUDIO, gb.sdl2.audio, [audio],
-   sdl2 ">= 2.0.2" SDL2_mixer ">= 2.0.0"
+   'sdl2 >= 2.0.2' 'SDL2_mixer >= 2.0.0'
 )
 
 AC_OUTPUT( \
-- 
2.17.1

