--- gambas-3.11.4/gb.pdf/src/CPdfDocument.cpp.orig	2018-11-01 20:05:27.379728413 +0000
+++ gambas-3.11.4/gb.pdf/src/CPdfDocument.cpp	2018-11-01 20:05:51.586459136 +0000
@@ -478,7 +478,7 @@
 	THIS->len=len;
 
 	white[0] = 0xFF; white[1] = 0xFF; white[2] = 0xFF;
-	THIS->dev=new SplashOutputDev(splashModeRGB8, 3, gFalse, white);
+	THIS->dev=new SplashOutputDev(splashModeRGB8, 3, false, white);
 
 	#if POPPLER_VERSION_0_20
 	THIS->dev->startDoc(THIS->doc);
@@ -922,17 +922,17 @@
 	#if POPPLER_VERSION_0_20
 	THIS->page->displaySlice(THIS->dev,72.0*scale,72.0*scale,
 			   rotation,
-			   gFalse,
-			   gTrue,
+			   false,
+			   true,
 			   x,y,w,h,
-			   gFalse);
+			   false);
 	#else
 	THIS->page->displaySlice(THIS->dev,72.0*scale,72.0*scale,
 			   rotation,
-			   gFalse,
-			   gTrue,
+			   false,
+			   true,
 			   x,y,w,h,
-			   gFalse,
+			   false,
 			   THIS->doc->getCatalog ());
 	#endif
 	
@@ -994,11 +994,11 @@
 	h = VARGOPT(H, (int32_t)THIS->page->getMediaHeight());
 
 	#if POPPLER_VERSION_0_20
-	dev = new TextOutputDev (NULL, gTrue, 0, gFalse, gFalse);
-	gfx = THIS->page->createGfx(dev,72.0,72.0,0,gFalse,gTrue,-1, -1, -1, -1, gFalse, NULL, NULL);
+	dev = new TextOutputDev (NULL, true, 0, false, false);
+	gfx = THIS->page->createGfx(dev,72.0,72.0,0,false,true,-1, -1, -1, -1, false, NULL, NULL);
 	#else
-	dev = new TextOutputDev (NULL, gTrue, gFalse, gFalse);
-	gfx = THIS->page->createGfx(dev,72.0,72.0,0,gFalse,gTrue,-1, -1, -1, -1, gFalse,THIS->doc->getCatalog (),NULL, NULL, NULL, NULL);
+	dev = new TextOutputDev (NULL, true, false, false);
+	gfx = THIS->page->createGfx(dev,72.0,72.0,0,false,true,-1, -1, -1, -1, false,THIS->doc->getCatalog (),NULL, NULL, NULL, NULL);
 	#endif
 
 	THIS->page->display(gfx);
@@ -1226,9 +1226,9 @@
 
 	count = 0;
 	#if POPPLER_VERSION_0_20
-	while (textdev->findText (block,nlen,gFalse,gTrue,gTrue,gFalse,sensitive,gFalse,gFalse,&x0,&y0,&x1,&y1))
+	while (textdev->findText (block,nlen,false,true,true,false,sensitive,false,false,&x0,&y0,&x1,&y1))
 	#else
-	while (textdev->findText (block,nlen,gFalse,gTrue,gTrue,gFalse,sensitive,gFalse,&x0,&y0,&x1,&y1))
+	while (textdev->findText (block,nlen,false,true,true,false,sensitive,false,&x0,&y0,&x1,&y1))
 	#endif
 	{
 		if (!THIS->Found)
