diff --git a/gb.qt4/src/CCheckBox.h b/gb.qt4/src/CCheckBox.h
index d20a11d94..b6de0fef7 100644
--- a/gb.qt4/src/CCheckBox.h
+++ b/gb.qt4/src/CCheckBox.h
@@ -24,10 +24,10 @@
 #ifndef __CCHECKBOX_H
 #define __CCHECKBOX_H
 
-#include "gambas.h"
-
 #include <QCheckBox>
 
+#include "gambas.h"
+
 #include "CWidget.h"
 
 #ifndef __CCHECKBOX_CPP
diff --git a/gb.qt4/src/CClipboard.h b/gb.qt4/src/CClipboard.h
index 01b8048f5..b199c03fe 100644
--- a/gb.qt4/src/CClipboard.h
+++ b/gb.qt4/src/CClipboard.h
@@ -24,11 +24,12 @@
 #ifndef __CCLIPBOARD_H
 #define __CCLIPBOARD_H
 
+#include <QMimeData>
+#include <QDropEvent>
+
 #include "gambas.h"
 #include "CWidget.h"
 #include "CPicture.h"
-#include <QMimeData>
-#include <QDropEvent>
 
 typedef
   struct {
diff --git a/gb.qt4/src/CDrawingArea.h b/gb.qt4/src/CDrawingArea.h
index b693b5eb0..9884644bb 100644
--- a/gb.qt4/src/CDrawingArea.h
+++ b/gb.qt4/src/CDrawingArea.h
@@ -24,13 +24,13 @@
 #ifndef __CDRAWINGAREA_H
 #define __CDRAWINGAREA_H
 
-#include "gambas.h"
-
 #include <QPaintEvent>
 #include <QPixmap>
 #include <QEvent>
 //#include <QFrame>
 
+#include "gambas.h"
+
 #include "CWidget.h"
 #include "CContainer.h"
 
diff --git a/gb.qt4/src/CFrame.h b/gb.qt4/src/CFrame.h
index 1075ca99b..d2012d1b6 100644
--- a/gb.qt4/src/CFrame.h
+++ b/gb.qt4/src/CFrame.h
@@ -24,13 +24,13 @@
 #ifndef __CFRAME_H
 #define __CFRAME_H
 
+#include <QGroupBox>
+
 #include "gambas.h"
 
 #include "CWidget.h"
 #include "CContainer.h"
 
-#include <QGroupBox>
-
 #ifndef __CFRAME_CPP
 extern GB_DESC CFrameDesc[];
 #else
diff --git a/gb.qt4/src/CLabel.h b/gb.qt4/src/CLabel.h
index 0ebc5d31c..530b56b51 100644
--- a/gb.qt4/src/CLabel.h
+++ b/gb.qt4/src/CLabel.h
@@ -24,13 +24,13 @@
 #ifndef __CLABEL_H
 #define __CLABEL_H
 
-#include "gambas.h"
-
 #include <QLabel>
 #include <QEvent>
 #include <QPaintEvent>
 #include <QResizeEvent>
 
+#include "gambas.h"
+
 #include "CWidget.h"
 #include "CPicture.h"
 #include "CContainer.h"
diff --git a/gb.qt4/src/CMenu.h b/gb.qt4/src/CMenu.h
index 50cb9b552..7d5d0d897 100644
--- a/gb.qt4/src/CMenu.h
+++ b/gb.qt4/src/CMenu.h
@@ -24,8 +24,6 @@
 #ifndef __CMENU_H
 #define __CMENU_H
 
-#include "gambas.h"
-
 #include <QAction>
 #include <QMenu>
 #include <QMenuBar>
@@ -33,6 +31,8 @@
 #include <QList>
 #include <QPoint>
 
+#include "gambas.h"
+
 #include "CWidget.h"
 #include "CPicture.h"
 
diff --git a/gb.qt4/src/CRadioButton.h b/gb.qt4/src/CRadioButton.h
index f2f65c1dd..6ec0d6f54 100644
--- a/gb.qt4/src/CRadioButton.h
+++ b/gb.qt4/src/CRadioButton.h
@@ -24,10 +24,10 @@
 #ifndef __CRADIOBUTTON_H
 #define __CRADIOBUTTON_H
 
-#include "gambas.h"
-
 #include <QRadioButton>
 
+#include "gambas.h"
+
 #include "CWidget.h"
 
 #ifndef __CRADIOBUTTON_CPP
diff --git a/gb.qt4/src/CSlider.h b/gb.qt4/src/CSlider.h
index d6e1a4d9a..5f0187907 100644
--- a/gb.qt4/src/CSlider.h
+++ b/gb.qt4/src/CSlider.h
@@ -24,11 +24,13 @@
 #ifndef __CSLIDER_H
 #define __CSLIDER_H
 
-#include "gambas.h"
-#include "CWidget.h"
 #include <QSlider>
 #include <QResizeEvent>
 
+#include "gambas.h"
+
+#include "CWidget.h"
+
 #ifndef __CSLIDER_CPP
 
 extern GB_DESC CSliderDesc[];
diff --git a/gb.qt4/src/CTabStrip.h b/gb.qt4/src/CTabStrip.h
index d01d0c117..ca7ba2abb 100644
--- a/gb.qt4/src/CTabStrip.h
+++ b/gb.qt4/src/CTabStrip.h
@@ -24,6 +24,10 @@
 #ifndef __CTABSTRIP_H
 #define __CTABSTRIP_H
 
+#include <QTabWidget>
+#include <QTabBar>
+#include <QEvent>
+
 #include "gambas.h"
 
 #include "CWidget.h"
@@ -31,10 +35,6 @@
 #include "CPicture.h"
 #include "CFont.h"
 
-#include <QTabWidget>
-#include <QTabBar>
-#include <QEvent>
-
 #ifndef __CTABSTRIP_CPP
 extern GB_DESC CTabStripDesc[];
 extern GB_DESC CTabStripContainerDesc[];
diff --git a/gb.qt4/src/CTextArea.h b/gb.qt4/src/CTextArea.h
index 7abefa6fb..a2ffe9f3e 100644
--- a/gb.qt4/src/CTextArea.h
+++ b/gb.qt4/src/CTextArea.h
@@ -24,12 +24,12 @@
 #ifndef __CTEXTAREA_H
 #define __CTEXTAREA_H
 
+#include <QPlainTextEdit>
+
 #include "gambas.h"
 
 #include "CWidget.h"
 
-#include <QPlainTextEdit>
-
 #ifndef __CTEXTAREA_CPP
 extern GB_DESC CTextAreaDesc[];
 extern GB_DESC CTextAreaSelectionDesc[];
diff --git a/gb.qt4/src/CTextBox.h b/gb.qt4/src/CTextBox.h
index 7df59df82..4b9f6e5ce 100644
--- a/gb.qt4/src/CTextBox.h
+++ b/gb.qt4/src/CTextBox.h
@@ -24,11 +24,11 @@
 #ifndef __CTEXTBOX_H
 #define __CTEXTBOX_H
 
-#include "gambas.h"
-
 #include <QComboBox>
 #include <QEvent>
 
+#include "gambas.h"
+
 #include "CWidget.h"
 
 #ifndef __CTEXTBOX_CPP
diff --git a/gb.qt4/src/ctrayicon.h b/gb.qt4/src/ctrayicon.h
index d0781ccc4..a6bd9293b 100644
--- a/gb.qt4/src/ctrayicon.h
+++ b/gb.qt4/src/ctrayicon.h
@@ -24,11 +24,12 @@
 #ifndef __CTRAYICON_H
 #define __CTRAYICON_H
 
+#include <QSystemTrayIcon>
+
 #include "gambas.h"
 #include "main.h"
 
 #include "CPicture.h"
-#include <QSystemTrayIcon>
 
 #ifndef __CTRAYICON_CPP
 extern GB_DESC TrayIconsDesc[];
diff --git a/gb.qt4/src/ext/CDial.cpp b/gb.qt4/src/ext/CDial.cpp
index f572ba4bf..6fb783da1 100644
--- a/gb.qt4/src/ext/CDial.cpp
+++ b/gb.qt4/src/ext/CDial.cpp
@@ -26,9 +26,6 @@
 #include "main.h"
 #include "gambas.h"
 
-#include <qapplication.h>
-#include <qdial.h>
-
 #include "CDial.h"
 
 DECLARE_EVENT(EVENT_Change);
diff --git a/gb.qt4/src/ext/CDial.h b/gb.qt4/src/ext/CDial.h
index eb396a5bb..723a486c2 100644
--- a/gb.qt4/src/ext/CDial.h
+++ b/gb.qt4/src/ext/CDial.h
@@ -24,6 +24,9 @@
 #ifndef __CDIAL_H
 #define __CDIAL_H
 
+#include <qapplication.h>
+#include <qdial.h>
+
 #include "gambas.h"
 #include "../gb.qt.h"
 
diff --git a/gb.qt4/src/ext/CTextEdit.h b/gb.qt4/src/ext/CTextEdit.h
index 98418e1be..9f2528032 100644
--- a/gb.qt4/src/ext/CTextEdit.h
+++ b/gb.qt4/src/ext/CTextEdit.h
@@ -24,11 +24,11 @@
 #ifndef __CTEXTEDIT_H
 #define __CTEXTEDIT_H
 
+#include <QTextEdit>
+
 #include "gambas.h"
 #include "../gb.qt.h"
 
-#include <QTextEdit>
-
 #ifndef __CTEXTEDIT_CPP
 extern GB_DESC CTextEditSelectionDesc[];
 extern GB_DESC CTextEditFormatDesc[];
diff --git a/gb.qt4/src/main.h b/gb.qt4/src/main.h
index d6311876d..568663c97 100644
--- a/gb.qt4/src/main.h
+++ b/gb.qt4/src/main.h
@@ -24,9 +24,6 @@
 #ifndef __MAIN_H
 #define __MAIN_H
 
-#include "gb_common.h"
-#include "gambas.h"
-
 #include <QEvent>
 #include <QApplication>
 #include <QTimerEvent>
@@ -35,6 +32,9 @@
 #include <QClipboard>
 #include <QFont>
 
+#include "gb_common.h"
+#include "gambas.h"
+
 #define DO_NOT_USE_QT_INTERFACE
 #include "gb.qt.h"
 #include "gb.image.h"
diff --git a/gb.qt4/src/webkit/ccookiejar.h b/gb.qt4/src/webkit/ccookiejar.h
index 2de32cd8f..468b7a99b 100644
--- a/gb.qt4/src/webkit/ccookiejar.h
+++ b/gb.qt4/src/webkit/ccookiejar.h
@@ -24,12 +24,12 @@
 #ifndef __CCOOKIEJAR_H
 #define __CCOOKIEJAR_H
 
-#include "main.h"
-
 #include <QUrl>
 #include <QNetworkCookie>
 #include <QNetworkCookieJar>
 
+#include "main.h"
+
 #ifndef __CCOOKIEJAR_CPP
 
 extern GB_DESC CookieDesc[];
diff --git a/gb.qt5/src/opengl/CGLarea.h b/gb.qt5/src/opengl/CGLarea.h
index 0c6020cef..c8e5aa835 100644
--- a/gb.qt5/src/opengl/CGLarea.h
+++ b/gb.qt5/src/opengl/CGLarea.h
@@ -24,10 +24,10 @@
 #ifndef __CGLAREA_H
 #define __CGLAREA_H
 
-#include "main.h"
-
 #include <QOpenGLWidget>
 
+#include "main.h"
+
 typedef
 	struct {
 		QT_WIDGET widget;
