diff -urB QtCurve-Gtk2-1.8.15.orig/style/shadowhelper.c QtCurve-Gtk2-1.8.15/style/shadowhelper.c
--- QtCurve-Gtk2-1.8.15.orig/style/shadowhelper.c	2012-04-24 17:19:54.000000000 -0400
+++ QtCurve-Gtk2-1.8.15/style/shadowhelper.c	2012-09-14 08:29:55.828897057 -0400
@@ -58,6 +58,7 @@
         cairo_rectangle(cr, 0, 0, shadowSize, shadowSize);
         cairo_fill(cr);
         cairo_destroy(cr);
+        cairo_surface_destroy(dest);
         g_object_unref(pixbuf);
         return pixmap;
     }
