diff -wbBur vdccm-0.10.1/lib/continousnode.cpp vdccm-0.10.1.my/lib/continousnode.cpp
--- vdccm-0.10.1/lib/continousnode.cpp	2006-03-28 02:40:15.000000000 +0400
+++ vdccm-0.10.1.my/lib/continousnode.cpp	2011-08-03 14:05:21.000000000 +0400
@@ -1,3 +1,4 @@
+#include <stdio.h>
 /***************************************************************************
  * Copyright (c) 2003 Volker Christian <voc@users.sourceforge.net>         *
  *                                                                         *
diff -wbBur vdccm-0.10.1/lib/localclientsocket.cpp vdccm-0.10.1.my/lib/localclientsocket.cpp
--- vdccm-0.10.1/lib/localclientsocket.cpp	2006-03-22 22:59:14.000000000 +0300
+++ vdccm-0.10.1.my/lib/localclientsocket.cpp	2011-08-03 14:09:42.000000000 +0400
@@ -68,6 +68,7 @@
     raddr.sun_family = AF_LOCAL;
     strncpy(raddr.sun_path, path.c_str(), sizeof(raddr.sun_path));
 
+#define offsetof(type, member)  __builtin_offsetof (type, member)
     size_t size = (offsetof (struct sockaddr_un, sun_path) + strlen(raddr.sun_path) + 1);
 
     if (::connect(fd, (struct sockaddr *) &raddr, size) < 0) {
diff -wbBur vdccm-0.10.1/lib/localsocket.cpp vdccm-0.10.1.my/lib/localsocket.cpp
--- vdccm-0.10.1/lib/localsocket.cpp	2006-03-28 02:40:15.000000000 +0400
+++ vdccm-0.10.1.my/lib/localsocket.cpp	2011-08-03 14:09:16.000000000 +0400
@@ -78,6 +78,7 @@
     localAddress.sun_family = AF_LOCAL;
     strncpy(localAddress.sun_path, path.c_str(), sizeof(localAddress.sun_path));
 
+#define offsetof(type, member)  __builtin_offsetof (type, member)
     size_t size = (offsetof (struct sockaddr_un, sun_path) + strlen(localAddress.sun_path) + 1);
 
     if (::bind(getDescriptor(), (struct sockaddr *) &localAddress, size) < 0) {
diff -wbBur vdccm-0.10.1/lib/tcpclientsocket.cpp vdccm-0.10.1.my/lib/tcpclientsocket.cpp
--- vdccm-0.10.1/lib/tcpclientsocket.cpp	2006-03-22 22:59:14.000000000 +0300
+++ vdccm-0.10.1.my/lib/tcpclientsocket.cpp	2011-08-03 14:05:21.000000000 +0400
@@ -1,3 +1,4 @@
+#include <string.h>
 /***************************************************************************
  * Copyright (c) 2003 Volker Christian <voc@users.sourceforge.net>         *
  *                                                                         *
diff -wbBur vdccm-0.10.1/lib/tcpconnectedsocket.cpp vdccm-0.10.1.my/lib/tcpconnectedsocket.cpp
--- vdccm-0.10.1/lib/tcpconnectedsocket.cpp	2006-06-29 23:01:48.000000000 +0400
+++ vdccm-0.10.1.my/lib/tcpconnectedsocket.cpp	2011-08-03 14:05:21.000000000 +0400
@@ -1,3 +1,4 @@
+#include <string.h>
 /***************************************************************************
  * Copyright (c) 2003 Volker Christian <voc@users.sourceforge.net>         *
  *                                                                         *
diff -wbBur vdccm-0.10.1/lib/tcpsocket.cpp vdccm-0.10.1.my/lib/tcpsocket.cpp
--- vdccm-0.10.1/lib/tcpsocket.cpp	2006-06-29 23:01:48.000000000 +0400
+++ vdccm-0.10.1.my/lib/tcpsocket.cpp	2011-08-03 14:05:21.000000000 +0400
@@ -1,3 +1,4 @@
+#include <string.h>
 /***************************************************************************
  * Copyright (c) 2003 Volker Christian <voc@users.sourceforge.net>         *
  *                                                                         *
diff -wbBur vdccm-0.10.1/lib/udpsocket.cpp vdccm-0.10.1.my/lib/udpsocket.cpp
--- vdccm-0.10.1/lib/udpsocket.cpp	2006-05-07 19:49:36.000000000 +0400
+++ vdccm-0.10.1.my/lib/udpsocket.cpp	2011-08-03 14:05:21.000000000 +0400
@@ -1,3 +1,4 @@
+#include <string.h>
 //
 // C++ Implementation: udpsocket
 //
diff -wbBur vdccm-0.10.1/lib/udpsocket.h vdccm-0.10.1.my/lib/udpsocket.h
--- vdccm-0.10.1/lib/udpsocket.h	2006-05-06 17:07:32.000000000 +0400
+++ vdccm-0.10.1.my/lib/udpsocket.h	2011-08-03 14:05:21.000000000 +0400
@@ -1,3 +1,4 @@
+#include <stdint.h>
 //
 // C++ Interface: udpsocket
 //
diff -wbBur vdccm-0.10.1/Makefile.in vdccm-0.10.1.my/Makefile.in
--- vdccm-0.10.1/Makefile.in	2007-12-19 22:59:02.000000000 +0300
+++ vdccm-0.10.1.my/Makefile.in	2011-08-03 14:05:21.000000000 +0400
@@ -186,9 +186,9 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 SUBDIRS = lib src tools
-AM_CFLAGS = -O2 -Werror -pedantic-errors -Wall
-AM_CXXFLAGS = -O2 -Werror -pedantic-errors -Wall
-AM_FFLAGS = -O2 -Werror -pedantic-errors -Wall
+AM_CFLAGS = -O2 -Werror  -Wall
+AM_CXXFLAGS = -O2 -Werror  -Wall
+AM_FFLAGS = -O2 -Werror  -Wall
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
diff -wbBur vdccm-0.10.1/src/Makefile.in vdccm-0.10.1.my/src/Makefile.in
--- vdccm-0.10.1/src/Makefile.in	2007-12-19 22:59:01.000000000 +0300
+++ vdccm-0.10.1.my/src/Makefile.in	2011-08-03 14:05:21.000000000 +0400
@@ -250,7 +250,7 @@
 	rapiproxy.cpp rapiproxyfactory.cpp rapiproxyconnection.cpp \
 	$(am__append_3)
 AM_CFLAGS = -Wall
-AM_CXXFLAGS = -pedantic-errors -Wall
+AM_CXXFLAGS =  -Wall
 man_MANS = vdccm.1
 EXTRA_DIST = $(man_MANS)
 all: $(BUILT_SOURCES)
diff -wbBur vdccm-0.10.1/src/rapiclient.cpp vdccm-0.10.1.my/src/rapiclient.cpp
--- vdccm-0.10.1/src/rapiclient.cpp	2006-05-06 17:07:32.000000000 +0400
+++ vdccm-0.10.1.my/src/rapiclient.cpp	2011-08-03 14:05:21.000000000 +0400
@@ -1,3 +1,4 @@
+#include <string.h>
 //
 // C++ Implementation: rapiclient
 //
diff -wbBur vdccm-0.10.1/src/rapiclient.h vdccm-0.10.1.my/src/rapiclient.h
--- vdccm-0.10.1/src/rapiclient.h	2006-05-06 17:07:32.000000000 +0400
+++ vdccm-0.10.1.my/src/rapiclient.h	2011-08-03 14:05:21.000000000 +0400
@@ -1,3 +1,5 @@
+#include <limits.h>
+#include <stdint.h>
 //
 // C++ Interface: rapiclient
 //
diff -wbBur vdccm-0.10.1/src/rapiconnection.cpp vdccm-0.10.1.my/src/rapiconnection.cpp
--- vdccm-0.10.1/src/rapiconnection.cpp	2007-05-10 22:27:03.000000000 +0400
+++ vdccm-0.10.1.my/src/rapiconnection.cpp	2011-08-03 14:05:21.000000000 +0400
@@ -1,3 +1,4 @@
+#include <stdio.h>
 //
 // C++ Implementation: rapiconnection
 //
