diff -rupN libretro-blastem.orig/Makefile libretro-blastem/Makefile
--- libretro-blastem.orig/Makefile	2019-04-17 09:08:11.641638576 +0200
+++ libretro-blastem/Makefile	2019-04-17 09:23:33.088550471 +0200
@@ -31,8 +31,8 @@ SDL2_PREFIX:="sdl/x86_64-w64-mingw32"
 GLUDIR:=x64
 endif
 GLEW32S_LIB:=$(GLEW_PREFIX)/lib/Release/$(GLUDIR)/glew32s.lib
-CFLAGS:=-std=gnu99 -Wreturn-type -Werror=return-type -Werror=implicit-function-declaration
-LDFLAGS:=-lm -lmingw32 -lws2_32 -mwindows
+CFLAGS:=$(CFLAGS) -std=gnu99 -Wreturn-type -Werror=return-type -Werror=implicit-function-declaration
+LDFLAGS:=$(CFLAGS) -lm -lmingw32 -lws2_32 -mwindows
 ifneq ($(MAKECMDGOALS),libblastem.dll)
 CFLAGS+= -I"$(SDL2_PREFIX)/include/SDL2" -I"$(GLEW_PREFIX)/include" -DGLEW_STATIC
 LDFLAGS+= $(GLEW32S_LIB) -L"$(SDL2_PREFIX)/lib" -lSDL2main -lSDL2 -lopengl32 -lglu32
@@ -47,7 +47,7 @@ NET:=net.o
 EXE:=
 
 HAS_PROC:=$(shell if [ -d /proc ]; then /bin/echo -e -DHAS_PROC; fi)
-CFLAGS:=-std=gnu99 -Wreturn-type -Werror=return-type -Werror=implicit-function-declaration -Wno-unused-value $(HAS_PROC) -DHAVE_UNISTD_H
+CFLAGS:=$(CFLAGS) -std=gnu99 -Wreturn-type -Werror=return-type -Werror=implicit-function-declaration -Wno-unused-value $(HAS_PROC) -DHAVE_UNISTD_H
 
 ifeq ($(OS),Darwin)
 LIBS=sdl2 glew
@@ -89,10 +89,10 @@ ifdef USE_GLES
 ifndef GLES_LIB
 GLES_LIB:=$(shell pkg-config --libs glesv2)
 endif
-LDFLAGS:=-lm $(GLES_LIB)
+LDFLAGS:=$(LDFLAGS) -lm $(GLES_LIB)
 else
 CFLAGS+= -DGLEW_STATIC -Iglew/include
-LDFLAGS:=-lm glew/lib/libGLEW.a
+LDFLAGS:=$(LDFLAGS) -lm glew/lib/libGLEW.a
 endif
 
 ifeq ($(OS),Darwin)
@@ -109,10 +109,10 @@ endif #Darwin
 
 else
 ifeq ($(MAKECMDGOALS),libblastem.$(SO))
-LDFLAGS:=-lm
+LDFLAGS:=$(LDFLAGS) -lm
 else
-CFLAGS:=$(shell pkg-config --cflags-only-I $(LIBS)) $(CFLAGS)
-LDFLAGS:=-lm $(shell pkg-config --libs $(LIBS))
+CFLAGS:=$(CFLAGS) $(shell pkg-config --cflags-only-I $(LIBS)) $(CFLAGS)
+LDFLAGS:=$(LDFLAGS) -lm $(shell pkg-config --libs $(LIBS))
 ifdef USE_FBDEV
 LDFLAGS+= -pthread
 endif
