From 6f70b2c43d9d2f0acf224e7f93db54ff9456000c Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz93@gmail.com>
Date: Mon, 22 Jan 2018 13:14:54 -0500
Subject: [PATCH] Fix manages not being installed in the correct directory.

---
 wscript | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wscript b/wscript
index 2819b59..187f518 100644
--- a/wscript
+++ b/wscript
@@ -165,7 +165,7 @@ def build(bld):
 			source = bld.path.ant_glob('docs/pflask.rst') +
 			         bld.path.ant_glob('build/docs/conf.py'),
 			target = 'docs/man/pflask.1 docs/man/pflask-debuild.1',
-			install_path = bld.env.MANDIR
+			install_path = bld.env.MANDIR + '/man1'
 		)
 
 		bld(
-- 
2.16.1

