diff -Naur deja-dup-24.0.orig/common/BackendU1.vala deja-dup-24.0/common/BackendU1.vala
--- deja-dup-24.0.orig/common/BackendU1.vala	2012-08-07 00:20:51.000000000 +0200
+++ deja-dup-24.0/common/BackendU1.vala	2012-11-05 16:39:34.140104977 +0100
@@ -75,11 +75,14 @@
 
 class U1Checker : Checker
 {
-  PythonChecker pyu1;
   construct {
     try {
       var proxy = BackendU1.get_creds_proxy();
-      if (proxy.get_name_owner() == null) {
+      if (proxy.get_name_owner() != null) {
+        available = true;
+        complete = true;
+      }
+      else {
         available = false;
         complete = true;
       }
@@ -89,24 +92,6 @@
       available = false;
       complete = true;
     }
-
-    if (!complete) {
-      // A bit of abstraction leakage here; we have to keep these imports in
-      // line with what duplicity uses.  Maybe we should add to duplicity a way
-      // to ask 'can I use this backend?'
-      pyu1 = PythonChecker.get_checker("ubuntuone.platform.credentials, ubuntuone.couch.auth");
-      if (pyu1.complete) {
-        available = pyu1.available;
-        complete = pyu1.complete;
-      }
-      else {
-        pyu1.notify["complete"].connect(() => {
-          available = pyu1.available;
-          complete = pyu1.complete;
-          pyu1 = null;
-        });
-      }
-    }
   }
 }
 
