diff -Nurp guitarix-0.35.0.mod/libgxwmm/gxwmm/gainline.h guitarix-0.35.0/libgxwmm/gxwmm/gainline.h
--- guitarix-0.35.0.mod/libgxwmm/gxwmm/gainline.h	2016-04-24 06:18:37.000000000 +0200
+++ guitarix-0.35.0/libgxwmm/gxwmm/gainline.h	2016-04-24 21:22:44.846076833 +0200
@@ -22,7 +22,7 @@
 #include <gxw/gainpoints.h>
 #include <vector>
 
-inline bool operator==(const gain_points& p1, const gain_points& p2) { return p1.i == p2.i && abs(p1.g - p2.g) < 1e-4 * (p1.g + p2.g); }
+inline bool operator==(const gain_points& p1, const gain_points& p2) { return p1.i == p2.i && fabs(p1.g - p2.g) < 1e-4 * (p1.g + p2.g); }
 inline bool operator!=(const gain_points& p1, const gain_points& p2) { return !(p1 == p2); }
 
 class Gainline: public std::vector<gain_points>
