diff -ruN a/Makefile b/Makefile
--- a/Makefile	2018-03-07 16:42:09.000000000 +0100
+++ b/Makefile	2020-07-09 12:15:56.182015666 +0200
@@ -62,7 +62,7 @@
 LIBS =
 
 LUA_CFLAGS = -I/usr/include/lua
-LUA_LIBS   = -llua
+LUA_LIBS   = $(shell pkg-config --libs lua-5.3)
 
 # get library version from meta file
 LIBRARY_VERSION = $(shell sed -n 's|^\#X text [0-9][0-9]* [0-9][0-9]* VERSION \(.*\);|\1|p' $(LIBRARY_META))
diff -ruN a/pdlua.c b/pdlua.c
--- a/pdlua.c	2018-03-07 16:42:09.000000000 +0100
+++ b/pdlua.c	2020-07-09 12:10:25.631400140 +0200
@@ -42,9 +42,9 @@
 #include <unistd.h>
 #endif
 /* we use Lua */
-#include <lua.h>
-#include <lauxlib.h>
-#include <lualib.h>
+#include <lua5.3/lua.h>
+#include <lua5.3/lauxlib.h>
+#include <lua5.3/lualib.h>
 
 /* we use Pd */
 #include "m_pd.h"
