diff -urN hgview-1.10.2.orig/hgviewlib/__init__.py hgview-1.10.2/hgviewlib/__init__.py
--- hgview-1.10.2.orig/hgviewlib/__init__.py	2018-08-11 18:17:12.701368614 +0200
+++ hgview-1.10.2/hgviewlib/__init__.py	2018-08-11 18:17:26.204879111 +0200
@@ -23,4 +23,4 @@
 from mercurial import demandimport
 
 # this should help docutils, see hgrepoview.rst2html
-demandimport.ignore.append('roman')
+demandimport.IGNORES.add('roman')
