=== modified file 'common/BackendU1.vala'
--- common/BackendU1.vala	2012-11-05 15:10:54 +0000
+++ common/BackendU1.vala	2012-11-05 15:17:28 +0000
@@ -94,7 +94,7 @@
       // A bit of abstraction leakage here; we have to keep these imports in
       // line with what duplicity uses.  Maybe we should add to duplicity a way
       // to ask 'can I use this backend?'
-      pyu1 = PythonChecker.get_checker("ubuntuone.platform.credentials, ubuntuone.couch.auth");
+      pyu1 = PythonChecker.get_checker("oauthlib, httplib2");
       if (pyu1.complete) {
         available = pyu1.available;
         complete = pyu1.complete;

