--- os/linux/sta_ioctl.c.orig	2010-12-17 12:44:47.000000000 +0100
+++ os/linux/sta_ioctl.c	2011-08-02 14:52:37.000000000 +0200
@@ -1812,7 +1812,7 @@
             wrq->length = strlen(extra) + 1; // 1: size of '\0'
             break;
         case SHOW_DRVIER_VERION:
-            sprintf(extra, "Driver version-%s, %s %s\n", STA_DRIVER_VERSION, __DATE__, __TIME__ );
+            sprintf(extra, "Driver version-%s, %s %s\n", STA_DRIVER_VERSION, NULL, NULL );
             wrq->length = strlen(extra) + 1; // 1: size of '\0'
             break;
 #ifdef DOT11_N_SUPPORT
