Description: Fix Background Module for python2-pillow 2.5.0-1.
Author: Claire Farron <diesal3@googlemail.com>
Origin: upstream
--- a/files/usr/lib/cinnamon-settings/bin/imtools.py
+++ b/files/usr/lib/cinnamon-settings/bin/imtools.py
@@ -859,7 +859,7 @@
     :type force: bool
     """
     # Paste on top
-    if source == mask:
+    if mask and source == mask:
         if has_alpha(source):
             # invert_alpha = the transparant pixels of the destination
             if has_alpha(destination) and (destination.size == source.size
