From 9397158303284bff0b85a3b329e4e8b2fff7eba7 Mon Sep 17 00:00:00 2001
From: Michael Webster <miketwebster@gmail.com>
Date: Mon, 17 Nov 2014 11:49:27 -0500
Subject: [PATCH] xlets: silently ignore uninstalled items (like bluetooth)

---
 js/ui/extension.js | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/js/ui/extension.js b/js/ui/extension.js
index 3de89a6..80b5734 100644
--- a/js/ui/extension.js
+++ b/js/ui/extension.js
@@ -377,8 +377,8 @@ function loadExtension(uuid, type) {
         var forgetMeta = true;
         try {
             let dir = findExtensionDirectory(uuid, type);
-            if(dir == null) {
-                throw (type.name + ' ' + uuid + ' not found.');
+            if (dir == null) {
+                throw ("not-found");
             }
             extension = new Extension(dir, type);
             forgetMeta = false;
@@ -388,7 +388,17 @@ function loadExtension(uuid, type) {
 
             extension.finalize();
             Main.cinnamonDBusService.EmitXletAddedComplete(true, uuid);
-        } catch(e) {
+        } catch (e) {
+            /* Silently fail to load xlets that aren't actually installed - 
+               but no error, since the user can't do anything about it anyhow
+               (short of editing gsettings).  Silent failure is consistent with
+               other reactions in Cinnamon to missing items (e.g. panel launchers
+               just don't show up if their program isn't installed, but we don't
+               remove them or anything) */
+            if (e == "not-found") {
+                forgetExtension(uuid, forgetMeta);
+                return null;
+            }
             Main.cinnamonDBusService.EmitXletAddedComplete(false, uuid);
             Main.xlet_startup_error = true;
             forgetExtension(uuid, forgetMeta);
