--- QScintilla-gpl-2.8.3/Qt4Qt5/features/qscintilla2.prf~	2014-07-03 14:15:46.000000000 +0400
+++ QScintilla-gpl-2.8.3/Qt4Qt5/features/qscintilla2.prf	2014-08-11 18:04:44.371414609 +0400
@@ -17,9 +17,18 @@
         win32: {
             LIBS += -lqscintilla2d
         } else {
-            LIBS += -lqscintilla2
+            greaterThan(QT_MAJOR_VERSION, 4) {
+                LIBS += -lqt5scintilla2
+            } else {
+                LIBS += -lqscintilla2
+            }
         }
     }
 } else {
+    greaterThan(QT_MAJOR_VERSION, 4) {
+        LIBS += -lqt5scintilla2
+        }
+        else {
     LIBS += -lqscintilla2
+        }
 }
--- QScintilla-gpl-2.8.3/Qt4Qt5/qscintilla.pro~	2014-07-03 14:15:46.000000000 +0400
+++ QScintilla-gpl-2.8.3/Qt4Qt5/qscintilla.pro	2014-08-11 18:00:05.598079386 +0400
@@ -41,6 +41,7 @@
 
 greaterThan(QT_MAJOR_VERSION, 4) {
 	QT += widgets printsupport
+	TARGET = qt5scintilla2
 
     greaterThan(QT_MINOR_VERSION, 1) {
 	    macx:QT += macextras
--- QScintilla-gpl-2.8.3/Qt4Qt5/designer-Qt4Qt5/designer.pro~	2014-07-03 14:15:46.000000000 +0400
+++ QScintilla-gpl-2.8.3/designer-Qt4Qt5/designer.pro	2014-08-11 18:00:05.598079386 +0400
@@ -8,11 +8,14 @@
 
 greaterThan(QT_MAJOR_VERSION, 4) {
     QT += designer
+    TARGET = qt5scintillaplugin
+    LIBS += -lqt5scintilla2
 
     # Work around QTBUG-39300.
     CONFIG -= android_install
 } else {
     CONFIG += designer
+    LIBS += -lqscintilla2
 }
 
 macx {
 