diff -rupN dolphin-4.0.2.orig/CMakeLists.txt dolphin-4.0.2/CMakeLists.txt
--- dolphin-4.0.2.orig/CMakeLists.txt	2015-03-31 15:05:43.953995860 +0200
+++ dolphin-4.0.2/CMakeLists.txt	2015-03-31 15:06:53.379078333 +0200
@@ -417,6 +417,7 @@ if(NOT ANDROID)
 
 		# Note: We do not need to explicitly check for X11 as it is done in the cmake
 		# FindOpenGL module on linux.
+		include(FindX11)
 		if(USE_X11 AND X11_FOUND)
 			set(USE_X11 1)
 			add_definitions(-DHAVE_X11=1)
