diff --git a/mk/golang.mk b/mk/golang.mk
index 0b2a2c55a..2c7f24d75 100644
--- a/mk/golang.mk
+++ b/mk/golang.mk
@@ -12,7 +12,7 @@ GOTFLAGS ?=
 unexport GOFLAGS
 # Override so we can combine with the user's go flags.
 # Try to make building as reproducible as possible by stripping the go path.
-override GOFLAGS += "-asmflags=all='-trimpath=$(GOPATH)'" "-gcflags=all='-trimpath=$(GOPATH)'"
+override GOFLAGS += "-trimpath"
 
 ifeq ($(tarball-is),1)
        GOFLAGS += -mod=vendor
