diff -Naur xarchiver-0.5.2.orig/src/main.c xarchiver-0.5.2/src/main.c
--- xarchiver-0.5.2.orig/src/main.c	2014-01-24 04:31:20.288863040 +0100
+++ xarchiver-0.5.2/src/main.c	2014-01-24 04:31:45.252847778 +0100
@@ -420,7 +420,7 @@
 		//ArchiveSuffix = g_list_append(ArchiveSuffix, "");
 	}
 
-	absolute_path = g_find_program_in_path ("rar");
+	/* absolute_path = g_find_program_in_path ("rar");
     if ( absolute_path )
 	{
 		ArchiveType = g_list_append(ArchiveType, "rar");
@@ -437,7 +437,7 @@
 			ArchiveSuffix = g_list_append(ArchiveSuffix, "*.rar");
 			g_free (absolute_path);
 		}
-	}
+	} */
 
 	absolute_path = g_find_program_in_path("cpio");
     if ( absolute_path )
diff -Naur xarchiver-0.5.2.orig/src/window.c xarchiver-0.5.2/src/window.c
--- xarchiver-0.5.2.orig/src/window.c	2014-01-24 04:05:33.455228000 +0100
+++ xarchiver-0.5.2/src/window.c	2014-01-24 04:15:30.624117028 +0100
@@ -1117,8 +1117,9 @@
 		xx = XARCHIVETYPE_ZIP;
 	else if (memcmp (magic,"\x60\xea",2) == 0)
 		xx = XARCHIVETYPE_ARJ;
+        /* Tempolary disable rar support due to incompatiblity with rar 5.0
 	else if (memcmp ( magic,"\x52\x61\x72\x21",4) == 0)
-		xx = XARCHIVETYPE_RAR;
+		xx = XARCHIVETYPE_RAR; */
 	else if (memcmp ( magic,"\x42\x5a\x68",3) == 0)
 		xx = XARCHIVETYPE_BZIP2;
 	else if (memcmp ( magic,"\x1f\x8b",2) == 0 || memcmp ( magic,"\x1f\x9d",2) == 0)
