From 8b92940fe6780155fbafb46782f6720fd6822af1 Mon Sep 17 00:00:00 2001
From: Christopher Arndt <chris@chrisarndt.de>
Date: Sat, 19 Feb 2022 16:24:20 +0100
Subject: [PATCH] fix: add -lGL to UILIBS for non-static unix LV2 builds

Signed-off-by: Christopher Arndt <chris@chrisarndt.de>
---
 common.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common.mak b/common.mak
index 1c12f32..3ae8c86 100644
--- a/common.mak
+++ b/common.mak
@@ -167,7 +167,7 @@ ifeq ($(LV2AVAIL)$(HAVE_UI)$(HAVE_IDLE), yesyesyes)
         UILIBS+=`pkg-config --variable=libdir ftgl`/libftgl.a `pkg-config --variable=libdir ftgl`/libfreetype.a
         UILIBS+=`pkg-config --libs zlib`
       else
-        UILIBS+=`pkg-config --libs glu ftgl`
+        UILIBS+=`pkg-config --libs glu gl ftgl`
       endif
       UICFLAGS+=-DFONTFILE=\"$(FONTFILE)\"
     endif
