diff -ruN a/tests/test_cli.py b/tests/test_cli.py
--- a/tests/test_cli.py	2019-12-21 19:29:52.000000000 +0100
+++ b/tests/test_cli.py	2020-03-14 11:53:20.505357215 +0100
@@ -390,7 +390,7 @@
         'ExtraContextWithNoEqualsSoInvalid',
     )
     assert result.exit_code == 2
-    assert 'Error: Invalid value for "[EXTRA_CONTEXT]..."' in result.output
+    assert "Error: Invalid value for '[EXTRA_CONTEXT]...'" in result.output
     assert 'should contain items of the form key=value' in result.output
 
 
diff -ruN a/tests/test_cookiecutter_invocation.py b/tests/test_cookiecutter_invocation.py
--- a/tests/test_cookiecutter_invocation.py	2019-12-21 19:29:52.000000000 +0100
+++ b/tests/test_cookiecutter_invocation.py	2020-03-14 11:52:17.332550458 +0100
@@ -23,7 +23,7 @@
         subprocess.check_call([sys.executable, '-m', 'cookiecutter.cli'])
 
     _, err = capfd.readouterr()
-    exp_message = 'Error: Missing argument "TEMPLATE".'
+    exp_message = "Error: Missing argument 'TEMPLATE'."
     assert exp_message in err
 
 
