=== modified file 'daemon/tile.vala'
--- daemon/tile.vala	2010-12-01 23:39:56 +0000
+++ daemon/tile.vala	2011-01-18 04:30:10 +0000
@@ -185,10 +185,6 @@
       {
         try
         {
-#if VALA_0_12
-#else
-          unowned
-#endif
           Gdk.Pixbuf temp_pb;
           unowned IconTheme it = IconTheme.get_default ();
           try
@@ -204,10 +200,6 @@
                                     IconLookupFlags.FORCE_SIZE);
           }
           pixbuf = temp_pb.copy ();
-#if VALA_0_12
-#else
-          temp_pb.unref (); // careful here!
-#endif
         }
         catch (Error err)
         {

