diff -rupN TLP-0.5.orig/tlp-stat TLP-0.5/tlp-stat
--- TLP-0.5.orig/tlp-stat	2014-09-27 15:33:19.090823700 +0200
+++ TLP-0.5/tlp-stat	2014-09-27 15:33:43.507854300 +0200
@@ -498,8 +498,8 @@ if [ "$show_all" = "1" ]; then
     if [ -d $I915D ]; then
         echo "+++ Intel Graphics"
         printparm_i915 $I915D/powersave
-        printparm_i915 $I915D/i915_enable_rc6
-        printparm_i915 $I915D/i915_enable_fbc
+        printparm_i915 $I915D/enable_rc6
+        printparm_i915 $I915D/enable_fbc
         printparm_i915 $I915D/lvds_downclock
         printparm_i915 $I915D/semaphores
         echo
