diff -ruN ASL-0.1.7-orig/CMakeLists.txt ASL-0.1.7/CMakeLists.txt
--- ASL-0.1.7-orig/CMakeLists.txt	2017-08-07 17:47:37.522927767 +0300
+++ ASL-0.1.7/CMakeLists.txt	2017-08-07 17:47:58.382410158 +0300
@@ -37,7 +37,7 @@
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fpermissive -std=c++11")
 else()
 	set(CMAKE_CXX_STANDARD 11)
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fpermissive")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fpermissive -Wno-deprecated -Wno-deprecated-declarations -Wno-ignored-attributes")
 endif()
 
 # Enable supplied cmake includes and modules
