diff --git a/CMakeLists.txt b/CMakeLists.txt
index 876da06..d3cf4e1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -99,7 +99,6 @@ install(
 
 include_directories(
     ${PROJECT_BINARY_DIR}/include
-    ${PROJECT_SOURCE_DIR}/gmock/gtest/include
     ${PROJECT_SOURCE_DIR}/src/exporter
     ${PROJECT_SOURCE_DIR}/src/importer
     ${PROJECT_SOURCE_DIR}/src/model
@@ -253,7 +253,7 @@ endif()
 
 # testing
 enable_testing()
-add_subdirectory(gmock)
+find_package(GTest REQUIRED)
 
 file(GLOB run-test_src
     test/*.cpp
@@ -262,7 +262,8 @@ add_executable(run-test
     ${run-test_src}
 )
 target_link_libraries(run-test
-    gmock
+    GTest::GTest
+    GTest::Main
 
     exporter
     importer
diff --git a/test/testChewingExporter.cpp b/test/testChewingExporter.cpp
index 911a453..b8bebf0 100644
--- a/test/testChewingExporter.cpp
+++ b/test/testChewingExporter.cpp
@@ -17,7 +17,7 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#include "gtest/gtest.h"
+#include <gtest/gtest.h>
 
 #include <QDebug>
 #include <QFile>
diff --git a/test/testChewingImporter.cpp b/test/testChewingImporter.cpp
index 412c513..3b600d7 100644
--- a/test/testChewingImporter.cpp
+++ b/test/testChewingImporter.cpp
@@ -17,7 +17,7 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#include "gtest/gtest.h"
+#include <gtest/gtest.h>
 
 #include <QDebug>
 #include <QDir>
diff --git a/test/testHashImporter.cpp b/test/testHashImporter.cpp
index 8f10a96..5c5be6b 100644
--- a/test/testHashImporter.cpp
+++ b/test/testHashImporter.cpp
@@ -17,7 +17,7 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#include "gtest/gtest.h"
+#include <gtest/gtest.h>
 
 #include <QDebug>
 #include <QDir>
diff --git a/test/testUserphraseImporter.cpp b/test/testUserphraseImporter.cpp
index 17f3e27..54f2c0f 100644
--- a/test/testUserphraseImporter.cpp
+++ b/test/testUserphraseImporter.cpp
@@ -17,7 +17,7 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#include "gtest/gtest.h"
+#include <gtest/gtest.h>
 
 #include <QDebug>
 
diff --git a/test/testUserphraseSet.cpp b/test/testUserphraseSet.cpp
index f80d2d9..f73f33d 100644
--- a/test/testUserphraseSet.cpp
+++ b/test/testUserphraseSet.cpp
@@ -17,7 +17,7 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#include "gtest/gtest.h"
+#include <gtest/gtest.h>
 
 #include "UserphraseSet.h"
 
diff --git a/test/testmain.cpp b/test/testmain.cpp
index 0d58d49..e269919 100644
--- a/test/testmain.cpp
+++ b/test/testmain.cpp
@@ -19,7 +19,7 @@
 #include <QApplication>
 #include <QFileInfo>
 
-#include "gtest/gtest.h"
+#include <gtest/gtest.h>
 
 bool verbose = false;
 
