diff --git a/bindings/python/setup.py.in b/bindings/python/setup.py.in
index cea13f3a9..dcfc46b05 100644
--- a/bindings/python/setup.py.in
+++ b/bindings/python/setup.py.in
@@ -8,15 +8,15 @@ import subprocess
 # Remove the "-Wstrict-prototypes" compiler option, which isn't valid for C++.
 cfg_vars = sysconfig.get_config_vars()
 opt = cfg_vars["OPT"]
-cfg_vars["OPT"] = " ".join( flag for flag in opt.split() if flag not in ['-Wstrict-prototypes' ${CLANG_PROHIBITED} ] ) + " --std=c++0x"
+cfg_vars["OPT"] = " ".join( flag for flag in opt.split() if flag not in ['-Wstrict-prototypes' ${CLANG_PROHIBITED} ] ) + " --std=c++17"
 
 cflags = cfg_vars["CFLAGS"]
-cfg_vars["CFLAGS"] = " ".join( flag for flag in cflags.split() if flag not in ['-Wstrict-prototypes' ${CLANG_PROHIBITED} ] ) + " --std=c++0x"
+cfg_vars["CFLAGS"] = " ".join( flag for flag in cflags.split() if flag not in ['-Wstrict-prototypes' ${CLANG_PROHIBITED} ] ) + " --std=c++17"
 
 # pypy doesn't define PY_CFLAGS so skip it if it's missing
 if "PY_CFLAGS" in cfg_vars:
   py_cflags = cfg_vars["PY_CFLAGS"]
-  cfg_vars["PY_CFLAGS"] = " ".join( flag for flag in py_cflags.split() if flag not in ['-Wstrict-prototypes' ${CLANG_PROHIBITED} ] ) + " --std=c++0x"
+  cfg_vars["PY_CFLAGS"] = " ".join( flag for flag in py_cflags.split() if flag not in ['-Wstrict-prototypes' ${CLANG_PROHIBITED} ] ) + " --std=c++17"
 
 ccl=cfg_vars["CC"].split()
 ccl[0]="${CMAKE_C_COMPILER}"
diff --git a/cmake/XRootDOSDefs.cmake b/cmake/XRootDOSDefs.cmake
index ee4525490..0257989f2 100644
--- a/cmake/XRootDOSDefs.cmake
+++ b/cmake/XRootDOSDefs.cmake
@@ -21,9 +21,9 @@ endif()
 add_definitions( -DUSE_LIBC_SEMAPHORE=${USE_LIBC_SEMAPHORE} )
 
 #-------------------------------------------------------------------------------
-# Enable c++0x / c++11
+# Enable c++17
 #-------------------------------------------------------------------------------
-set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x -DOPENSSL_NO_FILENAMES" )
+set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -DOPENSSL_NO_FILENAMES" )
 
 #-------------------------------------------------------------------------------
 # Enable XrdCl::Pipelines for clang compiler
@@ -37,7 +37,7 @@ endif()
 # GCC
 #-------------------------------------------------------------------------------
 if( CMAKE_COMPILER_IS_GNUCXX )
-  set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x" )
+  set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17" )
   set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra" )
   #-----------------------------------------------------------------------------
   # Set -Werror only for Debug (or undefined) build type or if we have been
diff --git a/src/XrdSsi/XrdSsiAtomics.hh b/src/XrdSsi/XrdSsiAtomics.hh
index 151f2b628..3f191db55 100644
--- a/src/XrdSsi/XrdSsiAtomics.hh
+++ b/src/XrdSsi/XrdSsiAtomics.hh
@@ -34,12 +34,12 @@
 #undef NEED_ATOMIC_MUTEX
 
 //-----------------------------------------------------------------------------
-//! Use native atomics at the c11 or higher level (-std=c++0x -lstdc++)
+//! Use native atomics at the c17 or higher level (-std=c++17 -lstdc++)
 //-----------------------------------------------------------------------------
 #if __cplusplus >= 201103L
 #include <atomic>
 #define Atomic(type)    std::atomic<type>
-#define Atomic_IMP "C++11"
+#define Atomic_IMP "C++17"
 #define Atomic_BEG(x)
 #define Atomic_DEC(x)          x.fetch_sub(1,std::memory_order_relaxed)
 #define Atomic_GET(x)          x.load(std::memory_order_relaxed)
diff --git a/tests/XrdClTests/tls/CMakeLists.txt b/tests/XrdClTests/tls/CMakeLists.txt
index bf6ee2317..f9ce03ca0 100644
--- a/tests/XrdClTests/tls/CMakeLists.txt
+++ b/tests/XrdClTests/tls/CMakeLists.txt
@@ -1,7 +1,7 @@
 
 include( XRootDCommon )
 
-set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x" )
+set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17" )
 
 #-------------------------------------------------------------------------------
 # xrdcopy
diff --git a/src/XrdCeph/cmake/XRootDOSDefs.cmake b/src/XrdCeph/cmake/XRootDOSDefs.cmake
index eadc2495..2969c543 100644
--- a/src/XrdCeph/cmake/XRootDOSDefs.cmake
+++ b/src/XrdCeph/cmake/XRootDOSDefs.cmake
@@ -11,7 +11,7 @@ set( LIBRARY_PATH_PREFIX "lib" )
 # GCC
 #-------------------------------------------------------------------------------
 if( CMAKE_COMPILER_IS_GNUCXX )
-  set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11" )
+  set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17" )
   set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror" )
   set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-parameter" )
   # gcc 4.1 is retarded
diff --git a/src/XrdClHttp/CMakeLists.txt b/src/XrdClHttp/CMakeLists.txt
index 3b38522..968811a 100644
--- a/src/XrdClHttp/CMakeLists.txt
+++ b/src/XrdClHttp/CMakeLists.txt
@@ -6,7 +6,7 @@ list(APPEND CMAKE_MODULE_PATH
     ${PROJECT_SOURCE_DIR}/src
     ${PROJECT_SOURCE_DIR}/cmake)

-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")

 # Defaults
 if( NOT XRDCLHTTP_SUBMODULE )
