diff --git a/libgnac/libgnac-converter.c b/libgnac/libgnac-converter.c
index 35d9bc4..e96cf97 100644
--- a/libgnac/libgnac-converter.c
+++ b/libgnac/libgnac-converter.c
@@ -78,7 +78,7 @@ G_DEFINE_TYPE(LibgnacConverter, libgnac_converter, G_TYPE_OBJECT);
     LIBGNAC_TYPE_CONVERTER, LibgnacConverterPrivate))
 
 
-LibgnacMetadata *metadata;
+extern LibgnacMetadata *metadata;
 
 
 static void
diff --git a/src/profiles/gnac-profiles-manager.c b/src/profiles/gnac-profiles-manager.c
index ba00b29..5952302 100644
--- a/src/profiles/gnac-profiles-manager.c
+++ b/src/profiles/gnac-profiles-manager.c
@@ -69,6 +69,7 @@ typedef struct {
 } CopyData;
 
 GtkBuilder *profiles_mgr_builder = NULL;
+gchar *saved_profiles_dir;
 
 static GtkTargetEntry target_list[] =  {
   { "text/uri-list", 0, DND_TARGET_URI   },
diff --git a/src/profiles/gnac-profiles-xml-engine.h b/src/profiles/gnac-profiles-xml-engine.h
index dcf3de5..865142c 100755
--- a/src/profiles/gnac-profiles-xml-engine.h
+++ b/src/profiles/gnac-profiles-xml-engine.h
@@ -35,7 +35,7 @@
 
 G_BEGIN_DECLS
 
-gchar *saved_profiles_dir;
+extern gchar *saved_profiles_dir;
 
 typedef struct
 {
