diff --git a/Makefile.am b/Makefile.am
index 4809bd5..6043fbf 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -14,11 +14,9 @@ INTLTOOL_BUILT = \
 	intltool-update
 
 EXTRA_DIST = \
-	autogen.sh \
-	gnome-doc-utils.make
+	autogen.sh
 
 DISTCLEANFILES = \
-	gnome-doc-utils.make \
 	po/.intltool-merge-cache \
 	$(INTLTOOL_BUILT)
 
diff --git a/configure.ac b/configure.ac
index b2c5605..20a2081 100644
--- a/configure.ac
+++ b/configure.ac
@@ -55,7 +55,7 @@ AM_GLIB_GNU_GETTEXT
 dnl Check for GSettings macros
 GLIB_GSETTINGS
 
-GNOME_DOC_INIT([0.17.2])
+YELP_HELP_INIT
 GNOME_DEBUG_CHECK
 GNOME_COMPILE_WARNINGS([maximum])
 GNOME_CXX_WARNINGS([yes])
diff --git a/help/Makefile.am b/help/Makefile.am
index dff2f18..760fc92 100644
--- a/help/Makefile.am
+++ b/help/Makefile.am
@@ -1,15 +1,14 @@
-include $(top_srcdir)/gnome-doc-utils.make
-dist-hook: doc-dist-hook
+@YELP_HELP_RULES@
 
-DOC_ID = gnac
+HELP_ID = gnac
 
-DOC_FIGURES = \
+HELP_MEDIA = \
 	figures/gnac-logo.png \
 	figures/gnac-main-window.png
 
-DOC_PAGES = \
+HELP_FILES = \
 	index.page \
 	intro.page \
 	license.page
 
-DOC_LINGUAS = cs da de es fr ro ru sl te
+HELP_LINGUAS = cs da de es fr ro ru sl te
diff --git a/src/gnac-ui.c b/src/gnac-ui.c
index 76f6401..200c7db 100644
--- a/src/gnac-ui.c
+++ b/src/gnac-ui.c
@@ -855,7 +855,7 @@ gnac_ui_on_help_cb(GtkAction *action,
   GError *error = NULL;
 
   ret = gtk_show_uri(gtk_widget_get_screen(gnac_ui_get_widget("main_window")),
-      "ghelp:gnac", gtk_get_current_event_time(), &error);
+      "help:gnac", gtk_get_current_event_time(), &error);
   if (!ret && error) {
     g_printerr("%s: %s\n", _("Could not display help"), error->message);
     g_clear_error(&error);
