From c6f3175c00a7e2b30e4ff5ab0e61c39967366c81 Mon Sep 17 00:00:00 2001
From: Tsuyoshi Hombashi <tsuyoshi.hombashi@gmail.com>
Date: Sun, 31 Jul 2022 21:44:07 +0900
Subject: [PATCH 1/2] Remove messages that were removed at pyflakes 2.5.0

---
 pylama/lint/pylama_pyflakes.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/pylama/lint/pylama_pyflakes.py b/pylama/lint/pylama_pyflakes.py
index 81f695a..c828fdb 100644
--- a/pylama/lint/pylama_pyflakes.py
+++ b/pylama/lint/pylama_pyflakes.py
@@ -9,7 +9,6 @@

 checker.messages.UnusedImport.message = "W0611 %r imported but unused"
 checker.messages.RedefinedWhileUnused.message = "W0404 redefinition of unused %r from line %r"
-checker.messages.RedefinedInListComp.message = "W0621 list comprehension redefines %r from line %r"
 checker.messages.ImportShadowedByLoopVar.message = "W0621 import %r from line %r shadowed by loop variable"
 checker.messages.ImportStarUsed.message = "W0401 'from %s import *' used; unable to detect undefined names"
 checker.messages.ImportStarUsage.message = "W0401 '%s may be undefined, or defined from star imports: %s'"
@@ -20,7 +19,6 @@
 checker.messages.DuplicateArgument.message = "E1122 duplicate argument %r in function definition"
 checker.messages.LateFutureImport.message = "W0410 future import(s) %r after other statements"
 checker.messages.UnusedVariable.message = "W0612 local variable %r is assigned to but never used"
-checker.messages.ReturnWithArgsInsideGenerator.message = "E0106 'return' with argument inside generator"
 checker.messages.ReturnOutsideFunction.message = "E0104 'return' outside function"
 
 

From 897b3fb9f923fd6eed77e50f22ecb0b9dfb8e472 Mon Sep 17 00:00:00 2001
From: Tsuyoshi Hombashi <tsuyoshi.hombashi@gmail.com>
Date: Sat, 6 Aug 2022 15:45:06 +0900
Subject: [PATCH 2/2] Modify to keep backward compatibility

---
 pylama/lint/pylama_pyflakes.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/pylama/lint/pylama_pyflakes.py b/pylama/lint/pylama_pyflakes.py
index c828fdb..8a1a606 100644
--- a/pylama/lint/pylama_pyflakes.py
+++ b/pylama/lint/pylama_pyflakes.py
@@ -22,6 +22,13 @@
 
 checker.messages.ReturnOutsideFunction.message = "E0104 'return' outside function"
 
+# RedefinedInListComp and ReturnWithArgsInsideGenerator were removed at pyflakes 2.5.0:
+#   https://github.com/PyCQA/pyflakes/commit/2246217295dc8cb30ef4a7b9d8dc449ce32e603a
+if hasattr(checker.messages, "RedefinedInListComp"):
+    CODES[checker.messages.RedefinedInListComp.message] = "W0621"
+if hasattr(checker.messages, "ReturnWithArgsInsideGenerator"):
+    CODES[checker.messages.ReturnWithArgsInsideGenerator.message] = "E0106"
+
 
 class Linter(Abstract):
     """Pyflakes runner."""
