diff -ruN a/src/gigedit/compat.h b/src/gigedit/compat.h
--- a/src/gigedit/compat.h	2021-02-21 15:41:48.000000000 +0100
+++ b/src/gigedit/compat.h	2021-06-03 19:28:13.603235786 +0200
@@ -190,11 +190,6 @@
 
 #if HAS_PANGOMM_CPP11_ENUMS
 # include <pangomm/layout.h>
-namespace Pango {
-    const Alignment ALIGN_LEFT = Alignment::LEFT;
-    const Alignment ALIGN_CENTER = Alignment::CENTER;
-    const Alignment ALIGN_RIGHT = Alignment::RIGHT;
-}
 #endif // HAS_PANGOMM_CPP11_ENUMS
 
 #if HAS_GTKMM_CPP11_ENUMS
