diff -rup ../prelink.orig/src/arch-x86_64.c ./src/arch-x86_64.c
--- ../prelink.orig/src/arch-x86_64.c	2010-12-22 12:03:06.725004002 +0100
+++ ./src/arch-x86_64.c	2010-12-22 12:04:08.193712039 +0100
@@ -532,7 +532,7 @@ PL_ARCH = {
   .R_COPY = R_X86_64_COPY,
   .R_RELATIVE = R_X86_64_RELATIVE,
   .rtype_class_valid = RTYPE_CLASS_VALID,
-  .dynamic_linker = "/lib64/ld-linux-x86-64.so.2",
+  .dynamic_linker = "/lib/ld-linux-x86-64.so.2",
   .adjust_dyn = x86_64_adjust_dyn,
   .adjust_rel = x86_64_adjust_rel,
   .adjust_rela = x86_64_adjust_rela,
