From 340ef9d973ffd533b17548e3a9f50418501b0681 Mon Sep 17 00:00:00 2001
From: Omer Katz <omer.drow@gmail.com>
Date: Mon, 28 Oct 2019 20:57:59 +0200
Subject: [PATCH] Add Python 3.8 Support (#5785)

* Workaround patching problem in test.

diff --git a/celery/contrib/testing/app.py b/celery/contrib/testing/app.py
index 3580c43165..60c6462135 100644
--- a/celery/contrib/testing/app.py
+++ b/celery/contrib/testing/app.py
@@ -30,6 +30,10 @@ class Trap(object):
     """
 
     def __getattr__(self, name):
+        # Workaround to allow unittest.mock to patch this object
+        # in Python 3.8 and above.
+        if name == '_is_coroutine':
+            return None
         raise RuntimeError('Test depends on current_app')
 
 
