diff -wbBur libsieve-2.3.1/src/sv_parser/header.y libsieve-2.3.1.q/src/sv_parser/header.y
--- libsieve-2.3.1/src/sv_parser/header.y	2012-05-31 13:16:50.000000000 +0400
+++ libsieve-2.3.1.q/src/sv_parser/header.y	2013-09-02 20:21:26.000000000 +0400
@@ -70,7 +70,7 @@
                 }
         | body WRAP             {
                 TRACE_DEBUG( "body: body WRAP: %s %s", $1, $2 );
-                $$ = libsieve_strbuf(context->strbuf, libsieve_strconcat( $1, $2, NULL ), strlen($1)+strlen($2), FREEME);
+                $$ = libsieve_strbuf(context->strbuf, libsieve_strconcat( $1 ? $1 : "", $2, NULL ), strlen($1 ? $1 : "")+strlen($2), FREEME);
                 };
 
 %%
diff -wbBur libsieve-2.3.1/src/sv_parser/addr.y libsieve-2.3.1.q/src/sv_parser/addr.y
--- libsieve-2.3.1/src/sv_parser/addr.y	2012-05-31 13:16:50.000000000 +0400
+++ libsieve-2.3.1.q/src/sv_parser/addr.y	2013-09-02 21:42:15.267979609 +0400
@@ -38,6 +38,7 @@
 
 /* sv_util */
 #include "src/sv_util/util.h"
+#include "src/sv_interface/context2.h"
 
 /* sv_parser */
 #include "addr.h"
