--- a/amitools/tools/geotool.py
+++ b/amitools/tools/geotool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.7
+#!/usr/bin/env python
 
 from __future__ import absolute_import
 from __future__ import print_function
--- a/amitools/tools/rdbtool.py
+++ b/amitools/tools/rdbtool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.7
+#!/usr/bin/env python
 # rdbtool
 # swiss army knife for rdb disk images or devices
 
--- a/amitools/tools/xdfscan.py
+++ b/amitools/tools/xdfscan.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.7
+#!/usr/bin/env python
 # xdfscan
 # quickly scan large sets of Amiga disk image files
 
--- a/amitools/tools/xdftool.py
+++ b/amitools/tools/xdftool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.7
+#!/usr/bin/env python
 # xdftool
 # swiss army knife for adf and hdf amiga disk images
 
--- a/amitools/util/BlkDevTools.py
+++ b/amitools/util/BlkDevTools.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.7
+#!/usr/bin/env python
 # based heavily on "iops" by Benjamin Schweizer
 # https://github.com/gopher/iops
 
