--- piper-cpp/vamp-capnp/piper.capnp.h	2017-05-25 10:26:10.000000000 +0200
+++ piper-cpp/vamp-capnp/piper.capnp.h.new	2017-12-05 13:46:42.929667773 +0100
@@ -6,7 +6,7 @@
 
 #include <capnp/generated-header-support.h>
 
-#if CAPNP_VERSION != 6000
+#if CAPNP_VERSION < 6000
 #error "Version mismatch between generated code and library headers.  You must use the same version of the Cap'n Proto compiler and library."
 #endif
 
