diff --git a/open-vm-tools/services/plugins/dndcp/copyPasteUIX11.cpp b/open-vm-tools/services/plugins/dndcp/copyPasteUIX11.cpp
index be8369e..e4d6877 100644
--- a/open-vm-tools/services/plugins/dndcp/copyPasteUIX11.cpp
+++ b/open-vm-tools/services/plugins/dndcp/copyPasteUIX11.cpp
@@ -92,7 +92,7 @@ extern "C" {
  * matter, we just create the atom we need directly in main().
  *
  * This is for V1 text copy paste only!
- */
+ *
 #ifndef GDK_SELECTION_CLIPBOARD
 GdkAtom GDK_SELECTION_CLIPBOARD;
 #endif
@@ -104,6 +104,7 @@ GdkAtom GDK_SELECTION_TYPE_TIMESTAMP;
 #ifndef GDK_SELECTION_TYPE_UTF8_STRING
 GdkAtom GDK_SELECTION_TYPE_UTF8_STRING;
 #endif
+ */
 
 
 /*
