--- network	2010-04-28 12:30:51.000000000 +0200
+++ network.bak	2010-05-20 12:12:50.000000000 +0200
@@ -76,7 +76,7 @@
          else
             echo `date` "[rescue_nic] activating $nic ..."
 
-            ifup $nic
+            ifconfig $nic up
             exitCode=`expr $exitCode \| $?`
          fi
       done < $activeList
@@ -103,7 +103,7 @@
 TranquilizeNetworkManager() {
    # `which' may be a bit noisy, so we'll shush it.
    dbusSend=`which dbus-send 2>/dev/null`
-   if [ $? -eq 0 ]; then
+   if [ $? -eq 0 ] && [ `pidof dbus-daemon` ]; then
       # NetworkManager 0.6
       $dbusSend --system --dest=org.freedesktop.NetworkManager          \
          /org/freedesktop/NetworkManager                                \
@@ -133,7 +133,7 @@
 WakeNetworkManager() {
    # `which' may be a bit noisy, so we'll shush it.
    dbusSend=`which dbus-send 2>/dev/null`
-   if [ $? -eq 0 ]; then
+   if [ $? -eq 0 ] && [ `pidof dbus-daemon` ]; then
       # NetworkManager 0.6
       $dbusSend --system --dest=org.freedesktop.NetworkManager          \
          /org/freedesktop/NetworkManager                                \
@@ -165,8 +165,8 @@
 
    # XXX Are these really necessary?  If so, we should have seen customer
    # complaints by now.
-   which ifup >/dev/null 2>&1      || Panic "ifup not in search path."
-   which ifconfig >/dev/null 2>&1  || Panic "ifconfig not in search path."
+#   which ifup >/dev/null 2>&1     || Panic "ifup not in search path."
+   which ifconfig >/dev/null 2>&1 || Panic "ifconfig not in search path."
 
    case "$1" in
       poweron-vm)
