diff --git a/test/test_streams.py b/test/test_streams.py
index 6985aec..8b292f9 100644
--- a/test/test_streams.py
+++ b/test/test_streams.py
@@ -121,10 +121,10 @@ def test_writer_bad_message(wfile, writer):
         second=1,
     ))
 
-    assert wfile.getvalue() in [
-        b'',
+    assert wfile.getvalue() == (
+        b''
         b'Content-Length: 10\r\n'
         b'Content-Type: application/vscode-jsonrpc; charset=utf8\r\n'
         b'\r\n'
-        b'1546304461'
-    ]
+        b'1546300861'
+        )
