diff -wbBur gsoap-2.8/gsoap/stdsoap2.h gsoap-2.8.my/gsoap/stdsoap2.h
--- gsoap-2.8/gsoap/stdsoap2.h	2018-03-08 18:45:36.000000000 +0300
+++ gsoap-2.8.my/gsoap/stdsoap2.h	2018-03-28 15:51:38.342848826 +0300
@@ -338,7 +338,6 @@
 #  define HAVE_MBTOWC
 #  define HAVE_INTTYPES_H
 #  define HAVE_LOCALE_H
-#  define HAVE_XLOCALE_H
 #  define HAVE_RANDOM
 # elif defined(_AIX43)
 #  define HAVE_SNPRINTF
@@ -419,7 +418,6 @@
 #  define HAVE_ISNAN
 #  define HAVE_ISINF
 #  define HAVE_LOCALE_H
-#  define HAVE_XLOCALE_H
 #  define HAVE_RANDOM
 # elif defined(__VMS)
 #  include <ioctl.h>
@@ -720,11 +718,6 @@
 # endif
 #endif
 
-/* force inclusion of xlocale.h */
-#if defined(WITH_INCLUDE_XLOCALE_H) && !defined(HAVE_XLOCALE_H)
-# define HAVE_XLOCALE_H
-#endif
-
 #ifdef WITH_C_LOCALE
 # include <locale.h>
 # if defined(WIN32) && !defined(CYGWIN)
@@ -732,9 +725,6 @@
 #  define SOAP_LOCALE(soap) ((soap)->c_locale ? (soap)->c_locale : ((soap)->c_locale = _create_locale(LC_ALL, "C")))
 #  define SOAP_FREELOCALE(soap) (void)((soap)->c_locale && (_free_locale((soap)->c_locale), ((soap)->c_locale = NULL)))
 # else
-#  if defined(HAVE_XLOCALE_H)
-#   include <xlocale.h>
-#  endif
 #  define SOAP_LOCALE_T locale_t
 #  define SOAP_LOCALE(soap) ((soap)->c_locale ? (soap)->c_locale : ((soap)->c_locale = newlocale(LC_ALL_MASK, "C", NULL)))
 #  define SOAP_FREELOCALE(soap) (void)((soap)->c_locale && (freelocale((soap)->c_locale), ((soap)->c_locale = NULL)))
