diff -Naur gecko-mediaplayer-1.0.5b1-old/src/np_entry.cpp gecko-mediaplayer-1.0.5b1-new/src/np_entry.cpp
--- gecko-mediaplayer-1.0.5b1-old/src/np_entry.cpp	2011-09-20 20:48:04.000000000 +0300
+++ gecko-mediaplayer-1.0.5b1-new/src/np_entry.cpp	2011-11-22 15:57:46.751354874 +0200
@@ -80,9 +80,9 @@
 
 #endif                          /* XP_WIN */
 
-char *NPP_GetMIMEDescription();
+const char *NPP_GetMIMEDescription();
 
-char *NP_GetMIMEDescription()
+const char *NP_GetMIMEDescription()
 {
     return NPP_GetMIMEDescription();
 }
diff -Naur gecko-mediaplayer-1.0.5b1-old/src/npp_gate.cpp gecko-mediaplayer-1.0.5b1-new/src/npp_gate.cpp
--- gecko-mediaplayer-1.0.5b1-old/src/npp_gate.cpp	2011-09-20 20:48:04.000000000 +0300
+++ gecko-mediaplayer-1.0.5b1-new/src/npp_gate.cpp	2011-11-22 15:58:02.473353591 +0200
@@ -44,7 +44,7 @@
 #include "plugin_types.h"
 #include "plugin_setup.h"
 
-char *NPP_GetMIMEDescription(void)
+const char *NPP_GetMIMEDescription(void)
 {
     return GetMIMEDescription();
 }
diff -Naur gecko-mediaplayer-1.0.5b1-old/src/npupp.h gecko-mediaplayer-1.0.5b1-new/src/npupp.h
--- gecko-mediaplayer-1.0.5b1-old/src/npupp.h	2011-09-20 20:48:04.000000000 +0300
+++ gecko-mediaplayer-1.0.5b1-new/src/npupp.h	2011-11-22 15:56:32.262360950 +0200
@@ -684,7 +684,7 @@
 
     NPError OSCALL NP_Shutdown();
 
-    char *NP_GetMIMEDescription();
+    const char *NP_GetMIMEDescription();
 
 #ifdef __cplusplus
 }
@@ -700,7 +700,7 @@
 
 /* plugin meta member functions */
 
-    NP_EXPORT(char *) NP_GetMIMEDescription(void);
+    NP_EXPORT(const char *) NP_GetMIMEDescription(void);
      NP_EXPORT(NPError) NP_Initialize(NPNetscapeFuncs *, NPPluginFuncs *);
      NP_EXPORT(NPError) NP_Shutdown(void);
      NP_EXPORT(NPError) NP_GetValue(void *future, NPPVariable aVariable, void *aValue);
