From d3f47d51ab68054ceb3415971455ca2c036b395f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bart=C5=82omiej=20Piotrowski?= <b@bpiotrowski.pl>
Date: Thu, 12 Apr 2012 16:43:41 +0200
Subject: [PATCH] Don't mark desktop file as executable.

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index be1567b..50973d6 100644
--- a/Makefile
+++ b/Makefile
@@ -78,7 +78,7 @@ install:
 install -d $(DESTDIR)/usr/share/pixmaps
 install extras/luakit.png $(DESTDIR)/usr/share/pixmaps/
 install -d $(DESTDIR)/usr/share/applications
- install extras/luakit.desktop $(DESTDIR)/usr/share/applications/
+ install -m0644 extras/luakit.desktop $(DESTDIR)/usr/share/applications/
 install -d $(MANPREFIX)/man1/
 install -m644 luakit.1 $(MANPREFIX)/man1/

---
1.7.10
