From 81875f9409cb306ec518d61812a6f3f075e16339 Mon Sep 17 00:00:00 2001
From: Tyler Erickson <me@tylererickson.com>
Date: Tue, 27 Dec 2016 12:09:46 -0800
Subject: [PATCH] Fixed regression caused by previous merge

---
 Text/Hamlet/Parse.hs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Text/Hamlet/Parse.hs b/Text/Hamlet/Parse.hs
index d0c9961..27b65a7 100644
--- a/Text/Hamlet/Parse.hs
+++ b/Text/Hamlet/Parse.hs
@@ -257,6 +257,7 @@ parseLine set = do
         case x of
             Left "#" -> case cr of
                           InContent -> return (ContentRaw "#", False)
+                          NotInQuotesAttr -> return (ContentRaw "#", False)
                           _ -> fail "Expected hash at end of line, got Id"
             Left str -> return (ContentRaw str, null str)
             Right deref -> return (ContentVar deref, False)
From b43ceba3f033e1aa83c3c424a661996a3e272194 Mon Sep 17 00:00:00 2001
From: Tyler Erickson <me@tylererickson.com>
Date: Tue, 27 Dec 2016 12:22:44 -0800
Subject: [PATCH] Reversed Test cases; NotInQuotes -> fail; otherwise ->
 succeed

---
 Text/Hamlet/Parse.hs | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/Text/Hamlet/Parse.hs b/Text/Hamlet/Parse.hs
index 27b65a7..3f77df6 100644
--- a/Text/Hamlet/Parse.hs
+++ b/Text/Hamlet/Parse.hs
@@ -256,9 +256,8 @@ parseLine set = do
         x <- parseHash
         case x of
             Left "#" -> case cr of
-                          InContent -> return (ContentRaw "#", False)
-                          NotInQuotesAttr -> return (ContentRaw "#", False)
-                          _ -> fail "Expected hash at end of line, got Id"
+                          NotInQuotes -> fail "Expected hash at end of lin, got Id"
+                          _ -> return (ContentRaw "#", False)
             Left str -> return (ContentRaw str, null str)
             Right deref -> return (ContentVar deref, False)
     contentAt = do
