From 81875f9409cb306ec518d61812a6f3f075e16339 Mon Sep 17 00:00:00 2001
From: Tyler Erickson <me@tylererickson.com>
Date: Tue, 27 Dec 2016 12:09:46 -0800
Subject: [PATCH] Fixed regression caused by previous merge

---
 Text/Hamlet/Parse.hs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Text/Hamlet/Parse.hs b/Text/Hamlet/Parse.hs
index d0c9961..27b65a7 100644
--- a/Text/Hamlet/Parse.hs
+++ b/Text/Hamlet/Parse.hs
@@ -257,6 +257,7 @@ parseLine set = do
         case x of
             Left "#" -> case cr of
                           InContent -> return (ContentRaw "#", False)
+                          NotInQuotesAttr -> return (ContentRaw "#", False)
                           _ -> fail "Expected hash at end of line, got Id"
             Left str -> return (ContentRaw str, null str)
             Right deref -> return (ContentVar deref, False)
