--- src/setup.py.orig	2014-11-16 10:34:05.521441703 +0100
+++ src/setup.py	2014-11-16 10:34:16.054812569 +0100
@@ -581,20 +581,6 @@
 
 
 #########################################################
-### Clean
-#########################################################
-
-print('Cleaning up stale installed files....')
-t = time.time()
-from sage_setup.clean import clean_install_dir
-output_dirs = SITE_PACKAGES + glob.glob(os.path.join(SAGE_SRC, 'build', 'lib*'))
-for output_dir in output_dirs:
-    print('- cleaning {0}'.format(output_dir))
-    clean_install_dir(output_dir, python_packages, python_modules, ext_modules)
-print('Finished cleaning, time: %.2f seconds.' % (time.time() - t))
-
-
-#########################################################
 ### Distutils
 #########################################################
 
