diff -ruN pil-1.1.6.orig/_imagingft.c pil-1.1.6/_imagingft.c
--- pil-1.1.6.orig/_imagingft.c	2013-12-23 17:45:39.991183855 +0400
+++ pil-1.1.6/_imagingft.c	2013-12-23 17:49:06.297861197 +0400
@@ -65,7 +65,7 @@
     const char* message;
 } ft_errors[] =
 
-#include <freetype/fterrors.h>
+#include <freetype2/fterrors.h>
 
 /* -------------------------------------------------------------------- */
 /* font objects */
