--- src/module_list.py.orig	2015-02-15 00:13:10.878070678 +0100
+++ src/module_list.py	2015-02-15 00:13:26.514831552 +0100
@@ -2092,11 +2092,6 @@
     Extension('sage.tests.interrupt',
               sources = ['sage/tests/interrupt.pyx', 'sage/tests/c_lib.c']),
 
-    Extension('sage.tests.parallel',
-              sources = ['sage/tests/parallel.pyx'],
-              extra_compile_args=["-fopenmp"],
-              extra_link_args=["-fopenmp"]),
-
     Extension('sage.tests.stl_vector',
               sources = ['sage/tests/stl_vector.pyx'],
               libraries = ['gmp'],
