diff --color -aur tensorboard_plugin_wit-1.8.1-old/WORKSPACE tensorboard_plugin_wit-1.8.1-new/WORKSPACE
--- tensorboard_plugin_wit-1.8.1-old/WORKSPACE	2022-11-25 22:30:42.405732907 +0200
+++ tensorboard_plugin_wit-1.8.1-new/WORKSPACE	2022-11-25 22:31:51.022397184 +0200
@@ -13,6 +13,14 @@
     ],
 )
 
+http_archive(
+    name = "zlib",
+    build_file = "@com_google_protobuf//:third_party/zlib.BUILD",
+    sha256 = "c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1",
+    strip_prefix = "zlib-1.2.11",
+    urls = ["https://zlib.net/fossils/zlib-1.2.11.tar.gz"],
+)
+
 load("@bazel_skylib//lib:versions.bzl", "versions")
 # Keep this version in sync with the BAZEL environment variable defined
 # in our .travis.yml config.
