diff -U 3 -d -r -N -- xsensors-0.70/src/gui.c xsensors-0.70-fixed/src/gui.c
--- xsensors-0.70/src/gui.c	2010-02-03 04:55:57.000000000 +0100
+++ xsensors-0.70-fixed/src/gui.c	2010-07-14 10:51:42.161429477 +0200
@@ -84,7 +84,7 @@
     while ( *digit ) {
         get_pm_location( *digit, &x, &y, &w );
         gdk_draw_drawable( widget->window,
-                           widget->style->fg_gc[ GTK_WIDGET_STATE
+                           widget->style->fg_gc[ gtk_widget_get_state
                            (widget) ], theme, x, y + highLow,
                            pos, 0, w, 30 );
         pos += w;
@@ -121,7 +121,7 @@
 
             /* Display RPM */
             gdk_draw_drawable( widget->window, 
-                               widget->style->fg_gc[ GTK_WIDGET_STATE 
+                               widget->style->fg_gc[ gtk_widget_get_state 
                                (widget) ], theme, 0, 120 + highLow, 
                                90, 0, 57, 30 );
             break;
@@ -142,7 +142,7 @@
             else
                 x = 57;
             gdk_draw_drawable( widget->window, 
-                             widget->style->fg_gc[ GTK_WIDGET_STATE 
+                             widget->style->fg_gc[ gtk_widget_get_state 
                              (widget) ], theme, x, 60 + highLow, 
                              96, 0, 57, 30 );
             
@@ -158,7 +158,7 @@
 
             /* Display V */
             gdk_draw_drawable( widget->window, 
-                             widget->style->fg_gc[ GTK_WIDGET_STATE 
+                             widget->style->fg_gc[ gtk_widget_get_state 
                              (widget) ], theme, 114, 60 + highLow, 
                              96, 0, 57, 30 );
 
