diff --git a/src/caja-file-management-properties.ui b/src/caja-file-management-properties.ui
index 68c9c94..05f9446 100644
--- a/src/caja-file-management-properties.ui
+++ b/src/caja-file-management-properties.ui
@@ -271,6 +271,8 @@
     <property name="border_width">5</property>
     <property name="title" translatable="yes">File Management Preferences</property>
     <property name="window_position">center</property>
+    <property name="default_width">600</property>
+    <property name="default_height">600</property>
     <property name="type_hint">dialog</property>
     <child internal-child="vbox">
       <object class="GtkVBox" id="dialog-vbox1">
