diff --git a/src/utils/net/upnp_stubs.c b/src/utils/net/upnp_stubs.c
index 41942f3..272b0cb 100644
--- a/src/utils/net/upnp_stubs.c
+++ b/src/utils/net/upnp_stubs.c
@@ -640,7 +640,7 @@ upnpPulse( ml_upnpmp_t * map )
     {
         struct UPNPDev * devlist;
         errno = 0;
-        devlist = upnpDiscover( 2000, NULL, NULL, 0, 0, 0 );
+        devlist = upnpDiscover( 2000, NULL, NULL, 0, 0, 0, NULL );
         if( devlist == NULL )
         {
             dbg_printf( "upnpDiscover failed (errno %d - %s)\n", errno,  str_errno( errno ) );
