Index: tvtime-1.0.2/src/videoinput.c
===================================================================
--- tvtime-1.0.2.orig/src/videoinput.c
+++ tvtime-1.0.2/src/videoinput.c
@@ -35,8 +35,9 @@
 #ifdef HAVE_CONFIG_H
 # include "config.h"
 #endif
-#include "videodev.h"
-#include "videodev2.h"
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
+#define BASE_VIDIOCPRIVATE      192             /* 192-255 are private */
 #include "videoinput.h"
 #include "mixer.h"
 
