--- src/module_list.py.orig	2015-01-02 01:37:15.614093528 +0100
+++ src/module_list.py	2015-01-02 01:37:26.140827483 +0100
@@ -2126,7 +2126,8 @@
 # These extensions are to be compiled only if the
 # corresponding packages have been installed
 
-from sage.misc.package import is_package_installed
+def is_package_installed(name):
+    return False
 
 if is_package_installed('fes'):
     ext_modules.extend([
