--- src/sage/groups/perm_gps/permgroup.py.orig	2014-12-23 10:14:09.687176992 +0100
+++ src/sage/groups/perm_gps/permgroup.py	2014-12-23 10:14:53.930669821 +0100
@@ -180,8 +180,6 @@
     """
     @wraps(f)
     def wrapped(self, n, p=0):
-        if not is_package_installed('gap_packages'):
-            raise RuntimeError("You must install the optional gap_packages package.")
         load_hap()
         from sage.rings.arith import is_prime
         if not (p == 0 or is_prime(p)):
