diff -rup luxrender-luxrays-7c7127ee1fa4/cmake/Dependencies.cmake luxrender-luxrays-7c7127ee1fa4.new/cmake/Dependencies.cmake
--- luxrender-luxrays-7c7127ee1fa4/cmake/Dependencies.cmake	2015-01-20 23:45:32.000000000 +0100
+++ luxrender-luxrays-7c7127ee1fa4.new/cmake/Dependencies.cmake	2015-02-15 22:19:40.775351414 +0100
@@ -58,7 +58,7 @@ set(Boost_MINIMUM_VERSION       "1.44.0"
 
 set(Boost_ADDITIONAL_VERSIONS "1.47.0" "1.46.1" "1.46" "1.46.0" "1.45" "1.45.0" "1.44" "1.44.0")
 
-set(LUXRAYS_BOOST_COMPONENTS thread program_options filesystem serialization iostreams regex system python)
+set(LUXRAYS_BOOST_COMPONENTS thread program_options filesystem serialization iostreams regex system python3)
 find_package(Boost ${Boost_MINIMUM_VERSION} COMPONENTS ${LUXRAYS_BOOST_COMPONENTS})
 if (NOT Boost_FOUND)
         # Try again with the other type of libs
