diff --git a/wscript b/wscript
--- a/wscript
+++ b/wscript
@@ -95,8 +95,8 @@
     conf.env.debug = conf.options.debug
     conf.env.msvc_relwithdebug = conf.options.msvc_relwithdebug
 
-    # Ensure that the headers in siplib and Phoenix's src dir can be found
-    conf.env.INCLUDES_WXPY = ['sip/siplib', 'src']
+    # Ensure that the headers in Phoenix's src dir can be found
+    conf.env.INCLUDES_WXPY = ['src']
 
     if isWindows:
         # Windows/MSVC specific stuff
@@ -505,26 +505,6 @@
         copy_file(name, cfg.PKGDIR, update=1, verbose=1)
 
 
-    # Create the build tasks for each of our extension modules.
-    addRelwithdebugFlags(bld, 'siplib')
-    siplib = bld(
-        features = 'c cxx cshlib cxxshlib pyext',
-        target   = makeTargetName(bld, 'siplib'),
-        source   = ['sip/siplib/apiversions.c',
-                    'sip/siplib/array.c',
-                    'sip/siplib/bool.cpp',
-                    'sip/siplib/descriptors.c',
-                    'sip/siplib/int_convertors.c',
-                    'sip/siplib/objmap.c',
-                    'sip/siplib/qtlib.c',
-                    'sip/siplib/siplib.c',
-                    'sip/siplib/threads.c',
-                    'sip/siplib/voidptr.c',
-                    ],
-        uselib   = 'siplib WX WXPY',
-    )
-    makeExtCopyRule(bld, 'siplib')
-
     # Add build rules for each of our ETG generated extension modules
     makeETGRule(bld, 'etg/_core.py',       '_core',      'WX')
     makeETGRule(bld, 'etg/_adv.py',        '_adv',       'WXADV')
@@ -613,14 +593,14 @@
 
 # Make extension module build rules using info gleaned from an ETG script
 def makeETGRule(bld, etgScript, moduleName, libFlags):
-    from buildtools.config   import loadETG, getEtgSipCppFiles
+    from buildtools.config   import loadETG
 
     addRelwithdebugFlags(bld, moduleName)
     rc = ['src/wxc.rc'] if isWindows else []
     etg = loadETG(etgScript)
     bld(features='c cxx cxxshlib pyext',
         target=makeTargetName(bld, moduleName),
-        source=getEtgSipCppFiles(etg) + rc,
+        source=rc,
         uselib='{} {} WXPY'.format(moduleName, libFlags),
         )
     makeExtCopyRule(bld, moduleName)
