From 760832ff3b23a0edf5969b2cb307b55a863c424d Mon Sep 17 00:00:00 2001
From: Sam Lunt <samuel.j.lunt@gmail.com>
Date: Thu, 13 May 2021 16:48:22 -0500
Subject: [PATCH] set colors to be an int, not a string

---
 tmuxp/cli.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tmuxp/cli.py b/tmuxp/cli.py
index 74f5142c..9f7b91a9 100644
--- a/tmuxp/cli.py
+++ b/tmuxp/cli.py
@@ -564,7 +564,7 @@ def load_workspace(
         ``tmux -S <socket-path>``
     new_session_name: str, options
         ``tmux new -s <new_session_name>``
-    colors : str, optional
+    colors : int, optional
         '-2'
             Force tmux to support 256 colors
     detached : bool
@@ -1013,6 +1013,7 @@ def command_freeze(session_name, socket_name, socket_path, force):
 @click.option(
     'colors',
     '-2',
+    type=int,
     flag_value=256,
     default=True,
     help='Force tmux to assume the terminal supports 256 colours.',
@@ -1020,6 +1021,7 @@ def command_freeze(session_name, socket_name, socket_path, force):
 @click.option(
     'colors',
     '-8',
+    type=int,
     flag_value=88,
     help='Like -2, but indicates that the terminal supports 88 colours.',
 )
