--- backuppc-3.2.1.orig/lib/BackupPC/Lib.pm
+++ backuppc-3.2.1/lib/BackupPC/Lib.pm
@@ -495,7 +495,8 @@
     }
     if ( $IODirentOk ) {
         @entries = sort({ $a->{inode} <=> $b->{inode} } readdirent($fh));
-        map { $_->{type} = 0 + $_->{type} } @entries;   # make type numeric
+        #map { $_->{type} = 0 + $_->{type} } @entries;   # make type numeric
+        map { $_->{type} = 0 + $_->{type}; $_->{type} = undef if ($_->{type} eq BPC_DT_UNKNOWN); } @entries;   # make type numeric, unset unknown types
     } else {
         @entries = map { { name => $_} } readdir($fh);
     }
