--- a/depends/packages/libsnark.mk
+++ b/depends/packages/libsnark.mk
@@ -6,8 +6,6 @@
 $(package)_sha256_hash=dad153fe46e2e1f33557a195cbe7d69aed8b19ed9befc08ddcb8c6d3c025941f
 $(package)_git_commit=9ada3f84ab484c57b2247c2f41091fd6a0916573
 
-$(package)_dependencies=libgmp libsodium
-
 define $(package)_build_cmds
   CXXFLAGS="-fPIC -DBINARY_OUTPUT -DNO_PT_COMPRESSION=1" $(MAKE) lib DEPINST=$(host_prefix) CURVE=ALT_BN128 MULTICORE=1 NO_PROCPS=1 NO_GTEST=1 NO_DOCS=1 STATIC=1 NO_SUPERCOP=1 FEATUREFLAGS=-DMONTGOMERY_OUTPUT OPTFLAGS="-O2 -march=x86-64"
 endef
