=== modified file 'gwibber/microblog/plugins/facebook/__init__.py'
--- old/gwibber/microblog/plugins/facebook/__init__.py	2012-09-24 20:55:20 +0000
+++ new/gwibber/microblog/plugins/facebook/__init__.py	2013-01-14 15:41:35 +0000
@@ -326,7 +326,10 @@
     
     if data.has_key("privacy"):
       m["privacy"] = {}
-      m["privacy"]["description"] = data["privacy"]["description"]
+      if data["privacy"].has_key("description"):
+        m["privacy"]["description"] = data["privacy"]["description"]
+      else:
+        m["privacy"]["description"] = ""
       m["privacy"]["value"] = data["privacy"]["value"]
 
     # Handle target for wall posts with a specific recipient

