diff --git a/src/libpcp/src/check-statics b/src/libpcp/src/check-statics
index f6a8058..8db3aac 100755
--- a/src/libpcp/src/check-statics
+++ b/src/libpcp/src/check-statics
@@ -619,7 +619,7 @@ do
 		# extras from the last object code file
 		sed <$tmp/out \
 		    -e 's/^[^ ]* //' \
-		    -e "s/^\(.\) \(.*\)/$obj: \1 \2 : Error: additional symbol/"
+		    -e "s!^\(.\) \(.*\)!$obj: \1 \2 : Error: additional symbol!"
 		touch $tmp/fail
 	    fi
 	fi
