--- a/modules/ruby/manifests/source.pp
+++ b/modules/ruby/manifests/source.pp
@@ -73,6 +73,12 @@
     require => Exec["download-ruby"],
   }
 
+  exec { "patch-ruby":
+    command => "patch -Np0 -i ../../../ruby23_gcc7.patch",
+    cwd     => $file_cache_dir,
+    require => Exec["untar-ruby"],
+  }
+
   autotools { "ruby":
     configure_flags  => "--prefix=${prefix} --disable-debug --disable-dependency-tracking --disable-install-doc --enable-shared --with-opt-dir=${prefix} --enable-load-relative${extra_configure_flags}",
     cwd              => $source_dir_path,
@@ -80,7 +86,7 @@
     install_sentinel => "${prefix}/bin/ruby",
     make_notify      => $make_notify,
     make_sentinel    => "${source_dir_path}/ruby",
-    require          => Exec["untar-ruby"],
+    require          => Exec["untar-ruby", "patch-ruby"],
   }
 
   if $operatingsystem == 'Darwin' {
