diff -ur ruby-2.3.4.orig/include/ruby/ruby.h ruby-2.3.4/include/ruby/ruby.h
--- ruby-2.3.4.orig/include/ruby/ruby.h	2016-03-29 07:03:30.000000000 +0100
+++ ruby-2.3.4/include/ruby/ruby.h	2017-07-10 20:00:18.699618133 +0100
@@ -555,7 +555,10 @@
  * itself.  don't use it directly */
 #ifdef __GNUC__
 #define RB_GC_GUARD_PTR(ptr) \
-    __extension__ ({volatile VALUE *rb_gc_guarded_ptr = (ptr); rb_gc_guarded_ptr;})
+    __extension__ ({volatile VALUE *rb_gc_guarded_ptr = (ptr); \
+ __asm__("" : : "m"(rb_gc_guarded_ptr)); \
+ rb_gc_guarded_ptr; \
+    })
 #else
 #ifdef _MSC_VER
 #pragma optimize("", off)
diff -ur ruby-2.3.4.orig/marshal.c ruby-2.3.4/marshal.c
--- ruby-2.3.4.orig/marshal.c	2017-01-16 19:08:08.000000000 +0000
+++ ruby-2.3.4/marshal.c	2017-07-10 20:01:36.016807763 +0100
@@ -1024,7 +1024,7 @@
     struct dump_arg *arg;
     VALUE wrapper; /* used to avoid memory leak in case of exception */
 
-    wrapper = TypedData_Make_Struct(rb_cData, struct dump_arg, &dump_arg_data, arg);
+    wrapper = TypedData_Make_Struct(0, struct dump_arg, &dump_arg_data, arg);
     arg->dest = 0;
     arg->symbols = st_init_numtable();
     arg->data    = rb_init_identtable();
@@ -2059,7 +2059,7 @@
     else {
 	io_needed();
     }
-    wrapper = TypedData_Make_Struct(rb_cData, struct load_arg, &load_arg_data, arg);
+    wrapper = TypedData_Make_Struct(0, struct load_arg, &load_arg_data, arg);
     arg->infection = infection;
     arg->src = port;
     arg->offset = 0;
diff -ur ruby-2.3.4.orig/test/ruby/test_marshal.rb ruby-2.3.4/test/ruby/test_marshal.rb
--- ruby-2.3.4.orig/test/ruby/test_marshal.rb	2017-01-16 19:08:08.000000000 +0000
+++ ruby-2.3.4/test/ruby/test_marshal.rb	2017-07-10 20:02:08.710362086 +0100
@@ -645,6 +645,9 @@
     c = Bug9523.new
     assert_raise_with_message(RuntimeError, /Marshal\.dump reentered at marshal_dump/) do
       Marshal.dump(c)
+      GC.start
+      1000.times {"x"*1000}
+      GC.start
       c.cc.call
     end
   end
