diff -aur xmobar-0.23/xmobar.cabal xmobar-0.23.new/xmobar.cabal
--- xmobar-0.23/xmobar.cabal	2015-03-08 14:08:58.000000000 +0100
+++ xmobar-0.23.new/xmobar.cabal	2015-03-31 21:05:35.035293814 +0200
@@ -128,12 +128,12 @@
        cpp-options: -DGHC6
 
     if flag(with_xft) || flag(all_extensions)
-       build-depends: utf8-string == 0.3.*, X11-xft >= 0.2 && < 0.4
+       build-depends: utf8-string == 1.*, X11-xft >= 0.2 && < 0.4
        other-modules: MinXft
        cpp-options: -DXFT
 
     if flag(with_utf8) || flag(all_extensions)
-       build-depends: utf8-string == 0.3.*
+       build-depends: utf8-string == 1.*
        cpp-options: -DUTF8
 
     if flag(with_inotify) || flag(all_extensions)
