diff --git a/plugins/backend/local/localInterface.vala b/plugins/backend/local/localInterface.vala
index 121be42a..9508f73a 100644
--- a/plugins/backend/local/localInterface.vala
+++ b/plugins/backend/local/localInterface.vala
@@ -603,7 +603,10 @@ public class FeedReader.localInterface : Peas.ExtensionBase, FeedServerInterface
 					if(item.pub_date != null)
 					{
 						GLib.Time time = GLib.Time();
+						var savedlocale = GLib.Intl.setlocale(GLib.LocaleCategory.ALL);
+						GLib.Intl.setlocale(GLib.LocaleCategory.ALL, "C");
 						time.strptime(item.pub_date, "%a, %d %b %Y %H:%M:%S %Z");
+						GLib.Intl.setlocale(GLib.LocaleCategory.ALL, savedlocale);
 						date = new GLib.DateTime.local(1900 + time.year, 1 + time.month, time.day, time.hour, time.minute, time.second);
 
 						if(date == null)
