diff -rup clive-2.3.0.1.orig/examples/cliverc clive-2.3.0.1/examples/cliverc
--- clive-2.3.0.1.orig/examples/cliverc	2011-03-03 14:07:26.561534928 +0100
+++ clive-2.3.0.1/examples/cliverc	2011-03-03 14:07:37.407390980 +0100
@@ -17,7 +17,7 @@
 #   %u  Video download URL
 #   %f  Full path to video file (to write to)
 #   %n  File name (basename of the above)
---get-with "/usr/bin/curl -L -C - o %f %u"
+--get-with "/usr/bin/curl -L -C - -o %f %u"
 
 # Play downloaded video when done.
 #--exec "/usr/bin/vlc --quiet %f"
