diff -upr clang-2.8.orig/tools/libclang/CIndexer.cpp clang-2.8/tools/libclang/CIndexer.cpp
--- clang-2.8.orig/tools/libclang/CIndexer.cpp	2010-05-27 23:16:37.000000000 +0300
+++ clang-2.8/tools/libclang/CIndexer.cpp	2011-02-11 01:14:20.000000000 +0200
@@ -68,7 +68,8 @@ const llvm::sys::Path& CIndexer::getClan
 
   // We now have the CIndex directory, locate clang relative to it.
   CIndexPath.eraseComponent();
-  CIndexPath.appendComponent("..");
+  CIndexPath.eraseComponent();
+  CIndexPath.eraseComponent();
   CIndexPath.appendComponent("bin");
   CIndexPath.appendComponent("clang");
 #endif
