diff --color -aur root-6.26.02-old/tmva/sofie/inc/TMVA/SOFIE_common.hxx root-6.26.02-new/tmva/sofie/inc/TMVA/SOFIE_common.hxx
--- root-6.26.02-old/tmva/sofie/inc/TMVA/SOFIE_common.hxx	2022-05-17 16:47:05.503723641 +0300
+++ root-6.26.02-new/tmva/sofie/inc/TMVA/SOFIE_common.hxx	2022-05-17 16:47:41.023722760 +0300
@@ -6,6 +6,7 @@
 
 #include <type_traits>
 #include <cstdint>
+#include <cstring>
 #include <string>
 #include <vector>
 #include <memory>
