diff -wbBur shotwell-0.14.1/src/photos/GRaw.vala shotwell-0.14.1/src/photos/GRaw.vala
--- /home/sergej/tmp/BUILD/staging-i686/sergej/build/shotwell/src/shotwell-0.14.1/src/photos/GRaw.vala	2013-06-17 16:26:12.247179310 +0400
+++ shotwell-0.14.1/src/photos/GRaw.vala	2013-04-03 04:54:23.000000000 +0400
@@ -168,7 +168,7 @@
     }
     
     public void document_mode_processing() throws Exception {
-        throw_exception("document_mode_processing", 0);
+        throw_exception("document_mode_processing", proc.document_mode_processing());
     }
     
     public unowned LibRaw.ImageOther get_image_other() {
@@ -244,7 +244,7 @@
         // threshold
         output_params->half_size = half_size;
         // four_color_rgb
-//        output_params->document_mode = GRaw.DocMode.STANDARD;
+        output_params->document_mode = GRaw.DocMode.STANDARD;
         output_params->highlight = GRaw.HighlightMode.CLIP;
         output_params->use_auto_wb = true;
         output_params->use_camera_wb = true;
diff -wbBur shotwell-0.14.1/vapi/libraw.vapi shotwell-0.14.1/vapi/libraw.vapi
--- /home/sergej/tmp/BUILD/staging-i686/sergej/build/shotwell/src/shotwell-0.14.1/vapi/libraw.vapi	2013-06-17 16:25:23.623846536 +0400
+++ shotwell-0.14.1/vapi/libraw.vapi	2013-04-03 04:54:23.000000000 +0400
@@ -184,6 +184,8 @@
     public Processor(Options flags = Options.NONE);
     
     public Result adjust_sizes_info_only();
+    [CCode (cname="libraw_dcraw_document_mode_processing")]
+    public Result document_mode_processing();
     public unowned ImageOther get_image_other() { return other; }
     public unowned ImageParams get_image_params() { return idata; }
     public Progress get_progress_flags() { return progress_flags; }
