From 004cf52db5ff79fe1a083277c743fea4ac2e8321 Mon Sep 17 00:00:00 2001
From: Yevgeniy Filatov <itspec.ru@gmail.com>
Date: Sun, 28 Jun 2020 14:44:37 +0400
Subject: [PATCH] Fixed compilation on Qt 5.15

---
 src/paintcanvas.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/paintcanvas.cpp b/src/paintcanvas.cpp
index 7d3abe25..b2d9d76a 100644
--- a/src/paintcanvas.cpp
+++ b/src/paintcanvas.cpp
@@ -25,6 +25,7 @@
 #include <QImage>
 #include <QPainter>
 #include <QVariant>
+#include <QPainterPath>
 
 PaintCanvas::PaintCanvas(QWidget *parent)
     : QWidget(parent)
