--- IPython/core/magic.py.orig	2012-04-16 09:54:54.424655306 -0400
+++ IPython/core/magic.py	2012-04-16 09:58:50.701308885 -0400
@@ -1695,9 +1695,9 @@
                         # Start file run
                         print "NOTE: Enter 'c' at the",
                         print "%s prompt to start your script." % deb.prompt
+                        ns = {'execfile': py3compat.execfile, 'prog_ns': prog_ns}
                         try:
-                            deb.run('execfile("%s")' % filename, prog_ns)
-
+                            deb.run('execfile("%s", prog_ns)' % filename, ns)
                         except:
                             etype, value, tb = sys.exc_info()
                             # Skip three frames in the traceback: the %run one,
