diff --git a/test/test_copy.py b/test/test_copy.py
index e46b8d2..847fef3 100644
--- a/test/test_copy.py
+++ b/test/test_copy.py
@@ -71,10 +71,10 @@ def test_copy_from_with_error(db_table):
         arg = {
             'S': 'ERROR',
             'C': '22P02',
-            'M': 'invalid input syntax for integer: ""',
+            'M': 'invalid input syntax for type integer: ""',
             'W': 'COPY t1, line 2, column f1: ""',
             'F': 'numutils.c',
-            'R': 'pg_atoi'
+            'R': 'pg_strtoint32'
         }
         earg = e.value.args[0]
         for k, v in arg.items():
