From ff6875f6e45ca2fca7a21feaa537eb3fa715ea37 Mon Sep 17 00:00:00 2001
From: Andrew Bell <andrew.bell.ia@gmail.com>
Date: Tue, 24 Jul 2018 08:49:40 -0400
Subject: [PATCH] Constify python strings. Close #2099

---
 plugins/python/plang/Environment.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/plugins/python/plang/Environment.cpp b/plugins/python/plang/Environment.cpp
index 19f5c1ae7..0bbcf2d70 100644
--- a/plugins/python/plang/Environment.cpp
+++ b/plugins/python/plang/Environment.cpp
@@ -190,9 +190,9 @@ std::string getTraceback()
                 throw pdal::pdal_error("unable to get repr in getTraceback");
 #if PY_MAJOR_VERSION >= 3
             Py_ssize_t size;
-            char *d = PyUnicode_AsUTF8AndSize(r, &size);
+            const char *d = PyUnicode_AsUTF8AndSize(r, &size);
 #else
-            char *d = PyString_AsString(r);
+            const char *d = PyString_AsString(r);
 #endif
             mssg << d;
         }
@@ -208,9 +208,9 @@ std::string getTraceback()
             throw pdal::pdal_error("couldn't make string representation of traceback value");
 #if PY_MAJOR_VERSION >= 3
         Py_ssize_t size;
-        char *d = PyUnicode_AsUTF8AndSize(r, &size);
+        const char *d = PyUnicode_AsUTF8AndSize(r, &size);
 #else
-        char *d = PyString_AsString(r);
+        const char *d = PyString_AsString(r);
 #endif
         mssg << d;
     }
@@ -266,9 +266,9 @@ std::string readPythonString(PyObject* dict, const std::string& key)
         throw pdal::pdal_error("unable to get repr in readPythonString");
 #if PY_MAJOR_VERSION >= 3
     Py_ssize_t size;
-    char *d = PyUnicode_AsUTF8AndSize(r, &size);
+    const char *d = PyUnicode_AsUTF8AndSize(r, &size);
 #else
-    char *d = PyString_AsString(r);
+    const char *d = PyString_AsString(r);
 #endif
     ss << d;
 
