diff -drupN pkg/usr/lib/systemd/system/bacula-dir.service system/bacula-dir.service
--- pkg/usr/lib/systemd/system/bacula-dir.service	2012-07-09 14:00:54.067169449 -0400
+++ system/bacula-dir.service	2012-07-09 13:57:31.290315816 -0400
@@ -14,8 +14,8 @@
 [Unit]
 Description=Bacula Director Daemon service
 Alias=bacula-dir
-Requires=var-run.mount nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
-After=var-run.mount nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
+Requires=nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
+After=postgresql.service mysqld.service nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
 # Dependency about the database
 # We let administrators decide if they need it (if local db instance)
 # Wants=postgresql.service
diff -drupN pkg/usr/lib/systemd/system/bacula-fd.service system/bacula-fd.service
--- pkg/usr/lib/systemd/system/bacula-fd.service	2012-07-09 14:00:54.030502329 -0400
+++ system/bacula-fd.service	2012-07-09 13:54:50.240171480 -0400
@@ -13,8 +13,8 @@
 #
 [Unit]
 Description=Bacula File Daemon service
-Requires=var-run.mount nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
-After=var-run.mount nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
+Requires=nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
+After=nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
 # Wants=
 # Before=
 # Conflicts=
diff -drupN pkg/usr/lib/systemd/system/bacula-sd.service system/bacula-sd.service
--- pkg/usr/lib/systemd/system/bacula-sd.service	2012-07-09 14:00:54.043836219 -0400
+++ system/bacula-sd.service	2012-07-09 13:56:47.040276256 -0400
@@ -13,8 +13,8 @@
 #
 [Unit]
 Description=Bacula Storage Daemon service
-Requires=var-run.mount nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
-After=var-run.mount nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
+Requires=nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
+After=nss-lookup.target network.target remote-fs.target syslog.target time-sync.target
 # Wants=
 # Before=
 # Conflicts=
