--- plugins/libnotify/ario-notifier-libnotify.c
+++ plugins/libnotify/ario-notifier-libnotify.c
@@ -30,6 +30,10 @@
 #include "lib/ario-conf.h"
 #include "ario-debug.h"
 
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
+
 struct ArioNotifierLibnotifyPrivate
 {
         NotifyNotification *notification;
@@ -122,9 +126,13 @@
         ARIO_LOG_FUNCTION_START;
         notifier_libnotify->priv = ARIO_NOTIFIER_LIBNOTIFY_GET_PRIVATE (notifier_libnotify);
 
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+        notifier_libnotify->priv->notification = notify_notification_new ("Ario",  NULL, NULL);
+#else
         notifier_libnotify->priv->notification = notify_notification_new ("Ario",  NULL, NULL, NULL);
         notify_notification_attach_to_status_icon (notifier_libnotify->priv->notification,
                                                    GTK_STATUS_ICON (ario_tray_icon_get_instance ()));
+#endif
 }
 
 static void
