diff --git a/src/profiles.c b/src/profiles.c
index 4622c56..ab40ef4 100644
--- a/src/profiles.c
+++ b/src/profiles.c
@@ -287,7 +287,7 @@ dlna_guess_media_profile (dlna_t *dlna, const char *filename)
     return NULL;
   }
 
-  if (av_find_stream_info (ctx) < 0)
+  if (avformat_find_stream_info (ctx, NULL) < 0)
   {
     if (dlna->verbosity)
       fprintf (stderr, "can't find stream info\n");
@@ -334,7 +334,7 @@ dlna_guess_media_profile (dlna_t *dlna, const char *filename)
     p = p->next;
   }
 
-  av_close_input_file (ctx);
+  avformat_close_input (&ctx);
   free (codecs);
   return profile;
 }
