diff --git i/gio/tests/meson.build w/gio/tests/meson.build
index 5c0e03365..d69346216 100644
--- i/gio/tests/meson.build
+++ w/gio/tests/meson.build
@@ -586,6 +586,7 @@ if not meson.is_cross_build() or meson.has_exe_wrapper()
                  '-r',
                  '-b','binary',
                  '@INPUT@',
+                 '--oformat', 'elf32-i386',
                  '-o','@OUTPUT@'])
 
 	# Rename symbol to match the one in the C file
