diff --git i/lib/colord/meson.build w/lib/colord/meson.build
index dbb3d19..b3e102b 100644
--- i/lib/colord/meson.build
+++ w/lib/colord/meson.build
@@ -146,6 +146,7 @@ pkgg.generate(
   description : 'colord is a system daemon for managing color devices',
 )
 
+if false
 libcolord_girtarget = gnome.generate_gir(colord,
   sources : [
     'cd-client.c',
@@ -210,6 +211,7 @@ gnome.generate_vapi('colord',
   install: true,
 )
 endif
+endif
 
 if get_option('installed_tests')
 con2 = configuration_data()
diff --git i/lib/colorhug/meson.build w/lib/colorhug/meson.build
index 9526ea4..71a326b 100644
--- i/lib/colorhug/meson.build
+++ w/lib/colorhug/meson.build
@@ -73,6 +73,7 @@ pkgg.generate(
   description : 'ColorHug is a simple display hardware colorimeter',
 )
 
+if false
 libcolorhug_gir = gnome.generate_gir(colorhug,
   sources : [
     'ch-common.c',
@@ -107,6 +108,7 @@ libcolorhug_gir = gnome.generate_gir(colorhug,
   ],
   install : true
 )
+endif
 
 if get_option('tests')
   e = executable(
