diff -ruN a/src/daemon.c b/src/daemon.c
--- a/src/daemon.c	2019-05-10 20:53:51.000000000 +0200
+++ b/src/daemon.c	2020-04-26 13:34:02.064214277 +0200
@@ -1159,11 +1159,6 @@
 	ctx.backup_cms->log_priv = &ctx;
 	ctx.sd = -1;
 
-	if (getuid() != 0) {
-		fprintf(stderr, "pesignd must be started as root");
-		exit(1);
-	}
-
 	check_socket(&ctx);
 
 	openlog("pesignd", LOG_PID, LOG_DAEMON);
diff -ruN a/src/daemon.h b/src/daemon.h
--- a/src/daemon.h	2019-05-10 20:53:51.000000000 +0200
+++ b/src/daemon.h	2020-04-26 13:51:56.580675620 +0200
@@ -49,8 +49,8 @@
 } pesignd_cmd;
 
 #define PESIGND_VERSION 0x2a9edaf0
-#define SOCKPATH	"/var/run/pesign/socket"
-#define PIDFILE		"/var/run/pesign.pid"
+#define SOCKPATH	"/run/pesign/socket"
+#define PIDFILE		"/run/pesign/pesign.pid"
 
 static inline uint32_t UNUSED
 pesignd_string_size(char *buffer)
