--- volumeicon-0.4.6/src/alsa_backend.backup	2012-05-15 09:12:13.469335762 +0200
+++ volumeicon-0.4.6/src/alsa_backend.c	2012-05-15 09:12:31.862905251 +0200
@@ -22,10 +22,7 @@
 //##############################################################################
 
 #include <alsa/asoundlib.h>
-#include <glib/gstring.h>
-#include <glib/gstdio.h>
-#include <glib/glist.h>
-#include <glib/giochannel.h>
+#include <glib.h>
 
 #include "alsa_backend.h"
 
