diff -rupN intellij-community-pycharm-183.5429.31.orig/build/dependencies/setupJbre.gradle intellij-community-pycharm-183.5429.31/build/dependencies/setupJbre.gradle
--- intellij-community-pycharm-183.5429.31.orig/build/dependencies/setupJbre.gradle	2019-01-31 10:59:48.043742670 +0100
+++ intellij-community-pycharm-183.5429.31/build/dependencies/setupJbre.gradle	2019-01-31 11:02:00.181362763 +0100
@@ -1,7 +1,7 @@
 def setupJbre = task setupJbre {}
 
 def jdkRepo = 'https://cache-redirector.jetbrains.com/intellij-jdk'
-def targetOs = System.getProperty("intellij.build.target.os", "all")
+def targetOs = System.getProperty("intellij.build.target.os", "linux")
 def bundledJreVersion = System.getProperty("intellij.build.bundled.jre.version", "8").toInteger()
 def isBundledJreModular = bundledJreVersion >= 9
 def jreArtifactBuild = System.getProperty("intellij.build.bundled.jre.build", jdkBuild)
@@ -169,4 +169,4 @@ static def archToDownload(platform, is32
 static def modularJdkSpecificFiles(platform) {
   def prefix = platform == 'osx' ? 'jdk/Contents/Home/' : ''
   ["${prefix}jmods/**", "${prefix}include/**", '**/src.zip', "${prefix}lib/src.zip"]
-}
\ No newline at end of file
+}
diff -rupN intellij-community-pycharm-183.5429.31.orig/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy intellij-community-pycharm-183.5429.31/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy
--- intellij-community-pycharm-183.5429.31.orig/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy	2019-01-31 11:00:02.063843803 +0100
+++ intellij-community-pycharm-183.5429.31/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy	2019-01-31 11:00:42.494135478 +0100
@@ -19,7 +19,7 @@ class BuildOptions {
   /**
    * Specifies for which operating systems distributions should be built.
    */
-  String targetOS = System.getProperty("intellij.build.target.os", OS_ALL)
+  String targetOS = System.getProperty("intellij.build.target.os", OS_LINUX)
   static final String OS_LINUX = "linux"
   static final String OS_WINDOWS = "windows"
   static final String OS_MAC = "mac"
@@ -100,7 +100,7 @@ class BuildOptions {
    * Build number without product code (e.g. '162.500.10'), if {@code null} '&lt;baseline&gt;.SNAPSHOT' will be used. Use {@link BuildContext#buildNumber} to
    * get the actual build number in build scripts.
    */
-  String buildNumber = System.getProperty("build.number")
+  String buildNumber = System.getProperty("idea.build.number")
 
   /**
    * By default build process produces temporary and resulting files under projectHome/out/productName directory, use this property to
