diff -wbBur rawstudio-2.0/plugins/load-gdk/exiv2-colorspace.cpp rawstudio-2.0.my/plugins/load-gdk/exiv2-colorspace.cpp
--- rawstudio-2.0/plugins/load-gdk/exiv2-colorspace.cpp	2011-02-24 01:35:18.000000000 +0300
+++ rawstudio-2.0.my/plugins/load-gdk/exiv2-colorspace.cpp	2012-01-23 15:03:22.000000000 +0400
@@ -102,8 +102,8 @@
 	{
 		*linear_guess = FALSE;
 		RSColorSpace* profile = NULL;
-		const gchar *icc_profile_title;
-		const gchar *icc_profile;
+		png_charp icc_profile_title;
+		png_bytep icc_profile;
 		guint icc_profile_size;
 		png_structp png_ptr = png_create_read_struct(
 				PNG_LIBPNG_VER_STRING,
@@ -125,11 +125,11 @@
 
 					int compression_type;
 					/* Extract embedded ICC profile */
-					if (info_ptr->valid & PNG_INFO_iCCP)
+					if (png_get_valid(png_ptr, info_ptr, PNG_INFO_iCCP))
 					{
 						png_uint_32 retval = png_get_iCCP (png_ptr, info_ptr,
-													(png_charpp) &icc_profile_title, &compression_type,
-													(png_charpp) &icc_profile, (png_uint_32*) &icc_profile_size);
+													&icc_profile_title, &compression_type,
+													&icc_profile, (png_uint_32*) &icc_profile_size);
 						if (retval != 0)
 						{
 							RSIccProfile *icc = rs_icc_profile_new_from_memory((gchar*)icc_profile, icc_profile_size, TRUE);
diff -wbBur rawstudio-2.0/plugins/output-pngfile/output-pngfile.c rawstudio-2.0.my/plugins/output-pngfile/output-pngfile.c
--- rawstudio-2.0/plugins/output-pngfile/output-pngfile.c	2011-03-26 04:52:29.000000000 +0300
+++ rawstudio-2.0.my/plugins/output-pngfile/output-pngfile.c	2012-01-23 15:04:59.000000000 +0400
@@ -23,6 +23,7 @@
 #include <gettext.h>
 #include "config.h"
 #include <png.h>
+#include <zlib.h>
 
 #define RS_TYPE_PNGFILE (rs_pngfile_type)
 #define RS_PNGFILE(obj) (G_TYPE_CHECK_INSTANCE_CAST ((obj), RS_TYPE_PNGFILE, RSPngfile))
