diff -rup minidlna-1.0.24/metadata.c minidlna-1.0.24.new/metadata.c
--- minidlna-1.0.24/metadata.c	2012-02-14 19:44:56.000000000 +0100
+++ minidlna-1.0.24.new/metadata.c	2012-07-10 10:28:01.373163002 +0200
@@ -1514,10 +1548,10 @@ GetVideoMetadata(const char * path, char
 	{
 		if( ctx->metadata )
 		{
-			AVMetadataTag *tag = NULL;
+			AVDictionaryEntry *tag = NULL;
 
 			//DEBUG DPRINTF(E_DEBUG, L_METADATA, "Metadata:\n");
-			while( (tag = av_metadata_get(ctx->metadata, "", tag, AV_METADATA_IGNORE_SUFFIX)) )
+			while( (tag = av_dict_get(ctx->metadata, "", tag, AV_DICT_IGNORE_SUFFIX)) )
 			{
 				//DEBUG DPRINTF(E_DEBUG, L_METADATA, "  %-16s: %s\n", tag->key, tag->value);
 				if( strcmp(tag->key, "title") == 0 )
