diff -wbBur sarg-2.3.11/index.c sarg-2.3.11.my/index.c
--- sarg-2.3.11/index.c	2018-01-14 21:00:22.000000000 +0300
+++ sarg-2.3.11.my/index.c	2018-02-19 12:20:15.896203347 +0300
@@ -208,7 +208,7 @@
 			m1=month / 16;
 			if(month % 16 != 0) {
 				m2=month % 16;
-				sprintf(monthnum,"%02d-%02d",m1,m2);
+				sprintf(monthnum,"%02u-%02u",(unsigned int)m1,(unsigned int)m2);
 				sprintf(monthname1,"%02d",m1);
 				sprintf(monthname2,"%02d",m2);
 				name_month(monthname1,sizeof(monthname1));
@@ -269,7 +269,7 @@
 				d1=day / 32;
 				if(day % 32 != 0) {
 					d2=day % 32;
-					sprintf(daynum,"%02d-%02d",d1,d2);
+					sprintf(daynum,"%02u-%02u",(unsigned int)d1,(unsigned int)d2);
 				} else {
 					sprintf(daynum,"%02d",d1);
 				}
diff -wbBur sarg-2.3.11/report.c sarg-2.3.11.my/report.c
--- sarg-2.3.11/report.c	2018-01-14 21:00:23.000000000 +0300
+++ sarg-2.3.11.my/report.c	2018-02-19 12:18:45.151207192 +0300
@@ -54,7 +54,7 @@
 	char accsmart[MAXLEN];
 	char crc2[MAXLEN/2 -1];
 	char siteind[MAX_TRUNCATED_URL];
-	char arqtt[256];
+	char arqtt[267];
 	char *oldurltt=NULL;
 	char oldaccdiatt[11],oldacchoratt[9];
 	char tmp3[MAXLEN];
diff -wbBur sarg-2.3.11/userinfo.c sarg-2.3.11.my/userinfo.c
--- sarg-2.3.11/userinfo.c	2013-06-01 22:02:04.000000000 +0400
+++ sarg-2.3.11.my/userinfo.c	2018-02-19 12:21:16.103200796 +0300
@@ -67,7 +67,7 @@
 	int skip;
 	int flen;
 	int count, clen;
-	char cstr[9];
+	char cstr[11];
 
 	last=NULL;
 	for (group=first_user_group ; group ; group=group->next) {
