diff -urN mysql-workbench-oss-5.1.12.orig/ext/cppconn/driver/mysql_art_resultset.cpp mysql-workbench-oss-5.1.12/ext/cppconn/driver/mysql_art_resultset.cpp
--- mysql-workbench-oss-5.1.12.orig/ext/cppconn/driver/mysql_art_resultset.cpp	2009-04-23 19:00:53.000000000 +0300
+++ mysql-workbench-oss-5.1.12/ext/cppconn/driver/mysql_art_resultset.cpp	2009-06-04 22:23:06.751993650 +0300
@@ -16,6 +16,7 @@
 
 #include <cppconn/exception.h>
 #include <sstream>
+#include <cstdio>
 #include <math.h>
 
 
diff -urN mysql-workbench-oss-5.1.12.orig/ext/cppconn/driver/mysql_resultbind.cpp mysql-workbench-oss-5.1.12/ext/cppconn/driver/mysql_resultbind.cpp
--- mysql-workbench-oss-5.1.12.orig/ext/cppconn/driver/mysql_resultbind.cpp	2009-04-23 19:00:53.000000000 +0300
+++ mysql-workbench-oss-5.1.12/ext/cppconn/driver/mysql_resultbind.cpp	2009-06-04 22:23:06.755298749 +0300
@@ -14,6 +14,7 @@
 #include "mysql_resultbind.h"
 
 #include <string.h>
+#include <cstdio>
 
 namespace sql
 {
diff -urN mysql-workbench-oss-5.1.12.orig/library/canvas/src/mdc_common.h mysql-workbench-oss-5.1.12/library/canvas/src/mdc_common.h
--- mysql-workbench-oss-5.1.12.orig/library/canvas/src/mdc_common.h	2009-04-23 19:00:55.000000000 +0300
+++ mysql-workbench-oss-5.1.12/library/canvas/src/mdc_common.h	2009-06-04 22:23:06.755298749 +0300
@@ -27,6 +27,7 @@
 #endif
 #endif
 
+#include <cstdio>
 
 #include "mdc_canvas_public.h"
 
diff -urN mysql-workbench-oss-5.1.12.orig/library/forms/filechooser.cpp mysql-workbench-oss-5.1.12/library/forms/filechooser.cpp
--- mysql-workbench-oss-5.1.12.orig/library/forms/filechooser.cpp	2009-04-23 19:00:55.000000000 +0300
+++ mysql-workbench-oss-5.1.12/library/forms/filechooser.cpp	2009-06-04 22:23:06.755298749 +0300
@@ -1,6 +1,8 @@
 
 #include "stdafx.h"
 
+#include <cstdio>
+
 #include <mforms/mforms.h>
 
 
diff -urN mysql-workbench-oss-5.1.12.orig/library/forms/mforms.cpp mysql-workbench-oss-5.1.12/library/forms/mforms.cpp
--- mysql-workbench-oss-5.1.12.orig/library/forms/mforms.cpp	2009-04-23 19:00:55.000000000 +0300
+++ mysql-workbench-oss-5.1.12/library/forms/mforms.cpp	2009-06-04 22:23:06.755298749 +0300
@@ -3,7 +3,7 @@
 #include <string.h>
 #include "mforms/mforms.h"
 #include "mforms/simpleform.h"
-
+#include <cstdio>
 using namespace mforms;
 
 ControlFactory *ControlFactory::get_instance()
