--- configure.in.orig	2010-12-08 16:58:57.456378836 -0800
+++ configure.in	2010-12-08 16:59:42.179715448 -0800
@@ -280,13 +280,13 @@
 AC_CHECK_PROG(PYTHON27, python2.7, [yes])
 # Python headers
 AC_MSG_CHECKING(for Python libraries)
-PYTHON_LIBS=[$(python -c "from distutils import sysconfig
+PYTHON_LIBS=[$(python2 -c "from distutils import sysconfig
 import os 
 cfg=sysconfig.get_config_vars()
 lib = os.path.join(cfg['LIBDIR'], cfg['LDLIBRARY'])
 if os.path.exists(lib):
     print lib")]
-PYTHON_CFLAGS=[$(python -c "from distutils import sysconfig
+PYTHON_CFLAGS=[$(python2 -c "from distutils import sysconfig
 print sysconfig.get_python_inc()")]
 PYTHON_CFLAGS="-I$PYTHON_CFLAGS"
 if test "$PYTHON_LIBS" = ""; then
