diff -wbBur phpLDAPadmin-1.2.6.4/lib/functions.php phpLDAPadmin-1.2.6.4.my/lib/functions.php
--- phpLDAPadmin-1.2.6.4/lib/functions.php	2022-12-21 18:43:40.596996754 +0300
+++ phpLDAPadmin-1.2.6.4.my/lib/functions.php	2022-12-21 18:43:30.240329853 +0300
@@ -2649,7 +2649,7 @@
  * encoded) into \<hexcode>.
  */
 function ldap_explode_dn_patch($dn,$with_attrib) {
-	$result = ldap_explode_dn($dn,$with_attrib);
+	$result = ldap_explode_dn((string)$dn,$with_attrib);
 	if (! $result)
 		return null;
 
