--- configure~	2010-04-27 22:21:56.902812388 +0300
+++ configure	2010-04-27 22:22:24.369476961 +0300
@@ -394,7 +394,7 @@
       elif [ $cc_maj -lt 3 \
              -o \( $cc_maj -eq 3 -a $cc_min -lt 2 \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 1 -a "$OS" != "darwin" \) \
-             -o \( $cc_maj -eq 4 -a $cc_min -gt 4 \) \
+             -o \( $cc_maj -eq 4 -a $cc_min -gt 5 \) \
              -o $cc_maj -gt 4 ]; then
         log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<5"
         fail really
