--- setup.py.orig	2010-05-03 11:02:53.551508139 +0200
+++ setup.py	2010-05-03 11:03:09.681918067 +0200
@@ -6,17 +6,6 @@
 import sys
 
 if os.name == 'posix':
-    for arg in sys.argv:
-        if arg == 'install': 
-            print 'Hello.\nYou are trying to do a system-wide install of emesene '\
-                  'using this script, which is a very bad thing to do.\n'\
-                  'Seriously, you do NOT want to do this, since it can break '\
-                  'other python apps, and emesene too!\n'\
-                  'Follow my advice: just run the \"emesene\" script that is '\
-                  'in this very same directory and you\'re done. emesene is running '\
-                  'and your system is safe. It\'s a win-win, don\'t you think?\n'\
-                  'Thanks for trying emesene.'
-            quit()
 
     # From apport's setup.py
     mo_files = []
