--- setup.py.old	2010-01-07 19:04:39.000000000 +0100
+++ setup.py	2010-01-07 19:04:50.000000000 +0100
@@ -5,17 +5,6 @@
 import sys
 
 if os.name == 'posix':
-    for arg in sys.argv:
-        if arg == 'install': 
-            print 'Hello.\nYou are trying to do a system-wide install of emesene '\
-                  'using this script, which is a very bad thing to do.\n'\
-                  'Seriously, you do NOT want to do this, since it can break '\
-                  'other python apps, and emesene too!\n'\
-                  'Follow my advice: just run the \"emesene\" script that is '\
-                  'in this very same directory and you\'re done. emesene is running '\
-                  'and your system is safe. It\'s a win-win, don\'t you think?\n'\
-                  'Thanks for trying emesene.'
-            quit()
             
     # From apport's setup.py
     mo_files = []
