diff -Naurp a/build/CMakeLists.txt b/build/CMakeLists.txt
--- a/build/CMakeLists.txt	2020-02-25 17:08:50.000000000 +0000
+++ b/build/CMakeLists.txt	2020-02-29 14:47:28.000000000 +0000
@@ -488,20 +488,7 @@ if (WITH_BINARY)
   endif ()
 endif ()
 
-install(FILES ../LICENSE
-  DESTINATION .
-  COMPONENT Runtime
-  )
-# Take a copy on Appveyor
-install(FILES "C:/projects/openssl-$ENV{OPENSSL}/LICENSE"
-  DESTINATION .
-  COMPONENT Runtime
-  RENAME LICENSE_OPENSSL
-  OPTIONAL                      # for local builds only!
-  )
-
 file(GLOB_RECURSE I2PD_SOURCES "../libi2pd/*.cpp" "../libi2pd_client/*.cpp" "../daemon/*.cpp" "../build" "../Win32" "../Makefile*")
-install(FILES ${I2PD_SOURCES} DESTINATION src/ COMPONENT Source)
 # install(DIRECTORY ../ DESTINATION src/
 #   # OPTIONAL
 #   COMPONENT Source FILES_MATCHING
@@ -509,8 +496,8 @@ install(FILES ${I2PD_SOURCES} DESTINATIO
 #   PATTERN "*.cpp"
 #   )
 
-file(GLOB I2PD_HEADERS "../libi2pd/*.h" "../libi2pd_client/*.h" "../daemon/*.h")
-install(FILES ${I2PD_HEADERS} DESTINATION src/ COMPONENT Headers)
+file(GLOB I2PD_HEADERS "../libi2pd/*.h" "../libi2pd_client/*.h")
+install(FILES ${I2PD_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/i2pd COMPONENT Headers)
 # install(DIRECTORY ../ DESTINATION src/
 #   # OPTIONAL
 #   COMPONENT Headers FILES_MATCHING
