diff -wbBur SimGear-2.0.0/simgear/scene/model/SGClipGroup.cxx SimGear-2.0.0.my/simgear/scene/model/SGClipGroup.cxx
--- SimGear-2.0.0/simgear/scene/model/SGClipGroup.cxx	2010-02-17 18:39:21.000000000 +0300
+++ SimGear-2.0.0.my/simgear/scene/model/SGClipGroup.cxx	2011-07-04 13:02:57.000000000 +0400
@@ -70,9 +70,6 @@
   osg::ref_ptr<osg::RefMatrix> mModelView;
 };
 
-osgUtil::RegisterRenderBinProxy
-SGClipGroup::clipBinProxy("ClipRenderBin", new SGClipGroup::ClipRenderBin);
-
 class SGClipGroup::CullCallback : public osg::NodeCallback {
 public:
   virtual void operator()(osg::Node* node, osg::NodeVisitor* nv)
diff -wbBur SimGear-2.0.0/simgear/scene/model/SGClipGroup.hxx SimGear-2.0.0.my/simgear/scene/model/SGClipGroup.hxx
--- SimGear-2.0.0/simgear/scene/model/SGClipGroup.hxx	2010-02-17 18:39:21.000000000 +0300
+++ SimGear-2.0.0.my/simgear/scene/model/SGClipGroup.hxx	2011-07-04 13:03:01.000000000 +0400
@@ -51,8 +51,6 @@
   class ClipRenderBin;
 
   std::vector<osg::ref_ptr<osg::ClipPlane> > mClipPlanes;
-
-  static osgUtil::RegisterRenderBinProxy clipBinProxy;
 };
 
 #endif
diff -wbBur SimGear-2.0.0/simgear/scene/model/SGPagedLOD.cxx SimGear-2.0.0.my/simgear/scene/model/SGPagedLOD.cxx
--- SimGear-2.0.0/simgear/scene/model/SGPagedLOD.cxx	2010-02-17 18:39:21.000000000 +0300
+++ SimGear-2.0.0.my/simgear/scene/model/SGPagedLOD.cxx	2011-07-12 15:12:21.000000000 +0400
@@ -21,6 +21,7 @@
 #include <osgDB/ReadFile>
 #include <osgDB/Input>
 #include <osgDB/ParameterOutput>
+#include <osgDB/DatabasePager>
 
 #include <simgear/debug/logstream.hxx>
 #include <simgear/structure/OSGVersion.hxx>
@@ -67,9 +68,12 @@
     unsigned childNum = getNumChildren();
     setTimeStamp(childNum, 0);
     double priority=1.0;
+/*
+FIXME:!!!
     dbp->requestNodeFile(getFileName(childNum),this,priority,0,
                          getDatabaseRequest(childNum),
                          _readerWriterOptions.get());
+*/
 }
 
 bool SGPagedLOD_writeLocalData(const Object& obj, osgDB::Output& fw)
diff -wbBur SimGear-2.0.0/simgear/scene/sky/bbcache.cxx SimGear-2.0.0.my/simgear/scene/sky/bbcache.cxx
--- SimGear-2.0.0/simgear/scene/sky/bbcache.cxx	2010-02-17 18:39:21.000000000 +0300
+++ SimGear-2.0.0.my/simgear/scene/sky/bbcache.cxx	2011-07-12 15:14:16.000000000 +0400
@@ -31,6 +31,7 @@
 #include <simgear/screen/extensions.hxx>
 #include <simgear/screen/RenderTexture.h>
 #include <osg/GLU>
+#include <GL/glu.h>
 
 #include "bbcache.hxx"
 
diff -wbBur SimGear-2.0.0/simgear/screen/tr.cxx SimGear-2.0.0.my/simgear/screen/tr.cxx
--- SimGear-2.0.0/simgear/screen/tr.cxx	2010-02-17 18:39:22.000000000 +0300
+++ SimGear-2.0.0.my/simgear/screen/tr.cxx	2011-07-12 15:16:01.000000000 +0400
@@ -145,6 +145,7 @@
 #endif
 
 #include <osg/GLU>
+#include <GL/glu.h>
 
 #include "tr.h"
 
