--- gnutls/library/__init__.py	2013-12-12 20:29:45.000000000 +0400
+++ gnutls.fixed/library/__init__.py	2013-12-17 20:56:48.637256402 +0400
@@ -104,10 +104,11 @@
         libgcrypt = load_library(name='gcrypt', version=11)
         gcry_control = libgcrypt.gcry_control
     else:
-        gcry_control = libgnutls.gcry_control
+        libgcrypt = load_library(name='gcrypt', version=11)
+        gcry_control = libgcrypt.gcry_control
 
     gcry_control(GCRYCTL_SET_THREAD_CBS, c_void_p(gcrypt_thread_callbacks_ptr))
-    libgnutls.gcry_check_version('1.2.4')    # GNUTLS_MIN_LIBGCRYPT_VERSION
+    libgcrypt.gcry_check_version('1.2.4')    # GNUTLS_MIN_LIBGCRYPT_VERSION
     if system == 'cygwin':
         gcry_control(GCRYCTL_DISABLE_SECMEM, 0)
     else:
@@ -118,12 +119,10 @@
 
 
 
-libgnutls = load_library(name='gnutls', version=26)
-libgnutls_extra = load_library(name='gnutls-extra', version=26)
+libgnutls = load_library(name='gnutls', version=28)
 
 initialize_gcrypt()
 libgnutls.gnutls_global_init()
-libgnutls_extra.gnutls_global_init_extra()
 
 
 from gnutls.library import constants
@@ -137,9 +136,6 @@
 if functions.gnutls_check_version(__need_version__) is None:
     version = functions.gnutls_check_version(None)
     raise RuntimeError("Found GNUTLS library version %s, but at least version %s is required" % (version, __need_version__))
-if functions.gnutls_extra_check_version(__need_version__) is None:
-    version = functions.gnutls_extra_check_version(None)
-    raise RuntimeError("Found GNUTLS extra library version %s, but at least version %s is required" % (version, __need_version__))
 
 
 del get_system_name, library_locations, load_library, initialize_gcrypt
