commit 5931bc0231c91ecdfc5dd313d8cce578eae426bb
Author: Eliot Horowitz <eliot@10gen.com>
Date:   Wed Mar 30 01:01:18 2011 -0400

    remove unused variable

diff --git a/db/commands.cpp b/db/commands.cpp
index c301fb3..0bbd765 100644
--- a/db/commands.cpp
+++ b/db/commands.cpp
@@ -127,7 +127,6 @@ namespace mongo {
         if ( strcmp(p, ".$cmd") != 0 ) return false;
 
         bool ok = false;
-        bool valid = false;
 
         BSONElement e = jsobj.firstElement();
         map<string,Command*>::iterator i;
@@ -138,7 +137,6 @@ namespace mongo {
            migrated over to the command object format.
            */
         else if ( (i = _commands->find(e.fieldName())) != _commands->end() ) {
-            valid = true;
             string errmsg;
             Command *c = i->second;
             if ( c->adminOnly() && !startsWith(ns, "admin.") ) {
