--- 85-wqy-bitmapsong.conf.org	2007-11-04 08:55:50.000000000 +0300
+++ 85-wqy-bitmapsong.conf	2012-09-25 19:14:27.000000000 +0400
@@ -25,10 +25,20 @@
     <match target="pattern">
         <test equal="any" compare="eq" name="lang">
                 <string>en</string>
+        </test> 
+        <test equal="any" compare="eq" name="lang">
                 <string>en-us</string>
+        </test> 
+        <test equal="any" compare="eq" name="lang">
                 <string>zh-cn</string>
+        </test> 
+        <test equal="any" compare="eq" name="lang">
                 <string>zh-tw</string>
+        </test> 
+        <test equal="any" compare="eq" name="lang">
                 <string>zh-hk</string>
+        </test> 
+        <test equal="any" compare="eq" name="lang">
                 <string>zh-sg</string>
         </test> 
         <test compare="more_eq" name="pixelsize">
@@ -47,10 +57,20 @@
     <match target="pattern">
         <test equal="any" compare="eq" name="lang">
                 <string>en</string>
+        </test> 
+        <test equal="any" compare="eq" name="lang">
                 <string>en-us</string>
+        </test> 
+        <test equal="any" compare="eq" name="lang">
                 <string>zh-cn</string>
+        </test> 
+        <test equal="any" compare="eq" name="lang">
                 <string>zh-tw</string>
+        </test> 
+        <test equal="any" compare="eq" name="lang">
                 <string>zh-hk</string>
+        </test> 
+        <test equal="any" compare="eq" name="lang">
                 <string>zh-sg</string>
         </test>
         <test compare="more_eq" name="size">
