diff -up kftpgrabber-0.8.1/kftpgrabber/src/engine/ssl.cpp.openssl kftpgrabber-0.8.1/kftpgrabber/src/engine/ssl.cpp
--- kftpgrabber-0.8.1/kftpgrabber/src/engine/ssl.cpp.openssl	2007-04-28 13:35:22.000000000 +0200
+++ kftpgrabber-0.8.1/kftpgrabber/src/engine/ssl.cpp	2009-08-22 15:29:28.000000000 +0200
@@ -169,7 +169,11 @@ bool Ssl::setClientCertificate(KSSLPKCS1
 
 void Ssl::setConnectionInfo()
 {
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+  const SSL_CIPHER *cipher;
+#else
   SSL_CIPHER *cipher;
+#endif
   char buffer[1024];
   
   buffer[0] = 0;
