diff -Naur ./nemo-fileroller.orig/configure.in ./nemo-fileroller/configure.in
--- ./nemo-fileroller.orig/configure.in	2012-09-20 19:48:01.298714400 -0400
+++ ./nemo-fileroller/configure.in	2012-09-20 19:48:08.375136044 -0400
@@ -29,10 +29,10 @@
 PKG_CHECK_MODULES(NEMO, libnemo-extension >= $NEMO_REQUIRED)
 PKG_CHECK_MODULES(GLIB, glib-2.0 >= $GLIB_REQUIRED)
 
-AC_PATH_PROG([PYTHON], [python])
+#AC_PATH_PROG([PYTHON], [python])
 
-AC_PATH_PROG([RST2MAN], [rst2man], [python rst2man.py])
-AC_SUBST(RST2MAN)
+#AC_PATH_PROG([RST2MAN], [rst2man], [python rst2man.py])
+#AC_SUBST(RST2MAN)
 
 # define module checking macro
 AC_DEFUN([PYTHON_CHECK_MODULE], [
@@ -55,9 +55,9 @@
 fi
 ])
 
-PYTHON_CHECK_MODULE(pygtk, gtk)
-PYTHON_CHECK_MODULE(gobject, gobject)
-PYTHON_CHECK_MODULE(docutils, docutils)
+#PYTHON_CHECK_MODULE(pygtk, gtk)
+#PYTHON_CHECK_MODULE(gobject, gobject)
+#PYTHON_CHECK_MODULE(docutils, docutils)
 
 # Make dependency CFLAGS and LIBS available
 AC_SUBST(NEMO_CFLAGS)
