From a6e2ac9d06cc95a33450e65e82ece128c9f7f782 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 24 Apr 2017 17:46:54 +0200
Subject: [PATCH] brightness: update D-Bus service name for GNOME 3.24

---
 brightness/src/brightness-applet.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/brightness/src/brightness-applet.c b/brightness/src/brightness-applet.c
index 6378c90f6..5970a9420 100644
--- a/brightness/src/brightness-applet.c
+++ b/brightness/src/brightness-applet.c
@@ -63,7 +63,7 @@ typedef struct{
 
 GType                gpm_brightness_applet_get_type  (void);
 
-#define GSD_DBUS_SERVICE	"org.gnome.SettingsDaemon"
+#define GSD_DBUS_SERVICE	"org.gnome.SettingsDaemon.Power"
 #define GSD_DBUS_PATH_POWER	"/org/gnome/SettingsDaemon/Power"
 
 G_DEFINE_TYPE (GpmBrightnessApplet, gpm_brightness_applet, PANEL_TYPE_APPLET)
-- 
2.12.2

