diff -up gnome-applets-3.1.90/configure.in.cpupower gnome-applets-3.1.90/configure.in
--- gnome-applets-3.1.90/configure.in.cpupower	2011-08-31 18:49:00.628945807 -0400
+++ gnome-applets-3.1.90/configure.in	2011-08-31 18:49:17.397946966 -0400
@@ -518,7 +518,7 @@ AC_CHECK_HEADER(cpufreq.h, have_libcpufr
 LIBCPUFREQ_LIBS=
 if test "x$have_libcpufreq" = "xyes"; then
    AC_DEFINE([HAVE_LIBCPUFREQ], [1], [Have libcpufreq.])
-   LIBCPUFREQ_LIBS=-lcpufreq
+   LIBCPUFREQ_LIBS=-lcpupower
 fi
 AM_CONDITIONAL(HAVE_LIBCPUFREQ, test x$have_libcpufreq = xyes)
 AC_SUBST(LIBCPUFREQ_LIBS)
