diff -upr fltk-2.0.x-r6829.orig/images/fl_png.cxx fltk-2.0.x-r6829/images/fl_png.cxx
--- fltk-2.0.x-r6829.orig/images/fl_png.cxx	2010-01-23 04:42:12.000000000 +0200
+++ fltk-2.0.x-r6829/images/fl_png.cxx	2010-01-23 04:58:31.000000000 +0200
@@ -62,7 +62,7 @@ bool pngImage::test(const uchar* datas, 
 #if !HAVE_LIBPNG
   return 0;
 #else
-  return png_check_sig((png_byte*)datas, (int)size)!=0;
+  return png_sig_cmp((png_byte*)datas, 0, (int)size)==0;
 #endif
 }
 
