diff --git a/gb.pdf/src/CPdfDocument.cpp b/gb.pdf/src/CPdfDocument.cpp
index 0610a65d8..f9319389b 100644
--- a/gb.pdf/src/CPdfDocument.cpp
+++ b/gb.pdf/src/CPdfDocument.cpp
@@ -40,7 +40,6 @@
 #include <TextOutputDev.h>
 #include <SplashOutputDev.h>
 #include <splash/SplashBitmap.h>
-#include <goo/GooList.h>
 #include <Outline.h>
 #include <Link.h>
 #include <Gfx.h>
@@ -833,7 +832,7 @@ BEGIN_METHOD_VOID(PDFINDEX_parent)
 
 	if (!THIS->pindex) { GB.ReturnBoolean(true); return; }
 
-	THIS->index=(GooList*)THIS->pindex[GB.Count(POINTER(THIS->pindex))-1];
+	THIS->index=(const std::vector<OutlineItem*>*)THIS->pindex[GB.Count(POINTER(THIS->pindex))-1];
 	THIS->currindex=THIS->oldindex[GB.Count(POINTER(THIS->pindex))-1];
 	if (GB.Count(POINTER(THIS->pindex))==1)
 	{
diff --git a/gb.pdf/src/CPdfDocument.h b/gb.pdf/src/CPdfDocument.h
index fa80d47a9..e9afdbf81 100644
--- a/gb.pdf/src/CPdfDocument.h
+++ b/gb.pdf/src/CPdfDocument.h
@@ -29,7 +29,7 @@
 #include <PDFDoc.h>
 #include <SplashOutputDev.h>
 #include <Page.h>
-#include <goo/GooList.h>
+#include <Outline.h>
 #include <stdint.h>
 
 #if POPPLER_VERSION_0_64
@@ -93,7 +92,7 @@ typedef
 		uint currpage;
 
 		void **pindex;            // Parent of current index entries
-		const_GooList *index;     // Current entries
+		const std::vector<OutlineItem*> *index;     // Current entries
 		
 		uint currindex;           // Current entry
 		uint *oldindex;           // Parent entry
