diff -upr whelk-2.7.1.orig/whelk/subprocess_34.py whelk-2.7.1/whelk/subprocess_34.py
--- whelk-2.7.1.orig/whelk/subprocess_34.py	2017-04-18 16:01:38.000000000 +0300
+++ whelk-2.7.1/whelk/subprocess_34.py	2019-11-06 17:46:00.946759802 +0200
@@ -92,7 +92,10 @@ class Popen(subprocess.Popen):
                         raise TimeoutExpired(self.args, orig_timeout)
 
                     ready = selector.select(timeout)
-                    self._check_timeout(endtime, orig_timeout)
+                    if sys.version_info[:2] >= (3,8):
+                        self._check_timeout(endtime, orig_timeout, stdout, stderr)
+                    else:
+                        self._check_timeout(endtime, orig_timeout)
 
                     # XXX Rewrite these to use non-blocking I/O on the file
                     # objects; they are no longer using C stdio!
