diff -ruN a/.flake8 b/.flake8
--- a/.flake8	1970-01-01 01:00:00.000000000 +0100
+++ b/.flake8	2022-08-12 19:43:01.553759779 +0200
@@ -0,0 +1,14 @@
+[flake8]
+max-line-length = 79
+output-file = flake8.txt
+max-complexity = 10
+exclude =
+    .git
+    .tox
+    .venv
+    .cache
+    .eggs
+    .pytest*
+    *.egg-info
+    examples
+    doc
diff -ruN a/.gitignore b/.gitignore
--- a/.gitignore	2019-03-25 02:16:20.000000000 +0100
+++ b/.gitignore	2022-08-12 19:43:01.560426406 +0200
@@ -1,4 +1,5 @@
 *.eggs/
+*.egg-info/
 *__pycache__/
 *.db
 /mailman-web.toml
diff -ruN a/.gitlab-ci.yml b/.gitlab-ci.yml
--- a/.gitlab-ci.yml	2019-03-25 02:16:20.000000000 +0100
+++ b/.gitlab-ci.yml	2022-08-12 19:43:01.560426406 +0200
@@ -1,6 +1,11 @@
+---
+
 image: maxking/mailman-ci-runner
 
 
 docs:
   script:
     - tox -e docs
+lint:
+  script:
+    - tox -e lint
diff -ruN a/docs/conf.py b/docs/conf.py
--- a/docs/conf.py	2021-01-20 01:10:36.000000000 +0100
+++ b/docs/conf.py	2022-08-12 19:43:01.560426406 +0200
@@ -65,7 +65,7 @@
 #
 # This is also used if you do content translation via gettext catalogs.
 # Usually you set "language" from the command line for these cases.
-language = None
+language = 'en'
 
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
@@ -194,6 +194,7 @@
 
 add_module_names = False
 
+
 # -- Modify the docstring
 def process_docstring(app, what, name, obj, options, lines):
     if what not in ('module'):
@@ -205,5 +206,6 @@
 """.format(pp.pformat(obj)).splitlines())
     return
 
+
 def setup(app):
     app.connect('autodoc-process-docstring', process_docstring)
diff -ruN a/docs/settings.py b/docs/settings.py
--- a/docs/settings.py	2021-01-20 01:10:36.000000000 +0100
+++ b/docs/settings.py	2022-08-12 19:43:01.560426406 +0200
@@ -1,8 +1,8 @@
 # Mailman Web configuration file.
 # /etc/mailman3/settings.py
 
-from mailman_web.settings.base import *
-from mailman_web.settings.mailman import *
+from mailman_web.settings.base import *  # noqa: F403
+from mailman_web.settings.mailman import *  # noqa: F403
 
 
 #: Default list of admins who receive the emails from error logging.
@@ -14,7 +14,7 @@
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.postgresql_psycopg2',
-        'NAME': ‘mailmanweb’,
+        'NAME': 'mailmanweb',
         'USER': '<db_username>',
         'PASSWORD': '<password>',
         'HOST': 'localhost',
@@ -28,7 +28,7 @@
 
 
 # Make sure that this directory is created or Django will fail on start.
-LOGGING['handlers']['file']['filename'] = '/opt/mailman/web/logs/mailmanweb.log'
+LOGGING['handlers']['file']['filename'] = '/opt/mailman/web/logs/mailmanweb.log'  # noqa: F405,E501
 
 #: See https://docs.djangoproject.com/en/dev/ref/settings/#allowed-hosts
 ALLOWED_HOSTS = [
diff -ruN a/mailman_web/manage.py b/mailman_web/manage.py
--- a/mailman_web/manage.py	2021-02-03 06:09:38.000000000 +0100
+++ b/mailman_web/manage.py	2022-08-12 19:43:01.560426406 +0200
@@ -3,12 +3,16 @@
 import sys
 from pathlib import Path
 
+
 def setup():
     """Setup environment for Mailman web."""
     if os.getenv('DJANGO_SETTINGS_MODULE') is not None:
         return
 
-    MAILMAN_WEB_CONFIG = os.getenv('MAILMAN_WEB_CONFIG', '/etc/mailman3/settings.py')
+    MAILMAN_WEB_CONFIG = os.getenv(
+        'MAILMAN_WEB_CONFIG',
+        '/etc/mailman3/settings.py',
+    )
 
     if not os.path.exists(MAILMAN_WEB_CONFIG):
         print('Mailman web configuration file at {} does not exist'.format(
@@ -22,6 +26,7 @@
 
     os.environ['DJANGO_SETTINGS_MODULE'] = config_path.stem
 
+
 def main():
     setup()
 
diff -ruN a/mailman_web/settings/base.py b/mailman_web/settings/base.py
--- a/mailman_web/settings/base.py	2021-01-22 22:32:26.000000000 +0100
+++ b/mailman_web/settings/base.py	2022-08-12 19:43:01.560426406 +0200
@@ -1,5 +1,6 @@
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 import os
+from django.contrib.messages import constants as messages
 from pathlib import Path
 
 #: The base directory for logs and database.
@@ -130,8 +131,9 @@
 #:     'OPTIONS': {'charset': 'utf8mb4'}  # Enable utf8 4-byte encodings.
 #:
 #: Check out
-#: `Django documentation <https://docs.djangoproject.com/en/3.0/ref/settings/#databases>`_
-#: for more details.
+#: `Django documentation
+#: <https://docs.djangoproject.com/en/3.0/ref/settings/#databases>`_ for
+#: more details.
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.sqlite3',
@@ -146,7 +148,7 @@
 AUTH_PASSWORD_VALIDATORS = [
     {
         'NAME':
-        'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
+        'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',  # noqa: E501
     },
     {
         'NAME':
@@ -226,8 +228,6 @@
 #: https://docs.djangoproject.com/en/dev/topics/email/#email-backends
 EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
 
-# Compatibility with Bootstrap 3
-from django.contrib.messages import constants as messages  # flake8: noqa
 MESSAGE_TAGS = {
     messages.ERROR: 'danger'
 }
@@ -248,9 +248,8 @@
             'filters': ['require_debug_false'],
             'class': 'django.utils.log.AdminEmailHandler'
         },
-        'file':{
+        'file': {
             'level': 'INFO',
-            #'class': 'logging.handlers.RotatingFileHandler',
             'class': 'logging.handlers.WatchedFileHandler',
             'filename': os.path.join(BASE_DIR, 'logs', 'mailmanweb.log'),
             'formatter': 'verbose',
@@ -283,16 +282,12 @@
     },
     'formatters': {
         'verbose': {
-            'format': '%(levelname)s %(asctime)s %(process)d %(name)s %(message)s'
+            'format': '%(levelname)s %(asctime)s %(process)d %(name)s %(message)s'  # noqa: E501
         },
         'simple': {
             'format': '%(levelname)s %(message)s'
         },
     },
-    #'root': {
-    #    'handlers': ['file'],
-    #    'level': 'INFO',
-    #},
 }
 
 #: Current Django Site being served. This is used to customize the web host
diff -ruN a/mailman_web/settings/mailman.py b/mailman_web/settings/mailman.py
--- a/mailman_web/settings/mailman.py	2021-04-11 07:33:12.000000000 +0200
+++ b/mailman_web/settings/mailman.py	2022-08-12 19:43:01.560426406 +0200
@@ -1,5 +1,3 @@
-import os
-
 #: Mailman Core default API Path
 MAILMAN_REST_API_URL = 'http://localhost:8001'
 #: Mailman Core API user
@@ -33,7 +31,7 @@
 ACCOUNT_AUTHENTICATION_METHOD = "username_email"
 ACCOUNT_EMAIL_REQUIRED = True
 ACCOUNT_EMAIL_VERIFICATION = "mandatory"
-ACCOUNT_UNIQUE_EMAIL  = True
+ACCOUNT_UNIQUE_EMAIL = True
 
 #: Protocol for URLs generated for authentication, like email
 #: confirmation.
@@ -42,7 +40,8 @@
 
 #: Extra configuration for Social auth. For these configuration to be used.
 #: each of the social account providers must be first added in INSTALLED_APPS.
-#: See :py:data:`mailman_web.settings.base.INSTALLED_APPS` for more configuration.
+#: See :py:data:`mailman_web.settings.base.INSTALLED_APPS` for more
+#: configuration.
 SOCIALACCOUNT_PROVIDERS = {
     'openid': {
         'SERVERS': [
diff -ruN a/mailman_web/urls.py b/mailman_web/urls.py
--- a/mailman_web/urls.py	2020-05-26 03:34:31.000000000 +0200
+++ b/mailman_web/urls.py	2022-08-12 19:43:01.560426406 +0200
@@ -17,19 +17,19 @@
 # Postorius.  If not, see <http://www.gnu.org/licenses/>.
 
 
-from django.conf.urls import include, url
+from django.conf.urls import include
 from django.contrib import admin
-from django.urls import reverse_lazy
+from django.urls import path, reverse_lazy
 from django.views.generic import RedirectView
 
 urlpatterns = [
-    url(r'^$', RedirectView.as_view(
-        url=reverse_lazy('list_index'),
-        permanent=True)),
-    url(r'^mailman3/', include('postorius.urls')),
-    url(r'^archives/', include('hyperkitty.urls')),
-    url(r'', include('django_mailman3.urls')),
-    url(r'^accounts/', include('allauth.urls')),
-    # Django admin
-    url(r'^admin/', admin.site.urls),
+    path(
+        '',
+        RedirectView.as_view(url=reverse_lazy('list_index'), permanent=True),
+    ),
+    path('mailman3/', include('postorius.urls')),
+    path('archives/', include('hyperkitty.urls')),
+    path('', include('django_mailman3.urls')),
+    path('accounts/', include('allauth.urls')),
+    path('admin/', admin.site.urls),
 ]
diff -ruN a/mailman_web/wsgi.py b/mailman_web/wsgi.py
--- a/mailman_web/wsgi.py	2019-10-01 03:56:13.000000000 +0200
+++ b/mailman_web/wsgi.py	2022-08-12 19:43:01.560426406 +0200
@@ -7,8 +7,6 @@
 https://docs.djangoproject.com/en/2.2/howto/deployment/wsgi/
 """
 
-import os
-
 from django.core.wsgi import get_wsgi_application
 from mailman_web.manage import setup
 
diff -ruN a/setup.cfg b/setup.cfg
--- a/setup.cfg	2021-09-17 17:03:22.707365000 +0200
+++ b/setup.cfg	2022-08-12 19:43:40.623515642 +0200
@@ -22,7 +22,7 @@
 	setuptools
 	setuptools-scm
 install_requires = 
-	Django<3.1
+	django >=3.2, <4.1
 	postorius
 	hyperkitty
 	whoosh
diff -ruN a/tox.ini b/tox.ini
--- a/tox.ini	2021-01-20 01:10:36.000000000 +0100
+++ b/tox.ini	2022-08-12 19:43:01.560426406 +0200
@@ -1,5 +1,10 @@
 [tox]
-envlist = docs
+envlist = docs, lint
+
+[testenv:lint]
+deps = flake8
+basepython = python3
+commands = flake8 .
 
 [testenv:docs]
 deps = sphinx
