diff --git a/fileutils.gemspec b/fileutils.gemspec
index 76baea3..dd23935 100644
--- a/fileutils.gemspec
+++ b/fileutils.gemspec
@@ -1,18 +1,8 @@
 # frozen_string_literal: true
 
-source_version = ["", "lib/"].find do |dir|
-  begin
-    break File.open(File.join(__dir__, "#{dir}fileutils.rb")) {|f|
-      f.gets("\n  VERSION = ")
-      f.gets[/\s*"(.+)"/, 1]
-    }
-  rescue Errno::ENOENT
-  end
-end
-
 Gem::Specification.new do |s|
   s.name = "fileutils"
-  s.version = source_version
+  s.version = "1.6.0"
   s.summary = "Several file utility methods for copying, moving, removing, etc."
   s.description = "Several file utility methods for copying, moving, removing, etc."
 
