diff -wbBur virt-manager-0.8.5/src/virtManager/console.py virt-manager-0.8.5.my/src/virtManager/console.py
--- virt-manager-0.8.5/src/virtManager/console.py	2010-08-23 20:37:13.000000000 +0000
+++ virt-manager-0.8.5.my/src/virtManager/console.py	2010-11-17 17:09:56.000000000 +0000
@@ -550,11 +550,11 @@
         #
         nc_params = "%s %s" % (vncaddr, str(vncport))
         nc_cmd = (
-            """nc -q 2>&1 | grep -q "requires an argument";"""
+            """nc.openbsd -q 2>&1 | grep -q "requires an argument";"""
             """if [ $? -eq 0 ] ; then"""
-            """   CMD="nc -q 0 %(nc_params)s";"""
+            """   CMD="nc.openbsd -q 0 %(nc_params)s";"""
             """else"""
-            """   CMD="nc %(nc_params)s";"""
+            """   CMD="nc.openbsd %(nc_params)s";"""
             """fi;"""
             """eval "$CMD";""" %
             {'nc_params': nc_params})
