--- a/src/virtManager/manager.py	Fri Jan 21 11:30:38 2011 -0500
+++ b/src/virtManager/manager.py	Fri Jan 21 14:43:19 2011 -0500
@@ -721,6 +721,7 @@
         if (self.config.get_console_popup() != 2 or
             vm.is_management_domain()):
             self.emit("action-refresh-console", uri, vmuuid)
+            return
 
         # user has requested consoles on all vms
         gtype = vm.get_graphics_console()[0]

