--- a/build.sh	2022-10-18 00:54:24.319840199 +0800
+++ b/build.sh	2022-10-18 00:54:33.839454372 +0800
@@ -104,8 +104,8 @@
 function render {
   name="x$1"
   variant="$2"
-  size=$(echo "$SVG_DIM*$1" | bc)
-  dpi=$(echo "$SVG_DPI*$1" | bc)
+  size=$(echo "$SVG_DIM*$1/1" | bc)
+  dpi=$(echo "$SVG_DPI*$1/1" | bc)
 
   mkdir -p "$BUILD_DIR/$variant/$name"
   find "$SRC/svg/$variant" -name "*.svg" -type f \
