diff -Naur encfs-1.5.orig/encfs/Makefile.am encfs-1.5/encfs/Makefile.am
--- encfs-1.5.orig/encfs/Makefile.am	2009-03-16 08:11:56.330904107 +0100
+++ encfs-1.5/encfs/Makefile.am	2009-03-16 08:13:13.164266321 +0100
@@ -3,7 +3,6 @@
 
 ALL_INCLUDES = @RLOG_CFLAGS@ @OPENSSL_CFLAGS@ @BOOST_CPPFLAGS@
 ALL_LDFLAGS = @RLOG_LIBS@ @OPENSSL_LIBS@ @FUSE_LIBS@ 
-ALL_LDFLAGS += @BOOST_SERIALIZATION_LIB@ @BOOST_FILESYSTEM_LIB@
 
 INCLUDES = $(all_includes) -I../intl
 
@@ -43,7 +42,7 @@
 #       :  +1 :           => internal changes, nothing breaks
 #
 libencfs_la_LDFLAGS = -version-info 5:0:1
-libencfs_la_LIBADD = -lrlog
+libencfs_la_LIBADD = -lrlog @OPENSSL_LIBS@ @BOOST_SERIALIZATION_LIB@ @BOOST_FILESYSTEM_LIB@ @BOOST_SYSTEM_LIB@
 
 EXTRASRC = ../intl/autosprintf.cpp
 if BUILD_OPENSSL

