*** includes/sorting.php.old	2011-04-03 06:05:31.000000000 -0500
--- includes/sorting.php	2012-03-27 05:05:51.000000000 -0500
***************
*** 123,129 ****
      function by_user_choice(&$a, &$b) {
          foreach ($GLOBALS['user_sort_choice'] as $sort) {
              $function = 'by_'.$sort['field'];
!             $response = $function(&$a, &$b);
          // Identical response, go on to the next sort choice
              if (!$response)
                  continue;
--- 123,129 ----
      function by_user_choice(&$a, &$b) {
          foreach ($GLOBALS['user_sort_choice'] as $sort) {
              $function = 'by_'.$sort['field'];
!             $response = $function($a, $b);
          // Identical response, go on to the next sort choice
              if (!$response)
                  continue;
*** modules/tv/tmpl/default/schedules.php.old	2011-04-03 06:05:31.000000000 -0500
--- modules/tv/tmpl/default/schedules.php	2012-03-27 05:37:40.000000000 -0500
***************
*** 90,95 ****
--- 90,100 ----
  </tr>
  </thead>
  <?php
+         // Hack:  Disable warnings about "Creating default
+         // object from empty value" from below line
+         // "$schedule->channel->name = '[ '.t('Any').' ]';"
+         ini_set('error_reporting', E_ERROR);
+ 
          $prev_group = '';
          $cur_group  = '';
          foreach ($the_schedules as $schedule) {
