--- Makefile.orig	2009-12-23 12:37:52.328137461 +0200
+++ Makefile	2009-12-23 12:38:20.231372376 +0200
@@ -101,7 +101,7 @@
 	install -m644 README       $(INSTALLDIR)/share/uzbl/docs
 	sed -i 's#^set prefix.*=.*#set prefix     = $(RUN_PREFIX)#' $(INSTALLDIR)/share/uzbl/examples/config/uzbl/config
 
-install-uzbl-browser: install-uzbl-core
+install-uzbl-browser:
 	install -d $(INSTALLDIR)/bin
 	install -m755 uzbl-browser $(INSTALLDIR)/bin/uzbl-browser
 	install -m755 examples/data/uzbl/scripts/uzbl-cookie-daemon $(INSTALLDIR)/bin/uzbl-cookie-daemon
