diff -wbBur ies4linux-2.99.0.1/lib/functions.sh ies4linux-2.99.0.1.my/lib/functions.sh
--- ies4linux-2.99.0.1/lib/functions.sh	2007-11-26 06:28:16.000000000 +0000
+++ ies4linux-2.99.0.1.my/lib/functions.sh	2010-06-03 15:01:02.000000000 +0000
@@ -242,8 +242,6 @@
 function create_wine_prefix {
 	if which wineprefixcreate &> /dev/null; then
 		( wineprefixcreate 2>&1 ) | debugPipe
-	else
-		error $MSG_ERROR_NO_WINEPREFIXCREATE
 	fi
 }
 
diff -wbBur ies4linux-2.99.0.1/lib/install.sh ies4linux-2.99.0.1.my/lib/install.sh
--- ies4linux-2.99.0.1/lib/install.sh	2007-11-26 06:28:16.000000000 +0000
+++ ies4linux-2.99.0.1.my/lib/install.sh	2010-06-03 15:01:09.000000000 +0000
@@ -423,7 +423,6 @@
 
 	subsection $MSG_CREATING_PREFIX
 		set_wine_prefix "$BASEDIR/ie1/"
-		wineprefixcreate &> /dev/null
 		clean_tmp
 
 	subsection $MSG_EXTRACTING_CABS
