Index: freehdl-0.0.8/v2cc/gvhdl.in
===================================================================
--- freehdl-0.0.8.orig/v2cc/gvhdl.in	2011-03-12 14:16:47.000000000 +0100
+++ freehdl-0.0.8/v2cc/gvhdl.in	2014-08-25 13:00:00.086254980 +0200
@@ -5,7 +5,7 @@
 
 my $cc = "@CXX@";
 my $libtool = "@SYSTEM_LIBTOOL@";
-my $libtool_options = "--mode=link";
+my $libtool_options = "--mode=link --tag CXX";
 my $vhdl_source_name = "";
 my $source = "";
 my $includes = "";
