diff -ruN eric5-5.5.1_py2.orig/install.py eric5-5.5.1_py2/install.py
--- eric5-5.5.1_py2.orig/install.py	2014-12-01 03:56:19.000000000 +0300
+++ eric5-5.5.1_py2/install.py	2014-12-01 03:56:46.332570642 +0300
@@ -237,7 +237,7 @@
     text = text.replace("@MARKER@", marker)
     
     if sys.version_info[0] == 2:
-        f = codecs.open(src, "w", "utf-8")
+        f = codecs.open(dst, "w", "utf-8")
     else:
         f = open(dst, "w")
     f.write(text)
