Index: ati803/common/etc/ati/authatieventsd.sh
===================================================================
--- ati803/common/etc/ati/authatieventsd.sh
+++ ati803/common/etc/ati/authatieventsd.sh	2008-04-18 23:25:27.000000000 +0300
@@ -34,8 +34,9 @@
 
     # Check xdm/kdm
 
-    XDM_AUTH_MASK=/var/lib/xdm/authdir/authfiles/A$1*
-    XDM_AUTH_FILE=`ls -t $XDM_AUTH_MASK 2>/dev/null | head -n 1`   # Choose the newest file
+    XDM_AUTH_MASK=/var/lib/xdm/authfiles/A$1*
+    KDM_AUTH_MASK=/var/run/xauth/A$1*
+    XDM_AUTH_FILE=`ls -t $XDM_AUTH_MASK $KDM_AUTH_MASK 2>/dev/null | head -n 1`   # Choose the newest file
     if [ -n "$XDM_AUTH_FILE" ]; then
         SERVER_AUTH_FILE=$XDM_AUTH_FILE
         DISP_SEARCH_STRING="#ffff#"
