Seulement dans shotwell-0.11.5-fixed/misc: gschemas.compiled
diff -ru shotwell-0.11.5/src/photos/GRaw.vala shotwell-0.11.5-fixed/src/photos/GRaw.vala
--- shotwell-0.11.5/src/photos/GRaw.vala	2011-09-24 00:52:27.000000000 +0200
+++ shotwell-0.11.5-fixed/src/photos/GRaw.vala	2011-11-01 00:25:05.976289495 +0100
@@ -163,9 +163,9 @@
         proc = new LibRaw.Processor(options);
     }
     
-    public void add_masked_borders_to_bitmap() {
-        proc.add_masked_borders_to_bitmap();
-    }
+//    public void add_masked_borders_to_bitmap() {
+//        proc.add_masked_borders_to_bitmap();
+//    }
     
     public void adjust_sizes_info_only() throws Exception {
         throw_exception("adjust_sizes_info_only", proc.adjust_sizes_info_only());
@@ -215,9 +215,9 @@
         throw_exception("ppm_tiff_writer", proc.ppm_tiff_writer(filename));
     }
     
-    public void rotate_fuji_raw() throws Exception {
-        throw_exception("rotate_fuji_raw", proc.rotate_fuji_raw());
-    }
+//    public void rotate_fuji_raw() throws Exception {
+//        throw_exception("rotate_fuji_raw", proc.rotate_fuji_raw());
+//    }
     
     public void thumb_writer(string filename) throws Exception {
         throw_exception("thumb_writer", proc.thumb_writer(filename));
@@ -262,7 +262,7 @@
         // camera_profile
         // bad_pixels
         // dark_frame
-        output_params->filtering_mode = LibRaw.Filtering.AUTOMATIC;
+        // output_params->filtering_mode = LibRaw.Filtering.AUTOMATIC;
         output_params->output_bps = 8;
         // output_tiff
         output_params->user_flip = GRaw.Flip.FROM_SOURCE;
@@ -303,8 +303,8 @@
         case LibRaw.Result.UNSUPPORTED_THUMBNAIL:
             throw new Exception.UNSUPPORTED_THUMBNAIL(msg);
         
-        case LibRaw.Result.CANNOT_ADDMASK:
-            throw new Exception.CANNOT_ADDMASK(msg);
+//        case LibRaw.Result.CANNOT_ADDMASK:
+//            throw new Exception.CANNOT_ADDMASK(msg);
         
         case LibRaw.Result.UNSUFFICIENT_MEMORY:
             throw new Exception.OUT_OF_MEMORY(msg);
