diff --git a/dev-requirements.txt b/dev-requirements.txt
index b2f5288..22fc710 100644
--- a/dev-requirements.txt
+++ b/dev-requirements.txt
@@ -1,5 +1,4 @@
 coverage
 codecov
-mock
 nose
 setuptools-scm
diff --git a/test/test_unix_connect.py b/test/test_unix_connect.py
index 7680ba5..82e9f01 100644
--- a/test/test_unix_connect.py
+++ b/test/test_unix_connect.py
@@ -6,7 +6,7 @@ import socket
 import sys
 import unittest
 
-from mock import patch
+from unittest.mock import patch
 
 from Xlib.support import unix_connect
 from Xlib.error import DisplayConnectionError, DisplayNameError
