# Copyright: Sébastien Luttringer <seblu@seblu.net>
--- a/dkms	2015-10-23 13:57:36.624685834 +0200
+++ b/dkms	2015-10-23 17:54:39.495367417 +0200
@@ -3465,6 +3465,12 @@
 
 # Sanity checking
 
+# No action, we exit
+if [[ -z $action ]]; then
+    show_usage
+    die 4 $"No action was specified."
+fi
+
 # Error out if binaries-only is set and source-only is set
 if [[ $binaries_only && $source_only ]]; then
     die 8 $" You have specified both --binaries-only and --source-only." \
