 Makefile.am  | 6 ++----
 configure.ac | 4 +---
 2 files changed, 3 insertions(+), 7 deletions(-)

diff --git c/Makefile.am i/Makefile.am
index a5a16220..10c4bb57 100644
--- c/Makefile.am
+++ i/Makefile.am
@@ -114,21 +114,19 @@ else
 CARGO_RELEASE_ARGS=--release
 endif
 
-if CROSS_COMPILING
-CARGO_TARGET_ARGS=--target=$(host)
-endif
+CARGO_TARGET_ARGS=--target=i686-unknown-linux-gnu
 
 CARGO_VERBOSE = $(cargo_verbose_$(V))
 cargo_verbose_ = $(cargo_verbose_$(AM_DEFAULT_VERBOSITY))
 cargo_verbose_0 =
 cargo_verbose_1 = --verbose
 
 RUST_LIB=@abs_top_builddir@/target/@RUST_TARGET_SUBDIR@/librsvg_internals.a
 CARGO_TARGET_DIR=@abs_top_builddir@/target
 
 check-local:
 	cd $(srcdir) && \
-	CARGO_TARGET_DIR=$(CARGO_TARGET_DIR) cargo test
+	CARGO_TARGET_DIR=$(CARGO_TARGET_DIR) cargo test $(CARGO_VERBOSE) $(CARGO_TARGET_ARGS) $(CARGO_RELEASE_ARGS)
 
 clean-local:
 	cd $(top_srcdir) && \
diff --git c/configure.ac i/configure.ac
index 2fae2362..9d4004f8 100644
--- c/configure.ac
+++ i/configure.ac
@@ -289,9 +289,7 @@ fi
 AM_CONDITIONAL([DEBUG_RELEASE], [test "x$debug_release" = "xyes"])
 
 AM_CONDITIONAL(CROSS_COMPILING, test $cross_compiling = yes)
-if test "x$cross_compiling" = "xyes" ; then
-	RUST_TARGET_SUBDIR="$host/$RUST_TARGET_SUBDIR"
-fi
+RUST_TARGET_SUBDIR="i686-unknown-linux-gnu/$RUST_TARGET_SUBDIR"
 AC_SUBST([RUST_TARGET_SUBDIR])
 
 dnl ===========================================================================
