diff -wbBur freewrl-1.22.9.org/src/lib/non_web3d_formats/ColladaParser.c freewrl-1.22.9/src/lib/non_web3d_formats/ColladaParser.c
--- freewrl-1.22.9.org/src/lib/non_web3d_formats/ColladaParser.c	2010-08-19 02:20:36.000000000 +0000
+++ freewrl-1.22.9/src/lib/non_web3d_formats/ColladaParser.c	2010-10-18 13:31:25.000000000 +0000
@@ -54,7 +54,7 @@
 #include "ColladaParser.h"
 
 #if HAVE_EXPAT_H
-# include <expat.h>
+# include </usr/include/expat.h>
 #endif
 
 #define PROTOINSTANCE_MAX_LEVELS 10
diff -wbBur freewrl-1.22.9.org/src/lib/x3d_parser/X3DParser.c freewrl-1.22.9/src/lib/x3d_parser/X3DParser.c
--- freewrl-1.22.9.org/src/lib/x3d_parser/X3DParser.c	2010-08-19 02:20:36.000000000 +0000
+++ freewrl-1.22.9/src/lib/x3d_parser/X3DParser.c	2010-10-18 13:30:01.000000000 +0000
@@ -54,7 +54,7 @@
 #include "X3DProtoScript.h"
 
 #if HAVE_EXPAT_H
-# include <expat.h>
+# include </usr/include/expat.h>
 #endif
 //#define X3DPARSERVERBOSE 1
 
