--- ./src/lib/main/ProdCon.c.orig	2009-10-01 21:35:36.000000000 +0200
+++ ./src/lib/main/ProdCon.c	2009-11-24 22:08:12.986224700 +0100
@@ -526,7 +526,8 @@
 
 	if (checkNetworkFile(fname)) {
 		/* if we are running as a plugin, ask the HTML browser for ALL files, as it'll know proxies, etc. */
-		if (RUNNINGASPLUGIN) {
+/*		if (RUNNINGASPLUGIN) {*/
+		if ( FALSE ) {
 			/* printf ("fileExists, runningasplugin\n"); */
 
 			/* are we running as a plugin? If so, ask the HTML browser to get the file, and place
