diff -aur multiget.orig/Makefile.am multiget.new/Makefile.am
--- multiget.orig/Makefile.am	2009-11-24 14:23:26.000000000 +0100
+++ multiget.new/Makefile.am	2009-11-24 15:29:44.000000000 +0100
@@ -3,7 +3,7 @@
 
 SUBDIRS = src po
 
-multigetdocdir = ${prefix}/doc/multiget
+multigetdocdir = ${prefix}/share/doc/multiget
 multigetdoc_DATA = \
 	README\
 	COPYING\
diff -aur multiget.orig/src/batch.cpp multiget.new/src/batch.cpp
--- multiget.orig/src/batch.cpp	2009-11-24 14:23:26.000000000 +0100
+++ multiget.new/src/batch.cpp	2009-11-24 14:31:58.000000000 +0100
@@ -25,6 +25,8 @@
 #include "batch.h"
 #include "common.h"
 #include <iostream>
+#include <cstdio>
+#include <cstdlib>
 
 CBatch::CBatch( std::string str )
 {
diff -aur multiget.orig/src/common.cpp multiget.new/src/common.cpp
--- multiget.orig/src/common.cpp	2009-11-24 14:23:26.000000000 +0100
+++ multiget.new/src/common.cpp	2009-11-24 14:33:22.000000000 +0100
@@ -26,7 +26,7 @@
 #include "speedctrl.h"
 #include "mirroradmin.h"
 
-
+#include <cstdlib>
 #include <pthread.h>
 
 #ifdef WIN32
diff -aur multiget.orig/src/mgftpbase.cpp multiget.new/src/mgftpbase.cpp
--- multiget.orig/src/mgftpbase.cpp	2009-11-24 14:23:26.000000000 +0100
+++ multiget.new/src/mgftpbase.cpp	2009-11-24 14:52:04.000000000 +0100
@@ -22,6 +22,9 @@
  *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
+#include <cstdlib>
+#include <cstdio>
+
 #ifdef WIN32
 #include <winsock2.h>
 #endif
diff -aur multiget.orig/src/mghttpbase.cpp multiget.new/src/mghttpbase.cpp
--- multiget.orig/src/mghttpbase.cpp	2009-11-24 14:23:26.000000000 +0100
+++ multiget.new/src/mghttpbase.cpp	2009-11-24 14:57:55.000000000 +0100
@@ -42,6 +42,9 @@
 -100 : app connection not established
  
 */
+
+#include <cstdio>
+
 #ifdef WIN32
 #include <winsock2.h>
 #endif
diff -aur multiget.orig/src/mgurlparser.cpp multiget.new/src/mgurlparser.cpp
--- multiget.orig/src/mgurlparser.cpp	2009-11-24 14:23:26.000000000 +0100
+++ multiget.new/src/mgurlparser.cpp	2009-11-24 15:05:24.000000000 +0100
@@ -26,6 +26,9 @@
 #include "mgurlparser.h"
 #include "common.h"
 #include <iostream>
+#include <cstring>
+#include <cstdlib>
+
 extern std::string gDefFtpPass;
 
 using namespace std;
