diff -wbBur sK1-0.9.1pre/setup.py sK1-0.9.1pre.my/setup.py
--- sK1-0.9.1pre/setup.py	2010-01-10 06:24:54.000000000 +0000
+++ sK1-0.9.1pre.my/setup.py	2010-02-24 21:19:02.000000000 +0000
@@ -149,8 +149,8 @@
 					filter_src+'stringfilter.c', filter_src+'binfile.c', filter_src+'hexfilter.c'])
 	
 	#Fix for Debian based distros
-	tcl_include_dirs = []
-	tcl_ver=''
+	tcl_include_dirs = ['/usr/include']
+	tcl_ver='8.5'
 	if os.path.isdir('/usr/include/tcl8.5'):
 		tcl_include_dirs = ['/usr/include/tcl8.5']
 		tcl_ver ='8.5'
