diff -aur sK1-0.9.1pre/src/app/UI/pluginpanels/properties/gradient_fill.py sK1-0.9.1pre.new/src/app/UI/pluginpanels/properties/gradient_fill.py
--- sK1-0.9.1pre/src/app/UI/pluginpanels/properties/gradient_fill.py	2009-07-16 23:26:15.000000000 +0000
+++ sK1-0.9.1pre.new/src/app/UI/pluginpanels/properties/gradient_fill.py	2013-09-22 09:52:26.543457833 +0000
@@ -24,7 +24,7 @@
 
 import tooltips
 
-import Image, ImageTk, ImageDraw
+from PIL import Image, ImageTk, ImageDraw
 
 DEFAULT_WIDTH=0.283286
 DEFAULT_CORNER=const.JoinMiter
diff -aur sK1-0.9.1pre/src/app/UI/pluginpanels/properties/outlineprop.py sK1-0.9.1pre.new/src/app/UI/pluginpanels/properties/outlineprop.py
--- sK1-0.9.1pre/src/app/UI/pluginpanels/properties/outlineprop.py	2009-05-15 22:11:39.000000000 +0000
+++ sK1-0.9.1pre.new/src/app/UI/pluginpanels/properties/outlineprop.py	2013-09-22 09:52:13.216885815 +0000
@@ -24,7 +24,7 @@
 
 import tooltips
 
-import Image, ImageTk, ImageDraw
+from PIL import Image, ImageTk, ImageDraw
 
 DEFAULT_WIDTH=0.283286
 DEFAULT_CORNER=const.JoinMiter
diff -aur sK1-0.9.1pre/src/app/UI/widgets/colorbutton.py sK1-0.9.1pre.new/src/app/UI/widgets/colorbutton.py
--- sK1-0.9.1pre/src/app/UI/widgets/colorbutton.py	2009-04-30 02:58:41.000000000 +0000
+++ sK1-0.9.1pre.new/src/app/UI/widgets/colorbutton.py	2013-09-22 09:52:38.150044124 +0000
@@ -8,7 +8,7 @@
 # The color sample size is 31x20 px
 
 from Ttk import TButton
-import Image, ImageTk
+from PIL import Image, ImageTk
 
 class TColorButton(TButton):
