--- src/params.h	2014-12-18 15:14:33.000000000 +0000
+++ src/params.h	2015-01-03 13:49:57.994775328 +0000
@@ -135,8 +135,12 @@
 /*
  * File names.
  */
-#define CFG_FULL_NAME			"$JOHN/john.conf"
-#define CFG_ALT_NAME			"$JOHN/john.ini"
+#ifndef CFG_FULL_NAME
+#define CFG_FULL_NAME			"$JOHN/john.conf"
+#endif
+#ifndef CFG_ALT_NAME
+#define CFG_ALT_NAME			"$JOHN/john.ini"
+#endif
 #if JOHN_SYSTEMWIDE
 #define CFG_PRIVATE_FULL_NAME		JOHN_PRIVATE_HOME "/john.conf"
 #define CFG_PRIVATE_ALT_NAME		JOHN_PRIVATE_HOME "/john.ini"
