Index: src/plugins/Input/ffmpeg/decoder_ffmpeg.cpp
===================================================================
--- src/plugins/Input/ffmpeg/decoder_ffmpeg.cpp	(revision 2728)
+++ src/plugins/Input/ffmpeg/decoder_ffmpeg.cpp	(revision 2729)
@@ -133,9 +133,6 @@
     m_stream->seekable = !input()->isSequential();
     m_stream->max_packet_size = INPUT_BUFFER_SIZE;
 
-    AVFormatParameters ap;
-    memset(&ap, 0, sizeof(ap));
-
     if(avformat_open_input(&ic, m_path.toLocal8Bit().constData(), fmt, 0) != 0)
     {
         qDebug("DecoderFFmpeg: av_open_input_stream() failed");
