diff --git a/CMakeLists.txt b/CMakeLists.txt
index 76a886e..773d804 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,8 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/build/cmake/Modu
 
 find_package(PkgConfig REQUIRED)
 
+find_package(X11)
+
 find_package(GTK2 2.20 REQUIRED gtk gtkmm)
 pkg_check_modules(GTHREAD REQUIRED gthread-2.0)
 pkg_check_modules(GMODULE REQUIRED gmodule-2.0)
diff --git a/frontend/linux/workbench/CMakeLists.txt b/frontend/linux/workbench/CMakeLists.txt
index b9c16c8..d52c34a 100644
--- a/frontend/linux/workbench/CMakeLists.txt
+++ b/frontend/linux/workbench/CMakeLists.txt
@@ -92,6 +92,7 @@ target_link_libraries(mysql-workbench-bin
     mysqlparser
     ${MYSQLCPPCONN_LIBRARIES}
     ${CTEMPLATE_LIBRARIES}
+    ${X11_LIBRARIES}
     ${GTK2_LIBRARIES}
     ${SIGC++_LIBRARIES}
     ${GRT_LIBRARIES}
