--- configure.in.orig	2010-10-06 16:50:47.356685634 -0700
+++ configure.in	2010-10-06 16:51:14.990017077 -0700
@@ -303,13 +303,13 @@
 dnl 	enable_python=$enableval, enable_python=no)
 
 AC_MSG_CHECKING(for Python)
-PYTHON_LIBS=[$(python -c "from distutils import sysconfig
+PYTHON_LIBS=[$(python2 -c "from distutils import sysconfig
 import os 
 cfg=sysconfig.get_config_vars()
 lib = os.path.join(cfg['LIBDIR'], cfg['LDLIBRARY'])
 if os.path.exists(lib):
     print lib")]
-PYTHON_CFLAGS=[$(python -c "from distutils import sysconfig
+PYTHON_CFLAGS=[$(python2 -c "from distutils import sysconfig
 print sysconfig.get_python_inc()")]
 PYTHON_CFLAGS="-I$PYTHON_CFLAGS"
 if test "$PYTHON_LIBS" = ""; then
