--- python2/httplib2/certs.py.orig	2018-11-15 20:09:04.605507207 +0800
+++ python2/httplib2/certs.py	2018-11-15 20:10:35.122107327 +0800
@@ -19,9 +19,7 @@
     pass
 
 
-BUILTIN_CA_CERTS = os.path.join(
-    os.path.dirname(os.path.abspath(__file__)), "cacerts.txt"
-)
+BUILTIN_CA_CERTS = "/etc/ssl/certs/ca-certificates.crt"
 
 
 def where():
--- python3/httplib2/certs.py.orig	2018-11-15 20:09:15.708832692 +0800
+++ python3/httplib2/certs.py	2018-11-15 20:11:16.888741464 +0800
@@ -19,9 +19,7 @@
     pass
 
 
-BUILTIN_CA_CERTS = os.path.join(
-    os.path.dirname(os.path.abspath(__file__)), "cacerts.txt"
-)
+BUILTIN_CA_CERTS = "/etc/ssl/certs/ca-certificates.crt"
 
 
 def where():
--- tests/test_cacerts_from_env.py.orig	2018-11-15 20:21:46.148162984 +0800
+++ tests/test_cacerts_from_env.py	2018-11-15 20:21:58.051484347 +0800
@@ -6,7 +6,7 @@
 import httplib2
 
 
-CA_CERTS_BUILTIN = os.path.join(os.path.dirname(httplib2.__file__), "cacerts.txt")
+CA_CERTS_BUILTIN = "/etc/ssl/certs/ca-certificates.crt"
 CERTIFI_CERTS_FILE = "unittest_certifi_file"
 CUSTOM_CA_CERTS = "unittest_custom_ca_certs"
 
