diff -Nur nspluginwrapper-1.3.0.orig//src/npw-viewer.c nspluginwrapper-1.3.0//src/npw-viewer.c
--- nspluginwrapper-1.3.0.orig//src/npw-viewer.c	2009-01-02 15:22:29.000000000 +0100
+++ nspluginwrapper-1.3.0//src/npw-viewer.c	2010-08-26 18:46:50.574976221 +0200
@@ -4187,7 +4187,8 @@
 #define high_offset		offsetof(struct _XtAppStruct, __maybe__input_max)
 #define n_offsets_max	(high_offset - low_offset)/2
   int i, ofs, n_offsets = 0;
-  int offsets[n_offsets_max] = { 0, };
+  int offsets[n_offsets_max];
+  offsets[0] = 0;
 
 #define n_inputs_max	4 /* number of refinements/input sources */
   int fd, id, n_inputs = 0;
