diff -Naur ubuntuone-client-4.0.0.orig/ubuntuone/platform/notification/linux.py ubuntuone-client-4.0.0/ubuntuone/platform/notification/linux.py
--- ubuntuone-client-4.0.0.orig/ubuntuone/platform/notification/linux.py	2012-06-19 15:51:01.000000000 +0200
+++ ubuntuone-client-4.0.0/ubuntuone/platform/notification/linux.py	2012-10-15 01:14:01.019476818 +0200
@@ -40,7 +40,7 @@
 
 if 'gi' in sys.modules and sys.modules['gi'] is not None:
     try:
-        from gi.repository import Notify
+        from gi.repository import GLib, Notify
         Notify  # pyflakes
         NOTIFY_MODULE = 'gi'
     except ImportError:
@@ -85,5 +85,6 @@
         if append:
             self.notification.set_hint_string('x-canonical-append', '')
 
-        self.notification.set_hint_int32('transient', int(True))
+        if NOTIFY_MODULE == 'gi':
+            self.notification.set_hint('transient', GLib.Variant.new_boolean(1))
         self.notification.show()
