From e532fc97da64a646bb706a9fa812a047cc557987 Mon Sep 17 00:00:00 2001
From: pad <pad@fb.com>
Date: Mon, 8 Sep 2014 10:03:20 -0700
Subject: [PATCH] * h_program-lang/database_code.ml: another issue #94 fix

---
 h_program-lang/database_code.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/h_program-lang/database_code.ml b/h_program-lang/database_code.ml
index 850e699..0745f58 100644
--- a/h_program-lang/database_code.ml
+++ b/h_program-lang/database_code.ml
@@ -360,7 +360,7 @@ let save_database database file =
   if File_type.is_json_filename file
   then
     database +> json_of_database 
-    +> Json_io.string_of_json ~compact:false ~recursive:false
+    +> Json_io.string_of_json ~compact:false ~recursive:false ~allow_nan:true
     +> Common.write_file ~file
   else Common2.write_value database file
 
