diff --no-dereference --ignore-all-space --unified --recursive --text glulxe-glulxe-0.5.4-old/Makefile glulxe-glulxe-0.5.4-new/Makefile
--- glulxe-glulxe-0.5.4-old/Makefile	2017-01-23 22:14:37.000000000 +0300
+++ glulxe-glulxe-0.5.4-new/Makefile	2021-09-04 15:54:28.668302548 +0300
@@ -40,7 +40,7 @@
 
 include $(GLKINCLUDEDIR)/$(GLKMAKEFILE)
 
-CFLAGS = $(OPTIONS) -I$(GLKINCLUDEDIR) $(XMLLIBINCLUDEDIR)
+CFLAGS += $(OPTIONS) -I$(GLKINCLUDEDIR) $(XMLLIBINCLUDEDIR)
 LIBS = -L$(GLKLIBDIR) $(GLKLIB) $(LINKLIBS) -lm $(XMLLIB)
 
 OBJS = main.o files.o vm.o exec.o funcs.o operand.o string.o glkop.o \
@@ -50,7 +50,7 @@
 all: glulxe
 
 glulxe: $(OBJS) unixstrt.o
-	$(CC) $(OPTIONS) -o glulxe $(OBJS) unixstrt.o $(LIBS)
+	$(CC) ${CFLAGS} ${LDFLAGS} -o glulxe $(OBJS) unixstrt.o $(LIBS)
 
 glulxdump: glulxdump.o
 	$(CC) -o glulxdump glulxdump.o
@@ -62,4 +62,3 @@
 
 clean:
 	rm -f *~ *.o glulxe glulxdump profile-raw
-
