From 2c15f97b0e682f8b7616f14ea3f3caab653a78f7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Johannes=20L=C3=B6thberg?= <johannes@kyriasis.com>
Date: Mon, 4 May 2020 17:57:29 +0200
Subject: [PATCH 4/4] Disable auto updates
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Johannes Löthberg <johannes@kyriasis.com>
---
 qt/aqt/update.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/qt/aqt/update.py b/qt/aqt/update.py
index 2513d84c..37656733 100644
--- a/qt/aqt/update.py
+++ b/qt/aqt/update.py
@@ -34,6 +34,8 @@ class LatestVersionFinder(QThread):
         return d
 
     def run(self):
+        return
+
         if not self.config["updates"]:
             return
         d = self._data()
-- 
2.26.2

