From bd0d5b6fde2ff0d3ca2fe8a6bdf59ce6e0fa1058 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Johannes=20L=C3=B6thberg?= <johannes@kyriasis.com>
Date: Mon, 4 May 2020 17:57:29 +0200
Subject: [PATCH 4/5] Disable auto updates
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Johannes Löthberg <johannes@kyriasis.com>
---
 qt/aqt/update.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/qt/aqt/update.py b/qt/aqt/update.py
index 8b179f65..a46618b8 100644
--- a/qt/aqt/update.py
+++ b/qt/aqt/update.py
@@ -33,6 +33,8 @@ class LatestVersionFinder(QThread):
         }
 
     def run(self):
+        return
+
         if not self.config["updates"]:
             return
         d = self._data()
-- 
2.29.2

