--- setup/build_environment.py.old	2014-08-02 17:49:06.363876595 +0200
+++ setup/build_environment.py	2014-08-02 17:49:34.427191537 +0200
@@ -103,7 +103,7 @@
         if os.path.exists(os.path.join(base, 'QtWidgets')):
             return base
     return q
-pyqt['pyqt_sip_dir'] = get_sip_dir(c.default_sip_dir)
+pyqt['pyqt_sip_dir'] = '/usr/share/sip/Py2-PyQt5/'
 pyqt['sip_inc_dir'] = c.sip_inc_dir
 
 glib_flags = subprocess.check_output([PKGCONFIG, '--libs', 'glib-2.0']).strip() if islinux else ''
