diff --no-dereference --ignore-all-space --unified --recursive --text glkterm-glkterm-1.0.4-widechar/gtncursesw.c glkterm-glkterm-1.0.4-widechar-new/gtncursesw.c
--- glkterm-glkterm-1.0.4-widechar/gtncursesw.c	2012-10-14 21:21:09.000000000 +0400
+++ glkterm-glkterm-1.0.4-widechar-new/gtncursesw.c	2021-09-04 13:48:43.952957333 +0300
@@ -92,7 +92,7 @@
 #else /* LOCAL_NCURSESW */
 
 #define _XOPEN_SOURCE_EXTENDED /* ncursesw *wch* and *wstr* functions */
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 
 int local_get_wch (wint_t *ch)
 {
diff --no-dereference --ignore-all-space --unified --recursive --text glkterm-glkterm-1.0.4-widechar/Makefile glkterm-glkterm-1.0.4-widechar-new/Makefile
--- glkterm-glkterm-1.0.4-widechar/Makefile	2012-10-14 21:21:09.000000000 +0400
+++ glkterm-glkterm-1.0.4-widechar-new/Makefile	2021-09-04 14:24:41.171037436 +0300
@@ -13,8 +13,8 @@
 CC = gcc -ansi
 
 # You may need to set directories to pick up the ncursesw library.
-#INCLUDEDIRS = -I/usr/5include
-#LIBDIRS = -L/usr/5lib
+INCLUDEDIRS = -I/usr/include
+LIBDIRS = -L/usr/lib
 LIBS = -lncursesw
 
 # Use this LIBS line instead of the above if you do not have ncursesw
@@ -22,11 +22,10 @@
 # probably also have to set LOCAL_NCURSESW in CFLAGS or in gtoption.h.)
 #LIBS = -lncurses
 
-OPTIONS = -g -Wall
+OPTIONS = -g -Wall -fPIC
 
-CFLAGS = $(OPTIONS) $(INCLUDEDIRS)
-
-GLKLIB = libglktermw.a
+CFLAGS += $(OPTIONS) $(INCLUDEDIRS)
+GLKLIB = libglkterm.so
 
 GLKTERM_OBJS = \
   main.o gtevent.o gtfref.o gtgestal.o gtinput.o \
@@ -39,17 +38,16 @@
   glkterm.h gtoption.h gtw_blnk.h gtw_buf.h \
   gtw_grid.h gtw_pair.h gi_dispa.h
 
-all: $(GLKLIB) Make.glktermw
+all: $(GLKLIB) Make.glkterm
 
 cgunicod.o: cgunigen.c
 
 $(GLKLIB): $(GLKTERM_OBJS)
-	ar r $(GLKLIB) $(GLKTERM_OBJS)
-	ranlib $(GLKLIB)
+	${CC} -shared ${LDFLAGS},-soname,${GLKLIB}.1.0.4 -o ${GLKLIB}.1.0.4 $(GLKTERM_OBJS)
 
-Make.glktermw:
-	echo LINKLIBS = $(LIBDIRS) $(LIBS) > Make.glktermw
-	echo GLKLIB = -lglktermw >> Make.glktermw
+Make.glkterm:
+	echo LINKLIBS = $(LIBDIRS) $(LIBS) > Make.glkterm
+	echo GLKLIB = -lglkterm >> Make.glkterm
 
 $(GLKTERM_OBJS): glk.h $(GLKTERM_HEADERS)
 
