diff --git a/buildSrc/build.gradle b/buildSrc/build.gradle
--- a/buildSrc/build.gradle	(revision a4eb7aa123dc53f8dac74d80b66a490f2d6b4a26)
+++ b/buildSrc/build.gradle	(date 1668163583892)
@@ -25,7 +25,6 @@
 }
 
 // Make sure the build environment is consistent.
-apply from: file('../gradle/validation/check-environment.gradle')
 
 // Load common buildSrc and script deps.
 apply from: file("scriptDepVersions.gradle")
diff --git a/solr/packaging/build.gradle b/solr/packaging/build.gradle
--- a/solr/packaging/build.gradle	(revision a4eb7aa123dc53f8dac74d80b66a490f2d6b4a26)
+++ b/solr/packaging/build.gradle	(date 1668163013921)
@@ -120,7 +120,6 @@
           "**/bin/post",
           "**/bin/postlogs",
           "**/bin/solr",
-          "**/bin/init.d/solr",
           "**/bin/solr-exporter",
       ]) { copy ->
         copy.setMode(0755)
