diff -Nur gnunet-gtk-0.9.0pre3.orig/configure.ac gnunet-gtk-0.9.0pre3/configure.ac
--- gnunet-gtk-0.9.0pre3.orig/configure.ac	2011-10-12 06:30:21.001993062 +0000
+++ gnunet-gtk-0.9.0pre3/configure.ac	2011-10-12 06:30:50.695578118 +0000
@@ -169,7 +169,6 @@
 AC_MSG_CHECKING(for gtk)
 without_gtk=true
 AM_PATH_GTK_3_0(3.0.0,without_gtk=false,)
-AM_PATH_GTK_2_0(2.6.0,without_gtk=false,)
 AM_CONDITIONAL(HAVE_GTK, test x$without_gtk != xtrue)
 if test $without_gtk != true
 then
