diff -Naur guake-0.5.0.orig/src/guake guake-0.5.0/src/guake
--- guake-0.5.0.orig/src/guake	2014-10-05 18:48:34.680242628 +0000
+++ guake-0.5.0/src/guake	2014-10-05 19:06:14.629370046 +0000
@@ -1039,10 +1039,6 @@
         if not self.window.get_property('visible'):
             self.show()
             self.set_terminal_focus()
-        elif (self.client.get_bool(KEY('/general/focus_if_open')) and
-                self.window.window and
-                not self.window.window.get_state()):
-            self.window.window.focus()
         else:
             self.hide()
 
