diff -ruN a/AbletonLinkConfig.cmake b/AbletonLinkConfig.cmake
--- a/AbletonLinkConfig.cmake	2018-05-29 12:37:09.000000000 +0200
+++ b/AbletonLinkConfig.cmake	2019-03-14 16:26:10.468519266 +0100
@@ -42,12 +42,6 @@
   )
 endif()
 
-include(${CMAKE_CURRENT_LIST_DIR}/cmake_include/AsioStandaloneConfig.cmake)
-set_property(TARGET Ableton::Link APPEND PROPERTY
-  INTERFACE_LINK_LIBRARIES
-  AsioStandalone::AsioStandalone
-)
-
 set_property(TARGET Ableton::Link APPEND PROPERTY
   INTERFACE_SOURCES
   ${CMAKE_CURRENT_LIST_DIR}/include/ableton/Link.hpp
