--- clipgrab-3.9.0/youtube_dl.cpp.orig	2020-11-14 11:28:07.576057009 +0100
+++ clipgrab-3.9.0/youtube_dl.cpp	2020-11-14 12:04:53.362705720 +0100
@@ -24,7 +24,7 @@
 
     // Try system-wide youtube-dl installation
     QString globalPath = QStandardPaths::findExecutable("youtube-dl");
-    process = instance(localPath, QStringList() << "--version");
+    process = instance(globalPath, QStringList() << "--version");
     process->start();
     process->waitForFinished();
     process->deleteLater();
