Index: src/ogmrip-main.c
===================================================================
--- src/ogmrip-main.c	(révision 1960)
+++ src/ogmrip-main.c	(copie de travail)
@@ -1452,7 +1452,7 @@
 
   ogmrip_encoding_set_relative (encoding,
 #if GTK_CHECK_VERSION(2,18,0)
-      gtk_widget_get_sensitive (data->relative_check) &
+      gtk_widget_is_sensitive (data->relative_check) &
 #else
       GTK_WIDGET_IS_SENSITIVE (data->relative_check) &
 #endif
Index: src/ogmrip-options-dialog.c
===================================================================
--- src/ogmrip-options-dialog.c	(révision 1960)
+++ src/ogmrip-options-dialog.c	(copie de travail)
@@ -971,13 +971,21 @@
   gtk_dialog_add_action_widget (GTK_DIALOG (dialog), widget, OGMRIP_RESPONSE_ENQUEUE);
   gtk_widget_show (widget);
 
+#if GTK_CHECK_VERSION(2,18,0)
+  gtk_widget_set_can_focus (widget, TRUE);
+#else
   GTK_WIDGET_SET_FLAGS (widget, GTK_CAN_DEFAULT);
+#endif
 
   widget = gtk_button_new_with_image (_("E_xtract"), GTK_STOCK_CONVERT);
   gtk_dialog_add_action_widget (GTK_DIALOG (dialog), widget, OGMRIP_RESPONSE_EXTRACT);
   gtk_widget_show (widget);
 
+#if GTK_CHECK_VERSION(2,18,0)
+  gtk_widget_set_can_focus (widget, TRUE);
+#else
   GTK_WIDGET_SET_FLAGS (widget, GTK_CAN_DEFAULT);
+#endif
 
   ogmrip_settings_bind_custom (settings, OGMRIP_GCONF_GENERAL, OGMRIP_GCONF_PROFILE,
       G_OBJECT (OGMRIP_OPTIONS_DIALOG (dialog)->priv->profile_combo), "active",
@@ -1294,7 +1302,7 @@
 
   return gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (dialog->priv->test_check)) &&
 #if GTK_CHECK_VERSION(2,18,0)
-    gtk_widget_get_sensitive (dialog->priv->test_check);
+    gtk_widget_is_sensitive (dialog->priv->test_check);
 #else
     GTK_WIDGET_IS_SENSITIVE (dialog->priv->test_check);
 #endif
Index: libogmrip-gtk/ogmrip-source-chooser-widget.c
===================================================================
--- libogmrip-gtk/ogmrip-source-chooser-widget.c	(révision 1960)
+++ libogmrip-gtk/ogmrip-source-chooser-widget.c	(copie de travail)
@@ -543,7 +543,11 @@
 
         toplevel = gtk_widget_get_toplevel (GTK_WIDGET (chooser));
 
+#if GTK_CHECK_VERSION(2,18,0)
+        if (gtk_widget_is_toplevel (toplevel) && GTK_IS_WINDOW (toplevel))
+#else
         if (GTK_WIDGET_TOPLEVEL (toplevel) && GTK_IS_WINDOW (toplevel))
+#endif
         {
           if (GTK_WINDOW (toplevel) != gtk_window_get_transient_for (GTK_WINDOW (chooser->priv->dialog)))
             gtk_window_set_transient_for (GTK_WINDOW (chooser->priv->dialog), GTK_WINDOW (toplevel));
