--- src/PDFDocument.cxx	2009-02-28 22:00:35.000000000 +0100
+++ src/PDFDocument.cxx	2011-01-30 18:28:13.650000158 +0100
@@ -678,19 +678,14 @@
     gdouble pageWidth, pageHeight;
     poppler_page_get_size(page, &pageWidth, &pageHeight);
 
-    //for get text we must exchange y coordinate, don't ask me where logic here.
     PopplerRectangle textRect = { rect->getX1() / getZoom(),
-                                  (pageHeight - rect->getY2() / getZoom()),
-                                  rect->getX2() * getZoom() / getZoom(),
-                                  (pageHeight - rect->getY1() / getZoom())};
+                                  rect->getY1() / getZoom(),
+                                  rect->getX2() / getZoom(),
+                                  rect->getY2() / getZoom()};
     repairEmpty(textRect);
 
-#if defined (HAVE_POPPLER_0_6_0)
-    gchar *text = poppler_page_get_text(page, POPPLER_SELECTION_GLYPH,
+    gchar *text = poppler_page_get_selected_text(page, POPPLER_SELECTION_GLYPH,
             &textRect);
-#else // !HAVE_POPPLER_0_6_0
-    gchar *text = poppler_page_get_text(page, &textRect);
-#endif // HAVE_POPPLER_0_6_0
     if(!text)
         goto cleanup;
 
