--- src/IDocument.cxx	2011-02-06 14:45:34.173334040 +0100
+++ src/IDocument.cxx    2011-02-06 14:51:23.173334040 +0100
@@ -135,7 +135,11 @@
     m_FindPage = 0;
     m_Format = NULL;
     m_Keywords = NULL;
+#if defined (HAVE_POPPLER_0_15_1)
+    m_Linearized = FALSE;
+#else
     m_Linearized = NULL;
+#endif
     m_ModifiedDate = NULL;
     m_PageCache = NULL;
     m_PageCacheAge = 0;
@@ -164,7 +168,9 @@
     g_free (m_FileName);
     g_free (m_Format);
     g_free (m_Keywords);
+#if !defined (HAVE_POPPLER_0_15_1)
     g_free (m_Linearized);
+#endif
     g_free (m_ModifiedDate);
     g_free (m_Password);
     g_free (m_Producer);
