--- cwiid-0.6.00.orig/libcwiid/bluetooth.c.old	2008-09-30 16:52:55.000000000 -0500
+++ cwiid-0.6.00.orig/libcwiid/bluetooth.c	2008-09-30 16:53:19.000000000 -0500
@@ -122,7 +122,7 @@
 		}
 
 		/* timeout (10000) in milliseconds */
-		if (hci_remote_name(sock, &dev_list[i].bdaddr, BT_NAME_LEN,
+		if (hci_read_remote_name(sock, &dev_list[i].bdaddr, BT_NAME_LEN,
 		                    (*bdinfo)[bdinfo_count].name, 10000)) {
 			cwiid_err(NULL, "Bluetooth name read error");
 			err = 1;
