--- flumotion/trunk/flumotion/common/package.py (revision 7162)
+++ flumotion/trunk/flumotion/common/package.py (revision 7927)
@@ -34,4 +34,27 @@
 
 __version__ = "$Rev$"
+
+
+class _PatchedModuleImporter(ihooks.ModuleImporter):
+    """
+    I am overriding ihook's ModuleImporter's import_module() method to
+    accept (and ignore) the 'level' keyword argument that appeared in
+    the built-in __import__() function in python2.5.
+
+    While no built-in modules in python2.5 seem to use that keyword
+    argument, 'encodings' module in python2.6 does and so it breaks if
+    used together with ihooks.
+
+    I make no attempt to properly support the 'level' argument -
+    ihooks didn't make it into py3k, and the only use in python2.6
+    we've seen so far, in 'encodings', serves as a performance hint
+    and it seems that can be ignored with no difference in behaviour.
+    """
+
+    def import_module(self, name, globals=None, locals=None, fromlist=None,
+                      level=-1):
+        # all we do is drop 'level' as ihooks don't support it, anyway
+        return ihooks.ModuleImporter.import_module(self, name, globals,
+                                                   locals, fromlist)
 
 
@@ -88,5 +111,10 @@
         self._hooks = PackageHooks()
         self._hooks.packager = self
-        self._importer = ihooks.ModuleImporter()
+        if sys.version_info < (2, 6):
+            self._importer = ihooks.ModuleImporter()
+        else:
+            self.debug('python2.6 or later detected - using patched'
+                       ' ModuleImporter')
+            self._importer = _PatchedModuleImporter()
         self._importer.set_hooks(self._hooks)
         self._importer.install()

