diff -Naur gnome-applets-3.5.92.orig/configure.in gnome-applets-3.5.92/configure.in
--- gnome-applets-3.5.92.orig/configure.in	2012-09-20 03:07:34.000000000 +0200
+++ gnome-applets-3.5.92/configure.in	2013-04-09 10:39:54.214243015 +0200
@@ -251,7 +251,7 @@
 
  dnl -- check for libgweather (required for gweather applet) ------------------
 build_libgweather_applets=false
-PKG_CHECK_MODULES(LIBGWEATHER, gweather-3.0 >= $GWEATHER_REQUIRED,
+PKG_CHECK_MODULES(LIBGWEATHER, gweather-3.0 >= $GWEATHER_REQUIRED gweather-3.0 <= 3.7,
 	          build_libgweather_applets=true,
 		  AC_MSG_WARN([libgweather not found. Not building the weather applet.]))
 AC_SUBST(LIBGWEATHER_CFLAGS)
