From bd4938bd943bc3810dc95ef7ced92674bcfe9797 Mon Sep 17 00:00:00 2001
From: anthraxx <levente@leventepolyak.net>
Date: Tue, 12 Sep 2017 19:45:51 +0200
Subject: [PATCH] don't pass IV in pycrypto CTR mode as counter must be
 exclusive

---
 axolotl/sessioncipher.py | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/axolotl/sessioncipher.py b/axolotl/sessioncipher.py
index 3b80ccb..6f41cea 100644
--- a/axolotl/sessioncipher.py
+++ b/axolotl/sessioncipher.py
@@ -229,11 +229,7 @@ class SessionCipher:
         # counterint = int.from_bytes(counterbytes, byteorder='big')
         ctr = Counter.new(128, initial_value=counter)
 
-        # cipher = AES.new(key, AES.MODE_CTR, counter=ctr)
-        ivBytes = bytearray(16)
-        ByteUtil.intToByteArray(ivBytes, 0, counter)
-
-        cipher = AES.new(key, AES.MODE_CTR, IV=bytes(ivBytes), counter=ctr)
+        cipher = AES.new(key, AES.MODE_CTR, counter=ctr)
 
         return cipher
 
-- 
2.14.1

