diff --git a/gfx/webrender_bindings/src/bindings.rs b/gfx/webrender_bindings/src/bindings.rs
index d3c44d69a150..c88285dae715 100644
--- a/gfx/webrender_bindings/src/bindings.rs
+++ b/gfx/webrender_bindings/src/bindings.rs
@@ -69,7 +69,7 @@ impl Into<WrExternalImageId> for ExternalImageId {
 const GL_FORMAT_BGRA_GL: gl::GLuint = gl::BGRA;
 const GL_FORMAT_BGRA_GLES: gl::GLuint = gl::BGRA_EXT;
 
-fn get_gl_format_bgra(gl: &gl::Gl) -> gl::GLuint {
+fn get_gl_format_bgra(gl: &dyn gl::Gl) -> gl::GLuint {
     match gl.get_type() {
         gl::GlType::Gl => {
             GL_FORMAT_BGRA_GL
@@ -787,7 +787,7 @@ pub extern "C" fn wr_window_new(window_id: WrWindowId,
                                 -> bool {
     assert!(unsafe { is_in_render_thread() });
 
-    let recorder: Option<Box<ApiRecordingReceiver>> = if ENABLE_RECORDING {
+    let recorder: Option<Box<dyn ApiRecordingReceiver>> = if ENABLE_RECORDING {
         let name = format!("wr-record-{}.bin", window_id.0);
         Some(Box::new(BinaryRecorder::new(&PathBuf::from(name))))
     } else {
