diff -wbBur lshw-B.02.14/src/core/abi.cc lshw-B.02.14.my//src/core/abi.cc
--- lshw-B.02.14/src/core/abi.cc	2009-02-12 22:53:09.000000000 +0000
+++ lshw-B.02.14.my//src/core/abi.cc	2010-05-24 14:30:18.000000000 +0000
@@ -10,6 +10,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <dirent.h>
+#include <stdlib.h>
 
 __ID("@(#) $Id: mem.cc 1352 2006-05-27 23:54:13Z ezix $");
 
diff -wbBur lshw-B.02.14/src/core/scsi.cc lshw-B.02.14.my//src/core/scsi.cc
--- lshw-B.02.14/src/core/scsi.cc	2009-02-12 22:53:09.000000000 +0000
+++ lshw-B.02.14.my//src/core/scsi.cc	2010-05-24 14:32:13.000000000 +0000
@@ -842,7 +842,7 @@
 
           if (!controller)
           {
-            controller = node.addChild(hwNode::hwNode("scsi", hw::storage));
+            controller = node.addChild(hwNode("scsi", hw::storage));
             if (controller)
             {
               controller->setLogicalName(host_logicalname(number));
