--- ./linbox/Makefile.in.orig	2012-06-07 15:19:27.000000000 -0600
+++ ./linbox/Makefile.in	2012-09-26 11:26:58.091786035 -0600
@@ -154,7 +154,7 @@ LTCXXCOMPILE = $(LIBTOOL) --tag=CXX $(AM
 CXXLD = $(CXX)
 CXXLINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=link $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) $(AM_LDFLAGS) \
-	$(LDFLAGS) -o $@
+	$(liblinbox_la_LDFLAGS) $(LDFLAGS) -o $@
 SOURCES = $(liblinbox_la_SOURCES)
 DIST_SOURCES = $(liblinbox_la_SOURCES)
 RECURSIVE_TARGETS = all-recursive check-recursive dvi-recursive \
@@ -390,6 +390,7 @@ liblinbox_la_LIBADD = \
     util/libutil.la             \
     randiter/libranditer.la     \
     algorithms/libalgorithms.la
+liblinbox_la_LDFLAGS = $(MPFR_LIBS) $(IML_LIBS) $(BLAS_LIBS)
 
 all: all-recursive
 
--- ./interfaces/sage/Makefile.in.orig	2012-06-07 15:19:27.000000000 -0600
+++ ./interfaces/sage/Makefile.in	2012-09-26 11:37:52.303447240 -0600
@@ -334,7 +334,7 @@ top_srcdir = @top_srcdir@
 #liblinboxsage_la_LIBADD = -llinbox $(BLAS_LIBS)
 #gentoo's linbox-1.1.6-fix-undefined-symbols.patch
 @LINBOX_HAVE_SAGE_TRUE@liblinboxsage_la_LIBADD = $(top_builddir)/linbox/liblinbox.la
-@LINBOX_HAVE_SAGE_TRUE@liblinboxsage_la_LDFLAGS = $(GIVARO_LIBS) $(GMP_LIBS) $(NTL_LIBS) $(BLAS_LIBS) $(MAPLE_LIBS) $(LDFLAGS) -version-info 0:0:0 #-Wl,-zmuldefs
+@LINBOX_HAVE_SAGE_TRUE@liblinboxsage_la_LDFLAGS = -Wl,--as-needed -L$(libdir)/atlas ../../linbox/liblinbox.la $(GIVARO_LIBS) $(NTL_LIBS) $(BLAS_LIBS) $(MAPLE_LIBS) $(LDFLAGS) -version-info 0:0:0 #-Wl,-zmuldefs
 all: all-am
 
 .SUFFIXES:
--- ./interfaces/driver/Makefile.in.orig	2012-06-07 15:19:27.000000000 -0600
+++ ./interfaces/driver/Makefile.in	2012-09-26 11:35:50.467638808 -0600
@@ -370,7 +370,7 @@ top_srcdir = @top_srcdir@
 
 # \
 #			#  lb-solve.C
-@LINBOX_COMPILE_DRIVERS_TRUE@liblbdriver_la_LDFLAGS = $(GIVARO_LIBS) $(GMP_LIBS) $(NTL_LIBS) $(BLAS_LIBS) $(LDFLAGS) $(top_srcdir)/linbox/liblinbox.la -Wl,-zmuldefs
+@LINBOX_COMPILE_DRIVERS_TRUE@liblbdriver_la_LDFLAGS = -Wl,--as-needed -L$(libdir)/atlas $(GIVARO_LIBS) $(NTL_LIBS) $(BLAS_LIBS) $(LDFLAGS) $(top_srcdir)/linbox/liblinbox.la -Wl,-zmuldefs
 @LINBOX_COMPILE_DRIVERS_TRUE@pkginclude_HEADERS = \
 @LINBOX_COMPILE_DRIVERS_TRUE@		lb-driver.h              \
 @LINBOX_COMPILE_DRIVERS_TRUE@		lb-blackbox-abstract.h   \
