diff -wbBur freeradius-server-2.1.10/src/modules/rlm_krb5/rlm_krb5.c freeradius-server-2.1.10.my/src/modules/rlm_krb5/rlm_krb5.c
--- freeradius-server-2.1.10/src/modules/rlm_krb5/rlm_krb5.c	2010-09-28 11:03:56.000000000 +0000
+++ freeradius-server-2.1.10.my/src/modules/rlm_krb5/rlm_krb5.c	2011-04-29 09:26:10.000000000 +0000
@@ -375,7 +375,7 @@
 	 * Heimdal krb5 verification
 	 */
 	radlog(L_AUTH, "rlm_krb5: Parsed name is: %s@%s\n",
-	       *userP->name.name_string.val,
+	       "-" /* *userP->name.name_string.val*/,
 	       userP->realm);
 
 	krb5_cc_default(context, &id);
@@ -390,7 +390,7 @@
 
        radlog(L_AUTH, "rlm_krb5: failed verify_user: %s (%s@%s )",
 	      error_message(ret),
-	      *userP->name.name_string.val,
+	      "-" /* *userP->name.name_string.val */,
 	      userP->realm);
 
        return RLM_MODULE_REJECT;
