diff -ur openresolv-3.3.4.orig/Makefile openresolv-3.3.4/Makefile
--- openresolv-3.3.4.orig/Makefile	2009-12-04 14:01:30.000000000 +0000
+++ openresolv-3.3.4/Makefile	2010-07-28 14:36:48.000000000 +0100
@@ -27,6 +27,7 @@
 # Try to embed correct service restart commands
 _CMD1=		\\1 status >/dev/null 2>\\&1
 _CMD2=		\\1 restart
+_CMD3=		\\1
 _CMD_SH=if [ -x /sbin/rc-service ]; then \
 		printf '/sbin/rc-service -e \\1 \\&\\& /sbin/rc-service \\1 -- -Ds restart'; \
 	elif [ -x /sbin/service ]; then \
@@ -35,6 +36,8 @@
 		printf 'if /usr/local/etc/rc.d/${_CMD1}; then'; \
 		printf ' /usr/local/etc/rc.d/${_CMD2}; '; \
 		printf 'elif /etc/rc.d/${_CMD1}; then /etc/rc.d/${_CMD2}; fi'; \
+	elif [ -d /etc/rc.d ] && grep -q "Arch Linux" /etc/rc.sysinit; then \
+		printf '[ -e /var/run/daemons/${_CMD3} ] \\&\\& /etc/rc.d/${_CMD2}'; \
 	elif [ -d /etc/rc.d ]; then \
 		printf '/etc/rc.d/${_CMD1} \\&\\& /etc/rc.d/${_CMD2}'; \
 	elif [ -d /etc/init.d ]; then \
