--- a/build/lib/git-environment.js
+++ b/build/lib/git-environment.js
@@ -5,14 +5,7 @@
  *  Find the path to the embedded Git environment
  */
 function resolveGitDir() {
-    if (process.env.LOCAL_GIT_DIRECTORY) {
-        return path.resolve(process.env.LOCAL_GIT_DIRECTORY);
-    }
-    else {
-        const s = path.sep;
-        return path.resolve(__dirname, '..', '..', 'git')
-            .replace(/[\\\/]app.asar[\\\/]/, `${s}app.asar.unpacked${s}`);
-    }
+    return '/usr';
 }
 /**
  *  Find the path to the embedded Git binary
@@ -33,7 +26,7 @@
 function resolveGitExecPath() {
     const gitDir = resolveGitDir();
     if (process.platform === 'darwin' || process.platform === 'linux') {
-        return path.join(gitDir, 'libexec', 'git-core');
+        return path.join(gitDir, 'lib', 'git-core');
     }
     else if (process.platform === 'win32') {
         return path.join(gitDir, 'mingw64', 'libexec', 'git-core');
@@ -80,10 +73,6 @@
         // an arbitrary location, you should set PREFIX for the
         // process to ensure that it knows how to resolve things
         env.PREFIX = gitDir;
-        // bypass whatever certificates might be set and use
-        // the bundle included in the distibution
-        const sslCABundle = `${gitDir}/ssl/cacert.pem`;
-        env.GIT_SSL_CAINFO = sslCABundle;
     }
     return { env, gitLocation };
 }
