--- a/src/package-manager.coffee
+++ b/src/package-manager.coffee
@@ -145,12 +145,7 @@
     return configPath if configPath
     return @apmPath if @apmPath?
 
-    commandName = 'apm'
-    commandName += '.cmd' if process.platform is 'win32'
-    apmRoot = path.join(process.resourcesPath, 'app', 'apm')
-    @apmPath = path.join(apmRoot, 'bin', commandName)
-    unless fs.isFileSync(@apmPath)
-      @apmPath = path.join(apmRoot, 'node_modules', 'atom-package-manager', 'bin', commandName)
+    @apmPath = '/usr/lib/node_modules/atom-package-manager/bin/apm'
     @apmPath
 
   # Public: Get the paths being used to look for packages.
