diff --git a/src/ukws_wayland_handler.h b/src/ukws_wayland_handler.h
index 0dceb25..f913d42 100644
--- a/src/ukws_wayland_handler.h
+++ b/src/ukws_wayland_handler.h
@@ -21,8 +21,6 @@
 #define UKWSWAYLANDHANDLER_H
 
 #include <QList>
-#include <KWayland/Client/plasmawindowmanagement.h>
-#include <KWayland/Client/plasmavirtualdesktop.h>
 
 #define XDG_TYPE "XDG_SESSION_TYPE"
 
diff --git a/ukui-window-switch.pro b/ukui-window-switch.pro
index f141557..346ad86 100644
--- a/ukui-window-switch.pro
+++ b/ukui-window-switch.pro
@@ -1,4 +1,4 @@
-QT  += core gui x11extras dbus KGlobalAccel KWaylandClient KWaylandServer
+QT  += core gui x11extras dbus KGlobalAccel
 greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
 
 CONFIG += c++11 console
