--- Electrum-1.9.8/gui/qt/installwizard.py.orig	2014-09-19 22:05:42.536699757 +0000
+++ Electrum-1.9.8/gui/qt/installwizard.py	2014-09-19 22:06:48.993369698 +0000
@@ -31,8 +31,8 @@
     def set_layout(self, layout):
         w = QWidget()
         w.setLayout(layout)
-        self.stack.setCurrentIndex(self.stack.addWidget(w))
-
+        self.stack.addWidget(w)
+        self.stack.setCurrentWidget(w)
 
     def restore_or_create(self):
 
