--- ./setup.py	2014-03-16 12:53:25.136657771 -0700
+++ ./setup.py	2014-03-16 12:53:42.736657586 -0700
@@ -16,8 +16,6 @@
     sys.exit("Error: Electrum requires Python version >= 2.6.0...")
 
 usr_share = '/usr/share'
-if not os.access(usr_share, os.W_OK):
-    usr_share = os.getenv("XDG_DATA_HOME", os.path.join(os.getenv("HOME"), ".local", "share"))
 
 data_files = []
 if (len(sys.argv) > 1 and (sys.argv[1] == "sdist")) or (platform.system() != 'Windows' and platform.system() != 'Darwin'):
