diff --git a/dephell_discover/_line.py b/dephell_discover/_line.py
index 6739c34..31c82e4 100644
--- a/dephell_discover/_line.py
+++ b/dephell_discover/_line.py
@@ -53,7 +53,7 @@ class Line:
         value = tree.body[0].value  # type: ignore
 
         # get string
-        if type(value) is ast.Str:
+        if type(value) is ast.Constant:
             return cls(
                 target=target.id,
                 value=value.s,
@@ -65,7 +65,7 @@ class Line:
         # get list
         if type(value) is ast.List or type(value) is ast.Tuple:
             for element in value.elts:
-                if type(element) is not ast.Str:
+                if type(element) is not ast.Constant:
                     return None
             return cls(
                 target=target.id,
