diff --git a/src/gs-window-x11.c b/src/gs-window-x11.c
index d267f59..fbfbd8a 100644
--- a/src/gs-window-x11.c
+++ b/src/gs-window-x11.c
@@ -1283,6 +1283,11 @@ shake_dialog (GSWindow *window)
                 g_usleep (10000);
         }
 
+        gtk_alignment_set_padding (GTK_ALIGNMENT (window->priv->lock_box),
+                                   0, 0,
+                                   0,
+                                   0);
+
         window->priv->dialog_shake_in_progress = FALSE;
         maybe_kill_dialog (window);
 }
