diff -Naur blam-1.8.11.orig/gtk-gui/Imendio.Blam.AddChannelDialog.cs blam-1.8.11/gtk-gui/Imendio.Blam.AddChannelDialog.cs
--- blam-1.8.11.orig/gtk-gui/Imendio.Blam.AddChannelDialog.cs	2013-03-16 19:24:08.000000000 +0100
+++ blam-1.8.11/gtk-gui/Imendio.Blam.AddChannelDialog.cs	2013-03-28 22:54:57.084680327 +0100
@@ -92,7 +92,7 @@
 			this.PasswordEntry.CanFocus = true;
 			this.PasswordEntry.Name = "PasswordEntry";
 			this.PasswordEntry.IsEditable = true;
-			this.PasswordEntry.InvisibleChar = '•';
+			this.PasswordEntry.InvisibleChar = '\u25CF';
 			this.table1.Add (this.PasswordEntry);
 			global::Gtk.Table.TableChild w8 = ((global::Gtk.Table.TableChild)(this.table1 [this.PasswordEntry]));
 			w8.TopAttach = ((uint)(2));
@@ -105,7 +105,7 @@
 			this.UrlEntry.CanFocus = true;
 			this.UrlEntry.Name = "UrlEntry";
 			this.UrlEntry.IsEditable = true;
-			this.UrlEntry.InvisibleChar = '•';
+			this.UrlEntry.InvisibleChar = '\u25CF';
 			this.table1.Add (this.UrlEntry);
 			global::Gtk.Table.TableChild w9 = ((global::Gtk.Table.TableChild)(this.table1 [this.UrlEntry]));
 			w9.LeftAttach = ((uint)(1));
@@ -116,7 +116,7 @@
 			this.UsernameEntry.CanFocus = true;
 			this.UsernameEntry.Name = "UsernameEntry";
 			this.UsernameEntry.IsEditable = true;
-			this.UsernameEntry.InvisibleChar = '•';
+			this.UsernameEntry.InvisibleChar = '\u25CF';
 			this.table1.Add (this.UsernameEntry);
 			global::Gtk.Table.TableChild w10 = ((global::Gtk.Table.TableChild)(this.table1 [this.UsernameEntry]));
 			w10.TopAttach = ((uint)(1));
