From 8d3c335b578e7c33942a92194f01ad95c227809f Mon Sep 17 00:00:00 2001
From: anthraxx <levente@leventepolyak.net>
Date: Wed, 19 Dec 2018 01:10:25 +0100
Subject: [PATCH] python: de-vendor venv paths

---
 runtime/Server.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/runtime/Server.cpp b/runtime/Server.cpp
index 15344c11..203fe1a9 100644
--- a/runtime/Server.cpp
+++ b/runtime/Server.cpp
@@ -135,11 +135,11 @@ Server::Server(quint16 port, QString key, QString logFileName)
     add_to_path(pythonHome, venvPath.canonicalFilePath());
 #else
     // Build (and canonicalise) the virtual environment path
-    QFileInfo venvBinPath(app_dir + "/../venv/bin");
-    QFileInfo venvLibPath(app_dir + "/../venv/lib/python");
-    QFileInfo venvDynLibPath(app_dir + "/../venv/lib/python/lib-dynload");
-    QFileInfo venvSitePackagesPath(app_dir + "/../venv/lib/python/site-packages");
-    QFileInfo venvPath(app_dir + "/../venv");
+    QFileInfo venvBinPath("/usr/bin");
+    QFileInfo venvLibPath("/usr/lib/python##PYTHONVERSION##");
+    QFileInfo venvDynLibPath("/usr/lib/python##PYTHONVERSION##/lib-dynload");
+    QFileInfo venvSitePackagesPath("/usr/lib/python##PYTHONVERSION##/site-packages");
+    QFileInfo venvPath("/usr/lib/python##PYTHONVERSION##");
 
     // Prepend the bin directory to the path
     add_to_path(path_env, venvBinPath.canonicalFilePath(), true);
-- 
2.20.1

