Index: cinelerra-20090210/quicktime/qth264.c
===================================================================
--- cinelerra-20090210.orig/quicktime/qth264.c
+++ cinelerra-20090210/quicktime/qth264.c
@@ -239,10 +239,15 @@ static int encode(quicktime_t *file, uns
 	codec->buffer_size = 0;
 	for(i = 0; i < nnal; i++)
 	{
+#if X264_BUILD >= 76
+		int size = nals[i].i_payload;
+		memcpy(codec->work_buffer + codec->buffer_size, nals[i].p_payload, nals[i].i_payload);
+#else
 		int size = x264_nal_encode(codec->work_buffer + codec->buffer_size, 
 			&allocation, 
 			1, 
 			nals + i);
+#endif
 		unsigned char *ptr = codec->work_buffer + codec->buffer_size;
 
 		if(size > 0)
