diff -Naur emesene-emesene-b89aa5e.orig/emesene/gui/gtkui/Preferences.py emesene-emesene-b89aa5e/emesene/gui/gtkui/Preferences.py
--- emesene-emesene-b89aa5e.orig/emesene/gui/gtkui/Preferences.py	2013-09-05 10:51:23.394961971 +0000
+++ emesene-emesene-b89aa5e/emesene/gui/gtkui/Preferences.py	2013-09-05 10:52:29.541187002 +0000
@@ -1287,6 +1287,7 @@
         self.append_check(_('Open mail in default desktop client'),
                           'session.config.b_open_mail_in_desktop')
 
+        self.session.config.get_or_set('b_check_for_updates', True)
         self.append_markup('<b>'+_('Updates')+'</b>')
         self.append_check(_('Weekly check for plugins/themes updates on startup'),
                           'session.config.b_check_for_updates')
