commit 6ab20d48d85638acb3a811c8676f713bd26f0df9
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Sat May 28 15:08:44 2016 +0800

    add setTimeZone in testGetBestPattern (fixes #19)

diff --git a/test/test_DateTimeParserGenerator.py b/test/test_DateTimeParserGenerator.py
index 5838e32..de8eb88 100644
--- a/test/test_DateTimeParserGenerator.py
+++ b/test/test_DateTimeParserGenerator.py
@@ -71,6 +71,7 @@ class TestDateTimePatternGenerator(TestCase):
             for index, skeleton in enumerate(skeletons):
                 pattern = dtpg.getBestPattern(skeleton)
                 sdf = SimpleDateFormat(pattern, locale)
+                sdf.setTimeZone(self.tz)
                 self.assertEqual(sdf.format(self.date), locale_data[index])
 
     def testReplaceFieldType(self):
