From 2f083aba12994c1c8be7a85f543b38c2c2662335 Mon Sep 17 00:00:00 2001
From: Moritz Lipp <mlq@pwmt.org>
Date: Thu, 21 Jan 2016 11:38:31 +0100
Subject: [PATCH] Use correct color space

---
 render.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/render.c b/render.c
index 99d3b9e..2a92a23 100644
--- a/render.c
+++ b/render.c
@@ -36,7 +36,7 @@ pdf_page_render_to_buffer(mupdf_document_t* mupdf_document, mupdf_page_t* mupdf_
   fz_irect irect = { .x1 = page_width, .y1 = page_height };
   fz_rect rect = { .x1 = page_width, .y1 = page_height };
 
-  fz_colorspace* colorspace = fz_device_rgb(mupdf_document->ctx);
+  fz_colorspace* colorspace = fz_device_bgr(mupdf_document->ctx);
   fz_pixmap* pixmap = fz_new_pixmap_with_bbox_and_data(mupdf_page->ctx, colorspace, &irect, image);
   fz_clear_pixmap_with_value(mupdf_page->ctx, pixmap, 0xFF);
 
--
libgit2 0.23.3

