diff --git i/pynitrokey/cli/__init__.py w/pynitrokey/cli/__init__.py
index 3dd8ca3..a22ff1f 100644
--- i/pynitrokey/cli/__init__.py
+++ w/pynitrokey/cli/__init__.py
@@ -17,7 +17,6 @@ import pynitrokey
 import pynitrokey.fido2.operations
 from pynitrokey.cli.fido2 import fido2
 from pynitrokey.cli.nethsm import nethsm
-from pynitrokey.cli.nk3 import nk3
 from pynitrokey.cli.pro import pro
 from pynitrokey.cli.start import start
 from pynitrokey.cli.storage import storage
@@ -57,7 +56,6 @@ def nitropy():
 
 nitropy.add_command(fido2)
 nitropy.add_command(nethsm)
-nitropy.add_command(nk3)
 nitropy.add_command(start)
 nitropy.add_command(storage)
 nitropy.add_command(pro)
@@ -78,7 +76,6 @@ def ls():
 
     fido2.commands["list"].callback()
     start.commands["list"].callback()
-    nk3.commands["list"].callback()
 
 
 nitropy.add_command(ls)
diff --git i/pyproject.toml w/pyproject.toml
index 8c69386..c61c206 100644
--- i/pyproject.toml
+++ w/pyproject.toml
@@ -21,7 +21,6 @@ requires = [
   "requests",
   "pygments",
   "python-dateutil",
-  "spsdk >= 1.5.0",
   "tqdm",
   "urllib3",
 	"cffi",
