--- rst2pdf-0.91/rst2pdf/math_flowable.py	Tue Mar  6 02:44:00 2012
+++ rst2pdf-0.91/rst2pdf/math_flowable.py	Wed May  2 00:42:05 2012
@@ -85,13 +85,14 @@
                 canv.setFillColorRGB(1,0,0)
                 canv.drawString(0,0,self.s)
             canv.restoreState()
-        if self.label:
-            log.info('Drawing equation-%s'%self.label)
-            canv.bookmarkHorizontal('equation-%s'%self.label,0,height)
         else:
             canv.saveState()
             canv.drawString(x, y, self.s)
             canv.restoreState()
+
+        if self.label:
+            log.info('Drawing equation-%s'%self.label)
+            canv.bookmarkHorizontal('equation-%s'%self.label,0,height)
 
     def descent(self):
         """Return the descent of this flowable,
