--- files.h.orig	2004-11-19 22:00:56.000000000 +0100
+++ files.h	2015-05-18 14:06:21.109931758 +0200
@@ -108,17 +108,17 @@
 template <class H>
   void printHeckeElt(FILE* file, const H& h, const SchubertContext& p, 
 		     const Interface& I, OutputTraits& traits, 
-		     const Length& l = undef_length);
+		     const Length& l);
 template <class H>
   void printHeckeElt(FILE* file, const H& h, const SchubertContext& p, 
 		     const Interface& I, OutputTraits& traits, 
-		     const Length& l = undef_length);
+		     const Length& l);
 template <class H>
   void printHeckeElt(FILE* file, const H& h, const Permutation& a, 
 		     const SchubertContext& p, const Interface& I,
 		     HeckeTraits& hTraits,
 		     PolynomialTraits& pTraits,
-		     const Length& l = undef_length);
+		     const Length& l);
 void printHomology(FILE* file, const Homology& h, OutputTraits& traits);
 template <class KL>
   void printIHBetti(FILE* file, const CoxNbr& y, KL& kl, OutputTraits& traits);
@@ -187,7 +187,7 @@
 template <class H>
   bool setTwoSided(const H& h, const Permutation& a, const SchubertContext& p,
 		   const Interface& I, HeckeTraits& hTraits,
-		   PolynomialTraits& pTraits, const Length& l = undef_length);
+		   PolynomialTraits& pTraits, const Length& l);
 void sortLists(List<List<CoxNbr> >& lc, schubert::NFCompare& nfc, 
 	       Permutation& a);
 void writeClasses(List<List<CoxNbr> >& lc, const Partition& pi);
