--- a/doc/tools/coqrst/coqdomain.py
+++ b/doc/tools/coqrst/coqdomain.py
@@ -21,6 +21,7 @@ from itertools import chain
 from collections import defaultdict
 
 from docutils import nodes, utils
+from docutils.nodes import math_block
 from docutils.transforms import Transform
 from docutils.parsers.rst import Directive, directives
 from docutils.parsers.rst.roles import code_role #, set_classes
@@ -34,7 +35,6 @@ from sphinx.util.logging import getLogge
 from sphinx.directives import ObjectDescription
 from sphinx.domains import Domain, ObjType, Index
 from sphinx.domains.std import token_xrefs
-from sphinx.ext import mathbase
 
 from . import coqdoc
 from .repl import ansicolors
@@ -74,8 +74,8 @@ def make_target(objtype, targetid):
     return "coq:{}.{}".format(objtype, targetid)
 
 def make_math_node(latex, docname, nowrap):
-    node = mathbase.displaymath()
-    node['latex'] = latex
+    node = math_block()
+    node['text'] = latex
     node['label'] = None # Otherwise equations are numbered
     node['nowrap'] = nowrap
     node['docname'] = docname
@@ -1249,11 +1249,11 @@ def setup(app):
     app.connect('doctree-resolved', CoqtopBlocksTransform.merge_consecutive_coqtop_blocks)
 
     # Add extra styles
-    app.add_stylesheet("ansi.css")
-    app.add_stylesheet("coqdoc.css")
-    app.add_javascript("notations.js")
-    app.add_stylesheet("notations.css")
-    app.add_stylesheet("pre-text.css")
+    app.add_css_file("ansi.css")
+    app.add_css_file("coqdoc.css")
+    app.add_js_file("notations.js")
+    app.add_css_file("notations.css")
+    app.add_css_file("pre-text.css")
 
     # Tell Sphinx about extra settings
     app.add_config_value("report_undocumented_coq_objects", None, 'env')
