diff -Naur tasque-0.1.9.orig/configure.ac tasque-0.1.9/configure.ac
--- tasque-0.1.9.orig/configure.ac	2010-01-12 20:45:49.000000000 +0100
+++ tasque-0.1.9/configure.ac	2012-02-27 12:10:20.441455163 +0100
@@ -42,8 +42,8 @@
 PKG_CHECK_MODULES([GTK_SHARP_20], [gtk-sharp-2.0 >= 2.10.0])
 PKG_CHECK_MODULES([GTK_SHARP_2_12], gtk-sharp-2.0 >= 2.12.0, GTK_2_12="yes", GTK_2_12="no")
 AM_CONDITIONAL(GTK_2_12, test "$GTK_2_12" = "yes")
-PKG_CHECK_MODULES([NDESK_DBUS_10], [ndesk-dbus-1.0])
-PKG_CHECK_MODULES([NDESK_DBUS_GLIB_10], [ndesk-dbus-glib-1.0])
+PKG_CHECK_MODULES([DBUS_SHARP_10], [dbus-sharp-1.0])
+PKG_CHECK_MODULES([DBUS_SHARP_GLIB_10], [dbus-sharp-glib-1.0])
 
 #
 # Allow the project to build without notify-sharp
diff -Naur tasque-0.1.9.orig/src/Backends/IceCore/IceBackend.cs tasque-0.1.9/src/Backends/IceCore/IceBackend.cs
--- tasque-0.1.9.orig/src/Backends/IceCore/IceBackend.cs	2010-01-12 20:45:49.000000000 +0100
+++ tasque-0.1.9/src/Backends/IceCore/IceBackend.cs	2012-02-27 12:08:23.195416220 +0100
@@ -3,7 +3,7 @@
 
 using System;
 using System.Collections.Generic;
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 using Novell.IceDesktop;
 using Tasque;
diff -Naur tasque-0.1.9.orig/src/Backends/IceCore/Makefile.am tasque-0.1.9/src/Backends/IceCore/Makefile.am
--- tasque-0.1.9.orig/src/Backends/IceCore/Makefile.am	2010-01-12 20:45:49.000000000 +0100
+++ tasque-0.1.9/src/Backends/IceCore/Makefile.am	2012-02-27 12:08:41.080970148 +0100
@@ -11,8 +11,8 @@
 	$(GNOME_SHARP_20_LIBS) \
 	$(GTK_DOTNET_20_LIBS) \
 	$(ICE_DESKTOP_LIBS) \
-	$(NDESK_DBUS_10_LIBS) \
-	$(NDESK_DBUS_GLIB_10_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
+	$(DBUS_SHARP_GLIB_10_LIBS) \
 	-r:System \
 	-r:Mono.Posix
 	-r:System.xml
diff -Naur tasque-0.1.9.orig/src/Makefile.am tasque-0.1.9/src/Makefile.am
--- tasque-0.1.9.orig/src/Makefile.am	2010-01-12 20:45:49.000000000 +0100
+++ tasque-0.1.9/src/Makefile.am	2012-02-27 12:07:55.520471332 +0100
@@ -133,8 +133,8 @@
 	$(GNOME_SHARP_20_LIBS) \
 	$(GTK_DOTNET_20_LIBS) \
 	$(NOTIFY_SHARP_LIBS) \
-	$(NDESK_DBUS_10_LIBS) \
-	$(NDESK_DBUS_GLIB_10_LIBS) \
+	$(DBUS_SHARP_10_LIBS) \
+	$(DBUS_SHARP_GLIB_10_LIBS) \
 	$(ICE_DESKTOP_LIBS) \
 	$(SQLITE_LIBS) \
 	$(EVOLUTION_SHARP_LIBS)
diff -Naur tasque-0.1.9.orig/src/RemoteControl.cs tasque-0.1.9/src/RemoteControl.cs
--- tasque-0.1.9.orig/src/RemoteControl.cs	2010-01-12 20:45:49.000000000 +0100
+++ tasque-0.1.9/src/RemoteControl.cs	2012-02-27 12:07:18.036136730 +0100
@@ -14,7 +14,7 @@
 #endif
 
 using org.freedesktop.DBus;
-using NDesk.DBus;
+using DBus;
 
 namespace Tasque
 {
diff -Naur tasque-0.1.9.orig/src/RemoteControlProxy.cs tasque-0.1.9/src/RemoteControlProxy.cs
--- tasque-0.1.9.orig/src/RemoteControlProxy.cs	2010-01-12 20:45:49.000000000 +0100
+++ tasque-0.1.9/src/RemoteControlProxy.cs	2012-02-27 12:09:05.522782848 +0100
@@ -1,5 +1,5 @@
 using System;
-using NDesk.DBus;
+using DBus;
 using org.freedesktop.DBus;
 
 namespace Tasque
