--- src/setup.py.orig	2014-11-01 23:49:10.000000000 +0100
+++ src/setup.py	2014-11-07 00:40:58.414838636 +0100
@@ -40,7 +40,7 @@
 
 # search for dependencies and add to gcc -I<path>
 include_dirs = [SAGE_INC,
-                os.path.join(SAGE_INC, 'csage'),
+                'c_lib/include',
                 SAGE_SRC,
                 os.path.join(SAGE_SRC, 'sage', 'ext')]
 
@@ -171,10 +171,12 @@
     m.libraries = ['csage'] + m.libraries
     if m.language == 'c++':
         m.libraries.append('stdc++')
+        m.libraries.append('ntl')
+    m.libraries = m.libraries + ['gmp', 'm', 'dl']
 
     m.extra_compile_args = m.extra_compile_args + extra_compile_args
     m.extra_link_args = m.extra_link_args + extra_link_args
-    m.library_dirs = m.library_dirs + [os.path.join(SAGE_LOCAL, "lib")]
+    m.library_dirs += ['c_lib', '%s/lib' % SAGE_LOCAL]
     m.include_dirs = m.include_dirs + include_dirs

 
 
