diff --git a/cgi.gemspec b/cgi.gemspec
index 3ba62b9..5809afc 100644
--- a/cgi.gemspec
+++ b/cgi.gemspec
@@ -1,15 +1,8 @@
 # frozen_string_literal: true
 
-name = File.basename(__FILE__, ".gemspec")
-version = ["lib", Array.new(name.count("-")+1, "..").join("/")].find do |dir|
-  break File.foreach(File.join(__dir__, dir, "#{name.tr('-', '/')}.rb")) do |line|
-    /^\s*VERSION\s*=\s*"(.*)"/ =~ line and break $1
-  end rescue nil
-end
-
 Gem::Specification.new do |spec|
-  spec.name          = name
-  spec.version       = version
+  spec.name          = 'cgi'
+  spec.version       = '0.3.2'
   spec.authors       = ["Yukihiro Matsumoto"]
   spec.email         = ["matz@ruby-lang.org"]
 
