--- a/depends/packages/librustzcash.mk
+++ b/depends/packages/librustzcash.mk
@@ -5,7 +5,7 @@
 $(package)_download_file=$($(package)_git_commit).tar.gz
 $(package)_sha256_hash=4d022b66e554efbf6db01b2a282e312e8a1b492c4680299ae8c26629882eb46b
 $(package)_git_commit=f5d2afb4eabac29b1b1cc860d66e45a5b48b4f88
-$(package)_dependencies=rust $(rust_crates)
+$(package)_dependencies=$(rust_crates)
 $(package)_patches=cargo.config
 
 define $(package)_preprocess_cmds
