--- a/qa/zcash/full_test_suite.py
+++ b/qa/zcash/full_test_suite.py
@@ -149,9 +149,9 @@
         return False
 
     rust_env = os.environ.copy()
-    rust_env['RUSTC'] = os.path.join(arch_dir, 'native', 'bin', 'rustc')
+    rust_env['RUSTC'] = 'rustc'
     return subprocess.call([
-        os.path.join(arch_dir, 'native', 'bin', 'cargo'),
+        'cargo',
         'test',
         '--manifest-path',
         os.path.join(REPOROOT, 'Cargo.toml'),
