--- a/depends/packages/libsnark.mk
+++ b/depends/packages/libsnark.mk
@@ -6,8 +6,6 @@
 $(package)_sha256_hash=9422b1a2a94e6b8be61f07af7f146087c2a7d70b208d07ad076622225aa7f0e4
 $(package)_git_commit=2e6314a9f7efcd9af1c77669d7d9a229df86a777
 
-$(package)_dependencies=libgmp libsodium
-
 define $(package)_build_cmds
   CXXFLAGS="-fPIC -DBINARY_OUTPUT -DNO_PT_COMPRESSION=1" $(MAKE) lib DEPINST=$(host_prefix) CURVE=ALT_BN128 MULTICORE=1 NO_PROCPS=1 NO_GTEST=1 NO_DOCS=1 STATIC=1 NO_SUPERCOP=1 FEATUREFLAGS=-DMONTGOMERY_OUTPUT OPTFLAGS="-O2 -march=x86-64"
 endef
