Description: Make fakedri_drv build with XServer 1.9.
Origin: upstream, http://www.virtualbox.org/changeset/31549

diff -Nur virtualbox-ose-3.2.8-dfsg.orig/src/VBox/Additions/common/crOpenGL/fakedri_drv.c virtualbox-ose-3.2.8-dfsg/src/VBox/Additions/common/crOpenGL/fakedri_drv.c
--- virtualbox-ose-3.2.8-dfsg.orig/src/VBox/Additions/common/crOpenGL/fakedri_drv.c	2010-08-05 13:42:12.000000000 +0200
+++ virtualbox-ose-3.2.8-dfsg/src/VBox/Additions/common/crOpenGL/fakedri_drv.c	2010-08-10 12:08:26.944142812 +0200
@@ -334,7 +334,11 @@
 void __attribute__ ((constructor)) vbox_install_into_mesa(void)
 {
     {
+#ifdef _X_ATTRIBUTE_PRINTF
+        void (*pxf86Msg)(MessageType type, const char *format, ...) _X_ATTRIBUTE_PRINTF(2,3);
+#else
         void (*pxf86Msg)(MessageType type, const char *format, ...) _printf_attribute(2,3);
+#endif
 
         pxf86Msg = dlsym(RTLD_DEFAULT, "xf86Msg");
         if (pxf86Msg)
