diff --git a/frontend/linux/workbench/mysql-workbench.in b/frontend/linux/workbench/mysql-workbench.in
index 77d60ac..6199b46 100755
--- a/frontend/linux/workbench/mysql-workbench.in
+++ b/frontend/linux/workbench/mysql-workbench.in
@@ -4,6 +4,12 @@
 # This will cause passwords to be stored only temporarily for the session.
 #WB_NO_GNOME_KEYRING=1
 
+# GNOME keyring does no longer export GNOME_KEYRING_CONTROL and MySQL Workbench
+# fails to detect it. So simply export here when executable is found.
+if test -x "/usr/bin/gnome-keyring-daemon"; then
+	export GNOME_KEYRING_CONTROL=1
+fi
+
 # force disable the Mac style single menu hack in Ubuntu Unity
 UBUNTU_MENUPROXY=0
 
