diff --git a/plugins/wb.admin/backend/wb_admin_ssh.py b/plugins/wb.admin/backend/wb_admin_ssh.py
index 11401aa..fa0ff71 100644
--- a/plugins/wb.admin/backend/wb_admin_ssh.py
+++ b/plugins/wb.admin/backend/wb_admin_ssh.py
@@ -88,6 +88,9 @@ if paramiko and server_version_str2tuple(paramiko.__version__) >= (1, 7, 4):
             if (server_version_str2tuple(paramiko.__version__) <= (1, 14, 99)):
                 m.add_int(self.window_size)
                 m.add_int(self.max_packet_size)
+            else:
+                m.add_int(self.default_window_size)
+                m.add_int(self.default_max_packet_size)
             if (kind == 'forwarded-tcpip') or (kind == 'direct-tcpip'):
                 m.add_string(dest_addr[0])
                 m.add_int(dest_addr[1])
@@ -103,6 +106,8 @@ if paramiko and server_version_str2tuple(paramiko.__version__) >= (1, 7, 4):
             chan._set_transport(self)
             if (server_version_str2tuple(paramiko.__version__) <= (1, 14, 99)):
                 chan._set_window(self.window_size, self.max_packet_size)
+            else:
+                chan._set_window(self.default_window_size, self.default_max_packet_size)
         finally:
             self.lock.release()
         self._send_user_message(m)
