diff --git a/frontend/linux/workbench/mysql-workbench.in b/frontend/linux/workbench/mysql-workbench.in
index 6e85f19..4db5edc 100755
--- a/frontend/linux/workbench/mysql-workbench.in
+++ b/frontend/linux/workbench/mysql-workbench.in
@@ -22,7 +22,7 @@ wblibdir="$destdir@WB_INSTALL_LIB_DIR@"
 
 # Set the PROJSO env variable so gdal can find proj cause it's using dlopen instead ld 
 PROJLOC=""
-for lib in libproj.so.0 libproj.so.9
+for lib in libproj.so.0 libproj.so.9 libproj.so.12
 do
   for loc in /usr/lib /usr/lib64 /usr/lib/x86_64-linux-gnu
   do
