diff --git a/tools/faust2appls/faust2md b/tools/faust2appls/faust2md
index 2e460fcb9..5831a86a9 100755
--- a/tools/faust2appls/faust2md
+++ b/tools/faust2appls/faust2md
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #---------------------- faust2md -----------------------
 # Usage: `faust2md [-t 4] [-c] [-f] foo.dsp > foo.md`
diff --git a/tools/faust2appls/faust2atomsnippets b/tools/faust2appls/faust2atomsnippets
index 183436022..989618a30 100755
--- a/tools/faust2appls/faust2atomsnippets
+++ b/tools/faust2appls/faust2atomsnippets
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #---------------------- faust2atomsnippets -----------------------
 # Usage: `faust2atomsnippets *.lib > faust-library.cson`
