diff -aur libSavitar-4.1.0.old/CMakeLists.txt libSavitar-4.1.0/CMakeLists.txt
--- libSavitar-4.1.0.old/CMakeLists.txt	2019-06-14 21:54:00.101539146 +0200
+++ libSavitar-4.1.0/CMakeLists.txt	2019-06-14 21:54:34.518676200 +0200
@@ -14,7 +14,7 @@
     find_package(Threads QUIET)
 endif()
 
-add_subdirectory(pugixml)
+find_package(pugixml REQUIRED)
 
 if(BUILD_PYTHON)
     list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
diff -aur libSavitar-4.1.0.old/src/MeshData.cpp libSavitar-4.1.0/src/MeshData.cpp
--- libSavitar-4.1.0.old/src/MeshData.cpp	2019-06-14 21:54:00.101539146 +0200
+++ libSavitar-4.1.0/src/MeshData.cpp	2019-06-14 21:56:16.930076069 +0200
@@ -17,7 +17,7 @@
  */
 
 #include "MeshData.h"
-#include "../pugixml/src/pugixml.hpp"
+#include <pugixml.hpp>
 #include <cstring>
 #include <iostream>
 #include <stdexcept> //For std::runtime_error.
diff -aur libSavitar-4.1.0.old/src/Scene.cpp libSavitar-4.1.0/src/Scene.cpp
--- libSavitar-4.1.0.old/src/Scene.cpp	2019-06-14 21:54:00.101539146 +0200
+++ libSavitar-4.1.0/src/Scene.cpp	2019-06-14 21:56:09.553308560 +0200
@@ -17,7 +17,7 @@
  */
 
 #include "Scene.h"
-#include "../pugixml/src/pugixml.hpp"
+#include <pugixml.hpp>
 #include <iostream>
 #include <string>
 using namespace Savitar;
diff -aur libSavitar-4.1.0.old/src/SceneNode.cpp libSavitar-4.1.0/src/SceneNode.cpp
--- libSavitar-4.1.0.old/src/SceneNode.cpp	2019-06-14 21:54:00.101539146 +0200
+++ libSavitar-4.1.0/src/SceneNode.cpp	2019-06-14 21:56:12.723351894 +0200
@@ -17,7 +17,7 @@
  */
 
 #include "SceneNode.h"
-#include "../pugixml/src/pugixml.hpp"
+#include <pugixml.hpp>
 #include <iostream>
 using namespace Savitar;
 
diff -aur libSavitar-4.1.0.old/src/ThreeMFParser.h libSavitar-4.1.0/src/ThreeMFParser.h
--- libSavitar-4.1.0.old/src/ThreeMFParser.h	2019-06-14 21:54:00.101539146 +0200
+++ libSavitar-4.1.0/src/ThreeMFParser.h	2019-06-14 21:55:32.532802504 +0200
@@ -21,7 +21,7 @@
 
 #include "SavitarExport.h"
 #include "SceneNode.h"
-#include "../pugixml/src/pugixml.hpp"
+#include <pugixml.hpp>
 
 #include <string>
 namespace Savitar
@@ -44,4 +44,4 @@
         std::string sceneToString(Scene scene);
     };
 }
-#endif
\ No newline at end of file
+#endif
