diff -wbBur pdf2djvu-0.7.12/configure pdf2djvu-0.7.12.my/configure
--- pdf2djvu-0.7.12/configure	2012-01-22 03:17:01.000000000 +0400
+++ pdf2djvu-0.7.12.my/configure	2012-05-17 20:44:06.000000000 +0400
@@ -6653,7 +6653,7 @@
 
 if test "$GXX" = yes
 then
-  CXXFLAGS="$CXXFLAGS -Werror=overloaded-virtual"
+  CXXFLAGS="$CXXFLAGS "
 fi
 
 # Output files
diff -wbBur pdf2djvu-0.7.12/configure.ac pdf2djvu-0.7.12.my/configure.ac
--- pdf2djvu-0.7.12/configure.ac	2012-01-22 03:16:50.000000000 +0400
+++ pdf2djvu-0.7.12.my/configure.ac	2012-05-17 20:43:37.000000000 +0400
@@ -225,7 +225,7 @@
 AX_CXXFLAGS_WARN_ALL(CXXFLAGS)
 if test "$GXX" = yes
 then
-  CXXFLAGS="$CXXFLAGS -Werror=overloaded-virtual"
+  CXXFLAGS="$CXXFLAGS "
 fi
 
 # Output files
diff -wbBur pdf2djvu-0.7.12/pdf-backend.cc pdf2djvu-0.7.12.my/pdf-backend.cc
--- pdf2djvu-0.7.12/pdf-backend.cc	2012-01-22 03:16:50.000000000 +0400
+++ pdf2djvu-0.7.12.my/pdf-backend.cc	2012-05-17 20:47:24.000000000 +0400
@@ -32,10 +32,10 @@
  * ======================
  */
 
-static void poppler_error_handler(int pos, char *message, va_list args)
+static void poppler_error_handler(void *data, ErrorCategory ec, int pos, char *message)
 {
   std::string format;
-  std::string expanded_message = string_vprintf(message, args);
+  std::string expanded_message = message;
   const char *c_message = expanded_message.c_str();
   if (pos >= 0)
   {
@@ -61,7 +61,7 @@
   Cwd cwd(argv0_dir_name);
 #endif
   globalParams = new GlobalParams();
-  setErrorFunction(poppler_error_handler);
+  setErrorCallback(poppler_error_handler, 0);
 }
 
 void pdf::Environment::set_antialias(bool value)
diff -wbBur pdf2djvu-0.7.12/pdf2djvu.cc pdf2djvu-0.7.12.my/pdf2djvu.cc
--- pdf2djvu-0.7.12/pdf2djvu.cc	2012-01-22 03:16:50.000000000 +0400
+++ pdf2djvu-0.7.12.my/pdf2djvu.cc	2012-05-17 20:55:07.000000000 +0400
@@ -1387,13 +1387,13 @@
         debug(0)++;
       }
       out1.reset(new pdf::Renderer(paper_color, config.monochrome));
-      out1->startDoc(doc->getXRef());
+      out1->startDoc((PDFDoc*)doc->getXRef());
       outm.reset(new MutedRenderer(paper_color, config.monochrome, *page_files));
-      outm->startDoc(doc->getXRef());
+      outm->startDoc((PDFDoc*)doc->getXRef());
       if (!config.monochrome)
       {
         outs.reset(new MutedRenderer(paper_color, config.monochrome, *page_files));
-        outs->startDoc(doc->getXRef());
+        outs->startDoc((PDFDoc*)doc->getXRef());
       }
     }
     assert(doc.get() != NULL);
