--- VirtualBox-6.0.14/src/VBox/Additions/common/crOpenGL/dri_glx.h.orig	2019-10-26 13:39:24.704688716 +0000
+++ VirtualBox-6.0.14/src/VBox/Additions/common/crOpenGL/dri_glx.h	2019-10-26 13:40:40.635814833 +0000
@@ -114,7 +114,7 @@
 extern DECLEXPORT(GLXPbufferSGIX) VBOXGLXTAG(glXCreateGLXPbufferSGIX)
 (Display *dpy, GLXFBConfigSGIX config, unsigned int width, unsigned int height, int *attrib_list);
 
-extern DECLEXPORT(int) VBOXGLXTAG(glXQueryGLXPbufferSGIX)
+extern DECLEXPORT(void) VBOXGLXTAG(glXQueryGLXPbufferSGIX)
 (Display *dpy, GLXPbuffer pbuf, int attribute, unsigned int *value);
 
 extern DECLEXPORT(GLXFBConfigSGIX *) VBOXGLXTAG(glXChooseFBConfigSGIX)
--- VirtualBox-6.0.14/src/VBox/Additions/common/crOpenGL/glx.c.orig     2019-10-26 13:47:03.041385389 +0000
+++ VirtualBox-6.0.14/src/VBox/Additions/common/crOpenGL/glx.c  2019-10-26 13:47:57.958842175 +0000
@@ -763,7 +763,7 @@
     (void) mask;
 }
 
-DECLEXPORT(int) VBOXGLXTAG(glXQueryGLXPbufferSGIX)(Display *dpy, GLXPbuffer pbuf,
+DECLEXPORT(void) VBOXGLXTAG(glXQueryGLXPbufferSGIX)(Display *dpy, GLXPbuffer pbuf,
                                                    int attribute, unsigned int *value)
 {
     (void) dpy;
@@ -771,7 +771,7 @@
     (void) attribute;
     (void) value;
     crWarning("glXQueryGLXPbufferSGIX not implemented by Chromium");
-    return 0;
+    return;
 }
 
 DECLEXPORT(int) VBOXGLXTAG(glXGetFBConfigAttribSGIX)(Display *dpy, GLXFBConfig config,

