--- configure.orig	2012-03-14 08:40:02.447527213 +0000
+++ configure	2012-03-14 09:05:01.535432453 +0000
@@ -1959,7 +1959,7 @@
           else
             cnf_append "VBOX_GSOAP_CXX_SOURCES" ""
           fi
-          cnf_append "VBOX_GSOAP_CXX_LIBS"    "libgsoap++"
+          cnf_append "VBOX_GSOAP_CXX_LIBS"    "libgsoapssl++ libz"
           log_success "found"
         else
           log_failure "stlvector.h not found -- disabling webservice"
