--- traitsui/toolkit.py.orig	2015-03-21 11:54:29.883107683 +0100
+++ traitsui/toolkit.py	2015-03-21 11:56:52.905373040 +0100
@@ -48,6 +48,9 @@
 #-------------------------------------------------------------------------------
 
 def _import_toolkit ( name ):
+    if name == "wx":
+        import wxversion
+        wxversion.select("2.8")
     return __import__( name, globals=globals(), level=1 ).toolkit
 
 
