diff -rupN firewalld-0.3.10.orig/config/firewalld.service.in firewalld-0.3.10/config/firewalld.service.in
--- firewalld-0.3.10.orig/config/firewalld.service.in	2014-06-10 11:07:27.864758482 +0200
+++ firewalld-0.3.10/config/firewalld.service.in	2014-06-10 11:07:39.582420959 +0200
@@ -6,7 +6,7 @@ Before=NetworkManager.service
 Conflicts=iptables.service ip6tables.service ebtables.service
 
 [Service]
-EnvironmentFile=-/etc/sysconfig/firewalld
+EnvironmentFile=-/etc/conf.d/firewalld
 ExecStart=@sbindir@/firewalld --nofork --nopid $FIREWALLD_ARGS
 ExecReload=/bin/kill -HUP $MAINPID
 # supress to log debug and error output also to /var/log/messages
diff -rupN firewalld-0.3.10.orig/config/Makefile.am firewalld-0.3.10/config/Makefile.am
--- firewalld-0.3.10.orig/config/Makefile.am	2014-06-10 11:07:27.851758856 +0200
+++ firewalld-0.3.10/config/Makefile.am	2014-06-10 11:07:39.580421016 +0200
@@ -164,12 +164,12 @@ firewalld.service: firewalld.service.in
 	$(edit) $< >$@
 
 install-sysconfig:
-	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/sysconfig
-	$(INSTALL_DATA) $(srcdir)/firewalld.sysconfig $(DESTDIR)$(sysconfdir)/sysconfig/firewalld
+	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/conf.d
+	$(INSTALL_DATA) $(srcdir)/firewalld.sysconfig $(DESTDIR)$(sysconfdir)/conf.d/firewalld
 
 uninstall-sysconfig:
-	rm -f $(DESTDIR)$(sysconfdir)/sysconfig/firewalld
-	rmdir $(DESTDIR)$(sysconfdir)/sysconfig || :
+	rm -f $(DESTDIR)$(sysconfdir)/conf.d/firewalld
+	rmdir $(DESTDIR)$(sysconfdir)/conf.d || :
 
 install-init: install-sysconfig
 	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/rc.d/init.d
diff -rupN firewalld-0.3.10.orig/config/Makefile.in firewalld-0.3.10/config/Makefile.in
--- firewalld-0.3.10.orig/config/Makefile.in	2014-06-10 11:07:27.782760844 +0200
+++ firewalld-0.3.10/config/Makefile.in	2014-06-10 11:07:39.581420988 +0200
@@ -761,12 +761,12 @@ firewalld.service: firewalld.service.in
 	$(edit) $< >$@
 
 install-sysconfig:
-	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/sysconfig
-	$(INSTALL_DATA) $(srcdir)/firewalld.sysconfig $(DESTDIR)$(sysconfdir)/sysconfig/firewalld
+	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/conf.d
+	$(INSTALL_DATA) $(srcdir)/firewalld.sysconfig $(DESTDIR)$(sysconfdir)/conf.d/firewalld
 
 uninstall-sysconfig:
-	rm -f $(DESTDIR)$(sysconfdir)/sysconfig/firewalld
-	rmdir $(DESTDIR)$(sysconfdir)/sysconfig || :
+	rm -f $(DESTDIR)$(sysconfdir)/conf.d/firewalld
+	rmdir $(DESTDIR)$(sysconfdir)/conf.d || :
 
 install-init: install-sysconfig
 	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/rc.d/init.d
diff -rupN firewalld-0.3.10.orig/src/firewall/core/ebtables.py firewalld-0.3.10/src/firewall/core/ebtables.py
--- firewalld-0.3.10.orig/src/firewall/core/ebtables.py	2014-06-10 11:07:27.340773575 +0200
+++ firewalld-0.3.10/src/firewall/core/ebtables.py	2014-06-10 11:07:39.585420873 +0200
@@ -37,7 +37,7 @@ class ebtables:
     ipv = "ipv4"
 
     def __init__(self):
-        self._command = "/sbin/ebtables"
+        self._command = "/usr/bin/ebtables"
 
     def __run(self, args):
         # convert to string list
diff -rupN firewalld-0.3.10.orig/src/firewall/core/ipXtables.py firewalld-0.3.10/src/firewall/core/ipXtables.py
--- firewalld-0.3.10.orig/src/firewall/core/ipXtables.py	2014-06-10 11:07:27.340773575 +0200
+++ firewalld-0.3.10/src/firewall/core/ipXtables.py	2014-06-10 11:07:39.586420844 +0200
@@ -25,8 +25,8 @@ from firewall.core.prog import runProg
 from firewall.core.logger import log
 
 COMMAND = {
-    "ipv4": "/sbin/iptables",
-    "ipv6": "/sbin/ip6tables",
+    "ipv4": "/usr/bin/iptables",
+    "ipv6": "/usr/bin/ip6tables",
 }
 
 PROC_IPxTABLE_NAMES = {
diff -rupN firewalld-0.3.10.orig/src/firewall-offline-cmd firewalld-0.3.10/src/firewall-offline-cmd
--- firewalld-0.3.10.orig/src/firewall-offline-cmd	2014-06-10 11:07:27.339773604 +0200
+++ firewalld-0.3.10/src/firewall-offline-cmd	2014-06-10 11:09:17.747594086 +0200
@@ -44,7 +44,7 @@ def __usage():
     print ("""
 Usage: firewall-offline-cmd [OPTIONS...]
 
-If no options are given, configuration from '/etc/sysconfig/system-config-firewall' will be migrated.
+If no options are given, configuration from '/etc/conf.d/system-config-firewall' will be migrated.
 
 General Options
   -h, --help           Prints a short help text and exists
@@ -70,7 +70,7 @@ Lokkit Compatibility Options
   --custom-rules=[<type>:][<table>:]<filename>
                         Ignored option. Was used to add custom rules to the
                         firewall (Example:
-                        ipv4:filter:/etc/sysconfig/ipv4_filter_addon)
+                        ipv4:filter:/etc/conf.d/ipv4_filter_addon)
   --forward-port=if=<interface>:port=<port>:proto=<protocol>[:toport=<destination port>][:toaddr=<destination address>]
                         Forward the port with protocol for the interface to
                         either another local destination port (no destination
@@ -376,7 +376,7 @@ def __pk_symlink(product='server'):
         __fail('no such file '+_PK_DIR+_PK_NAME+product+'.policy')
 
 # system-config-firewall: fw_sysconfig
-CONFIG = '/etc/sysconfig/system-config-firewall'
+CONFIG = '/etc/conf.d/system-config-firewall'
 def read_sysconfig_args():
     filename = None
     if os.path.exists(CONFIG) and os.path.isfile(CONFIG):
@@ -562,7 +562,7 @@ if len(sys.argv) > 1:
         args.append(joinArgs(aux_args[i+1:])) # add <args> as one arg
     a = parser.parse_args(args)
 else:
-    # migrate configuration from /etc/sysconfig/system-config-firewall
+    # migrate configuration from /etc/conf.d/system-config-firewall
     args = read_sysconfig_args()
     if args:
         a = parser.parse_args(args)
