--- Makefile.orig	2014-11-07 21:12:13.411106352 +0100
+++ Makefile	2014-11-07 21:12:21.854468444 +0100
@@ -29,7 +29,7 @@
 
 INSTALL_DIR = /usr/local
 
-CCFLAGS1 = -Wall -O2 -fomit-frame-pointer -DRATPOINTS_MAX_BITS_IN_PRIME=${PRIME_SIZE} -DUSE_SSE
+CCFLAGS1 = -fPIC -Wall -O2 -fomit-frame-pointer -DRATPOINTS_MAX_BITS_IN_PRIME=${PRIME_SIZE} -DUSE_SSE
 # for gcc on Apple, may have to add '-fnested-functions' to CCFLAGS1
 CCFLAGS2 = -lgmp -lgcc -lc -lm
 CCFLAGS3 = -L. -lratpoints
