From 6eabbd87690948f7aa489e28f2c216130218d465 Mon Sep 17 00:00:00 2001
From: Bernie Hackett <bernie@10gen.com>
Date: Thu, 25 Feb 2016 15:22:51 -0800
Subject: [PATCH] Don't free stack allocated buffer #15

---
 src/kerberosgss.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/kerberosgss.c b/src/kerberosgss.c
index 86788da..99436c8 100644
--- a/src/kerberosgss.c
+++ b/src/kerberosgss.c
@@ -603,7 +603,7 @@ int authenticate_gss_client_wrap(gss_client_state* state, const char* challenge,
 	if (output_token.value)
 		gss_release_buffer(&min_stat, &output_token);
 
-    if (input_token.value)
+    if (!user && input_token.value)
         gss_release_buffer(&min_stat, &input_token);
 
 	return ret;
