--- a/configure.ac	2021-05-14 16:58:39.897389483 +0200
+++ b/configure.ac	2021-05-14 16:58:51.890750976 +0200
@@ -72,17 +72,6 @@
 AC_MSG_RESULT([$enable_docs])
 AM_CONDITIONAL([DOCS], [test x${enable_docs} = xyes])

-dnl Enable Compiler Warnings
-AX_CFLAGS_WARN_ALL
-dnl Disable a warning with many false positives; perhaps rethink later
-AX_CHECK_COMPILE_FLAG([-Wno-write-strings],
-[CFLAGS="$CFLAGS -Wno-write-strings"
-CXXFLAGS="$CXXFLAGS -Wno-write-strings"])
-dnl Disable voluminous cloud of rather harmless warnings
-AX_CHECK_COMPILE_FLAG([-Wno-unused-variable],
-[CFLAGS="$CFLAGS -Wno-unused-variable"
-CXXFLAGS="$CXXFLAGS -Wno-unused-variable"])
-
 dnl Checks for libraries.
 AC_SEARCH_LIBS(sqrt,[m])
 AC_SEARCH_LIBS(atan,[m])
