diff --git a/doublex/matchers.py b/doublex/matchers.py
index e6ccf3c..2ecc987 100644
--- a/doublex/matchers.py
+++ b/doublex/matchers.py
@@ -97,7 +97,7 @@ class MethodCalled(OperationMatcher):
 
         if self._async_timeout:
             if self._times != any_time:
-                raise WrongApiUsage("'times' and 'async' are exclusive")
+                raise WrongApiUsage("'times' and 'wait' are exclusive")
             self.method._event.wait(self._async_timeout)
 
         return method._was_called(self.context, self._times)
@@ -126,7 +126,7 @@ class MethodCalled(OperationMatcher):
         self.context.check_some_args = True
         return self
 
-    def async(self, timeout):
+    def wait(self, timeout):
         self._async_timeout = timeout
         return self
 
diff --git a/doublex/test/async_race_condition_test.py b/doublex/test/async_race_condition_test.py
index c16e3a3..375a5cf 100644
--- a/doublex/test/async_race_condition_test.py
+++ b/doublex/test/async_race_condition_test.py
@@ -36,4 +36,4 @@ class AsyncTests(unittest.TestCase):
         sut.some_method()
 
         # then
-        assert_that(spy.write, called().async(1))
+        assert_that(spy.write, called().wait(1))
diff --git a/doublex/test/unit_tests.py b/doublex/test/unit_tests.py
index d1e1b6d..378fe94 100644
--- a/doublex/test/unit_tests.py
+++ b/doublex/test/unit_tests.py
@@ -1340,7 +1340,7 @@ class AsyncTests(TestCase):
         sut.send_data()
 
         # then
-        assert_that(spy.write, called().async(timeout=1))
+        assert_that(spy.write, called().wait(timeout=1))
 
     def test_spy_async_support_1_call_only(self):
         # given
@@ -1353,7 +1353,7 @@ class AsyncTests(TestCase):
 
         # then
         with self.assertRaises(WrongApiUsage):
-            assert_that(spy.write, called().async(timeout=1).with_args(3).times(2))
+            assert_that(spy.write, called().wait(timeout=1).with_args(3).times(2))
 
     def test_spy_async_stubbed(self):
         # given
@@ -1366,7 +1366,7 @@ class AsyncTests(TestCase):
         sut.send_data(3)
 
         # then
-        assert_that(spy.write, called().async(timeout=1))
+        assert_that(spy.write, called().wait(timeout=1))
 
 
 # new on 1.7
