From 85d38e50a9d2e5435f25a6915df5e34df3b2d2f6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 17 Apr 2019 22:08:06 +0200
Subject: [PATCH] Don't use legacy path for appdata file

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index bfb0cf5..4be0101 100755
--- a/setup.py
+++ b/setup.py
@@ -118,7 +118,7 @@ setup(name=PACKAGE_NAME,
 	data_files=[('share/mailnag', glob.glob('data/*.ui')),
 		('share/mailnag', ['data/mailnag.ogg']),
 		('share/mailnag', ['data/mailnag.png']),
-		('share/appdata', ['data/mailnag.appdata.xml']),
+		('share/metainfo', ['data/mailnag.appdata.xml']),
 		('share/applications', [os.path.join(BUILD_PATCH_DIR, 'mailnag.desktop'), os.path.join(BUILD_PATCH_DIR, 'mailnag-config.desktop')])],
 	cmdclass={'build': BuildData, 
 			'install_data': InstallData,
-- 
2.21.0

