From de5dd8f33dbc643d7e55087eef29f5a59b6dda1c Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Mon, 19 Jun 2017 13:26:47 +0100
Subject: [PATCH] trivial: Include the name in fwupd_device_to_string()

---
 libfwupd/fwupd-device.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libfwupd/fwupd-device.c b/libfwupd/fwupd-device.c
index af2af77..091879d 100644
--- a/libfwupd/fwupd-device.c
+++ b/libfwupd/fwupd-device.c
@@ -1011,6 +1011,7 @@ fwupd_device_to_string (FwupdDevice *device)
 		fwupd_pad_kv_str (str, FWUPD_RESULT_KEY_GUID, guid);
 	}
 	fwupd_pad_kv_str (str, FWUPD_RESULT_KEY_DEVICE_ID, priv->id);
+	fwupd_pad_kv_str (str, FWUPD_RESULT_KEY_DEVICE_NAME, priv->name);
 	fwupd_pad_kv_str (str, FWUPD_RESULT_KEY_DEVICE_DESCRIPTION, priv->description);
 	fwupd_pad_kv_str (str, FWUPD_RESULT_KEY_DEVICE_PLUGIN, priv->provider);
 	fwupd_pad_kv_dfl (str, FWUPD_RESULT_KEY_DEVICE_FLAGS, priv->flags);
