Thanks to Fernando Lopez-Lezcano <nando@ccrma.stanford.edu>

diff -rupN dssi-vst-0.8.old/dssi-vst_gui.cpp dssi-vst-0.8/dssi-vst_gui.cpp
--- dssi-vst-0.8.old/dssi-vst_gui.cpp	2008-12-12 07:34:31.000000000 -0500
+++ dssi-vst-0.8/dssi-vst_gui.cpp	2009-03-25 18:40:46.000000000 -0400
@@ -20,7 +20,7 @@
 #include <unistd.h>
 #include <time.h>
 #include <stdlib.h>
-
+#include <cstdio>
 #include <lo/lo.h>
 #include <lo/lo_lowlevel.h>

diff -rupN dssi-vst-0.8.old/rdwrops.cpp dssi-vst-0.8/rdwrops.cpp
--- dssi-vst-0.8.old/rdwrops.cpp	2008-12-12 07:34:31.000000000 -0500
+++ dssi-vst-0.8/rdwrops.cpp	2009-03-25 18:36:52.000000000 -0400
@@ -8,7 +8,7 @@
 */

 #include "rdwrops.h"
-
+#include <cstdio>
 #include <errno.h>
 #include <unistd.h>

diff -rupN dssi-vst-0.8.old/remotepluginclient.cpp dssi-vst-0.8/remotepluginclient.cpp
--- dssi-vst-0.8.old/remotepluginclient.cpp	2008-12-12 07:34:31.000000000 -0500
+++ dssi-vst-0.8/remotepluginclient.cpp	2009-03-25 18:37:50.000000000 -0400
@@ -18,7 +18,7 @@
 #include <errno.h>
 #include <stdlib.h>
 #include <string.h>
-
+#include <cstdio>
 #include "rdwrops.h"

 RemotePluginClient::RemotePluginClient() :
diff -rupN dssi-vst-0.8.old/remotepluginserver.cpp dssi-vst-0.8/remotepluginserver.cpp
--- dssi-vst-0.8.old/remotepluginserver.cpp	2008-12-12 07:34:31.000000000 -0500
+++ dssi-vst-0.8/remotepluginserver.cpp	2009-03-25 18:39:43.000000000 -0400
@@ -19,7 +19,7 @@
 #include <iostream>
 #include <stdlib.h>
 #include <string.h>
-
+#include <cstdio>
 #include "rdwrops.h"

 RemotePluginServer::RemotePluginServer(std::string fileIdentifiers) :
diff -rupN dssi-vst-0.8.old/remotevstclient.cpp dssi-vst-0.8/remotevstclient.cpp
--- dssi-vst-0.8.old/remotevstclient.cpp	2008-12-12 07:34:31.000000000 -0500
+++ dssi-vst-0.8/remotevstclient.cpp	2009-03-25 18:40:00.000000000 -0400
@@ -18,7 +18,7 @@
 #include <sys/wait.h>
 #include <errno.h>
 #include <stdlib.h>
-
+#include <cstdio>
 #include "rdwrops.h"
 #include "paths.h"

diff -rupN dssi-vst-0.8.old/vsthost.cpp dssi-vst-0.8/vsthost.cpp
--- dssi-vst-0.8.old/vsthost.cpp	2008-12-12 07:34:31.000000000 -0500
+++ dssi-vst-0.8/vsthost.cpp	2009-03-25 18:36:52.000000000 -0400
@@ -7,7 +7,7 @@

 #include <ctype.h>
 #include <signal.h>
-
+#include <cstdio>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/mman.h>
