diff -Naur cinnamon-control-center-1.7.1.orig/panels/user-accounts/um-photo-dialog.c cinnamon-control-center-1.7.1/panels/user-accounts/um-photo-dialog.c
--- cinnamon-control-center-1.7.1.orig/panels/user-accounts/um-photo-dialog.c	2013-02-19 17:37:17.000000000 +0100
+++ cinnamon-control-center-1.7.1/panels/user-accounts/um-photo-dialog.c	2013-02-27 19:17:18.249329833 +0100
@@ -406,7 +406,7 @@
         for (i = 0; dirs[i] != NULL; i++) {
                 char *path;
 
-                path = g_build_filename (dirs[i], "pixmaps", "faces", NULL);
+                path = g_build_filename (dirs[i], "pixmaps", "cinnamon", "faces", NULL);
                 dir = g_dir_open (path, 0, NULL);
                 if (dir == NULL) {
                         g_free (path);
