From 00984c339f524319b38786710528fecbeb1397a1 Mon Sep 17 00:00:00 2001
From: Larry Gritz <lg@larrygritz.com>
Date: Thu, 12 Sep 2019 13:11:27 -0700
Subject: [PATCH] LLVM 9 / clang 9 compatibility (#1058)

diff --git a/src/liboslexec/llvm_util.cpp b/src/liboslexec/llvm_util.cpp
index 1746ec628..838df1e21 100644
--- a/src/liboslexec/llvm_util.cpp
+++ b/src/liboslexec/llvm_util.cpp
@@ -682,7 +682,12 @@ LLVM_Util::make_function (const std::string &name, bool fastcall,
                           bool varargs)
 {
     llvm::FunctionType *functype = type_function (rettype, params, varargs);
+#if OSL_LLVM_VERSION >= 90
+    auto funccallee = module()->getOrInsertFunction(name, functype);
+    llvm::Value* c = funccallee.getCallee();
+#else
     llvm::Constant *c = module()->getOrInsertFunction (name, functype);
+#endif
     ASSERT (c && "getOrInsertFunction returned NULL");
     ASSERT_MSG (llvm::isa<llvm::Function>(c),
                 "Declaration for %s is wrong, LLVM had to make a cast", name.c_str());
