--- audex.orig/utils/encoderassistant.h	2010-09-09 15:41:30.000000000 +0200
+++ audex/utils/encoderassistant.h	2010-09-09 15:42:08.000000000 +0200
@@ -91,7 +91,7 @@
 
 #define ENCODER_OGGENC_NAME		i18n("Ogg Vorbis")
 #define ENCODER_OGGENC_BIN		"oggenc"
-#define ENCODER_OGGENC_VERSION_PARA	"-v"
+#define ENCODER_OGGENC_VERSION_PARA	"-V"
 #define ENCODER_OGGENC_SUFFIX		"ogg"
 
 /* preset normal quality */
