diff -ruN a/extras/Projucer/Source/Settings/jucer_StoredSettings.cpp b/extras/Projucer/Source/Settings/jucer_StoredSettings.cpp
--- a/extras/Projucer/Source/Settings/jucer_StoredSettings.cpp	2021-09-20 14:26:04.000000000 +0200
+++ b/extras/Projucer/Source/Settings/jucer_StoredSettings.cpp	2021-09-26 00:35:56.119917679 +0200
@@ -358,15 +358,15 @@
 {
     if (key == Ids::jucePath)
     {
-        return (os == TargetOS::windows ? "C:\\JUCE" : "~/JUCE");
+        return (os == TargetOS::windows ? "C:\\JUCE" : "/usr/share/doc/juce");
     }
     else if (key == Ids::defaultJuceModulePath)
     {
-        return (os == TargetOS::windows ? "C:\\JUCE\\modules" : "~/JUCE/modules");
+        return (os == TargetOS::windows ? "C:\\JUCE\\modules" : "/usr/share/juce/modules");
     }
     else if (key == Ids::defaultUserModulePath)
     {
-        return (os == TargetOS::windows ? "C:\\modules" : "~/modules");
+        return (os == TargetOS::windows ? "C:\\modules" : "~/.local/share/juce/modules");
     }
     else if (key == Ids::vstLegacyPath)
     {
