diff -wbBur FreeMat-4.0.1-Source/libs/libCore/helpwidget.moc.cpp FreeMat-4.0.1-Source.new/libs/libCore/helpwidget.moc.cpp
--- FreeMat-4.0.1-Source/libs/libCore/helpwidget.moc.cpp	2009-10-09 06:23:29.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libCore/helpwidget.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'helpwidget.hpp'
 **
-** Created: Thu Oct 8 21:52:25 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "helpwidget.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'helpwidget.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_HelpSearcher[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       1,   12, // methods
+       1,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       14,   13,   13,   13, 0x08,
@@ -43,9 +45,13 @@
       qt_meta_data_HelpSearcher, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &HelpSearcher::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *HelpSearcher::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *HelpSearcher::qt_metacast(const char *_clname)
@@ -73,13 +79,15 @@
 static const uint qt_meta_data_HelpWidget[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
        0,    0, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
        0        // eod
 };
@@ -93,9 +101,13 @@
       qt_meta_data_HelpWidget, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &HelpWidget::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *HelpWidget::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *HelpWidget::qt_metacast(const char *_clname)
@@ -116,13 +128,15 @@
 static const uint qt_meta_data_HelpWindow[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       6,   12, // methods
+       6,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       1,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       12,   11,   11,   11, 0x05,
@@ -150,9 +164,13 @@
       qt_meta_data_HelpWindow, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &HelpWindow::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *HelpWindow::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *HelpWindow::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libCore/XML.moc.cpp FreeMat-4.0.1-Source.new/libs/libCore/XML.moc.cpp
--- FreeMat-4.0.1-Source/libs/libCore/XML.moc.cpp	2009-10-09 06:23:28.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libCore/XML.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'XML.hpp'
 **
-** Created: Thu Oct 8 21:52:25 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "XML.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'XML.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_URLRetriever[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       1,   12, // methods
+       1,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       21,   14,   13,   13, 0x0a,
@@ -43,9 +45,13 @@
       qt_meta_data_URLRetriever, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &URLRetriever::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *URLRetriever::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *URLRetriever::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libFreeMat/DebugStream.moc.cpp FreeMat-4.0.1-Source.new/libs/libFreeMat/DebugStream.moc.cpp
--- FreeMat-4.0.1-Source/libs/libFreeMat/DebugStream.moc.cpp	2009-10-09 06:24:22.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libFreeMat/DebugStream.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'DebugStream.hpp'
 **
-** Created: Thu Oct 8 21:51:54 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "DebugStream.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'DebugStream.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_DebugStream[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       1,   12, // methods
+       1,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       1,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       17,   13,   12,   12, 0x05,
@@ -43,9 +45,13 @@
       qt_meta_data_DebugStream, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &DebugStream::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *DebugStream::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *DebugStream::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libFreeMat/Intrepreter.moc.cpp FreeMat-4.0.1-Source.new/libs/libFreeMat/Intrepreter.moc.cpp
--- FreeMat-4.0.1-Source/libs/libFreeMat/Intrepreter.moc.cpp	2009-10-09 06:24:21.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libFreeMat/Intrepreter.moc.cpp	2010-03-03 13:24:53.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'Interpreter.hpp'
 **
-** Created: Thu Oct 8 21:51:55 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:24:53 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "Interpreter.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'Interpreter.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_Interpreter[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-      20,   12, // methods
+      20,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+      16,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       13,   12,   12,   12, 0x05,
@@ -77,9 +79,13 @@
       qt_meta_data_Interpreter, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &Interpreter::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *Interpreter::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *Interpreter::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libGraphics/GLWindow.moc.cpp FreeMat-4.0.1-Source.new/libs/libGraphics/GLWindow.moc.cpp
--- FreeMat-4.0.1-Source/libs/libGraphics/GLWindow.moc.cpp	2009-10-09 06:24:11.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libGraphics/GLWindow.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'GLWindow.hpp'
 **
-** Created: Thu Oct 8 21:52:04 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "GLWindow.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'GLWindow.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_GLWidget[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
        0,    0, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
        0        // eod
 };
@@ -40,9 +42,13 @@
       qt_meta_data_GLWidget, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &GLWidget::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *GLWidget::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *GLWidget::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libGraphics/HandleMultiEdit.moc.cpp FreeMat-4.0.1-Source.new/libs/libGraphics/HandleMultiEdit.moc.cpp
--- FreeMat-4.0.1-Source/libs/libGraphics/HandleMultiEdit.moc.cpp	2009-10-09 06:24:11.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libGraphics/HandleMultiEdit.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'HandleMultiEdit.hpp'
 **
-** Created: Thu Oct 8 21:52:05 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "HandleMultiEdit.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'HandleMultiEdit.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_HandleMultiEdit[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       1,   12, // methods
+       1,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       1,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       17,   16,   16,   16, 0x05,
@@ -43,9 +45,13 @@
       qt_meta_data_HandleMultiEdit, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &HandleMultiEdit::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *HandleMultiEdit::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *HandleMultiEdit::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libGraphics/HandleUIControl.moc.cpp FreeMat-4.0.1-Source.new/libs/libGraphics/HandleUIControl.moc.cpp
--- FreeMat-4.0.1-Source/libs/libGraphics/HandleUIControl.moc.cpp	2009-10-09 06:24:11.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libGraphics/HandleUIControl.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'HandleUIControl.hpp'
 **
-** Created: Thu Oct 8 21:52:05 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "HandleUIControl.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'HandleUIControl.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_HandleUIControl[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       1,   12, // methods
+       1,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       17,   16,   16,   16, 0x08,
@@ -43,9 +45,13 @@
       qt_meta_data_HandleUIControl, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &HandleUIControl::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *HandleUIControl::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *HandleUIControl::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libGraphics/HandleWindow.moc.cpp FreeMat-4.0.1-Source.new/libs/libGraphics/HandleWindow.moc.cpp
--- FreeMat-4.0.1-Source/libs/libGraphics/HandleWindow.moc.cpp	2009-10-09 06:24:11.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libGraphics/HandleWindow.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'HandleWindow.hpp'
 **
-** Created: Thu Oct 8 21:52:05 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "HandleWindow.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'HandleWindow.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_HandleWindow[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       6,   12, // methods
+       6,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       14,   13,   13,   13, 0x0a,
@@ -50,9 +52,13 @@
       qt_meta_data_HandleWindow, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &HandleWindow::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *HandleWindow::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *HandleWindow::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/DataTable.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/DataTable.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/DataTable.moc.cpp	2009-10-09 06:24:19.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/DataTable.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'DataTable.hpp'
 **
-** Created: Thu Oct 8 21:52:15 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "DataTable.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'DataTable.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_DataTable[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
        0,    0, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
        0        // eod
 };
@@ -40,9 +42,13 @@
       qt_meta_data_DataTable, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &DataTable::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *DataTable::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *DataTable::qt_metacast(const char *_clname)
@@ -63,13 +69,15 @@
 static const uint qt_meta_data_DataView[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
        0,    0, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
        0        // eod
 };
@@ -83,9 +91,13 @@
       qt_meta_data_DataView, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &DataView::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *DataView::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *DataView::qt_metacast(const char *_clname)
@@ -106,13 +118,15 @@
 static const uint qt_meta_data_DataHeaderView[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       1,   12, // methods
+       1,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       16,   15,   15,   15, 0x08,
@@ -129,9 +143,13 @@
       qt_meta_data_DataHeaderView, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &DataHeaderView::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *DataHeaderView::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *DataHeaderView::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/DebugWin.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/DebugWin.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/DebugWin.moc.cpp	2009-10-09 06:24:18.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/DebugWin.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'DebugWin.hpp'
 **
-** Created: Thu Oct 8 21:52:15 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "DebugWin.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'DebugWin.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_DebugWin[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       3,   12, // methods
+       3,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       12,   10,    9,    9, 0x09,
@@ -46,9 +48,13 @@
       qt_meta_data_DebugWin, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &DebugWin::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *DebugWin::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *DebugWin::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/Editor.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/Editor.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/Editor.moc.cpp	2009-10-09 06:24:12.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/Editor.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'Editor.hpp'
 **
-** Created: Thu Oct 8 21:52:14 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "Editor.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'Editor.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_FMFindDialog[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       2,   12, // methods
+       2,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       1,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       39,   14,   13,   13, 0x05,
@@ -47,9 +49,13 @@
       qt_meta_data_FMFindDialog, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FMFindDialog::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FMFindDialog::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FMFindDialog::qt_metacast(const char *_clname)
@@ -85,13 +91,15 @@
 static const uint qt_meta_data_FMReplaceDialog[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       6,   12, // methods
+       6,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       3,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       42,   17,   16,   16, 0x05,
@@ -120,9 +128,13 @@
       qt_meta_data_FMReplaceDialog, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FMReplaceDialog::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FMReplaceDialog::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FMReplaceDialog::qt_metacast(const char *_clname)
@@ -176,13 +188,15 @@
 static const uint qt_meta_data_FMTextEdit[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       6,   12, // methods
+       6,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       3,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       12,   11,   11,   11, 0x05,
@@ -210,9 +224,13 @@
       qt_meta_data_FMTextEdit, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FMTextEdit::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FMTextEdit::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FMTextEdit::qt_metacast(const char *_clname)
@@ -264,13 +282,15 @@
 static const uint qt_meta_data_FMIndent[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       2,   12, // methods
+       2,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       10,    9,    9,    9, 0x08,
@@ -288,9 +308,13 @@
       qt_meta_data_FMIndent, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FMIndent::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FMIndent::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FMIndent::qt_metacast(const char *_clname)
@@ -319,13 +343,15 @@
 static const uint qt_meta_data_BreakPointIndicator[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
        0,    0, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
        0        // eod
 };
@@ -339,9 +365,13 @@
       qt_meta_data_BreakPointIndicator, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &BreakPointIndicator::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *BreakPointIndicator::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *BreakPointIndicator::qt_metacast(const char *_clname)
@@ -362,13 +392,15 @@
 static const uint qt_meta_data_FMLineNumber[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
        0,    0, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
        0        // eod
 };
@@ -382,9 +414,13 @@
       qt_meta_data_FMLineNumber, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FMLineNumber::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FMLineNumber::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FMLineNumber::qt_metacast(const char *_clname)
@@ -405,13 +441,15 @@
 static const uint qt_meta_data_FMEditPane[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       1,   12, // methods
+       1,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       1,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       23,   12,   11,   11, 0x05,
@@ -428,9 +466,13 @@
       qt_meta_data_FMEditPane, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FMEditPane::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FMEditPane::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FMEditPane::qt_metacast(const char *_clname)
@@ -465,13 +507,15 @@
 static const uint qt_meta_data_FMIndentConf[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       1,   12, // methods
+       1,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       14,   13,   13,   13, 0x08,
@@ -488,9 +532,13 @@
       qt_meta_data_FMIndentConf, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FMIndentConf::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FMIndentConf::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FMIndentConf::qt_metacast(const char *_clname)
@@ -518,13 +566,15 @@
 static const uint qt_meta_data_FMSynLightConf[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       5,   12, // methods
+       5,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       16,   15,   15,   15, 0x08,
@@ -547,9 +597,13 @@
       qt_meta_data_FMSynLightConf, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FMSynLightConf::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FMSynLightConf::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FMSynLightConf::qt_metacast(const char *_clname)
@@ -581,13 +635,15 @@
 static const uint qt_meta_data_FMEditor[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-      46,   12, // methods
+      46,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       3,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       10,    9,    9,    9, 0x05,
@@ -675,9 +731,13 @@
       qt_meta_data_FMEditor, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FMEditor::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FMEditor::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FMEditor::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/FileTool.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/FileTool.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/FileTool.moc.cpp	2009-10-09 06:24:20.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/FileTool.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'FileTool.hpp'
 **
-** Created: Thu Oct 8 21:52:15 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "FileTool.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'FileTool.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_FileTool[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       3,   12, // methods
+       3,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       1,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       12,   10,    9,    9, 0x05,
@@ -49,9 +51,13 @@
       qt_meta_data_FileTool, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FileTool::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FileTool::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FileTool::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/findform.ui.h FreeMat-4.0.1-Source.new/libs/libXP/findform.ui.h
--- FreeMat-4.0.1-Source/libs/libXP/findform.ui.h	2009-10-09 06:24:12.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/findform.ui.h	2010-03-03 13:17:50.000000000 +0000
@@ -1,10 +1,10 @@
 /********************************************************************************
-** Form generated from reading ui file 'findform.ui'
+** Form generated from reading UI file 'findform.ui'
 **
-** Created: Thu Oct 8 21:52:13 2009
-**      by: Qt User Interface Compiler version 4.5.2
+** Created: Wed Mar 3 13:17:50 2010
+**      by: Qt User Interface Compiler version 4.6.2
 **
-** WARNING! All changes made in this file will be lost when recompiling ui file!
+** WARNING! All changes made in this file will be lost when recompiling UI file!
 ********************************************************************************/
 
 #ifndef FINDFORM_H
@@ -69,7 +69,7 @@
 #ifndef Q_OS_MAC
         vboxLayout->setSpacing(6);
 #endif
-        vboxLayout->setMargin(0);
+        vboxLayout->setContentsMargins(0, 0, 0, 0);
         vboxLayout->setObjectName(QString::fromUtf8("vboxLayout"));
         vboxLayout->setContentsMargins(0, 0, 0, 0);
         hboxLayout = new QHBoxLayout();
@@ -77,7 +77,7 @@
         hboxLayout->setSpacing(6);
 #endif
 #ifndef Q_OS_MAC
-        hboxLayout->setMargin(0);
+        hboxLayout->setContentsMargins(0, 0, 0, 0);
 #endif
         hboxLayout->setObjectName(QString::fromUtf8("hboxLayout"));
         label = new QLabel(layoutWidget);
@@ -119,7 +119,7 @@
 #ifndef Q_OS_MAC
         vboxLayout1->setSpacing(6);
 #endif
-        vboxLayout1->setMargin(0);
+        vboxLayout1->setContentsMargins(0, 0, 0, 0);
         vboxLayout1->setObjectName(QString::fromUtf8("vboxLayout1"));
         vboxLayout1->setContentsMargins(0, 0, 0, 0);
         cbSensitive = new QCheckBox(layoutWidget1);
@@ -149,7 +149,7 @@
 #ifndef Q_OS_MAC
         vboxLayout2->setSpacing(6);
 #endif
-        vboxLayout2->setMargin(0);
+        vboxLayout2->setContentsMargins(0, 0, 0, 0);
         vboxLayout2->setObjectName(QString::fromUtf8("vboxLayout2"));
         vboxLayout2->setContentsMargins(0, 0, 0, 0);
         btFind = new QPushButton(layoutWidget2);
@@ -191,7 +191,6 @@
         lbStatus->setText(QString());
         btFind->setText(QApplication::translate("FMFindDialog", "Find", 0, QApplication::UnicodeUTF8));
         btClose->setText(QApplication::translate("FMFindDialog", "Close", 0, QApplication::UnicodeUTF8));
-        Q_UNUSED(FMFindDialog);
     } // retranslateUi
 
 };
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/FMFontDialog.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/FMFontDialog.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/FMFontDialog.moc.cpp	2009-10-09 06:24:20.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/FMFontDialog.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'FMFontDialog.hpp'
 **
-** Created: Thu Oct 8 21:52:14 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "FMFontDialog.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'FMFontDialog.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_FMFontDialog[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       3,   12, // methods
+       3,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       19,   14,   13,   13, 0x08,
@@ -46,9 +48,13 @@
       qt_meta_data_FMFontDialog, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FMFontDialog::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FMFontDialog::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FMFontDialog::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/fontform.ui.h FreeMat-4.0.1-Source.new/libs/libXP/fontform.ui.h
--- FreeMat-4.0.1-Source/libs/libXP/fontform.ui.h	2009-10-09 06:24:19.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/fontform.ui.h	2010-03-03 13:17:50.000000000 +0000
@@ -1,10 +1,10 @@
 /********************************************************************************
-** Form generated from reading ui file 'fontform.ui'
+** Form generated from reading UI file 'fontform.ui'
 **
-** Created: Thu Oct 8 21:52:13 2009
-**      by: Qt User Interface Compiler version 4.5.2
+** Created: Wed Mar 3 13:17:50 2010
+**      by: Qt User Interface Compiler version 4.6.2
 **
-** WARNING! All changes made in this file will be lost when recompiling ui file!
+** WARNING! All changes made in this file will be lost when recompiling UI file!
 ********************************************************************************/
 
 #ifndef FONTFORM_H
@@ -80,7 +80,6 @@
         label_2->setText(QApplication::translate("FixedFontDialog", "Size", 0, QApplication::UnicodeUTF8));
         groupBox->setTitle(QApplication::translate("FixedFontDialog", "Sample", 0, QApplication::UnicodeUTF8));
         sample->setText(QApplication::translate("FixedFontDialog", "AaBbZz123", 0, QApplication::UnicodeUTF8));
-        Q_UNUSED(FixedFontDialog);
     } // retranslateUi
 
 };
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/highlighter.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/highlighter.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/highlighter.moc.cpp	2009-10-09 06:24:19.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/highlighter.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'highlighter.hpp'
 **
-** Created: Thu Oct 8 21:52:15 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "highlighter.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'highlighter.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_Highlighter[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
        0,    0, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
        0        // eod
 };
@@ -40,9 +42,13 @@
       qt_meta_data_Highlighter, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &Highlighter::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *Highlighter::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *Highlighter::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/HistoryWidget.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/HistoryWidget.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/HistoryWidget.moc.cpp	2009-10-09 06:24:20.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/HistoryWidget.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'HistoryWidget.hpp'
 **
-** Created: Thu Oct 8 21:52:15 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "HistoryWidget.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'HistoryWidget.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_HistoryWidget[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       8,   12, // methods
+       8,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       3,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       17,   15,   14,   14, 0x05,
@@ -56,9 +58,13 @@
       qt_meta_data_HistoryWidget, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &HistoryWidget::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *HistoryWidget::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *HistoryWidget::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/KeyManager.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/KeyManager.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/KeyManager.moc.cpp	2009-10-09 06:24:12.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/KeyManager.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'KeyManager.hpp'
 **
-** Created: Thu Oct 8 21:52:14 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "KeyManager.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'KeyManager.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_KeyManager[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-      39,   12, // methods
+      39,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+      21,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       12,   11,   11,   11, 0x05,
@@ -101,9 +103,13 @@
       qt_meta_data_KeyManager, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &KeyManager::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *KeyManager::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *KeyManager::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/PathTool.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/PathTool.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/PathTool.moc.cpp	2009-10-09 06:24:12.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/PathTool.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'PathTool.hpp'
 **
-** Created: Thu Oct 8 21:52:14 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "PathTool.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'PathTool.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_PathTool[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       7,   12, // methods
+       7,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       10,    9,    9,    9, 0x08,
@@ -50,9 +52,13 @@
       qt_meta_data_PathTool, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &PathTool::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *PathTool::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *PathTool::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/QTTerm.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/QTTerm.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/QTTerm.moc.cpp	2009-10-09 06:24:18.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/QTTerm.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'QTTerm.hpp'
 **
-** Created: Thu Oct 8 21:52:14 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "QTTerm.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'QTTerm.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_QTTerm[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-      15,   12, // methods
+      15,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       4,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       10,    8,    7,    7, 0x05,
@@ -65,9 +67,13 @@
       qt_meta_data_QTTerm, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &QTTerm::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *QTTerm::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *QTTerm::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/replaceform.ui.h FreeMat-4.0.1-Source.new/libs/libXP/replaceform.ui.h
--- FreeMat-4.0.1-Source/libs/libXP/replaceform.ui.h	2009-10-09 06:24:12.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/replaceform.ui.h	2010-03-03 13:17:50.000000000 +0000
@@ -1,10 +1,10 @@
 /********************************************************************************
-** Form generated from reading ui file 'replaceform.ui'
+** Form generated from reading UI file 'replaceform.ui'
 **
-** Created: Thu Oct 8 21:52:13 2009
-**      by: Qt User Interface Compiler version 4.5.2
+** Created: Wed Mar 3 13:17:50 2010
+**      by: Qt User Interface Compiler version 4.6.2
 **
-** WARNING! All changes made in this file will be lost when recompiling ui file!
+** WARNING! All changes made in this file will be lost when recompiling UI file!
 ********************************************************************************/
 
 #ifndef REPLACEFORM_H
@@ -74,7 +74,7 @@
 #ifndef Q_OS_MAC
         vboxLayout->setSpacing(6);
 #endif
-        vboxLayout->setMargin(0);
+        vboxLayout->setContentsMargins(0, 0, 0, 0);
         vboxLayout->setObjectName(QString::fromUtf8("vboxLayout"));
         vboxLayout->setContentsMargins(0, 0, 0, 0);
         hboxLayout = new QHBoxLayout();
@@ -82,7 +82,7 @@
         hboxLayout->setSpacing(6);
 #endif
 #ifndef Q_OS_MAC
-        hboxLayout->setMargin(0);
+        hboxLayout->setContentsMargins(0, 0, 0, 0);
 #endif
         hboxLayout->setObjectName(QString::fromUtf8("hboxLayout"));
         label = new QLabel(layoutWidget);
@@ -113,7 +113,7 @@
 #ifndef Q_OS_MAC
         hboxLayout1->setSpacing(6);
 #endif
-        hboxLayout1->setMargin(0);
+        hboxLayout1->setContentsMargins(0, 0, 0, 0);
         hboxLayout1->setObjectName(QString::fromUtf8("hboxLayout1"));
         label_2 = new QLabel(layoutWidget);
         label_2->setObjectName(QString::fromUtf8("label_2"));
@@ -146,7 +146,7 @@
 #ifndef Q_OS_MAC
         vboxLayout1->setSpacing(6);
 #endif
-        vboxLayout1->setMargin(0);
+        vboxLayout1->setContentsMargins(0, 0, 0, 0);
         vboxLayout1->setObjectName(QString::fromUtf8("vboxLayout1"));
         vboxLayout1->setContentsMargins(0, 0, 0, 0);
         cbSensitive = new QCheckBox(layoutWidget1);
@@ -176,7 +176,7 @@
 #ifndef Q_OS_MAC
         vboxLayout2->setSpacing(6);
 #endif
-        vboxLayout2->setMargin(0);
+        vboxLayout2->setContentsMargins(0, 0, 0, 0);
         vboxLayout2->setObjectName(QString::fromUtf8("vboxLayout2"));
         vboxLayout2->setContentsMargins(0, 0, 0, 0);
         btReplace = new QPushButton(layoutWidget2);
@@ -234,7 +234,6 @@
         btReplaceAll->setText(QApplication::translate("FMReplaceDialog", "Replace All", 0, QApplication::UnicodeUTF8));
         btFind->setText(QApplication::translate("FMReplaceDialog", "Find Next", 0, QApplication::UnicodeUTF8));
         btClose->setText(QApplication::translate("FMReplaceDialog", "Close", 0, QApplication::UnicodeUTF8));
-        Q_UNUSED(FMReplaceDialog);
     } // retranslateUi
 
 };
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/StackTool.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/StackTool.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/StackTool.moc.cpp	2009-10-09 06:24:19.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/StackTool.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'StackTool.hpp'
 **
-** Created: Thu Oct 8 21:52:13 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "StackTool.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'StackTool.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_StackTool[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       3,   12, // methods
+       3,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       1,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       11,   10,   10,   10, 0x05,
@@ -49,9 +51,13 @@
       qt_meta_data_StackTool, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &StackTool::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *StackTool::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *StackTool::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/ToolDock.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/ToolDock.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/ToolDock.moc.cpp	2009-10-09 06:24:19.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/ToolDock.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'ToolDock.hpp'
 **
-** Created: Thu Oct 8 21:52:15 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "ToolDock.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'ToolDock.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_ToolDock[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
        0,    0, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
        0        // eod
 };
@@ -40,9 +42,13 @@
       qt_meta_data_ToolDock, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &ToolDock::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *ToolDock::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *ToolDock::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/libs/libXP/VariablesTool.moc.cpp FreeMat-4.0.1-Source.new/libs/libXP/VariablesTool.moc.cpp
--- FreeMat-4.0.1-Source/libs/libXP/VariablesTool.moc.cpp	2009-10-09 06:24:20.000000000 +0000
+++ FreeMat-4.0.1-Source.new/libs/libXP/VariablesTool.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'VariablesTool.hpp'
 **
-** Created: Thu Oct 8 21:52:15 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "VariablesTool.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'VariablesTool.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_VariablesTool[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       1,   12, // methods
+       1,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       0,       // signalCount
 
  // slots: signature, parameters, type, tag, flags
       15,   14,   14,   14, 0x09,
@@ -43,9 +45,13 @@
       qt_meta_data_VariablesTool, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &VariablesTool::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *VariablesTool::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *VariablesTool::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/src/application.moc.cpp FreeMat-4.0.1-Source.new/src/application.moc.cpp
--- FreeMat-4.0.1-Source/src/application.moc.cpp	2009-10-09 06:27:01.000000000 +0000
+++ FreeMat-4.0.1-Source.new/src/application.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'application.hpp'
 **
-** Created: Thu Oct 8 21:53:56 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "application.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'application.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_ApplicationWindow[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-      31,   12, // methods
+      31,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       6,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       19,   18,   18,   18, 0x05,
@@ -86,9 +88,13 @@
       qt_meta_data_ApplicationWindow, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &ApplicationWindow::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *ApplicationWindow::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *ApplicationWindow::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/src/DumbTerminal.moc.cpp FreeMat-4.0.1-Source.new/src/DumbTerminal.moc.cpp
--- FreeMat-4.0.1-Source/src/DumbTerminal.moc.cpp	2009-10-09 06:27:01.000000000 +0000
+++ FreeMat-4.0.1-Source.new/src/DumbTerminal.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'DumbTerminal.hpp'
 **
-** Created: Thu Oct 8 21:53:56 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "DumbTerminal.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'DumbTerminal.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_DumbTerminal[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-      13,   12, // methods
+      13,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       3,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       16,   14,   13,   13, 0x05,
@@ -63,9 +65,13 @@
       qt_meta_data_DumbTerminal, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &DumbTerminal::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *DumbTerminal::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *DumbTerminal::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/src/FuncMode.moc.cpp FreeMat-4.0.1-Source.new/src/FuncMode.moc.cpp
--- FreeMat-4.0.1-Source/src/FuncMode.moc.cpp	2009-10-09 06:27:01.000000000 +0000
+++ FreeMat-4.0.1-Source.new/src/FuncMode.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'FuncMode.hpp'
 **
-** Created: Thu Oct 8 21:53:56 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "FuncMode.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'FuncMode.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_FuncMode[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-       2,   12, // methods
+       2,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       1,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       10,    9,    9,    9, 0x05,
@@ -46,9 +48,13 @@
       qt_meta_data_FuncMode, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &FuncMode::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *FuncMode::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *FuncMode::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/src/MainApp.moc.cpp FreeMat-4.0.1-Source.new/src/MainApp.moc.cpp
--- FreeMat-4.0.1-Source/src/MainApp.moc.cpp	2009-10-09 06:27:01.000000000 +0000
+++ FreeMat-4.0.1-Source.new/src/MainApp.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'MainApp.hpp'
 **
-** Created: Thu Oct 8 21:53:56 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "MainApp.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'MainApp.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_MainApp[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-      21,   12, // methods
+      21,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       3,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
        9,    8,    8,    8, 0x05,
@@ -75,9 +77,13 @@
       qt_meta_data_MainApp, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &MainApp::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *MainApp::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *MainApp::qt_metacast(const char *_clname)
diff -wbBur FreeMat-4.0.1-Source/src/Terminal.moc.cpp FreeMat-4.0.1-Source.new/src/Terminal.moc.cpp
--- FreeMat-4.0.1-Source/src/Terminal.moc.cpp	2009-10-09 06:27:01.000000000 +0000
+++ FreeMat-4.0.1-Source.new/src/Terminal.moc.cpp	2010-03-03 13:22:19.000000000 +0000
@@ -1,8 +1,8 @@
 /****************************************************************************
 ** Meta object code from reading C++ file 'Terminal.hpp'
 **
-** Created: Thu Oct 8 21:53:56 2009
-**      by: The Qt Meta Object Compiler version 61 (Qt 4.5.2)
+** Created: Wed Mar 3 13:22:19 2010
+**      by: The Qt Meta Object Compiler version 62 (Qt 4.6.2)
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
@@ -10,8 +10,8 @@
 #include "Terminal.hpp"
 #if !defined(Q_MOC_OUTPUT_REVISION)
 #error "The header file 'Terminal.hpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 61
-#error "This file was generated using the moc from 4.5.2. It"
+#elif Q_MOC_OUTPUT_REVISION != 62
+#error "This file was generated using the moc from 4.6.2. It"
 #error "cannot be used with the include files from this version of Qt."
 #error "(The moc has changed too much.)"
 #endif
@@ -20,13 +20,15 @@
 static const uint qt_meta_data_Terminal[] = {
 
  // content:
-       2,       // revision
+       4,       // revision
        0,       // classname
        0,    0, // classinfo
-      13,   12, // methods
+      13,   14, // methods
        0,    0, // properties
        0,    0, // enums/sets
        0,    0, // constructors
+       0,       // flags
+       3,       // signalCount
 
  // signals: signature, parameters, type, tag, flags
       12,   10,    9,    9, 0x05,
@@ -62,9 +64,13 @@
       qt_meta_data_Terminal, 0 }
 };
 
+#ifdef Q_NO_DATA_RELOCATION
+const QMetaObject &Terminal::getStaticMetaObject() { return staticMetaObject; }
+#endif //Q_NO_DATA_RELOCATION
+
 const QMetaObject *Terminal::metaObject() const
 {
-    return &staticMetaObject;
+    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
 }
 
 void *Terminal::qt_metacast(const char *_clname)
