diff -rupN a/Source/NSBitmapImageRep+GIF.m b/Source/NSBitmapImageRep+GIF.m
--- a/Source/NSBitmapImageRep+GIF.m	2013-01-13 14:57:37.000000000 +0100
+++ b/Source/NSBitmapImageRep+GIF.m	2013-07-30 11:30:19.630935640 +0200
@@ -142,7 +142,7 @@ static int gs_gif_output(GifFileType *fi
     }
 
   gs_gif_init_input_source(&src, imageData);
-  file = DGifOpen(&src, gs_gif_input);
+  file = DGifOpen(&src, gs_gif_input, NULL);
   if (file == NULL)
     {
       /* we do not use giferror here because it doesn't
@@ -216,7 +216,7 @@ static int gs_gif_output(GifFileType *fi
 
   /* open the image */
   gs_gif_init_input_source(&src, imageData);
-  file = DGifOpen(&src, gs_gif_input);
+  file = DGifOpen(&src, gs_gif_input, NULL);
   if (file == NULL)
     {
       /* we do not use giferror here because it doesn't
