--- configure	2010-03-22 10:12:48.000000000 -0400
+++ configure.new	2010-07-16 00:05:30.551966715 -0400
@@ -18125,7 +18125,7 @@
 xiphos_exec=xiphos
 
 
-REQUIRED_LIBS="glib-2.0 gtk+-2.0 >= 2.14 libxml-2.0 libglade-2.0 libgtkhtml-3.14 >= 3.23  sword dbus-glib-1 libgsf-1 gtkhtml-editor"
+REQUIRED_LIBS="glib-2.0 gtk+-2.0 >= 2.14 libxml-2.0 libglade-2.0 libgtkhtml-3.14 >= 3.23  sword dbus-glib-1 libgsf-1 gtkhtml-editor-3.14"
 
 
 # Check whether --with-gecko was given.
