diff -ur ../gdmap-0.8.1/src/gui_support.c .//src/gui_support.c
--- ../gdmap-0.8.1/src/gui_support.c	2008-05-23 09:54:28.000000000 -0500
+++ .//src/gui_support.c	2010-07-22 02:52:43.489842704 -0500
@@ -85,7 +85,7 @@
   const char* text;
 
   (void)event;
-  if (GTK_WIDGET_SENSITIVE(label)) {
+  if (gtk_widget_get_sensitive(label)) {
     char* temp;
 
     text = g_object_get_data(G_OBJECT(label), "label");
@@ -106,7 +106,7 @@
 
   (void)event;
   //  if (GTK_WIDGET_HAS_FOCUS(gtk_widget_get_parent(GTK_WIDGET(label)))) return TRUE;
-/*   if (GTK_WIDGET_SENSITIVE(label)) { */
+/*   if (gtk_widget_get_sensitive(label)) { */
     text = g_object_get_data(G_OBJECT(label), "label");
     temp = g_strdup_printf(EVENT_FMT, text);
     gtk_label_set_markup(label, temp);
@@ -129,7 +129,7 @@
 void ui_event_label_set_sensitive(GtkWidget* ebox, gboolean set) {
   gpointer callback = g_object_get_data(G_OBJECT(ebox), "callback");
   GtkWidget* child = GTK_BIN(ebox)->child;
-  if (GTK_WIDGET_SENSITIVE(child) == set) return;
+  if (gtk_widget_get_sensitive(child) == set) return;
 
   if (set && !callback) return;
   gtk_widget_set_sensitive(child, set);
