commit 96f1c00cfb5b92419e188248c6e672dc10990d9e
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Mon Aug 14 09:14:08 2017 +0800

    Fix compatibility with protolude-0.2

diff --git a/main/Main.hs b/main/Main.hs
index 902195e..cbe5a9f 100644
--- a/main/Main.hs
+++ b/main/Main.hs
@@ -11,7 +11,7 @@ import           PostgREST.DbStructure    (getDbStructure)

 module Main where

-import           Protolude
+import           Protolude hiding         (replace, hPutStrLn)
 import           PostgREST.App
 import           PostgREST.Config                     (AppConfig (..),
 PgVersion (..),
diff --git a/postgrest.cabal b/postgrest.cabal
index e727116..ce97ff2 100644
--- a/postgrest.cabal
+++ b/postgrest.cabal
@@ -72,7 +72,7 @@ library
                      , network-uri
                      , optparse-applicative >= 0.13 && < 0.14
                      , parsec
-                     , protolude
+                     , protolude >= 0.2
                      , Ranged-sets == 0.3.0
                      , regex-tdfa
                      , safe
diff --git a/src/PostgREST/Config.hs b/src/PostgREST/Config.hs
index 9c7bd5d..d7dd680 100644
--- a/src/PostgREST/Config.hs
+++ b/src/PostgREST/Config.hs
@@ -45,7 +45,7 @@ import           System.IO                   (hPrint)
 import           Text.Heredoc
 import           Text.PrettyPrint.ANSI.Leijen hiding ((<>), (<$>))
 import qualified Text.PrettyPrint.ANSI.Leijen as L
-import           Protolude hiding            (intercalate, (<>))
+import           Protolude hiding            (intercalate, (<>), hPutStrLn)

 -- | Config file settings for the server
 data AppConfig = AppConfig {
diff --git a/src/PostgREST/Parsers.hs b/src/PostgREST/Parsers.hs
index e71abf5..0168603 100644
--- a/src/PostgREST/Parsers.hs
+++ b/src/PostgREST/Parsers.hs
@@ -1,6 +1,6 @@
 module PostgREST.Parsers where

-import           Protolude                     hiding (try, intercalate)
+import           Protolude                     hiding (try, intercalate, replace)
 import           Control.Monad                 ((>>))
 import           Data.Foldable                 (foldl1)
 import qualified Data.HashMap.Strict           as M
diff --git a/src/PostgREST/QueryBuilder.hs b/src/PostgREST/QueryBuilder.hs
index 4533b68..05dbe0d 100644
--- a/src/PostgREST/QueryBuilder.hs
+++ b/src/PostgREST/QueryBuilder.hs
@@ -47,7 +47,7 @@ import           Data.Scientific         ( FPFormat (..)
                                          , formatScientific
                                          , isInteger
                                          )
-import           Protolude hiding        (from, intercalate, ord, cast)
+import           Protolude hiding        (from, intercalate, ord, cast, replace)
 import           PostgREST.ApiRequest    (PreferRepresentation (..))

 {-| The generic query result format used by API responses. The location header
