--- ./source/plotutils.c.orig	2012-04-05 00:44:52.000000000 +0200
+++ ./source/plotutils.c	2012-04-28 05:27:04.000000000 +0200
@@ -1018,7 +1018,7 @@
     X_COORD (&DEVICE (f)) = 0;
     Y_COORD (&DEVICE (f)) = 0;
     return (PLOTTER (&DEVICE (f)));
-  } else if (!strcmp (device_type, "gif")) {
+  } else if (!strcmp (device_type, "gif") || !strcmp (device_type, "png")) {
 /*------------------------------------+
 | Supported plotter type - pseudo GIF |
 +------------------------------------*/
@@ -1059,7 +1059,7 @@
     (void) pl_setplparam (PLOTTER_PARAMS (&DEVICE (f)), "BITMAPSIZE", size);
     (void) pl_setplparam (PLOTTER_PARAMS (&DEVICE (f)), "BG_COLOR", (void *) "black");
     (void) pl_setplparam (PLOTTER_PARAMS (&DEVICE (f)), "GIF_ANIMATION", (void *) "no");
-    PLOTTER (&DEVICE (f)) = pl_newpl_r ("gif", NULL, STREAM (&DEVICE (f)), stderr, PLOTTER_PARAMS (&DEVICE (f)));
+    PLOTTER (&DEVICE (f)) = pl_newpl_r (device_type, NULL, STREAM (&DEVICE (f)), stderr, PLOTTER_PARAMS (&DEVICE (f)));
     if (PLOTTER (&DEVICE (f)) == NULL) {
       diagnostic_node (A68_RUNTIME_ERROR, p, ERROR_DEVICE_CANNOT_OPEN);
       exit_genie (p, A68_RUNTIME_ERROR);
