diff -uNr src/network/md5.cpp src/network/md5.cpp
--- src/network/md5.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/network/md5.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -76,6 +76,7 @@
 
 #include "md5.h"
 
+#include <cstring>
 #include <string>
 #include <memory>
 
diff -uNr src/network/nKrawall.cpp src/network/nKrawall.cpp
--- src/network/nKrawall.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/network/nKrawall.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -31,7 +31,7 @@
 
 
 */
-
+#include <cstring>
 #include "nKrawall.h"
 #include "nNetwork.h"
 #include "nServerInfo.h"
diff -uNr src/network/nNetObject.h src/network/nNetObject.h
--- src/network/nNetObject.h	2008-08-16 21:01:46.000000000 +0200
+++ src/network/nNetObject.h	2008-08-16 19:08:55.000000000 +0200
@@ -28,6 +28,7 @@
 #ifndef ArmageTron_NETOBJECT_H
 #define ArmageTron_NETOBJECT_H
 
+#include <cstring>
 #include "nNetwork.h"
 #include "tArray.h"
 #include "tConsole.h"
diff -uNr src/network/nNetwork.cpp src/network/nNetwork.cpp
--- src/network/nNetwork.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/network/nNetwork.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -24,7 +24,7 @@
 ***************************************************************************
 
 */
-
+#include <cstring>
 #include "tMemManager.h"
 #include "tInitExit.h"
 #include "nSimulatePing.h"
diff -uNr src/network/nSocket.cpp src/network/nSocket.cpp
--- src/network/nSocket.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/network/nSocket.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -30,7 +30,7 @@
 // loosely based on the GNU Quake 1 base network code, so technicaly:
 // Copyright (C) 1996-1997 Id Software, Inc.
 // Modified for Armagetron by Manuel Moos (manuel@moosnet.de)
-
+#include <cstring>
 #include "nSocket.h"
 
 #include "config.h"
diff -uNr src/render/rModel.cpp src/render/rModel.cpp
--- src/render/rModel.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/render/rModel.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -26,6 +26,7 @@
 */
 
 #include "rModel.h"
+#include <cstring>
 #include <string>
 #include <fstream>
 #include <stdlib.h>
diff -uNr src/thirdparty/particles/ParticleDLL/actions.h src/thirdparty/particles/ParticleDLL/actions.h
--- src/thirdparty/particles/ParticleDLL/actions.h	2008-08-16 21:01:46.000000000 +0200
+++ src/thirdparty/particles/ParticleDLL/actions.h	2008-08-16 19:08:55.000000000 +0200
@@ -8,6 +8,7 @@
 #define PARTICLEACTIONS_H
 
 // domain includes particle group and vector.
+#include <typeinfo>
 #include "pDomain.h"
 #include "ParticleGroup.h"
 
diff -uNr src/thirdparty/particles/pDomain.h src/thirdparty/particles/pDomain.h
--- src/thirdparty/particles/pDomain.h	2008-08-16 21:01:46.000000000 +0200
+++ src/thirdparty/particles/pDomain.h	2008-08-16 19:08:55.000000000 +0200
@@ -2,7 +2,7 @@
 #define PDOMAIN_H_INCLUDED
 
 #include "pVec.h"
-
+#include <stdio.h>
 class pDomain
 {
 public:
diff -uNr src/thirdparty/particles/pVec.h src/thirdparty/particles/pVec.h
--- src/thirdparty/particles/pVec.h	2008-08-16 21:01:46.000000000 +0200
+++ src/thirdparty/particles/pVec.h	2008-08-16 19:08:55.000000000 +0200
@@ -11,7 +11,7 @@
 #define PVEC_H_INCLUDED
 
 #include <iostream>
-
+#include <stdlib.h>
 #include <math.h>
 
 #ifndef M_PI
diff -uNr src/thirdparty/shttpd/shttpd.c src/thirdparty/shttpd/shttpd.c
--- src/thirdparty/shttpd/shttpd.c	2008-08-16 21:01:46.000000000 +0200
+++ src/thirdparty/shttpd/shttpd.c	2008-08-16 19:08:55.000000000 +0200
@@ -44,6 +44,7 @@
 #define EMBEDDED
 #endif
 
+#include <linux/limits.h>
 #include <sys/types.h>				/* Common */
 #include <sys/stat.h>
 #include <stdio.h>
diff -uNr src/tools/tArray.cpp src/tools/tArray.cpp
--- src/tools/tArray.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tArray.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -31,7 +31,9 @@
 #include "tMemManager.h"
 #include <iostream>
 #include <stdlib.h>
+
 #include <string>
+#include <cstring>
 #include "tArray.h"
 
 void GrowingArrayBase::ComplainIfFull(){
diff -uNr src/tools/tCommandLine.cpp src/tools/tCommandLine.cpp
--- src/tools/tCommandLine.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tCommandLine.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -27,6 +27,7 @@
 */
 
 // declaration
+#include <cstring>
 #ifndef		TCOMMANDLINE_H_INCLUDED
 #include	"tCommandLine.h"
 #endif
diff -uNr src/tools/tConfiguration.cpp src/tools/tConfiguration.cpp
--- src/tools/tConfiguration.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tConfiguration.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -25,6 +25,7 @@
 
 */
 
+#include <cstring>
 #include "config.h"
 #include <fstream>
 #include <iomanip>
diff -uNr src/tools/tCrypt.cpp src/tools/tCrypt.cpp
--- src/tools/tCrypt.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tCrypt.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -27,7 +27,7 @@
 
 #include "tCrypt.h"
 #include "tMemManager.h"
-
+#include<cstring>
 #include <string>
 
 class tCryptDummy: public tCrypt
diff -uNr src/tools/tDirectories.cpp src/tools/tDirectories.cpp
--- src/tools/tDirectories.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tDirectories.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -27,7 +27,7 @@
 */
 
 #include "config.h"
-
+#include <cstring>
 #include <errno.h>
 #include <sys/types.h>
 #include <stdio.h>
diff -uNr src/tools/tError.cpp src/tools/tError.cpp
--- src/tools/tError.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tError.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -25,6 +25,7 @@
 
 */
 
+#include <cstdlib>
 #include "config.h"
 #include <iostream>
 #if HAVE_UNISTD_H
diff -uNr src/tools/tLocale.cpp src/tools/tLocale.cpp
--- src/tools/tLocale.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tLocale.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -30,7 +30,7 @@
 #include "tConsole.h"
 #include "tDirectories.h"
 #include "tSafePTR.h"
-
+#include <cstring>
 #include <fstream>
 #include <string>
 #include <map>
diff -uNr src/tools/tMemManager.cpp src/tools/tMemManager.cpp
--- src/tools/tMemManager.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tMemManager.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -27,7 +27,7 @@
 
 // disable malloc replacement
 #define NO_MALLOC_REPLACEMENT
-
+#include <cstring>
 #include "defs.h"
 
 #include <iostream>
diff -uNr src/tools/tRandom.cpp src/tools/tRandom.cpp
--- src/tools/tRandom.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tRandom.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -26,6 +26,7 @@
 */
 
 // declaration
+#include<cstdlib>
 #ifndef		TRANDOM_H_INCLUDED
 #include	"tRandom.h"
 #endif
diff -uNr src/tools/tString.cpp src/tools/tString.cpp
--- src/tools/tString.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tString.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -31,6 +31,7 @@
 #include "tConfiguration.h"
 #include "tException.h"
 #include <ctype.h>
+#include <cstring>
 #include <string>
 #include <iostream>
 
diff -uNr src/tools/tXmlParser.cpp src/tools/tXmlParser.cpp
--- src/tools/tXmlParser.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tXmlParser.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -26,6 +26,7 @@
 
 */
 
+#include <cstring>
 #include "tXmlParser.h"
 #include "tResourceManager.h"
 #include "tDirectories.h"
diff -uNr src/tools/tXmlParser.h src/tools/tXmlParser.h
--- src/tools/tXmlParser.h	2008-08-16 21:01:46.000000000 +0200
+++ src/tools/tXmlParser.h	2008-08-16 19:08:55.000000000 +0200
@@ -34,7 +34,7 @@
 #include "defs.h"
 #include <libxml/parser.h>
 #include <libxml/tree.h>
-
+#include <typeinfo>
 #include "tString.h"
 #include "tError.h"
 #include "tConsole.h"
diff -uNr src/tron/gAIBase.cpp src/tron/gAIBase.cpp
--- src/tron/gAIBase.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tron/gAIBase.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -43,7 +43,7 @@
 #include "tRandom.h"
 #include "tRecorder.h"
 #include <stdlib.h>
-
+#include <memory>
 #define AI_REACTION          0 
 #define AI_EMERGENCY         1 
 #define AI_RANGE             2 
diff -uNr src/tron/gCycle.cpp src/tron/gCycle.cpp
--- src/tron/gCycle.cpp	2008-08-16 21:01:46.000000000 +0200
+++ src/tron/gCycle.cpp	2008-08-16 19:08:55.000000000 +0200
@@ -24,7 +24,7 @@
 ***************************************************************************
 
 */
-
+#include <memory>
 #include "gCycle.h"
 #include "nConfig.h"
 #include "rModel.h"

