--- pyface/ui/qt4/code_editor/pygments_highlighter.py.orig	2015-07-07 21:18:51.174941435 +0200
+++ pyface/ui/qt4/code_editor/pygments_highlighter.py	2015-07-07 21:19:29.448031232 +0200
@@ -33,11 +33,12 @@
         for rexmatch, action, new_state in statetokens:
             m = rexmatch(text, pos)
             if m:
-                if type(action) is _TokenType:
-                    yield pos, action, m.group()
-                else:
-                    for item in action(self, m):
-                        yield item
+                if action is not None:
+                    if type(action) is _TokenType:
+                        yield pos, action, m.group()
+                    else:
+                        for item in action(self, m):
+                            yield item
                 pos = m.end()
                 if new_state is not None:
                     # state transition
