diff -wbBur gmic-1.4.7.0/src/Makefile gmic-1.4.7.0.my/src/Makefile
--- gmic-1.4.7.0/src/Makefile	2010-12-18 09:38:29.000000000 +0000
+++ gmic-1.4.7.0.my/src/Makefile	2010-12-23 14:01:47.000000000 +0000
@@ -76,11 +76,11 @@
 #---------------------------------------------------
 
 # Flags that are mandatory to compile 'gmic'.
-MANDATORY_CFLAGS = -Dgmic_build -I$(USR)/include
-MANDATORY_LDFLAGS = -L$(USR)/lib
+MANDATORY_CFLAGS = -Dgmic_build -I$(USR)/include `pkg-config --cflags opencv`
+MANDATORY_LDFLAGS = -L$(USR)/lib `pkg-config --libs opencv`
 ifeq ($(IS_GCC),yes)
-MANDATORY_CFLAGS += -Wall -W -I$(USR)/include
-MANDATORY_LDFLAGS = -lm -L$(USR)/lib
+MANDATORY_CFLAGS += -Wall -W -I$(USR)/include `pkg-config --cflags opencv`
+MANDATORY_LDFLAGS = -lm -L$(USR)/lib `pkg-config --libs opencv`
 endif
 
 # Flags to enable debugging.
@@ -129,7 +129,7 @@
 
 # Flags to enable native support of webcams, using the OpenCV library.
 OPENCV_CFLAGS = -Dcimg_use_opencv -I$(USR)/include/opencv
-OPENCV_LDFLAGS = -lcv -lhighgui
+OPENCV_LDFLAGS = `pkg-config --libs opencv`
 # OPENCV_LDFLAGS = -lopencv_core -lopencv_highgui   # Use this for OpenCV 2.2.0 !
 
 # Flags to enable native support of most classical image file formats, using the GraphicsMagick++ library.
