diff -rupN libretro-bsnes-hd.orig/bsnes/GNUmakefile libretro-bsnes-hd/bsnes/GNUmakefile
--- libretro-bsnes-hd.orig/bsnes/GNUmakefile	2020-01-29 12:32:57.736128837 +0100
+++ libretro-bsnes-hd/bsnes/GNUmakefile	2020-01-29 12:40:58.993295237 +0100
@@ -39,8 +39,8 @@ else ifneq ($(filter $(platform),linux b
     options += -Wl,-export-dynamic
     options += -lX11 -lXext
   else ifeq ($(binary),library)
-    flags += -fPIC
-    options += -shared
+    flags += $(CXXFLAGS) -fPIC
+    options += $(LDFLAGS) -shared
   endif
 endif
 
