diff -aur Amaya11.3.1/Amaya/amaya/HTMLedit.c Amaya11.3.1.new/Amaya/amaya/HTMLedit.c
--- Amaya11.3.1/Amaya/amaya/HTMLedit.c	2009-12-09 11:11:22.000000000 +0100
+++ Amaya11.3.1.new/Amaya/amaya/HTMLedit.c	2010-08-17 21:39:23.696495709 +0200
@@ -1689,7 +1689,7 @@
                 {
 #ifdef _WX
                   wxString tmpFile = TtaConvMessageToWX(tempURL);
-                  wxFile::wxFile (tmpFile, wxFile::write);
+                  wxFile (tmpFile, wxFile::write);
 #endif /* _WX */
                 }
             }

