diff -aur Amaya11.3.1/Amaya/thotlib/dialogue/AmayaPage.cpp Amaya11.3.1.new/Amaya/thotlib/dialogue/AmayaPage.cpp
--- Amaya11.3.1/Amaya/thotlib/dialogue/AmayaPage.cpp	2009-09-09 12:07:49.000000000 +0200
+++ Amaya11.3.1.new/Amaya/thotlib/dialogue/AmayaPage.cpp	2010-08-17 20:54:13.606772367 +0200
@@ -383,7 +383,7 @@
 					    /*| wxSP_PERMIT_UNSPLIT*/
 #endif /* 0 */
 					    );
-  SetSplitMode(0); // undefined
+  SetSplitMode(wxSPLIT_VERTICAL); // undefined not allowed
   p_sizerTop2->Add( m_pSplitterWindow, 1, wxEXPAND, 0 );
