diff -aur Amaya11.3.1/Amaya/thotlib/dialogue/appdialogue_wx.c Amaya11.3.1.new/Amaya/thotlib/dialogue/appdialogue_wx.c
--- Amaya11.3.1/Amaya/thotlib/dialogue/appdialogue_wx.c	2009-09-15 17:17:02.000000000 +0200
+++ Amaya11.3.1.new/Amaya/thotlib/dialogue/appdialogue_wx.c	2010-08-17 21:07:40.973352206 +0200
@@ -993,7 +993,7 @@
    * this function waits for complete widgets initialisation */
   /* notice: no not use safe yield here because it use a
      wxWindowDisabler and it makes menus blinking */
-  wxYield();
+  wxTheApp->Yield(TRUE);
   return TRUE;
 }
 
@@ -1824,7 +1824,7 @@
   TtaHandlePendingEvents();
   /* wait for frame initialisation (needed by opengl) 
    * this function waits for complete widgets initialisation */
-  wxYield();
+  wxTheApp->Yield(TRUE);
   /* fix accessibility problem : force the created frame to get the focus */
   TtaRedirectFocus();

