diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-11-01 18:23:59.000000000 +0100
+++ b/CMakeLists.txt	2021-11-11 10:09:50.134726938 +0100
@@ -50,7 +50,13 @@
 # I'm not sure at this point whether this has any effect. I know etting it to "Instrument" causes it to show as
 # VSTi in reaper whereas setting it to "Effect" causes it to show as VST
 set(STOCHAS_CATEGORY "Effect") # original was "Instrument"
-add_subdirectory(lib/JUCE)
+find_package(JUCE COMPONENTS REQUIRED)
+include(FindPkgConfig)
+pkg_search_module(flac REQUIRED flac)
+pkg_search_module(ogg REQUIRED ogg)
+pkg_search_module(vorbis REQUIRED vorbis)
+pkg_search_module(vorbisenc REQUIRED vorbisenc)
+pkg_search_module(vorbisfile REQUIRED vorbisfile)
 
 add_custom_target( git-info BYPRODUCTS ${CMAKE_BINARY_DIR}/geninclude/version.cpp
   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
@@ -174,6 +180,11 @@
   assets                             # to include our images
   juce::juce_audio_utils
   juce::juce_audio_plugin_client
+  ${ogg_LIBRARIES}
+  ${flac_LIBRARIES}
+  ${vorbis_LIBRARIES}
+  ${vorbisenc_LIBRARIES}
+  ${vorbisfile_LIBRARIES}
   )
 
 if( APPLE )
