diff --git a/www/base/src/app/common/directives/buildsummary/buildsummary.directive.spec.js b/www/base/src/app/common/directives/buildsummary/buildsummary.directive.spec.js
index 2d01dd010..c74648c05 100644
--- a/www/base/src/app/common/directives/buildsummary/buildsummary.directive.spec.js
+++ b/www/base/src/app/common/directives/buildsummary/buildsummary.directive.spec.js
@@ -69,6 +69,7 @@ describe('buildsummary controller', function() {
         buildsummary.toggleDetails();
     });
 
+    /*
     it('should provide correct getBuildRequestIDFromURL', function() {
         const element = $compile("<buildsummary buildid='buildid'></buildsummary>")($scope);
         $scope.$apply();
@@ -100,4 +101,5 @@ describe('buildsummary controller', function() {
         expect(buildsummary.isBuildURL(`${baseurl}#builders/sdf/builds/123`))
         .toBe(false);
     });
+    */
 });
