--- acinclude.m4.original	2012-01-24 10:42:16.066632295 +0100
+++ acinclude.m4	2012-01-24 10:43:15.800274030 +0100
@@ -1411,7 +1411,7 @@
 if test x$ax_python_bin != x; then
    AC_CHECK_LIB($ax_python_bin, main, ax_python_lib=$ax_python_bin, ax_python_lib=no)
    AC_CHECK_HEADER([$ax_python_bin/Python.h],
-   [[ax_python_header=`locate $ax_python_bin/Python.h | sed -e s,/Python.h,,`]],
+   [[ax_python_header="/usr/include/$ax_python_bin"]],
    ax_python_header=no)
    if test x$ax_python_lib != xno; then
      if test x$ax_python_header != xno; then
