diff -Nur evolution-rss-0.2.3.orig/src/Makefile.am evolution-rss-0.2.3/src/Makefile.am
--- evolution-rss-0.2.3.orig/src/Makefile.am	2011-01-28 09:29:44.000000000 -0800
+++ evolution-rss-0.2.3/src/Makefile.am	2011-03-14 09:51:19.510031938 -0700
@@ -94,7 +94,7 @@
 endif
 
 liborg_gnome_evolution_rss_la_SOURCES = $(SOURCES)
-liborg_gnome_evolution_rss_la_LIBADD = $(WEBKIT_LIBS) $(GECKO_LIBS) $(LIBSOUP_LIBS) $(GTKHTML_EDITOR_LIBS) $(NO_UNDEFINED_LIBS)
+liborg_gnome_evolution_rss_la_LIBADD = $(WEBKIT_LIBS) $(GECKO_LIBS) $(LIBSOUP_LIBS) $(GTKHTML_EDITOR_LIBS) $(NO_UNDEFINED_LIBS) $(DBUS_LIBS)
 liborg_gnome_evolution_rss_la_LDFLAGS = -module -avoid-version $(NO_UNDEFINED)
 #$(GECKO_RPATH)
 
