diff -ru jupyterlab_widgets-1.1.0.orig/pyproject.toml jupyterlab_widgets-1.1.0/pyproject.toml
--- jupyterlab_widgets-1.1.0.orig/pyproject.toml	2022-03-16 08:53:32.364140285 +0100
+++ jupyterlab_widgets-1.1.0/pyproject.toml	2022-03-16 08:54:00.704173882 +0100
@@ -1,3 +1,3 @@
 [build-system]
-requires = ["jupyter_packaging~=0.7.9", "jupyterlab~=3.0", "setuptools>=40.8.0", "wheel"]
+requires = ["jupyter_packaging", "jupyterlab", "setuptools>=40.8.0", "wheel"]
 build-backend = "setuptools.build_meta"
