diff -Naur gwibber-3.6.0.orig/configure.ac gwibber-3.6.0/configure.ac
--- gwibber-3.6.0.orig/configure.ac	2012-10-03 02:09:08.000000000 +0200
+++ gwibber-3.6.0/configure.ac	2013-03-28 21:34:39.227995294 +0100
@@ -159,10 +159,10 @@
 
 if test "x$enable_spell" = "xyes"; then
         PKG_CHECK_MODULES(GTKSPELL,
-                  gtkspell-3.0)
+                  gtkspell3-3.0)
         AC_SUBST(GTKSPELL_CFLAGS)
         AC_SUBST(GTKSPELL_CFLAGS)
-	GTKSPELL_PKG="--pkg gtkspell-3.0"
+	GTKSPELL_PKG="--pkg gtkspell3-3.0"
         AC_SUBST(GTKSPELL_PKG)
 fi
 
diff -Naur gwibber-3.6.0.orig/libgwibber-gtk/entry.vala gwibber-3.6.0/libgwibber-gtk/entry.vala
--- gwibber-3.6.0.orig/libgwibber-gtk/entry.vala	2012-08-21 06:14:19.000000000 +0200
+++ gwibber-3.6.0/libgwibber-gtk/entry.vala	2013-03-28 21:36:32.830637428 +0100
@@ -141,7 +141,8 @@
 #if HAVE_GTKSPELL
             try
             {
-              var spell = new Gtk.Spell.attach(this, null);
+              var spell = new GtkSpell.Checker();
+              spell.attach(this);
             }
             catch (Error e)
             {
