--- guake-0.4.2-orig/data/guake-prefs.desktop.in	2009-12-01 12:51:08.000000000 +0100
+++ guake-0.4.2/data/guake-prefs.desktop.in	2012-05-21 17:07:05.982022360 +0200
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Name=Guake Preferences
 Name[pt_BR]=Preferências do Guake
-Comment=Comment=Configure your Guake sessions
+Comment=Configure your Guake sessions
 Comment[pt_BR]=Configurações do terminal Guake
 TryExec=guake-prefs
 Exec=guake-prefs
