commit 0353da88e550031b4694a4168bd6d489f873d68a
Author: max ulidtko <ulidtko@gmail.com>
Date:   Fri Mar 2 07:51:38 2012 +0200

    globalhotkeys build fix for glib 2.31
    
    apply patch from #425

diff --git a/src/globalhotkeys/keybinder.h b/src/globalhotkeys/keybinder.h
index 00e532d..69a50c0 100644
--- a/src/globalhotkeys/keybinder.h
+++ b/src/globalhotkeys/keybinder.h
@@ -2,7 +2,7 @@
 #ifndef __KEY_BINDER_H__
 #define __KEY_BINDER_H__
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
