diff --git a/plugins/devices/shortcut/shortcut.cpp b/plugins/devices/shortcut/shortcut.cpp
index 0aa8609..80bf602 100644
--- a/plugins/devices/shortcut/shortcut.cpp
+++ b/plugins/devices/shortcut/shortcut.cpp
@@ -191,7 +191,7 @@ void Shortcut::appendGeneralItems(){
     for (KeyEntry * gkeyEntry: generalEntries){
         if (showList.contains(gkeyEntry->keyStr)){
             DefineShortcutItem * singleWidget = new DefineShortcutItem(gkeyEntry->keyStr, gkeyEntry->valueStr);
-            singleWidget->setUserData(Qt::UserRole, gkeyEntry);
+            //singleWidget->setUserData(Qt::UserRole, gkeyEntry);

             CustomLineEdit * line = singleWidget->lineeditComponent();
             connect(line, &CustomLineEdit::shortcutCodeSignals, this, [=](QList<int> keyCode){
@@ -216,7 +216,7 @@ void Shortcut::buildCustomItem(KeyEntry * nkeyEntry){
     DefineShortcutItem * singleWidget = new DefineShortcutItem(nkeyEntry->nameStr, nkeyEntry->bindingStr);
     singleWidget->setDeleteable(true);
     singleWidget->setUpdateable(true);
-    singleWidget->setUserData(Qt::UserRole, nkeyEntry);
+    //singleWidget->setUserData(Qt::UserRole, nkeyEntry);
     connect(singleWidget, &DefineShortcutItem::updateShortcutSignal, [=]{
         addDialog->setTitleText(QObject::tr("Update Shortcut"));
         addDialog->setUpdateEnv(nkeyEntry->gsPath, nkeyEntry->nameStr, nkeyEntry->actionStr);
@@ -337,7 +337,7 @@ void Shortcut::createNewShortcut(QString path, QString name, QString exec){
                 DefineShortcutItem * widgetItem = dynamic_cast<DefineShortcutItem *>(ui->customListWidget->itemWidget(item));
                 widgetItem->setShortcutName(name);
                 KeyEntry * uKeyentry = customEntries.at(i);
-                widgetItem->setUserData(Qt::UserRole, uKeyentry);
+                //widgetItem->setUserData(Qt::UserRole, uKeyentry);
             }
         }
     }
diff --git a/plugins/devices/shortcut/shortcut.h b/plugins/devices/shortcut/shortcut.h
index f3d72fa..178266a 100644
--- a/plugins/devices/shortcut/shortcut.h
+++ b/plugins/devices/shortcut/shortcut.h
@@ -38,7 +38,7 @@ QT_END_NAMESPACE

 typedef struct _KeyEntry KeyEntry;

-struct _KeyEntry : QObjectUserData{
+struct _KeyEntry{
 //    int keyval;
     QString gsSchema;
     QString keyStr;
diff --git a/shell/utils/functionselect.h b/shell/utils/functionselect.h
index 54efe44..3c2df92 100644
--- a/shell/utils/functionselect.h
+++ b/shell/utils/functionselect.h
@@ -20,12 +20,13 @@
 #ifndef FUNCTIONSELECT_H
 #define FUNCTIONSELECT_H

+#include <QObject>
 #include <QList>
 #include <QStack>
 //#include <QStringList>


-typedef struct _FuncInfo : QObjectUserData
+typedef struct _FuncInfo
 {
     int type;
     int index;
@@ -34,7 +35,7 @@ typedef struct _FuncInfo : QObjectUserData
     QString namei18nString;
 }FuncInfo;

-typedef struct _RecordFunc : QObjectUserData
+typedef struct _RecordFunc
 {
     int type;
     QString namei18nString;
