diff --git a/plugins/system/display/outputconfig.h b/plugins/system/display/outputconfig.h
index 89d61f0b..afe86424 100644
--- a/plugins/system/display/outputconfig.h
+++ b/plugins/system/display/outputconfig.h
@@ -23,6 +23,7 @@
 #include <QComboBox>
 #include <QWidget>
 
+#include <KF5/KScreen/kscreen/mode.h>
 #include <KF5/KScreen/kscreen/output.h>
 
 #include <QGSettings>
diff --git a/plugins/system/display/resolutionslider.cpp b/plugins/system/display/resolutionslider.cpp
index e0bd3d36..3a9b88fa 100644
--- a/plugins/system/display/resolutionslider.cpp
+++ b/plugins/system/display/resolutionslider.cpp
@@ -29,6 +29,7 @@
 
 #include <QDBusInterface>
 
+#include <KF5/KScreen/kscreen/mode.h>
 #include <KF5/KScreen/kscreen/output.h>
 
 static bool sizeLessThan(const QSize &sizeA, const QSize &sizeB)
diff --git a/plugins/system/display/widget.cpp b/plugins/system/display/widget.cpp
index 6bfc22c9..a4bb8f17 100644
--- a/plugins/system/display/widget.cpp
+++ b/plugins/system/display/widget.cpp
@@ -219,7 +219,7 @@ void Widget::setConfig(const KScreen::ConfigPtr &config, bool showBrightnessFram
         outputRemoved(outputId, false);
     });
 
-    connect(mConfig.data(), &KScreen::Config::primaryOutputChanged,
+    connect(mConfig.data(), &KScreen::Config::prioritiesChanged,
             this, &Widget::primaryOutputChanged);
 
     for (const KScreen::OutputPtr &output : mConfig->outputs()) {
@@ -1026,9 +1026,10 @@ void Widget::primaryOutputSelected(int index)
 }
 
 // 主输出
-void Widget::primaryOutputChanged(const KScreen::OutputPtr &output)
+void Widget::primaryOutputChanged()
 {
     Q_ASSERT(mConfig);
+    const KScreen::OutputPtr &output = mConfig->primaryOutput();
     int index = output.isNull() ? 0 : ui->primaryCombo->findData(output->id());
     if (index == -1 || index == ui->primaryCombo->currentIndex()) {
         return;
diff --git a/plugins/system/display/widget.h b/plugins/system/display/widget.h
index 9790c59c..83c6351c 100644
--- a/plugins/system/display/widget.h
+++ b/plugins/system/display/widget.h
@@ -131,7 +131,7 @@ private Q_SLOTS:
     void outputAdded(const KScreen::OutputPtr &output, bool connectChanged);
     void outputRemoved(int outputId, bool connectChanged);
     void primaryOutputSelected(int index);
-    void primaryOutputChanged(const KScreen::OutputPtr &output);
+    void primaryOutputChanged();
 
     void showNightWidget(bool judge);
     void showCustomWiget(int index);
diff --git a/plugins/system/touchscreen/touchscreen.cpp b/plugins/system/touchscreen/touchscreen.cpp
index b318d296..307c616b 100644
--- a/plugins/system/touchscreen/touchscreen.cpp
+++ b/plugins/system/touchscreen/touchscreen.cpp
@@ -21,6 +21,7 @@
 
 
 #include <KF5/KScreen/kscreen/getconfigoperation.h>
+#include <KF5/KScreen/kscreen/mode.h>
 #include <KF5/KScreen/kscreen/output.h>
 #include <QDebug>
 #include <QThread>
