--- a/data/hydrogen.default.conf	2016-11-05 09:22:50.000000000 +0100
+++ b/data/hydrogen.default.conf	2017-07-28 11:05:14.481532464 +0200
@@ -19,7 +19,7 @@
 	<hearNewNotes>true</hearNewNotes>
 	<recordEvents>false</recordEvents>
 	<quantizeEvents>true</quantizeEvents>
-	<path_to_rubberband>Path to Rubberband-CLI</path_to_rubberband>
+	<path_to_rubberband>/usr/bin/rubberband</path_to_rubberband>
 	<recentUsedSongs />
 	<recentlyUsedEffects />
 	<serverList>
