diff --color -aur pytorch-1.9.0-old/c10/core/TensorImpl.h pytorch-1.9.0-new/c10/core/TensorImpl.h
--- pytorch-1.9.0-old/c10/core/TensorImpl.h	2021-08-17 19:33:40.324974399 +0300
+++ pytorch-1.9.0-new/c10/core/TensorImpl.h	2021-08-18 01:25:00.005901707 +0300
@@ -2177,7 +2177,7 @@
 //    DispatchKeySet
 //
 static_assert(
-    sizeof(void*) != sizeof(int64_t) || // if 64-bit...
+    sizeof(void*) <= sizeof(int64_t) || // if 64-bit...
         sizeof(TensorImpl) == sizeof(int64_t) * 24,
     "You changed the size of TensorImpl on 64-bit arch."
     "See Note [TensorImpl size constraints] on how to proceed.");
