diff -up Gtk2-MozEmbed-0.08/Makefile.PL.pixbuf Gtk2-MozEmbed-0.08/Makefile.PL
--- Gtk2-MozEmbed-0.08/Makefile.PL.pixbuf	2010-08-11 13:34:16.000000000 +0200
+++ Gtk2-MozEmbed-0.08/Makefile.PL	2010-08-11 13:35:36.000000000 +0200
@@ -54,6 +54,8 @@ my %pkgcfg = ExtUtils::PkgConfig->find(
 	"firefox-gtkmozembed >= $build_reqs{'Firefox'}",
 	"mozilla-firefox-gtkmozembed >= $build_reqs{'Firefox'}");
 
+my %pkgcfg_pixbuf = ExtUtils::PkgConfig->find("gdk-pixbuf-2.0");
+
 mkdir 'build', 0777;
 
 our @xs_files = <xs/*.xs>;
@@ -73,8 +75,8 @@ ExtUtils::PkgConfig->write_version_macro
   $pkg => "GTK_MOZ_EMBED",
 );
 
-my $inc = $pkgcfg{cflags};
-my $libs = $pkgcfg{libs};
+my $inc = $pkgcfg{cflags} ." ". $pkgcfg_pixbuf{cflags};
+my $libs = $pkgcfg{libs} . " ". $pkgcfg_pixbuf{libs};
 
 if ($libs =~ m#(lib|lib64)/xulrunner-(devel|sdk)-([^/]+)#) {
 	$inc .= qq( -DGTK_MOZ_EMBED_PERL_XULRUNNER_PATH=)
