diff --git a/uri.gemspec b/uri.gemspec
index 584a4fa..accceb1 100644
--- a/uri.gemspec
+++ b/uri.gemspec
@@ -1,12 +1,6 @@
-begin
-  require_relative "lib/uri/version"
-rescue LoadError # Fallback to load version file in ruby core repository
-  require_relative "version"
-end
-
 Gem::Specification.new do |spec|
   spec.name          = "uri"
-  spec.version       = URI::VERSION
+  spec.version       = "0.11.0"
   spec.authors       = ["Akira Yamada"]
   spec.email         = ["akira@ruby-lang.org"]
 
