diff -Naur cinnamon-settings-daemon-2.0.8.orig/plugins/power/csd-power-manager.c cinnamon-settings-daemon-2.0.8/plugins/power/csd-power-manager.c
--- cinnamon-settings-daemon-2.0.8.orig/plugins/power/csd-power-manager.c	2014-04-11 06:00:33.852563000 +0200
+++ cinnamon-settings-daemon-2.0.8/plugins/power/csd-power-manager.c	2014-04-11 06:52:24.516345019 +0200
@@ -1171,6 +1171,7 @@
         gboolean ret;
         GError *error = NULL;
 
+#if ! UP_CHECK_VERSION(0,99,0) 
         /* get devices from UPower */
         ret = up_client_enumerate_devices_sync (manager->priv->up_client, NULL, &error);
         if (!ret) {
@@ -1178,6 +1179,7 @@
                 g_error_free (error);
                 goto out;
         }
+#endif
 
         /* connected mobile phones */
         gpm_phone_coldplug (manager->priv->phone);
@@ -1317,7 +1319,7 @@
                              gboolean         is_ups)
 {
         CsdPowerActionType policy;
-
+#if ! UP_CHECK_VERSION(0,99,0)
         policy = g_settings_get_enum (manager->priv->settings, "critical-battery-action");
         if (policy == CSD_POWER_ACTION_SUSPEND) {
                 if (is_ups == FALSE &&
@@ -1329,7 +1331,17 @@
                         return policy;
                 return CSD_POWER_ACTION_SHUTDOWN;
         }
+#else
+        char *action;
 
+        action = up_client_get_critical_action (manager->priv->up_client);
+        /* We don't make the difference between HybridSleep and Hibernate */
+        if (g_strcmp0 (action, "PowerOff") == 0)
+                policy = GSD_POWER_ACTION_SHUTDOWN;
+        else
+                policy = GSD_POWER_ACTION_HIBERNATE;
+        g_free (action);
+#endif
         return policy;
 }
 
@@ -2314,6 +2326,7 @@
                                                    "lid-close-ac-action");
         }
 
+#if ! UP_CHECK_VERSION(0,99,0)
         /* check we won't melt when the lid is closed */
         if (action_type != CSD_POWER_ACTION_SUSPEND &&
             action_type != CSD_POWER_ACTION_HIBERNATE) {
@@ -2326,6 +2339,7 @@
                         lock_screensaver (manager);
                 }
         }
+#endif
 
         /* ensure we turn the panel back on after resume */
         ret = gnome_rr_screen_set_dpms_mode (manager->priv->x11_screen,
@@ -3886,6 +3900,7 @@
         inhibit_suspend (manager);
 }
 
+#if ! UP_CHECK_VERSION(0,99,0)
 static void
 upower_notify_sleep_cb (UpClient *client,
                         UpSleepKind sleep_kind,
@@ -3901,6 +3916,7 @@
 {
         handle_resume_actions (manager);
 }
+#endif
 
 static void
 logind_proxy_signal_cb (GDBusProxy  *proxy,
@@ -4021,10 +4037,12 @@
                           G_CALLBACK (engine_settings_key_changed_cb), manager);
         manager->priv->settings_screensaver = g_settings_new ("org.cinnamon.desktop.screensaver");
         manager->priv->up_client = up_client_new ();
+#if ! UP_CHECK_VERSION(0,99,0)
         g_signal_connect (manager->priv->up_client, "notify-sleep",
                           G_CALLBACK (upower_notify_sleep_cb), manager);
         g_signal_connect (manager->priv->up_client, "notify-resume",
                           G_CALLBACK (upower_notify_resume_cb), manager);
+#endif
         manager->priv->lid_is_closed = up_client_get_lid_is_closed (manager->priv->up_client);
         g_signal_connect (manager->priv->up_client, "device-added",
                           G_CALLBACK (engine_device_added_cb), manager);
