--- build/groovy/org/jetbrains/intellij/build/IdeaCommunityBuilder.groovy	(revision 0ea5972cdad569407078fb27070c80e2b9235c53)
+++ build/groovy/org/jetbrains/intellij/build/IdeaCommunityBuilder.groovy	(date 1658877188674)
@@ -41,11 +41,6 @@
     context.options.incrementalCompilation = true
     def tasks = BuildTasks.create(context)
     tasks.buildDistributions()
-    context.messages.block("Build standalone JPS") {
-      Path jpsArtifactDir = context.paths.artifactDir.resolve("jps")
-      new CommunityStandaloneJpsBuilder(context)
-        .processJpsLayout(jpsArtifactDir, context.fullBuildNumber, new ProjectStructureMapping(), true, {})
-    }
   }
 
   void buildUnpackedDistribution(Path targetDirectory) {
--- jps/standalone-builder/src/org/jetbrains/jps/build/Standalone.java	(revision 0ea5972cdad569407078fb27070c80e2b9235c53)
+++ jps/standalone-builder/src/org/jetbrains/jps/build/Standalone.java	(date 1663195749722)
@@ -190,6 +190,8 @@
     ProjectDescriptor descriptor = buildRunner.load(messageHandler, dataStorageRoot, new BuildFSState(true));
     try {
       buildRunner.runBuild(descriptor, CanceledStatus.NULL, messageHandler, BuildType.BUILD, scopes, includeDependenciesToScope);
+    } catch (Exception e) {
+      System.out.println("LEO EXCEPTION: " + Arrays.toString(e.getStackTrace()));
     }
     finally {
       descriptor.release();
