diff -aur xmonad-0.12/xmonad.cabal xmonad-0.12.new/xmonad.cabal
--- xmonad-0.12/xmonad.cabal	2015-12-18 21:23:38.000000000 +0100
+++ xmonad-0.12.new/xmonad.cabal	2017-01-08 12:08:19.195824197 +0100
@@ -71,7 +71,7 @@
                    process,
                    unix,
                    utf8-string >= 0.3 && < 1.1,
-                   X11>=1.5 && < 1.7
+                   X11>=1.5 && < 1.8
 
     if true
         ghc-options:        -funbox-strict-fields -Wall
