diff -wbBur scantailor/filters/deskew/Filter.cpp scantailor.my/filters/deskew/Filter.cpp
--- scantailor/filters/deskew/Filter.cpp	2018-11-16 15:30:37.542900262 +0300
+++ scantailor.my/filters/deskew/Filter.cpp	2018-11-16 15:23:20.859572011 +0300
@@ -83,9 +83,9 @@
 	
 	QDomElement filter_el(doc.createElement("deskew"));
 	writer.enumPages(
-		bind(
+		boost::lambda::bind(
 			&Filter::writePageSettings,
-			this, boost::ref(doc), var(filter_el), _1, _2
+			this, boost::ref(doc), var(filter_el), boost::lambda::_1, boost::lambda::_2
 		)
 	);
 	
diff -wbBur scantailor/filters/fix_orientation/Filter.cpp scantailor.my/filters/fix_orientation/Filter.cpp
--- scantailor/filters/fix_orientation/Filter.cpp	2018-11-16 15:30:37.542900262 +0300
+++ scantailor.my/filters/fix_orientation/Filter.cpp	2018-11-16 15:21:07.436240246 +0300
@@ -98,9 +98,9 @@
 	
 	QDomElement filter_el(doc.createElement("fix-orientation"));
 	writer.enumImages(
-		bind(
+		boost::lambda::bind(
 			&Filter::writeImageSettings,
-			this, boost::ref(doc), var(filter_el), _1, _2
+			this, boost::ref(doc), var(filter_el), boost::lambda::_1, boost::lambda::_2
 		)
 	);
 	
diff -wbBur scantailor/filters/output/ChangeDewarpingDialog.cpp scantailor.my/filters/output/ChangeDewarpingDialog.cpp
--- scantailor/filters/output/ChangeDewarpingDialog.cpp	2018-11-16 15:30:37.616233595 +0300
+++ scantailor.my/filters/output/ChangeDewarpingDialog.cpp	2018-11-16 14:27:12.996278059 +0300
@@ -20,6 +20,7 @@
 #include "ChangeDewarpingDialog.moc"
 #include "PageSelectionAccessor.h"
 #include "QtSignalForwarder.h"
+#include <QButtonGroup>
 #include <boost/function.hpp>
 #include <boost/lambda/lambda.hpp>
 
diff -wbBur scantailor/filters/output/Filter.cpp scantailor.my/filters/output/Filter.cpp
--- scantailor/filters/output/Filter.cpp	2018-11-16 15:30:37.546233596 +0300
+++ scantailor.my/filters/output/Filter.cpp	2018-11-16 14:24:31.816279912 +0300
@@ -91,9 +91,9 @@
 	
 	QDomElement filter_el(doc.createElement("output"));
 	writer.enumPages(
-		bind(
+		boost::lambda::bind(
 			&Filter::writePageSettings,
-			this, boost::ref(doc), var(filter_el), _1, _2
+			this, boost::ref(doc), var(filter_el), boost::lambda::_1, boost::lambda::_2
 		)
 	);
 	
diff -wbBur scantailor/filters/page_layout/Filter.cpp scantailor.my/filters/page_layout/Filter.cpp
--- scantailor/filters/page_layout/Filter.cpp	2018-11-16 15:30:37.549566930 +0300
+++ scantailor.my/filters/page_layout/Filter.cpp	2018-11-16 15:24:58.396237535 +0300
@@ -134,9 +134,9 @@
 	
 	QDomElement filter_el(doc.createElement("page-layout"));
 	writer.enumPages(
-		bind(
+		boost::lambda::bind(
 			&Filter::writePageSettings,
-			this, boost::ref(doc), var(filter_el), _1, _2
+			this, boost::ref(doc), var(filter_el), boost::lambda::_1, boost::lambda::_2
 		)
 	);
 	
diff -wbBur scantailor/filters/page_split/Filter.cpp scantailor.my/filters/page_split/Filter.cpp
--- scantailor/filters/page_split/Filter.cpp	2018-11-16 15:30:37.549566930 +0300
+++ scantailor.my/filters/page_split/Filter.cpp	2018-11-16 15:22:03.612906249 +0300
@@ -108,9 +108,9 @@
 	);
 	
 	writer.enumImages(
-		bind(
+		boost::lambda::bind(
 			&Filter::writeImageSettings,
-			this, boost::ref(doc), var(filter_el), _1, _2
+			this, boost::ref(doc), var(filter_el), boost::lambda::_1, boost::lambda::_2
 		)
 	);
 	
diff -wbBur scantailor/filters/page_split/OptionsWidget.cpp scantailor.my/filters/page_split/OptionsWidget.cpp
--- scantailor/filters/page_split/OptionsWidget.cpp	2018-11-16 15:30:37.616233595 +0300
+++ scantailor.my/filters/page_split/OptionsWidget.cpp	2018-11-16 15:22:32.029572581 +0300
@@ -29,6 +29,7 @@
 #include <QPixmap>
 #include <boost/foreach.hpp>
 #include <assert.h>
+#include <QButtonGroup>
 
 namespace page_split
 {
diff -wbBur scantailor/filters/select_content/Filter.cpp scantailor.my/filters/select_content/Filter.cpp
--- scantailor/filters/select_content/Filter.cpp	2018-11-16 15:30:37.552900262 +0300
+++ scantailor.my/filters/select_content/Filter.cpp	2018-11-16 15:24:01.626238198 +0300
@@ -117,9 +117,9 @@
 	
 	QDomElement filter_el(doc.createElement("select-content"));
 	writer.enumPages(
-		bind(
+		boost::lambda::bind(
 			&Filter::writePageSettings,
-			this, boost::ref(doc), var(filter_el), _1, _2
+			this, boost::ref(doc), var(filter_el), boost::lambda::_1, boost::lambda::_2
 		)
 	);
 	
diff -wbBur scantailor/math/MatrixCalc.h scantailor.my/math/MatrixCalc.h
--- scantailor/math/MatrixCalc.h	2018-11-16 15:30:37.619566929 +0300
+++ scantailor.my/math/MatrixCalc.h	2018-11-16 15:19:32.572908025 +0300
@@ -103,7 +103,7 @@
 	Mat operator-() const;
 
 	T const* rawData() const { return data; }
-private:
+//private:
 	Mat(AbstractAllocator<T>* alloc, T const* data, int rows, int cols)
 		: alloc(alloc), data(data), rows(rows), cols(cols) {}
 
diff -wbBur scantailor/ThumbnailSequence.cpp scantailor.my/ThumbnailSequence.cpp
--- scantailor/ThumbnailSequence.cpp	2018-11-16 15:30:37.612900262 +0300
+++ scantailor.my/ThumbnailSequence.cpp	2018-11-16 15:28:42.442901592 +0300
@@ -489,7 +489,7 @@
 	m_pSelectionLeader(0)
 {
 	m_graphicsScene.setContextMenuEventCallback(
-		bind(&Impl::sceneContextMenuEvent, this, _1)
+		boost::lambda::bind(&Impl::sceneContextMenuEvent, this, boost::lambda::_1)
 	);
 }
 
@@ -603,7 +603,7 @@
 {
 	ItemsById::iterator const id_it(m_itemsById.find(page_info.id()));
 	if (id_it != m_itemsById.end()) {
-		m_itemsById.modify(id_it, bind(&Item::pageInfo, _1) = page_info);
+		m_itemsById.modify(id_it, boost::lambda::bind(&Item::pageInfo, boost::lambda::_1) = page_info);
 		invalidateThumbnailImpl(id_it);
 	}
 }
@@ -723,10 +723,10 @@
 	// Sort pages in m_itemsInOrder using m_ptrOrderProvider.
 	if (m_ptrOrderProvider.get()) {
 		m_itemsInOrder.sort(
-			bind(
+			boost::lambda::bind(
 				&PageOrderProvider::precedes, m_ptrOrderProvider.get(),
-				bind(&Item::pageId, _1), bind(&Item::incompleteThumbnail, _1),
-				bind(&Item::pageId, _2), bind(&Item::incompleteThumbnail, _2) 
+				boost::lambda::bind(&Item::pageId, boost::lambda::_1), boost::lambda::bind(&Item::incompleteThumbnail, boost::lambda::_1),
+				boost::lambda::bind(&Item::pageId, boost::lambda::_2), boost::lambda::bind(&Item::incompleteThumbnail, boost::lambda::_2) 
 			)
 		);
 	}
