From 3b414306b0e4f8531cbf0bb8cd6d763fa21ed88b Mon Sep 17 00:00:00 2001
From: Guido Scholz <gscholz@users.sourceforge.net>
Date: Wed, 7 Nov 2018 22:35:31 +0100
Subject: [PATCH 11/11] Remove obsolete typecast

---
 src/m_vocoder.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/m_vocoder.cpp b/src/m_vocoder.cpp
index 371e2cf..9d2ec8b 100644
--- a/src/m_vocoder.cpp
+++ b/src/m_vocoder.cpp
@@ -166,7 +166,7 @@ M_vocoder::M_vocoder(QWidget* parent, int id)
 
   whichwin = 0;
   window = (float *) malloc (sizeof (float) * fftsize);
-  for (l2 = 0; l2 < (unsigned int) fftsize; l2++)
+  for (l2 = 0; l2 < fftsize; l2++)
     window[l2] = windowcurve (whichwin, fftsize, l2, 0.25);
 
   //  FFTW setup stuff
-- 
2.21.0

