diff --git a/platformplugin/dplatformbackingstore.cpp b/platformplugin/dplatformbackingstore.cpp
index 7f0db30..31cefef 100644
--- a/platformplugin/dplatformbackingstore.cpp
+++ b/platformplugin/dplatformbackingstore.cpp
@@ -32,7 +32,13 @@
 #include <QtGui/QOpenGLShaderProgram>
 #include <QtGui/QOpenGLContext>
 #include <QtGui/QOpenGLFunctions>
+
+#if QT_VERSION < QT_VERSION_CHECK(5, 9, 0)
 #include <QtGui/private/qopengltextureblitter_p.h>
+#else
+#include <QtGui/qopengltextureblitter.h>
+#endif
+
 #include <QtGui/qpa/qplatformgraphicsbuffer.h>
 #include <QtGui/qpa/qplatformgraphicsbufferhelper.h>
 #endif
@@ -804,7 +810,11 @@ void DPlatformBackingStore::composeAndFlushHelper(QWindow *window, const QRegion
 
     if (textureId) {
         if (m_needsSwizzle)
+#if QT_VERSION < QT_VERSION_CHECK(5, 9, 0)
             m_blitter->setSwizzleRB(true);
+#else
+            m_blitter->setRedBlueSwizzle(true);
+#endif
         // The backingstore is for the entire tlw.
         // In case of native children offset tells the position relative to the tlw.
         // reverse translate backward for widgets 
@@ -814,7 +824,11 @@ void DPlatformBackingStore::composeAndFlushHelper(QWindow *window, const QRegion
                                                                          origin);
         m_blitter->blit(textureId, QMatrix4x4(), source);
         if (m_needsSwizzle)
+#if QT_VERSION < QT_VERSION_CHECK(5, 9, 0)
             m_blitter->setSwizzleRB(false);
+#else
+            m_blitter->setRedBlueSwizzle(false);
+#endif
     }
 
     // Textures for renderToTexture widgets that have WA_AlwaysStackOnTop set.
