diff --git a/SConstruct b/SConstruct
index 81d54a4..66cc632 100755
--- a/SConstruct
+++ b/SConstruct
@@ -661,10 +661,7 @@ if conf.env['IS_CYGWIN']:
 else:
     conf.env.Append(CCFLAGS=['-fPIC'])
 
-# check _mm_crc32_u64 (SSE4.2) support:
-conf.check_mm_crc32_u64()
-if conf.env['HAVE_MM_CRC32_U64']:
-    conf.env.Append(CCFLAGS=['-msse4.2'])
+conf.env['HAVE_MM_CRC32_U64'] = False
 
 if 'clang' in os.path.basename(conf.env['CC']):
     conf.env.Append(CCFLAGS=['-fcolor-diagnostics'])  # Colored warnings
