From 3dc0338e4a36a3092720ebaa5b908ba3dca467d9 Mon Sep 17 00:00:00 2001
From: Young Xiao <YangX92@hotmail.com>
Date: Sat, 16 Mar 2019 17:27:18 +0800
Subject: [PATCH 1/3] SWFShape_setLeftFillStyle: prevent fill overflow

---
 src/blocks/shape.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/blocks/shape.c b/src/blocks/shape.c
index 67c81dcf..cfab7bca 100644
--- a/src/blocks/shape.c
+++ b/src/blocks/shape.c
@@ -1169,6 +1169,11 @@ SWFShape_setLeftFillStyle(SWFShape shape, SWFFillStyle fill)
 			return;		
 		idx = getFillIdx(shape, fill);
 	}
+	else if (idx >= 255 && shape->useVersion == SWF_SHAPE1)
+	{
+		SWF_error("Too many fills for SWFShape V1.\n" 
+			  "Use a higher SWFShape version\n");
+	}
 				
 	record = addStyleRecord(shape);
 	record.record.stateChange->leftFill = idx;

From 2be22fcf56a223dafe8de0e8a20fe20e8bbdb0b9 Mon Sep 17 00:00:00 2001
From: Young Xiao <YangX92@hotmail.com>
Date: Sat, 16 Mar 2019 17:42:14 +0800
Subject: [PATCH 2/3] decompileAction: Prevent heap buffer overflow and
 underflow with using OpCode

---
 util/decompile.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/util/decompile.c b/util/decompile.c
index e9341356..1df56da5 100644
--- a/util/decompile.c
+++ b/util/decompile.c
@@ -3202,7 +3202,6 @@ decompileCAST(int n, SWF_ACTION *actions, int maxn)
 int
 decompileAction(int n, SWF_ACTION *actions, int maxn)
 {
-	if( n > maxn ) SWF_error("Action overflow!!");
 
 #ifdef DEBUG
 	fprintf(stderr,"%d:\tACTION[%3.3d]: %s\n",
@@ -3210,7 +3209,7 @@ decompileAction(int n, SWF_ACTION *actions, int maxn)
 	        actionName(actions[n].SWF_ACTIONRECORD.ActionCode));
 #endif
 
-	switch(actions[n].SWF_ACTIONRECORD.ActionCode)
+	switch(OpCode(actions, n, maxn))
 	{
 	case SWFACTION_END:
 		return 0;

From 2223f7a1e431455a1411bee77c90db94a6f8e8fe Mon Sep 17 00:00:00 2001
From: Young Xiao <YangX92@hotmail.com>
Date: Sat, 16 Mar 2019 18:50:24 +0800
Subject: [PATCH 3/3] Fix left shift of a negative value in SWFInput_readSBits.
 Check for number before before left-shifting by (number-1).

---
 src/blocks/input.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/blocks/input.c b/src/blocks/input.c
index 478a0fe1..6e4a8320 100644
--- a/src/blocks/input.c
+++ b/src/blocks/input.c
@@ -113,7 +113,7 @@ SWFInput_readSBits(SWFInput input, int number)
 {
 	int num = SWFInput_readBits(input, number);
 
-	if ( num & (1<<(number-1)) )
+	if(number && num & (1<<(number-1)))
 		return num - (1<<number);
 	else
 		return num;
