diff -Naur old/desktop-widgets/groupedlineedit.cpp new/desktop-widgets/groupedlineedit.cpp
--- old/desktop-widgets/groupedlineedit.cpp	2020-06-12 09:44:56.000000000 -1000
+++ new/desktop-widgets/groupedlineedit.cpp	2020-06-12 15:56:58.558388952 -1000
@@ -32,6 +32,7 @@
 #include <QScrollBar>
 #include <QTextBlock>
 #include <QPainter>
+#include <QPainterPath>
 #include <QApplication>
 #include <QStyle>
 #include <QStyleOptionFocusRect>
