--- src/module_list.py.orig	2015-06-07 11:29:30.710612146 +0200
+++ src/module_list.py	2015-06-07 11:30:21.877506106 +0200
@@ -685,11 +685,11 @@
               include_dirs = [SAGE_INC + '/ecl'],
               depends = [SAGE_INC + '/ecl/ecl.h']),
 
-    OptionalExtension("sage.libs.fes",
-             ["sage/libs/fes.pyx"],
-             language = "c",
-             libraries = ['fes'],
-             package = 'fes'),
+#    OptionalExtension("sage.libs.fes",
+#             ["sage/libs/fes.pyx"],
+#             language = "c",
+#             libraries = ['fes'],
+#             package = 'fes'),
 
     Extension('sage.libs.flint.flint',
               sources = ["sage/libs/flint/flint.pyx"],
