--- sage-7.4/src/sage/symbolic/constants_c.pyx.orig	2016-11-25 09:34:32.665044510 +0000
+++ sage-7.4/src/sage/symbolic/constants_c.pyx	2016-11-25 09:38:13.777888000 +0000
@@ -60,6 +60,8 @@
             self.pointer = <GConstant *>&g_Catalan
         elif self._name == "euler_gamma":
             self.pointer = <GConstant *>&g_Euler
+        elif self._name == "NaN":
+            self.pointer = <GConstant *>&g_NaN
         else:
             GConstant_construct(&self.object, name, texname, domain)
             self.pointer = &self.object
--- sage-7.4/src/sage/symbolic/ginac.pxd.orig	2016-11-25 09:34:06.145103155 +0000
+++ sage-7.4/src/sage/symbolic/ginac.pxd	2016-11-25 09:37:14.294686607 +0000
@@ -216,6 +216,7 @@
     GEx g_Pi "Pi"
     GEx g_Catalan "Catalan"
     GEx g_Euler "Euler"
+    GEx g_NaN "NaN"
 
     GConstant* GConstant_construct(void *mem, char* name, char* texname, unsigned domain)
     bint is_a_constant "is_a<constant>" (GEx e)
@@ -293,13 +294,13 @@
     void g_list_symbols "list_symbols" (GEx e, GExSet s)
 
     # more is_a tests
-    bint is_a_add "is_a<add>" (GEx e)
-    bint is_a_mul "is_a<mul>" (GEx e)
-    bint is_a_power "is_a<power>" (GEx e)
-    bint is_a_fderivative "is_a<fderivative>" (GEx e)
-    bint is_a_function "is_a<function>" (GEx e)
-    bint is_exactly_a_function "is_exactly_a<function>" (GEx e)
-    bint is_a_ncmul "is_a<ncmul>" (GEx e)
+    bint is_a_add "is_a<GiNaC::add>" (GEx e)
+    bint is_a_mul "is_a<GiNaC::mul>" (GEx e)
+    bint is_a_power "is_a<GiNaC::power>" (GEx e)
+    bint is_a_fderivative "is_a<GiNaC::fderivative>" (GEx e)
+    bint is_a_function "is_a<GiNaC::function>" (GEx e)
+    bint is_exactly_a_function "is_exactly_a<GiNaC::function>" (GEx e)
+    bint is_a_ncmul "is_a<GiNaC::ncmul>" (GEx e)
 
     # Arithmetic
     int ginac_error()
