diff --git a/src/sage/databases/cremona.py b/src/sage/databases/cremona.py
index 3f8dd62..ed5726a 100644
--- a/src/sage/databases/cremona.py
+++ b/src/sage/databases/cremona.py
@@ -53,7 +53,7 @@ from sage.misc.prandom import randint
 
 import sage.schemes.elliptic_curves.constructor as elliptic
 from .sql_db import SQLDatabase, verify_column
-from sage.misc.package import is_package_installed
+from sage.features.databases import DatabaseCremona
 from sage.env import CREMONA_MINI_DATA_DIR, CREMONA_LARGE_DATA_DIR
 from sage.misc.all import walltime
 
@@ -835,7 +835,7 @@ class MiniCremonaDatabase(SQLDatabase):
             if N < self.largest_conductor():
                 message = "There is no elliptic curve with label " + label \
                     + " in the database"
-            elif is_package_installed('database_cremona_ellcurve'):
+            elif DatabaseCremona().is_present():
                 message = "There is no elliptic curve with label " + label \
                     + " in the currently available databases"
             else:
@@ -1683,7 +1683,7 @@ def CremonaDatabase(name=None,mini=None,set_global=None):
     if name is None and not set_global:
         return _db
     if set_global and name is None:
-        if is_package_installed('database_cremona_ellcurve'):
+        if DatabaseCremona().is_present():
             name = 'cremona'
         else:
             name = 'cremona mini'
