diff --git a/src/sage/libs/gap/gap_includes.pxd b/src/sage/libs/gap/gap_includes.pxd
index 864a13b385..8a2c22ed4a 100644
--- a/src/sage/libs/gap/gap_includes.pxd
+++ b/src/sage/libs/gap/gap_includes.pxd
@@ -77,8 +77,8 @@ cdef extern from "<gap/io.h>":
 
 cdef extern from "<gap/libgap-api.h>":
     ctypedef void (*CallbackFunc)()
-    void GAP_Initialize(int argc, char ** argv, char ** env,
-        CallbackFunc, CallbackFunc)
+    void GAP_Initialize(int argc, char ** argv,
+        CallbackFunc, CallbackFunc, int handleSignals)
     Obj GAP_EvalString(const char *) except *
     Obj GAP_EvalStringNoExcept "GAP_EvalString"(const char *)
     Obj GAP_ValueGlobalVariable(const char *)
diff --git a/src/sage/libs/gap/util.pyx b/src/sage/libs/gap/util.pyx
index 97c383fdbe..20ecc82976 100644
--- a/src/sage/libs/gap/util.pyx
+++ b/src/sage/libs/gap/util.pyx
@@ -320,8 +320,8 @@
             # Need to save/restore current SIGINT handling since GAP_Initialize
             # currently clobbers it; it doesn't matter what we set SIGINT to
             # temporarily.
-            GAP_Initialize(argc, argv, env, &gasman_callback,
-                           &error_handler)
+            GAP_Initialize(argc, argv, &gasman_callback,
+                           &error_handler, 1)
     except RuntimeError as msg:
         raise RuntimeError('libGAP initialization failed\n' + msg)
     finally:
 
