diff -wbBur synfigstudio-1.0.2/src/gui/canvasview.cpp synfigstudio-1.0.2.my/src/gui/canvasview.cpp
--- synfigstudio-1.0.2/src/gui/canvasview.cpp	2015-07-21 13:01:51.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/canvasview.cpp	2015-10-12 14:54:47.342272650 +0300
@@ -92,7 +92,7 @@
 //#include <synfigapp/action_deletewaypoint.h>
 
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
+//#include <sigc++/retype.h>
 //#include <sigc++/hide.h>
 
 #include "canvasview.h"
diff -wbBur synfigstudio-1.0.2/src/gui/compview.cpp synfigstudio-1.0.2.my/src/gui/compview.cpp
--- synfigstudio-1.0.2/src/gui/compview.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/compview.cpp	2015-10-12 14:56:56.132271169 +0300
@@ -39,7 +39,7 @@
 #include <iostream>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "canvasview.h"
 #include <synfigapp/action.h>
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dockable.cpp synfigstudio-1.0.2.my/src/gui/docks/dockable.cpp
--- synfigstudio-1.0.2/src/gui/docks/dockable.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dockable.cpp	2015-10-12 14:57:19.102270906 +0300
@@ -30,7 +30,7 @@
 #endif
 
 #include "app.h"
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 
 #include "docks/dockable.h"
 #include "docks/dockmanager.h"
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dock_children.cpp synfigstudio-1.0.2.my/src/gui/docks/dock_children.cpp
--- synfigstudio-1.0.2/src/gui/docks/dock_children.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dock_children.cpp	2015-10-12 14:57:15.132270952 +0300
@@ -37,7 +37,7 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/retype_return.h>
 #include <sigc++/slot.h>
 #include "trees/childrentreestore.h"
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dock_curves.cpp synfigstudio-1.0.2.my/src/gui/docks/dock_curves.cpp
--- synfigstudio-1.0.2/src/gui/docks/dock_curves.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dock_curves.cpp	2015-10-12 14:57:09.835604346 +0300
@@ -36,7 +36,7 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "canvasview.h"
 #include "trees/layerparamtreestore.h"
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dockdialog.cpp synfigstudio-1.0.2.my/src/gui/docks/dockdialog.cpp
--- synfigstudio-1.0.2/src/gui/docks/dockdialog.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dockdialog.cpp	2015-10-12 14:57:12.995604310 +0300
@@ -42,10 +42,10 @@
 #include <synfig/general.h>
 #include <synfig/uniqueid.h>
 #include <gtkmm/table.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
+//#include <sigc++/retype.h>
 #include "canvasview.h"
 #include <gtkmm/paned.h>
 #include <gtkmm/box.h>
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dock_history.cpp synfigstudio-1.0.2.my/src/gui/docks/dock_history.cpp
--- synfigstudio-1.0.2/src/gui/docks/dock_history.cpp	2015-07-21 13:01:51.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dock_history.cpp	2015-10-12 14:57:25.388937500 +0300
@@ -39,7 +39,7 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include <synfigapp/action.h>
 #include "trees/historytreestore.h"
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dock_keyframes.cpp synfigstudio-1.0.2.my/src/gui/docks/dock_keyframes.cpp
--- synfigstudio-1.0.2/src/gui/docks/dock_keyframes.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dock_keyframes.cpp	2015-10-12 14:57:03.548937751 +0300
@@ -36,7 +36,7 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "trees/keyframetreestore.h"
 #include "trees/keyframetree.h"
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dock_layergroups.cpp synfigstudio-1.0.2.my/src/gui/docks/dock_layergroups.cpp
--- synfigstudio-1.0.2/src/gui/docks/dock_layergroups.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dock_layergroups.cpp	2015-10-12 14:57:17.102270929 +0300
@@ -36,7 +36,7 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/retype_return.h>
 #include <sigc++/slot.h>
 #include "canvasview.h"
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dock_layers.cpp synfigstudio-1.0.2.my/src/gui/docks/dock_layers.cpp
--- synfigstudio-1.0.2/src/gui/docks/dock_layers.cpp	2015-07-21 13:01:51.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dock_layers.cpp	2015-10-12 14:57:01.478937774 +0300
@@ -39,7 +39,7 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/retype_return.h>
 #include <sigc++/slot.h>
 #include "trees/layertreestore.h"
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dockmanager.h synfigstudio-1.0.2.my/src/gui/docks/dockmanager.h
--- synfigstudio-1.0.2/src/gui/docks/dockmanager.h	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dockmanager.h	2015-10-12 14:35:08.685619532 +0300
@@ -32,7 +32,7 @@
 #include <map>
 #include <synfig/string.h>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 #include <ETL/smart_ptr>
 
 #include <gtkmm/widget.h>
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dock_metadata.cpp synfigstudio-1.0.2.my/src/gui/docks/dock_metadata.cpp
--- synfigstudio-1.0.2/src/gui/docks/dock_metadata.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dock_metadata.cpp	2015-10-12 14:57:05.435604395 +0300
@@ -37,7 +37,7 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "trees/metadatatreestore.h"
 #include "trees/metadatatree.h"
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dock_params.cpp synfigstudio-1.0.2.my/src/gui/docks/dock_params.cpp
--- synfigstudio-1.0.2/src/gui/docks/dock_params.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dock_params.cpp	2015-10-12 14:57:21.325604213 +0300
@@ -36,7 +36,7 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "canvasview.h"
 #include "trees/layerparamtreestore.h"
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dock_timetrack.cpp synfigstudio-1.0.2.my/src/gui/docks/dock_timetrack.cpp
--- synfigstudio-1.0.2/src/gui/docks/dock_timetrack.cpp	2015-08-17 19:28:00.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dock_timetrack.cpp	2015-10-12 14:57:23.445604190 +0300
@@ -38,7 +38,7 @@
 #include <cassert>
 #include "instance.h"
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include "canvasview.h"
 #include "trees/layerparamtreestore.h"
diff -wbBur synfigstudio-1.0.2/src/gui/docks/dock_toolbox.cpp synfigstudio-1.0.2.my/src/gui/docks/dock_toolbox.cpp
--- synfigstudio-1.0.2/src/gui/docks/dock_toolbox.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/docks/dock_toolbox.cpp	2015-10-12 14:57:07.748937703 +0300
@@ -38,10 +38,10 @@
 #include <gtkmm/accelmap.h>
 
 #include <sigc++/signal.h>
-#include <sigc++/hide.h>
+//#include <sigc++/hide.h>
 #include <sigc++/slot.h>
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
+//#include <sigc++/retype.h>
 
 #include "general.h"
 #include <sstream>
diff -wbBur synfigstudio-1.0.2/src/gui/duck.h synfigstudio-1.0.2.my/src/gui/duck.h
--- synfigstudio-1.0.2/src/gui/duck.h	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/duck.h	2015-10-12 14:34:43.148953160 +0300
@@ -38,7 +38,7 @@
 #include <synfig/string.h>
 #include <synfig/real.h>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 #include <synfig/time.h>
 #include <ETL/smart_ptr>
 #include <synfigapp/value_desc.h>
diff -wbBur synfigstudio-1.0.2/src/gui/duckmatic.cpp synfigstudio-1.0.2.my/src/gui/duckmatic.cpp
--- synfigstudio-1.0.2/src/gui/duckmatic.cpp	2015-07-21 13:01:51.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/duckmatic.cpp	2015-10-12 14:56:59.172271135 +0300
@@ -68,8 +68,8 @@
 #include <synfig/layers/layer_pastecanvas.h>
 
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
-#include <sigc++/hide.h>
+//#include <sigc++/retype.h>
+//#include <sigc++/hide.h>
 #include <sigc++/bind.h>
 
 #include "ducktransform_matrix.h"
diff -wbBur synfigstudio-1.0.2/src/gui/duckmatic.h synfigstudio-1.0.2.my/src/gui/duckmatic.h
--- synfigstudio-1.0.2/src/gui/duckmatic.h	2015-07-21 13:01:51.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/duckmatic.h	2015-10-12 14:34:38.638953211 +0300
@@ -40,7 +40,7 @@
 #include <synfig/string.h>
 #include <synfig/real.h>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 #include <synfig/time.h>
 #include <synfig/color.h>
 #include <ETL/smart_ptr>
diff -wbBur synfigstudio-1.0.2/src/gui/instance.cpp synfigstudio-1.0.2.my/src/gui/instance.cpp
--- synfigstudio-1.0.2/src/gui/instance.cpp	2015-07-21 13:01:51.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/instance.cpp	2015-10-12 14:54:42.525606039 +0300
@@ -54,7 +54,7 @@
 
 #include "autorecover.h"
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
+//#include <sigc++/retype.h>
 //#include <sigc++/hide.h>
 #include <synfig/valuenodes/valuenode_composite.h>
 #include <synfig/valuenodes/valuenode_duplicate.h>
diff -wbBur synfigstudio-1.0.2/src/gui/instance.h synfigstudio-1.0.2.my/src/gui/instance.h
--- synfigstudio-1.0.2/src/gui/instance.h	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/instance.h	2015-10-12 14:34:41.015619850 +0300
@@ -33,7 +33,7 @@
 #include <gtkmm/treeview.h>
 #include <gtkmm/treestore.h>
 #include <synfigapp/instance.h>
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 #include <synfigapp/value_desc.h>
 #include "trees/historytreestore.h"
 #include <synfig/canvas.h>
diff -wbBur synfigstudio-1.0.2/src/gui/states/state_stroke.h synfigstudio-1.0.2.my/src/gui/states/state_stroke.h
--- synfigstudio-1.0.2/src/gui/states/state_stroke.h	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/states/state_stroke.h	2015-10-12 14:35:03.862286254 +0300
@@ -29,7 +29,7 @@
 
 #include "canvasview.h"
 #include "workarea.h"
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 #include "duckmatic.h"
 #include <synfig/blinepoint.h>
 #include <list>
diff -wbBur synfigstudio-1.0.2/src/gui/states/state_zoom.cpp synfigstudio-1.0.2.my/src/gui/states/state_zoom.cpp
--- synfigstudio-1.0.2/src/gui/states/state_zoom.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/states/state_zoom.cpp	2015-10-12 14:58:59.892269747 +0300
@@ -31,7 +31,7 @@
 #endif
 
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 
 #include <ETL/handle>
 #include <synfig/vector.h>
diff -wbBur synfigstudio-1.0.2/src/gui/widgets/widget_curves.cpp synfigstudio-1.0.2.my/src/gui/widgets/widget_curves.cpp
--- synfigstudio-1.0.2/src/gui/widgets/widget_curves.cpp	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/widgets/widget_curves.cpp	2015-10-12 14:58:57.915603103 +0300
@@ -38,7 +38,7 @@
 #include <map>
 #include <vector>
 #include <ETL/misc>
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 
 #include "general.h"
 
diff -wbBur synfigstudio-1.0.2/src/gui/workarea.cpp synfigstudio-1.0.2.my/src/gui/workarea.cpp
--- synfigstudio-1.0.2/src/gui/workarea.cpp	2015-07-21 13:01:51.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/workarea.cpp	2015-10-12 14:56:52.485604544 +0300
@@ -45,8 +45,8 @@
 #include <gtkmm/scrollbar.h>
 #include <cmath>
 #include <sigc++/retype_return.h>
-#include <sigc++/retype.h>
-#include <sigc++/hide.h>
+//#include <sigc++/retype.h>
+//#include <sigc++/hide.h>
 #include <ETL/misc>
 
 #include <synfig/target_scanline.h>
diff -wbBur synfigstudio-1.0.2/src/gui/workarearenderer/workarearenderer.h synfigstudio-1.0.2.my/src/gui/workarearenderer/workarearenderer.h
--- synfigstudio-1.0.2/src/gui/workarearenderer/workarearenderer.h	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/gui/workarearenderer/workarearenderer.h	2015-10-12 14:35:06.422286225 +0300
@@ -29,7 +29,7 @@
 
 #include <ETL/handle>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 #include <synfig/vector.h>
 #include <gdkmm/window.h>
 #include <gdkmm/rectangle.h>
diff -wbBur synfigstudio-1.0.2/src/synfigapp/action_system.h synfigstudio-1.0.2.my/src/synfigapp/action_system.h
--- synfigstudio-1.0.2/src/synfigapp/action_system.h	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/synfigapp/action_system.h	2015-10-12 14:33:56.655620360 +0300
@@ -29,7 +29,7 @@
 
 #include "action.h"
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 #include <ETL/handle>
 #include <synfig/canvas.h>
 #include "uimanager.h"
diff -wbBur synfigstudio-1.0.2/src/synfigapp/canvasinterface.h synfigstudio-1.0.2.my/src/synfigapp/canvasinterface.h
--- synfigstudio-1.0.2/src/synfigapp/canvasinterface.h	2015-07-21 13:01:51.000000000 +0300
+++ synfigstudio-1.0.2.my/src/synfigapp/canvasinterface.h	2015-10-12 14:34:22.262286734 +0300
@@ -31,7 +31,7 @@
 //#include <synfig/canvas.h>
 #include <synfig/value.h>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 #include <list>
 #include "selectionmanager.h"
 #include "uimanager.h"
diff -wbBur synfigstudio-1.0.2/src/synfigapp/instance.h synfigstudio-1.0.2.my/src/synfigapp/instance.h
--- synfigstudio-1.0.2/src/synfigapp/instance.h	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/synfigapp/instance.h	2015-10-12 14:34:52.485619719 +0300
@@ -36,7 +36,7 @@
 #include <list>
 #include <set>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 #include "action_system.h"
 #include "selectionmanager.h"
 #include "cvs.h"
diff -wbBur synfigstudio-1.0.2/src/synfigapp/main.h synfigstudio-1.0.2.my/src/synfigapp/main.h
--- synfigstudio-1.0.2/src/synfigapp/main.h	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/synfigapp/main.h	2015-10-12 14:34:59.175619642 +0300
@@ -35,7 +35,7 @@
 #include <synfig/real.h>
 #include <synfig/waypoint.h>
 #include <sigc++/signal.h>
-#include <sigc++/object.h>
+//#include <sigc++/object.h>
 #include "inputdevice.h"
 #include "settings.h"
 
diff -wbBur synfigstudio-1.0.2/src/synfigapp/uimanager.h synfigstudio-1.0.2.my/src/synfigapp/uimanager.h
--- synfigstudio-1.0.2/src/synfigapp/uimanager.h	2015-07-09 10:33:04.000000000 +0300
+++ synfigstudio-1.0.2.my/src/synfigapp/uimanager.h	2015-10-12 14:37:44.075617745 +0300
@@ -30,7 +30,7 @@
 #include <ETL/handle>
 #include <synfig/general.h>
 #include <synfig/string.h>
-#include <sigc++/object.h>
+#include <sigc++/sigc++.h>
 
 /* === M A C R O S ========================================================= */
 
