diff -Nur cinelerra.old/configure.in cinelerra/configure.in
--- cinelerra.old/configure.in	2010-01-12 16:38:50.000000000 +0200
+++ cinelerra/configure.in	2010-01-12 16:41:55.000000000 +0200
@@ -269,12 +269,7 @@
 
 ############## libx264
 LIBX264_LIBS=""
-AC_CHECK_LIB(x264, x264_encoder_close,
-	[libx264=yes LIBX264_LIBS=-lx264],,-lpthread -lm)
-if test "x$with_pic" = "xyes"; then
-	AC_CHECK_LIB(x264_pic, x264_encoder_open,
-		[libx264=yes LIBX264_LIBS=-lx264_pic],,-lpthread -lm)
-fi
+PKG_CHECK_MODULES([LIBX264], [x264], [libx264=yes], [libx264=no])
 AC_CHECK_HEADER(x264.h, [libx264h=yes])
 AC_SUBST(LIBX264_LIBS)
 ##############
