diff --git a/src/code/modules/image2text/src/ocs.cpp b/src/code/modules/image2text/src/ocs.cpp
index 386dd84..624075e 100644
--- a/src/code/modules/image2text/src/ocs.cpp
+++ b/src/code/modules/image2text/src/ocs.cpp
@@ -4,6 +4,7 @@
 
 #include <tesseract/baseapi.h>
 #include <leptonica/allheaders.h>
+#include <leptonica/pix_internal.h>
 #include "OCRLanguageModel.h"
 #if TESSERACT_MAJOR_VERSION < 5
 #include <tesseract/strngs.h>
