From 0199a7ee717f0848e336fda8dc1f5b9364ce3478 Mon Sep 17 00:00:00 2001
From: Allan McRae <allan@archlinux.org>
Date: Sun, 15 Nov 2009 20:24:24 +1000
Subject: [PATCH 01/26] contrib/bacman: fix checking if file has been added

Fixes FS#17140.

Signed-off-by: Allan McRae <allan@archlinux.org>
Signed-off-by: Dan McGee <dan@archlinux.org>
---
 contrib/bacman |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/contrib/bacman b/contrib/bacman
index 22940ec..dfd53e8 100755
--- a/contrib/bacman
+++ b/contrib/bacman
@@ -21,7 +21,7 @@
 #
 
 readonly progname="bacman"
-readonly progver="0.2.0"
+readonly progver="0.2.1"
 
 #
 # User Friendliness
@@ -137,7 +137,7 @@ while read i; do
                 bsdtar -cnf - "/$i" 2> /dev/null | bsdtar -xpf -
 
                 # Workaround to bsdtar not reporting a missing file as an error
-                if [ ! -e "$work_dir/$i" ] && [ -L "$work_dir/$i"]; then
+                if [ ! -e "$work_dir/$i" -a ! -L "$work_dir/$i" ]; then
                     echo ""
                     echo "ERROR: unable to add /$i to the package"
                     echo "       If your user does not have permssion to read this file then"
-- 
1.6.5.3

