diff --git a/tests/integration-test.py b/tests/integration-test.py
index b03e6b9..a3905d2 100755
--- a/tests/integration-test
+++ b/tests/integration-test
@@ -592,7 +592,6 @@ class Tests(dbusmock.DBusTestCase):
             []
         )
         env = os.environ.copy()
-        env['LC_NUMERIC'] = 'fr_FR.UTF-8'
         self.start_daemon(env=env)
 
         self.assertEqual(self.get_dbus_property('HasAccelerometer'), True)
@@ -601,9 +600,8 @@ class Tests(dbusmock.DBusTestCase):
             mock_file.write(data)
         self.proxy.ClaimAccelerometer()
         self.assertEventually(lambda: self.have_text_in_log('Accel sent by driver'))
-        # If the 2nd test fails, it's likely that fr_FR.UTF-8 locale isn't supported
-        self.assertEqual(self.have_text_in_log('scale: 0,000000,0,000000,0,000000'), False)
-        self.assertEqual(self.have_text_in_log('scale: 0,000010,0,000010,0,000010'), True)
+        self.assertEqual(self.have_text_in_log('scale: 0.000000,0.000000,0.000000'), False)
+        self.assertEqual(self.have_text_in_log('scale: 0.000010,0.000010,0.000010'), True)
 
         self.stop_daemon()
 
@@ -659,14 +657,12 @@ class Tests(dbusmock.DBusTestCase):
              'IIO_SENSOR_PROXY_TYPE', 'iio-poll-accel']
         )
         env = os.environ.copy()
-        env['LC_NUMERIC'] = 'fr_FR.UTF-8'
         self.start_daemon(env=env)
 
         self.proxy.ClaimAccelerometer()
         self.assertEventually(lambda: self.have_text_in_log('Accel read from IIO on'))
-        # If the 2nd test fails, it's likely that fr_FR.UTF-8 locale isn't supported
-        self.assertEqual(self.have_text_in_log('scale 1,000000,1,000000,1,000000'), False)
-        self.assertEqual(self.have_text_in_log('scale 0,000001,0,000001,0,000001'), True)
+        self.assertEqual(self.have_text_in_log('scale 1.000000,1.000000,1.000000'), False)
+        self.assertEqual(self.have_text_in_log('scale 0.000001,0.000001,0.000001'), True)
 
         self.assertEventually(lambda: self.get_dbus_property('AccelerometerOrientation') == 'normal')
 
