diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in	2020-10-02 11:02:02.400414722 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in	2020-10-02 11:02:31.870972519 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in	2020-10-02 11:02:47.074593828 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in	2020-10-02 11:02:53.631384734 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in	2020-10-02 11:02:59.278158416 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in	2020-10-02 11:03:06.771633795 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in	2020-10-02 11:03:11.888397474 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in	2020-10-02 11:03:17.211831753 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_6.0_to_7.0.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_6.0_to_7.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_6.0_to_7.0.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_6.0_to_7.0.sh.in	2020-10-02 11:03:22.781937394 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_7.0_to_8.0.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_7.0_to_8.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_7.0_to_8.0.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_7.0_to_8.0.sh.in	2020-10-02 11:03:27.845366775 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_8.0_to_8.1.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_8.0_to_8.1.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_8.0_to_8.1.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_8.0_to_8.1.sh.in	2020-10-02 11:03:41.438958033 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_8.1_to_8.2.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_8.1_to_8.2.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_8.1_to_8.2.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_8.1_to_8.2.sh.in	2020-10-02 11:03:50.945805145 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_8.2_to_9.0.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_8.2_to_9.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_8.2_to_9.0.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_8.2_to_9.0.sh.in	2020-10-02 11:04:02.296020643 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_9.0_to_9.1.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_9.0_to_9.1.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_9.0_to_9.1.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_9.0_to_9.1.sh.in	2020-10-02 11:04:08.132798154 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_9.1_to_9.2.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_9.1_to_9.2.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_9.1_to_9.2.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_9.1_to_9.2.sh.in	2020-10-02 11:04:13.879573974 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_9.2_to_9.3.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_9.2_to_9.3.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/upgrade_9.2_to_9.3.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/upgrade_9.2_to_9.3.sh.in	2020-10-02 11:04:25.629797209 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/mysql/wipe_data.sh.in kea-1.8.0-new/src/share/database/scripts/mysql/wipe_data.sh.in
--- kea-1.8.0-old/src/share/database/scripts/mysql/wipe_data.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/mysql/wipe_data.sh.in	2020-10-02 11:05:55.311503213 +0300
@@ -12,14 +12,8 @@
 # including leases, reservations, etc... Use at your own peril.
 # Reference tables will be left in-tact.
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 # First argument is must be the expected schema version <major>.<minor>
 exp_version="$1"
diff -aur --color kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in	2020-10-02 11:04:39.893401621 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in	2020-10-02 11:00:03.148163954 +0300
+++ kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in	2020-10-02 11:04:45.243503326 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in
--- kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in	2020-10-02 11:00:03.148163954 +0300
+++ kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in	2020-10-02 11:04:50.880277161 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in
--- kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in	2020-10-02 11:04:55.337028573 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in
--- kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in	2020-10-02 11:05:00.917134693 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in	2020-10-02 11:00:03.151497350 +0300
+++ kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in	2020-10-02 11:05:09.357295228 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_4.0_to_5.0.sh.in kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_4.0_to_5.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_4.0_to_5.0.sh.in	2020-10-02 11:00:03.148163954 +0300
+++ kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_4.0_to_5.0.sh.in	2020-10-02 11:05:14.294055811 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_5.0_to_5.1.sh.in kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_5.0_to_5.1.sh.in
--- kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_5.0_to_5.1.sh.in	2020-10-02 11:00:03.148163954 +0300
+++ kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_5.0_to_5.1.sh.in	2020-10-02 11:05:19.260816976 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_5.1_to_6.0.sh.in kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_5.1_to_6.0.sh.in
--- kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_5.1_to_6.0.sh.in	2020-10-02 11:00:03.148163954 +0300
+++ kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_5.1_to_6.0.sh.in	2020-10-02 11:05:23.947572823 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_6.0_to_6.1.sh.in kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_6.0_to_6.1.sh.in
--- kea-1.8.0-old/src/share/database/scripts/pgsql/upgrade_6.0_to_6.1.sh.in	2020-10-02 11:00:03.148163954 +0300
+++ kea-1.8.0-new/src/share/database/scripts/pgsql/upgrade_6.0_to_6.1.sh.in	2020-10-02 11:05:29.001002326 +0300
@@ -1,13 +1,7 @@
 #!/bin/sh
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -aur --color kea-1.8.0-old/src/share/database/scripts/pgsql/wipe_data.sh.in kea-1.8.0-new/src/share/database/scripts/pgsql/wipe_data.sh.in
--- kea-1.8.0-old/src/share/database/scripts/pgsql/wipe_data.sh.in	2020-10-02 11:00:03.148163954 +0300
+++ kea-1.8.0-new/src/share/database/scripts/pgsql/wipe_data.sh.in	2020-10-02 11:05:37.991173444 +0300
@@ -12,14 +12,8 @@
 # including leases, reservations, etc... Use at your own peril.
 # Reference tables will be left in-tact.
 
-prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 # First argument is must be the expected schema version <major>.<minor>
 exp_version="$1"
