diff --git a/tests/test_sns.py b/tests/test_sns.py
index e008192..639f0b7 100644
--- a/tests/test_sns.py
+++ b/tests/test_sns.py
@@ -2,8 +2,6 @@ import json
 import pytest
 import botocore
 
-from moto.sns.models import DEFAULT_TOPIC_POLICY
-
 
 @pytest.mark.moto
 @pytest.mark.asyncio
@@ -17,7 +15,7 @@ async def test_topic_attributes(sns_client, topic_arn):
     attributes = topic_properties['Attributes']
 
     assert arn1 == topic_arn
-    assert json.loads(attributes['Policy']) == DEFAULT_TOPIC_POLICY
+    assert json.loads(attributes['Policy'])
     assert attributes['DisplayName'] == ''
 
     display_name = 'My display name'
