diff -Naur spyder-2.3.5.orig/spyderlib/userconfig.py spyder-2.3.5/spyderlib/userconfig.py
--- spyder-2.3.5.orig/spyderlib/userconfig.py	2015-06-22 16:47:56.000000000 +0200
+++ spyder-2.3.5/spyderlib/userconfig.py	2015-07-12 15:17:03.639522116 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 #    userconfig License Agreement (MIT License)
diff -Naur spyder-2.3.5.orig/spyderlib/utils/external/pickleshare.py spyder-2.3.5/spyderlib/utils/external/pickleshare.py
--- spyder-2.3.5.orig/spyderlib/utils/external/pickleshare.py	2015-06-22 16:47:56.000000000 +0200
+++ spyder-2.3.5/spyderlib/utils/external/pickleshare.py	2015-07-12 15:17:18.216188678 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """ PickleShare - a small 'shelve' like datastore with concurrency support
 
diff -Naur spyder-2.3.5.orig/spyderplugins/widgets/pylintgui.py spyder-2.3.5/spyderplugins/widgets/pylintgui.py
--- spyder-2.3.5.orig/spyderplugins/widgets/pylintgui.py	2015-06-22 16:47:56.000000000 +0200
+++ spyder-2.3.5/spyderplugins/widgets/pylintgui.py	2015-07-12 15:16:22.346189082 +0200
@@ -40,7 +40,7 @@
 _ = get_translation("p_pylint", dirname="spyderplugins")
 
 
-PYLINT = 'pylint'
+PYLINT = 'pylint2'
 if PY3:
     if programs.find_program('pylint3'):
         PYLINT = 'pylint3'
@@ -61,7 +61,7 @@
                                shell=True if os.name == 'nt' else False)
     lines = to_unicode_from_fs(process.stdout.read()).splitlines()
     if lines:
-        regex = '({0}*|pylint-script.py) ([0-9\.]*)'.format(PYLINT)
+        regex = '({0}*|pylint2-script.py) ([0-9\.]*)'.format(PYLINT)
         match = re.match(regex, lines[0])
         if match is not None:
             return match.groups()[1]
