diff -Naur spyder-2.3.0/spyderlib/userconfig.py spyder-2.3.0-patched/spyderlib/userconfig.py
--- spyder-2.3.0/spyderlib/userconfig.py	2014-07-10 08:06:40.000000000 +0200
+++ spyder-2.3.0-patched/spyderlib/userconfig.py	2014-08-08 12:30:53.544984165 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 #    userconfig License Agreement (MIT License)
diff -Naur spyder-2.3.0/spyderlib/utils/external/pickleshare.py spyder-2.3.0-patched/spyderlib/utils/external/pickleshare.py
--- spyder-2.3.0/spyderlib/utils/external/pickleshare.py	2014-07-10 08:06:40.000000000 +0200
+++ spyder-2.3.0-patched/spyderlib/utils/external/pickleshare.py	2014-08-08 12:31:00.181651117 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """ PickleShare - a small 'shelve' like datastore with concurrency support
 
