diff -Naur spyder-3.1.1.orig/scripts/spyder spyder-3.1.1/scripts/spyder
--- spyder-3.1.1.orig/scripts/spyder	2017-01-22 18:28:03.000000000 +0100
+++ spyder-3.1.1/scripts/spyder	2017-01-22 19:01:59.364263881 +0100
@@ -1,3 +1,3 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from spyder.app import start
 start.main()
diff -Naur spyder-3.1.1.orig/spyder/app/restart.py spyder-3.1.1/spyder/app/restart.py
--- spyder-3.1.1.orig/spyder/app/restart.py	2017-01-22 18:28:03.000000000 +0100
+++ spyder-3.1.1/spyder/app/restart.py	2017-01-22 19:00:54.574451747 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 #
 # Copyright © Spyder Project Contributors
diff -Naur spyder-3.1.1.orig/spyder_pylint/widgets/pylintgui.py spyder-3.1.1/spyder_pylint/widgets/pylintgui.py
--- spyder-3.1.1.orig/spyder_pylint/widgets/pylintgui.py	2017-01-22 18:28:03.000000000 +0100
+++ spyder-3.1.1/spyder_pylint/widgets/pylintgui.py	2017-01-22 19:05:13.522630971 +0100
@@ -47,7 +47,7 @@
     _ = gettext.gettext
 
 
-PYLINT = 'pylint'
+PYLINT = 'pylint2'
 if PY3:
     if programs.find_program('pylint3'):
         PYLINT = 'pylint3'
