From 0cc65e2f420af45f78d190ac2af64810a148bb31 Mon Sep 17 00:00:00 2001
From: Luis Menina <liberforce@freeside.fr>
Date: Mon, 11 Mar 2013 00:00:44 +0000
Subject: Don't crash on an empty print preview, just provide an empty preview.

https://bugzilla.gnome.org/show_bug.cgi?id=581282
---
diff --git a/src/planner-print-job.c b/src/planner-print-job.c
index 6334037..96267a0 100644
--- a/src/planner-print-job.c
+++ b/src/planner-print-job.c
@@ -237,17 +237,7 @@ print_job_draw_page (GtkPrintOperation *operation,
 
 	l = priv->views;
 
-	/* Abort if there is nothing to print.
-	 *
-	 * A better solution would be to set the number of pages to print to 0
-	 * in print_job_begin_print, but 0 is not a valid value for
-	 * gtk_print_operation_set_n_pages.
-	 */
-	if(!l) {
-		return;
-	}
-
-	while (!page_found) {
+	while (! page_found && l != NULL) {
 		v = PLANNER_VIEW (l->data);
 
 		pages_in_view = planner_view_print_get_n_pages (v);
@@ -258,8 +248,6 @@ print_job_draw_page (GtkPrintOperation *operation,
 		} else {
 			page_nr -= pages_in_view;
 			l = l->next;
-
-			g_assert (l != NULL);
 		}
 	}
 }
--
cgit v0.9.2
