--- pango/modules.c.orig	2010-08-26 06:45:49.329259966 +0200
+++ pango/modules.c	2010-08-26 06:46:13.786685177 +0200
@@ -529,7 +529,7 @@
 
   if (!file_str)
     file_str = g_build_filename (pango_get_sysconf_subdirectory (),
-				 "pango.modules",
+				 "pango.modules-32",
 				 NULL);
 
   files = pango_split_file_list (file_str);
@@ -640,7 +640,7 @@
       if (!no_module_warning)
 	{
 	  gchar *filename = g_build_filename (pango_get_sysconf_subdirectory (),
-					      "pango.modules",
+					      "pango.modules-32",
 					      NULL);
 	  g_critical ("No modules found:\n"
 		      "No builtin or dynamically loaded modules were found.\n"
