diff -ruN a/extras/AudioPerformanceTest/CMakeLists.txt b/extras/AudioPerformanceTest/CMakeLists.txt
--- a/extras/AudioPerformanceTest/CMakeLists.txt	2021-12-08 16:31:22.000000000 +0100
+++ b/extras/AudioPerformanceTest/CMakeLists.txt	2021-12-12 00:06:59.963753084 +0100
@@ -32,6 +32,11 @@
     JUCE_USE_CURL=0 JUCE_WEB_BROWSER=0)
 
 target_link_libraries(AudioPerformanceTest PRIVATE
+    "FLAC"
+    "ogg"
+    "vorbis"
+    "vorbisenc"
+    "vorbisfile"
     juce::juce_audio_utils
     juce::juce_recommended_config_flags
     juce::juce_recommended_lto_flags
diff -ruN a/extras/AudioPluginHost/CMakeLists.txt b/extras/AudioPluginHost/CMakeLists.txt
--- a/extras/AudioPluginHost/CMakeLists.txt	2021-12-08 16:31:22.000000000 +0100
+++ b/extras/AudioPluginHost/CMakeLists.txt	2021-12-12 00:08:03.473979636 +0100
@@ -63,6 +63,10 @@
     JUCE_WEB_BROWSER=0)
 
 target_link_libraries(AudioPluginHost PRIVATE
+    "ogg"
+    "vorbis"
+    "vorbisenc"
+    "vorbisfile"
     AudioPluginHostData
     juce::juce_audio_utils
     juce::juce_cryptography
diff -ruN a/extras/Build/juceaide/CMakeLists.txt b/extras/Build/juceaide/CMakeLists.txt
--- a/extras/Build/juceaide/CMakeLists.txt	2021-12-08 16:31:22.000000000 +0100
+++ b/extras/Build/juceaide/CMakeLists.txt	2021-12-12 00:06:59.970419777 +0100
@@ -37,6 +37,9 @@
         JUCE_USE_CURL=0)
 
     target_link_libraries(juceaide PRIVATE
+        "jpeg"
+        "png16"
+        "z"
         juce::juce_build_tools
         juce::juce_recommended_config_flags
         juce::juce_recommended_lto_flags
diff -ruN a/extras/NetworkGraphicsDemo/CMakeLists.txt b/extras/NetworkGraphicsDemo/CMakeLists.txt
--- a/extras/NetworkGraphicsDemo/CMakeLists.txt	2021-12-08 16:31:22.000000000 +0100
+++ b/extras/NetworkGraphicsDemo/CMakeLists.txt	2021-12-12 00:06:59.963753084 +0100
@@ -31,6 +31,13 @@
     JUCE_USE_CURL=0 JUCE_WEB_BROWSER=0)
 
 target_link_libraries(NetworkGraphicsDemo PRIVATE
+    "FLAC"
+    "ogg"
+    "png"
+    "vorbis"
+    "vorbisenc"
+    "vorbisfile"
+    "z"
     juce::juce_audio_utils
     juce::juce_cryptography
     juce::juce_opengl
diff -ruN a/extras/Projucer/CMakeLists.txt b/extras/Projucer/CMakeLists.txt
--- a/extras/Projucer/CMakeLists.txt	2021-12-08 16:31:22.000000000 +0100
+++ b/extras/Projucer/CMakeLists.txt	2021-12-12 00:06:59.970419777 +0100
@@ -158,6 +158,8 @@
     ../Build/CMake/juce_runtime_arch_detection.cpp)
 
 target_link_libraries(Projucer PRIVATE
+    "png16"
+    "jpeg"
     ProjucerData
     juce::juce_build_tools
     juce::juce_cryptography
diff -ruN a/extras/UnitTestRunner/CMakeLists.txt b/extras/UnitTestRunner/CMakeLists.txt
--- a/extras/UnitTestRunner/CMakeLists.txt	2021-12-08 16:31:22.000000000 +0100
+++ b/extras/UnitTestRunner/CMakeLists.txt	2021-12-12 00:06:59.963753084 +0100
@@ -33,6 +33,13 @@
     JUCE_WEB_BROWSER=0)
 
 target_link_libraries(UnitTestRunner PRIVATE
+    "FLAC"
+    "ogg"
+    "png16"
+    "vorbis"
+    "vorbisenc"
+    "vorbisfile"
+    "z"
     juce::juce_analytics
     juce::juce_audio_utils
     juce::juce_dsp
