diff -ruN a/modules/juce_audio_plugin_client/VST/juce_VST_Wrapper.cpp b/modules/juce_audio_plugin_client/VST/juce_VST_Wrapper.cpp
--- a/modules/juce_audio_plugin_client/VST/juce_VST_Wrapper.cpp	2021-09-20 14:26:04.000000000 +0200
+++ b/modules/juce_audio_plugin_client/VST/juce_VST_Wrapper.cpp	2021-09-26 00:58:02.047887074 +0200
@@ -84,8 +84,7 @@
 // field in the Projucer. The VST2 SDK can be obtained from the
 // vstsdk3610_11_06_2018_build_37 (or older) VST3 SDK or JUCE version 5.3.2. You
 // also need a VST2 license from Steinberg to distribute VST2 plug-ins.
-#include "pluginterfaces/vst2.x/aeffect.h"
-#include "pluginterfaces/vst2.x/aeffectx.h"
+#include "../../juce_audio_processors/format_types/juce_VSTInterface.h"
 }
 
 JUCE_END_IGNORE_WARNINGS_MSVC
diff -ruN a/modules/juce_audio_processors/format_types/juce_VSTPluginFormat.cpp b/modules/juce_audio_processors/format_types/juce_VSTPluginFormat.cpp
--- a/modules/juce_audio_processors/format_types/juce_VSTPluginFormat.cpp	2021-09-20 14:26:04.000000000 +0200
+++ b/modules/juce_audio_processors/format_types/juce_VSTPluginFormat.cpp	2021-09-26 01:03:15.142161924 +0200
@@ -46,8 +46,8 @@
 // paths or use the "VST (Legacy) SDK Folder" field in the Projucer. The VST2
 // SDK can be obtained from the vstsdk3610_11_06_2018_build_37 (or older) VST3
 // SDK or JUCE version 5.3.2.
-#include <pluginterfaces/vst2.x/aeffect.h>
-#include <pluginterfaces/vst2.x/aeffectx.h>
+#define FST2VST 1
+#include <pluginterfaces/fst/fst.h>
 }
 
 #include "juce_VSTCommon.h"
