diff -ruN a/extras/Projucer/Source/Application/jucer_AutoUpdater.cpp b/extras/Projucer/Source/Application/jucer_AutoUpdater.cpp
--- a/extras/Projucer/Source/Application/jucer_AutoUpdater.cpp	2021-09-20 14:26:04.000000000 +0200
+++ b/extras/Projucer/Source/Application/jucer_AutoUpdater.cpp	2021-09-26 00:16:57.933901901 +0200
@@ -41,6 +41,7 @@
 
 void LatestVersionCheckerAndUpdater::checkForNewVersion (bool background)
 {
+    return;
     if (! isThreadRunning())
     {
         backgroundCheck = background;
