diff --git a/Makefile b/Makefile
index f891255..1899f34 100644
--- a/Makefile
+++ b/Makefile
@@ -50,15 +50,7 @@ else
   HAVE_LZO=0
 endif
 
-ifeq ($(shell grep 'HAVE_LIBCRYPTO 1' config.h >/dev/null 2>&1 && echo 1), 1)
-  OTHTARGETS += pbkdf2
-  AES_OSSL_PO = aes_ossl.po
-  AES_OSSL_O = aes_ossl.o
-  CRYPTOLIB = -lcrypto
-  HAVE_OPENSSL=1
-else
-  HAVE_OPENSSL=0
-endif
+HAVE_OPENSSL=0
 
 ifeq ($(CC),wcl386)
   CFLAGS = "-ox -wx $(EXTRA_CFLAGS)"
