diff -Naur compiz-0.8.8-orig/kde/window-decorator-kde4/window.cpp compiz-0.8.8/kde/window-decorator-kde4/window.cpp
--- compiz-0.8.8-orig/kde/window-decorator-kde4/window.cpp	2012-01-29 22:05:54.000000000 -0500
+++ compiz-0.8.8/kde/window-decorator-kde4/window.cpp	2012-01-29 22:10:38.000000000 -0500
@@ -986,10 +986,10 @@
 {
     Atom    atom = Atoms::compizWindowBlurDecor;
     QRegion topQRegion, bottomQRegion, leftQRegion, rightQRegion;
-    Region  topRegion = NULL;
-    Region  bottomRegion = NULL;
-    Region  leftRegion = NULL;
-    Region  rightRegion = NULL;
+    ::Region  topRegion = NULL;
+    ::Region  bottomRegion = NULL;
+    ::Region  leftRegion = NULL;
+    ::Region  rightRegion = NULL;
     int     size = 0;
     int     w, h;
 
