diff -wbBur sniproxy-0.6.0/src/http.h sniproxy-0.6.0.my/src/http.h
--- sniproxy-0.6.0/src/http.h	2018-12-06 07:14:58.000000000 +0300
+++ sniproxy-0.6.0.my/src/http.h	2021-08-11 21:27:03.140773585 +0300
@@ -29,6 +29,6 @@
 #include <stdio.h>
 #include "protocol.h"
 
-const struct Protocol *const http_protocol;
+extern const struct Protocol *const http_protocol;
 
 #endif
diff -wbBur sniproxy-0.6.0/src/tls.h sniproxy-0.6.0.my/src/tls.h
--- sniproxy-0.6.0/src/tls.h	2018-12-06 07:14:58.000000000 +0300
+++ sniproxy-0.6.0.my/src/tls.h	2021-08-11 21:26:42.937439807 +0300
@@ -28,6 +28,6 @@
 
 #include "protocol.h"
 
-const struct Protocol *const tls_protocol;
+extern const struct Protocol *const tls_protocol;
 
 #endif
