diff -Naur speakup-3.1.3-old/src/speakup_soft.c speakup-3.1.3/src/speakup_soft.c
--- speakup-3.1.3-old/src/speakup_soft.c	2009-08-05 10:59:01.000000000 -0500
+++ speakup-3.1.3/src/speakup_soft.c	2009-12-07 05:28:21.000000000 -0600
@@ -24,6 +24,7 @@
 #include <linux/unistd.h>
 #include <linux/miscdevice.h> /* for misc_register, and SYNTH_MINOR */
 #include <linux/poll.h> /* for poll_wait() */
+#include <linux/sched.h>
 
 #include "spk_priv.h"
 #include "speakup.h"
