From 024f228c45a1df3f01dada4e72e99dd7a76ecf26 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Johannes=20L=C3=B6thberg?= <johannes@kyriasis.com>
Date: Sun, 27 May 2018 02:11:24 +0200
Subject: [PATCH] meson.build: Don't do broken synctex version check
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Johannes Löthberg <johannes@kyriasis.com>
---
 meson.build | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/meson.build b/meson.build
index 1862fa0..63246e8 100644
--- a/meson.build
+++ b/meson.build
@@ -82,9 +82,6 @@ endif
 if get_option('enable-synctex') and synctex.found()
   build_dependencies += synctex
   defines += '-DWITH_SYNCTEX'
-  if synctex.version() < '2.0.0'
-    defines += '-DWITH_SYNCTEX1'
-  endif
 endif
 
 if get_option('enable-magic') and magic.found()
-- 
2.17.0

