From 86cd73f292894f6cf55d3a3050e52a3f6705072b Mon Sep 17 00:00:00 2001
From: Ildar Mulyukov <ildar@users.sourceforge.net>
Date: Tue, 04 Aug 2009 21:26:38 +0000
Subject: Fixes use of LDADD for building and updates desktop file metadata

---
diff --git a/data/gnome-subtitles.desktop.in b/data/gnome-subtitles.desktop.in
index cd7ecdd..77eda17 100644
--- a/data/gnome-subtitles.desktop.in
+++ b/data/gnome-subtitles.desktop.in
@@ -9,7 +9,7 @@ Icon=gnome-subtitles.png
 StartupNotify=true
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;Video;AudioVideo;
+Categories=GNOME;GTK;Video;AudioVideo;AudioVideoEditing;
 MimeType=application/x-subrip;text/x-mpsub;text/x-ssa;text/x-microdvd;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-subtitles
diff --git a/src/Makefile.am b/src/Makefile.am
index 453771e..c967d6e 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -10,7 +10,8 @@ ASSEMBLY_CONFIG = $(ASSEMBLY).config
 AM_CFLAGS = $(gstreamer_CFLAGS) -Wall -g -fPIC
 noinst_PROGRAMS = libgstreamer_playbin.so
 libgstreamer_playbin_so_SOURCES = $(srcdir)/External/GStreamerPlaybin/main.c
-libgstreamer_playbin_so_LDFLAGS = $(gstreamer_LIBS) -lgstinterfaces-0.10 -shared -Wl,--as-needed,-soname,libgstreamer_playbin.so
+libgstreamer_playbin_so_LDFLAGS = -shared -Wl,--as-needed,-soname,libgstreamer_playbin.so
+libgstreamer_playbin_so_LDADD = -lgstinterfaces-0.10 $(gstreamer_LIBS)
 
 GS_SOURCES = \
 	$(srcdir)/External/GStreamerPlaybin/*.cs \
--
cgit v0.8.2

