diff -rupN runtime.4822e3c3aa77eb82b2fb33c9321f923cf11ddde6.orig/eng/packaging.targets runtime.4822e3c3aa77eb82b2fb33c9321f923cf11ddde6/eng/packaging.targets
--- runtime.4822e3c3aa77eb82b2fb33c9321f923cf11ddde6.orig/eng/packaging.targets	2021-11-12 12:56:02.738439496 +0000
+++ runtime.4822e3c3aa77eb82b2fb33c9321f923cf11ddde6/eng/packaging.targets	2021-11-12 12:56:33.683705462 +0000
@@ -1,7 +1,7 @@
 <Project>
 
   <PropertyGroup>
-    <EnablePackageValidation>true</EnablePackageValidation>
+    <EnablePackageValidation>false</EnablePackageValidation>
     <!-- Don't restore prebuilt packages during sourcebuild. -->
     <DisablePackageBaselineValidation Condition="'$(DotNetBuildFromSource)' == 'true'">true</DisablePackageBaselineValidation>
     <PackageValidationBaselineVersion Condition="'$(PackageValidationBaselineVersion)' == ''">$([MSBuild]::Subtract($(MajorVersion), 1)).0.0</PackageValidationBaselineVersion>
