Thanks to Fernando Lopez-Lezcano <nando@ccrma.stanford.edu>

diff -rupN dssi-vst-0.8.old/Makefile dssi-vst-0.8/Makefile
--- dssi-vst-0.8.old/Makefile	2008-12-12 07:34:31.000000000 -0500
+++ dssi-vst-0.8/Makefile	2009-04-30 04:49:04.000000000 -0400
@@ -43,7 +43,7 @@ install:	all
 		mkdir -p $(BINDIR)
 		install dssi-vst.so $(DSSIDIR)
 		install dssi-vst.so $(LADSPADIR)
-		install dssi-vst-server.exe.so dssi-vst-server dssi-vst-scanner.exe.so dssi-vst-scanner dssi-vst_gui $(DSSIDIR)/dssi-vst
+		install dssi-vst-server.exe.so dssi-vst-server.exe dssi-vst-scanner.exe.so dssi-vst-scanner.exe dssi-vst_gui $(DSSIDIR)/dssi-vst
 		install vsthost $(BINDIR)

 clean:
