From 05ba23be296085313abd8d7d5075d962084f5d38 Mon Sep 17 00:00:00 2001
From: Alexandre Franke <alexandre.franke@gmail.com>
Date: Wed, 14 Mar 2012 10:52:25 +0000
Subject: Better use of fields in the .desktop file

https://bugzilla.gnome.org/show_bug.cgi?id=672033
---
diff --git a/data/planner.desktop.in.in b/data/planner.desktop.in.in
index af61d34..30eb921 100644
--- a/data/planner.desktop.in.in
+++ b/data/planner.desktop.in.in
@@ -1,6 +1,10 @@
 [Desktop Entry]
-_Name=Project Management
-_Comment=Planner Project Management
+Version=1.0
+_Name=Planner
+_GenericName=Project Management
+_X-GNOME-FullName=Planner Project Management
+_Comment=Handle the tasks, resources, and cost of your projects
+_Keywords=Gantt;Chart;Critical path;Relationship;Dependency;WBS;
 Exec=planner %F
 Icon=gnome-planner.png
 Terminal=false
--
cgit v0.9.2
