diff -upr a/configure.ac b/configure.ac
--- a/configure.ac	2021-01-28 18:35:29.000000000 +0100
+++ b/configure.ac	2021-02-24 15:03:51.290443097 +0100
@@ -139,15 +139,9 @@ m4_define([AC_LANG_CALL], [AC_LANG_PROGR
 AC_SEARCH_LIBS(res_ninit, resolv,
 	AC_DEFINE(HAVE_RES_NINIT, 1,
 	[Define to 1 if you have the `res_ninit()' function.]))
-AC_SEARCH_LIBS(res_ndestroy, resolv,
-        AC_DEFINE(HAVE_RES_NDESTROY, 1,
-        [Define to 1 if you have the `res_ndestroy()' function.]))
 AC_SEARCH_LIBS(__res_ninit, resolv,
         AC_DEFINE(HAVE_RES_NINIT, 1,
         [Define to 1 if you have the `__res_ninit()' function.]))
-AC_SEARCH_LIBS(__res_ndestroy, resolv,
-        AC_DEFINE(HAVE_RES_NDESTROY, 1,
-        [Define to 1 if you have the `__res_ndestroy()' function.]))
  
 m4_rename_force([saved_AC_LANG_CALL], [AC_LANG_CALL])
 AC_CHECK_LIB(idn, idn_free)
