diff -wbBur open-vm-tools-2010.08.24-292196.org/lib/include/vm_basic_types.h open-vm-tools-2010.08.24-292196/lib/include/vm_basic_types.h
--- open-vm-tools-2010.08.24-292196.org/lib/include/vm_basic_types.h	2010-08-24 19:59:22.000000000 +0000
+++ open-vm-tools-2010.08.24-292196/lib/include/vm_basic_types.h	2010-08-30 09:38:48.000000000 +0000
@@ -83,8 +83,11 @@
 
 /* STRICT ANSI means the Xserver build and X defines Bool differently. */
 #if !defined(__STRICT_ANSI__) || defined(__FreeBSD__) || defined(__MINGW32__)
+#  ifndef _XTYPEDEF_BOOL
+#  define _XTYPEDEF_BOOL
 typedef char           Bool;
 #endif
+#endif
 
 #ifndef FALSE
 #define FALSE          0
