diff -ruN a/src/Makefile b/src/Makefile
--- a/src/Makefile	2018-11-30 23:30:41.000000000 +0100
+++ b/src/Makefile	2021-09-24 22:38:42.060503051 +0200
@@ -16,7 +16,7 @@
   $(error "libpng is required - install libpng-dev")
 endif
 
-ifeq ($(shell $(ECHO) "\#include <stdio.h>\n\#include <jpeglib.h>\nint main() { struct jpeg_error_mgr jerr; jpeg_std_error(&jerr); return 0; }" | $(CC) -pipe -x c -o $(CONFIGTEMP) $(ARCHINCLUDES) $(LDFLAGS) - -ljpeg 2>/dev/null || echo no; $(RM) -f $(CONFIGTEMP)), no)
+ifeq ($(shell $(ECHO) "#include <stdio.h>\n#include <jpeglib.h>\nint main() { struct jpeg_error_mgr jerr; jpeg_std_error(&jerr); return 0; }" | $(CC) -pipe -x c -o $(CONFIGTEMP) $(ARCHINCLUDES) $(LDFLAGS) - -ljpeg 2>/dev/null || echo no; $(RM) -f $(CONFIGTEMP)), no)
   $(error "libjpeg is required - install libjpeg-dev, libjpeg8-dev or libjpeg62-dev")
 endif
 
