From 160e2b7b7d1edc7630db0eb0ebbd796a66aab89a Mon Sep 17 00:00:00 2001
From: Sfan5 <sfan5@live.de>
Date: Fri, 3 Jan 2014 10:17:01 +0100
Subject: [PATCH] Add option to link to OpenGL ES, fixes #1051

---
 src/CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 90f5867..6418b3b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -170,7 +170,12 @@ endif()
 
 find_package(Sqlite3 REQUIRED)
 find_package(Json REQUIRED)
-find_package(OpenGLES2)
+
+option(ENABLE_GLES "Enable OpenGL ES support" 0)
+mark_as_advanced(ENABLE_GLES)
+if(ENABLE_GLES)
+	find_package(OpenGLES2)
+endif(ENABLE_GLES)
 
 if(USE_FREETYPE)
 	find_package(Freetype REQUIRED)
-- 
1.8.5.1

