--- component.am	2010-03-15 20:54:43.000000000 +0100
+++ ../component.am	2010-04-30 15:00:47.218700833 +0200
@@ -6,17 +6,17 @@
 	@if test -d $(COMPONENT); then \
 		echo "Compiling the $(COMPONENT) project..."; \
 		( \
-			$(DESTDIR)$(bindir)/gbi$(GAMBAS_VERSION) -r $(DESTDIR)$(prefix) $(COMPONENT); \
+			$(GBFILES)/gbi$(GAMBAS_VERSION) -r $(DESTDIR)$(prefix) $(COMPONENT); \
 			cd $(COMPONENT); \
-			$(DESTDIR)$(bindir)/gbc$(GAMBAS_VERSION) -ag -r $(DESTDIR)$(prefix); \
-			$(DESTDIR)$(bindir)/gba$(GAMBAS_VERSION); \
+			$(GBFILES)/gbc$(GAMBAS_VERSION) -ag -r $(DESTDIR)$(prefix); \
+			$(GBFILES)/gba$(GAMBAS_VERSION); \
 			rm -rf .gambas; \
 			$(INSTALL) $(COMPONENT).gambas $(DESTDIR)$(gblibdir); \
 		) \
 	fi
 	@echo
 	@echo "Creating the information files for $(COMPONENT) component..."
-	@$(DESTDIR)$(bindir)/gbi$(GAMBAS_VERSION) -r $(DESTDIR)$(prefix) $(COMPONENT)
+	@$(GBFILES)/gbi$(GAMBAS_VERSION) -r $(DESTDIR)$(prefix) $(COMPONENT)
 	@echo
 
 uninstall-hook:
