diff -Naur gnome-panel-3.6.2.orig/applets/clock/clock-location.c gnome-panel-3.6.2/applets/clock/clock-location.c
--- gnome-panel-3.6.2.orig/applets/clock/clock-location.c	2013-09-27 02:39:03.315952789 +0000
+++ gnome-panel-3.6.2/applets/clock/clock-location.c	2013-09-27 02:40:24.552036953 +0000
@@ -542,9 +542,8 @@
 	}
 
 	priv->weather_info =
-		gweather_info_new_for_world (priv->world,
-					     priv->loc,
-					     GWEATHER_FORECAST_LIST);
+		gweather_info_new (priv->loc,
+				   GWEATHER_FORECAST_LIST);
 	g_signal_connect (priv->weather_info, "updated",
 			  G_CALLBACK (weather_info_updated), loc);
 
