diff -Naur gnome-panel-3.6.2.orig/gnome-panel/panel-session.c gnome-panel-3.6.2/gnome-panel/panel-session.c
--- gnome-panel-3.6.2.orig/gnome-panel/panel-session.c	2012-09-03 17:41:42.000000000 +0200
+++ gnome-panel-3.6.2/gnome-panel/panel-session.c	2013-04-11 00:59:42.125864185 +0200
@@ -69,11 +69,6 @@
 
 	egg_sm_client_set_mode (mode);
 
-	client = egg_sm_client_get ();
-
-	g_signal_connect (client, "quit",
-			  G_CALLBACK (panel_session_handle_quit), NULL);
-
 	/* We don't want the WM to try and save/restore our
 	 * window position */
 	gdk_x11_set_sm_client_id (NULL);
