--- os/linux/rt_linux.c.orig	2010-11-26 04:18:41.000000000 +0100
+++ os/linux/rt_linux.c	2010-12-01 14:57:53.000000000 +0100
@@ -479,9 +479,9 @@
 		MEM_DBG_PKT_ALLOC_INC(pAd);
 
 		skb_reserve(skb, 2);				
-		NdisMoveMemory(skb->tail, pHeader802_3, HdrLen);
+		NdisMoveMemory(skb_tail_pointer(skb), pHeader802_3, HdrLen); //
 		skb_put(skb, HdrLen);
-		NdisMoveMemory(skb->tail, pData, DataSize);
+		NdisMoveMemory(skb_tail_pointer(skb), pData, DataSize); //
 		skb_put(skb, DataSize);
 		skb->dev = get_netdev_from_bssid(pAd, FromWhichBSSID);
 		pPacket = OSPKT_TO_RTPKT(skb);
