--- include/os/rt_linux.h.orig	2010-11-26 04:18:41.000000000 +0100
+++ include/os/rt_linux.h	2010-12-01 14:58:46.000000000 +0100
@@ -111,10 +111,10 @@
 
 #ifdef CONFIG_STA_SUPPORT
 #ifdef RTMP_MAC_PCI
-#define STA_PROFILE_PATH			"/etc/Wireless/RT2860STA/RT2860STA.dat"
+#define STA_PROFILE_PATH			"/etc/Wireless/RT3562STA/RT3562STA.dat"
 #define STA_DRIVER_VERSION			"2.4.1.1"
 #ifdef MULTIPLE_CARD_SUPPORT
-#define CARD_INFO_PATH			"/etc/Wireless/RT2860STA/RT2860STACard.dat"
+#define CARD_INFO_PATH			"/etc/Wireless/RT3562STA/RT3562STACard.dat"
 #endif // MULTIPLE_CARD_SUPPORT //
 #endif // RTMP_MAC_PCI //
 
--- os/linux/pci_main_dev.c.orig	2010-12-17 19:59:25.000000000 +0100
+++ os/linux/pci_main_dev.c	2010-12-17 20:00:12.000000000 +0100
@@ -90,14 +90,15 @@
 MODULE_VERSION(STA_DRIVER_VERSION);
 #endif
 #endif // CONFIG_STA_SUPPORT //
-
+MODULE_DESCRIPTION("RT3562 Wireless Lan Linux Driver");
+MODULE_LICENSE("GPL");
 
 //
 // Our PCI driver structure
 //
 static struct pci_driver rt2860_driver =
 {
-    name:       "rt2860",
+    name:       "rt3562",
     id_table:   rt2860_pci_tbl,
     probe:      rt2860_probe,
 #if LINUX_VERSION_CODE >= 0x20412
@@ -317,9 +318,9 @@
 	}
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,0)
-	print_name = pci_dev ? pci_name(pci_dev) : "rt2860";
+	print_name = pci_dev ? pci_name(pci_dev) : "rt3562";
 #else
-	print_name = pci_dev ? pci_dev->slot_name : "rt2860";
+	print_name = pci_dev ? pci_dev->slot_name : "rt3562";
 #endif // LINUX_VERSION_CODE //
 
 	if ((rv = pci_request_regions(pci_dev, print_name)) != 0)
