diff --git a/rwgif.c b/rwgif.c
index e67ac3c..6b19a28 100644
--- a/rwgif.c
+++ b/rwgif.c
@@ -54,8 +54,7 @@ open_gif_file (const char *filename, int *width, int *height)
     
     assert(data != 0);
     
-    int error;
-    data->file = DGifOpenFileName(filename, &error);
+    data->file = DGifOpenFileName(filename, NULL);
     
     assert(data->file !=0);
         
@@ -138,7 +137,7 @@ open_gif_file (const char *filename, int *width, int *height)
     }
     free(buffer);
     
-    assert(DGifCloseFile(data->file) == GIF_OK);
+    assert(DGifCloseFile(data->file, NULL) == GIF_OK);
     
     return data;
 }
