From 83d6c1513787c0cd156cb6c2a1c6dd03e8f0b811 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Luttringer?= <seblu@seblu.net>
Date: Mon, 6 Mar 2017 23:11:44 +0100
Subject: [PATCH] Fix https://bugs.archlinux.org/task/32642

---
 lib/BackupPC/DirOps.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/BackupPC/DirOps.pm b/lib/BackupPC/DirOps.pm
index 6d713c1..a490750 100644
--- a/lib/BackupPC/DirOps.pm
+++ b/lib/BackupPC/DirOps.pm
@@ -132,7 +132,7 @@ sub dirRead
     }
     if ( $IODirentOk ) {
         @entries = sort({ $a->{inode} <=> $b->{inode} } readdirent($fh));
-        map { $_->{type} = 0 + $_->{type} } @entries;   # make type numeric
+        map { $_->{type} = 0 + $_->{type}; $_->{type} = undef if ($_->{type} eq BPC_DT_UNKNOWN); } @entries;   # make type numeric, unset unknown types
     } else {
         @entries = map { { name => $_} } readdir($fh);
     }
-- 
2.12.0

