diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-12-11 13:07:48.000000000 +0100
+++ b/CMakeLists.txt	2021-08-01 13:48:17.937034619 +0200
@@ -11,9 +11,7 @@
 
 ##ADD_DEFINITIONS(-Wall -O0 -g -fPIC -DPIC) ###USED FOR DEBUG
 
-#Set Install prefix
-set (CMAKE_INSTALL_PREFIX  "/usr/local/lib/lv2") #Install prefix
-#set (CMAKE_INSTALL_PREFIX  "$ENV{HOME}/.lv2") #Install prefix
+set(LV2_LIB_DIR "lib/lv2")
 set (LV2_BUNDLE "sapistaEQv2.lv2") #LV2 Bundel dir
 set(EXECUTABLE_OUTPUT_PATH "${PROJECT_SOURCE_DIR}/bin")
 set(LIBRARY_OUTPUT_PATH "${PROJECT_SOURCE_DIR}/bin")
@@ -171,8 +169,8 @@
 
 ##===========================THE LV2 INSTALLER================================================
 #INSTALL( DIRECTORY "${PROJECT_SOURCE_DIR}/${LV2_BUNDLE}" DESTINATION "${LV2_BUNDLE}")
-INSTALL( FILES "manifest.ttl" "eq1qm.ttl" "eq1qs.ttl" "eq4qm.ttl" "eq4qs.ttl" "eq6qm.ttl" "eq6qs.ttl" "eq10qm.ttl" "eq10qs.ttl" "gate.ttl" "compressor.ttl" "compressor_sc.ttl" "gate_stereo.ttl" "compressor_stereo.ttl" "compressor_stereo_sc.ttl" "bassup.ttl" "lr2ms.ttl" "ms2lr.ttl" DESTINATION "${LV2_BUNDLE}")
-INSTALL( TARGETS eq1qm eq1qs eq4qm eq4qs eq6qm eq6qs eq10qm eq10qs gate compressor compressor_sc gate_stereo compressor_stereo compressor_stereo_sc bassup matrix_lr2ms matrix_ms2lr DESTINATION "${LV2_BUNDLE}")
+INSTALL( FILES "manifest.ttl" "eq1qm.ttl" "eq1qs.ttl" "eq4qm.ttl" "eq4qs.ttl" "eq6qm.ttl" "eq6qs.ttl" "eq10qm.ttl" "eq10qs.ttl" "gate.ttl" "compressor.ttl" "compressor_sc.ttl" "gate_stereo.ttl" "compressor_stereo.ttl" "compressor_stereo_sc.ttl" "bassup.ttl" "lr2ms.ttl" "ms2lr.ttl" DESTINATION "${LV2_LIB_DIR}/${LV2_BUNDLE}")
+INSTALL( TARGETS eq1qm eq1qs eq4qm eq4qs eq6qm eq6qs eq10qm eq10qs gate compressor compressor_sc gate_stereo compressor_stereo compressor_stereo_sc bassup matrix_lr2ms matrix_ms2lr DESTINATION "${LV2_LIB_DIR}/${LV2_BUNDLE}")
 
 INCLUDE(CPack)
 # cpack -G TGZ
diff -ruN a/gui/CMakeLists.txt b/gui/CMakeLists.txt
--- a/gui/CMakeLists.txt	2016-12-11 13:07:53.000000000 +0100
+++ b/gui/CMakeLists.txt	2021-08-01 13:52:19.484456051 +0200
@@ -43,8 +43,8 @@
 set_target_properties(eq10q_gui dynamics_gui bassup_gui matrix_ms_gui PROPERTIES PREFIX "")
 
 ##=================================INSTALLER=======================================================================
-INSTALL( TARGETS eq10q_gui dynamics_gui bassup_gui matrix_ms_gui DESTINATION "${LV2_BUNDLE}/gui")
-INSTALL( DIRECTORY "${PROJECT_SOURCE_DIR}/gui/knobs" DESTINATION "${LV2_BUNDLE}/gui")
-INSTALL( DIRECTORY "${PROJECT_SOURCE_DIR}/gui/combopix" DESTINATION "${LV2_BUNDLE}/gui")
-INSTALL( DIRECTORY "${PROJECT_SOURCE_DIR}/gui/icons" DESTINATION "${LV2_BUNDLE}/gui")
+INSTALL( TARGETS eq10q_gui dynamics_gui bassup_gui matrix_ms_gui DESTINATION "${LV2_LIB_DIR}/${LV2_BUNDLE}/gui")
+INSTALL( DIRECTORY "${PROJECT_SOURCE_DIR}/gui/knobs" DESTINATION "${LV2_LIB_DIR}/${LV2_BUNDLE}/gui")
+INSTALL( DIRECTORY "${PROJECT_SOURCE_DIR}/gui/combopix" DESTINATION "${LV2_LIB_DIR}/${LV2_BUNDLE}/gui")
+INSTALL( DIRECTORY "${PROJECT_SOURCE_DIR}/gui/icons" DESTINATION "${LV2_LIB_DIR}/${LV2_BUNDLE}/gui")
 
