Subject: [PATCH] fix-scala-compilation-on-gradle-8.0.1
---
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision 2e1947d240607d53f071f61c875cfffc3fec47fe)
+++ b/build.gradle	(date 1677026530111)
@@ -671,6 +671,7 @@
       scalaCompileOptions.additionalParameters += ["-opt-warnings", "-Xlint:strict-unsealed-patmat"]
       // Scala 2.13.2 introduces compiler warnings suppression, which is a pre-requisite for -Xfatal-warnings
       scalaCompileOptions.additionalParameters += ["-Xfatal-warnings"]
+      scalaCompileOptions.additionalParameters += ["-release:" + minJavaVersion]
     }
 
     // these options are valid for Scala versions < 2.13 only
@@ -684,7 +685,7 @@
     }
 
     // Scalac's `-release` requires Java 9 or higher
-    if (JavaVersion.current().isJava9Compatible())
+    if ((JavaVersion.current().isJava9Compatible()) && (versions.baseScala == '2.12'))
       scalaCompileOptions.additionalParameters += ["-release", minJavaVersion]
 
     configure(scalaCompileOptions.forkOptions) {
