Subject: [PATCH] support-gradle-7.6
---
Index: gradle/dependencies.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
--- a/gradle/dependencies.gradle	(revision b66af662e61082cb8def576ded1fe5cee37e155f)
+++ b/gradle/dependencies.gradle	(date 1675032287344)
@@ -62,7 +62,7 @@
   checkstyle: "8.36.2",
   commonsCli: "1.4",
   dropwizardMetrics: "4.1.12.1",
-  gradle: "7.4.2",
+  gradle: "7.6",
   grgit: "4.1.1",
   httpclient: "4.5.13",
   easymock: "4.3",
@@ -119,7 +119,7 @@
   spotbugs: "4.2.2",
   swaggerAnnotations: "2.2.0",
   swaggerJaxrs2: "2.2.0",
-  zinc: "1.3.5",
+  zinc: "1.6.1",
   zookeeper: "3.6.3",
   zstd: "1.5.2-1"
 ]
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision b66af662e61082cb8def576ded1fe5cee37e155f)
+++ b/build.gradle	(date 1675032216066)
@@ -66,8 +66,10 @@
   if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_16))
     defaultJvmArgs.addAll(
       "--add-opens=java.base/java.io=ALL-UNNAMED",
+      "--add-opens=java.base/java.lang=ALL-UNNAMED",
       "--add-opens=java.base/java.nio=ALL-UNNAMED",
       "--add-opens=java.base/java.nio.file=ALL-UNNAMED",
+      "--add-opens=java.base/java.util=ALL-UNNAMED",
       "--add-opens=java.base/java.util.concurrent=ALL-UNNAMED",
       "--add-opens=java.base/java.util.regex=ALL-UNNAMED",
       "--add-opens=java.base/java.util.stream=ALL-UNNAMED",
