diff --git a/gdal/frmts/pdf/pdfobject.cpp b/gdal/frmts/pdf/pdfobject.cpp
index ae734a74e4..c5a9935c00 100644
--- a/gdal/frmts/pdf/pdfobject.cpp
+++ b/gdal/frmts/pdf/pdfobject.cpp
@@ -1055,7 +1055,7 @@ const CPLString& GDALPDFObjectPoppler::GetString()
 {
     if (GetType() == PDFObjectType_String)
     {
-        GooString* gooString = m_po->getString();
+        const GooString* gooString = m_po->getString();
         return (osStr = GDALPDFGetUTF8StringFromBytes(reinterpret_cast<const GByte*>(gooString->getCString()),
                                                       static_cast<int>(gooString->getLength())));
     }
