diff -Naur gnome-screensaver-3.6.1.orig/configure.ac gnome-screensaver-3.6.1/configure.ac
--- gnome-screensaver-3.6.1.orig/configure.ac	2014-12-05 20:38:42.386401000 +0100
+++ gnome-screensaver-3.6.1/configure.ac	2014-12-05 20:42:07.474443655 +0100
@@ -761,6 +761,7 @@
 po/Makefile.in
 src/Makefile
 src/gnome-screensaver.desktop.in
+src/gnome-screensaver-autostart.desktop.in
 data/Makefile
 doc/Makefile
 doc/gnome-screensaver.xml
diff -Naur gnome-screensaver-3.6.1.orig/po/POTFILES.in gnome-screensaver-3.6.1/po/POTFILES.in
--- gnome-screensaver-3.6.1.orig/po/POTFILES.in	2012-06-11 17:05:23.000000000 +0200
+++ gnome-screensaver-3.6.1/po/POTFILES.in	2014-12-05 20:39:45.921183441 +0100
@@ -4,6 +4,7 @@
 src/gnome-screensaver-dialog.c
 src/gnome-screensaver.c
 src/gnome-screensaver.desktop.in.in
+src/gnome-screensaver-autostart.desktop.in.in
 src/gs-auth-pam.c
 src/gs-listener-dbus.c
 src/gs-lock-plug.c
diff -Naur gnome-screensaver-3.6.1.orig/po/POTFILES.skip gnome-screensaver-3.6.1/po/POTFILES.skip
--- gnome-screensaver-3.6.1.orig/po/POTFILES.skip	2012-06-11 17:05:23.000000000 +0200
+++ gnome-screensaver-3.6.1/po/POTFILES.skip	2014-12-05 20:39:47.271214330 +0100
@@ -1 +1,2 @@
 src/gnome-screensaver.desktop.in
+src/gnome-screensaver-autostart.desktop.in
diff -Naur gnome-screensaver-3.6.1.orig/src/gnome-screensaver-autostart.desktop.in.in gnome-screensaver-3.6.1/src/gnome-screensaver-autostart.desktop.in.in
--- gnome-screensaver-3.6.1.orig/src/gnome-screensaver-autostart.desktop.in.in	1970-01-01 01:00:00.000000000 +0100
+++ gnome-screensaver-3.6.1/src/gnome-screensaver-autostart.desktop.in.in	2014-12-05 20:41:04.679658977 +0100
@@ -0,0 +1,14 @@
+[Desktop Entry]
+Type=Application
+_Name=Screensaver
+_Comment=Launch screensaver and locker program
+Icon=preferences-desktop-screensaver
+Exec=gnome-screensaver
+OnlyShowIn=GNOME-Flashback;
+NoDisplay=true
+X-GNOME-Autostart-Phase=Application
+X-GNOME-Autostart-Notify=true
+X-GNOME-Bugzilla-Bugzilla=GNOME
+X-GNOME-Bugzilla-Product=gnome-screensaver
+X-GNOME-Bugzilla-Component=general
+X-GNOME-Bugzilla-Version=@VERSION@
diff -Naur gnome-screensaver-3.6.1.orig/src/gnome-screensaver.desktop.in.in gnome-screensaver-3.6.1/src/gnome-screensaver.desktop.in.in
--- gnome-screensaver-3.6.1.orig/src/gnome-screensaver.desktop.in.in	2014-12-05 20:38:42.383068000 +0100
+++ gnome-screensaver-3.6.1/src/gnome-screensaver.desktop.in.in	2014-12-05 20:41:22.556738056 +0100
@@ -4,7 +4,6 @@
 _Comment=Launch screensaver and locker program
 Icon=preferences-desktop-screensaver
 Exec=gnome-screensaver
-OnlyShowIn=GNOME;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Application
 X-GNOME-Autostart-Notify=true
diff -Naur gnome-screensaver-3.6.1.orig/src/Makefile.am gnome-screensaver-3.6.1/src/Makefile.am
--- gnome-screensaver-3.6.1.orig/src/Makefile.am	2014-12-05 20:38:42.383068000 +0100
+++ gnome-screensaver-3.6.1/src/Makefile.am	2014-12-05 20:52:28.649076834 +0100
@@ -47,6 +47,10 @@
 	test-window	\
 	$(NULL)
 
+autostartdir = $(sysconfdir)/xdg/autostart
+autostart_in_files = gnome-screensaver-autostart.desktop.in
+autostart_DATA = $(autostart_in_files:.desktop.in=.desktop)
+
 desktopdir = $(datadir)/applications
 desktop_in_files = gnome-screensaver.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
@@ -197,15 +201,18 @@
 
 gnome_screensaver_LDFLAGS = -export-dynamic
 
-EXTRA_DIST =				\
-	debug-screensaver.sh		\
-	gs-marshal.list			\
-	gnome-screensaver.desktop.in	\
+EXTRA_DIST =					\
+	debug-screensaver.sh			\
+	gs-marshal.list				\
+	gnome-screensaver.desktop.in		\
+	gnome-screensaver-autostart.desktop.in	\
 	$(NULL)
 
-CLEANFILES = 				\
-	$(desktop_DATA) 		\
-	gnome-screensaver.desktop.in	\
+CLEANFILES = 					\
+	$(autostart_DATA) 			\
+	$(desktop_DATA) 			\
+	gnome-screensaver.desktop.in		\
+	gnome-screensaver-autostart.desktop.in	\
 	$(BUILT_SOURCES)
 
 MAINTAINERCLEANFILES =			\
