diff -aur Impressive-0.10.3/impressive.py Impressive-0.10.3.new/impressive.py
--- Impressive-0.10.3/impressive.py	2013-09-20 09:34:16.477993905 +0000
+++ Impressive-0.10.3.new/impressive.py	2013-09-20 09:33:59.468099703 +0000
@@ -182,7 +182,7 @@
     from OpenGL.GL import *
     import pygame
     from pygame.locals import *
-    import Image, ImageDraw, ImageFont, ImageFilter
+    from PIL import Image, ImageDraw, ImageFont, ImageFilter
     import TiffImagePlugin, BmpImagePlugin, JpegImagePlugin, PngImagePlugin, PpmImagePlugin
 except (ValueError, ImportError), err:
     print >>sys.stderr, "Oops! Cannot load necessary modules:", err
