diff -rupN bamf.orig/configure.ac bamf/configure.ac
--- bamf.orig/configure.ac	2020-02-03 18:55:29.602584778 +0100
+++ bamf/configure.ac	2020-02-03 18:55:45.776052493 +0100
@@ -143,7 +143,7 @@ AC_SUBST(COVERAGE_LDFLAGS)
 AM_PATH_GTK_3_0
 
 dnl CFLAGS
-CFLAGS="$CFLAGS -Wall -Werror -Wno-error=deprecated-declarations -Wempty-body -Wformat-security -Winit-self -Warray-bounds -Wimplicit-function-declaration -lm"
+CFLAGS="$CFLAGS -Wall -Wempty-body -Wformat-security -Winit-self -Warray-bounds -Wimplicit-function-declaration -lm"
 
 AC_SUBST(BAMFDAEMON_CFLAGS)
 AC_SUBST(BAMFDAEMON_LIBS)


