diff -Naur mscore-0.9.6.3.orig/mscore/scriptgen/qtbindings/core/qtscript_QEvent.cpp mscore-0.9.6.3.new/mscore/scriptgen/qtbindings/core/qtscript_QEvent.cpp
--- mscore-0.9.6.3.orig/mscore/scriptgen/qtbindings/core/qtscript_QEvent.cpp	2010-10-12 22:01:35.000000000 +0200
+++ mscore-0.9.6.3.new/mscore/scriptgen/qtbindings/core/qtscript_QEvent.cpp	2010-10-12 22:03:11.000000000 +0200
@@ -214,7 +214,6 @@
     , QEvent::UngrabMouse
     , QEvent::GrabKeyboard
     , QEvent::UngrabKeyboard
-    , QEvent::CocoaRequestModal
     , QEvent::User
     , QEvent::MaxUser
 };
@@ -362,7 +361,6 @@
     , "UngrabMouse"
     , "GrabKeyboard"
     , "UngrabKeyboard"
-    , "CocoaRequestModal"
     , "User"
     , "MaxUser"
 };
