From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Mon, 12 Dec 2022 00:53:27 +0000
Subject: [PATCH] Linux 6.1

---
 xt_FULLCONENAT.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xt_FULLCONENAT.c b/xt_FULLCONENAT.c
index 5f4cc5390629..48598d697048 100644
--- a/xt_FULLCONENAT.c
+++ b/xt_FULLCONENAT.c
@@ -428,7 +428,11 @@ static uint16_t find_appropriate_port(struct net *net, const struct nf_conntrack
     /* for now we do the same thing for both --random and --random-fully */
 
     /* select a random starting point */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0)
+    start = (uint16_t)(get_random_u32() % (u32)range_size);
+#else
     start = (uint16_t)(prandom_u32() % (u32)range_size);
+#endif
   } else {
 
     if ((original_port >= min && original_port <= min + range_size - 1)
