--- 44-wqy-zenhei.conf.org	2009-05-18 00:25:41.000000000 +0400
+++ 44-wqy-zenhei.conf	2012-09-25 19:21:44.000000000 +0400
@@ -6,7 +6,11 @@
   <match target="font">
         <test qual="any" name="family">
                 <string>WenQuanYi Zen Hei</string>
+        </test> 
+        <test qual="any" name="family">
                 <string>文泉驿正黑</string>
+        </test> 
+        <test qual="any" name="family">
                 <string>文泉驛正黑</string>
         </test> 
         <edit name="globaladvance"><bool>false</bool></edit>
