diff --git c/test_haystack/settings.py w/test_haystack/settings.py
index dd426f1..30015eb 100644
--- c/test_haystack/settings.py
+++ w/test_haystack/settings.py
@@ -16,6 +16,7 @@ INSTALLED_APPS = [
     "django.contrib.admin",
     "django.contrib.auth",
     "django.contrib.contenttypes",
+    "django.contrib.messages",
     "django.contrib.sessions",
     "haystack",
     "test_haystack.discovery",
@@ -36,7 +37,8 @@ TEMPLATES = [
         "BACKEND": "django.template.backends.django.DjangoTemplates",
         "APP_DIRS": True,
         "OPTIONS": {
-            "context_processors": ["django.contrib.auth.context_processors.auth"]
+            "context_processors": ["django.contrib.auth.context_processors.auth",
+                                   "django.contrib.messages.context_processors.messages"]
         },
     }
 ]
