diff -rupN nv-codec-headers.orig/ffnvcodec.pc.in nv-codec-headers/ffnvcodec.pc.in
--- nv-codec-headers.orig/ffnvcodec.pc.in	2019-04-30 11:44:46.313950977 +0200
+++ nv-codec-headers/ffnvcodec.pc.in	2019-04-30 11:47:35.311985459 +0200
@@ -1,5 +1,5 @@
 prefix=@@PREFIX@@
-includedir=${prefix}/include
+includedir=${prefix}/include/ffnvcodec8.1
 
 Name: ffnvcodec
 Description: FFmpeg version of Nvidia Codec SDK headers
diff -rupN nv-codec-headers.orig/Makefile nv-codec-headers/Makefile
--- nv-codec-headers.orig/Makefile	2019-04-30 11:44:46.313950977 +0200
+++ nv-codec-headers/Makefile	2019-04-30 11:48:37.275820562 +0200
@@ -11,13 +11,13 @@ else
 endif
 
 install: all
-	$(INSTALL) -m 0755 -d '$(DESTDIR)$(PREFIX)/include/ffnvcodec'
-	$(INSTALL) -m 0644 include/ffnvcodec/*.h '$(DESTDIR)$(PREFIX)/include/ffnvcodec'
-	$(INSTALL) -m 0755 -d '$(DESTDIR)$(PREFIX)/$(LIBDIR)/pkgconfig'
-	$(INSTALL) -m 0644 ffnvcodec.pc '$(DESTDIR)$(PREFIX)/$(LIBDIR)/pkgconfig'
+	$(INSTALL) -m 0755 -d '$(DESTDIR)$(PREFIX)/include/ffnvcodec8.1/ffnvcodec'
+	$(INSTALL) -m 0644 include/ffnvcodec/*.h '$(DESTDIR)$(PREFIX)/include/ffnvcodec8.1/ffnvcodec'
+	$(INSTALL) -m 0755 -d '$(DESTDIR)$(PREFIX)/$(LIBDIR)/ffnvcodec8.1/pkgconfig'
+	$(INSTALL) -m 0644 ffnvcodec.pc '$(DESTDIR)$(PREFIX)/$(LIBDIR)/ffnvcodec8.1/pkgconfig'
 
 uninstall:
-	rm -rf '$(DESTDIR)$(PREFIX)/include/ffnvcodec' '$(DESTDIR)$(PREFIX)/$(LIBDIR)/pkgconfig/ffnvcodec.pc'
+	rm -rf '$(DESTDIR)$(PREFIX)/include/ffnvcodec8.1' '$(DESTDIR)$(PREFIX)/$(LIBDIR)/ffnvcodec8.1/pkgconfig/ffnvcodec.pc'
 
 .PHONY: all install uninstall
 
