diff -urN scikit-build-0.10.0.orig/requirements.txt scikit-build-0.10.0/requirements.txt
--- scikit-build-0.10.0.orig/requirements.txt	2020-02-04 13:44:07.340306400 +0100
+++ scikit-build-0.10.0/requirements.txt	2020-02-04 13:45:00.054455198 +0100
@@ -1,3 +1,4 @@
 wheel>=0.29.0
 setuptools>=28.0.0
-packaging
\ Brak znaku nowej linii na końcu pliku
+packaging
+distro
diff -urN scikit-build-0.10.0.orig/skbuild/platform_specifics/linux.py scikit-build-0.10.0/skbuild/platform_specifics/linux.py
--- scikit-build-0.10.0.orig/skbuild/platform_specifics/linux.py	2020-02-04 13:44:07.343639785 +0100
+++ scikit-build-0.10.0/skbuild/platform_specifics/linux.py	2020-02-04 13:45:35.738340612 +0100
@@ -1,5 +1,6 @@
 """This module defines object specific to Linux platform."""
 
+import distro
 import platform
 import sys
 import textwrap
@@ -24,7 +25,7 @@
         """
         # gentoo, slackware: Compiler is available by default.
 
-        distribution_name = platform.linux_distribution()[0]
+        distribution_name = distro.id()
         cmd = ""
         if distribution_name in [
                 'debian', 'Ubuntu', 'mandrake', 'mandriva']:
