diff -upr python-heatclient-2.4.0.orig/heatclient/common/template_utils.py python-heatclient-2.4.0/heatclient/common/template_utils.py
--- python-heatclient-2.4.0.orig/heatclient/common/template_utils.py	2021-07-06 15:24:33.000000000 +0300
+++ python-heatclient-2.4.0/heatclient/common/template_utils.py	2021-12-07 00:15:41.635946307 +0200
@@ -203,10 +203,10 @@ def deep_update(old, new):
         old = {}
 
     for k, v in new.items():
-        if isinstance(v, collections.Mapping):
+        if isinstance(v, collections.abc.Mapping):
             r = deep_update(old.get(k, {}), v)
             old[k] = r
-        elif v is None and isinstance(old.get(k), collections.Mapping):
+        elif v is None and isinstance(old.get(k), collections.abc.Mapping):
             # Don't override empty data, to work around yaml syntax issue
             pass
         else:
