diff -ruN a/allauth/socialaccount/providers/apple/views.py b/allauth/socialaccount/providers/apple/views.py
--- a/allauth/socialaccount/providers/apple/views.py	2021-11-15 17:23:50.000000000 +0100
+++ b/allauth/socialaccount/providers/apple/views.py	2021-11-15 19:38:17.285418437 +0100
@@ -68,7 +68,6 @@
                 id_token,
                 public_key,
                 algorithms=["RS256"],
-                verify=True,
                 audience=allowed_auds,
                 issuer="https://appleid.apple.com",
             )
