diff -ruN a/pdm/project/core.py b/pdm/project/core.py
--- a/pdm/project/core.py	2022-06-14 09:10:17.230656100 +0200
+++ b/pdm/project/core.py	2022-06-27 18:43:26.516225686 +0200
@@ -42,7 +42,7 @@
 if TYPE_CHECKING:
     from resolvelib.reporters import BaseReporter
 
-    from pdm._vendor import halo
+    import halo
     from pdm.core import Core
     from pdm.resolver.providers import BaseProvider
 
diff -ruN a/pdm/resolver/reporters.py b/pdm/resolver/reporters.py
--- a/pdm/resolver/reporters.py	2022-06-14 09:10:17.234656300 +0200
+++ b/pdm/resolver/reporters.py	2022-06-27 18:43:26.516225686 +0200
@@ -10,7 +10,7 @@
 if TYPE_CHECKING:
     from resolvelib.resolvers import RequirementInformation, State  # type: ignore
 
-    from pdm._vendor import halo
+    import halo
     from pdm.models.candidates import Candidate
     from pdm.models.requirements import Requirement
 
diff -ruN a/pdm/termui.py b/pdm/termui.py
--- a/pdm/termui.py	2022-06-14 09:10:17.234656300 +0200
+++ b/pdm/termui.py	2022-06-27 18:43:26.516225686 +0200
@@ -14,8 +14,9 @@
 import click
 from click._compat import strip_ansi
 
-from pdm._vendor import colorama, halo
-from pdm._vendor.log_symbols.symbols import is_supported as supports_unicode
+import colorama
+import halo
+from log_symbols.symbols import is_supported as supports_unicode
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
diff -ruN a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml	2022-06-14 09:10:17.234656300 +0200
+++ b/pyproject.toml	2022-06-27 19:08:55.515337628 +0200
@@ -22,6 +22,9 @@
     "tomlkit>=0.8.0,<1",
     "typing-extensions; python_version < \"3.8\"",
     "wheel>=0.36.2",
+    "colorama",
+    "halo",
+    "log_symbols",
 ]
 name = "pdm"
 description = "Python Development Master"
