diff --git a/engine/Makefile.am b/engine/Makefile.am
index 1334876..1836439 100644
--- a/engine/Makefile.am
+++ b/engine/Makefile.am
@@ -21,6 +21,7 @@
 
 DOC2MAN = docbook2man
 SGML = ibus-table-createdb.sgml
+XML = ibus-table-createdb.xml
 BUILT_MANS = ibus-table-createdb.1
 man_MANS = $(BUILT_MANS)
 
@@ -65,8 +66,9 @@ MAINTAINERCLEANFILES = \
 
 ${man_MANS}: ${SGML}
 	$(AM_V_GEN) $(RM) $@; \
-	$(DOC2MAN) ${SGML}; \
-	$(RM) manpage.*
+	SGML_CATALOG_FILES=/etc/sgml/catalog sgml2xml -xlower -xid ${SGML} > ${XML}; \
+	$(DOC2MAN) ${XML}; \
+	$(RM) manpage.* ${XML}
 
 table.xml: table.xml.in
 	( \
