diff -Naur spyder-3.3.6.orig/scripts/spyder3.desktop spyder-3.3.6/scripts/spyder3.desktop
--- spyder-3.3.6.orig/scripts/spyder3.desktop	2019-07-14 12:49:39.000000000 +0200
+++ spyder-3.3.6/scripts/spyder3.desktop	2019-10-12 19:32:47.496911043 +0200
@@ -1,12 +1,11 @@
 [Desktop Entry]
 Type=Application
 Version=1.0
-Name=Spyder3
-GenericName=Spyder3
+Name=Spyder
 Comment=The Scientific Python Development Environment - Python 3
-Icon=spyder3
-TryExec=spyder3
-Exec=spyder3 %F
+Icon=spyder
+TryExec=spyder
+Exec=spyder %F
 Terminal=false
 MimeType=text/x-python;
 Categories=Development;Science;IDE;Qt;
diff -Naur spyder-3.3.6.orig/setup.py spyder-3.3.6/setup.py
--- spyder-3.3.6.orig/setup.py	2019-07-14 12:49:39.000000000 +0200
+++ spyder-3.3.6/setup.py	2019-10-12 19:40:23.339975060 +0200
@@ -85,9 +85,9 @@
     """Return data_files in a platform dependent manner"""
     if sys.platform.startswith('linux'):
         if PY3:
-            data_files = [('share/applications', ['scripts/spyder3.desktop']),
-                          ('share/icons', ['img_src/spyder3.png']),
-                          ('share/metainfo', ['scripts/spyder3.appdata.xml'])]
+            data_files = [('share/applications', ['scripts/spyder.desktop']),
+                          ('share/icons', ['img_src/spyder.png']),
+                          ('share/metainfo', ['scripts/spyder.appdata.xml'])]
         else:
             data_files = [('share/applications', ['scripts/spyder.desktop']),
                           ('share/icons', ['img_src/spyder.png'])]
@@ -134,7 +134,7 @@
 # platforms due to a bug in pip installation process (see Issue 1158)
 SCRIPTS = ['%s_win_post_install.py' % NAME]
 if PY3 and sys.platform.startswith('linux'):
-    SCRIPTS.append('spyder3')
+    SCRIPTS.append('spyder')
 else:
     SCRIPTS.append('spyder')
 
