diff --git a/tests/testsuite.py b/tests/testsuite.py
index 0b0cb7a..5de5814 100755
--- a/tests/testsuite.py
+++ b/tests/testsuite.py
@@ -306,8 +306,7 @@ class XMLTestRunnerTestCase(unittest.TestCase):
         outdir.seek(0)
         output = outdir.read()
         self.assertIn(
-            u'<skipped message="demonstrating non-ascii skipping: éçà" '
-            u'type="skip"/>'.encode('utf8'),
+            u'<skipped type="skip" message="demonstrating non-ascii skipping: éçà"/>'.encode('utf8'),
             output)
 
     def test_xmlrunner_safe_xml_encoding_name(self):
