diff -ur ipython-0.10.orig/IPython/kernel/enginefc.py ipython-0.10/IPython/kernel/enginefc.py
--- ipython-0.10.orig/IPython/kernel/enginefc.py	2009-04-16 02:58:05.000000000 +0200
+++ ipython-0.10/IPython/kernel/enginefc.py	2010-03-20 20:38:44.000000000 +0100
@@ -31,7 +31,10 @@
 from twisted.internet.base import DelayedCall
 DelayedCall.debug = True
 
-from foolscap import Referenceable, DeadReferenceError
+try:
+  from foolscap.api import Referenceable, DeadReferenceError
+except ImportError:
+  from foolscap import Referenceable, DeadReferenceError
 from foolscap.referenceable import RemoteReference
 
 from IPython.kernel.pbutil import packageFailure, unpackageFailure
diff -ur ipython-0.10.orig/IPython/kernel/fcutil.py ipython-0.10/IPython/kernel/fcutil.py
--- ipython-0.10.orig/IPython/kernel/fcutil.py	2009-04-16 02:58:05.000000000 +0200
+++ ipython-0.10/IPython/kernel/fcutil.py	2010-03-20 20:35:04.000000000 +0100
@@ -17,7 +17,10 @@
 
 import os
 
-from foolscap import Tub, UnauthenticatedTub
+try:
+  from foolscap.api import Tub, UnauthenticatedTub
+except ImportError:
+  from foolscap import Tub, UnauthenticatedTub
 
 def check_furl_file_security(furl_file, secure):
     """Remove the old furl_file if changing security modes."""
diff -ur ipython-0.10.orig/IPython/kernel/multienginefc.py ipython-0.10/IPython/kernel/multienginefc.py
--- ipython-0.10.orig/IPython/kernel/multienginefc.py	2009-04-16 02:58:05.000000000 +0200
+++ ipython-0.10/IPython/kernel/multienginefc.py	2010-03-20 20:36:40.000000000 +0100
@@ -24,7 +24,10 @@
 from twisted.internet import defer
 from twisted.python import components, failure, log
 
-from foolscap import Referenceable
+try:
+  from foolscap.api import Referenceable
+except ImportError:
+  from foolscap import Referenceable
 
 from IPython.kernel import error 
 from IPython.kernel.util import printer
diff -ur ipython-0.10.orig/IPython/kernel/taskfc.py ipython-0.10/IPython/kernel/taskfc.py
--- ipython-0.10.orig/IPython/kernel/taskfc.py	2009-04-16 02:58:05.000000000 +0200
+++ ipython-0.10/IPython/kernel/taskfc.py	2010-03-20 20:37:37.000000000 +0100
@@ -25,7 +25,10 @@
 from twisted.internet import defer
 from twisted.python import components, failure
 
-from foolscap import Referenceable
+try:
+  from foolscap.api import Referenceable
+except ImportError:
+  from foolscap import Referenceable
 
 from IPython.kernel.twistedutil import blockingCallFromThread
 from IPython.kernel import error, task as taskmodule, taskclient
