--- /var/cvs/color-theme/color-theme/color-theme.el	2006/07/06 04:45:45	1.17
+++ /var/cvs/color-theme/color-theme/color-theme.el	2007/09/10 17:07:24	1.18
@@ -73,9 +73,10 @@
   "Non-nil if running XEmacs.")
 
 ;; Add this since it appears to miss in emacs-2x
-(or (fboundp 'replace-in-string)
-    (defun replace-in-string (target old new)
-      (replace-regexp-in-string old new  target)))
+(if (fboundp 'replace-in-string)
+    (defalias 'color-theme-replace-in-string 'replace-in-string)
+  (defsubst color-theme-replace-in-string (target old new &optional literal)
+    (replace-regexp-in-string old new target nil literal)))
 
 ;; face-attr-construct has a problem in Emacs 20.7 and older when
 ;; dealing with inverse-video faces.  Here is a short test to check
@@ -1628,8 +1629,8 @@ frame-parameter settings of previous col
        (add-to-list 'color-themes
                     (list ',n
                           (upcase-initials
-                           (replace-in-string
-                            (replace-in-string 
+                           (color-theme-replace-in-string
+                            (color-theme-replace-in-string 
                              (symbol-name ',n) "^color-theme-" "") "-" " "))
                           ,author))
        (defun ,n ()