diff --git a/Makefile.am b/Makefile.am
index 1634c6af..2f9b7583 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -297,7 +297,7 @@ INSTALL_SYSTEMD_TARGET = install-systemd-local
 # will be stored in a scripts folder in the _build_ folder
 # which will be the source folder if you're not using a separate build folder
 
-$(INSTALL_SYSTEMD_TARGET): scripts/$(SYSTEMD_SERVICE) $(INSTALL_USER_TARGET)
+$(INSTALL_SYSTEMD_TARGET): scripts/$(SYSTEMD_SERVICE)
 	install -d $(DESTDIR)$(systemdsystemunitdir)
 	[ -e $(DESTDIR)$(systemdsystemunitdir)/shairport-sync.service ] || install -m 0644 scripts/$(SYSTEMD_SERVICE) $(DESTDIR)$(systemdsystemunitdir)/shairport-sync.service
 
