diff -u -r cadaver-0.23.3/lib/neon/ne_openssl.c cadaver-0.23.3-nossl2/lib/neon/ne_openssl.c
--- cadaver-0.23.3/lib/neon/ne_openssl.c	2009-09-13 13:26:52.000000000 +0200
+++ cadaver-0.23.3-nossl2/lib/neon/ne_openssl.c	2016-03-03 13:13:14.207713538 +0100
@@ -564,11 +564,8 @@
         /* enable workarounds for buggy SSL server implementations */
         SSL_CTX_set_options(ctx->ctx, SSL_OP_ALL);
         SSL_CTX_set_verify(ctx->ctx, SSL_VERIFY_PEER, verify_callback);
-    } else if (mode == NE_SSL_CTX_SERVER) {
-        ctx->ctx = SSL_CTX_new(SSLv23_server_method());
-        SSL_CTX_set_session_cache_mode(ctx->ctx, SSL_SESS_CACHE_CLIENT);
     } else {
-        ctx->ctx = SSL_CTX_new(SSLv2_server_method());
+        ctx->ctx = SSL_CTX_new(SSLv23_server_method());
         SSL_CTX_set_session_cache_mode(ctx->ctx, SSL_SESS_CACHE_CLIENT);
     }
     return ctx;
