diff -aur dataenc-0.14.0.5/dataenc.cabal dataenc-0.14.0.5-new/dataenc.cabal
--- dataenc-0.14.0.5/dataenc.cabal	2012-12-30 11:30:27.000000000 +0100
+++ dataenc-0.14.0.5-new/dataenc.cabal	2014-04-27 22:01:21.640940945 +0200
@@ -21,7 +21,7 @@
 
 library
     hs-source-dirs: src
-    build-depends: array >= 0.1.0 && < 0.5, base >= 3.0.0 && < 4.7, containers >= 0.1.0 && < 0.6
+    build-depends: array >= 0.1.0 && <= 0.5.0.0, base >= 3.0.0 && <= 4.7.0.1, containers >= 0.1.0 && < 0.6
     exposed-modules:
             Codec.Binary.Base16
             Codec.Binary.Base32
