From 800ae670e2927128ea4fd0a3011a4a5b029e92ef Mon Sep 17 00:00:00 2001
From: Daniel Wagner <me@dmwit.com>
Date: Tue, 15 Dec 2015 07:44:36 -0800
Subject: [PATCH] use a record pattern to be robust against additions to the
 X11 library

---
 XMonad/Util/DebugWindow.hs | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/XMonad/Util/DebugWindow.hs b/XMonad/Util/DebugWindow.hs
index d4f5138..283297b 100644
--- a/XMonad/Util/DebugWindow.hs
+++ b/XMonad/Util/DebugWindow.hs
@@ -41,7 +41,15 @@ debugWindow w =  do
   case w' of
     Nothing                                   ->
       return $ "(deleted window " ++ wx ++ ")"
-    Just (WindowAttributes x y wid ht bw m o) -> do
+    Just (WindowAttributes
+      { wa_x                 = x
+      , wa_y                 = y
+      , wa_width             = wid
+      , wa_height            = ht
+      , wa_border_width      = bw
+      , wa_map_state         = m
+      , wa_override_redirect = o
+      }) -> do
       c' <- withDisplay $ \d ->
             io (getWindowProperty8 d wM_CLASS w)
       let c = case c' of
