diff --git a/pandoc-citeproc.cabal b/pandoc-citeproc.cabal
index 40c27af..595bb53 100644
--- a/pandoc-citeproc.cabal
+++ b/pandoc-citeproc.cabal
@@ -148,7 +148,7 @@ executable pandoc-citeproc
                       pandoc-types >= 1.16 && < 1.17,
                       pandoc >= 1.16 && < 1.18,
                       aeson,
-                      aeson-pretty, yaml, bytestring, syb, attoparsec, text,
+                      aeson-pretty >= 0.8, yaml, bytestring, syb, attoparsec, text,
                       filepath
     other-modules:    Paths_pandoc_citeproc
                       Prelude
diff --git a/pandoc-citeproc.hs b/pandoc-citeproc.hs
index f72fa41..2c1ce27 100644
--- a/pandoc-citeproc.hs
+++ b/pandoc-citeproc.hs
@@ -12,7 +12,7 @@ import qualified Data.ByteString.Char8 as B8
 import Data.Attoparsec.ByteString.Char8 as Attoparsec
 import qualified Data.Text as T
 import Data.Text.Encoding (encodeUtf8)
-import Data.Aeson.Encode.Pretty (encodePretty', Config(..))
+import Data.Aeson.Encode.Pretty (encodePretty', Config(..), Indent(Spaces), NumberFormat(Generic))
 import System.Console.GetOpt
 import Control.Monad
 import System.IO
@@ -62,8 +62,9 @@ main = do
                  B8.intercalate (B.singleton 10) .
                  map (unescapeTags . toByteString . (:[]))
             else B8.putStrLn . unescapeUnicode . B.concat . BL.toChunks .
-              encodePretty' Config{ confIndent = 2
-                                  , confCompare = compare }
+              encodePretty' Config{ confIndent = Spaces 2
+                                  , confCompare = compare
+                                  , confNumFormat = Generic }
      else toJSONFilter doCites
 
 formatFromExtension :: FilePath -> Maybe BibFormat
