diff -wbBur mailutils-2.2/lib/Makefile.in mailutils-2.2.q/lib/Makefile.in
--- mailutils-2.2/lib/Makefile.in	2010-09-08 13:58:58.000000000 +0400
+++ mailutils-2.2.q/lib/Makefile.in	2013-05-29 14:51:14.064888989 +0400
@@ -1861,7 +1861,8 @@
 	      -e 's|@''REPLACE_VSPRINTF''@|$(REPLACE_VSPRINTF)|g' \
 	      -e '/definitions of _GL_FUNCDECL_RPL/r $(CXXDEFS_H)' \
 	      -e '/definition of _GL_ARG_NONNULL/r $(ARG_NONNULL_H)' \
-	      -e '/definition of _GL_WARN_ON_USE/r $(WARN_ON_USE_H)'; \
+	      -e '/definition of _GL_WARN_ON_USE/r $(WARN_ON_USE_H)' \
+	      -e 's|_GL_WARN_ON_USE (gets.*||'; \
 	} > $@-t && \
 	mv $@-t $@
 
