--- a/chrome/browser/ui/libgtk2ui/gtk2_ui.cc
+++ b/chrome/browser/ui/libgtk2ui/gtk2_ui.cc
@@ -454,12 +454,18 @@
 
 // Queries GTK for its font DPI setting and returns the number of pixels in a
 // point.
+#if GTK_MAJOR_VERSION == 2
 double GetPixelsInPoint(float device_scale_factor) {
+#else
+double GetPixelsInPoint() {
+#endif
   double dpi = GetDPI();
 
+#if GTK_MAJOR_VERSION == 2
   // Take device_scale_factor into account — if Chrome already scales the
   // entire UI up by 2x, we should not also scale up.
   dpi /= device_scale_factor;
+#endif
 
   // There are 72 points in an inch.
   return dpi / 72.0;
@@ -1371,8 +1377,13 @@
     // Round the value when converting to pixels to match GTK's logic.
     const double size_points = pango_font_description_get_size(desc) /
         static_cast<double>(PANGO_SCALE);
+#if GTK_MAJOR_VERSION == 2
     default_font_size_pixels_ = static_cast<int>(
         GetPixelsInPoint(device_scale_factor_) * size_points + 0.5);
+#else
+    default_font_size_pixels_ = static_cast<int>(
+        GetPixelsInPoint() * size_points + 0.5);
+#endif
     query.point_size = static_cast<int>(size_points);
   }
 
@@ -1405,8 +1416,13 @@
 float Gtk2UI::GetDeviceScaleFactor() const {
   if (display::Display::HasForceDeviceScaleFactor())
     return display::Display::GetForcedDeviceScaleFactor();
+#if GTK_MAJOR_VERSION == 2
   const int kCSSDefaultDPI = 96;
   const float scale = GetDPI() / kCSSDefaultDPI;
+#else
+  const float scale = static_cast<float>(gdk_window_get_scale_factor(
+      gdk_screen_get_root_window(gdk_screen_get_default())));
+#endif
 
   // Blacklist scaling factors <130% (crbug.com/484400) and round
   // to 1 decimal to prevent rendering problems (crbug.com/485183).
