diff -rup apache-couchdb-1.2.0/configure apache-couchdb-1.2.0.new/configure
--- apache-couchdb-1.2.0/configure	2012-03-29 23:05:41.000000000 +0200
+++ apache-couchdb-1.2.0.new/configure	2012-04-08 13:50:14.923693056 +0200
@@ -18234,7 +18234,7 @@ echo "$as_me: error: $erlang_version_err
     fi
 fi
 
-otp_release="`${ERL} -noshell -eval 'io:put_chars(erlang:system_info(otp_release)).' -s erlang halt`"
+otp_release="`${ERL} -smp disable -noshell -eval 'io:put_chars(erlang:system_info(otp_release)).' -s erlang halt`"
 
  if test x$otp_release \> xR13B03; then
   USE_OTP_NIFS_TRUE=
@@ -18253,7 +18253,7 @@ else
 fi
 
 
-has_crypto=`${ERL} -eval "case application:load(crypto) of ok -> ok; _ -> exit(no_crypto) end." -noshell -s init stop`
+has_crypto=`${ERL} -smp disable -eval "case application:load(crypto) of ok -> ok; _ -> exit(no_crypto) end." -noshell -s init stop`
 
 if test -n "$has_crypto"; then
     { { echo "$as_me:$LINENO: error: Could not find the Erlang crypto library. Has Erlang been compiled with OpenSSL support?" >&5
