diff -Nur HandBrake-0.9.5.orig/gtk/configure.ac HandBrake-0.9.5/gtk/configure.ac
--- HandBrake-0.9.5.orig/gtk/configure.ac	2011-03-07 13:06:20.333338164 -0800
+++ HandBrake-0.9.5/gtk/configure.ac	2011-03-07 13:07:46.346670456 -0800
@@ -94,7 +94,7 @@
 	mingw_flag=yes
     ;;
   *)
-	GHB_PACKAGES="gtk+-2.0 >= 2.10 gthread-2.0 gstreamer-0.10 gstreamer-interfaces-0.10 gstreamer-video-0.10 gstreamer-pbutils-0.10 gio-2.0 libnotify gudev-1.0"
+	GHB_PACKAGES="gtk+-2.0 >= 2.10 gthread-2.0 gstreamer-0.10 gstreamer-interfaces-0.10 gstreamer-video-0.10 gstreamer-pbutils-0.10 gio-2.0 libnotify gudev-1.0 dbus-glib-1"
 	mingw_flag=no
     ;;
 esac
