diff -wbBur xloadimage.4.1/imagetypes.h xloadimage.4.1.my/imagetypes.h
--- xloadimage.4.1/imagetypes.h	2011-02-11 11:47:37.000000000 +0000
+++ xloadimage.4.1.my/imagetypes.h	2011-02-11 16:04:35.000000000 +0000
@@ -94,15 +94,9 @@
   {niffIdent,      niffLoad,      niffDump,    "niff",      "Native Image File Format (NIFF)"},
   {sunRasterIdent, sunRasterLoad, NULL,        "sunraster", "Sun Rasterfile"},
   {gifIdent,       gifLoad,       NULL,        "gif",       "GIF Image"},
-#ifdef HAVE_LIBJPEG
   {jpegIdent,      jpegLoad,      jpegDump,    "jpeg",      "JFIF-style JPEG Image"},
-#endif
-#ifdef HAVE_LIBTIFF
   {tiffIdent,      tiffLoad,      tiffDump,    "tiff",      "TIFF image"},
-#endif
-#ifdef HAVE_LIBPNG
   {pngIdent,       pngLoad,       NULL,        "png",       "PNG image"},
-#endif
   {fbmIdent,       fbmLoad,       NULL,        "fbm",       "FBM Image"},
   {cmuwmIdent,     cmuwmLoad,     NULL,        "cmuraster", "CMU WM Raster"},
   {pbmIdent,       pbmLoad,       pbmDump,     "pbm",       "Portable Bit Map (PBM, PGM, PPM)"},
diff -wbBur xloadimage.4.1/Imakefile xloadimage.4.1.my/Imakefile
--- xloadimage.4.1/Imakefile	2011-02-11 12:05:53.000000000 +0000
+++ xloadimage.4.1.my/Imakefile	2011-02-11 16:06:21.000000000 +0000
@@ -40,7 +40,7 @@
  PNG_SRC = png.c
  PNG_OBJ = png.o
 
-  EXTRA_DEFINES = $(TIFF_DEFS) $(JPEG_DEFS) $(PNG_DEFS)
+  EXTRA_DEFINES = $(TIFF_DEFS) $(JPEG_DEFS) $(PNG_DEFS) -DHAVE_LIBJPEG -DHAVE_LIBTIFF -DHAVE_LIBPNG
 LOCAL_LIBRARIES = $(TIFF_LIB)  $(JPEG_LIB) $(PNG_LIB) $(XLIB)
 
 SRCS = new.c niff.c value.c zio.c cmuwmraster.c faces.c fbm.c gif.c imagetypes.c img.c  mac.c mcidas.c mc_tables.c pbm.c pcx.c pdsuncomp.c  rle.c rlelib.c sunraster.c vff.c vicar.c xbitmap.c xpixmap.c  xwd.c bright.c clip.c compress.c dither.c fill.c halftone.c  merge.c reduce.c rotate.c smooth.c undither.c zoom.c config.c misc.c options.c root.c send.c window.c  xloadimage.c  $(TIFF_SRC) $(JPEG_SRC) $(PNG_SRC)
diff -wbBur xloadimage.4.1/Makefile xloadimage.4.1.my/Makefile
--- xloadimage.4.1/Makefile	2011-02-11 12:06:16.000000000 +0000
+++ xloadimage.4.1.my/Makefile	2011-02-11 16:06:24.000000000 +0000
@@ -2,7 +2,7 @@
 # $Xorg: imake.c,v 1.6 2001/02/09 02:03:15 xorgcvs Exp $
 
 # ----------------------------------------------------------------------
-# Makefile generated from "Imake.tmpl" and </tmp/IIf.yUZBVe>
+# Makefile generated from "Imake.tmpl" and </tmp/IIf.7u155R>
 # $Xorg: Imake.tmpl,v 1.4 2000/08/17 19:41:46 cpqbld Exp $
 # $XdotOrg: xc/config/cf/Imake.tmpl,v 1.9 2005/01/24 06:37:31 daniels Exp $
 #
@@ -1063,7 +1063,7 @@
  PNG_SRC = png.c
  PNG_OBJ = png.o
 
-  EXTRA_DEFINES = $(TIFF_DEFS) $(JPEG_DEFS) $(PNG_DEFS)
+  EXTRA_DEFINES = $(TIFF_DEFS) $(JPEG_DEFS) $(PNG_DEFS) -DHAVE_LIBJPEG -DHAVE_LIBTIFF -DHAVE_LIBPNG
 LOCAL_LIBRARIES = $(TIFF_LIB)  $(JPEG_LIB) $(PNG_LIB) $(XLIB)
 
 SRCS = new.c niff.c value.c zio.c cmuwmraster.c faces.c fbm.c gif.c imagetypes.c img.c  mac.c mcidas.c mc_tables.c pbm.c pcx.c pdsuncomp.c  rle.c rlelib.c sunraster.c vff.c vicar.c xbitmap.c xpixmap.c  xwd.c bright.c clip.c compress.c dither.c fill.c halftone.c  merge.c reduce.c rotate.c smooth.c undither.c zoom.c config.c misc.c options.c root.c send.c window.c  xloadimage.c  $(TIFF_SRC) $(JPEG_SRC) $(PNG_SRC)
diff -wbBur xloadimage.4.1/png.c xloadimage.4.1.my/png.c
--- xloadimage.4.1/png.c	2011-02-11 11:47:37.000000000 +0000
+++ xloadimage.4.1.my/png.c	2011-02-11 16:06:50.000000000 +0000
@@ -18,6 +18,7 @@
  * 
  */
 
+#define _GETOPT_H
 
 #include "image.h"	/* xloadimage declarations */
 #ifdef HAVE_LIBPNG
diff -wbBur xloadimage.4.1/jpeg.c xloadimage.4.1.my/jpeg.c
--- xloadimage.4.1/jpeg.c	2011-02-11 14:47:37.000000000 +0300
+++ xloadimage.4.1.my/jpeg.c	2011-11-21 14:32:30.000000000 +0400
@@ -319,6 +319,7 @@
       if (verbose)
 	printf("  Using arithmetic coding.\n");
       cinfo->arith_code = TRUE;
+    }
 #else
       fprintf(stderr, "jpegDump: sorry, arithmetic coding not supported\n");
     }
