diff -rupN pcsx2-1.0.0-r5350-sources.orig/common/include/PS2Eext.h pcsx2-1.0.0-r5350-sources/common/include/PS2Eext.h
--- pcsx2-1.0.0-r5350-sources.orig/common/include/PS2Eext.h	2013-06-02 18:24:59.369581900 +0200
+++ pcsx2-1.0.0-r5350-sources/common/include/PS2Eext.h	2013-06-02 18:27:39.091107300 +0200
@@ -144,10 +144,9 @@ struct PluginConf
     int ReadInt(const std::string& item, int defval)
     {
         int value = defval;
-	int err = 0;
         std::string buf = item + " = %d\n";
 
-        if (ConfFile) err=fscanf(ConfFile, buf.c_str(), &value);
+        if (ConfFile) fscanf(ConfFile, buf.c_str(), &value);
 
         return value;
     }
@@ -157,7 +156,7 @@ struct PluginConf
 	int err = 0;
         std::string buf = item + " = %d\n";
 
-        if (ConfFile) err=fprintf(ConfFile, buf.c_str(), value);
+        if (ConfFile) fprintf(ConfFile, buf.c_str(), value);
     }
 };
 
diff -rupN pcsx2-1.0.0-r5350-sources.orig/plugins/GSdx/GSDeviceSW.cpp pcsx2-1.0.0-r5350-sources/plugins/GSdx/GSDeviceSW.cpp
--- pcsx2-1.0.0-r5350-sources.orig/plugins/GSdx/GSDeviceSW.cpp	2013-06-02 18:24:59.922788200 +0200
+++ pcsx2-1.0.0-r5350-sources/plugins/GSdx/GSDeviceSW.cpp	2013-06-02 18:29:56.088850400 +0200
@@ -50,7 +50,7 @@ bool GSDeviceSW::Reset(int w, int h)
 
 GSTexture* GSDeviceSW::CreateSurface(int type, int w, int h, bool msaa, int format)
 {
-	if(format != 0) return false; // there is only one format
+	if(format != 0) return NULL; // there is only one format
 
 	return new GSTextureSW(type, w, h);
 }
diff -rupN pcsx2-1.0.0-r5350-sources.orig/plugins/GSdx/GS.h pcsx2-1.0.0-r5350-sources/plugins/GSdx/GS.h
--- pcsx2-1.0.0-r5350-sources.orig/plugins/GSdx/GS.h	2013-06-02 18:24:59.916311000 +0200
+++ pcsx2-1.0.0-r5350-sources/plugins/GSdx/GS.h	2013-06-02 18:29:28.940530500 +0200
@@ -807,7 +807,8 @@ REG_END2
 			if(PSM == PSM_PSMT4) return TW > 7 || TH > 7;
 		}
 
-		return (TBW << 6) < (1u << TW);
+    // The recast of TBW seems useless but it avoid tons of warning from GCC...
+    return ((uint32)TBW << 6u) < (1u << TW);
 	}
 REG_END2
 
diff -rupN pcsx2-1.0.0-r5350-sources.orig/plugins/GSdx/stdafx.h pcsx2-1.0.0-r5350-sources/plugins/GSdx/stdafx.h
--- pcsx2-1.0.0-r5350-sources.orig/plugins/GSdx/stdafx.h	2013-06-02 18:24:59.951638400 +0200
+++ pcsx2-1.0.0-r5350-sources/plugins/GSdx/stdafx.h	2013-06-02 18:32:05.092233800 +0200
@@ -430,7 +430,11 @@ struct aligned_free_second {template<cla
 
 	#ifdef __GNUC__
 
-	__forceinline unsigned long long __rdtsc()
+	// gcc 4.8 define __rdtsc but unfortunately the compiler crash...
+	// The redefine allow to skip the gcc __rdtsc version -- Gregory
+	#define __rdtsc _lnx_rdtsc
+	//__forceinline unsigned long long __rdtsc()
+	__forceinline unsigned long long _lnx_rdtsc()
 	{
 		#if defined(__amd64__) || defined(__x86_64__)
 		unsigned long long low, high;
