diff -ruN matio-1.5.6.orig/src/mat73.c matio-1.5.6/src/mat73.c
--- matio-1.5.6.orig/src/mat73.c	2016-04-12 16:19:26.714810306 +0300
+++ matio-1.5.6/src/mat73.c	2016-04-12 16:20:03.509186059 +0300
@@ -607,7 +607,7 @@
                 cells[i] = Mat_VarCalloc();
                 cells[i]->internal->hdf5_ref = ref_ids[i];
                 /* Closing of ref_id is done in Mat_H5ReadNextReferenceInfo */
-                ref_id = H5Rdereference(dset_id,H5R_OBJECT,ref_ids+i);
+                ref_id = H5Rdereference1(dset_id,H5R_OBJECT,ref_ids+i);
                 cells[i]->internal->id = ref_id;
                 cells[i]->internal->fp = matvar->internal->fp;
                 Mat_H5ReadNextReferenceInfo(ref_id,cells[i],mat);
@@ -902,7 +902,7 @@
                         /* Closing of ref_id is done in
                          * Mat_H5ReadNextReferenceInfo
                          */
-                        ref_id = H5Rdereference(field_id,H5R_OBJECT,
+                        ref_id = H5Rdereference1(field_id,H5R_OBJECT,
                                                 ref_ids+l);
                         fields[l*nfields+k]->internal->id=ref_id;
                         Mat_H5ReadNextReferenceInfo(ref_id,fields[l*nfields+k],mat);
@@ -1073,7 +1073,7 @@
                         cells[i] = Mat_VarCalloc();
                         cells[i]->internal->hdf5_ref = ref_ids[i];
                         /* Closing of ref_id is done in Mat_H5ReadNextReferenceInfo */
-                        ref_id = H5Rdereference(dset_id,H5R_OBJECT,ref_ids+i);
+                        ref_id = H5Rdereference1(dset_id,H5R_OBJECT,ref_ids+i);
                         cells[i]->internal->id=ref_id;
                         cells[i]->internal->fp=matvar->internal->fp;
                         Mat_H5ReadNextReferenceInfo(ref_id,cells[i],mat);
@@ -2273,7 +2273,7 @@
                 H5Iinc_ref(ref_id);
             }
             if ( 0 < matvar->internal->hdf5_ref ) {
-                dset_id = H5Rdereference(ref_id,H5R_OBJECT,&matvar->internal->hdf5_ref);
+                dset_id = H5Rdereference1(ref_id,H5R_OBJECT,&matvar->internal->hdf5_ref);
             } else {
                 dset_id = ref_id;
                 H5Iinc_ref(dset_id);
@@ -2526,7 +2526,7 @@
                 H5Iinc_ref(ref_id);
             }
             if ( 0 < matvar->internal->hdf5_ref ) {
-                dset_id = H5Rdereference(ref_id,H5R_OBJECT,&matvar->internal->hdf5_ref);
+                dset_id = H5Rdereference1(ref_id,H5R_OBJECT,&matvar->internal->hdf5_ref);
             } else {
                 dset_id = ref_id;
                 H5Iinc_ref(dset_id);
@@ -2842,7 +2842,7 @@
                         /* Closing of ref_id is done in
                          * Mat_H5ReadNextReferenceInfo
                          */
-                        ref_id = H5Rdereference(dset_id,H5R_OBJECT,ref_ids+i);
+                        ref_id = H5Rdereference1(dset_id,H5R_OBJECT,ref_ids+i);
                         cells[i]->internal->id=ref_id;
                         cells[i]->internal->fp=matvar->internal->fp;
                         Mat_H5ReadNextReferenceInfo(ref_id,cells[i],mat);
