diff -Naur cinnamon-control-center-1.8.2.orig/panels/power/cc-power-panel.c cinnamon-control-center-1.8.2/panels/power/cc-power-panel.c
--- cinnamon-control-center-1.8.2.orig/panels/power/cc-power-panel.c	2013-08-14 22:10:57.491152667 +0200
+++ cinnamon-control-center-1.8.2/panels/power/cc-power-panel.c	2013-08-14 21:58:36.533247961 +0200
@@ -978,9 +978,6 @@
   has_lid = up_client_get_lid_is_present (self->priv->up_client);
 
 out:
-  gtk_widget_set_visible (WID (priv->builder, "combobox_lid_ac"), has_lid);
-  gtk_widget_set_visible (WID (priv->builder, "label_lid_action"), has_lid);
-  gtk_widget_set_visible (WID (priv->builder, "combobox_lid_battery"), has_batteries && has_lid);    
   gtk_widget_set_visible (WID (priv->builder, "label_header_battery"), has_batteries);
   gtk_widget_set_visible (WID (priv->builder, "label_header_ac"), has_batteries);
   gtk_widget_set_visible (WID (priv->builder, "combobox_sleep_battery"), has_batteries);
@@ -1067,26 +1064,6 @@
   g_signal_connect (widget, "changed",
                     G_CALLBACK (combo_enum_changed_cb),
                     self);
-
-  value = g_settings_get_enum (self->priv->gsd_settings, "lid-close-ac-action");
-  widget = GTK_WIDGET (gtk_builder_get_object (self->priv->builder,
-                                               "combobox_lid_ac"));
-  disable_unavailable_combo_items (self, GTK_COMBO_BOX (widget));
-  set_value_for_combo (GTK_COMBO_BOX (widget), value);
-  g_object_set_data (G_OBJECT(widget), "_gsettings_key", "lid-close-ac-action");
-  g_signal_connect (widget, "changed",
-                    G_CALLBACK (combo_enum_changed_cb),
-                    self);
-
-  value = g_settings_get_enum (self->priv->gsd_settings, "lid-close-battery-action");
-  widget = GTK_WIDGET (gtk_builder_get_object (self->priv->builder,
-                                               "combobox_lid_battery"));
-  disable_unavailable_combo_items (self, GTK_COMBO_BOX (widget));
-  set_value_for_combo (GTK_COMBO_BOX (widget), value);
-  g_object_set_data (G_OBJECT(widget), "_gsettings_key", "lid-close-battery-action");
-  g_signal_connect (widget, "changed",
-                    G_CALLBACK (combo_enum_changed_cb),
-                    self);
 
   widget = WID (self->priv->builder, "vbox_power");
   gtk_widget_reparent (widget, (GtkWidget *) self);
diff -Naur cinnamon-control-center-1.8.2.orig/panels/power/power.ui cinnamon-control-center-1.8.2/panels/power/power.ui
--- cinnamon-control-center-1.8.2.orig/panels/power/power.ui	2013-05-22 18:25:19.000000000 +0200
+++ cinnamon-control-center-1.8.2/panels/power/power.ui	2013-08-14 21:57:54.872240364 +0200
@@ -222,50 +222,6 @@
               </packing>
             </child>
             <child>
-              <object class="GtkLabel" id="label_lid_action">
-                <property name="visible">True</property>
-                <property name="can_focus">False</property>
-                <property name="halign">end</property>
-                <property name="label" translatable="yes">When the lid is closed</property>
-              </object>
-              <packing>
-                <property name="left_attach">0</property>
-                <property name="top_attach">3</property>
-              </packing>
-            </child>
-            <child>
-              <object class="GtkComboBox" id="combobox_lid_battery">
-                <property name="visible">True</property>
-                <property name="can_focus">False</property>
-                <property name="model">liststore_lid</property>
-                <property name="hexpand">True</property>
-                <accessibility>
-                  <relation type="labelled-by" target="label_header_battery"/>
-                  <relation type="labelled-by" target="label_lid_action"/>
-                </accessibility>
-              </object>
-              <packing>
-                <property name="left_attach">1</property>
-                <property name="top_attach">3</property>
-              </packing>
-            </child>
-            <child>
-              <object class="GtkComboBox" id="combobox_lid_ac">
-                <property name="visible">True</property>
-                <property name="can_focus">False</property>
-                <property name="model">liststore_lid</property>
-                <property name="hexpand">True</property>
-                <accessibility>
-                  <relation type="labelled-by" target="label_header_ac"/>
-                  <relation type="labelled-by" target="label_lid_action"/>
-                </accessibility>
-              </object>
-              <packing>
-                <property name="left_attach">2</property>
-                <property name="top_attach">3</property>
-              </packing>
-            </child>
-            <child>
               <object class="GtkSeparator" id="separator_indicator">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
@@ -468,12 +424,10 @@
   </object>
   <object class="GtkSizeGroup" id="sizegroup_combos">
     <widgets>
-      <widget name="combobox_lid_battery"/>
-      <widget name="combobox_lid_ac"/>    
       <widget name="combobox_critical"/>
       <widget name="combobox_sleep_battery"/>
       <widget name="combobox_sleep_ac"/>
       <widget name="combobox_sleep_ac"/>
     </widgets>
   </object>
-</interface>
\ No newline at end of file
+</interface>
diff -Naur cinnamon-control-center-1.8.2.orig/panels/screen/cc-screen-panel.c cinnamon-control-center-1.8.2/panels/screen/cc-screen-panel.c
--- cinnamon-control-center-1.8.2.orig/panels/screen/cc-screen-panel.c	2013-08-14 22:10:57.541153876 +0200
+++ cinnamon-control-center-1.8.2/panels/screen/cc-screen-panel.c	2013-08-14 21:55:41.905692072 +0200
@@ -352,10 +352,6 @@
                       1, &value,
                       -1);
 
-  /* set both battery and ac keys */
-  g_settings_set_int (self->priv->gsd_settings, "sleep-display-ac", value);
-  g_settings_set_int (self->priv->gsd_settings, "sleep-display-battery", value);
-
   set_idle_delay_from_dpms (self, value);
 }
 
@@ -385,7 +381,7 @@
 {
   GtkTreeIter iter;
   GtkTreeModel *model;
-  gint value;
+  guint value;
   gint value_tmp, value_prev;
   gboolean ret;
   guint i;
@@ -400,7 +396,7 @@
   i = 0;
 
   /* try to make the UI match the AC setting */
-  value = g_settings_get_int (self->priv->gsd_settings, "sleep-display-ac");
+  g_settings_get (self->priv->session_settings, "idle-delay", "u", &value);
   do
     {
       gtk_tree_model_get (model, &iter,
@@ -511,7 +507,7 @@
   /* bind the auto dim checkbox */
   widget = WID ("screen_auto_reduce_checkbutton");
   g_settings_bind (self->priv->gsd_settings,
-                   "idle-dim-battery",
+                   "idle-dim",
                    widget, "active",
                    G_SETTINGS_BIND_DEFAULT);
 
