diff -ur mixxx-2.0.0/build/mixxx.py mixxx-2.0.0-patched/build/mixxx.py
--- ./build/depends.py.orig	2016-02-09 04:30:54.860754912 +0000
+++ ./build/depends.py	2016-02-09 04:29:53.286383033 +0000
@@ -359,7 +359,6 @@ class Qt(Dependence):
             qtdir = build.env['QTDIR']
             framework_path = Qt.find_framework_libdir(qtdir, qt5)
             if os.path.isdir(framework_path):
-                build.env.Append(LINKFLAGS="-Wl,-rpath," + framework_path)
                 build.env.Append(LINKFLAGS="-L" + framework_path)

