diff --git a/rts/System/ThreadPool.cpp b/rts/System/ThreadPool.cpp
index 0ce4b11..34e53ab 100644
--- a/rts/System/ThreadPool.cpp
+++ b/rts/System/ThreadPool.cpp
@@ -136,12 +136,11 @@ static bool DoTask(boost::shared_lock<boost::shared_mutex>& lk_)
 static bool DoTask(std::shared_ptr<ITaskGroup> tg)
 {
 	auto p = tg->GetTask();
-	const bool f = p;
-	if (f) {
+	if (p) {
 		SCOPED_MT_TIMER("::ThreadWorkers (accumulated)");
 		(*p)();
 	}
-	return f;
+	return static_cast<bool>(p);
 }
 
 
