--- src/sage/env.py.orig	2014-06-25 10:47:25.827203554 +1200
+++ src/sage/env.py	2014-06-25 10:49:07.667402821 +1200
@@ -90,10 +90,10 @@
 
 # bunch of sage directories and files
 _add_variable_or_fallback('SAGE_ROOT',       None)
-_add_variable_or_fallback('SAGE_LOCAL',      opj('$SAGE_ROOT', 'local'))
-_add_variable_or_fallback('SAGE_ETC',        opj('$SAGE_LOCAL', 'etc'))
+_add_variable_or_fallback('SAGE_LOCAL',      '/usr')
+_add_variable_or_fallback('SAGE_ETC',        '/etc/sage')
 _add_variable_or_fallback('SAGE_INC',        opj('$SAGE_LOCAL', 'include'))
-_add_variable_or_fallback('SAGE_SHARE',      opj('$SAGE_LOCAL', 'share'))
+_add_variable_or_fallback('SAGE_SHARE',      opj('$SAGE_LOCAL', 'share', 'sage'))
 
 _add_variable_or_fallback('SAGE_SRC',        opj('$SAGE_ROOT', 'src'))
 
@@ -102,13 +102,13 @@
 except AttributeError:  # in case of use inside virtualenv
     sitepackages_dirs = [os.path.join(os.path.dirname(site.__file__),
                                      'site-packages')]
-_add_variable_or_fallback('SITE_PACKAGES',   sitepackages_dirs)
+_add_variable_or_fallback('SITE_PACKAGES',   sitepackages_dirs + ['/usr/lib/sage/site-packages'])
 
 _add_variable_or_fallback('SAGE_LIB',        SITE_PACKAGES[0])
 
 _add_variable_or_fallback('SAGE_CYTHONIZED', opj('$SAGE_SRC', 'build', 'cythonized'))
 
-_add_variable_or_fallback('SAGE_EXTCODE',    opj('$SAGE_SHARE', 'sage', 'ext'))
+_add_variable_or_fallback('SAGE_EXTCODE',    opj('$SAGE_SHARE', 'ext'))
 _add_variable_or_fallback('SAGE_LOGS',       opj('$SAGE_ROOT', 'logs', 'pkgs'))
 _add_variable_or_fallback('SAGE_SPKG_INST',  opj('$SAGE_LOCAL', 'var', 'lib', 'sage', 'installed'))
 _add_variable_or_fallback('SAGE_DOC_SRC',    opj('$SAGE_SRC', 'doc'))
