diff -rupN SFML-1.6.orig/include/SFML/System/Resource.hpp SFML-1.6/include/SFML/System/Resource.hpp
--- SFML-1.6.orig/include/SFML/System/Resource.hpp	2010-01-27 14:53:54.000000000 +0100
+++ SFML-1.6/include/SFML/System/Resource.hpp	2013-10-11 15:22:44.416388808 +0200
@@ -29,7 +29,7 @@
 // Headers
 ////////////////////////////////////////////////////////////
 #include <set>
-
+#include <cstddef>
 
 namespace sf
 {
diff -rupN SFML-1.6.orig/src/SFML/Audio/Makefile SFML-1.6/src/SFML/Audio/Makefile
--- SFML-1.6.orig/src/SFML/Audio/Makefile	2010-01-27 15:00:19.000000000 +0100
+++ SFML-1.6/src/SFML/Audio/Makefile	2013-10-11 15:24:44.588725944 +0200
@@ -25,7 +25,7 @@ $(OBJ): %.o: %.cpp
 	$(CPP) -o $@ -c $< $(CFLAGS)
 
 $(OBJVORBIS): %.o: %.c
-	$(CC) -o $@ -c $< $(CFLAGSEXT)
+	$(CC) -o $@ -c $< $(CFLAGS)
 
 .PHONY: clean mrproper
 
diff -rupN SFML-1.6.orig/src/SFML/Graphics/GraphicsContext.hpp SFML-1.6/src/SFML/Graphics/GraphicsContext.hpp
--- SFML-1.6.orig/src/SFML/Graphics/GraphicsContext.hpp	2010-01-27 15:00:05.000000000 +0100
+++ SFML-1.6/src/SFML/Graphics/GraphicsContext.hpp	2013-10-11 15:22:44.419722114 +0200
@@ -30,7 +30,7 @@
 ////////////////////////////////////////////////////////////
 #include <SFML/Config.hpp>
 #include <SFML/System/NonCopyable.hpp>
-#include <SFML/Graphics/GLEW/glew.h>
+#include <GL/glew.h>
 #include <iostream>
 #include <string>
 
diff -rupN SFML-1.6.orig/src/SFML/Graphics/ImageLoader.cpp SFML-1.6/src/SFML/Graphics/ImageLoader.cpp
--- SFML-1.6.orig/src/SFML/Graphics/ImageLoader.cpp	2010-01-27 15:00:05.000000000 +0100
+++ SFML-1.6/src/SFML/Graphics/ImageLoader.cpp	2013-10-11 15:22:44.419722114 +0200
@@ -28,11 +28,12 @@
 #include <SFML/Graphics/ImageLoader.hpp>
 extern "C"
 {
-    #include <SFML/Graphics/libjpeg/jpeglib.h>
-    #include <SFML/Graphics/libjpeg/jerror.h>
+    #include <jpeglib.h>
+    #include <jerror.h>
 }
-#include <SFML/Graphics/libpng/png.h>
-#include <SFML/Graphics/SOIL/SOIL.h>
+#include <png.h>
+#include <SOIL/SOIL.h>
+#include <cstring>
 #include <iostream>
 
 
@@ -44,7 +45,7 @@ namespace
     void PngErrorHandler(png_structp Png, png_const_charp Message)
     {
         std::cerr << "Failed to write PNG image. Reason : " << Message << std::endl;
-        longjmp(Png->jmpbuf, 1);
+        longjmp(png_jmpbuf(Png), 1);
     }
 }
 
@@ -289,7 +290,7 @@ bool ImageLoader::WritePng(const std::st
     }
 
     // For proper error handling...
-    if (setjmp(Png->jmpbuf))
+    if (setjmp(png_jmpbuf(Png)))
     {
         png_destroy_write_struct(&Png, &PngInfo);
         return false;
diff -rupN SFML-1.6.orig/src/SFML/Graphics/Makefile SFML-1.6/src/SFML/Graphics/Makefile
--- SFML-1.6.orig/src/SFML/Graphics/Makefile	2010-01-27 15:00:05.000000000 +0100
+++ SFML-1.6/src/SFML/Graphics/Makefile	2013-10-11 15:26:37.971432986 +0200
@@ -1,15 +1,5 @@
 SRC     = $(wildcard *.cpp)
-SRCGLEW = $(wildcard ./GLEW/*.c)
-SRCJPEG = $(wildcard ./libjpeg/*.c)
-SRCPNG  = $(wildcard ./libpng/*.c)
-SRCSOIL = $(wildcard ./SOIL/*.c)
-SRCZLIB = $(wildcard ./zlib/*.c)
 OBJ     = $(SRC:.cpp=.o)
-OBJGLEW = $(SRCGLEW:.c=.o)
-OBJJPEG = $(SRCJPEG:.c=.o)
-OBJPNG  = $(SRCPNG:.c=.o)
-OBJSOIL = $(SRCSOIL:.c=.o)
-OBJZLIB = $(SRCZLIB:.c=.o)
 
 ifeq ($(STATIC), yes)
     LIB     = libsfml-graphics-s.a
@@ -23,22 +13,19 @@ endif
 
 all: $(LIB)
 
-libsfml-graphics-s.a: $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
-	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
+libsfml-graphics-s.a: $(OBJ)
+	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ)
 
-libsfml-graphics.so: $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
-	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB) -lfreetype -lX11 -lGL
+libsfml-graphics.so: $(OBJ)
+	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) -lGLEW -ljpeg -lpng -lz -lfreetype -lX11 -lGL
 
 $(OBJ): %.o: %.cpp
 	$(CPP) -o $@ -c $< $(CFLAGS) -I/usr/include/freetype2
 
-$(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB): %.o: %.c
-	$(CC) -o $@ -c $< $(CFLAGSEXT) -DSTBI_FAILURE_USERMSG
-
 .PHONY: clean mrproper
 
 clean:
-	@rm -rf $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
+	@rm -rf $(OBJ)
 
 mrproper: clean
 	@rm -rf $(LIBNAME)
diff -rupN SFML-1.6.orig/src/SFML/Makefile SFML-1.6/src/SFML/Makefile
--- SFML-1.6.orig/src/SFML/Makefile	2010-01-27 15:00:37.000000000 +0100
+++ SFML-1.6/src/SFML/Makefile	2013-10-11 15:29:14.687273136 +0200
@@ -1,10 +1,10 @@
 export DEBUGBUILD = no
-export STATIC     = no
+export STATIC     = yes
 
 ifeq ($(DEBUGBUILD), yes)
     DEBUGFLAGS = -g -DDEBUG
 else
-    DEBUGFLAGS = -O2 -DNDEBUG
+    DEBUGFLAGS = -DNDEBUG
 endif
 
 ifeq ($(STATIC), no)
@@ -17,9 +17,9 @@ endif
 
 export CC         = gcc
 export CPP        = g++
-export CFLAGS     = -W -Wall -pedantic -I../../../include -I../../ $(DEBUGFLAGS) $(BUILDFLAGS)
-export CFLAGSEXT  = -I../../../include -I../.. $(DEBUGFLAGS) $(BUILDFLAGS)
-export LDFLAGS    = $(LINKFLAGS)
+export CFLAGS     += -W -Wall -pedantic -I../../../include -I../../../../staging/usr/include -I../../ $(DEBUGFLAGS) $(BUILDFLAGS)
+export CFLAGSEXT  = -I../../../include -I../../../../staging/usr/include -I../.. $(DEBUGFLAGS) $(BUILDFLAGS)
+export LDFLAGS    += $(LINKFLAGS)
 export LIBPATH    = ../../../lib
 export VERSION    = 1.6
 export CP         = cp
diff -rupN SFML-1.6.orig/src/SFML/Window/Linux/Joystick.cpp SFML-1.6/src/SFML/Window/Linux/Joystick.cpp
--- SFML-1.6.orig/src/SFML/Window/Linux/Joystick.cpp	2010-01-27 15:00:27.000000000 +0100
+++ SFML-1.6/src/SFML/Window/Linux/Joystick.cpp	2013-10-11 15:22:44.423055420 +0200
@@ -27,7 +27,7 @@
 ////////////////////////////////////////////////////////////
 #include <SFML/Window/Joystick.hpp>
 #include <sstream>
-
+#include <unistd.h>
 
 namespace sf
 {
diff -rupN SFML-1.6.orig/src/SFML/Window/Makefile SFML-1.6/src/SFML/Window/Makefile
--- SFML-1.6.orig/src/SFML/Window/Makefile	2010-01-27 15:00:37.000000000 +0100
+++ SFML-1.6/src/SFML/Window/Makefile	2013-10-11 15:31:52.909728371 +0200
@@ -1,6 +1,5 @@
 SRC     = $(wildcard *.cpp ./Linux/*.cpp)
 OBJ     = $(SRC:.cpp=.o)
-OBJGLEW = $(SRCGLEW:.c=.o)
 
 ifeq ($(STATIC), yes)
     LIB     = libsfml-window-s.a
@@ -18,7 +17,7 @@ libsfml-window-s.a: $(OBJ)
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ)
 
 libsfml-window.so: $(OBJ)
-	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJGLEW) -lX11 -lXrandr -lGL
+	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) -lGLEW -lX11 -lXrandr -lGL
 
 $(OBJ): %.o: %.cpp
 	$(CPP) -o $@ -c $< $(CFLAGS)
