--- sql/chartablesql.cpp	2010-05-08 14:53:48.000000000 +0200
+++ sql/chartablesql.cpp	2010-05-08 14:53:59.000000000 +0200
@@ -50,7 +50,7 @@
 			// Read the unicode value
 			unicode = value.toInt();
 			// Convert unicode to a char
-			unicodeToChar = QString::QString(QChar(unicode)); //"\'" + QString::QString(QChar(unicode)) + "\'";
+			unicodeToChar = QString(QChar(unicode)); //"\'" + QString::QString(QChar(unicode)) + "\'";
 			return unicodeToChar;
 		} else {
 			// Last column (error weight)
--- sql/startsql.cpp	2010-05-08 14:54:24.000000000 +0200
+++ sql/startsql.cpp	2010-05-08 14:54:49.000000000 +0200
@@ -338,7 +338,7 @@
 		for (i = 0; i < content.size(); i++) {
 			//simplifiedContent = QString::QString(
 			//	content.at(i)).replace(QChar(0x27), "''", Qt::CaseSensitive).simplified();
-			simplifiedContent = trim(QString::QString(
+			simplifiedContent = trim(QString(
 				content.at(i)).replace(QChar(0x27), "''", Qt::CaseSensitive));
 
 			if (!query.exec("INSERT INTO own_content VALUES(NULL,'" +
--- widget/lessondialog.cpp	2010-05-08 14:52:48.000000000 +0200
+++ widget/lessondialog.cpp	2010-05-08 14:53:11.000000000 +0200
@@ -89,7 +89,7 @@
 			*lessonData = lineLessonContent->toPlainText().split("\n", QString::SkipEmptyParts);
 			// Delete empty lines
 			for (int i = 0; i < lessonData->size(); i++) {
-				if (QString::QString(lessonData->at(i).toLocal8Bit().constData()).simplified() == "") {
+				if (QString(lessonData->at(i).toLocal8Bit().constData()).simplified() == "") {
 					lessonData->removeAt(i);
 				}
 			}
@@ -268,7 +268,7 @@
 	contentList = lineLessonContent->toPlainText().split("\n", QString::SkipEmptyParts);
 	// Delete empty lines
 	for (i = 0; i < contentList.size(); i++) {
-		if (QString::QString(contentList.at(i).toLocal8Bit().constData()).simplified() == "") {
+		if (QString(contentList.at(i).toLocal8Bit().constData()).simplified() == "") {
 			contentList.removeAt(i);
 		}
 	}
--- widget/settingspages.cpp	2010-05-08 14:45:41.000000000 +0200
+++ widget/settingspages.cpp	2010-05-08 14:45:57.000000000 +0200
@@ -617,7 +617,7 @@
 	QSettings settings;
 	#endif
 	settings.beginGroup("database");
-	settings.setValue("pathpro", lineDatabasePath->text() + "/" + QString::QString(APP_USER_DB));
+	settings.setValue("pathpro", lineDatabasePath->text() + "/" + QString(APP_USER_DB));
 	settings.endGroup();
 }
 
--- widget/tickerboard.cpp	2010-05-08 14:28:38.000000000 +0200
+++ widget/tickerboard.cpp	2010-05-08 14:44:11.000000000 +0200
@@ -96,7 +96,7 @@
 
 		if (tickerSpeed == 50) {
 			scrollOffset = 290;
-			scroll(-290, 0, QRect::QRect(10, 15, 590, 35)); //contentsRect());
+			scroll(-290, 0, QRect(10, 15, 590, 35)); //contentsRect());
 		}
 
 		startFlag = true;
@@ -150,7 +150,7 @@
 			scrollOffset = 0;
 		} else {
 			scrollOffset = 290;
-			scroll(-290, 0, QRect::QRect(10, 15, 590, 35)); //contentsRect());
+			scroll(-290, 0, QRect(10, 15, 590, 35)); //contentsRect());
 		}
 		splitLesson();
 	}
@@ -239,7 +239,7 @@
 
 				// Move ticker 1 pixel to left
 				scrollOffset++;
-				scroll(-1, 0, QRect::QRect(10, 15, 590, 35)); //contentsRect());
+				scroll(-1, 0, QRect(10, 15, 590, 35)); //contentsRect());
 
 				if ((lessonOffset - scrollOffset) <= 30) {
 					setSpeed(tickerSpeed);
@@ -262,14 +262,14 @@
 			// 160 pixels overage (because the user must see at least the next word)
 			if ((lessonOffset - scrollOffset) > 200) {
 				scrollOffset += (lessonOffset - scrollOffset) - 200;
-				scroll(-((lessonOffset - scrollOffset) - 200), 0, QRect::QRect(10, 15, 590, 35)); //contentsRect());
+				scroll(-((lessonOffset - scrollOffset) - 200), 0, QRect(10, 15, 590, 35)); //contentsRect());
 			}
 		} else {
 			// If the user types faster than the ticker, move ticker faster after
 			// 160 pixels overage (because the user must see at least the next word)
 			if ((lessonOffset - scrollOffset) > 280) {
 				scrollOffset += 570;
-				scroll(-570, 0, QRect::QRect(10, 15, 590, 35)); //contentsRect());
+				scroll(-570, 0, QRect(10, 15, 590, 35)); //contentsRect());
 			}
 
 		}
