diff --git a/src/addonloader/luahelper.h b/src/addonloader/luahelper.h
index 6c57851..0452e6b 100644
--- a/src/addonloader/luahelper.h
+++ b/src/addonloader/luahelper.h
@@ -91,13 +91,6 @@ constexpr char kLuaModuleName[] = "__fcitx_luaaddon";
 
 LuaAddonState *GetLuaAddonState(lua_State *lua);
 
-// These functions are required for GetLuaAddonState.
-extern decltype(&::lua_getglobal) _fcitx_lua_getglobal;
-extern decltype(&::lua_touserdata) _fcitx_lua_touserdata;
-extern decltype(&::lua_settop) _fcitx_lua_settop;
-extern decltype(&::lua_close) _fcitx_lua_close;
-extern decltype(&::luaL_newstate) _fcitx_luaL_newstate;
-
 FCITX_DECLARE_LOG_CATEGORY(lua_log);
 #define FCITX_LUA_INFO() FCITX_LOGC(::fcitx::lua_log, Info)
 #define FCITX_LUA_WARN() FCITX_LOGC(::fcitx::lua_log, Warn)
