--- src/Makefile
+++ src/Makefile
@@ -12,10 +12,10 @@
 CXX	=	g++
 CFLAGS	=	-pipe $(PKG_CFLAGS) $(AL_CFLAGS) $(SDL_CFLAGS) $(SMPEG_CFLAGS) -O2 -DOLD_OPENAL -DAUDIO_OPENAL -D_REENTRANT
 CXXFLAGS=	-pipe $(PKG_CFLAGS) $(AL_CFLAGS) $(SDL_CFLAGS) $(SMPEG_CFLAGS) -O2 -DOLD_OPENAL -DAUDIO_OPENAL -D_REENTRANT
-INCPATH	=	-I../support/include -I../support/glpng/include -I/usr/X11R6/include
+INCPATH	=	-I../support/glpng/include -I/usr/X11R6/include
 LINK	=	g++
 LFLAGS	=	
-LIBS	=	$(SUBLIBS) -L../support/glpng/lib -L/usr/lib -L/usr/X11R6/lib -L/usr/local/lib $(GL_LIBS) $(AL_LIBS) $(SDL_LIBS) $(SMPEG_LIBS) $(VORBIS_LIBS)
+LIBS	=	$(SUBLIBS) -L../support/glpng/lib -L/usr/lib -L/usr/X11R6/lib -L/usr/local/lib $(GL_LIBS) $(AL_LIBS) $(SDL_LIBS) $(SMPEG_LIBS) $(VORBIS_LIBS) -lpng
 MOC	=	$(QTDIR)/bin/moc
 UIC	=	$(QTDIR)/bin/uic
 
--- support/glpng/src/glpng.c
+++ support/glpng/src/glpng.c
@@ -3,11 +3,12 @@
 	#include <windows.h>
 #endif
 
+#define GL_GLEXT_PROTOTYPES 1
 #include <GL/glpng.h>
 #include <GL/gl.h>
 #include <stdlib.h>
 #include <math.h>
-#include "png/png.h"
+#include <png.h>
 
 /* Used to decide if GL/gl.h supports the paletted extension */
 #ifdef GL_COLOR_INDEX1_EXT
--- support/glpng/src/Makefile
+++ support/glpng/src/Makefile
@@ -54,27 +54,7 @@
 		zlib/inftrees.c \
 		zlib/infutil.c \
 		zlib/zutil.c
-OBJECTS =	glpng.o \
-		png/png.o \
-		png/pngerror.o \
-		png/pngget.o \
-		png/pngmem.o \
-		png/pngpread.o \
-		png/pngread.o \
-		png/pngrio.o \
-		png/pngrtran.o \
-		png/pngrutil.o \
-		png/pngset.o \
-		png/pngtrans.o \
-		zlib/adler32.o \
-		zlib/crc32.o \
-		zlib/infblock.o \
-		zlib/infcodes.o \
-		zlib/inffast.o \
-		zlib/inflate.o \
-		zlib/inftrees.o \
-		zlib/infutil.o \
-		zlib/zutil.o
+OBJECTS =	glpng.o
 INTERFACES =	
 UICDECLS =	
 UICIMPLS =	

