From 961892779b5f5a2f46d6baa44c31d06c39e56470 Mon Sep 17 00:00:00 2001
From: Thomas Dziedzic <gostrc@gmail.com>
Date: Tue, 11 Sep 2012 05:06:59 -0700
Subject: [PATCH] update build deps for ghc 7.6.1

---
 dataenc.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dataenc.cabal b/dataenc.cabal
index 1ad42ca..3a4f5bb 100644
--- a/dataenc.cabal
+++ b/dataenc.cabal
@@ -21,7 +21,7 @@ flag BuildTests
 
 library
     hs-source-dirs: src
-    build-depends: array >= 0.1.0 && < 0.5, base >= 3.0.0 && < 4.6, containers >= 0.1.0 && < 0.5
+    build-depends: array >= 0.1.0 && < 0.5, base >= 3.0.0 && < 4.7, containers >= 0.1.0 && < 0.6
     exposed-modules:
             Codec.Binary.Base16
             Codec.Binary.Base32
-- 
1.7.12

