From b31b0a468ddbc9be4fc78289e5a0a108c31ec620 Mon Sep 17 00:00:00 2001
From: Sumit Bose <sbose@redhat.com>
Date: Fri, 18 Jul 2014 22:39:56 +0200
Subject: [PATCH] AD: add missing debug-to-stderr option to gpo_child

Reviewed-by: Jakub Hrozek <jhrozek@redhat.com>
---
 src/providers/ad/ad_gpo_child.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/providers/ad/ad_gpo_child.c b/src/providers/ad/ad_gpo_child.c
index 0b6a4cd..31c5c99 100644
--- a/src/providers/ad/ad_gpo_child.c
+++ b/src/providers/ad/ad_gpo_child.c
@@ -498,6 +498,9 @@ main(int argc, const char *argv[])
          _("Show timestamps with microseconds"), NULL},
         {"debug-fd", 0, POPT_ARG_INT, &debug_fd, 0,
          _("An open file descriptor for the debug logs"), NULL},
+        {"debug-to-stderr", 0, POPT_ARG_NONE | POPT_ARGFLAG_DOC_HIDDEN,
+         &debug_to_stderr, 0,
+         _("Send the debug output to stderr directly."), NULL },
         POPT_TABLEEND
     };
 
-- 
2.0.3

