diff --git a/capplet/csm-app-dialog.c b/capplet/csm-app-dialog.c
index df177f0..c1001e8 100644
--- a/capplet/csm-app-dialog.c
+++ b/capplet/csm-app-dialog.c
@@ -270,8 +270,6 @@ csm_app_dialog_constructor (GType                  type,
 
         setup_dialog (dialog);
 
-        gtk_widget_show_all (GTK_WIDGET (dialog));
-
         return G_OBJECT (dialog);
 }
 
diff --git a/capplet/csm-properties-dialog.c b/capplet/csm-properties-dialog.c
index 5ccb815..5c2cc56 100644
--- a/capplet/csm-properties-dialog.c
+++ b/capplet/csm-properties-dialog.c
@@ -374,6 +374,7 @@ on_add_app_clicked (GtkWidget           *widget,
         gtk_window_set_transient_for (GTK_WINDOW (add_dialog),
                                       GTK_WINDOW (dialog));
 
+        gtk_widget_show_all (add_dialog);
         if (csm_app_dialog_run (CSM_APP_DIALOG (add_dialog),
                                 &name, &exec, &comment, &delay)) {
                 csp_app_create (name, comment, exec, delay);
@@ -444,6 +445,7 @@ on_edit_app_clicked (GtkWidget           *widget,
                 gtk_window_set_transient_for (GTK_WINDOW (edit_dialog),
                                               GTK_WINDOW (dialog));
 
+                gtk_widget_show_all (edit_dialog);
                 if (csm_app_dialog_run (CSM_APP_DIALOG (edit_dialog),
                                         &name, &exec, &comment, &delay)) {
                         csp_app_update (app, name, comment, exec, delay);
@@ -678,8 +680,6 @@ csm_properties_dialog_constructor (GType                  type,
 
         setup_dialog (dialog);
 
-        gtk_widget_show (GTK_WIDGET (dialog));
-
         return G_OBJECT (dialog);
 }
 
-- 
1.9.3

