From c30911ee2846a7ab544552651e679786483f2168 Mon Sep 17 00:00:00 2001
From: Andrea Florio <andrea@opensuse.org>
Date: Mon, 25 Jul 2011 00:55:29 +0200
Subject: [PATCH] fixed error in tooltipS function call

---
 src/plugins/netstatus/netstatus-icon.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/plugins/netstatus/netstatus-icon.c b/src/plugins/netstatus/netstatus-icon.c
index 2cd3ccb..b3bfc19 100644
--- a/src/plugins/netstatus/netstatus-icon.c
+++ b/src/plugins/netstatus/netstatus-icon.c
@@ -26,6 +26,7 @@
 
 #include <gtk/gtk.h>
 #include <glib/gi18n.h>
+#include <gtk/gtktooltips.h>
 
 #include "netstatus-util.h"
 #include "netstatus-enums.h"
@@ -899,7 +900,7 @@ netstatus_icon_instance_init (NetstatusIcon      *icon,
   gtk_container_add (GTK_CONTAINER (icon), icon->priv->signal_image);
   gtk_widget_hide (icon->priv->signal_image);
 
-  icon->priv->tooltips = gtk_tooltip_new ();
+  icon->priv->tooltips = gtk_tooltips_new ();
   g_object_ref (icon->priv->tooltips);
   gtk_object_sink (GTK_OBJECT (icon->priv->tooltips));
 
-- 
1.7.0.1
