diff -Nur mysql-workbench-gpl-5.2.25/res/mysql.profiles/Makefile.am mysql-workbench-gpl-5.2.25a/res/mysql.profiles/Makefile.am
--- mysql-workbench-gpl-5.2.25/res/mysql.profiles/Makefile.am	2010-06-30 11:14:27.000000000 -0500
+++ mysql-workbench-gpl-5.2.25a/res/mysql.profiles/Makefile.am	2010-07-04 21:36:34.425999721 -0500
@@ -2,6 +2,7 @@
 workbenchdir=$(pkgdatadir)/mysql.profiles
 
 workbench_DATA=\
+ArchLinux.xml\
 Fedora_Linux_(MySQL_Package).xml\
 Fedora_Linux_(Vendor_Package).xml\
 FreeBSD_(MySQL_Package).xml\
diff -Nur mysql-workbench-gpl-5.2.25/res/mysql.profiles/Makefile.in mysql-workbench-gpl-5.2.25a/res/mysql.profiles/Makefile.in
--- mysql-workbench-gpl-5.2.25/res/mysql.profiles/Makefile.in	2010-06-30 11:14:38.000000000 -0500
+++ mysql-workbench-gpl-5.2.25a/res/mysql.profiles/Makefile.in	2010-07-04 21:36:59.396862401 -0500
@@ -213,6 +213,7 @@
 top_srcdir = @top_srcdir@
 workbenchdir = $(pkgdatadir)/mysql.profiles
 workbench_DATA = \
+ArchLinux.xml\
 Fedora_Linux_(MySQL_Package).xml\
 Fedora_Linux_(Vendor_Package).xml\
 FreeBSD_(MySQL_Package).xml\
