=== modified file duplicity/backends/botobackend.py
--- duplicity/backends/botobackend.py	2011-11-25 17:47:57 +0000
+++ duplicity/backends/botobackend.py	2011-12-01 16:15:25 +0000
@@ -26,6 +26,6 @@
     if sys.version_info[:2] < (2,6):
         print "Sorry, S3 multiprocessing requires version 2.6 or later of python"
         sys.exit(1)
-    import _boto_multi.py
+    import _boto_multi
 else:
-    import _boto_single.py
+    import _boto_single

