commit 46f08625416d79fbc319b42f8a361f69c3e7a315
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Tue Jun 27 21:47:54 2017 +0800

    Update brick to 0.19

diff --git a/hledger-ui/Hledger/UI/UIState.hs b/hledger-ui/Hledger/UI/UIState.hs
index 1968d978..1e2ac939 100644
--- a/hledger-ui/Hledger/UI/UIState.hs
+++ b/hledger-ui/Hledger/UI/UIState.hs
@@ -6,7 +6,6 @@
 module Hledger.UI.UIState
 where
 
-import Brick
 import Brick.Widgets.Edit
 import Data.List
 import Data.Text.Zipper (gotoEOL)
@@ -235,7 +234,7 @@ getDepth UIState{aopts=UIOpts{cliopts_=CliOpts{reportopts_=ropts}}} = depth_ rop
 showMinibuffer :: UIState -> UIState
 showMinibuffer ui = setMode (Minibuffer e) ui
   where
-    e = applyEdit gotoEOL $ editor MinibufferEditor (str . unlines) (Just 1) oldq
+    e = applyEdit gotoEOL $ editor MinibufferEditor (Just 1) oldq
     oldq = query_ $ reportopts_ $ cliopts_ $ aopts ui
 
 -- | Close the minibuffer, discarding any edit in progress.
diff --git a/hledger-ui/Hledger/UI/UIUtils.hs b/hledger-ui/Hledger/UI/UIUtils.hs
index 48afeccc..a4e39b79 100644
--- a/hledger-ui/Hledger/UI/UIUtils.hs
+++ b/hledger-ui/Hledger/UI/UIUtils.hs
@@ -136,7 +136,7 @@ minibuffer :: Editor String Name -> Widget Name
 minibuffer ed =
   forceAttr (borderAttr <> "minibuffer") $
   hBox $
-  [txt "filter: ", renderEditor True ed]
+  [txt "filter: ", renderEditor (str . unlines) True ed]
 
 -- | Wrap a widget in the default hledger-ui screen layout.
 defaultLayout :: Widget Name -> Widget Name -> Widget Name -> Widget Name
diff --git a/hledger-ui/hledger-ui.cabal b/hledger-ui/hledger-ui.cabal
index 9c148e95..cc8e9a35 100644
--- a/hledger-ui/hledger-ui.cabal
+++ b/hledger-ui/hledger-ui.cabal
@@ -85,7 +85,7 @@ executable hledger-ui
     buildable: False
   else
     build-depends:
-        brick >= 0.12 && < 0.19
+        brick >= 0.19 && < 0.20
       , vty >= 5.5 && < 5.16
   if flag(threaded)
     ghc-options: -threaded
