diff -wbBur synfig-0.62.01/src/modules/mod_libavcodec/trgt_av.cpp synfig-0.62.01.my/src/modules/mod_libavcodec/trgt_av.cpp
--- synfig-0.62.01/src/modules/mod_libavcodec/trgt_av.cpp	2010-05-31 18:12:14.000000000 +0000
+++ synfig-0.62.01.my/src/modules/mod_libavcodec/trgt_av.cpp	2010-06-03 15:56:04.000000000 +0000
@@ -33,6 +33,10 @@
 #	include <config.h>
 #endif
 
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
+#include <libswscale/swscale.h>
+
 #include "trgt_av.h"
 
 extern "C"
@@ -118,14 +122,14 @@
     picture = avcodec_alloc_frame();
     if (!picture)
         return NULL;
-    size = avpicture_get_size(pix_fmt, width, height);
+    size = avpicture_get_size((enum ::PixelFormat)pix_fmt, width, height);
     picture_buf = (uint8_t *)malloc(size);
     if (!picture_buf) {
         av_free(picture);
         return NULL;
     }
     avpicture_fill((AVPicture *)picture, picture_buf,
-                   pix_fmt, width, height);
+                   (enum ::PixelFormat)pix_fmt, width, height);
     return picture;
 }
 
diff -wbBur synfig-0.62.01/src/modules/mod_png/mptr_png.cpp synfig-0.62.01.my/src/modules/mod_png/mptr_png.cpp
--- synfig-0.62.01/src/modules/mod_png/mptr_png.cpp	2010-05-31 18:12:14.000000000 +0000
+++ synfig-0.62.01.my/src/modules/mod_png/mptr_png.cpp	2010-06-03 15:43:39.000000000 +0000
@@ -309,7 +309,7 @@
 				float b=gamma().b_U8_to_F32((unsigned char)png_ptr->palette[row_pointers[y][x]].blue);
 				float a=1.0;
 				if(info_ptr->valid & PNG_INFO_tRNS)
-				    a = (float)(unsigned char)png_ptr->trans[row_pointers[y][x]]*(1.0/255.0);
+				    a = (float)(unsigned char)png_ptr->trans_alpha[row_pointers[y][x]]*(1.0/255.0);
 				surface_buffer[y][x]=Color(
 					r,
 					g,
diff -wbBur synfig-0.62.01/src/synfig/canvas.cpp synfig-0.62.01.my/src/synfig/canvas.cpp
--- synfig-0.62.01/src/synfig/canvas.cpp	2010-05-31 18:12:14.000000000 +0000
+++ synfig-0.62.01.my/src/synfig/canvas.cpp	2010-06-03 15:47:00.000000000 +0000
@@ -716,7 +716,7 @@
 	Layer::LooseHandle loose_layer(x);
 
 	add_connection(loose_layer,
-				   sigc::connection::connection(
+				   sigc::connection(
 					   x->signal_added_to_group().connect(
 						   sigc::bind(
 							   sigc::mem_fun(
@@ -724,7 +724,7 @@
 								   &Canvas::add_group_pair),
 							   loose_layer))));
 	add_connection(loose_layer,
-				   sigc::connection::connection(
+				   sigc::connection(
 					   x->signal_removed_from_group().connect(
 						   sigc::bind(
 							   sigc::mem_fun(
diff -wbBur synfig-0.62.01/src/synfig/time.cpp synfig-0.62.01.my/src/synfig/time.cpp
--- synfig-0.62.01/src/synfig/time.cpp	2010-05-31 18:12:14.000000000 +0000
+++ synfig-0.62.01.my/src/synfig/time.cpp	2010-06-03 15:43:39.000000000 +0000
@@ -1,4 +1,6 @@
+#include <stdio.h>
 /* === S Y N F I G ========================================================= */
+#include <stdio.h>
 /*!	\file time.cpp
 **	\brief Template File
 **
