diff -wbBur mg-20130922/Makefile mg-20130922.y/Makefile
--- mg-20130922/Makefile	2013-09-22 18:06:43.000000000 +0400
+++ mg-20130922.y/Makefile	2013-10-29 15:18:18.795139731 +0400
@@ -4,7 +4,7 @@
 NOGCCERROR= true
 PROG=	mg
 
-LDADD+=	-lcurses -lutil -lclens
+LDADD+=	-lcurses -lutil -lclens -lbsd
 DPADD+=	${LIBCURSES} ${LIBUTIL}
 
 # (Common) compile-time options:
@@ -17,7 +17,7 @@
 #
 CFLAGS  = -O2
 CFLAGS += -Wall -DFKEYS -DREGEX -DXKEYS
-CFLAGS += -I/usr/local/include/clens
+CFLAGS += -I/usr/include/clens -isystem /usr/include/bsd -DLIBBSD_OVERLAY
 
 SRCS=	autoexec.c basic.c bell.c buffer.c cinfo.c dir.c display.c \
 	echo.c extend.c file.c fileio.c funmap.c help.c kbd.c keymap.c \
