diff -rupN Clementine-1.2.2.orig/src/internet/spotifyblobdownloader.cpp Clementine-1.2.2/src/internet/spotifyblobdownloader.cpp
--- Clementine-1.2.2.orig/src/internet/spotifyblobdownloader.cpp	2014-02-23 16:21:49.478100000 +0100
+++ Clementine-1.2.2/src/internet/spotifyblobdownloader.cpp	2014-02-23 16:22:57.048258400 +0100
@@ -32,6 +32,10 @@
   #include <QtCrypto>
 #endif // HAVE_QCA
 
+#ifdef Q_OS_LINUX
+  #include <unistd.h>
+#endif
+
 const char* SpotifyBlobDownloader::kSignatureSuffix = ".sha1";
 
 
diff -rupN Clementine-1.2.2.orig/src/core/utilities.cpp Clementine-1.2.2/src/core/utilities.cpp
--- Clementine-1.2.2.orig/src/core/utilities.cpp	2014-02-23 16:14:06.919406700 +0100
+++ Clementine-1.2.2/src/core/utilities.cpp	2014-02-23 16:16:12.107064900 +0100
@@ -50,6 +50,7 @@
 #endif
 
 #ifdef Q_OS_LINUX
+#  include <unistd.h>
 #  include <sys/syscall.h>
 #endif
 #ifdef Q_OS_DARWIN
