diff -up clementine-1.1.1/src/devices/imobiledeviceconnection.h.fix clementine-1.1.1/src/devices/imobiledeviceconnection.h
--- clementine-1.1.1/src/devices/imobiledeviceconnection.h.fix	2013-04-23 14:44:56.932661018 -0400
+++ clementine-1.1.1/src/devices/imobiledeviceconnection.h	2013-04-23 14:46:44.033657190 -0400
@@ -51,7 +51,7 @@ private:
   idevice_t device_;
   afc_client_t afc_;
 
-  uint16_t afc_port_;
+  lockdownd_service_descriptor_t afc_port_;
 };
 
 #endif // IMOBILEDEVICECONNECTION_H
