diff --git a/src/Config.hs b/src/Config.hs
index 59d6bd4..c788e6a 100644
--- a/src/Config.hs
+++ b/src/Config.hs
@@ -34,7 +34,6 @@ import Plugins.MarqueePipeReader
 import Plugins.CommandReader
 import Plugins.StdinReader
 import Plugins.XMonadLog
-import Plugins.EWMH
 import Plugins.Kbd
 import Plugins.Locks

@@ -147,7 +146,7 @@ infixr :*:
 -- the 'Runnable.Runnable' Read instance. To install a plugin just add
 -- the plugin's type to the list of types (separated by ':*:') appearing in
 -- this function's type signature.
-runnableTypes :: Command :*: Monitors :*: Date :*: PipeReader :*: BufferedPipeReader :*: CommandReader :*: StdinReader :*: XMonadLog :*: EWMH :*: Kbd :*: Locks :*:
+runnableTypes :: Command :*: Monitors :*: Date :*: PipeReader :*: BufferedPipeReader :*: CommandReader :*: StdinReader :*: XMonadLog :*: Kbd :*: Locks :*:
                  Mail :*: MBox :*:
                  DateZone :*: MarqueePipeReader :*:
                  ()
diff --git a/src/Plugins/Date.hs b/src/Plugins/Date.hs
index a263536..37e603f 100644
--- a/src/Plugins/Date.hs
+++ b/src/Plugins/Date.hs
@@ -20,7 +20,6 @@ module Plugins.Date (Date(..)) where

 import Plugins

-import System.Locale
 import Control.Monad (liftM)
 import Data.Time

diff --git a/xmobar.cabal b/xmobar.cabal
index db0d130..c76b355 100644
--- a/xmobar.cabal
+++ b/xmobar.cabal
@@ -81,7 +81,7 @@ executable xmobar
       Xmobar, Actions, Bitmap, Config, Parsers, Commands, Localize,
       XUtil, XPMFile, StatFS, Runnable, ColorCache, Window, Signal,
       Plugins, Plugins.BufferedPipeReader,
-      Plugins.CommandReader, Plugins.Date, Plugins.EWMH,
+      Plugins.CommandReader, Plugins.Date,
       Plugins.PipeReader, Plugins.MarqueePipeReader,
       Plugins.StdinReader, Plugins.XMonadLog,
       Plugins.Utils, Plugins.Kbd, Plugins.Locks, Plugins.Monitors,
