diff --git a/Makefile b/Makefile
index 8cb5b6c..11c3ca0 100644
--- a/Makefile
+++ b/Makefile
@@ -19,6 +19,12 @@ FNTEXES=ioncore.exports mod_tiling.exports \
 	mod_query.exports de.exports mod_menu.exports \
 	mod_dock.exports mod_sp.exports mod_statusbar.exports
 
+TEXSOURCES=conf-bindings.tex confintro.tex conf-menus.tex \
+	conf-statusbar.tex conf.tex conf-winprops.tex cstyle.tex designnotes.tex \
+	de.tex fdl.tex fnref.tex fullhierarchy.tex hookref.tex luaif.tex \
+	macros.tex miscref.tex notionconf.tex notionnotes.tex objectsimpl.tex objects.tex \
+	prelim.tex statusd.tex tricks.tex
+
 RUBBER_DVI=rubber
 RUBBER_PS=rubber -p
 RUBBER_PDF=rubber -d
@@ -31,14 +37,17 @@ TARGETS = notionconf notionnotes
 nothing:
 	@ echo "Please read the README first."
 
-%-dvi:
+%-dvi: $(TEXSOURCES)
 	$(RUBBER_DVI) $*
-	
-%-ps:
+	touch $@
+
+%-ps: $(TEXSOURCES)
 	$(RUBBER_PS) $*
+	touch $@
 
-%-pdf:
+%-pdf: $(TEXSOURCES)
 	$(RUBBER_PDF) $*
+	touch $@
 
 # Install
 ######################################
@@ -48,30 +57,38 @@ install:
 	for d in $(DOCS); do \
 	    for e in ps pdf dvi; do \
 	      test -f $$d.$$e && $(INSTALL) -m $(DATA_MODE) $$d.$$e $(DOCDIR); \
-            done; \
+	    done; \
 	    $(INSTALLDIR) $(DOCDIR)/$$d; \
-            for i in $$d/*; do \
-                $(INSTALL) -m $(DATA_MODE) $$i $(DOCDIR)/$$i; \
+	    for i in $$d/*; do \
+	        $(INSTALL) -m $(DATA_MODE) $$i $(DOCDIR)/$$i; \
 	    done; \
-        done
+	    rm -f $(DOCDIR)/$$d/*.log; \
+	    rm -f $(DOCDIR)/$$d/WARNINGS; \
+	    rm -f $(DOCDIR)/$$d/*.aux; \
+	    rm -f $(DOCDIR)/$$d/*.idx; \
+	    rm -f $(DOCDIR)/$$d/*.tex; \
+	    rm -f $(DOCDIR)/$$d/*.pl; \
+	done
 
 # notionconf rules
 ######################################
 
-notionconf-dvi: fnlist.tex
-notionconf-ps: fnlist.tex
-notionconf-pdf: fnlist.tex
+notionconf-dvi: fnlist.tex $(TEXSOURCES)
+notionconf-ps: fnlist.tex $(TEXSOURCES)
+notionconf-pdf: fnlist.tex $(TEXSOURCES)
 
-notionconf-html: $(FNTEXES)
+notionconf-html: $(FNTEXES) $(TEXSOURCES)
 	$(L2H) -split 3 notionconf
 	cp notion.css notionconf
+	touch $@
 
 # notionnotes rules
 ######################################
 
-notionnotes-html: 
+notionnotes-html: $(TEXSOURCES)
 	$(L2H) -split 4 notionnotes
 	cp notion.css notionnotes
+	touch $@
 
 # More generic rules
 ######################################
@@ -94,7 +111,7 @@ all-html: $(patsubst %, %-html, $(TARGETS))
 
 clean:
 	rm -f $(FNTEXES) fnlist.tex
-	rm -f *.aux *.toc *.log
+	rm -f *.aux *.toc *.log *.out
 	rm -f *.idx *.ild *.ilg *.ind
         
 realclean: clean
@@ -111,7 +128,8 @@ $(TOPDIR)/%/exports.tex:
 	$(MAKE) -C $$(dirname $@) _exports_doc
 
 %.exports: $(TOPDIR)/%/exports.tex
-	cp $< $@
+	cat $< | sed 's|UTF8_STRING|UTF8\\_STRING|' >$@
+#	cp $< $@
 
 # Function list
 ######################################
