diff -Naur gpsbabel-1.5.4.orig/gui/app.pro gpsbabel-1.5.4/gui/app.pro
--- gpsbabel-1.5.4.orig/gui/app.pro	2017-01-02 06:05:19.000000000 +0100
+++ gpsbabel-1.5.4/gui/app.pro	2017-02-22 05:52:04.650462935 +0100
@@ -15,13 +15,7 @@
       gui \
       network \
       xml \
-
-greaterThan(QT_MINOR_VERSION, 5) {
-  QT += webenginewidgets
-  DEFINES += HAVE_WEBENGINE
-} else {
-  QT += webkit webkitwidgets 
-}
+      widgets \
 
 unix:DESTDIR = objects
 unix:MOC_DIR = objects
@@ -72,13 +66,11 @@
 SOURCES += filterwidgets.cc
 SOURCES += format.cc
 SOURCES += formatload.cc
-SOURCES += gmapdlg.cc
 SOURCES += gpx.cc
 SOURCES += help.cc
 SOURCES += latlng.cc
 SOURCES += main.cc
 SOURCES += mainwindow.cc
-SOURCES += map.cc
 SOURCES += optionsdlg.cc
 SOURCES += preferences.cc
 SOURCES += processwait.cc
@@ -98,11 +90,9 @@
 HEADERS += filterwidgets.h
 HEADERS += format.h
 HEADERS += formatload.h
-HEADERS += gmapdlg.h
 HEADERS += gpx.h
 HEADERS += help.h
 HEADERS += mainwindow.h
-HEADERS += map.h
 HEADERS += optionsdlg.h
 HEADERS += preferences.h
 HEADERS += processwait.h
diff -Naur gpsbabel-1.5.4.orig/gui/main.cc gpsbabel-1.5.4/gui/main.cc
--- gpsbabel-1.5.4.orig/gui/main.cc	2017-01-02 06:05:19.000000000 +0100
+++ gpsbabel-1.5.4/gui/main.cc	2017-02-22 05:20:03.449272478 +0100
@@ -29,7 +29,6 @@
 #include <QTextCodec>
 
 #include "mainwindow.h"
-#include "gmapdlg.h"
 
 #ifdef _WIN32
 const char *pathSeparator = ";";
diff -Naur gpsbabel-1.5.4.orig/gui/mainwindow.cc gpsbabel-1.5.4/gui/mainwindow.cc
--- gpsbabel-1.5.4.orig/gui/mainwindow.cc	2017-01-02 06:05:19.000000000 +0100
+++ gpsbabel-1.5.4/gui/mainwindow.cc	2017-02-22 06:03:27.026789219 +0100
@@ -38,7 +38,6 @@
 #include "donate.h"
 #include "filterdlg.h"
 #include "formatload.h"
-#include "gmapdlg.h"
 #include "help.h"
 #include "optionsdlg.h"
 #include "preferences.h"
@@ -976,24 +975,6 @@
     formatList_[fidx].bumpWriteUseCount(1);
   }
 
-  // Now output for preview in google maps
-  QString tempName;
-  if (babelData_.previewGmap_) {
-    QTemporaryFile ftemp;
-    ftemp.open();
-    tempName = ftemp.fileName();
-    ftemp.close();
-
-    // Ideally, expost this in the UI.  For now, just split the track
-    // if we've no recorded fixes for > 5 mins and we've moved > 300 meters.
-    //args << "-x";
-    //args << "track,pack,sdistance=0.3k,split=5m";
-
-    args << "-o";
-    args << "gpx";
-    args << "-F" << tempName;
-  }
-
   ui_.outputWindow->clear();
   ui_.outputWindow->appendPlainText("gpsbabel " + args.join(" "));
 
@@ -1005,14 +986,6 @@
   ui_.outputWindow->appendPlainText(outputString);
   if (x) {
     ui_.outputWindow->appendPlainText(tr("Translation successful"));
-    if (babelData_.previewGmap_) {
-      this->hide();
-      GMapDialog dlg(0, tempName, babelData_.debugLevel_ >=1 ? ui_.outputWindow : 0);
-      dlg.show();
-      dlg.exec();
-      QFile(tempName).remove();
-      this->show();
-    }
   }
   else
     ui_.outputWindow->appendPlainText(tr("Error running gpsbabel: %1\n").arg(errorString));
