diff -Naur etherape-0.9.15.orig/configure.ac etherape-0.9.15/configure.ac
--- etherape-0.9.15.orig/configure.ac	2017-04-27 05:53:49.226303987 +0000
+++ etherape-0.9.15/configure.ac	2017-04-27 06:02:09.328964318 +0000
@@ -26,7 +26,7 @@
 #AC_PROG_INTLTOOL(0.27)
 
 # Checks for libraries.
-PKG_CHECK_MODULES(ETHERAPE, glib-2.0 libglade-2.0 libgnomeui-2.0)
+PKG_CHECK_MODULES(ETHERAPE, glib-2.0 libglade-2.0 libgnomecanvas-2.0 popt)
 AC_SUBST(ETHERAPE_CFLAGS)
 AC_SUBST(ETHERAPE_LDFLAGS)
 AC_SUBST(ETHERAPE_LIBS)
diff -Naur etherape-0.9.15.orig/src/callbacks.c etherape-0.9.15/src/callbacks.c
--- etherape-0.9.15.orig/src/callbacks.c	2017-04-27 05:53:49.236303974 +0000
+++ etherape-0.9.15/src/callbacks.c	2017-04-27 05:56:33.176083792 +0000
@@ -21,7 +21,8 @@
 #include <config.h>
 #endif
 
-#include <gnome.h>
+#include <gtk/gtk.h>
+#include <libgnomecanvas/libgnomecanvas.h>
 #include "appdata.h"
 #include "callbacks.h"
 #include "diagram.h"
diff -Naur etherape-0.9.15.orig/src/diagram.c etherape-0.9.15/src/diagram.c
--- etherape-0.9.15.orig/src/diagram.c	2017-04-27 05:53:49.226303987 +0000
+++ etherape-0.9.15/src/diagram.c	2017-04-27 05:58:19.665940538 +0000
@@ -21,7 +21,8 @@
 #include <config.h>
 #endif
 
-#include <gnome.h>
+#include <gtk/gtk.h>
+#include <libgnomecanvas/libgnomecanvas.h>
 
 #include "appdata.h"
 #include "diagram.h"
diff -Naur etherape-0.9.15.orig/src/main.c etherape-0.9.15/src/main.c
--- etherape-0.9.15.orig/src/main.c	2017-04-27 05:53:49.236303974 +0000
+++ etherape-0.9.15/src/main.c	2017-04-27 06:17:23.701057857 +0000
@@ -28,8 +28,10 @@
 #include <netinet/in.h>
 #include <signal.h>
 #include <glib.h>
-#include <gnome.h>
-#include <libgnomeui/gnome-client.h>
+#include <popt.h>
+#include <glib/gi18n.h>
+#include <gtk/gtk.h>
+#include <libgnomecanvas/libgnomecanvas.h>
 #include "appdata.h"
 #include "names/ip-cache.h"
 #include "main.h"
@@ -60,11 +62,6 @@
  **************************************************************************/
 static void free_static_data(void);
 static void set_debug_level (void);
-static void session_die (GnomeClient * client, gpointer client_data);
-static gint save_session (GnomeClient * client, gint phase, 
-                          GnomeSaveStyle save_style, gint is_shutdown, 
-                          GnomeInteractStyle interact_style, gint is_fast, 
-                          gpointer client_data);
 static void log_handler (gchar * log_domain, GLogLevelFlags mask, 
                          const gchar * message, gpointer user_data);
 static GPtrArray *parse_position_file(const gchar *path);
@@ -81,7 +78,6 @@
 int main (int argc, char *argv[])
 {
   GtkWidget *widget;
-  GnomeClient *client;
   gchar *mode_string = NULL;
   gchar *cl_filter = NULL;
   gchar *cl_interface = NULL;
@@ -188,9 +184,7 @@
 #else
   version = g_strdup(VERSION);
 #endif
-  gnome_program_init ("EtherApe", version, 
-                      LIBGNOMEUI_MODULE, argc, argv,
-		      GNOME_PARAM_POPT_TABLE, optionsTable, GNOME_PARAM_NONE);
+  gtk_init (&argc, &argv);
   g_free(version);
 
   set_debug_level();
@@ -311,12 +305,6 @@
       gtk_check_menu_item_set_active(GTK_CHECK_MENU_ITEM(widget), FALSE);
     }
 
-  /* Session handling */
-  client = gnome_master_client ();
-  g_signal_connect (G_OBJECT (client), "save_yourself",
-		    GTK_SIGNAL_FUNC (save_session), argv[0]);
-  g_signal_connect (G_OBJECT (client), "die",
-		    GTK_SIGNAL_FUNC (session_die), NULL);
   gtk_widget_show (appdata.app1);
 
   install_handlers();
@@ -496,46 +484,6 @@
     g_log_default_handler("EtherApe", mask, message, user_data);
 }
 
-/* the gnome session manager may call this function */
-static void
-session_die (GnomeClient * client, gpointer client_data)
-{
-  g_message ("in die");
-  gtk_main_quit ();
-}				/* session_die */
-
-/* the gnome session manager may call this function */
-static gboolean
-save_session (GnomeClient * client, gint phase, GnomeSaveStyle save_style,
-	      gboolean is_shutdown, GnomeInteractStyle interact_style,
-	      gboolean is_fast, gpointer client_data)
-{
-  gchar **argv;
-  guint argc;
-
-  /* allocate 0-filled, so it will be NULL-terminated */
-  argv = g_malloc0 (sizeof (gchar *) * 4);
-  g_assert(argv);
-
-  argc = 1;
-
-  argv[0] = client_data;
-
-  g_message ("In save_session");
-#if 0
-  if (message)
-    {
-      argv[1] = "--message";
-      argv[2] = message;
-      argc = 3;
-    }
-#endif
-
-  gnome_client_set_clone_command (client, argc, argv);
-  gnome_client_set_restart_command (client, argc, argv);
-  return TRUE;
-}				/* save_session */
-
 
 /***************************************************************************
  *
diff -Naur etherape-0.9.15.orig/src/Makefile.am etherape-0.9.15/src/Makefile.am
--- etherape-0.9.15.orig/src/Makefile.am	2017-04-27 05:53:49.236303974 +0000
+++ etherape-0.9.15/src/Makefile.am	2017-04-27 06:04:43.992088744 +0000
@@ -65,7 +65,7 @@
 etherape_SOURCES += names/thread_resolve.c names/thread_resolve.h
 endif
 
-etherape_LDADD = $(ETHERAPE_LIBS) $(PCAP_LIBS) 
+etherape_LDADD = $(ETHERAPE_LIBS) $(PCAP_LIBS) -lpthread
 etherape_LDFLAGS = $(ETHERAPE_LDFLAGS)
 
 .indent: $(etherape_SOURCES) $(etherape_INCLUDES)
