From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 5 Dec 2009 18:11:00 +0100
Subject: [PATCH] explicitly include posix_types.h

to get the correct type for __kernel_old_dev_t

Closes: #558990
---
 kpartx/lopart.c |   13 +------------
 1 files changed, 1 insertions(+), 12 deletions(-)

Index: mpath/kpartx/lopart.c
===================================================================
--- mpath.orig/kpartx/lopart.c	2012-02-09 18:02:23.808800332 +0530
+++ mpath/kpartx/lopart.c	2012-02-09 18:15:24.400671080 +0530
@@ -26,18 +26,7 @@
 #include <sys/stat.h>
 #include <sys/mman.h>
 #include <sysmacros.h>
-
-#if defined(__hppa__) || defined(__powerpc64__) || defined (__alpha__) \
- || defined (__x86_64__)
-typedef unsigned long __kernel_old_dev_t;
-#elif defined(__powerpc__) || defined(__ia64__) || (defined(__sparc__) && defined (__arch64__))
-typedef unsigned int __kernel_old_dev_t;
-#else
-typedef unsigned short __kernel_old_dev_t;
-#endif
-
-#define dev_t __kernel_old_dev_t
-
+#include <asm/posix_types.h>
 #include <linux/loop.h>
 
 #include "lopart.h"
