diff --git a/Makefile b/Makefile
index b16531e..f0f0e10 100644
--- a/Makefile
+++ b/Makefile
@@ -11,11 +11,7 @@ ifeq ($(UNAME_S),Linux)
 	# Linux (e.g. Ubuntu)
 	MOZJPEG_PREFIX ?= /opt/mozjpeg
 	CFLAGS += -I$(MOZJPEG_PREFIX)/include
-	ifeq ($(UNAME_M),x86_64)
-		LIBJPEG = $(MOZJPEG_PREFIX)/lib64/libjpeg.a
-	else
-		LIBJPEG = $(MOZJPEG_PREFIX)/lib/libjpeg.a
-	endif
+	LIBJPEG = $(MOZJPEG_PREFIX)/lib/libjpeg.a
 else
 	ifeq ($(UNAME_S),Darwin)
 		# Mac OS X
