--- cinnamon-screensaver-2.2.2.orig/src/gs-window-x11.c	2014-05-02 10:46:28.000000000 -0500
+++ cinnamon-screensaver-2.2.2/src/gs-window-x11.c	2014-05-03 13:04:32.024323984 -0500
@@ -2139,7 +2139,6 @@
 {
         cairo_set_source_rgba (cr, 0.0, 0.0, 0.0, 0.7);
         cairo_paint (cr);
-        gs_window_clear_to_background_surface (window);
         gtk_widget_queue_draw (widget);
         return FALSE;
 }
