diff -Naur guake-0.5.1.orig/src/guake.py guake-0.5.1/src/guake.py
--- guake-0.5.1.orig/src/guake.py	2014-11-13 00:07:02.000000000 +0100
+++ guake-0.5.1/src/guake.py	2014-11-20 16:28:19.583666637 +0100
@@ -1121,14 +1121,6 @@
             print "DBG: Showing the terminal"
             self.show()
             self.set_terminal_focus()
-        elif (self.client.get_bool(KEY('/general/focus_if_open')) and
-                self.window.window and
-                (int(self.window.window.get_state()) & GDK_WINDOW_STATE_STICKY or
-                 int(self.window.window.get_state()) & GDK_WINDOW_STATE_WITHDRAWN
-                 )):
-            print "DBG: Restoring the focus to the terminal"
-            self.window.window.focus()
-            self.set_terminal_focus()
         else:
             print "DBG: hiding the terminal"
             self.hide()
