diff -Naur goobox-3.0.1.orig/src/main.c goobox-3.0.1/src/main.c
--- goobox-3.0.1.orig/src/main.c	2011-08-22 19:55:45.000000000 +0200
+++ goobox-3.0.1/src/main.c	2012-04-15 02:59:34.476683867 +0200
@@ -729,15 +729,15 @@
 						window,
 						NULL);
 
-		notify_notification_set_hint_byte (notification,
-						   "action-icons",
-						   TRUE);
+		notify_notification_set_hint (notification,
+					      "action-icons",
+					      g_variant_new_boolean (TRUE));
 	}
 
 	if (notification_supports_persistence)
-		notify_notification_set_hint_byte (notification,
-						    "resident" /* "transient" */,
-						    TRUE);
+		notify_notification_set_hint (notification,
+					      "resident" /* "transient" */,
+					      g_variant_new_boolean (TRUE));
 
 	notify_notification_show (notification, NULL);
 
