diff -Naur cinnamon-control-center-1.7.1.orig/panels/sound-nua/gvc-mixer-dialog.c cinnamon-control-center-1.7.1/panels/sound-nua/gvc-mixer-dialog.c
--- cinnamon-control-center-1.7.1.orig/panels/sound-nua/gvc-mixer-dialog.c	2013-02-19 17:37:17.000000000 +0100
+++ cinnamon-control-center-1.7.1/panels/sound-nua/gvc-mixer-dialog.c	2013-02-26 23:12:37.192399508 +0100
@@ -1929,9 +1929,9 @@
         gvc_channel_bar_set_name (GVC_CHANNEL_BAR (self->priv->input_bar),
                                   _("_Input volume:"));
         gvc_channel_bar_set_low_icon_name (GVC_CHANNEL_BAR (self->priv->input_bar),
-                                           "audio-input-microphone-low-symbolic");
+                                           "cin-audio-input-microphone-low-symbolic");
         gvc_channel_bar_set_high_icon_name (GVC_CHANNEL_BAR (self->priv->input_bar),
-                                            "audio-input-microphone-high-symbolic");
+                                            "cin-audio-input-microphone-high-symbolic");
         gtk_widget_set_sensitive (self->priv->input_bar, FALSE);
 
         if (self->priv->size_group != NULL) {
