diff -Naur activity-log-manager-0.9.7.orig/src/unified-privacy.vala activity-log-manager-0.9.7/src/unified-privacy.vala
--- activity-log-manager-0.9.7.orig/src/unified-privacy.vala	2013-07-10 08:28:23.000000000 +0200
+++ activity-log-manager-0.9.7/src/unified-privacy.vala	2014-05-13 15:15:51.042857849 +0200
@@ -291,7 +291,7 @@
 			// Add/Remove buttons
 			var exception_toolbar = new Toolbar();
 			exception_toolbar.toolbar_style = ToolbarStyle.ICONS;
-			exception_toolbar.icon_size = 1;
+			exception_toolbar.icon_size = Gtk.IconSize.MENU;
 			exception_toolbar.icon_size_set = true;
 			exception_toolbar.visible = true;
 			
