diff -aur MediaInfoLib/Source/MediaInfo/Reader/Reader_libcurl.cpp MediaInfoLib.patched/Source/MediaInfo/Reader/Reader_libcurl.cpp
--- MediaInfoLib/Source/MediaInfo/Reader/Reader_libcurl.cpp	2012-04-08 20:25:35.000000000 +0200
+++ MediaInfoLib.patched/Source/MediaInfo/Reader/Reader_libcurl.cpp	2012-04-13 05:14:46.924995168 +0200
@@ -249,15 +249,15 @@
             #else
                 libcurl_Module=dlopen(MEDIAINFODLL_NAME, RTLD_LAZY);
                 if (!libcurl_Module)
-                    libcurl_Module=dlopen("./"MEDIAINFODLL_NAME, RTLD_LAZY);
+                    libcurl_Module=dlopen("./" MEDIAINFODLL_NAME, RTLD_LAZY);
                 if (!libcurl_Module)
-                    libcurl_Module=dlopen("/usr/local/lib/"MEDIAINFODLL_NAME, RTLD_LAZY);
+                    libcurl_Module=dlopen("/usr/local/lib/" MEDIAINFODLL_NAME, RTLD_LAZY);
                 if (!libcurl_Module)
-                    libcurl_Module=dlopen("/usr/local/lib64/"MEDIAINFODLL_NAME, RTLD_LAZY);
+                    libcurl_Module=dlopen("/usr/local/lib64/" MEDIAINFODLL_NAME, RTLD_LAZY);
                 if (!libcurl_Module)
-                    libcurl_Module=dlopen("/usr/lib/"MEDIAINFODLL_NAME, RTLD_LAZY);
+                    libcurl_Module=dlopen("/usr/lib/" MEDIAINFODLL_NAME, RTLD_LAZY);
                 if (!libcurl_Module)
-                    libcurl_Module=dlopen("/usr/lib64/"MEDIAINFODLL_NAME, RTLD_LAZY);
+                    libcurl_Module=dlopen("/usr/lib64/" MEDIAINFODLL_NAME, RTLD_LAZY);
             #endif
             if (!libcurl_Module)
             {
diff -aur MediaInfoLib/Source/MediaInfoDLL/MediaInfoDLL.h MediaInfoLib.patched/Source/MediaInfoDLL/MediaInfoDLL.h
--- MediaInfoLib/Source/MediaInfoDLL/MediaInfoDLL.h	2012-04-08 12:08:50.000000000 +0200
+++ MediaInfoLib.patched/Source/MediaInfoDLL/MediaInfoDLL.h	2012-04-13 05:14:05.481020899 +0200
@@ -248,24 +248,24 @@
 
 #ifdef MEDIAINFO_GLIBC
 #define MEDIAINFO_ASSIGN(_Name,_Name2) \
-    if (!g_module_symbol (MediaInfo_Module, "MediaInfo"MEDIAINFO_Ansi"_"_Name2, (gpointer*)&MediaInfo_##_Name)) \
+    if (!g_module_symbol (MediaInfo_Module, "MediaInfo" MEDIAINFO_Ansi "_" _Name2, (gpointer*)&MediaInfo_##_Name)) \
         Errors++;
 #define MEDIAINFOLIST_ASSIGN(_Name,_Name2) \
-    if (!g_module_symbol (MediaInfo_Module, "MediaInfoList"MEDIAINFO_Ansi"_"_Name2, (gpointer*)&MediaInfoList_##_Name)) \
+    if (!g_module_symbol (MediaInfo_Module, "MediaInfoList" MEDIAINFO_Ansi "_" _Name2, (gpointer*)&MediaInfoList_##_Name)) \
         Errors++;
 #elif defined (_WIN32) || defined (WIN32)
 #define MEDIAINFO_ASSIGN(_Name,_Name2) \
-    MediaInfo_##_Name=(MEDIAINFO_##_Name)GetProcAddress(MediaInfo_Module, "MediaInfo"MEDIAINFO_Ansi"_"_Name2); \
+    MediaInfo_##_Name=(MEDIAINFO_##_Name)GetProcAddress(MediaInfo_Module, "MediaInfo" MEDIAINFO_Ansi "_" _Name2); \
     if (MediaInfo_##_Name==NULL) Errors++;
 #define MEDIAINFOLIST_ASSIGN(_Name,_Name2) \
-    MediaInfoList_##_Name=(MEDIAINFOLIST_##_Name)GetProcAddress(MediaInfo_Module, "MediaInfoList"MEDIAINFO_Ansi"_"_Name2); \
+    MediaInfoList_##_Name=(MEDIAINFOLIST_##_Name)GetProcAddress(MediaInfo_Module, "MediaInfoList" MEDIAINFO_Ansi "_" _Name2); \
     if (MediaInfoList_##_Name==NULL) Errors++;
 #else
 #define MEDIAINFO_ASSIGN(_Name,_Name2) \
-    MediaInfo_##_Name=(MEDIAINFO_##_Name)dlsym(MediaInfo_Module, "MediaInfo"MEDIAINFO_Ansi"_"_Name2); \
+    MediaInfo_##_Name=(MEDIAINFO_##_Name)dlsym(MediaInfo_Module, "MediaInfo" MEDIAINFO_Ansi "_" _Name2); \
     if (MediaInfo_##_Name==NULL) Errors++;
 #define MEDIAINFOLIST_ASSIGN(_Name,_Name2) \
-    MediaInfoList_##_Name=(MEDIAINFOLIST_##_Name)dlsym(MediaInfo_Module, "MediaInfoList"MEDIAINFO_Ansi"_"_Name2); \
+    MediaInfoList_##_Name=(MEDIAINFOLIST_##_Name)dlsym(MediaInfo_Module, "MediaInfoList" MEDIAINFO_Ansi "_" _Name2); \
     if (MediaInfoList_##_Name==NULL) Errors++;
 #endif
 
@@ -317,15 +317,15 @@
     #else
         MediaInfo_Module=dlopen(MEDIAINFODLL_NAME, RTLD_LAZY);
         if (!MediaInfo_Module)
-            MediaInfo_Module=dlopen("./"MEDIAINFODLL_NAME, RTLD_LAZY);
+            MediaInfo_Module=dlopen("./" MEDIAINFODLL_NAME, RTLD_LAZY);
         if (!MediaInfo_Module)
-            MediaInfo_Module=dlopen("/usr/local/lib/"MEDIAINFODLL_NAME, RTLD_LAZY);
+            MediaInfo_Module=dlopen("/usr/local/lib/" MEDIAINFODLL_NAME, RTLD_LAZY);
         if (!MediaInfo_Module)
-            MediaInfo_Module=dlopen("/usr/local/lib64/"MEDIAINFODLL_NAME, RTLD_LAZY);
+            MediaInfo_Module=dlopen("/usr/local/lib64/" MEDIAINFODLL_NAME, RTLD_LAZY);
         if (!MediaInfo_Module)
-            MediaInfo_Module=dlopen("/usr/lib/"MEDIAINFODLL_NAME, RTLD_LAZY);
+            MediaInfo_Module=dlopen("/usr/lib/" MEDIAINFODLL_NAME, RTLD_LAZY);
         if (!MediaInfo_Module)
-            MediaInfo_Module=dlopen("/usr/lib64/"MEDIAINFODLL_NAME, RTLD_LAZY);
+            MediaInfo_Module=dlopen("/usr/lib64/" MEDIAINFODLL_NAME, RTLD_LAZY);
     #endif
     if (!MediaInfo_Module)
         return (size_t)-1;
