Index: rubberband-1.4/src/system/sysutils.h
===================================================================
--- rubberband-1.4.orig/src/system/sysutils.h
+++ rubberband-1.4/src/system/sysutils.h
@@ -81,6 +81,7 @@ inline float princargf(float a) { return
 
 #include <sys/mman.h>
 #include <dlfcn.h>
+#include <stdio.h> // perror
 
 #define MLOCK(a,b)   ::mlock((char *)(a),(b))
 #define MUNLOCK(a,b) (::munlock((char *)(a),(b)) ? (::perror("munlock failed"), 0) : 0)
