diff --git a/configure.ac b/configure.ac
index 2d807f5..46353a2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -62,7 +62,7 @@ AM_CONDITIONAL(BUILD_RECORDER, $build_recorder)
 
 CLUTTER_MIN_VERSION=1.7.5
 GOBJECT_INTROSPECTION_MIN_VERSION=0.10.1
-GJS_MIN_VERSION=1.29.18
+GJS_MIN_VERSION=1.9.0
 MUFFIN_MIN_VERSION=1.0.5
 GTK_MIN_VERSION=3.0.0
 GIO_MIN_VERSION=2.29.10
diff --git a/files/usr/share/cinnamon/applets/sound@cinnamon.org/applet.js b/files/usr/share/cinnamon/applets/sound@cinnamon.org/applet.js
index e80d97b..73fb87c 100644
--- a/files/usr/share/cinnamon/applets/sound@cinnamon.org/applet.js
+++ b/files/usr/share/cinnamon/applets/sound@cinnamon.org/applet.js
@@ -108,8 +108,8 @@ let compatible_players = [
     'deadbeef', 'smplayer', 'tomahawk'];
 let support_seek = [
     'clementine', 'banshee', 'rhythmbox', 'rhythmbox3', 'pragha', 'quodlibet',
-    'amarok', 'noise', 'xnoise', 'gmusicbrowser', 'spotify', 'vlc', 'beatbox',
-    'gnome-mplayer', 'qmmp', 'deadbeef', 'tomahawk'];
+    'amarok', 'xnoise', 'gmusicbrowser', 'spotify', 'vlc', 'beatbox',
+    'gnome-mplayer', 'qmmp', 'deadbeef'];
 /* dummy vars for translation */
 let x = _("Playing");
 x = _("Paused");
diff --git a/files/usr/share/cinnamon/applets/user@cinnamon.org/applet.js b/files/usr/share/cinnamon/applets/user@cinnamon.org/applet.js
index c2a4c61..0bdd671 100644
--- a/files/usr/share/cinnamon/applets/user@cinnamon.org/applet.js
+++ b/files/usr/share/cinnamon/applets/user@cinnamon.org/applet.js
@@ -94,7 +94,20 @@ MyApplet.prototype = {
                 }       
             }));
 
-            if (GLib.file_test("/usr/bin/mdmflexiserver", GLib.FileTest.EXISTS)) {
+            if (GLib.getenv("XDG_SEAT_PATH")) {
+                // LightDM
+                this.menu.addAction(_("Switch User"), Lang.bind(this, function() {
+                    Util.spawnCommandLine("cinnamon-screensaver-command --lock");
+                    Util.spawnCommandLine("dm-tool switch-to-greeter");
+                }));
+
+                this.menu.addAction(_("Guest Session"), Lang.bind(this, function() {
+                    Util.spawnCommandLine("cinnamon-screensaver-command --lock");
+                    Util.spawnCommandLine("dm-tool switch-to-guest");
+                }));
+            }
+            else if (GLib.file_test("/usr/bin/mdmflexiserver", GLib.FileTest.EXISTS)) {
+                // MDM                
                 this.menu.addAction(_("Switch User"), Lang.bind(this, function() {
                     Util.spawnCommandLine("mdmflexiserver");
                 }));
diff --git a/js/ui/hotCorner.js b/js/ui/hotCorner.js
index 82dafab..af91dda 100644
--- a/js/ui/hotCorner.js
+++ b/js/ui/hotCorner.js
@@ -230,17 +230,6 @@ HotCorner.prototype = {
         this._animRipple(this._ripple3, 0.35,  1.0,   0.0,   0.3,     1);
     },
 
-    handleDragOver: function(source, actor, x, y, time) {
-        if (source != Main.xdndHandler)
-            return;
-
-        if (!Main.overview.visible && !Main.overview.animationInProgress && !Main.expo.visible) {
-            this.rippleAnimation();
-            Main.overview.showTemporarily();
-            Main.overview.beginItemDrag(actor);
-        }
-    },
-
     runAction: function(){
         this._activationTime = Date.now() / 1000;
 
