diff -ruN glob2-0.9.3.orig/src/AIEcho.cpp glob2-0.9.3/src/AIEcho.cpp
--- glob2-0.9.3.orig/src/AIEcho.cpp	2008-04-06 02:13:11.000000000 +0200
+++ glob2-0.9.3/src/AIEcho.cpp	2008-06-21 15:57:45.000000000 +0200
@@ -29,6 +29,7 @@
 #include "GlobalContainer.h"
 #include "Order.h"
 #include <iterator>
+#include <typeinfo>
 #include "Utilities.h"
 #include "boost/tuple/tuple_io.hpp"
 #include "Brush.h"
diff -ruN glob2-0.9.3.orig/src/FertilityCalculatorThreadMessage.cpp glob2-0.9.3/src/FertilityCalculatorThreadMessage.cpp
--- glob2-0.9.3.orig/src/FertilityCalculatorThreadMessage.cpp	2008-03-08 06:27:17.000000000 +0100
+++ glob2-0.9.3/src/FertilityCalculatorThreadMessage.cpp	2008-06-21 15:57:22.000000000 +0200
@@ -19,6 +19,7 @@
 #include "FertilityCalculatorThreadMessage.h"
 
 #include <sstream>
+#include <typeinfo>
 
 FCTUpdateCompletionPercent::FCTUpdateCompletionPercent(float percent)
 	: percent(percent)
diff -ruN glob2-0.9.3.orig/src/NetMessage.cpp glob2-0.9.3/src/NetMessage.cpp
--- glob2-0.9.3.orig/src/NetMessage.cpp	2008-03-18 07:59:03.000000000 +0100
+++ glob2-0.9.3/src/NetMessage.cpp	2008-06-21 15:57:07.000000000 +0200
@@ -20,6 +20,7 @@
 #include <algorithm>
 #include <iostream>
 #include <sstream>
+#include <typeinfo>
 #include "Version.h"
 #include "BinaryStream.h"
 
