diff -wbBur frei0r-plugins-1.2/src/filter/vertigo/vertigo.c frei0r-plugins-1.2.my//src/filter/vertigo/vertigo.c
--- frei0r-plugins-1.2/src/filter/vertigo/vertigo.c	2010-09-13 10:00:31.000000000 +0000
+++ frei0r-plugins-1.2.my//src/filter/vertigo/vertigo.c	2010-09-29 10:50:02.000000000 +0000
@@ -67,7 +67,7 @@
   vertigoInfo->frei0r_version = FREI0R_MAJOR_VERSION;
   vertigoInfo->major_version = 1;
   vertigoInfo->minor_version = 0;
-  vertigoInfo->num_params =  3;
+  vertigoInfo->num_params =  2;
   vertigoInfo->explanation = "alpha blending with zoomed and rotated images";
 }
 
