diff -wbBur online-libreoffice-5.3.4.2/kit/Kit.cpp online-libreoffice-5.3.4.2.q/kit/Kit.cpp
--- online-libreoffice-5.3.4.2/kit/Kit.cpp	2017-07-12 19:24:55.656651908 +0300
+++ online-libreoffice-5.3.4.2.q/kit/Kit.cpp	2017-07-12 19:24:32.139985511 +0300
@@ -150,8 +150,13 @@
             File(newPath.parent()).createDirectories();
             if (link(fpath, newPath.toString().c_str()) == -1)
             {
+                if(errno != EEXIST)
+                {
+                    if(errno != EXDEV)
+                    {
                 LOG_SYS("link(\"" << fpath << "\", \"" <<
                         newPath.toString() << "\") failed. Will copy.");
+                    }
                 try
                 {
                     File(fpath).copyTo(newPath.toString());
@@ -163,6 +168,7 @@
                     std::_Exit(Application::EXIT_SOFTWARE);
                 }
             }
+            }
             break;
         case FTW_D:
             {
