From 4d2b79048573dd2fa1dee853230ccc45b1d9c8df Mon Sep 17 00:00:00 2001
From: Andriy Grytsenko <andrej@rep.kiev.ua>
Date: Tue, 25 Feb 2014 15:24:59 +0200
Subject: [PATCH] [LP#1284401]The lxshortcut should not be present in the
 menus.

---
 data/lxshortcut.desktop.in |    1 +
 1 file changed, 1 insertion(+)

diff --git a/data/lxshortcut.desktop.in b/data/lxshortcut.desktop.in
index 8ef7bca..f60a63e 100644
--- a/data/lxshortcut.desktop.in
+++ b/data/lxshortcut.desktop.in
@@ -8,3 +8,4 @@ Exec=lxshortcut -i %f
 StartupNotify=true
 Terminal=false
 MimeType=application/x-desktop;
+NoDisplay=true
-- 
1.7.10.4

