--- inc/InterfaceGraphic_X11.hxx	2014-04-29 11:40:35.000000000 +0200
+++ inc/InterfaceGraphic_X11.hxx.new	2014-10-10 01:16:49.274370801 +0200
@@ -21,6 +21,8 @@
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <X11/Xatom.h>
+typedef ptrdiff_t GLsizeiptr;
+typedef ptrdiff_t GLintptr;
 #include <GL/glx.h>
 
 #if defined (sun) || defined (SUNOS) || defined (__alpha) || defined (DECOSF1) || defined (sgi) || defined (IRIX) || defined (__hpux)|| defined (HPUX)
