diff -aur vanilla/VirtualBox-4.1.18/src/VBox/Frontends/VBoxHeadless/FramebufferVNC.cpp changed/VirtualBox-4.1.18/src/VBox/Frontends/VBoxHeadless/FramebufferVNC.cpp
--- vanilla/VirtualBox-4.1.18/src/VBox/Frontends/VBoxHeadless/FramebufferVNC.cpp	2012-06-20 15:15:53.000000000 +0200
+++ changed/VirtualBox-4.1.18/src/VBox/Frontends/VBoxHeadless/FramebufferVNC.cpp	2012-07-16 16:24:22.448349886 +0200
@@ -88,8 +88,16 @@
 
     vncServer = rfbGetScreen(0, NULL, mWidth, mHeight, 8, 3, 1);
     vncServer->screenData = (void*)this;
+#ifdef LIBVNCSERVER_IPv6
+    if (mVncPort) {
+	    vncServer->port = mVncPort;
+	    vncServer->ipv6port = mVncPort;
+    }
+#else
     if (mVncPort)
         vncServer->port = mVncPort;
+#endif
+
     char *pszDesktopName;
     rc = RTStrAPrintf(&pszDesktopName, "%s - VirtualBox", pszName);
     if (rc >= 0)
