diff -Nru lmms-0.4.4.orig/cmake/modules/DetectMachine.cmake lmms-0.4.4/cmake/modules/DetectMachine.cmake
--- lmms-0.4.4.orig/cmake/modules/DetectMachine.cmake	2009-05-05 02:13:12.000000000 +0300
+++ lmms-0.4.4/cmake/modules/DetectMachine.cmake	2009-05-31 18:22:42.018152824 +0300
@@ -32,7 +32,7 @@
 	SET(LIB_DIR "${CMAKE_INSTALL_LIBDIR}")
 ELSE(CMAKE_INSTALL_LIBDIR)
 	IF(LMMS_HOST_X86_64)
-		SET(LIB_DIR lib64)
+		SET(LIB_DIR lib)
 	ELSE(LMMS_HOST_X86_64)
 		SET(LIB_DIR lib)
 	ENDIF(LMMS_HOST_X86_64)
