diff -rupN noise-0.4.orig/data/org.pantheon.noise.desktop.in.in noise-0.4/data/org.pantheon.noise.desktop.in.in
--- noise-0.4.orig/data/org.pantheon.noise.desktop.in.in	2016-08-12 12:42:36.415710999 +0200
+++ noise-0.4/data/org.pantheon.noise.desktop.in.in	2016-08-12 12:46:39.418412632 +0200
@@ -3,7 +3,7 @@ Type=Application
 _Name=@APP_NAME@
 _GenericName=Music Player
 _Comment=Listen to music
-Exec=noise %U
+Exec=pantheon-music %U
 Icon=multimedia-audio-player
 Terminal=false
 Categories=Audio;Music;Player;AudioVideo;GNOME;GTK;
@@ -15,5 +15,5 @@ X-PulseAudio-Properties=media.role=music
 Actions=AboutDialog;
 
 [Desktop Action AboutDialog]
-Exec=noise --about
+Exec=pantheon-music --about
 _Name=About @APP_NAME@
diff -rupN noise-0.4.orig/src/CMakeLists.txt noise-0.4/src/CMakeLists.txt
--- noise-0.4.orig/src/CMakeLists.txt	2016-08-12 12:42:36.425710687 +0200
+++ noise-0.4/src/CMakeLists.txt	2016-08-12 12:45:25.250582352 +0200
@@ -88,7 +88,7 @@ vala_precompile (CLIENT_VALA_C ${SRC_TAR
   ${CLIENT_VAPI}
 )
 
-set (EXEC_NAME ${SRC_TARGET})
+set (EXEC_NAME pantheon-music)
 
 add_definitions (${DEPS_CFLAGS})
 link_directories (${DEPS_LIBRARY_DIRS})
