diff -u -r libmirage-3.0.0/filters/filter-daa/CMakeLists.txt libmirage-3.0.0-xml/filters/filter-daa/CMakeLists.txt
--- libmirage-3.0.0/filters/filter-daa/CMakeLists.txt	2014-06-29 15:25:04.000000000 +0200
+++ libmirage-3.0.0-xml/filters/filter-daa/CMakeLists.txt	2014-06-29 23:31:23.432780158 +0200
@@ -35,6 +35,7 @@
 
     # Install MIME type
     install(FILES "libmirage-${filter_short}.xml" DESTINATION ${CMAKE_INSTALL_DATADIR}/mime/packages)
+    install(FILES "libmirage-gbi.xml" DESTINATION ${CMAKE_INSTALL_DATADIR}/mime/packages)
     if (POST_INSTALL_HOOKS)
         install(CODE "execute_process (COMMAND ${UPDATE_MIME_DATABASE_EXECUTABLE} ${CMAKE_INSTALL_FULL_DATADIR}/mime)")
     endif ()
diff -u -r libmirage-3.0.0/images/image-mds/CMakeLists.txt libmirage-3.0.0-xml/images/image-mds/CMakeLists.txt
--- libmirage-3.0.0/images/image-mds/CMakeLists.txt	2014-06-29 15:25:04.000000000 +0200
+++ libmirage-3.0.0-xml/images/image-mds/CMakeLists.txt	2014-06-29 23:32:31.092485925 +0200
@@ -19,6 +19,7 @@
 
     # Install MIME type
     install(FILES "libmirage-${image_short}.xml" DESTINATION ${CMAKE_INSTALL_DATADIR}/mime/packages)
+    install(FILES "libmirage-xmd.xml" DESTINATION ${CMAKE_INSTALL_DATADIR}/mime/packages)
     if (POST_INSTALL_HOOKS)
         install(CODE "execute_process (COMMAND ${UPDATE_MIME_DATABASE_EXECUTABLE} ${CMAKE_INSTALL_FULL_DATADIR}/mime)")
     endif ()
