diff --git i/libmirage/src/filters/filter-daa/CMakeLists.txt w/libmirage/src/filters/filter-daa/CMakeLists.txt
index ebd2890..e7021c1 100644
--- i/libmirage/src/filters/filter-daa/CMakeLists.txt
+++ w/libmirage/src/filters/filter-daa/CMakeLists.txt
@@ -35,6 +35,7 @@ if (${ZLIB_FOUND})
 
     # Install MIME type
     install(FILES "libmirage-${filter_short}.xml" DESTINATION ${CMAKE_INSTALL_DATADIR}/mime/packages)
+    install(FILES "libmirage-gbi.xml" DESTINATION ${CMAKE_INSTALL_DATADIR}/mime/packages)
     if (POST_INSTALL_HOOKS)
         install(CODE "execute_process (COMMAND ${UPDATE_MIME_DATABASE_EXECUTABLE} ${CMAKE_INSTALL_FULL_DATADIR}/mime)")
     endif ()
diff --git i/libmirage/src/parsers/image-mds/CMakeLists.txt w/libmirage/src/parsers/image-mds/CMakeLists.txt
index c97be8f..bb0ee2b 100644
--- i/libmirage/src/parsers/image-mds/CMakeLists.txt
+++ w/libmirage/src/parsers/image-mds/CMakeLists.txt
@@ -20,6 +20,7 @@ if (TRUE)
 
     # Install MIME type
     install(FILES "libmirage-${parser_short}.xml" DESTINATION ${CMAKE_INSTALL_DATADIR}/mime/packages)
+    install(FILES "libmirage-xmd.xml" DESTINATION ${CMAKE_INSTALL_DATADIR}/mime/packages)
     if (POST_INSTALL_HOOKS)
         install(CODE "execute_process (COMMAND ${UPDATE_MIME_DATABASE_EXECUTABLE} ${CMAKE_INSTALL_FULL_DATADIR}/mime)")
     endif ()
