--- tig.c	2011-03-02 10:38:53.529056596 -0500
+++ tig.c	2011-03-02 10:40:08.448056588 -0500
@@ -2727,13 +2727,13 @@
 
 	foreach_displayed_view (view, i) {
 		if (!view->win) {
-			view->win = newwin(view->height, 0, offset, 0);
+			view->win = newwin(view->height, view->width, offset, 0);
 			if (!view->win)
 				die("Failed to create %s view", view->name);
 
 			scrollok(view->win, FALSE);
 
-			view->title = newwin(1, 0, offset + view->height, 0);
+			view->title = newwin(1, view->width, offset + view->height, 0);
 			if (!view->title)
 				die("Failed to create title window");
 
@@ -7103,7 +7103,7 @@
 		init_colors();
 
 	getmaxyx(stdscr, y, x);
-	status_win = newwin(1, 0, y - 1, 0);
+	status_win = newwin(1, x, y - 1, 0);
 	if (!status_win)
 		die("Failed to create status window");
 
