--- a/configure.ac
+++ b/configure.ac
@@ -129,14 +129,14 @@ else
 fi
 AC_SUBST(JAW_CFLAGS)
 
-GNOME_COMPILE_WARNINGS([error])
-case "$WARN_CFLAGS" in
-    *-Werror*)
-        WARN_CFLAGS="$WARN_CFLAGS -Wno-error=deprecated-declarations"
-        ;;
-esac
-AM_CFLAGS="$AM_CFLAGS $WARN_CFLAGS"
-AC_SUBST(AM_CFLAGS)
+#GNOME_COMPILE_WARNINGS([error])
+#case "$WARN_CFLAGS" in
+#    *-Werror*)
+#        WARN_CFLAGS="$WARN_CFLAGS -Wno-error=deprecated-declarations"
+#        ;;
+#esac
+#AM_CFLAGS="$AM_CFLAGS $WARN_CFLAGS"
+#AC_SUBST(AM_CFLAGS)
 
 CHECK_JNI
 
