diff -wbBur luasec-luasec-0.6-51/src/context.c luasec-luasec-0.6/src/context.c
--- luasec-luasec-0.6-51/src/context.c	2016-03-03 22:11:46.000000000 +0300
+++ luasec-luasec-0.6/src/context.c	2016-03-07 16:47:50.328080321 +0300
@@ -96,11 +96,6 @@
  */
 static LSEC_SSL_METHOD* str2method(const char *method)
 {
-  if (!strcmp(method, "any"))     return SSLv23_method();
-  if (!strcmp(method, "sslv23"))  return SSLv23_method();  // deprecated
-#ifndef OPENSSL_NO_SSL3
-  if (!strcmp(method, "sslv3"))   return SSLv3_method();
-#endif
   if (!strcmp(method, "tlsv1"))   return TLSv1_method();
 #if (OPENSSL_VERSION_NUMBER >= 0x1000100fL)
   if (!strcmp(method, "tlsv1_1")) return TLSv1_1_method();
