--- otp/lib/crypto/c_src/crypto.c	2017-04-28 09:39:16.685119582 +0000
+++ otp/lib/crypto/c_src/crypto.c	2017-04-28 09:41:19.371604045 +0000
@@ -1720,7 +1720,7 @@
     ErlNifBinary     key, ivec, text;
 #ifdef HAVE_EVP_AES_CTR
     const EVP_CIPHER *cipher;
-    EVP_CIPHER_CTX   ctx;
+    EVP_CIPHER_CTX   *ctx;
     unsigned char    *out;
     int              outl = 0;
 #else
@@ -1749,15 +1749,15 @@
     }
 
     out = enif_make_new_binary(env,text.size,&ret);
-    EVP_CIPHER_CTX_init(&ctx);
-    EVP_CipherInit_ex(&ctx, cipher, NULL,
+    ctx = EVP_CIPHER_CTX_new();
+    EVP_CipherInit_ex(ctx, cipher, NULL,
                       key.data, ivec.data, (argv[3] == atom_true));
     EVP_CIPHER_CTX_set_padding(&ctx, 0);
-    EVP_CipherUpdate(&ctx, out, &outl, text.data, text.size);
+    EVP_CipherUpdate(ctx, out, &outl, text.data, text.size);
     ASSERT(outl == text.size);
-    EVP_CipherFinal_ex(&ctx, out + outl, &outl);
+    EVP_CipherFinal_ex(ctx, out + outl, &outl);
     ASSERT(outl == 0);
-    EVP_CIPHER_CTX_cleanup(&ctx);
+    EVP_CIPHER_CTX_free(ctx);
 #else
     memcpy(ivec_clone, ivec.data, 16);    
     memset(ecount_buf, 0, sizeof(ecount_buf));
