diff -Naur openbox-3.5.2.orig/data/gnome-session/openbox-gnome.session openbox-3.5.2/data/gnome-session/openbox-gnome.session
--- openbox-3.5.2.orig/data/gnome-session/openbox-gnome.session	2013-09-02 16:06:03.076984284 +0000
+++ openbox-3.5.2/data/gnome-session/openbox-gnome.session	2013-09-02 16:07:32.696385427 +0000
@@ -1,9 +1,3 @@
 [GNOME Session]
 Name=GNOME/Openbox
-RequiredComponents=gnome-settings-daemon;
-# Try load with the gnome-panel and use the fallback if we can't load a panel
-RequiredProviders=windowmanager;panel
-DefaultProvider-windowmanager=openbox
-DefaultProvider-panel=gnome-panel
-FallbackSession=openbox-gnome-fallback
-DesktopName=GNOME
+RequiredComponents=gnome-panel;gnome-settings-daemon;gnome-screensaver;openbox;notification-daemon;polkit-gnome-authentication-agent-1;gnome-fallback-background-helper;gnome-fallback-media-keys-helper;gnome-fallback-mount-helper;
