--- osmo-pim/configure.ac	2011-08-15 19:06:25.184165142 -0700
+++ osmo-pim/configure.ac.new	2011-08-15 19:35:19.285312355 -0700
@@ -4,7 +4,7 @@
 AC_PREREQ(2.61)
 AC_INIT([Osmo], [0.2.12], [pasp@users.sf.net])
 AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Werror foreign])
 AC_CONFIG_SRCDIR([src/contacts.c])
 AC_CONFIG_HEADER([config.h])
 AC_MSG_CHECKING(for type of build)
@@ -245,9 +245,9 @@
 			[configdir=$withval AC_SUBST([configdir])])
 AM_CONDITIONAL([CONFIGDIR], [test x$configdir != x])
 
-CFLAGS=${CFLAGS:="$BUILD_CFLAGS -Wall $PLATFORM_CFLAGS -D_GNU_SOURCE"}
-CXXFLAGS=${CXXFLAGS:="$CFLAGS"}
-CPPFLAGS=""
+CFLAGS="$BUILD_CFLAGS -Wall $PLATFORM_CFLAGS $CFLAGS"
+CXXFLAGS="$CFLAGS $CXXFLAGS"
+CPPFLAGS="-D_GNU_SOURCE"
 LIBS="$gringotts_LIBS"
 
 AC_CONFIG_FILES([
