diff -Nur qbittorrent-2.4.8.orig/src/searchengine/searchengine.cpp qbittorrent-2.4.8/src/searchengine.cpp
--- qbittorrent-2.4.8.orig/src/searchengine/searchengine.cpp	2010-10-01 13:52:53.000000000 -0700
+++ qbittorrent-2.4.8/src/searchengine/searchengine.cpp	2010-10-25 01:20:20.046700620 -0700
@@ -354,7 +354,7 @@
   //on change le texte du label courrant
   currentSearchTab->getCurrentLabel()->setText(tr("Results")+" <i>(0)</i>:");
   // Launch search
-  searchProcess->start("python", params, QIODevice::ReadOnly);
+  searchProcess->start("python2", params, QIODevice::ReadOnly);
   searchTimeout->start(180000); // 3min
 }
 
@@ -421,7 +421,7 @@
     params << engine_url;
     params << torrent_url;
     // Launch search
-    downloadProcess->start("python", params, QIODevice::ReadOnly);
+    downloadProcess->start("python2", params, QIODevice::ReadOnly);
   }
 }
 
diff -Nur qbittorrent-2.4.8.orig/src/searchengine/supportedengines.h qbittorrent-2.4.8/src/supportedengines.h
--- qbittorrent-2.4.8.orig/src/searchengine/supportedengines.h	2010-10-24 01:32:39.000000000 -0700
+++ qbittorrent-2.4.8/src/searchengine/supportedengines.h	2010-10-25 01:18:42.230032834 -0700
@@ -144,7 +144,7 @@
     QStringList params;
     params << misc::searchEngineLocation()+QDir::separator()+"nova2.py";
     params << "--capabilities";
-    nova.start("python", params, QIODevice::ReadOnly);
+    nova.start("python2", params, QIODevice::ReadOnly);
     nova.waitForStarted();
     nova.waitForFinished();
     QString capabilities = QString(nova.readAll());
