diff -Nur mysql-workbench-gpl-5.2.36-src.old/configure.in mysql-workbench-gpl-5.2.36-src/configure.in
--- mysql-workbench-gpl-5.2.36-src.old/configure.in	2011-12-05 15:04:56.894410445 +0000
+++ mysql-workbench-gpl-5.2.36-src/configure.in	2011-12-05 15:05:45.818268997 +0000
@@ -245,13 +245,13 @@
 AC_CHECK_PROG(PYTHON27, python2.7, [yes])
 # Python headers
 AC_MSG_CHECKING(for Python libraries)
-PYTHON_LIBS=[$(python -c "from distutils import sysconfig
+PYTHON_LIBS=[$(python2 -c "from distutils import sysconfig
 import os 
 cfg=sysconfig.get_config_vars()
 lib = os.path.join(cfg['LIBDIR'], cfg['LDLIBRARY'])
 if os.path.exists(lib):
     print \"-L\" + cfg['LIBDIR'] + \" -l\" + cfg['LDLIBRARY'].lstrip(\"lib\").rstrip(\".so\")")]
-PYTHON_CFLAGS=[$(python -c "from distutils import sysconfig
+PYTHON_CFLAGS=[$(python2 -c "from distutils import sysconfig
 print sysconfig.get_python_inc()")]
 PYTHON_CFLAGS="-I$PYTHON_CFLAGS"
 if test "$PYTHON_LIBS" = ""; then
