commit c07d348bae9bbb96dcecd682ff50e2e31fdffc48
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Tue Jul 11 12:46:16 2017 +0800

    Fix compatibility with Qt 5.9
    
    Change-Id: I53f07a89ff49c69a40e06ae91fc4b6e431280630

diff --git a/platformplugin/dplatformwindowhelper.cpp b/platformplugin/dplatformwindowhelper.cpp
index 3145836..1700d6a 100644
--- a/platformplugin/dplatformwindowhelper.cpp
+++ b/platformplugin/dplatformwindowhelper.cpp
@@ -314,10 +314,17 @@ bool DPlatformWindowHelper::isExposed() const
     return me()->m_frameWindow->handle()->isExposed();
 }
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 9, 0)
+bool DPlatformWindowHelper::isEmbedded() const
+{
+    return me()->m_frameWindow->handle()->isEmbedded();
+}
+#else
 bool DPlatformWindowHelper::isEmbedded(const QPlatformWindow *parentWindow) const
 {
     return me()->m_frameWindow->handle()->isEmbedded(parentWindow);
 }
+#endif
 
 void DPlatformWindowHelper::propagateSizeHints()
 {
diff --git a/platformplugin/dplatformwindowhelper.h b/platformplugin/dplatformwindowhelper.h
index 4c2f367..843aecc 100644
--- a/platformplugin/dplatformwindowhelper.h
+++ b/platformplugin/dplatformwindowhelper.h
@@ -60,7 +60,11 @@ public:
 
     bool isExposed() const;
     bool isActive() const;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 9, 0)
+    bool isEmbedded() const;
+#else
     bool isEmbedded(const QPlatformWindow *parentWindow = 0) const;
+#endif
 
     void propagateSizeHints();
 
