commit 774ffb89733232b8728b897b58db3eab6c7a6e05
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Wed Oct 25 21:48:23 2017 +0800

    fix: compatibility with Qt 5.7.1+
    
    Change-Id: I484234a2008597a728dbe167f578e84c550602c3

diff --git a/platformplugin/dplatformintegration.cpp b/platformplugin/dplatformintegration.cpp
index caa36c4..ca84e4a 100644
--- a/platformplugin/dplatformintegration.cpp
+++ b/platformplugin/dplatformintegration.cpp
@@ -543,7 +543,11 @@ static void overrideChangeCursor(QPlatformCursor *cursorHandle, QCursor * cursor
             it = xcb_cursor->m_cursorHash.insert(key, overrideCreateFontCursor(xcb_cursor, cursor, widget));
         }
         c = it.value();
+#if QT_VERSION < QT_VERSION_CHECK(5, 7, 1)
         w->setCursor(c);
+#else
+        w->setCursor(c, false);
+#endif
     }
 
     VtableHook::callOriginalFun(cursorHandle, &QPlatformCursor::changeCursor, cursor, widget);
