--- Utility/QuickCheck.hs.orig	2016-01-21 14:13:28.661285249 +0800
+++ Utility/QuickCheck.hs	2016-01-21 14:13:34.534674457 +0800
@@ -21,12 +21,6 @@
 import Control.Applicative
 import Prelude
 
-instance (Arbitrary k, Arbitrary v, Eq k, Ord k) => Arbitrary (M.Map k v) where
-	arbitrary = M.fromList <$> arbitrary
-
-instance (Arbitrary v, Eq v, Ord v) => Arbitrary (S.Set v) where
-	arbitrary = S.fromList <$> arbitrary
-
 {- Times before the epoch are excluded. -}
 instance Arbitrary POSIXTime where
 	arbitrary = fromInteger <$> nonNegative arbitrarySizedIntegral
