diff --git a/redis.conf b/redis.conf
index 6efb6ac..344e021 100644
--- a/redis.conf
+++ b/redis.conf
@@ -87,7 +87,7 @@ timeout 0
 # On other kernels the period depends on the kernel configuration.
 #
 # A reasonable value for this option is 60 seconds.
-tcp-keepalive 0
+tcp-keepalive 60
 
 # Specify the server verbosity level.
 # This can be one of:
@@ -184,7 +184,7 @@ dbfilename dump.rdb
 # The Append Only File will also be created inside this directory.
 # 
 # Note that you must specify a directory here, not a file name.
-dir ./
+dir /var/lib/redis/
 
 ################################# REPLICATION #################################
 
