--- wrapper/Makefile.am
+++ wrapper/Makefile.am
@@ -16,7 +16,7 @@
 EXTRA_DIST = $(properties_DATA)
 
 $(JARFILES) : $(DEP_CLASSES)
-	$(JAR) cfm $(JARFILES) manifest.txt org/GNOME/Accessibility/*.class
+	$(JAR) cf $(JARFILES) org/GNOME/Accessibility/*.class
 
 all-local : $(DATA) $(JARFILES)
 
--- wrapper/Makefile.in
+++ wrapper/Makefile.in
@@ -717,7 +717,7 @@
 
 
 $(JARFILES) : $(DEP_CLASSES)
-	$(JAR) cfm $(JARFILES) manifest.txt org/GNOME/Accessibility/*.class
+	$(JAR) cf $(JARFILES) org/GNOME/Accessibility/*.class
 
 all-local : $(DATA) $(JARFILES)
 
