--- a/src/setup.py	2016-11-05 19:34:44.701025334 +0000
+++ b/src/setup.py	2016-11-05 19:46:00.706210289 +0000
@@ -562,16 +562,6 @@
     import Cython.Compiler.Options
     import Cython.Compiler.Main
 
-    # Sage uses these directives (mostly for historical reasons).
-    Cython.Compiler.Options.embed_pos_in_docstring = True
-    Cython.Compiler.Options.directive_defaults['autotestdict'] = False
-    Cython.Compiler.Options.directive_defaults['cdivision'] = True
-    Cython.Compiler.Options.directive_defaults['fast_getattr'] = True
-    # The globals() builtin in Cython was fixed to return to the current scope,
-    # but Sage relies on the broken behavior of returning to the nearest
-    # enclosing Python scope (e.g. to perform variable injection).
-    Cython.Compiler.Options.old_style_globals = True
-
     debug = False
     if os.environ.get('SAGE_DEBUG', None) != 'no':
         print('Enabling Cython debugging support')
@@ -608,7 +598,10 @@
         force=force,
         aliases=aliases,
         compiler_directives={
+            'autotestdict': False,
+            'cdivision': True,
             'embedsignature': True,
+            'fast_getattr': True,
             'profile': profile,
         })
 
