diff --git a/panels/sound-nua/cc-sound-panel.c b/panels/sound-nua/cc-sound-panel.c
index 1c115f0..092f319 100644
--- a/panels/sound-nua/cc-sound-panel.c
+++ b/panels/sound-nua/cc-sound-panel.c
@@ -120,7 +120,7 @@ enum {
         cc_sound_panel_register_type (G_TYPE_MODULE (module));
         g_io_extension_point_implement (CC_SHELL_PANEL_EXTENSION_POINT,
                                         CC_TYPE_SOUND_PANEL,
-                                        "sound-nua", 0);
+                                        "sound", 0);
 }
 
 /* GIO extension stuff */
diff --git a/panels/sound/cc-sound-panel.c b/panels/sound/cc-sound-panel.c
index 2f69c0f..3b78ce2 100644
--- a/panels/sound/cc-sound-panel.c
+++ b/panels/sound/cc-sound-panel.c
@@ -127,7 +127,7 @@ enum {
         cc_sound_panel_register_type (G_TYPE_MODULE (module));
         g_io_extension_point_implement (CC_SHELL_PANEL_EXTENSION_POINT,
                                         CC_TYPE_SOUND_PANEL,
-                                        "sound", 0);
+                                        "gnome-sound", 0);
 }
 
 /* GIO extension stuff */
