diff -aur cdemu-daemon-1.5.0.orig/session/cdemu-daemon-session.sh cdemu-daemon-1.5.0/session/cdemu-daemon-session.sh
--- cdemu-daemon-1.5.0.orig/session/cdemu-daemon-session.sh	2012-03-25 16:35:40.577837996 +0800
+++ cdemu-daemon-1.5.0/session/cdemu-daemon-session.sh	2012-03-25 16:39:12.180024983 +0800
@@ -1,15 +1,15 @@
 #!/bin/sh
 # Starts the CDEmu daemon instance on D-Bus *session* bus. Optional
 # configuration (number of devices, audio driver, log file) are read
-# from ~/.cdemu-daemon
+# from ~/.cdemud
 
 # Default settings
 NUM_DEVICES=2
-AUDIO_DRIVER=pulse
-LOG_FILE=~/.cdemu-daemon.log
+AUDIO_DRIVER=null
+LOG_FILE=~/.cdemud.log
 
 # Read the settings
-CONFIG_FILE=~/.cdemu-daemon
+CONFIG_FILE=~/.cdemud
 
 if [ -f ${CONFIG_FILE} ]; then
     . ${CONFIG_FILE};
diff -aur cdemu-daemon-1.5.0.orig/system/cdemu-daemon-system.sh cdemu-daemon-1.5.0/system/cdemu-daemon-system.sh
--- cdemu-daemon-1.5.0.orig/system/cdemu-daemon-system.sh	2012-03-25 16:35:40.581171312 +0800
+++ cdemu-daemon-1.5.0/system/cdemu-daemon-system.sh	2012-03-25 16:36:48.810801678 +0800
@@ -1,15 +1,15 @@
 #!/bin/sh
 # Starts the CDEmu daemon instance on D-Bus *system* bus. Optional
 # configuration (number of devices, audio driver, log file) are read
-# from /etc/sysconfig/cdemu-daemon
+# from /etc/conf.d/cdemud
 
 # Default settings
 NUM_DEVICES=2
 AUDIO_DRIVER=null
-LOG_FILE=/tmp/cdemu-daemon.log
+LOG_FILE=/var/log/cdemud.log
 
 # Read the settings
-CONFIG_FILE=/etc/sysconfig/cdemu-daemon
+CONFIG_FILE=/etc/conf.d/cdemud
 
 if [ -f ${CONFIG_FILE} ]; then
     . ${CONFIG_FILE};
