--- src/key_grabber.c	2008-04-08 00:12:31.000000000 +0200
+++ src/key_grabber.c	2012-05-17 19:43:25.227699421 +0200
@@ -257,7 +257,7 @@
 gboolean tilda_keygrabber_bind (const gchar *keystr, tilda_window *tw)
 {
     /* Empty strings are no good */
-    if (strcmp ("", keystr) == 0)
+    if (!keystr || strcmp ("", keystr) == 0)
         return FALSE;
 
     return tomboy_keybinder_bind (keystr, onKeybindingPull, tw);
