diff --git a/applets/wncklet/workspace-switcher.c b/applets/wncklet/workspace-switcher.c
index bf6e365..4a568bd 100644
--- a/applets/wncklet/workspace-switcher.c
+++ b/applets/wncklet/workspace-switcher.c
@@ -144,7 +144,7 @@ static void window_manager_changed(WnckScreen* screen, PagerData* pager)
 
 	if (!wm_name)
 		pager->wm = PAGER_WM_UNKNOWN;
-	else if (strcmp(wm_name, "Marco") == 0)
+	else if (strcmp(wm_name, "Metacity (Marco)") == 0)
 		pager->wm = PAGER_WM_MARCO;
 	else if (strcmp(wm_name, "Compiz") == 0)
 		pager->wm = PAGER_WM_COMPIZ;
