diff -Naur linuxmint-nemo-63f7b44.orig/libnemo-private/nemo-global-preferences.c linuxmint-nemo-63f7b44/libnemo-private/nemo-global-preferences.c
--- linuxmint-nemo-63f7b44.orig/libnemo-private/nemo-global-preferences.c	2013-02-19 17:30:18.000000000 +0100
+++ linuxmint-nemo-63f7b44/libnemo-private/nemo-global-preferences.c	2013-03-04 04:05:13.472897136 +0100
@@ -77,5 +77,6 @@
 	nemo_tree_sidebar_preferences = g_settings_new("org.nemo.sidebar-panels.tree");
 	gnome_lockdown_preferences = g_settings_new("org.gnome.desktop.lockdown");
 	gnome_background_preferences = g_settings_new("org.gnome.desktop.background");
-    gnome_media_handling_preferences = g_settings_new("org.gnome.desktop.media-handling");
+	gnome_media_handling_preferences = g_settings_new("org.gnome.desktop.media-handling");
+	gnome_terminal_preferences = g_settings_new("org.gnome.desktop.default-applications.terminal");
 }
diff -Naur linuxmint-nemo-63f7b44.orig/libnemo-private/nemo-global-preferences.h linuxmint-nemo-63f7b44/libnemo-private/nemo-global-preferences.h
--- linuxmint-nemo-63f7b44.orig/libnemo-private/nemo-global-preferences.h	2013-02-19 17:30:18.000000000 +0100
+++ linuxmint-nemo-63f7b44/libnemo-private/nemo-global-preferences.h	2013-03-04 04:49:20.581767625 +0100
@@ -211,6 +211,9 @@
 #define GNOME_DESKTOP_MEDIA_HANDLING_AUTOMOUNT_OPEN "automount-open"
 #define GNOME_DESKTOP_MEDIA_HANDLING_AUTORUN        "autorun-never"
 
+/* Terminal */
+#define GNOME_DESKTOP_TERMINAL_EXEC        "exec"
+
 void nemo_global_preferences_init                      (void);
 char *nemo_global_preferences_get_default_folder_viewer_preference_as_iid (void);
 
@@ -224,6 +227,7 @@
 GSettings *gnome_lockdown_preferences;
 GSettings *gnome_background_preferences;
 GSettings *gnome_media_handling_preferences;
+GSettings *gnome_terminal_preferences;
 
 G_END_DECLS
 
diff -Naur linuxmint-nemo-63f7b44.orig/src/nemo-view.c linuxmint-nemo-63f7b44/src/nemo-view.c
--- linuxmint-nemo-63f7b44.orig/src/nemo-view.c	2013-02-19 17:30:18.000000000 +0100
+++ linuxmint-nemo-63f7b44/src/nemo-view.c	2013-03-04 04:14:47.222310992 +0100
@@ -6716,7 +6716,8 @@
 open_in_terminal (gchar *location)
 {	
     gchar *argv[2];
-    argv[0] = "x-terminal-emulator";
+    argv[0] = g_settings_get_string (gnome_terminal_preferences,
+				     GNOME_DESKTOP_TERMINAL_EXEC);
     argv[1] = NULL;
     g_spawn_async(location, argv, NULL, G_SPAWN_SEARCH_PATH, NULL, NULL, NULL, NULL);
 }
