Description: don't include unnecessary and nonexistent header file.
Origin: upstream, http://websvn.kde.org/?view=revision&revision=1042286

Index: audex-0.72b1/utils/upload.h
===================================================================
--- audex-0.72b1.orig/utils/upload.h	2010-03-07 13:46:20.000000000 +0300
+++ audex-0.72b1/utils/upload.h	2010-03-07 13:46:14.000000000 +0300
@@ -28,7 +28,6 @@
 #include <KLocale>
 #include <KUrl>
 #include <KIO/Job>
-#include <KIO/MkdirJob>
 #include <KIO/SimpleJob>
 #include <KIO/CopyJob>
 
