diff -Nur VirtualBox-4.1.6_OSE.orig/src/VBox/Frontends/VirtualBox/src/main.cpp VirtualBox-4.1.6_OSE/src/VBox/Frontends/VirtualBox/src/main.cpp
--- VirtualBox-4.1.6_OSE.orig/src/VBox/Frontends/VirtualBox/src/main.cpp	2011-11-04 17:22:13.636083176 +0000
+++ VirtualBox-4.1.6_OSE/src/VBox/Frontends/VirtualBox/src/main.cpp	2011-11-04 17:48:58.915541368 +0000
@@ -75,10 +75,9 @@
   "The VirtualBox Linux kernel driver (vboxdrv) is either not loaded or "
   "there is a permission problem with /dev/vboxdrv. Please reinstall the kernel "
   "module by executing<br/><br/>"
-  "  <font color=blue>'/etc/init.d/vboxdrv setup'</font><br/><br/>"
-  "as root. If it is available in your distribution, you should install the "
-  "DKMS package first. This package keeps track of Linux kernel changes and "
-  "recompiles the vboxdrv kernel module if necessary."
+  "  <font color=blue>'pacman -S virtualbox-modules'</font><br/><br/>"
+  "as root. If you don't use our stock kernel, install virtualbox-source and"
+  "execute /usr/bin/vboxbuild ."
   );
 
 QString g_QStrHintOtherWrongDriverVersion = QApplication::tr(
@@ -90,11 +89,8 @@
 
 QString g_QStrHintLinuxWrongDriverVersion = QApplication::tr(
   "The VirtualBox kernel modules do not match this version of "
-  "VirtualBox. The installation of VirtualBox was apparently not "
-  "successful. Executing<br/><br/>"
-  "  <font color=blue>'/etc/init.d/vboxdrv setup'</font><br/><br/>"
-  "may correct this. Make sure that you do not mix the "
-  "OSE version and the PUEL version of VirtualBox."
+  "VirtualBox. Reload the modules or if you don't use our stock kernel execute<br/><br/> "
+  "  <font color=blue>'/usr/bin/vboxbuild'</font><br/><br/>"
   );
 
 QString g_QStrHintOtherNoDriver = QApplication::tr(
diff -Nur VirtualBox-4.1.6_OSE.orig/src/VBox/Installer/linux/VBox.sh VirtualBox-4.1.6_OSE/src/VBox/Installer/linux/VBox.sh
--- VirtualBox-4.1.6_OSE.orig/src/VBox/Installer/linux/VBox.sh	2011-11-04 17:22:13.549415814 +0000
+++ VirtualBox-4.1.6_OSE/src/VBox/Installer/linux/VBox.sh	2011-11-04 17:56:50.735874087 +0000
@@ -32,17 +32,18 @@
     cat << EOF
 WARNING: The vboxdrv kernel module is not loaded. Either there is no module
          available for the current kernel (`uname -r`) or it failed to
-         load. Please recompile the kernel module and install it by
+         load. Please reinstall the kernel module virtualbox-modules or
+         if you don't use our stock kernel compile the modules with
 
-           sudo /etc/init.d/vboxdrv setup
+           sudo /usr/bin/vboxbuild
 
          You will not be able to start VMs until this problem is fixed.
 EOF
 elif [ ! -c /dev/vboxdrv ]; then
     cat << EOF
-WARNING: The character device /dev/vboxdrv does not exist. Try
+WARNING: The character device /dev/vboxdrv does not exist. Try to load the module
 
-           sudo /etc/init.d/vboxdrv restart
+           sudo modprobe vboxdrv
 
          and if that is not successful, try to re-install the package.
 
