--- CMakeLists.txt 2015-11-26 03:24:35.605626000 +0200
+++ CMakeLists.txt  2015-11-26 03:26:36.134042424 +0200
@@ -228,12 +228,6 @@
     set(CURL_LIBRARY "")
     add_definitions(-DLIBCURL_RUNTIME)

-    set(JPEG_LIBRARIES "")
-    add_definitions(-DLIBJPEG_RUNTIME)
-
-    set(PNG_LIBRARIES "")
-    add_definitions(-DLIBPNG_RUNTIME)
-
     set(OGG_LIBRARY "")
     add_definitions(-DOGGLIB_RUNTIME)

