diff -ur a/libxnoise/BackgroundWorker/xnoise-worker.vala b/libxnoise/BackgroundWorker/xnoise-worker.vala
--- a/libxnoise/BackgroundWorker/xnoise-worker.vala
+++ b/libxnoise/BackgroundWorker/xnoise-worker.vala
@@ -65,7 +65,7 @@
     
     
     public bool is_same_thread() {
-        return (void*)Thread.self<int>() == (void*)_thread;
+        return Thread.self<void*>() == (void*)_thread;
     }
     
     public enum Priority {
diff -ur a/libxnoise/Parameter/xnoise-parameter.vala b/libxnoise/Parameter/xnoise-parameter.vala
--- a/libxnoise/Parameter/xnoise-parameter.vala
+++ b/libxnoise/Parameter/xnoise-parameter.vala
@@ -37,7 +37,7 @@
     private static const string settings_int    = "settings_int";
     private static const string settings_double = "settings_double";
     private static const string settings_string = "settings_string";
-    private static HashTable<string,int>     ht_int;
+    private static HashTable<string,int?>    ht_int;
     private static HashTable<string,double?> ht_double;
     private static HashTable<string,string>  ht_string;
     private static bool inited = false;
@@ -50,7 +50,7 @@
         if(inited == true)
             return;
         IParams_implementers = new GLib.List<IParams>();
-        ht_int    = new GLib.HashTable<string,int>(str_hash, str_equal);
+        ht_int    = new GLib.HashTable<string,int?>(str_hash, str_equal);
         ht_double = new GLib.HashTable<string,double?>(str_hash, str_equal);
         ht_string = new GLib.HashTable<string,string>(str_hash, str_equal);
         read_all_parameters_from_file();
@@ -133,16 +133,16 @@
     //  GETTERS FOR THE HASH TABLE
     //Type bool
     public static bool get_bool_value(string key) {
-        int val = ht_int.lookup(key);
-        if(val!=0)
+        int? val = ht_int.lookup(key);
+        if(val!=null)
             return true;
         else
             return false;
     }
     //Type int
     public static int get_int_value(string key) {
-        int val = ht_int.lookup(key);
-        if(val!=0)
+        int? val = ht_int.lookup(key);
+        if(val!=null)
             return val;
         else
             return 0;
diff -ur a/libxnoise/Utils/xnoise-utilities.vala b/libxnoise/Utils/xnoise-utilities.vala
--- a/libxnoise/Utils/xnoise-utilities.vala
+++ b/libxnoise/Utils/xnoise-utilities.vala
@@ -253,7 +253,8 @@
         
         try {
             GLib.Regex r = new GLib.Regex("\n");
-            return (owned)r.replace(val, -1, 0, " ");
+            string result = r.replace(val, -1, 0, " ");
+            return (owned)result;
         }
         catch(GLib.RegexError e) {
             print("%s\n", e.message);
@@ -312,7 +313,8 @@
     public static string replace_underline_with_blank_encoded(string value) {
         try {
             GLib.Regex r = new GLib.Regex("_");
-            return (owned)r.replace(value, -1, 0, "%20");
+            string result = r.replace(value, -1, 0, "%20");
+            return (owned)result;
         }
         catch(GLib.RegexError e) {
             print("%s\n", e.message);
