--- a/src/bblogger.c
+++ b/src/bblogger.c
@@ -144,7 +144,11 @@
   /* Error lines are errors. */
   if (strncmp(string, "(EE)", 4) == 0){
     if (strstr(string, "Failed to load module \"kbd\"") ||
-            strstr(string, "No input driver matching")) {
+            strstr(string, "Failed to load module \"mouse\"") ||
+            strstr(string, "failed to set DRM interface version 1.4:") ||
+            strstr(string, "No input driver matching") ||
+            strstr(string, "systemd-logind: failed to get session:") ||
+            strstr(string, "Server terminated successfully")) {
       /* non-fatal errors */
       prio = LOG_DEBUG;
     } else {
