diff --git a/tests/test_latex.py b/tests/test_latex.py
index bcef06f..175b3ba 100644
--- a/tests/test_latex.py
+++ b/tests/test_latex.py
@@ -8,7 +8,7 @@ import unittest
 
 CR = "\r?\n"
 
-blockdiag_fontpath = '/usr/share/fonts/truetype/ipafont/ipagp.ttf'
+blockdiag_fontpath = '/usr/share/fonts/OTF/ipagp.ttf'
 with_png_app = with_app(srcdir='tests/docs/basic',
                         buildername='latex',
                         write_docstring=True,
