diff -u -r ndiswrapper-1.62-5.4/driver/lin2win.S ndiswrapper-1.62-5.5/driver/lin2win.S
--- ndiswrapper-1.62-5.4/driver/lin2win.S	2019-02-11 04:11:14.000000000 +0000
+++ ndiswrapper-1.62-5.5/driver/lin2win.S	2020-01-27 22:33:17.363779223 +0000
@@ -14,6 +14,7 @@
  */
 
 #include <linux/linkage.h>
+#include <linux/version.h>
 
 	.text
 
@@ -71,8 +72,12 @@
  * We get (ARGC + 1) arguments.
  */
 .macro lin2win name, argc
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,5,0)
 	.type \name, @function
 	ENTRY(\name)
+#else
+	SYM_FUNC_START(\name)
+#endif
 
 	/* Create a call frame - it's optional, but good for debugging */
 	.cfi_startproc
@@ -125,7 +130,11 @@
 	.cfi_restore %rbp
 	ret
 	.cfi_endproc
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,5,0)
 	.size	\name, (. - \name)
+#else
+	SYM_FUNC_END(\name)
+#endif
 .endm
 
 /* Define lin2winN functions */
diff -u -r ndiswrapper-1.62-5.4/driver/win2lin_stubs.S ndiswrapper-1.62-5.5/driver/win2lin_stubs.S
--- ndiswrapper-1.62-5.4/driver/win2lin_stubs.S	2019-02-11 04:11:14.000000000 +0000
+++ ndiswrapper-1.62-5.5/driver/win2lin_stubs.S	2020-01-27 22:33:20.867163026 +0000
@@ -14,6 +14,7 @@
  */
 
 #include <linux/linkage.h>
+#include <linux/version.h>
 
 #ifdef CONFIG_X86_64
 
@@ -141,8 +142,12 @@
 
 /* Declare function LONGNAME, call function SHORTNAME with ARGC arguments */
 .macro win2linm longname, shortname, argc
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,5,0)
 	.type \longname, @function
 	ENTRY(\longname)
+#else
+	SYM_FUNC_START(\longname)
+#endif
 
 	/* Create a call frame - it's optional, but good for debugging */
 	.cfi_startproc
@@ -243,7 +248,11 @@
 	.cfi_restore %rbp
 	ret
 	.cfi_endproc
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,5,0)
 	.size \longname, (. - \longname)
+#else
+	SYM_FUNC_END(\longname)
+#endif
 .endm
 
 #define win2lin(name, argc) win2linm win2lin_ ## name ## _ ## argc, name, argc
