--- a/makefile
+++ b/makefile
@@ -364,109 +364,111 @@ all:	lout prg2lout
 install: lout prg2lout
 	@echo ""
 	@echo "(a) Installing lout and prg2lout binaries into BINDIR $(BINDIR)"
-	cp lout $(BINDIR)/lout
-	chmod 755 $(BINDIR)/lout
-	cp prg2lout $(BINDIR)/prg2lout
-	chmod 755 $(BINDIR)/prg2lout
+	if [ ! -d $(DESTDIR)$(BINDIR) ] ; then $(MKDIR) $(DESTDIR)$(BINDIR) ; fi
+	chmod 755 $(DESTDIR)$(BINDIR)
+	cp lout $(DESTDIR)$(BINDIR)/lout
+	chmod 755 $(DESTDIR)$(BINDIR)/lout
+	cp prg2lout $(DESTDIR)$(BINDIR)/prg2lout
+	chmod 755 $(DESTDIR)$(BINDIR)/prg2lout
 	@echo ""
 	@echo "(b) Installing library files into LOUTLIBDIR $(LOUTLIBDIR)"
-	$(MKDIR) $(LOUTLIBDIR)
-	chmod 755 $(LOUTLIBDIR)
+	$(MKDIR) $(DESTDIR)$(LOUTLIBDIR)
+	chmod 755 $(DESTDIR)$(LOUTLIBDIR)
 	@echo ""
-	$(MKDIR) $(LOUTLIBDIR)/$(LIBINCL)
-	chmod 755 $(LOUTLIBDIR)/$(LIBINCL)
-	cp include/* $(LOUTLIBDIR)/$(LIBINCL)
-	chmod 644 $(LOUTLIBDIR)/$(LIBINCL)/*
+	$(MKDIR) $(DESTDIR)$(LOUTLIBDIR)/$(LIBINCL)
+	chmod 755 $(DESTDIR)$(LOUTLIBDIR)/$(LIBINCL)
+	cp include/* $(DESTDIR)$(LOUTLIBDIR)/$(LIBINCL)
+	chmod 644 $(DESTDIR)$(LOUTLIBDIR)/$(LIBINCL)/*
 	@echo ""
-	$(MKDIR) $(LOUTLIBDIR)/$(LIBDATA)
-	chmod 755 $(LOUTLIBDIR)/$(LIBDATA)
-	cp data/* $(LOUTLIBDIR)/$(LIBDATA)
-	chmod 644 $(LOUTLIBDIR)/$(LIBDATA)/*
+	$(MKDIR) $(DESTDIR)$(LOUTLIBDIR)/$(LIBDATA)
+	chmod 755 $(DESTDIR)$(LOUTLIBDIR)/$(LIBDATA)
+	cp data/* $(DESTDIR)$(LOUTLIBDIR)/$(LIBDATA)
+	chmod 644 $(DESTDIR)$(LOUTLIBDIR)/$(LIBDATA)/*
 	@echo ""
-	$(MKDIR) $(LOUTLIBDIR)/$(LIBHYPH)
-	chmod 755 $(LOUTLIBDIR)/$(LIBHYPH)
-	cp hyph/* $(LOUTLIBDIR)/$(LIBHYPH)
-	chmod 644 $(LOUTLIBDIR)/$(LIBHYPH)/*
+	$(MKDIR) $(DESTDIR)$(LOUTLIBDIR)/$(LIBHYPH)
+	chmod 755 $(DESTDIR)$(LOUTLIBDIR)/$(LIBHYPH)
+	cp hyph/* $(DESTDIR)$(LOUTLIBDIR)/$(LIBHYPH)
+	chmod 644 $(DESTDIR)$(LOUTLIBDIR)/$(LIBHYPH)/*
 	@echo ""
-	$(MKDIR) $(LOUTLIBDIR)/$(LIBFONT)
-	chmod 755 $(LOUTLIBDIR)/$(LIBFONT)
-	cp font/* $(LOUTLIBDIR)/$(LIBFONT)
-	chmod 644 $(LOUTLIBDIR)/$(LIBFONT)/*
+	$(MKDIR) $(DESTDIR)$(LOUTLIBDIR)/$(LIBFONT)
+	chmod 755 $(DESTDIR)$(LOUTLIBDIR)/$(LIBFONT)
+	cp font/* $(DESTDIR)$(LOUTLIBDIR)/$(LIBFONT)
+	chmod 644 $(DESTDIR)$(LOUTLIBDIR)/$(LIBFONT)/*
 	@echo ""
-	$(MKDIR) $(LOUTLIBDIR)/$(LIBMAPS)
-	chmod 755 $(LOUTLIBDIR)/$(LIBMAPS)
-	cp maps/* $(LOUTLIBDIR)/$(LIBMAPS)
-	chmod 644 $(LOUTLIBDIR)/$(LIBMAPS)/*
+	$(MKDIR) $(DESTDIR)$(LOUTLIBDIR)/$(LIBMAPS)
+	chmod 755 $(DESTDIR)$(LOUTLIBDIR)/$(LIBMAPS)
+	cp maps/* $(DESTDIR)$(LOUTLIBDIR)/$(LIBMAPS)
+	chmod 644 $(DESTDIR)$(LOUTLIBDIR)/$(LIBMAPS)/*
 	@echo ""
-	$(MKDIR) $(LOUTLIBDIR)/$(LIBLOCA)
-	chmod 755 $(LOUTLIBDIR)/$(LIBLOCA)
+	$(MKDIR) $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)
+	chmod 755 $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)
 	@echo ""
 	@echo "(c) Initializing run (should be silent, no errors expected)"
-	$(BINDIR)/lout -x -s $(LOUTLIBDIR)/$(LIBINCL)/init -I $(LOUTLIBDIR)/$(LIBINCL) -H $(LOUTLIBDIR)/$(LIBHYPH) -D $(LOUTLIBDIR)/$(LIBDATA)
+	$(DESTDIR)$(BINDIR)/lout -x -s $(DESTDIR)$(LOUTLIBDIR)/$(LIBINCL)/init -I $(DESTDIR)$(LOUTLIBDIR)/$(LIBINCL) -H $(DESTDIR)$(LOUTLIBDIR)/$(LIBHYPH) -D $(DESTDIR)$(LOUTLIBDIR)/$(LIBDATA)
 	@echo ""
 	@echo "(d) Changing mode of files just created by initializing run"
-	chmod 644 $(LOUTLIBDIR)/$(LIBDATA)/*
-	chmod 644 $(LOUTLIBDIR)/$(LIBHYPH)/*
+	chmod 644 $(DESTDIR)$(LOUTLIBDIR)/$(LIBDATA)/*
+	chmod 644 $(DESTDIR)$(LOUTLIBDIR)/$(LIBHYPH)/*
 
 installman:
 	@echo ""
 	@echo "Installing manual entries into MANDIR $(MANDIR)"
-	if [ ! -d $(MANDIR) ] ; then $(MKDIR) $(MANDIR) ; fi
-	chmod 755 $(MANDIR)
+	if [ ! -d $(DESTDIR)$(MANDIR) ] ; then $(MKDIR) $(DESTDIR)$(MANDIR) ; fi
+	chmod 755 $(DESTDIR)$(MANDIR)
 	sed -e "s@<BINDIR>@$(BINDIR)@" -e "s@<LIBDIR>@$(LOUTLIBDIR)@"	\
 	    -e "s@<LOUTDOCDIR>@$(LOUTDOCDIR)@" -e "s@<MANDIR>@$(MANDIR)@"	\
-	man/lout.1 > $(MANDIR)/lout.1
-	chmod 644 $(MANDIR)/lout.1
-	cp man/prg2lout.1 $(MANDIR)/prg2lout.1
-	chmod 644 $(MANDIR)/prg2lout.1
+	man/lout.1 > $(DESTDIR)$(MANDIR)/lout.1
+	chmod 644 $(DESTDIR)$(MANDIR)/lout.1
+	cp man/prg2lout.1 $(DESTDIR)$(MANDIR)/prg2lout.1
+	chmod 644 $(DESTDIR)$(MANDIR)/prg2lout.1
 
 installdoc:
 	@echo ""
 	@echo "Creating LOUTDOCDIR $(LOUTDOCDIR) and copying documentation into it"
-	if [ ! -d $(LOUTDOCDIR) ] ; then $(MKDIR) $(LOUTDOCDIR) ; fi
-	$(RCOPY) doc/* $(LOUTDOCDIR)
-	chmod 755 $(LOUTDOCDIR)
-	chmod 755 $(LOUTDOCDIR)/*
-	chmod 644 $(LOUTDOCDIR)/*/*
+	if [ ! -d $(DESTDIR)$(LOUTDOCDIR) ] ; then $(MKDIR) $(DESTDIR)$(LOUTDOCDIR) ; fi
+	$(RCOPY) doc/* $(DESTDIR)$(LOUTDOCDIR)
+	chmod 755 $(DESTDIR)$(LOUTDOCDIR)
+	chmod 755 $(DESTDIR)$(LOUTDOCDIR)/*
+	chmod 644 $(DESTDIR)$(LOUTDOCDIR)/*/*
 
 allinstall:	install installman installdoc
 
 installfr:
 	@echo ""
 	@echo "Putting French error messages into $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)"
-	$(MKDIR) $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)
-	chmod 755 $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)
-	$(MKDIR) $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES
-	chmod 755 $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES
-	cp locale/msgs.fr $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES/msgs.$(LOC_FR)
-	gencat $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES/errors.$(LOC_FR)	\
-	       $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES/msgs.$(LOC_FR)
-	chmod 644 $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES/*
+	$(MKDIR) $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)
+	chmod 755 $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)
+	$(MKDIR) $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES
+	chmod 755 $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES
+	cp locale/msgs.fr $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES/msgs.$(LOC_FR)
+	gencat $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES/errors.$(LOC_FR)	\
+	       $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES/msgs.$(LOC_FR)
+	chmod 644 $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_FR)/LC_MESSAGES/*
 
 installde:
 	@echo ""
 	@echo "Putting German error messages into $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)"
-	$(MKDIR) $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)
-	chmod 755 $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)
-	$(MKDIR) $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES
-	chmod 755 $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES
-	cp locale/msgs.de $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES/msgs.$(LOC_DE)
-	gencat $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES/errors.$(LOC_DE)	\
-	       $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES/msgs.$(LOC_DE)
-	chmod 644 $(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES/*
+	$(MKDIR) $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)
+	chmod 755 $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)
+	$(MKDIR) $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES
+	chmod 755 $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES
+	cp locale/msgs.de $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES/msgs.$(LOC_DE)
+	gencat $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES/errors.$(LOC_DE)	\
+	       $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES/msgs.$(LOC_DE)
+	chmod 644 $(DESTDIR)$(LOUTLIBDIR)/$(LIBLOCA)/$(LOC_DE)/LC_MESSAGES/*
 
 uninstall:
-	-rm -f "$(BINDIR)/lout" "$(BINDIR)/prg2lout"
-	-rm -f "$(MANDIR)/lout.1" "$(MANDIR)/prg2lout.1"
+	-rm -f "$(DESTDIR)$(BINDIR)/lout" "$(DESTDIR)$(BINDIR)/prg2lout"
+	-rm -f "$(DESTDIR)$(MANDIR)/lout.1" "$(DESTDIR)$(MANDIR)/prg2lout.1"
 	{ \
-		for dir in "$(LOUTLIBDIR)" "$(LOUTDOCDIR)" ; do \
+		for dir in "$(DESTDIR)$(LOUTLIBDIR)" "$(DESTDIR)$(LOUTDOCDIR)" ; do \
 			if [ -n "$$dir" ] && [ -d "$$dir" ] ; then \
 				case "$$dir" in \
 				*lout*) rm -fr "$$dir" ;; \
 				esac ; \
 			fi ; \
 		done ; \
-		for dir in "$(MANDIR)" "$(LIBDIR)" ; do \
+		for dir in "$(DESTDIR)$(MANDIR)" "$(DESTDIR)$(LIBDIR)" ; do \
 			if [ -n "$$dir" ] && [ -d "$$dir" ] ; then \
 				case "$$dir" in \
 				*lout*) rmdir "$$dir" ;; \

