diff -ruNa a/migrate/versioning/schema.py b/migrate/versioning/schema.py
--- a/migrate/versioning/schema.py
+++ b/migrate/versioning/schema.py
@@ -7,7 +7,7 @@
 from sqlalchemy import (Table, Column, MetaData, String, Text, Integer,
     create_engine)
 from sqlalchemy.sql import and_
-from sqlalchemy import exceptions as sa_exceptions
+from sqlalchemy import exc as sa_exceptions
 from sqlalchemy.sql import bindparam
 
 from migrate import exceptions
