From 3423639d000f0e9ef9d01f4b91742dc4acf8d270 Mon Sep 17 00:00:00 2001
From: Keno Fischer <keno@juliacomputing.com>
Date: Mon, 25 Apr 2022 22:49:30 -0400
Subject: [PATCH] Header compat for LLVM15 (#45089)

---
 src/disasm.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/disasm.cpp b/src/disasm.cpp
index fa2e3e819a0b..4d21f49e4af2 100644
--- a/src/disasm.cpp
+++ b/src/disasm.cpp
@@ -60,6 +60,7 @@
 // for outputting disassembly
 #include <llvm/ADT/Triple.h>
 #include <llvm/AsmParser/Parser.h>
+#include <llvm/Analysis/TargetTransformInfo.h>
 #include <llvm/BinaryFormat/COFF.h>
 #include <llvm/BinaryFormat/MachO.h>
 #include <llvm/DebugInfo/DIContext.h>
