diff --git a/configure.ac b/configure.ac
index be832f7..485c574 100644
--- a/configure.ac
+++ b/configure.ac
@@ -155,10 +155,11 @@ dnl Gettext stuff.
 dnl ================================================================
 
 IT_PROG_INTLTOOL([0.40.0])
-AC_SUBST([GETTEXT_PACKAGE], [pluma])
+GETTEXT_PACKAGE=pluma
+AC_SUBST([GETTEXT_PACKAGE])
 AM_GNU_GETTEXT_VERSION([0.17])
 AM_GLIB_GNU_GETTEXT
-AC_DEFINE([GETTEXT_PACKAGE],[PACKAGE_TARNAME], [Define to the gettext package name.])
+AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Define to the Gettext package name])
 
 dnl ================================================================
 dnl Use gvfs metadata or the old xml file store.
