diff --git a/widgets/ccheckbutton.h b/widgets/ccheckbutton.h
index c219285..0273c59 100644
--- a/widgets/ccheckbutton.h
+++ b/widgets/ccheckbutton.h
@@ -21,6 +21,7 @@
 
 #include <QMap>
 #include <DToolButton>
+#include <DWidget>
 
 DWIDGET_USE_NAMESPACE
 
diff --git a/widgets/cclickbutton.h b/widgets/cclickbutton.h
index 7da95b4..d1f0eed 100644
--- a/widgets/cclickbutton.h
+++ b/widgets/cclickbutton.h
@@ -20,6 +20,7 @@
 #define CCLICKBUTTON_H
 
 #include <DToolButton>
+#include <DWidget>
 
 DWIDGET_USE_NAMESPACE
 
