From: Jakub Wilk <jwilk@debian.org>
Subject: fix typo in sourcecode
Origin: vendor, http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=598021
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=598021

---
 src/fuseiso.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- fuseiso.orig/src/fuseiso.c
+++ fuseiso/src/fuseiso.c
@@ -295,7 +295,7 @@ void usage(const char* prog) {
         "    -f                 -- run in foreground, do not daemonize\n"
         "    -d                 -- run in foreground and print debug information\n"
         "    -s                 -- run single-threaded\n"
-        "\nPlease consult with FUSE ducumentation for more information\n",
+        "\nPlease consult with FUSE documentation for more information\n",
         VERSION, 
         prog);
 };
