diff -wbBur Umlet/umlet.sh Umlet.my/umlet.sh
--- Umlet/umlet.sh	2011-02-18 18:27:14.000000000 +0000
+++ Umlet.my/umlet.sh	2011-03-03 15:20:48.000000000 +0000
@@ -2,11 +2,7 @@
 
 # Shell script for running umlet from the command prompt in linux
 
-# If you want to put umlet.sh in your home bin directory ($HOME/bin/) to start it from anywhere with
-#    $ umlet.sh myDiagram.uxf
-# you must specify the programDir directly instead
-#programDir=/path/to/umlet
-programDir=$(cd $(dirname $0);pwd)
+programDir=/usr/share/umlet
 
 if [ $# -gt 0 ]
  then java -jar ${programDir}/umlet.jar -filename="$1"
