diff -wbBur UMLet.org//umlet.sh UMLet/umlet.sh
--- UMLet.org//umlet.sh	2010-03-01 20:20:21.000000000 +0000
+++ UMLet/umlet.sh	2010-03-01 20:20:44.000000000 +0000
@@ -17,6 +17,7 @@
 # Now you can use umlet from the command line from any directory.
 #    $ umlet.sh myDiagram.uxf 
 
+cd /usr/share/umlet
 currentDir=`pwd`
 
 # uncomment this to be able to start umlet from anywhere (UMLET HOME has to be set)
