diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2021-07-01 20:32:19.000000000 +0200
+++ b/CMakeLists.txt	2021-10-10 00:56:11.485842867 +0200
@@ -14,7 +14,6 @@
 
 
 add_executable(tuning-library-tests)
-target_include_directories(tuning-library-tests PRIVATE libs/catch2)
 target_link_libraries(tuning-library-tests PRIVATE ${PROJECT_NAME})
 target_sources(tuning-library-tests PRIVATE
         tests/alltests.cpp)
diff -ruN a/tests/alltests.cpp b/tests/alltests.cpp
--- a/tests/alltests.cpp	2021-07-01 20:32:19.000000000 +0200
+++ b/tests/alltests.cpp	2021-10-10 00:57:22.469112753 +0200
@@ -1,5 +1,5 @@
 #define CATCH_CONFIG_RUNNER
-#include "catch2.hpp"
+#include <catch2/catch.hpp>
 
 #include "Tunings.h"
 #include <iomanip>
