diff -Naur acerhk-0.5.35/Makefile acerhk-0.5.35-new/Makefile
--- acerhk-0.5.35/Makefile	2006-08-16 11:11:08.000000000 -0500
+++ acerhk-0.5.35-new/Makefile	2010-04-09 15:59:40.000000000 -0500
@@ -3,7 +3,7 @@
 #KERNELSRC=/usr/src/linux
 KERNELSRC?=/lib/modules/`uname -r`/build
 # Starting with 2.6.18, the kernel version is in utsrelease.h instead of version.h, accomodate both cases
-KERNELVERSION=$(shell awk -F\" '/REL/ {print $$2}' $(shell grep -s -l REL $(KERNELSRC)/include/linux/version.h $(KERNELSRC)/include/linux/utsrelease.h))
+KERNELVERSION=$(shell awk -F\" '/REL/ {print $$2}' $(shell grep -s -l REL $(KERNELSRC)/include/linux/version.h $(KERNELSRC)/include/generated/utsrelease.h))
 KERNELMAJOR=$(shell echo $(KERNELVERSION)|head -c3)
 
 # next line is for kernel 2.6, if you integrate the driver in the kernel tree
