--- configure	2011-08-05 13:04:41.000000000 +0200
+++ configure.new	2011-08-17 14:02:07.303473086 +0200
@@ -19120,7 +19120,7 @@
 TRACKER_NAUTILUS_EXTENSION_REQUIRED="glib-2.0       >= $GLIB_REQUIRED
                                      gio-unix-2.0   >= $GLIB_REQUIRED
                                      gthread-2.0    >= $GLIB_REQUIRED
-                                     gtk+-2.0       >= $GTK_REQUIRED
+                                     gtk+-3.0       >= $GTK_REQUIRED
                                      libnautilus-extension"
 
 
