--- nomacs-3.8.0/ImageLounge/CMakeLists.txt.orig	2017-11-14 08:43:20.157769041 +0000
+++ nomacs-3.8.0/ImageLounge/CMakeLists.txt	2017-11-14 08:43:49.094497479 +0000
@@ -22,12 +22,6 @@
 set(NOMACS_VERSION ${NOMACS_VERSION_MAJOR}.${NOMACS_VERSION_MINOR})
 set(NOMACS_FULL_VERSION ${NOMACS_VERSION}.${NOMACS_VERSION_PATCH})
 
-# Codecov
-if(CMAKE_COMPILER_IS_GNUCXX)
-    include("cmake/CodeCoverage.cmake")
-    setup_target_for_coverage(${PROJECT_NAME}_coverage tests coverage)
-endif()
-
 if(CMAKE_CL_64)
 	SET(NMC_ARCHITECTURE "x64")
 else()
