diff -urN ppsspp-0.9.9.1/Core/HW/SimpleAudioDec.cpp ppsspp-0.9.9.1-patched/Core/HW/SimpleAudioDec.cpp
--- ppsspp-0.9.9.1/Core/HW/SimpleAudioDec.cpp	2014-08-19 07:35:25.000000000 +1100
+++ ppsspp-0.9.9.1-patched/Core/HW/SimpleAudioDec.cpp	2014-11-02 20:37:58.000000000 +1000
@@ -94,6 +94,8 @@
 	}
 	codecCtx_->channels = channels_;
 	codecCtx_->channel_layout = channels_ == 2 ? AV_CH_LAYOUT_STEREO : AV_CH_LAYOUT_MONO;
+	if (!codecCtx_->block_align)
+		codecCtx_->block_align = sample_rate_;
 	codecCtx_->sample_rate = sample_rate_;
 	OpenCodec();
 #endif  // USE_FFMPEG
