diff -wbBur xfmedia-0.9.2/src/mainwin.c xfmedia-0.9.2.my/src/mainwin.c
--- xfmedia-0.9.2/src/mainwin.c	2006-11-27 07:21:23.000000000 +0000
+++ xfmedia-0.9.2.my/src/mainwin.c	2010-11-01 13:01:17.000000000 +0000
@@ -1865,6 +1865,10 @@
     gboolean use_playlist_q  = xfmedia_playlist_queue_n_entries(mwin->playlist_q);
     gboolean ret = FALSE;
     
+    if(xfmedia_playlist_get_n_entries(mwin->plist) == 0)
+//    if(xfmedia_playlist_queue_is_empty(mwin->plist))
+	return;
+
     if(mwin->cur_playing
        && xfmedia_settings_get_bool("/xfmedia/playlist/shuffle")
        && !use_playlist_q )
@@ -1899,6 +1903,10 @@
 {
     gint tot_items, idx = -1;
     
+    if(xfmedia_playlist_get_n_entries(mwin->plist) == 0)
+//    if(xfmedia_playlist_queue_is_empty(mwin->plist))
+	return;
+
     if(xfmedia_settings_get_bool("/xfmedia/playlist/shuffle")) {
         XfmediaPlaylistEntryRef *ref = NULL;
         gboolean playing = FALSE;
