diff -wbBur xfmedia-0.9.2/src/xfmedia-xine.c xfmedia-0.9.2.my/src/xfmedia-xine.c
--- xfmedia-0.9.2/src/xfmedia-xine.c	2006-11-27 10:21:23.000000000 +0300
+++ xfmedia-0.9.2.my/src/xfmedia-xine.c	2012-01-05 01:23:21.000000000 +0400
@@ -1450,17 +1450,6 @@
 }
 
 gboolean
-xfmedia_xine_trick_mode(XfmediaXine * xfx, gint mode, gint value)
-{
-
-    g_return_val_if_fail(xfx != NULL, 0);
-    g_return_val_if_fail(XFMEDIA_IS_XINE(xfx), 0);
-    g_return_val_if_fail(xfx->priv->stream != NULL, 0);
-
-    return xine_trick_mode(xfx->priv->stream, mode, value);
-}
-
-gboolean
 xfmedia_xine_get_pos_length(XfmediaXine *xfx, gint *pos_stream,
     gint *pos_time, gint *length_time)
 {
diff -wbBur xfmedia-0.9.2/src/xfmedia-xine.h xfmedia-0.9.2.my/src/xfmedia-xine.h
--- xfmedia-0.9.2/src/xfmedia-xine.h	2006-11-27 10:21:23.000000000 +0300
+++ xfmedia-0.9.2.my/src/xfmedia-xine.h	2012-01-05 01:20:47.000000000 +0400
@@ -87,9 +87,6 @@
 gboolean     xfmedia_xine_play                       (XfmediaXine *xfx,
                                                   gint pos,
                                                   gint start_time);
-gboolean     xfmedia_xine_trick_mode                 (XfmediaXine *xfx,
-                                                  gint mode,
-                                                  gint value);
 gint         xfmedia_xine_get_stream_info            (XfmediaXine *xfx,
                                                   gint info);
 G_CONST_RETURN gchar *xfmedia_xine_get_meta_info     (XfmediaXine *xfx,
