--- src/ogmrip-progress-dialog.c
+++ src/ogmrip-progress-dialog.c
@@ -32,6 +32,9 @@
 
 #ifdef HAVE_LIBNOTIFY_SUPPORT
 #include <libnotify/notify.h>
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
 #endif /* HAVE_LIBNOTIFY_SUPPORT */
 
 #define OGMRIP_PROGRESS_DIALOG_GET_PRIVATE(o) \
@@ -418,8 +421,12 @@
   g_signal_connect_swapped (dialog->priv->status_icon, "popup_menu",
       G_CALLBACK (ogmrip_progress_dialog_status_icon_popup_menu), dialog);
 
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+  /* notify_notification_new_with_status_icon was removed */
+#else
   dialog->priv->notification = notify_notification_new_with_status_icon ("Dummy", "Dummy",
       OGMRIP_DATA_DIR G_DIR_SEPARATOR_S OGMRIP_ICON_FILE, dialog->priv->status_icon);
+#endif
   g_signal_connect (dialog, "window-state-event",
       G_CALLBACK (ogmrip_progress_dialog_state_changed), NULL);
 
