diff -u src/third_party/protobuf/3.7.0/BUILD src/third_party/protobuf/3.7.0/BUILD
--- src/third_party/protobuf/3.7.0/BUILD	2019-03-03 01:22:00.483246330 +0000
+++ src/third_party/protobuf/3.7.0/BUILD	2019-03-02 21:16:11.539170067 +0000
@@ -16,6 +16,57 @@
 )
 
 ################################################################################
+# Modifications made by Bazel
+################################################################################
+
+load(":protobuf.bzl", "py_proto_library")
+load(":compiler_config_setting.bzl", "create_compiler_config_setting")
+
+filegroup(
+    name = "srcs",
+    srcs = glob(
+        [
+            "**",
+            "protobuf.bzl",
+            "python/google/protobuf/**/*.py",
+            "src/google/protobuf/**/*.proto",
+        ],
+    ) + [
+        "//third_party/protobuf/3.7.0/examples:srcs",
+        "//third_party/protobuf/3.7.0/util/python:srcs",
+        "//third_party/protobuf/3.7.0/third_party/googletest:srcs",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+java_import(
+    name = "protobuf_java_util",
+    jars = ["libprotobuf_java_util.jar"],
+    visibility = ["//visibility:public"],
+    exports = [
+        "@io_bazel//third_party:gson",
+    ],
+)
+
+java_import(
+    name = "protobuf_java",
+    jars = ["libprotobuf_java.jar"],
+    visibility = ["//visibility:public"],
+)
+
+# For bootstrapping JavaBuilder
+filegroup(
+    name = "protobuf-jars",
+    srcs = ["libprotobuf_java.jar"],
+    visibility = ["//visibility:public"],
+)
+
+create_compiler_config_setting(
+    name = "msvc",
+    value = "msvc-cl",
+)
+
+################################################################################
 # ZLIB configuration
 ################################################################################
 
@@ -109,10 +160,6 @@
     ],
 })
 
-load(":compiler_config_setting.bzl", "create_compiler_config_setting")
-
-create_compiler_config_setting(name = "msvc", value = "msvc-cl")
-
 config_setting(
     name = "android",
     values = {
@@ -173,7 +220,7 @@
 
 PROTOBUF_DEPS = select({
     ":msvc": [],
-    "//conditions:default": ZLIB_DEPS,
+    "//conditions:default": [],
 })
 
 cc_library(
@@ -613,39 +660,8 @@
 )
 
 ################################################################################
-# Java support
+# Java support -- deleted as per Bazel instructions
 ################################################################################
-internal_gen_well_known_protos_java(
-    srcs = WELL_KNOWN_PROTOS,
-)
-
-java_library(
-    name = "protobuf_java",
-    srcs = glob([
-        "java/core/src/main/java/com/google/protobuf/*.java",
-    ]) + [
-        ":gen_well_known_protos_java",
-    ],
-    javacopts = select({
-       "//:jdk9": ["--add-modules=jdk.unsupported"],
-       "//conditions:default": ["-source 7", "-target 7"],
-    }),
-    visibility = ["//visibility:public"],
-)
-
-java_library(
-    name = "protobuf_java_util",
-    srcs = glob([
-        "java/util/src/main/java/com/google/protobuf/util/*.java",
-    ]),
-    javacopts = ["-source 7", "-target 7"],
-    visibility = ["//visibility:public"],
-    deps = [
-        "protobuf_java",
-        "//external:gson",
-        "//external:guava",
-    ],
-)
 
 ################################################################################
 # Python support
