diff -wbBur mlt-0.8.2/src/modules/avformat/consumer_avformat.c mlt-0.8.2.my/src/modules/avformat/consumer_avformat.c
--- mlt-0.8.2/src/modules/avformat/consumer_avformat.c	2012-08-28 20:25:58.000000000 +0400
+++ mlt-0.8.2.my/src/modules/avformat/consumer_avformat.c	2012-11-06 16:03:07.269782436 +0400
@@ -237,11 +237,7 @@
 		mlt_properties_set_data( properties, "acodec", codecs, 0, (mlt_destructor) mlt_properties_close, NULL );
 		mlt_properties_set_data( doc, "audio_codecs", codecs, 0, NULL, NULL );
 		while ( ( codec = av_codec_next( codec ) ) )
-#if LIBAVCODEC_VERSION_INT >= ((54<<16)+(0<<8)+0)
-			if ( ( codec->encode || codec->encode2 ) && codec->type == CODEC_TYPE_AUDIO )
-#else
-			if ( codec->encode && codec->type == CODEC_TYPE_AUDIO )
-#endif
+			if ( ( codec->encode2 ) && codec->type == CODEC_TYPE_AUDIO )
 			{
 				snprintf( key, sizeof(key), "%d", mlt_properties_count( codecs ) );
 				mlt_properties_set( codecs, key, codec->name );
@@ -261,11 +257,7 @@
 		mlt_properties_set_data( properties, "vcodec", codecs, 0, (mlt_destructor) mlt_properties_close, NULL );
 		mlt_properties_set_data( doc, "video_codecs", codecs, 0, NULL, NULL );
 		while ( ( codec = av_codec_next( codec ) ) )
-#if LIBAVCODEC_VERSION_INT >= ((54<<16)+(0<<8)+0)
-			if ( (codec->encode || codec->encode2) && codec->type == CODEC_TYPE_VIDEO )
-#else
-			if ( codec->encode && codec->type == CODEC_TYPE_VIDEO )
-#endif
+			if ( (codec->encode2) && codec->type == CODEC_TYPE_VIDEO )
 			{
 				snprintf( key, sizeof(key), "%d", mlt_properties_count( codecs ) );
 				mlt_properties_set( codecs, key, codec->name );
