From 07b5ce0a12f8fadba9bb7941b0874db5493ef034 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 26 May 2017 09:30:59 +0100
Subject: [PATCH] raven: Only deal with a device card when it actually exists

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/raven/sound.vala | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/raven/sound.vala b/src/raven/sound.vala
index becd7128..9c74203b 100644
--- a/src/raven/sound.vala
+++ b/src/raven/sound.vala
@@ -318,6 +318,10 @@ public class SoundWidget : Gtk.Box
         }
         var device = this.mixer.lookup_output_id(id);
 
+        if (device.card == null) {
+            return;
+        }
+
         var card = device.card as Gvc.MixerCard;
         var check = new Gtk.RadioButton.with_label_from_widget(this.output_leader, "%s - %s".printf(device.description, card.name));
         (check.get_child() as Gtk.Label).set_ellipsize(Pango.EllipsizeMode.END);
