diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2021-06-16 19:21:46.000000000 +0200
+++ b/CMakeLists.txt	2021-12-19 16:09:47.936130466 +0100
@@ -45,8 +45,6 @@
   # Headers and license files
   install(DIRECTORY "${PROJECT_SOURCE_DIR}/include/function2"
           DESTINATION "include")
-  install(FILES "LICENSE.txt" DESTINATION .)
-  install(FILES "Readme.md" DESTINATION .)
 
   # Config.cmake
   write_basic_package_version_file(
