diff --git a/src/Makefile b/src/Makefile
index f40e786..5c2deac 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -918,7 +918,7 @@ $(BIN_INSTALL): $(DEPENDENCIES_BIN)
 	$(MAKE) -C "$($(BIN_INST_ARTIFACT_ID)_PATH)" install DESTDIR="$(DESTDIR)"
 
 $(ARTIFACT_XDG_INST): install_xdg_msg
-	$(MAKE) -C "$($(ARTIFACT_XDG_INST_ID)_PATH)/res/main/xdg" install_xdg DESTDIR="$(DESTDIR)" FILES="$(ARTIFACT_XDG_FILES)"
+	$(MAKE) -C "$($(ARTIFACT_XDG_INST_ID)_PATH)/res/main/xdg" install_xdg DESTDIR="$(DESTDIR)" FILES="$(ARTIFACT_XDG_FILES)" CONFIG=$(ROOTDIR)/.config.mk
 
 install_jack: jack
 	echo "Install jack"
