diff -ruN a/source/Makefile b/source/Makefile
--- a/source/Makefile	2022-05-04 10:22:40.000000000 +0200
+++ b/source/Makefile	2022-05-05 11:34:28.881814839 +0200
@@ -26,7 +26,6 @@
 VERSION = 0.10.3
 CPPFLAGS += -MMD -MP -DVERSION=\"$(VERSION)\" -DLIBDIR=\"$(LIBDIR)\"
 CXXFLAGS += -O2 -Wall
-CXXFLAGS += -march=native
 
 
 all:	aeolus aeolus_x11.so aeolus_txt.so
@@ -75,7 +74,7 @@
 	install -m 755 aeolus $(DESTDIR)$(BINDIR)
 	install -m 755 aeolus_x11.so $(DESTDIR)$(LIBDIR)
 	install -m 755 aeolus_txt.so $(DESTDIR)$(LIBDIR)
-	ldconfig $(PREFIX)/$(LIBDIR)
+	ldconfig -N $(DESTDIR)/$(LIBDIR)
 
 
 clean:
