diff -wbBur tomoyo-tools/usr_lib_tomoyo/tomoyo-editpolicy-agent.c tomoyo-tools.my/usr_lib_tomoyo/tomoyo-editpolicy-agent.c
--- tomoyo-tools/usr_lib_tomoyo/tomoyo-editpolicy-agent.c	2011-05-11 04:00:00.000000000 +0400
+++ tomoyo-tools.my/usr_lib_tomoyo/tomoyo-editpolicy-agent.c	2011-09-30 12:55:43.000000000 +0400
@@ -35,6 +35,7 @@
 #include <dirent.h>
 #define _GNU_SOURCE
 #include <sched.h>
+#include <linux/sched.h>
 #include <sys/mount.h>
 
 static _Bool wait_data(const int fd)
diff -wbBur tomoyo-tools/usr_lib_tomoyo/tomoyo-notifyd.c tomoyo-tools.my/usr_lib_tomoyo/tomoyo-notifyd.c
--- tomoyo-tools/usr_lib_tomoyo/tomoyo-notifyd.c	2011-05-11 04:00:00.000000000 +0400
+++ tomoyo-tools.my/usr_lib_tomoyo/tomoyo-notifyd.c	2011-09-30 12:55:21.000000000 +0400
@@ -31,6 +31,7 @@
 #include <time.h>
 #define _GNU_SOURCE
 #include <sched.h>
+#include <linux/sched.h>
 #include <sys/mount.h>
 
 int main(int argc, char *argv[])
