diff -Naur budgie-desktop-10.4.orig/src/panel/settings/settings_main.vala budgie-desktop-10.4/src/panel/settings/settings_main.vala
--- budgie-desktop-10.4.orig/src/panel/settings/settings_main.vala	2017-08-15 00:00:04.000000000 +0200
+++ budgie-desktop-10.4/src/panel/settings/settings_main.vala	2018-03-29 03:33:48.068212130 +0200
@@ -102,12 +102,6 @@
     void build_content()
     {
         this.add_page(new Budgie.StylePage());
-
-        /* So we don't bust a nut on start up */
-        if (Environment.find_program_in_path("nautilus") != null) {
-            this.add_page(new Budgie.DesktopPage());
-        }
-
         this.add_page(new Budgie.FontPage());
         this.add_page(new Budgie.WindowsPage());
         this.add_page(new Budgie.AutostartPage());
