diff -wbBur scantailor-0.9.10/zones/EditableZoneSet.h scantailor-0.9.10.my/zones/EditableZoneSet.h
--- scantailor-0.9.10/zones/EditableZoneSet.h	2011-07-31 00:47:25.000000000 +0400
+++ scantailor-0.9.10.my/zones/EditableZoneSet.h	2011-08-03 01:59:40.000000000 +0400
@@ -76,6 +77,12 @@
 
 	EditableZoneSet();
 
+	EditableZoneSet(const EditableZoneSet& zs)
+	{
+		m_splineMap = zs.m_splineMap;
+		m_defaultProps = zs.m_defaultProps;
+	}
+
 	const_iterator begin() const { return iterator(m_splineMap.begin()); }
 
 	const_iterator end() const { return iterator(m_splineMap.end()); }
