--- a/script/update
+++ b/script/update
@@ -48,8 +48,8 @@
   if sys.platform == 'win32' and not args.skip_depot_tools_update:
     update_depot_tools()
 
-  if args.clean and os.path.isdir(SRC_DIR):
-    git_clean_recursive(SRC_DIR)
+  # if args.clean and os.path.isdir(SRC_DIR):
+  #   git_clean_recursive(SRC_DIR)
 
   # Warning about using a network share as git cache from Windows 7+: The
   # gclient script may experience errors unless you disable SMBv2 cache by
@@ -60,11 +60,11 @@
   git_cache = args.git_cache or os.getenv('LIBCHROMIUMCONTENT_GIT_CACHE', '')
   target_arch = args.target_arch
   nohooks = target_arch == 'mips64el' and IS_MIPS64EL_HOST
-  if not args.skip_gclient:
-    if (args.source_only):
-      return gclient_sync(chromium_version(), args.clean, git_cache, nohooks)
-    else:
-      gclient_sync(chromium_version(), args.clean, git_cache, nohooks)
+  # if not args.skip_gclient:
+  #   if (args.source_only):
+  #     return gclient_sync(chromium_version(), args.clean, git_cache, nohooks)
+  #   else:
+  #     gclient_sync(chromium_version(), args.clean, git_cache, nohooks)
 
   if target_arch == 'mips64el' and IS_MIPS64EL_HOST:
     os.system('sh script/mips64el/runhooks-mips64el')
