--- a/chromiumcontent/build_libs.py
+++ b/chromiumcontent/build_libs.py
@@ -13,7 +13,8 @@ def gen_list(out, name, obj_dirs):
     out.write(name + " = [\n")
     for base_dir in obj_dirs:
         for dir, subdirs, files in os.walk(os.path.join('obj', base_dir)):
-            for f in files:
+            subdirs.sort()
+            for f in sorted(files):
                 if f.endswith('.obj') or f.endswith('.o'):
                     out.write('"' + os.path.abspath(os.path.join(dir, f)) + '",\n')
     out.write("]\n")
--- a/tools/generate_filenames_gypi.py
+++ b/tools/generate_filenames_gypi.py
@@ -98,7 +98,7 @@ def main(target_file, code_dir, shared_dir, static_dir):
 
 
 def searh_files(src, suffix, exclude):
-  files = glob.glob(os.path.join(src, '*.' + suffix))
+  files = sorted(glob.glob(os.path.join(src, '*.' + suffix)))
   files = [f for f in files if os.path.basename(f) not in exclude]
   return ([os.path.abspath(f) for f in files if not is_v8_library(f)],
           [os.path.abspath(f) for f in files if is_v8_library(f)])
