From b6e44705c0fb68e526ef66d56d26fa677d885193 Mon Sep 17 00:00:00 2001
From: Corbin <RavetcoFX@gmail.com>
Date: Fri, 18 Apr 2014 17:11:15 -0600
Subject: [PATCH] Fix sticky screensaver issue

---
 src/gs-window-x11.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/gs-window-x11.c b/src/gs-window-x11.c
index eb4fe5f..127eda5 100644
--- a/src/gs-window-x11.c
+++ b/src/gs-window-x11.c
@@ -2140,7 +2140,6 @@ shade_background (GtkWidget    *widget,
 {
         cairo_set_source_rgba (cr, 0.0, 0.0, 0.0, 0.7);
         cairo_paint (cr);
-        gs_window_clear_to_background_surface (window);
 
         return FALSE;
 }
-- 
1.9.1

