diff -ur acetoneiso_2.3.orig/acetoneiso/acetoneiso.pro acetoneiso_2.3/acetoneiso/acetoneiso.pro
--- acetoneiso_2.3.orig/acetoneiso/acetoneiso.pro	2010-11-11 19:52:33.000000000 +0100
+++ acetoneiso_2.3/acetoneiso/acetoneiso.pro	2017-02-07 09:40:46.751204646 +0100
@@ -59,7 +59,6 @@
 QT += phonon
 QT += dbus
 #QT += svg
-QT += webkit
 
 #install#
     target.path = /usr/bin
diff -ur acetoneiso_2.3.orig/acetoneiso/sources/acetoneiso.cpp acetoneiso_2.3/acetoneiso/sources/acetoneiso.cpp
--- acetoneiso_2.3.orig/acetoneiso/sources/acetoneiso.cpp	2010-11-11 21:36:19.000000000 +0100
+++ acetoneiso_2.3/acetoneiso/sources/acetoneiso.cpp	2017-02-07 09:36:27.715760057 +0100
@@ -16,7 +16,6 @@
 
 #include <QtGui>
 #include <QTabWidget>
-#include <QWebView>
 #include "acetoneiso.h"
 #include "connections.h"
 #include <fcntl.h>
diff -ur acetoneiso_2.3.orig/acetoneiso/sources/acetoneiso.h acetoneiso_2.3/acetoneiso/sources/acetoneiso.h
--- acetoneiso_2.3.orig/acetoneiso/sources/acetoneiso.h	2010-11-11 00:56:58.000000000 +0100
+++ acetoneiso_2.3/acetoneiso/sources/acetoneiso.h	2017-02-07 09:36:44.429505254 +0100
@@ -21,8 +21,6 @@
 #include <QListWidget>
 #include <QTabWidget>
 #include <QDebug>
-#include <QtWebKit>
-#include <QWebView>
 #include <QProcess>
 #include"progress.h"
 #include"QFileSystemModel"
diff -ur acetoneiso_2.3.orig/acetoneiso/ui/acetoneiso.ui acetoneiso_2.3/acetoneiso/ui/acetoneiso.ui
--- acetoneiso_2.3.orig/acetoneiso/ui/acetoneiso.ui	2010-11-15 02:34:50.000000000 +0100
+++ acetoneiso_2.3/acetoneiso/ui/acetoneiso.ui	2017-02-07 09:37:46.144278448 +0100
@@ -466,47 +466,6 @@
         </property>
        </widget>
       </widget>
-      <widget class="QWidget" name="tab_2">
-       <attribute name="icon">
-        <iconset resource="../acetoneiso.qrc">
-         <normaloff>:/images/splash.png</normaloff>:/images/splash.png</iconset>
-       </attribute>
-       <attribute name="title">
-        <string>Updates</string>
-       </attribute>
-       <layout class="QGridLayout" name="gridLayout_4">
-        <item row="0" column="0">
-         <widget class="QWebView" name="webView" native="true">
-          <property name="sizePolicy">
-           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
-            <horstretch>0</horstretch>
-            <verstretch>0</verstretch>
-           </sizepolicy>
-          </property>
-          <property name="minimumSize">
-           <size>
-            <width>260</width>
-            <height>100</height>
-           </size>
-          </property>
-          <property name="maximumSize">
-           <size>
-            <width>6000</width>
-            <height>6000</height>
-           </size>
-          </property>
-          <property name="toolTip">
-           <string>Real time updates from the net</string>
-          </property>
-          <property name="url" stdset="0">
-           <url>
-            <string>http://www.acetoneteam.org/clients.html</string>
-           </url>
-          </property>
-         </widget>
-        </item>
-       </layout>
-      </widget>
      </widget>
     </item>
     <item row="1" column="0">
@@ -1810,13 +1769,6 @@
    </property>
   </action>
  </widget>
- <customwidgets>
-  <customwidget>
-   <class>QWebView</class>
-   <extends>QWidget</extends>
-   <header>QtWebKit/QWebView</header>
-  </customwidget>
- </customwidgets>
  <tabstops>
   <tabstop>listWidget_2</tabstop>
  </tabstops>
