diff -Naur ubuntuone-client-4.2.0.orig/ubuntuone/platform/notification/linux.py ubuntuone-client-4.2.0/ubuntuone/platform/notification/linux.py
--- ubuntuone-client-4.2.0.orig/ubuntuone/platform/notification/linux.py	2013-04-01 16:37:48.530956424 +0000
+++ ubuntuone-client-4.2.0/ubuntuone/platform/notification/linux.py	2013-04-01 16:38:32.333945773 +0000
@@ -32,7 +32,7 @@
 USE_NOTIFY = False
 
 try:
-    from gi.repository import Notify
+    from gi.repository import GLib, Notify
     USE_NOTIFY = True
 except ImportError:
     pass
@@ -67,5 +67,5 @@
         if append:
             self.notification.set_hint_string('x-canonical-append', '')
 
-        self.notification.set_hint_int32('transient', int(True))
+        self.notification.set_hint('transient', GLib.Variant.new_boolean(1))
         self.notification.show()
