=== modified file 'libawn/awn-applet-simple.c'
--- libawn/awn-applet-simple.c	2010-01-10 14:43:43 +0000
+++ libawn/awn-applet-simple.c	2010-04-11 21:08:28 +0000
@@ -393,16 +393,16 @@
                                  GStrv            states,
                                  GStrv            icon_names)
 {
-  gchar *applet_name;
+  gchar *applet_name=NULL;
   
   g_return_if_fail (AWN_IS_APPLET_SIMPLE (applet));
-  g_return_if_fail (applet_name);
   g_return_if_fail (states);
   g_return_if_fail (icon_names);
 
   g_object_get (applet,
                 "canonical-name",&applet_name,
                 NULL);  
+  g_return_if_fail (applet_name);
   applet->priv->last_set_icon = ICON_THEMED_MANY;
   awn_themed_icon_set_size (AWN_THEMED_ICON (applet->priv->icon),
                             awn_applet_get_size (AWN_APPLET (applet)));

