--- src/makefile.linux~	2009-11-03 17:57:37.978184080 +0500
+++ src/makefile.linux	2009-11-03 18:00:05.228169458 +0500
@@ -37,20 +37,12 @@
 	cp $(OUTPUT) $(LIBDIR)
 	cp $(STATIC) $(LIBDIR)
 	chmod 755 $(LIBDIR)/$(OUTPUT)
-	ln -s $(LIBDIR)/$(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION).$(MINOR_VERSION)
-	ln -s $(LIBDIR)/$(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION)
-	ln -s $(LIBDIR)/$(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so
-	ln -s $(LIBDIR)/$(OUTPUT) $(LIBDIR)/$(LIBNAME).so
+	ln -s $(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION).$(MINOR_VERSION)
+	ln -s $(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION)
+	ln -s $(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so
+	ln -s $(OUTPUT) $(LIBDIR)/$(LIBNAME).so
 	cp ../include/nl.h $(INCDIR)/nl.h
 	chmod 644 $(INCDIR)/nl.h
-	@echo ""
-	@echo "*************************************************"
-	@echo "* Installed HawkNL.                             *"
-	@echo "* Remember to run /sbin/ldconfig before using   *"
-	@echo "* the library, you may also want to check that  *"
-	@echo "* $(LIBDIR) is included in /etc/ld.so.conf      *"
-	@echo "* You must be root to run ldconfig.             *"
-	@echo "*************************************************"
 
 uninstall:
 	rm -f $(LIBDIR)/$(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION).$(MINOR_VERSION)
