diff -ruN a/spsdk/crypto/__init__.py b/spsdk/crypto/__init__.py
--- a/spsdk/crypto/__init__.py	2022-07-29 15:19:47.000000000 +0200
+++ b/spsdk/crypto/__init__.py	2022-07-31 20:31:00.457065560 +0200
@@ -48,7 +48,6 @@
 # Explicit import due to MYPY issue
 from cryptography.hazmat.backends import *
 from cryptography.hazmat.backends import default_backend
-from cryptography.hazmat.backends.interfaces import *
 from cryptography.hazmat.backends.openssl.rsa import *
 
 # Explicit import due to MYPY issue
