commit 8fd2fa1b0c02fda6f2f5f3df96092922ba55f5da
Author: Anatol Pomozov <anatol.pomozov@gmail.com>
Date:   Tue Jun 21 09:37:26 2022 -0700

    Pin sources to ruby2.7 package

diff --git a/Makefile b/Makefile
index e80ab5d72..5d3cd312c 100644
--- a/Makefile
+++ b/Makefile
@@ -402,7 +402,7 @@ race-go: test-go
 .PHONY: rspec
 ## Run Ruby tests.
 rspec: build prepare-tests
-	${Q}cd ${GITALY_RUBY_DIR} && PATH='${SOURCE_DIR}/internal/testhelper/testdata/home/bin:${PATH}' bundle exec rspec
+	${Q}cd ${GITALY_RUBY_DIR} && PATH='${SOURCE_DIR}/internal/testhelper/testdata/home/bin:${PATH}' bundle-2.7 exec rspec
 
 .PHONY: verify
 ## Verify that various files conform to our expectations.
@@ -460,7 +460,7 @@ clean-ruby-vendor-go:
 .PHONY: rubocop
 ## Run Rubocop.
 rubocop: ${RUBY_BUNDLE_FILE}
-	${Q}cd ${GITALY_RUBY_DIR} && bundle exec rubocop --parallel --config ${GITALY_RUBY_DIR}/.rubocop.yml ${GITALY_RUBY_DIR} ${SOURCE_DIR}/_support/test-boot
+	${Q}cd ${GITALY_RUBY_DIR} && bundle-2.7 exec rubocop --parallel --config ${GITALY_RUBY_DIR}/.rubocop.yml ${GITALY_RUBY_DIR} ${SOURCE_DIR}/_support/test-boot
 
 .PHONY: cover
 ## Generate coverage report via Go tests.
@@ -542,7 +542,7 @@ libgit2: ${LIBGIT2_INSTALL_DIR}/lib/libgit2.a
 # step. Both Omnibus and CNG assume it is in the Gitaly root, not in
 # _build. Hence the '../' in front.
 ${RUBY_BUNDLE_FILE}: ${GITALY_RUBY_DIR}/Gemfile.lock ${GITALY_RUBY_DIR}/Gemfile
-	${Q}cd ${GITALY_RUBY_DIR} && bundle install
+	${Q}cd ${GITALY_RUBY_DIR} && bundle-2.7 install
 	${Q}touch $@
 
 ${SOURCE_DIR}/NOTICE: ${BUILD_DIR}/NOTICE
diff --git a/internal/gitaly/linguist/linguist.go b/internal/gitaly/linguist/linguist.go
index 30ba2d612..702b7c777 100644
--- a/internal/gitaly/linguist/linguist.go
+++ b/internal/gitaly/linguist/linguist.go
@@ -86,7 +86,7 @@ func (inst *Instance) startGitLinguist(ctx context.Context, commitID string) (*c
 		return nil, fmt.Errorf("get repo path: %w", err)
 	}
 
-	bundle, err := exec.LookPath("bundle")
+	bundle, err := exec.LookPath("bundle-2.7")
 	if err != nil {
 		return nil, fmt.Errorf("finding bundle executable: %w", err)
 	}
diff --git a/internal/gitaly/rubyserver/rubyserver.go b/internal/gitaly/rubyserver/rubyserver.go
index 7e661954a..4650546ff 100644
--- a/internal/gitaly/rubyserver/rubyserver.go
+++ b/internal/gitaly/rubyserver/rubyserver.go
@@ -191,7 +191,7 @@ func (s *Server) start() error {
 		// Use 'ruby-cd' to make sure gitaly-ruby has the same working directory
 		// as the current process. This is a hack to sort-of support relative
 		// Unix socket paths.
-		args := []string{"bundle", "exec", "bin/ruby-cd", wd, gitalyRuby, strconv.Itoa(os.Getpid()), socketPath}
+		args := []string{"bundle-2.7", "exec", "bin/ruby-cd", wd, gitalyRuby, strconv.Itoa(os.Getpid()), socketPath}
 
 		events := make(chan supervisor.Event)
 		check := func() error { return ping(socketPath) }
diff --git a/ruby/bin/gitaly-ruby b/ruby/bin/gitaly-ruby
index 11cfa231b..d7493f723 100755
--- a/ruby/bin/gitaly-ruby
+++ b/ruby/bin/gitaly-ruby
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby-2.7
 
 # Gitaly always spawns this executable with LANG=en_US.UTF-8. In case the
 # system doesn't have this local, Ruby will fall back to the C locale and as a
diff --git a/ruby/bin/ruby-cd b/ruby/bin/ruby-cd
index 9c60fcd42..15f0cb05a 100755
--- a/ruby/bin/ruby-cd
+++ b/ruby/bin/ruby-cd
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby-2.7
 
 # This script lets you run `bundle exec` in one directory, and then changes into another.
 
diff --git a/tools/protogem/build-proto-gem b/tools/protogem/build-proto-gem
index ee5fc94df..6b36fafc5 100755
--- a/tools/protogem/build-proto-gem
+++ b/tools/protogem/build-proto-gem
@@ -41,7 +41,7 @@ def generate_sources(output_dir, version)
   proto_files = Dir[File.join(proto_dir, '*.proto')].sort
 
   run!(
-    %W[bundle exec grpc_tools_ruby_protoc -I #{proto_dir} --ruby_out=#{proto_output_dir} --grpc_out=#{proto_output_dir}] + proto_files,
+    %W[bundle-2.7 exec grpc_tools_ruby_protoc -I #{proto_dir} --ruby_out=#{proto_output_dir} --grpc_out=#{proto_output_dir}] + proto_files,
     File.join(SOURCE_DIR, 'tools', 'protogem')
   )
 
