commit 93c8397df36651110096f1bb373da8ce3b05efee
Author: Anatol Pomozov <anatol.pomozov@gmail.com>
Date:   Tue Jun 21 09:37:26 2022 -0700

    Pin sources to ruby2.7 package

diff --git a/ruby/bin/gitaly-ruby b/ruby/bin/gitaly-ruby
index 11cfa231b..d7493f723 100755
--- a/ruby/bin/gitaly-ruby
+++ b/ruby/bin/gitaly-ruby
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby-2.7
 
 # Gitaly always spawns this executable with LANG=en_US.UTF-8. In case the
 # system doesn't have this local, Ruby will fall back to the C locale and as a
diff --git a/ruby/bin/ruby-cd b/ruby/bin/ruby-cd
index 9c60fcd42..15f0cb05a 100755
--- a/ruby/bin/ruby-cd
+++ b/ruby/bin/ruby-cd
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby-2.7
 
 # This script lets you run `bundle exec` in one directory, and then changes into another.
 
diff --git a/tools/protogem/build-proto-gem b/tools/protogem/build-proto-gem
index 9825719a7..fcadbdc34 100755
--- a/tools/protogem/build-proto-gem
+++ b/tools/protogem/build-proto-gem
@@ -69,7 +69,7 @@ def generate_sources(output_dir, version)
   proto_files = Dir[File.join(proto_dir, '*.proto')].sort
 
   run!(
-    %W[bundle exec grpc_tools_ruby_protoc -I #{proto_dir} --ruby_out=#{proto_output_dir} --grpc_out=#{proto_output_dir}] + proto_files,
+    %W[bundle-2.7 exec grpc_tools_ruby_protoc -I #{proto_dir} --ruby_out=#{proto_output_dir} --grpc_out=#{proto_output_dir}] + proto_files,
     File.join(SOURCE_DIR, 'tools', 'protogem')
   )
 
