diff --git a/_support/Makefile.template b/_support/Makefile.template
index 592c1e7f..a1eef028 100644
--- a/_support/Makefile.template
+++ b/_support/Makefile.template
@@ -55,8 +55,8 @@ build: ../.ruby-bundle
 # step. Both Omnibus and CNG assume it is in the Gitaly root, not in
 # _build. Hence the '../' in front.
 ../.ruby-bundle:  {{ .GitalyRubyDir }}/Gemfile.lock  {{ .GitalyRubyDir }}/Gemfile
-	cd  {{ .GitalyRubyDir }} && bundle config # for debugging
-	cd  {{ .GitalyRubyDir }} && bundle install $(BUNDLE_FLAGS)
+	cd  {{ .GitalyRubyDir }} && bundle-2.5 config # for debugging
+	cd  {{ .GitalyRubyDir }} && bundle-2.5 install $(BUNDLE_FLAGS)
 	touch $@
 
 .PHONY: install
@@ -135,12 +135,12 @@ race-go: prepare-tests
 
 .PHONY: rspec
 rspec: assemble-go prepare-tests
-	cd  {{ .GitalyRubyDir }} && bundle exec rspec
+	cd  {{ .GitalyRubyDir }} && bundle-2.5 exec rspec
 
 .PHONY: rspec-gitlab-shell
 rspec-gitlab-shell: {{ .GitlabShellDir }}/config.yml assemble-go prepare-tests
 	# rspec in {{ .GitlabShellRelDir }}
-	@cd  {{ .GitalyRubyDir }} && bundle exec bin/ruby-cd {{ .GitlabShellDir }} rspec
+	@cd  {{ .GitalyRubyDir }} && bundle-2.5 exec bin/ruby-cd {{ .GitlabShellDir }} rspec
 
 .PHONY: verify
 verify: check-mod-tidy lint check-formatting staticcheck notice-up-to-date check-proto rubocop
@@ -217,7 +217,7 @@ check-proto: proto no-changes
 
 .PHONY: rubocop
 rubocop: ../.ruby-bundle
-	cd  {{ .GitalyRubyDir }} && bundle exec rubocop --parallel
+	cd  {{ .GitalyRubyDir }} && bundle-2.5 exec rubocop --parallel
 
 .PHONY: cover
 cover: prepare-tests {{ .GoCovMerge }}
diff --git a/go.mod b/go.mod
index 3302cc02..86cabc19 100644
--- a/go.mod
+++ b/go.mod
@@ -23,3 +23,5 @@ require (
 	google.golang.org/grpc v1.16.0
 	gopkg.in/yaml.v2 v2.2.2
 )
+
+go 1.13
diff --git a/internal/linguist/linguist.go b/internal/linguist/linguist.go
index 419f2299..86cdb2ff 100644
--- a/internal/linguist/linguist.go
+++ b/internal/linguist/linguist.go
@@ -103,7 +103,7 @@ func LoadColors(cfg config.Cfg) error {
 }
 
 func startGitLinguist(ctx context.Context, repoPath string, commitID string, linguistCommand string) (io.Reader, error) {
-	cmd := exec.Command("bundle", "exec", "bin/ruby-cd", repoPath, "git-linguist", "--commit="+commitID, linguistCommand)
+	cmd := exec.Command("bundle-2.5", "exec", "bin/ruby-cd", repoPath, "git-linguist", "--commit="+commitID, linguistCommand)
 	cmd.Dir = config.Config.Ruby.Dir
 
 	var env []string
@@ -136,7 +136,7 @@ func openLanguagesJSON(cfg config.Cfg) (io.ReadCloser, error) {
 	// We use a symlink because we cannot trust Bundler to not print garbage
 	// on its stdout.
 	rubyScript := `FileUtils.ln_sf(Bundler.rubygems.find_name('github-linguist').first.full_gem_path, ARGV.first)`
-	cmd := exec.Command("bundle", "exec", "ruby", "-rfileutils", "-e", rubyScript, linguistPathSymlink.Name())
+	cmd := exec.Command("bundle-2.5", "exec", "ruby-2.5", "-rfileutils", "-e", rubyScript, linguistPathSymlink.Name())
 	cmd.Dir = config.Config.Ruby.Dir
 
 	// We have learned that in practice the command we are about to run is a
diff --git a/internal/rubyserver/rubyserver.go b/internal/rubyserver/rubyserver.go
index b9cc7d19..845060db 100644
--- a/internal/rubyserver/rubyserver.go
+++ b/internal/rubyserver/rubyserver.go
@@ -145,7 +145,7 @@ func (s *Server) start() error {
 		// Use 'ruby-cd' to make sure gitaly-ruby has the same working directory
 		// as the current process. This is a hack to sort-of support relative
 		// Unix socket paths.
-		args := []string{"bundle", "exec", "bin/ruby-cd", wd, gitalyRuby, strconv.Itoa(os.Getpid()), socketPath}
+		args := []string{"bundle-2.5", "exec", "bin/ruby-cd", wd, gitalyRuby, strconv.Itoa(os.Getpid()), socketPath}
 
 		events := make(chan supervisor.Event)
 		check := func() error { return ping(socketPath) }
diff --git a/ruby/bin/gitaly-ruby b/ruby/bin/gitaly-ruby
index 4e608739..8e725c66 100755
--- a/ruby/bin/gitaly-ruby
+++ b/ruby/bin/gitaly-ruby
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby-2.5
 
 require 'fileutils'
 
diff --git a/ruby/bin/ruby-cd b/ruby/bin/ruby-cd
index 9c60fcd4..c1b23f5c 100755
--- a/ruby/bin/ruby-cd
+++ b/ruby/bin/ruby-cd
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby-2.5
 
 # This script lets you run `bundle exec` in one directory, and then changes into another.
 
diff --git a/ruby/gitlab-shell/bin/check b/ruby/gitlab-shell/bin/check
index d2224a64..5de42e5b 100755
--- a/ruby/gitlab-shell/bin/check
+++ b/ruby/gitlab-shell/bin/check
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby-2.5
 
 require_relative '../lib/gitlab_init'
 require_relative '../lib/gitlab_net'
diff --git a/ruby/gitlab-shell/hooks/post-receive b/ruby/gitlab-shell/hooks/post-receive
index 2b6538f0..835e3065 100755
--- a/ruby/gitlab-shell/hooks/post-receive
+++ b/ruby/gitlab-shell/hooks/post-receive
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby-2.5
 
 # This file was placed here by GitLab. It makes sure that your pushed commits
 # will be processed properly.
diff --git a/ruby/gitlab-shell/hooks/pre-receive b/ruby/gitlab-shell/hooks/pre-receive
index 6ce58795..3f49689f 100755
--- a/ruby/gitlab-shell/hooks/pre-receive
+++ b/ruby/gitlab-shell/hooks/pre-receive
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby-2.5
 
 # This file was placed here by GitLab. It makes sure that your pushed commits
 # will be processed properly.
diff --git a/ruby/gitlab-shell/hooks/update b/ruby/gitlab-shell/hooks/update
index 4c2fc08b..0bc81ce7 100755
--- a/ruby/gitlab-shell/hooks/update
+++ b/ruby/gitlab-shell/hooks/update
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby-2.5
 
 # This file was placed here by GitLab. It makes sure that your pushed commits
 # will be processed properly.
