Description: fix FTBFS with boost 1.49
Author: Dominique Belhachemi <domibel@debian.org>
Origin: http://anonscm.debian.org/gitweb/?p=collab-maint/vtk.git;a=commitdiff;h=55084a11b450e53b2e8786e23dd64a6a6b846e6b
Last-Update: 2012-05-07

--- a/VTK/Infovis/vtkBoostBreadthFirstSearchTree.cxx
+++ b/VTK/Infovis/vtkBoostBreadthFirstSearchTree.cxx
@@ -49,11 +49,17 @@
 
 #if BOOST_VERSION >= 104800      // Boost 1.48.x
 namespace {
-  vtkIdType unwrap_edge_id(vtkEdgeType const &e) {
+  vtkIdType unwrap_edge_id(vtkEdgeType const &e) 
+  {
     return e.Id;
   }
-  vtkIdType unwrap_edge_id(boost::detail::reverse_graph_edge_descriptor<vtkEdgeType> const &e) {
+  vtkIdType unwrap_edge_id(boost::detail::reverse_graph_edge_descriptor<vtkEdgeType> const &e) 
+  {
+# if BOOST_VERSION == 104800
     return e.underlying_desc.Id;
+# else
+    return e.underlying_descx.Id;
+# endif
   }
 }
 #endif
