diff --git i/library/rbconfig.rb w/library/rbconfig.rb
index 611ad6a..533ef00 100644
--- i/library/rbconfig.rb
+++ w/library/rbconfig.rb
@@ -19,9 +19,9 @@ module RbConfig
   CONFIG["libexecdir"]         = "$(exec_prefix)/libexec"
   CONFIG["datarootdir"]        = "$(prefix)/share"
   CONFIG["datadir"]            = "$(datarootdir)"
-  CONFIG["sysconfdir"]         = "$(prefix)/etc"
+  CONFIG["sysconfdir"]         = "/etc"
   CONFIG["sharedstatedir"]     = "$(prefix)/com"
-  CONFIG["localstatedir"]      = "$(prefix)/var"
+  CONFIG["localstatedir"]      = "/var"
   CONFIG["includedir"]         = "$(prefix)/include"
   CONFIG["oldincludedir"]      = "/usr/include"
   CONFIG["docdir"]             = "$(datarootdir)/doc/$(PACKAGE)"
