--- gambas/branches/3.2/main/lib/signal/csignal.c	2012/07/07 22:18:14	4902
+++ gambas/branches/3.2/main/lib/signal/csignal.c	2012/07/07 22:46:07	4903
@@ -40,10 +40,12 @@
 #define SIGPWR -1
 #endif
 
-#if !defined(OS_BSD) && !defined(OS_CYGWIN)
+// How could the following fix a BSD warning if it is not compiled on BSD systems?
+
+/*#if !defined(OS_BSD) && !defined(OS_CYGWIN)
 typedef
 	struct siginfo siginfo_t;
-#endif
+#endif*/
 
 // The -1 signal is used for ignored signal numbers
 
