From 73f2ac8b2fe5c709de52ea4d4f0aa4c8d3926506 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rapha=C3=ABl=20Rochet?= <raphael@rri.fr>
Date: Fri, 17 Jan 2014 16:25:17 +0100
Subject: [PATCH] Adding app_paintable=true on pie window

---
 src/renderers/pieWindow.vala | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/renderers/pieWindow.vala b/src/renderers/pieWindow.vala
index 66ed863..4b8973d 100644
--- a/src/renderers/pieWindow.vala
+++ b/src/renderers/pieWindow.vala
@@ -85,6 +85,7 @@ public class PieWindow : Gtk.Window {
         this.set_resizable(false);
         this.icon_name = "gnome-pie";
         this.set_accept_focus(false);
+        this.app_paintable = true;
         
         // check for compositing
         if (this.screen.is_composited()) {
-- 
2.0.3

