--- src/gf_util.c	2011-03-14 21:19:46.987545000 +0100
+++ src/gf_util.c	2011-04-18 03:36:24.970676551 +0200
@@ -612,7 +612,7 @@
 	if(!gfire_notify_init() || !p_title)
 		return;
 
-	NotifyNotification *notification = notify_notification_new(p_title, p_msg, NULL, NULL);
+	NotifyNotification *notification = notify_notification_new(p_title, p_msg, NULL);
 	notify_notification_set_urgency(notification, NOTIFY_URGENCY_NORMAL);
 	notify_notification_set_timeout(notification, NOTIFY_EXPIRES_DEFAULT);
 	g_signal_connect(notification, "closed", G_CALLBACK(gfire_notify_closed_cb), NULL);
@@ -637,7 +637,7 @@
 		return;
 	}
 
-	notification = notify_notification_new(p_title, p_msg, NULL, NULL);
+	notification = notify_notification_new(p_title, p_msg, NULL);
 
 	// Get Buddy Icon
 	PurpleBuddyIcon *icon = purple_buddy_get_icon(p_buddy);

