--- pyface-5.0.0/pyface/ipython_widget.py.orig	2015-11-21 16:13:42.171018433 +0100
+++ pyface-5.0.0/pyface/ipython_widget.py	2015-11-21 16:14:03.557532935 +0100
@@ -13,6 +13,7 @@
 #------------------------------------------------------------------------------
 """ The implementation of an IPython shell. """
 
+from __future__ import absolute_import
 
 # Import the toolkit specific version.
 try:
@@ -23,8 +24,5 @@
 Could not load the Wx frontend for ipython.
 You need to have ipython >= 0.9 installed to use the ipython widget.'''
 
-
-from __future__ import absolute_import
-
 from .toolkit import toolkit_object
 IPythonWidget= toolkit_object('ipython_widget:IPythonWidget')
