--- modules/libiconv/manifests/init.pp.orig	2015-07-20 09:32:21.162683278 +0100
+++ modules/libiconv/manifests/init.pp	2015-07-20 12:28:16.068969435 +0100
@@ -46,6 +46,12 @@
     require => Wget::Fetch["libiconv"],
   }
 
+  exec { "patch-libiconv":
+    command => 'patch -Np0 -i ../../../libiconv.patch',
+    cwd     => $file_cache_dir,
+    require => Exec["untar-libiconv"],
+  }
+
   autotools { "libiconv":
     configure_flags  => "--prefix=${prefix} --disable-dependency-tracking",
     cwd              => $source_dir_path,
@@ -53,6 +59,6 @@
     install_sentinel => "${prefix}/lib/libiconv.a",
     make_notify      => $make_notify,
     make_sentinel    => "${source_dir_path}/lib/.libs/iconv.o",
-    require          => Exec["untar-libiconv"],
+    require          => Exec["untar-libiconv", "patch-libiconv"],
   }
 }
