From c09f272d9f27e4c0d6671a70a9bee2f77ac9687b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Corentin=20No=C3=ABl?= <corentin@elementary.io>
Date: Sat, 4 Nov 2017 16:15:02 +0100
Subject: [PATCH] GLib.Application.set_default is useless when there is only
 one instance (#47)

---
 src/Audience.vala | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/Audience.vala b/src/Audience.vala
index 15dcb3b..557fd76 100644
--- a/src/Audience.vala
+++ b/src/Audience.vala
@@ -59,7 +59,6 @@ namespace Audience {
             Gtk.Settings.get_default ().gtk_application_prefer_dark_theme = true;
             this.flags |= GLib.ApplicationFlags.HANDLES_OPEN;
             settings = new Settings ();
-            set_default ();
         }
 
         private static App app; // global App instance
