diff -ru VirtualBox-4.3.20/src/VBox/Additions/x11/vboxvideo/edid.c VirtualBox-4.3.20b/src/VBox/Additions/x11/vboxvideo/edid.c
--- VirtualBox-4.3.20/src/VBox/Additions/x11/vboxvideo/edid.c	2014-11-21 16:16:53.000000000 +0100
+++ VirtualBox-4.3.20b/src/VBox/Additions/x11/vboxvideo/edid.c	2015-02-10 16:37:25.656612649 +0100
@@ -45,6 +45,7 @@
  */
 
 #include <misc.h>
+#include <xorg/xorg-server.h>
 #include <xf86DDC.h>
 #include <xf86Crtc.h>
 #include "vboxvideo.h"
diff -ru VirtualBox-4.3.20/src/VBox/Additions/x11/vboxvideo/pointer.c VirtualBox-4.3.20b/src/VBox/Additions/x11/vboxvideo/pointer.c
--- VirtualBox-4.3.20/src/VBox/Additions/x11/vboxvideo/pointer.c	2014-11-21 16:16:53.000000000 +0100
+++ VirtualBox-4.3.20b/src/VBox/Additions/x11/vboxvideo/pointer.c	2015-02-10 16:31:25.677093405 +0100
@@ -27,6 +27,7 @@
 #include <iprt/string.h>
 #include "compiler.h"
 #include "cursorstr.h"
+#include <xorg/xorg-server.h>
 #include "servermd.h"
 
 #include "vboxvideo.h"
diff -ru VirtualBox-4.3.20/src/VBox/Additions/x11/vboxvideo/vboxvideo.c VirtualBox-4.3.20b/src/VBox/Additions/x11/vboxvideo/vboxvideo.c
--- VirtualBox-4.3.20/src/VBox/Additions/x11/vboxvideo/vboxvideo.c	2014-11-21 16:16:53.000000000 +0100
+++ VirtualBox-4.3.20b/src/VBox/Additions/x11/vboxvideo/vboxvideo.c	2015-02-10 16:31:48.163313719 +0100
@@ -72,6 +72,7 @@
 # include "xf86Pci.h"
 #endif
 
+#include <xorg/xorg-server.h>
 #include "fb.h"
 
 #include "vboxvideo.h"
