diff -wbBur /home/sergej/tmp/BUILD/extra-i686/sergej/build/hedgewars/src/hedgewars-src-0.9.19/hedgewars/uPhysFSLayer.pas hedgewars-src-0.9.19/hedgewars/uPhysFSLayer.pas
--- /home/sergej/tmp/BUILD/extra-i686/sergej/build/hedgewars/src/hedgewars-src-0.9.19/hedgewars/uPhysFSLayer.pas	2013-06-03 20:01:14.212121255 +0400
+++ hedgewars-src-0.9.19/hedgewars/uPhysFSLayer.pas	2013-05-31 00:07:33.000000000 +0400
@@ -9,7 +9,6 @@
 {$IFNDEF WIN32}
     {$linklib physfs}
     {$linklib physlayer}
-    {$linklib gcc_s}
 
     {statically linking physfs brings IOKit dependency on OSX}
     {divdi3 is found in stdc++ on linux x86 and in gcc_s.1 on osx ppc32}
