diff --git a/lib/ivis_opengl/screen.cpp b/lib/ivis_opengl/screen.cpp
index d240a63..861265d 100644
--- a/lib/ivis_opengl/screen.cpp
+++ b/lib/ivis_opengl/screen.cpp
@@ -261,7 +261,7 @@ void screen_EnableMissingFunctions()
 		__glewUseProgram = __glewUseProgramObjectARB;
 		__glewGetShaderiv = __glewGetObjectParameterivARB;
 		__glewLinkProgram = __glewLinkProgramARB;
-		__glewShaderSource = __glewShaderSourceARB;
+		__glewShaderSource = (PFNGLSHADERSOURCEPROC) __glewShaderSourceARB;
 		__glewUniform1f = __glewUniform1fARB;
 		__glewUniform1i = __glewUniform1iARB;
 		__glewUniform4fv = __glewUniform4fvARB;
