diff -wbBur synfig-0.62.00/src/modules/mod_libavcodec/trgt_av.cpp synfig-0.62.00.my/src/modules/mod_libavcodec/trgt_av.cpp
--- synfig-0.62.00/src/modules/mod_libavcodec/trgt_av.cpp	2009-10-12 09:33:44.000000000 +0000
+++ synfig-0.62.00.my/src/modules/mod_libavcodec/trgt_av.cpp	2010-01-25 12:08:16.000000000 +0000
@@ -118,14 +118,14 @@
     picture = avcodec_alloc_frame();
     if (!picture)
         return NULL;
-    size = avpicture_get_size(pix_fmt, width, height);
+    size = avpicture_get_size((enum ::PixelFormat)pix_fmt, width, height);
     picture_buf = (uint8_t *)malloc(size);
     if (!picture_buf) {
         av_free(picture);
         return NULL;
     }
     avpicture_fill((AVPicture *)picture, picture_buf,
-                   pix_fmt, width, height);
+                   (enum ::PixelFormat)pix_fmt, width, height);
     return picture;
 }
 
diff -wbBur synfig-0.62.00/src/modules/mod_png/mptr_png.cpp synfig-0.62.00.my/src/modules/mod_png/mptr_png.cpp
--- synfig-0.62.00/src/modules/mod_png/mptr_png.cpp	2009-10-12 09:33:44.000000000 +0000
+++ synfig-0.62.00.my/src/modules/mod_png/mptr_png.cpp	2010-01-25 13:17:21.000000000 +0000
@@ -309,7 +309,7 @@
 				float b=gamma().b_U8_to_F32((unsigned char)png_ptr->palette[row_pointers[y][x]].blue);
 				float a=1.0;
 				if(info_ptr->valid & PNG_INFO_tRNS)
-				    a = (float)(unsigned char)png_ptr->trans[row_pointers[y][x]]*(1.0/255.0);
+				    a = (float)(unsigned char)png_ptr->trans_alpha[row_pointers[y][x]]*(1.0/255.0);
 				surface_buffer[y][x]=Color(
 					r,
 					g,
diff -wbBur synfig-0.62.00/src/synfig/time.cpp synfig-0.62.00.my/src/synfig/time.cpp
--- synfig-0.62.00/src/synfig/time.cpp	2009-10-12 09:33:44.000000000 +0000
+++ synfig-0.62.00.my/src/synfig/time.cpp	2010-01-25 11:54:02.000000000 +0000
@@ -1,4 +1,6 @@
+#include <stdio.h>
 /* === S Y N F I G ========================================================= */
+#include <stdio.h>
 /*!	\file time.cpp
 **	\brief Template File
 **
