diff -Naur clementine-0.7.ori/src/devices/giolister.cpp clementine-0.7/src/devices/giolister.cpp
--- clementine-0.7.ori/src/devices/giolister.cpp	2010-11-20 05:27:10.000000000 -0800
+++ clementine-0.7/src/devices/giolister.cpp	2011-03-31 04:47:30.580003940 -0700
@@ -243,13 +243,10 @@
   {
     QMutexLocker l(&mutex_);
 
-    const DeviceInfo* old_info = NULL;
-
     // The volume might already exist - either mounted or unmounted.
     foreach (const QString& id, devices_.keys()) {
       if (devices_[id].volume == info.volume) {
         old_id = id;
-        old_info = &devices_[id];
         break;
       }
     }
@@ -260,7 +257,6 @@
       devices_.remove(old_id);
       emit DeviceRemoved(old_id);
 
-      old_info = NULL;
       old_id = QString();
     }
     devices_[info.unique_id()] = info;
diff -Naur clementine-0.7.ori/src/scripting/uiinterface.h clementine-0.7/src/scripting/uiinterface.h
--- clementine-0.7.ori/src/scripting/uiinterface.h	2011-01-02 10:10:26.000000000 -0800
+++ clementine-0.7/src/scripting/uiinterface.h	2011-03-31 04:45:56.436672791 -0700
@@ -45,7 +45,7 @@
 
 private:
   struct Location {
-    Location() {}
+    Location() : menu_(NULL), before_(NULL) {}
     Location(QMenu* menu, QAction* before) : menu_(menu), before_(before) {}
 
     QMenu* menu_;
