diff -Naur clementine-1.0.1.ori/src/main.cpp clementine-1.0.1/src/main.cpp
--- clementine-1.0.1.ori/src/main.cpp	2012-01-22 07:43:26.000000000 -0500
+++ clementine-1.0.1/src/main.cpp	2012-07-05 13:58:49.040596906 -0400
@@ -280,13 +280,6 @@
   UniversalEncodingHandler handler;
   TagLib::ID3v1::Tag::setStringHandler(&handler);
 
-#ifdef Q_OS_LINUX
-  // Force Clementine's menu to be shown in the Clementine window and not in
-  // the Unity global menubar thing.  See:
-  // https://bugs.launchpad.net/unity/+bug/775278
-  setenv("QT_X11_NO_NATIVE_MENUBAR", "1", true);
-#endif
-
   // Initialise logging
   logging::Init();
   logging::SetLevels(options.log_levels());
