diff -urN tables-3.1.1.orig/setup.py tables-3.1.1/setup.py
--- tables-3.1.1.orig/setup.py	2014-09-21 21:11:18.612906437 +0200
+++ tables-3.1.1/setup.py	2014-09-21 21:11:38.562703246 +0200
@@ -123,7 +123,7 @@
 # Check if Cython is installed or not (requisite)
 try:
     from Cython.Distutils import build_ext
-    from Cython.Compiler.Main import Version
+    from Cython.Compiler import Version
     cmdclass['build_ext'] = build_ext
 except ImportError:
     exit_with_error(
