diff --git a/src/sage/libs/polybori/decl.pxd b/src/sage/libs/polybori/decl.pxd
index b8fcc6e..6ec4bce 100644
--- a/src/sage/libs/polybori/decl.pxd
+++ b/src/sage/libs/polybori/decl.pxd
@@ -10,8 +10,8 @@ cdef extern from "sage/libs/polybori/pb_wrap.h":
         pbdp_asc        "CTypes::dp_asc"
         pbblock_dlex    "CTypes::block_dlex"
         pbblock_dp_asc  "CTypes::block_dp_asc"
-        pbdp            "17"
-        pbblock_dp      "19"
+        pbdp            "static_cast<COrderEnums::ordercodes>(17)"
+        pbblock_dp      "static_cast<COrderEnums::ordercodes>(19)"
 
     cdef enum comparecodes "CCompareEnums::comparecodes":
         less_than               "CTypes::less_than"

