diff -up kphotoalbum-4.1.1/doc/index.docbook.docbook_fix kphotoalbum-4.1.1/doc/index.docbook
--- kphotoalbum-4.1.1/doc/index.docbook.docbook_fix	2009-11-15 15:16:29.000000000 -0600
+++ kphotoalbum-4.1.1/doc/index.docbook	2010-06-13 13:03:02.815174744 -0500
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kphotoalbum "<application>KPhotoAlbum</application>">
   <!ENTITY kappname "&kphotoalbum;">
   <!ENTITY package "kdeextragear-2">
@@ -41,12 +41,6 @@
     <date>2004-08-06</date>
     <releaseinfo>2.0</releaseinfo>
     
-    <keywordset>
-      <keyword>KDE</keyword>
-      <keyword>kdeextragear-2</keyword>
-      <keyword>KPhotoAlbum</keyword>
-      <keyword>Image Database</keyword>
-    </keywordset>
     <abstract><para>
   This is the user manual for &kphotoalbum;. If you are
   not the type who likes reading documentation, but rather just want to try
@@ -64,6 +58,12 @@ menu and select <guimenuitem>Run &kphoto
 on long HTML document at <ulink
 url="http://www.kphotoalbum.org/">&kphotoalbum;'s Homepage</ulink></para>
 </abstract>
+    <keywordset>
+      <keyword>KDE</keyword>
+      <keyword>kdeextragear-2</keyword>
+      <keyword>KPhotoAlbum</keyword>
+      <keyword>Image Database</keyword>
+    </keywordset>
   </bookinfo>
   
   &introduction;
diff -up kphotoalbum-4.1.1/doc-translations/de_kphotoalbum/kphotoalbum/index.docbook.docbook_fix kphotoalbum-4.1.1/doc-translations/de_kphotoalbum/kphotoalbum/index.docbook
--- kphotoalbum-4.1.1/doc-translations/de_kphotoalbum/kphotoalbum/index.docbook.docbook_fix	2009-11-15 15:17:07.000000000 -0600
+++ kphotoalbum-4.1.1/doc-translations/de_kphotoalbum/kphotoalbum/index.docbook	2010-06-13 13:10:06.145165665 -0500
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kphotoalbum "<application
 >KPhotoAlbum</application
 >">
@@ -65,16 +65,6 @@
     <releaseinfo
 >2.0</releaseinfo>
     
-    <keywordset>
-      <keyword
->KDE</keyword>
-      <keyword
->kdeextragear-2</keyword>
-      <keyword
->KPhotoAlbum</keyword>
-      <keyword
->Bilddatenbank</keyword>
-    </keywordset>
     <abstract
 ><para
 >Dies ist das Benutzerhandbuch für &kphotoalbum;. Wenn Sie nicht gerne Handbücher lesen, sondern lieber gleich Funktionen ausprobieren möchten, haben Sie folgende Alternativen zu diesem Handbuch: <itemizedlist>
@@ -102,6 +92,12 @@
 >HTML</acronym
 >-Dokument zum Herunterladen.</para>
 </abstract>
+    <keywordset>
+      <keyword>KDE</keyword>
+      <keyword>kdeextragear-2</keyword>
+      <keyword>KPhotoAlbum</keyword>
+      <keyword>Bilddatenbank</keyword>
+    </keywordset>
   </bookinfo>
   
   &introduction;
diff -up kphotoalbum-4.1.1/doc-translations/et_kphotoalbum/kphotoalbum/index.docbook.docbook_fix kphotoalbum-4.1.1/doc-translations/et_kphotoalbum/kphotoalbum/index.docbook
--- kphotoalbum-4.1.1/doc-translations/et_kphotoalbum/kphotoalbum/index.docbook.docbook_fix	2009-11-15 15:17:20.000000000 -0600
+++ kphotoalbum-4.1.1/doc-translations/et_kphotoalbum/kphotoalbum/index.docbook	2010-06-13 13:11:18.860165964 -0500
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kphotoalbum "<application
 >KPhotoAlbum</application
 >">
@@ -65,16 +65,6 @@
     <releaseinfo
 >2.0</releaseinfo>
     
-    <keywordset>
-      <keyword
->KDE</keyword>
-      <keyword
->kdeextragear-2</keyword>
-      <keyword
->KPhotoAlbum</keyword>
-      <keyword
->Piltide andmebaas</keyword>
-    </keywordset>
     <abstract
 ><para
 >See on &kphotoalbum;i käsiraamat. Kui sa ei armasta eriti käsiraamatuid lugeda, vaid soovid lihtsalt proovida, mis mida teeb, siis on sinu ees lahti mitu võimalust: <itemizedlist>
@@ -100,6 +90,12 @@
 >&kphotoalbum;i koduleheküljelt</ulink
 >.</para>
 </abstract>
+    <keywordset>
+      <keyword>KDE</keyword>
+      <keyword>kdeextragear-2</keyword>
+      <keyword>KPhotoAlbum</keyword>
+      <keyword>Piltide andmebaas</keyword>
+    </keywordset>
   </bookinfo>
   
   &introduction;
diff -up kphotoalbum-4.1.1/doc-translations/it_kphotoalbum/kphotoalbum/index.docbook.docbook_fix kphotoalbum-4.1.1/doc-translations/it_kphotoalbum/kphotoalbum/index.docbook
--- kphotoalbum-4.1.1/doc-translations/it_kphotoalbum/kphotoalbum/index.docbook.docbook_fix	2009-11-15 15:17:57.000000000 -0600
+++ kphotoalbum-4.1.1/doc-translations/it_kphotoalbum/kphotoalbum/index.docbook	2010-06-13 13:12:00.978165382 -0500
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kphotoalbum "<application
 >KPhotoAlbum</application
 >">
@@ -65,16 +65,6 @@
     <releaseinfo
 >2.0</releaseinfo>
     
-    <keywordset>
-      <keyword
->KDE</keyword>
-      <keyword
->kdeextragear-2</keyword>
-      <keyword
->KPhotoAlbum</keyword>
-      <keyword
->Banca dati immagini</keyword>
-    </keywordset>
     <abstract
 ><para
 >Questo è il manuale utente di &kphotoalbum;. Se non sei il tipo a cui piace leggere la documentazione, ma che preferisce provare a far funzionare le cose, eccoti delle alternative: <itemizedlist>
@@ -100,6 +90,12 @@
 >homepage di &kphotoalbum;</ulink
 ></para>
 </abstract>
+    <keywordset>
+       <keyword>KDE</keyword>
+       <keyword>kdeextragear-2</keyword>
+       <keyword>KPhotoAlbum</keyword>
+       <keyword>Banca dati immagini</keyword>
+    </keywordset>
   </bookinfo>
   
   &introduction;
diff -up kphotoalbum-4.1.1/doc-translations/sv_kphotoalbum/kphotoalbum/index.docbook.docbook_fix kphotoalbum-4.1.1/doc-translations/sv_kphotoalbum/kphotoalbum/index.docbook
--- kphotoalbum-4.1.1/doc-translations/sv_kphotoalbum/kphotoalbum/index.docbook.docbook_fix	2009-11-15 15:19:13.000000000 -0600
+++ kphotoalbum-4.1.1/doc-translations/sv_kphotoalbum/kphotoalbum/index.docbook	2010-06-13 13:12:45.765165840 -0500
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kphotoalbum "<application
 >KPhotoAlbum</application
 >">
@@ -65,16 +65,6 @@
     <releaseinfo
 >2.0</releaseinfo>
     
-    <keywordset>
-      <keyword
->KDE</keyword>
-      <keyword
->kdeextragear-2</keyword>
-      <keyword
->Kfotoalbum</keyword>
-      <keyword
->Bilddatabas</keyword>
-    </keywordset>
     <abstract
 ><para
 >Det här är användarmanualen för Kfotoalbum. Om du inte är den som gillar att läsa dokumentation, utan istället bara vill försöka dig på saker och ting, ges här några alternativ: <itemizedlist>
@@ -100,6 +90,12 @@
 >Kfotoalbums hemsida</ulink
 ></para>
 </abstract>
+    <keywordset>
+      <keyword>KDE</keyword>
+      <keyword>kdeextragear-2</keyword>
+      <keyword>Kfotoalbum</keyword>
+      <keyword>Bilddatabas</keyword>
+    </keywordset>
   </bookinfo>
   
   &introduction;
diff -up kphotoalbum-4.1.1/doc-translations/uk_kphotoalbum/kphotoalbum/index.docbook.docbook_fix kphotoalbum-4.1.1/doc-translations/uk_kphotoalbum/kphotoalbum/index.docbook
--- kphotoalbum-4.1.1/doc-translations/uk_kphotoalbum/kphotoalbum/index.docbook.docbook_fix	2009-11-15 15:19:26.000000000 -0600
+++ kphotoalbum-4.1.1/doc-translations/uk_kphotoalbum/kphotoalbum/index.docbook	2010-06-13 13:13:28.217166055 -0500
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kphotoalbum "<application
 >KPhotoAlbum</application
 >">
@@ -65,16 +65,6 @@
     <releaseinfo
 >2.0</releaseinfo>
     
-    <keywordset>
-      <keyword
->KDE</keyword>
-      <keyword
->kdeextragear-2</keyword>
-      <keyword
->KPhotoAlbum</keyword>
-      <keyword
->База даних зображень</keyword>
-    </keywordset>
     <abstract
 ><para
 >Це інструкція користувача для &kphotoalbum;. Якщо ви не любитель або любителька читання документації і надаєте перевагу самостійному вивченню програми шляхом експериментів, ви можете піти одним з таких шляхів: <itemizedlist>
@@ -100,6 +90,12 @@
 >домашній сторінці &kphotoalbum;</ulink
 ></para>
 </abstract>
+    <keywordset>
+      <keyword>KDE</keyword>
+      <keyword>kdeextragear-2</keyword>
+      <keyword>KPhotoAlbum</keyword>
+      <keyword>База даних зображень</keyword>
+    </keywordset>
   </bookinfo>
   
   &introduction;
