--- audex-0.74b1/utils/coverfetcher.cpp	2011-01-15 12:06:25.000000000 +0100
+++ audex-0.74b1/utils/coverfetcher_new.cpp	2011-07-20 18:43:37.554588883 +0200
@@ -162,7 +162,10 @@
   cover_tbnids.clear();
   cover_thumbnails.clear();
 
-  QRegExp rx("<a\\shref=(\\/imgres\\?imgurl=[a-zA-Z0-9\\&\\_\\%\\/\\=\\.\\:\\-\\?]+)>[\\s\\n]*<img\\ssrc=([a-zA-Z0-9\\&\\_\\%\\/\\=\\.\\:\\-\\?]+).*>[\\s\\n]*</a>");
+  QRegExp rx("<a\\shref=\"(\\/imgres\\?imgurl=[^\"]+)\">[\\s\\n]*<img[^>]+src=\"([^>]+)\"></a>");
+  QString html = xml;
+  html.replace( QLatin1String("&amp;"), QLatin1String("&") );
+
   rx.setMinimal(TRUE);
   
   int pos = 0; int i = 0;
