diff -upr mapnik-v3.0.13.orig/include/mapnik/text/harfbuzz_shaper.hpp mapnik-v3.0.13/include/mapnik/text/harfbuzz_shaper.hpp
--- mapnik-v3.0.13.orig/include/mapnik/text/harfbuzz_shaper.hpp	2017-02-08 15:13:58.000000000 +0200
+++ mapnik-v3.0.13/include/mapnik/text/harfbuzz_shaper.hpp	2017-04-26 12:14:46.156921035 +0300
@@ -55,11 +55,7 @@ static inline hb_script_t _icu_script_to
 static inline const uint16_t * uchar_to_utf16(const UChar* src)
 {
    static_assert(sizeof(UChar) == sizeof(uint16_t),"UChar is eq size to uint16_t");
-#if defined(_MSC_VER)
    return reinterpret_cast<const uint16_t *>(src);
-#else
-   return src;
-#endif
 }
 
 struct harfbuzz_shaper
diff -upr mapnik-v3.0.13.orig/src/unicode.cpp mapnik-v3.0.13/src/unicode.cpp
--- mapnik-v3.0.13.orig/src/unicode.cpp	2017-02-08 15:13:58.000000000 +0200
+++ mapnik-v3.0.13/src/unicode.cpp	2017-04-26 12:14:20.346507663 +0300
@@ -30,6 +30,7 @@
 #pragma GCC diagnostic push
 #include <mapnik/warning_ignore.hpp>
 #include <unicode/ucnv.h>
+#include <unicode/unistr.h>
 #pragma GCC diagnostic pop
 
 namespace mapnik {
