--- keepalived-1.2.2/configure	2011-01-06 01:51:33.000000000 +0100
+++ keepalived-1.2.2.orig/configure	2011-11-24 19:36:32.051486626 +0100
@@ -3973,7 +3973,7 @@
 
 IPVS_SUPPORT="_WITHOUT_LVS_"
 if test "$enable_lvs" != "no"; then
-  ac_fn_c_check_header_mongrel "$LINENO" "net/ip_vs.h" "ac_cv_header_net_ip_vs_h" "$ac_includes_default"
+  ac_fn_c_check_header_mongrel "$LINENO" "linux/ip_vs.h" "ac_cv_header_net_ip_vs_h" "$ac_includes_default"
 if test "x$ac_cv_header_net_ip_vs_h" = x""yes; then :
   IPVS_SUPPORT="_WITH_LVS_"
 else
diff -ru keepalived-1.2.2/configure.in keepalived-1.2.2.orig/configure.in
--- keepalived-1.2.2/configure.in	2011-01-06 01:51:30.000000000 +0100
+++ keepalived-1.2.2.orig/configure.in	2011-11-24 19:36:11.797860077 +0100
@@ -124,7 +124,7 @@
 dnl ----[ Checks for LVS and VRRP support ]----
 IPVS_SUPPORT="_WITHOUT_LVS_"
 if test "$enable_lvs" != "no"; then
-  AC_CHECK_HEADER([net/ip_vs.h],
+  AC_CHECK_HEADER([linux/ip_vs.h],
     [IPVS_SUPPORT="_WITH_LVS_"],
     [
       IPVS_SUPPORT="_WITHOUT_LVS_"
