--- src/sage/repl/interpreter.py.orig	2015-08-16 14:57:55.571128473 +0200
+++ src/sage/repl/interpreter.py	2015-08-16 14:58:06.647849803 +0200
@@ -104,8 +104,8 @@
 import sys
 from sage.repl.preparse import preparse
 
-from IPython import Config
-from IPython.utils.traitlets import Bool, Type
+from traitlets.config.loader import Config
+from traitlets import Bool, Type
 
 from sage.env import SAGE_LOCAL
 
