diff -upr r8168-8.035.00.orig/src/r8168_n.c r8168-8.035.00/src/r8168_n.c
--- r8168-8.035.00.orig/src/r8168_n.c	2013-02-21 03:31:04.000000000 +0200
+++ r8168-8.035.00/src/r8168_n.c	2013-02-21 03:31:56.000000000 +0200
@@ -14541,7 +14541,7 @@ rtl8168_phy_power_down (struct net_devic
 	spin_unlock_irqrestore(&tp->phy_lock, flags);
 }
 
-static int __devinit
+static int
 rtl8168_init_board(struct pci_dev *pdev,
 		   struct net_device **dev_out,
 		   void __iomem **ioaddr_out)
@@ -14711,7 +14711,7 @@ err_out:
 	goto out;
 }
 
-static void __devinit
+static void
 rtl8168_init_sequence(struct rtl8168_private *tp)
 {
 	void __iomem *ioaddr = tp->mmio_addr;
@@ -14964,7 +14964,7 @@ static const struct net_device_ops rtl81
 };
 #endif
 
-static int __devinit
+static int
 rtl8168_init_one(struct pci_dev *pdev,
 		 const struct pci_device_id *ent)
 {
@@ -15128,7 +15128,7 @@ rtl8168_init_one(struct pci_dev *pdev,
 	return 0;
 }
 
-static void __devexit
+static void
 rtl8168_remove_one(struct pci_dev *pdev)
 {
 	struct net_device *dev = pci_get_drvdata(pdev);
@@ -17649,7 +17649,7 @@ static struct pci_driver rtl8168_pci_dri
 	.name		= MODULENAME,
 	.id_table	= rtl8168_pci_tbl,
 	.probe		= rtl8168_init_one,
-	.remove		= __devexit_p(rtl8168_remove_one),
+	.remove		= rtl8168_remove_one,
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,11)
 	.shutdown	= rtl8168_shutdown,
 #endif
