--- autogen.sh.orig	2013-01-27 10:38:55.000000000 +0100
+++ autogen.sh	2013-01-27 10:59:56.897030226 +0100
@@ -6,9 +6,10 @@
 
 if [ "$AM_INSTALLED_VERSION" != "1.10" \
     -a "$AM_INSTALLED_VERSION" != "1.11" \
-    -a "$AM_INSTALLED_VERSION" != "1.12" ];then
+    -a "$AM_INSTALLED_VERSION" != "1.12" \
+	-a "$AM_INSTALLED_VERSION" != "1.14" ];then
 	echo
-	echo "You must have automake 1.10, 1.11, or 1.12 installed to compile lxpanel."
+	echo "You must have automake >=1.10 installed to compile lxpanel."
 	echo "Install the appropriate package for your distribution,"
 	echo "or get the source tarball at http://ftp.gnu.org/gnu/automake/"
 	exit 1
