--- Makefile.in	2014-07-26 10:41:58.086137103 +0200
+++ Makefile.in	2014-07-26 10:42:06.232803695 +0200
@@ -45,7 +45,7 @@
 	$(QUIET)echo "**********************************************************************"
 	$(QUIET)echo ""
 
-maninstall:
+maninstall: makedirectories
 	$(MKPATH) $(mandir)/$(man1dir)
 	for i in $(MAN1PAGES) ; do		\
 	  $(DOCINSTALL) docs/$$i $(DESTDIR)/$(mandir)/$(man1dir) ;	\
