diff -Naur stk-4.4.2.orig//projects/demo/StkDemo stk-4.4.2/projects/demo/StkDemo
--- stk-4.4.2.orig//projects/demo/StkDemo	2010-09-01 08:51:05.000000000 +0800
+++ stk-4.4.2/projects/demo/StkDemo	2010-09-01 08:51:48.000000000 +0800
@@ -1 +1,2 @@
+cd /usr/lib/stk
 wish < tcl/Demo.tcl | ./demo Clarinet -or -ip
diff -Naur stk-4.4.2.orig//projects/effects/StkEffects stk-4.4.2/projects/effects/StkEffects
--- stk-4.4.2.orig//projects/effects/StkEffects	2010-09-01 08:51:05.000000000 +0800
+++ stk-4.4.2/projects/effects/StkEffects	2010-09-01 08:51:54.000000000 +0800
@@ -1 +1,2 @@
+cd /usr/lib/stk
 wish < tcl/Effects.tcl | ./effects -ip
diff -Naur stk-4.4.2.orig//projects/ragamatic/Raga stk-4.4.2/projects/ragamatic/Raga
--- stk-4.4.2.orig//projects/ragamatic/Raga	2010-09-01 08:51:05.000000000 +0800
+++ stk-4.4.2/projects/ragamatic/Raga	2010-09-01 08:52:04.000000000 +0800
@@ -1 +1,2 @@
+cd /usr/lib/stk
 wish < tcl/Raga.tcl | ./ragamat -ip
