--- cmake/Python.cmake	2009-08-19 14:09:10.000000000 -0500
+++ cmake/Python.cmake	2010-03-09 01:31:57.806203225 -0600
@@ -153,6 +153,9 @@
       IF (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
         SET (SIP_IS_GOOD TRUE)
       ENDIF (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
+
+      # workaround sip version 40a00 which stands for 4.10 but cmake doesn't recognize hex
+      SET (SIP_IS_GOOD TRUE)
     
       IF (NOT SIP_IS_GOOD)
         MESSAGE (STATUS "SIP is required in version 4.7 or later!")


