--- src/bin/sage.orig	2014-12-18 09:47:27.354829070 +0100
+++ src/bin/sage	2014-12-18 09:48:00.041563401 +0100
@@ -352,18 +352,6 @@
 
 # Prepare for running Sage, either interactively or non-interactively.
 sage_setup() {
-    # Check that we're not in a source tarball which hasn't been built yet (#13561).
-    if [ ! -d "$SAGE_LOCAL/lib/python/site-packages/sage" ]; then
-        echo >&2 '************************************************************************'
-        echo >&2 'It seems that you are attempting to run Sage from an unpacked source'
-        echo >&2 'tarball, but you have not compiled it yet (or maybe the build has not'
-        echo >&2 'finished). You should run `make` in the Sage root directory first.'
-        echo >&2 'If you did not intend to build Sage from source, you should download'
-        echo >&2 'a binary tarball instead. Read README.txt for more information.'
-        echo >&2 '************************************************************************'
-        exit 1
-    fi
-
     # Display the startup banner (unless SAGE_BANNER is explictly "no")
     if [ "$SAGE_BANNER" != "no" ]; then
         # can be 'bare', or 'yes', or unspecified

