diff -Nur gambas3-3.4.0/main/Makefile.am gambas3-3.4.0b/main/Makefile.am
--- gambas3-3.4.0/main/Makefile.am	2013-02-03 19:20:59.000000000 +0000
+++ gambas3-3.4.0b/main/Makefile.am	2013-02-21 15:28:16.941250567 +0000
@@ -27,5 +27,7 @@
 uninstall-local:
 	@rm -f $(DESTDIR)$(bindir)/gbr$(GAMBAS_VERSION)
 	@rm -rf $(DESTDIR)$(gblibdir)/info
-	@xdg-mime uninstall $(srcdir)/mime/application-x-gambas3.xml;
-	@xdg-icon-resource uninstall --context mimetypes --size 64 application-x-gambas3;
+	@if test x"$(XDG_UTILS)" != x; then \
+		xdg-mime uninstall $(srcdir)/mime/application-x-gambas3.xml; \
+		xdg-icon-resource uninstall --context mimetypes --size 64 application-x-gambas3; \
+	fi
