diff --git a/platformplugin/linux.pri b/platformplugin/linux.pri
index a36b405..9e4fce8 100644
--- a/platformplugin/linux.pri
+++ b/platformplugin/linux.pri
@@ -22,7 +22,7 @@ SOURCES += \
     $$PWD/utility_x11.cpp \
     $$PWD/dxcbwmsupport.cpp
 
-INCLUDEPATH += $$PWD/libqt5xcbqpa-dev
+INCLUDEPATH += /usr/include/qtxcb-private

 contains(QT_CONFIG, xcb-xlib)|qtConfig(xcb-xlib) {
     DEFINES += XCB_USE_XLIB
@@ -63,13 +63,3 @@ contains(QT_CONFIG, xcb-sm)|qtConfig(xcb-sm) {
 contains(QT_CONFIG, xcb-qt) {
     DEFINES += XCB_USE_RENDER
 }
-
-exists($$PWD/libqt5xcbqpa-dev) {
-    !system(cd $$PWD/libqt5xcbqpa-dev && git checkout $$VERSION) {
-        !system(cd $$PWD/libqt5xcbqpa-dev && git fetch -p):error(update libqt5xcbqpa header sources failed)
-        !system(cd $$PWD/libqt5xcbqpa-dev && git checkout $$VERSION):error(Not support Qt Version: $$VERSION)
-    }
-} else {
-    !system(git clone https://cr.deepin.io/libqt5xcbqpa-dev):error(clone libqt5xcbqpa header sources failed)
-    !system(cd $$PWD/libqt5xcbqpa-dev && git checkout $$VERSION):error(Not support Qt Version: $$VERSION)
-}
