diff -wbBur xbmc-10.1/xbmc/KeyboardStat.cpp xbmc-10.1.my/xbmc/KeyboardStat.cpp
--- xbmc-10.1/xbmc/KeyboardStat.cpp	2011-03-05 06:33:37.000000000 +0300
+++ xbmc-10.1.my/xbmc/KeyboardStat.cpp	2011-12-24 00:27:02.000000000 +0400
@@ -120,6 +120,7 @@
 { { 121, 0xad } // Volume mute
 , { 122, 0xae } // Volume down
 , { 123, 0xaf } // Volume up
+, { 124, 0x82 } // Power
 , { 127, 0x20 } // Pause
 , { 135, 0x5d } // Right click
 , { 136, 0xb2 } // Stop
