diff -wbBur kdenlive-0.8.2/src/videoglwidget.cpp kdenlive-0.8.2.my/src/videoglwidget.cpp
--- kdenlive-0.8.2/src/videoglwidget.cpp	2011-10-31 22:46:21.000000000 +0400
+++ kdenlive-0.8.2.my/src/videoglwidget.cpp	2011-11-02 14:54:10.000000000 +0400
@@ -2,6 +2,7 @@
 #include <QtGui>
 #include <QtOpenGL>
 #include "videoglwidget.h"
+#include <GL/glu.h>
 
 #ifndef GL_TEXTURE_RECTANGLE_EXT
 #define GL_TEXTURE_RECTANGLE_EXT GL_TEXTURE_RECTANGLE_NV
