From 01cb07d2d97f621a510c63157e94092d98856b4d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Johannes=20L=C3=B6thberg?= <johannes@kyriasis.com>
Date: Mon, 4 May 2020 17:57:29 +0200
Subject: [PATCH 4/4] Disable auto updates
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Johannes Löthberg <johannes@kyriasis.com>
---
 qt/aqt/profiles.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qt/aqt/profiles.py b/qt/aqt/profiles.py
index b8dff8ab..8946e80c 100644
--- a/qt/aqt/profiles.py
+++ b/qt/aqt/profiles.py
@@ -28,7 +28,7 @@ from aqt.utils import TR, locale_dir, showWarning, tr
 
 metaConf = dict(
     ver=0,
-    updates=True,
+    updates=False,
     created=intTime(),
     id=random.randrange(0, 2 ** 63),
     lastMsg=-1,
-- 
2.26.2

