diff -aur anki-1.2.8/ankiqt/ui/facteditor.py anki-1.2.8.new//ankiqt/ui/facteditor.py
--- anki-1.2.8/ankiqt/ui/facteditor.py	2011-03-28 07:33:31.000000000 +0000
+++ anki-1.2.8.new//ankiqt/ui/facteditor.py	2011-05-28 11:09:18.167545733 +0000
@@ -14,7 +14,7 @@
 import ankiqt
 from ankiqt.ui.utils import mungeQA, saveGeom, restoreGeom
 from anki.hooks import addHook, removeHook, runHook, runFilter
-from sqlalchemy.exceptions import InvalidRequestError
+from sqlalchemy.exc import InvalidRequestError
 
 clozeColour = "#0000ff"
 
diff -aur anki-1.2.8/libanki/anki/db.py anki-1.2.8.new//libanki/anki/db.py
--- anki-1.2.8/libanki/anki/db.py	2011-03-28 07:14:17.000000000 +0000
+++ anki-1.2.8.new//libanki/anki/db.py	2011-05-28 11:01:55.423520523 +0000
@@ -30,7 +30,7 @@
 from sqlalchemy.orm import mapper, sessionmaker as _sessionmaker, relation, backref, \
      object_session as _object_session, class_mapper
 from sqlalchemy.sql import select, text, and_
-from sqlalchemy.exceptions import DBAPIError, OperationalError
+from sqlalchemy.exc import DBAPIError, OperationalError
 from sqlalchemy.pool import NullPool
 import sqlalchemy
 
