diff -Naur Cinnamon-2.6.13.old/files/usr/share/cinnamon/applets/keyboard@cinnamon.org/applet.js Cinnamon-2.6.13/files/usr/share/cinnamon/applets/keyboard@cinnamon.org/applet.js
--- Cinnamon-2.6.13.old/files/usr/share/cinnamon/applets/keyboard@cinnamon.org/applet.js	2015-08-11 14:00:42.008882266 +0000
+++ Cinnamon-2.6.13/files/usr/share/cinnamon/applets/keyboard@cinnamon.org/applet.js	2015-08-11 14:09:14.683803907 +0000
@@ -139,7 +139,7 @@
         for (let i = 0; i < groups.length; i++) {
             let icon_name = this._config.get_group_name(i);
             let actor;
-            if (this._showFlags && this.icon_theme.has_icon(icon_name))
+            if (this._showFlags)
                 actor = new St.Icon({ icon_name: icon_name, icon_type: St.IconType.FULLCOLOR, style_class: 'popup-menu-icon' });
             else
                 actor = new St.Label({ text: short_names[i] });
@@ -171,7 +171,7 @@
         let selectedLabel = this._labelActors[selected];

         this.set_applet_tooltip(item._long_name)
-        if (this._showFlags && this.icon_theme.has_icon(item._icon_name)) {
+        if (this._showFlags) {
             this.set_applet_icon_name(item._icon_name);
             this.set_applet_label("");
         } else {
