diff --git a/ImageLounge/src/DkCore/DkImageStorage.cpp b/ImageLounge/src/DkCore/DkImageStorage.cpp
index f121c88e..c6b0c894 100644
--- a/ImageLounge/src/DkCore/DkImageStorage.cpp
+++ b/ImageLounge/src/DkCore/DkImageStorage.cpp
@@ -1419,7 +1419,7 @@ void DkImage::mapGammaTable(cv::Mat& img, const QVector<unsigned short>& gammaTa
 	qDebug() << "gamma computation takes: " << dt;
 }
 
-void DkImage::logPolar(const cv::Mat& src, cv::Mat& dst, CvPoint2D32f center, double scaleLog, double angle, double scale) {
+void DkImage::logPolar(const cv::Mat& src, cv::Mat& dst, cv::Point2d center, double scaleLog, double angle, double scale) {
 
 	cv::Mat mapx, mapy;
 
diff --git a/ImageLounge/src/DkCore/DkImageStorage.h b/ImageLounge/src/DkCore/DkImageStorage.h
index 751057e3..62a87971 100644
--- a/ImageLounge/src/DkCore/DkImageStorage.h
+++ b/ImageLounge/src/DkCore/DkImageStorage.h
@@ -105,7 +105,7 @@ public:
 	static void mapGammaTable(cv::Mat& img, const QVector<unsigned short>& gammaTable);
 	static void gammaToLinear(cv::Mat& img);
 	static void linearToGamma(cv::Mat& img);
-	static void logPolar(const cv::Mat& src, cv::Mat& dst, CvPoint2D32f center, double scaleLog, double angle, double scale = 1.0);
+	static void logPolar(const cv::Mat& src, cv::Mat& dst, cv::Point2d center, double scaleLog, double angle, double scale = 1.0);
 	static void tinyPlanet(QImage& img, double scaleLog, double angle, QSize s, bool invert = false);
 #endif
 
