--- src/cairo-compmgr-0.3.0/configure.ac.orig	2010-03-18 00:13:26.000000000 +0200
+++ src/cairo-compmgr-0.3.0/configure.ac	2010-09-24 23:09:27.777000000 +0300
@@ -85,7 +85,7 @@
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
         gtk+-2.0 >= $GTK_REQUIRED
-        vala-1.0 >= $VALA_REQUIRED])
+        vala-0.10 >= $VALA_REQUIRED])
 AC_SUBST(CAIRO_COMPMGR_CFLAGS)
 AC_SUBST(CAIRO_COMPMGR_LIBS)
 