diff -aur Amaya11.3.1/Amaya/thotlib/base/platform.c Amaya11.3.1.new/Amaya/thotlib/base/platform.c
--- Amaya11.3.1/Amaya/thotlib/base/platform.c	2009-12-02 12:23:07.000000000 +0100
+++ Amaya11.3.1.new/Amaya/thotlib/base/platform.c	2010-08-17 21:37:25.699840595 +0200
@@ -329,9 +329,9 @@
     {
       tmp = (char *)TtaGetMemory (strlen(targetFileName)+10);
       sprintf (tmp, "%s.tmp", targetFileName);
-      wxFile::wxFile (targetFile, wxFile::write);
+      wxFile (targetFile, wxFile::write);
       tmpFile = TtaConvMessageToWX(tmp);
-      wxFile::wxFile (tmpFile, wxFile::write);
+      wxFile (tmpFile, wxFile::write);
       result = wxFile::Exists(targetFile);
       if (result)
         result = wxConcatFiles (tmpFile, sourceFile, targetFile);

