commit 7c0acbef88624e04adcd7ebd267a64d65f6896a0
Author: Michail Vourlakos <mvourlakos@gmail.com>
Date:   Thu Oct 26 22:16:42 2017 +0300

    fix #730

diff --git a/plasmoid/package/contents/ui/task/TaskDelegate.qml b/plasmoid/package/contents/ui/task/TaskDelegate.qml
index 4585fe2..5bab8c3 100644
--- a/plasmoid/package/contents/ui/task/TaskDelegate.qml
+++ b/plasmoid/package/contents/ui/task/TaskDelegate.qml
@@ -1218,8 +1218,15 @@ MouseArea{
     }
 
     Component.onDestruction: {
-        wrapper.sendEndOfNeedBothAxisAnimation();
+        root.mouseWasEntered.disconnect(signalMouseWasEntered);
+        root.draggingFinished.disconnect(handlerDraggingFinished);
+        root.clearZoomSignal.disconnect(clearZoom);
+        root.publishTasksGeometries.disconnect(slotPublishGeometries);
+        root.showPreviewForTasks.disconnect(slotShowPreviewForTasks);
+        root.updateScale.disconnect(wrapper.signalUpdateScale);
         root.waitingLauncherRemoved.disconnect(slotWaitingLauncherRemoved);
+
+        wrapper.sendEndOfNeedBothAxisAnimation();
     }
 
     ///REMOVE
diff --git a/plasmoid/package/contents/ui/task/animations/TaskRealRemovalAnimation.qml b/plasmoid/package/contents/ui/task/animations/TaskRealRemovalAnimation.qml
index 3bbf37a..316b79c 100644
--- a/plasmoid/package/contents/ui/task/animations/TaskRealRemovalAnimation.qml
+++ b/plasmoid/package/contents/ui/task/animations/TaskRealRemovalAnimation.qml
@@ -41,13 +41,6 @@ SequentialAnimation {
 
     ScriptAction{
         script:{
-            root.mouseWasEntered.disconnect(signalMouseWasEntered);
-            root.draggingFinished.disconnect(handlerDraggingFinished);
-            root.clearZoomSignal.disconnect(clearZoom);
-            root.publishTasksGeometries.disconnect(slotPublishGeometries);
-            root.showPreviewForTasks.disconnect(slotShowPreviewForTasks);
-            root.updateScale.disconnect(wrapper.signalUpdateScale);
-
             mainItemContainer.inAnimation = true;
             icList.delayingRemoval = true;
             mainItemContainer.inAddRemoveAnimation = true;
