diff -ruN shfs-0.35.orig/shfs/Linux-2.6/inode.c shfs-0.35/shfs/Linux-2.6/inode.c
--- shfs-0.35.orig/shfs/Linux-2.6/inode.c	2004-06-01 15:16:19.000000000 +0200
+++ shfs-0.35/shfs/Linux-2.6/inode.c	2006-04-05 16:59:55.000000000 +0200
@@ -14,6 +14,7 @@
 #include <asm/uaccess.h>
 #include <linux/file.h>
 #include <linux/smp_lock.h>
+#include <linux/version.h>
 
 #include "shfs_fs.h"
 #include "shfs_fs_sb.h"
@@ -118,6 +119,9 @@
 	}
 	KMEM_FREE("inode", inode_cache, i);
 out:
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,13))
+	truncate_inode_pages(&inode->i_data, 0);
+#endif
 	clear_inode(inode);
 }
 
