commit 8902d7ad27ebf0efd4d1886dc1ff0420840c3d94
Author: Anatol Pomozov <anatol.pomozov@gmail.com>
Date:   Sat Mar 21 09:13:31 2020 -0700

    ArchLinux fixes

diff --git a/Gemfile b/Gemfile
index 83b1b0ba25e..0e7a83213b2 100644
--- a/Gemfile
+++ b/Gemfile
@@ -324,6 +324,9 @@ gem 'gettext', '~> 3.3', require: false, group: :development
 
 gem 'batch-loader', '~> 2.0.1'
 
+gem 'irb' # https://bugs.archlinux.org/task/68569
+gem 'rake'
+
 # Perf bar
 gem 'peek', '~> 1.1'
 
