--- librakikpimsync/synceengine.h.old	2006-02-07 21:52:42.000000000 +0200
+++ librakikpimsync/synceengine.h	2006-07-29 14:28:53.000000000 +0300
@@ -86,7 +86,7 @@
 
   private:
     void doSync();
-    template<class T> T  *SynCEEngine::templateSyncee(SynceeList *synceeList) const;
+    template<class T> T  *templateSyncee(SynceeList *synceeList) const;
 
     Konnector::List mOpenedKonnectors;
     Konnector::List mProcessedKonnectors;
