From 2c482276ed39112c650ed886c66d2c7b7d5e3783 Mon Sep 17 00:00:00 2001
From: Jim Crist-Harif <jcristharif@gmail.com>
Date: Tue, 10 Nov 2020 16:29:57 -0600
Subject: [PATCH] Python 3.9 compatibility

---
 distributed/utils.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/distributed/utils.py b/distributed/utils.py
index 77487f8ce..c1a0d2caa 100644
--- a/distributed/utils.py
+++ b/distributed/utils.py
@@ -72,6 +72,10 @@
 
 
 def _initialize_mp_context():
+    if not WINDOWS:
+        # For some reason this is required in python >= 3.9
+        import multiprocessing.popen_spawn_posix
+
     if WINDOWS or PYPY:
         return multiprocessing
     else:
