diff -wbBur mailutils-2.99.98.org/libmu_auth/tls.c mailutils-2.99.98/libmu_auth/tls.c
--- mailutils-2.99.98.org/libmu_auth/tls.c	2012-01-05 22:21:42.000000000 +0400
+++ mailutils-2.99.98/libmu_auth/tls.c	2015-04-16 16:00:39.803642048 +0300
@@ -428,20 +428,9 @@
   struct _mu_tls_stream *sp = (struct _mu_tls_stream *) stream;
   int rc;
   mu_transport_t transport[2];
-  static int protocol_priority[] = {GNUTLS_TLS1, GNUTLS_SSL3, 0};
-  static int kx_priority[] = {GNUTLS_KX_RSA, 0};
-  static int cipher_priority[] = {GNUTLS_CIPHER_3DES_CBC,
-				  GNUTLS_CIPHER_ARCFOUR_128,
-				  0};
-  static int comp_priority[] = {GNUTLS_COMP_NULL, 0};
-  static int mac_priority[] = {GNUTLS_MAC_SHA, GNUTLS_MAC_MD5, 0};
 
   gnutls_init (&sp->session, GNUTLS_CLIENT);
-  gnutls_protocol_set_priority (sp->session, protocol_priority);
-  gnutls_cipher_set_priority (sp->session, cipher_priority);
-  gnutls_compression_set_priority (sp->session, comp_priority);
-  gnutls_kx_set_priority (sp->session, kx_priority);
-  gnutls_mac_set_priority (sp->session, mac_priority);
+  gnutls_set_default_priority (sp->session);
 
   gnutls_certificate_allocate_credentials (&x509_cred);
   if (mu_tls_module_config.ssl_cafile)
