diff -upr gmerlin-avdecoder-1.0.1.orig/lib/pngreader.c gmerlin-avdecoder-1.0.1/lib/pngreader.c
--- gmerlin-avdecoder-1.0.1.orig/lib/pngreader.c	2010-01-25 00:30:55.000000000 +0200
+++ gmerlin-avdecoder-1.0.1/lib/pngreader.c	2010-01-25 00:31:21.000000000 +0200
@@ -139,7 +139,7 @@ int bgav_png_reader_read_header(bgav_png
         bits = 16;
         }
       if(bit_depth < 8)
-        png_set_gray_1_2_4_to_8(png->png_ptr);
+        png_set_expand_gray_1_2_4_to_8(png->png_ptr);
       if (png_get_valid(png->png_ptr, png->info_ptr, PNG_INFO_tRNS))
         {
         png_set_tRNS_to_alpha(png->png_ptr);
