diff --git a/erb.gemspec b/erb.gemspec
index 43ffc89..5d51446 100644
--- a/erb.gemspec
+++ b/erb.gemspec
@@ -1,13 +1,6 @@
-begin
-  require_relative 'lib/erb/version'
-rescue LoadError
-  # for Ruby core repository
-  require_relative 'erb/version'
-end
-
 Gem::Specification.new do |spec|
   spec.name          = 'erb'
-  spec.version       = ERB.const_get(:VERSION, false)
+  spec.version       = '2.2.3'
   spec.authors       = ['Masatoshi SEKI']
   spec.email         = ['seki@ruby-lang.org']
 
