--- aaa/src/adapters/pandoc.rs	2023-03-22 14:32:02.599432479 +0100
+++ bbb/src/adapters/pandoc.rs	2023-03-22 14:32:54.759429382 +0100
@@ -80,7 +80,7 @@
             //.arg("--to=commonmark-header_attributes-link_attributes-fenced_divs-markdown_in_html_blocks-raw_html-native_divs-native_spans-bracketed_spans")
             .arg("--to=plain")
             .arg("--wrap=none")
-            .arg("--atx-headers");
+            .arg("--markdown-headings=atx");
         cmd
     }
 }
