diff -aur griffith-0.12.1/lib/add.py griffith-0.12.1.new//lib/add.py
--- griffith-0.12.1/lib/add.py	2010-10-03 12:22:04.000000000 +0000
+++ griffith-0.12.1.new//lib/add.py	2011-06-12 20:24:18.010607268 +0000
@@ -26,7 +26,7 @@
 import os
 
 import gtk
-from sqlalchemy.exceptions import IntegrityError
+from sqlalchemy.exc import IntegrityError
 
 import quick_filter
 import db
diff -aur griffith-0.12.1/lib/preferences.py griffith-0.12.1.new//lib/preferences.py
--- griffith-0.12.1/lib/preferences.py	2010-09-30 18:49:05.000000000 +0000
+++ griffith-0.12.1.new//lib/preferences.py	2011-06-12 20:24:07.793848977 +0000
@@ -492,7 +492,7 @@
             old['name'] != c.get('name', section='database'):
         log.info('DATABASE: connecting to new db server...')
         import sql
-        from sqlalchemy.exceptions import InvalidRequestError
+        from sqlalchemy.exc import InvalidRequestError
         from initialize import dictionaries, people_treeview
 
         # new database connection
diff -aur griffith-0.12.1/lib/sql.py griffith-0.12.1.new//lib/sql.py
--- griffith-0.12.1/lib/sql.py	2010-09-20 17:50:30.000000000 +0000
+++ griffith-0.12.1.new//lib/sql.py	2011-06-12 20:23:18.713413792 +0000
@@ -29,7 +29,7 @@
 import os.path
 
 from sqlalchemy import create_engine, or_, and_, not_, exists, asc, desc
-from sqlalchemy.exceptions import OperationalError
+from sqlalchemy.exc import OperationalError
 from sqlalchemy.orm import sessionmaker, scoped_session
 from sqlalchemy.sql.expression import Update, Delete
 
