diff -Nru launchy-2.5/src/src.pro launchy-2.5a/src/src.pro
--- launchy-2.5/src/src.pro	2010-03-13 18:29:11.000000000 +0000
+++ launchy-2.5a/src/src.pro	2012-03-13 12:00:35.249250229 +0000
@@ -70,6 +70,7 @@
                ../platforms/unix/platform_x11_hotkey.h \
                 platform_base_hotkey.h \
                 platform_base_hottrigger.h
+    LIBS += -lX11
 
     PREFIX = /usr
     DEFINES += SKINS_PATH=\\\"$$PREFIX/share/launchy/skins/\\\" \
