--- a/Makefile.am
+++ b/Makefile.am
@@ -6,7 +6,7 @@ pkgconfig_DATA = ffms2.pc
 
 dist_doc_DATA = doc/ffms2-api.html doc/ffms2-changelog.html doc/style.css
 
-INCLUDES = -I. -I$(top_srcdir)/include -I$(top_srcdir)/src/config @LIBAV_CFLAGS@ @ZLIB_CPPFLAGS@ -include config.h
+AM_CPPFLAGS = -I. -I$(top_srcdir)/include -I$(top_srcdir)/src/config @LIBAV_CFLAGS@ @ZLIB_CPPFLAGS@ -include config.h
 
 lib_LTLIBRARIES = src/core/libffms2.la
 src_core_libffms2_la_LIBADD = @LIBAV_LIBS@ @ZLIB_LDFLAGS@ -lz @LTUNDEF@
--- a/configure.ac
+++ b/configure.ac
@@ -30,7 +30,7 @@ if test -z "$CXXFLAGS"; then
     CXXFLAGS="$OPT_FLAGS"
 fi
 
-AM_CONFIG_HEADER([src/config/config.h])
+AC_CONFIG_HEADERS([src/config/config.h])
 AC_PROG_CC
 AC_PROG_CXX
 LT_INIT([static disable-shared])
