--- pdfslicer/third-party/catch2/catch.hpp	2022-09-13 17:05:47.939757624 +0200
+++ pdfslicer/third-party/catch2/catch.hpp.new	2022-09-13 17:09:33.785363992 +0200
@@ -8746,7 +8746,8 @@
 
     // 32kb for the alternate stack seems to be sufficient. However, this value
     // is experimentally determined, so that's not guaranteed.
-    constexpr static std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
+    // Update: MINSIGSTKSZ is not const anymore with recent glibc
+    static constexpr std::size_t sigStackSize = 32768;
 
     static SignalDefs signalDefs[] = {
         { SIGINT,  "SIGINT - Terminal interrupt signal" },
