diff --git a/application/CustomControl/DFMGlobal.cpp b/application/CustomControl/DFMGlobal.cpp
index 99be58f6..ddab64c2 100644
--- a/application/CustomControl/DFMGlobal.cpp
+++ b/application/CustomControl/DFMGlobal.cpp
@@ -1,6 +1,8 @@
 #include "DFMGlobal.h"
 #include <private/qtextengine_p.h>
 
+#include <QPainterPath>
+
 #define MAX_FILE_NAME_CHAR_COUNT 255
 #define TEXT_LINE_HEIGHT 18
 
diff --git a/application/CustomControl/ImageLabel.cpp b/application/CustomControl/ImageLabel.cpp
index 1870e5c5..51e2cca2 100644
--- a/application/CustomControl/ImageLabel.cpp
+++ b/application/CustomControl/ImageLabel.cpp
@@ -21,6 +21,7 @@
 #include <DGuiApplicationHelper>
 #include <QDebug>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPalette>
 
 #include "utils/utils.h"
diff --git a/application/CustomControl/RoundColorWidget.cpp b/application/CustomControl/RoundColorWidget.cpp
index 06f1e593..41f29739 100644
--- a/application/CustomControl/RoundColorWidget.cpp
+++ b/application/CustomControl/RoundColorWidget.cpp
@@ -24,6 +24,7 @@
 
 #include <QColor>
 #include <QPainter>
+#include <QPainterPath>
 #include <QMouseEvent>
 #include <QApplication>
 #include <QDebug>
diff --git a/application/document/model.h b/application/document/model.h
index 91ed1f4e..15f1ffa6 100644
--- a/application/document/model.h
+++ b/application/document/model.h
@@ -29,6 +29,7 @@ along with qpdfview.  If not, see <http://www.gnu.org/licenses/>.
 #include <QStandardItemModel>
 #include <QString>
 #include <QWidget>
+#include <QPainterPath>
 
 class QColor;
 class QImage;
diff --git a/application/pdfControl/bookmark/bookmarkdelegate.cpp b/application/pdfControl/bookmark/bookmarkdelegate.cpp
index 7145d877..e8c18480 100644
--- a/application/pdfControl/bookmark/bookmarkdelegate.cpp
+++ b/application/pdfControl/bookmark/bookmarkdelegate.cpp
@@ -18,6 +18,7 @@
 #include "pdfControl/imageviewmodel.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 #include <QItemSelectionModel>
 #include <QAbstractItemView>
diff --git a/application/pdfControl/note/notesdelegate.cpp b/application/pdfControl/note/notesdelegate.cpp
index ecbda8b2..eff67b07 100644
--- a/application/pdfControl/note/notesdelegate.cpp
+++ b/application/pdfControl/note/notesdelegate.cpp
@@ -18,6 +18,7 @@
 #include "pdfControl/imageviewmodel.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 #include <QItemSelectionModel>
 #include <QAbstractItemView>
diff --git a/application/pdfControl/search/searchresdelegate.cpp b/application/pdfControl/search/searchresdelegate.cpp
index d39dec80..217a0190 100644
--- a/application/pdfControl/search/searchresdelegate.cpp
+++ b/application/pdfControl/search/searchresdelegate.cpp
@@ -18,6 +18,7 @@
 #include "pdfControl/imageviewmodel.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 #include <QItemSelectionModel>
 #include <QAbstractItemView>
diff --git a/application/pdfControl/thumbnail/ThumbnailDelegate.cpp b/application/pdfControl/thumbnail/ThumbnailDelegate.cpp
index d6b7a20d..33d11258 100644
--- a/application/pdfControl/thumbnail/ThumbnailDelegate.cpp
+++ b/application/pdfControl/thumbnail/ThumbnailDelegate.cpp
@@ -19,6 +19,7 @@
 #include "utils/utils.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 #include <QItemSelectionModel>
 #include <QAbstractItemView>
