--- a/build/linux/unbundle/re2.gyp
+++ b/build/linux/unbundle/re2.gyp
@@ -8,13 +8,12 @@
       'target_name': 're2',
       'type': 'none',
       'variables': {
-        'headers_root_path': '.',
+        'headers_root_path': 'src',
         'header_filenames': [
           're2/filtered_re2.h',
           're2/re2.h',
           're2/set.h',
           're2/stringpiece.h',
-          're2/variadic_function.h',
         ],
         'shim_generator_additional_args': [
           # Chromium copy of re2 is patched to rename POSIX to POSIX_SYNTAX
