diff --git a/script/build-libchromiumcontent.py b/script/build-libchromiumcontent.py
index e0a95f6..1383e70 100755
--- a/script/build-libchromiumcontent.py
+++ b/script/build-libchromiumcontent.py
@@ -44,7 +44,7 @@ def main():
   build = os.path.join(script_dir, 'build')
   create_dist = os.path.join(script_dir, 'create-dist')
   if args.force_update or libchromiumcontent_outdated():
-    execute_stdout([sys.executable, bootstrap])
+    # execute_stdout([sys.executable, bootstrap])
     execute_stdout([sys.executable, update, '-t', args.target_arch,
                     '--defines', args.defines])
     update_gclient_done_marker()
