--- a/sites.h
+++ b/sites.h
@@ -62,3 +62,18 @@
 /* ORDB: Open Relay DataBase
    http://www.ordb.org/ */
 /* SITE("relays.ordb.org") */
+
+
+SITE("zen.spamhaus.org");
+SITE("sbl.spamhaus.org");
+SITE("xbl.spamhaus.org");
+SITE("pbl.spamhaus.org");
+SITE("bl.spamcop.net");
+SITE("psbl.surriel.com");
+SITE("dnsbl.njabl.org");
+SITE("dul.dnsbl.sorbs.net");
+SITE("all.spam-rbl.fr");
+SITE("spam.spam-rbl.fr");
+SITE("dsl.spam-rbl.fr");
+SITE("bogon.spam-rbl.fr");
+
