diff --git a/lib/Text/Pandoc/CrossRef/Util/Settings/Template.hs b/lib/Text/Pandoc/CrossRef/Util/Settings/Template.hs
index c0f5499..b899f10 100644
--- a/lib/Text/Pandoc/CrossRef/Util/Settings/Template.hs
+++ b/lib/Text/Pandoc/CrossRef/Util/Settings/Template.hs
@@ -24,8 +24,8 @@ fromRecDef t cname f c = do
                   TyConI dec -> return dec
                   _ -> fail "No cons"
   (_, cons) <- case reified of
-               DataD _ _ params cons' _ -> return (params, cons')
-               NewtypeD _ _ params con' _ -> return (params, [con'])
+               DataD _ _ params _ cons' _ -> return (params, cons')
+               NewtypeD _ _ params _ con' _ -> return (params, [con'])
                _ -> fail "No cons"
   decs <- liftM concat . mapM (\ (name,_,_) -> f t name) . nub $ concatMap namedFields cons
   return $ c cname decs
@@ -50,7 +50,7 @@ makeCon t cname = fromRecDef t cname makeCon' RecConE
 
 makeCon' :: Name -> Name -> Q [(Name, Exp)]
 makeCon' t accName = do
-    VarI _ t' _ _ <- reify accName
+    VarI _ t' _ <- reify accName
     funT <- [t|$(conT t) -> Bool -> Int -> [Inline]|]
     inlT <- [t|$(conT t) -> [Inline]|]
     blkT <- [t|$(conT t) -> [Block]|]
