diff -Naur gwibber-3.6.0.orig/gwibber/microblog/util/imagehandler.py gwibber-3.6.0/gwibber/microblog/util/imagehandler.py
--- gwibber-3.6.0.orig/gwibber/microblog/util/imagehandler.py	2012-08-21 06:14:09.000000000 +0200
+++ gwibber-3.6.0/gwibber/microblog/util/imagehandler.py	2013-09-15 15:03:07.056371930 +0200
@@ -1,6 +1,7 @@
 #!/usr/bin/env python
 
-import os, hashlib, urllib2, Image
+import os, hashlib, urllib2
+from PIL import Image
 
 DEFAULT_AVATAR = 'http://digg.com/img/udl.png'
 
diff -Naur gwibber-3.6.0.orig/gwibber/microblog/util/resources.py gwibber-3.6.0/gwibber/microblog/util/resources.py
--- gwibber-3.6.0.orig/gwibber/microblog/util/resources.py	2012-08-21 06:14:09.000000000 +0200
+++ gwibber-3.6.0/gwibber/microblog/util/resources.py	2013-09-15 15:02:36.825648020 +0200
@@ -8,7 +8,7 @@
 from hashlib import sha1
 from os import makedirs, remove, environ
 from os.path import join, isdir, realpath, exists
-import Image
+from PIL import Image
 from datetime import datetime
 from gwibber.microblog import network
 from gwibber.microblog.util.const import *
