diff --git a/tools/system_libs.py b/tools/system_libs.py
index a1e616fa4..0b483d835 100755
--- a/tools/system_libs.py
+++ b/tools/system_libs.py
@@ -856,6 +856,9 @@ class libcxxabi(CXXLibrary, NoExceptLibrary, MTLibrary):
       # revision: https://reviews.llvm.org/D64961
       '-D_LIBCXXABI_GUARD_ABI_ARM',
     ]
+  includes = [
+    ['system', 'lib', 'libunwind', 'include'],
+  ]
 
   def get_cflags(self):
     cflags = super(libcxxabi, self).get_cflags()
