--- branches/v5-30-00-patches/configure	2011/06/27 13:38:32	39993
+++ branches/v5-30-00-patches/configure	2011/07/04 15:43:59	40128
@@ -2430,8 +2430,9 @@
     check_header "lzma.h" "" \
         $LZMA ${LZMA:+$LZMA/include} \
         ${finkdir:+$finkdir/include} \
-        /usr/local/include /usr/include/lzma /usr/local/include/lzma \
-        /opt/lzma/include /usr/include
+        /usr/local/include /usr/include \
+        /usr/local/include/lzma /usr/include/lzma \
+        /opt/lzma/include
     if test "x$found_dir" = "x" ; then
         enable_builtin_lzma=yes
     else
