diff -u -r luxrender-lux-b3f85cf7742f/CMakeLists.txt luxrender-lux-b3f85cf7742f-lib/CMakeLists.txt
--- luxrender-lux-b3f85cf7742f/CMakeLists.txt	2016-05-09 23:21:17.000000000 +0200
+++ luxrender-lux-b3f85cf7742f-lib/CMakeLists.txt	2018-07-29 16:09:18.682728166 +0200
@@ -224,7 +224,6 @@
 	STRING(REGEX MATCH "(x86_64-*)|(X86_64-*)|(AMD64-*)|(amd64-*)" _mach_x86_64 ${MACHINE})
 	IF (_mach_x86_64)
 		SET(ARCH_X86_64 1)
-		SET(LIB_SUFFIX 64)
 		#jromang - Hack to avoid boost bug on x64  Ubuntu 8.10 and Fedora 10 (http://www.luxrender.net/mantis/view.php?id=433)
 		ADD_DEFINITIONS(-DBOOST_NO_INTRINSIC_INT64_T)
 	ENDIF (_mach_x86_64)
