diff --git a/filedigger/main.cpp b/filedigger/main.cpp
index 6682985..43fe9ae 100644
--- a/filedigger/main.cpp
+++ b/filedigger/main.cpp
@@ -21,11 +21,7 @@
 #include "filedigger.h"
 #include "mergedvfs.h"
 
-#if LIBGIT2_VER_MAJOR == 0 && LIBGIT2_VER_MINOR >= 24
 #include <git2/global.h>
-#else
-#include <git2/threads.h>
-#endif
 
 #include <KAboutData>
 #include <KLocalizedString>
@@ -66,19 +62,11 @@ int main(int pArgCount, char **pArgArray) {
 	}
 
 	// This needs to be called first thing, before any other calls to libgit2.
-	#if LIBGIT2_VER_MAJOR == 0 && LIBGIT2_VER_MINOR >= 24
 	git_libgit2_init();
-	#else
-	git_threads_init();
-	#endif
 
 	FileDigger *lFileDigger = new FileDigger(lRepoPath, lParser.value(QStringLiteral("branch")));
 	lFileDigger->show();
 	int lRetVal = lApp.exec();
-	#if LIBGIT2_VER_MAJOR == 0 && LIBGIT2_VER_MINOR >= 24
 	git_libgit2_shutdown();
-	#else
-	git_threads_shutdown();
-	#endif
 	return lRetVal;
 }
diff --git a/kioslave/bupslave.cpp b/kioslave/bupslave.cpp
index 1a7e1a3..0812658 100644
--- a/kioslave/bupslave.cpp
+++ b/kioslave/bupslave.cpp
@@ -64,22 +64,14 @@ BupSlave::BupSlave(const QByteArray &pPoolSocket, const QByteArray &pAppSocket)
 {
 	mRepository = nullptr;
 	mOpenFile = nullptr;
-	#if LIBGIT2_VER_MAJOR == 0 && LIBGIT2_VER_MINOR >= 24
 	git_libgit2_init();
-	#else
-	git_threads_init();
-	#endif
 }
 
 BupSlave::~BupSlave() {
 	if(mRepository != nullptr) {
 		delete mRepository;
 	}
-	#if LIBGIT2_VER_MAJOR == 0 && LIBGIT2_VER_MINOR >= 24
 	git_libgit2_shutdown();
-	#else
-	git_threads_shutdown();
-	#endif
 }
 
 void BupSlave::close() {
