--- recoll-1.13.04.orig/rcldb/stemdb.cpp
+++ recoll-1.13.04/rcldb/stemdb.cpp
@@ -12,7 +12,6 @@ static char rcsid[] = "@(#$Id: stemdb.cp
 #include <map>

 #include <xapian.h>
-#include <xapian/stem.h>

 #include "stemdb.h"
 #include "wipedir.h"
--- recoll-1.13.04.orig/rcldb/rclquery.cpp
+++ recoll-1.13.04/rcldb/rclquery.cpp
@@ -8,7 +8,7 @@ static char rcsid[] = "@(#$ID: rclquery.
 #include <list>
 #include <vector>

-#include "xapian/sorter.h"
+#include <xapian.h>

 #include "rcldb.h"
 #include "rcldb_p.h"
--- recoll-1.13.04.orig/rcldb/rcldb.cpp
+++ recoll-1.13.04/rcldb/rcldb.cpp
@@ -33,7 +33,7 @@ static char rcsid[] = "@(#$Id: rcldb.cpp
 using namespace std;
 #endif /* NO_NAMESPACES */

-#include "xapian/version.h"
+#include <xapian.h>

 #include "rclconfig.h"
 #include "rcldb.h"
