From 30b775a7b36ce7fbc04656c7eec4809f99d3e178 Mon Sep 17 00:00:00 2001
From: Hai Zhu <35182391+cocolato@users.noreply.github.com>
Date: Tue, 28 Feb 2023 02:52:28 +0000
Subject: [PATCH] remove asyncio.coroutine

---
 fire/test_components_py3.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/fire/test_components_py3.py b/fire/test_components_py3.py
index 5140921d..17fb932c 100644
--- a/fire/test_components_py3.py
+++ b/fire/test_components_py3.py
@@ -57,8 +57,7 @@ def lru_cache_decorated(arg1):
 
 class WithAsyncio(object):
 
-  @asyncio.coroutine
-  def double(self, count=0):
+  async def double(self, count=0):
     return 2 * count
 
 
