diff --git a/src/Text/Mustache/Internal/Types.hs b/src/Text/Mustache/Internal/Types.hs
index d824ba7..1552d09 100644
--- a/src/Text/Mustache/Internal/Types.hs
+++ b/src/Text/Mustache/Internal/Types.hs
@@ -24,7 +24,8 @@ import           Data.Text
 import           Data.Text
 import qualified Data.Text.Lazy           as LT
 import qualified Data.Vector              as V
-import           Language.Haskell.TH.Lift (Lift (lift), deriveLift)
+import           Language.Haskell.TH.Lift (deriveLift)
+import Language.Haskell.TH.Syntax


 -- | Type of errors we may encounter during substitution.
@@ -380,16 +381,20 @@ data Template = Template
   , partials :: TemplateCache
   } deriving (Show)

+
+deriveLift ''DataIdentifier
+deriveLift ''Node
+deriveLift ''Template
+
 instance Lift TemplateCache where
+#if MIN_VERSION_template_haskell(2,16,0)
+  liftTyped m = [|| HM.fromList $$(liftTyped $ HM.toList m) ||]
+#else
   lift m = [| HM.fromList $(lift $ HM.toList m) |]
+#endif

 --Data.Text 1.2.4.0 introduces its own Lift Text instance
 #if !MIN_VERSION_text(1,2,4)
 instance Lift Text where
   lift = lift . unpack
 #endif
-
-deriveLift ''DataIdentifier
-deriveLift ''Node
-deriveLift ''Template
-
