diff -u -r gmic-1.4.3.1-orig/configure gmic-1.4.3.1/configure
--- gmic-1.4.3.1-orig/configure	2010-10-01 15:46:19.000000000 +0200
+++ gmic-1.4.3.1/configure	2010-10-06 23:36:07.613267761 +0200
@@ -4980,9 +4980,9 @@
 }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_GraphicsMagickpp_InitializeMagick=yes
-else
   ac_cv_lib_GraphicsMagickpp_InitializeMagick=no
+else
+  ac_cv_lib_GraphicsMagickpp_InitializeMagick=yes
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
@@ -5032,9 +5032,9 @@
 }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_IlmImf_ImfInputReadPixels=yes
-else
   ac_cv_lib_IlmImf_ImfInputReadPixels=no
+else
+  ac_cv_lib_IlmImf_ImfInputReadPixels=yes
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
@@ -5078,9 +5078,9 @@
 }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_cv_main=yes
-else
   ac_cv_lib_cv_main=no
+else
+  ac_cv_lib_cv_main=yes
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
@@ -5124,9 +5124,9 @@
 }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_highgui_main=yes
-else
   ac_cv_lib_highgui_main=no
+else
+  ac_cv_lib_highgui_main=yes
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
