diff -rupN clementine-1.0.1.old/src/covers/albumcoverfetcher.cpp clementine-1.0.1/src/covers/albumcoverfetcher.cpp
--- clementine-1.0.1.old/src/covers/albumcoverfetcher.cpp	2011-12-02 16:24:44.000000000 -0500
+++ clementine-1.0.1/src/covers/albumcoverfetcher.cpp	2012-02-26 11:37:56.351044572 -0500
@@ -108,12 +108,16 @@ void AlbumCoverFetcher::StartRequests()
 
 void AlbumCoverFetcher::SingleSearchFinished(quint64 request_id, CoverSearchResults results) {
   AlbumCoverFetcherSearch* search = active_requests_.take(request_id);
+  if (!search)
+    return;
   search->deleteLater();
   emit SearchFinished(request_id, results, search->statistics());
 }
 
 void AlbumCoverFetcher::SingleCoverFetched(quint64 request_id, const QImage& image) {
   AlbumCoverFetcherSearch* search = active_requests_.take(request_id);
+  if (!search)
+    return;
   search->deleteLater();
   emit AlbumCoverFetched(request_id, image, search->statistics());
 }
