diff --git a/WORKSPACE b/WORKSPACE
index 30007945..479886aa 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -82,11 +82,10 @@ http_archive(
     name = "org_tensorflow",
     # NOTE: when updating this, MAKE SURE to also update the protobuf_js runtime version
     # in third_party/workspace.bzl to >= the protobuf/protoc version provided by TF.
-    sha256 = "48ddba718da76df56fd4c48b4bbf4f97f254ba269ec4be67f783684c75563ef8",
-    strip_prefix = "tensorflow-2.0.0-rc0",
+    sha256 = "364fecb68c48dd4c4fc89fdaede51f29550b612e0c70be33c8ff880e3d535128",
+    strip_prefix = "tensorflow-a6ba397ac0bb2de666709dfc33322b9111414aa7",
     urls = [
-        "http://mirror.tensorflow.org/github.com/tensorflow/tensorflow/archive/v2.0.0-rc0.tar.gz",  # 2019-08-23
-        "https://github.com/tensorflow/tensorflow/archive/v2.0.0-rc0.tar.gz",
+        "https://github.com/tensorflow/tensorflow/archive/a6ba397ac0bb2de666709dfc33322b9111414aa7.tar.gz",
     ],
 )
 
