diff --git a/CMakeLists.txt b/CMakeLists.txt
index de695d2..4c71030 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,10 @@ if (DEFINED DISABLE_DRAG_ANIMATION)
     add_definitions(-DDISABLE_DRAG_ANIMATION)
 endif ()
 
+if (DEFINED WITHOUT_UNINSTALL_APP)
+    add_definitions(-DWITHOUT_UNINSTALL_APP)
+endif ()
+
 set(BIN_NAME dde-launcher)
 
 # Sources files
diff --git a/src/worker/menuworker.cpp b/src/worker/menuworker.cpp
index 91e226d..c00452f 100644
--- a/src/worker/menuworker.cpp
+++ b/src/worker/menuworker.cpp
@@ -122,7 +122,10 @@ void MenuWorker::showMenuByAppItem(QPoint pos, const QModelIndex &index) {
     }
 
     uninstall->setEnabled(m_isRemovable);
+
+#ifndef WITHOUT_UNINSTALL_APP
     menu->addAction(uninstall);
+#endif
 
     connect(open, &QAction::triggered, signalMapper, static_cast<void (QSignalMapper::*)()>(&QSignalMapper::map));
     connect(desktop, &QAction::triggered, signalMapper, static_cast<void (QSignalMapper::*)()>(&QSignalMapper::map));
