Index: aiccu-20070115/Makefile
===================================================================
--- aiccu-20070115.orig/Makefile	2009-10-25 22:25:45.000000000 +0100
+++ aiccu-20070115/Makefile	2009-10-25 22:25:58.000000000 +0100
@@ -84,9 +84,9 @@
 	@cp doc/README ${DESTDIR}${dirdoc}
 	@cp doc/LICENSE ${DESTDIR}${dirdoc}
 	@cp doc/HOWTO  ${DESTDIR}${dirdoc}
-	@echo "Installing Debian-style init.d"
-	@mkdir -p ${DESTDIR}${diretc}init.d
-	@cp doc/${PROJECT}.init.debian ${DESTDIR}${diretc}init.d/${PROJECT}
+#	@echo "Installing Debian-style init.d"
+#	@mkdir -p ${DESTDIR}${diretc}init.d
+#	@cp doc/${PROJECT}.init.debian ${DESTDIR}${diretc}init.d/${PROJECT}
 else
 	@echo "Installing Redhat-style init.d"
 	@mkdir -p ${DESTDIR}${diretc}init.d
