Fix some spelling errors to satisfy lintian

--- aiccu.orig/common/aiccu_test.c
+++ aiccu/common/aiccu_test.c
@@ -133,7 +133,7 @@
 	printf("### This confirms the reachability of the other side of the tunnel\n");
 	printf("### If it doesn't reply then check your interface and routing tables\n");
 	printf("### Don't forget to check your firewall of course\n");
-	printf("### If the previous test was succesful then this could be both\n");
+	printf("### If the previous test was successful then this could be both\n");
 	printf("### a firewalling and a routing/interface problem\n");
 	printf("\n");
 	system_arg(PING6, PINGCOUNT, hTunnel->sIPv6_POP);
--- aiccu.orig/common/common.c
+++ aiccu/common/common.c
@@ -484,7 +484,7 @@
 		return false;
 	}
 
-	dolog(LOG_DEBUG, "TLS Handshake completed succesfully\n");
+	dolog(LOG_DEBUG, "TLS Handshake completed successfully\n");
 
 	sock->tls_active = true;
 	return true;
--- aiccu.orig/common/tic.c
+++ aiccu/common/tic.c
@@ -517,8 +517,8 @@
 		struct in6_addr ipv6_ll, ipv6_local;
 		char ll[100];
 
-		/* Log that the fetch was succesful */
-		dolog(LOG_INFO, "Succesfully retrieved tunnel information for %s\n", sId);
+		/* Log that the fetch was successful */
+		dolog(LOG_INFO, "Successfully retrieved tunnel information for %s\n", sId);
 
 		/*
 		 * Some TUN/TAP devices don't have any
@@ -652,7 +652,7 @@
 	/* All went okay? */
 	if (buf[0] == '2' && buf[1] == '0' && buf[2] == '2')
 	{
-		dolog(LOG_INFO, "Succesfully retrieved POP information for %s\n", sId);
+		dolog(LOG_INFO, "Successfully retrieved POP information for %s\n", sId);
 		return pop;
 	}
 
