--- src/module_list.py.orig	2014-11-23 15:58:13.000000000 +0100
+++ src/module_list.py	2014-12-23 10:43:44.412576412 +0100
@@ -2126,7 +2126,8 @@
 # These extensions are to be compiled only if the
 # corresponding packages have been installed
 
-from sage.misc.package import is_package_installed
+def is_package_installed(name):
+    return False
 
 if is_package_installed('fes'):
     ext_modules.extend([
@@ -2208,7 +2209,7 @@
         )
 
 
-if is_package_installed('lrcalc'):
+if 1:
     ext_modules.append(
         Extension('sage.libs.lrcalc.lrcalc',
                   sources = ["sage/libs/lrcalc/lrcalc.pyx"],
