diff -up src/module_list.py.orig src/module_list.py
--- src/module_list.py.orig	2014-05-26 13:34:05.597783829 -0300
+++ src/module_list.py	2014-05-26 13:34:15.908784224 -0300
@@ -2070,7 +2070,8 @@ ext_modules = [
 # These extensions are to be compiled only if the
 # corresponding packages have been installed
 
-from sage.misc.package import is_package_installed
+def is_package_installed(name):
+    return False
 
 if is_package_installed('fes'):
     ext_modules.extend([