diff --git a/src/widgets/ColorPicker.cpp b/src/widgets/ColorPicker.cpp
index 25df2e3..fd9ce08 100644
--- a/src/widgets/ColorPicker.cpp
+++ b/src/widgets/ColorPicker.cpp
@@ -3,6 +3,7 @@
 
 #include <QPaintEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QMouseEvent>
 #include <QDesktopWidget>
 #include <QPixmap>
diff --git a/src/widgets/ColorThemeListView.cpp b/src/widgets/ColorThemeListView.cpp
index d79b7eb..6013fa4 100644
--- a/src/widgets/ColorThemeListView.cpp
+++ b/src/widgets/ColorThemeListView.cpp
@@ -3,6 +3,7 @@
 #include <QJsonArray>
 #include <QMap>
 #include <QPainter>
+#include <QPainterPath>
 #include <QFontMetrics>
 #include <QScreen>
 #include <QJsonArray>
diff --git a/src/widgets/DisassemblyWidget.cpp b/src/widgets/DisassemblyWidget.cpp
index ef5ad7e..82bb0b6 100644
--- a/src/widgets/DisassemblyWidget.cpp
+++ b/src/widgets/DisassemblyWidget.cpp
@@ -14,6 +14,7 @@
 #include <QRegularExpression>
 #include <QTextBlockUserData>
 #include <QPainter>
+#include <QPainterPath>
 #include <QSplitter>
 
 
