diff -ruN a/app/gui/qt/CMakeLists.txt b/app/gui/qt/CMakeLists.txt
--- a/app/gui/qt/CMakeLists.txt	2021-02-01 12:10:30.000000000 +0100
+++ b/app/gui/qt/CMakeLists.txt	2021-09-20 00:34:44.730830353 +0200
@@ -195,7 +195,6 @@
 target_include_directories(${APP_NAME}
     PRIVATE
     ${QTAPP_ROOT}
-    ${APP_ROOT}/external/scsynth-boost-1.74.0
     ${APP_ROOT}/external/TLSF-2.4.6/src
     ${APP_ROOT}/external
     ${QTAPP_ROOT}/osc
