diff -ruN a/app/gui/qt/mainwindow.cpp b/app/gui/qt/mainwindow.cpp
--- a/app/gui/qt/mainwindow.cpp	2020-02-25 01:23:45.000000000 +0100
+++ b/app/gui/qt/mainwindow.cpp	2020-03-08 13:53:17.604508214 +0100
@@ -353,10 +353,10 @@
         ruby_path = "ruby";
     }
 
-    ruby_server_path = QDir::toNativeSeparators(root_path + "/app/server/ruby/bin/sonic-pi-server.rb");
-    port_discovery_path = QDir::toNativeSeparators(root_path + "/app/server/ruby/bin/port-discovery.rb");
-    fetch_url_path = QDir::toNativeSeparators(root_path + "/app/server/ruby/bin/fetch-url.rb");
-    sample_path = QDir::toNativeSeparators(root_path + "/etc/samples");
+    ruby_server_path = QDir::toNativeSeparators(root_path + "/usr/lib/sonic-pi/server/bin/sonic-pi-server.rb");
+    port_discovery_path = QDir::toNativeSeparators(root_path + "/usr/lib/sonic-pi/server/bin/port-discovery.rb");
+    fetch_url_path = QDir::toNativeSeparators(root_path + "/usr/lib/sonic-pi/server/bin/fetch-url.rb");
+    sample_path = QDir::toNativeSeparators(root_path + "/usr/share/sonic-pi/samples");
 
     sp_user_path           = QDir::toNativeSeparators(sonicPiHomePath() + "/.sonic-pi");
     sp_user_tmp_path       = QDir::toNativeSeparators(sp_user_path + "/.writableTesterPath");
@@ -367,14 +367,14 @@
     process_log_path       = QDir::toNativeSeparators(log_path + "/processes.log");
     scsynth_log_path       = QDir::toNativeSeparators(log_path + QDir::separator() + "scsynth.log");
 
-    init_script_path       = QDir::toNativeSeparators(root_path + "/app/server/ruby/bin/init-script.rb");
-    exit_script_path       = QDir::toNativeSeparators(root_path + "/app/server/ruby/bin/exit-script.rb");
+    init_script_path       = QDir::toNativeSeparators(root_path + "/usr/lib/sonic-pi/server/bin/init-script.rb");
+    exit_script_path       = QDir::toNativeSeparators(root_path + "/usr/lib/sonic-pi/server/bin/exit-script.rb");
 
-    qt_app_theme_path      = QDir::toNativeSeparators(root_path + "/app/gui/qt/theme/app.qss");
+    qt_app_theme_path      = QDir::toNativeSeparators(root_path + "/usr/share/sonic-pi/theme/app.qss");
 
-    qt_browser_dark_css    = QDir::toNativeSeparators(root_path + "/app/gui/qt/theme/dark/doc-styles.css");
-    qt_browser_light_css   = QDir::toNativeSeparators(root_path + "/app/gui/qt/theme/light/doc-styles.css");
-    qt_browser_hc_css      = QDir::toNativeSeparators(root_path + "/app/gui/qt/theme/high_contrast/doc-styles.css");
+    qt_browser_dark_css    = QDir::toNativeSeparators(root_path + "/usr/share/sonic-pi/theme/dark/doc-styles.css");
+    qt_browser_light_css   = QDir::toNativeSeparators(root_path + "/usr/share/sonic-pi/theme/light/doc-styles.css");
+    qt_browser_hc_css      = QDir::toNativeSeparators(root_path + "/usr/share/sonic-pi/theme/high_contrast/doc-styles.css");
 
     // attempt to create log directory
     QDir logDir(log_path);
@@ -1056,7 +1056,7 @@
     return QCoreApplication::applicationDirPath() + "/../../../../..";
 #else
     // On linux, CMake builds app into the build folder
-    return QCoreApplication::applicationDirPath() + "/../../../..";
+    return QCoreApplication::applicationDirPath() + "/../..";
 #endif
 }
 
@@ -1098,7 +1098,7 @@
     // Register server pid for potential zombie clearing
     QStringList regServerArgs;
 #if QT_VERSION >= QT_VERSION_CHECK(5, 3, 0)
-    regServerArgs << QDir::toNativeSeparators(rootPath() + "/app/server/ruby/bin/task-register.rb")<< QString::number(serverProcess->processId());
+    regServerArgs << QDir::toNativeSeparators(rootPath() + "/usr/lib/sonic-pi/server/bin/task-register.rb")<< QString::number(serverProcess->processId());
 #endif
     QProcess *regServerProcess = new QProcess();
     regServerProcess->start(ruby_path, regServerArgs);
diff -ruN a/app/gui/qt/model/sonicpitheme.cpp b/app/gui/qt/model/sonicpitheme.cpp
--- a/app/gui/qt/model/sonicpitheme.cpp	2020-02-25 01:23:45.000000000 +0100
+++ b/app/gui/qt/model/sonicpitheme.cpp	2020-03-08 19:59:00.840045897 +0100
@@ -25,11 +25,11 @@
     this->customSettingsFilename = customSettingsFilename;
     this->rootPath = rootPath;
 
-    qt_app_theme_path      = QDir::toNativeSeparators(rootPath + "/app/gui/qt/theme/app.qss");
+    qt_app_theme_path      = QDir::toNativeSeparators(rootPath + "/usr/share/sonic-pi/theme/app.qss");
 
-    qt_browser_dark_css    = QDir::toNativeSeparators(rootPath + "/app/gui/qt/theme/dark/doc-styles.css");
-    qt_browser_light_css   = QDir::toNativeSeparators(rootPath + "/app/gui/qt/theme/light/doc-styles.css");
-    qt_browser_hc_css      = QDir::toNativeSeparators(rootPath + "/app/gui/qt/theme/high_contrast/doc-styles.css");
+    qt_browser_dark_css    = QDir::toNativeSeparators(rootPath + "/usr/share/sonic-pi/theme/dark/doc-styles.css");
+    qt_browser_light_css   = QDir::toNativeSeparators(rootPath + "/usr/share/sonic-pi/theme/light/doc-styles.css");
+    qt_browser_hc_css      = QDir::toNativeSeparators(rootPath + "/usr/share/sonic-pi/theme/high_contrast/doc-styles.css");
 
     loadToolBarIcons();
 
