diff -aur old/configure.in new/configure.in
--- old/configure.in	2011-09-29 20:38:56.725672831 +1000
+++ new/configure.in	2011-09-29 20:39:07.975658829 +1000
@@ -2810,11 +2810,13 @@
 then
   SAVE_CFLAGS="$CFLAGS"
   SAVE_LDFLAGS="$LDFLAGS"
+  SAVE_LIBS="$LIBS"
 dnl ARCHFLAGS="" -> disable multi -arch on OSX (see Config_heavy.pl:fetch_string)
   PERL_CFLAGS=`ARCHFLAGS="" $perl_interpreter -MExtUtils::Embed -e ccopts`
   PERL_LDFLAGS=`ARCHFLAGS="" $perl_interpreter -MExtUtils::Embed -e ldopts`
   CFLAGS="$CFLAGS $PERL_CFLAGS"
   LDFLAGS="$LDFLAGS $PERL_LDFLAGS"
+  LIBS="$LIBS -L/usr/lib/perl5/core_perl/CORE -lperl $PERL_LDFLAGS"
 
   AC_CACHE_CHECK([for libperl],
     [c_cv_have_libperl],
@@ -2833,7 +2835,10 @@
 			 Nullsv);
       ]]),
       [c_cv_have_libperl="yes"],
-      [c_cv_have_libperl="no"]
+      [ 
+        c_cv_have_libperl="no" 
+        LIBS="$SAVE_LIBS"
+      ]
     )
   )
 
