diff -uprw tea-43.1.0.orig/spellchecker.h tea-43.1.0/spellchecker.h
--- tea-43.1.0.orig/spellchecker.h	2016-09-16 22:56:10.000000000 +0300
+++ tea-43.1.0/spellchecker.h	2016-11-27 16:39:56.471555601 +0200
@@ -100,7 +100,7 @@ public:
   QString user_dir;
   QString lng;
 
-  char *encoding;
+  const char *encoding;
   QString dict_dir;
   
   CHunspellChecker (const QString &lang, const QString &path = "", const QString &user_path = "");
