diff -Naur sugar-0.110.0.orig/src/jarabe/main.py sugar-0.110.0/src/jarabe/main.py
--- sugar-0.110.0.orig/src/jarabe/main.py	2016-10-07 12:46:55.000000000 +0200
+++ sugar-0.110.0/src/jarabe/main.py	2017-03-30 19:35:47.528931067 +0200
@@ -186,11 +186,8 @@
 def _start_window_manager():
     global _metacity_process
 
-    settings = Gio.Settings.new('org.gnome.desktop.interface')
-    settings.set_string('cursor-theme', 'sugar')
-
     _metacity_process = subprocess.Popen(
-        ['metacity', '--no-force-fullscreen', '--no-composite'])
+        ['metacity', '--no-force-fullscreen'])
 
     screen = Wnck.Screen.get_default()
     screen.connect('window-manager-changed', __window_manager_changed_cb)
