From 57b9698ce56a78c783b5535910e2db81c0cc9e2c Mon Sep 17 00:00:00 2001
From: "Daniel M. Capella" <polyzen@archlinux.org>
Date: Sat, 29 Feb 2020 16:10:50 -0500
Subject: [PATCH] Fix test_datestimes for PyYAML >=5.3

---
 test/test.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/test/test.py b/test/test.py
index 12fa806..8897196 100755
--- a/test/test.py
+++ b/test/test.py
@@ -3,7 +3,7 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-import os, sys, unittest, tempfile, json, io, platform, subprocess
+import os, sys, unittest, tempfile, json, io, platform, subprocess, yaml
 
 sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
 from yq import yq, cli  # noqa
@@ -116,7 +116,10 @@ class TestYq(unittest.TestCase):
 
     def test_datetimes(self):
         self.assertEqual(self.run_yq("- 2016-12-20T22:07:36Z\n", ["."]), "")
-        self.assertEqual(self.run_yq("- 2016-12-20T22:07:36Z\n", ["-y", "."]), "- '2016-12-20T22:07:36'\n")
+        if yaml.__version__ < '5.3':
+            self.assertEqual(self.run_yq("- 2016-12-20T22:07:36Z\n", ["-y", "."]), "- '2016-12-20T22:07:36'\n")
+        else:
+            self.assertEqual(self.run_yq("- 2016-12-20T22:07:36Z\n", ["-y", "."]), "- '2016-12-20T22:07:36+00:00'\n")
 
         self.assertEqual(self.run_yq("2016-12-20", ["."]), "")
         self.assertEqual(self.run_yq("2016-12-20", ["-y", "."]), "'2016-12-20'\n")
-- 
2.25.1

