From 8665ff5dc95900798bd8e935c61a10324115579a Mon Sep 17 00:00:00 2001
From: Konstantin Shalygin <k0ste@k0ste.ru>
Date: Mon, 18 Nov 2019 11:28:53 +0700
Subject: [PATCH] s3cmd: python 3.8 compat

---
 s3cmd | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/s3cmd b/s3cmd
index cef1e9c5..3e18af1b 100755
--- a/s3cmd
+++ b/s3cmd
@@ -317,9 +317,9 @@ def cmd_expiration_set(args):
             raise ParameterError("Expecting S3 URI with just the bucket name set instead of '%s'" % arg)
         try:
             response = s3.expiration_set(uri, cfg.bucket_location)
-            if response["status"] is 200:
+            if response["status"] == 200:
                 output(u"Bucket '%s': expiration configuration is set." % (uri.uri()))
-            elif response["status"] is 204:
+            elif response["status"] == 204:
                 output(u"Bucket '%s': expiration configuration is deleted." % (uri.uri()))
         except S3Error as e:
             if e.info["Code"] in S3.codes:
