diff --git a/aws_xray_sdk/ext/pynamodb/patch.py b/aws_xray_sdk/ext/pynamodb/patch.py
index 8815cc3..a5e462f 100644
--- a/aws_xray_sdk/ext/pynamodb/patch.py
+++ b/aws_xray_sdk/ext/pynamodb/patch.py
@@ -16,8 +16,8 @@ def patch():
     setattr(botocore.vendored.requests.sessions, '_xray_enabled', True)
 
     wrapt.wrap_function_wrapper(
-        'botocore.vendored.requests.sessions',
-        'Session.send',
+        'botocore.httpsession',
+        'URLLib3Session.send',
         _xray_traced_pynamodb,
     )
 
@@ -59,7 +59,7 @@ def pynamodb_meta_processor(wrapped, instance, args, kwargs, return_value,
         subsegment.add_error_flag()
         subsegment.add_exception(exception, stack, True)
 
-    resp = return_value.json() if return_value else None
+    resp = json.loads(return_value.text) if return_value else None
     _extract_whitelisted_params(subsegment.name, operation_name, aws_meta,
                                 [None, json.loads(args[0].body.decode('utf-8'))],
                                 None, resp)
