diff -wbBur awstats-7.8/wwwroot/cgi-bin/lib/mime.pm awstats-7.8.my/wwwroot/cgi-bin/lib/mime.pm
--- awstats-7.8/wwwroot/cgi-bin/lib/mime.pm	2020-04-16 12:18:35.000000000 +0300
+++ awstats-7.8.my/wwwroot/cgi-bin/lib/mime.pm	2020-04-18 19:27:04.242408222 +0300
@@ -56,7 +56,7 @@
 'diskimage', 'Disc and media file extensions',
 'vm',        'Virtual Machine image',
 'torrent',   'BitTorrent File',
-'gis',       'GIS File'
+'gis',       'GIS File',
 'ebook',     'Ebook File'
 );
 
