From aad41807c365ccef001b787407d7dc756e11de02 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gabriel=20F=C3=A9ron?= <g@leirbag.net>
Date: Tue, 11 Jan 2022 20:28:41 +0100
Subject: [PATCH] Remove unused import of sentry_sdk

---
 cura/CrashHandler.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/cura/CrashHandler.py b/cura/CrashHandler.py
index c911a423504..35e9aa832be 100644
--- a/cura/CrashHandler.py
+++ b/cura/CrashHandler.py
@@ -12,8 +12,6 @@
 import locale
 from typing import cast, Any
 
-import sentry_sdk
-
 try:
     from sentry_sdk.hub import Hub
     from sentry_sdk.utils import event_from_exception
