=== modified file 'zim/www.py'
--- zim/www.py	2014-07-21 20:03:15 +0000
+++ zim/www.py	2014-08-21 09:06:34 +0000
@@ -119,6 +119,10 @@
 		self.config = config or ConfigManager(profile=notebook.profile)
 
 		self.output = None
+
+		if template is None:
+			template = 'Default'
+
 		if isinstance(template, basestring):
 			from zim.templates import get_template
 			self.template = get_template('html', template)

