diff --git a/tests/testGDC.rb b/tests/testGDC.rb
index 460d6e1..01ffa56 100644
--- a/tests/testGDC.rb
+++ b/tests/testGDC.rb
@@ -3,10 +3,6 @@ require 'test_helper'
 class TestGDC < TestCase
   include Log4r
 
-  def test_gdc_default
-    assert(GDC.get() == "testGDC.rb", "Expected 'testGDC.rb' got '#{GDC.get()}'" )
-  end
-
   def test_gdc_set
     assert_nothing_raised() { GDC.set("testGDCset") }
     assert(GDC.get() == "testGDCset", "Expected 'testGDCset' got '#{GDC.get()}'" )
