diff -upr a/cheroot.egg-info/requires.txt b/cheroot.egg-info/requires.txt
--- a/cheroot.egg-info/requires.txt	2020-07-13 16:44:02.000000000 +0200
+++ b/cheroot.egg-info/requires.txt	2020-07-13 18:54:35.707376340 +0200
@@ -18,7 +18,6 @@ sphinxcontrib-spelling>=4.3.0
 pytest>=4.6.6
 pytest-mock>=1.11.0
 pytest-sugar>=0.9.3
-pytest-testmon<1.0.0
 pytest-watch==4.2.0
 pytest-xdist>=1.28.0
 jaraco.text>=3.1
diff -upr a/pytest.ini b/pytest.ini
--- a/pytest.ini	2020-07-13 16:43:42.000000000 +0200
+++ b/pytest.ini	2020-07-13 18:55:10.148315625 +0200
@@ -3,10 +3,6 @@ addopts =
     # `pytest-xdist`:
     --numprocesses=auto
 
-    # `pytest-mon`:
-    # useful for live testing with `pytest-watch` during development:
-    --testmon
-
     # show 10 slowest invocations:
     --durations=10
 
diff -upr a/setup.cfg b/setup.cfg
--- a/setup.cfg	2020-07-13 16:44:03.000000000 +0200
+++ b/setup.cfg	2020-07-13 18:54:42.334223734 +0200
@@ -79,7 +79,6 @@ testing =
 	pytest>=4.6.6
 	pytest-mock>=1.11.0
 	pytest-sugar>=0.9.3
-	pytest-testmon<1.0.0
 	pytest-watch==4.2.0
 	pytest-xdist>=1.28.0
 	
diff -upr a/tox.ini b/tox.ini
--- a/tox.ini	2020-07-13 16:43:42.000000000 +0200
+++ b/tox.ini	2020-07-13 18:54:52.601170405 +0200
@@ -10,7 +10,7 @@ deps =
     pip>=19.1
     wheel
 commands =
-    pytest --testmon-off {posargs}
+    pytest {posargs}
     codecov -f coverage.xml -X gcov
 usedevelop = True
 extras = testing
