--- a/wiki_lite.py
+++ b/wiki_lite.py
@@ -98,7 +98,7 @@ def remove_tags(html):
     flags = re.DOTALL | re.MULTILINE
     #body = re.compile('^<body class.*?^NewPP limit report$', flags)
     #body = re.compile('^<body class="mediawiki.*?"site":"loading","user":"ready","user.groups":"ready"', flags)
-    body = re.compile('^<body class="mediawiki.*?<li id="footer-info-copyright">', flags)
+    body = re.compile('^<body .*?<li id="footer-info-copyright">', flags)
     html = body.findall(html)[0]
     html = '\n'.join(html.split('\n')[:-1])
     toc  = re.compile('^<div class="toc noprint" style="text-align: center; margin-bottom: 1em">.*?</div>$', flags)
