diff --git a/config/cloud.cfg.d/05_logging.cfg b/config/cloud.cfg.d/05_logging.cfg
index 2e18073..e2ccd78 100644
--- a/config/cloud.cfg.d/05_logging.cfg
+++ b/config/cloud.cfg.d/05_logging.cfg
@@ -38,25 +38,18 @@ _log:
    format=%(asctime)s - %(filename)s[%(levelname)s]: %(message)s
    
    [formatter_simpleFormatter]
-   format=[CLOUDINIT] %(filename)s[%(levelname)s]: %(message)s
- - &log_file |
-   [handler_cloudLogHandler]
-   class=FileHandler
-   level=DEBUG
-   formatter=arg0Formatter
-   args=('/var/log/cloud-init.log',)
+   format=%(levelname)s %(filename)s: %(message)s
+
  - &log_syslog |
    [handler_cloudLogHandler]
    class=handlers.SysLogHandler
-   level=DEBUG
+   level=INFO
    formatter=simpleFormatter
-   args=("/dev/log", handlers.SysLogHandler.LOG_USER)
+   args=("/dev/log", handlers.SysLogHandler.LOG_DAEMON)
 
 log_cfgs:
 # These will be joined into a string that defines the configuration
  - [ *log_base, *log_syslog ]
-# These will be joined into a string that defines the configuration
- - [ *log_base, *log_file ]
 # A file path can also be used
 # - /etc/log.conf
 

