--- src/parallel.pod.orig
+++ src/parallel.pod
@@ -528,18 +528,18 @@
 
 =over 3
 
-=item 0
+=item 0Z<>
 
 Do not halt if a job fails. Exit status will be the number of jobs
 failed. This is the default.
 
-=item 1
+=item 1Z<>
 
 Do not start new jobs if a job fails, but complete the running jobs
 including cleanup. The exit status will be the exit status from the
 last failing job.
 
-=item 2
+=item 2Z<>
 
 Kill off all jobs immediately and exit without cleanup. The exit
 status will be the exit status from the failing job.
@@ -2760,7 +2760,7 @@
 
 =over 6
 
-=item 0
+=item 0Z<>
 
 All jobs ran without error.
 
@@ -2768,11 +2768,11 @@
 
 Some of the jobs failed. The exit status gives the number of failed jobs
 
-=item 254
+=item 254Z<>
 
 More than 253 jobs failed.
 
-=item 255
+=item 255Z<>
 
 Other error.
 
