commit dbd1d58ba6c350db3aa7d154e9a940d835ee52a4
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Sat Oct 10 09:34:35 2015 +0800

    Fix test_copyUnicode under PyPy

diff --git a/pyperclip/__init__.py b/pyperclip/__init__.py
index ab2b6df..1c3ec66 100644
--- a/pyperclip/__init__.py
+++ b/pyperclip/__init__.py
@@ -113,6 +113,8 @@ def _copyQt(text):
 
 
 def _copyXclip(text):
+    if not isinstance(text, text_type):
+        text = text.decode('utf-8')
     p = Popen(['xclip', '-selection', 'c'], stdin=PIPE, close_fds=True)
     p.communicate(input=text.encode('utf-8'))
 
@@ -124,6 +126,8 @@ def _pasteXclip():
 
 
 def _copyXsel(text):
+    if not isinstance(text, text_type):
+        text = text.decode('utf-8')
     p = Popen(['xsel', '-b', '-i'], stdin=PIPE, close_fds=True)
     p.communicate(input=text.encode('utf-8'))
 
@@ -134,6 +138,8 @@ def _pasteXsel():
     return stdout.decode('utf-8')
 
 def _copyKlipper(text):
+    if not isinstance(text, text_type):
+        text = text.decode('utf-8')
     p = Popen(['qdbus', 'org.kde.klipper', '/klipper',
             'setClipboardContents', text.encode('utf-8')],
              stdin=PIPE, close_fds=True)
