From e971e91fe82b90ba2931ba3be8cf704dd0e016fe Mon Sep 17 00:00:00 2001
From: Gabriel Burt <gabriel.burt@gmail.com>
Date: Thu, 28 Oct 2010 15:11:13 +0000
Subject: [build] Remove 'dnl' line form build.rules.mk (bgo#633333)

---
(limited to 'build/build.rules.mk')

diff --git a/build/build.rules.mk b/build/build.rules.mk
index 648a60d..31291ba 100644
--- a/build/build.rules.mk
+++ b/build/build.rules.mk
@@ -2,7 +2,6 @@ UNIQUE_FILTER_PIPE = tr [:space:] \\n | sort | uniq
 BUILD_DATA_DIR = $(top_builddir)/bin/share/$(PACKAGE)
 
 SOURCES_BUILD = $(addprefix $(srcdir)/, $(SOURCES))
-dnl SOURCES_BUILD += $(top_srcdir)/src/AssemblyInfo.cs
 
 RESOURCES_EXPANDED = $(addprefix $(srcdir)/, $(RESOURCES))
 RESOURCES_BUILD = $(foreach resource, $(RESOURCES_EXPANDED), \
--
cgit v0.9.0.2
