--- a/tests/test_core/test_request_mocking.py	2023-04-25 01:11:07.284562629 +0800
+++ b/tests/test_core/test_request_mocking.py	2023-04-25 01:10:53.118267009 +0800
@@ -13,7 +13,7 @@
     conn.create_queue(QueueName="queue1")
 
     res = requests.get("https://google.com/")
-    assert res.status_code < 400
+    assert res.status_code < 400 or res.status_code == 429
 
 
 if not settings.TEST_SERVER_MODE:
