diff -Naur SparkleShare-325bc848afe460911d596eff1e09a4d39cd0b638.orig/SparkleShare/Linux/UserInterface.cs SparkleShare-325bc848afe460911d596eff1e09a4d39cd0b638/SparkleShare/Linux/UserInterface.cs
--- SparkleShare-325bc848afe460911d596eff1e09a4d39cd0b638.orig/SparkleShare/Linux/UserInterface.cs	2016-12-16 19:27:32.000000000 +0100
+++ SparkleShare-325bc848afe460911d596eff1e09a4d39cd0b638/SparkleShare/Linux/UserInterface.cs	2017-01-21 18:07:36.298740982 +0100
@@ -62,7 +62,7 @@
 
         public void Run ()
         {   
-            (application as GLib.Application).Run ("org.sparkleshare.SparkleShare", new string [0]);
+            (application as GLib.Application).Run (0, null);
         }
 
 
