--- a/cherrypy/lib/sessions.py	2018-09-02 01:11:12.000000000 +0200
+++ b/cherrypy/lib/sessions.py	2018-09-04 15:24:32.137916798 +0200
@@ -109,7 +109,7 @@ import binascii
 
 import six
 from six.moves import cPickle as pickle
-import contextlib2
+import contextlib
 
 import zc.lockfile
 
@@ -572,7 +572,7 @@ class FileSession(Session):
     def release_lock(self, path=None):
         """Release the lock on the currently-loaded session data."""
         self.lock.close()
-        with contextlib2.suppress(FileNotFoundError):
+        with contextlib.suppress(FileNotFoundError):
             os.remove(self.lock._path)
         self.locked = False
 
--- a/setup.py	2018-09-02 01:11:12.000000000 +0200
+++ b/setup.py	2018-09-04 15:24:42.964449629 +0200
@@ -63,7 +63,6 @@ params = dict(
         'portend>=2.1.1',
         'more_itertools',
         'zc.lockfile',
-        'contextlib2',
     ],
     extras_require={
         'docs': [
