diff --git i/cdemu-client/CMakeLists.txt w/cdemu-client/CMakeLists.txt
index cdcff25232886440..5296229538cd64bf 100644
--- i/cdemu-client/CMakeLists.txt
+++ w/cdemu-client/CMakeLists.txt
@@ -38,7 +38,8 @@ install (
 
 install (
     FILES data/cdemu-bash-completion.sh
-    DESTINATION ${CMAKE_INSTALL_SYSCONFDIR}/bash_completion.d
+    DESTINATION ${CMAKE_INSTALL_DATADIR}/bash-completion/completions
+    RENAME cdemu
 )
 
 install (
diff --git i/cdemu-client/src/cdemu w/cdemu-client/src/cdemu
index 85bdcc5f30cdffef..a3fa132492266a7a 100755
--- i/cdemu-client/src/cdemu
+++ w/cdemu-client/src/cdemu
@@ -116,7 +116,7 @@ class cmd_load (object):
 
     def __call__ (self, proxy, arguments):
         # We need to pass absolute filenames to daemon
-        filenames = map(os.path.abspath, arguments.image_file)
+        filenames = [os.path.abspath(f) for f in arguments.image_file]
 
         # Gather parser parameters into a dictionary
         parser_params = {}
