diff --color -aur lesspipe-1.91-old/lesspipe.sh lesspipe-1.91-new/lesspipe.sh
--- lesspipe-1.91-old/lesspipe.sh	2021-11-10 20:11:39.000000000 +0200
+++ lesspipe-1.91-new/lesspipe.sh	2021-11-21 04:34:21.400743376 +0200
@@ -531,9 +531,9 @@
     msg "No suitable tool for HTML parsing found, install one of html2text, elinks, links, lynx or w3m"
     return
   elif cmd_exist html2text; then
-    if [[ "$1" = - ]]; then html2text -utf8 2>/dev/null ||
+    if [[ "$1" = - ]]; then html2text - utf8 2>/dev/null ||
       html2text -from_encoding utf-8
-    else html2text -utf8 "$1" 2>/dev/null ||
+    else html2text "$1" utf8 2>/dev/null ||
       html2text -from_encoding utf-8 "$1"; fi
   elif cmd_exist lynx; then
     if [[ "$1" = - ]]; then set - -stdin; fi
