--- a/tests/test_10_exceptions.py
+++ b/tests/test_10_exceptions.py
@@ -230,11 +230,11 @@ class TestPGPKey(object):
         rsa_pub.subkeys['EEE097A017B979CA'].encrypt(PGPMessage.new('asdf'),
                                                     cipher=SymmetricKeyAlgorithm.CAST5)
 
-        w = recwarn.pop(UserWarning)
+        w = recwarn.list[-2]
         assert str(w.message) == "Selected symmetric algorithm not in key preferences"
         assert w.filename == __file__
 
-        w = recwarn.pop(UserWarning)
+        w = recwarn.list[-1]
         assert str(w.message) == "Selected compression algorithm not in key preferences"
         assert w.filename == __file__

