diff -aur gramps-4.2.0/setup.py gramps-4.2.0.new/setup.py
--- gramps-4.2.0/setup.py	2015-07-29 20:53:06.000000000 +0200
+++ gramps-4.2.0.new/setup.py	2015-09-20 17:58:17.035215951 +0200
@@ -91,7 +91,7 @@
         except:
             return (0,0,0)
     else:
-        cmd = 'intltool-update --version | head -1 | cut -d" " -f3'
+        cmd = 'intltool-update --version 2> /dev/null | head -1 | cut -d" " -f3'
         retcode, version_str = subprocess.getstatusoutput(cmd)
         if retcode != 0:
             return None
