diff --git a/master/buildbot/test/__init__.py b/master/buildbot/test/__init__.py
index f7ba7b1e7..6d4b86eab 100644
--- a/master/buildbot/test/__init__.py
+++ b/master/buildbot/test/__init__.py
@@ -131,3 +131,7 @@ warnings.filterwarnings('ignore', ".*Not importing directory .*/zope: missing __
                         category=ImportWarning)
 warnings.filterwarnings('ignore', ".*Not importing directory .*/sphinxcontrib: missing __init__",
                         category=ImportWarning)
+
+# ignore warnings from setuptools 49.2.0
+warnings.filterwarnings('ignore', ".*Distutils was imported before Setuptools.*",
+                        category=UserWarning)
