From 968e8956110c6f0ad089ac3c8b11948ef0c1d023 Mon Sep 17 00:00:00 2001
From: Maxime Gauduin <alucryd@archlinux.org>
Date: Fri, 14 Jul 2017 14:54:56 +0200
Subject: [PATCH] Customize libexecdir

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 76c186b..2a0acf6 100644
--- a/meson.build
+++ b/meson.build
@@ -11,7 +11,7 @@ dependency('gtk+-3.0', version :'>=3.14')
 
 python_dir = join_paths(get_option('prefix'), python.sysconfig_path('purelib'))
 service_dir = join_paths(get_option('datadir'), 'dbus-1', 'services')
-libexec_dir = join_paths(get_option('prefix'), 'libexec')
+libexec_dir = join_paths(get_option('prefix'), get_option('libexecdir'))
 
 conf = configuration_data()
 conf.set('pythondir', python_dir)
