diff -wbBur Term-Animation-2.4/lib/Term/Animation.pm Term-Animation-2.4.my/lib/Term/Animation.pm
--- Term-Animation-2.4/lib/Term/Animation.pm	2006-12-17 06:29:51.000000000 +0000
+++ Term-Animation-2.4.my/lib/Term/Animation.pm	2011-03-24 21:42:47.000000000 +0000
@@ -136,7 +136,7 @@
 		}
 		$self->{FULLSCREEN} = 0;
 	} else {
-		$self->{WIN} = new Curses;
+		$self->{WIN} = Curses::initscr();
 		noecho();
 		curs_set(0);
 		$self->{FULLSCREEN} = 1;
