diff -ruN a/config/config.sample.php b/config/config.sample.php
--- a/config/config.sample.php	2021-02-19 09:49:36.000000000 +0100
+++ b/config/config.sample.php	2021-02-20 13:29:19.194514157 +0100
@@ -83,7 +83,7 @@
  *
  * Default to ``data/`` in the Nextcloud directory.
  */
-'datadirectory' => '/var/www/nextcloud/data',
+'datadirectory' => '/var/lib/nextcloud/data',
 
 /**
  * The current version number of your Nextcloud installation. This is set up
@@ -831,7 +831,7 @@
  *
  * Defaults to ``[datadirectory]/nextcloud.log``
  */
-'logfile' => '/var/log/nextcloud.log',
+'logfile' => '/var/log/nextcloud/nextcloud.log',
 
 /**
  * Log file mode for the Nextcloud loggin type in octal notation.
@@ -968,8 +968,13 @@
  */
 'apps_paths' => [
 	[
-		'path'=> '/var/www/nextcloud/apps',
+		'path'=> '/usr/share/webapps/nextcloud/apps',
 		'url' => '/apps',
+		'writable' => false,
+	],
+	[
+		'path'=> '/var/lib/nextcloud/apps',
+		'url' => '/wapps',
 		'writable' => true,
 	],
 ],
