diff -urN pandas-datareader-0.6.0.orig/pandas_datareader/compat/__init__.py pandas-datareader-0.6.0/pandas_datareader/compat/__init__.py
--- pandas-datareader-0.6.0.orig/pandas_datareader/compat/__init__.py	2018-08-11 19:23:13.526123366 +0200
+++ pandas-datareader-0.6.0/pandas_datareader/compat/__init__.py	2018-08-11 19:24:01.953342399 +0200
@@ -14,6 +14,7 @@
 PANDAS_0190 = (PANDAS_VERSION >= LooseVersion('0.19.0'))
 PANDAS_0200 = (PANDAS_VERSION >= LooseVersion('0.20.0'))
 PANDAS_0210 = (PANDAS_VERSION >= LooseVersion('0.21.0'))
+PANDAS_0230 = (PANDAS_VERSION >= LooseVersion('0.23.0'))
 
 if PANDAS_0190:
     from pandas.api.types import is_number
@@ -37,6 +38,12 @@
     from pandas.util.testing import assertRaisesRegexp as assert_raises_regex
     get_filepath_or_buffer = com.get_filepath_or_buffer
 
+if PANDAS_0230:
+    from pandas.core.dtypes.common import is_list_like
+else:
+    from pandas.core.common import is_list_like
+
+
 if compat.PY3:
     from urllib.error import HTTPError
 else:
diff -urN pandas-datareader-0.6.0.orig/pandas_datareader/fred.py pandas-datareader-0.6.0/pandas_datareader/fred.py
--- pandas-datareader-0.6.0.orig/pandas_datareader/fred.py	2018-08-11 19:23:13.529456737 +0200
+++ pandas-datareader-0.6.0/pandas_datareader/fred.py	2018-08-11 19:25:30.284351849 +0200
@@ -1,4 +1,5 @@
-from pandas.core.common import is_list_like
+from pandas_datareader.compat import is_list_like
+
 from pandas import concat, read_csv
 
 from pandas_datareader.base import _BaseReader
