diff -wbBur phpvirtualbox-5.0-4/endpoints/api.php phpvirtualbox-5.0-4.q/endpoints/api.php
--- phpvirtualbox-5.0-4/endpoints/api.php	2015-12-07 16:49:21.000000000 +0300
+++ phpvirtualbox-5.0-4.q/endpoints/api.php	2016-01-03 23:28:42.853589422 +0300
@@ -313,7 +313,7 @@
 			/*
 			 * Call to vboxconnector
 			 */
-			$vbox->$request['fn']($request['params'],array(&$response));
+			$vbox->{$request['fn']}($request['params'],array(&$response));
 
 
 			/*
diff -wbBur phpvirtualbox-5.0-4/endpoints/lib/auth/Builtin.php phpvirtualbox-5.0-4.q/endpoints/lib/auth/Builtin.php
--- phpvirtualbox-5.0-4/endpoints/lib/auth/Builtin.php	2015-12-07 16:49:21.000000000 +0300
+++ phpvirtualbox-5.0-4.q/endpoints/lib/auth/Builtin.php	2016-01-03 23:29:06.016923275 +0300
@@ -167,7 +167,7 @@
 		global $_SESSION;
 
 		// Must be an admin
-		if(!$_SESSION['admin']) break;
+		if(!$_SESSION['admin']) return;
 
 		// Use main / auth server
 		$vbox = new vboxconnector(true);
@@ -175,7 +175,7 @@
 
 		// See if it exists
 		if(!$skipExistCheck && $vbox->vbox->getExtraData('phpvb/users/'.$vboxRequest['u'].'/pass'))
-			break;
+			return;
 
 		if($vboxRequest['p'])
 			$vbox->vbox->setExtraData('phpvb/users/'.$vboxRequest['u'].'/pass', hash('sha512', $vboxRequest['p']));
