diff -wbBur src/webnews/MIME_Message.php src.my/webnews/MIME_Message.php
--- src/webnews/MIME_Message.php	2004-09-20 15:58:04.000000000 +0400
+++ src.my/webnews/MIME_Message.php	2016-02-16 18:02:11.415650635 +0300
@@ -64,7 +64,7 @@
 		function decode_header($headers) {
 			$header_want = "/^(From|Subject|Date|Newsgroups|References|Message-ID|Content-Type|Content-Transfer-Encoding|Content-Disposition|Content-ID): (.*$)/i";
 			
-			$headers = split("\r\n", $headers);
+			$headers = explode("\r\n", $headers);
 
 			// Parse the header
 			for ($line_count = 0; $line_count < sizeof($headers);$line_count++) {
@@ -88,7 +88,7 @@
 
 		// An article is a raw MIME message
 		function decode_article($article) {
-			list($header, $body) = split("\r\n\r\n", $article, 2);
+			list($header, $body) = explode("\r\n\r\n", $article, 2);
 						
 			$header = $this->decode_header($header);
 
diff -wbBur src/webnews/util.php src.my/webnews/util.php
--- src/webnews/util.php	2006-03-28 21:59:23.000000000 +0400
+++ src.my/webnews/util.php	2016-02-16 18:02:11.415650635 +0300
@@ -46,7 +46,10 @@
 			$str = $matches[1].quoted_printable_decode($str).$matches[3];
 		}
 		while (preg_match("/=\?.*\?b\?.*\?=/i", $str)) {
-			$str = preg_replace("/(.*)=\?.*\?b\?(.*)\?=(.*)/ie", "'$1'.base64_decode('$2').'$3'", $str);
+			$str = preg_replace_callback("/(.*)=\?.*\?b\?(.*)\?=(.*)/i",
+				function ($matches) {
+					return $matches[1].base64_decode($matches[2]).$matches[3];
+				}, $str);
 		}
 
 		return $str;
@@ -403,7 +406,7 @@
 						$ini[$section_name] = array();
 					}
 				} else if (strpos($buf, "=") !== FALSE) {
-					list($key, $value) = split("=", $buf, 2);
+					list($key, $value) = explode("=", $buf, 2);
 					$value = preg_replace("/^(['|\"])?(.*?)(?(1)['|\"])$/", "\\2", trim($value));
 
 					if ((strlen($key) != 0) && (strlen($value) != 0)) {					
