diff --git a/SoftLayer/API.py b/SoftLayer/API.py
index 3fbab72b..913752ce 100644
--- a/SoftLayer/API.py
+++ b/SoftLayer/API.py
@@ -298,7 +298,7 @@ class BaseClient(object):
                     results = transports.SoftLayerListResult(results, len(results))
                 else:
                     yield results
-                    raise StopIteration
+                    return
 
             for item in results:
                 yield item
@@ -313,7 +313,7 @@ class BaseClient(object):
 
             offset += limit
 
-        raise StopIteration
+        return
 
     def __repr__(self):
         return "Client(transport=%r, auth=%r)" % (self.transport, self.auth)
