diff --git a/src/views/window.py b/src/views/window.py
index 3aee43a..0870ed0 100644
--- a/src/views/window.py
+++ b/src/views/window.py
@@ -174,7 +174,6 @@ class Window(QQuickView):
     def setCursorVisible(self, visible):
         self.setCursor(QCursor(Qt.ArrowCursor if visible else Qt.BlankCursor))
 
-    @pyqtSlot("QVariant")
     def focusWindowChangedSlot(self, win):
         if not win: self.rootObject().hideTransientWindows()
 
