diff -upr h11-0.8.1.orig/h11/tests/test_events.py h11-0.8.1/h11/tests/test_events.py
--- h11-0.8.1.orig/h11/tests/test_events.py	2017-03-18 23:56:02.000000000 +0200
+++ h11-0.8.1/h11/tests/test_events.py	2019-11-01 10:17:40.211258036 +0200
@@ -43,7 +43,7 @@ def test_event_bundle():
     with pytest.raises(TypeError) as exc:
         T(b=0)
     # make sure we error on the right missing kwarg
-    assert 'kwarg a' in str(exc)
+    assert 'kwarg a' in str(exc.value)
 
     # _validate is called
     with pytest.raises(ValueError):
