From ef3e835125ab53157b45dbd6e95a5b104a3d4610 Mon Sep 17 00:00:00 2001
From: John Vandenberg <jayvdb@gmail.com>
Date: Mon, 27 Dec 2021 06:33:03 +0800
Subject: [PATCH 2/4] setup.py: Allow Django 4.0

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 51f3600..c6c27e4 100755
--- a/setup.py
+++ b/setup.py
@@ -38,7 +38,7 @@ setup(
     packages=find_packages(),
     include_package_data=True,
     install_requires=[
-        'django>=2.2,<3.3',
+        'django>=2.2,<4.1',
         'mailmanclient>=3.3.3',
         'django-allauth',
         'django-gravatar2 >= 1.0.6',
-- 
GitLab


From 6cfc4c87bc445592a1a020ee877bd8fead3d8e39 Mon Sep 17 00:00:00 2001
From: John Vandenberg <jayvdb@gmail.com>
Date: Mon, 27 Dec 2021 06:33:03 +0800
Subject: [PATCH 3/4] signals.py: Fix Django 4.0 breakage

---
 django_mailman3/signals.py | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/django_mailman3/signals.py b/django_mailman3/signals.py
index 54306eb..20f1b11 100644
--- a/django_mailman3/signals.py
+++ b/django_mailman3/signals.py
@@ -51,13 +51,13 @@ FIELD_LAST_NAME = 'last_name'
 #
 # Defined signals
 #
-domain_created = Signal(providing_args=["mail_host"])
-domain_deleted = Signal(providing_args=["mail_host"])
-mailinglist_created = Signal(providing_args=["list_id"])
-mailinglist_modified = Signal(providing_args=["list_id"])
-mailinglist_deleted = Signal(providing_args=["list_id", "delete_archives"])
-user_subscribed = Signal(providing_args=["list_id", "user_email", "role"])
-user_unsubscribed = Signal(providing_args=["list_id", "user_email", "role"])
+domain_created = Signal()
+domain_deleted = Signal()
+mailinglist_created = Signal()
+mailinglist_modified = Signal()
+mailinglist_deleted = Signal()
+user_subscribed = Signal()
+user_unsubscribed = Signal()
 
 #
 # Signals listened to
-- 
GitLab

