diff -Naur spyder-3.1.4.orig/spyder/config/ipython.py spyder-3.1.4/spyder/config/ipython.py
--- spyder-3.1.4.orig/spyder/config/ipython.py	2017-04-24 02:30:23.000000000 +0200
+++ spyder-3.1.4/spyder/config/ipython.py	2017-05-07 02:50:25.640734369 +0200
@@ -23,7 +23,7 @@
 dependencies.add("qtconsole", _("Integrate the IPython console"),
                  required_version=QTCONSOLE_REQVER)
 dependencies.add("nbconvert", _("Manipulate Jupyter notebooks on the Editor"),
-                 required_version=NBCONVERT_REQVER)
+                 required_version=NBCONVERT_REQVER, optional=True)
 
 
 # Auxiliary functions
