diff -ruN a/pdm/__init__.py b/pdm/__init__.py
--- a/pdm/__init__.py	2022-06-28 06:41:18.142208600 +0200
+++ b/pdm/__init__.py	2022-06-30 13:06:52.281583369 +0200
@@ -16,14 +16,3 @@
     "Synchronizer",
     "Core",
 )
-
-
-def _fix_pkg_resources() -> None:
-    import importlib
-    import sys
-
-    sys.modules["pkg_resources"] = importlib.import_module("pip._vendor.pkg_resources")
-
-
-_fix_pkg_resources()
-del _fix_pkg_resources
diff -ruN a/pdm/cli/actions.py b/pdm/cli/actions.py
--- a/pdm/cli/actions.py	2022-06-28 06:41:18.142208600 +0200
+++ b/pdm/cli/actions.py	2022-06-30 13:05:15.234561298 +0200
@@ -826,7 +826,7 @@
     import sys
     from shlex import quote
 
-    from pip._vendor.packaging.version import parse as parse_version
+    from packaging.version import parse as parse_version
 
     from pdm.cli.utils import (
         is_homebrew_installation,
diff -ruN a/pdm/models/caches.py b/pdm/models/caches.py
--- a/pdm/models/caches.py	2022-06-28 06:41:18.146208500 +0200
+++ b/pdm/models/caches.py	2022-06-30 13:04:26.081047734 +0200
@@ -12,7 +12,7 @@
 from pdm.utils import open_file
 
 if TYPE_CHECKING:
-    from pip._vendor import requests
+    import requests
 
 KT = TypeVar("KT")
 VT = TypeVar("VT")
diff -ruN a/pdm/models/markers.py b/pdm/models/markers.py
--- a/pdm/models/markers.py	2022-06-28 06:41:18.146208500 +0200
+++ b/pdm/models/markers.py	2022-06-30 13:01:37.863762738 +0200
@@ -4,7 +4,7 @@
 from functools import reduce
 from typing import Any, List, Optional, Set, Tuple, Union
 
-from pip._vendor.packaging.markers import Marker as PackageMarker
+from packaging.markers import Marker as PackageMarker
 
 from pdm.models.specifiers import PySpecSet
 from pdm.utils import join_list_with
diff -ruN a/pdm/models/repositories.py b/pdm/models/repositories.py
--- a/pdm/models/repositories.py	2022-06-28 06:41:18.146208500 +0200
+++ b/pdm/models/repositories.py	2022-06-30 13:02:07.257204323 +0200
@@ -6,7 +6,7 @@
 from typing import TYPE_CHECKING, Any, Callable, Iterable, Mapping, TypeVar, cast
 
 from packaging.version import parse as parse_version
-from pip._vendor.html5lib import parse
+from html5lib import parse
 
 from pdm import termui
 from pdm._types import CandidateInfo, Package, SearchResult, Source
diff -ruN a/pdm/models/requirements.py b/pdm/models/requirements.py
--- a/pdm/models/requirements.py	2022-06-28 06:41:18.146208500 +0200
+++ b/pdm/models/requirements.py	2022-06-30 13:04:45.914453815 +0200
@@ -12,11 +12,11 @@
 from pathlib import Path
 from typing import Any, Sequence, Type, TypeVar, cast
 
-from pip._vendor.packaging.markers import InvalidMarker
-from pip._vendor.packaging.requirements import InvalidRequirement
-from pip._vendor.packaging.specifiers import SpecifierSet
-from pip._vendor.pkg_resources import Requirement as PackageRequirement
-from pip._vendor.pkg_resources import RequirementParseError, safe_name
+from packaging.markers import InvalidMarker
+from packaging.requirements import InvalidRequirement
+from packaging.specifiers import SpecifierSet
+from pkg_resources import Requirement as PackageRequirement
+from pkg_resources import RequirementParseError, safe_name
 
 from pdm._types import RequirementDict
 from pdm.exceptions import ExtrasWarning, RequirementError
diff -ruN a/pdm/models/specifiers.py b/pdm/models/specifiers.py
--- a/pdm/models/specifiers.py	2022-06-28 06:41:18.146208500 +0200
+++ b/pdm/models/specifiers.py	2022-06-30 13:04:03.440964652 +0200
@@ -6,7 +6,7 @@
 from pathlib import Path
 from typing import Any, Iterable, List, Set, Tuple, Union, cast
 
-from pip._vendor.packaging.specifiers import SpecifierSet
+from packaging.specifiers import SpecifierSet
 
 from pdm.exceptions import InvalidPyVersion
 from pdm.models.versions import Version
diff -ruN a/pdm/project/core.py b/pdm/project/core.py
--- a/pdm/project/core.py	2022-06-28 06:41:18.150208500 +0200
+++ b/pdm/project/core.py	2022-06-30 12:59:13.863231124 +0200
@@ -42,7 +42,7 @@
 if TYPE_CHECKING:
     from resolvelib.reporters import BaseReporter
 
-    from pdm._vendor import halo
+    import halo
     from pdm.core import Core
     from pdm.resolver.providers import BaseProvider
 
diff -ruN a/pdm/resolver/reporters.py b/pdm/resolver/reporters.py
--- a/pdm/resolver/reporters.py	2022-06-28 06:41:18.150208500 +0200
+++ b/pdm/resolver/reporters.py	2022-06-30 12:59:13.863231124 +0200
@@ -10,7 +10,7 @@
 if TYPE_CHECKING:
     from resolvelib.resolvers import RequirementInformation, State  # type: ignore
 
-    from pdm._vendor import halo
+    import halo
     from pdm.models.candidates import Candidate
     from pdm.models.requirements import Requirement
 
diff -ruN a/pdm/termui.py b/pdm/termui.py
--- a/pdm/termui.py	2022-06-28 06:41:18.150208500 +0200
+++ b/pdm/termui.py	2022-06-30 12:59:13.863231124 +0200
@@ -14,8 +14,9 @@
 import click
 from click._compat import strip_ansi
 
-from pdm._vendor import colorama, halo
-from pdm._vendor.log_symbols.symbols import is_supported as supports_unicode
+import colorama
+import halo
+from log_symbols.symbols import is_supported as supports_unicode
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
diff -ruN a/pdm/utils.py b/pdm/utils.py
--- a/pdm/utils.py	2022-06-28 06:41:18.150208500 +0200
+++ b/pdm/utils.py	2022-06-30 13:01:17.743688589 +0200
@@ -33,8 +33,8 @@
 )
 
 from packaging.version import Version
-from pip._vendor.packaging.tags import Tag
-from pip._vendor.requests import Session
+from packaging.tags import Tag
+from requests import Session
 
 from pdm._types import Distribution, Source
 from pdm.models.pip_shims import (
diff -ruN a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml	2022-06-28 06:41:18.150208500 +0200
+++ b/pyproject.toml	2022-06-30 13:08:36.771965354 +0200
@@ -22,6 +22,12 @@
     "tomlkit>=0.8.0,<1",
     "typing-extensions; python_version < \"3.8\"",
     "wheel>=0.36.2",
+    "colorama",
+    "halo",
+    "html5lib",
+    "log_symbols",
+    "requests",
+    "setuptools",
 ]
 name = "pdm"
 description = "Python Development Master"
diff -ruN a/tests/conftest.py b/tests/conftest.py
--- a/tests/conftest.py	2022-06-28 06:41:18.150208500 +0200
+++ b/tests/conftest.py	2022-06-30 13:00:15.103457478 +0200
@@ -13,7 +13,7 @@
 import pytest
 from click.testing import CliRunner
 from pip._internal.vcs import versioncontrol
-from pip._vendor import requests
+import requests
 
 from pdm._types import CandidateInfo
 from pdm.cli.actions import do_init, do_use
