--- /usr/share/logwatch/scripts/shared/applystddate.orig	2009-12-15 14:36:04.000000000 +0000
+++ /usr/share/logwatch/scripts/shared/applystddate.orig	2010-04-19 14:25:55.000000000 +0000
@@ -15,6 +15,7 @@ 
 my $Debug = $ENV{'LOGWATCH_DEBUG'} || 0;
 
 $SearchDate = TimeFilter('%b %e %H:%M:%S');
+$SearchDateRsyslog = TimeFilter('%Y-%m-%dT%H:%M:%S\.[0-9]+[+-][0-9]{2}:[0-9]{2}');
 
 # The date might be "Dec 09", but it needs to be "Dec  9"...
 #$SearchDate =~ s/ 0/  /;
@@ -22,11 +23,15 @@ 
 if ( $Debug > 5 ) {
    print STDERR "DEBUG: Inside ApplyStdDate...\n";
    print STDERR "DEBUG: Looking For: " . $SearchDate . "\n";
+   print STDERR "DEBUG: Looking For: " . $SearchDateRsyslog . "\n";
 }
 
 while (defined($ThisLine = <STDIN>)) {
    if ($ThisLine =~ m/^$SearchDate /o) {
       print $ThisLine;
+   } elsif ($ThisLine =~ /^$SearchDateRsyslog /o) {
+      $ThisLine =~ s/^([0-9]{4})-([0-9]{2})-([0-9]{2})T([0-9]{2}):([0-9]{2}):([0-9]{2})\.[0-9]+[+-][0-9]{2}:[0-9]{2} //o;
+      print POSIX::strftime("%b %e %H:%M:%S", $6, $5, $4, $3+1, $2-1, $1 - 1900) . " " . $ThisLine;
    } elsif ($ThisLine =~ m/(Mon|Tue|Wed|Thu|Fri|Sat|Sun) $SearchDate \d{4}/o) {
       print $ThisLine;
    }
