diff --git a/src/sage_setup/docbuild/__init__.py b/src/sage_setup/docbuild/__init__.py
index fada71126c..72eb35ee4c 100644
--- a/src/sage_setup/docbuild/__init__.py
+++ b/src/sage_setup/docbuild/__init__.py
@@ -99,7 +99,7 @@ def builder_helper(type):
         logger.debug(build_command)
 
         # Run Sphinx with Sage's special logger
-        sys.argv = ["sphinx-build"] + build_command.split()
+        sys.argv = build_command.split()
         from .sphinxbuild import runsphinx
         try:
             runsphinx()
