diff --git a/src/sage_docbuild/sphinxbuild.py b/src/sage_docbuild/sphinxbuild.py
index d917c3e9d4..7e85a723ab 100644
--- a/src/sage_docbuild/sphinxbuild.py
+++ b/src/sage_docbuild/sphinxbuild.py
@@ -110,7 +110,8 @@ class SageSphinxLogger(object):
             re.compile('WARNING: Any IDs not assiend for figure node'),
             re.compile('WARNING: .* is not referenced'),
             re.compile('WARNING: Build finished'),
-            re.compile('WARNING: rST localisation for language .* not found')
+            re.compile('WARNING: rST localisation for language .* not found'),
+            re.compile('WARNING: hardcoded link .*')
         )
         # The warning "unknown config value 'multidoc_first_pass'..."
         # should only appear when building the documentation for a
