--- a/sympy/external/tests/test_pythonmpq.py	2021-09-30 16:23:28.000000000 -0600
+++ b/sympy/external/tests/test_pythonmpq.py	2021-10-08 11:06:59.453830335 -0600
@@ -42,7 +42,7 @@ def test_PythonMPQ():
         assert check_Q(Q(Q(3, 5))) == (3, 5)
         assert check_Q(Q(0.5)) == (1, 2)
         assert check_Q(Q('0.5')) == (1, 2)
-        assert check_Q(Q(Decimal('0.6'))) == (3, 5)
+        assert check_Q(Q(str(Decimal('0.6')))) == (3, 5)
         assert check_Q(Q(Fraction(3, 5))) == (3, 5)
 
         # Invalid types
--- a/sympy/polys/ring_series.py	2021-09-30 16:23:28.000000000 -0600
+++ b/sympy/polys/ring_series.py	2021-10-08 13:43:30.746068783 -0600
@@ -989,6 +989,11 @@ def rs_nth_root(p, n, x, prec):
             except ValueError:                # as exponent
                 raise DomainError("The given series can't be expanded in "
                     "this domain.")
+            except TypeError:
+                try:
+                    const = R(c**QQ(1, n))
+                except ValueError:
+                    raise DomainError("The given series can't be expanded in this domain.")
         res = rs_nth_root(p/c, n, x, prec)*const
     else:
         res = _nth_root1(p, n, x, prec)

