diff --git a/pysollib/mfxutil.py b/pysollib/mfxutil.py
index f461ed1..101b3e1 100644
--- a/pysollib/mfxutil.py
+++ b/pysollib/mfxutil.py
@@ -143,7 +143,11 @@ def getprefdir(package):
 
 
 # high resolution clock() and sleep()
-uclock = time.clock
+try:
+    uclock = time.perf_counter
+except Exception:
+    uclock = time.clock
+
 usleep = time.sleep
 if os.name == "posix":
     uclock = time.time
