diff -rupN TLP-0.4.1.orig/Makefile TLP-0.4.1/Makefile
--- TLP-0.4.1.orig/Makefile	2014-01-02 17:48:08.000000000 +0100
+++ TLP-0.4.1/Makefile	2014-01-27 11:13:12.038086913 +0100
@@ -1,11 +1,11 @@
 # Makefile for TLP
 
-SBIN  = $(DESTDIR)/usr/sbin
+SBIN  = $(DESTDIR)/usr/bin
 BIN   = $(DESTDIR)/usr/bin
 PMETC = $(DESTDIR)/etc/pm/power.d
 TLIB  = $(DESTDIR)/usr/lib/tlp-pm
 PLIB  = $(DESTDIR)/usr/lib/pm-utils
-ULIB  = $(DESTDIR)/lib/udev
+ULIB  = $(DESTDIR)/usr/lib/udev
 ACPI  = $(DESTDIR)/etc/acpi
 NMDSP = $(DESTDIR)/etc/NetworkManager/dispatcher.d
 
@@ -38,8 +38,6 @@ install-tlp:
 	install -D -m 755 tlp-usb-udev $(ULIB)/tlp-usb-udev
 	install -D -m 644 tlp.rules $(ULIB)/rules.d/40-tlp.rules
 	[ -f $(CONFFILE) ] || install -D -m 644 default $(CONFFILE)
-	install -D -m 755 tlp.init $(DESTDIR)/etc/init.d/tlp
-	install -D -m 755 49tlp $(PLIB)/sleep.d/49tlp
 	install -D -m 644 thinkpad-radiosw $(ACPI)/events/thinkpad-radiosw
 	install -m 755 thinkpad-radiosw.sh $(ACPI)/
 	install -D -m 644 tlp.bash_completion $(DESTDIR)/etc/bash_completion.d/tlp
@@ -69,9 +67,7 @@ uninstall-tlp:
 	rmdir $(TLIB)
 	rm $(ULIB)/tlp-usb-udev
 	rm $(ULIB)/rules.d/40-tlp.rules
-	rm $(DESTDIR)/etc/init.d/tlp
 	rm $(DESTDIR)/etc/network/if-up.d/tlp-ifup
-	rm $(PLIB)/sleep.d/49tlp
 	rm $(ACPI)/events/thinkpad-radiosw
 	rm $(ACPI)/thinkpad-radiosw.sh
 	rm $(DESTDIR)/etc/bash_completion.d/tlp
diff -rupN TLP-0.4.1.orig/tlp.rules TLP-0.4.1/tlp.rules
--- TLP-0.4.1.orig/tlp.rules	2014-01-02 17:48:08.000000000 +0100
+++ TLP-0.4.1/tlp.rules	2014-01-27 11:16:06.231719068 +0100
@@ -4,7 +4,7 @@
 # This software is licensed under the GPL v2 or later.
 
 # handle change of power source ac/bat
-ACTION=="change", SUBSYSTEM=="power_supply", ATTR{type}=="Mains", RUN+="/usr/sbin/tlp auto"
+ACTION=="change", SUBSYSTEM=="power_supply", ATTR{type}=="Mains", RUN+="/usr/bin/tlp auto"
 
 # handle added usb devices (exclude subdevices)
-ACTION=="add", SUBSYSTEM=="usb", DRIVER=="usb", ENV{DEVTYPE}=="usb_device", RUN+="/lib/udev/tlp-usb-udev %p"
+ACTION=="add", SUBSYSTEM=="usb", DRIVER=="usb", ENV{DEVTYPE}=="usb_device", RUN+="/usr/lib/udev/tlp-usb-udev %p"
diff -rupN TLP-0.4.1.orig/tlp.service TLP-0.4.1/tlp.service
--- TLP-0.4.1.orig/tlp.service	2014-01-02 17:48:08.000000000 +0100
+++ TLP-0.4.1/tlp.service	2014-01-27 11:13:34.737140892 +0100
@@ -11,8 +11,8 @@ Before=shutdown.target
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/usr/sbin/tlp init start
-ExecStop=/usr/sbin/tlp init stop
+ExecStart=/usr/bin/tlp init start
+ExecStop=/usr/bin/tlp init stop
 
 [Install]
 WantedBy=multi-user.target
diff -rupN TLP-0.4.1.orig/tlp-sleep.service TLP-0.4.1/tlp-sleep.service
--- TLP-0.4.1.orig/tlp-sleep.service	2014-01-02 17:48:08.000000000 +0100
+++ TLP-0.4.1/tlp-sleep.service	2014-01-27 11:16:53.463600400 +0100
@@ -11,8 +11,8 @@ StopWhenUnneeded=yes
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/usr/sbin/tlp suspend
-ExecStop=/usr/sbin/tlp resume
+ExecStart=/usr/bin/tlp suspend
+ExecStop=/usr/bin/tlp resume
 
 [Install]
 WantedBy=sleep.target
