diff -aur flask-mail-0.9.1/setup.py flask-mail-0.9.1.new/setup.py
--- flask-mail-0.9.1/setup.py	2014-09-29 01:35:16.000000000 +0200
+++ flask-mail-0.9.1.new/setup.py	2022-11-12 22:51:06.794782737 +0100
@@ -39,7 +39,6 @@
         'nose',
         'blinker',
         'speaklater',
-        'mock',
     ],
     classifiers=[
         'Development Status :: 4 - Beta',
Only in flask-mail-0.9.1: setup.py.rej
diff -aur flask-mail-0.9.1/tests.py flask-mail-0.9.1.new/tests.py
--- flask-mail-0.9.1/tests.py	2014-09-29 01:35:16.000000000 +0200
+++ flask-mail-0.9.1.new/tests.py	2022-11-12 22:51:15.748304470 +0100
@@ -7,7 +7,7 @@
 import unittest
 import time
 import re
-import mock
+from unittest import mock
 from contextlib import contextmanager
 
 from email.header import Header
Only in flask-mail-0.9.1: tests.py.rej
diff -aur flask-mail-0.9.1/tox.ini flask-mail-0.9.1.new/tox.ini
--- flask-mail-0.9.1/tox.ini	2014-09-29 01:35:16.000000000 +0200
+++ flask-mail-0.9.1.new/tox.ini	2022-11-12 22:51:41.515512656 +0100
@@ -3,7 +3,6 @@
 
 [testenv]
 deps =
-    mock
     nose
     speaklater
 
Only in flask-mail-0.9.1: tox.ini.rej
