From fc0397732ea87c2d641f2d2fdef0b13096d945b8 Mon Sep 17 00:00:00 2001
From: Dave Halter <davidhalter88@gmail.com>
Date: Sun, 5 Nov 2017 15:05:09 +0100
Subject: [PATCH] Update the parso dependency

---
 requirements.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/requirements.txt b/requirements.txt
index 2c72b5f42..2af26c8c8 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1 +1 @@
-parso==0.1.0
+parso==0.1.1
