diff -wbBur web-news/webnews/nntp.php Web-News/webnews/nntp.php
--- web-news/webnews/nntp.php	2006-03-28 20:31:41.000000000 +0400
+++ Web-News/webnews/nntp.php	2011-07-04 14:04:36.000000000 +0400
@@ -468,7 +468,7 @@
 						$boundary = "--".$boundary;
 					} else {
 						$boundary = "";
-						$send_message .= "Content-Type: text/plain\r\n";
+						$send_message .= "Content-Type: text/plain; charset=UTF-8;\r\n";
 					}
 
 					if ($references && (strlen($references) != 0)) {
diff -wbBur web-news/webnews/rss.php Web-News/webnews/rss.php
--- web-news/webnews/rss.php	2006-03-27 18:27:29.000000000 +0400
+++ Web-News/webnews/rss.php	2011-07-04 13:42:22.000000000 +0400
@@ -38,7 +38,7 @@
 
 	ob_clean();
 	header("Content-Type: application/xml");
-	echo '<?xml version="1.0" encoding="ISO-8859-1" ?>'."\n";
+	echo '<?xml version="1.0" encoding="UTF-8" ?>'."\n";
 ?>
 <rss version="2.0">
 	<channel>
diff -wbBur web-news/webnews/util.php Web-News/webnews/util.php
--- web-news/webnews/util.php	2006-03-28 20:40:21.000000000 +0400
+++ Web-News/webnews/util.php	2011-07-04 14:02:57.000000000 +0400
@@ -54,24 +54,7 @@
 	
 	
 	function encode_MIME_header($str) {
-		if (is_non_ASCII($str)) {
-			$result = "=?ISO-8859-1?Q?";
-			for ($i = 0;$i < strlen($str);$i++) {
-				$ascii = ord($str{$i});
-				if ($ascii == 0x20) {	// Space
-					$result .= "_";
-				} else if (($ascii == 0x3D) || ($ascii == 0x3F) || ($ascii == 0x5F) || ($ascii > 0x7F)) {	// =, ?, _, 8 bit
-					$result .= "=".dechex($ascii);
-				} else {
-					$result .= $str{$i};
-				}
-			}
-			$result .= "?=";
-		} else {
-			$result = $str;
-		}
-		
-		return $result;
+		return "=?UTF-8?B?".base64_encode($str)."?=";
 	}
 	
 	
@@ -212,7 +195,7 @@
 			$counter = 0;
 			$message_body .= "This is a multi-part message in MIME format\r\n";
 			$message_body .= $boundary."\r\n";
-			$message_body .= "Content-Type: text/plain\r\n";
+			$message_body .= "Content-Type: text/plain; charset=UTF-8;\r\n";
 			$message_body .= "\r\n";
 			$message_body .= $message;
 			$message_body .= "\r\n\r\n";
