diff --git i/source/vst/moduleinfo/moduleinfo.h w/source/vst/moduleinfo/moduleinfo.h
index ecf528b..49f2827 100644
--- i/source/vst/moduleinfo/moduleinfo.h
+++ w/source/vst/moduleinfo/moduleinfo.h
@@ -39,6 +39,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 //------------------------------------------------------------------------
 namespace Steinberg {
diff --git i/source/vst/moduleinfo/moduleinfoparser.h w/source/vst/moduleinfo/moduleinfoparser.h
index 910cbf5..e052ac2 100644
--- i/source/vst/moduleinfo/moduleinfoparser.h
+++ w/source/vst/moduleinfo/moduleinfoparser.h
@@ -41,6 +41,7 @@
 #include <iostream>
 #include <optional>
 #include <string_view>
+#include <limits>
 
 //------------------------------------------------------------------------
 namespace Steinberg::ModuleInfoLib {
