=== modified file 'src/nautilus_terminal.py'
--- src/nautilus_terminal.py	2011-10-17 11:59:24 +0000
+++ src/nautilus_terminal.py	2014-12-01 07:29:40 +0000
@@ -121,7 +121,7 @@
         #Term
         self.shell_pid = -1
         self.term = Vte.Terminal()
-        self.shell_pid = self.term.fork_command_full(Vte.PtyFlags.DEFAULT,
+        self.shell_pid = self.term.spawn_sync(Vte.PtyFlags.DEFAULT,
                 self._path, [CONF.get("terminal/shell")], None,
                 GLib.SpawnFlags.SEARCH_PATH, None, None)[1]
         self.term.connect_after("child-exited", self._on_term_child_exited)
@@ -308,7 +308,7 @@
             term -- The VTE terminal (self.term).
         """
         if not self._respawn_lock:
-            self.shell_pid = self.term.fork_command_full(Vte.PtyFlags.DEFAULT,
+            self.shell_pid = self.term.spawn_sync(Vte.PtyFlags.DEFAULT,
                 self._path, [CONF.get("terminal/shell")], None,
                 GLib.SpawnFlags.SEARCH_PATH, None, None)[1]
 

