diff -wbBur arandr-0.1.6/screenlayout/xrandr.py arandr-0.1.6.my/screenlayout/xrandr.py
--- arandr-0.1.6/screenlayout/xrandr.py	2012-06-02 17:55:36.000000000 +0400
+++ arandr-0.1.6.my/screenlayout/xrandr.py	2013-02-12 13:06:49.612013203 +0400
@@ -147,7 +147,7 @@
                     o.rotations.add(r)
 
             for d in details:
-                o.modes.append(Size(int(a) for a in d.strip().split(" ")[0].split("x")))
+                o.modes.append(Size(int(a.replace("i", "")) for a in d.strip().split(" ")[0].split("_")[0].split("x")))
 
             self.state.outputs[o.name] = o
             self.configuration.outputs[o.name] = self.configuration.OutputConfiguration(active, geometry, rotation)
