diff --git i/core/rbconfig.rb w/core/rbconfig.rb
index 8d23097..acd6ed8 100644
--- i/core/rbconfig.rb
+++ w/core/rbconfig.rb
@@ -14,12 +14,12 @@ module RbConfig
   CONFIG["exec_prefix"]        = "$(prefix)"
   CONFIG["bindir"]             = Rubinius::BIN_PATH.dup
   CONFIG["sbindir"]            = "$(exec_prefix)/sbin"
-  CONFIG["libexecdir"]         = "$(exec_prefix)/libexec"
+  CONFIG["libexecdir"]         = "/usr/lib/rubinius"
   CONFIG["datarootdir"]        = "$(prefix)/share"
   CONFIG["datadir"]            = "$(datarootdir)"
-  CONFIG["sysconfdir"]         = "$(prefix)/etc"
-  CONFIG["sharedstatedir"]     = "$(prefix)/com"
-  CONFIG["localstatedir"]      = "$(prefix)/var"
+  CONFIG["sysconfdir"]         = "/etc"
+  CONFIG["sharedstatedir"]     = "/var/lib"
+  CONFIG["localstatedir"]      = "/var"
   CONFIG["includedir"]         = "$(prefix)/include"
   CONFIG["oldincludedir"]      = "/usr/include"
   CONFIG["docdir"]             = "$(datarootdir)/doc/$(PACKAGE)"
