diff -wbBur phpldapadmin-1.2.3.org/lib/functions.php phpldapadmin-1.2.3/lib/functions.php
--- phpldapadmin-1.2.3.org/lib/functions.php	2012-10-01 10:54:14.000000000 +0400
+++ phpldapadmin-1.2.3/lib/functions.php	2020-04-25 10:41:00.000000000 +0300
@@ -1588,7 +1591,7 @@
 
 	# Return icon filename based upon objectClass value
 	if (in_array('sambaaccount',$object_classes) &&
-		'$' == $rdn{ strlen($rdn) - 1 })
+		'$' == $rdn[ strlen($rdn) - 1 ])
 		return 'nt_machine.png';
 
 	if (in_array('sambaaccount',$object_classes))
diff -wbBur phpldapadmin-1.2.3.org/lib/TemplateRender.php phpldapadmin-1.2.3/lib/TemplateRender.php
--- phpldapadmin-1.2.3.org/lib/TemplateRender.php	2012-10-01 10:54:14.000000000 +0400
+++ phpldapadmin-1.2.3/lib/TemplateRender.php	2020-04-25 10:43:00.000000000 +0300
@@ -200,7 +200,7 @@
 					$next_number = $vals;
 
 					foreach ($mod as $calc) {
-						$operand = $calc{0};
+						$operand = $calc[0];
 						$operator = substr ($calc,1);
 
 						switch ($operand) {
