diff --git a/hokey.hs.orig b/hokey.hs
index a78d5a2..b951fe1 100644
--- a/hokey.hs.orig
+++ b/hokey.hs
@@ -153,9 +153,11 @@ instance A.ToJSON b => A.ToJSON (FakeMap Text b) where
     toJSON = A.toJSON . Map.fromList . unFakeMap
 instance A.ToJSON RevocationStatus
 
+instance Semigroup UIDReport where
+    (<>) (UIDReport a b c d e) (UIDReport a' b' c' d' e') = UIDReport (a <> a') (b <> b') (c <> c') (d <> d') (e <> e')
+
 instance Monoid UIDReport where
     mempty = UIDReport [] [] [] [] []
-    mappend (UIDReport a b c d e) (UIDReport a' b' c' d' e') = UIDReport (a <> a') (b <> b') (c <> c') (d <> d') (e <> e')
 
 checkKey :: Maybe POSIXTime -> TK -> KeyReport
 checkKey mpt key = (\x -> x { keyBestOf = populateBestOf x, keyHasEncryptionCapableSubkey = hasEncryptionCapableSubkey (concatMap skUsageFlags (keySubkeys x)) }) KeyReport {
