--- tupkg	2007-02-22 13:26:44.000000000 +0100
+++ tupkg.new	2007-02-22 13:31:24.000000000 +0100
@@ -128,7 +128,7 @@
 	print "  -u, --user     Connect with username"
 	print "  -P, --password Connect with password"
 	print "  -h, --host     Connect to host"
-	print "  -p, --port     Connect to host on port (default 1034)"
+	print "  -p, --port     Connect to host on port (default 1035)"
 	print "May also use conf file: ~/.tupkg"
 
 def main(argv=None):
@@ -150,12 +150,12 @@
 		try:
 			confdict['port'] =  config.getint('tupkg','port')
 		except:
-			confdict['port'] = 1034
+			confdict['port'] = 1035
 	else:
 		confdict['user'] = ""
 		confdict['password'] = ""
 		confdict['host'] = 'aur.archlinux.org'
-		confdict['port'] = 1034
+		confdict['port'] = 1035
 		if len(argv) == 1: #no config file and no args, bail
 			usage()
 			return 1
