diff -upr planout-6ba0aba6addb08e03ed6e2d21f9b224b3d56d503.orig/python/planout/assignment.py planout-6ba0aba6addb08e03ed6e2d21f9b224b3d56d503/python/planout/assignment.py
--- planout-6ba0aba6addb08e03ed6e2d21f9b224b3d56d503.orig/python/planout/assignment.py	2016-04-14 21:25:24.000000000 +0300
+++ planout-6ba0aba6addb08e03ed6e2d21f9b224b3d56d503/python/planout/assignment.py	2021-12-07 05:47:31.017734967 +0200
@@ -7,7 +7,7 @@
 
 from .ops.random import *
 from .ops.base import PlanOutOp
-from collections import MutableMapping
+from collections.abc import MutableMapping
 
 
 # The Assignment class is the main work horse that lets you to execute
