--- a/src/pcscd.h.in	2010-12-13 16:29:40.328693266 +0200
+++ b/src/pcscd.h.in	2010-12-13 16:30:04.531954586 +0200
@@ -63,7 +63,7 @@ enum
 #define PCSCLITE_STATUS_EVENT_TIMEOUT 10*60*1000 /* 10 minutes */
 
 /* Uncomment the next line if you do NOT want to use auto power off */
-/* #define DISABLE_ON_DEMAND_POWER_ON */
+#define DISABLE_ON_DEMAND_POWER_ON
 
 /* Uncomment the next line if you do not want the card to be powered on
  * when inserted */
diff -up pcsc-lite-1.6.6/src/pcscd.h.no_auto_power_down pcsc-lite-1.6.6/src/pcscd.h
--- pcsc-lite-1.6.6/src/pcscd.h.no_auto_power_down	2010-12-13 16:29:31.064593222 +0200
+++ pcsc-lite-1.6.6/src/pcscd.h	2010-12-13 16:29:55.337855295 +0200
@@ -63,7 +63,7 @@ enum
 #define PCSCLITE_STATUS_EVENT_TIMEOUT 10*60*1000 /* 10 minutes */
 
 /* Uncomment the next line if you do NOT want to use auto power off */
-/* #define DISABLE_ON_DEMAND_POWER_ON */
+#define DISABLE_ON_DEMAND_POWER_ON
 
 /* Uncomment the next line if you do not want the card to be powered on
  * when inserted */
