diff -upr m2r-0.2.1.orig/tests/test_cli.py m2r-0.2.1/tests/test_cli.py
--- m2r-0.2.1.orig/tests/test_cli.py	2018-10-12 02:36:50.000000000 +0300
+++ m2r-0.2.1/tests/test_cli.py	2021-12-02 14:27:24.916039487 +0200
@@ -58,7 +58,8 @@ class TestConvert(TestCase):
         self.assertIn('underscore-emphasis', message)
         self.assertIn('anonymous-references', message)
         self.assertIn('inline-math', message)
-        self.assertIn('optional arguments:', message)
+        self.assertTrue(
+            'optional arguments:' in message or 'options:' in message)
 
     def test_parse_file(self):
         output = parse_from_file(test_md)
