From efcd1d8d3cdc354f46bc2485e7e3394f8578b7ba Mon Sep 17 00:00:00 2001
From: pingou <pingou@pingoured.fr>
Date: Mon, 23 Aug 2010 09:54:35 +0200
Subject: [PATCH] Retrieve port as int

---
 src/guake.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/guake.py b/src/guake.py
index 6214fb2..40e688f 100644
--- a/src/guake.py
+++ b/src/guake.py
@@ -1099,7 +1099,7 @@ class Guake(SimpleGladeApp):
                 ssl_port = port
             else:
                 ssl_host = self.client.get_string('/system/proxy/secure_host')
-                ssl_port = self.client.get_string('/system/proxy/secure_port')
+                ssl_port = self.client.get_int('/system/proxy/secure_port')
 
             if self.client.get_bool(proxy + 'use_authentication'):
                 auth_user = self.client.get_string(
-- 
1.7.2.1

