diff -Naur libgtksourceviewmm-0.3.1.orig/gtksourceview/gtksourceviewmm/sourcetagstyle.cc libgtksourceviewmm-0.3.1/gtksourceview/gtksourceviewmm/sourcetagstyle.cc
--- libgtksourceviewmm-0.3.1.orig/gtksourceview/gtksourceviewmm/sourcetagstyle.cc	2008-08-03 14:41:06.000000000 +0200
+++ libgtksourceviewmm-0.3.1/gtksourceview/gtksourceviewmm/sourcetagstyle.cc	2008-08-03 14:42:26.000000000 +0200
@@ -27,6 +27,8 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#include <cstring>
+
 #include <gtksourceviewmm/sourcetagstyle.h>
 
 namespace gtksourceview
diff -Naur libgtksourceviewmm-0.3.1.orig/tests/test-search.cpp libgtksourceviewmm-0.3.1/tests/test-search.cpp
--- libgtksourceviewmm-0.3.1.orig/tests/test-search.cpp	2008-08-03 14:41:05.000000000 +0200
+++ libgtksourceviewmm-0.3.1/tests/test-search.cpp	2008-08-03 14:44:44.000000000 +0200
@@ -1,4 +1,5 @@
 #include <cassert>
+#include <cstring>
 #include <iostream>
 #include <fstream>
 #include <stdexcept>
