diff -Naur shotwell-0.13.1.orig/Makefile shotwell-0.13.1/Makefile
--- shotwell-0.13.1.orig/Makefile	2012-10-05 20:07:37.000000000 +0200
+++ shotwell-0.13.1/Makefile	2012-10-31 22:27:30.382869896 +0100
@@ -558,8 +558,8 @@
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
 	$(INSTALL_PROGRAM) $(PROGRAM) $(DESTDIR)$(PREFIX)/bin
 	$(INSTALL_PROGRAM) $(THUMBNAILER_BIN) $(DESTDIR)$(PREFIX)/bin
-	mkdir -p $(DESTDIR)$(PREFIX)/libexec/shotwell
-	$(INSTALL_PROGRAM) $(MIGRATOR_BIN) $(DESTDIR)$(PREFIX)/libexec/shotwell
+	mkdir -p $(DESTDIR)$(PREFIX)/lib/shotwell
+	$(INSTALL_PROGRAM) $(MIGRATOR_BIN) $(DESTDIR)$(PREFIX)/lib/shotwell
 	mkdir -p $(DESTDIR)$(PREFIX)/share/shotwell/icons
 	$(INSTALL_DATA) icons/* $(DESTDIR)$(PREFIX)/share/shotwell/icons
 	mkdir -p $(DESTDIR)$(PREFIX)/share/icons/hicolor/scalable/apps
diff -Naur shotwell-0.13.1.orig/src/AppDirs.vala shotwell-0.13.1/src/AppDirs.vala
--- shotwell-0.13.1.orig/src/AppDirs.vala	2012-10-05 19:56:07.000000000 +0200
+++ shotwell-0.13.1/src/AppDirs.vala	2012-10-31 22:28:17.634388872 +0100
@@ -109,7 +109,7 @@
                 // not installed yet - use wherever we were run from
                 libexec_dir = get_exec_dir();
             } else {
-                libexec_dir = File.new_for_path(Resources.PREFIX + "/libexec/shotwell");
+                libexec_dir = File.new_for_path(Resources.PREFIX + "/lib/shotwell");
             }
         }
 
