--- src/sage/plot/contour_plot.py.orig	2016-01-20 15:01:00.077409793 +0100
+++ src/sage/plot/contour_plot.py	2016-01-20 15:01:22.591370847 +0100
@@ -568,7 +568,7 @@
     g.add_primitive(ContourPlot(xy_data_array, xrange, yrange, options))
     return g
 
-@options(plot_points=150, contours=(0,0), fill=False, cmap=["blue"])
+@options(plot_points=150, contours=(0,1e-20), fill=False, cmap=["blue"])
 def implicit_plot(f, xrange, yrange, **options):
     r"""
     ``implicit_plot`` takes a function of two variables, `f(x,y)`
