--- src/sage/repl/ipython_kernel/install.py.orig	2015-10-23 19:23:39.728456590 +0000
+++ src/sage/repl/ipython_kernel/install.py	2015-10-23 19:24:00.965018978 +0000
@@ -13,8 +13,7 @@
     SAGE_ROOT, SAGE_DOC, SAGE_LOCAL, SAGE_EXTCODE,
     SAGE_VERSION
 )
-from jupyter_core.paths import ENV_JUPYTER_PATH
-JUPYTER_PATH = ENV_JUPYTER_PATH[0]
+JUPYTER_PATH = os.environ.get('JUPYTER_PATH') 
 
 class SageKernelSpec(object):

@@ -208,7 +208,7 @@
             sage: spec._symlink_resources()    # not tested
         """
         path = os.path.join(SAGE_EXTCODE, 'notebook-ipython')
-        for filename in os.listdir(path):
+        for filename in os.listdir('ext/notebook-ipython'):
             self.symlink(
                 os.path.join(path, filename),
                 os.path.join(self.kernel_dir, filename)

