diff -rup mathmap-1.3.5/drawable.h mathmap-1.3.5.new/drawable.h
--- mathmap-1.3.5/drawable.h	2009-08-23 13:56:14.000000000 +0200
+++ mathmap-1.3.5.new/drawable.h	2016-11-09 22:15:44.264585707 +0100
@@ -98,16 +98,6 @@ typedef struct _image_t
     {
 	struct _input_drawable_t *drawable;
 	struct {
-	    /* for rendering */
-	    struct _mathfuncs_t *funcs;
-	    /* for getting single pixels - never called for the root closure */
-	    filter_func_t func;
-	    mathmap_pools_t *pools;
-	    void *xy_vars;
-	    int num_args;
-	    userval_t args[];
-	} closure;
-	struct {
 	    float ax;
 	    float bx;
 	    float ay;
@@ -119,6 +109,16 @@ typedef struct _image_t
 	    float x_factor;
 	    float y_factor;
 	} resize;
+	struct {
+	    /* for rendering */
+	    struct _mathfuncs_t *funcs;
+	    /* for getting single pixels - never called for the root closure */
+	    filter_func_t func;
+	    mathmap_pools_t *pools;
+	    void *xy_vars;
+	    int num_args;
+	    userval_t args[];
+	} closure;
     } v;
 } image_t;
 /* END */
