diff --git a/src/gui/libcloudproviders/libcloudproviders.cpp b/src/gui/libcloudproviders/libcloudproviders.cpp
index 99409e813..9e41e30f0 100644
--- a/src/gui/libcloudproviders/libcloudproviders.cpp
+++ b/src/gui/libcloudproviders/libcloudproviders.cpp
@@ -12,12 +12,10 @@
  * for more details.
  */
 
-extern "C" {
-    #include <cloudproviders/cloudprovidersaccountexporter.h>
-    #include <cloudproviders/cloudprovidersproviderexporter.h>
-    #include <gio/gio.h>
-    #include <glib.h>
-}
+#include <cloudproviders/cloudprovidersaccountexporter.h>
+#include <cloudproviders/cloudprovidersproviderexporter.h>
+#include <gio/gio.h>
+#include <glib.h>
 
 #include "libcloudproviders.h"
 #include "libcloudproviders_p.h"
