diff -ruN a/test/test_util.py b/test/test_util.py
--- a/test/test_util.py	2017-05-23 00:29:23.000000000 +0200
+++ b/test/test_util.py	2020-03-14 17:57:20.716681417 +0100
@@ -107,16 +107,16 @@
 
 def test_build_sh_cmd_simple_command():
     cmd = util.build_sh_cmd('ls')
-    assert b'/bin/ls' == cmd._path
+    assert b'/bin/ls' in cmd._path
 
 
 def test_build_sh_cmd_command_with_args():
     cmd = util.build_sh_cmd('ls /tmp')
-    assert b'/bin/ls' == cmd._path
+    assert b'/bin/ls' in cmd._path
     assert [b'/tmp'] == cmd._partial_baked_args
 
 
 def test_build_sh_cmd_command_with_cwd(temp_dir):
     cmd = util.build_sh_cmd('ls', cwd=temp_dir)
-    assert b'/bin/ls' == cmd._path
+    assert b'/bin/ls' in cmd._path
     assert temp_dir == cmd._partial_call_args['cwd']
