=== modified file 'Docky.Services/Docky.Services/SystemService.cs'
--- Docky.Services/Docky.Services/SystemService.cs	2010-11-22 07:42:41 +0000
+++ Docky.Services/Docky.Services/SystemService.cs	2012-04-16 01:55:03 +0000
@@ -118,10 +118,10 @@
 			if (Bus.System.NameHasOwner (NetworkManagerName)) {
 				try {
 					network = Bus.System.GetObject<INetworkManager> (NetworkManagerName, new ObjectPath (NetworkManagerPath));
-					NetworkConnected = State == NetworkState.Connected;
+					//NetworkConnected = State == NetworkState.Connected;
 					network.StateChanged += OnConnectionStatusChanged;
 					nmTimer = GLib.Timeout.Add (1 * 60 * 1000, () => { 
-						NetworkConnected = State == NetworkState.Connected;
+						//NetworkConnected = State == NetworkState.Connected;
 						return true;
 					});
 				} catch (Exception e) {
@@ -153,7 +153,7 @@
 		void OnConnectionStatusChanged (uint state)
 		{
 			NetworkState newState = (NetworkState) Enum.ToObject (typeof (NetworkState), state);
-			NetworkConnected = newState == NetworkState.Connected;
+			//NetworkConnected = newState == NetworkState.Connected;
 			
 			if (ConnectionStatusChanged != null) {
 				Delegate [] handlers = ConnectionStatusChanged.GetInvocationList ();

