diff -Naur docky-2.2.0.orig/StandardPlugins/WorkspaceSwitcher/src/Desk.cs docky-2.2.0/StandardPlugins/WorkspaceSwitcher/src/Desk.cs
--- docky-2.2.0.orig/StandardPlugins/WorkspaceSwitcher/src/Desk.cs	2010-08-20 11:27:22.000000000 +0200
+++ docky-2.2.0/StandardPlugins/WorkspaceSwitcher/src/Desk.cs	2014-11-20 04:38:09.829116494 +0100
@@ -130,7 +130,11 @@
 					grid [x, y] = desk;
 				}
 				if (gridsize.Y - 1 > y) {
-					desk = (grid [0, y] != null ? grid [0, y].GetNeighbor (Wnck.MotionDirection.Down) : null);
+					if (grid [0, y] != null) {
+						desk = grid [0, y].GetNeighbor (Wnck.MotionDirection.Down);
+					} else {
+						desk = null;
+					}
 					grid [0, y+1] = desk;
 				}
 			}
