=== modified file 'StandardPlugins/SessionManager/src/SystemManager.cs'
--- StandardPlugins/SessionManager/src/SystemManager.cs	2012-12-31 04:25:57 +0000
+++ StandardPlugins/SessionManager/src/SystemManager.cs	2013-05-02 16:08:57 +0000
@@ -86,11 +86,15 @@
 		[Interface (SystemdIface)]
 		interface ISystemd
 		{
+			string CanHibernate ();
+			string CanSuspend ();
 			string CanPowerOff ();
 			string CanReboot ();
 
 			void PowerOff (bool interactive);
 			void Reboot (bool interactive);
+			void Suspend (bool interactive);
+			void Hibernate (bool interactive);
 		}
 
 		[Interface (ConsoleKitIface)]
@@ -212,7 +216,9 @@
 		
 		public bool CanHibernate ()
 		{
-			if (upower != null)
+			if (systemd != null)
+				return String.Equals (systemd.CanHibernate (), "yes");
+			else if (upower != null)
 				return GetBoolean (upower, UPowerName, "CanHibernate") && upower.HibernateAllowed ();
 			else if (devicekit != null)
 				return GetBoolean (devicekit, DeviceKitPowerName, "CanHibernate");
@@ -223,7 +229,10 @@
 
 		public void Hibernate ()
 		{
-			if (upower != null) {
+			if (systemd != null) {
+				if (String.Equals (systemd.CanHibernate (), "yes"))
+					systemd.Hibernate (true);
+			} else if (upower != null) {
 				if (GetBoolean (upower, UPowerName, "CanHibernate") && upower.HibernateAllowed ())
 					upower.Hibernate ();
 			} else if (devicekit != null) {
@@ -236,7 +245,9 @@
 
 		public bool CanSuspend ()
 		{
-			if (upower != null)
+			if (systemd != null)
+				return String.Equals (systemd.CanSuspend (), "yes");
+			else if (upower != null)
 				return GetBoolean (upower, UPowerName, "CanSuspend") && upower.SuspendAllowed ();
 			else if (devicekit != null)
 				return GetBoolean (devicekit, DeviceKitPowerName, "CanSuspend");
@@ -247,7 +258,10 @@
 
 		public void Suspend ()
 		{
-			if (upower != null) {
+			if (systemd != null) {
+				if (String.Equals (systemd.CanSuspend (), "yes"))
+					systemd.Suspend (true);
+			} else if (upower != null) {
 				if (GetBoolean (upower, UPowerName, "CanSuspend") && upower.SuspendAllowed ())
 					upower.Suspend ();
 			} else if (devicekit != null) {

