Index: scribus/plugins/import/pdf/importpdfconfig.h
===================================================================
--- scribus/plugins/import/pdf/importpdfconfig.h	(revision 22797)
+++ scribus/plugins/import/pdf/importpdfconfig.h	(revision 22798)
@@ -37,4 +37,11 @@
 #define getCString  c_str
 #endif
 
+#if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(0, 73, 0)
+#define Guchar  unsigned char
+#define Gushort unsigned short
+#define Guint   unsigned int
+#define Gulong  unsigned long
 #endif
+
+#endif
Index: scribus/plugins/import/pdf/slaoutput.h
===================================================================
--- scribus/plugins/import/pdf/slaoutput.h	(revision 22805)
+++ scribus/plugins/import/pdf/slaoutput.h	(revision 22806)
@@ -28,7 +28,9 @@
 #include "selection.h"
 #include "vgradient.h"
 
+#if POPPLER_ENCODED_VERSION < POPPLER_VERSION_ENCODE(0, 73, 0)
 #include <poppler/goo/gtypes.h>
+#endif
 #include <poppler/Object.h>
 #include <poppler/OutputDev.h>
 #include <poppler/Gfx.h>
