diff --git a/sagenb/flask_version/admin.py b/sagenb/flask_version/admin.py
index 680d6fad..5a465d9e 100644
--- a/sagenb/flask_version/admin.py
+++ b/sagenb/flask_version/admin.py
@@ -1,7 +1,7 @@
 import os
 from flask import Blueprint, url_for, render_template, request, session, redirect, g, current_app
 from .decorators import login_required, admin_required, with_lock
-from flask.ext.babel import Babel, gettext, ngettext, lazy_gettext
+from flask_babel import Babel, gettext, ngettext, lazy_gettext
 _ = gettext
 
 admin = Blueprint('admin', 'sagenb.flask_version.admin')
diff --git a/sagenb/flask_version/authentication.py b/sagenb/flask_version/authentication.py
index f2f4f738..f344b34f 100644
--- a/sagenb/flask_version/authentication.py
+++ b/sagenb/flask_version/authentication.py
@@ -2,7 +2,7 @@ import os
 import random
 from flask import Blueprint, url_for, render_template, request, session, redirect, g, current_app
 from .decorators import with_lock
-from flask.ext.babel import gettext, ngettext, lazy_gettext
+from flask_babel import gettext, ngettext, lazy_gettext
 _ = gettext
 
 authentication = Blueprint('authentication',
diff --git a/sagenb/flask_version/base.py b/sagenb/flask_version/base.py
index da6a990c..fdc5382a 100755
--- a/sagenb/flask_version/base.py
+++ b/sagenb/flask_version/base.py
@@ -10,11 +10,11 @@ from .decorators import global_lock
 # Make flask use the old session foo from <=flask-0.9
 from flask_oldsessions import OldSecureCookieSessionInterface
 
-from flask.ext.autoindex import AutoIndex
+from flask_autoindex import AutoIndex
 from sage.env import SAGE_SRC, SAGE_DOC
 SRC = os.path.join(SAGE_SRC, 'sage')
-from flask.ext.openid import OpenID
-from flask.ext.babel import Babel, gettext, ngettext, lazy_gettext, get_locale
+from flask_openid import OpenID
+from flask_babel import Babel, gettext, ngettext, lazy_gettext, get_locale
 from sagenb.misc.misc import SAGENB_ROOT, DATA, translations_path, N_, nN_, unicode_str
 from json import dumps
 from sagenb.notebook.cell import number_of_rows
diff --git a/sagenb/flask_version/decorators.py b/sagenb/flask_version/decorators.py
index 8726a763..4f27bdfa 100644
--- a/sagenb/flask_version/decorators.py
+++ b/sagenb/flask_version/decorators.py
@@ -1,6 +1,6 @@
 from functools import wraps
 from flask import Flask, url_for, render_template, request, session, redirect, g, current_app
-from flask.ext.babel import Babel, gettext, ngettext, lazy_gettext
+from flask_babel import Babel, gettext, ngettext, lazy_gettext
 _ = gettext
 
 from threading import Lock
diff --git a/sagenb/flask_version/doc.py b/sagenb/flask_version/doc.py
index 0895f8a0..aa3f0278 100644
--- a/sagenb/flask_version/doc.py
+++ b/sagenb/flask_version/doc.py
@@ -18,7 +18,7 @@ import os
 from flask import Blueprint, url_for, render_template, request, session, redirect, g, current_app
 from .decorators import login_required, guest_or_login_required
 
-from flask.ext.babel import gettext, ngettext, lazy_gettext
+from flask_babel import gettext, ngettext, lazy_gettext
 _ = gettext
 
 doc = Blueprint('doc', 'sagenb.flask_version.doc')
diff --git a/sagenb/flask_version/settings.py b/sagenb/flask_version/settings.py
index b482c094..40051113 100644
--- a/sagenb/flask_version/settings.py
+++ b/sagenb/flask_version/settings.py
@@ -2,7 +2,7 @@ import os
 import random
 from flask import Blueprint, url_for, render_template, request, session, redirect, g, current_app
 from .decorators import login_required, with_lock
-from flask.ext.babel import gettext, ngettext, lazy_gettext
+from flask_babel import gettext, ngettext, lazy_gettext
 _ = gettext
 
 
diff --git a/sagenb/flask_version/worksheet.py b/sagenb/flask_version/worksheet.py
index ccc9c5e3..01e83341 100644
--- a/sagenb/flask_version/worksheet.py
+++ b/sagenb/flask_version/worksheet.py
@@ -8,7 +8,7 @@ from flask import Blueprint, make_response, url_for, render_template, request, s
 from .decorators import login_required, with_lock
 from collections import defaultdict
 from werkzeug.utils import secure_filename
-from flask.ext.babel import Babel, gettext, ngettext, lazy_gettext
+from flask_babel import Babel, gettext, ngettext, lazy_gettext
 _ = gettext
 
 from sagenb.notebook.interact import INTERACT_UPDATE_PREFIX
diff --git a/sagenb/flask_version/worksheet_listing.py b/sagenb/flask_version/worksheet_listing.py
index 75c71549..ad30a72e 100644
--- a/sagenb/flask_version/worksheet_listing.py
+++ b/sagenb/flask_version/worksheet_listing.py
@@ -5,7 +5,7 @@ import os
 import urllib, urlparse
 from flask import Blueprint, url_for, render_template, request, session, redirect, g, current_app
 from .decorators import login_required, guest_or_login_required, with_lock
-from flask.ext.babel import Babel, gettext, ngettext, lazy_gettext
+from flask_babel import Babel, gettext, ngettext, lazy_gettext
 _ = gettext
 
 worksheet_listing = Blueprint('worksheet_listing',
diff --git a/sagenb/notebook/challenge.py b/sagenb/notebook/challenge.py
index 4cdf7954..b99085ae 100644
--- a/sagenb/notebook/challenge.py
+++ b/sagenb/notebook/challenge.py
@@ -32,7 +32,7 @@ from six.moves.urllib.parse import urlencode
 from six.moves.urllib.request import urlopen, Request
 
 from sagenb.notebook.template import template
-from flask.ext.babel import gettext, lazy_gettext
+from flask_babel import gettext, lazy_gettext
 _ = lazy_gettext
 
 class ChallengeResponse(object):
diff --git a/sagenb/notebook/conf.py b/sagenb/notebook/conf.py
index 50dc2a84..a6b72205 100644
--- a/sagenb/notebook/conf.py
+++ b/sagenb/notebook/conf.py
@@ -10,7 +10,7 @@ from __future__ import absolute_import
 #  The full text of the GPL is available at:
 #                  http://www.gnu.org/licenses/
 #############################################################################
-from flask.ext.babel import gettext, lazy_gettext
+from flask_babel import gettext, lazy_gettext
 
 POS = 'pos'
 DESC = 'desc'
diff --git a/sagenb/notebook/notebook.py b/sagenb/notebook/notebook.py
index b287028c..2b4efdde 100644
--- a/sagenb/notebook/notebook.py
+++ b/sagenb/notebook/notebook.py
@@ -51,7 +51,7 @@ from . import server_conf  # server configuration
 from . import user_conf    # user configuration
 from . import user         # users
 from   .template import template, prettify_time_ago
-from flask.ext.babel import gettext, lazy_gettext
+from flask_babel import gettext, lazy_gettext
 
 try:
     # sage is installed
diff --git a/sagenb/notebook/register.py b/sagenb/notebook/register.py
index cc2377ed..8908efc3 100644
--- a/sagenb/notebook/register.py
+++ b/sagenb/notebook/register.py
@@ -12,7 +12,7 @@
 """
 Helper functions dealing with the verification of user  
 """
-from flask.ext.babel import gettext as _
+from flask_babel import gettext as _
 
 def build_msg(key, username, addr, port, secure):
     url_prefix = "https" if secure else "http"
diff --git a/sagenb/notebook/server_conf.py b/sagenb/notebook/server_conf.py
index 9a95ebd6..4c51606e 100644
--- a/sagenb/notebook/server_conf.py
+++ b/sagenb/notebook/server_conf.py
@@ -8,7 +8,7 @@ from . import conf
 from .conf import (POS, DESC, GROUP, TYPE, CHOICES, T_BOOL, T_INTEGER,
                    T_CHOICE, T_REAL, T_COLOR, T_STRING, T_LIST, T_INFO)
 from sagenb.misc.misc import get_languages, N_
-from flask.ext.babel import gettext, lazy_gettext
+from flask_babel import gettext, lazy_gettext
 _ = lazy_gettext
 
 defaults = {'word_wrap_cols':72,
diff --git a/sagenb/notebook/template.py b/sagenb/notebook/template.py
index f9a578a0..35966423 100644
--- a/sagenb/notebook/template.py
+++ b/sagenb/notebook/template.py
@@ -21,7 +21,7 @@ import jinja2
 import os, re, sys
 
 from sagenb.misc.misc import SAGE_VERSION, DATA
-from flask.ext.babel import gettext, ngettext, lazy_gettext
+from flask_babel import gettext, ngettext, lazy_gettext
 from flask import current_app as app
 
 if 'SAGENB_TEMPLATE_PATH' in os.environ:
diff --git a/sagenb/notebook/tutorial.py b/sagenb/notebook/tutorial.py
index 2f745dfd..99dfd141 100644
--- a/sagenb/notebook/tutorial.py
+++ b/sagenb/notebook/tutorial.py
@@ -348,7 +348,7 @@ the "save" command), and get back to where you were quickly.
 
 #####################################
 
-from flask.ext.babel import lazy_gettext as _
+from flask_babel import lazy_gettext as _
 
 notebook_help = [
     (_('Find Help and Documentation'),
diff --git a/sagenb/notebook/user_conf.py b/sagenb/notebook/user_conf.py
index d11dbaa9..54e7686e 100644
--- a/sagenb/notebook/user_conf.py
+++ b/sagenb/notebook/user_conf.py
@@ -8,7 +8,7 @@ from . import server_conf
 from .conf import (Configuration, POS, DESC, GROUP, TYPE, CHOICES, T_BOOL,
                    T_INTEGER, T_CHOICE, T_REAL, T_COLOR, T_STRING, T_LIST)
 from sagenb.misc.misc import SAGENB_ROOT, get_languages
-from flask.ext.babel import lazy_gettext
+from flask_babel import lazy_gettext
 
 defaults = {'max_history_length':1000,
             'default_system':'sage',
diff --git a/sagenb/notebook/worksheet.py b/sagenb/notebook/worksheet.py
index 40c70b62..d93b010b 100644
--- a/sagenb/notebook/worksheet.py
+++ b/sagenb/notebook/worksheet.py
@@ -57,7 +57,7 @@ from sagenb.misc.format import relocate_future_imports
 # Imports specifically relevant to the sage notebook
 from .cell import Cell, TextCell
 from .template import template, clean_name, prettify_time_ago
-from flask.ext.babel import gettext, lazy_gettext
+from flask_babel import gettext, lazy_gettext
 _ = gettext
 
 # Set some constants that will be used for regular expressions below.
@@ -4527,7 +4527,7 @@ def convert_time_to_string(t):
     Converts ``t`` (in Unix time) to a locale-specific string
     describing the time and date.
     """
-    from flask.ext.babel import format_datetime
+    from flask_babel import format_datetime
     import datetime, time
     try:
         return format_datetime(datetime.datetime.fromtimestamp(float(t)))
