From c60f2ab90f6e39fdb515cc0305467e534f718a7d Mon Sep 17 00:00:00 2001
From: Thomas Dziedzic <gostrc@gmail.com>
Date: Tue, 11 Sep 2012 07:43:41 -0700
Subject: [PATCH] fix incorrect types in documentation/haddock

---
 src/Documentation/Haddock.hs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/Documentation/Haddock.hs b/src/Documentation/Haddock.hs
index 60bb314..8fd9674 100644
--- a/src/Documentation/Haddock.hs
+++ b/src/Documentation/Haddock.hs
@@ -59,6 +59,7 @@ import Haddock.Types
 import Haddock.Options
 import Haddock.Utils
 import Main
+import GHC (Ghc)
 
 
 -- | Create 'Interface' structures from a given list of Haddock command-line
@@ -68,7 +69,7 @@ import Main
 createInterfaces
   :: [Flag]         -- ^ A list of command-line flags
   -> [String]       -- ^ File or module names
-  -> IO [Interface] -- ^ Resulting list of interfaces
+  -> Ghc [Interface] -- ^ Resulting list of interfaces
 createInterfaces flags modules = do
   (_, ifaces, _) <- readPackagesAndProcessModules flags modules
   return ifaces
-- 
1.7.12

