diff -wbBur wqy-bitmapfont/85-wqy-bitmapsong.conf wqy-bitmapfont.my/85-wqy-bitmapsong.conf
--- wqy-bitmapfont/85-wqy-bitmapsong.conf	2007-11-04 08:55:50.000000000 +0300
+++ wqy-bitmapfont.my/85-wqy-bitmapsong.conf	2013-02-05 13:46:31.485788374 +0400
@@ -23,12 +23,22 @@
         Updated by Qianqian Fang on May 9,2007 
 -->
     <match target="pattern">
-        <test equal="any" compare="eq" name="lang">
+        <test qual="any" compare="eq" name="lang">
                 <string>en</string>
+        </test> 
+        <test qual="any" compare="eq" name="lang">
                 <string>en-us</string>
+        </test> 
+        <test qual="any" compare="eq" name="lang">
                 <string>zh-cn</string>
+        </test> 
+        <test qual="any" compare="eq" name="lang">
                 <string>zh-tw</string>
+        </test> 
+        <test qual="any" compare="eq" name="lang">
                 <string>zh-hk</string>
+        </test> 
+        <test qual="any" compare="eq" name="lang">
                 <string>zh-sg</string>
         </test> 
         <test compare="more_eq" name="pixelsize">
@@ -45,12 +55,22 @@
         </edit>
     </match>
     <match target="pattern">
-        <test equal="any" compare="eq" name="lang">
+        <test qual="any" compare="eq" name="lang">
                 <string>en</string>
+        </test> 
+        <test qual="any" compare="eq" name="lang">
                 <string>en-us</string>
+        </test> 
+        <test qual="any" compare="eq" name="lang">
                 <string>zh-cn</string>
+        </test> 
+        <test qual="any" compare="eq" name="lang">
                 <string>zh-tw</string>
+        </test> 
+        <test qual="any" compare="eq" name="lang">
                 <string>zh-hk</string>
+        </test> 
+        <test qual="any" compare="eq" name="lang">
                 <string>zh-sg</string>
         </test>
         <test compare="more_eq" name="size">
