From 248c94feb79d6e4467bb0f9614bc0a1e2846f79c Mon Sep 17 00:00:00 2001
From: Felipe Escoto Aceves <feescoto@microsoft.com>
Date: Tue, 28 Apr 2020 09:35:41 -0700
Subject: [PATCH] Add size to charArray in libmarkdown.vapi

---
 vapi/libmarkdown.vapi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vapi/libmarkdown.vapi b/vapi/libmarkdown.vapi
index 81b5189..a6bf7e7 100644
--- a/vapi/libmarkdown.vapi
+++ b/vapi/libmarkdown.vapi
@@ -41,7 +41,7 @@ namespace Markdown
 	public void initialize ();
 	public void with_html5_tags ();
 	public void shlib_destructor ();
-	public char markdown_version[];
+	public char markdown_version[512];
 
 	[Compact]
 	[CCode (cname = "MMIOT", cprefix = "mkd_", free_function = "mkd_cleanup")]
