--- a/doc/dbus/example/example-conf-client.py
+++ b/doc/dbus/example/example-conf-client.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 import dbus
--- a/doc/dbus/example/example-conf-mechanism.py
+++ b/doc/dbus/example/example-conf-mechanism.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 import gobject
--- a/py_rules.mk
+++ b/py_rules.mk
@@ -33,15 +33,15 @@ $(_SETUP_PY):	$(_SETUP_PY).in $(PKGNAME).spec
 
 py-build-ext:	$(_SETUP_PY) $(PY_SOURCES)
 	cd $(PY_TOPDIR); \
-	python $(SETUP_PY) build_ext -i
+	python2 $(SETUP_PY) build_ext -i
 
 py-build:   $(_SETUP_PY) $(PY_SOURCES)
 	cd $(PY_TOPDIR); \
-	python $(SETUP_PY) build
+	python2 $(SETUP_PY) build
 
 py-install:	$(_SETUP_PY)
 	cd $(PY_TOPDIR); \
-	python $(SETUP_PY) install -O1 \
+	python2 $(SETUP_PY) install -O1 \
 		--root $(DESTDIR) \
 		--prefix $(PREFIX) \
 		--exec-prefix $(EXEC_PREFIX) \
@@ -49,7 +49,7 @@ py-install:	$(_SETUP_PY)
 
 py-clean:	$(_SETUP_PY)
 	cd $(PY_TOPDIR); \
-	python $(SETUP_PY) clean; \
+	python2 $(SETUP_PY) clean; \
 	rm -f $(SETUP_PY); \
 	rm -rf build
 
--- a/setup.py.in
+++ b/setup.py.in
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 from distutils.core import setup
