diff -Naur regexxer-0.10.orig/src/fileshared.h regexxer-0.10/src/fileshared.h
--- regexxer-0.10.orig/src/fileshared.h	2012-03-27 03:59:38.209864722 +0000
+++ regexxer-0.10/src/fileshared.h	2012-03-27 04:17:47.161371044 +0000
@@ -23,6 +23,7 @@
 
 #include "sharedptr.h"
 
+#include <glibmm/regex.h>
 #include <gtkmm/textbuffer.h>
 #include <functional>
 #include <utility>
diff -Naur regexxer-0.10.orig/src/globalstrings.h regexxer-0.10/src/globalstrings.h
--- regexxer-0.10.orig/src/globalstrings.h	2012-03-27 03:59:38.209864722 +0000
+++ regexxer-0.10/src/globalstrings.h	2012-03-27 04:21:44.906198757 +0000
@@ -21,7 +21,7 @@
 #ifndef REGEXXER_GLOBALSTRINGS_H_INCLUDED
 #define REGEXXER_GLOBALSTRINGS_H_INCLUDED
 
-#include <glib/gutils.h>  /* for G_DIR_SEPARATOR_S */
+#include <glib.h>  /* for G_DIR_SEPARATOR_S */
 
 namespace Regexxer
 {
diff -Naur regexxer-0.10.orig/src/translation.h regexxer-0.10/src/translation.h
--- regexxer-0.10.orig/src/translation.h	2012-03-27 03:59:38.206531386 +0000
+++ regexxer-0.10/src/translation.h	2012-03-27 04:21:39.209576021 +0000
@@ -21,7 +21,7 @@
 #ifndef REGEXXER_TRANSLATION_H_INCLUDED
 #define REGEXXER_TRANSLATION_H_INCLUDED
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glibmm/ustring.h>
 
 #ifndef gettext_noop
