From 8b35ba54ab31aab13a34c360a31d014da1f5c809 Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@tiptoe.de>
Date: Thu, 17 Oct 2019 18:22:09 +0200
Subject: [PATCH 1/2] Expect ordering NULLs to work on sqlite >= 3.30.

Signed-off-by: Nils Philippsen <nils@tiptoe.de>
---
 test/requirements.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/requirements.py b/test/requirements.py
index a0d984ef4..7dde12f56 100644
--- a/test/requirements.py
+++ b/test/requirements.py
@@ -772,7 +772,8 @@ def dbapi_lastrowid(self):
     @property
     def nullsordering(self):
         """Target backends that support nulls ordering."""
-        return fails_on_everything_except("postgresql", "oracle", "firebird")
+        return fails_on_everything_except("postgresql", "oracle", "firebird",
+                                          "sqlite >= 3.30.0")
 
     @property
     def reflects_pk_names(self):

From e18534a9045786efdaf4963515222838c62e0300 Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@tiptoe.de>
Date: Fri, 18 Oct 2019 10:09:16 +0200
Subject: [PATCH 2/2] Skip nested aggregate tests on sqlite.

This seems to have stopped working with SQLite 3.30.

Signed-off-by: Nils Philippsen <nils@tiptoe.de>
---
 test/requirements.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/requirements.py b/test/requirements.py
index 7dde12f56..388131cc9 100644
--- a/test/requirements.py
+++ b/test/requirements.py
@@ -788,7 +788,7 @@ def nested_aggregates(self):
         """target database can select an aggregate from a subquery that's
         also using an aggregate"""
 
-        return skip_if(["mssql"])
+        return skip_if(["mssql", "sqlite"])
 
     @property
     def array_type(self):
