--- synaptiks-0.8.1/synaptiks/monitors/mouses.py~	2012-08-28 18:58:09.540085410 -0500
+++ synaptiks-0.8.1/synaptiks/monitors/mouses.py	2012-08-28 21:38:46.060947638 -0500
@@ -74,7 +74,7 @@
         # device.  The parent represents the actual physical device.  The name
         # may be decorated with quotation marks, which are removed for the sake
         # of a clean represenation
-        return cls(device['ID_SERIAL'], device.parent['NAME'].strip('"'))
+        return cls(device.get('ID_SERIAL'), device.parent['NAME'].strip('"'))
 
 
 class MouseDevicesMonitor(QObject):
