--- src/dirnode.cpp~	2014-10-15 22:23:22.000000000 -0400
+++ src/dirnode.cpp	2016-02-25 10:11:26.838675629 -0500
@@ -582,7 +582,7 @@
     //this->dir_radius_sqrt = sqrt(dir_radius); //dir_radius_sqrt is not used
 
 //    this->parent_radius = std::max(1.0, parent_circ / PI);
-    this->parent_radius = std::max(1.0, sqrt(total_file_area) * gGourceDirPadding);
+    this->parent_radius = (1.0) > (sqrt(total_file_area) * gGourceDirPadding) ? (1.0) : (sqrt(total_file_area) * gGourceDirPadding);
 }
 
 float RDirNode::distanceToParent() const{
