From 1b7f09b9492fd9ccbacc4a16f6cc9a623653aba3 Mon Sep 17 00:00:00 2001
From: Caleb James DeLisle <cjd@cjdns.fr>
Date: Sun, 24 Nov 2019 23:47:30 +0100
Subject: [PATCH] New GCC wants strncpy limit to be less than the output size

---
 util/platform/netdev/NetPlatform_linux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/platform/netdev/NetPlatform_linux.c b/util/platform/netdev/NetPlatform_linux.c
index 8206ebad5..967ee261e 100644
--- a/util/platform/netdev/NetPlatform_linux.c
+++ b/util/platform/netdev/NetPlatform_linux.c
@@ -78,7 +78,7 @@ static int socketForIfName(const char* interfaceName,
     }
 
     memset(ifRequestOut, 0, sizeof(struct ifreq));
-    strncpy(ifRequestOut->ifr_name, interfaceName, IFNAMSIZ);
+    strncpy(ifRequestOut->ifr_name, interfaceName, IFNAMSIZ-1);
 
     if (ioctl(s, SIOCGIFINDEX, ifRequestOut) < 0) {
         int err = errno;
