From 0b66f14b812bb83ca149bc29906716c17f388da1 Mon Sep 17 00:00:00 2001
From: Pedro Castro <pedro@gnomesubtitles.org>
Date: Sat, 10 Mar 2012 11:32:28 +0000
Subject: Fix #671567 (Fix opening help), based on a patch from György Balló.

Fix keyboard shortcuts url.
---
diff --git a/src/GnomeSubtitles/Core/EventHandlers.cs b/src/GnomeSubtitles/Core/EventHandlers.cs
index 014fc54..6de82b0 100644
--- a/src/GnomeSubtitles/Core/EventHandlers.cs
+++ b/src/GnomeSubtitles/Core/EventHandlers.cs
@@ -392,15 +392,11 @@ public class EventHandlers {
 	/*	Help Menu */
 	
 	public void OnHelpContents (object o, EventArgs args) {
-		const string arg = "ghelp:gnome-subtitles";
-		if ((!Util.OpenUrl("gnome-help " + arg)) && (!Util.OpenUrl("yelp " + arg))) {
-			BasicErrorDialog errorDialog = new BasicErrorDialog(Catalog.GetString("The Gnome Subtitles Manual could not be found."), Catalog.GetString("Please verify that your installation has been completed successfully."));
-			errorDialog.Show();
-		}
+		Util.OpenUrl("ghelp:gnome-subtitles");
 	}
 	
 	public void OnHelpKeyboardShortcuts (object o, EventArgs args) {
-		Util.OpenUrl("http://gnome-subtitles.sourceforge.net/shortcuts");
+		Util.OpenUrl("http://gnomesubtitles.org/shortcuts");
 	}
 
 	public void OnHelpRequestFeature (object o, EventArgs args) {
--
cgit v0.9.0.2
