diff -Naur tango-0.99.9/tango/io/device/Device.d tango-0.99.9-new/tango/io/device/Device.d
--- tango-0.99.9/tango/io/device/Device.d	2010-02-02 13:14:51.000000000 -0600
+++ tango-0.99.9-new/tango/io/device/Device.d	2010-03-14 10:08:56.000000000 -0500
@@ -274,7 +274,7 @@
                 {
                         if (handle >= 0)
                            {
-                           if (scheduler)
+                           //if (scheduler)
                                // TODO Not supported on Posix
                                // scheduler.close (handle, toString);
                            posix.close (handle);
