--- omg/kittens/everywhere/lol/configure.orig	2010-07-21 17:31:30.000000000 +0300
+++ omg/kittens/everywhere/lol/configure	2010-10-21 00:05:41.000000000 +0300
@@ -28605,7 +28610,7 @@ if { (eval echo configure:28601: \"$ac_l
 #include "confdefs.h"
 
 #include "$THIS_INCLUDE"
-#if DB_VERSION_MAJOR == 4
+#if DB_VERSION_MAJOR == 4 || DB_VERSION_MAJOR == 5
             yes
 #endif
           
@@ -28649,7 +28654,7 @@ echo "configure:28647: checking for DB4
 #include "confdefs.h"
 
 #include "$THIS_INCLUDE"
-#if DB_VERSION_MINOR != 1 || DB_VERSION_PATCH >= 25
+#if DB_VERSION_MINOR != 1 || (DB_VERSION_MINOR == 1 && DB_VERSION_PATCH >= 25)  || DB_VERSION_MAJOR == 5
       yes
 #endif
     

Modified: php/php-src/branches/PHP_5_3/ext/dba/dba_db4.c
===================================================================
--- php/php-src/branches/PHP_5_3/ext/dba/dba_db4.c	2010-06-03 06:57:47 UTC (rev 300150)
+++ php/php-src/branches/PHP_5_3/ext/dba/dba_db4.c	2010-06-03 07:03:05 UTC (rev 300151)
@@ -37,14 +37,14 @@
 #endif

 static void php_dba_db4_errcall_fcn(
-#if (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 3)
+#if (DB_VERSION_MAJOR > 4 || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 3))
 	const DB_ENV *dbenv, 
 #endif
 	const char *errpfx, const char *msg)
 {
 	TSRMLS_FETCH();

-#if (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR == 8 && DB_VERSION_PATCH <= 26) 
+#if (DB_VERSION_MAJOR == 5 || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR == 8))
 /* Bug 51086, Berkeley DB 4.8.26 */
 /* This code suppresses a BDB 4.8 error message that BDB incorrectly emits */
 	{
@@ -124,7 +124,7 @@
 	if ((err=db_create(&dbp, NULL, 0)) == 0) {
 	    dbp->set_errcall(dbp, php_dba_db4_errcall_fcn);
 	    if (
-#if (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 1)
+#if (DB_VERSION_MAJOR > 4 || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 1))
 			(err=dbp->open(dbp, 0, info->path, NULL, type, gmode, filemode)) == 0) {
 #else
 			(err=dbp->open(dbp, info->path, NULL, type, gmode, filemode)) == 0) {
