diff -Naur glabels-3.0.0.orig/data/schemas/org.gnome.glabels-3.gschema.xml.in glabels-3.0.0/data/schemas/org.gnome.glabels-3.gschema.xml.in
--- glabels-3.0.0.orig/data/schemas/org.gnome.glabels-3.gschema.xml.in	2012-04-09 13:58:25.256413359 +0000
+++ glabels-3.0.0/data/schemas/org.gnome.glabels-3.gschema.xml.in	2012-04-09 13:59:40.495830152 +0000
@@ -2,10 +2,10 @@
 
 
   <schema id="org.gnome.glabels-3" path="/apps/glabels-3/" gettext-domain="glabels-3.0">
-    <child name="ui" schema="org.gnome.glabels.ui"/>
-    <child name="locale" schema="org.gnome.glabels.locale"/>
-    <child name="objects" schema="org.gnome.glabels.objects"/>
-    <child name="history" schema="org.gnome.glabels.history"/>
+    <child name="ui" schema="org.gnome.glabels-3.ui"/>
+    <child name="locale" schema="org.gnome.glabels-3.locale"/>
+    <child name="objects" schema="org.gnome.glabels-3.objects"/>
+    <child name="history" schema="org.gnome.glabels-3.history"/>
   </schema>
 
 
diff -Naur glabels-3.0.0.orig/data/schemas/org.gnome.glabels-3.gschema.xml.in.in glabels-3.0.0/data/schemas/org.gnome.glabels-3.gschema.xml.in.in
--- glabels-3.0.0.orig/data/schemas/org.gnome.glabels-3.gschema.xml.in.in	2012-04-09 13:58:25.256413359 +0000
+++ glabels-3.0.0/data/schemas/org.gnome.glabels-3.gschema.xml.in.in	2012-04-09 13:59:32.232560783 +0000
@@ -2,10 +2,10 @@
 
 
   <schema id="org.gnome.glabels-3" path="/apps/glabels-3/" gettext-domain="@GETTEXT_PACKAGE@">
-    <child name="ui" schema="org.gnome.glabels.ui"/>
-    <child name="locale" schema="org.gnome.glabels.locale"/>
-    <child name="objects" schema="org.gnome.glabels.objects"/>
-    <child name="history" schema="org.gnome.glabels.history"/>
+    <child name="ui" schema="org.gnome.glabels-3.ui"/>
+    <child name="locale" schema="org.gnome.glabels-3.locale"/>
+    <child name="objects" schema="org.gnome.glabels-3.objects"/>
+    <child name="history" schema="org.gnome.glabels-3.history"/>
   </schema>
 
 
