From c1b66cd1ceefcde9074d8a382f9b06a161109401 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 4 Jun 2020 18:49:13 +0000
Subject: [PATCH] Update id in appdata file to match with the desktop file name

---
 balsa.appdata.xml.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/balsa.appdata.xml.in b/balsa.appdata.xml
index 1f6e8a601..ab7c60453 100644
--- a/balsa.appdata.xml
+++ b/balsa.appdata.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop">
-    <id>balsa.desktop</id>
+    <id>org.desktop.Balsa.desktop</id>
     <metadata_license>CC0-1.0</metadata_license>
     <description>
         <p>Balsa is a graphical mail client. While it is in active development, Balsa has many features such as:</p>
diff --git a/balsa.appdata.xml.in b/balsa.appdata.xml.in
index 1f6e8a601..ab7c60453 100644
--- a/balsa.appdata.xml.in
+++ b/balsa.appdata.xml.in
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2015 The Balsa developers -->
 <component type="desktop">
-    <id>balsa.desktop</id>
+    <id>org.desktop.Balsa.desktop</id>
     <metadata_license>CC0-1.0</metadata_license>
     <description>
         <_p>Balsa is a graphical mail client. While it is in active development,
-- 
2.26.2

