From 6738583e77ca2f52760be8887c9aefebe8e89b3b Mon Sep 17 00:00:00 2001
From: Zygmunt Krynicki <me@zygoon.pl>
Date: Thu, 20 Oct 2016 10:14:21 +0200
Subject: [PATCH] dirs: FEDORA: use alternate snap mount directory

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>
---
 dirs/dirs.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dirs/dirs.go b/dirs/dirs.go
index 36e3b54..315df32 100644
--- a/dirs/dirs.go
+++ b/dirs/dirs.go
@@ -103,7 +103,7 @@ func SetRootDir(rootdir string) {
 	}
 	GlobalRootDir = rootdir
 
-	SnapMountDir = filepath.Join(rootdir, "/snap")
+	SnapMountDir = filepath.Join(rootdir, "/var/lib/snapd/snap")
 	SnapDataDir = filepath.Join(rootdir, "/var/snap")
 	SnapDataHomeGlob = filepath.Join(rootdir, "/home/*/snap/")
 	SnapAppArmorDir = filepath.Join(rootdir, snappyDir, "apparmor", "profiles")
-- 
2.7.4

