From 529c77918eab0742e645db3d97b62b55e9f6888a Mon Sep 17 00:00:00 2001
From: Timothy Redaelli <tredaelli@archlinux.info>
Date: Tue, 21 Feb 2017 11:38:00 +0000
Subject: [PATCH] debian: ARCHLINUX: use alternate snap mount directory

Signed-off-by: Timothy Redaelli <tredaelli@archlinux.info>
---
 debian/snapd.refresh.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/debian/snapd.refresh.service b/debian/snapd.refresh.service
index 4c3375786..dda02de0e 100644
--- a/debian/snapd.refresh.service
+++ b/debian/snapd.refresh.service
@@ -2,7 +2,7 @@
 Description=Automatically refresh installed snaps
 After=network-online.target snapd.socket
 Requires=snapd.socket
-ConditionPathExistsGlob=/snap/*/current
+ConditionPathExistsGlob=/var/lib/snapd/snap/*/current
 Documentation=man:snap(1)
 
 [Service]
-- 
2.11.1

