diff --git a/deepin-system-monitor.pro b/deepin-system-monitor.pro
index 4b41229..c30c6e9 100644
--- a/deepin-system-monitor.pro
+++ b/deepin-system-monitor.pro
@@ -39,7 +39,6 @@ HEADERS += src/utils.h \
         src/compact_network_monitor.h \
         src/compact_disk_monitor.h \
         src/network_traffic_filter.h \
-        src/hashqstring.h \
         src/find_window_title.h \
         src/smooth_curve_generator.h \
         src/interactive_kill.h \
diff --git a/src/gui/process_table_view.cpp b/src/gui/process_table_view.cpp
index 3a18329..ae0f4b5 100644
--- a/src/gui/process_table_view.cpp
+++ b/src/gui/process_table_view.cpp
@@ -20,6 +20,7 @@
 #include <QMessageBox>
 #include <QProcess>
 #include <QTimer>
+#include <QKeyEvent>
 
 #include "kill_process_confirm_dialog.h"
 #include "main_window.h"
diff --git a/src/gui/system_service_table_view.cpp b/src/gui/system_service_table_view.cpp
index 15d18b4..d1b1f8c 100644
--- a/src/gui/system_service_table_view.cpp
+++ b/src/gui/system_service_table_view.cpp
@@ -10,6 +10,7 @@
 #include <QFutureWatcher>
 #include <QScrollBar>
 #include <QtConcurrent>
+#include <QKeyEvent>
 
 #include "common/error_context.h"
 #include "main_window.h"
diff --git a/src/hashqstring.h b/src/hashqstring.h
deleted file mode 100644
index 20522b3..0000000
--- a/src/hashqstring.h
+++ /dev/null
@@ -1,40 +0,0 @@
-/* -*- Mode: C++; indent-tabs-mode: nil; tab-width: 4 -*-
- * -*- coding: utf-8 -*-
- *
- * Copyright (C) 2011 ~ 2018 Deepin, Inc.
- *               2011 ~ 2018 Wang Yong
- *
- * Author:     Wang Yong <wangyong@deepin.com>
- * Maintainer: Wang Yong <wangyong@deepin.com>
- *
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */ 
-
-#ifndef HASHQSTRING_H
-#define HASHQSTRING_H
-
-#include <QString>
-#include <string>
-
-namespace std {
-    template<> struct hash<QString>
-    {
-        size_t operator()(const QString& k) const
-            {
-                return hash<string>()(k.toStdString());
-            }
-    };
-}
-
-#endif // HASHQSTRING_H
diff --git a/src/utils.cpp b/src/utils.cpp
index 054025e..4d19745 100644
--- a/src/utils.cpp
+++ b/src/utils.cpp
@@ -49,7 +49,6 @@
 
 #include <X11/extensions/shape.h>
 
-#include "hashqstring.h"
 #include "utils.h"
 
 DCORE_USE_NAMESPACE
diff --git a/src/utils.h b/src/utils.h
index 747b445..7248711 100644
--- a/src/utils.h
+++ b/src/utils.h
@@ -32,6 +32,5 @@
 #include <QString>
 #include <unordered_set>
 #include "find_window_title.h"
-#include "hashqstring.h"
 #include <QDebug>
 
 #include <QVariant>
