diff -wbBur scantailor/filters/page_split/OptionsWidget.cpp scantailor.my/filters/page_split/OptionsWidget.cpp
--- scantailor/filters/page_split/OptionsWidget.cpp	2018-11-16 15:30:37.616233595 +0300
+++ scantailor.my/filters/page_split/OptionsWidget.cpp	2018-11-16 15:22:32.029572581 +0300
@@ -29,6 +29,7 @@
 #include <QPixmap>
 #include <boost/foreach.hpp>
 #include <assert.h>
+#include <QButtonGroup>
 
 namespace page_split
 {
