commit 49702aec5da73fb874483a3246299cfb304c55cd
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Fri Mar 20 21:41:34 2020 +0800

    Properly include resource files in installation

diff --git a/ukui-volume-control-applet-qt/ukui-volume-control-applet-qt.pro b/ukui-volume-control-applet-qt/ukui-volume-control-applet-qt.pro
index a1790a3..917b789 100644
--- a/ukui-volume-control-applet-qt/ukui-volume-control-applet-qt.pro
+++ b/ukui-volume-control-applet-qt/ukui-volume-control-applet-qt.pro
@@ -10,6 +10,8 @@ TEMPLATE = app
 TARGET = ukui-volume-control-applet-qt
 INCLUDEPATH += .
 
+PREFIX = /usr/share/ukui-media
+
 include(QtSingleApplication/qtsingleapplication.pri)
 
 DEFINES += QT_DEPRECATED_WARNINGS
@@ -26,9 +28,6 @@ target.path = /usr/bin
 #inst.files += translations/ukui-media-volume-control-applet-qt_zh_CN.qm
 #              translations/1
 
-INSTALLS += \
-    target
-
 CONFIG += \
     c++11 \
     no_keywords link_pkgconfig \
@@ -66,4 +65,15 @@ RESOURCES += \
     res.qrc
 
 TRANSLATIONS += \
-    translations/ukui-volume-control-applet-qt-zh_CN.ts
+    translations/*.ts
+
+system("lrelease translations/*.ts")
+
+qm_file.files = translations/*.qm
+qm_file.path = $${PREFIX}/translations/
+
+data_files.files = data/*
+data_files.path = $${PREFIX}/
+
+INSTALLS += \
+    target qm_file data_files
