diff -up matplotlib-1.2.0/setupext.py.orig matplotlib-1.2.0/setupext.py
--- matplotlib-1.2.0/setupext.py.orig	2012-11-23 14:50:48.954715965 -0200
+++ matplotlib-1.2.0/setupext.py	2012-11-23 14:55:53.731727636 -0200
@@ -898,12 +898,12 @@ def parse_tcl_config(tcl_lib_dir, tk_lib
 
     tcl_poss = [tcl_lib_dir,
                 os.path.normpath(os.path.join(tcl_lib_dir, '..')),
-                "/usr/lib/tcl"+str(Tkinter.TclVersion),
-                "/usr/lib"]
+                "@@libdir@@/tcl"+str(Tkinter.TclVersion),
+                "@@lib@@"]
     tk_poss = [tk_lib_dir,
                 os.path.normpath(os.path.join(tk_lib_dir, '..')),
-               "/usr/lib/tk"+str(Tkinter.TkVersion),
-               "/usr/lib"]
+               "@@libdir@@/tk"+str(Tkinter.TkVersion),
+               "@@libdir@@"]
     for ptcl, ptk in zip(tcl_poss, tk_poss):
         tcl_config = os.path.join(ptcl, "tclConfig.sh")
         tk_config = os.path.join(ptk, "tkConfig.sh")
@@ -974,10 +974,10 @@ def guess_tcl_config(tcl_lib_dir, tk_lib
     return tcl_lib, tcl_inc, 'tcl' + tk_ver, tk_lib, tk_inc, 'tk' + tk_ver
 
 def hardcoded_tcl_config():
-    tcl_inc = "/usr/local/include"
-    tk_inc = "/usr/local/include"
-    tcl_lib = "/usr/local/lib"
-    tk_lib = "/usr/local/lib"
+    tcl_inc = "/usr/include"
+    tk_inc = "/usr/include"
+    tcl_lib = "@@libdir@@"
+    tk_lib = "@@libdir@@"
     return tcl_lib, tcl_inc, 'tcl', tk_lib, tk_inc, 'tk'
 
 def add_tk_flags(module):
