diff --git a/src/io/encoderlatex.cpp b/src/io/encoderlatex.cpp
index 2d666970..2ba30ab0 100644
--- a/src/io/encoderlatex.cpp
+++ b/src/io/encoderlatex.cpp
@@ -1099,7 +1099,7 @@ QString EncoderLaTeX::convertToPlainAscii(const QString &ninput) const
     for (int i = 0; i < ninputLen; ++i)
         uChars[i] = ninput.at(i).unicode();
     /// Create an ICU-specific unicode string
-    UnicodeString uString = UnicodeString(uChars, ninputLen);
+    icu::UnicodeString uString = icu::UnicodeString(uChars, ninputLen);
     /// Perform the actual transliteration, modifying Unicode string
     m_trans->transliterate(uString);
     /// Create regular C++ string from Unicode string
