--- xmoto-0.5.3/src/image/tim_png.cpp.old	2010-05-19 21:13:09.401802546 +0200
+++ xmoto-0.5.3/src/image/tim_png.cpp	2010-05-19 21:15:13.008451656 +0200
@@ -166,7 +166,7 @@
 			png_set_palette_to_rgb(PngPtr);
 			
 		if(nColorType==PNG_COLOR_TYPE_GRAY && nBitDepth<8)
-			png_set_gray_1_2_4_to_8(PngPtr);
+			png_set_expand_gray_1_2_4_to_8(PngPtr);
 
 		if(png_get_valid(PngPtr,InfoPtr,PNG_INFO_tRNS))
 			png_set_tRNS_to_alpha(PngPtr);
