
diff -r 71388899ac09 -r 2ea243e2ca82 config.mk
--- a/config.mk	Tue Jun 08 09:06:10 2010 +0200
+++ b/config.mk	Sun Nov 20 16:06:38 2011 +0100
@@ -7,13 +7,15 @@
 PREFIX = /usr/local
 MANPREFIX = ${PREFIX}/share/man
 
-GTKINC=$(shell pkg-config --cflags gtk+-2.0 webkit-1.0)
-GTKLIB=$(shell pkg-config --libs gtk+-2.0 webkit-1.0)
+X11INC = /usr/X11R6/include
+X11LIB = /usr/X11R6/lib
 
+GTKINC = `pkg-config --cflags gtk+-2.0 webkit-1.0`
+GTKLIB = `pkg-config --libs gtk+-2.0 webkit-1.0`
 
 # includes and libs
-INCS = -I. -I/usr/include ${GTKINC}
-LIBS = -L/usr/lib -lc ${GTKLIB} -lgthread-2.0
+INCS = -I. -I/usr/include -I${X11INC} ${GTKINC}
+LIBS = -L/usr/lib -lc -L${X11LIB} -lX11 ${GTKLIB} -lgthread-2.0
 
 # flags
 CPPFLAGS = -DVERSION=\"${VERSION}\"




