--- src/vagrant-installers/package/support/install_vagrant.sh.orig	2014-03-20 19:49:56.853976851 +0000
+++ src/vagrant-installers/package/support/install_vagrant.sh	2014-03-20 19:53:25.921123996 +0000
@@ -55,28 +55,6 @@
 export PATH="${EMBEDDED_DIR}/bin:${PATH}"
 ${GEM_COMMAND} install vagrant.gem --no-ri --no-rdoc
 
-# Install extensions
-${GEM_COMMAND} install vagrant-login --no-ri --no-rdoc --source "http://gems.hashicorp.com"
-${GEM_COMMAND} install vagrant-share --no-ri --no-rdoc --source "http://gems.hashicorp.com"
-
-# Setup the system plugins
-cat <<EOF >${EMBEDDED_DIR}/plugins.json
-{
-    "version": "1",
-    "installed": {
-        "vagrant-login": {
-            "ruby_version": "0",
-            "vagrant_version": "${VERSION}"
-        },
-        "vagrant-share": {
-            "ruby_version": "0",
-            "vagrant_version": "${VERSION}"
-        }
-    }
-}
-EOF
-chmod 0644 ${EMBEDDED_DIR}/plugins.json
-
 # Exit the temporary directory
 popd
 rm -rf ${TMP_DIR}
