--- kvirc-4.2.0/src/kvirc/kernel/KviIrcConnection.cpp.orig	2016-05-12 09:06:26.860358905 +0000
+++ kvirc-4.2.0/src/kvirc/kernel/KviIrcConnection.cpp	2016-05-12 09:06:37.610323266 +0000
@@ -669,7 +669,7 @@
 	while(m_pChannelList->first())
 	{
 		m_pChannelList->first()->close();
-		QApplication::processEvents(QEventLoop::ExcludeSocketNotifiers & QEventLoop::ExcludeUserInputEvents);
+		QApplication::processEvents(QEventLoop::ExcludeSocketNotifiers | QEventLoop::ExcludeUserInputEvents);
 	}
 }
 
@@ -678,7 +678,7 @@
 	while(m_pQueryList->first())
 	{
 		m_pQueryList->first()->close();
-		QApplication::processEvents(QEventLoop::ExcludeSocketNotifiers & QEventLoop::ExcludeUserInputEvents);
+		QApplication::processEvents(QEventLoop::ExcludeSocketNotifiers | QEventLoop::ExcludeUserInputEvents);
 	}
 }
 
