diff --git a/master/buildbot/test/unit/www/test_graphql.py b/master/buildbot/test/unit/www/test_graphql.py
index 787c0daf4..68b1ea266 100644
--- a/master/buildbot/test/unit/www/test_graphql.py
+++ b/master/buildbot/test/unit/www/test_graphql.py
@@ -224,7 +224,6 @@ class V3RootResource(TestReactorMixin, www.WwwTestMixin, unittest.TestCase):
                 {
                     "message": "Cannot query field 'notexistant' on type 'Query'.",
                     "locations": [{"line": 1, "column": 2}],
-                    "path": None,
                 }
             ],
             200,
diff --git a/master/buildbot/test/unit/www/test_ws.py b/master/buildbot/test/unit/www/test_ws.py
index 6b2627f4d..21cdc3330 100644
--- a/master/buildbot/test/unit/www/test_ws.py
+++ b/master/buildbot/test/unit/www/test_ws.py
@@ -269,7 +269,6 @@ class WsResource(TestReactorMixin, www.WwwTestMixin, unittest.TestCase):
                         {
                             "locations": [{"column": 11, "line": 1}],
                             "message": "Cannot query field 'not_existing' on type 'Builder'.",
-                            "path": None,
                         }
                     ],
                 },
