commit c69690edb180b96081f3e75d0c23bbd5abcb3d65
Author: Robert Maynard <robert.maynard@kitware.com>
Date:   Mon Jan 9 12:50:32 2012 -0500

    Corrected build issues when using system tiff library.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b2ee9e..a83beb7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -216,6 +216,9 @@ VISIT_VTK_THIRD_PARTY_INCLUDE(NETCDF vtknetcdf/include)
 #setup png
 VISIT_VTK_THIRD_PARTY_INCLUDE(PNG vtkpng)
 
+#setup tiff
+VISIT_VTK_THIRD_PARTY_INCLUDE(TIFF vtktiff)
+
 
 #-----------------------------------------------------------------------------
 # Detect packages here. We could probably write macros that we can include from
diff --git a/visit_vtk/full/CMakeLists.txt b/visit_vtk/full/CMakeLists.txt
index e83e041..532dc62 100644
--- a/visit_vtk/full/CMakeLists.txt
+++ b/visit_vtk/full/CMakeLists.txt
@@ -115,6 +115,6 @@ set_directory_properties(PROPERTIES INCLUDE_DIRECTORIES
   "${VISIT_SOURCE_DIR}/parser;${tmp_include_dirs}")
 
 VTK_ADD_LIBRARY(visit_vtk ${VISIT_VTK_SOURCES})
-TARGET_LINK_LIBRARIES(visit_vtk lightweight_visit_vtk vtkGraphics vtkImaging vtktiff)
+TARGET_LINK_LIBRARIES(visit_vtk lightweight_visit_vtk vtkGraphics vtkImaging ${VTK_TIFF_LIBRARIES})
 
 VISIT_INSTALL_TARGETS(visit_vtk)
