From fac2aff7b099af6e678290e0ddd186c747bbd3d9 Mon Sep 17 00:00:00 2001
From: Nick Porter <nick@portercomputing.co.uk>
Date: Fri, 13 Jan 2023 18:05:38 +0000
Subject: Use accessor function for PyFrameObject members on python >= 3.10

---
 src/modules/rlm_python/rlm_python.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/modules/rlm_python/rlm_python.c b/src/modules/rlm_python/rlm_python.c
index 2adba0e..96992cb 100644
--- a/src/modules/rlm_python/rlm_python.c
+++ b/src/modules/rlm_python/rlm_python.c
@@ -234,13 +234,27 @@ static void python_error_log(void)
 
 		for (; ptb != NULL; ptb = ptb->tb_next, fnum++) {
 			PyFrameObject *cur_frame = ptb->tb_frame;
+#if PY_VERSION_HEX >= 0x030A0000
+			PyCodeObject *code = PyFrame_GetCode(cur_frame);
+#endif
 
 			ERROR("[%ld] %s:%d at %s()",
 				fnum,
+#if PY_VERSION_HEX >= 0x030A0000
+				PyUnicode_AsUTF8(code->co_filename),
+#else
 				PyString_AsString(cur_frame->f_code->co_filename),
+#endif
 				PyFrame_GetLineNumber(cur_frame),
+#if PY_VERSION_HEX >= 0x30A0000
+				PyUnicode_AsUTF8(code->co_name)
+#else
 				PyString_AsString(cur_frame->f_code->co_name)
+#endif
 			);
+#if PY_VERSION_HEX >= 0x030A0000
+			Py_XDECREF(code);
+#endif
 		}
 	}
 
