--- a/files/usr/lib/cinnamon-settings-users/cinnamon-settings-users.py
+++ b/files/usr/lib/cinnamon-settings-users/cinnamon-settings-users.py
@@ -734,9 +734,9 @@ class Module:
             pixbuf = GdkPixbuf.Pixbuf.new_from_file_at_size("/usr/share/cinnamon/faces/user-generic.png", 48, 48)
             description = "<b>%s</b>\n%s" % (fullname, username)
             piter = self.users.append(None, [new_user, pixbuf, description])
-            # Add the user to his/her own group and sudo if Administrator was selected
+            # Add the user to his/her own group and wheel if Administrator was selected
             if dialog.account_type_combo.get_active() == 1:
-                os.system("usermod %s -G %s,sudo" % (username, username)) 
+                os.system("usermod %s -G %s,wheel" % (username, username)) 
             else:
                 os.system("usermod %s -G %s" % (username, username))
             self.load_groups()               
