diff -Naur Cinnamon-2.0.14.orig/files/usr/share/cinnamon/applets/calendar@cinnamon.org/applet.js Cinnamon-2.0.14/files/usr/share/cinnamon/applets/calendar@cinnamon.org/applet.js
--- Cinnamon-2.0.14.orig/files/usr/share/cinnamon/applets/calendar@cinnamon.org/applet.js	2013-11-25 19:41:30.000000000 +0100
+++ Cinnamon-2.0.14/files/usr/share/cinnamon/applets/calendar@cinnamon.org/applet.js	2014-04-11 08:10:28.785899903 +0200
@@ -6,7 +6,6 @@
 const St = imports.gi.St;
 const Util = imports.misc.util;
 const PopupMenu = imports.ui.popupMenu;
-const UPowerGlib = imports.gi.UPowerGlib;
 const Settings = imports.ui.settings;
 const AppletDir = imports.ui.appletManager.applets['calendar@cinnamon.org'];
 const Calendar = AppletDir.calendar;
@@ -88,10 +87,6 @@
             this.settings.bindProperty(Settings.BindingDirection.IN, "use-custom-format", "use_custom_format", this.on_settings_changed, null);
             this.settings.bindProperty(Settings.BindingDirection.IN, "custom-format", "custom_format", this.on_settings_changed, null);        
 
-            // https://bugzilla.gnome.org/show_bug.cgi?id=655129
-            this._upClient = new UPowerGlib.Client();
-            this._upClient.connect('notify-resume', this._updateClockAndDate);
-
             // Start the clock
             this.on_settings_changed();
             this._updateClockAndDatePeriodic();
