From 0f6ceabab0a8849b47f67d727aa526c2656089ba Mon Sep 17 00:00:00 2001
From: Klas Lindfors <klas@yubico.com>
Date: Tue, 3 Apr 2018 09:19:53 +0200
Subject: [PATCH 1/1] util: make sure to close the authfile before returning
 success

fixes #136

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 util.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/util.c b/util.c
index 6cb4a79..32bca06 100644
--- a/util.c
+++ b/util.c
@@ -167,6 +167,8 @@ check_user_token (const char *authfile,
 		{
 		  if(verbose)
 		      D (debug_file, "Match user/token as %s/%s", username, otp_id);
+
+		  fclose(opwfile);
 		  return AUTH_FOUND;
 		}
 	    }
