diff -upr skype-call-recorder-0.8.orig/CMakeLists.txt skype-call-recorder-0.8/CMakeLists.txt
--- skype-call-recorder-0.8.orig/CMakeLists.txt	2013-02-19 08:42:53.000000000 +0200
+++ skype-call-recorder-0.8/CMakeLists.txt	2013-02-19 08:56:55.000000000 +0200
@@ -75,7 +75,7 @@ SET(LIBRARIES ${LIBRARIES} ${ID3_LIBRARY
 
 FIND_PACKAGE(vorbisenc REQUIRED)
 INCLUDE_DIRECTORIES(${VORBISENC_INCLUDE_DIR})
-SET(LIBRARIES ${LIBRARIES} ${VORBISENC_LIBRARY})
+SET(LIBRARIES ${LIBRARIES} ${VORBISENC_LIBRARY} vorbis ogg)
 
 # Qt
 
