--- guake.py	2010-01-20 07:23:00.000000000 -0300
+++ guake.py.new	2010-04-14 07:22:52.133932225 -0300
@@ -980,11 +980,11 @@
         entry.reparent(vbox)
 
         response = dialog.run()
-        dialog.destroy()
 
         if response == gtk.RESPONSE_ACCEPT:
             self.selected_tab.set_label(entry.get_text())
 
+        dialog.destroy()
         self.set_terminal_focus()
 
     def on_close_activate(self, *args):
