diff --git a/scribus/plugins/tools/2geomtools/CMakeLists.txt b/scribus/plugins/tools/2geomtools/CMakeLists.txt
index 9b6683cbb..d2e1ee5ea 100644
--- a/scribus/plugins/tools/2geomtools/CMakeLists.txt
+++ b/scribus/plugins/tools/2geomtools/CMakeLists.txt
@@ -3,6 +3,6 @@ include_directories(
   ${CMAKE_SOURCE_DIR}/scribus
 )
 
-include_directories( ${Boost_INCLUDE_DIR} )
+include_directories( ${Boost_INCLUDE_DIRS} )
 add_subdirectory (meshdistortion)
 add_subdirectory (pathalongpath)
