diff --git c/Makefile w/Makefile
index 1b4bed6e..1210d1f1 100644
--- c/Makefile
+++ w/Makefile
@@ -33,9 +33,10 @@ endif
 PROJECT := github.com/kubernetes-sigs/cri-tools
 BINDIR ?= /usr/local/bin
 
-VERSION := $(shell git describe --tags --dirty --always)
-VERSION := $(VERSION:v%=%)
-GO_LDFLAGS := -X $(PROJECT)/pkg/version.Version=$(VERSION)
+VERSION ?= $(shell git describe --tags --dirty --always | sed 's/^v//')
+CGO_ENABLED ?= 0
+GOFLAGS ?= -trimpath
+GO_LDFLAGS := $(GO_LDFLAGS) -X $(PROJECT)/pkg/version.Version=$(VERSION)
 
 BUILD_PATH := $(shell pwd)/build
 BUILD_BIN_PATH := $(BUILD_PATH)/bin/$(GOOS)/$(GOARCH)
@@ -64,18 +65,18 @@ critest:
 	@$(MAKE) -B $(CRITEST)
 
 $(CRITEST):
-	CGO_ENABLED=0 $(GO_TEST) -c -o $@ \
+	CGO_ENABLED=$(CGO_ENABLED) $(GO_TEST) -c -o $@ \
 		-ldflags '$(GO_LDFLAGS)' \
-		-trimpath \
+		$(GOFLAGS) \
 	     $(PROJECT)/cmd/critest
 
 crictl:
 	@$(MAKE) -B $(CRICTL)
 
 $(CRICTL):
-	CGO_ENABLED=0 $(GO_BUILD) -o $@ \
+	CGO_ENABLED=$(CGO_ENABLED) $(GO_BUILD) -o $@ \
 		-ldflags '$(GO_LDFLAGS)' \
-		-trimpath \
+		$(GOFLAGS) \
 		$(PROJECT)/cmd/crictl
 
 clean:
