diff -ruN a/src/Makefile.in b/src/Makefile.in
--- a/src/Makefile.in	2004-06-15 11:11:28.000000000 +0200
+++ b/src/Makefile.in	2020-03-31 20:30:16.365386072 +0200
@@ -194,144 +194,145 @@
 	square_1643_data.so$(EXEEXT) parabola_1649_data.so$(EXEEXT)
 PROGRAMS = $(noinst_PROGRAMS) $(pkgbin_PROGRAMS) $(wavedata_PROGRAMS)
 
+DEFS = @DEFS@
+DEFAULT_INCLUDES =  -I. -I$(srcdir) -I$(top_builddir)
+CPPFLAGS = @CPPFLAGS@
+LDFLAGS = @LDFLAGS@
+LIBS = @LIBS@
+
 adsr_1653_so_SOURCES = adsr_1653.so.c
 adsr_1653_so_OBJECTS = adsr_1653.so.$(OBJEXT)
 adsr_1653_so_LDADD = $(LDADD)
-adsr_1653_so_LDFLAGS =
+adsr_1653_so_LDFLAGS = $(LDFLAGS)
 adsr_1680_so_SOURCES = adsr_1680.so.c
 adsr_1680_so_OBJECTS = adsr_1680.so.$(OBJEXT)
 adsr_1680_so_LDADD = $(LDADD)
-adsr_1680_so_LDFLAGS =
+adsr_1680_so_LDFLAGS = $(LDFLAGS)
 amp_1654_so_SOURCES = amp_1654.so.c
 amp_1654_so_OBJECTS = amp_1654.so.$(OBJEXT)
 amp_1654_so_LDADD = $(LDADD)
-amp_1654_so_LDFLAGS =
+amp_1654_so_LDFLAGS = $(LDFLAGS)
 branch_1673_so_SOURCES = branch_1673.so.c
 branch_1673_so_OBJECTS = branch_1673.so.$(OBJEXT)
 branch_1673_so_LDADD = $(LDADD)
 branch_1673_so_DEPENDENCIES =
-branch_1673_so_LDFLAGS =
+branch_1673_so_LDFLAGS = $(LDFLAGS)
 dahdsr_2021_so_SOURCES = dahdsr_2021.so.c
 dahdsr_2021_so_OBJECTS = dahdsr_2021.so.$(OBJEXT)
 dahdsr_2021_so_LDADD = $(LDADD)
-dahdsr_2021_so_LDFLAGS =
+dahdsr_2021_so_LDFLAGS = $(LDFLAGS)
 difference_2030_so_SOURCES = difference_2030.so.c
 difference_2030_so_OBJECTS = difference_2030.so.$(OBJEXT)
 difference_2030_so_LDADD = $(LDADD)
 difference_2030_so_DEPENDENCIES =
-difference_2030_so_LDFLAGS =
+difference_2030_so_LDFLAGS = $(LDFLAGS)
 fmod_1656_so_SOURCES = fmod_1656.so.c
 fmod_1656_so_OBJECTS = fmod_1656.so.$(OBJEXT)
 fmod_1656_so_LDADD = $(LDADD)
-fmod_1656_so_LDFLAGS =
+fmod_1656_so_LDFLAGS = $(LDFLAGS)
 interpolator_1660_so_SOURCES = interpolator_1660.so.c
 interpolator_1660_so_OBJECTS = interpolator_1660.so.$(OBJEXT)
 interpolator_1660_so_LDADD = $(LDADD)
 interpolator_1660_so_DEPENDENCIES =
-interpolator_1660_so_LDFLAGS =
+interpolator_1660_so_LDFLAGS = $(LDFLAGS)
 am_lp4pole_1671_so_OBJECTS = lp4pole_1671.so.$(OBJEXT) \
 	lp4pole_filter.$(OBJEXT)
 lp4pole_1671_so_OBJECTS = $(am_lp4pole_1671_so_OBJECTS)
 lp4pole_1671_so_LDADD = $(LDADD)
-lp4pole_1671_so_LDFLAGS =
+lp4pole_1671_so_LDFLAGS = $(LDFLAGS)
 nodist_parabola_1649_data_so_OBJECTS = parabola_1649_data.so.$(OBJEXT)
 parabola_1649_data_so_OBJECTS = $(nodist_parabola_1649_data_so_OBJECTS)
 parabola_1649_data_so_LDADD = $(LDADD)
 parabola_1649_data_so_DEPENDENCIES =
-parabola_1649_data_so_LDFLAGS =
+parabola_1649_data_so_LDFLAGS = $(LDFLAGS)
 product_1668_so_SOURCES = product_1668.so.c
 product_1668_so_OBJECTS = product_1668.so.$(OBJEXT)
 product_1668_so_LDADD = $(LDADD)
 product_1668_so_DEPENDENCIES =
-product_1668_so_LDFLAGS =
+product_1668_so_LDFLAGS = $(LDFLAGS)
 am_pulse_1645_so_OBJECTS = pulse_1645.so.$(OBJEXT) wavedata.$(OBJEXT)
 pulse_1645_so_OBJECTS = $(am_pulse_1645_so_OBJECTS)
 pulse_1645_so_LDADD = $(LDADD)
-pulse_1645_so_LDFLAGS =
+pulse_1645_so_LDFLAGS = $(LDFLAGS)
 am_quantiser100_2029_so_OBJECTS = quantiser.so.$(OBJEXT)
 quantiser100_2029_so_OBJECTS = $(am_quantiser100_2029_so_OBJECTS)
 quantiser100_2029_so_LDADD = $(LDADD)
-quantiser100_2029_so_LDFLAGS =
+quantiser100_2029_so_LDFLAGS = $(LDFLAGS)
 am_quantiser20_2027_so_OBJECTS = quantiser.so.$(OBJEXT)
 quantiser20_2027_so_OBJECTS = $(am_quantiser20_2027_so_OBJECTS)
 quantiser20_2027_so_LDADD = $(LDADD)
-quantiser20_2027_so_LDFLAGS =
+quantiser20_2027_so_LDFLAGS = $(LDFLAGS)
 am_quantiser50_2028_so_OBJECTS = quantiser.so.$(OBJEXT)
 quantiser50_2028_so_OBJECTS = $(am_quantiser50_2028_so_OBJECTS)
 quantiser50_2028_so_LDADD = $(LDADD)
-quantiser50_2028_so_LDFLAGS =
+quantiser50_2028_so_LDFLAGS = $(LDFLAGS)
 random_1661_so_SOURCES = random_1661.so.c
 random_1661_so_OBJECTS = random_1661.so.$(OBJEXT)
 random_1661_so_LDADD = $(LDADD)
-random_1661_so_LDFLAGS =
+random_1661_so_LDFLAGS = $(LDFLAGS)
 ratio_2034_so_SOURCES = ratio_2034.so.c
 ratio_2034_so_OBJECTS = ratio_2034.so.$(OBJEXT)
 ratio_2034_so_LDADD = $(LDADD)
-ratio_2034_so_LDFLAGS =
+ratio_2034_so_LDFLAGS = $(LDFLAGS)
 am_sawtooth_1641_so_OBJECTS = sawtooth_1641.so.$(OBJEXT) \
 	wavedata.$(OBJEXT)
 sawtooth_1641_so_OBJECTS = $(am_sawtooth_1641_so_OBJECTS)
 sawtooth_1641_so_LDADD = $(LDADD)
-sawtooth_1641_so_LDFLAGS =
+sawtooth_1641_so_LDFLAGS = $(LDFLAGS)
 nodist_sawtooth_1641_data_so_OBJECTS = sawtooth_1641_data.so.$(OBJEXT)
 sawtooth_1641_data_so_OBJECTS = $(nodist_sawtooth_1641_data_so_OBJECTS)
 sawtooth_1641_data_so_LDADD = $(LDADD)
 sawtooth_1641_data_so_DEPENDENCIES =
-sawtooth_1641_data_so_LDFLAGS =
+sawtooth_1641_data_so_LDFLAGS = $(LDFLAGS)
 am_sequencer16_1677_so_OBJECTS = sequencer.so.$(OBJEXT)
 sequencer16_1677_so_OBJECTS = $(am_sequencer16_1677_so_OBJECTS)
 sequencer16_1677_so_LDADD = $(LDADD)
-sequencer16_1677_so_LDFLAGS =
+sequencer16_1677_so_LDFLAGS = $(LDFLAGS)
 am_sequencer32_1676_so_OBJECTS = sequencer.so.$(OBJEXT)
 sequencer32_1676_so_OBJECTS = $(am_sequencer32_1676_so_OBJECTS)
 sequencer32_1676_so_LDADD = $(LDADD)
-sequencer32_1676_so_LDFLAGS =
+sequencer32_1676_so_LDFLAGS = $(LDFLAGS)
 am_sequencer64_1675_so_OBJECTS = sequencer.so.$(OBJEXT)
 sequencer64_1675_so_OBJECTS = $(am_sequencer64_1675_so_OBJECTS)
 sequencer64_1675_so_LDADD = $(LDADD)
-sequencer64_1675_so_LDFLAGS =
+sequencer64_1675_so_LDFLAGS = $(LDFLAGS)
 am_square_1643_so_OBJECTS = square_1643.so.$(OBJEXT) wavedata.$(OBJEXT)
 square_1643_so_OBJECTS = $(am_square_1643_so_OBJECTS)
 square_1643_so_LDADD = $(LDADD)
-square_1643_so_LDFLAGS =
+square_1643_so_LDFLAGS = $(LDFLAGS)
 nodist_square_1643_data_so_OBJECTS = square_1643_data.so.$(OBJEXT)
 square_1643_data_so_OBJECTS = $(nodist_square_1643_data_so_OBJECTS)
 square_1643_data_so_LDADD = $(LDADD)
 square_1643_data_so_DEPENDENCIES =
-square_1643_data_so_LDFLAGS =
+square_1643_data_so_LDFLAGS = $(LDFLAGS)
 sum_1665_so_SOURCES = sum_1665.so.c
 sum_1665_so_OBJECTS = sum_1665.so.$(OBJEXT)
 sum_1665_so_LDADD = $(LDADD)
 sum_1665_so_DEPENDENCIES =
-sum_1665_so_LDFLAGS =
+sum_1665_so_LDFLAGS = $(LDFLAGS)
 sync_pulse_2023_so_SOURCES = sync_pulse_2023.so.c
 sync_pulse_2023_so_OBJECTS = sync_pulse_2023.so.$(OBJEXT)
 sync_pulse_2023_so_LDADD = $(LDADD)
-sync_pulse_2023_so_LDFLAGS =
+sync_pulse_2023_so_LDFLAGS = $(LDFLAGS)
 sync_square_1678_so_SOURCES = sync_square_1678.so.c
 sync_square_1678_so_OBJECTS = sync_square_1678.so.$(OBJEXT)
 sync_square_1678_so_LDADD = $(LDADD)
 sync_square_1678_so_DEPENDENCIES =
-sync_square_1678_so_LDFLAGS =
+sync_square_1678_so_LDFLAGS = $(LDFLAGS)
 tracker_2025_so_SOURCES = tracker_2025.so.c
 tracker_2025_so_OBJECTS = tracker_2025.so.$(OBJEXT)
 tracker_2025_so_LDADD = $(LDADD)
-tracker_2025_so_LDFLAGS =
+tracker_2025_so_LDFLAGS = $(LDFLAGS)
 am_triangle_1649_so_OBJECTS = triangle_1649.so.$(OBJEXT) \
 	wavedata.$(OBJEXT)
 triangle_1649_so_OBJECTS = $(am_triangle_1649_so_OBJECTS)
 triangle_1649_so_LDADD = $(LDADD)
-triangle_1649_so_LDFLAGS =
+triangle_1649_so_LDFLAGS = $(LDFLAGS)
 am_wavegen_OBJECTS = wavegen.$(OBJEXT) wdatutil.$(OBJEXT)
 wavegen_OBJECTS = $(am_wavegen_OBJECTS)
 wavegen_LDADD = $(LDADD)
-wavegen_LDFLAGS =
+wavegen_LDFLAGS = $(LDFLAGS)
 
-DEFS = @DEFS@
-DEFAULT_INCLUDES =  -I. -I$(srcdir) -I$(top_builddir)
-CPPFLAGS = @CPPFLAGS@
-LDFLAGS = @LDFLAGS@
-LIBS = @LIBS@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 @AMDEP_TRUE@DEP_FILES = $(DEPDIR)/adsr_1653.so.Po \
 @AMDEP_TRUE@	$(DEPDIR)/adsr_1680.so.Po $(DEPDIR)/amp_1654.so.Po \
@@ -797,22 +798,22 @@
 	$(COMPILE) -DWAVEDATA_SUBDIR=\"$(wavedata_subdir)\" -c wavedata.c
 
 sequencer64_1675.so$(EXEEXT): sequencer.so.c
-	$(LINK) -DSEQUENCER_MAX_INPUTS=64 -DSEQUENCER_BASE_ID=1675 sequencer.so.c
+	$(LINK) $(sequencer64_1675_so_LDFLAGS) -DSEQUENCER_MAX_INPUTS=64 -DSEQUENCER_BASE_ID=1675 sequencer.so.c
 
 sequencer32_1676.so$(EXEEXT): sequencer.so.c
-	$(LINK) -DSEQUENCER_MAX_INPUTS=32 -DSEQUENCER_BASE_ID=1676 sequencer.so.c
+	$(LINK) $(sequencer32_1676_so_LDFLAGS) -DSEQUENCER_MAX_INPUTS=32 -DSEQUENCER_BASE_ID=1676 sequencer.so.c
 
 sequencer16_1677.so$(EXEEXT): sequencer.so.c
-	$(LINK) -DSEQUENCER_MAX_INPUTS=16 -DSEQUENCER_BASE_ID=1677 sequencer.so.c
+	$(LINK) $(sequencer16_1677_so_LDFLAGS) -DSEQUENCER_MAX_INPUTS=16 -DSEQUENCER_BASE_ID=1677 sequencer.so.c
 
 quantiser20_2027.so$(EXEEXT): quantiser.so.c
-	$(LINK) -O0 -DQUANTISER_MAX_INPUTS=20 -DQUANTISER_BASE_ID=2027 -lm quantiser.so.c
+	$(LINK) $(quantiser20_2027_so_LDFLAGS) -O0 -DQUANTISER_MAX_INPUTS=20 -DQUANTISER_BASE_ID=2027 -lm quantiser.so.c
 
 quantiser50_2028.so$(EXEEXT): quantiser.so.c
-	$(LINK) -O0 -DQUANTISER_MAX_INPUTS=50 -DQUANTISER_BASE_ID=2028 -lm quantiser.so.c
+	$(LINK) $(quantiser50_2028_so_LDFLAGS) -O0 -DQUANTISER_MAX_INPUTS=50 -DQUANTISER_BASE_ID=2028 -lm quantiser.so.c
 
 quantiser100_2029.so$(EXEEXT): quantiser.so.c
-	$(LINK) -O0 -DQUANTISER_MAX_INPUTS=100 -DQUANTISER_BASE_ID=2029 -lm quantiser.so.c
+	$(LINK) $(quantiser100_2029_so_LDFLAGS) -O0 -DQUANTISER_MAX_INPUTS=100 -DQUANTISER_BASE_ID=2029 -lm quantiser.so.c
 
 parabola_1649_data.so.c: wavegen
 	./wavegen -r $(sample_rate) -f $(first_note) -s $(note_step) \
