diff --git i/pynitrokey/cli/__init__.py w/pynitrokey/cli/__init__.py
index dc27289..034cd60 100644
--- i/pynitrokey/cli/__init__.py
+++ w/pynitrokey/cli/__init__.py
@@ -18,7 +18,6 @@ import pynitrokey
 import pynitrokey.fido2.operations
 from pynitrokey.cli.fido2 import fido2
 from pynitrokey.cli.nethsm import nethsm
-from pynitrokey.cli.nk3 import nk3
 from pynitrokey.cli.pro import pro
 from pynitrokey.cli.start import start
 from pynitrokey.cli.storage import storage
@@ -58,7 +57,6 @@ def nitropy():
 
 nitropy.add_command(fido2)
 nitropy.add_command(nethsm)
-nitropy.add_command(nk3)
 nitropy.add_command(start)
 nitropy.add_command(storage)
 nitropy.add_command(pro)
@@ -76,7 +74,6 @@ nitropy.add_command(version)
 def _list():
     fido2.commands["list"].callback()
     start.commands["list"].callback()
-    nk3.commands["list"].callback()
 
 
 @click.command()
diff --git i/pyproject.toml w/pyproject.toml
index 36d76e4..87ae2e3 100644
--- i/pyproject.toml
+++ w/pyproject.toml
@@ -32,7 +32,6 @@ dependencies = [
   "python-dateutil",
   "pyusb",
   "requests",
-  "spsdk >=1.6.0",
   "tqdm",
   "urllib3",
 ]
