diff -upr text-icu-0.7.0.1.orig/cbits/text_icu.c text-icu-0.7.0.1/cbits/text_icu.c
--- text-icu-0.7.0.1.orig/cbits/text_icu.c	2015-01-09 06:58:03.000000000 +0200
+++ text-icu-0.7.0.1/cbits/text_icu.c	2020-12-19 08:34:48.487152611 +0200
@@ -1,3 +1,5 @@
+#include <stdbool.h>
+
 #include "hs_text_icu.h"
 
 UBreakIterator* __hs_ubrk_open(UBreakIteratorType type, const char *locale,
@@ -305,7 +307,7 @@ int32_t __hs_u_strFoldCase(UChar *dest,
 
 int32_t __hs_u_strCompareIter(UCharIterator *iter1, UCharIterator *iter2)
 {
-    return u_strCompareIter(iter1, iter2, TRUE);
+    return u_strCompareIter(iter1, iter2, true);
 }
 
 UBlockCode __hs_ublock_getCode(UChar32 c)
