diff -ruN a/python-inet_diag/inet_diag.c b/python-inet_diag/inet_diag.c
--- a/python-inet_diag/inet_diag.c	2016-04-07 16:39:56.000000000 +0200
+++ b/python-inet_diag/inet_diag.c	2020-11-15 22:20:23.580012069 +0100
@@ -525,7 +525,6 @@
 	.tp_name	= "inet_socket",
 	.tp_basicsize	= sizeof(struct inet_socket),
 	.tp_dealloc	= (destructor)inet_socket__dealloc,
-	.tp_print	= (printfunc)inet_socket__print,
 	.tp_getattr	= (getattrfunc)inet_socket__getattr,
 };
 
@@ -749,7 +748,6 @@
 	.tp_name	= "inet_diag",
 	.tp_basicsize	= sizeof(struct inet_diag),
 	.tp_dealloc	= (destructor)inet_diag__dealloc,
-	.tp_print	= (printfunc)inet_diag__print,
 	.tp_getattr	= (getattrfunc)inet_diag__getattr,
 };
 
