diff --unified --recursive --text googletest-release-1.11.0.orig/googlemock/CMakeLists.txt googletest-release-1.11.0/googlemock/CMakeLists.txt
--- googletest-release-1.11.0.orig/googlemock/CMakeLists.txt	2020-11-13 14:33:36.022632459 -0500
+++ googletest-release-1.11.0/googlemock/CMakeLists.txt	2020-11-13 14:35:01.806861990 -0500
@@ -40,6 +40,7 @@
   project(gmock CXX C)
 else()
   cmake_policy(SET CMP0048 NEW)
+  set(GOOGLETEST_VERSION __GOOGLETEST_VERSION__)
   project(gmock VERSION ${GOOGLETEST_VERSION} LANGUAGES CXX C)
 endif()
 cmake_minimum_required(VERSION 2.6.4)
diff --unified --recursive --text googletest-release-1.11.0.orig/googletest/CMakeLists.txt googletest-release-1.11.0/googletest/CMakeLists.txt
--- googletest-release-1.11.0.orig/googletest/CMakeLists.txt	2020-11-13 14:33:36.039299268 -0500
+++ googletest-release-1.11.0/googletest/CMakeLists.txt	2020-11-13 14:37:10.274915300 -0500
@@ -46,6 +46,7 @@
 
 # Project version:
 
+set(GOOGLETEST_VERSION __GOOGLETEST_VERSION__)
 if (CMAKE_VERSION VERSION_LESS 3.0)
   project(gtest CXX C)
   set(PROJECT_VERSION ${GOOGLETEST_VERSION})
