diff --git a/dbus/conn.go b/dbus/conn.go
index 20f65b1..01da4e3 100644
--- a/dbus/conn.go
+++ b/dbus/conn.go
@@ -26,6 +26,7 @@ import (
 	"reflect"
 	"strings"
 	"sync"
+	dbus1 "pkg.deepin.io/lib/dbus1"
 )
 
 const defaultSystemBusAddress = "unix:path=/var/run/dbus/system_bus_socket"
@@ -112,12 +113,12 @@ func SessionBus() (conn *Conn, err error) {
 
 // SessionBusPrivate returns a new private connection to the session bus.
 func SessionBusPrivate() (*Conn, error) {
-	address := os.Getenv("DBUS_SESSION_BUS_ADDRESS")
-	if address != "" && address != "autolaunch:" {
-		return Dial(address)
+	address, err := dbus1.GetSessionBusAddress()
+	if err != nil {
+		return nil, err
 	}
 
-	return sessionBusPlatform()
+	return Dial(address)
 }
 
 // SystemBus returns a shared connection to the system bus, connecting to it if
diff --git a/dbus1/conn.go b/dbus1/conn.go
index 2737621..ac89088 100644
--- a/dbus1/conn.go
+++ b/dbus1/conn.go
@@ -77,6 +77,10 @@ func SessionBus() (conn *Conn, err error) {
 	return
 }
 
+func GetSessionBusAddress() (string, error) {
+	return getSessionBusAddress()
+}
+
 func getSessionBusAddress() (string, error) {
 	if address := os.Getenv("DBUS_SESSION_BUS_ADDRESS"); address != "" && address != "autolaunch:" {
 		return address, nil
