--- lsb_release.orig	2012-11-13 05:12:02.000000000 -0800
+++ lsb_release	2013-11-24 17:13:55.943608718 -0800
@@ -216,7 +216,7 @@
             TMP_DISTRIB_DESC=$DISTRIB_DESCRIPTION
         fi
 
-        if [ -z "$TMP_DISTRIB_DESC" ]  # head or lsb-release init
+        if [ -z "$DISTRIB_DESCRIPTION" ]  # head or lsb-release init
         then                           # file contains no data
             DISTRIB_DESCRIPTION=$MSG_NONE
             NO="y"
