diff --git a/CMakeLists.txt b/CMakeLists.txt
index 58f9949..2303b24 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,7 +108,10 @@ set_package_properties(KQuickImageEditor PROPERTIES
     PURPOSE "Add image editing capability to image attachments"
 )
 
-find_package(QCoro REQUIRED)
+find_package(QCoro5 COMPONENTS Coro QUIET)
+if(NOT QCoro5_FOUND)
+    find_package(QCoro REQUIRED)
+endif()
 
 qcoro_enable_coroutines()
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7ad3c88..54aab85 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -62,7 +62,12 @@ if(NOT ANDROID)
 endif()
 
 target_include_directories(neochat PRIVATE ${CMAKE_BINARY_DIR})
-target_link_libraries(neochat PRIVATE Qt::Quick Qt::Qml Qt::Gui Qt::Network Qt::QuickControls2 KF5::I18n KF5::Kirigami2 KF5::Notifications KF5::ConfigCore KF5::ConfigGui KF5::CoreAddons Quotient cmark::cmark ${QTKEYCHAIN_LIBRARIES} QCoro::QCoro)
+target_link_libraries(neochat PRIVATE Qt::Quick Qt::Qml Qt::Gui Qt::Network Qt::QuickControls2 KF5::I18n KF5::Kirigami2 KF5::Notifications KF5::ConfigCore KF5::ConfigGui KF5::CoreAddons Quotient cmark::cmark ${QTKEYCHAIN_LIBRARIES})
+if(TARGET QCoro5::Coro)
+    target_link_libraries(neochat PRIVATE QCoro5::Coro)
+else()
+    target_link_libraries(neochat PRIVATE QCoro::QCoro)
+endif()
 kconfig_add_kcfg_files(neochat GENERATE_MOC neochatconfig.kcfgc)
 
 if(NEOCHAT_FLATPAK)
