diff -aur pgadmin4-4.30/web/pgadmin/browser/utils.py pgadmin4-4.30.old/web/pgadmin/browser/utils.py
--- pgadmin4-4.30/web/pgadmin/browser/utils.py	2021-10-03 19:12:11.176725436 +0200
+++ pgadmin4-4.30.old/web/pgadmin/browser/utils.py	2021-10-03 19:04:26.747074705 +0200
@@ -13,9 +13,8 @@
 
 import flask
 from flask import render_template, current_app
-from flask.views import View, MethodViewType
+from flask.views import View, MethodViewType, with_metaclass
 from flask_babelex import gettext
-from six import with_metaclass
 
 from config import PG_DEFAULT_DRIVER
 from pgadmin.utils.ajax import make_json_response, precondition_required,\
