diff -aur pgadmin4-4.30/web/pgadmin/browser/__init__.py pgadmin4-4.30.old/web/pgadmin/browser/__init__.py
--- pgadmin4-4.30/web/pgadmin/browser/__init__.py	2021-10-03 19:00:39.757955280 +0200
+++ pgadmin4-4.30.old/web/pgadmin/browser/__init__.py	2021-10-03 18:59:22.220450894 +0200
@@ -30,7 +30,7 @@
 from flask_security.signals import reset_password_instructions_sent
 from flask_security.utils import config_value, do_flash, get_url, \
     get_message, slash_url_suffix, login_user, send_mail, logout_user
-from flask_security.views import _security, view_commit, _ctx
+from flask_security.views import _security, _commit, _ctx
 from werkzeug.datastructures import MultiDict
 
 import config
@@ -1039,7 +1039,7 @@
                 has_error = True
 
             if request.json is None and not has_error:
-                after_this_request(view_commit)
+                after_this_request(_commit)
                 do_flash(*get_message('PASSWORD_CHANGE'))
 
                 old_key = get_crypt_key()[1]
@@ -1205,7 +1205,7 @@
                 has_error = True
 
             if not has_error:
-                after_this_request(view_commit)
+                after_this_request(_commit)
                 do_flash(*get_message('PASSWORD_RESET'))
                 login_user(user)
                 return redirect(get_url(_security.post_reset_view) or
