diff --git a/plugins/one-key-login/CMakeLists.txt b/plugins/one-key-login/CMakeLists.txt
index 2c532f76..47ee48da 100644
--- a/plugins/one-key-login/CMakeLists.txt
+++ b/plugins/one-key-login/CMakeLists.txt
@@ -1,9 +1,14 @@
+
 find_package(QT NAMES Qt5 COMPONENTS Widgets REQUIRED)
 find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Widgets REQUIRED)
+find_package(DtkCore REQUIRED)
+find_package(DtkWidget REQUIRED)
 
 link_libraries(
     ${Qt_LIBS}
     Qt${QT_VERSION_MAJOR}::Widgets
+    ${DtkCore_LIBRARIES}
+    ${DtkWidget_LIBRARIES}
 )
 
 set(LIB_NAME one-key-login)
