diff -Naur ubuntuone-client-gnome-3.0.0.orig/nautilus/utils.c ubuntuone-client-gnome-3.0.0/nautilus/utils.c
--- ubuntuone-client-gnome-3.0.0.orig/nautilus/utils.c	2012-02-22 16:40:56.000000000 +0100
+++ ubuntuone-client-gnome-3.0.0/nautilus/utils.c	2012-05-13 22:26:35.267409857 +0200
@@ -26,8 +26,7 @@
 #include <limits.h>
 #include <stdlib.h>
 #include <glib/gi18n.h>
-#include <glib/gutils.h>
-#include <glib/gfileutils.h>
+#include <glib.h>
 #include <libnautilus-extension/nautilus-file-info.h>
 #include <libsyncdaemon/libsyncdaemon.h>
 #include "ubuntuone-nautilus.h"
