diff -Naurp a/contrib/tunnels.d/README b/contrib/tunnels.d/README
--- a/contrib/tunnels.d/README	2019-08-27 14:17:32.000000000 +0000
+++ b/contrib/tunnels.d/README	2019-09-30 18:48:51.066404952 +0000
@@ -1,4 +1,4 @@
-# In that directory you can store separated config files for every tunnel.
+# In the /etc/i2pd/tunnels.d directory you can store separated config files for every tunnel.
 # Please read documentation for more info.
 #
 # You can find examples in /usr/share/doc/i2pd/tunnels.d directory
