diff --git a/hack/lib/golang.sh b/hack/lib/golang.sh
index b646bbe2edf..89d0a5cdba1 100755
--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -798,7 +798,7 @@ kube::golang::build_binaries() {
 
     # extract tags if any specified in GOFLAGS
     # shellcheck disable=SC2001
-    gotags="selinux,$(echo "${GOFLAGS:-}" | sed -e 's|.*-tags=\([^-]*\).*|\1|')"
+    gotags="selinux $(echo "${GOFLAGS:-}" | sed -e 's|.*-tags=\([^-]*\).*|\1|')"
 
     local -a targets=()
     local arg
