diff -upr cherokee-0.98.1.orig/cherokee.conf.sample.pre cherokee-0.98.1/cherokee.conf.sample.pre
--- cherokee-0.98.1.orig/cherokee.conf.sample.pre	2009-01-23 21:18:41.000000000 +0200
+++ cherokee-0.98.1/cherokee.conf.sample.pre	2009-02-20 16:39:40.000000000 +0200
@@ -17,6 +17,8 @@ server!keepalive_max_requests = 500
 server!server_tokens = full
 server!panic_action = %prefix%/bin/cherokee-panic
 server!pid_file = %localstatedir%/run/cherokee.pid
+server!user = http
+server!group = http

 # Default virtual server
 #
diff -upr cherokee-0.98.1.orig/performance.conf.sample.pre cherokee-0.98.1/performance.conf.sample.pre
--- cherokee-0.98.1.orig/performance.conf.sample.pre	2009-01-23 21:18:41.000000000 +0200
+++ cherokee-0.98.1/performance.conf.sample.pre	2009-02-20 16:39:41.000000000 +0200
@@ -15,6 +15,8 @@ server!keepalive_max_requests = 1000
 server!server_tokens = full
 server!panic_action = %prefix%/bin/cherokee-panic
 server!pid_file = %localstatedir%/run/cherokee.pid
+server!user = http
+server!group = http

 # Default virtual server
 #
