diff --git a/src/luxrender/core/__init__.py b/src/luxrender/core/__init__.py
--- a/src/luxrender/core/__init__.py
+++ b/src/luxrender/core/__init__.py
@@ -1212,7 +1212,7 @@
         """
         Convert AOVs to Blender images
         """
-        from ..outputs.luxcore_api import pyluxcore
+        import pyluxcore
 
         # Structure: {channelType: [pyluxcoreType, is HDR, arrayDepth, optional matching Blender pass]}
         attributes = {
@@ -1459,7 +1459,7 @@
             LuxLog('ERROR: LuxCore rendering requires pyluxcore')
             self.report({'ERROR'}, 'LuxCore rendering requires pyluxcore')
             return
-        from ..outputs.luxcore_api import pyluxcore
+        import pyluxcore
 
         try:
             scene.luxcore_rendering_controls.pause_render = False
@@ -1841,7 +1841,7 @@
         if not PYLUXCORE_AVAILABLE:
             LuxLog('ERROR: LuxCore preview rendering requires pyluxcore')
             return
-        from ..outputs.luxcore_api import pyluxcore
+        import pyluxcore
         from ..export.luxcore.materialpreview import MaterialPreviewExporter
 
         try:
diff --git a/src/luxrender/export/luxcore/__init__.py b/src/luxrender/export/luxcore/__init__.py
--- a/src/luxrender/export/luxcore/__init__.py
+++ b/src/luxrender/export/luxcore/__init__.py
@@ -28,7 +28,7 @@
 import bpy, time, os
 
 from ...outputs import LuxManager, LuxLog
-from ...outputs.luxcore_api import pyluxcore
+import pyluxcore
 from ...outputs.luxcore_api import ToValidLuxCoreName
 
 from ...extensions_framework import util as efutil
diff --git a/src/luxrender/export/luxcore/camera.py b/src/luxrender/export/luxcore/camera.py
--- a/src/luxrender/export/luxcore/camera.py
+++ b/src/luxrender/export/luxcore/camera.py
@@ -28,7 +28,8 @@
 import bpy, math, mathutils
 from mathutils import Vector
 
-from ...outputs.luxcore_api import pyluxcore, set_prop_cam
+import pyluxcore
+from ...outputs.luxcore_api import set_prop_cam
 from ...export import get_worldscale
 from ...export import object_anim_matrices
 from ...export import fix_matrix_order
diff --git a/src/luxrender/export/luxcore/config.py b/src/luxrender/export/luxcore/config.py
--- a/src/luxrender/export/luxcore/config.py
+++ b/src/luxrender/export/luxcore/config.py
@@ -27,7 +27,7 @@
 
 import os
 
-from ...outputs.luxcore_api import pyluxcore
+import pyluxcore
 from ...extensions_framework import util as efutil
 from ...export import get_output_filename
 from .utils import is_lightgroup_opencl_compatible
diff --git a/src/luxrender/export/luxcore/duplis.py b/src/luxrender/export/luxcore/duplis.py
--- a/src/luxrender/export/luxcore/duplis.py
+++ b/src/luxrender/export/luxcore/duplis.py
@@ -26,7 +26,7 @@
 #
 
 import math, mathutils, time
-from ...outputs.luxcore_api import pyluxcore
+import pyluxcore
 from ...outputs.luxcore_api import ToValidLuxCoreName
 from ...export import matrix_to_list, is_obj_visible
 
diff --git a/src/luxrender/export/luxcore/lights.py b/src/luxrender/export/luxcore/lights.py
--- a/src/luxrender/export/luxcore/lights.py
+++ b/src/luxrender/export/luxcore/lights.py
@@ -27,7 +27,7 @@
 
 import bpy, mathutils, math, os
 
-from ...outputs.luxcore_api import pyluxcore
+import pyluxcore
 from ...outputs.luxcore_api import ToValidLuxCoreName
 from ...export import is_obj_visible
 from ...export import get_worldscale
diff --git a/src/luxrender/export/luxcore/materialpreview.py b/src/luxrender/export/luxcore/materialpreview.py
--- a/src/luxrender/export/luxcore/materialpreview.py
+++ b/src/luxrender/export/luxcore/materialpreview.py
@@ -27,7 +27,7 @@
 
 from mathutils import Matrix
 
-from ...outputs.luxcore_api import pyluxcore
+import pyluxcore
 from ...export import matrix_to_list
 
 from . import LuxCoreExporter
diff --git a/src/luxrender/export/luxcore/materials.py b/src/luxrender/export/luxcore/materials.py
--- a/src/luxrender/export/luxcore/materials.py
+++ b/src/luxrender/export/luxcore/materials.py
@@ -27,7 +27,7 @@
 
 import bpy
 
-from ...outputs.luxcore_api import pyluxcore
+import pyluxcore
 from ...outputs.luxcore_api import ToValidLuxCoreName
 from ...export.materials import get_texture_from_scene
 from ...export import get_expanded_file_name
diff --git a/src/luxrender/export/luxcore/meshes.py b/src/luxrender/export/luxcore/meshes.py
--- a/src/luxrender/export/luxcore/meshes.py
+++ b/src/luxrender/export/luxcore/meshes.py
@@ -29,7 +29,7 @@
 
 import bpy
 
-from ...outputs.luxcore_api import pyluxcore
+import pyluxcore
 from ...outputs.luxcore_api import ToValidLuxCoreName
 
 
diff --git a/src/luxrender/export/luxcore/objects.py b/src/luxrender/export/luxcore/objects.py
--- a/src/luxrender/export/luxcore/objects.py
+++ b/src/luxrender/export/luxcore/objects.py
@@ -28,7 +28,7 @@
 import os
 
 from ...extensions_framework import util as efutil
-from ...outputs.luxcore_api import pyluxcore
+import pyluxcore
 from ...outputs.luxcore_api import ToValidLuxCoreName
 from ...export import is_obj_visible
 from ...export import object_anim_matrices
diff --git a/src/luxrender/export/luxcore/textures.py b/src/luxrender/export/luxcore/textures.py
--- a/src/luxrender/export/luxcore/textures.py
+++ b/src/luxrender/export/luxcore/textures.py
@@ -28,7 +28,7 @@
 import bpy, math, mathutils, os, tempfile
 
 from ...extensions_framework import util as efutil
-from ...outputs.luxcore_api import pyluxcore
+import pyluxcore
 from ...outputs.luxcore_api import ToValidLuxCoreName
 from ...export import matrix_to_list
 from ...export import get_expanded_file_name
diff --git a/src/luxrender/export/luxcore/utils.py b/src/luxrender/export/luxcore/utils.py
--- a/src/luxrender/export/luxcore/utils.py
+++ b/src/luxrender/export/luxcore/utils.py
@@ -27,7 +27,7 @@
 
 from  math import pi
 
-from ...outputs.luxcore_api import pyluxcore
+import pyluxcore
 from ...outputs.luxcore_api import ToValidLuxCoreName
 from ...export.materials import get_texture_from_scene
 from ...export import get_worldscale
diff --git a/src/luxrender/export/luxcore/volumes.py b/src/luxrender/export/luxcore/volumes.py
--- a/src/luxrender/export/luxcore/volumes.py
+++ b/src/luxrender/export/luxcore/volumes.py
@@ -27,7 +27,7 @@
 
 import math
 
-from ...outputs.luxcore_api import pyluxcore
+import pyluxcore
 from ...properties import find_node_in_volume
 
 from .utils import convert_texture_channel, generate_volume_name
diff --git a/src/luxrender/operators/__init__.py b/src/luxrender/operators/__init__.py
--- a/src/luxrender/operators/__init__.py
+++ b/src/luxrender/operators/__init__.py
@@ -279,7 +279,7 @@
             devs.remove(0)
 
         # Create the new list
-        from ..outputs.luxcore_api import pyluxcore
+        import pyluxcore
 
         deviceList = pyluxcore.GetOpenCLDeviceList()
         for dev in deviceList:
