=== modified file 'applets/taskmanager/task-manager.c'
--- applets/taskmanager/task-manager.c	2011-01-15 16:44:44 +0000
+++ applets/taskmanager/task-manager.c	2011-03-12 21:42:33 +0000
@@ -1611,11 +1611,6 @@
   g_debug ("%s: Window opened: %s",__func__,wnck_window_get_name (window));  
   g_debug ("xid = %lu, pid = %d",wnck_window_get_xid (window),wnck_window_get_pid (window));
 #endif  
-  /* 
-    for some reason the skip tasklist property for the taskmanager toggles briefly
-   off and on in certain circumstances.  Nip this in the bud.
-   TODO:  Investigate wth this is happening...  it bothers me.
-   */
 //  if ( wnck_window_get_pid (window) == getpid() || 
   if ( g_strcmp0 (wnck_window_get_name (window),"awn-applet")==0 )
   {
@@ -1626,11 +1621,6 @@
    */
   g_signal_connect (window, "state-changed", G_CALLBACK (on_window_state_changed), manager);
 
-  if (wnck_window_is_skip_tasklist (window))
-  {
-    return;
-  }
-
   g_signal_connect (window, "state-changed", 
                     G_CALLBACK (check_attention_requested), manager);    
   
@@ -1798,6 +1787,11 @@
   g_return_if_fail (TASK_IS_MANAGER (manager));
   g_return_if_fail (WNCK_IS_WINDOW (window));
 
+  if (wnck_window_is_skip_tasklist(window))
+  {
+    return;
+  }
+
   _wnck_get_wmclass (wnck_window_get_xid (window),
                      &res_name, &class_name);
   if (get_special_wait_from_window_data (res_name, 

