From bf62d389bdbe2085f765f648de536eb44979e448 Mon Sep 17 00:00:00 2001
From: Steven Myint <git@stevenmyint.com>
Date: Fri, 10 Jul 2015 07:17:51 -0700
Subject: [PATCH] Fix a test under Python 3.5

---
 pyflakes/test/test_api.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/pyflakes/test/test_api.py b/pyflakes/test/test_api.py
index ad61983..a83a7fd 100644
--- a/pyflakes/test/test_api.py
+++ b/pyflakes/test/test_api.py
@@ -385,12 +385,18 @@ def test_nonKeywordAfterKeywordSyntaxError(self):
         sourcePath = self.makeTempFile(source)
         last_line = '            ^\n' if sys.version_info >= (3, 2) else ''
         column = '13:' if sys.version_info >= (3, 2) else ''
+
+        if sys.version_info >= (3, 5):
+            message = 'positional argument follows keyword argument'
+        else:
+            message = 'non-keyword arg after keyword arg'
+
         self.assertHasErrors(
             sourcePath,
             ["""\
-%s:1:%s non-keyword arg after keyword arg
+%s:1:%s %s
 foo(bar=baz, bax)
-%s""" % (sourcePath, column, last_line)])
+%s""" % (sourcePath, column, message, last_line)])
 
     def test_invalidEscape(self):
         """
