diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2251518..7ab4a02 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,7 +76,7 @@ else (WIN32)
   find_package(Boost "1.33.1" COMPONENTS graph)
 endif (WIN32)
 # needed by libkgpgfile
-find_package(QGpgme REQUIRED)
+find_package(Gpgmepp REQUIRED)
 find_package(KdepimLibs REQUIRED)
 find_package(SharedMimeInfo REQUIRED)
 
diff --git a/libkgpgfile/CMakeLists.txt b/libkgpgfile/CMakeLists.txt
index a41a6a4..d60a93c 100644
--- a/libkgpgfile/CMakeLists.txt
+++ b/libkgpgfile/CMakeLists.txt
@@ -7,6 +7,6 @@ set(kgpgfile_STAT_SRCS
    )
 
 kde4_add_library(kgpgfile STATIC ${kgpgfile_STAT_SRCS})
-target_link_libraries(kgpgfile ${QGPGME_LIBRARIES})
+target_link_libraries(kgpgfile Gpgmepp)
 
 
diff --git a/libkgpgfile/kgpgfile.cpp b/libkgpgfile/kgpgfile.cpp
index b1870be..a5d3d4e 100644
--- a/libkgpgfile/kgpgfile.cpp
+++ b/libkgpgfile/kgpgfile.cpp
@@ -47,7 +47,6 @@
 #include <gpgme++/keylistresult.h>
 #include <gpgme++/key.h>
 #include <gpgme++/data.h>
-#include <qgpgme/dataprovider.h>
 
 class KGPGFile::Private
 {

