diff -ruNa sqlalchemy-migrate-0.7.2.orig/migrate/changeset/ansisql.py sqlalchemy-migrate-0.7.2/migrate/changeset/ansisql.py
--- sqlalchemy-migrate-0.7.2.orig/migrate/changeset/ansisql.py	2011-10-28 17:11:43.000000000 +0200
+++ sqlalchemy-migrate-0.7.2/migrate/changeset/ansisql.py	2014-01-13 13:02:08.456043268 +0100
@@ -7,7 +7,7 @@
 import StringIO
 
 import sqlalchemy as sa
-from sqlalchemy.schema import SchemaVisitor
+from sqlalchemy.sql.base import SchemaVisitor
 from sqlalchemy.engine.default import DefaultDialect
 from sqlalchemy.sql import ClauseElement
 from sqlalchemy.schema import (ForeignKeyConstraint,
