diff -ruN a/Documentation/libtracecmd/Makefile b/Documentation/libtracecmd/Makefile
--- a/Documentation/libtracecmd/Makefile	2021-05-14 16:20:12.000000000 +0200
+++ b/Documentation/libtracecmd/Makefile	2021-05-14 21:32:15.921462439 +0200
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 
 # Include the utils
-include $(src)/scripts/utils.mk
+include ../../scripts/utils.mk
 
 # This Makefile and manpage XSL files were taken from libtracefs
 # and modified for libtracecmd
@@ -134,7 +134,7 @@
 endif
 
 install-%.3: $(OUTPUT)%.3
-	$(Q)$(call do_install_docs,$<,$(man3dir),644);
+	$(Q)$(call do_install_docs,$<,$(DESTDIR)$(man3dir),644);
 
 do-install-man: man $(addprefix install-,$(wildcard $(OUTPUT)*.3))
 
@@ -142,7 +142,7 @@
 	$(Q)$(MAKE) -C . do-install-man
 
 install-%.txt: $(OUTPUT)%.html
-	$(Q)$(call do_install_docs,$<,$(htmldir),644);
+	$(Q)$(call do_install_docs,$<,$(DESTDIR)$(htmldir),644);
 
 do-install-html: html $(addprefix install-,$(wildcard *.txt))
 
diff -ruN a/Documentation/trace-cmd/Makefile b/Documentation/trace-cmd/Makefile
--- a/Documentation/trace-cmd/Makefile	2021-05-14 16:20:12.000000000 +0200
+++ b/Documentation/trace-cmd/Makefile	2021-05-14 21:26:53.224092234 +0200
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 
 # Include the utils
-include $(src)/scripts/utils.mk
+include ../../scripts/utils.mk
 
 bdir:=$(doc_dir)/trace-cmd
 
