diff --git a/src/filesystem/dpathbuf.h b/src/filesystem/dpathbuf.h
index 3aa7609..85df90d 100644
--- a/src/filesystem/dpathbuf.h
+++ b/src/filesystem/dpathbuf.h
@@ -26,7 +26,10 @@ DCORE_BEGIN_NAMESPACE
 class LIBDTKCORESHARED_EXPORT DPathBuf
 {
 public:
-    DPathBuf(const QString &path);
+    DPathBuf(const QString &path)
+    {
+        m_path = QDir(path).absolutePath();
+    }
 
     DPathBuf operator/(const QString &p) const
     {
diff --git a/src/filesystem/filesystem.pri b/src/filesystem/filesystem.pri
index c8d81a0..e561867 100644
--- a/src/filesystem/filesystem.pri
+++ b/src/filesystem/filesystem.pri
@@ -14,8 +14,7 @@ SOURCES += \
     $$PWD/dbasefilewatcher.cpp \
     $$PWD/dfilewatcher.cpp \
     $$PWD/dfilewatchermanager.cpp \
-    $$PWD/dstandardpaths.cpp \
-    $$PWD/dpathbuf.cpp
+    $$PWD/dstandardpaths.cpp
 
 linux {
     SOURCES += \
