diff -ur gtk+-2.20.1/gtk/gtkrc.c gtk+-2.20.1-32/gtk/gtkrc.c
--- gtk+-2.20.1/gtk/gtkrc.c	2010-05-03 01:28:21.000000000 +0200
+++ gtk+-2.20.1-32/gtk/gtkrc.c	2010-08-26 07:22:42.316920033 +0200
@@ -450,7 +450,7 @@
       if (im_module_file)
 	result = g_strdup (im_module_file);
       else
-	result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtk.immodules", NULL);
+	result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtk.immodules-32", NULL);
     }
 
   return result;
