diff -ur tigervnc-1.2.0.orig/CMakeLists.txt tigervnc-1.2.0/CMakeLists.txt
--- tigervnc-1.2.0.orig/CMakeLists.txt	2012-03-09 21:34:29.000000000 +0100
+++ tigervnc-1.2.0/CMakeLists.txt	2012-03-27 23:14:12.292068145 +0200
@@ -344,7 +344,7 @@
       # system's version (if available) to perform this test.
       set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES};-lz)
     endif()
-    check_function_exists(gnutls_transport_set_global_errno HAVE_OLD_GNUTLS)
+    check_function_exists(gnutls_transport_set_global_errno HAVE_GNUTLS_GLOBAL_ERRNO)
     check_function_exists(gnutls_x509_crt_print HAVE_GNUTLS_X509_CRT_PRINT)
     check_type_size(gnutls_x509_crt_t GNUTLS_X509_CRT_T)
     check_type_size(gnutls_datum_t GNUTLS_DATUM_T)
diff -ur tigervnc-1.2.0.orig/common/rdr/TLSInStream.cxx tigervnc-1.2.0/common/rdr/TLSInStream.cxx
--- tigervnc-1.2.0.orig/common/rdr/TLSInStream.cxx	2010-09-30 08:25:28.000000000 +0200
+++ tigervnc-1.2.0/common/rdr/TLSInStream.cxx	2012-03-27 23:14:11.166068223 +0200
@@ -27,7 +27,7 @@
 #include <rdr/TLSInStream.h>
 #include <errno.h>
 
-#ifdef HAVE_OLD_GNUTLS
+#ifndef HAVE_GNUTLS_GLOBAL_ERRNO
 #define gnutls_transport_set_global_errno(A) do { errno = (A); } while(0)
 #endif
 
diff -ur tigervnc-1.2.0.orig/common/rdr/TLSOutStream.cxx tigervnc-1.2.0/common/rdr/TLSOutStream.cxx
--- tigervnc-1.2.0.orig/common/rdr/TLSOutStream.cxx	2010-09-30 08:25:28.000000000 +0200
+++ tigervnc-1.2.0/common/rdr/TLSOutStream.cxx	2012-03-27 23:14:09.576068335 +0200
@@ -27,7 +27,7 @@
 #include <rdr/TLSOutStream.h>
 #include <errno.h>
 
-#ifdef HAVE_OLD_GNUTLS
+#ifndef HAVE_GNUTLS_GLOBAL_ERRNO
 #define gnutls_transport_set_global_errno(A) do { errno = (A); } while(0)
 #endif
 
