diff -wbBur xloadimage.4.1/png.c xloadimage.4.1.my/png.c
--- xloadimage.4.1/png.c	2011-02-11 14:47:37.000000000 +0300
+++ xloadimage.4.1.my/png.c	2012-01-19 11:58:44.000000000 +0400
@@ -75,7 +75,7 @@
 {
   debug(" #error ");
   output_warn( png_ptr, str);
-  longjmp(png_ptr->jmpbuf, 1);	/* return control to outer routine */
+  longjmp(png_jmpbuf(png_ptr), 1);	/* return control to outer routine */
 }
 
 
@@ -164,7 +164,7 @@
     png_destroy_read_struct(png_pp, info_pp, end_pp);
     return 0;
   }
-  if (setjmp((*png_pp)->jmpbuf)) {
+  if (setjmp(png_jmpbuf(*png_pp))) {
     /* On error */
     png_destroy_read_struct(png_pp, info_pp, end_pp);
     return 0;
@@ -220,7 +220,7 @@
     zclose(zinput_file);
     return 0;
   }
-  if (setjmp(png_ptr->jmpbuf)) {
+  if (setjmp(png_jmpbuf(png_ptr))) {
     /* On error */
     freeImage(image);
     png_destroy_read_struct(&png_ptr, &info_ptr, &end_info);
diff -wbBur xloadimage.4.1/tiff.c xloadimage.4.1.my/tiff.c
--- xloadimage.4.1/tiff.c	2011-02-11 14:47:38.000000000 +0300
+++ xloadimage.4.1.my/tiff.c	2012-01-19 12:00:00.000000000 +0400
@@ -34,14 +34,14 @@
      struct tiff_info *info;
 {
   ZFILE *zf;
-  TIFFHeader th;
+  TIFFHeaderCommon th;
   TIFF *tiff;
 
   zf = zopen(fullname);
 
   /* read TIFF header and see if it looks right
    */
-  if ((zread(zf, (byte *)&th, sizeof(TIFFHeader)) == sizeof(TIFFHeader)) &&
+  if ((zread(zf, (byte *)&th, sizeof(TIFFHeaderCommon)) == sizeof(TIFFHeaderCommon)) &&
       ((th.tiff_magic == TIFF_BIGENDIAN) ||
        (th.tiff_magic == TIFF_LITTLEENDIAN))) {
 
