From d10dd4a3100ec12d35ffbfa936d8553f7ecf0036 Mon Sep 17 00:00:00 2001
From: Fabian Affolter <mail@fabian-affolter.ch>
Date: Wed, 2 Nov 2022 08:29:03 +0100
Subject: [PATCH] Fix style

---
 pyproject.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index dff94b0..9b0c888 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -8,8 +8,8 @@ readme = "README.md"
 
 [tool.poetry.dependencies]
 python = "^3.6"
-pydantic = ">=0.32.2<2.0.0"
-hypothesis = ">=4.36<6.0.0"
+pydantic = ">=0.32.2, <2.0.0"
+hypothesis = ">=4.36, <6.0.0"
 pytest = { version = "^4.0.0", optional = true }
 
 [tool.poetry.extras]
