diff -wbBur kpacman-0.3.2/kpacman/keys.cpp kpacman-0.3.2.my/kpacman/keys.cpp
--- kpacman-0.3.2/kpacman/keys.cpp	2003-04-21 22:59:22.000000000 +0400
+++ kpacman-0.3.2.my/kpacman/keys.cpp	2007-02-13 12:49:37.000000000 +0300
@@ -13,7 +13,7 @@
 #include <qpixmap.h>
 #include <qstring.h>
 
-Keys::Keys( QWidget *parent, const char *name)
+KPKeys::KPKeys( QWidget *parent, const char *name)
     : QDialog( parent, name, TRUE )
 {
     KStandardDirs *dirs = KGlobal::dirs();
@@ -88,7 +88,7 @@
     init();
 }
 
-void Keys::keyPressEvent( QKeyEvent *e )
+void KPKeys::keyPressEvent( QKeyEvent *e )
 {
         uint kCode = e->key() & ~(SHIFT | CTRL | ALT);
         QString string = KAccel::keyToString(kCode);
@@ -103,27 +103,27 @@
             reject();
 }
 
-void Keys::butUp()
+void KPKeys::butUp()
 {
     getKey(0);
 }
 
-void Keys::butDown()
+void KPKeys::butDown()
 {
     getKey(1);
 }
 
-void Keys::butLeft()
+void KPKeys::butLeft()
 {
     getKey(2);
 }
 
-void Keys::butRight()
+void KPKeys::butRight()
 {
     getKey(3);
 }
 
-void Keys::getKey(int i)
+void KPKeys::getKey(int i)
 {
     if ( lab != 0)
             focusOut(lab);
@@ -131,14 +131,14 @@
     focusIn(labels[i]);
 }
 
-void Keys::focusOut(QLabel *l)
+void KPKeys::focusOut(QLabel *l)
 {
     l->setFrameStyle( QFrame::NoFrame );
     l->setBackgroundColor(backgroundColor());
     l->repaint();
 }
 
-void Keys::focusIn(QLabel *l)
+void KPKeys::focusIn(QLabel *l)
 {
     lab = l;
     lab->setFrameStyle( QFrame::Panel | QFrame::Sunken );
@@ -146,7 +146,7 @@
     lab->repaint();
 }
 
-void Keys::defaults()
+void KPKeys::defaults()
 {
     if ( lab != 0)
             focusOut(lab);
@@ -159,7 +159,7 @@
     labels[3]->setText("Right");
 }
 
-void Keys::init()
+void KPKeys::init()
 {
     QString up("Up");
     up = kapp->config()->readEntry("upKey", (const char*) up);
@@ -178,7 +178,7 @@
     labels[3]->setText(right);
 }
 
-void Keys::ok()
+void KPKeys::ok()
 {
     kapp->config()->writeEntry("upKey",   (const char*) labels[0]->text() );
     kapp->config()->writeEntry("downKey", (const char*) labels[1]->text() );
diff -wbBur kpacman-0.3.2/kpacman/keys.h kpacman-0.3.2.my/kpacman/keys.h
--- kpacman-0.3.2/kpacman/keys.h	2002-02-27 18:13:24.000000000 +0300
+++ kpacman-0.3.2.my/kpacman/keys.h	2007-02-13 12:50:09.000000000 +0300
@@ -13,11 +13,11 @@
 
 #include <kaccel.h>
 
-class Keys : public QDialog
+class KPKeys : public QDialog
 {
     Q_OBJECT
 public:
-    Keys( QWidget *parent=0, const char *name=0 );
+    KPKeys( QWidget *parent=0, const char *name=0 );
 
 private slots:
     void butRight();
diff -wbBur kpacman-0.3.2/kpacman/kpacman.cpp kpacman-0.3.2.my/kpacman/kpacman.cpp
--- kpacman-0.3.2/kpacman/kpacman.cpp	2003-06-18 20:01:27.000000000 +0400
+++ kpacman-0.3.2.my/kpacman/kpacman.cpp	2007-02-13 12:51:14.000000000 +0300
@@ -29,6 +29,7 @@
 
 // application specific includes
 #include "kpacman.h"
+#include <keys.h>
 
 #define ID_STATUS_MSG 1
 
@@ -445,7 +446,7 @@
   slotStatusMsg(i18n("Configure key bindings..."));
   ///////////////////////////////////////////////////////////////////
   // configure key bindings
-  Keys *keys = new Keys();
+  KPKeys *keys = new KPKeys();
   if (keys->exec() == QDialog::Accepted) {
     view->referee->initKeys();
     view->score->initKeys();
