--- M2Crypto-0.25.1/setup.py	2016-07-22 14:43:17.000000000 +0200
+++ M2Crypto-0.25.1.dirty/setup.py	2017-03-23 18:18:35.128517781 +0100
@@ -70,8 +70,8 @@
             self.libraries = ['ssleay32', 'libeay32']
             self.openssl = 'c:\\pkg'
         else:
-            self.libraries = ['ssl', 'crypto']
-            self.openssl = '/usr'
+            self.libraries = [':libssl.so.1.0.0', ':libcrypto.so.1.0.0']
+            self.openssl = '/usr/lib/openssl-1.0/'
 
     def finalize_options(self):
         '''Overloaded build_ext implementation to append custom openssl
@@ -87,16 +87,11 @@
         if _openssl and os.path.isdir(_openssl):
             self.openssl = _openssl
 
-        self.include_dirs.append(os.path.join(self.openssl, 'include'))
-        openssl_library_dir = os.path.join(self.openssl, 'lib')
+        self.include_dirs.append(os.path.join('/usr/include/', 'openssl-1.0'))
+
+        openssl_library_dir = os.path.join('/usr/lib/', 'openssl-1.0')
 
         if platform.system() == "Linux":
-            if _multiarch:  # on Fedora/RHEL it is an empty string
-                self.include_dirs.append(
-                    os.path.join(self.openssl, 'include', _multiarch))
-            else:
-                self.include_dirs.append(
-                    os.path.join(self.openssl, 'include', 'openssl'))
 
             # For RedHat-based distros, the '-D__{arch}__' option for
             # Swig needs to be normalized, particularly on i386.
