diff -wbBur MediaInfoLib/Source/MediaInfo/Reader/Reader_libmms.cpp MediaInfoLib.my/Source/MediaInfo/Reader/Reader_libmms.cpp
--- MediaInfoLib/Source/MediaInfo/Reader/Reader_libmms.cpp	2011-08-25 15:14:50.000000000 +0400
+++ MediaInfoLib.my/Source/MediaInfo/Reader/Reader_libmms.cpp	2011-10-05 17:29:12.000000000 +0400
@@ -48,10 +48,8 @@
 #else
     #ifdef MEDIAINFO_LIBMMS_FROMSOURCE
         #include "mmsx.h"
-        #include "mmsh.h"
     #else //MEDIAINFO_LIBMMS_FROMSOURCE
         #include "libmms/mmsx.h"
-        #include "libmms/mmsh.h"
     #endif //MEDIAINFO_LIBMMS_FROMSOURCE
 #endif
 #include <iostream>
@@ -71,34 +69,9 @@
 //---------------------------------------------------------------------------
 size_t Reader_libmms::Format_Test(MediaInfo_Internal* MI, const String &File_Name)
 {
-    mmsx_t* Handle;
-
-    //Opening the file
-    #if MEDIAINFO_LIBMMS_DESCRIBE_SUPPORT
-    if (MI->Config.File_Mmsh_Describe_Only_Get())
-    {
-        // Use MMSH & Send a DESCRIBE request
-        mmsh_t* MmshHandle;
-
-        MmshHandle=mmsh_describe_request(0, 0, Ztring(File_Name).To_Local().c_str());
-        if (MmshHandle==NULL)
-            return 0;
-
-        Handle=mmsx_set_mmsh_handle(MmshHandle);
-        if (Handle==NULL)
-        {
-            mmsh_close(MmshHandle);
-            return 0;
-        }
-    }
-    else
-    #endif //MEDIAINFO_LIBMMS_DESCRIBE_SUPPORT
-    {
-       // Use MMS or MMSH (Send a DESCRIBE & PLAY request)
-       Handle=mmsx_connect(0, 0, Ztring(File_Name).To_Local().c_str(), (int)-1);
+    mmsx_t* Handle=mmsx_connect(0, 0, Ztring(File_Name).To_Local().c_str(), (int)-1);
        if (Handle==NULL)
            return 0;
-    }
 
     //Init
     size_t Buffer_Size_Max;
