diff --git a/src/mcm-screen.c b/src/mcm-screen.c
index d555a37..aeec5b9 100644
--- a/src/mcm-screen.c
+++ b/src/mcm-screen.c
@@ -91,11 +91,13 @@ mcm_screen_ensure_instance (McmScreen *screen, GError **error)
 		goto out;
 
 	/* get screen (this is slow) */
-	priv->rr_screen = mate_rr_screen_new (gdk_screen_get_default (), (MateRRScreenChanged) mcm_screen_randr_event_cb, screen, error);
+	priv->rr_screen = mate_rr_screen_new (gdk_screen_get_default (), error);
 	if (priv->rr_screen == NULL) {
 		ret = FALSE;
 		goto out;
 	}
+
+	g_signal_connect (priv->rr_screen, "changed", G_CALLBACK (mcm_screen_randr_event_cb), screen);
 out:
 	return ret;
 }
@@ -196,7 +198,7 @@ mcm_screen_finalize (GObject *object)
 	McmScreenPrivate *priv = screen->priv;
 
 	if (priv->rr_screen != NULL)
-		mate_rr_screen_destroy (priv->rr_screen);
+		g_object_unref (priv->rr_screen);
 
 	G_OBJECT_CLASS (mcm_screen_parent_class)->finalize (object);
 }
