diff -ruN kchmviewer-6.1-orig/packages/kchmviewer.desktop kchmviewer-6.1/packages/kchmviewer.desktop
--- kchmviewer-6.1-orig/packages/kchmviewer.desktop	2009-12-15 07:02:57.000000000 +0100
+++ kchmviewer-6.1/packages/kchmviewer.desktop	2014-05-12 20:37:00.983182220 +0200
@@ -3,7 +3,6 @@
 Categories=Qt;KDE;Office;Viewer;
 Terminal=false
 Type=Application
-Encoding=UTF-8
 Exec=kchmviewer %f
 Icon=kchmviewer
 GenericName=CHM file viewer
diff -ruN kchmviewer-6.1-orig/src/CMakeLists.txt kchmviewer-6.1/src/CMakeLists.txt
--- kchmviewer-6.1-orig/src/CMakeLists.txt	2011-12-19 08:16:43.000000000 +0100
+++ kchmviewer-6.1/src/CMakeLists.txt	2014-05-12 20:35:51.195078972 +0200
@@ -8,7 +8,7 @@
 	dialog_chooseurlfromlist.cpp
 	dialog_setup.cpp
 	kde-qt.cpp
-	keyeventfilter.cpp
+#	keyeventfilter.cpp
 	main.cpp
 	mainwindow.cpp
 	navigationpanel.cpp
