diff -Nur compiz-0.8.4.orig/gtk/window-decorator/gtk-window-decorator.c compiz-0.8.4/gtk/window-decorator/gtk-window-decorator.c
--- compiz-0.8.4.orig/gtk/window-decorator/gtk-window-decorator.c	2009-08-15 10:24:40.000000000 +0300
+++ compiz-0.8.4/gtk/window-decorator/gtk-window-decorator.c	2010-03-31 14:46:30.860771323 +0300
@@ -4233,7 +4233,11 @@
 static void
 hide_tooltip (void)
 {
+#if GTK_CHECK_VERSION(2,19,7)
+    if (gtk_widget_get_realized(tip_window))
+#else
     if (GTK_WIDGET_VISIBLE (tip_window))
+#endif
 	g_get_current_time (&tooltip_last_popdown);
 
     gtk_widget_hide (tip_window);
