From c6781c4eae7f820c1d832177539bc90e1ec4e3e4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 11 Jan 2018 11:12:22 +0100
Subject: [PATCH] Fix string comparison

---
 src/nautilus-python.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/nautilus-python.c b/src/nautilus-python.c
index 694d555..bbffb97 100644
--- a/src/nautilus-python.c
+++ b/src/nautilus-python.c
@@ -248,7 +248,7 @@ nautilus_python_check_all_directories(GTypeModule *module) {
     while (*temp != NULL) {
         gchar *dir = g_build_filename(*temp,
             "nautilus-python", "extensions", NULL);
-        if (dir != prefix_extension_dir) {
+        if (g_strcmp0 (dir, prefix_extension_dir) != 0) {
             dirs = g_list_append(dirs, dir);
         }
 
-- 
2.15.1

