diff -ur gambas2-2.13.1.orig/gb.db.sqlite2/src/main.cpp gambas2-2.13.1/gb.db.sqlite2/src/main.cpp
--- gambas2-2.13.1.orig/gb.db.sqlite2/src/main.cpp	2009-05-26 21:41:45.000000000 +0300
+++ gambas2-2.13.1/gb.db.sqlite2/src/main.cpp	2009-06-15 20:43:18.411751913 +0300
@@ -932,7 +932,7 @@
 {
   char *fld;
 
-  fld = strchr(name, (int)FLD_SEP);
+  fld = strchr((char *) name, (int) FLD_SEP);
   if (fld){ //Includes table identity
      fld[0] = '.';
   }
