commit 73103a21fa8b09f4a2d0ee1886204e3bb9e17252
Author: Anatol Pomozov <anatol.pomozov@gmail.com>
Date:   Sat Apr 22 06:54:02 2017 -0700

    Remove tests that fail because of #4130

diff --git a/spec/std/socket_spec.cr b/spec/std/socket_spec.cr
index 56e565f86..f5554893b 100644
--- a/spec/std/socket_spec.cr
+++ b/spec/std/socket_spec.cr
@@ -536,18 +536,6 @@ describe TCPSocket do
       TCPSocket.new("localhost", port)
     end
   end
-
-  it "fails when host doesn't exist" do
-    expect_raises(Socket::Error, /No address found for doesnotexist.example.org.:12345/) do
-      TCPSocket.new("doesnotexist.example.org.", 12345)
-    end
-  end
-
-  it "fails (rather than segfault on darwin) when host doesn't exist and port is 0" do
-    expect_raises(Socket::Error, /No address found for doesnotexist.example.org.:0/) do
-      TCPSocket.new("doesnotexist.example.org.", 0)
-    end
-  end
 end
 
 describe UDPSocket do
