diff -ruN a/tests/test_core.py b/tests/test_core.py
--- a/tests/test_core.py	2020-01-22 10:39:39.000000000 +0100
+++ b/tests/test_core.py	2020-01-24 09:29:48.120558405 +0100
@@ -8,8 +8,6 @@
 from django.template.engine import Engine
 from django.template.loader import render_to_string
 
-import pep8
-
 from sekizai import context_processors
 from sekizai.context import SekizaiContext
 from sekizai.helpers import (
@@ -212,17 +210,6 @@
         self.assertTrue(result.status, result.message)
         return rendered
 
-    def test_pep8(self):
-        sekizai_dir = os.path.dirname(os.path.abspath(__file__))
-        pep8style = pep8.StyleGuide()
-        with CaptureStdout() as stdout:
-            result = pep8style.check_files([sekizai_dir])
-            errors = stdout.getvalue()
-        self.assertEqual(
-            result.total_errors, 0,
-            "Code not PEP8 compliant:\n{0}".format(errors)
-        )
-
     def test_basic_dual_block(self):
         """
         Basic dual block testing
