From 02823635bf3bc149892ce75efffcc33a33e184c5 Mon Sep 17 00:00:00 2001
From: Brett Cornwall <brett@i--b.com>
Date: Fri, 25 Dec 2020 16:29:03 -0800
Subject: [PATCH] make: Append, don't overwrite user defs

---
 config.mk | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/config.mk b/config.mk
index 57bf7a3..619121d 100644
--- a/config.mk
+++ b/config.mk
@@ -1,18 +1,16 @@
-ifndef VERSION
-  VERSION = "development-version"
-endif
+VERSION ?= "development-version"
 
-PREFIX = /usr/local
+PREFIX ?= /usr/local
 
-INCS = -I/usr/include/libevdev-1.0
+INCS += -I/usr/include/libevdev-1.0
 
-CPPFLAGS = $(INCS) -DVERSION=\"$(VERSION)\"
+CPPFLAGS += $(INCS) -DVERSION=\"$(VERSION)\"
 
-COMPFLAGS = -pedantic -Wall -Wextra -O3
-CFLAGS = $(COMPFLAGS) -std=c99
+COMPFLAGS += -pedantic -Wall -Wextra -O3
+CFLAGS += $(COMPFLAGS) -std=c99
 CXXFLAGS = $(COMPFLAGS) -std=c++11
 
-LDFLAGS = -rdynamic -lyaml-cpp -levdev
+LDFLAGS += -rdynamic -lyaml-cpp -levdev
 
 CC = cc
 CXX = c++
-- 
2.29.2

