diff --git a/Text/Show/ByteString/Integer.hs b/Text/Show/ByteString/Integer.hs
index e1a30cd..46d5115 100644
--- a/Text/Show/ByteString/Integer.hs
+++ b/Text/Show/ByteString/Integer.hs
@@ -19,7 +19,7 @@ import GHC.Base
 #if   __GLASGOW_HASKELL__ && __GLASGOW_HASKELL__ <  611 && INTEGER_GMP
 import GHC.Integer.Internals
 #elif __GLASGOW_HASKELL__ && __GLASGOW_HASKELL__ >= 611 && INTEGER_GMP
-import GHC.Integer.GMP.Internals
+import GHC.Integer.GMP.Internals hiding (quotRemInteger)
 #elif __GLASGOW_HASKELL__ && INTEGER_SIMPLE
 import GHC.Integer.Simple.Internals
 #endif
diff --git a/bytestring-show.cabal b/bytestring-show.cabal
index 0da6800..f37c649 100644
--- a/bytestring-show.cabal
+++ b/bytestring-show.cabal
@@ -16,7 +16,7 @@ flag integer-simple
         description: use with integer-simple build of GHC
 
 library
-    build-depends: base < 5, binary < 0.8, bytestring >= 0.9 && <= 1, array < 0.6, containers < 0.6
+    build-depends: base < 5, binary < 0.9, bytestring >= 0.9 && <= 1, array < 0.6, containers < 0.7
 
     exposed-modules:
         Text.Show.ByteString
