diff -ruN a/test/CMakeLists.txt b/test/CMakeLists.txt
--- a/test/CMakeLists.txt	2021-06-16 19:21:46.000000000 +0200
+++ b/test/CMakeLists.txt	2021-12-19 16:06:28.983252614 +0100
@@ -1,12 +1,12 @@
 add_library(gtest STATIC
-  "${CMAKE_CURRENT_LIST_DIR}/googletest/googletest/src/gtest-all.cc"
-  "${CMAKE_CURRENT_LIST_DIR}/googletest/googletest/src/gtest_main.cc")
+  "/usr/src/googletest/src/gtest-all.cc"
+  "/usr/src/googletest/src/gtest_main.cc")
 
 target_include_directories(gtest
   PRIVATE
-    "${CMAKE_CURRENT_LIST_DIR}/googletest/googletest"
+    "/usr/src/googletest"
   PUBLIC
-    "${CMAKE_CURRENT_LIST_DIR}/googletest/googletest/include")
+    "/usr/include/gtest")
 
 target_compile_definitions(gtest
   PUBLIC
