From 1f4b6d5c855b59e849b7228090981c520928c2a4 Mon Sep 17 00:00:00 2001
From: Ovidiu-Florin BOGDAN <ovidiub13@users.noreply.github.com>
Date: Fri, 2 Dec 2016 16:51:07 +0200
Subject: [PATCH] Remove prefix duplication in yaml-cpp.pc.cmake

---
 yaml-cpp.pc.cmake | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/yaml-cpp.pc.cmake b/yaml-cpp.pc.cmake
index 04d343f..044ef78 100644
--- a/yaml-cpp.pc.cmake
+++ b/yaml-cpp.pc.cmake
@@ -1,7 +1,5 @@
-prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=@CMAKE_INSTALL_PREFIX@
-libdir=${prefix}/@LIB_INSTALL_DIR@
-includedir=${prefix}/@INCLUDE_INSTALL_ROOT_DIR@
+libdir=@LIB_INSTALL_DIR@
+includedir=@INCLUDE_INSTALL_ROOT_DIR@
 
 Name: Yaml-cpp
 Description: A YAML parser and emitter for C++
