--- VTK/IO/vtkFFMPEGWriter.cxx.orig	2012-07-08 13:40:55.000000000 +0200
+++ VTK/IO/vtkFFMPEGWriter.cxx	2012-07-08 13:59:37.000000000 +0200
@@ -192,13 +192,6 @@
     c->bit_rate_tolerance = this->Writer->GetBitRateTolerance();
     }
 
-  //apply the chosen parameters
-  if (av_set_parameters(this->avFormatContext, NULL) < 0)
-    {
-    vtkGenericWarningMacro (<< "Invalid output format parameters." );
-    return 0;
-    }
-
   //manufacture a codec with the chosen parameters
   AVCodec *codec = avcodec_find_encoder(c->codec_id);
   if (!codec) 
@@ -259,14 +252,14 @@
 
 
   //Finally, open the file and start it off.
-  if (url_fopen(&this->avFormatContext->pb, this->avFormatContext->filename, URL_WRONLY) < 0) 
+  if (avio_open(&this->avFormatContext->pb, this->avFormatContext->filename, AVIO_FLAG_WRITE) < 0) 
     {
     vtkGenericWarningMacro (<< "Could not open " << this->Writer->GetFileName() << "." );
     return 0;
     }
   this->openedFile = 1;
 
-  av_write_header(this->avFormatContext);
+  avformat_write_header(this->avFormatContext, NULL);
   return 1;
 }
 
@@ -388,11 +381,7 @@
     if (this->openedFile)
       {
       av_write_trailer(this->avFormatContext);
-#ifdef VTK_FFMPEG_OLD_URL_FCLOSE
-      url_fclose(&this->avFormatContext->pb);
-#else
-      url_fclose(this->avFormatContext->pb);
-#endif
+      avio_close(this->avFormatContext->pb);
       this->openedFile = 0;
       }
 
