From bf103c05bea41902cf9186224a6fa9435cb4e849 Mon Sep 17 00:00:00 2001
From: pradeep <pradeep@arrayfire.com>
Date: Mon, 9 Jan 2017 12:58:52 +0530
Subject: [PATCH] Fix for CUDA usage with gcc > 5.x

---
 src/backend/cuda/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/backend/cuda/CMakeLists.txt b/src/backend/cuda/CMakeLists.txt
index 8ae3799..f5ff450 100644
--- a/src/backend/cuda/CMakeLists.txt
+++ b/src/backend/cuda/CMakeLists.txt
@@ -338,6 +338,10 @@ ELSE()
   ENDIF()
 ENDIF()
 
+IF("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_MWAITXINTRIN_H_INCLUDED -D_FORCE_INLINES")
+ENDIF()
+
 ## Copied from FindCUDA.cmake
 ## The target_link_library needs to link with the cuda libraries using
 ## PRIVATE
