diff --git a/gdal/frmts/pdf/pdfdataset.cpp b/gdal/frmts/pdf/pdfdataset.cpp
index 6e8223e470..16651ee2a3 100644
--- a/gdal/frmts/pdf/pdfdataset.cpp
+++ b/gdal/frmts/pdf/pdfdataset.cpp
@@ -3452,7 +3452,7 @@ void PDFDataset::FindLayersPoppler()
         {
             auto ocg = refOCGPair.second.get();
 #else
-        GooList* ocgList = optContentConfig->getOCGs();
+        std::vector<void*> ocgList = optContentConfig->getOCGs();
         for(int i=0;i<ocgList->getLength();i++)
         {
             OptionalContentGroup* ocg = (OptionalContentGroup*) ocgList->get(i);
@@ -3490,7 +3490,7 @@ void PDFDataset::TurnLayersOnOffPoppler()
         {
             auto ocg = refOCGPair.second.get();
 #else
-        GooList* ocgList = optContentConfig->getOCGs();
+        std::vector<void*> ocgList = optContentConfig->getOCGs();
         for(i=0;i<ocgList->getLength();i++)
         {
             OptionalContentGroup* ocg = (OptionalContentGroup*) ocgList->get(i);
diff --git a/gdal/frmts/pdf/pdfsdk_headers.h b/gdal/frmts/pdf/pdfsdk_headers.h
index 9150b0f410..1311636f7b 100644
--- a/gdal/frmts/pdf/pdfsdk_headers.h
+++ b/gdal/frmts/pdf/pdfsdk_headers.h
@@ -51,7 +51,6 @@
 #else
 typedef unsigned char Guchar;
 #endif
-#include <goo/GooList.h>
 
 /* begin of poppler xpdf includes */
 #include <poppler/Object.h>
