diff -upr gdal-3.4.0.orig/frmts/pdf/pdfdataset.cpp gdal-3.4.0/frmts/pdf/pdfdataset.cpp
--- gdal-3.4.0.orig/frmts/pdf/pdfdataset.cpp	2021-11-04 13:57:20.000000000 +0200
+++ gdal-3.4.0/frmts/pdf/pdfdataset.cpp	2022-03-02 13:17:56.964778481 +0200
@@ -4322,7 +4322,11 @@ PDFDataset *PDFDataset::Open( GDALOpenIn
         oObj.getObj()->initNull();
         auto poStream = new VSIPDFFileStream(fp, pszFilename, oObj.getObj());
 #endif
+#if POPPLER_MAJOR_VERSION > 22 || (POPPLER_MAJOR_VERSION == 22 && POPPLER_MINOR_VERSION >= 3)
+        poDocPoppler = new PDFDoc(poStream, std::nullopt, GooString(poUserPwd));
+#else
         poDocPoppler = new PDFDoc(poStream, nullptr, poUserPwd);
+#endif
         if( globalParamsCreatedByGDAL )
             registerErrorCallback();
         delete poUserPwd;
