--- lib/node_modules/connect/lib/middleware/static.js.orig	2013-03-23 13:30:36.479309260 +0100
+++ lib/node_modules/connect/lib/middleware/static.js	2013-03-23 13:30:13.661917828 +0100
@@ -126,7 +126,7 @@
   if (!root && ~path.indexOf('..')) return utils.forbidden(res);

   // join / normalize from optional root dir
-  path = normalize(join(root, path));
+  path = normalize(join(root || '', path));

   // malicious path
   if (root && 0 != path.indexOf(root)) return fn
