diff -ruN pymol.orig/layer0/os_python.h pymol/layer0/os_python.h
--- pymol.orig/layer0/os_python.h	2015-01-29 21:18:47.681749642 +0000
+++ pymol/layer0/os_python.h	2015-01-29 21:21:07.501148656 +0000
@@ -30,6 +30,7 @@
 #include<pythread.h>
 
 #ifdef _PYMOL_NUMPY
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include <numpy/arrayobject.h>
 #endif
 
--- pymol.orig/layer0/Field.cpp	2015-01-29 21:31:36.105141178 +0000
+++ pymol/layer0/Field.cpp	2015-01-29 21:31:52.655071349 +0000
@@ -73,7 +73,7 @@
 
   if(copy) {
     if((result = PyArray_SimpleNew(field->n_dim, dims, typenum)))
-      memcpy(PyArray_DATA(result), field->data, field->size);
+      memcpy(PyArray_DATA((PyArrayObject *)result), field->data, field->size);
   } else {
     result = PyArray_SimpleNewFromData(field->n_dim, dims, typenum, field->data);
   }
--- pymol.orig/layer2/CoordSet.cpp	2015-01-29 21:58:16.588313743 +0000
+++ pymol/layer2/CoordSet.cpp	2015-01-29 21:59:42.051283150 +0000
@@ -198,7 +198,7 @@
 
   if(copy) {
     if((result = PyArray_SimpleNew(2, dims, typenum)))
-      memcpy(PyArray_DATA(result), cs->Coord, cs->NIndex * 3 * base_size);
+      memcpy(PyArray_DATA((PyArrayObject *)result), cs->Coord, cs->NIndex * 3 * base_size);
   } else {
     result = PyArray_SimpleNewFromData(2, dims, typenum, cs->Coord);
   }
--- pymol.orig/layer3/Selector.cpp       2014-12-04 01:50:03.000000000 +0300
+++ pymol/layer3/Selector.cpp     2015-01-30 00:44:31.622293145 +0300
@@ -6608,7 +6608,7 @@
   }

   result = PyArray_SimpleNew(2, dims, typenum);
-  dataptr = (float*) PyArray_DATA(result);
+  dataptr = (float*) PyArray_DATA((PyArrayObject *)result);

   for(i = 0, iter.reset(); iter.next(); i++) {
     v_ptr = iter.getCoord();
@@ -6677,11 +6677,12 @@
   import_array1(false);

   if(PyArray_Check(coords)) {
-    if(PyArray_NDIM(coords) != 2 || PyArray_DIM(coords, 1) != 3) {
+    if(PyArray_NDIM((PyArrayObject *)coords) != 2 ||
+        PyArray_DIM((PyArrayObject *)coords, 1) != 3) {
       ErrMessage(G, "LoadCoords", "numpy array shape mismatch");
       return false;
     }
-    itemsize = PyArray_ITEMSIZE(coords);
+    itemsize = PyArray_ITEMSIZE((PyArrayObject *)coords);
     switch(itemsize) {
       case sizeof(double):
       case sizeof(float):
@@ -6700,7 +6701,7 @@
       // fast implementation for numpy arrays only
 #ifdef _PYMOL_NUMPY
       for(b = 0; b < 3; b++) {
-        ptr = PyArray_GETPTR2(coords, a, b);
+        ptr = PyArray_GETPTR2((PyArrayObject *)coords, a, b);

         switch(itemsize) {
           case sizeof(double):
