Index: /branches/kvirc/4.0/src/modules/mediaplayer/mp_mprisinterface.cpp
===================================================================
--- /branches/kvirc/4.0/src/modules/mediaplayer/mp_mprisinterface.cpp	(revision 4667)
+++ /branches/kvirc/4.0/src/modules/mediaplayer/mp_mprisinterface.cpp	(revision 6024)
@@ -190,6 +190,6 @@
 	QString artist;
 	QString title;
-	foreach (QVariant v, reply.arguments()) {
-		QDBusArgument arg = qvariant_cast<QDBusArgument>(v);
+	foreach (QVariant w, reply.arguments()) {
+		QDBusArgument arg = qvariant_cast<QDBusArgument>(w);
 		QVariant v = qdbus_cast<QVariantMap>(arg);
 		if (v.userType() == QVariant::Map) {
Index: /branches/kvirc/4.0/src/modules/mediaplayer/mp_mprisinterface.cpp
===================================================================
--- /branches/kvirc/4.0/src/modules/mediaplayer/mp_mprisinterface.cpp	(revision 6024)
+++ /branches/kvirc/4.0/src/modules/mediaplayer/mp_mprisinterface.cpp	(revision 6025)
@@ -139,6 +139,6 @@
 		return __return_if_fail; \
 	MPRIS_CALL_METHOD("GetMetadata", __return_if_fail) \
-	foreach (QVariant v, reply.arguments()) { \
-		QDBusArgument arg = qvariant_cast<QDBusArgument>(v); \
+	foreach (QVariant w, reply.arguments()) { \
+		QDBusArgument arg = qvariant_cast<QDBusArgument>(w); \
 		QVariant v = qdbus_cast<QVariantMap>(arg); \
 		if (v.userType() == QVariant::Map) { \
@@ -214,6 +214,6 @@
 	MPRIS_CALL_METHOD("GetMetadata", "")
 
-	foreach (QVariant v, reply.arguments()) {
-		QDBusArgument arg = qvariant_cast<QDBusArgument>(v);
+	foreach (QVariant w, reply.arguments()) {
+		QDBusArgument arg = qvariant_cast<QDBusArgument>(w);
 		QVariant v = qdbus_cast<QVariantMap>(arg);
 		if (v.userType() == QVariant::Map) {
@@ -292,6 +292,6 @@
 	MPRIS_CALL_METHOD("GetMetadata", -1)
 
-	foreach (QVariant v, reply.arguments()) {
-		QDBusArgument arg = qvariant_cast<QDBusArgument>(v);
+	foreach (QVariant w, reply.arguments()) {
+		QDBusArgument arg = qvariant_cast<QDBusArgument>(w);
 		QVariant v = qdbus_cast<QVariantMap>(arg);
 		if (v.userType() == QVariant::Map) {
@@ -353,6 +353,6 @@
 	MPRIS_CALL_METHOD("GetMetadata", "")
 
-	foreach (QVariant v, reply.arguments()) {
-		QDBusArgument arg = qvariant_cast<QDBusArgument>(v);
+	foreach (QVariant w, reply.arguments()) {
+		QDBusArgument arg = qvariant_cast<QDBusArgument>(w);
 		QVariant v = qdbus_cast<QVariantMap>(arg);
 		if (v.userType() == QVariant::Map) {
@@ -404,6 +404,6 @@
 	MPRIS_CALL_METHOD("GetMetadata", -1)
 
-	foreach (QVariant v, reply.arguments()) {
-		QDBusArgument arg = qvariant_cast<QDBusArgument>(v);
+	foreach (QVariant w, reply.arguments()) {
+		QDBusArgument arg = qvariant_cast<QDBusArgument>(w);
 		QVariant v = qdbus_cast<QVariantMap>(arg);
 		if (v.userType() == QVariant::Map) {
