diff --git a/source/libnormaliz/cone.cpp b/source/libnormaliz/cone.cpp
index 09e73631..706c3b46 100644
--- a/source/libnormaliz/cone.cpp
+++ b/source/libnormaliz/cone.cpp
@@ -474,7 +474,7 @@ template<typename Integer>
 Cone<Integer>::~Cone() {
     if(IntHullCone!=NULL)
         delete IntHullCone;
-    if(IntHullCone!=NULL)
+    if(SymmCone!=NULL)
         delete SymmCone;
     if(ProjCone!=NULL)
         delete ProjCone;
