--- oblogout/__init__.py	2009-04-13 16:48:11.000000000 -0400
+++ da/oblogout/__init__.py	2010-10-19 14:38:29.000000000 -0300
@@ -222,8 +222,8 @@ class OpenboxLogout():
         if self.parser.has_section("commands"):
             for key in self.parser.items("commands"):
                 self.logger.debug("Setting cmd_%s as %s" % (key[0], key[1]))
-                if key[1] in ['logout', 'restart', 'shutdown', 'suspend', 'hibernate', 'safesuspend', 'lock', 'switch']:
-                    if key[1]: setattr(self, "cmd_" + key[0], key[1])
+                if key[0] in ['logout', 'restart', 'shutdown', 'suspend', 'hibernate', 'safesuspend', 'lock', 'switch']:
+                    if key[0]: setattr(self, "cmd_" + key[0], key[1])
 
         # Load theme information from local directory if local mode is set
         if self.local_mode:
