diff -Nur cinelerra.orig/cinelerra/filepng.C cinelerra/cinelerra/filepng.C
--- cinelerra.orig/cinelerra/filepng.C	2010-01-21 01:13:00.000000000 +0200
+++ cinelerra/cinelerra/filepng.C	2010-01-21 01:13:35.000000000 +0200
@@ -56,7 +56,7 @@
 		fread(test, 16, 1, stream);
 		fclose(stream);
 
-		if(png_check_sig((unsigned char*)test, 8))
+		if(png_sig_cmp((unsigned char*)test, 0, 8) == 0)
 		{
 //printf("FilePNG::check_sig 1\n");
 			return 1;
diff -Nur cinelerra.orig/plugins/shapewipe/shapewipe.C cinelerra/plugins/shapewipe/shapewipe.C
--- cinelerra.orig/plugins/shapewipe/shapewipe.C	2010-01-21 01:13:00.000000000 +0200
+++ cinelerra/plugins/shapewipe/shapewipe.C	2010-01-21 01:46:46.000000000 +0200
@@ -386,7 +386,7 @@
 	}
 
 	png_ptr = png_create_read_struct(PNG_LIBPNG_VER_STRING,
-		png_voidp_NULL, png_error_ptr_NULL, png_error_ptr_NULL);
+		NULL, NULL, NULL);
 
 	if (!png_ptr)
 	{
@@ -399,14 +399,14 @@
 	info_ptr = png_create_info_struct(png_ptr);
 	if (!info_ptr)
 	{
-		png_destroy_read_struct(&png_ptr, png_infopp_NULL, png_infopp_NULL);
+		png_destroy_read_struct(&png_ptr, NULL, NULL);
 		return 1;
 	}
 
 	end_info = png_create_info_struct(png_ptr);
 	if (!end_info)
 	{
-		png_destroy_read_struct(&png_ptr, &info_ptr, png_infopp_NULL);
+		png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
 		return 1;
 	}
 
