From bc98e2896682ac831cd27142c7e76fe0f8280bee Mon Sep 17 00:00:00 2001
From: Joshua Lock <josh@linux.intel.com>
Date: Wed, 15 Dec 2010 14:07:50 +0000
Subject: Fix Makefile whitespace for the new improved strict make 3.82

Signed-off-by: Joshua Lock <josh@linux.intel.com>
---
diff --git a/Makefile.am b/Makefile.am
index e8a3c7f..51b4f1d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -7,5 +7,5 @@ DISTCLEANFILES = intltool-extract intltool-merge intltool-update
 MAINTAINERCLEANFILES = $(DISTCLEANFILES) aclocal.m4 compile config.guess config.sub configure depcomp install-sh ltmain.sh Makefile.in missing
 
 snapshot:
-        $(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
+	$(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
 
--
cgit v0.9.0.2
