From f3e8719489fc5db5ae87dd03d39aad176d65762f Mon Sep 17 00:00:00 2001
From: Paul Wise <pabs3@bonedaddy.net>
Date: Fri, 29 Apr 2011 13:10:15 +0800
Subject: [PATCH] Fix FTBFS with newer GCC

---
 configure.ac    |    2 +-
 src/Makefile.am |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7f009d9..0181292 100644
--- a/configure.ac
+++ b/configure.ac
@@ -17,7 +17,7 @@ AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Gettext package])
 AC_SUBST(GETTEXT_PACKAGE)
 AM_GLIB_GNU_GETTEXT
 
-PKG_CHECK_MODULES(DATES, glib-2.0 gtk+-2.0 libecal-1.2 gconf-2.0)
+PKG_CHECK_MODULES(DATES, glib-2.0 gtk+-2.0 libecal-1.2 gconf-2.0 pango)
 
 PKG_CHECK_MODULES(GTK, gtk+-2.0)
 AC_SUBST(GTK_CFLAGS)
diff --git a/src/Makefile.am b/src/Makefile.am
index 90c7049..3bb4f32 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -35,7 +35,7 @@ dates_SOURCES = $(platform) 	\
 	gconf-bridge.h		\
 	gconf-bridge.c
 
-dates_LDADD = $(DATES_LIBS) libgtkdatesview.la
+dates_LDADD =  libgtkdatesview.la -lm $(DATES_LIBS)
 
 if USE_OWL
 dates_LDADD += $(top_builddir)/libowl/libowl.la
-- 
1.7.4.4

