--- nauty26r12/makefile.basic.orig	2019-10-15 06:08:44.730370702 +0000
+++ nauty26r12/makefile.basic	2019-10-15 06:09:06.797358787 +0000
@@ -337,7 +337,7 @@
 		nautilL1.o naugraphL1.o schreierL.o naurng.o ${LDFLAGS}
 
 gentreeg : gentreeg.c gtools.o nauty.h gtools.h
-	${CC} -o gentreeg ${CFLAGS} gentreeg.c gtools.o
+	${CC} -o gentreeg ${CFLAGS} gentreeg.c gtools.o ${LDFLAGS}
 
 genrang : ${GTOOLSH} genrang.c gtools.o naututil.o ${NAUTYO}
 	${CC} -o genrang ${CFLAGS} genrang.c \
--- nauty26r12/makefile.in.orig	2019-10-15 06:08:37.656934310 +0000
+++ nauty26r12/makefile.in	2019-10-15 06:09:15.337483153 +0000
@@ -337,7 +337,7 @@
 		nautilL1.o naugraphL1.o schreierL.o naurng.o ${LDFLAGS}
 
 gentreeg : gentreeg.c gtools.o nauty.h gtools.h
-	${CC} -o gentreeg ${CFLAGS} gentreeg.c gtools.o
+	${CC} -o gentreeg ${CFLAGS} gentreeg.c gtools.o ${LDFLAGS}
 
 genrang : ${GTOOLSH} genrang.c gtools.o naututil.o ${NAUTYO}
 	${CC} -o genrang ${CFLAGS} genrang.c \
