diff -rupN noise-0.3.1.orig/CMakeLists.txt noise-0.3.1/CMakeLists.txt
--- noise-0.3.1.orig/CMakeLists.txt	2015-08-30 14:03:02.563699740 +0200
+++ noise-0.3.1/CMakeLists.txt	2015-08-30 14:05:57.528197582 +0200
@@ -13,11 +13,10 @@ set (VERSION "0.3.1")
 set (VERSION_INFO "Stable Release")
 
 include (GNUInstallDirs)
-set (DATADIR ${CMAKE_INSTALL_PREFIX}/share)
+set (DATADIR ${CMAKE_INSTALL_FULL_DATADIR})
 set (PKG_DATADIR ${DATADIR}/${CMAKE_PROJECT_NAME})
 set (ICON_DIR ${DATADIR}/${CMAKE_PROJECT_NAME}/icons)
-set (PLUGIN_DIR_UNPREFIXED ${CMAKE_INSTALL_LIBDIR}/${CMAKE_PROJECT_NAME}/plugins)
-set (PLUGIN_DIR ${CMAKE_INSTALL_PREFIX}/${PLUGIN_DIR_UNPREFIXED})
+set (PLUGIN_DIR ${CMAKE_INSTALL_FULL_LIBDIR}/${CMAKE_PROJECT_NAME}/plugins)
 
 add_definitions ("-DGETTEXT_PACKAGE=\"${GETTEXT_PACKAGE}\"")
 
diff -rupN noise-0.3.1.orig/core/noise-core.pc.cmake noise-0.3.1/core/noise-core.pc.cmake
--- noise-0.3.1.orig/core/noise-core.pc.cmake	2015-08-30 14:03:02.563699740 +0200
+++ noise-0.3.1/core/noise-core.pc.cmake	2015-08-30 14:03:14.083336997 +0200
@@ -1,12 +1,12 @@
 prefix=@PREFIX@
-exec_prefix=@DOLLAR@{prefix}
-libdir=@DOLLAR@{prefix}/@CMAKE_INSTALL_LIBDIR@
-includedir=@DOLLAR@{prefix}/@CMAKE_INSTALL_INCLUDEDIR@/
-plugindir=@DOLLAR@{prefix}/@PLUGIN_DIR_UNPREFIXED@
- 
+exec_prefix=@PREFIX@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+plugindir=@PLUGIN_DIR@
+
 Name: Noise
-Description: Noise headers  
-Version: @VERSION@  
+Description: Noise headers
+Version: @VERSION@
 Libs: -lnoise-core
 Cflags: -I@DOLLAR@{includedir}/noise-core
-Requires: glib-2.0 gio-2.0 gee-0.8 libpeas-1.0 libpeas-gtk-1.0 gtk+-3.0 granite gstreamer-1.0 gstreamer-pbutils-1.0 gstreamer-tag-1.0
\ No newline at end of file
+Requires: glib-2.0 gio-2.0 gee-0.8 libpeas-1.0 libpeas-gtk-1.0 gtk+-3.0 granite gstreamer-1.0 gstreamer-pbutils-1.0 gstreamer-tag-1.0
