diff --git a/hphp/runtime/vm/jit/vasm-xls.cpp b/hphp/runtime/vm/jit/vasm-xls.cpp
index b07f3f3..48bfd5b 100644
--- a/hphp/runtime/vm/jit/vasm-xls.cpp
+++ b/hphp/runtime/vm/jit/vasm-xls.cpp
@@ -2047,6 +2047,9 @@ void Vxls::insertCopiesAt(jit::vector<Vinstr>& code, unsigned& j,
     return sf_ivl && !sf_ivl->ranges.empty() && sf_ivl->covers(pos);
   };
   MovePlan moves;
+  for (auto reg : moves) {
+    moves[reg] = InvalidReg;
+  }
   jit::vector<Vinstr> loads;
   for (auto dst : copies) {
     auto ivl = copies[dst];
