--- mcomix/archive/rarfile.py	2012-02-13 15:35:10.000000000 +0100
+++ mcomix/archive/rarfile.py	2012-02-17 23:48:00.000000000 +0100
@@ -209,7 +209,6 @@
 
         handle = self._unrar.RAROpenArchiveEx(ctypes.byref(archivedata))
         if handle:
-            self._unrar.RARSetCallback(handle, self._callback_function, 0)
             return handle
         else:
             errormessage = UnrarException.get_error_message(archivedata.OpenResult)
