--- src/image-editor-1.0.11/libimagevisualresult/CMakeLists.txt.orig	2022-04-04 03:51:14.592674102 +0300
+++ src/image-editor-1.0.11/libimagevisualresult/CMakeLists.txt	2022-04-04 03:51:22.224740532 +0300
@@ -75,14 +75,3 @@
 install(DIRECTORY ${PROJECT_SOURCE_DIR}/filter_cube
     DESTINATION ${PREFIX}/share/libimagevisualresult
     FILES_MATCHING PATTERN "*.dat")
-
-# 加速编译优化参数
-if (CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "mips64")
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -Wl,-O1 -Wl,--as-needed -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections -ftree-vectorize -march=loongson3a -mhard-float -mno-micromips -mno-mips16 -flax-vector-conversions -mloongson-ext2 -mloongson-mmi -z noexecstack")
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -Wl,-O1 -Wl,--as-needed -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections -ftree-vectorize -march=loongson3a -mhard-float -mno-micromips -mno-mips16 -flax-vector-conversions -mloongson-ext2 -mloongson-mmi -z noexecstack")
-    SET(CMAKE_EXE_LINKER_FLAGS  "-pie")
-else()
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -Wl,-O1 -Wl,--as-needed -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections")
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3  -Wl,-O1 -Wl,--as-needed -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections")
-    SET(CMAKE_EXE_LINKER_FLAGS  "-pie")
-endif()
--- src/image-editor-1.0.11/libimageviewer/CMakeLists.txt.orig	2022-04-04 03:53:23.353737464 +0300
+++ src/image-editor-1.0.11/libimageviewer/CMakeLists.txt	2022-04-04 03:53:36.255837853 +0300
@@ -109,19 +109,6 @@
     DESTINATION ${PREFIX}/share/libimageviewer
     FILES_MATCHING PATTERN "*.qm")
 
-# 加速编译优化参数
-if (NOT(CMAKE_BUILD_TYPE STREQUAL "Debug"))
-    if (CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "mips64")
-        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -Wl,-O1 -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections -ftree-vectorize -march=loongson3a -mhard-float -mno-micromips -mno-mips16 -flax-vector-conversions -mloongson-ext2 -mloongson-mmi -z noexecstack")
-        SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -Wl,-O1 -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections -ftree-vectorize -march=loongson3a -mhard-float -mno-micromips -mno-mips16 -flax-vector-conversions -mloongson-ext2 -mloongson-mmi -z noexecstack")
-        SET(CMAKE_EXE_LINKER_FLAGS  "-pie")
-    else()
-        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -Wl,-O1 -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections")
-        SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3  -Wl,-O1 -Wl,-E -fPIE -ffunction-sections -fdata-sections -Wl,--gc-sections")
-        SET(CMAKE_EXE_LINKER_FLAGS  "-pie")
-    endif()
-endif()
-
 if (CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "sw_64")
     SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mieee")
     SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mieee")
