From 236c342dc8594c8d73849c670cd57946f2549451 Mon Sep 17 00:00:00 2001
From: condorcet <novikov.vz@gmail.com>
Date: Tue, 3 Sep 2019 20:09:04 +0300
Subject: [PATCH 1/2] Support asyncio in _start_io_loop method

---
 threadloop/threadloop.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/threadloop/threadloop.py b/threadloop/threadloop.py
index 8b584b6..1916baa 100644
--- a/threadloop/threadloop.py
+++ b/threadloop/threadloop.py
@@ -72,6 +72,12 @@ def mark_as_ready():
             self._ready.set()
 
         if not self._io_loop:
+            _io_loop = ioloop.IOLoop.current(instance=False)
+            if _io_loop:
+                self._io_loop = _io_loop
+                if self._io_loop.is_running():
+                    mark_as_ready()
+                    return
             self._io_loop = ioloop.IOLoop()
 
         self._io_loop.add_callback(mark_as_ready)
@@ -92,7 +98,7 @@ def is_ready(self):
 
     def stop(self):
         """Stop IOLoop & close daemonized thread."""
-        self._io_loop.stop()
+        self._io_loop.add_callback(self._io_loop.stop)
         self._thread.join()
 
     def submit(self, fn, *args, **kwargs):

From c15556dd79cdf5a5f7c8dc04015fa4a0ef3f2c56 Mon Sep 17 00:00:00 2001
From: condorcet <novikov.vz@gmail.com>
Date: Wed, 4 Sep 2019 11:03:26 +0300
Subject: [PATCH 2/2] Fix mistake

---
 threadloop/threadloop.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/threadloop/threadloop.py b/threadloop/threadloop.py
index 1916baa..d6847fb 100644
--- a/threadloop/threadloop.py
+++ b/threadloop/threadloop.py
@@ -75,9 +75,8 @@ def mark_as_ready():
             _io_loop = ioloop.IOLoop.current(instance=False)
             if _io_loop:
                 self._io_loop = _io_loop
-                if self._io_loop.is_running():
-                    mark_as_ready()
-                    return
+                mark_as_ready()
+                return
             self._io_loop = ioloop.IOLoop()
 
         self._io_loop.add_callback(mark_as_ready)
