diff -r 79b9af284286 cx_Freeze/finder.py
--- a/cx_Freeze/finder.py	Fri Dec 26 15:56:54 2014 -0700
+++ b/cx_Freeze/finder.py	Wed Jun 29 19:13:55 2016 +0200
@@ -189,12 +189,21 @@
         self.IncludePackage("encodings")
         if sys.version_info[0] >= 3:
             self.IncludeModule("io")
-        if sys.version_info[:2] >= (3, 3):
+        if sys.version_info[:2] >= (3, 3) and sys.version_info < (3, 5):
             self.AddAlias("_frozen_importlib", "importlib._bootstrap")
             self.IncludeModule("_frozen_importlib")
             # importlib itself must not be frozen
             del self._modules["importlib"]
             del self._modules["importlib._bootstrap"]
+        if sys.version_info[:2] >= (3, 5):
+            self.AddAlias("_frozen_importlib", "importlib._bootstrap")
+            self.AddAlias("_frozen_importlib_external", "importlib._bootstrap_external")
+            self.IncludeModule("_frozen_importlib_external")
+            self.IncludeModule("_frozen_importlib")
+            # importlib itself must not be frozen
+            del self._modules["importlib"]
+            del self._modules["importlib._bootstrap"]
+            del self._modules["importlib._bootstrap_external"]
         if self.copyDependentFiles:
             self.IncludeModule("os")
             self.IncludeModule("sys")
