diff -rupN mono-msbuild.orig/mono/build/sdks_and_nugets/update_sdks_and_nugets.proj mono-msbuild/mono/build/sdks_and_nugets/update_sdks_and_nugets.proj
--- mono-msbuild.orig/mono/build/sdks_and_nugets/update_sdks_and_nugets.proj	2020-07-03 19:37:07.112979921 +0200
+++ mono-msbuild/mono/build/sdks_and_nugets/update_sdks_and_nugets.proj	2020-07-03 19:37:59.966877151 +0200
@@ -61,7 +61,6 @@
             Targets="Restore;Build"
             Properties="OutputDirectory=$(DotNetOverlayDirectory)\nuget-support\msbuild-bin"/>
         
-        <Exec Command="$(MSBuildThisFileDirectory)/../extract_and_copy_hostfxr.sh $(DotNetSdkVersionForLibHostFxr) $(MSBuildSdkResolverOutDir)" />
         <Exec Command="$(MSBuildThisFileDirectory)/../get_sdk_files.sh $(DotNetOverlayDirectory)\msbuild-bin" />
     </Target>
 </Project>
