diff -Naur Gnome-Pie-0.5.5.orig/src/actionGroups/sessionGroup.vala Gnome-Pie-0.5.5/src/actionGroups/sessionGroup.vala
--- Gnome-Pie-0.5.5.orig/src/actionGroups/sessionGroup.vala	2013-03-17 12:15:02.000000000 +0100
+++ Gnome-Pie-0.5.5/src/actionGroups/sessionGroup.vala	2014-08-11 13:20:44.830325275 +0200
@@ -58,13 +58,13 @@
 //    	iface = GLib.Bus.get_proxy_sync(GLib.BusType.SESSION, "org.freedesktop.ConsoleKit", "/org/freedesktop/ConsoleKit/Manager");
     
         this.add_action(new AppAction(_("Shutdown"), "gnome-shutdown", 
-            "dbus-send --print-reply --dest=org.gnome.SessionManager /org/gnome/SessionManager org.gnome.SessionManager.RequestShutdown"));
+            "dbus-send --print-reply --dest=org.gnome.SessionManager /org/gnome/SessionManager org.gnome.SessionManager.Shutdown"));
             
-        this.add_action(new AppAction(_("Logout"), "gnome-session-logout", 
+        this.add_action(new AppAction(_("Logout"), "gnome-logout", 
             "dbus-send --print-reply --dest=org.gnome.SessionManager /org/gnome/SessionManager org.gnome.SessionManager.Logout uint32:1"));
             
-        this.add_action(new AppAction(_("Reboot"), "gnome-session-reboot", 
-            "dbus-send --print-reply --dest=org.gnome.SessionManager /org/gnome/SessionManager org.gnome.SessionManager.RequestReboot"));
+        this.add_action(new AppAction(_("Reboot"), "view-refresh", 
+            "dbus-send --print-reply --dest=org.gnome.SessionManager /org/gnome/SessionManager org.gnome.SessionManager.Reboot"));
     }
     
     // TODO: check for available interfaces --- these may work too:
diff -Naur Gnome-Pie-0.5.5.orig/src/pies/defaultConfig.vala Gnome-Pie-0.5.5/src/pies/defaultConfig.vala
--- Gnome-Pie-0.5.5.orig/src/pies/defaultConfig.vala	2013-03-17 12:15:02.000000000 +0100
+++ Gnome-Pie-0.5.5/src/pies/defaultConfig.vala	2014-08-11 13:22:50.953499473 +0200
@@ -47,11 +47,11 @@
             bookmarks.add_group(new DevicesGroup(bookmarks.id));
         
         // add a pie with session controls
-        var session = PieManager.create_persistent_pie(_("Session"), "gnome-session-halt", new Trigger.from_string("<Control><Alt>q"));
+        var session = PieManager.create_persistent_pie(_("Session"), "gnome-logout", new Trigger.from_string("<Control><Alt>q"));
             session.add_group(new SessionGroup(session.id));
         
         // add a pie with a main menu
-        var menu = PieManager.create_persistent_pie(_("Main Menu"), "alacarte", new Trigger.from_string("<Control><Alt>space"));
+        var menu = PieManager.create_persistent_pie(_("Main Menu"), "gnome-main-menu", new Trigger.from_string("<Control><Alt>space"));
             menu.add_group(new MenuGroup(menu.id));
         
         // add a pie with window controls
diff -Naur Gnome-Pie-0.5.5.orig/src/utilities/config.vala Gnome-Pie-0.5.5/src/utilities/config.vala
--- Gnome-Pie-0.5.5.orig/src/utilities/config.vala	2014-08-11 14:22:10.914071339 +0200
+++ Gnome-Pie-0.5.5/src/utilities/config.vala	2014-08-11 14:22:30.664573331 +0200
@@ -180,7 +180,7 @@
         
         if (themes.size > 0) {
             if (current == "") {
-                current = "Unity";
+                current = "Adwaita";
                 warning("No theme specified! Using default...");
             }
             foreach (var t in themes) {
