--- diff-so-fancy.old	2017-06-10 22:14:03.858070687 +0800
+++ diff-so-fancy	2017-06-10 22:15:00.762691077 +0800
@@ -572,29 +572,5 @@
 
 # Find the location of diff-highlight
 sub find_diff_highlight {
-    my $dh         = "diff-highlight";
-    my $dh_in_path = trim(`which $dh 2>/dev/null`);
-    my $ret;
-
-    # We check for diff-highlight in three places:
-    # 1) System path
-    # 2) Same dir as d-s-f
-    # 3) third_party/diff-highlight/
-
-    # This is #1
-    if ($dh_in_path) {
-        $ret = $dh_in_path;
-    # This is #2 and #3
-    } else {
-        $ret = dirname(Cwd::realpath($0)) . "/diff-highlight";
-        if (!-X $ret) {
-            $ret = dirname(Cwd::realpath($0)) . "/third_party/$dh/diff-highlight";
-        }
-    }
-
-    if (!-X $ret) {
-        die("Error locating diff-highlight\n");
-    }
-
-    return $ret;
+    return "/usr/share/git/diff-highlight/diff-highlight";
 }
