--- configure.ac	2012-05-23 21:21:14.000000000 +0200
+++ configure.ac	2012-06-02 12:29:24.836921683 +0200
@@ -378,6 +378,10 @@
 	AC_CHECK_HEADERS(sys/videodev2.h,FOUND_OS=solaris,FOUND_OS=linux)
 fi
 
+if test "$FOUND_OS" = "linux"; then
+    AC_CHECK_HEADERS(linux/videodev2.h)
+fi
+
 #---------------------------------------------------------------------------------------------
 
 dnl ---------------------------------------------------------------------
--- utils/linux/capture/capture.h	2011-04-12 02:20:59.000000000 +0200
+++ utils/linux/capture/capture.h	2012-06-02 12:29:24.836921683 +0200
@@ -34,7 +34,11 @@
 #ifdef HAVE_SYS_VIDEODEV2_H
 #   include <sys/videodev2.h>
 #else
-#   include <linux/videodev.h>
+	#ifdef HAVE_LINUX_VIDEODEV2_H
+		#include <linux/videodev2.h>
+	#else
+		#include <linux/videodev.h>
+	#endif
 #endif
 */
 
--- utils/linux/capture/libng/plugins/Rules.mk	2012-05-18 17:31:32.000000000 +0200
+++ utils/linux/capture/libng/plugins/Rules.mk	2012-06-02 12:51:59.523065502 +0200
@@ -2,6 +2,13 @@
 TARGETS-plugins := $(capture_dir)/libng/plugins/conv-mjpeg.so
 TARGETS-plugins += $(patsubst %,$(capture_dir)/libng/plugins/%.so,${LIBNG_PLUGINS})
 
+plugin_link_so         = $(CC) $(LDFLAGS) $^ $(LDLIBS) $(capture_dir)/capture.so $(SHARED) -o $@
+ifeq ($(verbose),no)
+  echo_plugin_link_so          = echo "  LDP    " $@
+else
+  echo_plugin_link_so          = echo $(plugin_link_so)
+endif
+  
 # global targets
 all:: $(TARGETS-plugins)
 