diff -wbBur google-gadgets-for-linux-0.11.2/extensions/linux_system_framework/network.cc google-gadgets-for-linux-0.11.2.my/extensions/linux_system_framework/network.cc
--- google-gadgets-for-linux-0.11.2/extensions/linux_system_framework/network.cc	2009-03-15 04:12:42.000000000 +0000
+++ google-gadgets-for-linux-0.11.2.my/extensions/linux_system_framework/network.cc	2011-04-08 20:57:16.000000000 +0000
@@ -23,17 +23,9 @@
 namespace framework {
 namespace linux_system {
 
-#ifdef NM_DEVICE_TYPE_WIFI
 static const int kDeviceTypeWifi = NM_DEVICE_TYPE_WIFI;
-#else
-static const int kDeviceTypeWifi = DEVICE_TYPE_802_11_WIRELESS;
-#endif
 
-#ifdef NM_DEVICE_TYPE_ETHERNET
 static const int kDeviceTypeEthernet = NM_DEVICE_TYPE_ETHERNET;
-#else
-static const int kDeviceTypeEthernet = DEVICE_TYPE_802_3_ETHERNET;
-#endif
 
 static const int kDeviceTypeUnknown = 0;
 
diff -wbBur google-gadgets-for-linux-0.11.2/extensions/linux_system_framework/wireless.cc google-gadgets-for-linux-0.11.2.my/extensions/linux_system_framework/wireless.cc
--- google-gadgets-for-linux-0.11.2/extensions/linux_system_framework/wireless.cc	2009-12-28 10:39:53.000000000 +0000
+++ google-gadgets-for-linux-0.11.2.my/extensions/linux_system_framework/wireless.cc	2011-04-08 20:57:39.000000000 +0000
@@ -77,11 +77,7 @@
 
 using namespace ggadget::dbus;
 
-#ifdef NM_DEVICE_TYPE_WIFI
 static const int kDeviceTypeWifi = NM_DEVICE_TYPE_WIFI;
-#else
-static const int kDeviceTypeWifi = DEVICE_TYPE_802_11_WIRELESS;
-#endif
 
 // Defined in nm 0.7
 static const int kDeviceStateActivated = 8;
