diff -wbBur simh-3.11-1/PDP11/pdp11_defs.h simh-3.11-1.my/PDP11/pdp11_defs.h
--- simh-3.11-1/PDP11/pdp11_defs.h	2020-03-14 17:43:09.000000000 +0300
+++ simh-3.11-1.my/PDP11/pdp11_defs.h	2020-11-12 23:22:59.363475165 +0300
@@ -900,7 +900,7 @@
 #define WrMemW(pa,d)    uc15_WrMemW (pa, d)
 #define WrMemB(pa, d)   uc15_WrMemB (pa, d)
 
-uint32 uc15_memsize;
+extern uint32 uc15_memsize;
 int32 uc15_RdMemW (int32 pa);
 int32 uc15_RdMemB (int32 pa);
 void uc15_WrMemW (int32 pa, int32 d);
