From a67fd739c266f1e61f43c3d7f18a0f3c64d636b6 Mon Sep 17 00:00:00 2001
From: Antonio Rojas <arojas@archlinux.org>
Date: Tue, 26 May 2020 19:38:37 +0200
Subject: [PATCH] Add missing endif

---
 src/calibre/headless/fontconfig_database.cpp | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/src/calibre/headless/fontconfig_database.cpp b/src/calibre/headless/fontconfig_database.cpp
index d867f02948..96a6365ba2 100644
--- a/src/calibre/headless/fontconfig_database.cpp
+++ b/src/calibre/headless/fontconfig_database.cpp
@@ -295,7 +295,26 @@ static const char specialLanguages[][6] = {
     "", // MasaramGondi
     "", // Nushu
     "", // Soyombo
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+    "", // ZanabazarSquare
+    "", // Dogra
+    "", // GunjalaGondi
+    "", // HanifiRohingya
+    "", // Makasar
+    "", // Medefaidrin
+    "", // OldSogdian
+    "", // Sogdian
+    "", // Elymaic
+    "", // Nandinagari
+    "", // NyiakengPuachueHmong
+    "", // Wancho
+    "", // Chorasmian
+    "", // DivesAkuru
+    "", // KhitanSmallScript
+    "" // Yezidi
+#else
     "" // ZanabazarSquare
+#endif // Qt >= 5.15
 #else
     ""  // SignWriting
 #endif // Qt >= 5.11
