diff -aur calibre-2.77.0/setup/build.py calibre-2.77.0.new/setup/build.py
--- calibre-2.77.0/setup/build.py	2017-01-13 03:38:35.000000000 +0000
+++ calibre-2.77.0.new/setup/build.py	2017-01-25 09:49:28.128533094 +0000
@@ -360,8 +360,9 @@
             TARGET = headless
             PLUGIN_TYPE = platforms
             PLUGIN_CLASS_NAME = HeadlessIntegrationPlugin
-            load(qt_plugin)
-            QT += core-private gui-private platformsupport-private
+            TEMPLATE = lib
+            CONFIG += plugin
+            QT += core-private gui-private fontdatabase_support_private service_support_private eventdispatcher_support_private
             HEADERS = {headers}
             SOURCES = {sources}
             OTHER_FILES = {others}
diff -aur calibre-2.77.0/src/calibre/headless/fontconfig_database.cpp calibre-2.77.0.new/src/calibre/headless/fontconfig_database.cpp
--- calibre-2.77.0/src/calibre/headless/fontconfig_database.cpp	2017-01-13 03:38:35.000000000 +0000
+++ calibre-2.77.0.new/src/calibre/headless/fontconfig_database.cpp	2017-01-24 21:12:25.995347259 +0000
@@ -41,7 +41,7 @@
 #include <qpa/qplatformintegration.h>
 #include <qpa/qplatformservices.h>
 
-#include <QtPlatformSupport/private/qfontenginemultifontconfig_p.h>
+#include <QtFontDatabaseSupport/private/qfontenginemultifontconfig_p.h>
 #include <QtGui/private/qfontengine_ft_p.h>
 #include <QtGui/private/qguiapplication_p.h>
 #if (QT_VERSION >= QT_VERSION_CHECK(5, 6, 0))
diff -aur calibre-2.77.0/src/calibre/headless/fontconfig_database.h calibre-2.77.0.new/src/calibre/headless/fontconfig_database.h
--- calibre-2.77.0/src/calibre/headless/fontconfig_database.h	2017-01-13 03:38:35.000000000 +0000
+++ calibre-2.77.0.new/src/calibre/headless/fontconfig_database.h	2017-01-24 21:12:25.998680585 +0000
@@ -7,7 +7,7 @@
 #pragma once
 
 #include <qpa/qplatformfontdatabase.h>
-#include <QtPlatformSupport/private/qbasicfontdatabase_p.h>
+#include <QtFontDatabaseSupport/private/qbasicfontdatabase_p.h>
 
 QT_BEGIN_NAMESPACE
 
diff -aur calibre-2.77.0/src/calibre/headless/headless_integration.cpp calibre-2.77.0.new/src/calibre/headless/headless_integration.cpp
--- calibre-2.77.0/src/calibre/headless/headless_integration.cpp	2017-01-13 03:38:35.000000000 +0000
+++ calibre-2.77.0.new/src/calibre/headless/headless_integration.cpp	2017-01-24 21:12:25.998680585 +0000
@@ -4,10 +4,10 @@
 #if (QT_VERSION >= QT_VERSION_CHECK(5, 4, 1))
 #include "fontconfig_database.h"
 #else
-#include <QtPlatformSupport/private/qfontconfigdatabase_p.h>
+#include <QtFontDatabaseSupport/private/qfontconfigdatabase_p.h>
 #endif
 #ifndef Q_OS_WIN
-#include <QtPlatformSupport/private/qgenericunixeventdispatcher_p.h>
+#include <QtEventDispatcherSupport/private/qgenericunixeventdispatcher_p.h>
 #else
 #include <QtCore/private/qeventdispatcher_win_p.h>
 #endif
diff -aur calibre-2.77.0/src/calibre/headless/headless_integration.h calibre-2.77.0.new/src/calibre/headless/headless_integration.h
--- calibre-2.77.0/src/calibre/headless/headless_integration.h	2017-01-13 03:38:35.000000000 +0000
+++ calibre-2.77.0.new/src/calibre/headless/headless_integration.h	2017-01-24 21:12:25.998680585 +0000
@@ -3,7 +3,7 @@
 #include <qpa/qplatformintegration.h>
 #include <qpa/qplatformscreen.h>
 #include <qpa/qplatformservices.h>
-#include <QtPlatformSupport/private/qgenericunixservices_p.h>
+#include <QtServiceSupport/private/qgenericunixservices_p.h>
 #include <QScopedPointer>
 
 QT_BEGIN_NAMESPACE
