diff -wbBur gwenhywfar-4.13.1/src/sio/syncio_tls.c gwenhywfar-4.13.1.my/src/sio/syncio_tls.c
--- gwenhywfar-4.13.1/src/sio/syncio_tls.c	2014-10-12 17:12:22.000000000 +0400
+++ gwenhywfar-4.13.1.my/src/sio/syncio_tls.c	2015-04-16 14:54:53.288513328 +0300
@@ -350,7 +350,7 @@
   }
 
   /* set default priority */
-#if 1 /* old code */
+#if 0 /* old code */
   GWEN_Gui_ProgressLog(0, GWEN_LoggerLevel_Error, I18N("Using old SSL preparation code."));
   rv=gnutls_set_default_priority(xio->session);
   if (rv) {
