--- extconf.rb~	2017-10-05 20:21:59.480315863 +0200
+++ extconf.rb	2017-10-05 20:23:55.077931560 +0200
@@ -6,12 +6,7 @@
 
 require 'mkmf'
 require 'rbconfig'
-
-$CFLAGS = case RUBY_VERSION
-          when /^1\.9/; '-DRUBY19'
-          when /^2\./; '-DRUBY19'
-          else; ''
-          end
+$CFLAGS = "#{$CFLAGS} -DRUBY19"
 
 implementation = case CONFIG['host_os']
                  when /linux/i; 'shadow'
