From c6c789f045182a95908d5013af190dfb2111043b Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 15 Sep 2015 23:20:15 +0200
Subject: [PATCH] Use TERM=tmux by default, since we have it

---
 options-table.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/options-table.c b/options-table.c
index c99937d..c25b0fd 100644
--- a/options-table.c
+++ b/options-table.c
@@ -62,7 +62,7 @@ const struct options_table_entry server_options_table[] = {
 
 	{ .name = "default-terminal",
 	  .type = OPTIONS_TABLE_STRING,
-	  .default_str = "screen"
+	  .default_str = "tmux"
 	},
 
 	{ .name = "escape-time",
-- 
2.5.2

