Description: Usage of new dbus-sharp bindings
Author: Rico Tzschichholz <ricotz@ubuntu.com>
Forwarded: no

Index: notify-sharp/Makefile.in
===================================================================
--- notify-sharp.orig/Makefile.in	2011-10-25 09:18:50.340768390 +0800
+++ notify-sharp/Makefile.in	2011-10-25 09:18:52.470780790 +0800
@@ -114,8 +114,8 @@
 MONODOC_LIBS = @MONODOC_LIBS@
 MONO_CFLAGS = @MONO_CFLAGS@
 MONO_LIBS = @MONO_LIBS@
-NDESK_DBUS_CFLAGS = @NDESK_DBUS_CFLAGS@
-NDESK_DBUS_LIBS = @NDESK_DBUS_LIBS@
+DBUS_SHARP_CFLAGS = @DBUS_SHARP_CFLAGS@
+DBUS_SHARP_LIBS = @DBUS_SHARP_LIBS@
 PACKAGE = @PACKAGE@
 PACKAGE_BUGREPORT = @PACKAGE_BUGREPORT@
 PACKAGE_NAME = @PACKAGE_NAME@
Index: notify-sharp/configure
===================================================================
--- notify-sharp.orig/configure	2011-10-25 09:18:50.310768215 +0800
+++ notify-sharp/configure	2011-10-25 09:18:52.474114142 +0800
@@ -601,8 +601,8 @@
 ac_unique_file="src/Global.cs"
 ac_subst_vars='LTLIBOBJS
 LIBOBJS
-NDESK_DBUS_LIBS
-NDESK_DBUS_CFLAGS
+DBUS_SHARP_LIBS
+DBUS_SHARP_CFLAGS
 GTK_SHARP_LIBS
 GTK_SHARP_CFLAGS
 BUILD_DOCS_FALSE
@@ -709,8 +709,8 @@
 MONODOC_LIBS
 GTK_SHARP_CFLAGS
 GTK_SHARP_LIBS
-NDESK_DBUS_CFLAGS
-NDESK_DBUS_LIBS'
+DBUS_SHARP_CFLAGS
+DBUS_SHARP_LIBS'
 
 
 # Initialize some variables set by options.
@@ -1356,10 +1356,10 @@
               C compiler flags for GTK_SHARP, overriding pkg-config
   GTK_SHARP_LIBS
               linker flags for GTK_SHARP, overriding pkg-config
-  NDESK_DBUS_CFLAGS
-              C compiler flags for NDESK_DBUS, overriding pkg-config
-  NDESK_DBUS_LIBS
-              linker flags for NDESK_DBUS, overriding pkg-config
+  DBUS_SHARP_CFLAGS
+              C compiler flags for DBUS_SHARP, overriding pkg-config
+  DBUS_SHARP_LIBS
+              linker flags for DBUS_SHARP, overriding pkg-config
 
 Use these variables to override the choices made by `configure' or to help
 it to find libraries and programs with nonstandard names/locations.
@@ -3237,27 +3237,27 @@
 fi
 
 
-NDESK_DBUS_REQ_VERSION=0.4
-NDESK_DBUS_GLIB_REQ_VERSION=0.3
+DBUS_SHARP_REQ_VERSION=0.7
+DBUS_SHARP_GLIB_REQ_VERSION=0.5
 
 pkg_failed=no
-{ $as_echo "$as_me:$LINENO: checking for NDESK_DBUS" >&5
-$as_echo_n "checking for NDESK_DBUS... " >&6; }
+{ $as_echo "$as_me:$LINENO: checking for DBUS_SHARP" >&5
+$as_echo_n "checking for DBUS_SHARP... " >&6; }
 
 if test -n "$PKG_CONFIG"; then
-    if test -n "$NDESK_DBUS_CFLAGS"; then
-        pkg_cv_NDESK_DBUS_CFLAGS="$NDESK_DBUS_CFLAGS"
+    if test -n "$DBUS_SHARP_CFLAGS"; then
+        pkg_cv_DBUS_SHARP_CFLAGS="$DBUS_SHARP_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"ndesk-dbus-1.0 >= \$NDESK_DBUS_REQ_VERSION \\
-                              ndesk-dbus-glib-1.0 >= \$NDESK_DBUS_GLIB_REQ_VERSION\"") >&5
-  ($PKG_CONFIG --exists --print-errors "ndesk-dbus-1.0 >= $NDESK_DBUS_REQ_VERSION \
-                              ndesk-dbus-glib-1.0 >= $NDESK_DBUS_GLIB_REQ_VERSION") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"dbus-sharp-1.0 >= \$DBUS_SHARP_REQ_VERSION \\
+                              dbus-sharp-glib-1.0 >= \$DBUS_SHARP_GLIB_REQ_VERSION\"") >&5
+  ($PKG_CONFIG --exists --print-errors "dbus-sharp-1.0 >= $DBUS_SHARP_REQ_VERSION \
+                              dbus-sharp-glib-1.0 >= $DBUS_SHARP_GLIB_REQ_VERSION") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_NDESK_DBUS_CFLAGS=`$PKG_CONFIG --cflags "ndesk-dbus-1.0 >= $NDESK_DBUS_REQ_VERSION \
-                              ndesk-dbus-glib-1.0 >= $NDESK_DBUS_GLIB_REQ_VERSION" 2>/dev/null`
+  pkg_cv_DBUS_SHARP_CFLAGS=`$PKG_CONFIG --cflags "dbus-sharp-1.0 >= $DBUS_SHARP_REQ_VERSION \
+                              dbus-sharp-glib-1.0 >= $DBUS_SHARP_GLIB_REQ_VERSION" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -3266,19 +3266,19 @@
 	pkg_failed=untried
 fi
 if test -n "$PKG_CONFIG"; then
-    if test -n "$NDESK_DBUS_LIBS"; then
-        pkg_cv_NDESK_DBUS_LIBS="$NDESK_DBUS_LIBS"
+    if test -n "$DBUS_SHARP_LIBS"; then
+        pkg_cv_DBUS_SHARP_LIBS="$DBUS_SHARP_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"ndesk-dbus-1.0 >= \$NDESK_DBUS_REQ_VERSION \\
-                              ndesk-dbus-glib-1.0 >= \$NDESK_DBUS_GLIB_REQ_VERSION\"") >&5
-  ($PKG_CONFIG --exists --print-errors "ndesk-dbus-1.0 >= $NDESK_DBUS_REQ_VERSION \
-                              ndesk-dbus-glib-1.0 >= $NDESK_DBUS_GLIB_REQ_VERSION") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"dbus-sharp-1.0 >= \$DBUS_SHARP_REQ_VERSION \\
+                              dbus-sharp-glib-1.0 >= \$DBUS_SHARP_GLIB_REQ_VERSION\"") >&5
+  ($PKG_CONFIG --exists --print-errors "dbus-sharp-1.0 >= $DBUS_SHARP_REQ_VERSION \
+                              dbus-sharp-glib-1.0 >= $DBUS_SHARP_GLIB_REQ_VERSION") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_NDESK_DBUS_LIBS=`$PKG_CONFIG --libs "ndesk-dbus-1.0 >= $NDESK_DBUS_REQ_VERSION \
-                              ndesk-dbus-glib-1.0 >= $NDESK_DBUS_GLIB_REQ_VERSION" 2>/dev/null`
+  pkg_cv_DBUS_SHARP_LIBS=`$PKG_CONFIG --libs "dbus-sharp-1.0 >= $DBUS_SHARP_REQ_VERSION \
+                              dbus-sharp-glib-1.0 >= $DBUS_SHARP_GLIB_REQ_VERSION" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -3297,37 +3297,37 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        NDESK_DBUS_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "ndesk-dbus-1.0 >= $NDESK_DBUS_REQ_VERSION \
-                              ndesk-dbus-glib-1.0 >= $NDESK_DBUS_GLIB_REQ_VERSION"`
+	        DBUS_SHARP_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "dbus-sharp-1.0 >= $DBUS_SHARP_REQ_VERSION \
+                              dbus-sharp-glib-1.0 >= $DBUS_SHARP_GLIB_REQ_VERSION"`
         else
-	        NDESK_DBUS_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "ndesk-dbus-1.0 >= $NDESK_DBUS_REQ_VERSION \
-                              ndesk-dbus-glib-1.0 >= $NDESK_DBUS_GLIB_REQ_VERSION"`
+	        DBUS_SHARP_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "dbus-sharp-1.0 >= $DBUS_SHARP_REQ_VERSION \
+                              dbus-sharp-glib-1.0 >= $DBUS_SHARP_GLIB_REQ_VERSION"`
         fi
 	# Put the nasty error message in config.log where it belongs
-	echo "$NDESK_DBUS_PKG_ERRORS" >&5
+	echo "$DBUS_SHARP_PKG_ERRORS" >&5
 
-	{ { $as_echo "$as_me:$LINENO: error: Package requirements (ndesk-dbus-1.0 >= $NDESK_DBUS_REQ_VERSION \
-                              ndesk-dbus-glib-1.0 >= $NDESK_DBUS_GLIB_REQ_VERSION) were not met:
+	{ { $as_echo "$as_me:$LINENO: error: Package requirements (dbus-sharp-1.0 >= $DBUS_SHARP_REQ_VERSION \
+                              dbus-sharp-glib-1.0 >= $DBUS_SHARP_GLIB_REQ_VERSION) were not met:
 
-$NDESK_DBUS_PKG_ERRORS
+$DBUS_SHARP_PKG_ERRORS
 
 Consider adjusting the PKG_CONFIG_PATH environment variable if you
 installed software in a non-standard prefix.
 
-Alternatively, you may set the environment variables NDESK_DBUS_CFLAGS
-and NDESK_DBUS_LIBS to avoid the need to call pkg-config.
+Alternatively, you may set the environment variables DBUS_SHARP_CFLAGS
+and DBUS_SHARP_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details.
 " >&5
-$as_echo "$as_me: error: Package requirements (ndesk-dbus-1.0 >= $NDESK_DBUS_REQ_VERSION \
-                              ndesk-dbus-glib-1.0 >= $NDESK_DBUS_GLIB_REQ_VERSION) were not met:
+$as_echo "$as_me: error: Package requirements (dbus-sharp-1.0 >= $DBUS_SHARP_REQ_VERSION \
+                              dbus-sharp-glib-1.0 >= $DBUS_SHARP_GLIB_REQ_VERSION) were not met:
 
-$NDESK_DBUS_PKG_ERRORS
+$DBUS_SHARP_PKG_ERRORS
 
 Consider adjusting the PKG_CONFIG_PATH environment variable if you
 installed software in a non-standard prefix.
 
-Alternatively, you may set the environment variables NDESK_DBUS_CFLAGS
-and NDESK_DBUS_LIBS to avoid the need to call pkg-config.
+Alternatively, you may set the environment variables DBUS_SHARP_CFLAGS
+and DBUS_SHARP_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details.
 " >&2;}
    { (exit 1); exit 1; }; }
@@ -3338,8 +3338,8 @@
 is in your PATH or set the PKG_CONFIG environment variable to the full
 path to pkg-config.
 
-Alternatively, you may set the environment variables NDESK_DBUS_CFLAGS
-and NDESK_DBUS_LIBS to avoid the need to call pkg-config.
+Alternatively, you may set the environment variables DBUS_SHARP_CFLAGS
+and DBUS_SHARP_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details.
 
 To get pkg-config, see <http://pkg-config.freedesktop.org/>.
@@ -3348,16 +3348,16 @@
 is in your PATH or set the PKG_CONFIG environment variable to the full
 path to pkg-config.
 
-Alternatively, you may set the environment variables NDESK_DBUS_CFLAGS
-and NDESK_DBUS_LIBS to avoid the need to call pkg-config.
+Alternatively, you may set the environment variables DBUS_SHARP_CFLAGS
+and DBUS_SHARP_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details.
 
 To get pkg-config, see <http://pkg-config.freedesktop.org/>.
 See \`config.log' for more details." >&2;}
    { (exit 1); exit 1; }; }; }
 else
-	NDESK_DBUS_CFLAGS=$pkg_cv_NDESK_DBUS_CFLAGS
-	NDESK_DBUS_LIBS=$pkg_cv_NDESK_DBUS_LIBS
+	DBUS_SHARP_CFLAGS=$pkg_cv_DBUS_SHARP_CFLAGS
+	DBUS_SHARP_LIBS=$pkg_cv_DBUS_SHARP_LIBS
         { $as_echo "$as_me:$LINENO: result: yes" >&5
 $as_echo "yes" >&6; }
 	:
Index: notify-sharp/configure.ac
===================================================================
--- notify-sharp.orig/configure.ac	2011-10-25 09:18:50.320768274 +0800
+++ notify-sharp/configure.ac	2011-10-25 09:18:52.810782766 +0800
@@ -66,11 +66,11 @@
 PKG_CHECK_MODULES(GTK_SHARP, gtk-sharp-2.0 >= $GTK_SHARP_REQ_VERSION)
 AC_SUBST(GTK_SHARP_LIBS)
 
-NDESK_DBUS_REQ_VERSION=0.4
-NDESK_DBUS_GLIB_REQ_VERSION=0.3
-PKG_CHECK_MODULES(NDESK_DBUS, ndesk-dbus-1.0 >= $NDESK_DBUS_REQ_VERSION \
-                              ndesk-dbus-glib-1.0 >= $NDESK_DBUS_GLIB_REQ_VERSION)
-AC_SUBST(NDESK_DBUS_LIBS)
+DBUS_SHARP_REQ_VERSION=0.7
+DBUS_SHARP_GLIB_REQ_VERSION=0.5
+PKG_CHECK_MODULES(DBUS_SHARP, dbus-sharp-1.0 >= $DBUS_SHARP_REQ_VERSION \
+                              dbus-sharp-glib-1.0 >= $DBUS_SHARP_GLIB_REQ_VERSION)
+AC_SUBST(DBUS_SHARP_LIBS)
 
 required_assemblies="Mono.Posix"
 
Index: notify-sharp/docs/Makefile.in
===================================================================
--- notify-sharp.orig/docs/Makefile.in	2011-10-25 09:18:50.254101218 +0800
+++ notify-sharp/docs/Makefile.in	2011-10-25 09:18:52.810782766 +0800
@@ -87,8 +87,8 @@
 MONODOC_LIBS = @MONODOC_LIBS@
 MONO_CFLAGS = @MONO_CFLAGS@
 MONO_LIBS = @MONO_LIBS@
-NDESK_DBUS_CFLAGS = @NDESK_DBUS_CFLAGS@
-NDESK_DBUS_LIBS = @NDESK_DBUS_LIBS@
+DBUS_SHARP_CFLAGS = @DBUS_SHARP_CFLAGS@
+DBUS_SHARP_LIBS = @DBUS_SHARP_LIBS@
 PACKAGE = @PACKAGE@
 PACKAGE_BUGREPORT = @PACKAGE_BUGREPORT@
 PACKAGE_NAME = @PACKAGE_NAME@
Index: notify-sharp/src/Global.cs
===================================================================
--- notify-sharp.orig/src/Global.cs	2011-10-25 09:18:50.277434687 +0800
+++ notify-sharp/src/Global.cs	2011-10-25 09:18:52.814116119 +0800
@@ -24,7 +24,7 @@
 using System.Reflection;
 using System.Collections.Generic;
 
-using NDesk.DBus;
+using DBus;
 using org.freedesktop;
 using org.freedesktop.DBus;
 
Index: notify-sharp/src/Makefile.am
===================================================================
--- notify-sharp.orig/src/Makefile.am	2011-10-25 09:18:50.267434629 +0800
+++ notify-sharp/src/Makefile.am	2011-10-25 09:18:52.814116119 +0800
@@ -7,7 +7,7 @@
 LIBFLAGS = \
 	-r:Mono.Posix \
 	$(GTK_SHARP_LIBS) \
-	$(NDESK_DBUS_LIBS)
+	$(DBUS_SHARP_LIBS)
 
 CSFILES = \
 	$(srcdir)/Global.cs \
Index: notify-sharp/src/Makefile.in
===================================================================
--- notify-sharp.orig/src/Makefile.in	2011-10-25 09:18:50.300768156 +0800
+++ notify-sharp/src/Makefile.in	2011-10-25 09:18:52.814116119 +0800
@@ -78,8 +78,8 @@
 MONODOC_LIBS = @MONODOC_LIBS@
 MONO_CFLAGS = @MONO_CFLAGS@
 MONO_LIBS = @MONO_LIBS@
-NDESK_DBUS_CFLAGS = @NDESK_DBUS_CFLAGS@
-NDESK_DBUS_LIBS = @NDESK_DBUS_LIBS@
+DBUS_SHARP_CFLAGS = @DBUS_SHARP_CFLAGS@
+DBUS_SHARP_LIBS = @DBUS_SHARP_LIBS@
 PACKAGE = @PACKAGE@
 PACKAGE_BUGREPORT = @PACKAGE_BUGREPORT@
 PACKAGE_NAME = @PACKAGE_NAME@
@@ -151,7 +151,7 @@
 LIBFLAGS = \
 	-r:Mono.Posix \
 	$(GTK_SHARP_LIBS) \
-	$(NDESK_DBUS_LIBS)
+	$(DBUS_SHARP_LIBS)
 
 CSFILES = \
 	$(srcdir)/Global.cs \
Index: notify-sharp/src/Notification.cs
===================================================================
--- notify-sharp.orig/src/Notification.cs	2011-10-25 09:18:52.457447379 +0800
+++ notify-sharp/src/Notification.cs	2011-10-25 09:18:52.814116119 +0800
@@ -28,7 +28,7 @@
 using Gdk;
 using Gtk;
 
-using NDesk.DBus;
+using DBus;
 using org.freedesktop;
 using org.freedesktop.DBus;
 
