commit 00c4ed6ed6231d30f211cd83062507448b7ffca4
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Mon Jun 21 06:01:10 2021 +0800

    Fix tests in GHC 9.0
    
    Implements #192

diff --git a/mono-traversable/test/Spec.hs b/mono-traversable/test/Main.hs
similarity index 99%
rename from mono-traversable/test/Spec.hs
rename to mono-traversable/test/Main.hs
index e8d2357..befda4c 100644
--- a/mono-traversable/test/Spec.hs
+++ b/mono-traversable/test/Main.hs
@@ -6,7 +6,7 @@
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE PatternSynonyms #-}
 
-module Spec where
+module Main where
 
 import Data.MonoTraversable
 import Data.Containers
diff --git a/mono-traversable/test/main.hs b/mono-traversable/test/main.hs
deleted file mode 100644
index a4754a0..0000000
--- a/mono-traversable/test/main.hs
+++ /dev/null
@@ -1 +0,0 @@
-import Spec (main)
