From d26f44b4aeaec63ae8ce0a4f34dea7439da48012 Mon Sep 17 00:00:00 2001
From: Jamie Hewland <jhewland@gmail.com>
Date: Thu, 13 May 2021 15:11:14 +0100
Subject: [PATCH] Use non-deprecated jinja2.pass_context in Jinja 3.0+ (#1183)

---
 starlette/templating.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/starlette/templating.py b/starlette/templating.py
index 631b6bfee..64fdbd14f 100644
--- a/starlette/templating.py
+++ b/starlette/templating.py
@@ -6,6 +6,12 @@
 
 try:
     import jinja2
+
+    # @contextfunction renamed to @pass_context in Jinja 3.0, to be removed in 3.1
+    if hasattr(jinja2, "pass_context"):
+        pass_context = jinja2.pass_context
+    else:  # pragma: nocover
+        pass_context = jinja2.contextfunction
 except ImportError:  # pragma: nocover
     jinja2 = None  # type: ignore
 
@@ -53,7 +59,7 @@ def __init__(self, directory: str) -> None:
         self.env = self.get_env(directory)
 
     def get_env(self, directory: str) -> "jinja2.Environment":
-        @jinja2.contextfunction
+        @pass_context
         def url_for(context: dict, name: str, **path_params: typing.Any) -> str:
             request = context["request"]
             return request.url_for(name, **path_params)
