diff --git a/3rdparty/deepin-pdfium/src/3rdparty/pdfium/pdfium.pri b/3rdparty/deepin-pdfium/src/3rdparty/pdfium/pdfium.pri
index 82f7354d..9bdd89f3 100755
--- a/3rdparty/deepin-pdfium/src/3rdparty/pdfium/pdfium.pri
+++ b/3rdparty/deepin-pdfium/src/3rdparty/pdfium/pdfium.pri
@@ -22,6 +22,7 @@ DEFINES +=  USE_SYSTEM_LIBJPEG \
 
 INCLUDEPATH += /usr/include/openjpeg-2.3 \
                /usr/include/openjpeg-2.4 \
+               /usr/include/openjpeg-2.5 \
                /usr/include/freetype2 \
                /usr/include/freetype2/freetype \
                /usr/include/freetype2/freetype/config
