diff --git a/cabal-install/Distribution/Client/HashValue.hs b/cabal-install/Distribution/Client/HashValue.hs
index 54b8aee9e..11e647c1c 100644
--- a/cabal-install/Distribution/Client/HashValue.hs
+++ b/cabal-install/Distribution/Client/HashValue.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP          #-}
 {-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE DeriveGeneric      #-}
 module Distribution.Client.HashValue (
@@ -72,10 +73,14 @@ hashFromTUF (Sec.Hash hashstr) =
     --TODO: [code cleanup] either we should get TUF to use raw bytestrings or
     -- perhaps we should also just use a base16 string as the internal rep.
     case Base16.decode (BS.pack hashstr) of
+#if MIN_VERSION_base16_bytestring(1,0,0)
+      Right hash -> HashValue hash
+      Left _ -> error "hashFromTUF: cannot decode base16"
+#else
       (hash, trailing) | not (BS.null hash) && BS.null trailing
         -> HashValue hash
       _ -> error "hashFromTUF: cannot decode base16 hash"
-
+#endif
 
 -- | Truncate a 32 byte SHA256 hash to
 --
diff --git a/cabal-install/src/Distribution/Client/HttpUtils.hs b/cabal-install/src/Distribution/Client/HttpUtils.hs
index 9b8a8ecb844..0c334bec44a 100644
--- a/cabal-install/Distribution/Client/HttpUtils.hs
+++ b/cabal-install/Distribution/Client/HttpUtils.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP          #-}
 -----------------------------------------------------------------------------
 -- | Separate module for HTTP actions, using a proxy server if one exists.
 -----------------------------------------------------------------------------
@@ -182,7 +183,11 @@ downloadURI transport verbosity uri path = do
     fragmentParser = do
         _ <- P.string "#sha256="
         str <- some P.hexDigit
+#if MIN_VERSION_base16_bytestring(1,0,0)
+        return  (Base16.decodeLenient (BS8.pack str))
+#else
         return (fst (Base16.decode (BS8.pack str)))
+#endif
 
 ------------------------------------------------------------------------------
 -- Utilities for repo url management
diff --git a/cabal-install/src/Distribution/Client/HttpUtils.hs b/cabal-install/src/Distribution/Client/HttpUtils.hs
index 0c334bec44a..6de14f3c661 100644
--- a/cabal-install/Distribution/Client/HttpUtils.hs
+++ b/cabal-install/Distribution/Client/HttpUtils.hs
@@ -183,10 +183,11 @@ downloadURI transport verbosity uri path = do
     fragmentParser = do
         _ <- P.string "#sha256="
         str <- some P.hexDigit
+        let bs = Base16.decode (BS8.pack str)
 #if MIN_VERSION_base16_bytestring(1,0,0)
-        return  (Base16.decodeLenient (BS8.pack str))
+        either fail return bs
 #else
-        return (fst (Base16.decode (BS8.pack str)))
+        return (fst bs)
 #endif
 
 ------------------------------------------------------------------------------
diff --git a/cabal-install/cabal-install.cabal b/cabal-install/cabal-install.cabal
index ee49fe245..6d423fdb3 100644
--- a/cabal-install/cabal-install.cabal
+++ b/cabal-install/cabal-install.cabal
@@ -319,7 +319,7 @@ executable cabal
         async      >= 2.0      && < 2.3,
         array      >= 0.4      && < 0.6,
         base       >= 4.8      && < 4.15,
-        base16-bytestring >= 0.1.1 && < 0.2,
+        base16-bytestring >= 0.1.1 && < 1.1.0.0,
         binary     >= 0.7.3    && < 0.9,
         bytestring >= 0.10.6.0 && < 0.11,
         Cabal      == 3.2.*,
