From ec130aeca009ec94cc1f3c119a282bb57f6577f2 Mon Sep 17 00:00:00 2001
From: Vincent Hanquez <vincent@snarc.org>
Date: Wed, 17 Feb 2016 07:05:25 +0000
Subject: [PATCH] x448: set WBITS to 32 bits on 32 bits architectures

---
 cbits/ed448/x448.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/cbits/ed448/x448.c b/cbits/ed448/x448.c
index 5824fe9..c401fe1 100644
--- a/cbits/ed448/x448.c
+++ b/cbits/ed448/x448.c
@@ -11,7 +11,12 @@
 #include <stdint.h>
 #include "x448.h"
 
-#define WBITS 64 /* TODO */
+#ifdef ARCH_X86_64
+#define WBITS 64
+#else
+#define WBITS 32
+#endif
+
 #define LBITS (WBITS * 7 / 8)
 #define X448_LIMBS (448/LBITS)
 
