diff -ruN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2020-06-22 14:51:45.000000000 +0200
+++ b/Makefile.am	2020-06-23 11:45:50.918537263 +0200
@@ -59,21 +59,9 @@
 test: _jack_mixer_c.so
 	@./test.py
 
-schemadir = @GCONF_SCHEMA_FILE_DIR@
-schema_DATA = jack_mixer.schemas
-
 install-exec-hook:
 	ln -vfs jack_mixer.py $(DESTDIR)$(bindir)/jack_mixer
 
-if GCONF_SCHEMAS_INSTALL
-install-data-hook:
-	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$(schema_DATA)
-
-uninstall-hook:
-	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-uninstall-rule $(srcdir)/$(schema_DATA)
-	rm -vf $(DESTDIR)$(bindir)/jack_mixer
-endif
-
 pacoinstall:
 	test `whoami` == root # test for root
 	-paco -rvB jack_mixer
diff -ruN a/gui.py b/gui.py
--- a/gui.py	2020-06-22 14:51:45.000000000 +0200
+++ b/gui.py	2020-06-23 11:51:38.699062969 +0200
@@ -16,10 +16,10 @@
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 
 import gi
-gi.require_version('GConf', '2.0')
 from gi.repository import GObject
 
 try:
+    gi.require_version('GConf', '2.0')
     from gi.repository import GConf
 except:
     print("Cannot load Python bindings for GConf, your preferences will not be preserved across jack_mixer invocations")
