--- cacti-0.8.8g/lib/adodb/adodb-lib.inc.php	2010-08-21 17:03:22.000000000 +0200
+++ cacti-0.8.8g-php7/lib/adodb/adodb-lib.inc.php	2016-04-21 12:30:45.707940772 +0200
@@ -534,7 +534,7 @@
 		//because we have to call MetaType.
 		//php can't do a $rsclass::MetaType()
 		$rsclass = $zthis->rsPrefix.$zthis->databaseType;
-		$recordSet =& new $rsclass(-1,$zthis->fetchMode);
+		$recordSet = new $rsclass(-1,$zthis->fetchMode);
 		$recordSet->connection = &$zthis;
 		
 		if (is_string($cacheRS) && $cacheRS == $rs) {
