diff -wbBur FreeMat-4.0.1-Source/libs/libCore/Loader.cpp FreeMat-4.0.1-Source.my/libs/libCore/Loader.cpp
--- FreeMat-4.0.1-Source/libs/libCore/Loader.cpp	2009-10-09 06:23:29.000000000 +0000
+++ FreeMat-4.0.1-Source.my/libs/libCore/Loader.cpp	2010-12-27 12:04:20.000000000 +0000
@@ -388,7 +388,7 @@
   context->addSpecialFunction("warning",WarningFunction,1,0,"msg",NULL);
   context->addFunction("error",ErrorFunction,1,0,"string",NULL);
   context->addSpecialFunction("eval",EvalFunction,2,-1,"try_clause","catch_clause",NULL);
-  context->addSpecialFunction("evalin",EvalInFunction,2,3,"workspace","expression",NULL);
+  context->addSpecialFunction("evalin",EvalInFunction,3,3,"workspace","expression","catch_clause",NULL);
   context->addSpecialFunction("assignin",AssignInFunction,3,0,"workspace","variablename","value",NULL);
   context->addSpecialFunction("feval",FevalFunction,-1,-1,NULL);
   context->addSpecialFunction("exist",ExistFunction,2,1,"item","kind",NULL);
