diff -ruN a/src/wheezy/template/engine.py b/src/wheezy/template/engine.py
--- a/src/wheezy/template/engine.py	2021-04-17 09:05:32.000000000 +0200
+++ b/src/wheezy/template/engine.py	2023-04-06 20:15:00.437384667 +0200
@@ -44,7 +44,7 @@
         self.global_vars = {"_r": self.render, "_i": self.import_name}
         self.loader = loader
         self.template_class = template_class or Template
-        self.compiler = Compiler(self.global_vars, -2)
+        self.compiler = Compiler(self.global_vars, 0)
         self.lexer = Lexer(**lexer_scan(extensions))
         self.parser = Parser(**parser_scan(extensions))
         self.builder = SourceBuilder(**builder_scan(extensions))
