diff -ruN a/librtosc.pc.cmake c/librtosc.pc.cmake
--- a/librtosc.pc.cmake	2019-04-16 01:18:21.000000000 +0200
+++ c/librtosc.pc.cmake	2019-04-18 23:43:41.030168614 +0200
@@ -3,9 +3,8 @@
 #
 
 prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=${prefix}
-libdir=${exec_prefix}
-includedir=${prefix}/include
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
 Name: rtosc
 Description: rtosc - a realtime safe open sound control serialization and dispatch system
