diff --git c/crates/rust-analyzer/src/cli/load_cargo.rs i/crates/rust-analyzer/src/cli/load_cargo.rs
index 5dba545b8..c9faeaa6b 100644
--- c/crates/rust-analyzer/src/cli/load_cargo.rs
+++ i/crates/rust-analyzer/src/cli/load_cargo.rs
@@ -69,10 +69,13 @@ pub fn load_workspace(
             if let Some(sysroot) = sysroot.as_ref() {
                 let standalone_server_name =
                     format!("rust-analyzer-proc-macro-srv{}", std::env::consts::EXE_SUFFIX);
-                let server_path = sysroot.root().join("libexec").join(&standalone_server_name);
-                if std::fs::metadata(&server_path).is_ok() {
-                    path = server_path;
-                    args = vec![];
+                for dir in ["libexec", "lib"] {
+                    let server_path = sysroot.root().join(dir).join(&standalone_server_name);
+                    if std::fs::metadata(&server_path).is_ok() {
+                        path = server_path;
+                        args = vec![];
+                        break;
+                    }
                 }
             }
         }
diff --git c/crates/rust-analyzer/src/reload.rs i/crates/rust-analyzer/src/reload.rs
index aa0510a4e..d7cf69c3a 100644
--- c/crates/rust-analyzer/src/reload.rs
+++ i/crates/rust-analyzer/src/reload.rs
@@ -327,21 +327,22 @@ impl GlobalState {
                             | ProjectWorkspace::Json { sysroot, .. } = ws
                             {
                                 if let Some(sysroot) = sysroot.as_ref() {
-                                    let server_path = sysroot
-                                        .root()
-                                        .join("libexec")
-                                        .join(&standalone_server_name);
-                                    if std::fs::metadata(&server_path).is_ok() {
-                                        tracing::debug!(
-                                            "Sysroot proc-macro server exists at {}",
-                                            server_path.display()
-                                        );
-                                        sysroot_server = Some(server_path);
-                                    } else {
-                                        tracing::debug!(
-                                            "Sysroot proc-macro server does not exist at {}",
-                                            server_path.display()
-                                        );
+                                    for dir in ["libexec", "lib"] {
+                                        let server_path =
+                                            sysroot.root().join(dir).join(&standalone_server_name);
+                                        if std::fs::metadata(&server_path).is_ok() {
+                                            tracing::debug!(
+                                                "Sysroot proc-macro server exists at {}",
+                                                server_path.display()
+                                            );
+                                            sysroot_server = Some(server_path);
+                                            break;
+                                        } else {
+                                            tracing::debug!(
+                                                "Sysroot proc-macro server does not exist at {}",
+                                                server_path.display()
+                                            );
+                                        }
                                     }
                                 }
                             }
