diff -aur pcapy-ng-1.0.5.old/setup.py pcapy-ng-1.0.5/setup.py
--- pcapy-ng-1.0.5.old/setup.py	2023-04-07 16:15:38.645831706 +0200
+++ pcapy-ng-1.0.5/setup.py	2023-04-07 16:16:49.277403503 +0200
@@ -44,25 +44,6 @@
     sources.append(os.path.join('win32', 'dllmain.cc'))
     macros.append(('WIN32', '1'))
 
-# HACK replace linker gcc with g++
-from distutils import sysconfig
-save_init_posix = sysconfig._init_posix
-
-
-def my_init_posix():
-    save_init_posix()
-    g = sysconfig._config_vars
-    compiler = g['LDSHARED'].split()[0]
-    flags = g['LDSHARED'].split()[1:]
-    if compiler == 'gcc':
-        g['LDSHARED'] = ' '.join(['g++'] + flags)
-    elif compiler == 'clang':
-        g['LDSHARED'] = ' '.join(['clang++'] + flags)
-        print('my_init_posix: changing LDSHARED =',
-              repr(g['LDSHARED']))
-        print('to', repr(g['LDSHARED']))
-sysconfig._init_posix = my_init_posix
-
 
 def read(fname):
     return open(os.path.join(os.path.dirname(__file__), fname)).read()
