diff --git a/ejabberd-dev/src/gen_mod.erl b/ejabberd-dev/src/gen_mod.erl
index f824ff7..5245dc6 100644
--- a/ejabberd-dev/src/gen_mod.erl
+++ b/ejabberd-dev/src/gen_mod.erl
@@ -39,6 +39,7 @@
 %%-export([behaviour_info/1]).
 
 -include("ejabberd.hrl").
+-include("logger.hrl").
 
 -record(ejabberd_module,
         {module_host = {undefined, <<"">>} :: {atom(), binary()},
diff --git a/mod_archive/src/mod_archive_odbc.erl b/mod_archive/src/mod_archive_odbc.erl
index d25ab19..a01ac32 100644
--- a/mod_archive/src/mod_archive_odbc.erl
+++ b/mod_archive/src/mod_archive_odbc.erl
@@ -83,6 +83,7 @@
          terminate/2, code_change/3]).
 
 -include("ejabberd.hrl").
+-include("logger.hrl").
 -include("jlib.hrl").
 
 -record(state, {host,
diff --git a/mod_archive/src/mod_archive_webview.erl b/mod_archive/src/mod_archive_webview.erl
index d8a5fa2..1ebdfd9 100644
--- a/mod_archive/src/mod_archive_webview.erl
+++ b/mod_archive/src/mod_archive_webview.erl
@@ -14,9 +14,10 @@
 	]).
 
 -include("ejabberd.hrl").
+-include("logger.hrl").
+-include("ejabberd_web_admin.hrl").
 -include("jlib.hrl").
--include("web/ejabberd_http.hrl").
--include("web/ejabberd_web_admin.hrl"). %for all the defines
+-include("ejabberd_http.hrl").
 
 -define(LINK(L) , "/archive/" ++ L).
 %-define(P(Els), ?XE("p", Els)).
diff --git a/mod_muc_admin/src/mod_muc_admin.erl b/mod_muc_admin/src/mod_muc_admin.erl
index 07af7cf..de06595 100644
--- a/mod_muc_admin/src/mod_muc_admin.erl
+++ b/mod_muc_admin/src/mod_muc_admin.erl
@@ -30,9 +30,9 @@
 
 -include("ejabberd.hrl").
 -include("jlib.hrl").
--include("mod_muc/mod_muc_room.hrl").
--include("web/ejabberd_http.hrl").
--include("web/ejabberd_web_admin.hrl").
+-include("mod_muc_room.hrl").
+-include("ejabberd_http.hrl").
+-include("ejabberd_web_admin.hrl").
 -include("ejabberd_commands.hrl").
 
 %% Copied from mod_muc/mod_muc.erl
diff --git a/mod_muc_log_http/src/mod_muc_log_http.erl b/mod_muc_log_http/src/mod_muc_log_http.erl
index f6b52ee..e39c88f 100644
--- a/mod_muc_log_http/src/mod_muc_log_http.erl
+++ b/mod_muc_log_http/src/mod_muc_log_http.erl
@@ -19,8 +19,8 @@
 
 -include("ejabberd.hrl").
 -include("jlib.hrl").
--include("web/ejabberd_http.hrl").
--include("mod_muc/mod_muc_room.hrl").
+-include("ejabberd_http.hrl").
+-include("mod_muc_room.hrl").
 -include_lib("kernel/include/file.hrl").
 
 -define(PROCNAME, mod_muc_log_http).
