diff --git a/tools/setup_helpers/cmake.py b/tools/setup_helpers/cmake.py
index d5db749d15..fd54cca6b8 100644
--- a/tools/setup_helpers/cmake.py
+++ b/tools/setup_helpers/cmake.py
@@ -295,6 +295,10 @@ class CMake:
         build_options.update(cmake__options)
 
         CMake.defines(args,
+                      DISABLE_AVX2=1,
+                      DISABLE_AVX512F=1,
+                      DISABLE_FMA4=1,
+                      DISABLE_SSE4=1,
                       PYTHON_EXECUTABLE=sys.executable,
                       PYTHON_LIBRARY=cmake_python_library,
                       PYTHON_INCLUDE_DIR=distutils.sysconfig.get_python_inc(),
