--- fglrx-install/common/etc/ati/authatieventsd.sh	2012-11-14 14:20:26.274922842 +0200
+++ fglrx-install/common/etc/ati/authatieventsd.sh	2012-11-14 14:20:10.591585406 +0200
@@ -34,8 +34,9 @@
 
     # Check xdm/kdm
 
-    XDM_AUTH_MASK=/var/lib/xdm/authdir/authfiles/A$1*
-    XDM_AUTH_FILE=`ls -t $XDM_AUTH_MASK 2>/dev/null | head -n 1`   # Choose the newest file
+    XDM_AUTH_MASK=/var/lib/xdm/authfiles/A$1*
+    KDM_AUTH_MASK=/var/run/xauth/A$1*
+    XDM_AUTH_FILE=`ls -t $XDM_AUTH_MASK $KDM_AUTH_MASK 2>/dev/null | head -n 1`   # Choose the newest file
     if [ -n "$XDM_AUTH_FILE" ]; then
         SERVER_AUTH_FILE=$XDM_AUTH_FILE
         DISP_SEARCH_STRING="#ffff#"
@@ -44,10 +45,10 @@
 
     # Check gdm
 
-    GDM_AUTH_FILE=/var/lib/gdm/$1.Xauth
+    GDM_AUTH_FILE=/var/run/gdm/auth-for-gdm-*/database
     if [ -e $GDM_AUTH_FILE ]; then
         SERVER_AUTH_FILE=$GDM_AUTH_FILE
-        DISP_SEARCH_STRING="$1"
+        DISP_SEARCH_STRING="unix$1"
         return 0
     fi
 
