From de048bbaa2c236914ab6e13008bbc1f434b05e62 Mon Sep 17 00:00:00 2001
From: Jonas Witschel <diabonas@archlinux.org>
Date: Sun, 5 Jul 2020 13:43:42 +0200
Subject: [PATCH] Disable Thunderbird 78 upgrade warning message

---
 package/configure.jsm | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/package/configure.jsm b/package/configure.jsm
index c230cdf0..8f09b54f 100644
--- a/package/configure.jsm
+++ b/package/configure.jsm
@@ -311,6 +311,13 @@ var EnigmailConfigure = {
   upgradeTo217: function(win) {
     if (EnigmailCompat.isPostbox() || (!EnigmailCompat.isAtLeastTb68())) return;
 
-    displayUpgradeInfo(win);
+    /* Enigmail obtained from addons.mozilla.org shows a warning that the extension will stop
+       working with Thunderbird >= 78 and that users should hold back on upgrading Thunderbird until
+       its new builtin OpenPGP support is mature (expected for Thunderbird 78.2). This makes sense for
+       users of the builtin Thunderbird auto-update function, but less so for user obtaining Thunderbird
+       from a distribution repository, since it encourages partial upgrades. Hence for Arch Linux we
+       remove this warning and add a versioned dependency on Thunderbird < 78 instead to prevent users
+       from accidentally upgrading to the next release while still having Enigmail installed. */
+    // displayUpgradeInfo(win);
   }
 };
-- 
2.28.0

