diff --git a/src/music-player/view/lyricwidget.cpp b/src/music-player/view/lyricwidget.cpp
index b522ae4d..bb2ffc8a 100644
--- a/src/music-player/view/lyricwidget.cpp
+++ b/src/music-player/view/lyricwidget.cpp
@@ -26,6 +26,7 @@
 #include <QScrollArea>
 #include <QHBoxLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QResizeEvent>
 #include <QPaintEvent>
 #include <QStringListModel>
diff --git a/src/music-player/view/widget/cover.cpp b/src/music-player/view/widget/cover.cpp
index 8ade80be..6987fec5 100644
--- a/src/music-player/view/widget/cover.cpp
+++ b/src/music-player/view/widget/cover.cpp
@@ -23,6 +23,7 @@
 
 #include <QDebug>
 #include <QPainter>
+#include <QPainterPath>
 #include <QGraphicsDropShadowEffect>
 
 static auto borderPenWidth = 1.0;
diff --git a/src/music-player/view/widget/delegate/musiclistdatadelegate.cpp b/src/music-player/view/widget/delegate/musiclistdatadelegate.cpp
index 752c6e15..da30078b 100644
--- a/src/music-player/view/widget/delegate/musiclistdatadelegate.cpp
+++ b/src/music-player/view/widget/delegate/musiclistdatadelegate.cpp
@@ -22,6 +22,7 @@
 #include "musiclistdatadelegate.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 #include <QDate>
 #include <QEvent>
diff --git a/src/music-player/view/widget/delegate/musicsearchlistdelegate.cpp b/src/music-player/view/widget/delegate/musicsearchlistdelegate.cpp
index 943eeca6..b9f40b22 100755
--- a/src/music-player/view/widget/delegate/musicsearchlistdelegate.cpp
+++ b/src/music-player/view/widget/delegate/musicsearchlistdelegate.cpp
@@ -25,6 +25,7 @@
 #include <DHiDPIHelper>
 #include <QDebug>
 #include <QPainter>
+#include <QPainterPath>
 #include <QTextDocument>
 #include <QStyleOptionViewItem>
 #include <QTextCursor>
diff --git a/src/music-player/view/widget/delegate/playitemdelegate.cpp b/src/music-player/view/widget/delegate/playitemdelegate.cpp
index 19ff14cf..8593c1a2 100644
--- a/src/music-player/view/widget/delegate/playitemdelegate.cpp
+++ b/src/music-player/view/widget/delegate/playitemdelegate.cpp
@@ -25,6 +25,7 @@
 #include <QDebug>
 #include <QFont>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStandardItemModel>
 
 #include <musicmeta.h>
diff --git a/src/music-player/view/widget/musicpixmapbutton.cpp b/src/music-player/view/widget/musicpixmapbutton.cpp
index 32cd78eb..d4f8c391 100644
--- a/src/music-player/view/widget/musicpixmapbutton.cpp
+++ b/src/music-player/view/widget/musicpixmapbutton.cpp
@@ -23,6 +23,7 @@
 
 #include <QDebug>
 #include <QPainter>
+#include <QPainterPath>
 #include <QRect>
 
 MusicPixmapButton::MusicPixmapButton(QWidget *parent)
diff --git a/src/music-player/view/widget/musictitleimage.cpp b/src/music-player/view/widget/musictitleimage.cpp
index 98744fdf..8a39e8d1 100644
--- a/src/music-player/view/widget/musictitleimage.cpp
+++ b/src/music-player/view/widget/musictitleimage.cpp
@@ -22,6 +22,7 @@
 #include "musictitleimage.h"
 
 #include <QPainter>
+#include <QPainterPath>
 
 MusicTitleImageWidget::MusicTitleImageWidget(QWidget *parent)
     : DWidget( parent)
diff --git a/src/music-player/view/widget/soundvolume.cpp b/src/music-player/view/widget/soundvolume.cpp
index 79507306..c4328305 100644
--- a/src/music-player/view/widget/soundvolume.cpp
+++ b/src/music-player/view/widget/soundvolume.cpp
@@ -23,6 +23,7 @@
 
 #include <QDebug>
 #include <QPainter>
+#include <QPainterPath>
 #include <QWheelEvent>
 #include <QGraphicsDropShadowEffect>
 #include <QVBoxLayout>
diff --git a/src/music-player/view/widget/tooltips.cpp b/src/music-player/view/widget/tooltips.cpp
index b5233b69..acb77924 100644
--- a/src/music-player/view/widget/tooltips.cpp
+++ b/src/music-player/view/widget/tooltips.cpp
@@ -33,6 +33,7 @@
 #include <QHBoxLayout>
 #include <QLabel>
 #include <QPainter>
+#include <QPainterPath>
 #include <QGraphicsDropShadowEffect>
 #include <QPropertyAnimation>
 #include <QGraphicsOpacityEffect>
diff --git a/src/music-player/view/widget/waveform.cpp b/src/music-player/view/widget/waveform.cpp
index e4822bee..832b28f1 100644
--- a/src/music-player/view/widget/waveform.cpp
+++ b/src/music-player/view/widget/waveform.cpp
@@ -26,6 +26,7 @@
 #include <QPaintEvent>
 #include <QApplication>
 #include <QPainter>
+#include <QPainterPath>
 #include <QAudioBuffer>
 #include <QAudioFormat>
 #include <QWidget>
diff --git a/src/music-player/view/widget/waveformscale.cpp b/src/music-player/view/widget/waveformscale.cpp
index 93ab3004..b7f40c7b 100755
--- a/src/music-player/view/widget/waveformscale.cpp
+++ b/src/music-player/view/widget/waveformscale.cpp
@@ -27,6 +27,7 @@
 #include <QEvent>
 #include <QPaintEvent>
 #include <QPainter>
+#include <QPainterPath>
 
 #include <mediameta.h>
 
