commit e55fcfa8e9a343aba4d146aba5d1a262f79c2be0
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Fri Mar 20 21:48:25 2020 +0800

    Fix installation target

diff --git a/ukui-menu.pro b/ukui-menu.pro
index c666653..f4eb67e 100644
--- a/ukui-menu.pro
+++ b/ukui-menu.pro
@@ -13,6 +13,8 @@ QMAKE_CXXFLAGS += -g
 TARGET = ukui-menu
 TEMPLATE = app
 
+target.path = /usr/bin
+
 TRANSLATIONS+=\
     ukui_menu_ch.ts
 
@@ -95,3 +97,9 @@ RESOURCES += \
 
 PKGCONFIG+=glib-2.0
 CONFIG += no_keywords link_pkgconfig
+
+desktop_file.files = ukui-menu.desktop
+desktop_file.path = /etc/xdg/autostart
+
+INSTALLS += \
+    target desktop_file
