commit 79c814ac445f21a50e7b9957f4a8ae449ee880a1
Author: Anatol Pomozov <anatol.pomozov@gmail.com>
Date:   Sat Mar 21 09:13:31 2020 -0700

    ArchLinux fixes

diff --git a/Gemfile b/Gemfile
index 87a0cff84c1..884bf737ba5 100644
--- a/Gemfile
+++ b/Gemfile
@@ -326,6 +326,9 @@ gem 'gettext', '~> 3.3', require: false, group: :development
 
 gem 'batch-loader', '~> 2.0.1'
 
+gem 'irb' # https://bugs.archlinux.org/task/68569
+gem 'rake'
+
 # Perf bar
 gem 'peek', '~> 1.1'
 
