commit 12f6321d679fb5aedef23680bb618a1443146c4f
Author: Anatol Pomozov <anatol.pomozov@gmail.com>
Date:   Sat Mar 21 09:13:31 2020 -0700

    ArchLinux fixes

diff --git a/Gemfile b/Gemfile
index 6f9340ebd7dc..1fb09413b91d 100644
--- a/Gemfile
+++ b/Gemfile
@@ -330,6 +330,9 @@ gem 'gettext', '~> 3.3', require: false, group: :development
 
 gem 'batch-loader', '~> 2.0.1'
 
+gem 'irb' # https://bugs.archlinux.org/task/68569
+gem 'rake'
+
 # Perf bar
 gem 'peek', '~> 1.1'
 
