commit ec0738c3091f91465308a15051ecb5fc4dcd38c3
Author: Anatol Pomozov <anatol.pomozov@gmail.com>
Date:   Sat Mar 21 09:13:31 2020 -0700

    ArchLinux fixes

diff --git a/Gemfile b/Gemfile
index 39e61564968..d2564260ed5 100644
--- a/Gemfile
+++ b/Gemfile
@@ -324,6 +324,9 @@ gem 'gettext', '~> 3.3', require: false, group: :development
 
 gem 'batch-loader', '~> 2.0.1'
 
+gem 'irb' # https://bugs.archlinux.org/task/68569
+gem 'rake'
+
 # Perf bar
 gem 'peek', '~> 1.1'
 
