diff --git a/lib/rack/builder.rb b/lib/rack/builder.rb
index 975cf1e1..c61e8421 100644
--- a/lib/rack/builder.rb
+++ b/lib/rack/builder.rb
@@ -79,6 +79,14 @@ module Rack
     # The +call+ method in this example sets an additional environment key which then can be
     # referenced in the application if required.
     def use(middleware, *args, &block)
+      # workaround for ruby 2.7 problem discussed https://gitlab.com/groups/gitlab-org/-/epics/2380
+      if not args.empty? and args[-1] == block
+        # for some reason ruby adds block defined in lib/api/api_guard.rb
+        # both to args and to 'block' parameter
+        # pop extra parameter from 'args'
+        args.pop
+      end
+
       if @map
         mapping, @map = @map, nil
         @use << proc { |app| generate_map app, mapping }
