diff -rupN noise-0.3.0.orig/data/noise.desktop noise-0.3.0/data/noise.desktop
--- noise-0.3.0.orig/data/noise.desktop	2014-07-14 15:32:41.859039779 +0200
+++ noise-0.3.0/data/noise.desktop	2014-07-14 15:33:18.950708937 +0200
@@ -3,7 +3,7 @@ Type=Application
 Name=Noise
 GenericName=Music Player
 Comment=Listen to music
-Exec=noise %U
+Exec=noise-player %U
 Icon=multimedia-audio-player
 Terminal=false
 Categories=Audio;Music;Player;AudioVideo;GNOME;GTK;
diff -rupN noise-0.3.0.orig/src/CMakeLists.txt noise-0.3.0/src/CMakeLists.txt
--- noise-0.3.0.orig/src/CMakeLists.txt	2014-07-14 15:32:41.852369815 +0200
+++ noise-0.3.0/src/CMakeLists.txt	2014-07-14 15:38:21.230137167 +0200
@@ -107,7 +107,7 @@ vala_precompile (CLIENT_VALA_C ${SRC_TAR
   ${CLIENT_VAPI}
 )
 
-set (EXEC_NAME ../${SRC_TARGET})
+set (EXEC_NAME ../${SRC_TARGET}-player)
 
 add_definitions (${DEPS_CFLAGS})
 link_directories (${DEPS_LIBRARY_DIRS})
