--- qt-recordmydesktop-0.3.8/configure.ac~	2013-06-17 18:22:27.747345090 +0000
+++ qt-recordmydesktop-0.3.8/configure.ac	2013-06-17 18:22:46.937225955 +0000
@@ -43,7 +43,7 @@
 
 export PYTHONPATH=$PYTHONPATH
 
-export PYQT4_VERSION=`python -c 'import PyQt4; from PyQt4 import QtCore; print QtCore.PYQT_VERSION_STR'  2>>/dev/null `
+export PYQT4_VERSION=`python2 -c 'import PyQt4; from PyQt4 import QtCore; print QtCore.PYQT_VERSION_STR'  2>>/dev/null `
 if test x$PYQT4_VERSION == x; then
     AC_MSG_ERROR(You need PyQt4>=4.1 installed to procceed);
 fi
