diff --git a/widgets/ccheckbutton.h b/widgets/ccheckbutton.h
index c219285..0273c59 100644
--- a/widgets/ccheckbutton.h
+++ b/widgets/ccheckbutton.h
@@ -21,6 +21,7 @@
 
 #include <QMap>
 #include <DToolButton>
+#include <DWidget>
 
 DWIDGET_USE_NAMESPACE
 
diff --git a/widgets/cclickbutton.h b/widgets/cclickbutton.h
index 7da95b4..d1f0eed 100644
--- a/widgets/cclickbutton.h
+++ b/widgets/cclickbutton.h
@@ -20,6 +20,7 @@
 #define CCLICKBUTTON_H
 
 #include <DToolButton>
+#include <DWidget>
 
 DWIDGET_USE_NAMESPACE
 
diff --git a/frame/linewidget.h b/frame/linewidget.h
index a8e28091..c9897e5b 100644
--- a/frame/linewidget.h
+++ b/frame/linewidget.h
@@ -21,7 +21,7 @@
 
 #include <DWidget>
 #include <DLabel>
-
+#include <DComboBox>
 #include "drawshape/globaldefine.h"
 
 class CSideWidthWidget;
diff --git a/widgets/csidewidthwidget.h b/widgets/csidewidthwidget.h
index f3027762..71cc94f1 100644
--- a/widgets/csidewidthwidget.h
+++ b/widgets/csidewidthwidget.h
@@ -25,6 +25,7 @@
 #include <DFrame>
 #include <QHBoxLayout>
 #include <DWidget>
+#include <DComboBox>
 
 DWIDGET_USE_NAMESPACE
 
