diff -rupN "Simple OpenGL Image Library.orig/projects/makefile/alternate Makefile.txt" "Simple OpenGL Image Library/projects/makefile/alternate Makefile.txt"
--- "Simple OpenGL Image Library.orig/projects/makefile/alternate Makefile.txt"	2008-07-07 18:13:28.000000000 +0200
+++ "Simple OpenGL Image Library/projects/makefile/alternate Makefile.txt"	2013-10-11 12:31:18.545860800 +0200
@@ -1,7 +1,7 @@
 MAKE = make
 CC = gcc
-INSTALL_FILE = install -p -o root -g root -m 644
-INSTALL_DIR = install -p -o root -g root -d
+INSTALL_FILE = install -p -m 644
+INSTALL_DIR = install -p -d
 LN = ln -s
 RM = rm -fv
 CFLAGS += -c -O2 -Wall
@@ -10,13 +10,12 @@ LDFLAGS +=
 CFILES = image_DXT.c image_helper.c SOIL.c stb_image_aug.c
 OFILES = $(CFILES:.c=.o)
 LIBNAME = libSOIL
-VERSION = 1.07-20071110
+VERSION = 1.16
 MAJOR = 1
 
 HFILES = SOIL.h image_DXT.h image_helper.h \
   stbi_DDS_aug.h stbi_DDS_aug_c.h stb_image_aug.h
 AFILE = libSOIL.a
-SOFILE = libSOIL.so.$(VERSION)
 INCLUDEDIR = /usr/include/SOIL
 LIBDIR = /usr/lib
 
@@ -29,22 +28,17 @@ lib: $(OFILES)
 	# create static library
 	ar -cvq $(LIBNAME).a $(OFILES)
 	# create shared library
-	gcc -shared -Wl,-soname,$(LIBNAME).so.$(MAJOR) -o $(LIBNAME).so.$(VERSION) $(OFILES)
 
 install:
 	$(INSTALL_DIR) $(DESTDIR)/$(INCLUDEDIR)
 	$(INSTALL_FILE) $(HFILES) $(DESTDIR)/$(INCLUDEDIR)
 	$(INSTALL_DIR) $(DESTDIR)/$(LIBDIR)
 	$(INSTALL_FILE) $(AFILE) $(DESTDIR)/$(LIBDIR)
-	$(INSTALL_FILE) $(SOFILE) $(DESTDIR)/$(LIBDIR)
-	( cd $(DESTDIR)/$(LIBDIR) && $(LN) $(SOFILE) $(LIBNAME).so.$(MAJOR) \
-	  && $(LN) $(SOFILE) $(LIBNAME).so )
 
 clean:
 	$(RM) *.o
 	$(RM) *~
 
 distclean:
-	$(RM) $(AFILE) $(SOFILE)
 
 .PHONY: all lib clean distclean
