diff --git a/src/sage/libs/cypari2/paripriv.pxd b/src/sage/libs/cypari2/paripriv.pxd
index b6ea381..921599d 100644
--- a/src/sage/libs/cypari2/paripriv.pxd
+++ b/src/sage/libs/cypari2/paripriv.pxd
@@ -26,5 +26,5 @@ cdef extern from "pari/paripriv.h":
 
     extern gp_data* GP_DATA
 
-cdef extern from "pari/anal.h":
-    char* closure_func_err()
+cdef extern:  # Declared in private PARI header file "anal.h"
+    const char* closure_func_err()
