--- a/src/mcomix-0.96/mcomix/archive/rar.py	2011-12-22 08:52:32.000000000 +0100
+++ b/src/mcomix-0.96/mcomix/archive/rar.py	2012-02-11 15:22:18.898980508 +0100
@@ -45,9 +45,6 @@
     def is_available():
         return bool(RarExecArchive._find_unrar_executable())
 
-if rarfile.UnrarDll.is_available():
-    RarArchive = rarfile.UnrarDll
-else:
-    RarArchive = RarExecArchive
+RarArchive = RarExecArchive
 
 # vim: expandtab:sw=4:ts=4
