diff --git a/mcomix-1.00/mcomix/event.py b/mcomix-1.00-mod/mcomix/event.py
index bc8cf79..bf80ec2 100644
--- a/mcomix/event.py
+++ b/mcomix/event.py
@@ -117,8 +117,11 @@ class EventHandler:
 
         # Zooming commands for manual zoom mode
         manager.register('zoom in',
-            ['equal'],
+            ['plus', 'KP_Add', 'equal'],
             self._window.actiongroup.get_action('zoom_in').activate)
+        manager.register('zoom out',
+            ['minus', 'KP_Subtract'],
+            self._window.actiongroup.get_action('zoom_out').activate)
         # Zoom out is already defined as GTK menu hotkey
         manager.register('zoom original',
             ['KP_0'],
