diff --git a/cmake/Modules/FindGTK3.cmake b/cmake/Modules/FindGTK3.cmake
index e07a4e90..9a4dbca3 100644
--- a/cmake/Modules/FindGTK3.cmake
+++ b/cmake/Modules/FindGTK3.cmake
@@ -409,7 +409,7 @@ endif()
 #
 
 find_package(Freetype QUIET)
-list(APPEND GTK3_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS})
+list(APPEND GTK3_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS} /usr/include/harfbuzz)
 list(APPEND GTK3_LIBRARIES ${FREETYPE_LIBRARIES})
 
 foreach(_GTK3_component ${GTK3_FIND_COMPONENTS})
