--- a/mythweb/includes/errors.php
+++ b/mythweb/includes/errors.php
@@ -103,6 +103,9 @@
 /**/
     function error_handler($errno, $errstr, $errfile, $errline, $vars) {
         global $db;
+    // Leave early if we haven't requested reports from this kind of error
+        if (!($errno & error_reporting()))
+            return;
         if (class_exists('Translate'))
             $errstr = Translate::find()->string($errstr);
     // Try to auto-repair damaged SQL tables
@@ -113,9 +116,6 @@
             add_error('Regular Expression Error:  '.$match[1]);
             return;
         }
-    // Leave early if we haven't requested reports from this kind of error
-        if (!($errno & error_reporting()))
-            return;
     // Fatal errors should report considerably more detail
         if (in_array($errno, array(E_USER_ERROR, E_ERROR))) {
         // What type of error?
