diff --git a/src/50-marco-desktop-key.xml.in b/src/50-marco-desktop-key.xml.in
index 00ebc91..f9e33c3 100644
--- a/src/50-marco-desktop-key.xml.in
+++ b/src/50-marco-desktop-key.xml.in
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" ?>
-<KeyListEntries _name="Desktop" wm_name="Marco" package="marco" schema="org.mate.Marco.global-keybindings">
+<KeyListEntries _name="Desktop" wm_name="Metacity (Marco)" package="marco" schema="org.mate.Marco.global-keybindings">
 
 	<KeyListEntry name="panel-run-dialog" _description="Show the panel's &quot;Run Application&quot; dialog box" />
 
diff --git a/src/50-marco-global-key.xml.in b/src/50-marco-global-key.xml.in
index 9a61c37..81a8db8 100644
--- a/src/50-marco-global-key.xml.in
+++ b/src/50-marco-global-key.xml.in
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" ?>
-<KeyListEntries _name="Window Management" wm_name="Marco" package="marco" schema="org.mate.Marco.global-keybindings">
+<KeyListEntries _name="Window Management" wm_name="Metacity (Marco)" package="marco" schema="org.mate.Marco.global-keybindings">
 
 	<KeyListEntry name="switch-windows" _description="Move between windows, using a popup window" />
 
diff --git a/src/50-marco-window-key.xml.in b/src/50-marco-window-key.xml.in
index 3aba645..5d79933 100644
--- a/src/50-marco-window-key.xml.in
+++ b/src/50-marco-window-key.xml.in
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" ?>
-<KeyListEntries _name="Window Management" wm_name="Marco" package="marco" schema="org.mate.Marco.window-keybindings">
+<KeyListEntries _name="Window Management" wm_name="Metacity (Marco)" package="marco" schema="org.mate.Marco.window-keybindings">
 
 	<KeyListEntry
 	name="activate-window-menu"
