diff -ruN a/tests/nxpcertgen/test_certgen.py b/tests/nxpcertgen/test_certgen.py
--- a/tests/nxpcertgen/test_certgen.py	2022-07-29 15:19:47.000000000 +0200
+++ b/tests/nxpcertgen/test_certgen.py	2022-07-31 21:09:38.614790577 +0200
@@ -12,6 +12,7 @@
 
 import pytest
 import yaml
+from yaml import Loader
 from click.testing import CliRunner
 
 from spsdk import SPSDKError
@@ -270,6 +271,6 @@
         assert result.exit_code == 0
         assert os.path.isfile(template)
         with open(template) as f:
-            data = yaml.load(f)
+            data = yaml.load(f, Loader=Loader)
         # there should be at least 5 items in the template
         assert len(data) > 5
