diff --git a/tests/test_regex.py b/tests/test_regex.py
index d131cdd..6f40efa 100644
--- a/tests/test_regex.py
+++ b/tests/test_regex.py
@@ -5,10 +5,10 @@ import pytest
 pytestmark = pytest.mark.asyncio
 
 async def test_regex_httpbin_default_user_agent(get_version):
-    assert await get_version("example", {
+    assert (await get_version("example", {
         "url": "https://httpbin.org/get",
         "regex": r'"User-Agent": "([^"]+)"',
-    }) == "lilydjwg/nvchecker"
+    })).startswith("lilydjwg/nvchecker")
 
 async def test_regex_httpbin_user_agent(get_version):
     assert await get_version("example", {
