From 4fc62d35888c3d695956793d5724a52ba3a99574 Mon Sep 17 00:00:00 2001
From: lilydjwg <lilydjwg@gmail.com>
Date: Wed, 8 Apr 2020 18:17:32 +0800
Subject: [PATCH] tests: update test_archpkg

---
 tests/test_archpkg.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/test_archpkg.py b/tests/test_archpkg.py
index 36f7222..3096394 100644
--- a/tests/test_archpkg.py
+++ b/tests/test_archpkg.py
@@ -18,7 +18,7 @@ async def test_archpkg_provided(get_version):
     assert await get_version("jsoncpp", {
         "archpkg": None,
         "provided": "libjsoncpp.so",
-    }) == "21-64"
+    }) == "22-64"
 
 @flaky
 async def test_archpkg_provided_strip(get_version):
@@ -26,5 +26,5 @@ async def test_archpkg_provided_strip(get_version):
         "archpkg": None,
         "provided": "libjsoncpp.so",
         "strip-release": True,
-    }) == "21"
+    }) == "22"
 
