diff --git a/WORKSPACE b/WORKSPACE
index 30007945..479886aa 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -82,11 +82,10 @@ http_archive(
     name = "org_tensorflow",
     # NOTE: when updating this, MAKE SURE to also update the protobuf_js runtime version
     # in third_party/workspace.bzl to >= the protobuf/protoc version provided by TF.
-    sha256 = "48ddba718da76df56fd4c48b4bbf4f97f254ba269ec4be67f783684c75563ef8",
-    strip_prefix = "tensorflow-2.0.0-rc0",
+    sha256 = "fe76e3c5dd021eca4b3c1f22393514e2019479640bcb7df9158d6990a4c83294",
+    strip_prefix = "tensorflow-c6daad319d78e08d2dacbc4aa58094ce541188cf",
     urls = [
-        "http://mirror.tensorflow.org/github.com/tensorflow/tensorflow/archive/v2.0.0-rc0.tar.gz",  # 2019-08-23
-        "https://github.com/tensorflow/tensorflow/archive/v2.0.0-rc0.tar.gz",
+        "https://github.com/tensorflow/tensorflow/archive/c6daad319d78e08d2dacbc4aa58094ce541188cf.tar.gz",
     ],
 )
 
