--- a/fonomobutton.c	2008-10-20 01:06:40.000000000 -0700
+++ b/fonomobutton.c	2009-11-22 10:40:48.000000000 -0800
@@ -88,7 +88,7 @@
 {
     GtkWidget *button, *image, *sep;
     GtkIconInfo *icoinfo;
-    gchar *iconames[]={"camera-web", "camera-fonomo"};
+    const gchar *iconames[]={"camera-web", "camera-fonomo"};
 
     button = gtk_button_new();
     gtk_button_set_relief(GTK_BUTTON(button), GTK_RELIEF_NONE);
