diff -Naur libsignon-glib-1.6.orig/Makefile.am libsignon-glib-1.6/Makefile.am
--- libsignon-glib-1.6.orig/Makefile.am	2012-08-13 10:09:23.000000000 +0200
+++ libsignon-glib-1.6/Makefile.am	2012-10-16 22:04:28.805736477 +0200
@@ -11,7 +11,7 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libsignon-glib.pc
 
-libsignondocdir = ${prefix}/doc/reference
+libsignondocdir = ${docdir}
 libsignondoc_DATA= README COPYING AUTHORS ChangeLog INSTALL NEWS
 
 EXTRA_DIST = \
