--- data/etc/sysactions.conf.bak	2011-02-25 18:25:13.210000005 +0100
+++ data/etc/sysactions.conf	2011-02-25 18:25:18.160000006 +0100
@@ -45,8 +45,8 @@
 
 action:   halt      /sbin/shutdown -h now
 action:   reboot    /sbin/shutdown -r now
-action:   suspend   /etc/acpi/sleep.sh force
-action:   hibernate /etc/acpi/hibernate.sh force
+action:   suspend   /usr/sbin/pm-suspend
+action:   hibernate /usr/sbin/pm-hibernate
 
 # on FreeBSD use this instead of the above.
 #action suspend  /usr/sbin/zzz 
