diff -Naur electricsheep-2.7b12.orig/electricsheep.c electricsheep-2.7b12/electricsheep.c
--- electricsheep-2.7b12.orig/electricsheep.c	2011-10-31 22:36:30.000000000 -0400
+++ electricsheep-2.7b12/electricsheep.c	2011-10-31 22:40:04.000000000 -0400
@@ -662,7 +662,7 @@
     input_stream_index = -1;
     for (j = 0; j < ictx->nb_streams; j++) {
 	AVCodecContext *enc = ictx->streams[j]->codec;
-	if (CODEC_TYPE_VIDEO == enc->codec_type) {
+	if (AVMEDIA_TYPE_VIDEO == enc->codec_type) {
 	    input_stream_index = j;
 	    break;
 	}
@@ -730,7 +730,7 @@
 	av_init_packet(&opkt);
 	if (av_parser_change(ictx->streams[input_stream_index]->parser, output_ctx->streams[0]->codec,
 			     &opkt.data, &opkt.size, ipkt.data, ipkt.size,
-			     ipkt.flags & PKT_FLAG_KEY))
+			     ipkt.flags & AV_PKT_FLAG_KEY))
 	    opkt.destruct= av_destruct_packet;
 		
 	if (-1 == av_interleaved_write_frame(output_ctx, &opkt)) {
diff -Naur electricsheep-2.7b12.orig/electricsheep.c electricsheep-2.7b12/electricsheep.c
--- electricsheep-2.7b12.orig/electricsheep.c	2011-11-01 00:21:51.000000000 -0400
+++ electricsheep-2.7b12/electricsheep.c	2011-11-01 00:25:10.000000000 -0400
@@ -686,9 +686,9 @@
 	    }
 	}
 
-	output_ctx = av_alloc_format_context();
+	output_ctx = avformat_alloc_context();
 
-	ofmt = guess_format(NULL, fname, NULL);
+	ofmt = av_guess_format(NULL, fname, NULL);
 	if (!ofmt) {
 	    fprintf(logout, "could not determine format from %s.\n", fname);
 	    exit(1);
