diff -Naur gloobus-preview-0.4.5-ubuntu11~ppa290.orig/src/gloobus-preview-interface-text.h gloobus-preview-0.4.5-ubuntu11~ppa290/src/gloobus-preview-interface-text.h
--- gloobus-preview-0.4.5-ubuntu11~ppa290.orig/src/gloobus-preview-interface-text.h	2012-04-06 15:07:29.000000000 +0200
+++ gloobus-preview-0.4.5-ubuntu11~ppa290/src/gloobus-preview-interface-text.h	2013-05-17 06:27:37.833360264 +0200
@@ -1,8 +1,7 @@
 #ifndef _ITEXT_
 #define _ITEXT_
 
-#include <gtksourceview/gtksourceview.h>
-#include <gtksourceview/gtksourcelanguagemanager.h>
+#include <gtksourceview/gtksource.h>
 //sudo ln -s /usr/include/gtksourceview-2.0/gtksourceview/ /usr/include/gtksourceview
 
 #include "gloobus-preview-interface-main.h"
diff -Naur gloobus-preview-0.4.5-ubuntu11~ppa290.orig/src/gloobus-preview-plugin-manager.cpp gloobus-preview-0.4.5-ubuntu11~ppa290/src/gloobus-preview-plugin-manager.cpp
--- gloobus-preview-0.4.5-ubuntu11~ppa290.orig/src/gloobus-preview-plugin-manager.cpp	2012-04-06 15:07:29.000000000 +0200
+++ gloobus-preview-0.4.5-ubuntu11~ppa290/src/gloobus-preview-plugin-manager.cpp	2013-05-17 06:28:19.300984059 +0200
@@ -1,6 +1,6 @@
 #include "gloobus-preview-plugin-manager.h"
 #include <dlfcn.h>
-#include <gtksourceview/gtksourcelanguagemanager.h>
+#include <gtksourceview/gtksource.h>
 
 vector<gp_plugin_container*> PluginManager::plugins;
 map<string, int> PluginManager::filetypes;
diff -Naur gloobus-preview-0.4.5-ubuntu11~ppa290.orig/src/plugin-text/plugin-text.h gloobus-preview-0.4.5-ubuntu11~ppa290/src/plugin-text/plugin-text.h
--- gloobus-preview-0.4.5-ubuntu11~ppa290.orig/src/plugin-text/plugin-text.h	2012-04-06 15:07:29.000000000 +0200
+++ gloobus-preview-0.4.5-ubuntu11~ppa290/src/plugin-text/plugin-text.h	2013-05-17 06:26:26.788384315 +0200
@@ -3,10 +3,7 @@
 
 #ifndef PLUGIN_LOADER
 
-#include <gtksourceview/gtksourceview.h>
-#include <gtksourceview/gtksourcelanguagemanager.h>
-#include <gtksourceview/gtksourcestylescheme.h>
-#include <gtksourceview/gtksourcestyleschememanager.h>
+#include <gtksourceview/gtksource.h>
 #include <gio/gio.h>
 
 #include "../gloobus-preview-config.h"
