diff -Naur gloobus-preview-0.4.5-ubuntu11~ppa290.orig/src/gloobus-preview-window.cpp gloobus-preview-0.4.5-ubuntu11~ppa290/src/gloobus-preview-window.cpp
--- gloobus-preview-0.4.5-ubuntu11~ppa290.orig/src/gloobus-preview-window.cpp	2012-04-06 15:07:29.000000000 +0200
+++ gloobus-preview-0.4.5-ubuntu11~ppa290/src/gloobus-preview-window.cpp	2013-05-18 09:49:24.833574800 +0200
@@ -560,12 +560,12 @@
     string label(gtk_menu_item_get_label(item));
     if(label ==  _("Settings"))
         cmd = "gloobus-preview-configuration";
-    else if(label ==  _("Report a Bug"))
-        cmd = "gnome-open \"https://bugs.launchpad.net/gloobus-preview/\"";
+    else if(label ==  _("Report a Problem"))
+        cmd = "xdg-open \"https://bugs.launchpad.net/gloobus-preview/\"";
     else if(label == _("Blog"))
-        cmd = "gnome-open \"http://gloobus.wordpress.com\"" ;
+        cmd = "xdg-open \"http://gloobus.wordpress.com\"" ;
     else if(label == _("Donate"))
-        cmd = "gnome-open \"https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=guitarboy000%40gmail%2ecom&item_name=Gloobus%2c%20A%20Quicklook%20for%20linux&no_shipping=0&no_note=1&tax=0&currency_code=EUR&lc=US&bn=PP%2dDonationsBF&chars\"" ;
+        cmd = "xdg-open \"https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=guitarboy000%40gmail%2ecom&item_name=Gloobus%2c%20A%20Quicklook%20for%20linux&no_shipping=0&no_note=1&tax=0&currency_code=EUR&lc=US&bn=PP%2dDonationsBF&chars\"" ;
     else if(label == _("About"))
         ui->create_about();
     else if(label == _("Quit"))
