diff -Naur signon-8.43.orig/src/extensions/extensions.pro signon-8.43/src/extensions/extensions.pro
--- signon-8.43.orig/src/extensions/extensions.pro	2012-09-18 14:59:58.000000000 +0200
+++ signon-8.43/src/extensions/extensions.pro	2012-10-17 20:02:52.846790974 +0200
@@ -1,7 +1,7 @@
 TEMPLATE = subdirs
 SUBDIRS =
 
-system(pkg-config --exists libcryptsetup) {
+CONFIG(cryptsetup) {
     SUBDIRS += cryptsetup
 }
 
