diff -ruN a/hack/lib/golang.sh b/hack/lib/golang.sh
--- a/hack/lib/golang.sh	2021-12-07 19:08:39.000000000 +0100
+++ b/hack/lib/golang.sh	2021-12-08 23:59:38.948699522 +0100
@@ -808,7 +808,7 @@
 
     # extract tags if any specified in GOFLAGS
     # shellcheck disable=SC2001
-    gotags="selinux,notest,$(echo "${GOFLAGS:-}" | sed -ne 's|.*-tags=\([^-]*\).*|\1|p')"
+    gotags="selinux notest $(echo "${GOFLAGS:-}" | sed -ne 's|.*-tags=\([^-]*\).*|\1|p')"
 
     local -a targets=()
     local arg
