diff -rupN cairo-dock-plugins-3.2.1.orig/Dbus/interfaces/bash/BashInstall.cmake.in cairo-dock-plugins-3.2.1/Dbus/interfaces/bash/BashInstall.cmake.in
--- cairo-dock-plugins-3.2.1.orig/Dbus/interfaces/bash/BashInstall.cmake.in	2013-07-11 17:09:45.165345967 +0200
+++ cairo-dock-plugins-3.2.1/Dbus/interfaces/bash/BashInstall.cmake.in	2013-07-11 17:10:22.238469121 +0200
@@ -2,7 +2,7 @@ IF ("@PYTHON2_FOUND@" STREQUAL "TRUE")
 	IF("@ROOT_PREFIX@" STREQUAL "")
 		EXEC_PROGRAM("@PYTHON2_EXECUTABLE@" "${CMAKE_CURRENT_BINARY_DIR}" ARGS
 						"setup.py" "install"
-						"--prefix=${CMAKE_INSTALL_PREFIX}"
+						"--prefix=$DESTDIR${CMAKE_INSTALL_PREFIX}"
 						"@DEBIAN_INSTALL_LAYOUT@")
 	ELSEIF(NOT "@ROOT_PREFIX@" STREQUAL "" AND "@python_no_deb_layout@" STREQUAL "")
 		EXEC_PROGRAM("@PYTHON2_EXECUTABLE@" "${CMAKE_CURRENT_BINARY_DIR}" ARGS
@@ -20,7 +20,7 @@ IF ("@PYTHON3_FOUND@" STREQUAL "TRUE")
 	IF("@ROOT_PREFIX@" STREQUAL "")
 		EXEC_PROGRAM("@PYTHON3_EXECUTABLE@" "${CMAKE_CURRENT_BINARY_DIR}" ARGS
 						"setup.py" "install"
-						"--prefix=${CMAKE_INSTALL_PREFIX}"
+						"--prefix=$DESTDIR${CMAKE_INSTALL_PREFIX}"
 						"@DEBIAN_INSTALL_LAYOUT@")
 	ELSEIF(NOT "@ROOT_PREFIX@" STREQUAL "" AND "@python_no_deb_layout@" STREQUAL "")
 		EXEC_PROGRAM("@PYTHON3_EXECUTABLE@" "${CMAKE_CURRENT_BINARY_DIR}" ARGS
diff -rupN cairo-dock-plugins-3.2.1.orig/Dbus/interfaces/python/PythonInstall.cmake.in cairo-dock-plugins-3.2.1/Dbus/interfaces/python/PythonInstall.cmake.in
--- cairo-dock-plugins-3.2.1.orig/Dbus/interfaces/python/PythonInstall.cmake.in	2013-07-11 17:09:45.162012653 +0200
+++ cairo-dock-plugins-3.2.1/Dbus/interfaces/python/PythonInstall.cmake.in	2013-07-11 17:10:56.581607933 +0200
@@ -2,7 +2,7 @@ if ("@PYTHON2_FOUND@" STREQUAL "TRUE")
 	IF("@ROOT_PREFIX@" STREQUAL "")
 		EXEC_PROGRAM("@PYTHON2_EXECUTABLE@" "${CMAKE_CURRENT_BINARY_DIR}" ARGS
 						"setup.py" "install"
-						"--prefix=${CMAKE_INSTALL_PREFIX}"
+						"--prefix=$DESTDIR${CMAKE_INSTALL_PREFIX}"
 						"@DEBIAN_INSTALL_LAYOUT@")
 	ELSEIF(NOT "@ROOT_PREFIX@" STREQUAL "" AND "@python_no_deb_layout@" STREQUAL "")
 		EXEC_PROGRAM("@PYTHON2_EXECUTABLE@" "${CMAKE_CURRENT_BINARY_DIR}" ARGS
@@ -20,7 +20,7 @@ if ("@PYTHON3_FOUND@" STREQUAL "TRUE")
 	IF("@ROOT_PREFIX@" STREQUAL "")
 		EXEC_PROGRAM("@PYTHON3_EXECUTABLE@" "${CMAKE_CURRENT_BINARY_DIR}" ARGS
 						"setup.py" "install"
-						"--prefix=${CMAKE_INSTALL_PREFIX}"
+						"--prefix=$DESTDIR${CMAKE_INSTALL_PREFIX}"
 						"@DEBIAN_INSTALL_LAYOUT@")
 	ELSEIF(NOT "@ROOT_PREFIX@" STREQUAL "" AND "@python_no_deb_layout@" STREQUAL "")
 		EXEC_PROGRAM("@PYTHON3_EXECUTABLE@" "${CMAKE_CURRENT_BINARY_DIR}" ARGS
@@ -32,4 +32,4 @@ if ("@PYTHON3_FOUND@" STREQUAL "TRUE")
 						"setup.py" "install"
 						"--root=@ROOT_PREFIX@")
 	ENDIF()
-ENDIF()
\ No newline at end of file
+ENDIF()
