diff -aur origsrc/ansible-lint-5.0.0/setup.cfg src/ansible-lint-5.0.0/setup.cfg
--- aaa/setup.cfg	2021-02-09 19:19:11.689518714 +0100
+++ bbb/setup.cfg	2021-02-09 19:19:30.538804812 +0100
@@ -54,18 +54,12 @@
   lint
 
 [options]
-use_scm_version = True
 python_requires = >=3.6
 package_dir =
   = src
 packages = find:
 zip_safe = False
 
-# These are required during `setup.py` run:
-setup_requires =
-  setuptools_scm>=1.15.0
-  setuptools_scm_git_archive>=1.0
-
 # These are required in actual runtime:
 install_requires =
   # ansible-lint does not list ansible as direct dependency in order to
diff -aur origsrc/ansible-lint-5.0.0/setup.py src/ansible-lint-5.0.0/setup.py
--- aaa/setup.py	2021-02-09 19:19:11.689518714 +0100
+++ bbb/setup.py	2021-02-09 19:19:42.271693765 +0100
@@ -9,5 +9,4 @@
 # https://github.com/jazzband/pip-tools/issues/1278
 setuptools.setup(
     use_scm_version={"local_scheme": "no-local-version"},
-    setup_requires=["setuptools_scm[toml]>=3.5.0"],
 )
