diff -aur libdlna-0.2.3/src/profiles.c libdlna-0.2.3.new/src/profiles.c
--- libdlna-0.2.3/src/profiles.c	2007-11-26 21:47:43.000000000 +0100
+++ libdlna-0.2.3.new/src/profiles.c	2011-10-31 16:06:34.000000000 +0100
@@ -205,13 +205,13 @@
   for (i = 0; i < ctx->nb_streams; i++)
   {
     if (audio_stream == -1 &&
-        ctx->streams[i]->codec->codec_type == CODEC_TYPE_AUDIO)
+        ctx->streams[i]->codec->codec_type == AVMEDIA_TYPE_AUDIO)
     {
       audio_stream = i;
       continue;
     }
     else if (video_stream == -1 &&
-             ctx->streams[i]->codec->codec_type == CODEC_TYPE_VIDEO)
+             ctx->streams[i]->codec->codec_type == AVMEDIA_TYPE_VIDEO)
     {
       video_stream = i;
       continue;
