diff -u -Nr libdesktop-agnostic-0.3.90/debian/control various-fixes/debian/control
--- libdesktop-agnostic-0.3.90/debian/control	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/debian/control	2010-10-02 04:42:20.187222000 +0200
@@ -6,6 +6,7 @@
 # for waf
                python,
 # "real" deps
+               intltool,
                valac (>= 0.7.10),
                gobject-introspection (>= 0.6.3),
                gobject-introspection-glib-2.0,
diff -u -Nr libdesktop-agnostic-0.3.90/debian/libdesktop-agnostic0.install various-fixes/debian/libdesktop-agnostic0.install
--- libdesktop-agnostic-0.3.90/debian/libdesktop-agnostic0.install	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/debian/libdesktop-agnostic0.install	2010-10-02 04:42:20.187222000 +0200
@@ -1,4 +1,5 @@
 usr/lib/desktop-agnostic/modules/libda-cfg-type-color.so
 usr/lib/desktop-agnostic/modules/libda-module-guesser.so
 usr/lib/*.so.0*
+usr/share/locale/
 usr/etc/xdg etc/
diff -u -Nr libdesktop-agnostic-0.3.90/docs/install.rst various-fixes/docs/install.rst
--- libdesktop-agnostic-0.3.90/docs/install.rst	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/docs/install.rst	2010-10-02 04:42:20.187222000 +0200
@@ -51,6 +51,7 @@
 * GObject Introspection 0.6.3 or later (requires the development files, to
   properly detect the correct version)
 * Vala 0.7.10
+* intltool
 
 Build/Runtime (*required*)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
diff -u -Nr libdesktop-agnostic-0.3.90/docs/known-issues.rst various-fixes/docs/known-issues.rst
--- libdesktop-agnostic-0.3.90/docs/known-issues.rst	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/docs/known-issues.rst	2010-10-02 04:42:20.187222000 +0200
@@ -19,7 +19,7 @@
     config-bridge.vala:164: warning: pointer targets in passing argument 2 of ‘g_object_class_list_properties’ differ in signedness
     desktop-entry-impl-glib.c:306: warning: pointer targets in passing argument 4 of ‘g_key_file_get_string_list’ differ in signedness
 
-.. [1] See `GNOME Bug #582092`_.
+.. [1] See `GNOME Bug #582092`_ (fixed in Vala 0.9.1).
 .. [2] See `GNOME Bug #529866`_ (fixed in Vala 0.7.6).
 .. [3] See `GNOME Bug #592108`_ (fixed in Vala 0.7.6).
 
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-bridge.vala various-fixes/libdesktop-agnostic/config-bridge.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-bridge.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/config-bridge.vala	2010-10-02 04:42:20.187222000 +0200
@@ -202,14 +202,13 @@
                                         string key, ParamSpec spec,
                                         NotifyFuncHandler func) throws GLib.Error
     {
-      unowned BindingNotifier notifier;
-      notifier = (BindingNotifier*) config.get_data ("lda-binding-notifier");
+      unowned BindingNotifier? notifier;
+      notifier = config.get_data ("lda-binding-notifier");
       if (notifier == null)
       {
         BindingNotifier new_notifier = new BindingNotifier (config);
         notifier = new_notifier;
-        config.set_data_full ("lda-binding-notifier", notifier.@ref (),
-                              Object.unref);
+        config.set_data ("lda-binding-notifier", notifier);
       }
 
       if (spec.value_type == typeof (bool) ||
@@ -353,8 +352,7 @@
       uint pos = -1;
       string binding_key;
 
-      unowned BindingListWrapper? obj_bindings = 
-        (BindingListWrapper*) obj.get_data ("lda-bindings");
+      unowned BindingListWrapper? obj_bindings = obj.get_data ("lda-bindings");
       binding_key = "%s/%s/%s".printf (config.instance_id, group, key);
       bindings_list = this.bindings.get_data (binding_key);
       bindings_to_remove = new SList<uint> ();
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-client.vala various-fixes/libdesktop-agnostic/config-client.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-client.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/config-client.vala	2010-10-02 04:42:20.187222000 +0200
@@ -101,13 +101,12 @@
     // constructors
     public Client (string schema_filename)
     {
-      this.schema_filename = schema_filename;
+      GLib.Object (schema_filename: schema_filename);
     }
     public Client.for_instance (string schema_filename,
                                 string instance_id) throws GLib.Error
     {
-      this.schema_filename = schema_filename;
-      this.instance_id = instance_id;
+      GLib.Object (schema_filename: schema_filename, instance_id: instance_id);
     }
     /**
      * Auto-determines whether an instance config object should be created.
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-impl-gconf.vala various-fixes/libdesktop-agnostic/config-impl-gconf.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-impl-gconf.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/config-impl-gconf.vala	2010-10-02 04:42:20.187222000 +0200
@@ -654,7 +654,6 @@
   }
 }
 
-[ModuleInit]
 public Type
 register_plugin ()
 {
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-impl-keyfile.vala various-fixes/libdesktop-agnostic/config-impl-keyfile.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-impl-keyfile.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/config-impl-keyfile.vala	2010-10-02 04:42:20.187222000 +0200
@@ -697,6 +697,10 @@
 
       if (value.n_values == 0)
       {
+        if (!this._data.has_group (group))
+        {
+          return;
+        }
         if (this._data.has_key (group, key))
         {
           // set_*_list() doesn't like NULL lists, so just unset the key.
@@ -764,7 +768,6 @@
     }
   }
 }
-[ModuleInit]
 public Type
 register_plugin ()
 {
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-impl-memory.vala various-fixes/libdesktop-agnostic/config-impl-memory.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-impl-memory.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/config-impl-memory.vala	2010-10-02 04:42:20.187222000 +0200
@@ -226,7 +226,6 @@
     }
   }
 }
-[ModuleInit]
 public Type
 register_plugin ()
 {
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-impl-null.vala various-fixes/libdesktop-agnostic/config-impl-null.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-impl-null.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/config-impl-null.vala	2010-10-02 04:42:20.187222000 +0200
@@ -121,7 +121,6 @@
     }
   }
 }
-[ModuleInit]
 public Type
 register_plugin ()
 {
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-schema.vala various-fixes/libdesktop-agnostic/config-schema.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-schema.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/config-schema.vala	2010-10-02 04:42:20.187222000 +0200
@@ -188,7 +188,8 @@
      */
     public Schema (string filename) throws GLib.Error
     {
-      this.filename = filename;
+      GLib.Object (filename: filename);
+
       unowned HashTable<string,Value?> backend_metadata_keys;
       this.options = Datalist<SchemaOption> ();
       this.keys = new HashTable<string,List<string>> (str_hash, str_equal);
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-type-color.vala various-fixes/libdesktop-agnostic/config-type-color.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/config-type-color.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/config-type-color.vala	2010-10-02 04:42:20.187222000 +0200
@@ -120,7 +120,6 @@
     dest_value = ct.deserialize ((string)src_value);
   }
 }
-[ModuleInit]
 public Type
 register_plugin ()
 {
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/desktop-entry-impl-gio.vala various-fixes/libdesktop-agnostic/desktop-entry-impl-gio.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/desktop-entry-impl-gio.vala	1970-01-01 01:00:00.000000000 +0100
+++ various-fixes/libdesktop-agnostic/desktop-entry-impl-gio.vala	2010-10-02 04:42:20.187222000 +0200
@@ -0,0 +1,394 @@
+/*
+ * Desktop Agnostic Library: Desktop Entry implementation using GLib.
+ *
+ * Copyright (C) 2010 Michal Hruby <michal.mhr@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * Author : Michal Hruby <michal.mhr@gmail.com>
+ */
+
+using DesktopAgnostic;
+
+namespace DesktopAgnostic.FDO
+{
+  private const string GROUP = "Desktop Entry";
+  public class DesktopEntryGio : DesktopEntry, Object
+  {
+    private KeyFile _keyfile = new KeyFile ();
+    private bool loaded = false;
+    private VFS.File _file = null;
+
+    public VFS.File? file
+    {
+      get
+      {
+        return this._file;
+      }
+      set construct
+      {
+        if (value != null)
+        {
+          if (this.loaded)
+          {
+            warning ("The desktop entry has already been initialized.");
+          }
+          else if (value.exists ())
+          {
+            string? path;
+
+            this._file = value;
+            path = value.path;
+            if (path == null)
+            {
+              string data;
+              size_t data_len;
+
+              this._file.load_contents (out data, out data_len);
+              this._keyfile.load_from_data (data, data_len,
+                                            KeyFileFlags.KEEP_TRANSLATIONS);
+            }
+            else
+            {
+              this._keyfile.load_from_file (path, KeyFileFlags.KEEP_TRANSLATIONS);
+            }
+            this.loaded = true;
+          }
+        }
+      }
+    }
+
+    public KeyFile keyfile
+    {
+      get
+      {
+        return this._keyfile;
+      }
+      set construct
+      {
+        if (value != null)
+        {
+          if (this.loaded)
+          {
+            warning ("The desktop entry has already been initialized.");
+          }
+          else
+          {
+            string data;
+            size_t length;
+
+            data = value.to_data (out length);
+            this._keyfile.load_from_data (data, length,
+                                          KeyFileFlags.KEEP_TRANSLATIONS);
+            this.loaded = true;
+          }
+        }
+      }
+    }
+
+    public string data
+    {
+      set construct
+      {
+        if (value != null && value != "")
+        {
+          if (this.loaded)
+          {
+            warning ("The desktop entry has already been initialized.");
+          }
+          else
+          {
+            this._keyfile.load_from_data (value, value.size (),
+                                          KeyFileFlags.KEEP_TRANSLATIONS);
+            this.loaded = true;
+          }
+        }
+      }
+    }
+
+    public DesktopEntryType entry_type
+    {
+      get
+      {
+        string type = this.get_string ("Type");
+        switch (type)
+        {
+          case "Application":
+            return DesktopEntryType.APPLICATION;
+          case "Link":
+            return DesktopEntryType.LINK;
+          case "Directory":
+            return DesktopEntryType.DIRECTORY;
+          default:
+            return DesktopEntryType.UNKNOWN;
+        }
+      }
+      set
+      {
+        this.set_string ("Type", desktop_entry_type_to_string (value));
+      }
+    }
+
+    public string name
+    {
+      owned get
+      {
+        return this.get_string ("Name");
+      }
+      set
+      {
+        this.set_string ("Name", value);
+      }
+    }
+
+    public string? icon
+    {
+      /**
+       * If a path is provided then return the given value. Otherwise,
+       * strip any extension (.xpm, .svg, .png).
+       */
+      owned get
+      {
+        string? icon_name = this.get_string ("Icon");
+
+        if (icon_name != null && Path.get_basename (icon_name) == icon_name)
+        {
+          icon_name = icon_name.split (".png", 2)[0];
+          icon_name = icon_name.split (".svg", 2)[0];
+          icon_name = icon_name.split (".xpm", 2)[0];
+        }
+
+        return icon_name;
+      }
+      set
+      {
+        if (value == null)
+        {
+          warning ("Cannot set a NULL value for 'Icon'.");
+        }
+        else
+        {
+          this.set_string ("Icon", value);
+        }
+      }
+    }
+
+    public bool
+    key_exists (string key)
+    {
+      return this._keyfile.has_group (GROUP) &&
+             this._keyfile.has_key (GROUP, key);
+    }
+
+    public bool
+    get_boolean (string key)
+    {
+      try
+      {
+        return this._keyfile.get_boolean (GROUP, key);
+      }
+      catch (KeyFileError err)
+      {
+        warning ("Error trying to retrieve '%s': %s", key, err.message);
+        return false;
+      }
+    }
+
+    public void
+    set_boolean (string key, bool value)
+    {
+      this._keyfile.set_boolean (GROUP, key, value);
+    }
+
+    public string?
+    get_string (string key)
+    {
+      try
+      {
+        return this._keyfile.get_string (GROUP, key);
+      }
+      catch (KeyFileError err)
+      {
+        warning ("Error trying to retrieve '%s': %s", key, err.message);
+        return null;
+      }
+    }
+
+    public void
+    set_string (string key, string value)
+    {
+      this._keyfile.set_string (GROUP, key, value);
+    }
+
+    public string?
+    get_localestring (string key, string? locale)
+    {
+      try
+      {
+        return this._keyfile.get_locale_string (GROUP, key, locale);
+      }
+      catch (KeyFileError err)
+      {
+        warning ("Error trying to retrieve '%s[%s]': %s", key, locale,
+                 err.message);
+        return null;
+      }
+    }
+
+    public void
+    set_localestring (string key, string locale, string value)
+    {
+      this._keyfile.set_locale_string (GROUP, key, locale, value);
+    }
+
+    [CCode (array_length = false, array_null_terminated = true)]
+    public string[]?
+    get_string_list (string key)
+    {
+      try
+      {
+        return this._keyfile.get_string_list (GROUP, key);
+      }
+      catch (KeyFileError err)
+      {
+        warning ("Error trying to retrieve '%s': %s", key, err.message);
+        return null;
+      }
+    }
+
+    public void
+    set_string_list (string key, [CCode (array_length = false, array_null_terminated = true)] string[] value)
+    {
+      this._keyfile.set_string_list (GROUP, key, value);
+    }
+
+    /**
+     * Based on EggDesktopFile's egg_desktop_file_can_launch().
+     */
+    public bool
+    exists ()
+    {
+      switch (this.entry_type)
+      {
+        case DesktopEntryType.APPLICATION:
+          if (this._keyfile.has_key (GROUP, "TryExec"))
+          {
+            if (Environment.find_program_in_path (this.get_string ("TryExec")) != null)
+            {
+              return true;
+            }
+          }
+          string? exec;
+          string[] argv = null;;
+          exec = this.get_string ("Exec");
+          if (exec == null || !Shell.parse_argv (exec, out argv))
+          {
+            return false;
+          }
+          return Environment.find_program_in_path (argv[0]) != null;
+        case DesktopEntryType.LINK:
+          if (this._keyfile.has_key (GROUP, "URL"))
+          {
+            string uri = this._keyfile.get_string (GROUP, "URL");
+            VFS.File file = VFS.file_new_for_uri (uri);
+            return file.exists ();
+          }
+          else
+          {
+            return false;
+          }
+        default:
+          return false;
+      }
+    }
+
+    /**
+     * Launch desktop entry.
+     * @return always zero.
+     */
+    public Pid
+    launch (DesktopEntryLaunchFlags flags,
+            SList<string>? documents) throws GLib.Error
+    {
+      List<unowned string> uris = new List<unowned string> ();
+      foreach (unowned string s in documents)
+      {
+        uris.append (s);
+      }
+
+      // interesting that GIO 2.26 supports only APPLICATION
+      switch (this.entry_type)
+      {
+        case DesktopEntryType.APPLICATION:
+          AppInfo info;
+          if (this._file != null)
+          {
+            info = new DesktopAppInfo.from_filename (this._file.path);
+          }
+          else
+          {
+            info = new DesktopAppInfo.from_keyfile (this._keyfile);
+          }
+
+          //var context = new AppLaunchContext ();
+          info.launch_uris (uris, null);
+
+          break;
+        case DesktopEntryType.LINK:
+          if (this._keyfile.has_key (GROUP, "URL"))
+          {
+            string uri = this._keyfile.get_string (GROUP, "URL");
+            AppInfo.launch_default_for_uri (uri, null);
+          }
+          else
+          {
+            throw new DesktopEntryError.NOT_LAUNCHABLE ("Invalid desktop entry.");
+          }
+          break;
+        default:
+          throw new DesktopEntryError.NOT_LAUNCHABLE ("Unknown desktop entry type.");
+      }
+
+      return (Pid) 0;
+    }
+
+    public void
+    save (VFS.File? new_file) throws GLib.Error
+    {
+      VFS.File? file = null;
+      if (new_file != null)
+      {
+        file = new_file;
+      }
+      else if (this._file != null)
+      {
+        file = this._file;
+      }
+      else
+      {
+        throw new DesktopEntryError.INVALID_FILE ("No filename specified.");
+      }
+      file.replace_contents (this._keyfile.to_data ());
+    }
+  }
+}
+
+public Type
+register_plugin ()
+{
+  return typeof (DesktopAgnostic.FDO.DesktopEntryGio);
+}
+
+// vim: set ts=2 sts=2 sw=2 et ai cindent :
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/desktop-entry-impl-glib.vala various-fixes/libdesktop-agnostic/desktop-entry-impl-glib.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/desktop-entry-impl-glib.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/desktop-entry-impl-glib.vala	2010-10-02 04:42:20.187222000 +0200
@@ -531,6 +531,59 @@
         throw new DesktopEntryError.NOT_LAUNCHABLE ("Could not parse Exec key.");
       }
 
+      if (this._keyfile.has_key (GROUP, "Terminal") &&
+          this.get_boolean ("Terminal"))
+      {
+        string[] term_argv = new string[argv.length + 2];
+
+        // based on the code for GDesktopAppInfo
+        string? check = null;
+        check = Environment.find_program_in_path ("gnome-terminal");
+        if (check != null)
+        {
+          term_argv[0] = check;
+          term_argv[1] = "-x";
+        }
+        else
+        {
+          if (check == null)
+          {
+            check = Environment.find_program_in_path ("nxterm");
+          }
+          if (check == null)
+          {
+            check = Environment.find_program_in_path ("color-xterm");
+          }
+          if (check == null)
+          {
+            check = Environment.find_program_in_path ("rxvt");
+          }
+          if (check == null)
+          {
+            check = Environment.find_program_in_path ("xterm");
+          }
+          if (check == null)
+          {
+            check = Environment.find_program_in_path ("dtterm");
+          }
+          if (check == null)
+          {
+            check = "xterm";
+            warning ("couldn't find a terminal, falling back to xterm");
+          }
+
+          term_argv[0] = check;
+          term_argv[1] = "-e";
+        }
+
+        for (int i = 0; i < argv.length; i++)
+        {
+          term_argv[i+2] = argv[i];
+        }
+
+        argv = (owned) term_argv;
+      }
+
       Process.spawn_async_with_pipes (working_dir, argv, null, flags, null, out pid);
       return pid;
     }
@@ -614,7 +667,6 @@
   }
 }
 
-[ModuleInit]
 public Type
 register_plugin ()
 {
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/desktop-entry-impl-gnome.vala various-fixes/libdesktop-agnostic/desktop-entry-impl-gnome.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/desktop-entry-impl-gnome.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/desktop-entry-impl-gnome.vala	2010-10-02 04:42:20.187222000 +0200
@@ -305,7 +305,6 @@
   }
 }
 
-[ModuleInit]
 public Type
 register_plugin ()
 {
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/ui-icon-chooser-dialog.vala various-fixes/libdesktop-agnostic/ui-icon-chooser-dialog.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/ui-icon-chooser-dialog.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/ui-icon-chooser-dialog.vala	2010-10-02 04:42:20.187222000 +0200
@@ -35,13 +35,40 @@
     FILE
   }
 
+  private class LazyPixbufRenderer : CellRendererPixbuf
+  {
+    public bool item_ready { get; set; default = false; }
+
+    public signal void prepare_pixbuf (TreePath path);
+
+    public override void render (Gdk.Window window,
+                                 Gtk.Widget widget,
+                                 Gdk.Rectangle background_area,
+                                 Gdk.Rectangle cell_area,
+                                 Gdk.Rectangle expose_area,
+                                 Gtk.CellRendererState flags)
+    {
+      if (!item_ready)
+      {
+        int x, y;
+        var view = widget as Gtk.IconView;
+        x = cell_area.x + cell_area.width / 2;
+        y = cell_area.y + cell_area.height / 2;
+        var path = view.get_path_at_pos (x, y);
+        prepare_pixbuf (path);
+      }
+      base.render (window, widget,
+                   background_area, cell_area, expose_area, flags);
+    }
+  }
+
   public class IconChooserDialog : Dialog
   {
     private RadioButton _file;
     private RadioButton _themed;
     private FileChooserButton _directory;
     private ComboBox _themed_context;
-    private IconView _file_viewer;
+    private IconView? _file_viewer = null;
     private IconView? _themed_viewer = null;
     private unowned IconView _viewer;
     public string selected_icon { get; private set; default = null; }
@@ -53,16 +80,26 @@
       PIXBUF,
       NAME,
       DATA,
+      PIXBUF_READY,
       COUNT
     }
 
     public signal void icon_selected ();
 
+    private static Gdk.Pixbuf NO_ICON;
+
+    static construct
+    {
+      var flags = IconLookupFlags.FORCE_SIZE | IconLookupFlags.GENERIC_FALLBACK;
+      NO_ICON = IconTheme.get_default ().load_icon ("gtk-file", 48, flags);
+    }
+
     construct
     {
       this.response.connect (this.on_response);
       this.title = _ ("Select Icon");
       this.icon_name = STOCK_FIND;
+      this.set_default_size (375, 375);
       this.create_ui ();
     }
 
@@ -72,27 +109,17 @@
       HBox choices;
 
       choices = new HBox (false, 5);
-      this._file = new RadioButton.with_mnemonic (null, _ ("From File"));
-      choices.add (this._file);
-      this._themed = new RadioButton.with_mnemonic_from_widget (this._file,
-                                                                _ ("From Theme"));
-      this._themed.active = false;
-      this._themed.toggled.connect (this.on_icon_type_toggled);
+      this._themed = new RadioButton.with_mnemonic (null, _ ("From Theme"));
       choices.add (this._themed);
+      this._file = new RadioButton.with_mnemonic_from_widget (this._themed,
+                                                              _ ("From File"));
+      this._themed.active = true;
+      this._themed.toggled.connect (this.on_icon_type_toggled);
+      choices.add (this._file);
       this.vbox.pack_start (choices, false, false, 5);
       choices.show_all ();
 
-      this._directory = new FileChooserButton (_ ("Select icon folder"),
-                                               FileChooserAction.SELECT_FOLDER);
-      this._directory.current_folder_changed.connect (this.on_folder_changed);
-      this.vbox.pack_start (this._directory, false, false, 5);
-      this._directory.show ();
-
-      this.add_icon_viewer (ref this._file_viewer, false);
-      this._file_viewer.parent.show_all ();
-      this._viewer = this._file_viewer;
-
-      this.on_folder_changed (this._directory);
+      this.on_icon_type_toggled ();
 
       this.add_buttons (STOCK_CANCEL, ResponseType.CANCEL,
                         STOCK_OK, ResponseType.OK);
@@ -115,30 +142,64 @@
     create_icon_viewer (bool themed)
     {
       IconView viewer;
-      CellRendererPixbuf cell_pixbuf;
+      LazyPixbufRenderer cell_pixbuf;
       CellRendererText cell_text;
 
       viewer = new IconView.with_model (this.create_model ());
-      viewer.item_width = 72;
-      viewer.columns = 4;
-      viewer.column_spacing = 5;
-      viewer.set_size_request (325, 300);
-      viewer.tooltip_column = Column.DATA;
-      cell_pixbuf = new CellRendererPixbuf ();
+      // without this the IconView is not shrinkable after expanding it
+      viewer.set_size_request (108, -1);
+      viewer.set_item_width (108);
+      viewer.set_column_spacing (5);
+      viewer.set_tooltip_column (Column.DATA);
+
+      cell_pixbuf = new LazyPixbufRenderer ();
       cell_pixbuf.xalign = 0.5f;
       cell_pixbuf.yalign = 0.5f;
       cell_pixbuf.width = 48;
+
       viewer.pack_start (cell_pixbuf, false);
       viewer.add_attribute (cell_pixbuf, "pixbuf", Column.PIXBUF);
+      viewer.add_attribute (cell_pixbuf, "item-ready", Column.PIXBUF_READY);
+
+      cell_pixbuf.prepare_pixbuf.connect ((p) =>
+      {
+        TreeIter iter;
+        Value val;
+        var store = this._viewer.model as ListStore;
+        store.get_iter (out iter, p);
+        store.get_value (iter, Column.DATA, out val);
+
+        string icon_name = val.get_string ();
+        IconTheme icon_theme = IconTheme.get_default ();
+        var info = icon_theme.lookup_icon (icon_name, 48, 0);
+        string? name = info.get_display_name ();
+        if (name == null)
+        {
+          name = icon_name.replace ("-", " ");
+        }
+        try
+        {
+          var pixbuf = info.load_icon ();
+          store.set (iter, Column.NAME, name, Column.PIXBUF, pixbuf,
+                     Column.PIXBUF_READY, true, -1);
+        }
+        catch (Error err)
+        {
+          warning ("Could not load %s: %s", icon_name, err.message);
+          store.set (iter, Column.NAME, name, Column.PIXBUF_READY, true, -1);
+        }
+      });
+
       cell_text = new CellRendererText ();
       cell_text.xalign = 0.5f;
-      cell_text.yalign = 0;
+      cell_text.yalign = 0.0f;
       cell_text.wrap_mode = Pango.WrapMode.WORD;
-      cell_text.wrap_width = 72;
-      cell_text.width = 72;
-      cell_text.ellipsize = Pango.EllipsizeMode.START;
+      int wrap_width = viewer.item_width - viewer.item_padding * 2;
+      cell_text.wrap_width = wrap_width;
+      cell_text.width = wrap_width;
+      cell_text.ellipsize = Pango.EllipsizeMode.MIDDLE;
       viewer.pack_start (cell_text, true);
-      viewer.add_attribute (cell_text, "text", 1);
+      viewer.add_attribute (cell_text, "text", Column.NAME);
       viewer.selection_mode = SelectionMode.SINGLE;
 
       return viewer;
@@ -148,20 +209,24 @@
     create_model ()
     {
       // icon, name, data
-      return new ListStore (Column.COUNT, typeof (Gdk.Pixbuf), typeof (string),
-                            typeof (string));
+      return new ListStore (Column.COUNT,
+                            typeof (Gdk.Pixbuf),
+                            typeof (string),
+                            typeof (string),
+                            typeof (bool));
     }
 
     private void
-    on_icon_type_toggled (ToggleButton themed)
+    on_icon_type_toggled ()
     {
       if (this._themed.active)
       {
         if (this._themed_viewer == null)
         {
           unowned IconTheme icon_theme;
-          unowned List<string> context_list;
+          List<string> context_list;
 
+          // "From Theme" widgets -> context combobox + icon view
           this._themed_context = new ComboBox.text ();
           this._themed_context.changed.connect (this.on_icon_context_changed);
           this.vbox.pack_start (this._themed_context, false, false, 5);
@@ -171,21 +236,52 @@
           icon_theme = IconTheme.get_default ();
           context_list = icon_theme.list_contexts ();
           context_list.sort ((CompareFunc)strcmp);
+
+          int active_index = 0;
+          int cur_index = 0;
           foreach (unowned string context in context_list)
           {
             this._themed_context.append_text (context);
+            // try to make "Applications" context active by default
+            if (context == "Applications")
+            {
+              active_index = cur_index;
+            }
+            cur_index++;
           }
+          this._themed_context.set_active (active_index);
+        }
+
+        if (this._file_viewer != null)
+        {
+          this._file_viewer.parent.hide ();
+          this._directory.hide ();
         }
-        this._file_viewer.parent.hide ();
-        this._directory.hide ();
         this._themed_viewer.parent.show ();
         this._themed_context.show ();
         this._viewer = this._themed_viewer;
       }
       else
       {
-        this._themed_viewer.parent.hide ();
-        this._themed_context.hide ();
+        if (this._file_viewer == null)
+        {
+          // "From File" widgets -> directory chooser + icon view
+          this._directory = new FileChooserButton (_ ("Select icon folder"),
+                                                   FileChooserAction.SELECT_FOLDER);
+          this._directory.current_folder_changed.connect (this.on_folder_changed);
+          this.vbox.pack_start (this._directory, false, false, 5);
+          this._directory.show ();
+
+          this.add_icon_viewer (ref this._file_viewer, false);
+
+          this.on_folder_changed (this._directory);
+        }
+
+        if (this._themed_viewer != null)
+        {
+          this._themed_viewer.parent.hide ();
+          this._themed_context.hide ();
+        }
         this._file_viewer.parent.show ();
         this._directory.show ();
         this._viewer = this._file_viewer;
@@ -225,7 +321,9 @@
             pixbuf = new Gdk.Pixbuf.from_file_at_scale (path, 48, -1, true);
 
             model.append (out iter);
-            model.set (iter, Column.PIXBUF, pixbuf,
+            model.set (iter,
+                       Column.PIXBUF, pixbuf,
+                       Column.PIXBUF_READY, true,
                        Column.NAME, Path.get_basename (path),
                        Column.DATA, path);
           }
@@ -246,7 +344,7 @@
     {
       unowned ListStore model;
       unowned IconTheme icon_theme;
-      unowned List<string> icon_list;
+      List<string> icon_list;
 
       model = this._themed_viewer.model as ListStore;
       model.clear ();
@@ -256,29 +354,14 @@
       icon_list.sort ((CompareFunc)strcmp);
       foreach (unowned string icon_name in icon_list)
       {
-        try
-        {
-          IconInfo info;
-          Gdk.Pixbuf pixbuf;
-          string? name;
-          TreeIter iter;
-
-          info = icon_theme.lookup_icon (icon_name, 48, 0);
-          pixbuf = info.load_icon ();
-          name = info.get_display_name ();
-          if (name == null)
-          {
-            name = icon_name.replace ("-", " ");
-          }
-          model.append (out iter);
-          model.set (iter, Column.PIXBUF, pixbuf,
-                     Column.NAME, name,
-                     Column.DATA, icon_name);
-        }
-        catch (Error err)
-        {
-          warning ("Could not load %s: %s", icon_name, err.message);
-        }
+        TreeIter iter;
+
+        model.append (out iter);
+        model.set (iter,
+                   Column.PIXBUF, NO_ICON,
+                   Column.PIXBUF_READY, false,
+                   Column.NAME, icon_name,
+                   Column.DATA, icon_name);
       }
     }
 
@@ -297,7 +380,7 @@
 
           msg = _ ("Please select an icon.");
           dialog = new MessageDialog (this, DialogFlags.MODAL, MessageType.ERROR,
-                                      ButtonsType.OK, msg);
+                                      ButtonsType.OK, "%s", msg);
           dialog.title = _ ("Error");
           dialog.run ();
           dialog.destroy ();
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/ui-launcher-editor-dialog.vala various-fixes/libdesktop-agnostic/ui-launcher-editor-dialog.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/ui-launcher-editor-dialog.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/ui-launcher-editor-dialog.vala	2010-10-02 04:42:20.187222000 +0200
@@ -70,20 +70,9 @@
 
     public LauncherEditorDialog (VFS.File file, VFS.File? output, bool standalone)
     {
-      WindowType type;
-
-      if (standalone)
-      {
-        type = WindowType.TOPLEVEL;
-      }
-      else
-      {
-        type = WindowType.POPUP;
-      }
-      this.type = type;
+      GLib.Object (type: standalone ? WindowType.TOPLEVEL : WindowType.POPUP,
+                   file: file, output: output);
       this._standalone = standalone;
-      this.file = file;
-      this.output = output;
     }
 
     private override void
@@ -210,6 +199,14 @@
       advanced.add (advanced_vbox);
       table.attach_fill (advanced, 0, 3, 3, 4);
 
+      List<unowned Widget> focus_chain_list = new List<unowned Widget> ();
+      focus_chain_list.append (this._name);
+      focus_chain_list.append (this._desc);
+      focus_chain_list.append (exec_hbox);
+      focus_chain_list.append (this._icon);
+      focus_chain_list.append (advanced);
+      table.set_focus_chain (focus_chain_list);
+
       this.vbox.add (table);
     }
 
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-bookmarks-gtk.vala various-fixes/libdesktop-agnostic/vfs-bookmarks-gtk.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-bookmarks-gtk.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/vfs-bookmarks-gtk.vala	2010-10-02 04:42:20.187222000 +0200
@@ -77,7 +77,7 @@
      */
     public GtkBookmarks (File? file = null, bool monitor = true)
     {
-      this.file = file;
+      GLib.Object (file: file);
       if (this._file.exists ())
       {
         this.parse ();
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-impl-gio.vala various-fixes/libdesktop-agnostic/vfs-impl-gio.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-impl-gio.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/vfs-impl-gio.vala	2010-10-02 04:42:20.187222000 +0200
@@ -89,7 +89,6 @@
     }
   }
 }
-[ModuleInit]
 public Type
 register_plugin ()
 {
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-impl-gnome-vfs.vala various-fixes/libdesktop-agnostic/vfs-impl-gnome-vfs.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-impl-gnome-vfs.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/vfs-impl-gnome-vfs.vala	2010-10-02 04:42:20.187222000 +0200
@@ -92,7 +92,6 @@
     }
   }
 }
-[ModuleInit]
 public Type
 register_plugin ()
 {
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-impl-thunar-vfs.vala various-fixes/libdesktop-agnostic/vfs-impl-thunar-vfs.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-impl-thunar-vfs.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/vfs-impl-thunar-vfs.vala	2010-10-02 04:42:20.187222000 +0200
@@ -92,7 +92,6 @@
     }
   }
 }
-[ModuleInit]
 public Type
 register_plugin ()
 {
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-trash-impl-gio.vala various-fixes/libdesktop-agnostic/vfs-trash-impl-gio.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-trash-impl-gio.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/vfs-trash-impl-gio.vala	2010-10-02 04:42:20.187222000 +0200
@@ -77,7 +77,7 @@
 
       try
       {
-        file_info = dir.query_info_finish (res);
+        file_info = dir.query_info_async.end (res);
         this._file_count = file_info.get_attribute_uint32 (FILE_ATTRIBUTE_TRASH_ITEM_COUNT);
         this.file_count_changed ();
       }
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-trash-impl-thunar-vfs.vala various-fixes/libdesktop-agnostic/vfs-trash-impl-thunar-vfs.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-trash-impl-thunar-vfs.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/vfs-trash-impl-thunar-vfs.vala	2010-10-02 04:42:20.187222000 +0200
@@ -53,7 +53,7 @@
     {
       Monitor monitor;
       this.trash = ThunarVfs.Path.get_for_trash ();
-      this.dbus = Bus.get (BusType.SESSION);
+      this.dbus = DBus.Bus.get (DBus.BusType.SESSION);
       this.xfce_trash =
         (Xfce.Trash)this.dbus.get_object ("org.xfce.Thunar",
                                           "/org/xfce/FileManager");
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-volume-impl-gio.vala various-fixes/libdesktop-agnostic/vfs-volume-impl-gio.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-volume-impl-gio.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/vfs-volume-impl-gio.vala	2010-10-02 04:42:20.187222000 +0200
@@ -121,7 +121,7 @@
       bool result = false;
       try
       {
-        result = this.vol.mount_finish (this.async_result);
+        result = this.vol.mount.end (this.async_result);
       }
       catch (GLib.Error err)
       {
@@ -156,7 +156,7 @@
       bool result = false;
       try
       {
-        result = this.vol.get_mount ().unmount_finish (this.async_result);
+        result = this.vol.get_mount ().unmount.end (this.async_result);
       }
       catch (GLib.Error err)
       {
@@ -192,7 +192,7 @@
       bool result = false;
       try
       {
-        result = this.vol.eject_finish (this.async_result);
+        result = this.vol.eject.end (this.async_result);
       }
       catch (GLib.Error err)
       {
@@ -211,7 +211,7 @@
       this.monitor = GLib.VolumeMonitor.get ();
       this._volumes = new HashTable<GLib.Volume,VFS.Volume> (direct_hash,
                                                              direct_equal);
-      unowned List<GLib.Volume> vols = this.monitor.get_volumes ();
+      List<GLib.Volume> vols = this.monitor.get_volumes ();
       foreach (unowned GLib.Volume gvol in vols)
       {
         VFS.Volume vol = this.create_volume (gvol);
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-volume-impl-gnome-vfs.vala various-fixes/libdesktop-agnostic/vfs-volume-impl-gnome-vfs.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-volume-impl-gnome-vfs.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/vfs-volume-impl-gnome-vfs.vala	2010-10-02 04:42:20.187222000 +0200
@@ -190,8 +190,7 @@
       this.monitor = GnomeVFS.get_volume_monitor ();
       this._volumes = new HashTable<GnomeVFS.Drive,VFS.Volume> (direct_hash,
                                                                 direct_equal);
-      unowned List<GnomeVFS.Drive> drives =
-        this.monitor.get_connected_drives ();
+      List<GnomeVFS.Drive> drives = this.monitor.get_connected_drives ();
       foreach (unowned GnomeVFS.Drive drive in drives)
       {
         VFS.Volume vol = this.create_volume (drive);
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-volume-impl-thunar-vfs.vala various-fixes/libdesktop-agnostic/vfs-volume-impl-thunar-vfs.vala
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/vfs-volume-impl-thunar-vfs.vala	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/vfs-volume-impl-thunar-vfs.vala	2010-10-02 04:42:20.187222000 +0200
@@ -63,7 +63,7 @@
     }
     public VolumeThunarVFS.for_implementation (ThunarVfs.Volume impl)
     {
-      this.implementation = impl;
+      GLib.Object (implementation: impl);
     }
     public bool
     is_mounted ()
diff -u -Nr libdesktop-agnostic-0.3.90/libdesktop-agnostic/wscript various-fixes/libdesktop-agnostic/wscript
--- libdesktop-agnostic-0.3.90/libdesktop-agnostic/wscript	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/libdesktop-agnostic/wscript	2010-10-02 04:42:20.187222000 +0200
@@ -34,6 +34,10 @@
         }
     },
     'fdo': {
+        'gio': {
+            'uselib': 'GIO_UNIX',
+            'packages': 'gio-unix-2.0',
+        },
         'gnome': {
             'uselib': 'GNOME_DESKTOP',
             'packages': 'gnome-desktop-2.0',
@@ -113,7 +117,7 @@
         ])
     lib.packages = 'gdk-2.0 gmodule-2.0'
     lib.target = 'desktop-agnostic'
-    lib.gir = 'DesktopAgnostic-1.0'
+    if bld.env['INTROSPECTION']: lib.gir = 'DesktopAgnostic-1.0'
     lib.uselib = 'M GDK GMODULE'
     lib.packages_private = 'build'
     lib.includes = '..'
@@ -133,7 +137,7 @@
     vfs.packages = 'desktop-agnostic posix'
     vfs.target = 'desktop-agnostic-vfs'
     vfs.header = 'vfs'
-    vfs.gir = 'DesktopAgnosticVFS-1.0'
+    if bld.env['INTROSPECTION']: vfs.gir = 'DesktopAgnosticVFS-1.0'
     vfs.uselib_local = 'desktop-agnostic'
     vfs.packages_private = 'posix-glob'
     vfs.includes = '..'
@@ -152,7 +156,7 @@
     cfg.packages = 'desktop-agnostic-vfs'
     cfg.target = 'desktop-agnostic-cfg'
     cfg.header = 'config'
-    cfg.gir = 'DesktopAgnosticConfig-1.0'
+    if bld.env['INTROSPECTION']: cfg.gir = 'DesktopAgnosticConfig-1.0'
     cfg.uselib_local = 'desktop-agnostic-vfs'
     cfg.packages_private = 'hashtable-gtype-key'
     cfg.includes = '..'
@@ -166,7 +170,7 @@
     fdo.packages = 'desktop-agnostic-vfs'
     fdo.target = 'desktop-agnostic-fdo'
     fdo.header = 'fdo'
-    fdo.gir = 'DesktopAgnosticFDO-1.0'
+    if bld.env['INTROSPECTION']: fdo.gir = 'DesktopAgnosticFDO-1.0'
     fdo.uselib_local = 'desktop-agnostic-vfs'
     fdo.vapi_dirs = '.'
     fdo.includes = '..'
@@ -183,7 +187,7 @@
     ui.packages_private = 'build'
     ui.target = 'desktop-agnostic-ui'
     ui.header = 'ui'
-    ui.gir = 'DesktopAgnosticUI-1.0'
+    if bld.env['INTROSPECTION']: ui.gir = 'DesktopAgnosticUI-1.0'
     ui.uselib = 'GTK'
     ui.uselib_local = 'desktop-agnostic-fdo'
     ui.vapi_dirs = '. ../vapi'
diff -u -Nr libdesktop-agnostic-0.3.90/po/en_US@piglatin.po various-fixes/po/en_US@piglatin.po
--- libdesktop-agnostic-0.3.90/po/en_US@piglatin.po	1970-01-01 01:00:00.000000000 +0100
+++ various-fixes/po/en_US@piglatin.po	2010-10-02 04:42:20.187222000 +0200
@@ -0,0 +1,103 @@
+# Translatable strings for libdesktop-agnostic.
+# Copyright (C) 2009, 2010 Mark Lee
+# This file is distributed under the BSD license.
+#
+# Translators:
+# Mark Lee <libdesktop-agnostic@lazymalevolence.com>, 2010.
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: libdesktop-agnostic-0.4\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2010-04-18 11:20-0700\n"
+"PO-Revision-Date: 2010-04-18 12:46-0800\n"
+"Last-Translator: Mark Lee <libdesktop-agnostic@lazymalevolence.com>\n"
+"Language-Team: Pig Latin <LL@li.org>\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=utf-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"X-Poedit-Language: English\n"
+"X-Poedit-Country: UNITED STATES\n"
+"X-Poedit-SourceCharset: utf-8\n"
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:64
+msgid "Select Icon"
+msgstr "Elect-say Icon-way"
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:75
+msgid "From File"
+msgstr "Om-fray Ile-fay"
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:78
+msgid "From Theme"
+msgstr "Om-fray Eme-thay"
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:85
+msgid "Select icon folder"
+msgstr "Elect-say icon-way older-fay"
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:298
+msgid "Please select an icon."
+msgstr "Ease-play elect-say an icon-way."
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:301
+msgid "Error"
+msgstr "Error-way"
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:92
+msgid "Desktop Entry Editor"
+msgstr "Esktop-day Entry-way Editor-way"
+
+#. Name
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:149
+msgid "_Name:"
+msgstr "Ame-_nay:"
+
+#. Description
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:161
+msgid "_Description:"
+msgstr "Escription-_day:"
+
+#. Exec
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:173
+msgid "_Command:"
+msgstr "Ommand-_cay:"
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:184
+msgid "_Browse..."
+msgstr "Owse-_bray..."
+
+#. Advanced options
+#. TODO look into ResizeMode so that the window shrinks when the expander
+#. is un-expanded.
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:194
+msgid "_Advanced"
+msgstr "_Advanced-way"
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:196
+msgid "Run in _terminal"
+msgstr "Un-ray in erminal-_tay"
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:203
+msgid "Use _startup notification"
+msgstr "Use-way artup-_stay otification-nay"
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:249
+msgid "Locate Command"
+msgstr "Ocate-lay Ommand-cay"
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:283
+msgid "Save As"
+msgstr "Ave-say As"
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:341
+#, c-format
+msgid ""
+"An error occurred while trying to save the desktop entry:\n"
+"\n"
+"%s"
+msgstr ""
+"An error-way occurred-way ile-whay ying-tray to ave-say the esktop-day entry-way:\n"
+"\n"
+"%s"
+
diff -u -Nr libdesktop-agnostic-0.3.90/po/libdesktop-agnostic.pot various-fixes/po/libdesktop-agnostic.pot
--- libdesktop-agnostic-0.3.90/po/libdesktop-agnostic.pot	1970-01-01 01:00:00.000000000 +0100
+++ various-fixes/po/libdesktop-agnostic.pot	2010-10-02 04:42:20.187222000 +0200
@@ -0,0 +1,96 @@
+# Translatable strings for libdesktop-agnostic.
+# Copyright (C) 2009, 2010 Mark Lee
+# This file is distributed under the BSD license.
+#
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: libdesktop-agnostic-0.4\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2010-04-18 11:20-0700\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=utf-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:64
+msgid "Select Icon"
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:75
+msgid "From File"
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:78
+msgid "From Theme"
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:85
+msgid "Select icon folder"
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:298
+msgid "Please select an icon."
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-icon-chooser-dialog.vala:301
+msgid "Error"
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:92
+msgid "Desktop Entry Editor"
+msgstr ""
+
+#. Name
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:149
+msgid "_Name:"
+msgstr ""
+
+#. Description
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:161
+msgid "_Description:"
+msgstr ""
+
+#. Exec
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:173
+msgid "_Command:"
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:184
+msgid "_Browse..."
+msgstr ""
+
+#. Advanced options
+#. TODO look into ResizeMode so that the window shrinks when the expander
+#. is un-expanded.
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:194
+msgid "_Advanced"
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:196
+msgid "Run in _terminal"
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:203
+msgid "Use _startup notification"
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:249
+msgid "Locate Command"
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:283
+msgid "Save As"
+msgstr ""
+
+#: ../libdesktop-agnostic/ui-launcher-editor-dialog.vala:341
+#, c-format
+msgid ""
+"An error occurred while trying to save the desktop entry:\n"
+"\n"
+"%s"
+msgstr ""
diff -u -Nr libdesktop-agnostic-0.3.90/po/LINGUAS various-fixes/po/LINGUAS
--- libdesktop-agnostic-0.3.90/po/LINGUAS	1970-01-01 01:00:00.000000000 +0100
+++ various-fixes/po/LINGUAS	2010-10-02 04:42:20.187222000 +0200
@@ -0,0 +1,2 @@
+# please keep this list sorted alphabetically
+en_US@piglatin
diff -u -Nr libdesktop-agnostic-0.3.90/po/POTFILES.in various-fixes/po/POTFILES.in
--- libdesktop-agnostic-0.3.90/po/POTFILES.in	1970-01-01 01:00:00.000000000 +0100
+++ various-fixes/po/POTFILES.in	2010-10-02 04:42:20.187222000 +0200
@@ -0,0 +1,5 @@
+# List of source files containing translatable strings.
+# Please keep this file sorted alphabetically.
+[encoding: UTF-8]
+libdesktop-agnostic/ui-icon-chooser-dialog.vala
+libdesktop-agnostic/ui-launcher-editor-dialog.vala
diff -u -Nr libdesktop-agnostic-0.3.90/po/wscript various-fixes/po/wscript
--- libdesktop-agnostic-0.3.90/po/wscript	1970-01-01 01:00:00.000000000 +0100
+++ various-fixes/po/wscript	2010-10-02 04:42:20.187222000 +0200
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+# vim: set ft=python
+
+
+def build(bld):
+    bld.new_task_gen(features='intltool_po', appname='libdesktop-agnostic')
diff -u -Nr libdesktop-agnostic-0.3.90/python/desktopagnostic.override various-fixes/python/desktopagnostic.override
--- libdesktop-agnostic-0.3.90/python/desktopagnostic.override	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/python/desktopagnostic.override	2010-10-02 04:42:20.187222000 +0200
@@ -98,19 +98,23 @@
 _wrap_desktop_agnostic_color_from_values (PyObject *self, PyObject *args,
                                           PyObject *kwargs)
 {
-    static char *kwlist[] = { "red", "green", "blue", "alpha", NULL };
-    int red, green, blue, alpha;
-    DesktopAgnosticColor *ret;
-
-    if (!PyArg_ParseTupleAndKeywords (args, kwargs,
-                                      "iiii:color_new_from_values", kwlist,
-                                      &red, &green, &blue, &alpha))
-        return NULL;
+  static char *kwlist[] = { "red", "green", "blue", "alpha", NULL };
+  int red, green, blue, alpha;
+  DesktopAgnosticColor *ret;
+  PyObject *py_ret;
+
+  if (!PyArg_ParseTupleAndKeywords (args, kwargs,
+                                    "iiii:color_new_from_values", kwlist,
+                                    &red, &green, &blue, &alpha))
+    return NULL;
 
-    ret = desktop_agnostic_color_new_from_values (red, green, blue, alpha);
+  ret = desktop_agnostic_color_new_from_values (red, green, blue, alpha);
 
-    /* pygobject_new handles NULL checking */
-    return pygobject_new ((GObject *)ret);
+  /* pygobject_new handles NULL checking */
+  py_ret = pygobject_new ((GObject *)ret);
+  if (ret != NULL)
+      g_object_unref (ret);
+  return py_ret;
 }
 %%
 define DesktopAgnosticColor.from_string kwargs staticmethod
@@ -122,6 +126,7 @@
   char *spec;
   GError *error = NULL;
   DesktopAgnosticColor *ret;
+  PyObject *py_ret;
 
   if (!PyArg_ParseTupleAndKeywords (args, kwargs, "s:color_new_from_string",
                                     kwlist, &spec))
@@ -137,7 +142,10 @@
   }
 
   /* pygobject_new handles NULL checking */
-  return pygobject_new ((GObject *)ret);
+  py_ret = pygobject_new ((GObject *)ret);
+  if (ret != NULL)
+    g_object_unref (ret);
+  return py_ret;
 }
 %%
 override-slot DesktopAgnosticColor.tp_str
diff -u -Nr libdesktop-agnostic-0.3.90/python/fdo.override various-fixes/python/fdo.override
--- libdesktop-agnostic-0.3.90/python/fdo.override	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/python/fdo.override	2010-10-02 04:42:20.187222000 +0200
@@ -197,6 +197,7 @@
 {
   DesktopAgnosticFDODesktopEntry *ret;
   GError *error = NULL;
+  PyObject *py_ret;
 
   ret = desktop_agnostic_fdo_desktop_entry_new (&error);
 
@@ -206,7 +207,10 @@
   }
 
   /* pygobject_new handles NULL checking */
-  return pygobject_new ((GObject *)ret);
+  py_ret = pygobject_new ((GObject *)ret);
+  if (ret != NULL)
+      g_object_unref (ret);
+  return py_ret;
 }
 %%
 define DesktopAgnosticFDODesktopEntry.for_data kwargs staticmethod
@@ -218,6 +222,7 @@
   static char *kwlist[] = { "data", NULL };
   char *data;
   DesktopAgnosticFDODesktopEntry *ret;
+  PyObject *py_ret;
   GError *error = NULL;
 
   if (!PyArg_ParseTupleAndKeywords (args, kwargs,
@@ -235,7 +240,10 @@
   }
 
   /* pygobject_new handles NULL checking */
-  return pygobject_new ((GObject *)ret);
+  py_ret = pygobject_new ((GObject *)ret);
+  if (ret != NULL)
+      g_object_unref (ret);
+  return py_ret;
 }
 %%
 define DesktopAgnosticFDODesktopEntry.for_file onearg staticmethod
@@ -244,6 +252,7 @@
                                                      PyGObject *file)
 {
   DesktopAgnosticFDODesktopEntry *ret;
+  PyObject *py_ret;
   GError *error = NULL;
 
   ret = desktop_agnostic_fdo_desktop_entry_new_for_file (DESKTOP_AGNOSTIC_VFS_FILE(file->obj),
@@ -255,7 +264,10 @@
   }
 
   /* pygobject_new handles NULL checking */
-  return pygobject_new ((GObject *)ret);
+  py_ret = pygobject_new ((GObject *)ret);
+  if (ret != NULL)
+      g_object_unref (ret);
+  return py_ret;
 }
 %%
 define DesktopAgnosticFDODesktopEntry.type_to_string onearg staticmethod
diff -u -Nr libdesktop-agnostic-0.3.90/python/ui.override various-fixes/python/ui.override
--- libdesktop-agnostic-0.3.90/python/ui.override	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/python/ui.override	2010-10-02 04:42:20.187222000 +0200
@@ -46,9 +46,13 @@
                                                     PyGObject *color)
 {
     DesktopAgnosticUIColorButton *ret;
+    PyObject *py_ret;
 
     ret = desktop_agnostic_ui_color_button_new_with_color (DESKTOP_AGNOSTIC_COLOR(color->obj));
 
     /* pygobject_new handles NULL checking */
-    return pygobject_new((GObject *)ret);
+    py_ret = pygobject_new ((GObject *)ret);
+    if (ret != NULL)
+        g_object_unref (ret);
+    return py_ret;
 }
diff -u -Nr libdesktop-agnostic-0.3.90/python/vfs.override various-fixes/python/vfs.override
--- libdesktop-agnostic-0.3.90/python/vfs.override	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/python/vfs.override	2010-10-02 04:42:20.187222000 +0200
@@ -117,6 +117,7 @@
 {
   char *path;
   DesktopAgnosticVFSFile *ret;
+  PyObject *py_ret;
   GError *error = NULL;
 
   if (!PyString_Check (arg))
@@ -136,16 +137,20 @@
   }
 
   /* pygobject_new handles NULL checking */
-  return pygobject_new ((GObject *)ret);
+  py_ret = pygobject_new ((GObject *)ret);
+  if (ret != NULL)
+      g_object_unref (ret);
+  return py_ret;
 }
 %%
 define DesktopAgnosticVFSFile.for_uri onearg staticmethod
 static PyObject *
 _wrap_desktop_agnostic_v_f_s_file_for_uri (PyObject *self, PyObject *arg)
 {
-    char *uri;
-    DesktopAgnosticVFSFile *ret;
-    GError *error = NULL;
+  char *uri;
+  DesktopAgnosticVFSFile *ret;
+  PyObject *py_ret;
+  GError *error = NULL;
 
   if (!PyString_Check (arg))
   {
@@ -164,7 +169,10 @@
   }
 
   /* pygobject_new handles NULL checking */
-  return pygobject_new ((GObject *)ret);
+  py_ret = pygobject_new ((GObject *)ret);
+  if (ret != NULL)
+      g_object_unref (ret);
+  return py_ret;
 }
 %%
 override desktop_agnostic_vfs_file_enumerate_children noargs
diff -u -Nr libdesktop-agnostic-0.3.90/sphinx.py various-fixes/sphinx.py
--- libdesktop-agnostic-0.3.90/sphinx.py	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/sphinx.py	2010-10-02 04:42:20.187222000 +0200
@@ -72,8 +72,15 @@
                              for x in glob.split()])
         glob_base = generate_glob('*', True)
         glob_static = generate_glob('_static/*')
+        glob_sources = ' '.join([os.path.join('docs', x)
+                                 for x in self.generator.path \
+                                              .ant_glob('*.rst').split()])
         self.generator.bld.install_files('${HTMLDIR}', glob_base)
         self.generator.bld.install_files('${HTMLDIR}/_static', glob_static)
+        for rst in glob_sources.split():
+            basename = os.path.splitext(os.path.basename(rst))[0]
+            self.generator.bld.install_as('${HTMLDIR}/_sources/%s.txt' % \
+                                          basename, rst)
 
 
 @TaskGen.extension('.rst')
diff -u -Nr libdesktop-agnostic-0.3.90/wscript various-fixes/wscript
--- libdesktop-agnostic-0.3.90/wscript	2010-04-11 09:32:01.000000000 +0200
+++ various-fixes/wscript	2010-10-02 04:42:20.187222000 +0200
@@ -1,6 +1,7 @@
 #! /usr/bin/env python
 # encoding: utf-8
 
+import intltool
 import Options
 import os
 import Scripting
@@ -69,6 +70,8 @@
                    dest='profiling', default=False,
                    help='Enables the library to be built so that it is '
                         'instrumented to measure performance.')
+    opt.add_option('--disable-gi', action='store_true',
+                   dest='no_gi', default=False)
 
 
 def configure(conf):
@@ -87,12 +90,13 @@
     conf.env['DEBUG'] = Options.options.debug
     conf.env['EXTRA_WARNINGS'] = Options.options.extra_warnings
     conf.env['PROFILING'] = Options.options.profiling
+    conf.env['INTROSPECTION'] = not Options.options.no_gi
     conf.env['VNUM'] = str(VNUM)
 
     conf.check_tool('gnu_dirs')
-    conf.check_tool('compiler_cc misc vala python')
+    conf.check_tool('compiler_cc intltool misc python vala')
 
-    MIN_VALA_VERSION = (0, 7, 10)
+    MIN_VALA_VERSION = (0, 8, 1)
 
     conf.check_cfg(package='gmodule-2.0', uselib_store='GMODULE',
                    atleast_version='2.6.0', mandatory=True,
@@ -130,12 +134,16 @@
         conf.check_cfg(package='gnome-vfs-2.0', uselib_store='GNOME_VFS',
                        atleast_version='2.6.0', mandatory=True,
                        args='--cflags --libs')
+    if 'gio' in conf.env['BACKENDS_DE']:
+        conf.check_cfg(package='gio-unix-2.0', uselib_store='GIO_UNIX',
+                       atleast_version='2.18.0', mandatory=True,
+                       args='--cflags --libs')
     if 'gnome' in conf.env['BACKENDS_DE']:
         conf.check_cfg(package='gnome-desktop-2.0',
                        uselib_store='GNOME_DESKTOP', mandatory=True,
                        args='--cflags --libs')
     # make sure we have the proper Vala version
-    if conf.env['VALAC_VERSION'] != MIN_VALA_VERSION and \
+    if conf.env['VALAC_VERSION'] < MIN_VALA_VERSION and \
         not os.path.isdir(os.path.join(conf.curdir, GEN_SRC_DIR)):
         conf.fatal('''\
 Your Vala compiler version %s is too old. The project requires
@@ -183,6 +191,9 @@
     # process subfolders from here
     bld.add_subdirs('libdesktop-agnostic tools tests data python docs')
 
+    if bld.env['INTLTOOL']:
+        bld.add_subdirs('po')
+
     cls = Task.TaskBase.classes['valac']
     old = cls.run
 
