diff --git a/components/keyrings/common/component_helpers/src/keyring_log_builtins_definition.cc b/components/keyrings/common/component_helpers/src/keyring_log_builtins_definition.cc
index b48013298e1..92b483ba8bc 100644
--- a/components/keyrings/common/component_helpers/src/keyring_log_builtins_definition.cc
+++ b/components/keyrings/common/component_helpers/src/keyring_log_builtins_definition.cc
@@ -25,6 +25,7 @@
 #include <cstring>
 #include <iostream>
 #include <locale>
+#include <memory>
 #include <sstream>
 
 #include <m_string.h>
