--- src/tuxpaint-stamps-2009.06.28/Makefile	2006-09-10 21:06:55.000000000 +0200
+++ tuxpaint-stamps-2009.06.28/Makefile	2011-11-04 11:16:22.956864808 +0100
@@ -79,9 +79,7 @@
 	@rm -rf "build/tuxpaint-stamps-$(VER_DATE)" "build/tuxpaint-stamps-$(VER_DATE).tar.gz"
 	@-if [ -d build ] ; then rmdir build ; fi
 
-releasedir: build/tuxpaint-stamps-$(VER_DATE)
-
-build/tuxpaint-stamps-$(VER_DATE):
+releasedir:
 	@echo
 	@echo "Creating release directory"
 	@echo
