diff --git a/src/Xrd/XrdConfig.cc b/src/Xrd/XrdConfig.cc
index 1be14323e..4b6d9caec 100644
--- a/src/Xrd/XrdConfig.cc
+++ b/src/Xrd/XrdConfig.cc
@@ -46,6 +46,7 @@
 #include <sys/resource.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/socket.h>
 #include <sys/un.h>

 #include "XrdVersion.hh"
@@ -793,12 +794,12 @@ int XrdConfig::ConfigXeq(char *var, XrdOucStream &Config, XrdSysError *eDest)

 int XrdConfig::ASocket(const char *path, const char *fname, mode_t mode)
 {
-   struct sockaddr_un unix;
+   struct sockaddr_un my_local_unix;
    int  plen = strlen(path), flen = strlen(fname);

 // Make sure we can fit everything in our buffer
 //
-   if ((plen + flen + 3) > (int)sizeof(unix.sun_path))
+   if ((plen + flen + 3) > (int)sizeof(my_local_unix.sun_path))
       {Log.Emsg("Config", "admin path", path, "too long");
        return 1;
       }
