Index: intern/cycles/util/util_path.cpp
===================================================================
--- intern/cycles/util/util_path.cpp	(revision 48822)
+++ intern/cycles/util/util_path.cpp	(working copy)
@@ -26,8 +26,6 @@
 
 #include <stdio.h>
 
-#define BOOST_FILESYSTEM_VERSION 2
-
 #include <boost/filesystem.hpp> 
 #include <boost/algorithm/string.hpp>
 
@@ -60,7 +58,7 @@
 
 string path_filename(const string& path)
 {
-	return boost::filesystem::path(path).filename();
+	return boost::filesystem::path(path).filename().string();
 }
 
 string path_dirname(const string& path)
Index: intern/cycles/util/util_cache.cpp
===================================================================
--- intern/cycles/util/util_cache.cpp	(revision 48822)
+++ intern/cycles/util/util_cache.cpp	(working copy)
@@ -26,8 +26,6 @@
 #include "util_path.h"
 #include "util_types.h"
 
-#define BOOST_FILESYSTEM_VERSION 2
-
 #include <boost/filesystem.hpp> 
 #include <boost/algorithm/string.hpp>
 
@@ -117,7 +115,7 @@
 		boost::filesystem::directory_iterator it(dir), it_end;
 
 		for(; it != it_end; it++) {
-			string filename = it->path().filename();
+			string filename = it->path().filename().string();
 
 			if(boost::starts_with(filename, name))
 				if(except.find(filename) == except.end())
