diff --git a/scripts/src/3rdparty.lua b/scripts/src/3rdparty.lua
index 5744e8d5a2f..4848d905d50 100644
--- a/scripts/src/3rdparty.lua
+++ b/scripts/src/3rdparty.lua
@@ -845,7 +845,6 @@ project "lua"
 	kind "StaticLib"
 
 	options {
-		"ForceCPP",
 	}
 
 	configuration { "gmake or ninja" }
@@ -933,7 +932,6 @@ project "lualibs"
 	kind "StaticLib"
 
 	options {
-		"ForceCPP",
 	}
 
 	configuration { "gmake or ninja" }
