=== modified file 'qreator/qrcodes/QRCodeWifiGtk.py'
--- qreator/qrcodes/QRCodeWifiGtk.py	2017-12-19 14:47:03 +0000
+++ qreator/qrcodes/QRCodeWifiGtk.py	2018-07-30 01:05:28 +0000
@@ -148,8 +148,10 @@
     for dev in devs:
         if dev.get_device_type() == NM.DeviceType.WIFI:
             for ap in dev.get_access_points():
-                ssid = NM.utils_ssid_to_utf8(ap.get_ssid().get_data());
-                ssids.append((ap.get_strength(), ssid))
+                ssid = ap.get_ssid()
+                if ssid is not None:
+                    ssid = NM.utils_ssid_to_utf8(ssid.get_data());
+                    ssids.append((ap.get_strength(), ssid))
     ssids.sort()
     ssids.reverse()   # strongest has the highest number -> should be first
     return ssids

