diff --git a/src/ukws_new_workspace_button.cpp b/src/ukws_new_workspace_button.cpp
index f8487a1..1121ef8 100644
--- a/src/ukws_new_workspace_button.cpp
+++ b/src/ukws_new_workspace_button.cpp
@@ -2,6 +2,7 @@
 #include "ukws_new_workspace_button.h"
 
 #include <QPainter>
+#include <QPainterPath>
 
 UkwsNewWorkspaceButton::UkwsNewWorkspaceButton(QWidget *parent) : QWidget(parent)
 {
diff --git a/src/ukws_window_box.cpp b/src/ukws_window_box.cpp
index 3079e9f..ee7ad52 100644
--- a/src/ukws_window_box.cpp
+++ b/src/ukws_window_box.cpp
@@ -29,6 +29,7 @@
 #include <QDebug>
 #include <QStyleOption>
 #include <QPainter>
+#include <QPainterPath>
 
 #include "ukws_common.h"
 #include "ukws_helper.h"
