diff -Naur madman-0.93.orig/utility/refcnt_ptr.h madman-0.93/utility/refcnt_ptr.h
--- madman-0.93.orig/utility/refcnt_ptr.h	2004-05-20 07:03:13.000000000 -0700
+++ madman-0.93/utility/refcnt_ptr.h	2004-07-12 12:09:48.000000000 -0700
@@ -109,7 +109,7 @@
     refcnt_ptr(const refcnt_ptr<T2>& rhs) throw()
 	{
 	object = rhs.object;
-	refcnt = rhs.refCnt;
+	this->refcnt = rhs.refCnt;
 	increaseCount();
 	}
 
