diff -Naur cinnamon-desktop-2.8.0.orig/schemas/org.cinnamon.desktop.background.gschema.xml.in.in cinnamon-desktop-2.8.0/schemas/org.cinnamon.desktop.background.gschema.xml.in.in
--- cinnamon-desktop-2.8.0.orig/schemas/org.cinnamon.desktop.background.gschema.xml.in.in	2015-10-16 13:00:30.000000000 +0000
+++ cinnamon-desktop-2.8.0/schemas/org.cinnamon.desktop.background.gschema.xml.in.in	2015-11-20 19:04:16.543027956 +0000
@@ -10,7 +10,7 @@
       </_description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/themes/Adwaita/backgrounds/adwaita-timed.xml'</default>
+      <default>'file://@datadir@/backgrounds/gnome/adwaita-timed.xml'</default>
       <_summary>Picture URI</_summary>
       <_description>
         URI to use for the background image. Not that the backend only supports
diff -Naur cinnamon-desktop-2.8.0.orig/schemas/org.cinnamon.desktop.interface.gschema.xml.in.in cinnamon-desktop-2.8.0/schemas/org.cinnamon.desktop.interface.gschema.xml.in.in
--- cinnamon-desktop-2.8.0.orig/schemas/org.cinnamon.desktop.interface.gschema.xml.in.in	2015-10-16 13:00:30.000000000 +0000
+++ cinnamon-desktop-2.8.0/schemas/org.cinnamon.desktop.interface.gschema.xml.in.in	2015-11-20 19:04:16.543027956 +0000
@@ -97,7 +97,7 @@
       </_description>
     </key>
     <key name="icon-theme" type="s">
-      <default>'gnome'</default>
+      <default>'Adwaita'</default>
       <_summary>Icon Theme</_summary>
       <_description>
         Icon theme to use for the panel, nautilus etc.
diff -Naur cinnamon-desktop-2.8.0.orig/schemas/org.cinnamon.desktop.session.gschema.xml.in.in cinnamon-desktop-2.8.0/schemas/org.cinnamon.desktop.session.gschema.xml.in.in
--- cinnamon-desktop-2.8.0.orig/schemas/org.cinnamon.desktop.session.gschema.xml.in.in	2015-10-16 13:00:30.000000000 +0000
+++ cinnamon-desktop-2.8.0/schemas/org.cinnamon.desktop.session.gschema.xml.in.in	2015-11-20 19:04:16.546361273 +0000
@@ -12,17 +12,17 @@
       "gnome" and "gnome-fallback".</description>
     </key>
     <key name="screensaver-uses-logind" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Whether or not cinnamon-screensaver listens to logind to lock/unlock itself</_summary>
       <_description>If true, it listens to logind. Otherwise it listens to consolekit.</_description>
     </key>
     <key name="session-manager-uses-logind" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Whether or not cinnamon-session uses logind to suspend/hibernate/shutdown/restart (usually from the shutdown dialog)</_summary>
       <_description>If true, it uses logind. Otherwise it uses consolekit to shutdown/restart and upower to suspend/hibernate. It is only compatible with upower 0.9 and lower versions (not 0.99 or later).</_description>
     </key>
     <key name="settings-daemon-uses-logind" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Whether or not cinnamon-settings-daemon uses logind to suspend/hibernate/shutdown (either from media-keys or power-manager)</_summary>
       <_description>If true, it uses logind for all three actions. Otherwise it uses consolekit for shutdown and upower for suspend/hibernate.</_description>
     </key>
