diff -rupN a/umlet.sh b/umlet.sh
--- a/umlet.sh	2015-06-13 10:58:05.076542241 +0200
+++ b/umlet.sh	2015-06-13 10:59:40.336309692 +0200
@@ -9,6 +9,6 @@
 programDir=$(cd $(dirname $0);pwd)
 
 if [ $# -eq 1 ]
- then java -jar ${programDir}/umlet.jar -filename="$1"
- else java -jar ${programDir}/umlet.jar "$@"
+ then java -Dsun.java2d.xrender=f -jar ${programDir}/umlet.jar -filename="$1"
+ else java -Dsun.java2d.xrender=f -jar ${programDir}/umlet.jar "$@"
 fi
