--- src/third_party/protobuf/3.7.0/util/python/BUILD	2019-02-28 20:55:14.000000000 +0000
+++ src/third_party/protobuf/3.7.0/util/python/BUILD	2019-03-02 21:35:32.359033541 +0000
@@ -12,6 +12,14 @@
 #   name = "python",
 #   path = __workspace_dir__ + "/util/python",
 # )
+licenses(["notice"])
+
+filegroup(
+    name = "srcs",
+    srcs = glob(["**"]),
+    visibility = ["//visibility:public"],
+)
+
 cc_library(
     name = "python_headers",
     visibility = ["//visibility:public"],
