--- src/third_party/protobuf/3.7.0/third_party/googletest/BUILD.bazel	2019-03-02 21:45:23.409589455 +0000
+++ src/third_party/protobuf/3.7.0/third_party/googletest/BUILD.bazel	1980-01-01 00:00:00.000000000 +0000
@@ -36,6 +36,12 @@
 
 licenses(["notice"])
 
+filegroup(
+    name = "srcs",
+    srcs = glob(["**/*"]),
+    visibility = ["//visibility:public"],
+)
+
 config_setting(
     name = "win",
     values = {"cpu": "x64_windows_msvc"},
