From 8c3743f8b75875dd712ab4e9f168c3a7cbbac20c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bart=C5=82omiej=20Piotrowski?= <bpiotrowski@archlinux.org>
Date: Mon, 31 Jul 2017 18:31:57 +0200
Subject: [PATCH] Replace ladon with GNU parallel

---
 jpeg-archive | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/jpeg-archive b/jpeg-archive
index 551255d..5ecdccd
--- a/jpeg-archive
+++ b/jpeg-archive
@@ -1,6 +1,8 @@
 #!/bin/bash
+shopt -s globstar
+shopt -s extglob
 
-# This script requires ladon, dcraw, exiftool, and jpeg-recompress
+# This script requires GNU parallel, dcraw, exiftool, and jpeg-recompress
 # Usage:
 #   $ cd path/to/photos
 #   $ jpeg-archive
@@ -22,18 +24,24 @@ fi
 
 set -e
 
+# Common parallel arguments
+PARALLEL_ARGS=(
+    '--profile /usr/share/jpeg-archive/parallel.config'
+    --no-notice
+)
+
 # Cleanup old files
 rm -rf Comp
 rm -rf /tmp/comp
 
 echo 'Converting RAW files...'
-ladon -m /tmp/comp/RELDIR '**/*.+(cr2|nef|dng)' -- "dcraw -w -q 3 -c RELPATH | jpeg-recompress ${@:--q high} --ppm - /tmp/comp/RELDIR/BASENAME.jpg"
+parallel ${PARALLEL_ARGS[@]} "mkdir -p /tmp/comp/RELDIR; dcraw -w -q 3 -c RELPATH | jpeg-recompress ${@:--q high} --ppm - /tmp/comp/RELDIR/BASENAME.jpg" ::: **/*.+(cr2|nef|dng)
 
 echo 'Copying EXIF data...'
-ladon '**/*.+(cr2|nef|dng)' -- exiftool -overwrite_original -TagsFromFile RELPATH -all:all /tmp/comp/RELDIR/BASENAME.jpg
+parallel ${PARALLEL_ARGS[@]} 'exiftool -overwrite_original -TagsFromFile RELPATH -all:all /tmp/comp/RELDIR/BASENAME.jpg' ::: **/*.+(cr2|nef|dng)
 
 echo 'Recompressing JPEG files'
-ladon -m /tmp/comp/RELDIR '**/*.jpg' -- jpeg-recompress ${@:--q high} RELPATH /tmp/comp/RELPATH
+parallel ${PARALLEL_ARGS[@]} "mkdir -p /tmp/comp/RELDIR; jpeg-recompress ${@:--q high} RELPATH /tmp/comp/RELPATH" ::: **/*.jpg
 
 mv /tmp/comp Comp
 
-- 
2.13.3

