diff -Naur sword-1.6.2.orig/src/mgr/curlftpt.cpp sword-1.6.2.new/src/mgr/curlftpt.cpp
--- sword-1.6.2.orig/src/mgr/curlftpt.cpp	2011-07-27 00:41:40.000000000 +0200
+++ sword-1.6.2.new/src/mgr/curlftpt.cpp	2011-07-27 00:42:05.000000000 +0200
@@ -26,7 +26,6 @@
 #include <fcntl.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include <swlog.h>
diff -Naur sword-1.6.2.orig/src/mgr/curlhttpt.cpp sword-1.6.2.new/src/mgr/curlhttpt.cpp
--- sword-1.6.2.orig/src/mgr/curlhttpt.cpp	2011-07-27 00:41:40.000000000 +0200
+++ sword-1.6.2.new/src/mgr/curlhttpt.cpp	2011-07-27 00:42:45.000000000 +0200
@@ -25,7 +25,6 @@
 #include <cctype>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include <swlog.h>
