diff --git a/platformplugin/dplatformwindowhelper.cpp b/platformplugin/dplatformwindowhelper.cpp
index 925c0ed..4c0bb57 100644
--- a/platformplugin/dplatformwindowhelper.cpp
+++ b/platformplugin/dplatformwindowhelper.cpp
@@ -320,9 +320,7 @@ void DPlatformWindowHelper::setVisible(bool visible)
         Utility::setMotifWmHints(window->m_window, mwmhints);
         Utility::setMotifWmHints(helper->m_nativeWindow->QNativeWindow::winId(), cw_hints);
 
-        if (helper->m_nativeWindow->window()->modality() != Qt::NonModal) {
-            window->setNetWmStates(window->netWmStates() | QNativeWindow::NetWmStateModal);
-        }
+        window->setNetWmStateOnUnmappedWindow();
 #endif
 
 #if QT_VERSION >= QT_VERSION_CHECK(5, 9, 0)
@@ -362,7 +360,7 @@ void DPlatformWindowHelper::setWindowState(Qt::WindowStates state)
     if (state == Qt::WindowMinimized
             && (window->m_windowState == Qt::WindowMaximized
                 || window->m_windowState == Qt::WindowFullScreen)) {
-        window->changeNetWmState(true, Utility::internAtom("_NET_WM_STATE_HIDDEN"));
+        window->setNetWmState(true, Utility::internAtom("_NET_WM_STATE_HIDDEN"));
         Utility::XIconifyWindow(window->connection()->xlib_display(),
                                 window->m_window,
                                 window->connection()->primaryScreenNumber());
