From ee6ebb4a416c386ec1d32460a2caa0d486d7547b Mon Sep 17 00:00:00 2001
From: zccrs <ccrr1314@live.com>
Date: Tue, 19 Jun 2018 16:57:26 +0800
Subject: [PATCH] fix: crash when add scroll device

Qt Version >= 5.10.0

Change-Id: I03dd073031f68c34061d817d2610b6dfd4316062
---

diff --git a/platformplugin/xcbnativeeventfilter.cpp b/platformplugin/xcbnativeeventfilter.cpp
index 62018c9..d032631 100644
--- a/platformplugin/xcbnativeeventfilter.cpp
+++ b/platformplugin/xcbnativeeventfilter.cpp
@@ -181,6 +181,7 @@
                     return false;
                 }
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
                 xXIDeviceChangedEvent *xiDCEvent = reinterpret_cast<xXIDeviceChangedEvent *>(xiEvent);
                 QHash<int, QXcbConnection::ScrollingDevice>::iterator device = xcb_connect->m_scrollingDevices.find(xiDCEvent->sourceid);
 
@@ -191,23 +192,22 @@
                     return false;
                 }
 
-
                 for (int c = 0; c < xiDeviceInfo->num_classes; ++c) {
                     if (xiDeviceInfo->classes[c]->type == XIScrollClass) {
                         XIScrollClassInfo *sci = reinterpret_cast<XIScrollClassInfo *>(xiDeviceInfo->classes[c]);
 
                         if (sci->scroll_type == XIScrollTypeVertical) {
-                            device->legacyOrientations = device->orientations;
-                            device->orientations |= Qt::Vertical;
-                            device->verticalIndex = sci->number;
+//                            device->legacyOrientations = device->orientations;
+//                            device->orientations |= Qt::Vertical;
+//                            device->verticalIndex = sci->number;
                             device->verticalIncrement = std::signbit(sci->increment)
                                     ? -std::abs(device->verticalIncrement)
                                     : std::abs(device->verticalIncrement);
                         }
                         else if (sci->scroll_type == XIScrollTypeHorizontal) {
-                            device->legacyOrientations = device->orientations;
-                            device->orientations |= Qt::Horizontal;
-                            device->horizontalIndex = sci->number;
+//                            device->legacyOrientations = device->orientations;
+//                            device->orientations |= Qt::Horizontal;
+//                            device->horizontalIndex = sci->number;
                             device->horizontalIncrement = std::signbit(sci->increment)
                                     ? -std::abs(device->horizontalIncrement)
                                     : std::abs(device->horizontalIncrement);
@@ -216,6 +216,7 @@
                 }
 
                 XIFreeDeviceInfo(xiDeviceInfo);
+#endif
             }
             break;
         }
