--- a/src/file-system-blob-store.js
+++ b/src/file-system-blob-store.js
@@ -20,7 +20,7 @@
 
   reset () {
     this.inMemoryBlobs = new Map()
-    this.storedBlob = new Buffer(0)
+    this.storedBlob = Buffer.alloc(0)
     this.storedBlobMap = {}
     this.usedKeys = new Set()
   }
--- a/src/compile-cache.js
+++ b/src/compile-cache.js
@@ -177,7 +177,7 @@
       var rawData = sourceMappingURL.slice(sourceMappingURL.indexOf(',') + 1)
 
       try {
-        var sourceMap = JSON.parse(new Buffer(rawData, 'base64'))
+        var sourceMap = JSON.parse(Buffer.from(rawData, 'base64'))
       } catch (error) {
         console.warn('Error parsing source map', error.stack)
         return null
