--- a/src/browser/atom-window.coffee
+++ b/src/browser/atom-window.coffee
@@ -24,11 +24,11 @@
     options =
       show: false
       title: 'Atom'
-      'web-preferences':
-        'direct-write': true
+      'webPreferences':
+        'directWrite': true
 
     if @isSpec
-      options['web-preferences']['page-visibility'] = true
+      options.webPreferences.backgroundThrottling = false
 
     # Don't set icon on Windows so the exe's ico will be used as window and
     # taskbar's icon. See https://github.com/atom/atom/issues/4811 for more.
