From 514c308ec71074228e7a32356646a7ea18576377 Mon Sep 17 00:00:00 2001
From: Xavier Roche <roche@httrack.com>
Date: Sun, 26 Jun 2016 14:26:37 +0200
Subject: [PATCH] Fixed FTBFS with openssl 1.1.0

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 3f43580..4268cb9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -160,7 +160,7 @@ AC_CHECK_LIB(crypto, EVP_get_digestbyname,
 	]
 	)
 LIBS="$LIBS $OPENSSL_LIBS"
-AC_CHECK_LIB(ssl, SSL_library_init,
+AC_CHECK_LIB(ssl, SSL_CTX_new,
 	[
 	OPENSSL_LIBS="$OPENSSL_LIBS -lssl"
 	AC_DEFINE(HTS_USEOPENSSL, 1, [Check for OpenSSL])
