--- a/Makefile
+++ b/Makefile
@@ -46,7 +46,6 @@ coverage:
 	coverage run --source rtmidi setup.py test
 	coverage report -m
 	coverage html
-	xdg-open htmlcov/index.html
 
 dist: clean release
 	ls -l dist
@@ -59,7 +58,6 @@ docs: release
 	cat docs/api.rst.inc >> docs/rtmidi.rst
 	$(MAKE) -C docs clean
 	$(MAKE) -C docs html
-	xdg-open docs/_build/html/index.html
 
 lint:
 	flake8 rtmidi tests examples
