diff -ruN a/extras/AudioPerformanceTest/CMakeLists.txt e/extras/AudioPerformanceTest/CMakeLists.txt
--- a/extras/AudioPerformanceTest/CMakeLists.txt	2021-09-20 14:26:04.000000000 +0200
+++ e/extras/AudioPerformanceTest/CMakeLists.txt	2021-09-25 20:11:11.782908700 +0200
@@ -32,6 +32,11 @@
     JUCE_USE_CURL=0 JUCE_WEB_BROWSER=0)
 
 target_link_libraries(AudioPerformanceTest PRIVATE
+    "FLAC"
+    "ogg"
+    "vorbis"
+    "vorbisenc"
+    "vorbisfile"
     juce::juce_audio_utils
     juce::juce_recommended_config_flags
     juce::juce_recommended_lto_flags
diff -ruN a/extras/AudioPluginHost/CMakeLists.txt e/extras/AudioPluginHost/CMakeLists.txt
--- a/extras/AudioPluginHost/CMakeLists.txt	2021-09-20 14:26:04.000000000 +0200
+++ e/extras/AudioPluginHost/CMakeLists.txt	2021-09-25 20:31:04.882600939 +0200
@@ -55,6 +55,10 @@
     JUCE_WEB_BROWSER=0)
 
 target_link_libraries(AudioPluginHost PRIVATE
+    "ogg"
+    "vorbis"
+    "vorbisenc"
+    "vorbisfile"
     juce::juce_audio_utils
     juce::juce_cryptography
     juce::juce_dsp
diff -ruN a/extras/NetworkGraphicsDemo/CMakeLists.txt e/extras/NetworkGraphicsDemo/CMakeLists.txt
--- a/extras/NetworkGraphicsDemo/CMakeLists.txt	2021-09-20 14:26:04.000000000 +0200
+++ e/extras/NetworkGraphicsDemo/CMakeLists.txt	2021-09-25 20:14:30.376577084 +0200
@@ -31,6 +31,13 @@
     JUCE_USE_CURL=0 JUCE_WEB_BROWSER=0)
 
 target_link_libraries(NetworkGraphicsDemo PRIVATE
+    "FLAC"
+    "ogg"
+    "png"
+    "vorbis"
+    "vorbisenc"
+    "vorbisfile"
+    "z"
     juce::juce_audio_utils
     juce::juce_cryptography
     juce::juce_opengl
diff -ruN a/extras/UnitTestRunner/CMakeLists.txt e/extras/UnitTestRunner/CMakeLists.txt
--- a/extras/UnitTestRunner/CMakeLists.txt	2021-09-20 14:26:04.000000000 +0200
+++ e/extras/UnitTestRunner/CMakeLists.txt	2021-09-25 20:22:11.631027145 +0200
@@ -33,6 +33,13 @@
     JUCE_WEB_BROWSER=0)
 
 target_link_libraries(UnitTestRunner PRIVATE
+    "FLAC"
+    "ogg"
+    "png16"
+    "vorbis"
+    "vorbisenc"
+    "vorbisfile"
+    "z"
     juce::juce_analytics
     juce::juce_audio_utils
     juce::juce_dsp
diff -ruN a/extras/Build/juceaide/CMakeLists.txt b/extras/Build/juceaide/CMakeLists.txt
--- a/extras/Build/juceaide/CMakeLists.txt	2021-09-20 14:26:04.000000000 +0200
+++ b/extras/Build/juceaide/CMakeLists.txt	2021-09-25 18:52:06.968927572 +0200
@@ -37,6 +37,9 @@
         JUCE_USE_CURL=0)
 
     target_link_libraries(juceaide PRIVATE
+        "jpeg"
+        "png16"
+        "z"
         juce::juce_build_tools
         juce::juce_recommended_config_flags
         juce::juce_recommended_lto_flags
diff -ruN a/extras/Projucer/CMakeLists.txt b/extras/Projucer/CMakeLists.txt
--- a/extras/Projucer/CMakeLists.txt	2021-09-20 14:26:04.000000000 +0200
+++ b/extras/Projucer/CMakeLists.txt	2021-09-25 19:02:39.550915895 +0200
@@ -158,6 +158,8 @@
     ../Build/CMake/juce_runtime_arch_detection.cpp)
 
 target_link_libraries(Projucer PRIVATE
+    "png16"
+    "jpeg"
     ProjucerData
     juce::juce_build_tools
     juce::juce_cryptography
