diff -ruN a/extras/Build/CMake/JUCEUtils.cmake b/extras/Build/CMake/JUCEUtils.cmake
--- a/extras/Build/CMake/JUCEUtils.cmake	2021-09-20 14:26:04.000000000 +0200
+++ b/extras/Build/CMake/JUCEUtils.cmake	2021-09-25 19:28:31.202649065 +0200
@@ -83,6 +83,7 @@
 if((CMAKE_SYSTEM_NAME STREQUAL "Linux") OR (CMAKE_SYSTEM_NAME MATCHES ".*BSD"))
     _juce_create_pkgconfig_target(JUCE_CURL_LINUX_DEPS libcurl)
     _juce_create_pkgconfig_target(JUCE_BROWSER_LINUX_DEPS webkit2gtk-4.0 gtk+-x11-3.0)
+    _juce_create_pkgconfig_target(JUCE_IMG_LINUX_DEPS libjpeg libpng)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     find_program(JUCE_XCRUN xcrun)
 
@@ -225,6 +226,8 @@
         if(needs_browser)
             target_link_libraries(${target} PRIVATE juce::pkgconfig_JUCE_BROWSER_LINUX_DEPS)
         endif()
+
+        target_link_libraries(${target} PRIVATE juce::pkgconfig_JUCE_IMG_LINUX_DEPS)
     elseif(APPLE)
         get_target_property(needs_storekit ${target} JUCE_NEEDS_STORE_KIT)
 
