diff -ruN a/extras/Projucer/Source/ProjectSaving/jucer_ProjectExporter.cpp b/extras/Projucer/Source/ProjectSaving/jucer_ProjectExporter.cpp
--- a/extras/Projucer/Source/ProjectSaving/jucer_ProjectExporter.cpp	2021-09-20 14:26:04.000000000 +0200
+++ b/extras/Projucer/Source/ProjectSaving/jucer_ProjectExporter.cpp	2021-09-26 21:02:34.760731676 +0200
@@ -528,6 +528,39 @@
         packages.add ("gtk+-x11-3.0");
     }
 
+    // as vendored libraries were removed, we need to link against system libraries
+    if (project.getEnabledModules().isModuleEnabled ("juce_core"))
+    {
+        packages.add ("zlib");
+    }
+    if (project.getEnabledModules().isModuleEnabled ("juce_graphics"))
+    {
+        packages.add ("libjpeg");
+        packages.add ("libpng");
+    }
+    if (project.getEnabledModules().isModuleEnabled ("juce_audio_formats"))
+    {
+        if(project.isConfigFlagEnabled ("JUCE_USE_FLAC", true))
+        {
+            packages.add ("flac");
+        }
+
+        if(project.isConfigFlagEnabled ("JUCE_USE_OGGVORBIS", true))
+        {
+            packages.add ("vorbis");
+            packages.add ("vorbisfile");
+            packages.add ("vorbisenc");
+            packages.add ("ogg");
+        }
+    }
+    if (project.getEnabledModules().isModuleEnabled ("juce_audio_devices"))
+    {
+        if(project.isConfigFlagEnabled ("JUCE_JACK", true))
+        {
+            packages.add ("jack");
+        }
+    }
+
     packages.removeEmptyStrings();
     packages.removeDuplicates (false);
 
diff -ruN a/modules/juce_core/juce_core.h b/modules/juce_core/juce_core.h
--- a/modules/juce_core/juce_core.h	2021-09-20 14:26:04.000000000 +0200
+++ b/modules/juce_core/juce_core.h	2021-09-26 21:16:41.501526112 +0200
@@ -65,6 +65,31 @@
 
 #include "system/juce_TargetPlatform.h"
 
+#ifdef JUCE_INCLUDE_ZLIB_CODE
+# undef JUCE_INCLUDE_ZLIB_CODE
+#endif
+#define JUCE_INCLUDE_ZLIB_CODE 0
+
+#ifdef JUCE_INCLUDE_FLAC_CODE
+# undef JUCE_INCLUDE_FLAC_CODE
+#endif
+#define JUCE_INCLUDE_FLAC_CODE 0
+
+#ifdef JUCE_INCLUDE_OGGVORBIS_CODE
+# undef JUCE_INCLUDE_OGGVORBIS_CODE
+#endif
+#define JUCE_INCLUDE_OGGVORBIS_CODE 0
+
+#ifdef JUCE_INCLUDE_JPEGLIB_CODE
+# undef JUCE_INCLUDE_JPEGLIB_CODE
+#endif
+#define JUCE_INCLUDE_JPEGLIB_CODE 0
+
+#ifdef JUCE_INCLUDE_PNGLIB_CODE
+# undef JUCE_INCLUDE_PNGLIB_CODE
+#endif
+#define JUCE_INCLUDE_PNGLIB_CODE 0
+
 //==============================================================================
 /** Config: JUCE_FORCE_DEBUG
 
@@ -124,7 +149,7 @@
     specify the path where your zlib headers live.
 */
 #ifndef JUCE_INCLUDE_ZLIB_CODE
- #define JUCE_INCLUDE_ZLIB_CODE 1
+ #define JUCE_INCLUDE_ZLIB_CODE 0
 #endif
 
 #ifndef JUCE_ZLIB_INCLUDE_PATH
