--- backend/comics/comicsdocument.atril-backend.in.orig	2018-12-19 21:56:58.967629664 -0600
+++ backend/comics/comicsdocument.atril-backend.in	2018-12-19 21:57:16.687554873 -0600
@@ -1,4 +1,4 @@
 [Atril Backend]
 Module=comicsdocument
 _TypeDescription=Comic Books
-MimeType=application/x-cbr;application/x-cbz;application/x-cb7;application/x-cbt;application/vnd.comicbook+zip;application/vnd.rar;
+MimeType=application/x-cbr;application/x-cbz;application/x-cb7;application/x-cbt;application/vnd.comicbook+zip;application/vnd.comicbook-rar;
