diff -rup luxrender-lux-3066b2161439/cmake/luxrender.cmake luxrender-lux-3066b2161439.new/cmake/luxrender.cmake
--- luxrender-lux-3066b2161439/cmake/luxrender.cmake	2013-03-16 19:32:24.000000000 +0000
+++ luxrender-lux-3066b2161439.new/cmake/luxrender.cmake	2013-03-26 18:26:46.194520697 +0000
@@ -98,7 +98,7 @@ IF(QT4_FOUND)
 
 	QT4_ADD_RESOURCES( LUXQTGUI_RC_SRCS ${LUXQTGUI_RCS})
 	QT4_WRAP_UI( LUXQTGUI_UI_HDRS ${LUXQTGUI_UIS} )
-	QT4_WRAP_CPP( LUXQTGUI_MOC_SRCS ${LUXQTGUI_MOC} )
+	QT4_WRAP_CPP( LUXQTGUI_MOC_SRCS ${LUXQTGUI_MOC} OPTIONS -DBOOST_TT_HAS_OPERATOR_HPP_INCLUDED -DBOOST_NO_TEMPLATE_PARTIAL_SPECIALIZATION )
 
 	#file (GLOB TRANSLATIONS_FILES qtgui/translations/*.ts)
 	#qt4_create_translation(QM_FILES ${FILES_TO_TRANSLATE} ${TRANSLATIONS_FILES})
diff -rup luxrender-lux-3066b2161439/core/film.cpp luxrender-lux-3066b2161439.new/core/film.cpp
--- luxrender-lux-3066b2161439/core/film.cpp	2013-03-16 19:32:24.000000000 +0000
+++ luxrender-lux-3066b2161439.new/core/film.cpp	2013-03-26 18:03:02.828222310 +0000
@@ -890,8 +890,8 @@ Film::Film(u_int xres, u_int yres, Filte
 	filename(filename1),
 	colorSpace(0.63f, 0.34f, 0.31f, 0.595f, 0.155f, 0.07f, 0.314275f, 0.329411f), // default is SMPTE
 	convTest(NULL), varianceBuffer(NULL),
-	noiseAwareMap(NULL), noiseAwareMapVersion(0),
-	userSamplingMapFileName(samplingmapfilename), userSamplingMap(NULL), userSamplingMapVersion(0),
+	noiseAwareMapVersion(0),
+	userSamplingMapFileName(samplingmapfilename), userSamplingMapVersion(0),
 	ZBuffer(NULL), use_Zbuf(useZbuffer),
 	debug_mode(debugmode), premultiplyAlpha(premult),
 	writeResumeFlm(w_resume_FLM), restartResumeFlm(restart_resume_FLM), writeFlmDirect(write_FLM_direct),
diff -rup luxrender-lux-3066b2161439/samplers/lowdiscrepancy.cpp luxrender-lux-3066b2161439.new/samplers/lowdiscrepancy.cpp
--- luxrender-lux-3066b2161439/samplers/lowdiscrepancy.cpp	2013-03-16 19:32:24.000000000 +0000
+++ luxrender-lux-3066b2161439.new/samplers/lowdiscrepancy.cpp	2013-03-26 18:03:02.828222310 +0000
@@ -29,7 +29,7 @@
 using namespace lux;
 
 LDSampler::LDData::LDData(const Sampler &sampler, int xPixelStart, int yPixelStart, u_int pixelSamples) :
-	samplingMap(NULL), noiseAwareMapVersion(0), userSamplingMapVersion(0) {
+	noiseAwareMapVersion(0), userSamplingMapVersion(0) {
 	xPos = xPixelStart - 1;
 	yPos = yPixelStart;
 	samplePos = pixelSamples - 1;
diff -rup luxrender-lux-3066b2161439/samplers/metrosampler.cpp luxrender-lux-3066b2161439.new/samplers/metrosampler.cpp
--- luxrender-lux-3066b2161439/samplers/metrosampler.cpp	2013-03-16 19:32:24.000000000 +0000
+++ luxrender-lux-3066b2161439.new/samplers/metrosampler.cpp	2013-03-26 18:13:13.146702871 +0000
@@ -39,7 +39,7 @@ static const u_int rngA = 884;
 MetropolisSampler::MetropolisData::MetropolisData(const MetropolisSampler &sampler) :
 	consecRejects(0), stamp(0), currentStamp(0), weight(0.f),
 	LY(0.f), alpha(0.f), totalLY(0.f), sampleCount(0.f),
-	samplingMap(NULL), noiseAwareMapVersion(0), userSamplingMapVersion(0),
+	noiseAwareMapVersion(0), userSamplingMapVersion(0),
 	large(true), cooldown(sampler.cooldownTime > 0)
 {
 	u_int i;
diff -rup luxrender-lux-3066b2161439/samplers/random.cpp luxrender-lux-3066b2161439.new/samplers/random.cpp
--- luxrender-lux-3066b2161439/samplers/random.cpp	2013-03-16 19:32:24.000000000 +0000
+++ luxrender-lux-3066b2161439.new/samplers/random.cpp	2013-03-26 18:12:34.378868365 +0000
@@ -30,7 +30,7 @@ using namespace lux;
 
 RandomSampler::RandomData::RandomData(const Sampler &sampler, int xPixelStart,
 	int yPixelStart, u_int pixelSamples) :
-	samplingMap(NULL), noiseAwareMapVersion(0), userSamplingMapVersion(0)
+	noiseAwareMapVersion(0), userSamplingMapVersion(0)
 {
 	xPos = xPixelStart;
 	yPos = yPixelStart;
