diff -wbBur psi-1.3.org/src/groupchatdlg.cpp psi-1.3/src/groupchatdlg.cpp
--- psi-1.3.org/src/groupchatdlg.cpp	2017-09-24 13:54:15.000000000 +0300
+++ psi-1.3/src/groupchatdlg.cpp	2018-02-20 20:33:06.279033368 +0300
@@ -1456,7 +1456,7 @@
 		QString nick = d->self;
 
 		if(!account()->groupChatJoin(host, room, nick, d->password)) {
-			appendSysMsg(tr("Error: You are in or joining this room already!"), true);
+//			appendSysMsg(tr("Error: You are in or joining this room already!"), true);
 			d->state = Private::Idle;
 		}
 	}
diff -wbBur psi-1.3.org/src/mucjoindlg.cpp psi-1.3/src/mucjoindlg.cpp
--- psi-1.3.org/src/mucjoindlg.cpp	2017-09-24 13:54:15.000000000 +0300
+++ psi-1.3/src/mucjoindlg.cpp	2018-02-20 20:33:09.399033235 +0300
@@ -181,7 +181,7 @@
 
 
 	if (!account_->groupChatJoin(host, room, nick, pass, !ui_.ck_history->isChecked())) {
-		QMessageBox::information(this, tr("Error"), tr("You are in or joining this room already!"));
+//		QMessageBox::information(this, tr("Error"), tr("You are in or joining this room already!"));
 		return;
 	}
 
