--- amule/src/ClientCreditsList.cpp.orig	2018-01-30 22:28:29.951863645 +0000
+++ amule/src/ClientCreditsList.cpp	2018-01-30 22:30:05.725410074 +0000
@@ -312,7 +312,7 @@
 		// calculate and store public key
 		CryptoPP::RSASSA_PKCS1v15_SHA_Verifier pubkey(*static_cast<CryptoPP::RSASSA_PKCS1v15_SHA_Signer *>(m_pSignkey));
 		CryptoPP::ArraySink asink(m_abyMyPublicKey, 80);
-		pubkey.DEREncode(asink);
+		pubkey.AccessMaterial().Save(asink);
 		m_nMyPublicKeyLen = asink.TotalPutLength();
 		asink.MessageEnd();
 	} catch (const CryptoPP::Exception& e) {
