diff --git a/lib/Makefile b/lib/Makefile
index fd1710cf..fb957ea8 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -245,6 +245,7 @@ libzstd.pc: libzstd.pc.in
 	@echo creating pkgconfig
 	@sed -e 's|@PREFIX@|$(PREFIX)|' \
              -e 's|@VERSION@|$(VERSION)|' \
+             -e 's|@LIBDIR@|$(LIBDIR)|' \
              $< >$@
 
 install: install-pc install-static install-shared install-includes
diff --git a/lib/libzstd.pc.in b/lib/libzstd.pc.in
index e7880be4..9309426e 100644
--- a/lib/libzstd.pc.in
+++ b/lib/libzstd.pc.in
@@ -5,7 +5,7 @@
 prefix=@PREFIX@
 exec_prefix=${prefix}
 includedir=${prefix}/include
-libdir=${exec_prefix}/lib
+libdir=@LIBDIR@
 
 Name: zstd
 Description: fast lossless compression algorithm library
