From c9c11a574fa6c10b8beb11fb87b90a6f86e006bf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 20 Jun 2019 22:30:43 +0200
Subject: [PATCH] Remove incompatible option

---
 src/Widgets/SplitPDF.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Widgets/SplitPDF.vala b/src/Widgets/SplitPDF.vala
index f5512ab..64893fc 100644
--- a/src/Widgets/SplitPDF.vala
+++ b/src/Widgets/SplitPDF.vala
@@ -369,7 +369,7 @@ namespace pdftricks {
             int exit_status = 0;
             string output_filename = output_file.replace(".pdf", "_" + label + ".pdf");
             try{
-                var cmd = "gs -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dNOPAUSE -dQUIET -dBATCH -dAutoFilterColorImages=false -dEncodeColorImages=true -dColorImageFilter=/DCTEncode -dColorConversionStrategy=/LeaveColorUnchange -dFirstPage=" + page_start.to_string() + " -dLastPage=" + page_end.to_string() + " -sOutputFile=" + output_filename +" " + input.replace(" ", "\\ ");
+                var cmd = "gs -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dNOPAUSE -dQUIET -dBATCH -dAutoFilterColorImages=false -dEncodeColorImages=true -dColorImageFilter=/DCTEncode -dFirstPage=" + page_start.to_string() + " -dLastPage=" + page_end.to_string() + " -sOutputFile=" + output_filename +" " + input.replace(" ", "\\ ");
                 Process.spawn_command_line_sync (cmd, out output, out stderr, out exit_status);
             } catch (Error e) {
                 critical (e.message);
-- 
2.22.0

