From 431c3891898a1696c6b497c633ee8d0a404a9d5d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 20 Jun 2019 20:44:15 +0200
Subject: [PATCH] Fix build with vala 0.44

---
 src/Widgets/MergePDF.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Widgets/MergePDF.vala b/src/Widgets/MergePDF.vala
index 389b0a9..0777ef8 100644
--- a/src/Widgets/MergePDF.vala
+++ b/src/Widgets/MergePDF.vala
@@ -59,7 +59,7 @@ namespace pdftricks {
             view = new Gtk.TreeView.with_model (list_store);
             view.hexpand = true;
             view.vexpand = true;
-            view.enable_model_drag_source( Gdk.BUTTON1_MASK,
+            view.enable_model_drag_source(Gdk.ModifierType.BUTTON1_MASK,
                                                 targets,
                                                 Gdk.DragAction.MOVE);
             view.enable_model_drag_dest(targets,
-- 
2.22.0

