diff --git a/digest.gemspec b/digest.gemspec
index 4a01c5f..25f8967 100644
--- a/digest.gemspec
+++ b/digest.gemspec
@@ -2,13 +2,8 @@
 # frozen_string_literal: true
 
 Gem::Specification.new do |spec|
-  version_module = Module.new do
-    version_rb = File.join(__dir__, "lib/digest/version.rb")
-    module_eval(File.read(version_rb), version_rb)
-  end
-
   spec.name          = "digest"
-  spec.version       = version_module::Digest::VERSION
+  spec.version       = "3.1.0"
   spec.authors       = ["Akinori MUSHA"]
   spec.email         = ["knu@idaemons.org"]
 
