diff --git a/plugins/onboard/onboarditem.cpp b/plugins/onboard/onboarditem.cpp
index e43a0917..a7b29345 100644
--- a/plugins/onboard/onboarditem.cpp
+++ b/plugins/onboard/onboarditem.cpp
@@ -23,6 +23,7 @@
 
 #include <QSvgRenderer>
 #include <QPainter>
+#include <QPainterPath>
 #include <QMouseEvent>
 #include <QApplication>
 #include <QIcon>
diff --git a/plugins/shutdown/shutdownwidget.cpp b/plugins/shutdown/shutdownwidget.cpp
index 8b62a4ab..a303df4e 100644
--- a/plugins/shutdown/shutdownwidget.cpp
+++ b/plugins/shutdown/shutdownwidget.cpp
@@ -23,6 +23,7 @@
 
 #include <QSvgRenderer>
 #include <QPainter>
+#include <QPainterPath>
 #include <QMouseEvent>
 #include <QApplication>
 #include <QIcon>
diff --git a/plugins/tray/fashiontray/containers/spliteranimated.cpp b/plugins/tray/fashiontray/containers/spliteranimated.cpp
index 92d4ca1c..3feb06f4 100644
--- a/plugins/tray/fashiontray/containers/spliteranimated.cpp
+++ b/plugins/tray/fashiontray/containers/spliteranimated.cpp
@@ -1,6 +1,7 @@
 #include "spliteranimated.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 
 #define OpacityMax 0.3
diff --git a/plugins/tray/fashiontray/fashiontraycontrolwidget.cpp b/plugins/tray/fashiontray/fashiontraycontrolwidget.cpp
index b7edc119..2238f559 100644
--- a/plugins/tray/fashiontray/fashiontraycontrolwidget.cpp
+++ b/plugins/tray/fashiontray/fashiontraycontrolwidget.cpp
@@ -23,6 +23,7 @@
 
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <DHiDPIHelper>
 #include <DStyle>
 #include <DGuiApplicationHelper>
diff --git a/plugins/tray/fashiontray/fashiontraywidgetwrapper.cpp b/plugins/tray/fashiontray/fashiontraywidgetwrapper.cpp
index 30e47172..833f1e1c 100644
--- a/plugins/tray/fashiontray/fashiontraywidgetwrapper.cpp
+++ b/plugins/tray/fashiontray/fashiontraywidgetwrapper.cpp
@@ -23,6 +23,7 @@
 #include "../xembedtraywidget.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 #include <QMouseEvent>
 #include <QDrag>
