diff -wbBur cdfs-2.6.27/root.c cdfs-2.6.27.my/root.c
--- cdfs-2.6.27/root.c	2009-12-21 16:38:31.000000000 +0300
+++ cdfs-2.6.27.my/root.c	2012-01-18 11:23:03.000000000 +0400
@@ -517,7 +517,7 @@
   
   i->i_uid        = this_cd->uid;
   i->i_gid        = this_cd->gid;
-  i->i_nlink      = 1;
+  i->__i_nlink      = 1;
   i->i_op         = &cdfs_inode_operations;
   i->i_fop        = NULL;
   i->i_data.a_ops = NULL;
