diff -upr cdfs-2.6.27.orig/daemon.c cdfs-2.6.27/daemon.c
--- cdfs-2.6.27.orig/daemon.c	2013-02-21 05:51:10.000000000 +0200
+++ cdfs-2.6.27/daemon.c	2013-02-21 05:56:55.000000000 +0200
@@ -158,7 +158,6 @@ int kcdfsd_thread(void *unused){
   exit_files(current);  /* daemonize doesn't do exit_files */
   daemonize();
 #else
-  daemonize("k"FSNAME"d");
 
   /* Allow SIGTERM to quit properly when removing module */
   /* By default with daemonize all signals are dropped */
