diff -up kftpgrabber-0.8.1/kftpgrabber/src/misc/libs/ssh/wrapper.c.comment kftpgrabber-0.8.1/kftpgrabber/src/misc/libs/ssh/wrapper.c
--- kftpgrabber-0.8.1/kftpgrabber/src/misc/libs/ssh/wrapper.c.comment	2007-04-28 13:35:19.000000000 +0200
+++ kftpgrabber-0.8.1/kftpgrabber/src/misc/libs/ssh/wrapper.c	2009-08-22 15:16:49.000000000 +0200
@@ -80,7 +80,7 @@ HMACCTX *hmac_init(const void *key, int 
     HMAC_CTX *ctx;
     ctx=malloc(sizeof(HMAC_CTX));
 #ifndef OLD_CRYPTO
-    HMAC_CTX_init(ctx); // openssl 0.9.7 requires it.
+    HMAC_CTX_init(ctx); /* openssl 0.9.7 requires it. */
 #endif
     switch(type){
         case HMAC_SHA1:
