From e72947c8fd50575d69cb005a52f99aeef66f4186 Mon Sep 17 00:00:00 2001
From: Marcel O'Neil <marcel@marceloneil.com>
Date: Wed, 7 Mar 2018 15:15:54 -0500
Subject: [PATCH] prevent requirements.txt from being installed to /usr

---
 MANIFEST.in | 2 ++
 setup.py    | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/MANIFEST.in b/MANIFEST.in
index 4fa5491a..b7355bd9 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -4,6 +4,8 @@ include electrum.conf.sample
 include electrum.desktop
 include *.py
 include electrum
+include contrib/requirements/requirements.txt
+include contrib/requirements/requirements-hw.txt
 recursive-include lib *.py
 recursive-include gui *.py
 recursive-include plugins *.py
diff --git a/setup.py b/setup.py
index 63581a61..030993b5 100755
--- a/setup.py
+++ b/setup.py
@@ -20,7 +20,7 @@ version = imp.load_source('version', 'lib/version.py')
 if sys.version_info[:3] < (3, 4, 0):
     sys.exit("Error: Electrum requires Python version >= 3.4.0...")
 
-data_files = ['contrib/requirements/' + r for r in ['requirements.txt', 'requirements-hw.txt']]
+data_files = []
 
 if platform.system() in ['Linux', 'FreeBSD', 'DragonFly']:
     parser = argparse.ArgumentParser()
-- 
2.16.2

