--- gajim/common/app.py	2022-10-05 16:25:51.964769181 +0200
+++ gajim/common/app.py	2022-10-05 16:26:15.344869976 +0200
@@ -233,14 +233,14 @@
         pass
 
     # UPNP
-    try:
-        gi.require_version('GUPnPIgd', '1.0')
-        from gi.repository import GUPnPIgd
-        global gupnp_igd  # pylint: disable=global-statement
-        gupnp_igd = GUPnPIgd.SimpleIgd()
-        _dependencies['UPNP'] = True
-    except ValueError:
-        pass
+#    try:
+#        gi.require_version('GUPnPIgd', '1.0')
+#        from gi.repository import GUPnPIgd
+#        global gupnp_igd  # pylint: disable=global-statement
+#        gupnp_igd = GUPnPIgd.SimpleIgd()
+#        _dependencies['UPNP'] = True
+#    except ValueError:
+#        pass
 
     # IDLE
     try:

