diff -Nawurp a/fs-uae-launcher b/fs-uae-launcher
--- a/fs-uae-launcher	2019-06-21 16:49:42.000000000 +0000
+++ b/fs-uae-launcher	2019-06-29 13:38:13.351496262 +0000
@@ -8,6 +8,8 @@ if sys.version_info[0] < 3 or sys.versio
 
 if os.environ.get("FSGS_PYTHONPATH", ""):
     sys.path.insert(0, os.environ.get("FSGS_PYTHONPATH"))
+else:
+    sys.path.insert(0, "/usr/share/fs-uae-launcher")
 
 if os.getcwd().startswith("C:\\msys64\\home\\"):
     os.environ["PATH"] = "C:\\msys64\\mingw64\\bin;" + os.environ["PATH"]
