diff -aur dnf-4.10.0/dnf/base.py dnf-4.10.0.new/dnf/base.py
--- dnf-4.10.0/dnf/base.py	2021-10-21 08:31:48.000000000 +0200
+++ dnf-4.10.0.new/dnf/base.py	2021-12-20 11:22:45.127917712 +0100
@@ -156,6 +156,8 @@
         if 'releasever' not in subst:
             subst['releasever'] = \
                 dnf.rpm.detect_releasever(conf.installroot)
+        if subst['releasever'] is None:
+            subst['releasever'] = '36'
         return conf
 
     def _setup_modular_excludes(self):
