diff -Naur spyder-3.2.0.orig/spyder/plugins/editor.py spyder-3.2.0/spyder/plugins/editor.py
--- spyder-3.2.0.orig/spyder/plugins/editor.py	2017-07-24 05:03:40.000000000 +0200
+++ spyder-3.2.0/spyder/plugins/editor.py	2017-08-12 21:45:09.455757449 +0200
@@ -56,7 +56,7 @@
 # Dependencies
 NBCONVERT_REQVER = ">=4.0"
 dependencies.add("nbconvert", _("Manipulate Jupyter notebooks on the Editor"),
-                 required_version=NBCONVERT_REQVER)
+                 required_version=NBCONVERT_REQVER, optional=True)
 
 
 def _load_all_breakpoints():
