diff -Naur spyder-3.1.4.orig/scripts/spyder3.desktop spyder-3.1.4/scripts/spyder3.desktop
--- spyder-3.1.4.orig/scripts/spyder3.desktop	2017-04-24 02:30:23.000000000 +0200
+++ spyder-3.1.4/scripts/spyder3.desktop	2017-08-12 20:49:55.349203674 +0200
@@ -1,9 +1,8 @@
 [Desktop Entry]
 Version=1.0
 Type=Application
-Name=Spyder3
-GenericName=Spyder3
-Comment=Scientific PYthon Development EnviRonment - Python3
+Name=Spyder for Python 3.x
+Comment=Scientific PYthon Development EnviRonment - Python 3.x
 TryExec=spyder3
 Exec=spyder3 %F
 Categories=Development;Science;IDE;Qt;
diff -Naur spyder-3.1.4.orig/scripts/spyder.desktop spyder-3.1.4/scripts/spyder.desktop
--- spyder-3.1.4.orig/scripts/spyder.desktop	2017-04-24 02:30:23.000000000 +0200
+++ spyder-3.1.4/scripts/spyder.desktop	2017-08-12 20:49:54.005881501 +0200
@@ -1,13 +1,12 @@
 [Desktop Entry]
 Version=1.0
 Type=Application
-Name=Spyder
-GenericName=Spyder
-Comment=Scientific PYthon Development EnviRonment
-TryExec=spyder
-Exec=spyder %F
+Name=Spyder for Python 2.x
+Comment=Scientific PYthon Development EnviRonment - Python 2.x
+TryExec=spyder2
+Exec=spyder2 %F
 Categories=Development;Science;IDE;Qt;
-Icon=spyder
+Icon=spyder2
 Terminal=false
 StartupNotify=true
 MimeType=text/x-python;
diff -Naur spyder-3.1.4.orig/setup.py spyder-3.1.4/setup.py
--- spyder-3.1.4.orig/setup.py	2017-04-24 02:30:23.000000000 +0200
+++ spyder-3.1.4/setup.py	2017-08-12 20:34:09.464160097 +0200
@@ -82,8 +82,8 @@
                           ('share/pixmaps', ['img_src/spyder3.png']),
                           ('share/metainfo', ['scripts/spyder3.appdata.xml'])]
         else:
-            data_files = [('share/applications', ['scripts/spyder.desktop']),
-                          ('share/pixmaps', ['img_src/spyder.png'])]
+            data_files = [('share/applications', ['scripts/spyder2.desktop']),
+                          ('share/pixmaps', ['img_src/spyder2.png'])]
     elif os.name == 'nt':
         data_files = [('scripts', ['img_src/spyder.ico',
                                    'img_src/spyder_reset.ico'])]
