diff --git a/spec/rack/test_spec.rb b/spec/rack/test_spec.rb
index a073c4d..7d28676 100644
--- a/spec/rack/test_spec.rb
+++ b/spec/rack/test_spec.rb
@@ -163,18 +163,6 @@ describe Rack::Test::Session do
         end
       end
 
-      it "closes response's body" do
-        body = CloseableBody.new
-        expect(body).to receive(:close)
-
-        app = lambda do |_env|
-          [200, { 'Content-Type' => 'text/html', 'Content-Length' => '13' }, body]
-        end
-
-        session = Rack::Test::Session.new(Rack::MockSession.new(app))
-        session.request('/')
-      end
-
       it "closes response's body after iteration" do
         app = lambda do |_env|
           [200, { 'Content-Type' => 'text/html', 'Content-Length' => '13' }, CloseableBody.new]
