diff -aur django-debug-toolbar-4.0.0/pyproject.toml django-debug-toolbar-4.0.0.new/pyproject.toml
--- django-debug-toolbar-4.0.0/pyproject.toml	2023-04-03 16:43:42.000000000 +0200
+++ django-debug-toolbar-4.0.0.new/pyproject.toml	2023-04-15 15:57:22.989915600 +0200
@@ -16,11 +16,6 @@
 classifiers = [
     "Development Status :: 5 - Production/Stable",
     "Environment :: Web Environment",
-    "Framework :: Django",
-    "Framework :: Django :: 3.2",
-    "Framework :: Django :: 4.0",
-    "Framework :: Django :: 4.1",
-    "Framework :: Django :: 4.2",
     "Intended Audience :: Developers",
     "License :: OSI Approved :: BSD License",
     "Operating System :: OS Independent",
