diff -rupN firewalld-0.5.0.orig/config/firewalld.service.in firewalld-0.5.0/config/firewalld.service.in
--- firewalld-0.5.0.orig/config/firewalld.service.in	2018-01-25 22:14:21.796600916 +0100
+++ firewalld-0.5.0/config/firewalld.service.in	2018-01-25 22:15:29.448820083 +0100
@@ -8,7 +8,7 @@ Conflicts=iptables.service ip6tables.ser
 Documentation=man:firewalld(1)
 
 [Service]
-EnvironmentFile=-/etc/sysconfig/firewalld
+EnvironmentFile=-/etc/conf.d/firewalld
 ExecStart=@sbindir@/firewalld --nofork --nopid $FIREWALLD_ARGS
 ExecReload=/bin/kill -HUP $MAINPID
 # supress to log debug and error output also to /var/log/messages
diff -rupN firewalld-0.5.0.orig/config/Makefile.am firewalld-0.5.0/config/Makefile.am
--- firewalld-0.5.0.orig/config/Makefile.am	2018-01-25 22:14:21.796600916 +0100
+++ firewalld-0.5.0/config/Makefile.am	2018-01-25 22:17:25.497003199 +0100
@@ -308,12 +308,12 @@ firewalld-sysctls.conf: firewalld-sysctl
 	$(edit) $< >$@
 
 install-sysconfig:
-	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/sysconfig
-	$(INSTALL_DATA) $(srcdir)/firewalld.sysconfig $(DESTDIR)$(sysconfdir)/sysconfig/firewalld
+	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/conf.d
+	$(INSTALL_DATA) $(srcdir)/firewalld.sysconfig $(DESTDIR)$(sysconfdir)/conf.d/firewalld
 
 uninstall-sysconfig:
-	rm -f $(DESTDIR)$(sysconfdir)/sysconfig/firewalld
-	rmdir $(DESTDIR)$(sysconfdir)/sysconfig || :
+	rm -f $(DESTDIR)$(sysconfdir)/conf.d/firewalld
+	rmdir $(DESTDIR)$(sysconfdir)/conf.d || :
 
 install-rpmmacros:
 	$(MKDIR_P) $(DESTDIR)$(prefix)/lib/rpm/macros.d
diff -rupN firewalld-0.5.0.orig/configure.ac firewalld-0.5.0/configure.ac
--- firewalld-0.5.0.orig/configure.ac	2018-01-25 22:14:21.796600916 +0100
+++ firewalld-0.5.0/configure.ac	2018-01-25 22:18:20.346179951 +0100
@@ -71,7 +71,7 @@ AC_SUBST(BASHCOMPLETIONDIR)
 
 AC_ARG_WITH([ifcfgdir],
        AS_HELP_STRING([--with-ifcfgdir=DIR], [The ifcfg configuration directory]),
-       [IFCFGDIR=$withval], [IFCFGDIR="/etc/sysconfig/network-scripts"])
+       [IFCFGDIR=$withval], [IFCFGDIR="/etc/conf.d/network-scripts"])
 AC_SUBST(IFCFGDIR)
 
 # Extend PATH to include /sbin etc in case we are building as non-root
diff -rupN firewalld-0.5.0.orig/doc/xml/firewall-offline-cmd.xml firewalld-0.5.0/doc/xml/firewall-offline-cmd.xml
--- firewalld-0.5.0.orig/doc/xml/firewall-offline-cmd.xml	2018-01-25 22:14:21.799934194 +0100
+++ firewalld-0.5.0/doc/xml/firewall-offline-cmd.xml	2018-01-25 22:18:43.412506403 +0100
@@ -68,7 +68,7 @@
   <refsect1 id="options">
     <title>Options</title>
     <para>
-      If no options are given, configuration from <command>/etc/sysconfig/system-config-firewall</command> will be migrated.
+      If no options are given, configuration from <command>/etc/conf.d/system-config-firewall</command> will be migrated.
     </para>
 
     <para>
diff -rupN firewalld-0.5.0.orig/src/firewall/config/__init__.py.in firewalld-0.5.0/src/firewall/config/__init__.py.in
--- firewalld-0.5.0.orig/src/firewall/config/__init__.py.in	2018-01-25 22:14:21.803267471 +0100
+++ firewalld-0.5.0/src/firewall/config/__init__.py.in	2018-01-25 22:19:44.971614279 +0100
@@ -96,7 +96,7 @@ FIREWALLD_PIDFILE = "/var/run/firewalld.
 
 FIREWALLD_TEMPDIR = '/run/firewalld'
 
-SYSCONFIGDIR = '/etc/sysconfig'
+SYSCONFIGDIR = '/etc/conf.d'
 IFCFGDIR = "@IFCFGDIR@"
 
 SYSCTL_CONFIG = '/etc/sysctl.conf'
diff -rupN firewalld-0.5.0.orig/src/tests/legacy/firewall-offline-cmd_test.sh firewalld-0.5.0/src/tests/legacy/firewall-offline-cmd_test.sh
--- firewalld-0.5.0.orig/src/tests/legacy/firewall-offline-cmd_test.sh	2018-01-25 22:14:21.806600748 +0100
+++ firewalld-0.5.0/src/tests/legacy/firewall-offline-cmd_test.sh	2018-01-25 22:19:19.201985041 +0100
@@ -218,7 +218,7 @@ fw_port2="port=333:proto=udp:toport=444"
 lokkit_opts="--enabled --addmodule=abc --addmodule=efg --removemodule=xyz
  --trust=${trusted_iface1} --trust=${trusted_iface2}
  --masq=tun+ --masq=tap+ --port=7:tcp --port=666:udp
- --custom-rules=ipv4:mangle:/etc/sysconfig/ebtables-config
+ --custom-rules=ipv4:mangle:/etc/conf.d/ebtables-config
  --service=${service1} --service=${service2} --remove-service=${service3}
  --block-icmp=${icmp1} --block-icmp=${icmp2}
  --forward-port=if=ippp+:${fw_port1}
@@ -236,7 +236,7 @@ cat << EOF > ./system-config-firewall
 --masq=tap+
 --port=7:tcp
 --port=666:udp
---custom-rules=ipv4:mangle:/etc/sysconfig/ebtables-config
+--custom-rules=ipv4:mangle:/etc/conf.d/ebtables-config
 --service=${service1}
 --service=${service2}
 --remove-service=${service3}
