Only in a: dist
diff -upr a/Makefile b/Makefile
--- a/Makefile	2022-03-21 09:45:51.000000000 +0100
+++ b/Makefile	2022-03-30 16:15:10.079243904 +0200
@@ -44,7 +44,7 @@ ifdef lib
 endif

 define list_modules
-	`ls -1 | sed -n '/egg-info/n; /pyroute2/p'`
+	`find . -maxdepth 1 -mindepth 1 -type d -iname 'pyroute2*' -printf '%f\n'`
 endef

 define make_modules
@@ -57,7 +57,7 @@ endef

 define clean_module
 	if [ -f $$module/setup.json ]; then \
-		for i in `ls -1 templates`; do rm -f $$module/$$i; done; \
+		for i in `find templates -maxdepth 1 -mindepth 1 -type f -printf '%f\n'`; do rm -f $$module/$$i; done; \
 	fi; \
 	rm -f $$module/LICENSE.*; \
 	rm -f $$module/README.license.md; \
@@ -71,7 +71,7 @@ endef
 define process_templates
 	for module in $(call list_modules); do \
 		if [ -f $$module/setup.json ]; then \
-			for template in `ls -1 templates`; do \
+			for template in `find templates -maxdepth 1 -mindepth 1 -type f -printf '%f\n'`; do \
 				${python} \
 					util/process_template.py \
 					templates/$$template \
@@ -207,6 +207,8 @@ upload: dist
 	${python} -m twine upload dist/*

 setup:
+	$(MAKE) clean
+	$(MAKE) VERSION
 	$(call process_templates)
 	@for module in $(call list_modules); do $(call deploy_license); done
 	@for module in pyroute2 pyroute2.minimal; do \
@@ -217,7 +219,7 @@ setup:
 			$$module/setup.cfg ; \
 	done

-dist: clean VERSION setup
+dist: setup
 	cd pyroute2; ${python} setup.py sdist
 	mkdir dist
 	$(call make_modules, dist)
@@ -231,7 +233,7 @@ install: dist
 install-minimal: dist
 	${python} -m pip install dist/pyroute2.minimal* dist/pyroute2.core* ${root}

-uninstall: clean VERSION setup
+uninstall: setup
 	$(call make_modules, uninstall)

 audit-imports:
