diff -ruN gprename-20140325/Makefile gprename-20140325-orig/Makefile
--- gprename-20140325/Makefile	2014-03-24 22:06:04.000000000 +0100
+++ gprename-20140325-orig/Makefile	2014-06-28 16:05:47.453008829 +0200
@@ -26,7 +26,7 @@
 	perl -ne 's!\@INSTALLDIR\@!$(PREFIX)!g ; print' < bin/gprename > build/gprename
 	perl -ne 's!\@INSTALLDIR\@!$(PREFIX)!g ; print' < bin/gprename.desktop > build/gprename.desktop
 
-install: uninstall build
+install: build
 	# Compile all .po files to .mo
 	msgfmt -o build/locale/ca.mo	locale/ca.po
 	msgfmt -o build/locale/de.mo	locale/de.po
@@ -45,8 +45,8 @@
 	# Create directories
 	install -d "$(DESTDIR)"
 	install -d "$(DESTDIR)/bin"
-	install -d "/usr/share"
-	install -d "/usr/share/icons"
+	install -d "$(DESTDIR)/share"
+	install -d "$(DESTDIR)/share/icons"
 	install -d "$(DESTDIR)/share/applications"
 	install -d "$(DESTDIR)/share/man"
 	install -d "$(DESTDIR)/share/man/man1"
@@ -67,7 +67,7 @@
 	# Copy all files in the filesystem
 	install -m 755 build/gprename                "$(DESTDIR)/bin/"
 	install -m 644 gprename-nautilus-actions.xml "$(DESTDIR)/share/applications/"
-	install -m 644 icon/gprename.png             "/usr/share/icons/"
+	install -m 644 icon/gprename.png             "$(DESTDIR)/share/icons/"
 	install -m 644 man/gprename.1                "$(DESTDIR)/share/man/man1/"
 	install -m 644 build/locale/ca.mo            "$(DESTDIR)/share/locale/ca/LC_MESSAGES/gprename.mo"
 	install -m 644 build/locale/de.mo            "$(DESTDIR)/share/locale/de/LC_MESSAGES/gprename.mo"
@@ -82,8 +82,8 @@
 	install -m 644 build/locale/ru.mo            "$(DESTDIR)/share/locale/ru/LC_MESSAGES/gprename.mo"
 	install -m 644 build/locale/sv.mo            "$(DESTDIR)/share/locale/sv/LC_MESSAGES/gprename.mo"
 	install -m 644 build/locale/zh_CN.mo         "$(DESTDIR)/share/locale/zh_CN/LC_MESSAGES/gprename.mo"
-	desktop-file-install bin/gprename.desktop
-	update-desktop-database
+	install -m 644 bin/gprename.desktop          "$(DESTDIR)/share/applications/gprename.desktop"
+#	update-desktop-database
 	@echo "Installation completed."
 
 uninstall: clean
