diff --git c/example_project/settings.py w/example_project/settings.py
index 1c965f5..70de5d4 100644
--- c/example_project/settings.py
+++ w/example_project/settings.py
@@ -75,7 +75,7 @@ INSTALLED_APPS = (
     'allauth.socialaccount.providers.github',
     'allauth.socialaccount.providers.gitlab',
     'allauth.socialaccount.providers.google',
-    'allauth.socialaccount.providers.facebook',
+    # 'allauth.socialaccount.providers.facebook',
     'allauth.socialaccount.providers.twitter',
     'allauth.socialaccount.providers.stackexchange',
 )
@@ -129,7 +129,7 @@ DATABASES = {
         # Use 'sqlite3', 'postgresql_psycopg2', 'mysql', 'sqlite3' or 'oracle'.
         'ENGINE': 'django.db.backends.sqlite3',
         # DB name or path to database file if using sqlite3.
-        'NAME': os.path.join(BASE_DIR, 'hyperkitty.db'),
+        'NAME': '/var/lib/hyperkitty/data/hyperkitty.db',
         # The following settings are not used with sqlite3:
         'USER': 'hyperkitty',
         'PASSWORD': 'hkpass',
@@ -195,7 +195,7 @@ USE_TZ = True
 # Don't put anything in this directory yourself; store your static files
 # in apps' "static/" subdirectories and in STATICFILES_DIRS.
 # Example: "/var/www/example.com/static/"
-STATIC_ROOT = os.path.join(BASE_DIR, 'static')
+STATIC_ROOT = '/var/lib/hyperkitty/static'
 
 # URL prefix for static files.
 # Example: "http://example.com/static/", "http://static.example.com/"
@@ -321,7 +321,8 @@ COMPRESS_PRECOMPILERS = (
 # recompiled on each requests. It means running an additional "compress"
 # management command after each code upgrade.
 # http://django-compressor.readthedocs.io/en/latest/usage/#offline-compression
-# COMPRESS_OFFLINE = True
+COMPRESS_OFFLINE = True
+COMPRESS_ENABLED = True
 
 # Needed for debug mode
 # INTERNAL_IPS = ('127.0.0.1',)
@@ -333,7 +334,7 @@ COMPRESS_PRECOMPILERS = (
 HAYSTACK_CONNECTIONS = {
     'default': {
         'ENGINE': 'haystack.backends.whoosh_backend.WhooshEngine',
-        'PATH': os.path.join(BASE_DIR, "fulltext_index"),
+        'PATH': "/var/lib/hyperkitty/fulltext_index",
         # You can also use the Xapian engine, it's faster and more accurate,
         # but requires another library.
         # http://django-haystack.readthedocs.io/en/v2.4.1/installing_search_engines.html#xapian
@@ -388,7 +389,7 @@ LOGGING = {
             'level': 'INFO',
             #'class': 'logging.handlers.RotatingFileHandler',
             'class': 'logging.handlers.WatchedFileHandler',
-            'filename': os.path.join(BASE_DIR, 'hyperkitty.log'),
+            'filename': '/var/log/hyperkitty/hyperkitty.log',
             'formatter': 'verbose',
         },
     },
@@ -439,7 +440,7 @@ LOGGING = {
 # while you're fiddling with the code.
 if DEBUG == True:
     EMAIL_BACKEND = 'django.core.mail.backends.filebased.EmailBackend'
-    EMAIL_FILE_PATH = os.path.join(BASE_DIR, 'emails')
+    EMAIL_FILE_PATH = '/var/lib/hyperkitty/emails'
 
 
 #
