diff -ur --ignore-all-space synapse-1.49.2/synapse/python_dependencies.py synapse-1.49.2-new/synapse/python_dependencies.py
--- synapse-1.49.2/synapse/python_dependencies.py	2021-12-21 20:41:34.000000000 +0300
+++ synapse-1.49.2-new/synapse/python_dependencies.py	2021-12-22 00:56:31.359112226 +0300
@@ -50,8 +50,7 @@
 REQUIREMENTS = [
     # we use the TYPE_CHECKER.redefine method added in jsonschema 3.0.0
     "jsonschema>=3.0.0",
-    # frozendict 2.1.2 is broken on Debian 10: https://github.com/Marco-Sulla/python-frozendict/issues/41
-    "frozendict>=1,<2.1.2",
+    "frozendict>=1",
     "unpaddedbase64>=1.1.0",
     "canonicaljson>=1.4.0",
     # we use the type definitions added in signedjson 1.1.
