diff -aur --color wakatime-13.0.7-old/wakatime/packages/tzlocal/unix.py wakatime-13.0.7-new/wakatime/packages/tzlocal/unix.py
--- wakatime-13.0.7-old/wakatime/packages/tzlocal/unix.py	2020-02-13 19:11:46.000000000 +0200
+++ wakatime-13.0.7-new/wakatime/packages/tzlocal/unix.py	2020-04-21 16:25:15.100231033 +0300
@@ -97,7 +97,7 @@
     if os.path.exists(tzpath) and os.path.islink(tzpath):
         tzpath = os.path.realpath(tzpath)
         start = tzpath.find("/")+1
-        while start is not 0:
+        while start != 0:
             tzpath = tzpath[start:]
             try:
                 return pytz.timezone(tzpath)
