--- a/htdocs/login.php	2020-06-17 18:17:18.926320916 -0600
+++ b/htdocs/login.php	2020-06-17 18:16:59.796222429 -0600
@@ -14,6 +14,7 @@
     $user = array();
     $user['login'] = get_request('login');
     $user['password'] = get_request('login_pass');
+    $user['password'] = str_replace('&amp;', '&', $user['password']);
 
     if ($user['login'] && ! strlen($user['password']))
         system_message(array(
