--- src/sage/env.py.orig	2014-06-25 10:47:25.827203554 +1200
+++ src/sage/env.py	2014-06-25 10:49:07.667402821 +1200
@@ -90,8 +90,8 @@
 
 # bunch of sage directories and files
 _add_variable_or_fallback('SAGE_ROOT',       None)
-_add_variable_or_fallback('SAGE_LOCAL',      opj('$SAGE_ROOT', 'local'))
-_add_variable_or_fallback('SAGE_ETC',        opj('$SAGE_LOCAL', 'etc'))
+_add_variable_or_fallback('SAGE_LOCAL',      '/usr')
+_add_variable_or_fallback('SAGE_ETC',        '/etc/sage')
 _add_variable_or_fallback('SAGE_INC',        opj('$SAGE_LOCAL', 'include'))
 _add_variable_or_fallback('SAGE_SHARE',      opj('$SAGE_LOCAL', 'share'))
 
