diff --git a/config.php-dist b/config.php-dist
index eed1a699d..c18886390 100644
--- a/config.php-dist
+++ b/config.php-dist
@@ -44,7 +44,7 @@
 	// *** Files and directories ***
 	// *****************************
 
-	define('PHP_EXECUTABLE', '/usr/bin/php');
+	define('PHP_EXECUTABLE', '/usr/bin/php7');
 	// Path to PHP *COMMAND LINE* executable, used for various command-line tt-rss
 	// programs and update daemon. Do not try to use CGI binary here, it won't work.
 	// If you see HTTP headers being displayed while running tt-rss scripts,
diff --git a/update.php b/update.php
index 2737b0456..721696d3a 100755
--- a/update.php
+++ b/update.php
@@ -73,7 +73,7 @@
 	}
 
 	if (!defined('PHP_EXECUTABLE'))
-		define('PHP_EXECUTABLE', '/usr/bin/php');
+		define('PHP_EXECUTABLE', '/usr/bin/php7');
 
 	$pdo = Db::pdo();
 
