diff --git a/piep/main.py b/piep/main.py
index f3b1b58..4b18cfc 100755
--- a/piep/main.py
+++ b/piep/main.py
@@ -127,7 +127,7 @@ def init_globals(opts, input_file):
 			sys.path.insert(0, path)
 	for import_mod in opts.imports:
 		import_node = ast.Import(names=[ast.alias(name=import_mod, asname=None)])
-		code = compile(ast.fix_missing_locations(ast.Module(body=[import_node])), 'import %s' % (import_mod,), 'exec')
+		code = compile(ast.fix_missing_locations(ast.Module(body=[import_node], type_ignores=[])), 'import %s' % (import_mod,), 'exec')
 		eval(code, globs)
 	for eval_str in opts.evals:
 		try:
@@ -319,6 +319,7 @@ def compile_pipe_exprs(exprs):
 		transform_def = ast.FunctionDef(
 			name='_transformer',
 			args=ast.arguments(
+				posonlyargs=[],
 				args=[arg('p'), arg('i')],
 				vararg=None,
 				kwarg=None,
@@ -376,7 +377,7 @@ def compile_pipe_exprs(exprs):
 				body.append(expr)
 				body.extend(post_pipe_check)
 	
-	mod = ast.Module(body=body)
+	mod = ast.Module(body=body, type_ignores=[])
 	ast.fix_missing_locations(mod)
 
 	#raise RuntimeError(ast.dump(mod))
diff --git a/test/test_statements.py b/test/test_statements.py
index accbd28..1bff5d3 100644
--- a/test/test_statements.py
+++ b/test/test_statements.py
@@ -4,12 +4,12 @@ from test.test_helper import run
 class TestEval(TestCase):
 	def test_import_assignment(self):
 		self.assertEqual(
-				run('--eval=from cgi import escape as quote', 'quote(p)', ['a&b']),
+				run('--eval=from html import escape as quote', 'quote(p)', ['a&b']),
 				['a&amp;b'])
 
 	def test_import_shorthand(self):
 		self.assertEqual(
-				run('--import=cgi', 'cgi.escape(p)', ['a&b']),
+				run('--import=html', 'html.escape(p)', ['a&b']),
 				['a&amp;b'])
 
 	def test_globals(self):
