diff -wbBur lire-2.1.1/all/lib/Lire/Config/ConfigFile.pm lire-2.1.1.my/all/lib/Lire/Config/ConfigFile.pm
--- lire-2.1.1/all/lib/Lire/Config/ConfigFile.pm	2006-07-23 17:16:30.000000000 +0400
+++ lire-2.1.1.my/all/lib/Lire/Config/ConfigFile.pm	2014-02-14 13:46:08.830257062 +0400
@@ -39,8 +39,6 @@
 
 The Lire::Config::TypeSpec object to which this configuration adheres.
 
-=end
-
 =cut
 
 sub new {
@@ -61,6 +59,8 @@
 
 =pod
 
+=back
+
 =head2 filename( [$new_filename] )
 
 Returns (and optionally changes) the filename where this configuration
diff -wbBur lire-2.1.1/all/lib/Lire/DlfStore.pm lire-2.1.1.my/all/lib/Lire/DlfStore.pm
--- lire-2.1.1/all/lib/Lire/DlfStore.pm	2011-12-31 17:08:51.000000000 +0400
+++ lire-2.1.1.my/all/lib/Lire/DlfStore.pm	2014-02-14 13:46:28.903590052 +0400
@@ -818,6 +818,8 @@
 
 =pod
 
+=back
+
 =head1 SEE ALSO
 
 Lire::DlfConverter(3pm)
diff -wbBur lire-2.1.1/all/lib/Lire/OutputJob.pm lire-2.1.1.my/all/lib/Lire/OutputJob.pm
--- lire-2.1.1/all/lib/Lire/OutputJob.pm	2006-07-23 17:16:29.000000000 +0400
+++ lire-2.1.1.my/all/lib/Lire/OutputJob.pm	2014-02-14 13:46:50.500256739 +0400
@@ -171,8 +171,6 @@
 An additional text file that will be appended to text report or attach
 to other report types.
 
-=item
-
 =back
 
 =head2 emails()
diff -wbBur lire-2.1.1/all/lib/Lire/Plugin.pm lire-2.1.1.my/all/lib/Lire/Plugin.pm
--- lire-2.1.1/all/lib/Lire/Plugin.pm	2006-07-23 17:16:29.000000000 +0400
+++ lire-2.1.1.my/all/lib/Lire/Plugin.pm	2014-02-14 13:47:04.733589845 +0400
@@ -105,6 +105,8 @@
 
 =pod
 
+=back
+
 =head1 SEE ALSO
 
 Lire::PluginManager(3pm), Lire::DlfConverter(3pm),
diff -wbBur lire-2.1.1/all/lib/Lire/Report/Subreport.pm lire-2.1.1.my/all/lib/Lire/Report/Subreport.pm
--- lire-2.1.1/all/lib/Lire/Report/Subreport.pm	2006-07-23 17:16:31.000000000 +0400
+++ lire-2.1.1.my/all/lib/Lire/Report/Subreport.pm	2014-02-14 13:44:26.350257994 +0400
@@ -203,8 +203,6 @@
     return $self->{'_schemas'};
 }
 
-=cut
-
 =pod
 
 =head2 field_schema( $field )
diff -wbBur lire-2.1.1/all/lib/Lire/ReportParser/LaTeXWriter.pm lire-2.1.1.my/all/lib/Lire/ReportParser/LaTeXWriter.pm
--- lire-2.1.1/all/lib/Lire/ReportParser/LaTeXWriter.pm	2006-07-29 21:57:34.000000000 +0400
+++ lire-2.1.1.my/all/lib/Lire/ReportParser/LaTeXWriter.pm	2014-02-14 13:44:57.700258016 +0400
@@ -450,6 +450,8 @@
 
 =pod
 
+=back
+
 =head1 SEE ALSO
 
 Lire::ReportParser::LaTeXDocBookFormatter(3pm)
diff -wbBur lire-2.1.1/all/lib/Lire/ReportParser.pm lire-2.1.1.my/all/lib/Lire/ReportParser.pm
--- lire-2.1.1/all/lib/Lire/ReportParser.pm	2006-07-23 17:16:29.000000000 +0400
+++ lire-2.1.1.my/all/lib/Lire/ReportParser.pm	2014-02-14 13:47:23.406922964 +0400
@@ -135,6 +135,8 @@
 specific or higher ones and can be used to hook before the other events
 are synthesized.
 
+=back
+
 =head1 HIGH-LEVEL EVENT METHODS
 
 For each element defined, an I<element_name>_start() and an
diff -wbBur lire-2.1.1/all/script/lr_env.in lire-2.1.1.my/all/script/lr_env.in
--- lire-2.1.1/all/script/lr_env.in	2006-07-23 17:16:32.000000000 +0400
+++ lire-2.1.1.my/all/script/lr_env.in	2014-02-14 13:41:55.000260120 +0400
@@ -103,6 +103,8 @@
 
 The directory where platform independant perl modules are installed.
 
+=back
+
 =head1 VERSION
 
 $Id: lr_env.in,v 1.7 2006/07/23 13:16:32 vanbaal Exp $
diff -wbBur lire-2.1.1/email/script/s1ms2dlf.in lire-2.1.1.my/email/script/s1ms2dlf.in
--- lire-2.1.1/email/script/s1ms2dlf.in	2006-07-23 17:16:34.000000000 +0400
+++ lire-2.1.1.my/email/script/s1ms2dlf.in	2014-02-14 13:47:41.230256107 +0400
@@ -65,6 +65,7 @@
 
  TO BE WRITTEN
 
+=back
 
 =head1 EXAMPLES
 
