diff --git c/example_project/settings.py w/example_project/settings.py
index c54af7e..d05535b 100644
--- c/example_project/settings.py
+++ w/example_project/settings.py
@@ -136,7 +136,7 @@ WSGI_APPLICATION = 'wsgi.application'
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'postorius.db'),
+        'NAME': '/var/lib/postorius/data/postorius.db',
     }
 }
 
@@ -180,7 +180,7 @@ USE_TZ = True
 # Don't put anything in this directory yourself; store your static files
 # in apps' "static/" subdirectories and in STATICFILES_DIRS.
 # Example: "/var/www/example.com/static/"
-STATIC_ROOT = os.path.join(BASE_DIR, 'static')
+STATIC_ROOT = '/var/lib/postorius/static'
 
 # URL prefix for static files.
 # Example: "http://example.com/static/", "http://static.example.com/"
@@ -259,7 +259,7 @@ LOGGING = {
             'level': 'INFO',
             #'class': 'logging.handlers.RotatingFileHandler',
             'class': 'logging.handlers.WatchedFileHandler',
-            'filename': os.path.join(BASE_DIR, 'logs', 'postorius.log'),
+            'filename': '/var/log/postorius/postorius.log',
             'formatter': 'verbose',
         },
     },
