diff -u -r treq-release-18.6.0/src/treq/test/test_multipart.py treq-release-18.6.0-py3.7/src/treq/test/test_multipart.py
--- treq-release-18.6.0/src/treq/test/test_multipart.py	2018-06-29 07:26:36.000000000 +0200
+++ treq-release-18.6.0-py3.7/src/treq/test/test_multipart.py	2018-07-29 18:37:15.842395815 +0200
@@ -594,8 +594,11 @@
             )
         )
 
-        form = cgi.parse_multipart(BytesIO(output), {"boundary": b"heyDavid"})
-        self.assertEqual(set([b'just a string\r\n', b'another string']),
+        form = cgi.parse_multipart(BytesIO(output), {
+            "boundary": b"heyDavid",
+            "CONTENT-LENGTH": str(len(output)),
+        })
+        self.assertEqual(set(['just a string\r\n', 'another string']),
                          set(form['cfield']))
 
         self.assertEqual(set([b'my lovely bytes2']), set(form['efield']))
