From 938e5d933aee963a6b28d6cf9e34e0634c8c4a29 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Chv=C3=A1tal?= <tomas.chvatal@gmail.com>
Date: Fri, 26 Jul 2019 01:20:41 +0200
Subject: [PATCH] Use new pytest api to keep building with pytest 5 (#353)

diff --git a/tests/test__default.py b/tests/test__default.py
index 3fb0fa1..d143479 100644
--- a/tests/test__default.py
+++ b/tests/test__default.py
@@ -92,7 +92,7 @@ def test__load_credentials_from_file_authorized_user_bad_format(tmpdir):
 
 
 def test__load_credentials_from_file_authorized_user_cloud_sdk():
-    with pytest.warns(UserWarning, matches='Cloud SDK'):
+    with pytest.warns(UserWarning, match='Cloud SDK'):
         credentials, project_id = _default._load_credentials_from_file(
             AUTHORIZED_USER_CLOUD_SDK_FILE)
     assert isinstance(credentials, google.oauth2.credentials.Credentials)
