From 8277b4232617c0433f80e9c2844452b9fae67a65 Mon Sep 17 00:00:00 2001
From: Fabian Affolter <mail@fabian-affolter.ch>
Date: Wed, 2 Nov 2022 08:22:03 +0100
Subject: [PATCH] Switch to poetry-core

---
 pyproject.toml | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index dff94b0..f4d5fce 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -39,7 +39,8 @@ favicon = "art/logo.png"
 logo = "art/logo.png"
 name = "material"
 palette = {primary = "teal", accent = "cyan"}
+
 [build-system]
-requires = ["poetry>=0.12"]
-build-backend = "poetry.masonry.api"
+requires = ["poetry-core>=1.0.0"]
+build-backend = "poetry.core.masonry.api"
 
