diff -Naur spyder-3.1.1.orig/spyder/plugins/editor.py spyder-3.1.1.pep8-python2/spyder/plugins/editor.py
--- spyder-3.1.1.orig/spyder/plugins/editor.py	2017-01-22 18:28:03.000000000 +0100
+++ spyder-3.1.1.pep8-python2/spyder/plugins/editor.py	2017-01-22 23:40:42.426230374 +0100
@@ -225,7 +225,7 @@
         pep8_label = QLabel(_("<i>(Refer to the {} page)</i>").format(pep_url))
         pep8_label.setOpenExternalLinks(True)
         is_pyflakes = codeanalysis.is_pyflakes_installed()
-        is_pep8 = codeanalysis.get_checker_executable('pep8') is not None
+        is_pep8 = codeanalysis.get_checker_executable('pycodestyle') is not None
         pyflakes_box = newcb(_("Real-time code analysis"),
                       'code_analysis/pyflakes', default=True,
                       tip=_("<p>If enabled, Python source code will be analyzed "
diff -Naur spyder-3.1.1.orig/spyder/utils/codeanalysis.py spyder-3.1.1.pep8-python2/spyder/utils/codeanalysis.py
--- spyder-3.1.1.orig/spyder/utils/codeanalysis.py	2017-01-22 18:28:03.000000000 +0100
+++ spyder-3.1.1.pep8-python2/spyder/utils/codeanalysis.py	2017-01-22 23:44:40.107465175 +0100
@@ -92,7 +92,7 @@
                  required_version=PYFLAKES_REQVER)
 
 PEP8_REQVER = '>=0.6'
-dependencies.add("pep8", _("Real-time code style analysis on the Editor"),
+dependencies.add("pycodestyle", _("Real-time code style analysis on the Editor"),
                  required_version=PEP8_REQVER)
 
 
@@ -167,7 +167,7 @@
 def check_with_pep8(source_code, filename=None):
     """Check source code with pep8"""
     try:
-        args = get_checker_executable('pep8')
+        args = get_checker_executable('pycodestyle2')
         results = check(args, source_code, filename=filename, options=['-r'])
     except Exception:
         # Never return None to avoid lock in spyder/widgets/editor.py
diff -Naur spyder-3.1.1.orig/spyder/widgets/editor.py spyder-3.1.1.pep8-python2/spyder/widgets/editor.py
--- spyder-3.1.1.orig/spyder/widgets/editor.py	2017-01-22 18:28:03.000000000 +0100
+++ spyder-3.1.1.pep8-python2/spyder/widgets/editor.py	2017-01-22 23:45:29.697031996 +0100
@@ -199,7 +199,7 @@
         """Run code analysis"""
         run_pyflakes = run_pyflakes and codeanalysis.is_pyflakes_installed()
         run_pep8 = run_pep8 and\
-                   codeanalysis.get_checker_executable('pep8') is not None
+                   codeanalysis.get_checker_executable('pycodestyle') is not None
         self.pyflakes_results = []
         self.pep8_results = []
         if self.editor.is_python():
