diff -upr flask-nav-0.6.orig/flask_nav/__init__.py flask-nav-0.6/flask_nav/__init__.py
--- flask-nav-0.6.orig/flask_nav/__init__.py	2016-11-23 21:21:38.000000000 +0200
+++ flask-nav-0.6/flask_nav/__init__.py	2021-12-04 01:22:38.179319543 +0200
@@ -46,7 +46,7 @@ class NavbarRenderingError(Exception):
     pass
 
 
-class ElementRegistry(collections.MutableMapping):
+class ElementRegistry(collections.abc.MutableMapping):
     def __init__(self):
         self._elems = {}
 
