diff --git a/tests/patcher_test.py b/tests/patcher_test.py
index dbf6e1c..1af4ac3 100644
--- a/tests/patcher_test.py
+++ b/tests/patcher_test.py
@@ -476,6 +476,8 @@ def test_patcher_existing_locks_late():
     tests.run_isolated('patcher_existing_locks_late.py')
 
 
+# https://github.com/eventlet/eventlet/issues/730
+@tests.skip_if(sys.version_info >= (3, 10))
 def test_patcher_existing_locks_locked():
     tests.run_isolated('patcher_existing_locks_locked.py')
 
diff --git a/tests/wsgi_test.py b/tests/wsgi_test.py
index 999d6f0..5c5bd92 100644
--- a/tests/wsgi_test.py
+++ b/tests/wsgi_test.py
@@ -556,6 +556,8 @@ class TestHttpd(_TestBase):
         self.assertEqual(1, len(
             [l for l in header_lines if l.lower().startswith(b'content-length')]))
 
+    # https://github.com/eventlet/eventlet/issues/739
+    @tests.skip_if(sys.version_info >= (3, 10))
     @tests.skip_if_no_ssl
     def test_017_ssl_zeroreturnerror(self):
 
