diff -upr aioitertools-0.8.0.orig/aioitertools/asyncio.py aioitertools-0.8.0/aioitertools/asyncio.py
--- aioitertools-0.8.0.orig/aioitertools/asyncio.py	2021-08-05 05:22:56.166504100 +0300
+++ aioitertools-0.8.0/aioitertools/asyncio.py	2021-12-02 13:35:33.052209921 +0200
@@ -59,7 +59,6 @@ async def as_completed(
             Tuple[Set[Awaitable[T]], Set[Awaitable[T]]],
             await asyncio.wait(
                 pending,
-                loop=loop,
                 timeout=remaining,
                 return_when=asyncio.FIRST_COMPLETED,
             ),
@@ -124,7 +123,7 @@ async def gather(
         if pending:
             try:
                 done, pending = await asyncio.wait(
-                    pending, loop=loop, return_when=asyncio.FIRST_COMPLETED
+                    pending, return_when=asyncio.FIRST_COMPLETED
                 )
                 for x in done:
                     if return_exceptions and x.exception():
@@ -136,7 +135,7 @@ async def gather(
                 for x in pending:
                     x.cancel()
                 # we insure that all tasks are cancelled before we raise
-                await asyncio.gather(*pending, loop=loop, return_exceptions=True)
+                await asyncio.gather(*pending, return_exceptions=True)
                 raise
 
         if not pending and next_arg == len(args):
@@ -162,7 +161,6 @@ async def gather_iter(
     """
     return await gather(
         *[maybe_await(i) async for i in aiter(itr)],
-        loop=loop,
         return_exceptions=return_exceptions,
         limit=limit,
     )
