diff --git a/mocha.gemspec b/mocha.gemspec
index e6e6e93..0c53d33 100644
--- a/mocha.gemspec
+++ b/mocha.gemspec
@@ -34,16 +34,8 @@ Gem::Specification.new do |s| # rubocop:disable Metrics/BlockLength
     # Rake >= v11 does not support Ruby < v1.9.3 so use
     s.add_development_dependency('rake', '~> 10.0')
   end
-  s.add_development_dependency('introspection', '~> 0.0.1')
   if RUBY_VERSION >= '2.2.0'
     # No test libraries in standard library
     s.add_development_dependency('minitest')
   end
-  if RUBY_VERSION >= '1.9.2'
-    s.add_development_dependency('rubocop', '<= 0.58.2')
-  end
-  if ENV['MOCHA_GENERATE_DOCS']
-    s.add_development_dependency('redcarpet')
-    s.add_development_dependency('yard')
-  end
 end
