diff --git a/csv.gemspec b/csv.gemspec
index 11c5b0f..203811c 100644
--- a/csv.gemspec
+++ b/csv.gemspec
@@ -1,15 +1,8 @@
 # frozen_string_literal: true
 
-begin
-  require_relative "lib/csv/version"
-rescue LoadError
-  # for Ruby core repository
-  require_relative "version"
-end
-
 Gem::Specification.new do |spec|
   spec.name          = "csv"
-  spec.version       = CSV::VERSION
+  spec.version       = "3.2.5"
   spec.authors       = ["James Edward Gray II", "Kouhei Sutou"]
   spec.email         = [nil, "kou@cozmixng.org"]
 
