diff --git a/forwardable.gemspec b/forwardable.gemspec
index 523b8ee..7fbc97a 100644
--- a/forwardable.gemspec
+++ b/forwardable.gemspec
@@ -1,15 +1,8 @@
 # frozen_string_literal: true
 
-name = File.basename(__FILE__, ".gemspec")
-version = ["lib", Array.new(name.count("-")+1, "..").join("/")].find do |dir|
-  break File.foreach(File.join(__dir__, dir, "#{name.tr('-', '/')}.rb")) do |line|
-    /^\s*VERSION\s*=\s*"(.*)"/ =~ line and break $1
-  end rescue nil
-end
-
 Gem::Specification.new do |spec|
-  spec.name          = name
-  spec.version       = version
+  spec.name          = "forwardable"
+  spec.version       = "1.3.2"
   spec.authors       = ["Keiju ISHITSUKA"]
   spec.email         = ["keiju@ruby-lang.org"]
 
