diff --git a/did_you_mean.gemspec b/did_you_mean.gemspec
index 8fe5723..35801dc 100644
--- a/did_you_mean.gemspec
+++ b/did_you_mean.gemspec
@@ -1,15 +1,8 @@
 # coding: utf-8
-lib = File.expand_path('../lib', __FILE__)
-$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
-begin
-  require_relative "lib/did_you_mean/version"
-rescue LoadError # Fallback to load version file in ruby core repository
-  require_relative "version"
-end
 
 Gem::Specification.new do |spec|
   spec.name          = "did_you_mean"
-  spec.version       = DidYouMean::VERSION
+  spec.version       = "1.6.1"
   spec.authors       = ["Yuki Nishijima"]
   spec.email         = ["mail@yukinishijima.net"]
   spec.summary       = '"Did you mean?" experience in Ruby'
