diff --git a/src/sage_setup/docbuild/__init__.py b/src/sage_setup/docbuild/__init__.py
index 5877b5b..7dc8ff7 100644
--- a/src/sage_setup/docbuild/__init__.py
+++ b/src/sage_setup/docbuild/__init__.py
@@ -265,7 +265,7 @@ class DocBuilder(object):
     inventory = builder_helper('inventory')
 
 
-if not (CYGWIN_VERSION and CYGWIN_VERSION[0] < 3):
+if False:
     def build_many(target, args):
         # Pool() uses an actual fork() to run each new instance. This is
         # important for performance reasons, i.e., don't use a forkserver when

