diff --git a/solr/distribution/build.gradle b/solr/distribution/build.gradle
--- a/solr/distribution/build.gradle	(revision a4eb7aa123dc53f8dac74d80b66a490f2d6b4a26)
+++ b/solr/distribution/build.gradle	(date 1668163583889)
@@ -70,7 +70,6 @@
 
   files = objects.fileCollection()
   [
-      tasks.assembleSourceTgz,
       distTarTask,
   ].each { dep ->
     dependsOn dep
@@ -129,7 +128,6 @@
   })
 
   from tasks.prepareGitRev
-  from tasks.assembleSourceTgz
   from distTarTask
 
   from tasks.computeChecksums
diff --git a/solr/distribution/source-release.gradle b/solr/distribution/source-release.gradle
--- a/solr/distribution/source-release.gradle	(revision a4eb7aa123dc53f8dac74d80b66a490f2d6b4a26)
+++ b/solr/distribution/source-release.gradle	(date 1668163583894)
@@ -23,7 +23,6 @@
     description "Assemble source Solr artifact as a .tgz file."
 
     // Make sure no unstaged/ dirty files are present.
-    dependsOn ":gitStatus", ":checkWorkingCopyClean"
 
     ext {
       destination = file("${buildDir}/packages/solr-${version}-src.tgz")
