--- a/tests/__init__.py	2022-10-17 14:59:15.734686304 +0800
+++ b/tests/__init__.py	2022-10-17 14:59:20.814593222 +0800
@@ -165,9 +165,10 @@
 
     def setUp(self):
         self.driver = ClientDriver()
-        env = None
+        env = os.environ.copy()
         if self.INJECT_DUMMY_CREDS:
-            env = {'AWS_ACCESS_KEY_ID': 'foo', 'AWS_SECRET_ACCESS_KEY': 'bar'}
+            env['AWS_ACCESS_KEY_ID'] = 'foo'
+            env['AWS_SECRET_ACCESS_KEY'] = 'bar'
         self.driver.start(env=env)
 
     def cmd(self, *args):
