diff --git i/tests/system/utils/test_main.py w/tests/system/utils/test_main.py
index 5a3942d..7121450 100644
--- i/tests/system/utils/test_main.py
+++ w/tests/system/utils/test_main.py
@@ -38,33 +38,6 @@ def _fingerprints_broken():
     return broken_urllib3
 
 
-@pytest.mark.skipif(
-    _fingerprints_broken(), reason="https://github.com/shazow/urllib3/issues/529"
-)
-@pytest.mark.parametrize(
-    "fingerprint",
-    [
-        "94:FD:7A:CB:50:75:A4:69:82:0A:F8:23:DF:07:FC:69:3E:CD:90:CA",
-        "19:90:F7:23:94:F2:EF:AB:2B:64:2D:57:3D:25:95:2D",
-    ],
-)
-def test_request_ssl_fingerprints(httpsserver, fingerprint):
-    httpsserver.serve_content("")  # we need to serve something
-
-    http.request("GET", httpsserver.url, verify=False, verify_fingerprint=fingerprint)
-    with pytest.raises(requests.exceptions.ConnectionError) as excinfo:
-        http.request("GET", httpsserver.url, verify_fingerprint=fingerprint)
-
-    with pytest.raises(requests.exceptions.ConnectionError) as excinfo:
-        http.request(
-            "GET",
-            httpsserver.url,
-            verify=False,
-            verify_fingerprint="".join(reversed(fingerprint)),
-        )
-    assert "Fingerprints did not match" in str(excinfo.value)
-
-
 def test_open_graphical_browser(monkeypatch):
     import webbrowser
 
