diff --git a/test/vspec/signatures.vim b/test/vspec/signatures.vim
index ca4248e..6090ee9 100644
--- a/test/vspec/signatures.vim
+++ b/test/vspec/signatures.vim
@@ -41,7 +41,7 @@ describe 'signatures'
         doautocmd CursorHoldI
         noautocmd normal istaticmethod()
         doautocmd CursorHoldI
-        Expect getline(1) == '?!?jedi=0, ?!?            (*_*f: Callable*_*) ?!?jedi?!?'
+        Expect getline(1) == '?!?jedi=0, ?!?            (*_*f: Callable[..., Any]*_*) ?!?jedi?!?'
     end
 
     it 'no signature'
@@ -68,7 +68,7 @@ describe 'signatures'
         redir => msg
         Python jedi_vim.show_call_signatures()
         redir END
-        Expect msg == "\nstaticmethod(f: Callable)"
+        Expect msg == "\nstaticmethod(f: Callable[..., Any])"
 
         redir => msg
         doautocmd InsertLeave
