diff -aur tensorboard-2.2.0-old/third_party/workspace.bzl tensorboard-2.2.0-new/third_party/workspace.bzl
--- tensorboard-2.2.0-old/third_party/workspace.bzl	2020-03-24 15:41:29.000000000 +0200
+++ tensorboard-2.2.0-new/third_party/workspace.bzl	2020-04-07 18:09:49.309586469 +0300
@@ -55,6 +55,18 @@
       actual = "@org_pythonhosted_six",
   )
 
+  # gRPC's BUILD file depends on //external:protobuf_clib.
+  native.bind(
+      name = "protobuf_clib",
+      actual = "@com_google_protobuf//:protoc_lib",
+  )
+
+  # gRPC's BUILD file depends on //external:protobuf_headers.
+  native.bind(
+      name = "protobuf_headers",
+      actual = "@com_google_protobuf//:protobuf_headers",
+  )
+
   platform_http_file(
       name = "org_chromium_chromium",
       licenses = ["notice"],  # BSD 3-clause (maybe more?)
diff -aur tensorboard-2.2.0-old/WORKSPACE tensorboard-2.2.0-new/WORKSPACE
--- tensorboard-2.2.0-old/WORKSPACE	2020-04-07 18:11:04.789583868 +0300
+++ tensorboard-2.2.0-new/WORKSPACE	2020-04-07 18:11:15.819583488 +0300
@@ -113,6 +113,21 @@
 load("@org_tensorflow//tensorflow:workspace.bzl", "tf_workspace")
 tf_workspace()
 
+# add upb dep
+http_archive(
+    name = "upb",
+    urls = [
+        "https://github.com/protocolbuffers/upb/archive/ef25ea52814b48f1d945e25b5b1b19b04eea3ad8.tar.gz"
+    ],
+    strip_prefix = "upb-ef25ea52814b48f1d945e25b5b1b19b04eea3ad8",
+    sha256 = "03dd5e19e1b653a3b2d7f50a3cc3ed913a72a0ecaf0d667c5f4df3ab3c8587c2",
+)
+# load("@upb//bazel:workspace_deps.bzl", "upb_deps")
+# load("@upb//bazel:upb_proto_library.bzl", "upb_deps")
+# upb_deps()
+# load("@upb//bazel:grpc_deps.bzl", "grpc_my_deps")
+# grpc_my_deps()
+
 # Please add all new dependencies in workspace.bzl.
 load("//third_party:workspace.bzl", "tensorboard_workspace")
 tensorboard_workspace()
