diff --git a/src/nemo-window.c b/src/nemo-window.c
--- a/src/nemo-window.c	2015-11-27 11:00:22.000000000 +0100
+++ b/src/nemo-window.c	2016-02-13 22:13:11.907508549 +0100
@@ -2066,6 +2066,11 @@ nemo_window_init (NemoWindow *window)
 
 	window->details = G_TYPE_INSTANCE_GET_PRIVATE (window, NEMO_TYPE_WINDOW, NemoWindowDetails);
 
+    GtkStyleContext *context;
+
+    context = gtk_widget_get_style_context (GTK_WIDGET (window));
+    gtk_style_context_add_class (context, "nemo-window");
+
 	window->details->panes = NULL;
 	window->details->active_pane = NULL;
 
diff --git a/src/nemo-window-pane.c b/src/nemo-window-pane.c
--- a/src/nemo-window-pane.c	2015-11-27 11:00:22.000000000 +0100
+++ b/src/nemo-window-pane.c	2016-02-13 22:35:49.996213630 +0100
@@ -983,6 +983,11 @@ nemo_window_pane_init (NemoWindowPane *p
 	pane->active_slot = NULL;
 
 	gtk_orientable_set_orientation (GTK_ORIENTABLE (pane), GTK_ORIENTATION_VERTICAL);
+
+	GtkStyleContext *context;
+
+	context = gtk_widget_get_style_context (GTK_WIDGET (pane));
+	gtk_style_context_add_class (context, "nemo-window-pane");
 }
 
 NemoWindowPane *

diff --git a/src/nemo-file-management-properties.glade b/src/nemo-file-management-properties.glade
--- a/src/nemo-file-management-properties.glade	2015-11-27 11:00:22.000000000 +0100
+++ b/src/nemo-file-management-properties.glade	2016-02-17 09:05:07.491011645 +0100
@@ -253,6 +253,9 @@
     <property name="window_position">center</property>
     <property name="default_width">800</property>
     <property name="default_height">600</property>
+    <style>
+      <class name="nemo-properties-dialog"/>
+    </style>
     <property name="type_hint">dialog</property>
     <child internal-child="vbox">
       <object class="GtkBox" id="dialog-vbox1">