--- pdfmixtool-v1.1/src/pdf_edit_lib/pdf_info.h	2022-09-13 16:34:05.000000000 +0200
+++ pdfmixtool-v1.1/src/pdf_edit_lib/pdf_info.h.new	2022-09-13 16:37:30.147859755 +0200
@@ -19,6 +19,8 @@
 #ifndef PDF_INFO_H
 #define PDF_INFO_H
 
+#define POINTERHOLDER_TRANSITION 0
+
 #include <string>
 #include <ctime>
 #include <qpdf/QUtil.hh>
--- pdfmixtool-v1.1/src/pdf_edit_lib/pdf_editor.h	2022-09-13 16:34:05.000000000 +0200
+++ pdfmixtool-v1.1/src/pdf_edit_lib/pdf_editor.h.new	2022-09-13 16:38:01.588630775 +0200
@@ -23,6 +23,8 @@
 #include <vector>
 #include <cmath>
 
+#define POINTERHOLDER_TRANSITION 0
+
 #include <qpdf/QPDF.hh>
 #include <qpdf/QUtil.hh>
 #include <qpdf/QPDFPageDocumentHelper.hh>
--- pdfmixtool-v1.1/src/pdf_edit_lib/pdf_editor.cpp	2022-09-13 16:40:11.000000000 +0200
+++ pdfmixtool-v1.1/src/pdf_edit_lib/pdf_editor.cpp.new	2022-09-13 16:43:41.636993637 +0200
@@ -28,10 +28,10 @@
 #include <qpdf/QPDFOutlineDocumentHelper.hh>
 #include "qpdf_version.h"
 
-QPDFOutlineDocumentHelper::~QPDFOutlineDocumentHelper()
+/*QPDFOutlineDocumentHelper::~QPDFOutlineDocumentHelper()
 {
 
-}
+}*/
 
 PdfEditor::PdfEditor() :
     m_old_locale{std::locale::global(std::locale::classic())},
