--- src/desmume-0.9.10/src/gtk/glx_3Demu.cpp	2013-11-28 01:36:53.712615822 +0100
+++ glx_3Demu.cpp	2013-11-28 22:08:52.762390897 +0100
@@ -41,11 +41,7 @@
         glXDestroyContext(dpy, ctx);
 
         XCloseDisplay(dpy);
-
-        return true;
     }
-
-    return false;
 }
 
 int init_glx_3Demu(void) 
--- src/desmume-0.9.10/src/gtk/glx_3Demu.cpp	2013-11-28 01:36:53.712615822 +0100
+++ glx_3Demu.cpp	2013-11-28 22:14:34.529740137 +0100
@@ -31,7 +31,7 @@
 static GLXContext ctx;
 static GLXPbuffer pbuf;
 
-void deinit_glx_3Demu(void)
+int deinit_glx_3Demu(void)
 {
     Display *dpy = glXGetCurrentDisplay();
 
