diff -u -r pycdio-0.21/cdio.py pycdio-0.21-fix/cdio.py
--- pycdio-0.21/cdio.py	2015-05-08 04:44:44.000000000 +0200
+++ pycdio-0.21-fix/cdio.py	2018-01-12 23:28:17.062063693 +0100
@@ -47,7 +47,6 @@
 drivers = {
     'Unknown'  : pycdio.DRIVER_UNKNOWN,
     'AIX'      : pycdio.DRIVER_AIX,
-    'BSDI'     : pycdio.DRIVER_BSDI,
     'FreeBSD'  : pycdio.DRIVER_FREEBSD,
     'GNU/Linux': pycdio.DRIVER_LINUX,
     'linux'    : pycdio.DRIVER_LINUX,
diff -u -r pycdio-0.21/pycdio.py pycdio-0.21-fix/pycdio.py
--- pycdio-0.21/pycdio.py	2017-08-16 14:24:10.000000000 +0200
+++ pycdio-0.21-fix/pycdio.py	2018-01-12 23:28:05.482036758 +0100
@@ -1139,9 +1139,6 @@
 _pycdio.DRIVER_AIX_swigconstant(_pycdio)
 DRIVER_AIX = _pycdio.DRIVER_AIX
 
-_pycdio.DRIVER_BSDI_swigconstant(_pycdio)
-DRIVER_BSDI = _pycdio.DRIVER_BSDI
-
 _pycdio.DRIVER_FREEBSD_swigconstant(_pycdio)
 DRIVER_FREEBSD = _pycdio.DRIVER_FREEBSD
 
diff -u -r pycdio-0.21/pyiso9660.py pycdio-0.21-fix/pyiso9660.py
--- pycdio-0.21/pyiso9660.py	2017-08-16 14:24:14.000000000 +0200
+++ pycdio-0.21-fix/pyiso9660.py	2018-01-12 23:34:38.799618728 +0100
@@ -518,7 +518,7 @@
 
     The b_mode2 parameter is not used.
     """
-    return _pyiso9660.fs_stat_translate(p_cdio, psz_path, b_mode2)
+    return _pyiso9660.fs_stat_translate(p_cdio, psz_path)
 
 def ifs_stat(p_iso, psz_path):
     """Return file status for pathname. None is returned on error."""
diff -u -r pycdio-0.21/swig/device_const.swg pycdio-0.21-fix/swig/device_const.swg
--- pycdio-0.21/swig/device_const.swg	2013-02-20 12:32:41.000000000 +0100
+++ pycdio-0.21-fix/swig/device_const.swg	2018-01-12 23:28:12.518719791 +0100
@@ -83,7 +83,6 @@
 /* driver_id_t enums. */
 %constant long int DRIVER_UNKNOWN = DRIVER_UNKNOWN;
 %constant long int DRIVER_AIX     = DRIVER_AIX;
-%constant long int DRIVER_BSDI    = DRIVER_BSDI;
 %constant long int DRIVER_FREEBSD = DRIVER_FREEBSD;
 %constant long int DRIVER_LINUX   = DRIVER_LINUX;
 %constant long int DRIVER_SOLARIS = DRIVER_SOLARIS;
diff -u -r pycdio-0.21/swig/pyiso9660.i pycdio-0.21-fix/swig/pyiso9660.i
--- pycdio-0.21/swig/pyiso9660.i	2013-02-20 12:32:41.000000000 +0100
+++ pycdio-0.21-fix/swig/pyiso9660.i	2018-01-12 23:30:03.645644975 +0100
@@ -550,13 +550,10 @@
 %feature("autodoc",
 "Return file status for path name psz_path. None is returned on error.
 pathname version numbers in the ISO 9660 name are dropped, i.e. ;1
-is removed and if level 1 ISO-9660 names are lowercased.
-
-The b_mode2 parameter is not used.");
+is removed and if level 1 ISO-9660 names are lowercased.");
 %rename iso9660_fs_stat_translate fs_stat_translate;
 IsoStat_t *iso9660_fs_stat_translate (CdIo_t *p_cdio, 
-				      const char psz_path[], 
-				      bool b_mode2=false);
+				      const char psz_path[]);
 
 %feature("autodoc",
 "Return file status for pathname. None is returned on error.");
@@ -580,7 +577,7 @@
 %inline %{
 IsoStatList_t *fs_readdir (CdIo_t *p_cdio, const char psz_path[])
 {
-  CdioList_t *p_statlist = iso9660_fs_readdir (p_cdio, psz_path, false);
+  CdioList_t *p_statlist = iso9660_fs_readdir (p_cdio, psz_path);
   return p_statlist;
 }
 %}
