--- php/php-src/branches/PHP_5_2/main/fopen_wrappers.c	2010/12/09 17:04:29	306136
+++ php/php-src/branches/PHP_5_2/main/fopen_wrappers.c	2010/12/10 12:50:25	306184
@@ -192,7 +192,7 @@
 		if (strncmp(resolved_basedir, resolved_name, resolved_basedir_len) == 0) {
 #endif
 			if (resolved_name_len > resolved_basedir_len &&
-				resolved_name[resolved_basedir_len] != PHP_DIR_SEPARATOR) {
+				resolved_name[resolved_basedir_len - 1] != PHP_DIR_SEPARATOR) {
 				return -1;
 			} else {
 				/* File is in the right directory */
