diff -wbBur xl2tpd-1.2.8/control.c xl2tpd-1.2.8.my/control.c
--- xl2tpd-1.2.8/control.c	2011-02-23 08:10:21.000000000 +0300
+++ xl2tpd-1.2.8.my/control.c	2011-03-25 21:29:29.896518014 +0300
@@ -892,8 +892,8 @@
                 po = add_opt (po, c->lac->pppoptfile);
             }
         };
-	po = add_opt (po, "ipparam");
-        po = add_opt (po, IPADDY (t->peer.sin_addr));
+//	po = add_opt (po, "ipparam");
+//        po = add_opt (po, IPADDY (t->peer.sin_addr));
         start_pppd (c, po);
         opt_destroy (po);
         if (c->lac)
@@ -966,8 +966,8 @@
             po = add_opt (po, "file");
             po = add_opt (po, c->lns->pppoptfile);
         }
-	po = add_opt (po, "ipparam");
-        po = add_opt (po, IPADDY (t->peer.sin_addr));
+//	po = add_opt (po, "ipparam");
+//        po = add_opt (po, IPADDY (t->peer.sin_addr));
         start_pppd (c, po);
         opt_destroy (po);
         l2tp_log (LOG_NOTICE,
@@ -1026,8 +1026,8 @@
                 po = add_opt (po, c->lac->pppoptfile);
             }
         };
-	po = add_opt (po, "ipparam");
-        po = add_opt (po, IPADDY (t->peer.sin_addr));
+//	po = add_opt (po, "ipparam");
+//        po = add_opt (po, IPADDY (t->peer.sin_addr));
         start_pppd (c, po);
 
         /*  jz: just show some information */
