diff -Naur foxtrotgps-1.2.0.orig/contrib/georss2foxtrotgps-poi foxtrotgps-1.2.0/contrib/georss2foxtrotgps-poi
--- foxtrotgps-1.2.0.orig/contrib/georss2foxtrotgps-poi	2014-10-24 06:57:10.000000000 +0200
+++ foxtrotgps-1.2.0/contrib/georss2foxtrotgps-poi	2017-06-25 16:26:22.472440359 +0200
@@ -30,7 +30,7 @@
 import random
 import sqlite3
 import feedparser
-from BeautifulSoup import BeautifulSoup, BeautifulStoneSoup
+from bs4 import BeautifulSoup
 
 def link2txt(a):
 	a_text = ''.join(a).strip()
@@ -76,11 +76,11 @@
 		rand2 = random.randint(100000000,1000000000)
 		rand = '%s%s' % (rand1,rand2)
 		desc = e.description
-		soup = BeautifulSoup(desc,convertEntities=BeautifulStoneSoup.HTML_ENTITIES,smartQuotesTo=None)
-		[img.extract() for img in soup.findAll('img')]
-		[br.replaceWith('\n') for br in soup.findAll('br')]
-		[a.replaceWith(link2txt(a)) for a in soup.findAll('a')]
-		desc = ''.join(soup.findAll(text=True)).strip()
+		soup = BeautifulSoup(desc)
+		[img.extract() for img in soup.find_all('img')]
+		[br.replaceWith('\n') for br in soup.find_all('br')]
+		[a.replaceWith(link2txt(a)) for a in soup.find_all('a')]
+		desc = ''.join(soup.find_all(text=True)).strip()
 		poi.append((rand,lat,lon,e.title,desc))
 
 cur.executemany('INSERT INTO poi (idmd5,lat,lon,keywords,desc,visibility,cat,subcat,price_range,extended_open) VALUES (?,?,?,?,?,1.0,0.0,0.0,1.0,0.0)', poi)
diff -Naur foxtrotgps-1.2.0.orig/contrib/osb2foxtrot foxtrotgps-1.2.0/contrib/osb2foxtrot
--- foxtrotgps-1.2.0.orig/contrib/osb2foxtrot	2014-10-24 06:57:10.000000000 +0200
+++ foxtrotgps-1.2.0/contrib/osb2foxtrot	2017-06-25 16:28:16.822909562 +0200
@@ -2,7 +2,7 @@
 
 from xml.etree.ElementTree import parse
 import sys, re, hashlib
-from pysqlite2 import dbapi2 as sqlite
+import sqlite3
 
 if len(sys.argv) <3:
   print "usage:"
@@ -18,7 +18,7 @@
 gpx = open(sys.argv[1])
 xml = parse(gpx).getroot()
 
-con = sqlite.connect(sys.argv[2]) 
+con = sqlite3.connect(sys.argv[2]) 
 cur = con.cursor()
 
 for wpt in xml:
