=== modified file 'src/tile_management.c'
--- src/tile_management.c	2014-11-08 17:55:34 +0000
+++ src/tile_management.c	2017-04-11 09:20:47 +0000
@@ -181,7 +181,7 @@
 	if(strcmp(repo->uri,"maps-for-free")==0)
 	{
 		g_snprintf(tile_data_tmp, sizeof tile_data_tmp,
-				"http://maps-for-free.com/layer/relief/z%d/row%d/%d_%d-%d.jpg"
+				"https://maps-for-free.com/layer/relief/z%d/row%d/%d_%d-%d.jpg"
 				"|%s/%d/%d/%d.png|%s/%d/%d/",
 				zoom,y,zoom,x,y,
 				repo->dir, zoom, x, y,

