diff --git a/blist/_sortedlist.py b/blist/_sortedlist.py
index b34f69e..d6143c4 100644
--- a/blist/_sortedlist.py
+++ b/blist/_sortedlist.py
@@ -420,7 +420,10 @@ class _setmixin(object):
     def __iter__(self):
         it = super(_setmixin, self).__iter__()
         while True:
-            item = next(it)
+            try:
+                item = next(it)
+            except StopIteration:
+                return
             n = len(self)
             yield item
             if n != len(self):
