diff -Naur spyder-2.3.0/spyderplugins/widgets/pylintgui.py spyder-2.3.0-patched/spyderplugins/widgets/pylintgui.py
--- spyder-2.3.0/spyderplugins/widgets/pylintgui.py	2014-07-10 08:06:40.000000000 +0200
+++ spyder-2.3.0-patched/spyderplugins/widgets/pylintgui.py	2014-08-08 12:26:07.164971831 +0200
@@ -40,7 +40,7 @@
 _ = get_translation("p_pylint", dirname="spyderplugins")
 
 
-PYLINT_PATH = programs.find_program('pylint')
+PYLINT_PATH = programs.find_program('pylint2')
 
 
 def get_pylint_version():
@@ -48,13 +48,13 @@
     global PYLINT_PATH
     if PYLINT_PATH is None:
         return
-    process = subprocess.Popen(['pylint', '--version'],
+    process = subprocess.Popen(['pylint2', '--version'],
                                stdout=subprocess.PIPE, stderr=subprocess.PIPE,
                                cwd=osp.dirname(PYLINT_PATH),
                                shell=True if os.name == 'nt' else False)
     lines = to_unicode_from_fs(process.stdout.read()).splitlines()
     if lines:
-        match = re.match('(pylint|pylint-script.py) ([0-9\.]*)', lines[0])
+        match = re.match('(pylint2|pylint2-script.py) ([0-9\.]*)', lines[0])
         if match is not None:
             return match.groups()[1]
 
