--- a/scripts/spyder3.desktop
+++ b/scripts/spyder3.desktop
@@ -1,12 +1,11 @@
 [Desktop Entry]
 Type=Application
 Version=1.0
-Name=Spyder3
-GenericName=Spyder3
+Name=Spyder
 Comment=The Scientific Python Development Environment - Python 3
-Icon=spyder3
-TryExec=spyder3
-Exec=spyder3 %F
+Icon=spyder
+TryExec=spyder
+Exec=spyder %F
 Terminal=false
 MimeType=text/x-python;
 Categories=Development;Science;IDE;Qt;
--- a/setup.py
+++ b/setup.py
@@ -88,9 +88,9 @@
     """Return data_files in a platform dependent manner"""
     if sys.platform.startswith('linux'):
         if PY3:
-            data_files = [('share/applications', ['scripts/spyder3.desktop']),
-                          ('share/icons', ['img_src/spyder3.png']),
-                          ('share/metainfo', ['scripts/spyder3.appdata.xml'])]
+            data_files = [('share/applications', ['scripts/spyder.desktop']),
+                          ('share/icons', ['img_src/spyder.png']),
+                          ('share/metainfo', ['scripts/spyder.appdata.xml'])]
         else:
             data_files = [('share/applications', ['scripts/spyder.desktop']),
                           ('share/icons', ['img_src/spyder.png'])]
@@ -131,7 +131,7 @@
 # See spyder-ide/spyder#1158.
 SCRIPTS = ['%s_win_post_install.py' % NAME]
 if PY3 and sys.platform.startswith('linux'):
-    SCRIPTS.append('spyder3')
+    SCRIPTS.append('spyder')
 else:
     SCRIPTS.append('spyder')
 
