diff -Nur gecko-mediaplayer-0.9.9.old/configure.in gecko-mediaplayer-0.9.9/configure.in
--- gecko-mediaplayer-0.9.9.old/configure.in	2010-02-01 18:00:42.000000000 +0200
+++ gecko-mediaplayer-0.9.9/configure.in	2010-02-07 21:16:15.000000000 +0200
@@ -96,11 +96,11 @@
 AC_ARG_WITH([gconf],
     [AS_HELP_STRING([--with-gconf],
         [Use Gconf to store preferences])],
-        [],
+        [AM_CONDITIONAL([GCONF_SCHEMAS_INSTALL],[1])],
         [with_gconf=check])
 AS_IF([test "x$with_gconf" != xno],        
     [PKG_CHECK_MODULES(GCONF, [gconf-2.0],[AM_GCONF_SOURCE_2
- GCONF_DEFINES="-DHAVE_GCONF" AC_DEFINE([HAVE_GCONF],[1],[Define to 1 to use GConf to store preferences])],AC_MSG_NOTICE([Using keystore for preferences]))
+ GCONF_DEFINES="-DHAVE_GCONF" AC_DEFINE([HAVE_GCONF],[1],[Define to 1 to use GConf to store preferences])],[AM_CONDITIONAL([GCONF_SCHEMAS_INSTALL],[1]) AC_MSG_NOTICE([Using keystore for preferences])])
     AC_SUBST(GCONF_DEFINES)
     AC_SUBST(GCONF_CFLAGS)
     AC_SUBST(GCONF_LIBS)
