From 1418b24bf99b10a3bd45258ed30613a8de4b8955 Mon Sep 17 00:00:00 2001
From: Jacques Garrigue <garrigue@math.nagoya-u.ac.jp>
Date: Wed, 25 Sep 2019 16:41:02 +0200
Subject: [PATCH] disable -warn-error on Obj.truncate

---
 src/dune | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/dune b/src/dune
index ed283fd4..def43f7a 100644
--- a/src/dune
+++ b/src/dune
@@ -68,7 +68,8 @@
 (library
  (name lablgtk3)
  (public_name lablgtk3)
- (flags :standard -w -6-7-9-10-27-32-33-34-35-36-50-52 -no-strict-sequence)
+ (flags :standard -w -6-7-9-10-27-32-33-34-35-36-50-52 -no-strict-sequence
+        -warn-error A-3) ; needed for Obj.truncate
  (wrapped false)
  (modules
    ; gtkBrokenProps ogtkBrokenProps gtkBroken gBroken
-- 
2.23.0

