diff --git a/src/compositor/compositor-xrender.c b/src/compositor/compositor-xrender.c
index dbd7bd9..9bf9d8d 100644
--- a/src/compositor/compositor-xrender.c
+++ b/src/compositor/compositor-xrender.c
@@ -911,6 +911,12 @@ window_has_shadow (MetaCompWindow *cw)
       }
     }
 
+  /* Do not add shadows to ARGB windows */
+  if (cw->mode == WINDOW_ARGB) {
+    meta_verbose ("Window has no shadow as it is ARGB\n");
+    return FALSE;
+  }
+
   /* Never put a shadow around shaped windows */
   if (cw->shaped) {
     meta_verbose ("Window has no shadow as it is shaped\n");
