From 1a9a05ef7942c91dafaf24023404aef9cbd77dbe Mon Sep 17 00:00:00 2001
From: Felix Yan <felixonmars@archlinux.org>
Date: Sat, 16 Feb 2019 00:14:39 +0800
Subject: [PATCH] fix: compatibility with Qt 5.12.1

Change-Id: I193b2c069e46251b6bed9fb0766d26287e46d0b4
---

diff --git a/platformplugin/dplatformwindowhelper.cpp b/platformplugin/dplatformwindowhelper.cpp
index d1b3772..09401e0 100644
--- a/platformplugin/dplatformwindowhelper.cpp
+++ b/platformplugin/dplatformwindowhelper.cpp
@@ -321,7 +321,11 @@
         Utility::setMotifWmHints(helper->m_nativeWindow->QNativeWindow::winId(), cw_hints);
 
         if (helper->m_nativeWindow->window()->modality() != Qt::NonModal) {
+#if QT_VERSION < QT_VERSION_CHECK(5, 12, 1)
             window->setNetWmStates(window->netWmStates() | QNativeWindow::NetWmStateModal);
+#else
+            window->setNetWmState(true, window->atom(QXcbAtom::_NET_WM_STATE_MODAL));
+#endif
         }
 #endif
 
@@ -362,7 +366,11 @@
     if (state == Qt::WindowMinimized
             && (window->m_windowState == Qt::WindowMaximized
                 || window->m_windowState == Qt::WindowFullScreen)) {
+#if QT_VERSION < QT_VERSION_CHECK(5, 12, 1)
         window->changeNetWmState(true, Utility::internAtom("_NET_WM_STATE_HIDDEN"));
+#else
+        window->setNetWmState(true, Utility::internAtom("_NET_WM_STATE_HIDDEN"));
+#endif
         Utility::XIconifyWindow(window->connection()->xlib_display(),
                                 window->m_window,
                                 window->connection()->primaryScreenNumber());
