From 64a870e1f94fd3d74c3c6af3897d52ebf4f6bdc7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Eric=20B=C3=A9langer?= <snowmaniscool@gmail.com>
Date: Fri, 16 Aug 2013 03:05:52 -0400
Subject: [PATCH] Update PKGBUILD template
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Eric Bélanger <snowmaniscool@gmail.com>
---
 pkgbuild-mode.el | 24 +++++++++++++++++++-----
 1 file changed, 19 insertions(+), 5 deletions(-)

diff --git a/pkgbuild-mode.el b/pkgbuild-mode.el
index deef5df..5c77a64 100644
--- a/pkgbuild-mode.el
+++ b/pkgbuild-mode.el
@@ -129,6 +129,7 @@
 pkgname=%s
 pkgver=VERSION
 pkgrel=1
+epoch=
 pkgdesc=\"\"
 arch=('i686' 'x86_64')
 url=\"\"
@@ -136,6 +137,7 @@ license=('GPL')
 groups=()
 depends=()
 makedepends=()
+checkdepends=()
 optdepends=()
 provides=()
 conflicts=()
@@ -144,22 +146,34 @@ backup=()
 options=()
 install=
 changelog=
-source=($pkgname-$pkgver.tar.gz)
+source=($pkgname-$pkgver.tar.gz
+        $pkgname-$pkgver.patch)
 noextract=()
 md5sums=()
 
+prepare() {
+  cd $pkgname-$pkgver
+
+  patch -p1 -i \"$srcdir/$pkgname-$pkgver.patch\"
+}
+
 build() {
-  cd $startdir/src/$pkgname-$pkgver
+  cd $pkgname-$pkgver
 
   ./configure --prefix=/usr
   make
 }
 
-package() {
+check() {
+  cd $pkgname-$pkgver
 
-  cd $startdir/src/$pkgname-$pkgver
+  make -k check
+}
+
+package() {
+  cd $pkgname-$pkgver
 
-  make DESTDIR=$startdir/pkg install
+  make DESTDIR=\"$pkgdir/\" install
 }
 
 # vim:set ts=2 sw=2 et:
-- 
1.8.5.5

