diff -r 7bd874f87f0e configure
--- a/configure	Wed Apr 22 17:18:49 2020 +0100
+++ b/configure	Fri Apr 24 10:15:07 2020 +0100
@@ -4508,7 +4508,10 @@
 fi
 
 CXXFLAGS="$CXXFLAGS_BUILD $SV_DEFINES_BUILD"
-LDFLAGS="$LDFLAGS_BUILD"
+
+if test x"LDFLAGS" = x; then
+   LDFLAGS="$LDFLAGS_BUILD"
+fi
 
 # On IRIX 5.3, sys/types and inttypes.h are conflicting.
 for ac_header in sys/types.h sys/stat.h stdlib.h string.h memory.h strings.h \
diff -r 7bd874f87f0e configure.ac
--- a/configure.ac	Wed Apr 22 17:18:49 2020 +0100
+++ b/configure.ac	Fri Apr 24 10:15:07 2020 +0100
@@ -87,7 +87,11 @@
 fi
 
 CXXFLAGS="$CXXFLAGS_BUILD $SV_DEFINES_BUILD"
-LDFLAGS="$LDFLAGS_BUILD"
+
+# Set our own LDFLAGS only if the user didn't request any
+if test x"LDFLAGS" = x; then
+   LDFLAGS="$LDFLAGS_BUILD"
+fi
 
 SV_MODULE_REQUIRED([bz2],[],[bzlib.h],[bz2],[BZ2_bzReadOpen])
 SV_MODULE_REQUIRED([fftw3],[fftw3 >= 3.0.0],[fftw3.h],[fftw3],[fftw_execute])
