--- lcalc-1.23/src/Makefile.orig	2019-08-05 09:09:40.888836276 +0000
+++ lcalc-1.23/src/Makefile	2019-08-05 09:10:16.912666730 +0000
@@ -58,7 +58,7 @@
    #MACHINE_SPECIFIC_FLAGS = -mpowerpc -mpowerpc64 -m64
 endif
 
-CCFLAGS =  -Wa,-W -O3 $(OPENMP_FLAG) -Wno-deprecated $(PREPROCESSOR_DEFINE) $(MACHINE_SPECIFIC_FLAGS) $(EXTRA)
+CCFLAGS =  -Wa,-W -O3 $(OPENMP_FLAG) -Wno-deprecated $(PREPROCESSOR_DEFINE) $(MACHINE_SPECIFIC_FLAGS) $(EXTRA) $(CXXFLAGS)
 #CCFLAGS =  -Wa,-W -O3 $(OPENMP_FLAG)  $(PREPROCESSOR_DEFINE) $(MACHINE_SPECIFIC_FLAGS) $(EXTRA)
 #CCFLAGS =  -Wa,-W -O2 -fno-exceptions -Wno-deprecated $(PREPROCESSOR_DEFINE) $(MACHINE_SPECIFIC_FLAGS) $(EXTRA)
 
@@ -104,9 +104,9 @@
 endif
 
 ifeq ($(PARI_DEFINE),-DINCLUDE_PARI)
-    LDFLAGS = $(LDFLAGS2) -L$(LOCATION_PARI_LIBRARY) -lpari
+    LDFLAGS += $(LDFLAGS2) -L$(LOCATION_PARI_LIBRARY) -lpari
 else
-    LDFLAGS = $(LDFLAGS2)
+    LDFLAGS += $(LDFLAGS2) 
 endif
 
 
@@ -262,7 +262,7 @@
 
 
 libLfunction.so: $(OBJ_L)
-	g++ -$(DYN_OPTION)  -o libLfunction.so $(OBJ_L)
+	g++ -$(DYN_OPTION)  -o libLfunction.so $(OBJ_L) $(LDFLAGS)
 
 clean:
 	rm -f *.o lcalc libLfunction.so example_programs/example
