--- gecko-dev-042b84af6020b1f2d8029a0dc36ac5955b7f325f.orig/memory/mozjemalloc/jemalloc.c	2020-12-19 15:42:06.871151870 +0000
+++ gecko-dev-042b84af6020b1f2d8029a0dc36ac5955b7f325f/memory/mozjemalloc/jemalloc.c	2020-12-19 15:40:18.222595396 +0000
@@ -332,7 +332,7 @@ __FBSDID("$FreeBSD: head/lib/libc/stdlib
 #include <sys/time.h>
 #include <sys/types.h>
 #if !defined(MOZ_MEMORY_SOLARIS) && !defined(MOZ_MEMORY_ANDROID)
-#include <sys/sysctl.h>
+#include <linux/sysctl.h>
 #endif
 #include <sys/uio.h>
 #ifndef MOZ_MEMORY
