diff --git a/src/gui/generalsettings.cpp b/src/gui/generalsettings.cpp
index 62381d36d..c69592c82 100644
--- a/src/gui/generalsettings.cpp
+++ b/src/gui/generalsettings.cpp
@@ -91,6 +91,7 @@ GeneralSettings::GeneralSettings(QWidget *parent)
     #else
         if (QOperatingSystemVersion::current() < QOperatingSystemVersion::Windows10)
     #endif
+#else
             _ui->showInExplorerNavigationPaneCheckBox->setVisible(false);
 #endif
 
