diff -rup fatrat-1.2.0_beta2/CMakeLists.txt fatrat-1.2.0_beta2.new/CMakeLists.txt
--- fatrat-1.2.0_beta2/CMakeLists.txt	2012-07-14 11:44:27.000000000 +0000
+++ fatrat-1.2.0_beta2.new/CMakeLists.txt	2013-03-25 16:21:03.143596960 +0000
@@ -552,7 +552,7 @@ if(WITH_DOCUMENTATION)
 	)
 endif(WITH_DOCUMENTATION)
 
-qt4_wrap_cpp(fatrat_MOC_SRCS ${fatrat_MOC_HDRS})
+qt4_wrap_cpp(fatrat_MOC_SRCS ${fatrat_MOC_HDRS} OPTIONS -DBOOST_TT_HAS_OPERATOR_HPP_INCLUDED -DBOOST_NO_TEMPLATE_PARTIAL_SPECIALIZATION)
 
 set(fatrat_RCS
 	gfx/resources.qrc
