--- ndiswrapper-1.56/driver/ntoskernel.h	2010-02-11 02:09:32.000000000 +0000
+++ ndiswrapper-1.56/driver/ntoskernel.h	2011-11-09 15:39:14.640555480 +0000
@@ -275,10 +275,6 @@
 #define PM_EVENT_SUSPEND 2
 #endif
 
-#if !defined(HAVE_NETDEV_PRIV)
-#define netdev_priv(dev)  ((dev)->priv)
-#endif
-
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,18)
 #define ISR_PT_REGS_PARAM_DECL
 #define ISR_PT_REGS_ARG
