diff -r ed726703c08c config/baseconfig.mk
--- a/config/baseconfig.mk	Tue Jan 24 14:49:05 2017 -0500
+++ b/config/baseconfig.mk	Fri Feb 03 12:07:10 2017 +0100
@@ -2,10 +2,10 @@
 # directly in python/mozbuild/mozbuild/base.py for gmake validation.
 # We thus use INCLUDED_AUTOCONF_MK to enable/disable some parts depending
 # whether a normal build is happening or whether the check is running.
-includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+includedir := $(includedir)/$(MOZ_APP_NAME)-developer-edition
+idldir = $(datadir)/idl/$(MOZ_APP_NAME)-developer-edition
+installdir = $(libdir)/$(MOZ_APP_NAME)-developer-edition
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-developer-edition-devel
 ifeq (.,$(DEPTH))
 DIST = dist
 else
diff -r ed726703c08c toolkit/mozapps/installer/packager.mk
--- a/toolkit/mozapps/installer/packager.mk	Tue Jan 24 14:49:05 2017 -0500
+++ b/toolkit/mozapps/installer/packager.mk	Fri Feb 03 12:07:10 2017 +0100
@@ -132,8 +132,8 @@
 	(cd $(DIST)/$(MOZ_PKG_DIR) && $(TAR) --exclude=precomplete $(TAR_CREATE_FLAGS) - .) | \
 	  (cd $(DESTDIR)$(installdir) && tar -xf -)
 	$(NSINSTALL) -D $(DESTDIR)$(bindir)
-	$(RM) -f $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)
-	ln -s $(installdir)/$(MOZ_APP_NAME) $(DESTDIR)$(bindir)
+	$(RM) -f $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)-developer-edition
+	ln -s $(installdir)/$(MOZ_APP_NAME) $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)-developer-edition
 
 checksum:
 	mkdir -p `dirname $(CHECKSUM_FILE)`
diff -r ed726703c08c toolkit/xre/nsAppRunner.cpp
--- a/toolkit/xre/nsAppRunner.cpp	Tue Jan 24 14:49:05 2017 -0500
+++ b/toolkit/xre/nsAppRunner.cpp	Fri Feb 03 12:07:10 2017 +0100
@@ -3699,7 +3699,11 @@
 #endif
 
   // Set program name to the one defined in application.ini.
-  {
+  if (strcmp(gAppData->name, "Firefox") == 0) {
+    // HACK to make GNOME Shell distinguish this
+    // app from regular Firefox
+    g_set_prgname("firefox-developer-edition");
+  } else {
     nsAutoCString program(gAppData->name);
     ToLowerCase(program);
     g_set_prgname(program.get());
