diff -Nur gtvg-0.3.orig/src/callbacks.c gtvg-0.3/src/callbacks.c
--- gtvg-0.3.orig/src/callbacks.c	2007-09-24 03:10:22.000000000 -0700
+++ gtvg-0.3/src/callbacks.c	2011-03-08 00:30:14.136672256 -0800
@@ -3959,11 +3959,7 @@
 						}
 					}
 					if(show_reminder) {
-						if(gtk_status_icon_get_visible(status_icon)) {
-							favsched->notification = notify_notification_new_with_status_icon(title_str, desc, favsched->prog->channel->iconpath, status_icon);
-						} else {
-							favsched->notification = notify_notification_new(title_str, desc, favsched->prog->channel->iconpath, NULL);
-						}
+					        favsched->notification = notify_notification_new(title_str, desc, favsched->prog->channel->iconpath);
 						g_signal_connect(G_OBJECT(favsched->notification), "closed", G_CALLBACK(on_notification_closed), (gpointer) favsched);
 						g_free(title_str);
 						g_free(desc);
