diff -Naur buoh-0.8.2.orig/data/buoh-ui.xml buoh-0.8.2/data/buoh-ui.xml
--- buoh-0.8.2.orig/data/buoh-ui.xml	2014-12-19 18:35:42.417309811 +0100
+++ buoh-0.8.2/data/buoh-ui.xml	2014-12-19 18:36:27.767267954 +0100
@@ -31,7 +31,7 @@
       </menu>
 
       <menu name="HelpMenu" action="Help">
-	 <menuitem name="HelpContentsMenu" action="HelpContents"/>
+<!--	 <menuitem name="HelpContentsMenu" action="HelpContents"/> -->
 	 <menuitem name="HelpAboutMenu" action="HelpAbout"/>
       </menu>
    </menubar>
diff -Naur buoh-0.8.2.orig/src/buoh-window.c buoh-0.8.2/src/buoh-window.c
--- buoh-0.8.2.orig/src/buoh-window.c	2006-10-22 17:21:37.000000000 +0200
+++ buoh-0.8.2/src/buoh-window.c	2014-12-19 18:31:33.124916691 +0100
@@ -206,9 +206,9 @@
 	  G_CALLBACK (buoh_window_cmd_go_last) },
 
 	/* Help menu */
-	{ "HelpContents", GTK_STOCK_HELP, N_("_Contents"), "F1",
+/*	{ "HelpContents", GTK_STOCK_HELP, N_("_Contents"), "F1",
 	  N_("Display help for the Buoh online comic reader"),
-	  G_CALLBACK (buoh_window_cmd_help_contents) },
+	  G_CALLBACK (buoh_window_cmd_help_contents) },*/
 	{ "HelpAbout", GTK_STOCK_ABOUT, N_("_About"), NULL,
 	  N_("Display credits for the Buoh online comic reader creators"),
 	  G_CALLBACK (buoh_window_cmd_help_about) }
