diff -ur gambas2-2.13.1/gb.db.sqlite3/src/main.cpp gambas-2.0/gb.db.sqlite3/src/main.cpp
--- gambas2-2.13.1/gb.db.sqlite3/src/main.cpp	2009-05-26 20:41:45.000000000 +0200
+++ gambas-2.0/gb.db.sqlite3/src/main.cpp	2009-06-09 17:48:24.119432456 +0200
@@ -1008,7 +1008,7 @@
 {
 	char *fld;
 
-	fld = strchr(name, (int) FLD_SEP);
+	fld = strchr((char *) name, (int) FLD_SEP);
 	if (fld)
 	{															//Includes table identity
 		fld[0] = '.';

