diff -wbBur libvirt-0.9.0/src/qemu/qemu_migration.c libvirt-0.9.0.my/src/qemu/qemu_migration.c
--- libvirt-0.9.0/src/qemu/qemu_migration.c	2011-03-29 14:18:53.000000000 +0000
+++ libvirt-0.9.0.my/src/qemu/qemu_migration.c	2011-04-04 14:34:43.000000000 +0000
@@ -819,7 +819,7 @@
                                                unixfile);
     }
     else if (qemuCapsGet(qemuCaps, QEMU_CAPS_MIGRATE_QEMU_EXEC)) {
-        const char *args[] = { "nc", "-U", unixfile, NULL };
+        const char *args[] = { "nc.openbsd", "-U", unixfile, NULL };
         internalret = qemuMonitorMigrateToCommand(priv->mon, QEMU_MONITOR_MIGRATE_BACKGROUND, args);
     } else {
         internalret = -1;
diff -wbBur libvirt-0.9.0/src/remote/remote_driver.c libvirt-0.9.0.my/src/remote/remote_driver.c
--- libvirt-0.9.0/src/remote/remote_driver.c	2011-03-29 12:57:01.000000000 +0000
+++ libvirt-0.9.0.my/src/remote/remote_driver.c	2011-04-04 14:34:23.000000000 +0000
@@ -777,7 +777,7 @@
             cmd_argv[j++] = strdup ("none");
         }
         cmd_argv[j++] = strdup (priv->hostname);
-        cmd_argv[j++] = strdup (netcat ? netcat : "nc");
+        cmd_argv[j++] = strdup (netcat ? netcat : "nc.openbsd");
         cmd_argv[j++] = strdup ("-U");
         cmd_argv[j++] = strdup (sockname ? sockname :
                                 (flags & VIR_CONNECT_RO
