diff -wbBur libvirt-0.9.3/src/qemu/qemu_migration.c libvirt-0.9.3.my/src/qemu/qemu_migration.c
--- libvirt-0.9.3/src/qemu/qemu_migration.c	2011-07-04 06:45:14.000000000 +0400
+++ libvirt-0.9.3.my/src/qemu/qemu_migration.c	2011-07-04 15:03:15.000000000 +0400
@@ -1763,7 +1763,7 @@
         ret = qemuMonitorMigrateToUnix(priv->mon, background_flags,
                                                unixfile);
     } else if (qemuCapsGet(priv->qemuCaps, QEMU_CAPS_MIGRATE_QEMU_EXEC)) {
-        const char *args[] = { "nc", "-U", unixfile, NULL };
+        const char *args[] = { "nc.openbsd", "-U", unixfile, NULL };
         ret = qemuMonitorMigrateToCommand(priv->mon, QEMU_MONITOR_MIGRATE_BACKGROUND, args);
     } else {
         ret = -1;
diff -wbBur libvirt-0.9.3/src/remote/remote_driver.c libvirt-0.9.3.my/src/remote/remote_driver.c
--- libvirt-0.9.3/src/remote/remote_driver.c	2011-07-01 10:59:10.000000000 +0400
+++ libvirt-0.9.3.my/src/remote/remote_driver.c	2011-07-04 15:03:32.000000000 +0400
@@ -571,7 +571,7 @@
                                                 command,
                                                 username,
                                                 no_tty,
-                                                netcat ? netcat : "nc",
+                                                netcat ? netcat : "nc.openbsd",
                                                 sockname)))
             goto failed;
 
diff -wbBur libvirt-0.9.3/src/rpc/virnetsocket.c libvirt-0.9.3.my/src/rpc/virnetsocket.c
--- libvirt-0.9.3/src/rpc/virnetsocket.c	2011-07-01 10:59:10.000000000 +0400
+++ libvirt-0.9.3.my/src/rpc/virnetsocket.c	2011-07-04 15:03:00.000000000 +0400
@@ -597,7 +597,7 @@
         virCommandAddArgList(cmd, "-T", "-o", "BatchMode=yes",
                              "-e", "none", NULL);
     virCommandAddArgList(cmd, nodename,
-                         netcat ? netcat : "nc",
+                         netcat ? netcat : "nc.openbsd",
                          "-U", path, NULL);
 
     return virNetSocketNewConnectCommand(cmd, retsock);
