diff -ruN a/FoxDot/lib/Effects/NewEffects.py b/FoxDot/lib/Effects/NewEffects.py
--- a/FoxDot/lib/Effects/NewEffects.py	2021-06-03 18:34:22.000000000 +0200
+++ b/FoxDot/lib/Effects/NewEffects.py	2023-04-06 19:42:44.980845124 +0200
@@ -14,7 +14,7 @@
         self.lines = []
     def __call__(self, order=0):
         def decorator(effect):
-            effect_data = inspect.getargspec(effect) # Original args and defaults
+            effect_data = inspect.getfullargspec(effect) # Original args and defaults
             
             # Get filename from function name
             filename = "{}.scd".format(effect.__name__)# filename
diff -ruN a/FoxDot/lib/Patterns/Main.py b/FoxDot/lib/Patterns/Main.py
--- a/FoxDot/lib/Patterns/Main.py	2021-06-03 18:34:22.000000000 +0200
+++ b/FoxDot/lib/Patterns/Main.py	2023-04-06 19:43:25.204310325 +0200
@@ -34,7 +34,7 @@
         for i in range(LCM(*[len(arg) for arg in args if (hasattr(arg, '__len__') and not isinstance(arg, PGroup))])):
             pat |= f(*[(arg[i] if isinstance(arg, Pattern) else arg) for arg in args])
         return pat
-    new_function.argspec = inspect.getargspec(f)
+    new_function.argspec = inspect.getfullargspec(f)
     return new_function
 
 # TODO -- if it isn't looped, return the original if it is a group
@@ -58,7 +58,7 @@
             pat |= f(self, *[(modi(arg, i) if not isinstance(arg, PGroup) else arg) for arg in args])
         return pat
 
-    new_function.argspec = inspect.getargspec(f)
+    new_function.argspec = inspect.getfullargspec(f)
     return new_function
 
 def PatternMethod(f):
diff -ruN a/FoxDot/lib/Repeat.py b/FoxDot/lib/Repeat.py
--- a/FoxDot/lib/Repeat.py	2021-06-03 18:34:22.000000000 +0200
+++ b/FoxDot/lib/Repeat.py	2023-04-06 19:41:56.644020595 +0200
@@ -424,7 +424,7 @@
 
         # Check if a method has the _beat_ keyword argument
 
-        if "_beat_" in inspect.getargspec(self.method).args:
+        if "_beat_" in inspect.getfullargspec(self.method).args:
 
             self.kwargs["_beat_"] = None
 
diff -ruN a/FoxDot/lib/TempoClock.py b/FoxDot/lib/TempoClock.py
--- a/FoxDot/lib/TempoClock.py	2021-06-03 18:34:22.000000000 +0200
+++ b/FoxDot/lib/TempoClock.py	2023-04-06 19:42:24.067443408 +0200
@@ -765,11 +765,11 @@
 
         try:
 
-            function = inspect.getargspec(item)
+            function = inspect.getfullargspec(item)
 
         except TypeError:
 
-            function = inspect.getargspec(item.__call__)
+            function = inspect.getfullargspec(item.__call__)
 
         # If the item can't take arbitrary keywords, check any kwargs are valid
 
