diff -Naur gloobus-preview-0.4.5-ubuntu11~ppa290.orig/src/gloobus-preview-configuration gloobus-preview-0.4.5-ubuntu11~ppa290/src/gloobus-preview-configuration
--- gloobus-preview-0.4.5-ubuntu11~ppa290.orig/src/gloobus-preview-configuration	2014-01-22 21:54:41.555697168 +0100
+++ gloobus-preview-0.4.5-ubuntu11~ppa290/src/gloobus-preview-configuration	2014-01-22 21:34:36.867164953 +0100
@@ -6,10 +6,9 @@
 import os
 import sys
 import glob
-import gtk
+from gi.repository import Gtk, GConf
 import signal
 import gettext
-import gconf
 #import config
 
 #=========================== CLASS COLORS =======================================================#
@@ -69,54 +68,54 @@
 	def __init__(self):
 		self.config = config_load()
 		
-		self.client = gconf.client_get_default()
+		self.client = GConf.Client.get_default()
 		self.gconf_win_layout = self.client.get_string('/apps/metacity/general/button_layout')
 
 		#================= Show In Taskbar ================= #
-		self.entry1 = gtk.CheckButton(label=_("Show in TaskBar"), use_underline=False)  
+		self.entry1 = Gtk.CheckButton(label=_("Show in TaskBar"), use_underline=False)  
 		self.entry1.set_tooltip_markup(_("When enabled gloobus-preview will be\nshown in the <b>taskbar</b>"))
 		self.entry1.set_active(self.config.getboolean("Main","taskbar"))
 
 		#================= Allways on top ================= #
-		self.entry2 = gtk.CheckButton(label=_("Always on top"), use_underline=False)  
+		self.entry2 = Gtk.CheckButton(label=_("Always on top"), use_underline=False)  
 		self.entry2.set_tooltip_markup(_("When enabled gloobus-preview will be\n<b>always on top</b> of the other windows"))
 		self.entry2.set_active(self.config.getboolean("Main","ontop"))
 		win.set_keep_above(self.config.getboolean("Main","ontop"))
 
 
 		#================= Quit on lose focus ================= #
-		self.entry3 = gtk.CheckButton(label=_("Quit on lose focus"), use_underline=False)  
+		self.entry3 = Gtk.CheckButton(label=_("Quit on lose focus"), use_underline=False)  
 		self.entry3.set_tooltip_markup(_("When enabled, gloobus-preview will <b>exit</b>\nwhen it loses focus"))
 		self.entry3.set_active(self.config.getboolean("Main","focus"))
 
 		#================= Win bar layout ================= #
-		self.entry4 = gtk.CheckButton(label=_("Window Bar Layout inverted"), use_underline=False)  
+		self.entry4 = Gtk.CheckButton(label=_("Window Bar Layout inverted"), use_underline=False)  
 		self.entry4.set_tooltip_markup(_("Layout of the buttons\nclose button on the right/left?"))
 		self.entry4.set_active(self.config.getboolean("Main","winbar_layout"))
 
 		#================= Use gtk theme ================= #
-		self.theme_gtk_entry = gtk.CheckButton(label=_("Use gtk theme"), use_underline=False)  
+		self.theme_gtk_entry = Gtk.CheckButton(label=_("Use gtk theme"), use_underline=False)  
 		self.theme_gtk_entry.set_tooltip_markup(_("When enabled, gloobus-preview will use system GTK theme"))
 		self.theme_gtk_entry.set_active(self.config.getboolean("Theme","gtk"))
 	
 		#=================== BUTTONS  ===================== #
-		container0 = gtk.HBox(True,1)  #Save and close buttons
+		container0 = Gtk.HBox(True,1)  #Save and close buttons
 
-		ok_button = gtk.Button(_("Save"))
-		cancel_button = gtk.Button(_("Cancel"))
+		ok_button = Gtk.Button(_("Save"))
+		cancel_button = Gtk.Button(_("Cancel"))
 
 		container0.add(cancel_button)			
 		container0.add(ok_button)				
-		cancel_button.connect('clicked', gtk.main_quit)
+		cancel_button.connect('clicked', Gtk.main_quit)
 		ok_button.connect('clicked', self.save_conf)
 
 		#============ GENERAL CONTAINER ================== #
 
-		general_container = gtk.VBox(False,1)
+		general_container = Gtk.VBox(False,1)
 
-		title = gtk.Label()
+		title = Gtk.Label()
 		title.set_markup(_("<b>Main settings</b>"))
-		align2 = gtk.Alignment(0,0,0,0)
+		align2 = Gtk.Alignment.new(0,0,0,0)
 		align2.set_padding(0,5,0,0)
 		align2.add(title)
 
@@ -128,11 +127,11 @@
 		if self.gconf_win_layout == None:
 			general_container.add(self.entry4)	#Always on top
 
-		general_container.add(gtk.HSeparator())
-		general_container.add(gtk.Label())	#White Space
+		general_container.add(Gtk.HSeparator())
+		general_container.add(Gtk.Label())	#White Space
 		general_container.add(container0)	#Close & save buttons
 
-		align = gtk.Alignment(0,0,0,0);
+		align = Gtk.Alignment.new(0,0,0,0);
 		align.set_padding(10,5,10,10)
 
 		align.add(general_container)
@@ -155,7 +154,7 @@
 
 		config_save(self.config)
 		
-		gtk.main_quit()
+		Gtk.main_quit()
 
 
 #==================================== GLOBAL FUNCTIONS ===============================================#		
@@ -232,13 +231,13 @@
 g.debug("Path: "   + abspath)
 g.debug("Config File: " + CONFIG_PATH)
 
-win = gtk.Window()
+win = Gtk.Window()
 win.set_title(_("Gloobus-Preview Configuration"))
 #win.set_default_size(400,150)
-win.connect('delete-event', gtk.main_quit)
+win.connect('delete-event', Gtk.main_quit)
 
 
 gui = GUI()
 
-gtk.main()
+Gtk.main()
 
