--- a/contrib/dist/tor.service	2020-10-08 21:12:40.414833733 +0200
+++ b/contrib/dist/tor.service	2020-10-08 21:12:42.751547264 +0200
@@ -29,7 +29,7 @@
 ReadWriteDirectories=-/var/lib/tor
 ReadWriteDirectories=-/var/log/tor
 NoNewPrivileges=yes
-CapabilityBoundingSet=CAP_SETUID CAP_SETGID CAP_NET_BIND_SERVICE
+CapabilityBoundingSet=CAP_SETUID CAP_SETGID CAP_NET_BIND_SERVICE CAP_DAC_READ_SEARCH
 
 [Install]
 WantedBy=multi-user.target
