diff --color -aur ipv6calc-2.2.0-old/ipv6logconv/ipv6logconv.c ipv6calc-2.2.0-new/ipv6logconv/ipv6logconv.c
--- ipv6calc-2.2.0-old/ipv6logconv/ipv6logconv.c	2019-10-11 08:15:53.000000000 +0300
+++ ipv6calc-2.2.0-new/ipv6logconv/ipv6logconv.c	2020-09-01 18:28:10.697766271 +0300
@@ -58,7 +58,7 @@
 /* LRU cache */
 
 #define CACHE_LRU_SIZE 200
-
+int cache_lru_limit;
 static int      cache_lru_max = 0;
 static int      cache_lru_last = 0;
 static char     cache_lru_key_token[CACHE_LRU_SIZE][NI_MAXHOST];
diff --color -aur ipv6calc-2.2.0-old/ipv6logconv/ipv6logconv.h ipv6calc-2.2.0-new/ipv6logconv/ipv6logconv.h
--- ipv6calc-2.2.0-old/ipv6logconv/ipv6logconv.h	2019-10-11 08:15:53.000000000 +0300
+++ ipv6calc-2.2.0-new/ipv6logconv/ipv6logconv.h	2020-09-01 18:28:24.011099499 +0300
@@ -20,8 +20,7 @@
 #define DEBUG_ipv6logconv_general      0x00000001l
 #define DEBUG_ipv6logconv_processing   0x00000002l
 
-/* prototyping */
-int cache_lru_limit;
+extern int cache_lru_limit;
 
 extern int feature_reg;
 extern int feature_ieee;
