--- ogmrip-1.0.1/src/ogmrip-spell-dialog.c.orig	2015-01-10 13:39:43.000000000 +0000
+++ ogmrip-1.0.1/src/ogmrip-spell-dialog.c	2017-11-23 08:47:19.352081545 +0000
@@ -315,14 +315,14 @@
         enchant_dict_add_to_session (dialog->priv->dict, word, len);
         break;
       case OGMRIP_SPELL_RESPONSE_ADD_WORD:
-        enchant_dict_add_to_personal (dialog->priv->dict, word, len);
+        enchant_dict_add (dialog->priv->dict, word, len);
         break;
       default:
         break;
     }
 
     if (suggs && n_suggs)
-      enchant_dict_free_suggestions (dialog->priv->dict, suggs);
+      enchant_dict_free_string_list (dialog->priv->dict, suggs);
   }
 
   return status;
