diff --git a/tests/asyncio/test_asyncio_client.py b/tests/asyncio/test_asyncio_client.py
index d34fd05..c47238d 100644
--- a/tests/asyncio/test_asyncio_client.py
+++ b/tests/asyncio/test_asyncio_client.py
@@ -241,7 +241,7 @@ class TestAsyncClient(unittest.TestCase):
 
         c = asyncio_client.AsyncClient()
         c.start_background_task(foo, 'bar')
-        pending = asyncio.Task.all_tasks()
+        pending = asyncio.all_tasks(loop=asyncio.get_event_loop())
         asyncio.get_event_loop().run_until_complete(asyncio.wait(pending))
         assert r == ['bar']
 
diff --git a/tests/asyncio/test_asyncio_server.py b/tests/asyncio/test_asyncio_server.py
index 2faecca..5f9272a 100644
--- a/tests/asyncio/test_asyncio_server.py
+++ b/tests/asyncio/test_asyncio_server.py
@@ -1004,7 +1004,7 @@ class TestAsyncServer(unittest.TestCase):
 
         s = asyncio_server.AsyncServer()
         s.start_background_task(foo, 'bar')
-        pending = asyncio.Task.all_tasks()
+        pending = asyncio.all_tasks(loop=asyncio.get_event_loop())
         asyncio.get_event_loop().run_until_complete(asyncio.wait(pending))
         assert r == ['bar']
 
