diff -aur midori-0.4.7/katze/midori-paths.vala midori-0.4.7.patch/katze/midori-paths.vala
--- midori-0.4.7/katze/midori-paths.vala	2012-09-19 23:17:43.000000000 +0200
+++ midori-0.4.7.patch/katze/midori-paths.vala	2012-09-24 20:36:04.585209437 +0200
@@ -74,7 +74,7 @@
             else if (mode == RuntimeMode.PRIVATE || mode == RuntimeMode.APP) {
                 /* Use mock folders in development builds */
                 if ("." in MIDORI_VERSION_SUFFIX)
-                    config_dir = cache_dir = user_data_dir = config_base;
+                    config_dir = cache_dir = user_data_dir = config_base ?? Path.DIR_SEPARATOR_S;
                 else
                     config_dir = cache_dir = user_data_dir = "/";
                 readonly_dir = config_base ?? Path.build_path (Path.DIR_SEPARATOR_S,
