--- src-setup/Audio.cpp
+++ src-setup/Audio.cpp
@@ -7,13 +7,13 @@
  */
 #include "Audio.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
 
 #ifndef _WIN32
 #include <unistd.h>
-#include <errno.h>
+#include <cerrno>
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
@@ -23,6 +23,7 @@
 #include <AL/alc.h>
 #include <AL/alut.h>
 
+
 //====================================================================
 Audio::Audio()
 {
--- src-setup/ConfigFile.cpp
+++ src-setup/ConfigFile.cpp
@@ -7,8 +7,8 @@
  */
 #include "ConfigFile.h"
 
-#include <stdio.h>
-#include <stdlib.h>
+#include <cstdio>
+#include <cstdlib>
 
 #include <qapp.h>
 #include <qfiledialog.h>
--- src-setup/HelpPage.cpp
+++ src-setup/HelpPage.cpp
@@ -7,7 +7,7 @@
  */
 #include "HelpPage.h"
 
-#include <stdio.h>
+#include <cstdio>
 
 #include <qwidget.h>
 #include <qapplication.h>
--- src-setup/HiScoreWidget.cpp
+++ src-setup/HiScoreWidget.cpp
@@ -1,7 +1,7 @@
 #include "HiScoreWidget.h"
 
-#include <stdio.h>
-#include <stdlib.h>
+#include <cstdio>
+#include <cstdlib>
 
 #include <qfileinfo.h>
 
--- src-setup/HiScoreWidget.h
+++ src-setup/HiScoreWidget.h
@@ -8,7 +8,7 @@
 #ifndef HiScoreWidget_h
 #define HiScoreWidget_h
 
-#include <time.h>
+#include <ctime>
 
 #include <qlabel.h>
 #include <qlayout.h>
--- src-setup/MainWidget.cpp
+++ src-setup/MainWidget.cpp
@@ -7,8 +7,8 @@
  */
 #include "MainWidget.h"
 
-#include <stdio.h>
-#include <stdlib.h>
+#include <cstdio>
+#include <cstdlib>
 
 #include <qapp.h>
 #include <qfile.h>
--- src-setup/PlayList.cpp
+++ src-setup/PlayList.cpp
@@ -7,8 +7,8 @@
  */
 #include "PlayList.h"
 
-#include <stdio.h>
-#include <stdlib.h>
+#include <cstdio>
+#include <cstdlib>
 
 #include <qapp.h>
 #include <qtooltip.h>
--- src/Ammo.cpp
+++ src/Ammo.cpp
@@ -7,7 +7,7 @@
  */
 #include "Ammo.h"
 
-#include <stdio.h>
+#include <cstdio>
 
 
 int ActiveAmmo::ammoCount = 0;
--- src/AudioOpenAL.cpp
+++ src/AudioOpenAL.cpp
@@ -15,14 +15,14 @@
 	void	alcDestroyContext(void*) { alExit(); }
 #endif //macintosh
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <stdarg.h>
-#include <string.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstdarg>
+#include <cstring>
 
 #ifndef _WIN32
 	#include <unistd.h>
-	#include <errno.h>
+	#include <cerrno>
 	#include <fcntl.h>
 	#include <sys/stat.h>
 	#include <sys/types.h>
--- src/AudioSDLMixer.cpp
+++ src/AudioSDLMixer.cpp
@@ -9,14 +9,14 @@
  
 #include "AudioSDLMixer.h"
 
-#include <stdio.h>
-#include <stdlib.h>
+#include <cstdio>
+#include <cstdlib>
 #include <stdarg.h>
-#include <string.h>
+#include <cstring>
 
 #ifndef _WIN32
 #include <unistd.h>
-#include <errno.h>
+#include <cerrno>
 #include <fcntl.h>
 #include <sys/stat.h>
 #endif // _WIN32
--- src/Config.cpp
+++ src/Config.cpp
@@ -10,9 +10,9 @@
 #include "HiScore.h"
 
 //====================================================================
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
 
 #include "NCString.h"
 
--- src/EnemyAircraft.cpp
+++ src/EnemyAircraft.cpp
@@ -7,9 +7,9 @@
  */
 #include "EnemyAircraft.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
 
 #include "Config.h"
 
--- src/EnemyAircraft_Boss00.cpp
+++ src/EnemyAircraft_Boss00.cpp
@@ -7,9 +7,9 @@
  */
 #include "EnemyAircraft_Boss00.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
 
 #include "Config.h"
 
--- src/EnemyAircraft_Boss01.cpp
+++ src/EnemyAircraft_Boss01.cpp
@@ -7,9 +7,9 @@
  */
 #include "EnemyAircraft_Boss01.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
 
 #include "Config.h"
 
--- src/EnemyAircraft_Gnat.cpp
+++ src/EnemyAircraft_Gnat.cpp
@@ -7,9 +7,9 @@
  */
 #include "EnemyAircraft_Gnat.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
 
 #include "Config.h"
 
--- src/EnemyAircraft_Omni.cpp
+++ src/EnemyAircraft_Omni.cpp
@@ -7,9 +7,9 @@
  */
 #include "EnemyAircraft_Omni.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
 
 #include "Config.h"
 
--- src/EnemyAircraft_RayGun.cpp
+++ src/EnemyAircraft_RayGun.cpp
@@ -7,9 +7,9 @@
  */
 #include "EnemyAircraft_RayGun.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
 
 #include "Config.h"
 
--- src/EnemyAircraft_Straight.cpp
+++ src/EnemyAircraft_Straight.cpp
@@ -7,9 +7,9 @@
  */
 #include "EnemyAircraft_Straight.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
 
 #include "Config.h"
 
--- src/EnemyAircraft_Tank.cpp
+++ src/EnemyAircraft_Tank.cpp
@@ -7,9 +7,9 @@
  */
 #include "EnemyAircraft_Tank.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
 
 #include "Config.h"
 
--- src/EnemyAmmo.cpp
+++ src/EnemyAmmo.cpp
@@ -7,9 +7,9 @@
  */
 #include "EnemyAmmo.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
 
 #include "compatibility.h"
 #include <GL/gl.h>
--- src/EnemyFleet.cpp
+++ src/EnemyFleet.cpp
@@ -7,8 +7,8 @@
  */
 #include "EnemyFleet.h"
 
-#include <stdio.h>
-#include <math.h>
+#include <cstdio>
+#include <cmath>
 
 #include "compatibility.h"
 #include <GL/gl.h>
--- src/Explosions.cpp
+++ src/Explosions.cpp
@@ -7,7 +7,7 @@
  */
 #include "Explosions.h"
 
-#include <stdlib.h>
+#include <cstdlib>
 
 #include "compatibility.h"
 #include <GL/gl.h>
--- src/Global.h
+++ src/Global.h
@@ -8,8 +8,8 @@
 #ifndef Global_h
 #define Global_h
 
-#include <stdio.h>
-#include <time.h>
+#include <cstdio>
+#include <ctime>
 
 #include "define.h"
 #include "TexFont.h"
--- src/GroundMetal.cpp
+++ src/GroundMetal.cpp
@@ -8,7 +8,7 @@
 #include "GroundMetal.h"
 
 #include <GL/glpng.h>
-#include <math.h>
+#include <cmath>
 
 #include "extern.h"
 #include "Global.h"
--- src/GroundMetalSegment.cpp
+++ src/GroundMetalSegment.cpp
@@ -7,7 +7,7 @@
  */
 #include "GroundMetalSegment.h"
 
-#include <math.h>
+#include <cmath>
 
 #include "compatibility.h"
 #include <GL/gl.h>
--- src/GroundSea.cpp
+++ src/GroundSea.cpp
@@ -7,7 +7,7 @@
  */
 #include "GroundSea.h"
 
-#include <stdlib.h>
+#include <cstdlib>
 #include <GL/glpng.h>
 
 #include "extern.h"
--- src/GroundSeaSegment.cpp
+++ src/GroundSeaSegment.cpp
@@ -7,7 +7,7 @@
  */
 #include "GroundSeaSegment.h"
 
-#include <math.h>
+#include <cmath>
 
 #include "compatibility.h"
 #include <GL/gl.h>
--- src/HeroAircraft.cpp
+++ src/HeroAircraft.cpp
@@ -7,9 +7,9 @@
  */
 #include "HeroAircraft.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
 
 #include "compatibility.h"
 #include <GL/gl.h>
--- src/HeroAmmo.cpp
+++ src/HeroAmmo.cpp
@@ -7,8 +7,8 @@
  */
 #include "HeroAmmo.h"
 
-#include <stdio.h>
-#include <stdlib.h>
+#include <cstdio>
+#include <cstdlib>
 
 #include "compatibility.h"
 #include <GL/gl.h>
--- src/HiScore.cpp
+++ src/HiScore.cpp
@@ -11,9 +11,9 @@
 #include "HeroAircraft.h"
 
 //====================================================================
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
 
 HiScore	*HiScore::instance = 0;
 
--- src/HiScore.h
+++ src/HiScore.h
@@ -8,8 +8,8 @@
 #ifndef HiScore_h
 #define HiScore_h
 
-#include <stdio.h>
-#include <time.h>
+#include <cstdio>
+#include <ctime>
 
 #include "define.h"
 
--- src/MainGL.cpp
+++ src/MainGL.cpp
@@ -7,10 +7,10 @@
  */
 #include "MainGL.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <cmath>
 
 #include "compatibility.h"
 #include <GL/gl.h>
--- src/MainGLUT.cpp
+++ src/MainGLUT.cpp
@@ -9,7 +9,7 @@
 
 #ifdef USE_GLUT
 
-#include <stdio.h>
+#include <cstdio>
 #include <unistd.h>
 
 #include <GL/glut.h>
--- src/MainSDL.cpp
+++ src/MainSDL.cpp
@@ -9,8 +9,8 @@
 
 #ifdef USE_SDL
 
-#include <stdlib.h>
-#include <math.h>
+#include <cstdlib>
+#include <cmath>
 #include <SDL.h>
 
 #include "compatibility.h"
--- src/MainSDL_Event.cpp
+++ src/MainSDL_Event.cpp
@@ -9,9 +9,9 @@
 
 #ifdef USE_SDL
 
-#include <stdlib.h>
-#include <stdio.h>
-#include <string.h>
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
 
 #include "Config.h"
 
--- src/MenuGL.cpp
+++ src/MenuGL.cpp
@@ -7,9 +7,9 @@
  */
 #include "MenuGL.h"
 
-#include <stdlib.h>
-#include <string.h>
-#include <math.h>
+#include <cstdlib>
+#include <cstring>
+#include <cmath>
 
 #include "compatibility.h"
 #include <GL/gl.h>
--- src/NCString.cpp
+++ src/NCString.cpp
@@ -1,10 +1,11 @@
 #include "NCString.h"
 
-#include <stdio.h>
-#include <ctype.h>
-#include <stdlib.h>
-#include <string.h>
-#include <limits.h>
+#include <cstdio>
+#include <cctype>
+#include <cstdlib>
+#include <cstring>
+#include <climits>
+#include <cstdarg>
 
 /**
  * ctor
@@ -1463,7 +1464,7 @@
 NCString operator + (const char 	*s1, const NCString &s2)   {   NCString tmp(s1);   tmp += s2;  return tmp; }
 NCString operator + (const NCString &s1, const char 	*s2)   {   NCString tmp(s1);   tmp += s2;  return tmp; }
 //-----------------------------------------------------
-ostream& operator << (ostream &os, const NCString &s1)
+std::ostream& operator << (std::ostream &os, const NCString &s1)
 {
 	return os << (const char*)s1;
 }
--- src/NCString.h
+++ src/NCString.h
@@ -1,7 +1,7 @@
 #ifndef NCString_h
 #define NCString_h
 
-#include <stream.h>
+#include <iostream>
 
 typedef unsigned int uint;
 typedef unsigned short ushort;
@@ -123,7 +123,7 @@
 /** add     @relates NCString */	NCString operator + (const NCString &s1, const NCString &s2);
 /** add     @relates NCString */	NCString operator + (const char     *s1, const NCString &s2);
 /** add     @relates NCString */	NCString operator + (const NCString &s1, const char     *s2);
-/** stream  @relates NCString */	ostream& operator << (ostream &os, const NCString &s1);
+/** stream  @relates NCString */	std::ostream& operator << (std::ostream &os, const NCString &s1);
 
 #endif // NCString_h
 
--- src/PowerUps.cpp
+++ src/PowerUps.cpp
@@ -7,9 +7,9 @@
  */
 #include "PowerUps.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cmath>
 
 #include "compatibility.h"
 #include <GL/gl.h>
--- src/ScreenItem.cpp
+++ src/ScreenItem.cpp
@@ -7,7 +7,7 @@
  */
 #include "ScreenItem.h"
 
-#include <stdio.h>
+#include <cstdio>
 
 
 int ScreenItem::itemCount = 0;
--- src/ScreenItemAdd.cpp
+++ src/ScreenItemAdd.cpp
@@ -7,8 +7,8 @@
  */
 #include "ScreenItemAdd.h"
 
-#include <stdio.h>
-#include <stdlib.h>
+#include <cstdio>
+#include <cstdlib>
 
 #include "define.h"
 #include "Global.h"
--- src/StatusDisplay.cpp
+++ src/StatusDisplay.cpp
@@ -7,10 +7,10 @@
  */
 #include "StatusDisplay.h"
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
-#include <math.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <cmath>
 
 #include "compatibility.h"
 #include <GL/gl.h>
--- src/TexFont.cpp
+++ src/TexFont.cpp
@@ -5,12 +5,12 @@
    provided without guarantee or warrantee expressed or  implied. This
    program is -not- in the public domain. */
 #include "compatibility.h"
-#include <assert.h>
-#include <ctype.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <string.h>
-#include <math.h>
+#include <cassert>
+#include <cctype>
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
+#include <cmath>
 #include <GL/glu.h>
 #include "TexFont.h"
 
--- src/extern.h
+++ src/extern.h
@@ -8,7 +8,7 @@
 #ifndef extern_h
 #define extern_h
 
-#include <stdio.h>
+#include <cstdio>
 
 const char *dataLoc(const char *filename, bool doCheck = true);
 const char *alterPathForPlatform(char *filename);
--- src/main.cpp
+++ src/main.cpp
@@ -8,15 +8,15 @@
 #include "compatibility.h"
 #include <GL/gl.h>
 
-#include <stdlib.h>
-#include <stdio.h>
-#include <string.h>
-#include <time.h>
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
+#include <ctime>
 
 #ifndef _WIN32
 #include <unistd.h>
 #include <fcntl.h>
-#include <errno.h>
+#include <cerrno>
 #include <sys/stat.h>
 #endif // _WIN32
 
--- support/glpng/include/GL/glpng.h
+++ support/glpng/include/GL/glpng.h
@@ -23,7 +23,11 @@
 #ifndef _GLPNG_H_
 #define _GLPNG_H_
 
-#include <stdio.h>
+#ifdef __cplusplus
+#  include <cstdio>
+#else
+#  include <stdio.h>
+#endif
 
 #ifdef __cplusplus
 extern "C" {
--- support/glpng/src/glpng.c
+++ support/glpng/src/glpng.c
@@ -4,6 +4,8 @@
 #endif
 
 #include <GL/glpng.h>
+/* needed to get the glext prototypes */
+#define GL_GLEXT_PROTOTYPES
 #include <GL/gl.h>
 #include <stdlib.h>
 #include <math.h>
@@ -47,7 +49,7 @@
 
 static char gammaExplicit = 0;	/*if  */
 
-static void checkForGammaEnv()
+static void checkForGammaEnv(void)
 {
 	double viewingGamma;
 	char *gammaEnv = getenv("VIEWING_GAMMA");
@@ -91,6 +93,7 @@
 	}
 }
 
+#ifdef _WIN32
 static int ExtSupported(const char *x) {
 	static const GLubyte *ext = NULL;
 	const char *c;
@@ -107,6 +110,7 @@
 
 	return 0;
 }
+#endif
 
 #define GET(o) ((int)*(data + (o)))
 
