diff --git a/configure.ac b/configure.ac
index af66e0e..0a0c391 100644
--- a/configure.ac
+++ b/configure.ac
@@ -133,6 +133,17 @@ fi
 AC_SUBST(LIBNOTIFY_CFLAGS)
 AC_SUBST(LIBNOTIFY_LIBS)
 
+dnl -- Check for D-Bus -----------------------------------------
+AC_ARG_WITH(dbus-sys,
+        [AC_HELP_STRING([--with-dbus-sys=<dir>],
+        [where D-BUS system.d directory is])])
+if ! test -z "$with_dbus_sys" ; then
+        DBUS_SYS_DIR="$with_dbus_sys"
+else
+        DBUS_SYS_DIR='${sysconfdir}/dbus-1/system.d'
+fi
+AC_SUBST(DBUS_SYS_DIR)
+
 dnl -- check for glib/gobject (required) --------------------------------------
 PKG_CHECK_MODULES(CPUFREQ_SELECTOR, glib-2.0 >= $GLIB_REQUIRED gobject-2.0 >= $GLIB_REQUIRED)
 AC_SUBST(CPUFREQ_SELECTOR_CFLAGS)
diff --git a/cpufreq/src/cpufreq-selector/Makefile.am b/cpufreq/src/cpufreq-selector/Makefile.am
index 27dfac6..52f2d6e 100644
--- a/cpufreq/src/cpufreq-selector/Makefile.am
+++ b/cpufreq/src/cpufreq-selector/Makefile.am
@@ -50,7 +50,7 @@ dbus_services_in_files = org.mate.CPUFreqSelector.service.in
 polkit_in_files = org.mate.cpufreqselector.policy.in
 
 dbus_servicesdir = $(datadir)/dbus-1/system-services
-dbus_confdir = $(sysconfdir)/dbus-1/system.d
+dbus_confdir = $(DBUS_SYS_DIR)
 polkitdir = $(datadir)/polkit-1/actions
 
 if HAVE_POLKIT
