From 06d7914bd5da89335c95793a3bc1903675f7a3b8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Luttringer?= <seblu@seblu.net>
Date: Sat, 8 Oct 2022 17:31:34 +0200
Subject: [PATCH] Disable online version check

---
 internal/core/build_info.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/internal/core/build_info.go b/internal/core/build_info.go
index c3f7799e..be80db7d 100644
--- a/internal/core/build_info.go
+++ b/internal/core/build_info.go
@@ -57,6 +57,7 @@ func (b *BuildInfo) GetUserAgent() string {
 }
 
 func (b *BuildInfo) checkVersion(ctx context.Context) {
+	return
 	if !b.IsRelease() || ExtractEnv(ctx, scwDisableCheckVersionEnv) == "true" {
 		ExtractLogger(ctx).Debug("skipping check version")
 		return
-- 
2.37.3

