From 9815ef3b02dfc05ed2e2dbe40d017b91f4f4fa41 Mon Sep 17 00:00:00 2001
From: James Booth <boothj5@gmail.com>
Date: Mon, 24 Mar 2014 21:02:10 +0000
Subject: [PATCH] Change libotr 4.0.0 code to use only OTR spec v2

---
 src/otr/otrlibv4.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/otr/otrlibv4.c b/src/otr/otrlibv4.c
index 80013af..01afa86 100644
--- a/src/otr/otrlibv4.c
+++ b/src/otr/otrlibv4.c
@@ -29,13 +29,13 @@
 OtrlPolicy
 otrlib_policy(void)
 {
-    return OTRL_POLICY_ALLOW_V1 | OTRL_POLICY_ALLOW_V2 | OTRL_POLICY_ALLOW_V3;
+    return OTRL_POLICY_ALLOW_V1 | OTRL_POLICY_ALLOW_V2;
 }
 
 char *
 otrlib_start_query(void)
 {
-    return "?OTR?v23?";
+    return "?OTR?v2?";
 }
 
 static const char*
-- 
1.9.1

