From e6e2837764ae430662220cb0df18089a681a0364 Mon Sep 17 00:00:00 2001
From: leigh123linux <leigh123linux@googlemail.com>
Date: Thu, 9 Jun 2016 10:06:34 +0100
Subject: [PATCH] fix build with kernel-4.7

---
 cpufreq/src/cpufreq-monitor.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/cpufreq/src/cpufreq-monitor.c b/cpufreq/src/cpufreq-monitor.c
index 36fbb4e15..e25d0d506 100644
--- a/cpufreq/src/cpufreq-monitor.c
+++ b/cpufreq/src/cpufreq-monitor.c
@@ -23,6 +23,7 @@
 #include "config.h"
 
 #include <cpufreq.h>
+#include <linux/version.h>
 #include <stdlib.h>
 
 #include "cpufreq-monitor.h"
@@ -76,7 +77,11 @@ monitor_run (CPUFreqMonitor *monitor)
   if (!policy)
     {
       /* Check whether it failed because cpu is not online. */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 7, 0)
       if (!cpufreq_cpu_exists (monitor->cpu))
+#else
+      if (cpupower_is_cpu_online (monitor->cpu))
+#endif
         {
           monitor->online = FALSE;
           return TRUE;
-- 
2.14.2

