diff -aur HTTP-4000.2.12/HTTP.cabal HTTP-4000.2.12.new/HTTP.cabal
--- HTTP-4000.2.12/HTTP.cabal	2005-05-06 06:32:53.000000000 +0200
+++ HTTP-4000.2.12.new/HTTP.cabal	2014-07-12 16:22:44.083433261 +0200
@@ -85,7 +85,7 @@
                  Network.HTTP.Utils
                  Paths_HTTP
   GHC-options: -fwarn-missing-signatures -Wall
-  Build-depends: base >= 2 && < 4.8, network < 2.5, parsec
+  Build-depends: base >= 2 && < 4.8, network < 2.6, parsec
   Extensions: FlexibleInstances
   if flag(old-base)
     Build-depends: base < 3
