From 8a9001bdf8a367968922ca8193af56f5e22ec885 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 17 Feb 2016 15:56:26 +0100
Subject: [PATCH] Better handling the background of logout dialog

---
 lxsession-logout/lxsession-logout.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/lxsession-logout/lxsession-logout.c b/lxsession-logout/lxsession-logout.c
index 8141b19..698b7d2 100644
--- a/lxsession-logout/lxsession-logout.c
+++ b/lxsession-logout/lxsession-logout.c
@@ -629,9 +629,7 @@ int main(int argc, char * argv[])
     gtk_container_add(GTK_CONTAINER(alignment), center_area);
 
 #ifdef USE_GTK3
-    GtkStyle* style = gtk_widget_get_style (window);
-    GdkColor color = style->bg[GTK_STATE_NORMAL];
-    gtk_widget_modify_bg(center_area, GTK_STATE_NORMAL, &color);
+    gtk_style_context_add_class (gtk_widget_get_style_context (center_area), GTK_STYLE_CLASS_BACKGROUND);
 #endif
 
     GtkWidget* center_vbox = gtk_vbox_new(FALSE, 6);
-- 
2.7.1

