From daed110fb6f5d4309aaac4982f8e8712b809f81f Mon Sep 17 00:00:00 2001
From: Jiachen YANG <farseerfc@gmail.com>
Date: Tue, 15 Dec 2020 11:42:26 +0900
Subject: [PATCH] fix install_fcitx translation files

---
 scripts/install_fcitx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/install_fcitx b/scripts/install_fcitx
index d4aa204e..493f1e1c 100755
--- a/scripts/install_fcitx
+++ b/scripts/install_fcitx
@@ -3,7 +3,7 @@
 _bldtype="${_bldtype:-Debug}"
 PREFIX="${PREFIX:-/usr}"
 
-for mofile in "out_linux/${_bldtype}/gen/unix/fcitx/po/*.mo"
+for mofile in out_linux/${_bldtype}/gen/unix/fcitx/po/*.mo
 do
     filename=`basename $mofile`
     lang=${filename/.mo/}
-- 
2.29.2

