From 55d4f1ed968aac62914065b331ddeeeb7da3a860 Mon Sep 17 00:00:00 2001
From: "Carsten Haitzler (Rasterman)" <raster@rasterman.com>
Date: Sat, 22 May 2021 11:54:17 +0100
Subject: [PATCH 01/13] alpm - remove unused error PK_ERROR_ENUM_REPO_NOT_FOUND

---
 backends/alpm/pk-alpm-error.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/backends/alpm/pk-alpm-error.c b/backends/alpm/pk-alpm-error.c
index 31f1fab17..300110f36 100644
--- a/backends/alpm/pk-alpm-error.c
+++ b/backends/alpm/pk-alpm-error.c
@@ -72,7 +72,6 @@ pk_alpm_error_emit (PkBackendJob *job, GError *error)
 		break;
 	case ALPM_ERR_DB_OPEN:
 	case ALPM_ERR_DB_NOT_FOUND:
-	case ALPM_ERR_PKG_REPO_NOT_FOUND:
 		code = PK_ERROR_ENUM_REPO_NOT_FOUND;
 		break;
 	case ALPM_ERR_DB_CREATE:
-- 
2.31.1

