--- a/script/bootstrap.py
+++ b/script/bootstrap.py
@@ -678,6 +678,7 @@
             f'--prefix={posix_install_dir}',
             '--enable-static',
             '--enable-pic',
+            '--disable-lavf',
             shell='bash')
         cmd('make', '-j', CPU_COUNT)
         cmd('make', 'install')
@@ -802,6 +803,7 @@
         '--enable-filter=select',
         '--enable-filter=concat',
         '--enable-filter=ssim',
+        '--enable-lto',
     ]
     if os.name == 'nt':
         result.extend([
