diff -ruN mailman-2.1.25-a/src/common.c mailman-2.1.25-b/src/common.c
--- mailman-2.1.25-a/src/common.c	2017-10-26 23:07:01.000000000 +0200
+++ mailman-2.1.25-b/src/common.c	2018-01-27 19:25:52.870021686 +0100
@@ -192,6 +192,12 @@
         "SERVER_NAME=",
         "SERVER_PORT=",
         "USER=",
+        "CONTENT_LENGTH=",
+        "DOCUMENT_ROOT=",
+        "SERVER_PROTOCOL=",
+        "REQUEST_SCHEME=",
+        "HTTPS=",
+        "REMOTE_PORT=",
         NULL
 };
 
