From 34e2d2f4f7224ac24dec5f0461d80cb9dfddb827 Mon Sep 17 00:00:00 2001
From: Marcelo Fernandez <marcelofg55@gmail.com>
Date: Mon, 5 Nov 2018 21:34:36 -0300
Subject: [PATCH] Fix binaries incorrectly detected as shared libraries on some
 linux distros

---
 platform/x11/detect.py | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/platform/x11/detect.py b/platform/x11/detect.py
index ee59e9b5a17..dc5c22b4b3b 100644
--- a/platform/x11/detect.py
+++ b/platform/x11/detect.py
@@ -149,6 +149,18 @@ def configure(env):
     env.Append(CCFLAGS=['-pipe'])
     env.Append(LINKFLAGS=['-pipe'])
 
+    # Check for gcc version >= 6 before adding -no-pie
+    version = get_compiler_version(env) or [-1, -1]
+    if using_gcc(env):
+        if version[0] >= 6:
+            env.Append(CCFLAGS=["-fpie"])
+            env.Append(LINKFLAGS=["-no-pie"])
+    # Do the same for clang should be fine with Clang 4 and higher
+    if using_clang(env):
+        if version[0] >= 4:
+            env.Append(CCFLAGS=["-fpie"])
+            env.Append(LINKFLAGS=["-no-pie"])
+
     ## Dependencies
 
     env.ParseConfig('pkg-config x11 --cflags --libs')
