From 08662fa09a14c68b1048ee4384b25dfd9ad0463f Mon Sep 17 00:00:00 2001
From: Taylor Hornby <taylor@defuse.ca>
Date: Sun, 24 May 2020 16:49:13 -0600
Subject: [PATCH] Add missing <stdexcept> header for std::invalid_argument

---
 src/crypto/equihash.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/crypto/equihash.h b/src/crypto/equihash.h
index 556b87d090..7cd97ff600 100644
--- a/src/crypto/equihash.h
+++ b/src/crypto/equihash.h
@@ -13,6 +13,7 @@
 
 #include <cstring>
 #include <exception>
+#include <stdexcept>
 #include <functional>
 #include <memory>
 #include <set>
