diff -rupN corefx.620cea9ccf0359993e803c900059932966399584.orig/src/Native/Unix/CMakeLists.txt corefx.620cea9ccf0359993e803c900059932966399584/src/Native/Unix/CMakeLists.txt
--- corefx.620cea9ccf0359993e803c900059932966399584.orig/src/Native/Unix/CMakeLists.txt	2020-06-23 18:07:14.111464986 +0200
+++ corefx.620cea9ccf0359993e803c900059932966399584/src/Native/Unix/CMakeLists.txt	2020-06-23 18:08:56.147153789 +0200
@@ -18,6 +18,8 @@ add_compile_options(-Wno-empty-translati
 add_compile_options(-Wno-cast-align)
 add_compile_options(-Wno-typedef-redefinition)
 add_compile_options(-Wno-c11-extensions)
+add_compile_options(-Wno-sign-conversion)
+add_compile_options(-Wno-unused-result)
 add_compile_options(-fPIC)
 add_compile_options(-I${CMAKE_CURRENT_SOURCE_DIR}/Common)
 add_compile_options(-I${CMAKE_CURRENT_BINARY_DIR}/Common)

