diff --git a/src/amuleDlg.cpp b/src/amuleDlg.cpp
index 03bc83c3..a1dc07b2 100644
--- a/src/amuleDlg.cpp
+++ b/src/amuleDlg.cpp
@@ -1380,7 +1380,7 @@ void CamuleDlg::Create_Toolbar(bool orientation)
 	if (!m_wndToolbar) {
 		m_wndToolbar = CreateToolBar(
 			(orientation ? wxTB_VERTICAL : wxTB_HORIZONTAL) |
-			wxNO_BORDER | wxTB_TEXT | wxTB_3DBUTTONS |
+			wxNO_BORDER | wxTB_TEXT | 
 			wxTB_FLAT | wxCLIP_CHILDREN | wxTB_NODIVIDER);
 
 
diff --git a/src/extern/wxWidgets/listctrl.cpp b/src/extern/wxWidgets/listctrl.cpp
index 46070069..4483d0f1 100644
--- a/src/extern/wxWidgets/listctrl.cpp
+++ b/src/extern/wxWidgets/listctrl.cpp
@@ -4860,7 +4860,7 @@ void wxListMainWindow::SortItems( MuleListCtrlCompare fn, long data )
 void wxListMainWindow::OnScroll(wxScrollWinEvent& event)
 {
     // FIXME
-#if ( defined(__WXGTK__) || defined(__WXMAC__) ) && !defined(__WXUNIVERSAL__)
+#if ( defined(__WXMAC__) ) && !defined(__WXUNIVERSAL__)
     wxScrolledWindow::OnScroll(event);
 #else
     HandleOnScroll( event );
