diff --git a/Apper/CMakeLists.txt b/Apper/CMakeLists.txt
index 2b71603..8c24305 100644
--- a/Apper/CMakeLists.txt
+++ b/Apper/CMakeLists.txt
@@ -49,7 +49,7 @@ target_link_libraries(apper
 
 set_target_properties(apper PROPERTIES INSTALL_RPATH ${CMAKE_INSTALL_FULL_LIBDIR}/apper)
 
-install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/apper DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS apper DESTINATION ${CMAKE_INSTALL_BINDIR})
 install(FILES org.kde.apper.desktop DESTINATION ${CMAKE_INSTALL_APPDIR})
 install(FILES org.kde.apper_installer.desktop DESTINATION ${CMAKE_INSTALL_APPDIR})
 install(FILES org.kde.apper_settings.desktop DESTINATION ${CMAKE_INSTALL_APPDIR})
