diff --git a/tools/settings_template_readonly.py b/tools/settings_template_readonly.py
index f1ca646ea..3e85c7ab1 100644
--- a/tools/settings_template_readonly.py
+++ b/tools/settings_template_readonly.py
@@ -12,9 +12,9 @@ import os
 
 # This is used by external projects in order to find emscripten.  It is not used
 # by emscripten itself.
-EMSCRIPTEN_ROOT = os.path.expanduser(os.getenv('EMSCRIPTEN', '{{{ EMSCRIPTEN_ROOT }}}')) # directory
+EMSCRIPTEN_ROOT = "/usr/lib/emscripten"
 
-LLVM_ROOT = os.path.expanduser(os.getenv('LLVM', '{{{ LLVM_ROOT }}}')) # directory
+LLVM_ROOT = "/usr/lib/emscripten-llvm"
 BINARYEN_ROOT = os.path.expanduser(os.getenv('BINARYEN', '')) # directory
 
 # Add this if you have manually built the JS optimizer executable (in
@@ -43,7 +43,7 @@ JAVA = 'java' # executable
 # All JS engines to use when running the automatic tests. Not all the engines in
 # this list must exist (if they don't, they will be skipped in the test runner).
 #
-# JS_ENGINES = [NODE_JS] # add V8_ENGINE or SPIDERMONKEY_ENGINE if you have them installed too.
+JS_ENGINES = [NODE_JS] # add V8_ENGINE or SPIDERMONKEY_ENGINE if you have them installed too.
 #
 # WASMER = os.path.expanduser(os.path.join('~', '.wasmer', 'bin', 'wasmer'))
 # WASMTIME = os.path.expanduser(os.path.join('~', 'wasmtime'))
