diff --unified --recursive --text src.ori/Arduino-master-1.8.9/arduino-core/src/cc/arduino/contributions/packages/ContributionsIndexer.java src.new/Arduino-master-1.8.9/arduino-core/src/cc/arduino/contributions/packages/ContributionsIndexer.java
--- src.ori/Arduino-master-1.8.9/arduino-core/src/cc/arduino/contributions/packages/ContributionsIndexer.java	2019-03-15 15:14:01.000000000 +0100
+++ src.new/Arduino-master-1.8.9/arduino-core/src/cc/arduino/contributions/packages/ContributionsIndexer.java	2019-05-20 17:20:47.402857561 +0200
@@ -215,7 +215,7 @@
   }
 
   private void syncBuiltInHardware() throws IOException {
-    if (index == null) {
+    if (index == null || builtInHardwareFolder == null) {
       return;
     }
     for (File folder : builtInHardwareFolder.listFiles(ONLY_DIRS)) {
diff --unified --recursive --text src.ori/Arduino-master-1.8.9/arduino-core/src/processing/app/BaseNoGui.java src.new/Arduino-master-1.8.9/arduino-core/src/processing/app/BaseNoGui.java
--- src.ori/Arduino-master-1.8.9/arduino-core/src/processing/app/BaseNoGui.java	2019-03-15 15:14:01.000000000 +0100
+++ src.new/Arduino-master-1.8.9/arduino-core/src/processing/app/BaseNoGui.java	2019-05-20 17:21:55.776950094 +0200
@@ -676,7 +676,9 @@
     // Libraries located in the latest folders on the list can override
     // other libraries with the same name.
     librariesIndexer.setLibrariesFolders(librariesFolders);
-    librariesIndexer.setArchitecturePriority(getTargetPlatform().getId());
+    if (getTargetPlatform() != null) {
+      librariesIndexer.setArchitecturePriority(getTargetPlatform().getId());
+    }
     librariesIndexer.rescanLibraries();
 
     populateImportToLibraryTable();
