diff --git a/cmake/Modules/FindGTK3.cmake b/cmake/Modules/FindGTK3.cmake
index ebbf0400..245fde13 100644
--- a/cmake/Modules/FindGTK3.cmake
+++ b/cmake/Modules/FindGTK3.cmake
@@ -388,7 +388,7 @@ endif()
 #
 
 find_package(Freetype)
-list(APPEND GTK3_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS})
+list(APPEND GTK3_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS} /usr/include/harfbuzz)
 list(APPEND GTK3_LIBRARIES ${FREETYPE_LIBRARIES})
 
 foreach(_GTK3_component ${GTK3_FIND_COMPONENTS})
