From d2a1db8834bf0a9007bf6924b006a6f2c94ce88d Mon Sep 17 00:00:00 2001
From: Jasper St. Pierre <jstpierre@mecheye.net>
Date: Mon, 13 Jan 2014 18:03:19 +0000
Subject: Update for Cogl API breaks

---
(limited to 'src/compositor')

diff --git a/src/compositor/meta-texture-rectangle.c b/src/compositor/meta-texture-rectangle.c
index 3fc9430..450155d 100644
--- a/src/compositor/meta-texture-rectangle.c
+++ b/src/compositor/meta-texture-rectangle.c
@@ -35,10 +35,8 @@ CoglTexture *
 meta_texture_rectangle_new (unsigned int width,
                             unsigned int height,
                             CoglPixelFormat format,
-                            CoglPixelFormat internal_format,
                             unsigned int rowstride,
-                            const guint8 *data,
-                            GError **error)
+                            const guint8 *data)
 {
   ClutterBackend *backend =
     clutter_get_default_backend ();
@@ -46,10 +44,7 @@ meta_texture_rectangle_new (unsigned int width,
     clutter_backend_get_cogl_context (backend);
   CoglTextureRectangle *tex_rect;
 
-  tex_rect = cogl_texture_rectangle_new_with_size (context,
-                                                   width, height,
-                                                   internal_format,
-                                                   error);
+  tex_rect = cogl_texture_rectangle_new_with_size (context, width, height);
   if (tex_rect == NULL)
     return NULL;
 
diff --git a/src/compositor/meta-texture-rectangle.h b/src/compositor/meta-texture-rectangle.h
index 30f60d3..7b84229 100644
--- a/src/compositor/meta-texture-rectangle.h
+++ b/src/compositor/meta-texture-rectangle.h
@@ -34,10 +34,8 @@ CoglTexture *
 meta_texture_rectangle_new (unsigned int width,
                             unsigned int height,
                             CoglPixelFormat format,
-                            CoglPixelFormat internal_format,
                             unsigned int rowstride,
-                            const guint8 *data,
-                            GError **error);
+                            const guint8 *data);
 
 gboolean
 meta_texture_rectangle_check (CoglTexture *texture);
diff --git a/src/compositor/meta-texture-tower.c b/src/compositor/meta-texture-tower.c
index 9a30de0..019aaf4 100644
--- a/src/compositor/meta-texture-tower.c
+++ b/src/compositor/meta-texture-tower.c
@@ -365,13 +365,9 @@ texture_tower_create_texture (MetaTextureTower *tower,
         meta_texture_rectangle_new (width, height,
                                     /* data format */
                                     TEXTURE_FORMAT,
-                                    /* internal cogl format */
-                                    TEXTURE_FORMAT,
                                     /* rowstride */
                                     width * 4,
                                     /* data */
-                                    NULL,
-                                    /* error */
                                     NULL);
     }
   else
--- a/src/compositor/meta-shaped-texture.c
+++ b/src/compositor/meta-shaped-texture.c
@@ -295,10 +295,7 @@ meta_shaped_texture_ensure_mask (MetaSha
       if (meta_texture_rectangle_check (paint_tex))
         priv->mask_texture = meta_texture_rectangle_new (tex_width, tex_height,
                                                          COGL_PIXEL_FORMAT_A_8,
-                                                         COGL_PIXEL_FORMAT_A_8,
-                                                         stride,
-                                                         mask_data,
-                                                         NULL /* error */);
+                                                         stride, mask_data);
       else
         priv->mask_texture = cogl_texture_new_from_data (tex_width, tex_height,
                                                          COGL_TEXTURE_NONE,
--
cgit v0.9.2
