diff -Naur cinnamon-desktop-2.8.0.orig/schemas/org.cinnamon.desktop.background.gschema.xml.in cinnamon-desktop-2.8.0/schemas/org.cinnamon.desktop.background.gschema.xml.in.in
--- cinnamon-desktop-2.8.0.orig/schemas/org.cinnamon.desktop.background.gschema.xml.in	2015-10-16 13:00:30.000000000 +0000
+++ cinnamon-desktop-2.8.0/schemas/org.cinnamon.desktop.background.gschema.xml.in	2015-11-20 19:04:16.543027956 +0000
@@ -10,7 +10,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/themes/Adwaita/backgrounds/adwaita-timed.xml'</default>
+      <default>'file://@datadir@/backgrounds/gnome/adwaita-timed.xml'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Not that the backend only supports
diff -Naur cinnamon-desktop-2.8.0.orig/schemas/org.cinnamon.desktop.interface.gschema.xml.in cinnamon-desktop-2.8.0/schemas/org.cinnamon.desktop.interface.gschema.xml.in.in
--- cinnamon-desktop-2.8.0.orig/schemas/org.cinnamon.desktop.interface.gschema.xml.in	2015-10-16 13:00:30.000000000 +0000
+++ cinnamon-desktop-2.8.0/schemas/org.cinnamon.desktop.interface.gschema.xml.in	2015-11-20 19:04:16.543027956 +0000
@@ -97,7 +97,7 @@
       </description>
     </key>
     <key name="icon-theme" type="s">
-      <default>'gnome'</default>
+      <default>'Adwaita'</default>
       <summary>Icon Theme</summary>
       <description>
         Icon theme to use for the panel, nautilus etc.
