--- configure.sic	2019-09-30 15:48:53.742897037 +0200
+++ configure	2019-09-30 15:49:12.492545288 +0200
@@ -39,8 +39,8 @@
     echo "The standalone Camlp4 requires OCaml >= 4.02."
     echo "For previous versions of OCaml use the Camlp4 distributed with OCaml."
     exit 2
-elif [ $major -ne 4 -o $minor -ne 8 ]; then
-    echo "This version of Camlp4 is for OCaml 4.07 but you are using OCaml $ocaml_version."
+elif [ $major -ne 4 -o $minor -ne 9 ]; then
+    echo "This version of Camlp4 is for OCaml 4.09 but you are using OCaml $ocaml_version."
     if [ -d .git ] ; then
       if [ $PINNED -eq 1 ] ; then
         echo "You've used the wrong opam pin URL for this switch.">&2

