--- config.default.orig	2011-11-10 21:48:01.000000000 +0100
+++ config.default	2011-11-10 21:51:58.383221979 +0100
@@ -6,15 +6,15 @@
 #BACKEND = sdl
 #BACKEND = sdl2
 #BACKEND = mixer_sdl
-#BACKEND = mixer_sdl2
+BACKEND = mixer_sdl2
 #BACKEND = posix
 
 #OSTYPE = amiga
 #OSTYPE = beos
 #OSTYPE = cygwin
 #OSTYPE = freebsd
 #OSTYPE = haiku
-#OSTYPE = linux
+OSTYPE = linux
 #OSTYPE = mingw
 #OSTYPE = mac
 
@@ -37,6 +37,6 @@
 #SDL2_CONFIG    = sdl2-config
 #FREETYPE_CONFIG    = freetype-config
 
-#VERBOSE = 1
+VERBOSE = 1
 
 # Use static libraries instead
 # STATIC = 1
