diff -ru converseen-0.9.9.2.orig/src/formats.cpp converseen-0.9.9.2/src/formats.cpp
--- converseen-0.9.9.2.orig/src/formats.cpp	2021-05-03 12:00:24.000000000 +0200
+++ converseen-0.9.9.2/src/formats.cpp	2021-10-12 00:43:47.998537864 +0200
@@ -37,11 +37,16 @@
 {
 //#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD)
     list<CoderInfo> coderList;
+    try {
     coderInfoList(&coderList,
                   CoderInfo::TrueMatch,
                   CoderInfo::AnyMatch,
                   CoderInfo::AnyMatch);
-
+    }
+    catch (ErrorModule &e) {
+        cerr << e.what() << endl;
+    }
+    
     list<CoderInfo>::iterator entry = coderList.begin();
     QString readableExts;
 
