--- libfilezilla-0.9.2/m4/cppunit.m4.ddd	2015-11-02 14:47:31.000000000 +0100
+++ libfilezilla-0.9.2/m4/cppunit.m4	2017-06-06 19:08:00.329550653 +0200
@@ -29,12 +29,14 @@ AC_ARG_WITH(cppunit-exec-prefix,[  --wit
   no_cppunit=""
   if test "$CPPUNIT_CONFIG" = "no" ; then
     AC_MSG_RESULT(no)
-    no_cppunit=yes
+    CPPUNIT_CFLAGS=`pkg-config --cflags cppunit`
+    CPPUNIT_LIBS=`pkg-config --libs cppunit`
+    cppunit_version=`pkg-config --modversion cppunit`
   else
     CPPUNIT_CFLAGS=`$CPPUNIT_CONFIG --cflags`
     CPPUNIT_LIBS=`$CPPUNIT_CONFIG --libs`
     cppunit_version=`$CPPUNIT_CONFIG --version`
-
+  fi
     cppunit_major_version=`echo $cppunit_version | \
            sed 's/\([[0-9]]*\).\([[0-9]]*\).\([[0-9]]*\)/\1/'`
     cppunit_minor_version=`echo $cppunit_version | \
@@ -74,7 +76,6 @@ AC_ARG_WITH(cppunit-exec-prefix,[  --wit
       AC_MSG_RESULT(no)
       no_cppunit=yes
     fi
-  fi
 
   if test "x$no_cppunit" = x ; then
      ifelse([$2], , :, [$2])
