diff --git i/Makefile w/Makefile
index f5a11604..6dec06a9 100644
--- i/Makefile
+++ w/Makefile
@@ -34,7 +34,7 @@ BINDIR ?= /usr/local/bin
 
 VERSION := $(shell git describe --tags --dirty --always)
 VERSION := $(VERSION:v%=%)
-GO_LDFLAGS := -X $(PROJECT)/pkg/version.Version=$(VERSION)
+GO_LDFLAGS := $(GO_LDFLAGS) -X $(PROJECT)/pkg/version.Version=$(VERSION)
 
 BUILD_PATH := $(shell pwd)/build
 BUILD_BIN_PATH := $(BUILD_PATH)/bin
@@ -63,7 +63,7 @@ critest:
 	@$(MAKE) -B $(CRITEST)
 
 $(CRITEST):
-	CGO_ENABLED=0 $(GO_TEST) -c -o $@ \
+	CGO_ENABLED=1 $(GO_TEST) -c -o $@ \
 		-ldflags '$(GO_LDFLAGS)' \
 		-trimpath \
 	     $(PROJECT)/cmd/critest
@@ -72,7 +72,7 @@ crictl:
 	@$(MAKE) -B $(CRICTL)
 
 $(CRICTL):
-	CGO_ENABLED=0 $(GO_BUILD) -o $@ \
+	CGO_ENABLED=1 $(GO_BUILD) -o $@ \
 		-ldflags '$(GO_LDFLAGS)' \
 		-trimpath \
 		$(PROJECT)/cmd/crictl
