diff -ruN a/Makefile b/Makefile
--- a/Makefile	2020-08-28 11:44:53.000000000 +0200
+++ b/Makefile	2020-11-05 13:32:43.543054387 +0100
@@ -33,7 +33,7 @@
 
 VERSION := $(shell git describe --tags --dirty --always)
 VERSION := $(VERSION:v%=%)
-GO_LDFLAGS := -X $(PROJECT)/pkg/version.Version=$(VERSION)
+GO_LDFLAGS := $(GO_LDFLAGS) -X $(PROJECT)/pkg/version.Version=$(VERSION)
 BUILDTAGS := selinux
 
 BUILD_PATH := $(shell pwd)/build
@@ -57,13 +57,13 @@
 	@echo " * 'clean' - Clean artifacts."
 
 critest:
-	CGO_ENABLED=0 $(GO_TEST) -c -o $(CURDIR)/_output/critest$(BIN_EXT) \
+	CGO_ENABLED=1 $(GO_TEST) -c -o $(CURDIR)/_output/critest$(BIN_EXT) \
 		-ldflags '$(GO_LDFLAGS)' \
 		-tags '$(BUILDTAGS)' \
-	     $(PROJECT)/cmd/critest
+		$(PROJECT)/cmd/critest
 
 crictl:
-	CGO_ENABLED=0 $(GO_BUILD) -o $(CURDIR)/_output/crictl$(BIN_EXT) \
+	CGO_ENABLED=1 $(GO_BUILD) -o $(CURDIR)/_output/crictl$(BIN_EXT) \
 		-ldflags '$(GO_LDFLAGS)' \
 		-tags '$(BUILDTAGS)' \
 		$(PROJECT)/cmd/crictl
