diff --git a/plugins/background/csd-background-manager.c b/plugins/background/csd-background-manager.c
index c6388e0..fe62f5b 100644
--- a/plugins/background/csd-background-manager.c
+++ b/plugins/background/csd-background-manager.c
@@ -137,7 +137,6 @@ settings_change_event_cb (GSettings            *settings,
 {
         gnome_bg_load_from_preferences (manager->priv->bg,
                                         manager->priv->settings);
-        gnome_bg_set_accountsservice_background(gnome_bg_get_filename(manager->priv->bg));
         return FALSE;
 }
 
