--- src/mgr/curlftpt.cpp	(revision 2505)
+++ src/mgr/curlftpt.cpp	(revision 2506)
@@ -177,6 +177,9 @@
 		res = curl_easy_perform(session);
 		SWLog::getSystemLog()->logDebug("***** Finished performing curl easy action. \n");
 
+		// it seems CURL tries to use this option data later for some reason, so we unset here
+		curl_easy_setopt(session, CURLOPT_PROGRESSDATA, (void*)NULL);
+
 		if(CURLE_OK != res) {
 			retVal = -1;
 		}
