--- src/hd/kbd.c.old	2010-03-15 23:37:52.000000000 +0800
+++ src/hd/kbd.c	2010-03-22 10:50:58.000000000 +0800
@@ -146,11 +146,13 @@
   }
 
   if(!dev && (fd = open(DEV_CONSOLE, O_RDWR | O_NONBLOCK | O_NOCTTY)) >= 0) {
+    /* Removing since TIOCGDEV is not supported by the mainline kernel
     if(ioctl(fd, TIOCGDEV, &u) != -1) {
       tty_major = (u >> 8) & 0xfff;
       tty_minor = (u & 0xff) | ((u >> 12) & 0xfff00);
       ADD2LOG(DEV_CONSOLE ": major %u, minor %u\n", tty_major, tty_minor);
     }
+    */
 
     if (0)
 	    ;
