diff -ruN cuyo-2.~-1.1-orig/icons/cuyo.desktop cuyo-2.~-1.1/icons/cuyo.desktop
--- cuyo-2.~-1.1-orig/icons/cuyo.desktop	2007-12-20 18:22:32.000000000 +0100
+++ cuyo-2.~-1.1/icons/cuyo.desktop	2009-11-08 14:08:56.000000000 +0100
@@ -1,11 +1,13 @@
 [Desktop Entry]
+Type=Application
 Name=Cuyo
+GenericName=Cuyo
 Icon=cuyo
-Type=Application
+Exec=cuyo
+StartupNotify=true
 Terminal=false
-MimeType=application/x-executable
+MimeType=application/x-executable;
 Categories=Game;BlocksGame;
-Encoding=UTF-8
 Comment=A tetris like game with many levels
+Comment[cs]=Hra podobná tetrisu s mnoha úrovněmi
 Comment[de]=Ein Tetris-artiges Spiel mit vielen Leveln
-Exec=cuyo
diff -ruN cuyo-2.~-1.1-orig/src/stringzeug.h cuyo-2.~-1.1/src/stringzeug.h
--- cuyo-2.~-1.1-orig/src/stringzeug.h	2006-07-03 19:26:23.000000000 +0200
+++ cuyo-2.~-1.1/src/stringzeug.h	2009-11-08 14:08:56.000000000 +0100
@@ -19,6 +19,7 @@
 #define STRINGZEUG_H
 
 #include <config.h>
+#include <cstdarg>
 #include <string>
 
 
