diff -ruN a/Makefile b/Makefile
--- a/Makefile	2009-12-06 18:55:28.000000000 +0100
+++ b/Makefile	2023-03-21 19:17:35.516241775 +0100
@@ -1,15 +1,15 @@
 CC = gcc -pipe
 VERSION = 0.47
-CFLAGS = -O2 -fno-rtti -fno-exceptions -DCYCFX2PROG_VERSION=\"$(VERSION)\" \
+CUSTOM_CFLAGS = -O2 -fno-rtti -fno-exceptions -DCYCFX2PROG_VERSION=\"$(VERSION)\" \
 	-W -Wall -Wformat
-LDFLAGS = -lusb
+CUSTOM_LDFLAGS = -lusb
 DIST_DEST = cycfx2prog-$(VERSION)
 
 all: cycfx2prog
 
 # NOTE: Also add sources to the "dist:" target!
 cycfx2prog: cycfx2prog.o cycfx2dev.o
-	$(CC) $(LDFLAGS) cycfx2prog.o cycfx2dev.o -o cycfx2prog
+	$(CC) $(CFLAGS) $(CUSTOM_LDFLAGS) $(LDFLAGS) cycfx2prog.o cycfx2dev.o -o cycfx2prog
 
 clean:
 	-rm -f *.o
@@ -26,7 +26,7 @@
 	rm -r "$(DIST_DEST)"
 
 .cc.o:
-	$(CC) -c $(CFLAGS) $<
+	$(CC) -c $(CFLAGS) $(CUSTOM_CFLAGS) $<
 
 cycfx2dev.o: cycfx2dev.cc cycfx2dev.h
 cycfx2prog.o: cycfx2prog.cc cycfx2dev.h
