--- freehdl-0.0.8.orig/vaul/expr.cc	2007-10-23 22:33:45.000000000 +0300
+++ freehdl-0.0.8/vaul/expr.cc	2018-03-04 17:47:04.703969598 +0200
@@ -2631,6 +2631,9 @@
   // XXX - every constant interface thing expect in a subprogram is a
   // generic?
 
+  assert(d);
+  if (!d->declarative_region) return IR_NOT_STATIC;
+  assert(d->declarative_region);
   if (!d->declarative_region->is (IR_SUBPROGRAM_DECLARATION))
     return IR_GLOBALLY_STATIC;
   return IR_NOT_STATIC;
