diff --git a/frame/toptoolbar.cpp b/frame/toptoolbar.cpp
index f6f8073..a1dc84e 100644
--- a/frame/toptoolbar.cpp
+++ b/frame/toptoolbar.cpp
@@ -35,6 +35,7 @@
 #include <QDebug>
 #include <QHBoxLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QProcess>
 #include <QResizeEvent>
 #include <QShortcut>
diff --git a/module/view/contents/ttbcontent.cpp b/module/view/contents/ttbcontent.cpp
index 1f95a63..7d03cdc 100644
--- a/module/view/contents/ttbcontent.cpp
+++ b/module/view/contents/ttbcontent.cpp
@@ -38,6 +38,7 @@
 #include <DApplicationHelper>
 #include <DSpinner>
 #include <QtMath>
+#include <QPainterPath>
 
 DWIDGET_USE_NAMESPACE
 namespace {
diff --git a/thumbnail/thumbnaildelegate.cpp b/thumbnail/thumbnaildelegate.cpp
index a789f41..3b2b50a 100644
--- a/thumbnail/thumbnaildelegate.cpp
+++ b/thumbnail/thumbnaildelegate.cpp
@@ -24,6 +24,7 @@
 #include <QHBoxLayout>
 #include <QLineEdit>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPixmapCache>
 #include <QStandardItemModel>
 #include <QThread>
diff --git a/widgets/popupmenustyle.cpp b/widgets/popupmenustyle.cpp
index ddb509c..0fd91f7 100644
--- a/widgets/popupmenustyle.cpp
+++ b/widgets/popupmenustyle.cpp
@@ -19,6 +19,7 @@
 #include <QPixmap>
 #include <QPixmapCache>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStyleOption>
 #include <QStyleOptionMenuItem>
 
