diff --git a/riakc/src/riakc_obj.erl b/riakc/src/riakc_obj.erl
index 8086533..c95fdcf 100644
--- a/riakc/src/riakc_obj.erl
+++ b/riakc/src/riakc_obj.erl
@@ -79,7 +79,7 @@
 -type key() :: binary() | 'undefined'. %% A key name
 -type id() :: {bucket(), key()}.
 -type vclock() :: binary(). %% An opaque vector clock
--type metadata() :: dict(). %% Value metadata
+-type metadata() :: dict:dict(). %% Value metadata
 -type content_type() :: string(). %% The media type of a value
 -type value() :: binary(). %% An opaque value
 -type contents() :: [{metadata(), value()}]. %% All metadata/value pairs in a `riakc_obj'.
@@ -103,7 +103,7 @@
           key :: key(),
           vclock :: vclock(),
           contents :: contents(),
-          updatemetadata :: dict(),
+          updatemetadata :: dict:dict(),
           updatevalue :: value()
          }).
 
@@ -284,7 +284,7 @@ get_update_value(#riakc_obj{updatevalue=UV}=Object) ->
     end.
 
 %% @doc  Return the content type from metadata
--spec md_ctype(dict()) -> undefined | content_type().
+-spec md_ctype(dict:dict()) -> undefined | content_type().
 md_ctype(MetaData) ->
     case dict:find(?MD_CTYPE, MetaData) of
         error ->
diff --git a/riakc/src/riakc_pb_socket.erl b/riakc/src/riakc_pb_socket.erl
index 77d27eb..4eb043f 100644
--- a/riakc/src/riakc_pb_socket.erl
+++ b/riakc/src/riakc_pb_socket.erl
@@ -117,7 +117,7 @@
                 queue_if_disconnected = false :: boolean(), % if true, add requests to queue if disconnected
                 sock :: port(),       % gen_tcp socket
                 active :: #request{} | undefined,     % active request
-                queue :: queue() | undefined,      % queue of pending requests
+                queue :: queue:queue() | undefined,      % queue of pending requests
                 connects=0 :: non_neg_integer(), % number of successful connects
                 failed=[] :: [connection_failure()],  % breakdown of failed connects
                 connect_timeout=infinity :: timeout(), % timeout of TCP connection
diff --git a/riak_pb/src/riak_pb_kv_codec.erl b/riak_pb/src/riak_pb_kv_codec.erl
index c95fde6..0bcab88 100644
--- a/riak_pb/src/riak_pb_kv_codec.erl
+++ b/riak_pb/src/riak_pb_kv_codec.erl
@@ -49,7 +49,7 @@
 
 -type symbolic_quorum() :: one | quorum | all | default.
 -type value() :: binary().
--type metadata() :: dict().
+-type metadata() :: dict:dict().
 -type contents() :: [{metadata(), value()}].
 
 %% @doc Convert a list of object {MetaData,Value} pairs to protocol
