--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -1,9 +1,11 @@
 # vi:set sw=2 et:
 project(surge-common)
 
+find_package(PkgConfig REQUIRED)
+
 surge_add_lib_subdirectory(airwindows)
 surge_add_lib_subdirectory(eurorack)
-surge_add_lib_subdirectory(fmt)
+pkg_search_module(fmt REQUIRED fmt)
 
 # Make MTS optional
 if (NOT SURGE_SKIP_ODDSOUND_MTS)
@@ -17,7 +19,7 @@ endif()
 if(MINGW)
   set(HAVE_VISIBILITY 0 CACHE INTERNAL "Force-disable libsamplerate's visibility check on MinGW")
 endif()
-surge_add_lib_subdirectory(libsamplerate)
+pkg_search_module(samplerate REQUIRED samplerate)
 surge_add_lib_subdirectory(pffft)
 surge_add_lib_subdirectory(tuning-library)
 surge_add_lib_subdirectory(sqlite-3.23.3)
--- a/src/surge-fx/CMakeLists.txt
+++ b/src/surge-fx/CMakeLists.txt
@@ -60,6 +60,7 @@ target_link_libraries(${PROJECT_NAME} PRIVATE
   surge-juce
   juce::juce_audio_utils
   juce::juce_audio_processors
+  fmt
   )
 
 if(SURGE_BUILD_CLAP)
--- a/src/surge-xt/CMakeLists.txt
+++ b/src/surge-xt/CMakeLists.txt
@@ -204,6 +204,7 @@ target_link_libraries(${PROJECT_NAME} PRIVATE
   juce::juce_audio_plugin_client
   surge-xt-binary
   sst-filters-extras
+  fmt
   )
 
 target_include_directories(${PROJECT_NAME}
