diff --git a/plugins/CMakeLists.txt b/plugins/CMakeLists.txt
index d1f688c2..8c713614 100644
--- a/plugins/CMakeLists.txt
+++ b/plugins/CMakeLists.txt
@@ -19,6 +19,5 @@ add_subdirectory (word-completion)
 add_subdirectory (outline)
 add_subdirectory (vim-emulation)
 add_subdirectory (highlight-word-selection)
-add_subdirectory (clipboard-history)
 add_subdirectory (detect-indent)
 add_subdirectory (preserve-indent)
