--- a/tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/PublishCoreSetupBinaries.cs
+++ b/tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/PublishCoreSetupBinaries.cs
@@ -46,6 +46,7 @@ namespace Microsoft.DotNet.Build.Tasks
                                               ".fedora.32", "-fedora.32",
                                               ".fedora.33", "-fedora.33",
                                               ".fedora.34", "-fedora.34",
+                                              "-arch",".arch",
                                               "-linux", ".linux",
                                               "-osx", ".osx",
                                               "-OSX", ".OSX",

