 gui/icons/Makefile.am | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gui/icons/Makefile.am b/gui/icons/Makefile.am
index e67c987..678d43f 100644
--- a/gui/icons/Makefile.am
+++ b/gui/icons/Makefile.am
@@ -6,3 +6,8 @@ EXTRA_DIST = $(icons48_DATA) $(icons72_DATA) $(icons128_DATA)
 icon48dir = $(datadir)/icons/hicolor/48x48/apps
 icon72dir = $(datadir)/icons/hicolor/72x72/apps
 icon128dir = $(datadir)/icons/hicolor/128x128/apps
+
+install-data-hook:
+	if [ -n "`which gtk-update-icon-cache`" ]; then \
+		gtk-update-icon-cache $(datadir)/icons/hicolor; \
+	fi
