diff -r 0c5a115449a3 config/baseconfig.mk
--- a/config/baseconfig.mk	Wed Jan 17 17:20:41 2018 -0500
+++ b/config/baseconfig.mk	Sat Jan 20 21:54:40 2018 -0500
@@ -2,7 +2,7 @@
 # directly in python/mozbuild/mozbuild/base.py for gmake validation.
 # We thus use INCLUDED_AUTOCONF_MK to enable/disable some parts depending
 # whether a normal build is happening or whether the check is running.
-installdir = $(libdir)/$(MOZ_APP_NAME)
+installdir = $(libdir)/$(MOZ_APP_NAME)-developer-edition
 ifeq (.,$(DEPTH))
 DIST = dist
 else
diff -r 0c5a115449a3 toolkit/mozapps/installer/packager.mk
--- a/toolkit/mozapps/installer/packager.mk	Wed Jan 17 17:20:41 2018 -0500
+++ b/toolkit/mozapps/installer/packager.mk	Sat Jan 20 21:54:40 2018 -0500
@@ -140,8 +140,8 @@
 	(cd $(DIST)/$(MOZ_PKG_DIR) && $(TAR) --exclude=precomplete $(TAR_CREATE_FLAGS) - .) | \
 	  (cd $(DESTDIR)$(installdir) && tar -xf -)
 	$(NSINSTALL) -D $(DESTDIR)$(bindir)
-	$(RM) -f $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)
-	ln -s $(installdir)/$(MOZ_APP_NAME) $(DESTDIR)$(bindir)
+	$(RM) -f $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)-developer-edition
+	ln -s $(installdir)/$(MOZ_APP_NAME) $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)-developer-edition
 
 checksum:
 	mkdir -p `dirname $(CHECKSUM_FILE)`
diff -r 0c5a115449a3 toolkit/xre/nsAppRunner.cpp
--- a/toolkit/xre/nsAppRunner.cpp	Wed Jan 17 17:20:41 2018 -0500
+++ b/toolkit/xre/nsAppRunner.cpp	Sat Jan 20 21:54:40 2018 -0500
@@ -3866,7 +3866,11 @@
   // consistently.
 
   // Set program name to the one defined in application.ini.
-  {
+  if (strcmp(gAppData->name, "Firefox") == 0) {
+    // HACK to make GNOME Shell distinguish this
+    // app from regular Firefox
+    g_set_prgname("firefox-developer-edition");
+  } else {
     nsAutoCString program(gAppData->name);
     ToLowerCase(program);
     g_set_prgname(program.get());
