From a5045dc99d7fc3baad364cafa5a5a06cec69e29c Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Sun, 5 Jul 2015 20:38:44 +0200
Subject: [PATCH] fix build with qt 5.5

---
 lxqt-config-monitor/monitor.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lxqt-config-monitor/monitor.h b/lxqt-config-monitor/monitor.h
index 9f87bc0..b5df05b 100644
--- a/lxqt-config-monitor/monitor.h
+++ b/lxqt-config-monitor/monitor.h
@@ -20,6 +20,7 @@
 #ifndef _MONITOR_H_
 #define _MONITOR_H_
 
+#include <QObject>
 #include <QStringList>
 #include <QHash>
 #include <QList>
