commit 963dea395cc58eae940b85e242f0d84fb7d2eaa5
Author: dgod <dgod.osa@gmail.com>
Date:   Mon Apr 22 19:08:58 2013 +0800

    Patch that resolves -1% CPU for processes on armhf

diff --git a/src/xfce-taskmanager-linux.c b/src/xfce-taskmanager-linux.c
index 116fec1..1267dfc 100644
--- a/src/xfce-taskmanager-linux.c
+++ b/src/xfce-taskmanager-linux.c
@@ -267,6 +267,7 @@ gboolean get_system_status (system_status *sys_stat)
             }
         }
         fclose (file);
+        if(!cpu_count) cpu_count++; /* cpu_count should be at least 1 */
     }
     sys_stat->cpu_count=cpu_count;
     return TRUE;
