From 0eaca2da65d376ddd3ec170286feee5e88d68433 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Mon, 19 Jun 2017 14:01:27 +0100
Subject: [PATCH] trivial: Fix test failure with de5dd8f3

---
 libfwupd/fwupd-self-test.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libfwupd/fwupd-self-test.c b/libfwupd/fwupd-self-test.c
index 7597fff..f3b14b8 100644
--- a/libfwupd/fwupd-self-test.c
+++ b/libfwupd/fwupd-self-test.c
@@ -139,6 +139,7 @@ fwupd_result_func (void)
 		"  Guid:                 2082b5e0-7a64-478a-b1b2-e3404fab6dad\n"
 		"  Guid:                 00000000-0000-0000-0000-000000000000\n"
 		"  DeviceID:             USB:foo\n"
+		"  DisplayName:          ColorHug2\n"
 		"  Flags:                allow-offline|require-ac\n"
 		"  FirmwareHash:         SHA1(beefdead)\n"
 		"  Created:              1970-01-01\n"
