From e92ce8b06b298622253fe2a55b2fa524aa342914 Mon Sep 17 00:00:00 2001
From: haruyukilxz <lxz@ilxz.me>
Date: Wed, 11 Jul 2018 13:07:23 +0800
Subject: [PATCH] fix(osd): null pointer exception notify

Change-Id: I082595e578839700117e006f344e948ce125e131
---

diff --git a/dde-osd/notification/bubble.cpp b/dde-osd/notification/bubble.cpp
index 3998e37..46f16e7 100644
--- a/dde-osd/notification/bubble.cpp
+++ b/dde-osd/notification/bubble.cpp
@@ -175,6 +175,8 @@
 {
     DBlurEffectWidget::hideEvent(event);
 
+    m_outAnimation->stop();
+
     m_quitTimer->start();
 }
 
@@ -210,7 +212,10 @@
 
 void Bubble::onOutAnimFinished()
 {
-    Q_EMIT expired(m_entity->id().toInt());
+    // FIXME: There should be no empty pointers here
+    if (m_entity) {
+        Q_EMIT expired(m_entity->id().toInt());
+    }
 }
 
 void Bubble::updateContent()
