diff -Naur activity-log-manager-0.9.7.orig/src/blacklist-dbus.vala activity-log-manager-0.9.7/src/blacklist-dbus.vala
--- activity-log-manager-0.9.7.orig/src/blacklist-dbus.vala	2015-07-17 02:38:04.290396888 +0000
+++ activity-log-manager-0.9.7/src/blacklist-dbus.vala	2015-07-17 02:38:33.883637191 +0000
@@ -137,7 +137,7 @@
 			return false;
 		}
 
-		public async void find_events (string id, TreeIter iter, ListStore store)
+		public async void find_events (string id, TreeIter iter, Gtk.ListStore store)
 		{
 	     	Event event = new Event();
 	    	event.manifestation = Zeitgeist.ZG.USER_ACTIVITY;
@@ -165,7 +165,7 @@
 			store.set_value(iter, 5, counter);
 		}
 
-		public void get_count_for_app (string app_id, TreeIter iter, ListStore store) {
+		public void get_count_for_app (string app_id, TreeIter iter, Gtk.ListStore store) {
 			find_events(app_id, iter, store);
 		}
 		
diff -Naur activity-log-manager-0.9.7.orig/src/security-widget.vala activity-log-manager-0.9.7/src/security-widget.vala
--- activity-log-manager-0.9.7.orig/src/security-widget.vala	2015-07-17 02:38:04.290396888 +0000
+++ activity-log-manager-0.9.7/src/security-widget.vala	2015-07-17 02:38:33.950303649 +0000
@@ -39,7 +39,7 @@
 
 		private void set_up_ui () {
 
-			var liststore_delay = new ListStore (2, typeof (string), typeof (int));
+			var liststore_delay = new Gtk.ListStore (2, typeof (string), typeof (int));
 			TreeIter iter;
 			liststore_delay.append (out iter);
 			liststore_delay.set (iter, 0, "1 second", 1, 0);
diff -Naur activity-log-manager-0.9.7.orig/src/unified-privacy-applications.vala activity-log-manager-0.9.7/src/unified-privacy-applications.vala
--- activity-log-manager-0.9.7.orig/src/unified-privacy-applications.vala	2015-07-17 02:38:04.290396888 +0000
+++ activity-log-manager-0.9.7/src/unified-privacy-applications.vala	2015-07-17 02:38:33.936970357 +0000
@@ -60,7 +60,7 @@
 			}
 		}
 
-		public void get_count_for_app(string id, TreeIter iter, ListStore store) {
+		public void get_count_for_app(string id, TreeIter iter, Gtk.ListStore store) {
 			this.blacklist_interface.get_count_for_app(id, iter, store);
 		}
 
@@ -131,7 +131,7 @@
 	private class ApplicationsTreeView : Gtk.Box {
 	
 		private ApplicationBlacklist app_blacklist;
-		private ListStore store;
+		private Gtk.ListStore store;
 		private TreeView treeview;
 		
 		HashTable<string, AppChooseInfo> blocked_apps;
@@ -145,7 +145,7 @@
 			N_COLS
 		}
 
-		public ListStore liststore {
+		public Gtk.ListStore liststore {
 			get {
 				return this.store;
 			}
@@ -160,7 +160,7 @@
 			this.blocked_apps = blocked;
 			this.unblocked_apps = unblocked;
 
-			this.store = new ListStore (TreeViewCols.N_COLS,
+			this.store = new Gtk.ListStore (TreeViewCols.N_COLS,
 										typeof (string),
 										typeof (Gdk.Pixbuf),
 										typeof (string));
@@ -292,7 +292,7 @@
 	private class AppSelectionTreeView : Gtk.Box {
 
 		private ApplicationBlacklist app_blacklist;
-		private ListStore store;
+		private Gtk.ListStore store;
 		private TreeView treeview;
 		
 		enum TreeViewCols
@@ -312,7 +312,7 @@
 			}
 		}
 
-		public ListStore liststore {
+		public Gtk.ListStore liststore {
 			get {
 				return store;
 			}
@@ -323,7 +323,7 @@
 			
 			this.app_blacklist = app_blacklist;
 			
-			this.store = new ListStore (TreeViewCols.N_COLS,
+			this.store = new Gtk.ListStore (TreeViewCols.N_COLS,
 										typeof (string),
 										typeof (Gdk.Pixbuf),
 										typeof (string),
@@ -426,7 +426,7 @@
 			}
 		}
 
-		public ListStore liststore {
+		public Gtk.ListStore liststore {
 			get {
 				return treeview.liststore;
 			}
diff -Naur activity-log-manager-0.9.7.orig/src/unified-privacy.vala activity-log-manager-0.9.7/src/unified-privacy.vala
--- activity-log-manager-0.9.7.orig/src/unified-privacy.vala	2015-07-17 02:38:04.290396888 +0000
+++ activity-log-manager-0.9.7/src/unified-privacy.vala	2015-07-17 02:38:33.886970514 +0000
@@ -55,7 +55,7 @@
 		
 		//General stuffs
 		private TreeView exception_list;
-		private ListStore exception_list_store;
+		private Gtk.ListStore exception_list_store;
 		
 		private Switch record_button;
 		
@@ -262,7 +262,7 @@
 			exception_vbox.pack_start (al2, false, false, 5);
 
 			// Exception TreeView
-			this.exception_list_store = new ListStore (4, 
+			this.exception_list_store = new Gtk.ListStore (4, 
 				typeof(string),      // Full file path of application desktop file.
 				typeof(Gdk.Pixbuf?), // Pixbuf
 				typeof(string),      // File basename or ApplicationName
@@ -291,7 +291,7 @@
 			// Add/Remove buttons
 			var exception_toolbar = new Toolbar();
 			exception_toolbar.toolbar_style = ToolbarStyle.ICONS;
-			exception_toolbar.icon_size = 1;
+			exception_toolbar.icon_size = Gtk.IconSize.MENU;
 			exception_toolbar.icon_size_set = true;
 			exception_toolbar.visible = true;
 			
