diff -Naur guake-0.5.0.orig/src/guake guake-0.5.0/src/guake
--- guake-0.5.0.orig/src/guake	2014-10-05 17:49:44.240402904 +0000
+++ guake-0.5.0/src/guake	2014-10-05 17:51:22.280038750 +0000
@@ -634,6 +634,9 @@
         scale_factor = 2 ** (self.font_scale_index / 6)
         new_size = int(scale_factor * font.get_size())
 
+        if new_size == 0:
+            return
+
         if font.get_size_is_absolute():
             font.set_absolute_size(new_size)
         else:
