diff --git a/src/widgets/dstyle.cpp b/src/widgets/dstyle.cpp
index fe80df89..ddd2b238 100644
--- a/src/widgets/dstyle.cpp
+++ b/src/widgets/dstyle.cpp
@@ -1995,7 +1995,7 @@ int DStyle::pixelMetric(QStyle::PixelMetric m, const QStyleOption *opt, const QW
     case PM_ListViewIconSize:
     case PM_LargeIconSize:
         return 24;
-#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 2, 0))
     case PM_LineEditIconSize:
         return widget ? (widget->height() < 34 ? 16 : 32) : 24;
 #endif
