diff --git a/src/editor/dtextedit.cpp b/src/editor/dtextedit.cpp
index 473a557d..da322f69 100644
--- a/src/editor/dtextedit.cpp
+++ b/src/editor/dtextedit.cpp
@@ -33,9 +33,9 @@
 #include "insertblockbytextcommond.h"
 
 
-#include <KF5/KSyntaxHighlighting/definition.h>
-#include <KF5/KSyntaxHighlighting/syntaxhighlighter.h>
-#include <KF5/KSyntaxHighlighting/theme.h>
+#include <KSyntaxHighlighting/definition.h>
+#include <KSyntaxHighlighting/syntaxhighlighter.h>
+#include <KSyntaxHighlighting/theme.h>
 
 #include <QAbstractTextDocumentLayout>
 #include <QTextDocumentFragment>
diff --git a/src/editor/showflodcodewidget.h b/src/editor/showflodcodewidget.h
index def67adc..5d2e5b7b 100644
--- a/src/editor/showflodcodewidget.h
+++ b/src/editor/showflodcodewidget.h
@@ -23,10 +23,10 @@
 #include <DFrame>
 #include <DPlainTextEdit>
 #include <DApplicationHelper>
-#include <KF5/KSyntaxHighlighting/repository.h>
-#include <KF5/KSyntaxHighlighting/definition.h>
-#include <KF5/KSyntaxHighlighting/syntaxhighlighter.h>
-#include <KF5/KSyntaxHighlighting/theme.h>
+#include <KSyntaxHighlighting/repository.h>
+#include <KSyntaxHighlighting/definition.h>
+#include <KSyntaxHighlighting/syntaxhighlighter.h>
+#include <KSyntaxHighlighting/theme.h>
 
 DWIDGET_USE_NAMESPACE
 
