diff -ur gcin-2.8.9.orig/gtk-im/Makefile gcin-2.8.9/gtk-im/Makefile
--- gcin-2.8.9.orig/gtk-im/Makefile	2019-03-14 21:31:09.553500793 +0800
+++ gcin-2.8.9/gtk-im/Makefile	2019-03-14 22:07:00.917828776 +0800
@@ -1,7 +1,7 @@
 include ../config.mak
 
 OBJS = imgcin.o gtkimcontextgcin.o
-GTK2IM=gtk-2.0/immodules
+GTK2IM=gtk-2.0/`pkg-config --variable=gtk_binary_version gtk+-2.0`/immodules
 GTKINC=`pkg-config --cflags gtk+-2.0`
 LDFLAGS=`pkg-config --libs gtk+-2.0`
 IMMODULES=$(libdir)/$(GTK2IM)
diff -ur gcin-2.8.9.orig/gtk3-im/Makefile gcin-2.8.9/gtk3-im/Makefile
--- gcin-2.8.9.orig/gtk3-im/Makefile	2019-03-14 21:31:09.553500793 +0800
+++ gcin-2.8.9/gtk3-im/Makefile	2019-03-14 22:06:40.175065314 +0800
@@ -1,7 +1,7 @@
 include ../config.mak
 
 OBJS = imgcin.o gtkimcontextgcin.o
-GTK3IM=gtk-3.0/immodules
+GTK3IM=gtk-3.0/`pkg-config --variable=gtk_binary_version gtk+-3.0`/immodules
 GTKINC=`pkg-config --cflags gtk+-3.0`
 LDFLAGS=`pkg-config --libs gtk+-3.0`
 LDFLAGS+='-Wl,--as-needed'
