--- tilda-0.9.6/src/key_grabber.c	2008-04-08 01:12:31.000000000 +0300
+++ tilda-0.9.6/src/key_grabber.c	2009-10-16 17:39:16.399679408 +0300
@@ -190,7 +190,7 @@
          * Overriding the user time here seems to work a lot better than calling
          * gtk_window_present_with_time() here, or at the end of the function. I have
          * no idea why, they should do the same thing. */
-        gdk_x11_window_set_user_time (GTK_WIDGET(tw->window)->window,
+        gdk_x11_window_set_user_time (GTK_WIDGET(tw->window),
                                       tomboy_keybinder_get_current_event_time());
         gtk_window_move (GTK_WINDOW(tw->window), config_getint ("x_pos"), config_getint ("y_pos"));
         gtk_widget_show (GTK_WIDGET(tw->window));

--- tilda-0.9.6/tilda.desktop.in	2007-03-16 04:53:34.000000000 +0100
+++ tilda-0.9.6/tilda.desktop.in	2011-08-28 19:08:53.000000000 +0200
@@ -2 +1,0 @@
-Encoding=UTF-8
@@ -4,2 +3,2 @@
-Exec=@BINDIR@/tilda
-Icon=@PIXMAPSDIR@/tilda.png
+Exec=tilda
+Icon=tilda
@@ -8 +7 @@
-Categories=GNOME;GTK;Application;Utility;TerminalEmulator;
+Categories=GNOME;GTK;Utility;TerminalEmulator;System;
