commit 7bb2e4711410cdff372e486b91a01e110b8331d0
Author: Anatol Pomozov <anatol.pomozov@gmail.com>
Date:   Sat Mar 21 09:13:31 2020 -0700

    ArchLinux fixes

diff --git a/Gemfile b/Gemfile
index 68e4f2df1ff..32395440994 100644
--- a/Gemfile
+++ b/Gemfile
@@ -324,6 +324,9 @@ gem 'gettext', '~> 3.3', require: false, group: :development
 
 gem 'batch-loader', '~> 2.0.1'
 
+gem 'irb' # https://bugs.archlinux.org/task/68569
+gem 'rake'
+
 # Perf bar
 gem 'peek', '~> 1.1'
 
