diff -Naur steadyflow-0.2.0.orig/cmake/FindExtraDeps.cmake steadyflow-0.2.0/cmake/FindExtraDeps.cmake
--- steadyflow-0.2.0.orig/cmake/FindExtraDeps.cmake	2012-06-30 14:39:35.000000000 +0200
+++ steadyflow-0.2.0/cmake/FindExtraDeps.cmake	2017-07-16 09:54:03.871584093 +0200
@@ -1,5 +1,5 @@
 message("-- Checking for libgee...")
-pkg_check_modules(GEE gee-1.0)
+pkg_check_modules(GEE gee-0.8)
 message("-- Checking for libnotify >= 0.7...")
 pkg_check_modules(LIBNOTIFY libnotify>=0.7)
 
diff -Naur steadyflow-0.2.0.orig/CMakeLists.txt steadyflow-0.2.0/CMakeLists.txt
--- steadyflow-0.2.0.orig/CMakeLists.txt	2012-06-30 14:39:35.000000000 +0200
+++ steadyflow-0.2.0/CMakeLists.txt	2017-07-16 09:54:03.871584093 +0200
@@ -48,7 +48,7 @@
 )
 
 set(VALA_PACKAGES
-	gee-1.0
+	gee-0.8
 	gio-2.0
 	glib-2.0
 	gtk+-3.0
diff -Naur steadyflow-0.2.0.orig/Steadyflow/FileListController.vala steadyflow-0.2.0/Steadyflow/FileListController.vala
--- steadyflow-0.2.0.orig/Steadyflow/FileListController.vala	2012-06-30 14:39:35.000000000 +0200
+++ steadyflow-0.2.0/Steadyflow/FileListController.vala	2017-07-16 09:53:53.937975668 +0200
@@ -27,7 +27,7 @@
 	private static const double REDRAW_SEC = 0.2;
 
 	private TreeView tree;
-	private ListStore model;
+	private Gtk.ListStore model;
 	private string filter = "";
 	private Timer redraw_timer;
 
@@ -46,7 +46,7 @@
 		column.pack_start (renderer, true);
 		column.set_cell_data_func (renderer, set_cell_data);
 		
-		model = new ListStore.newv ({ typeof (IDownloadFile) });
+		model = new Gtk.ListStore.newv ({ typeof (IDownloadFile) });
 		tree.set_model (model);
 		tree.append_column (column);
 		
diff -Naur steadyflow-0.2.0.orig/Steadyflow/MainWindow.vala steadyflow-0.2.0/Steadyflow/MainWindow.vala
--- steadyflow-0.2.0.orig/Steadyflow/MainWindow.vala	2012-06-30 14:39:35.000000000 +0200
+++ steadyflow-0.2.0/Steadyflow/MainWindow.vala	2017-07-16 10:41:53.590124439 +0200
@@ -230,6 +230,7 @@
 					return;
 			}
 			
+			file.pause ();
 			Services.download.remove_file (file);
 		}
 
diff -Naur steadyflow-0.2.0.orig/Steadyflow/Services.vala steadyflow-0.2.0/Steadyflow/Services.vala
--- steadyflow-0.2.0.orig/Steadyflow/Services.vala	2012-06-30 14:39:35.000000000 +0200
+++ steadyflow-0.2.0/Steadyflow/Services.vala	2017-07-16 10:07:00.382761532 +0200
@@ -61,7 +61,7 @@
 				 "information to the developer:\n\nError Code %d:\n\n%s").printf (e.code, message));
 			
 		md.run ();
-		exit(1);
+		GLib.Process.exit(1);
 	}
 }
 
diff -Naur steadyflow-0.2.0.orig/Steadyflow.Core/GioDownloadFile.vala steadyflow-0.2.0/Steadyflow.Core/GioDownloadFile.vala
--- steadyflow-0.2.0.orig/Steadyflow.Core/GioDownloadFile.vala	2012-06-30 14:39:35.000000000 +0200
+++ steadyflow-0.2.0/Steadyflow.Core/GioDownloadFile.vala	2017-07-16 09:53:53.934642242 +0200
@@ -160,7 +160,12 @@
 				// Not mounted
 				MountOperation mount_op;
 				get_mount_operation (out mount_op);
-				yield remote.mount_enclosing_volume (MountMountFlags.NONE, mount_op);
+				try {
+					yield remote.mount_enclosing_volume (MountMountFlags.NONE, mount_op);
+				}
+				catch (Error e) {
+					// The volume might already be mounted. Quietly continue.
+				}
 			}
 			
 			try {
diff -Naur steadyflow-0.2.0.orig/Steadyflow.UI/GtkBuilderDialog.vala steadyflow-0.2.0/Steadyflow.UI/GtkBuilderDialog.vala
--- steadyflow-0.2.0.orig/Steadyflow.UI/GtkBuilderDialog.vala	2012-06-30 14:39:35.000000000 +0200
+++ steadyflow-0.2.0/Steadyflow.UI/GtkBuilderDialog.vala	2021-04-16 18:41:51.622494074 +0200
@@ -24,7 +24,7 @@
 public abstract class GtkBuilderDialog : Dialog, IGtkBuilderContainer {
 	private Builder builder;
 
-	public GtkBuilderDialog (string file_id, Window? parent, bool modal) {
+	protected GtkBuilderDialog (string file_id, Window? parent, bool modal) {
 		if (parent != null)
 			set_transient_for (parent);
 		
diff -Naur steadyflow-0.2.0.orig/Steadyflow.UI/GtkBuilderWindow.vala steadyflow-0.2.0/Steadyflow.UI/GtkBuilderWindow.vala
--- steadyflow-0.2.0.orig/Steadyflow.UI/GtkBuilderWindow.vala	2012-06-30 14:39:35.000000000 +0200
+++ steadyflow-0.2.0/Steadyflow.UI/GtkBuilderWindow.vala	2021-04-16 18:41:49.602450522 +0200
@@ -24,7 +24,7 @@
 public abstract class GtkBuilderWindow : Window, IGtkBuilderContainer {
 	private Builder builder;
 
-	public GtkBuilderWindow (string file_id) throws Error {
+	protected GtkBuilderWindow (string file_id) throws Error {
 		builder = init_builder (this, file_id);
 	}
 	

