diff --git a/libphobos/configure.ac b/libphobos/configure.ac
index 69dfc35..89b8136 100644
--- a/libphobos/configure.ac
+++ b/libphobos/configure.ac
@@ -215,9 +215,9 @@ dnl (# Default case for install directory for include files.) and on
 # will have to modify gcc/configure.ac ..
 # For now, basic workaround for cross compilers ..
 if test "${host}" != "${target}"; then
-    gdc_include_dir='${libdir}/gcc/${host_alias}'/${d_gcc_ver}/include/d
+    gdc_include_dir='${libdir}/gcc/${host_alias}'/${d_gcc_ver}/include/dlang
 else
-    gdc_include_dir='${prefix}'/include/d/${d_gcc_ver}
+    gdc_include_dir='${prefix}'/include/dlang/gcc
 fi
 AC_SUBST(gdc_include_dir)
 AC_ARG_WITH([cross-host],
