--- src/main.c	2014-06-24 08:15:04.211951141 +0200
+++ src/main.c.new	2014-06-24 08:15:16.216043962 +0200
@@ -97,7 +97,7 @@
 				      GTK_DIALOG_MODAL|GTK_DIALOG_DESTROY_WITH_PARENT, /* GtkDialogFlags */
                                       GTK_MESSAGE_ERROR,
                                       GTK_BUTTONS_OK,
-				      message);
+				      "%s", message);
   g_assert(d);
   /*gint result =*/ gtk_dialog_run(GTK_DIALOG(d));
 #if 0
@@ -120,7 +120,7 @@
 				      GTK_DIALOG_MODAL|GTK_DIALOG_DESTROY_WITH_PARENT, /* GtkDialogFlags */
                                       GTK_MESSAGE_INFO,
                                       GTK_BUTTONS_OK,
-				      message);
+				      "%s", message);
   g_assert(d);
   gtk_dialog_run(GTK_DIALOG(d));
   gtk_widget_destroy(d);
