--- src/src/github.com/syncthing/syncthing/build.go~	2023-01-02 11:59:44.508141672 +0300
+++ src/src/github.com/syncthing/syncthing/build.go	2023-01-02 12:00:13.064820683 +0300
@@ -397,7 +397,7 @@
 	if runtime.GOARCH == "amd64" {
 		switch runtime.GOOS {
 		case buildpkg.Darwin, buildpkg.Linux, buildpkg.FreeBSD: // , "windows": # See https://github.com/golang/go/issues/27089
-			args = append(args, "-race")
+			//args = append(args, "-race")
 		}
 	}
 
