From c62e03344fe3af8d061f3ff881ab3cb7bfa0e1e9 Mon Sep 17 00:00:00 2001
From: Tom Hughes <tom@compton.nu>
Date: Mon, 23 Jan 2023 19:01:17 +0000
Subject: [PATCH] Include cstdint for uint16_t

---
 include/mapnik/wkb.hpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/mapnik/wkb.hpp b/include/mapnik/wkb.hpp
index 939fc17b74..52b902ae97 100644
--- a/include/mapnik/wkb.hpp
+++ b/include/mapnik/wkb.hpp
@@ -28,6 +28,9 @@
 #include <mapnik/geometry.hpp>
 #include <mapnik/util/noncopyable.hpp>
 
+// stl
+#include <cstdint>
+
 namespace mapnik
 {
 
