diff -upr android-tools-31.0.3p1.orig/vendor/adb/tls/include/adb/tls/tls_connection.h android-tools-31.0.3p1/vendor/adb/tls/include/adb/tls/tls_connection.h
--- android-tools-31.0.3p1.orig/vendor/adb/tls/include/adb/tls/tls_connection.h	2021-03-22 19:48:10.000000000 +0200
+++ android-tools-31.0.3p1/vendor/adb/tls/include/adb/tls/tls_connection.h	2022-06-11 19:39:36.472011172 +0300
@@ -19,6 +19,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <functional>
 #include <string_view>
 #include <vector>
 
