--- onboard-1.2.0/Onboard/pypredict/lm/lm_dynamic_cached.h	2016-02-18 14:31:51.000000000 +0100
+++ /tmp/lm_dynamic_cached.h.new	2016-06-14 19:04:23.924349370 +0200
@@ -224,7 +224,7 @@
     public:
         typedef _DynamicModelKN<TNGRAMS> Base;
         static const Smoothing DEFAULT_SMOOTHING = ABS_DISC_I;
-        static const double DEFAULT_LAMBDA = 0.3;  // Jelinek-Mercer weights
+        const double DEFAULT_LAMBDA = 0.3;  // Jelinek-Mercer weights
 
     public:
         _CachedDynamicModel()
