--- tigervnc-1.0.90/unix/xserver/hw/vnc/Makefile.am~	2011-04-20 13:46:16.993047050 +0200
+++ tigervnc-1.0.90/unix/xserver/hw/vnc/Makefile.am	2011-04-20 14:03:35.942724608 +0200
@@ -1,11 +1,12 @@
 TIGERVNC_SRCDIR=${top_srcdir}/../..
 LIB_DIR=${top_builddir}/../../common
 
+OS_LIB=$(LIB_DIR)/os/libos.a
 RFB_LIB=$(LIB_DIR)/rfb/librfb.la
 RDR_LIB=$(LIB_DIR)/rdr/librdr.la
 NETWORK_LIB=$(LIB_DIR)/network/libnetwork.la
 XREGION_LIB=$(LIB_DIR)/Xregion/libXregion.la
-COMMON_LIBS=$(NETWORK_LIB) $(RFB_LIB) $(RDR_LIB) $(XREGION_LIB)
+COMMON_LIBS=$(NETWORK_LIB) $(OS_LIB) $(RFB_LIB) $(RDR_LIB) $(XREGION_LIB)
 
 noinst_LTLIBRARIES = libvnccommon.la
 
