Index: trunk/geogebra/giac/src/giac/cpp/subst.cc
===================================================================
--- a/src/subst.cc
+++ b/src/subst.cc
@@ -3277,4 +3277,6 @@
   }
   gen invfracpow(const gen & e,GIAC_CONTEXT){
+    if (e.is_symb_of_sommet(at_same))
+      return symbolic(at_same,makesequence(invfracpow(e._SYMBptr->feuille[0],contextptr),invfracpow(e._SYMBptr->feuille[1],contextptr)));
     vector< gen_op_context > invfracpow_v(1,do_invfracpow);
     vector<const unary_function_ptr *> inv_v(1,at_inv);
