diff -ru src.orig/lisp/interrupt.c src/lisp/interrupt.c
--- src.orig/lisp/interrupt.c	2022-03-11 21:13:20.249071179 +0100
+++ src/lisp/interrupt.c	2022-03-11 21:14:27.152779948 +0100
@@ -405,7 +405,7 @@
 * Noise to install handlers.                                     *
 \****************************************************************/
 
-char altstack[SIGNAL_STACK_SIZE];
+char altstack[8192];
 
 void
 interrupt_install_low_level_handler(int signal, void handler(HANDLER_ARGS))
