diff -Naur Gnome-Pie-0.6.1.orig/src/actionGroups/devicesGroup.vala Gnome-Pie-0.6.1/src/actionGroups/devicesGroup.vala
--- Gnome-Pie-0.6.1.orig/src/actionGroups/devicesGroup.vala	2015-04-20 14:16:01.012535306 +0000
+++ Gnome-Pie-0.6.1/src/actionGroups/devicesGroup.vala	2015-04-20 14:34:13.181211266 +0000
@@ -33,7 +33,7 @@
     public static GroupRegistry.TypeDescription register() {
         var description = new GroupRegistry.TypeDescription();
         description.name = _("Group: Devices");
-        description.icon = "harddrive";
+        description.icon = "drive-harddisk";
         description.description = _("Shows a Slice for each plugged in devices, like USB-Sticks.");
         description.id = "devices";
         return description;
@@ -82,7 +82,7 @@
 
     private void load() {
         // add root device
-        this.add_action(new UriAction(_("Root"), "harddrive", "file:///"));
+        this.add_action(new UriAction(_("Root"), "drive-harddisk", "file:///"));
 
         // add all other devices
         foreach(var mount in this.monitor.get_mounts()) {
diff -Naur Gnome-Pie-0.6.1.orig/src/actionGroups/menuGroup.vala Gnome-Pie-0.6.1/src/actionGroups/menuGroup.vala
--- Gnome-Pie-0.6.1.orig/src/actionGroups/menuGroup.vala	2015-04-20 14:16:01.012535306 +0000
+++ Gnome-Pie-0.6.1/src/actionGroups/menuGroup.vala	2015-04-20 14:22:30.410914120 +0000
@@ -33,7 +33,7 @@
     public static GroupRegistry.TypeDescription register() {
         var description = new GroupRegistry.TypeDescription();
         description.name = _("Group: Main menu");
-        description.icon = "gnome-main-menu";
+        description.icon = "start-here";
         description.description = _("Displays your main menu structure.");
         description.id = "menu";
         return description;
diff -Naur Gnome-Pie-0.6.1.orig/src/actionGroups/sessionGroup.vala Gnome-Pie-0.6.1/src/actionGroups/sessionGroup.vala
--- Gnome-Pie-0.6.1.orig/src/actionGroups/sessionGroup.vala	2015-04-20 14:16:01.012535306 +0000
+++ Gnome-Pie-0.6.1/src/actionGroups/sessionGroup.vala	2015-04-20 14:39:11.883213936 +0000
@@ -33,7 +33,7 @@
     public static GroupRegistry.TypeDescription register() {
         var description = new GroupRegistry.TypeDescription();
         description.name = _("Group: Session Control");
-        description.icon = "gnome-logout";
+        description.icon = "system-log-out";
         description.description = _("Shows a Slice for Shutdown, Reboot, and Hibernate.");
         description.id = "session";
         return description;
@@ -57,14 +57,14 @@
 //        iface = GLib.Bus.get_proxy_sync(GLib.BusType.SESSION, "org.kde.ksmserver", "/KSMServer");
 //        iface = GLib.Bus.get_proxy_sync(GLib.BusType.SESSION, "org.freedesktop.ConsoleKit", "/org/freedesktop/ConsoleKit/Manager");
 
-        this.add_action(new AppAction(_("Shutdown"), "gnome-shutdown",
-            "dbus-send --print-reply --dest=org.gnome.SessionManager /org/gnome/SessionManager org.gnome.SessionManager.RequestShutdown"));
+        this.add_action(new AppAction(_("Shutdown"), "system-shutdown",
+            "dbus-send --print-reply --dest=org.gnome.SessionManager /org/gnome/SessionManager org.gnome.SessionManager.Shutdown"));
 
-        this.add_action(new AppAction(_("Logout"), "gnome-session-logout",
+        this.add_action(new AppAction(_("Logout"), "system-log-out",
             "dbus-send --print-reply --dest=org.gnome.SessionManager /org/gnome/SessionManager org.gnome.SessionManager.Logout uint32:1"));
 
-        this.add_action(new AppAction(_("Reboot"), "gnome-session-reboot",
-            "dbus-send --print-reply --dest=org.gnome.SessionManager /org/gnome/SessionManager org.gnome.SessionManager.RequestReboot"));
+        this.add_action(new AppAction(_("Reboot"), "view-refresh",
+            "dbus-send --print-reply --dest=org.gnome.SessionManager /org/gnome/SessionManager org.gnome.SessionManager.Reboot"));
     }
 
     // TODO: check for available interfaces --- these may work too:
diff -Naur Gnome-Pie-0.6.1.orig/src/actionGroups/windowListGroup.vala Gnome-Pie-0.6.1/src/actionGroups/windowListGroup.vala
--- Gnome-Pie-0.6.1.orig/src/actionGroups/windowListGroup.vala	2015-04-20 14:16:01.012535306 +0000
+++ Gnome-Pie-0.6.1/src/actionGroups/windowListGroup.vala	2015-04-20 14:44:10.345223458 +0000
@@ -32,7 +32,7 @@
     public static GroupRegistry.TypeDescription register() {
         var description = new GroupRegistry.TypeDescription();
         description.name = _("Group: Window List");
-        description.icon = "gnome-window-manager";
+        description.icon = "preferences-system-windows";
         description.description = _("Shows a Slice for each of your opened Windows. Almost like Alt-Tab.");
         description.id = "window_list";
         return description;
diff -Naur Gnome-Pie-0.6.1.orig/src/gui/piePreviewAddSign.vala Gnome-Pie-0.6.1/src/gui/piePreviewAddSign.vala
--- Gnome-Pie-0.6.1.orig/src/gui/piePreviewAddSign.vala	2015-04-20 14:16:01.012535306 +0000
+++ Gnome-Pie-0.6.1/src/gui/piePreviewAddSign.vala	2015-04-20 14:32:42.304950652 +0000
@@ -87,7 +87,7 @@
     /////////////////////////////////////////////////////////////////////
 
     public void load() {
-        this.icon = new Icon("add", 36);
+        this.icon = new Icon("list-add", 36);
     }
 
     /////////////////////////////////////////////////////////////////////
diff -Naur Gnome-Pie-0.6.1.orig/src/gui/piePreviewDeleteSign.vala Gnome-Pie-0.6.1/src/gui/piePreviewDeleteSign.vala
--- Gnome-Pie-0.6.1.orig/src/gui/piePreviewDeleteSign.vala	2015-04-20 14:16:01.012535306 +0000
+++ Gnome-Pie-0.6.1/src/gui/piePreviewDeleteSign.vala	2015-04-20 14:36:32.490589922 +0000
@@ -85,7 +85,7 @@
     /////////////////////////////////////////////////////////////////////
 
     public void load() {
-        this.icon = new Icon("stock_delete", PiePreviewDeleteSign.radius*2);
+        this.icon = new Icon("edit-delete", PiePreviewDeleteSign.radius*2);
     }
 
     /////////////////////////////////////////////////////////////////////
diff -Naur Gnome-Pie-0.6.1.orig/src/pies/defaultConfig.vala Gnome-Pie-0.6.1/src/pies/defaultConfig.vala
--- Gnome-Pie-0.6.1.orig/src/pies/defaultConfig.vala	2015-04-20 14:16:01.012535306 +0000
+++ Gnome-Pie-0.6.1/src/pies/defaultConfig.vala	2015-04-20 15:04:38.770052475 +0000
@@ -26,11 +26,11 @@
     public void create_default_config() {
 
         // add a pie with playback controls
-        var multimedia = PieManager.create_persistent_pie(_("Multimedia"), "stock_media-play", new Trigger.from_string("<Control><Alt>m"));
-            multimedia.add_action(new KeyAction(_("Next Track"), "stock_media-next", "XF86AudioNext", true));
-            multimedia.add_action(new KeyAction(_("Stop"), "stock_media-stop", "XF86AudioStop"));
-            multimedia.add_action(new KeyAction(_("Previous Track"), "stock_media-prev", "XF86AudioPrev"));
-            multimedia.add_action(new KeyAction(_("Play/Pause"), "stock_media-play", "XF86AudioPlay"));
+        var multimedia = PieManager.create_persistent_pie(_("Multimedia"), "media-playback-start", new Trigger.from_string("<Control><Alt>m"));
+            multimedia.add_action(new KeyAction(_("Next Track"), "media-skip-forward", "XF86AudioNext", true));
+            multimedia.add_action(new KeyAction(_("Stop"), "media-playback-stop", "XF86AudioStop"));
+            multimedia.add_action(new KeyAction(_("Previous Track"), "media-skip-backward", "XF86AudioPrev"));
+            multimedia.add_action(new KeyAction(_("Play/Pause"), "media-playback-start", "XF86AudioPlay"));
 
         // add a pie with the users default applications
         var apps = PieManager.create_persistent_pie(_("Applications"), "applications-accessories", new Trigger.from_string("<Control><Alt>a"));
@@ -47,20 +47,20 @@
             bookmarks.add_group(new DevicesGroup(bookmarks.id));
 
         // add a pie with session controls
-        var session = PieManager.create_persistent_pie(_("Session"), "gnome-session-halt", new Trigger.from_string("<Control><Alt>q"));
+        var session = PieManager.create_persistent_pie(_("Session"), "system-log-out", new Trigger.from_string("<Control><Alt>q"));
             session.add_group(new SessionGroup(session.id));
 
         // add a pie with a main menu
-        var menu = PieManager.create_persistent_pie(_("Main Menu"), "alacarte", new Trigger.from_string("<Control><Alt>space"));
+        var menu = PieManager.create_persistent_pie(_("Main Menu"), "start-here", new Trigger.from_string("<Control><Alt>space"));
             menu.add_group(new MenuGroup(menu.id));
 
         // add a pie with window controls
-        var window = PieManager.create_persistent_pie(_("Window"), "gnome-window-manager", new Trigger.from_string("<Control><Alt>w"));
-            window.add_action(new KeyAction(_("Scale"), "top", "<Control><Alt>s"));
-            window.add_action(new KeyAction(_("Minimize"), "bottom", "<Alt>F9", true));
+        var window = PieManager.create_persistent_pie(_("Window"), "preferences-system-windows", new Trigger.from_string("<Control><Alt>w"));
+            window.add_action(new KeyAction(_("Scale"), "go-top", "<Control><Alt>s"));
+            window.add_action(new KeyAction(_("Minimize"), "go-bottom", "<Alt>F9", true));
             window.add_action(new KeyAction(_("Close"), "window-close", "<Alt>F4"));
-            window.add_action(new KeyAction(_("Maximize"), "window_fullscreen", "<Alt>F10"));
-            window.add_action(new KeyAction(_("Restore"), "window_nofullscreen", "<Alt>F5"));
+            window.add_action(new KeyAction(_("Maximize"), "view-fullscreen", "<Alt>F10"));
+            window.add_action(new KeyAction(_("Restore"), "view-restore", "<Alt>F5"));
 
         // save the configuration to file
         Pies.save();
diff -Naur Gnome-Pie-0.6.1.orig/src/utilities/config.vala Gnome-Pie-0.6.1/src/utilities/config.vala
--- Gnome-Pie-0.6.1.orig/src/utilities/config.vala	2015-04-20 14:16:01.012535306 +0000
+++ Gnome-Pie-0.6.1/src/utilities/config.vala	2015-04-20 14:16:34.909063576 +0000
@@ -192,7 +192,7 @@
 
         if (themes.size > 0) {
             if (current == "") {
-                current = "Unity";
+                current = "Adwaita";
                 warning("No theme specified! Using default...");
             }
             foreach (var t in themes) {
