--- intellij-community-pycharm-182.1177/build/dependencies/setupJdk.gradle	2019-04-01 13:07:28.258512665 +0200
+++ intellij-community-pycharm-182.1177/build/dependencies/setupJdk.gradle	2019-04-01 13:07:22.691827611 +0200
@@ -136,9 +136,7 @@
 
 // JDK 1.6
 repositories {
-  maven {
-    url 'https://repo.labs.intellij.net/java'
-  }
+  mavenCentral()
 }
 
 configurations {
diff -rupN intellij-community-pycharm-182.1177.orig/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy intellij-community-pycharm-182.1177/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy
--- intellij-community-pycharm-182.1177.orig/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy	2018-03-29 19:26:25.415841858 +0200
+++ intellij-community-pycharm-182.1177/platform/build-scripts/groovy/org/jetbrains/intellij/build/BuildOptions.groovy	2018-03-29 19:31:46.883868914 +0200
@@ -100,7 +100,7 @@ class BuildOptions {
    * Build number without product code (e.g. '162.500.10'), if {@code null} '&lt;baseline&gt;.SNAPSHOT' will be used. Use {@link BuildContext#buildNumber} to
    * get the actual build number in build scripts.
    */
-  String buildNumber = System.getProperty("build.number")
+  String buildNumber = System.getProperty("idea.build.number")
 
   /**
    * By default build process produces temporary and resulting files under projectHome/out/productName directory, use this property to
