diff -Naur cinnamon-control-center-1.7.3.orig/panels/power/cc-power-panel.c cinnamon-control-center-1.7.3/panels/power/cc-power-panel.c
--- cinnamon-control-center-1.7.3.orig/panels/power/cc-power-panel.c	2013-04-15 16:04:38.000000000 +0200
+++ cinnamon-control-center-1.7.3/panels/power/cc-power-panel.c	2013-04-22 05:04:26.466092668 +0200
@@ -1068,26 +1068,6 @@
                     G_CALLBACK (combo_enum_changed_cb),
                     self);
 
-  value = g_settings_get_enum (self->priv->gsd_settings, "lid-close-ac-action");
-  widget = GTK_WIDGET (gtk_builder_get_object (self->priv->builder,
-                                               "combobox_lid_ac"));
-  disable_unavailable_combo_items (self, GTK_COMBO_BOX (widget));
-  set_value_for_combo (GTK_COMBO_BOX (widget), value);
-  g_object_set_data (G_OBJECT(widget), "_gsettings_key", "lid-close-ac-action");
-  g_signal_connect (widget, "changed",
-                    G_CALLBACK (combo_enum_changed_cb),
-                    self);
-
-  value = g_settings_get_enum (self->priv->gsd_settings, "lid-close-battery-action");
-  widget = GTK_WIDGET (gtk_builder_get_object (self->priv->builder,
-                                               "combobox_lid_battery"));
-  disable_unavailable_combo_items (self, GTK_COMBO_BOX (widget));
-  set_value_for_combo (GTK_COMBO_BOX (widget), value);
-  g_object_set_data (G_OBJECT(widget), "_gsettings_key", "lid-close-battery-action");
-  g_signal_connect (widget, "changed",
-                    G_CALLBACK (combo_enum_changed_cb),
-                    self);
-
   widget = WID (self->priv->builder, "vbox_power");
   gtk_widget_reparent (widget, (GtkWidget *) self);
 
diff -Naur cinnamon-control-center-1.7.3.orig/panels/screen/cc-screen-panel.c cinnamon-control-center-1.7.3/panels/screen/cc-screen-panel.c
--- cinnamon-control-center-1.7.3.orig/panels/screen/cc-screen-panel.c	2013-04-15 16:04:38.000000000 +0200
+++ cinnamon-control-center-1.7.3/panels/screen/cc-screen-panel.c	2013-04-22 14:15:28.440488342 +0200
@@ -351,10 +351,6 @@
                       1, &value,
                       -1);
 
-  /* set both battery and ac keys */
-  g_settings_set_int (self->priv->gsd_settings, "sleep-display-ac", value);
-  g_settings_set_int (self->priv->gsd_settings, "sleep-display-battery", value);
-
   set_idle_delay_from_dpms (self, value);
 }
 
@@ -384,7 +380,7 @@
 {
   GtkTreeIter iter;
   GtkTreeModel *model;
-  gint value;
+  guint value;
   gint value_tmp, value_prev;
   gboolean ret;
   guint i;
@@ -399,7 +395,7 @@
   i = 0;
 
   /* try to make the UI match the AC setting */
-  value = g_settings_get_int (self->priv->gsd_settings, "sleep-display-ac");
+  g_settings_get (self->priv->session_settings, "idle-delay", "u", &value);
   do
     {
       gtk_tree_model_get (model, &iter,
@@ -510,7 +506,7 @@
   /* bind the auto dim checkbox */
   widget = WID ("screen_auto_reduce_checkbutton");
   g_settings_bind (self->priv->gsd_settings,
-                   "idle-dim-battery",
+                   "idle-dim",
                    widget, "active",
                    G_SETTINGS_BIND_DEFAULT);
 
