From 9c3c83d1a0cacde5f43c28752e1767cf9af802b9 Mon Sep 17 00:00:00 2001
From: Willem Toorop <willem@nlnetlabs.nl>
Date: Fri, 10 Nov 2017 10:52:50 +0100
Subject: [PATCH] Ignore SIGPIPE signal (for not suddenly stopping)

---
 src/stubby.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/stubby.c b/src/stubby.c
index 53996ad..a2f987f 100644
--- a/src/stubby.c
+++ b/src/stubby.c
@@ -819,13 +819,20 @@ main(int argc, char **argv)
 				        strerror(errno));
 				exit(EXIT_FAILURE);
 			}
-		} else
+		} else {
+#ifdef SIGPIPE
+			(void)signal(SIGPIPE, SIG_IGN);
+#endif
 			getdns_context_run(context);
+		}
 	} else
 #endif
 	{
 		stubby_local_log(NULL,GETDNS_LOG_UPSTREAM_STATS, GETDNS_LOG_DEBUG,
 			       "Starting DAEMON....\n");
+#ifdef SIGPIPE
+		(void)signal(SIGPIPE, SIG_IGN);
+#endif
 		getdns_context_run(context);
 	}
 
