diff -ru rw-0.9.orig/simplerw.c rw-0.9/simplerw.c
--- rw-0.9.orig/simplerw.c	2022-09-06 23:46:51.140196431 +0200
+++ rw-0.9/simplerw.c	2022-09-06 23:51:48.752989556 +0200
@@ -134,7 +134,7 @@
 		igraph_destroy(&igraph);
 		return(-1);
 	}
-	igraph_get_adjacency(&igraph, &imatrix, IGRAPH_GET_ADJACENCY_BOTH, 0);
+	igraph_get_adjacency(&igraph, &imatrix, IGRAPH_GET_ADJACENCY_BOTH, 0, 0);
 	igraph_destroy(&igraph);
 	if(igraph_matrix_nrow(&imatrix) > MAX_VERTICES)
 	{
