diff -aur visitors_0.7.old/Makefile visitors_0.7/Makefile
--- visitors_0.7.old/Makefile	2020-09-26 12:09:16.045972086 +0200
+++ visitors_0.7/Makefile	2020-09-26 12:09:48.899392633 +0200
@@ -14,10 +14,10 @@
 
 visitors.o: visitors.c blacklist.h
 visitors: $(OBJ)
-	$(CC) -o $(PRGNAME) $(CCOPT) $(DEBUG) $(OBJ)
+	$(CC) -o $(PRGNAME) $(CCOPT) $(LDFLAGS) $(DEBUG) $(OBJ)
 
 .c.o:
-	$(CC) -c $(CCOPT) $(DEBUG) $(COMPILE_TIME) $<
+	$(CC) -c $(CCOPT) $(LDFLAGS) $(DEBUG) $(COMPILE_TIME) $<
 
 clean:
 	rm -rf $(PRGNAME) *.o
