diff -wbBur sysstat-11.1.3.org/configure.in sysstat-11.1.3/configure.in
--- sysstat-11.1.3.org/configure.in	2015-02-13 17:05:44.000000000 +0300
+++ sysstat-11.1.3/configure.in	2015-02-16 13:25:26.201405593 +0300
@@ -209,21 +209,6 @@
 fi
 
 SADC_DIR=$AuxPrefix/lib
-if test -d $AuxPrefix/lib64; then
-    # Verify that this OS is really 64 bit
-    BITS=$(getconf LONG_BIT 2>/dev/null)
-    if test $? = 0; then
-        if test $BITS = 64; then
-            SADC_DIR=$AuxPrefix/lib64
-        fi
-    else
-        # Fallback: look for lm (long mode) flag to know if CPU is 64 bit
-        grep " lm " /proc/cpuinfo >/dev/null 2>&1
-        if test $? = 0; then
-            SADC_DIR=$AuxPrefix/lib64
-        fi
-    fi
-fi
 
 AC_MSG_CHECKING(sadc directory)
 AC_ARG_VAR([sa_lib_dir],[sadc directory])
