diff -Naur matplotlib-0.99.1.1-orig/src/_png.cpp matplotlib-0.99.1.1/src/_png.cpp
--- matplotlib-0.99.1.1-orig/src/_png.cpp	2010-01-25 01:42:32.000000000 -0500
+++ matplotlib-0.99.1.1/src/_png.cpp	2010-01-25 01:43:17.000000000 -0500
@@ -290,7 +290,7 @@
 
   //free the png memory
   png_read_end(png_ptr, info_ptr);
-  png_destroy_read_struct(&png_ptr, &info_ptr, png_infopp_NULL);
+  png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
   fclose(fp);
   for (row = 0; row < height; row++)
     delete [] row_pointers[row];
