diff -upr netsurf.orig/image/png.c netsurf/image/png.c
--- netsurf.orig/image/png.c	2010-01-25 02:52:03.000000000 +0200
+++ netsurf/image/png.c	2010-01-25 02:52:38.000000000 +0200
@@ -147,7 +147,7 @@ void info_callback(png_structp png, png_
 	if (color_type == PNG_COLOR_TYPE_PALETTE)
 		png_set_palette_to_rgb(png);
 	if (color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8)
-		png_set_gray_1_2_4_to_8(png);
+		png_set_expand_gray_1_2_4_to_8(png);
 	if (png_get_valid(png, info, PNG_INFO_tRNS))
 		png_set_tRNS_to_alpha(png);
 	if (bit_depth == 16)
