diff -ruN a/scripts/utils.mk b/scripts/utils.mk
--- a/scripts/utils.mk	2021-05-13 04:59:12.000000000 +0200
+++ b/scripts/utils.mk	2021-05-14 21:05:11.357878988 +0200
@@ -56,7 +56,7 @@
 
 do_compile_shared_library =			\
 	($(print_shared_lib_compile)		\
-	$(CC) --shared $^ '-Wl,-soname,$(1),-rpath=$$ORIGIN' -o $@ $(LIBS))
+	$(CC) --shared $^ '-Wl,-soname,$(1),-rpath=$$ORIGIN' $(LDFLAGS) -o $@ $(LIBS))
 
 do_compile_plugin_obj =				\
 	($(print_plugin_obj_compile)		\
