diff -ruN a/Documentation/Makefile b/Documentation/Makefile
--- a/Documentation/Makefile	2021-12-20 03:09:42.000000000 +0100
+++ b/Documentation/Makefile	2021-12-20 16:24:43.651313119 +0100
@@ -13,8 +13,6 @@
 _MAN_HTML=$(patsubst %.txt,%.html,$(MAN_TXT))
 _DOC_MAN3=$(patsubst %.txt,%.m,$(MAN3_TXT))
 
-OUTPUT		:= $(obj)/Documentation/
-
 MAN_XML=$(addprefix $(OUTPUT),$(_MAN_XML))
 MAN_HTML=$(addprefix $(OUTPUT),$(_MAN_HTML))
 DOC_MAN3=$(addprefix $(OUTPUT),$(_DOC_MAN3))
