From 9392146c07b93d8b90681dafbd74d03e15590f12 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 29 Feb 2016 03:38:39 +0000
Subject: [PATCH] raven: Restore lightdm (only) screen locking

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 raven/powerstrip.vala | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/raven/powerstrip.vala b/raven/powerstrip.vala
index 02edab9..e83d40f 100644
--- a/raven/powerstrip.vala
+++ b/raven/powerstrip.vala
@@ -12,24 +12,23 @@
 namespace Budgie
 {
 
-
-[DBus (name="org.gnome.ScreenSaver")]
-public interface ScreenSaver : Object
-{
-    public abstract void lock() throws Error;
-}
-
 [DBus (name="org.gnome.SessionManager")]
 public interface SessionManager : Object
 {
     public abstract async void Shutdown() throws Error;
 }
 
+[DBus (name="org.freedesktop.DisplayManager.Seat")]
+public interface DMSeat : Object
+{
+    public abstract void lock() throws IOError;
+}
+
 
 class PowerStrip : Gtk.EventBox
 {
 
-    private ScreenSaver? saver = null;
+    private DMSeat? saver = null;
     private SessionManager? session = null;
 
     private Gtk.Button? lock_btn = null;
@@ -37,8 +36,9 @@ class PowerStrip : Gtk.EventBox
 
     async void setup_dbus()
     {
+        var path = Environment.get_variable("XDG_SEAT_PATH");
         try {
-            saver = yield Bus.get_proxy(BusType.SESSION, "org.gnome.ScreenSaver", "/org/gnome/ScreenSaver");
+            saver = yield Bus.get_proxy(BusType.SYSTEM, "org.freedesktop.DisplayManager", path);
         } catch (Error e) {
             warning("Unable to contact login manager: %s", e.message);
             return;
