diff -ru linssid-3.6.orig/linssid-app/MainForm.cpp linssid-3.6/linssid-app/MainForm.cpp
--- linssid-3.6.orig/linssid-app/MainForm.cpp	2021-10-10 18:23:38.037465662 +0200
+++ linssid-3.6/linssid-app/MainForm.cpp	2021-10-10 18:32:43.396592202 +0200
@@ -28,6 +28,7 @@
 #include <qwt_plot_curve.h>
 #include <qwt_plot_grid.h>
 #include <qwt_symbol.h>
+#include <qwt_text.h>
 #include <qwt_plot_marker.h>
 #include <QPointF>
 #include <qwt_scale_draw.h>
@@ -1036,7 +1037,7 @@
                     MainForm::cellDataRay[tbi].signal ) );
             }
         } else {
-            MainForm::cellDataRay[tbi].pBandCurve->setSamples(0, 0, 0);
+            MainForm::cellDataRay[tbi].pBandCurve->setSamples((double*)0, (double*)0, 0);
         }
         // now the signal history plot
         int ixStart;
@@ -1055,7 +1056,7 @@
                     &(MainForm::cellDataRay[tbi].pHistory->signal[ixStart]), ixLength);
         } else {
 
-            MainForm::cellDataRay[tbi].pTimeCurve->setSamples(0, 0, 0);
+            MainForm::cellDataRay[tbi].pTimeCurve->setSamples((double*)0, (double*)0, 0);
         }
     }
     MainForm::mainFormWidget.chan24Plot->replot();
@@ -1424,4 +1425,4 @@
 //    const std::string whiteSpaces( " \f\n\r\t\v" );
 //    std::string::size_type pos = str.find_first_not_of( whiteSpaces );
 //    str.erase( pos + 1 );    
-//}
\ No hay ningún carácter de nueva línea al final del fichero
+//}
