diff --git a/scripts/src/3rdparty.lua b/scripts/src/3rdparty.lua
index 5744e8d5a2f..4848d905d50 100644
--- a/scripts/src/3rdparty.lua
+++ b/scripts/src/3rdparty.lua
@@ -845,7 +845,6 @@ project "lua"
 	kind "StaticLib"
 
 	options {
-		"ForceCPP",
 	}
 
 	configuration { "gmake or ninja" }
@@ -933,7 +932,6 @@ project "lualibs"
 	kind "StaticLib"
 
 	options {
-		"ForceCPP",
 	}
 
 	configuration { "gmake or ninja" }
diff --git a/src/frontend/mame/luaengine.cpp b/src/frontend/mame/luaengine.cpp
index ebf2f701e15..9adfd413321 100644
--- a/src/frontend/mame/luaengine.cpp
+++ b/src/frontend/mame/luaengine.cpp
@@ -41,11 +41,12 @@
 //  LUA ENGINE
 //**************************************************************************
 
+extern "C" {
 int luaopen_zlib(lua_State *const L);
-extern "C" int luaopen_lfs(lua_State *L);
+int luaopen_lfs(lua_State *L);
 int luaopen_linenoise(lua_State *L);
 int luaopen_lsqlite3(lua_State *L);
-
+}
 
 template <typename T>
 struct lua_engine::devenum
diff --git a/src/frontend/mame/luaengine.h b/src/frontend/mame/luaengine.h
index 0767a349ae2..c848952db14 100644
--- a/src/frontend/mame/luaengine.h
+++ b/src/frontend/mame/luaengine.h
@@ -20,7 +20,6 @@
 #include <tuple>
 #include <vector>
 
-#define SOL_USING_CXX_LUA 1
 #ifdef MAME_DEBUG
 #define SOL_ALL_SAFETIES_ON 1
 #else
