diff --git a/setup.py b/setup.py
index 7539b13..5127ece 100644
--- a/setup.py
+++ b/setup.py
@@ -154,6 +154,7 @@ def find_lua_build(no_luajit=False):
         for lua_version in ('5.2', '5.1')
         for name in ('lua%s' % lua_version, 'lua-%s' % lua_version, 'lua')
     ]
+    packages += [('lua52', '5.2')]
 
     for package_name, min_version in packages:
         print("Checking for installed %s library using pkg-config" %
