--- iftop.c.old 2014-01-20 04:21:19.000000000 +0800
+++ iftop.c 2014-03-03 12:49:57.383335066 +0800
@@ -713,7 +713,7 @@ void packet_init() {
     if(have_hw_addr) {
       fprintf(stderr, "MAC address is:");
       for (i = 0; i < 6; ++i)
-	fprintf(stderr, "%c%02x", i ? ':' : ' ', (unsigned int)if_hw_addr[i]);
+	fprintf(stderr, "%c%02x", i ? ':' : ' ', (unsigned char)if_hw_addr[i]);
       fprintf(stderr, "\n");
     }
 
