diff --git a/ming.i b/ming.i
index 1860f770..a90e7e0d 100644
--- a/ming.i
+++ b/ming.i
@@ -11,7 +11,10 @@
 
 %}
 
-%typemap(python,in) FILE * {
+typedef unsigned char byte;
+
+#ifdef SWIGPYTHON
+%typemap(in) FILE * {
   int temp = PyObject_AsFileDescriptor($input);
   if (temp == -1) {
     PyErr_SetString(PyExc_TypeError, "Need a file!");
@@ -20,7 +23,7 @@
   $1 = fdopen(temp, "r");
 }
 
-%typemap(python,in) char * {
+%typemap(in) char * {
   if (!PyString_Check($input)) {
     PyErr_SetString(PyExc_TypeError, "not a string");
     return NULL;
@@ -28,17 +31,17 @@
   $1 = PyString_AsString($input);
 }
 
-typedef unsigned char byte;
-
-%typemap(python,in) byte * {
+%typemap(in) byte * {
   if (!PyString_Check($input)) {
     PyErr_SetString(PyExc_TypeError, "not a string");
     return NULL;
   }
   $1 = PyString_AsString($input);
 }
+#endif
 
-%typemap(ruby,in) int * {
+#ifdef SWIGRUBY
+%typemap(in) int * {
   if (NIL_P($input)) {
          $1 = 0;
   } else {
@@ -46,6 +49,7 @@ typedef unsigned char byte;
     $1 = (int *)RARRAY($input)->ptr;
   }
 }
+#endif
 
 %pointer_functions(int, intp);
 %pointer_functions(float, floatp);
