diff -aur bonnie++-1.03e/Makefile bonnie++-1.03ef/Makefile
--- bonnie++-1.03e/Makefile	2013-05-13 08:40:45.122959831 +0000
+++ bonnie++-1.03ef/Makefile	2013-05-13 08:43:20.365248937 +0000
@@ -33,8 +33,8 @@
 	$(CXX) zcav.cpp -o zcav $(ZCAVOBJS) $(LFLAGS)
 
 install-bin: $(EXES)
-	mkdir -p $(eprefix)/bin $(eprefix)/sbin
-	${INSTALL} -s $(EXES) $(eprefix)/sbin
+	mkdir -p $(eprefix)/bin 
+	${INSTALL} -s $(EXES) $(eprefix)/bin
 	${INSTALL} $(SCRIPTS) $(eprefix)/bin
 
 install: install-bin
