--- a/mscore/mscore.cpp
+++ b/mscore/mscore.cpp
@@ -919,15 +919,15 @@
       if (debugMode)
             printf("open handbook for language <%s>\n", qPrintable(lang));

-      QFileInfo mscoreHelp(mscoreGlobalShare + QString("man/MuseScore-") + lang + QString(".pdf"));
+      QFileInfo mscoreHelp(("/usr/share/doc/musescore/MuseScore-") + lang + QString(".pdf"));
       if (!mscoreHelp.isReadable()) {
             if (debugMode) {
                   printf("cannot open doc <%s>\n", qPrintable(mscoreHelp.filePath()));
                   }
             lang = lang.left(2);
-            mscoreHelp.setFile(mscoreGlobalShare + QString("man/MuseScore-") + lang + QString(".pdf"));
+            mscoreHelp.setFile(("/usr/share/doc/musescore/MuseScore-") + lang + QString(".pdf"));
             if(!mscoreHelp.isReadable()){
-                mscoreHelp.setFile(mscoreGlobalShare + QString("man/MuseScore-en.pdf"));
+                mscoreHelp.setFile(("/usr/share/doc/musescore/MuseScore-en.pdf"));
                 if (!mscoreHelp.isReadable()) {
                       QString info(tr("MuseScore handbook not found at: \n"));
                       info += mscoreHelp.filePath();
