--- mysql-workbench-gpl-5.2.30-src/library/sql-parser/source/my_global.h.mysql55	2010-11-18 23:04:05.000000000 +0100
+++ mysql-workbench-gpl-5.2.30-src/library/sql-parser/source/my_global.h	2010-12-28 18:11:29.000000000 +0100
@@ -24,6 +24,20 @@
 #define HAVE_EXTERNAL_CLIENT
 #endif
 
+/* 
+  MYSQL_PLUGIN_IMPORT macro is used to export mysqld data
+  (i.e variables) for usage in storage engine loadable plugins.
+  Outside of Windows, it is dummy.
+*/
+#ifndef MYSQL_PLUGIN_IMPORT
+#if (defined(_WIN32) && defined(MYSQL_DYNAMIC_PLUGIN))
+#define MYSQL_PLUGIN_IMPORT __declspec(dllimport)
+#else
+#define MYSQL_PLUGIN_IMPORT
+#endif
+#endif
+
+
 #if defined( __EMX__) && !defined( MYSQL_SERVER)
 /* moved here to use below VOID macro redefinition */
 #define INCL_BASE
@@ -1013,8 +1027,8 @@
 typedef char		byte;	/* Smallest addressable unit */
 #endif
 typedef char		my_bool; /* Small bool */
-#if !defined(bool) && !defined(bool_defined) && (!defined(HAVE_BOOL) || !defined(__cplusplus))
-typedef char		bool;	/* Ordinary boolean values 0 1 */
+#if !defined(__cplusplus) && !defined(bool)
+#define bool In_C_you_should_use_my_bool_instead()
 #endif
 	/* Macros for converting *constants* to the right type */
 #define INT8(v)		(int8) (v)
