diff --git a/plugins/kwin-xcb/lib/kwinutils.cpp b/plugins/kwin-xcb/lib/kwinutils.cpp
index 7a0bf5c..eca85e2 100644
--- a/plugins/kwin-xcb/lib/kwinutils.cpp
+++ b/plugins/kwin-xcb/lib/kwinutils.cpp
@@ -79,16 +79,6 @@ public:
 public Q_SLOTS:
     void slotWindowMove();
     void slotWindowMaximize();
-
-    // kwin 5.8.6
-    void slotWindowQuickTileLeft();
-    void slotWindowQuickTileRight();
-    void slotWindowQuickTileTop();
-    void slotWindowQuickTileBottom();
-    void slotWindowQuickTileTopLeft();
-    void slotWindowQuickTileTopRight();
-    void slotWindowQuickTileBottomLeft();
-    void slotWindowQuickTileBottomRight();
 };
 class Scripting : public QObject {
 public:
@@ -388,43 +378,6 @@ void KWinUtils::QuickTileWindow(uint side)
     if (ws) {
         if (interface->quickTileWindow) {
             interface->quickTileWindow(ws, (KWin::Workspace::QuickTileFlag)side);
-        } else { // fallback for kwin 5.8.6
-            switch ((KWin::Workspace::QuickTileFlag)side) {
-            case KWin::Workspace::QuickTileFlag::Left:
-                ws->slotWindowQuickTileLeft();
-                break;
-            case KWin::Workspace::QuickTileFlag::Right:
-                ws->slotWindowQuickTileRight();;
-                break;
-            case KWin::Workspace::QuickTileFlag::Top:
-                ws->slotWindowQuickTileTop();
-                break;
-            case KWin::Workspace::QuickTileFlag::Bottom:
-                ws->slotWindowQuickTileBottom();
-                break;
-            case KWin::Workspace::QuickTileFlag::Horizontal:
-                ws->slotWindowQuickTileLeft();
-                ws->slotWindowQuickTileRight();
-                break;
-            case KWin::Workspace::QuickTileFlag::Vertical:
-                ws->slotWindowQuickTileTop();
-                ws->slotWindowQuickTileBottom();
-                break;
-            case KWin::Workspace::QuickTileFlag::TopLeft:
-                ws->slotWindowQuickTileTopLeft();
-                break;
-            case KWin::Workspace::QuickTileFlag::TopRight:
-                ws->slotWindowQuickTileTopRight();
-                break;
-            case KWin::Workspace::QuickTileFlag::BottomLeft:
-                ws->slotWindowQuickTileBottomLeft();
-                break;
-            case KWin::Workspace::QuickTileFlag::BottomRight:
-                ws->slotWindowQuickTileBottomRight();
-                break;
-            default:
-                break;
-            }
         }
     }
 }
