diff --git a/network/nm_setting_beans_extend.go b/network/nm_setting_beans_extend.go
index f969fc8a..b8bfce3c 100644
--- a/network/nm_setting_beans_extend.go
+++ b/network/nm_setting_beans_extend.go
@@ -140,6 +140,46 @@ func checkSettingOlpcMeshValues(data connectionData) (err sectionErrors) {
 	return
 }
 
+func getSettingOvsBridgeAvailableKeys(data connectionData) (keys []string) {
+	return
+}
+func getSettingOvsBridgeAvailableValues(data connectionData, key string) (values []kvalue) {
+	return
+}
+func checkSettingOvsBridgeValues(data connectionData) (err sectionErrors) {
+	return
+}
+
+func getSettingOvsInterfaceAvailableKeys(data connectionData) (keys []string) {
+	return
+}
+func getSettingOvsInterfaceAvailableValues(data connectionData, key string) (values []kvalue) {
+	return
+}
+func checkSettingOvsInterfaceValues(data connectionData) (err sectionErrors) {
+	return
+}
+
+func getSettingOvsPatchAvailableKeys(data connectionData) (keys []string) {
+	return
+}
+func getSettingOvsPatchAvailableValues(data connectionData, key string) (values []kvalue) {
+	return
+}
+func checkSettingOvsPatchValues(data connectionData) (err sectionErrors) {
+	return
+}
+
+func getSettingOvsPortAvailableKeys(data connectionData) (keys []string) {
+	return
+}
+func getSettingOvsPortAvailableValues(data connectionData, key string) (values []kvalue) {
+	return
+}
+func checkSettingOvsPortValues(data connectionData) (err sectionErrors) {
+	return
+}
+
 func getSettingProxyAvailableKeys(data connectionData) (keys []string) {
 	return
 }
