diff --git a/external/randomx/CMakeLists.txt b/external/randomx/CMakeLists.txt
index 40f0354..3bacafa 100644
--- a/external/randomx/CMakeLists.txt
+++ b/external/randomx/CMakeLists.txt
@@ -102,20 +102,6 @@ if (ARCH_ID STREQUAL "x86_64" OR ARCH_ID STREQUAL "x86-64" OR ARCH_ID STREQUAL "
   set_property(SOURCE src/jit_compiler_x86_static.S PROPERTY LANGUAGE C)
   set_property(SOURCE src/jit_compiler_x86_static.S PROPERTY XCODE_EXPLICIT_FILE_TYPE sourcecode.asm)
 
-  if(ARCH STREQUAL "native")
-    add_flag("-march=native")
-  else()
-    # default build has hardware AES enabled (software AES can be selected at runtime)
-    add_flag("-maes")
-    check_c_compiler_flag(-mssse3 HAVE_SSSE3)
-    if(HAVE_SSSE3)
-      set_source_files_properties(src/argon2_ssse3.c COMPILE_FLAGS -mssse3)
-    endif()
-    check_c_compiler_flag(-mavx2 HAVE_AVX2)
-    if(HAVE_AVX2)
-      set_source_files_properties(src/argon2_avx2.c COMPILE_FLAGS -mavx2)
-    endif()
-  endif()
 endif()
 
 # PowerPC
