diff --git a/help.ml b/help.ml
index 165d46e..b86f0a0 100644
--- a/help.ml
+++ b/help.ml
@@ -1,6 +1,6 @@
 let keys =
 {|-----Quitting-----
-escape/q                - quit
+q                       - quit
 Q                       - quit without saving the configuration or changes
 W                       - save changes
 
diff --git a/main.ml b/main.ml
index cd45361..5d66930 100644
--- a/main.ml
+++ b/main.ml
@@ -3365,7 +3365,7 @@ let viewkeyboard key mask =
      )
      else impmsg "keyboard link navigation does not work under rotation"
 
-  | Escape | Ascii 'q' ->
+  | Ascii 'q' ->
      begin match state.mstate with
      | Mzoomrect _ ->
         resetmstate ();
@@ -3782,7 +3782,7 @@ let viewkeyboard key mask =
      postRedisplay "|";
      state.mode <- Textentry (te, onleave);
 
-  | (Ascii _|Fn _|Enter|Left|Right|Code _|Ctrl _) ->
+  | (Escape|Ascii _|Fn _|Enter|Left|Right|Code _|Ctrl _) ->
      vlog "huh? %s" (Wsi.keyname key)
 ;;
 
