--- ntl-11.4.1/src/gf2x_version_1_2_or_later_required.cpp.orig	2019-12-10 15:33:19.317814419 +0000
+++ ntl-11.4.1/src/gf2x_version_1_2_or_later_required.cpp	2019-12-10 15:33:46.244836785 +0000
@@ -7,6 +7,8 @@
 #if (defined(NTL_GF2X_LIB) && defined(NTL_THREADS))
 // we require v1.2 or later
 
+#include <gf2x.h>
+
 #ifndef GF2X_VERSION_MAJOR
 // versions after v1.2 should define GF2X_VERSION_MAJOR
 
