diff -ruN a/Makefile b/Makefile
--- a/Makefile	2013-04-09 03:53:29.000000000 +0200
+++ b/Makefile	2020-05-07 23:35:37.658428769 +0200
@@ -16,7 +16,7 @@
 
 CXX := g++
 INCLUDES := -I.
-COPTS  := -Wall -Wextra -O3
+COPTS  := -Wall -Wextra -O3 -fPIC
 CDEBUG := -Wall -Wextra -g 
 CXXFLAGS := $(COPTS) $(INCLUDES) -D$(ENDIANESS)
 
@@ -111,23 +111,18 @@
 	$(CXX) -dynamiclib -Wl,-install_name,$(LIBSONAME) -o $(LIBFILENAME) $(LIBOBJECTS) -lc
 else
 	#GNU/Linux case
-	$(CXX) -shared -Wl,-soname,$(LIBSONAME) -o $(LIBFILENAME) $(LIBOBJECTS) -lc
+	$(CXX) -shared -Wl,-soname,$(LIBSONAME) -o $(LIBFILENAME) $(LIBOBJECTS) -lc $(LDFLAGS)
 endif
 
 lib: $(LIBFILENAME)
 
 #Installs the library on a system global location
 install: $(LIBFILENAME)
-	@$(INSTALL) -m 755 $(LIBFILENAME) $(PREFIX)/lib/$(LIBFILENAME)
-	@ln -s -f $(PREFIX)/lib/$(LIBFILENAME) $(PREFIX)/lib/$(LIBSONAME) 
-	@mkdir  -p $(PREFIX)/include/oscpack/ip $(PREFIX)/include/oscpack/osc
-	@$(INSTALL) -m 644 ip/*.h $(PREFIX)/include/oscpack/ip
-	@$(INSTALL) -m 644 osc/*.h $(PREFIX)/include/oscpack/osc
+	@$(INSTALL) -vDm 755 $(LIBFILENAME) $(DESTDIR)/$(PREFIX)/lib/$(LIBFILENAME)
+	@ln -sfv $(LIBFILENAME) $(DESTDIR)/$(PREFIX)/lib/$(LIBSONAME)
+	@$(INSTALL) -vDm 644 ip/*.h -t $(DESTDIR)/$(PREFIX)/include/oscpack/ip
+	@$(INSTALL) -vDm 644 osc/*.h -t $(DESTDIR)/$(PREFIX)/include/oscpack/osc
 	@echo "SUCCESS! oscpack has been installed in $(PREFIX)/lib and $(PREFIX)/include/ospack/"
-ifneq ($(UNAME), Darwin)
-	@echo "now doing ldconfig..."
-	@ldconfig
-endif
 
 #Installs the include/lib structure locally
 install-local: $(LIBFILENAME)
