diff -Naur musescore-svn.orig//mscore/mscore/mscore.cpp musescore-svn.new//mscore/mscore/mscore.cpp
--- musescore-svn.orig//mscore/mscore/mscore.cpp	2010-06-16 23:10:27.000000000 +0200
+++ musescore-svn.new//mscore/mscore/mscore.cpp	2010-06-16 23:13:59.000000000 +0200
@@ -897,15 +897,15 @@
       if (debugMode)
             printf("open handbook for language <%s>\n", qPrintable(lang));
 
-      QFileInfo mscoreHelp(mscoreGlobalShare + QString("man/MuseScore-") + lang + QString(".pdf"));
+      QFileInfo mscoreHelp(mscoreGlobalShare + QString("../doc/musescore/MuseScore-") + lang + QString(".pdf"));
       if (!mscoreHelp.isReadable()) {
             if (debugMode) {
                   printf("cannot open doc <%s>\n", qPrintable(mscoreHelp.filePath()));
                   }
             lang = lang.left(2);
-            mscoreHelp.setFile(mscoreGlobalShare + QString("man/MuseScore-") + lang + QString(".pdf"));
+            mscoreHelp.setFile(mscoreGlobalShare + QString("../doc/mscore/MuseScore-") + lang + QString(".pdf"));
             if(!mscoreHelp.isReadable()){
-                mscoreHelp.setFile(mscoreGlobalShare + QString("man/MuseScore-en.pdf"));
+                mscoreHelp.setFile(mscoreGlobalShare + QString("../doc/mscore/MuseScore-en.pdf"));
                 if (!mscoreHelp.isReadable()) {
                       QString info(tr("MuseScore handbook not found at: \n"));
                       info += mscoreHelp.filePath();
