From 421c3705107fab022631b93a2e70fc7c0ffc86ea Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Tue, 23 Jul 2019 18:02:31 +0200
Subject: [PATCH] Fix #295 ensure IM don't use multiple threads

---
 imagick.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/imagick.c b/imagick.c
index 8e58afe..e5a688a 100644
--- a/imagick.c
+++ b/imagick.c
@@ -3844,6 +3844,8 @@ PHP_RINIT_FUNCTION(imagick)
 	(void)module_number;
 
 	IMAGICK_G(progress_callback) = NULL;
+	MagickSetResourceLimit(ThreadResource, 1);
+
 	return SUCCESS;
 }
 
