diff --git a/php7/memcache.c b/php7/memcache.c
index c7d4e0a..1818db4 100644
--- a/php7/memcache.c
+++ b/php7/memcache.c
@@ -721,7 +721,7 @@ mmc_t *mmc_find_persistent(const char *host, int host_len, unsigned short port,
 		mmc = mmc_server_new(host, host_len, port, udp_port, 1, timeout, retry_interval);
 		le->type = le_memcache_server;
 		le->ptr  = mmc;
-		GC_REFCOUNT(le) = 1;
+		GC_SET_REFCOUNT(le, 1);
 
 		/* register new persistent connection */
 		if (zend_hash_str_update_mem(&EG(persistent_list), key, key_len, le, sizeof(*le)) == NULL) {
@@ -779,7 +779,7 @@ static mmc_t *php_mmc_pool_addserver(
 		pool->failure_callback = &php_mmc_failure_callback;
 		list_res = zend_register_resource(pool, le_memcache_pool);
 		add_property_resource(mmc_object, "connection", list_res);
-		GC_REFCOUNT(list_res)++;
+		GC_ADDREF(list_res);
 	}
 	else {
 		pool = zend_fetch_resource_ex(connection, "connection", le_memcache_pool);
@@ -863,7 +863,7 @@ static void php_mmc_connect(INTERNAL_FUNCTION_PARAMETERS, zend_bool persistent)
 		mmc_object = return_value;
 		object_init_ex(mmc_object, memcache_ce);
 		add_property_resource(mmc_object, "connection", list_res);
-		GC_REFCOUNT(list_res)++;
+		GC_ADDREF(list_res);
 	} else {
 		RETVAL_TRUE;
 	}
