diff --git i/vendor/github.com/containers/common/pkg/config/default.go w/vendor/github.com/containers/common/pkg/config/default.go
index 8979a406b..4736e0188 100644
--- i/vendor/github.com/containers/common/pkg/config/default.go
+++ w/vendor/github.com/containers/common/pkg/config/default.go
@@ -47,7 +47,7 @@ const (
 
 var (
 	// DefaultInitPath is the default path to the container-init binary
-	DefaultInitPath = "/usr/libexec/podman/catatonit"
+	DefaultInitPath = "/usr/lib/podman/catatonit"
 	// DefaultInfraImage to use for infra container
 	DefaultInfraImage = ""
 	// DefaultRootlessSHMLockPath is the default path for rootless SHM locks
