diff --git a/data/gen-regexps b/data/gen-regexps
index 7a8bfc0..f563794 100755
--- a/data/gen-regexps
+++ b/data/gen-regexps
@@ -49,8 +49,8 @@ def rx_opt (rx):
     return rx_repeat (rx, 0, 1)
 
 def rx_output (rx, name):
-    print "s/@%s@/%s/g\n" % (name,
-                             rx.replace('\\', '\\\\').replace('/', '\\/').replace('"', '\&quot;').replace('&', '\&amp;'))
+    print("s/@%s@/%s/g\n" % (name,
+                             rx.replace('\\', '\\\\').replace('/', '\\/').replace('"', '\&quot;').replace('&', '\&amp;')))
 
 ## Phone numbers
 
diff --git a/data/regextest2.py b/data/regextest2.py
index 35ad8b3..43b4a93 100644
--- a/data/regextest2.py
+++ b/data/regextest2.py
@@ -32,7 +32,7 @@ class LocalRegexTest (unittest.TestCase):
         parts = self.__spec.split (self.__delimiter)
         text = "".join (parts)
         expected = [ parts[i] for i in range (1, len (parts), 2) ]
-        result = map (lambda m: m.group(0), re.finditer (self.__regex, text))
+        result = list (map (lambda m: m.group(0), re.finditer (self.__regex, text)))
         self.assertEquals (result, expected)
 
 
@@ -96,7 +96,7 @@ def static_counter_gen ():
     while True:
         k += 1
         yield k
-static_counter = static_counter_gen ().next
+static_counter = lambda: next(static_counter_gen ())
 
 
 rxt_regex = None
