# https://github.com/fonttools/unicodedata2/commit/44409e0c0cd5

diff -upr unicodedata2-13.0.0-2.orig/unicodedata2/py3/unicodedata.c unicodedata2-13.0.0-2/unicodedata2/py3/unicodedata.c
--- unicodedata2-13.0.0-2.orig/unicodedata2/py3/unicodedata.c	2020-03-23 13:38:17.000000000 +0200
+++ unicodedata2-13.0.0-2/unicodedata2/py3/unicodedata.c	2021-12-03 02:54:14.709059346 +0200
@@ -16,7 +16,12 @@
 #define PY_SSIZE_T_CLEAN
 
 #include "Python.h"
+#if PY_MINOR_VERSION < 10
 #include "ucnhash.h"
+#else
+#define Py_BUILD_CORE
+#include "internal/pycore_ucnhash.h"
+#endif
 #include "structmember.h"
 #include "unicodectype.h"
 
