diff -Naur deja-dup-32.0.orig/libdeja/tools/duplicity/DuplicityPlugin.vala deja-dup-32.0/libdeja/tools/duplicity/DuplicityPlugin.vala
--- deja-dup-32.0.orig/libdeja/tools/duplicity/DuplicityPlugin.vala	2014-01-24 17:07:35.000000000 +0100
+++ deja-dup-32.0/libdeja/tools/duplicity/DuplicityPlugin.vala	2014-10-22 23:47:37.406082239 +0200
@@ -53,7 +53,6 @@
   public override DejaDup.ToolJob create_job () throws Error
   {
     if (!has_been_setup) {
-      do_initial_setup();
       has_been_setup = true;
     }
     return new DuplicityJob();
