diff --git a/main.go b/main.go
index 9f2b76ffd..03ff858f3 100644
--- a/main.go
+++ b/main.go
@@ -9,7 +9,6 @@ import (
 
 	"gitlab.com/gitlab-org/gitlab-runner/common"
 	cli_helpers "gitlab.com/gitlab-org/gitlab-runner/helpers/cli"
-	issue_28732_patch "gitlab.com/gitlab-org/gitlab-runner/helpers/patches/issue_28732"
 	"gitlab.com/gitlab-org/gitlab-runner/log"
 
 	_ "gitlab.com/gitlab-org/gitlab-runner/cache/azure"
@@ -40,8 +39,6 @@ func main() {
 		}
 	}()
 
-	issue_28732_patch.AssertFixPresent()
-
 	app := cli.NewApp()
 	app.Name = filepath.Base(os.Args[0])
 	app.Usage = "a GitLab Runner"
