--- a/Makefile
+++ b/Makefile
@@ -144,7 +144,7 @@ src/version.h: src/version
 	$(Q)echo '#define VERSION_H' >> ./src/version.h
 	$(Q)echo '#include "version_number.h"' >> ./src/version.h
 	$(Q)echo '#define REVISION "' | tr -d '\n'  >> ./src/version.h
-	$(Q)date +"%Y%m%d" | tr -d '\n' >> ./src/version.h
+	$(Q)git show --no-patch --format=%cd --date=format:%Y%m%d | tr -d '\n' >> ./src/version.h
 	$(Q)echo '"' >> ./src/version.h
 	$(Q)echo '#define VERSION_STRING "klystrack-plus " VERSION " " REVISION' >> ./src/version.h
 	$(Q)echo '#endif' >> ./src/version.h
