diff -aur ssr-0.5.0/configure.ac ssr-0.5.0.new/configure.ac
--- ssr-0.5.0/configure.ac	2018-11-16 17:15:54.000000000 +0100
+++ ssr-0.5.0.new/configure.ac	2021-05-08 23:43:42.856785555 +0200
@@ -16,7 +16,7 @@
 dnl arguments: full package name, version, email address for bug reports,
 dnl tarball name, homepage
 AC_INIT([SSR (SoundScape Renderer)],
-        [m4_esyscmd_s([git describe 2> /dev/null || date +$USER%Y%m%d%H%M])],
+        [m4_esyscmd_s([git describe 2> /dev/null || echo "builduser"])],
         [ssr@spatialaudio.net],
         [ssr],
         [http://spatialaudio.net/ssr/])
