--- a/src/sage/all.py
+++ b/src/sage/all.py
@@ -324,6 +324,9 @@ warnings.filterwarnings('ignore', category=DeprecationWarning,
 # However, be sure to keep OUR deprecation warnings
 warnings.filterwarnings('default', category=DeprecationWarning,
     message=r'[\s\S]*See https\?://trac\.sagemath\.org/[0-9]* for details.')
+# jedi 0.16 deprecation warnings
+warnings.filterwarnings('ignore', category=DeprecationWarning,
+    message='.*Use get_signatures().*')
 # Python 3.8 deprecation warnings
 warnings.filterwarnings('ignore', category=DeprecationWarning,
     message='.*PY_SSIZE_T_CLEAN.*')
diff --git a/src/sage/__init__.py b/src/sage/__init__.py
index c5ebb92..e2ae5b8 100644
--- a/src/sage/__init__.py
+++ b/src/sage/__init__.py
@@ -56,13 +56,3 @@ def isfunction(obj):
 
 import inspect
 inspect.isfunction = isfunction
-
-
-# Monkey-patch ExtensionFileLoader to allow IPython to find the sources
-# of Cython files. See https://trac.sagemath.org/ticket/24681
-try:
-    from importlib.machinery import ExtensionFileLoader
-except ImportError:
-    pass  # Python 2
-else:
-    del ExtensionFileLoader.get_source
