--- a/src/Darcs/Util/Hash.hs	2016-06-09 11:45:13.303203462 +0800
+++ b/src/Darcs/Util/Hash.hs	2016-06-09 11:45:29.330047632 +0800
@@ -27,8 +27,8 @@
 base16 :: BS.ByteString -> BS.ByteString
 debase16 :: BS.ByteString -> Maybe BS.ByteString
 
-base16 = BS8.map toLower . B16.b16_enc
-debase16 bs = case B16.b16_dec $ BS8.map toUpper bs of
+base16 = BS8.map toLower . B16.b16Enc
+debase16 bs = case B16.b16Dec $ BS8.map toUpper bs of
                 Right (s, _) -> Just s
                 Left _ -> Nothing
 
