diff --git a/scribus/plugins/import/pdf/slaoutput.cpp b/scribus/plugins/import/pdf/slaoutput.cpp
index fe9794a65..2012d98d3 100644
--- a/scribus/plugins/import/pdf/slaoutput.cpp
+++ b/scribus/plugins/import/pdf/slaoutput.cpp
@@ -430,7 +430,7 @@ bool SlaOutputDev::handleLinkAnnot(Annot* annota, double xCoor, double yCoor, do
 				if (dst->isPageRef())
 				{
 					Ref dstr = dst->getPageRef();
-					pagNum = pdfDoc->findPage(dstr.num, dstr.gen);
+					pagNum = pdfDoc->findPage(dstr);
 				}
 				else
 					pagNum = dst->getPageNum();
@@ -452,7 +452,7 @@ bool SlaOutputDev::handleLinkAnnot(Annot* annota, double xCoor, double yCoor, do
 						if (dstn->isPageRef())
 						{
 							Ref dstr = dstn->getPageRef();
-							pagNum = pdfDoc->findPage(dstr.num, dstr.gen);
+							pagNum = pdfDoc->findPage(dstr);
 						}
 						else
 							pagNum = dstn->getPageNum();
@@ -932,7 +932,7 @@ void SlaOutputDev::handleActions(PageItem* ite, AnnotWidget *ano)
 					if (dst->isPageRef())
 					{
 						Ref dstr = dst->getPageRef();
-						pagNum = pdfDoc->findPage(dstr.num, dstr.gen);
+						pagNum = pdfDoc->findPage(dstr);
 					}
 					else
 						pagNum = dst->getPageNum();
@@ -956,7 +956,7 @@ void SlaOutputDev::handleActions(PageItem* ite, AnnotWidget *ano)
 							if (dstn->isPageRef())
 							{
 								Ref dstr = dstn->getPageRef();
-								pagNum = pdfDoc->findPage(dstr.num, dstr.gen);
+								pagNum = pdfDoc->findPage(dstr);
 							}
 							else
 								pagNum = dstn->getPageNum();
