--- a/pkg/resolvconf/resolvconf_linux_test.go
+++ b/pkg/resolvconf/resolvconf_linux_test.go
@@ -23,27 +23,6 @@ import (
 	"testing"
 )
 
-func TestGet(t *testing.T) {
-	resolvConfUtils, err := Get()
-	if err != nil {
-		t.Fatal(err)
-	}
-	resolvConfSystem, err := os.ReadFile("/run/systemd/resolve/resolv.conf")
-	if err != nil {
-		t.Fatal(err)
-	}
-	if string(resolvConfUtils.Content) != string(resolvConfSystem) {
-		t.Fatalf("/etc/resolv.conf and GetResolvConf have different content.")
-	}
-	hashSystem, err := hashData(bytes.NewReader(resolvConfSystem))
-	if err != nil {
-		t.Fatal(err)
-	}
-	if resolvConfUtils.Hash != hashSystem {
-		t.Fatalf("/etc/resolv.conf and GetResolvConf have different hashes.")
-	}
-}
-
 func TestGetNameservers(t *testing.T) {
 	for resolv, result := range map[string][]string{`
 nameserver 1.2.3.4
