diff -ruN a/app/server/ruby/bin/task-clear.rb e/app/server/ruby/bin/task-clear.rb
--- a/app/server/ruby/bin/task-clear.rb	2020-02-25 01:23:45.000000000 +0100
+++ e/app/server/ruby/bin/task-clear.rb	2020-03-10 18:46:53.990877950 +0100
@@ -77,7 +77,7 @@
   end
 
   begin
-    info = Sys::ProcTable.ps(pid)
+    info = Sys::ProcTable.ps(pid: pid)
     raise unless info
   rescue
     log_process_info "  -- unable to get ProcTable info for: #{pid}"
diff -ruN a/app/server/ruby/bin/task-register.rb e/app/server/ruby/bin/task-register.rb
--- a/app/server/ruby/bin/task-register.rb	2020-02-25 01:23:45.000000000 +0100
+++ e/app/server/ruby/bin/task-register.rb	2020-03-10 18:48:39.273368749 +0100
@@ -34,12 +34,14 @@
 f = nil
 
 begin
-  if s = Sys::ProcTable.ps(pid)
+  if s = Sys::ProcTable.ps(pid: pid)
     f = File.open(pid_path, 'w')
     f.puts s.cmdline
     log_process_info "Started [#{pid}] [-] #{s.cmdline} [-] #{pid_path}"
   end
 rescue Exception => e
+  log_process_info "ERROR: Unable to write information for PID #{pid} to path #{pid_path}!"
+  log_process_info "#{e}"
 end
 
 f.close if f
