From addd1eb2995aed10fd9bcb5c97bbb00872728a85 Mon Sep 17 00:00:00 2001
From: Elperson <40611324+Elpersonn@users.noreply.github.com>
Date: Wed, 20 Jul 2022 17:02:07 +0200
Subject: [PATCH] Fix compile error (#574)

---
 rts/Lua/LuaObjectRendering.h | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/rts/Lua/LuaObjectRendering.h b/rts/Lua/LuaObjectRendering.h
index 13ecc7562bd..7e63ab5c4cb 100644
--- a/rts/Lua/LuaObjectRendering.h
+++ b/rts/Lua/LuaObjectRendering.h
@@ -11,13 +11,10 @@ struct lua_State;
 template<LuaObjType T> class LuaObjectRendering;
 
 class LuaObjectRenderingImpl {
-#if __GNUC__ == 11
 public:
-#else
-private:
 	friend class LuaObjectRendering<LUAOBJ_UNIT>;
 	friend class LuaObjectRendering<LUAOBJ_FEATURE>;
-#endif
+	
 	static void CreateMatRefMetatable(lua_State* L);
 	static void PushFunction(lua_State* L, int (*fnPntr)(lua_State*), const char* fnName);
 
