diff --git a/nginx/config b/nginx/config
index fb7fb92..474aaf1 100644
--- a/nginx/config
+++ b/nginx/config
@@ -29,7 +29,7 @@ if [ $STREAM != NO ]; then
     ngx_module_incs="$ngx_addon_dir/../src $ngx_addon_dir/../build"
     ngx_module_deps="$ngx_addon_dir/../build/libnjs.a $NJS_DEPS"
     ngx_module_srcs="$ngx_addon_dir/ngx_stream_js_module.c $NJS_SRCS"
-    ngx_module_libs="PCRE OPENSSL $ngx_addon_dir/../build/libnjs.a -lm"
+    ngx_module_libs="PCRE OPENSSL LIBXSLT $ngx_addon_dir/../build/libnjs.a -lm"
 
     . auto/module
 fi
