--- tw.color.c.orig	2008-10-17 16:57:33.000000000 -0300
+++ tw.color.c	2010-10-18 23:34:36.764372377 -0300
@@ -86,6 +86,10 @@
     VAR(NOS, "ow", ""),		/* Other writable dir (o+w) but not sticky */
     VAR(NOS, "st", ""),		/* Sticky dir (+t) but not other writable */
     VAR(NOS, "rs", "0"),	/* Reset to normal color */
+    VAR(NOS, "hl", ""),		/* Regular file with more than one link */
+    VAR(NOS, "ca", ""),		/* File with capability */
+    VAR(NOS, "mh", ""),		/* MULTIHARDLINK */
+    VAR(NOS, "cl", ""),		/* CLRTOEOL */
 };
 
 enum FileType {
