--- a/src/ft2_about.c
+++ b/src/ft2_about.c
@@ -194,7 +194,7 @@ void showAboutScreen(void) // called once when About screen is opened
 
 	showPushButton(PB_EXIT_ABOUT);
 
-	sprintf(customText3, "v%s (%s)", PROG_VER_STR, __DATE__);
+	sprintf(customText3, "v%s (%s)", PROG_VER_STR, PROG_VER_DATE);
 	customText1X = (SCREEN_W - textWidth(customText1)) >> 1;
 	customText2X = (SCREEN_W-8) - textWidth(customText2);
 	customText3X = (SCREEN_W-8) - textWidth(customText3);
--- a/src/ft2_header.h
+++ b/src/ft2_header.h
@@ -13,6 +13,7 @@
 #include "ft2_replayer.h"
 
 #define PROG_VER_STR "1.63"
+#define PROG_VER_DATE "@DATE@"
 
 // do NOT change these! It will only mess things up...
 
