--- Cinnamon-2.2.13.orig/files/usr/share/cinnamon/applets/calendar@cinnamon.org/applet.js	2014-05-25 15:10:49.000000000 -0500
+++ Cinnamon-2.2.13/files/usr/share/cinnamon/applets/calendar@cinnamon.org/applet.js	2014-05-30 17:05:16.099584860 -0500
@@ -6,7 +6,6 @@
 const St = imports.gi.St;
 const Util = imports.misc.util;
 const PopupMenu = imports.ui.popupMenu;
-const UPowerGlib = imports.gi.UPowerGlib;
 const Settings = imports.ui.settings;
 const AppletDir = imports.ui.appletManager.applets['calendar@cinnamon.org'];
 const Calendar = AppletDir.calendar;
@@ -92,10 +91,6 @@
             this.settings.bindProperty(Settings.BindingDirection.IN, "use-custom-format", "use_custom_format", this.on_settings_changed, null);
             this.settings.bindProperty(Settings.BindingDirection.IN, "custom-format", "custom_format", this.on_settings_changed, null);        
 
-            // https://bugzilla.gnome.org/show_bug.cgi?id=655129
-            this._upClient = new UPowerGlib.Client();
-            this._upClient.connect('notify-resume', this._updateClockAndDate);
-
             // Start the clock
             this.on_settings_changed();
             this._updateClockAndDatePeriodic();
