diff -aur chromium-bsu-0.9.16/src/MainSDL_Event.cpp chromium-bsu-0.9.16.new/src/MainSDL_Event.cpp
--- chromium-bsu-0.9.16/src/MainSDL_Event.cpp	2016-06-30 17:26:58.000000000 +0200
+++ chromium-bsu-0.9.16.new/src/MainSDL_Event.cpp	2016-08-14 21:37:55.746045707 +0200
@@ -515,14 +515,14 @@
 		#define KP8 SDLK_KP8
 		#define KP9 SDLK_KP9
 #endif
-		if( keystate[LEFT]  || keystate[KP4] ) key_speed_x -= 2.0 + abs(key_speed_x)*0.4;
-		if( keystate[RIGHT] || keystate[KP6] ) key_speed_x += 2.0 + abs(key_speed_x)*0.4;
-		if( keystate[UP]    || keystate[KP8] ) key_speed_y -= 2.0 + abs(key_speed_y)*0.4;
-		if( keystate[DOWN]  || keystate[KP2] ) key_speed_y += 2.0 + abs(key_speed_y)*0.4;
-		if( keystate[KP7] ){ key_speed_x -= 2.0 + abs(key_speed_x)*0.4; key_speed_y -= 2.0 + abs(key_speed_y)*0.4; }
-		if( keystate[KP9] ){ key_speed_x += 2.0 + abs(key_speed_x)*0.4; key_speed_y -= 2.0 + abs(key_speed_y)*0.4; }
-		if( keystate[KP3] ){ key_speed_x += 2.0 + abs(key_speed_x)*0.4; key_speed_y += 2.0 + abs(key_speed_y)*0.4; }
-		if( keystate[KP1] ){ key_speed_x -= 2.0 + abs(key_speed_x)*0.4; key_speed_y += 2.0 + abs(key_speed_y)*0.4; }
+		if( keystate[LEFT]  || keystate[KP4] ) key_speed_x -= 2.0 + fabs(key_speed_x)*0.4;
+		if( keystate[RIGHT] || keystate[KP6] ) key_speed_x += 2.0 + fabs(key_speed_x)*0.4;
+		if( keystate[UP]    || keystate[KP8] ) key_speed_y -= 2.0 + fabs(key_speed_y)*0.4;
+		if( keystate[DOWN]  || keystate[KP2] ) key_speed_y += 2.0 + fabs(key_speed_y)*0.4;
+		if( keystate[KP7] ){ key_speed_x -= 2.0 + fabs(key_speed_x)*0.4; key_speed_y -= 2.0 + fabs(key_speed_y)*0.4; }
+		if( keystate[KP9] ){ key_speed_x += 2.0 + fabs(key_speed_x)*0.4; key_speed_y -= 2.0 + fabs(key_speed_y)*0.4; }
+		if( keystate[KP3] ){ key_speed_x += 2.0 + fabs(key_speed_x)*0.4; key_speed_y += 2.0 + fabs(key_speed_y)*0.4; }
+		if( keystate[KP1] ){ key_speed_x -= 2.0 + fabs(key_speed_x)*0.4; key_speed_y += 2.0 + fabs(key_speed_y)*0.4; }
 		//float s = (1.0-game->speedAdj)+(game->speedAdj*0.7);
 		float s = 0.7;
 		key_speed_x *= s;
