diff -aur aaa/configure.ac bbb/configure.ac
--- aaa/configure.ac	2022-10-27 15:25:10.000000000 +0200
+++ bbb/configure.ac	2022-10-27 21:33:52.976284207 +0200
@@ -178,13 +178,6 @@
         equivalent variable and wxWidgets version is $MIN_WX_VERSION or above.
     ])
   fi
-  if test "${WX_VERSION_MAJOR}.${WX_VERSION_MINOR}" = "3.1"; then
-    AC_MSG_ERROR([You must use wxWidgets 3.0.x, development versions of wxWidgets are not supported.])
-  elif test "${WX_VERSION_MAJOR}.${WX_VERSION_MINOR}" != "3.0"; then
-    if ! echo "$WX_CPPFLAGS" | grep __WXMAC__ > /dev/null; then
-      AC_MSG_ERROR([You must use wxWidgets 3.0.x, wxWidgets 3.2 or higher is not yet supported.])
-    fi
-  fi
 
   if test "${WX_VERSION_MAJOR}.${WX_VERSION_MINOR}" = "3.0"; then
     MAC_REQUIRES_AQUA="YES"
