diff -ruN zita-at1-0.6.0/source/Makefile zita-at1-0.6.0-b/source/Makefile
--- zita-at1-0.6.0/source/Makefile	2017-08-09 23:04:06.000000000 +0200
+++ zita-at1-0.6.0-b/source/Makefile	2018-07-18 23:12:48.093318136 +0200
@@ -34,7 +34,7 @@
 
 ZITA-AT1_O = zita-at1.o styles.o jclient.o mainwin.o png2img.o guiclass.o \
              button.o rotary.o tmeter.o retuner.o
-zita-at1:	CPPFLAGS += -I/usr/X11R6/include `freetype-config --cflags`
+zita-at1:	CPPFLAGS += -I/usr/X11R6/include `pkgconf --cflags freetype2`
 zita-at1:	LDLIBS += -lcairo -lclxclient -lclthreads -lzita-resampler -lfftw3f -ljack -lpthread -lpng -lXft -lX11 -lrt
 zita-at1:	LDFLAGS += -L/usr/X11R6/lib
 zita-at1:	$(ZITA-AT1_O) 
