diff -aur synfig-0.63.02/src/modules/mod_libavcodec/trgt_av.cpp synfig-0.63.02.new/src/modules/mod_libavcodec/trgt_av.cpp
--- synfig-0.63.02/src/modules/mod_libavcodec/trgt_av.cpp	2011-09-29 10:51:41.000000000 +0200
+++ synfig-0.63.02.new/src/modules/mod_libavcodec/trgt_av.cpp	2011-10-31 16:39:03.000000000 +0100
@@ -441,7 +441,7 @@
 		if( context->coded_frame )
 			pkt.pts = context->coded_frame->pts;
 		if( context->coded_frame && context->coded_frame->key_frame)
-			pkt.flags |= PKT_FLAG_KEY;
+			pkt.flags |= AV_PKT_FLAG_KEY;
 
 		//kluge for raw picture format (they said they'd fix)
 		if (formatc->oformat->flags & AVFMT_RAWPICTURE)
@@ -463,7 +463,7 @@
 				if( context->coded_frame )
 					pkt.pts = context->coded_frame->pts;
 				if( context->coded_frame && context->coded_frame->key_frame)
-					pkt.flags |= PKT_FLAG_KEY;
+					pkt.flags |= AV_PKT_FLAG_KEY;
 
 				ret = av_write_frame(formatc, &pkt);
 
@@ -567,17 +567,17 @@
 		if (typestring)
 		{
 			//formatptr guess_format(type, filename, MIME type)
-			format = guess_format(typestring,NULL,NULL);
+			format = av_guess_format(typestring,NULL,NULL);
 		}
 		else
 		{
-			format = guess_format(NULL, filename, NULL);
+			format = av_guess_format(NULL, filename, NULL);
 		}
 
 		if(!format)
 		{
 			synfig::warning("Unable to Guess the output, defaulting to mpeg");
-			format = guess_format("mpeg", NULL, NULL);
+			format = av_guess_format("mpeg", NULL, NULL);
 		}
 
 		if(!format)
@@ -753,7 +753,7 @@
 
 		context = st->codec;
 		context->codec_id = (CodecID)codec_id;
-		context->codec_type = CODEC_TYPE_VIDEO;
+		context->codec_type = AVMEDIA_TYPE_VIDEO;
 
 		//PARAMETERS MUST BE PASSED IN SOMEHOW (ANOTHER FUNCTION PARAMETER???)
 
@@ -800,7 +800,7 @@
 
 		context = stream->codec;
 		context->codec_id = (CodecID)codec_id;
-		context->codec_type = CODEC_TYPE_AUDIO;
+		context->codec_type = AVMEDIA_TYPE_AUDIO;
 
 		/* put sample parameters */
 		context->bit_rate = 64000;
