From 30937f6e427153e85862afed41e69b470d52dc67 Mon Sep 17 00:00:00 2001
From: ntalex <alex@thuering.biz>
Date: Fri, 2 Aug 2019 22:49:37 +0200
Subject: changed to use configation file .dvdstyler/dvdstyler under unix


diff --git a/src/Config.cpp b/src/Config.cpp
index a1ccbf5..6ea06fc 100644
--- a/src/Config.cpp
+++ b/src/Config.cpp
@@ -10,9 +10,8 @@
 
 #include "Config.h"
 #include <wxVillaLib/utils.h>
-#ifdef __WXMSW__
 #include <wx/fileconf.h>
-#endif
+#include <wx/stdpaths.h>
 
 Config s_config;
 
@@ -27,8 +26,19 @@ void Config::Init() {
 		} else
 			wxConfig::Set(new wxFileConfig(wxT(""), wxT(""), fileName));
 	}
-#endif
+#elif defined(__WXMAC__)
 	cfg = wxConfig::Get();
+#else
+	// check if .dvdstyler exist and move it
+	wxString dataDir = wxStandardPaths::Get().GetUserLocalDataDir();
+	if (wxFileExists(dataDir)) {
+		wxRenameFile(dataDir, dataDir + ".tmp");
+		wxMkdir(dataDir);
+		wxRenameFile(dataDir + ".tmp", dataDir + wxFILE_SEP_PATH + "dvdstyler");
+	}
+	cfg = new wxFileConfig("", "", dataDir + wxFILE_SEP_PATH + "dvdstyler");
+	wxConfig::Set(cfg);
+#endif
 }
 
 bool Config::IsMainWinMaximized() {
