diff --git a/capplets/about-me/mate-about-me-dialog.ui b/capplets/about-me/mate-about-me-dialog.ui
index c57f267..fc22a13 100644
--- a/capplets/about-me/mate-about-me-dialog.ui
+++ b/capplets/about-me/mate-about-me-dialog.ui
@@ -6,6 +6,8 @@
     <property name="border_width">5</property>
     <property name="title" translatable="yes">About Me</property>
     <property name="type_hint">dialog</property>
+    <property name="width_request">400</property>
+    <property name="height_request">200</property>
     <child internal-child="vbox">
       <object class="GtkBox" id="dialog-vbox2">
         <property name="visible">True</property>
