--- libfilezilla-0.9.2/lib/libfilezilla/util.hpp.orig	2017-05-25 20:35:14.000000000 +0200
+++ libfilezilla-0.9.2/lib/libfilezilla/util.hpp	2017-06-06 20:53:40.457650687 +0200
@@ -28,7 +28,7 @@ int64_t FZ_PUBLIC_SYMBOL random_number(i
 
 /** \brief Get random uniformly distributed bytes
  */
-std::vector<uint8_t> random_bytes(size_t size);
+std::vector<uint8_t> FZ_PUBLIC_SYMBOL random_bytes(size_t size);
 
 }
 
