diff -wbBur xmms2-0.8DrO_o/src/clients/lib/python/xmmsapi.pyx xmms2-0.8DrO_o.q/src/clients/lib/python/xmmsapi.pyx
--- xmms2-0.8DrO_o/src/clients/lib/python/xmmsapi.pyx	2011-10-20 23:26:08.000000000 +0400
+++ xmms2-0.8DrO_o.q/src/clients/lib/python/xmmsapi.pyx	2015-03-03 17:23:55.858103590 +0300
@@ -1358,22 +1358,6 @@
 			res = xmmsc_medialib_import_path(self.conn, <char *>p)
 		return self.create_result(cb, res)
 
-	@deprecated
-	def medialib_path_import(self, path, cb = None, encoded=False):
-		"""
-		@deprecated
-		Use medialib_import_path(path, ...) instead
-		"""
-		return self.medialib_import_path(self, path, cb=cb, encoded=encoded)
-
-	@deprecated
-	def medialib_path_import_encoded(self, path, cb = None):
-		"""
-		@deprecated
-		Use medialib_import_path(path, ..., encoded=True) instead
-		"""
-		return self.medialib_import_path(self, path, cb=cb, encoded=True)
-
 	cpdef XmmsResult medialib_property_set(self, int id, key, value, source=None, cb=None):
 		"""
 		medialib_property_set(id, key, value, source=None, cb=None) -> XmmsResult
