diff -Naur old/src/basemath/buch2.c new/src/basemath/buch2.c
--- old/src/basemath/buch2.c	2020-03-07 21:03:06.784293669 -1000
+++ new/src/basemath/buch2.c	2020-03-07 21:03:14.577801531 -1000
@@ -2143,7 +2143,8 @@
   for (i = 1; i < RU; i++)
     if (gexpo(gel(emb,i)) > -1) break;
   p1 = imag_i( row_i(logunit,i, 1,RU-1) );
-  p1 = RgV_dotproduct(p1, ex); if (!R1) p1 = gmul2n(p1, -1);
+  p1 = RgV_dotproduct(p1, ex);
+  if (i > R1) p1 = gmul2n(p1, -1);
   p1 = gsub(garg(gel(emb,i),prec), p1);
   /* p1 = arg(the missing root of 1) */
 
diff -Naur old/src/test/32/nf new/src/test/32/nf
--- old/src/test/32/nf	2020-03-07 21:03:06.770947894 -1000
+++ new/src/test/32/nf	2020-03-07 21:03:14.577801531 -1000
@@ -494,6 +494,7 @@
 [1, 1/2*x - 1/2]
 Mod(0, x)
 Mod(-6/5, x)
+[0, 2, Mod(0, 2)]~
   ***   at top-level: nfinit([y^3+2,[1,x]])
   ***                 ^---------------------
   *** nfinit: incorrect type in nfbasic_init (t_VEC).
diff -Naur old/src/test/in/nf new/src/test/in/nf
--- old/src/test/in/nf	2020-03-07 21:03:06.780957225 -1000
+++ new/src/test/in/nf	2020-03-07 21:03:14.577801531 -1000
@@ -150,6 +150,10 @@
 nfinit(x, 3)[2]
 nfinit(1/2*x + 3/5, 3)[2]
 
+\\ #2164
+bnf = bnfinit(y^4-y-1);
+bnfisunit(bnf,-y^3+2*y^2-1)
+
 \\ ERRORS: keep at end of file
 
 nfinit([y^3+2,[1,x]])
