diff -Nur libquicktime-1.1.5.orig/utils/gtk/libquicktime_config.c libquicktime-1.1.5/utils/gtk/libquicktime_config.c
--- libquicktime-1.1.5.orig/utils/gtk/libquicktime_config.c	2008-12-30 08:05:36.000000000 -0800
+++ libquicktime-1.1.5/utils/gtk/libquicktime_config.c	2010-10-15 05:46:21.956697112 -0700
@@ -112,9 +112,9 @@
                    G_CALLBACK(main_window_button_callback),
                    (gpointer)ret);
 
-  GTK_WIDGET_SET_FLAGS (ret->close_button, GTK_CAN_DEFAULT);
-  GTK_WIDGET_SET_FLAGS (ret->save_button, GTK_CAN_DEFAULT);
-  
+  gtk_widget_set_can_default(GTK_WIDGET(ret->close_button), TRUE);
+  gtk_widget_set_can_default(GTK_WIDGET(ret->save_button), TRUE);
+
   gtk_widget_show(ret->close_button);
   gtk_widget_show(ret->save_button);
 
diff -Nur libquicktime-1.1.5.orig/utils/gtk/lqt_gtk.c libquicktime-1.1.5/utils/gtk/lqt_gtk.c
--- libquicktime-1.1.5.orig/utils/gtk/lqt_gtk.c	2009-03-11 07:25:50.000000000 -0700
+++ libquicktime-1.1.5/utils/gtk/lqt_gtk.c	2010-10-15 05:47:14.720013265 -0700
@@ -941,9 +941,9 @@
 		   G_CALLBACK(codec_config_window_button_callback),
 		   (gpointer)ret);
 
-  GTK_WIDGET_SET_FLAGS (ret->apply_button, GTK_CAN_DEFAULT);
-  GTK_WIDGET_SET_FLAGS (ret->close_button, GTK_CAN_DEFAULT);
-  GTK_WIDGET_SET_FLAGS (ret->restore_button, GTK_CAN_DEFAULT);
+  gtk_widget_set_can_default(GTK_WIDGET(ret->apply_button), TRUE);
+  gtk_widget_set_can_default(GTK_WIDGET(ret->close_button), TRUE);
+  gtk_widget_set_can_default(GTK_WIDGET(ret->restore_button), TRUE);
 
   gtk_widget_show(ret->apply_button);
   gtk_widget_show(ret->close_button);
@@ -1240,7 +1240,7 @@
   ret->mainbox = gtk_vbox_new(0, 10);
 
   ret->close_button = gtk_button_new_from_stock(GTK_STOCK_CLOSE);
-  GTK_WIDGET_SET_FLAGS (ret->close_button, GTK_CAN_DEFAULT);
+  gtk_widget_set_can_default(GTK_WIDGET(ret->close_button), TRUE);
 
   g_signal_connect(G_OBJECT(ret->close_button), "clicked",
                      G_CALLBACK(codec_info_window_button_callback),
