diff -Naurw gourmet-0.15.4.orig//src/lib/backends/db.py gourmet-0.15.4/src/lib/backends/db.py
--- gourmet-0.15.4.orig//src/lib/backends/db.py	2010-06-29 03:39:27.923509421 +0800
+++ gourmet-0.15.4/src/lib/backends/db.py	2010-06-29 03:40:19.776837096 +0800
@@ -794,7 +794,7 @@
 
     def get_unique_values (self, colname,table=None,**criteria):
         """Get list of unique values for column in table."""
-        if not table: table=self.recipe_table
+        if table is None: table=self.recipe_table
         if criteria: table = table.select(*make_simple_select_arg(criteria,table))
         if colname=='category' and table==self.recipe_table:
             print 'WARNING: you are using a hack to access category values.'
