diff --git a/plugins/save-param/xrandr-output.cpp b/plugins/save-param/xrandr-output.cpp
index 3497bf4..7ee0a89 100644
--- a/plugins/save-param/xrandr-output.cpp
+++ b/plugins/save-param/xrandr-output.cpp
@@ -30,6 +30,7 @@
 #include <QRect>
 #include <QStandardPaths>
 #include <QtXml>
+#include <KF5/KScreen/kscreen/mode.h>
 #include <KF5/KScreen/kscreen/output.h>
 #include <KF5/KScreen/kscreen/edid.h>
 #include "xrandr-config.h"
diff --git a/plugins/xrandr/xrandr-manager.cpp b/plugins/xrandr/xrandr-manager.cpp
index bbf12b5..b294c19 100644
--- a/plugins/xrandr/xrandr-manager.cpp
+++ b/plugins/xrandr/xrandr-manager.cpp
@@ -950,7 +950,7 @@ void XrandrManager::outputRemoved(int outputId)
 
 }
 
-void XrandrManager::primaryOutputChanged(const KScreen::OutputPtr &output)
+void XrandrManager::primaryOutputChanged()
 {
 //    USD_LOG(LOG_DEBUG,".");
 }
@@ -1197,7 +1197,7 @@ void XrandrManager::monitorsInit()
             mSaveConfigTimer->start(SAVE_CONFIG_TIME);
         });
 
-        connect(output.data(), &KScreen::Output::isPrimaryChanged, this, [this](){
+        connect(output.data(), &KScreen::Output::priorityChanged, this, [this](){
             KScreen::Output *senderOutput = static_cast<KScreen::Output*> (sender());
             USD_LOG(LOG_DEBUG,"PrimaryChanged:%s",senderOutput->name().toLatin1().data());
 
@@ -1289,7 +1289,7 @@ void XrandrManager::monitorsInit()
             this, &XrandrManager::outputRemoved,
             static_cast<Qt::ConnectionType>(Qt::QueuedConnection | Qt::UniqueConnection));
 
-    connect(mConfig.data(), &KScreen::Config::primaryOutputChanged,
+    connect(mConfig.data(), &KScreen::Config::prioritiesChanged,
             this, &XrandrManager::primaryOutputChanged);
 
     if (mMonitoredConfig->fileExists()) {
diff --git a/plugins/xrandr/xrandr-manager.h b/plugins/xrandr/xrandr-manager.h
index 33d0b60..c91c37c 100644
--- a/plugins/xrandr/xrandr-manager.h
+++ b/plugins/xrandr/xrandr-manager.h
@@ -36,6 +36,7 @@
 
 
 #include <KF5/KScreen/kscreen/config.h>
+#include <KF5/KScreen/kscreen/mode.h>
 #include <KF5/KScreen/kscreen/log.h>
 #include <KF5/KScreen/kscreen/output.h>
 #include <KF5/KScreen/kscreen/edid.h>
@@ -130,7 +131,7 @@ public Q_SLOTS:
     void RotationChangedEvent(const QString &rotation);
     void outputAddedHandle(const KScreen::OutputPtr &output);
     void outputRemoved(int outputId);
-    void primaryOutputChanged(const KScreen::OutputPtr &output);
+    void primaryOutputChanged();
    // void applyConfigTimerHandle();
     void setScreenMode(QString modeName);
     void setScreensParam(QString screensParam);
diff --git a/plugins/xrandr/xrandr-output.cpp b/plugins/xrandr/xrandr-output.cpp
index d18cf4f..4e125b8 100644
--- a/plugins/xrandr/xrandr-output.cpp
+++ b/plugins/xrandr/xrandr-output.cpp
@@ -30,6 +30,7 @@
 #include <QRect>
 #include <QStandardPaths>
 #include <QtXml>
+#include <KF5/KScreen/kscreen/mode.h>
 #include <KF5/KScreen/kscreen/output.h>
 #include <KF5/KScreen/kscreen/edid.h>
 #include "xrandr-config.h"
