diff -Naur linuxmint-muffin-1ad6bd4.orig/src/core/keybindings.c linuxmint-muffin-1ad6bd4/src/core/keybindings.c
--- linuxmint-muffin-1ad6bd4.orig/src/core/keybindings.c	2013-04-15 15:53:35.000000000 +0200
+++ linuxmint-muffin-1ad6bd4/src/core/keybindings.c	2013-04-24 18:37:54.028549911 +0200
@@ -3774,14 +3774,14 @@
                           handle_switch, META_TAB_LIST_GROUP);
 
   add_builtin_keybinding (display,
-                          "switch-windows",
+                          "switch-applications",
                           SCHEMA_COMMON_KEYBINDINGS,
                           META_KEY_BINDING_REVERSES,
                           META_KEYBINDING_ACTION_SWITCH_WINDOWS,
                           handle_switch, META_TAB_LIST_NORMAL);
 
   add_builtin_keybinding (display,
-                          "switch-windows-backward",
+                          "switch-applications-backward",
                           SCHEMA_COMMON_KEYBINDINGS,
                           REVERSES_AND_REVERSED,
                           META_KEYBINDING_ACTION_SWITCH_WINDOWS_BACKWARD,
