diff -Naur gambas3-3.9.2.orig/main/lib/gui/main.c gambas3-3.9.2/main/lib/gui/main.c
--- gambas3-3.9.2.orig/main/lib/gui/main.c	2016-12-17 13:38:03.000000000 +0100
+++ gambas3-3.9.2/main/lib/gui/main.c	2017-02-08 15:20:42.767501760 +0100
@@ -36,7 +36,7 @@
   NULL
 };
 
-char *GB_INCLUDE EXPORT = "gb.qt4";
+char *GB_INCLUDE EXPORT = "gb.qt5";
 
 int EXPORT GB_INIT(void)
 {
diff -Naur gambas3-3.9.2.orig/main/lib/gui.opengl/main.c gambas3-3.9.2/main/lib/gui.opengl/main.c
--- gambas3-3.9.2.orig/main/lib/gui.opengl/main.c	2016-12-17 13:38:04.000000000 +0100
+++ gambas3-3.9.2/main/lib/gui.opengl/main.c	2017-02-08 15:20:42.790835802 +0100
@@ -32,7 +32,7 @@
   NULL
 };
 
-char *GB_INCLUDE EXPORT = "gb.qt4.opengl";
+char *GB_INCLUDE EXPORT = "gb.qt5.opengl";
 
 int EXPORT GB_INIT(void)
 {
diff -Naur gambas3-3.9.2.orig/main/lib/gui.qt/main.c gambas3-3.9.2/main/lib/gui.qt/main.c
--- gambas3-3.9.2.orig/main/lib/gui.qt/main.c	2016-12-17 13:38:04.000000000 +0100
+++ gambas3-3.9.2/main/lib/gui.qt/main.c	2017-02-08 15:20:42.790835802 +0100
@@ -36,7 +36,7 @@
   NULL
 };
 
-char *GB_INCLUDE EXPORT = "gb.qt4";
+char *GB_INCLUDE EXPORT = "gb.qt5";
 
 int EXPORT GB_INIT(void)
 {
@@ -58,7 +58,7 @@
 	
 	if (use == USE_NOTHING)
 	{
-		use = USE_GB_QT4;
+		use = USE_GB_QT5;
 		
 		env = getenv("KDE_FULL_SESSION");
 		
diff -Naur gambas3-3.9.2.orig/main/lib/gui.qt.webkit/main.c gambas3-3.9.2/main/lib/gui.qt.webkit/main.c
--- gambas3-3.9.2.orig/main/lib/gui.qt.webkit/main.c	2016-12-17 13:38:04.000000000 +0100
+++ gambas3-3.9.2/main/lib/gui.qt.webkit/main.c	2017-02-08 15:20:42.790835802 +0100
@@ -32,7 +32,7 @@
   NULL
 };
 
-char *GB_INCLUDE EXPORT = "gb.qt4.webkit";
+char *GB_INCLUDE EXPORT = "gb.qt5.webkit";
 
 int EXPORT GB_INIT(void)
 {
