diff -r ee7da6873927 setup/Linux/oss/build/osscore.c
--- a/setup/Linux/oss/build/osscore.c   Mon Aug 09 04:43:06 2010 +0300
+++ b/setup/Linux/oss/build/osscore.c   Wed Sep 01 13:08:11 2010 +0300
@@ -933,7 +933,9 @@
   read_t tmp_read = (read_t) op->read;
   write_t tmp_write = (write_t) op->write;
   /* readdir_t tmp_readdir = (readdir_t)op->readdir; */
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,35)
   ioctl_t tmp_ioctl = (ioctl_t) op->ioctl;
+#endif
   mmap_t tmp_mmap = (mmap_t) op->mmap;
   open_t tmp_open = (open_t) op->open;
   release_t tmp_release = (release_t) op->release;
@@ -953,7 +955,9 @@
   fop->write = tmp_write;
   fop->readdir = NULL;      /* tmp_readdir; */
   fop->poll = tmp_poll;
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,35)
   fop->ioctl = tmp_ioctl;
+#endif
   fop->mmap = tmp_mmap;
   fop->open = tmp_open;
   fop->release = tmp_release;

