diff -rupN a/source/network/NetServer.cpp b/source/network/NetServer.cpp
--- a/source/network/NetServer.cpp	2013-12-18 16:08:56.000000000 +0000
+++ b/source/network/NetServer.cpp	2014-02-18 10:11:59.955214338 +0000
@@ -274,7 +274,7 @@ void* CNetServerWorker::SetupUPnP(void*)
 	// Check that the port was actually forwarded.
 	ret = UPNP_GetSpecificPortMappingEntry(urls.controlURL,
 									 data.first.servicetype,
-									 psPort, protocall,
+									 psPort, protocall, externalIPAddress,
 									 intClient, intPort, NULL/*desc*/,
 									 NULL/*enabled*/, duration);
 
