Index: dd_rescue-1.99.11/Makefile
===================================================================
--- a/Makefile
+++ b/Makefile
@@ -480,7 +480,7 @@ install: $(TARGETS)
 	ln -sf libddr_hash.so $(INSTALLLIBDIR)/libddr_MD5.so
 	mkdir -p $(MANDIR)/man1
 	$(INSTALL) $(INSTASROOT) -m 644 dd_rescue.1 ddr_lzo.1 ddr_crypt.1 $(MANDIR)/man1/
-	gzip -9f $(MANDIR)/man1/dd_rescue.1 $(MANDIR)/man1/ddr_lzo.1 $(MANDIR)/man1/ddr_crypt.1
+	gzip -n -9f $(MANDIR)/man1/dd_rescue.1 $(MANDIR)/man1/ddr_lzo.1 $(MANDIR)/man1/ddr_crypt.1
 
 check: $(TARGETS) find_nonzero md5 sha1 sha256 sha512 fmt_no
 	@echo "make check ... Pass VG=\"valgrind --options\" to use with valgrind"
@@ -751,4 +751,3 @@ make_check_crypt: check_crypt
 	# Cleanup
 	rm -f dd_rescue2 dd_rescue.dec dd_rescue.enc
 
-
