diff -u -r r8168-8.049.02/src/r8168_n.c r8168-8.049.02-linux517/src/r8168_n.c
--- r8168-8.049.02/src/r8168_n.c	2021-07-05 08:30:48.000000000 +0000
+++ r8168-8.049.02-linux517/src/r8168_n.c	2022-03-23 00:32:48.392881125 +0000
@@ -1733,7 +1733,7 @@
 static int rtl8168_proc_open(struct inode *inode, struct file *file)
 {
         struct net_device *dev = proc_get_parent_data(inode);
-        int (*show)(struct seq_file *, void *) = PDE_DATA(inode);
+        int (*show)(struct seq_file *, void *) = pde_data(inode);
 
         return single_open(file, show, dev);
 }
@@ -24061,6 +24061,7 @@
         struct rtl8168_private *tp = netdev_priv(dev);
         int i;
         u8 mac_addr[MAC_ADDR_LEN];
+        u8 addr[ETH_ALEN];
 
         for (i = 0; i < MAC_ADDR_LEN; i++)
                 mac_addr[i] = RTL_R8(tp, MAC0 + i);
@@ -24129,9 +24130,10 @@
         rtl8168_rar_set(tp, mac_addr);
 
         for (i = 0; i < MAC_ADDR_LEN; i++) {
-                dev->dev_addr[i] = RTL_R8(tp, MAC0 + i);
+                addr[i] = RTL_R8(tp, MAC0 + i);
                 tp->org_mac_addr[i] = dev->dev_addr[i]; /* keep the original MAC address */
         }
+        eth_hw_addr_set(dev, addr);
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,13)
         memcpy(dev->perm_addr, dev->dev_addr, dev->addr_len);
 #endif
