diff --git a/dde-file-manager-lib/views/dfiledialog.cpp b/dde-file-manager-lib/views/dfiledialog.cpp
index c0eef394..1a1d775f 100644
--- a/dde-file-manager-lib/views/dfiledialog.cpp
+++ b/dde-file-manager-lib/views/dfiledialog.cpp
@@ -99,7 +99,7 @@ DFileDialog::DFileDialog(QWidget *parent)
 
     DFMEventDispatcher::instance()->installEventFilter(this);
 
-    QPlatformFileDialogHelper::filterRegExp = "^(.*)\\(([^()]*)\\)$";
+    // QPlatformFileDialogHelper::filterRegExp = "^(.*)\\(([^()]*)\\)$";
 
     connect(statusBar()->acceptButton(), &QPushButton::clicked, this, &DFileDialog::onAcceptButtonClicked);
     connect(statusBar()->rejectButton(), &QPushButton::clicked, this, &DFileDialog::onRejectButtonClicked);
@@ -236,7 +236,7 @@ void DFileDialog::addDisableUrlScheme(const QString &scheme)
 QStringList qt_strip_filters(const QStringList &filters)
 {
     QStringList strippedFilters;
-    QRegExp r(QString::fromLatin1(QPlatformFileDialogHelper::filterRegExp));
+    QRegExp r(QString::fromLatin1("^(.*)\\(([^()]*)\\)$"));
     const int numFilters = filters.count();
     strippedFilters.reserve(numFilters);
     for (int i = 0; i < numFilters; ++i) {
