diff -wbBur gpsdrive-2.10pre7/src/gps_handler.c gpsdrive-2.10pre7.my/src/gps_handler.c
--- gpsdrive-2.10pre7/src/gps_handler.c	2009-06-14 13:57:25.000000000 +0000
+++ gpsdrive-2.10pre7.my/src/gps_handler.c	2010-04-23 08:52:32.000000000 +0000
@@ -319,7 +319,7 @@
 	if (mydebug > 40)
 	{
 		g_print ("  Fix Status: %d\n  Fix Mode: %d\n", data->status, data->fix.mode);
-		g_print ("  Sats used: %d of %d\n", data->satellites_used, data->satellites);
+		g_print ("  Sats used: %d of %d\n", data->satellites_used, data->satellites_visible);
 		g_print ("  Position: %.6f / %.6f\n", data->fix.latitude, data->fix.longitude);
 		g_print ("  Course: %.2f\n  GPS time: %.0f\n",
 			data->fix.track, data->fix.time);
@@ -333,13 +333,13 @@
 	coords.current_lon = data->fix.longitude;
 	current.altitude = data->fix.altitude;
 	current.gps_sats_used = data->satellites_used;
-	current.gps_sats_in_view = data->satellites;
+	current.gps_sats_in_view = data->satellites_visible;
 	if (data->set & TRACK_SET) 
 		current.course = data->fix.track * DEG_2_RAD;
 	if (data->set & SPEED_SET)
 		current.groundspeed = data->fix.speed * MPS_TO_KPH * local_config.distfactor;
-	current.gps_hdop = data->hdop;
-	current.gps_eph = data->fix.eph;
+	current.gps_hdop = 0; //data->hdop;
+	current.gps_eph = 0; //data->fix.eph;
 	current.gps_epv = data->fix.epv;
 
 	t_gpstime = (time_t) data->fix.time;
@@ -348,9 +348,9 @@
 	localtime_r(&t_gpstime, &t_tim);
 	strftime (current.loc_time, sizeof (current.loc_time), "%H:%M", &t_tim);
 
-	if ((data->set & SATELLITE_SET) && (data->satellites > 0))
+	if ((data->set & SATELLITE_SET) && (data->satellites_visible > 0))
 	{
-		for (i=0;i<data->satellites;i++)
+		for (i=0;i<data->satellites_visible;i++)
 		{
 			gps_sats[i].used = data->used[i];
 			gps_sats[i].prn = data->PRN[i];
@@ -381,7 +381,7 @@
 	 * s - get fix status (must be after "o" else its value is overwritten)
 	 */
 	gpsdata->set = 0;
-	gps_query (gpsdata, "oys\n");
+	gps_stream(gpsdata, WATCH_ENABLE, NULL);//gps_query (gpsdata, "oys\n");
 
 	return (TRUE);
 }
