diff -aur src/ApvlvFile.cpp.orig src/ApvlvFile.cpp
--- src/ApvlvFile.cpp.orig	2010-09-14 07:28:44.000000000 +0100
+++ src/ApvlvFile.cpp		2010-10-29 22:29:21.128949529 +0100
@@ -275,7 +275,7 @@
   {
     PopplerRectangle rect = { x1, y1, x2, y2 };
     PopplerPage *page = poppler_document_get_page (mDoc, pn);
-    *out = poppler_page_get_text (page, POPPLER_SELECTION_WORD, &rect);
+    *out = poppler_page_get_selected_text (page, POPPLER_SELECTION_WORD, &rect);
     if (*out != NULL)
       {
 	return true;
