diff -crB ibam-0.5.2/ibam.inl ibam-0.5.2-patch/ibam.inl
*** ibam-0.5.2/ibam.inl	2009-03-09 12:53:20.000000000 -0400
--- ibam-0.5.2-patch/ibam.inl	2010-12-13 16:17:51.243333333 -0500
***************
*** 959,965 ****
        acpi.close();
        apm = new acpi_status();
     } else {
!       sysfs.open((sysfs_path+"/BAT1/charge_full").c_str());
        if(sysfs.is_open()) {
  #ifdef IBAM_DEBUG
        cout << "using sysfs" << endl;
--- 959,965 ----
        acpi.close();
        apm = new acpi_status();
     } else {
!       sysfs.open((sysfs_path+"/BAT0/energy_full").c_str());
        if(sysfs.is_open()) {
  #ifdef IBAM_DEBUG
        cout << "using sysfs" << endl;
