diff -ruN eric6-19.6-orig/install.py eric6-19.6/install.py
--- eric6-19.6-orig/install.py	2019-06-13 19:56:12.530281194 +0300
+++ eric6-19.6/install.py	2019-06-13 19:56:58.593612194 +0300
@@ -828,7 +828,8 @@
     # set install prefix, if not None
     if distDir:
         for key in list(cfg.keys()):
-            cfg[key] = os.path.normpath(os.path.join(distDir, cfg[key]))
+            relpath = os.path.relpath(cfg[key], start="/") if cfg[key] else ""
+            cfg[key] = os.path.normpath(os.path.join(distDir, relpath))
     
     try:
         # Install the files
