Description: Use pkg-config for finding freetype
Author: Adrian Bunk <bunk@debian.org>
Bug-Debian: https://bugs.debian.org/892455

--- tvtime-1.0.11.orig/configure.ac
+++ tvtime-1.0.11/configure.ac
@@ -71,10 +71,7 @@ dnl ------------------------------------
 dnl freetype
 dnl ---------------------------------------------
 dnl Test for freetype
-AC_PATH_PROG(FREETYPE_CONFIG, freetype-config, no)
-if test "$FREETYPE_CONFIG" = "no" ; then
-  AC_MSG_ERROR(freetype2 needed and freetype-config not found)
-fi
+PKG_CHECK_MODULES(TTF, freetype2)
 
 dnl ---------------------------------------------
 dnl libxml2
--- tvtime-1.0.11.orig/src/Makefile.am
+++ tvtime-1.0.11/src/Makefile.am
@@ -19,9 +19,6 @@ pkgsysconfdir = $(sysconfdir)/@PACKAGE@
 tmpdir = /tmp
 localedir = $(datadir)/locale
 
-TTF_CFLAGS = `$(FREETYPE_CONFIG) --cflags`
-TTF_LIBS = `$(FREETYPE_CONFIG) --libs`
-
 # Set the following if you want to specify an additional font directory
 # FONT_CFLAGS = -DFONTDIR='/usr/share/fonts/truetype/freefont/'
 
