--- whipper-0.10.0/whipper/test/test_common_accurip.py	2021-05-17 17:46:08.000000000 +0200
+++ whipper-0.10.0.new/whipper/test/test_common_accurip.py	2023-04-23 16:40:15.123021975 +0200
@@ -42,8 +42,8 @@
         self.assertEqual(responses[1].discId1, '0000f21c')
         self.assertEqual(responses[1].discId2, '00027ef8')
         self.assertEqual(responses[1].cddbDiscId, '05021002')
-        self.assertEqual(responses[1].confidences[0], 7)
-        self.assertEqual(responses[1].confidences[1], 7)
+        self.assertEqual(responses[1].confidences[0], 8)
+        self.assertEqual(responses[1].confidences[1], 8)
         self.assertEqual(responses[1].checksums[0], 'dc77f9ab')
         self.assertEqual(responses[1].checksums[1], 'dd97d2c3')
 
