diff -rupN dwb-2014.03.07.original/exar/Makefile dwb-2014.03.07/exar/Makefile
--- dwb-2014.03.07.original/exar/Makefile	2014-03-07 15:29:57.737105453 +0100
+++ dwb-2014.03.07/exar/Makefile	2014-03-07 15:31:41.171106513 +0100
@@ -1,6 +1,6 @@
 ORIG_CFLAGS := $(CFLAGS)
 
-CFLAGS := -Wall -pedantic -Werror -Wextra -std=c99 -O2
+CFLAGS := -Wall -pedantic -Wextra -std=c99 -O2
 CFLAGS += $(ORIG_CFLAGS)
 
 DCFLAGS += -g -O0 -Wall -pedantic -Werror -Wextra -std=c99 

--- dwb-2014.03.07/dwbremote/Makefile	2014-03-07 12:39:43.000000000 +0100
+++ dwb-2014.03.07/dwbremote/Makefile.new	2015-04-17 22:59:41.289518737 +0200
@@ -7,7 +7,7 @@
 include $(BASE_DIR)/version.mk
 
 
-CFLAGS += -Wall -pedantic -Werror -Wextra -std=c99 -Os
+CFLAGS += -Wall -pedantic -Wextra -std=c99 -Os
 CFLAGS += -DVERSION=\"$(VERSION)\"
 CFLAGS += -DNAME=\"$(NAME)\"
 CFLAGS += -DCOPYRIGHT=\"$(COPYRIGHT)\"
