diff --git a/symengine/prime_sieve.h b/symengine/prime_sieve.h
index e54a6185..61a02d43 100644
--- a/symengine/prime_sieve.h
+++ b/symengine/prime_sieve.h
@@ -2,6 +2,7 @@
 #define SYMENGINE_PRIME_SIEVE_H
 
 #include <vector>
+#include <symengine/symengine_config.h>
 
 // Sieve class stores all the primes upto a limit. When a prime or a list of
 // prime
diff --git a/symengine/prime_sieve.cpp b/symengine/prime_sieve.cpp
index a5549d2b..22058a25 100644
--- a/symengine/prime_sieve.cpp
+++ b/symengine/prime_sieve.cpp
@@ -5,6 +5,9 @@
 #include <algorithm>
 #include <vector>
 #include <iterator>
+#ifdef HAVE_SYMENGINE_PRIMESIEVE
+#include <primesieve.hpp>
+#endif
 
 namespace SymEngine
 {
