diff --git a/dkms b/dkms
index abb01a4..265b69a 100644
--- a/dkms
+++ b/dkms
@@ -162,11 +162,14 @@ set_kernel_source_dir()
 
 # A little test function for DKMS commands that only work on one kernel.
 have_one_kernel() {
-    (( ${#kernelver[@]} > 1 )) && \
-	die 4 $"The action $1 does not support multiple kernel version" \
-	$"parameters on the command line."
-    [[ $all ]] && die 5 $"The action $1 does not support the --all" \
-	$"parameter."
+       if (( ${#kernelver[@]} != 1 )); then
+	  die 4 $"The action $1 does not support multiple kernel version" \
+	  $"parameters on the command line."
+       fi
+       if [[ $all ]]; then  
+           die 5 $"The action $1 does not support the --all" \
+	   $"parameter."
+       fi 
 }
