diff -r fb12dbcd0ad1 -r 01e87b730bb3 src/hugin1/hugin/AboutDialog.cpp
--- a/src/hugin1/hugin/AboutDialog.cpp	Wed May 15 17:41:28 2019 +0200
+++ b/src/hugin1/hugin/AboutDialog.cpp	Wed May 15 17:47:40 2019 +0200
@@ -42,7 +42,7 @@
 // workaround for a conflict between exiv2 and wxWidgets/CMake built
 #define HAVE_PID_T 1
 #endif
-#include "exiv2/exiv2.hpp"
+#include <exiv2/exiv2.hpp>
 #include "lensdb/LensDB.h"
 #include "sqlite3.h"
 #include <lcms2.h>
diff -r fb12dbcd0ad1 -r 01e87b730bb3 src/hugin1/hugin/PanoOperation.cpp
--- a/src/hugin1/hugin/PanoOperation.cpp	Wed May 15 17:41:28 2019 +0200
+++ b/src/hugin1/hugin/PanoOperation.cpp	Wed May 15 17:47:40 2019 +0200
@@ -36,8 +36,7 @@
 // workaround for a conflict between exiv2 and wxWidgets/CMake built
 #define HAVE_PID_T 1
 #endif
-#include <exiv2/exif.hpp>
-#include <exiv2/image.hpp>
+#include <exiv2/exiv2.hpp>
 #include "base_wx/LensTools.h"
 #include "base_wx/wxLensDB.h"
 #include "hugin/ResetDialog.h"
diff -r fb12dbcd0ad1 -r 01e87b730bb3 src/hugin1/hugin/RawImport.cpp
--- a/src/hugin1/hugin/RawImport.cpp	Wed May 15 17:41:28 2019 +0200
+++ b/src/hugin1/hugin/RawImport.cpp	Wed May 15 17:47:40 2019 +0200
@@ -35,10 +35,7 @@
 // workaround for a conflict between exiv2 and wxWidgets/CMake built
 #define HAVE_PID_T 1
 #endif
-#include <exiv2/exif.hpp>
-#include <exiv2/image.hpp>
-#include <exiv2/easyaccess.hpp>
-#include <exiv2/xmpsidecar.hpp>
+#include <exiv2/exiv2.hpp>
 #ifdef __WXMSW__
 #include <wx/msw/registry.h>
 #endif
diff -r fb12dbcd0ad1 -r 01e87b730bb3 src/hugin1/hugin/TextureManager.cpp
--- a/src/hugin1/hugin/TextureManager.cpp	Wed May 15 17:41:28 2019 +0200
+++ b/src/hugin1/hugin/TextureManager.cpp	Wed May 15 17:47:40 2019 +0200
@@ -65,8 +65,7 @@
 // workaround for a conflict between exiv2 and wxWidgets/CMake built
 #define HAVE_PID_T 1
 #endif
-#include "exiv2/exiv2.hpp"
-#include "exiv2/preview.hpp"
+#include <exiv2/exiv2.hpp>
 
 TextureManager::TextureManager(HuginBase::Panorama *pano, ViewState *view_state_in)
 {
diff -r fb12dbcd0ad1 -r 01e87b730bb3 src/hugin1/ptbatcher/FindPanoDialog.cpp
--- a/src/hugin1/ptbatcher/FindPanoDialog.cpp	Wed May 15 17:41:28 2019 +0200
+++ b/src/hugin1/ptbatcher/FindPanoDialog.cpp	Wed May 15 17:47:40 2019 +0200
@@ -36,8 +36,7 @@
 // workaround for a conflict between exiv2 and wxWidgets/CMake built
 #define HAVE_PID_T 1
 #endif
-#include "exiv2/exiv2.hpp"
-#include "exiv2/preview.hpp"
+#include <exiv2/exiv2.hpp>
 #ifdef _WIN32
 #include <commctrl.h>
 #endif
diff -r fb12dbcd0ad1 -r 01e87b730bb3 src/hugin_base/panodata/Exiv2Helper.cpp
--- a/src/hugin_base/panodata/Exiv2Helper.cpp	Wed May 15 17:41:28 2019 +0200
+++ b/src/hugin_base/panodata/Exiv2Helper.cpp	Wed May 15 17:47:40 2019 +0200
@@ -28,8 +28,7 @@
 #include "Exiv2Helper.h"
 #include "hugin_math/hugin_math.h"
 #include "hugin_utils/utils.h"  
-#include "exiv2/easyaccess.hpp"
-#include "exiv2/version.hpp"
+#include <exiv2/exiv2.hpp>
 
 namespace HuginBase
 {
diff -r fb12dbcd0ad1 -r 01e87b730bb3 src/hugin_base/panodata/SrcPanoImage.cpp
--- a/src/hugin_base/panodata/SrcPanoImage.cpp	Wed May 15 17:41:28 2019 +0200
+++ b/src/hugin_base/panodata/SrcPanoImage.cpp	Wed May 15 17:47:40 2019 +0200
@@ -40,9 +40,7 @@
 #include <vigra/diff2d.hxx>
 #include <vigra/imageinfo.hxx>
 #include <hugin_utils/utils.h>
-#include <exiv2/exif.hpp>
-#include <exiv2/image.hpp>
-#include <exiv2/easyaccess.hpp>
+#include <exiv2/exiv2.hpp>
 #include <lensdb/LensDB.h>
 #include "Exiv2Helper.h"
 
