diff --git a/src/i18n.c b/src/i18n.c
index 0bae3b0..4d12d0a 100644
--- a/src/i18n.c
+++ b/src/i18n.c
@@ -1,6 +1,7 @@
 /* Internationalization implementation.  Includes definitions of English
  * string arrays, and the i18n pointer. */
 
+#include <linux/slab.h>		/* For kmalloc. */
 #include <linux/ctype.h>
 #include <linux/module.h>
 #include <linux/string.h>
diff --git a/src/kobjects.c b/src/kobjects.c
index e27ca3f..1cd4566 100644
--- a/src/kobjects.c
+++ b/src/kobjects.c
@@ -11,6 +11,7 @@
  * Released under the GPL version 2 only.
  *
  */
+#include <linux/slab.h>		/* For kmalloc. */
 #include <linux/kernel.h>
 #include <linux/kobject.h>
 #include <linux/string.h>
diff --git a/src/selection.c b/src/selection.c
index 499604f..0f3ea4f 100644
--- a/src/selection.c
+++ b/src/selection.c
@@ -1,3 +1,4 @@
+#include <linux/slab.h> /* for kmalloc */
 #include <linux/consolemap.h>
 #include <linux/interrupt.h>
 #include <linux/sched.h>
