diff --git a/sure/__init__.py b/sure/__init__.py
index ef50bd4..d4c2afb 100644
--- a/sure/__init__.py
+++ b/sure/__init__.py
@@ -126,7 +126,7 @@ class CallBack(object):
             err = traceback.format_exc().splitlines()[-1]
             err = err.replace("{0}:".format(exc_klass.__name__), "").strip()
 
-            if err.startswith(self.callback_name) and (
+            if self.callback_name in err and (
                 "takes no arguments (1 given)" in err
                 or "takes 0 positional arguments but 1 was given" in err
             ):
