diff --git a/src/utils/lib/CryptoPP.h b/src/utils/lib/CryptoPP.h
index 775a898..af1621b 100644
--- a/src/utils/lib/CryptoPP.h
+++ b/src/utils/lib/CryptoPP.h
@@ -695,6 +695,7 @@ class CRYPTOPP_DLL NullNameValuePairs : public NameValuePairs
 {
 public:
 	bool GetVoidValue(const char* /* name */, const std::type_info& /* valueType */, void* /* pValue */) const {return false;}
+	NullNameValuePairs() {}
 };
 
 //! _
