From: Debian CLI Applications Team <pkg-cli-apps-team@lists.alioth.debian.org>
Date: Tue, 18 Nov 2014 10:07:49 +0800
Subject: use_dbussharp_2

===================================================================
---
 configure.ac | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 166dfd4..db2b685 100644
--- a/configure.ac
+++ b/configure.ac
@@ -44,7 +44,7 @@ fi
 AC_MSG_RESULT([$BANSHEE_COLLECTION_INDEXER])
 
 AC_MSG_CHECKING([which DBus implementation Banshee.CollectionIndexer requires])
-if $PKG_CONFIG --print-requires $BANSHEE_COLLECTION_INDEXER 2>&1 | grep dbus-sharp-1.0 > /dev/null ; then
+if $PKG_CONFIG --print-requires $BANSHEE_COLLECTION_INDEXER 2>&1 | grep dbus-sharp-2.0 > /dev/null ; then
 	BANSHEE_USE_DBUS_SHARP=yes
 	AC_MSG_RESULT([dbus-sharp])
 else
@@ -53,8 +53,8 @@ else
 fi
 
 dnl -- DBus# required for Banshee and Telepathy support
-PKG_CHECK_MODULES([DBUS_SHARP], [dbus-sharp-1.0], HAVE_DBUS_SHARP=yes, HAVE_DBUS_SHARP=no)
-PKG_CHECK_MODULES([DBUS_SHARP_GLIB], [dbus-sharp-glib-1.0], HAVE_DBUS_SHARP_GLIB=yes, HAVE_DBUS_SNARP_GLIB=no)
+PKG_CHECK_MODULES([DBUS_SHARP], [dbus-sharp-2.0], HAVE_DBUS_SHARP=yes, HAVE_DBUS_SHARP=no)
+PKG_CHECK_MODULES([DBUS_SHARP_GLIB], [dbus-sharp-glib-2.0], HAVE_DBUS_SHARP_GLIB=yes, HAVE_DBUS_SNARP_GLIB=no)
 
 if test "x$HAVE_DBUS_SHARP" = "xyes" ; then
 	DBUS_BUILD_DEFINES=-define:USE_DBUS_SHARP
