=== modified file 'Status-Notifier/src/applet-host-ias.c'
--- old/Status-Notifier/src/applet-host-ias.c	2012-03-18 02:41:47 +0000
+++ new/Status-Notifier/src/applet-host-ias.c	2012-05-27 12:50:54 +0000
@@ -238,10 +238,12 @@
 		cIconThemePath,
 		#if (INDICATOR_OLD_NAMES == 0)
 		cAccessbleDesc && *cAccessbleDesc != '\0' ? cAccessbleDesc :
+		#if (INDICATOR_APPLICATIONADDED_HAS_HINT == 1)
 		#if (INDICATOR_APPLICATIONADDED_HAS_TITLE == 1)
 		cTitle && *cTitle != '\0' ? cTitle :
 		#endif
 		#endif
+		#endif
 		cLabel
 		);
 	

