--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 CXX ?= g++
-CFLAGS = -Wall -Wconversion -O3 -fPIC
+CXXFLAGS += -Wall -Wconversion -fPIC
 SHVER = 3
 OS = $(shell uname)
 ifeq ($(OS),Darwin)
@@ -15,14 +15,14 @@ endif
 all: svm-train svm-predict svm-scale
 
 lib: svm.o
-	$(CXX) $(SHARED_LIB_FLAG) svm.o -o libsvm.so.$(SHVER)
+	$(CXX) $(SHARED_LIB_FLAG) svm.o -o libsvm.so.$(SHVER) $(LDFLAGS)
 svm-predict: svm-predict.c svm.o
-	$(CXX) $(CFLAGS) svm-predict.c svm.o -o svm-predict -lm
+	$(CXX) $(CXXFLAGS) svm-predict.c svm.o -o svm-predict -lm $(LDFLAGS)
 svm-train: svm-train.c svm.o
-	$(CXX) $(CFLAGS) svm-train.c svm.o -o svm-train -lm
+	$(CXX) $(CXXFLAGS) svm-train.c svm.o -o svm-train -lm $(LDFLAGS)
 svm-scale: svm-scale.c
-	$(CXX) $(CFLAGS) svm-scale.c -o svm-scale
+	$(CXX) $(CXXFLAGS) svm-scale.c -o svm-scale $(LDFLAGS)
 svm.o: svm.cpp svm.h
-	$(CXX) $(CFLAGS) -c svm.cpp
+	$(CXX) $(CXXFLAGS) -c svm.cpp
 clean:
 	rm -f *~ svm.o svm-train svm-predict svm-scale libsvm.so.$(SHVER)
--- a/svm-toy/qt/Makefile
+++ b/svm-toy/qt/Makefile
@@ -2,12 +2,12 @@
 # may need to adjust the path of header files
 CXX? = g++
 INCLUDE = /usr/include/qt
-CFLAGS = -Wall -O3 -I$(INCLUDE) -I$(INCLUDE)/QtWidgets -I$(INCLUDE)/QtGui -I$(INCLUDE)/QtCore -fPIC -std=c++11
+CXXFLAGS += -Wall -I$(INCLUDE) -I$(INCLUDE)/QtWidgets -I$(INCLUDE)/QtGui -I$(INCLUDE)/QtCore -fPIC -std=c++11
 LIB = -lQt5Widgets -lQt5Gui -lQt5Core
 MOC = /usr/bin/moc
 
 svm-toy: svm-toy.cpp svm-toy.moc ../../svm.o
-	$(CXX) $(CFLAGS) svm-toy.cpp ../../svm.o -o svm-toy $(LIB)
+	$(CXX) $(CXXFLAGS) svm-toy.cpp ../../svm.o -o svm-toy $(LIB) $(LDFLAGS)
 
 svm-toy.moc: svm-toy.cpp
 	$(MOC) svm-toy.cpp -o svm-toy.moc
