--- a/src/library/trackcollection.cpp
+++ b/src/library/trackcollection.cpp
@@ -2,10 +2,6 @@
 #include <QtDebug>
 
 #include "library/trackcollection.h"
-
-#ifdef __SQLITE3__
-#include <sqlite3.h>
-#endif
 
 #include "library/librarytablemodel.h"
 #include "library/schemamanager.h"

--- a/src/library/trackcollection.h
+++ b/src/library/trackcollection.h
@@ -34,8 +34,7 @@
 #include "library/dao/libraryhashdao.h"
 
 #ifdef __SQLITE3__
-typedef struct sqlite3_context sqlite3_context;
-typedef struct Mem sqlite3_value;
+#include <sqlite3.h>
 #endif
 
 class TrackInfoObject;

