--- a/src/sage/interfaces/r.py
+++ b/src/sage/interfaces/r.py
@@ -331,7 +331,7 @@
                   prompt = '> ', #default, later comes the change
 
                   # This is the command that starts up your program
-                  command = "R --vanilla --quiet",
+                  command = "R --no-readline --vanilla --quiet",
 
                   server=server,
                   server_tmpdir=server_tmpdir,
