diff --git a/pkgs/sagemath-standard/setup.py b/pkgs/sagemath-standard/setup.py
index 6744236593..a038006449 100755
--- a/pkgs/sagemath-standard/setup.py
+++ b/pkgs/sagemath-standard/setup.py
@@ -81,8 +81,7 @@ else:
 optional_packages_with_extensions = ['mcqd', 'bliss', 'tdlib',
                                      'coxeter3', 'sirocco', 'meataxe']
 distributions += ['sagemath-{}'.format(pkg)
-                  for pkg in optional_packages_with_extensions
-                  if is_package_installed_and_updated(pkg)]
+                  for pkg in optional_packages_with_extensions]
 log.warn('distributions = {0}'.format(distributions))
 from sage_setup.find import find_python_sources, find_extra_files
 python_packages, python_modules, cython_modules = find_python_sources(
