--- guitarix-0.35.1/src/gx_head/engine/gx_system.cpp.orig	2016-10-07 19:41:19.707493651 +0000
+++ guitarix-0.35.1/src/gx_head/engine/gx_system.cpp	2016-10-07 19:42:18.128197974 +0000
@@ -349,7 +349,7 @@
 	file->enumerate_children(G_FILE_ATTRIBUTE_STANDARD_NAME
 				 "," G_FILE_ATTRIBUTE_STANDARD_DISPLAY_NAME);
     Glib::RefPtr<Gio::FileInfo> file_info;
-    while ((file_info = child_enumeration->next_file()) != 0) {
+    while (file_info = child_enumeration->next_file()) {
 	if (file_info->get_file_type() == Gio::FILE_TYPE_DIRECTORY) {
 	    Glib::RefPtr<Gio::File> child = file->get_child(
 		file_info->get_attribute_byte_string(G_FILE_ATTRIBUTE_STANDARD_NAME));
diff --git a/trunk/src/gx_head/engine/ladspaback.cpp b/trunk/src/gx_head/engine/ladspaback.cpp
index 2a6bd01..be934a7 100644
--- guitarix-0.35.1/src/gx_head/engine/ladspaback.cpp
+++ guitarix-0.35.1/src/gx_head/engine/ladspaback.cpp
@@ -1622,7 +1622,7 @@ void LadspaPluginList::load(gx_system::CmdlineOptions& options, std::vector<std:
                                      "," G_FILE_ATTRIBUTE_STANDARD_FAST_CONTENT_TYPE);
         Glib::RefPtr<Gio::FileInfo> file_info;
 
-        while ((file_info = child_enumeration->next_file()) != 0) {
+        while ((file_info = child_enumeration->next_file())) {
             if (file_info->get_attribute_string(G_FILE_ATTRIBUTE_STANDARD_FAST_CONTENT_TYPE) == "application/x-sharedlib") {
                 std::string nm = file_info->get_attribute_byte_string(G_FILE_ATTRIBUTE_STANDARD_NAME);
                 if (lib_is_blacklisted(nm)) {
@@ -1650,7 +1650,7 @@ void LadspaPluginList::load(gx_system::CmdlineOptions& options, std::vector<std:
                                      "," G_FILE_ATTRIBUTE_STANDARD_FAST_CONTENT_TYPE);
         Glib::RefPtr<Gio::FileInfo> file_info;
 
-        while ((file_info = child_enumeration->next_file()) != 0) {
+        while ((file_info = child_enumeration->next_file())) {
 	    //if not f.endswith((".rdf",".rdfs")):
             if (file_info->get_attribute_string(G_FILE_ATTRIBUTE_STANDARD_FAST_CONTENT_TYPE) == "application/rdf+xml") {
                 std::string nm = file_info->get_attribute_byte_string(G_FILE_ATTRIBUTE_STANDARD_NAME);
--- guitarix-0.35.1/src/gx_head/engine/gx_system.cpp.orig	2016-10-07 19:53:44.599882708 +0000
+++ guitarix-0.35.1/src/gx_head/engine/gx_system.cpp	2016-10-07 19:53:56.963366468 +0000
@@ -329,7 +329,7 @@
 				       "," G_FILE_ATTRIBUTE_STANDARD_DISPLAY_NAME
 				       "," G_FILE_ATTRIBUTE_STANDARD_FAST_CONTENT_TYPE);
         Glib::RefPtr<Gio::FileInfo> file_info;
-        while ((file_info = child_enumeration->next_file()) != 0) {
+        while (file_info = child_enumeration->next_file()) {
 	    if (file_info->get_attribute_string(G_FILE_ATTRIBUTE_STANDARD_FAST_CONTENT_TYPE) == "audio/x-wav") {
 		listing.push_back(
 		    FileName(
