diff --git a/a/setup.py b/b/setup.py
index 30bcddb..6b37d3b 100644
--- a/a/setup.py
+++ b/b/setup.py
@@ -118,8 +118,8 @@ setup(
 		'scripts/powerline-daemon',
 		'scripts/powerline-render',
 		'scripts/powerline-config',
-	] + (['scripts/powerline'] if can_use_scripts else []),
-	data_files=(None if can_use_scripts else (('bin', ['scripts/powerline']),)),
+	],
+	data_files=(('bin', ['scripts/powerline']),),
 	keywords='',
 	packages=find_packages(exclude=('tests', 'tests.*')),
 	include_package_data=True,
