diff -Nur VirtualBox-3.1.0_OSE.orig/src/VBox/Frontends/VirtualBox/Makefile.kmk VirtualBox-3.1.0_OSE/src/VBox/Frontends/VirtualBox/Makefile.kmk
--- VirtualBox-3.1.0_OSE.orig/src/VBox/Frontends/VirtualBox/Makefile.kmk	2009-11-20 06:46:47.000000000 +0000
+++ VirtualBox-3.1.0_OSE/src/VBox/Frontends/VirtualBox/Makefile.kmk	2009-11-30 19:47:22.956848035 +0000
@@ -36,7 +36,7 @@
 
 # Show the update notifier dialog during startup. If this setting is disabled
 # the user can still update using the menu if desired.
-VBOX_WITH_UPDATE_REQUEST := 1
+#VBOX_WITH_UPDATE_REQUEST := 1
 
 
 #
