diff -Nur VirtualBox-3.2.12_OSE.orig//src/VBox/Main/ConsoleImpl.cpp VirtualBox-3.2.12_OSE/src/VBox/Main/ConsoleImpl.cpp
--- VirtualBox-3.2.12_OSE.orig//src/VBox/Main/ConsoleImpl.cpp	2010-12-01 09:09:35.000000000 -0800
+++ VirtualBox-3.2.12_OSE/src/VBox/Main/ConsoleImpl.cpp	2010-12-01 14:10:59.846374315 -0800
@@ -1074,8 +1074,10 @@
     updateGuestPropertiesVRDPDisconnect(u32ClientId);
 #endif /* VBOX_WITH_GUEST_PROPS */
 
+#ifdef VBOX_WITH_VRDP
     if (u32Clients == 0)
         mcGuestCredentialsProvided = false;
+#endif /* VBOX_WITH_VRDP */
 
     LogFlowFuncLeave();
     return;
