diff -uprb clang-2.7.orig/lib/Frontend/InitHeaderSearch.cpp clang-2.7/lib/Frontend/InitHeaderSearch.cpp
--- clang-2.7.orig/lib/Frontend/InitHeaderSearch.cpp	2010-03-06 21:38:10.000000000 +0200
+++ clang-2.7/lib/Frontend/InitHeaderSearch.cpp	2010-04-30 19:05:24.000000000 +0300
@@ -539,10 +539,10 @@ void InitHeaderSearch::AddDefaultCPlusPl
                                 "i586-suse-linux", "", "", triple);
     AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4",
                                 "x86_64-suse-linux", "", "", triple);
-    // Arch Linux 2008-06-24
-    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.3.1",
+    // Arch Linux 2010-04-30
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.5.0",
                                 "i686-pc-linux-gnu", "", "", triple);
-    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.3.1",
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.5.0",
                                 "x86_64-unknown-linux-gnu", "", "", triple);
     // Gentoo x86 2009.1 stable
     AddGnuCPlusPlusIncludePaths(
