diff --git a/Rakefile b/Rakefile
index bbcea3b..95a77bd 100644
--- a/Rakefile
+++ b/Rakefile
@@ -1,7 +1,3 @@
-require 'bundler'
-Bundler::GemHelper.install_tasks
-require 'bundler/setup'
-
 require 'rake/testtask'

 desc 'Run all tests'
@@ -9,12 +5,8 @@ task 'default' => ['test', 'test:performance']

 desc 'Run tests'
 task 'test' do
-  if (test_library = ENV['MOCHA_RUN_INTEGRATION_TESTS'])
-    Rake::Task["test:integration:#{test_library}"].invoke
-  else
-    Rake::Task['test:units'].invoke
-    Rake::Task['test:acceptance'].invoke
-  end
+  Rake::Task['test:units'].invoke
+  Rake::Task['test:acceptance'].invoke
 end

 namespace 'test' do # rubocop:disable Metrics/BlockLength
@@ -83,15 +75,6 @@ namespace 'test' do # rubocop:disable Metrics/BlockLength
   end
 end

-begin
-  require 'rubocop/rake_task'
-  if RUBY_VERSION >= '2.2.0' && (defined?(RUBY_ENGINE) && RUBY_ENGINE == 'ruby') && ENV['MOCHA_RUN_INTEGRATION_TESTS'].nil?
-    RuboCop::RakeTask.new
-    task 'test' => 'rubocop'
-  end
-rescue LoadError # rubocop:disable Lint/HandleExceptions
-end
-
 # rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
 def benchmark_test_case(klass, iterations)
   require 'benchmark'
@@ -121,37 +104,3 @@ def benchmark_test_case(klass, iterations)
   end
 end
 # rubocop:enable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
-
-if ENV['MOCHA_GENERATE_DOCS']
-  require 'yard'
-
-  desc 'Remove generated documentation'
-  task 'clobber_yardoc' do
-    `rm -rf ./docs`
-  end
-
-  task 'docs_environment' do
-    unless ENV['GOOGLE_ANALYTICS_WEB_PROPERTY_ID']
-      puts "\nWarning: GOOGLE_ANALYTICS_WEB_PROPERTY_ID was not defined\n\n"
-    end
-  end
-
-  desc 'Generate documentation'
-  YARD::Rake::YardocTask.new('yardoc' => 'docs_environment') do |task|
-    task.options = ['--title', "Mocha #{Mocha::VERSION}", '--fail-on-warning']
-  end
-
-  task 'checkout_docs_cname' do
-    `git checkout docs/CNAME`
-  end
-
-  task 'checkout_docs_js' do
-    `git checkout docs/js/app.js`
-    `git checkout docs/js/jquery.js`
-  end
-
-  desc 'Generate documentation'
-  task 'generate_docs' => %w[clobber_yardoc yardoc checkout_docs_cname checkout_docs_js]
-end
-
-task 'release' => 'default'
