From a798e2e41dc71951175749a6ac34a07a068871a1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 26 Jan 2018 20:53:23 +0100
Subject: [PATCH] Fix icon on about dialog

---
 about.glade | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/about.glade b/about.glade
index 85cad9d..d0d02ae 100644
--- a/about.glade
+++ b/about.glade
@@ -5,7 +5,7 @@
   <object class="GtkDialog" id="dialog">
     <property name="can_focus">False</property>
     <property name="title" translatable="yes">About Syncthing-GTK</property>
-    <property name="icon">icons/st-gtk-logo.png</property>
+    <property name="icon_name">syncthing-gtk</property>
     <property name="type_hint">dialog</property>
     <signal name="response" handler="on_dialog_response" swapped="no"/>
     <child internal-child="vbox">
@@ -53,7 +53,8 @@
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="margin_bottom">10</property>
-                <property name="pixbuf">icons/st-gtk-logo.png</property>
+                <property name="pixel_size">128</property>
+                <property name="icon_name">syncthing-gtk</property>
               </object>
               <packing>
                 <property name="expand">False</property>
