diff -wbBur udev/hso.udev udev.my/hso.udev
--- udev/hso.udev	2008-06-19 14:34:07.000000000 +0400
+++ udev.my/hso.udev	2012-01-07 15:31:02.000000000 +0400
@@ -1,8 +1,5 @@
 ACTION!="add", GOTO="hso_end"
 
-# Syntax switch, need possibly for upcoming Distros
-ATTR{dev}!="?*", GOTO="hso_old_syntax"
-
 ############################### NEW SYNTAX ###############################
 
 LABEL="hso_new_syntax"
@@ -63,65 +60,11 @@
 SUBSYSTEM=="tty", SUBSYSTEMS=="usb", ATTR{hsotype}=="PCSC",	   SYMLINK+="wpcsc0"
 GOTO="hso_tty_permission"
 
-
-############################### OLD SYNTAX ###############################
-
-LABEL="hso_old_syntax"
-#RUN+="/binb/touch /tmp/udev-syntax-old"
-
-# enable Selective suspend in USB core system by a helper script
-SUBSYSTEM=="usb", SYSFS{idVendor}=="0af0", SYSFS{bDeviceClass}=="ff", RUN+="/usr/sbin/osetsuspend udev %k"
-
-# send rezero command via ozerocdoff, which uses usblib for all known ZCOPTION devices, supported by bundled driver
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="6711", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="6731", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="6751", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="6771", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="6791", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="6811", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="6911", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="6951", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="6971", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7011", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7031", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7051", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7071", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7111", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7211", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7251", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7271", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7311", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="c031", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="d031", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="d033", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7301", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7361", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7401", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7501", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-SUBSYSTEM=="usb_device", SYSFS{idVendor}=="0af0", SYSFS{idProduct}=="7601", SYSFS{bDeviceClass}=="00", RUN+="/usr/sbin/ozerocdoff -wi 0x%s{idProduct}"
-# Note SYSTEMS and ATTRS is not supported by udev from Xandros/EeePC
-
-# add device node links to all possible device
-SUBSYSTEM=="tty", SYSFS{hsotype}=="Control",	 SYMLINK+="wctrl0"
-SUBSYSTEM=="tty", SYSFS{hsotype}=="Application", SYMLINK+="wapp0"
-SUBSYSTEM=="tty", SYSFS{hsotype}=="Application", SYMLINK+="wappa0"
-SUBSYSTEM=="tty", SYSFS{hsotype}=="Application2",SYMLINK+="wappb0"
-SUBSYSTEM=="tty", SYSFS{hsotype}=="Diagnostic",	 SYMLINK+="wdiag0"
-SUBSYSTEM=="tty", SYSFS{hsotype}=="Diagnostic",	 SYMLINK+="wdiaga0"
-SUBSYSTEM=="tty", SYSFS{hsotype}=="Diagnostic2", SYMLINK+="wdiagb0"
-SUBSYSTEM=="tty", SYSFS{hsotype}=="Modem",	 SYMLINK+="wmodem0"
-SUBSYSTEM=="tty", SYSFS{hsotype}=="GPS",	 SYMLINK+="wgps0"
-SUBSYSTEM=="tty", SYSFS{hsotype}=="GPS_Control", SYMLINK+="wgpsc0"
-SUBSYSTEM=="tty", SYSFS{hsotype}=="PCSC",	 SYMLINK+="wpcsc0"
-# Note use here SYSFS instead of ATTR to be compatible to Xandros/EeePC
-# Note use no extra check for  SUBSYSTEMS=="usb", ATTRS{DRIVERS}="hso", because it is not supported by Xandros/EeePC
-
-
 ############################### COMMON SYNTAX ###############################
 
 LABEL="hso_tty_permission"
 # setup group proper group rights 
-KERNEL=="ttyHS[0-9]*", NAME="%k", GROUP="dialout", MODE="0660"
+KERNEL=="ttyHS[0-9]*", GROUP="network", MODE="0660"
 
 
 SUBSYSTEM!="usb", GOTO="hso_end"
