diff -rupN a/plugins/gg/commands.c b/plugins/gg/commands.c
--- a/plugins/gg/commands.c	2011-03-17 18:08:11.000000000 +0100
+++ b/plugins/gg/commands.c	2013-08-31 11:31:42.521377436 +0200
@@ -1029,14 +1029,15 @@ static int token_gif_load (char *fname,
 	GifFileType *file;
 	int fd;
 	fd = open(fname, O_RDONLY);
+	int giferror;
 	if (fd == -1) {
 		snprintf(errbuf, sizeof(errbuf), "open(%s): %m", fname);
 		goto err;
 	}
 	
-	if (!(file = DGifOpenFileHandle(fd))) {
+	if (!(file = DGifOpenFileHandle(fd, giferror))) {
 		snprintf(errbuf, sizeof(errbuf), "DGifOpenFileHandle(): %d", 
-			GifLastError());
+			GifErrorString(giferror));
 		goto err2;
 	}
 
@@ -1046,7 +1047,7 @@ static int token_gif_load (char *fname,
 	}
 	
 	if (DGifSlurp(file) != GIF_OK) {
-		snprintf(errbuf, sizeof(errbuf), "DGifSlurp(): %d", GifLastError());
+		snprintf(errbuf, sizeof(errbuf), "DGifSlurp(): %d", GifErrorString(GIF_ERROR));
 		goto err3;
 	}
 
