# A seblu patch :)
--- rst2pdf-0.91/rst2pdf/createpdf.py	2012-03-04 16:59:51.000000000 +0100
+++ rst2pdf-0.91-patch/rst2pdf/createpdf.py	2012-05-06 20:45:30.905822400 +0200
@@ -67,7 +67,7 @@
 from docutils.readers import standalone
 from docutils.transforms import Transform
 
-from roman import toRoman
+from docutils.utils.roman import toRoman
 
 from reportlab.platypus import *
 from reportlab.platypus.doctemplate import IndexingFlowable
