commit 6ae91cd88fe276eb6d5dd517760395b2d0d180a0
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Thu Oct 26 14:48:08 2017 +0800

    Support conduit-extra 1.2.0

diff --git a/doc/GUIDE.md b/doc/GUIDE.md
index c741338b..8dfbfc1f 100644
--- a/doc/GUIDE.md
+++ b/doc/GUIDE.md
@@ -1085,7 +1085,7 @@ The following changes will be made to stack.yaml:
     - aeson-0.10.0.0
     - aeson-compat-0.3.0.0
     - attoparsec-0.13.0.1
-    - conduit-extra-1.1.9.2
+    - conduit-extra-1.2.0
     - email-validate-2.2.0
     - hex-0.1.2
     - http-api-data-0.2.2
@@ -2039,11 +2039,11 @@ image:
 and then run `stack image container` and then `docker images` to list
 the images.
 
-Note that the executable will be built in the development environment 
-and copied to the container, so the dev OS must match that of the 
+Note that the executable will be built in the development environment
+and copied to the container, so the dev OS must match that of the
 container OS. This is easily accomplished using [Docker integration](docker_integration.md),
-under which the exe emitted by `stack build` will be built on the 
-Docker container, not the local OS. 
+under which the exe emitted by `stack build` will be built on the
+Docker container, not the local OS.
 
 The executable will be stored under `/usr/local/bin/<your-project>-exe`
 in the running container.
diff --git a/src/Data/Attoparsec/Interpreter.hs b/src/Data/Attoparsec/Interpreter.hs
index ca9285e1..02b5a971 100644
--- a/src/Data/Attoparsec/Interpreter.hs
+++ b/src/Data/Attoparsec/Interpreter.hs
@@ -146,9 +146,9 @@ getInterpreterArgs file = do
 
     decodeError e =
       case e of
-        ParseError ctxs _ (Position line col) ->
+        ParseError ctxs _ (Position line col offset) ->
           if null ctxs
           then "Parse error"
           else ("Expecting " ++ intercalate " or " ctxs)
-          ++ " at line " ++ show line ++ ", column " ++ show col
+          ++ " at line " ++ show line ++ ", column " ++ show col ++ ", offset " ++ show offset
         DivergentParser -> "Divergent parser"
diff --git a/stack.cabal b/stack.cabal
index d29ab9df..1e0418cc 100644
--- a/stack.cabal
+++ b/stack.cabal
@@ -198,7 +198,7 @@ library
                    , bytestring >= 0.10.4.0
                    , clock >= 0.7.2
                    , conduit >= 1.2.8
-                   , conduit-extra >= 1.1.14
+                   , conduit-extra >= 1.2.0
                    , containers >= 0.5.5.1
                    , cryptonite >= 0.19 && < 0.25
                    , cryptonite-conduit >= 0.1 && < 0.3
@@ -389,7 +389,7 @@ test-suite stack-integration-test
                 , base >= 4.7 && < 5
                 , bytestring >= 0.10.4.0
                 , conduit >= 1.2.8
-                , conduit-extra >= 1.1.14
+                , conduit-extra >= 1.2.0
                 , containers >= 0.5.5.1
                 , directory >= 1.2.1.0 && < 1.4
                 , filepath >= 1.3.0.2
