diff --git a/src/Options/Applicative/Builder/Extra.hs b/src/Options/Applicative/Builder/Extra.hs
index d72e43af..c02dfd85 100644
--- a/src/Options/Applicative/Builder/Extra.hs
+++ b/src/Options/Applicative/Builder/Extra.hs
@@ -164,7 +164,7 @@ execExtraHelp args helpOpt parser pd =
                                 some (strArgument (metavar "OTHER ARGUMENTS") :: Parser String)))
                         (fullDesc <> progDesc pd))
         return ()
-  where hiddenHelper = abortOption ShowHelpText (long "help" <> hidden <> internal)
+  where hiddenHelper = abortOption (ShowHelpText Nothing) (long "help" <> hidden <> internal)
 
 -- | 'option', specialized to 'Text'.
 textOption :: Mod OptionFields Text -> Parser Text
diff --git a/src/Options/Applicative/Complicated.hs b/src/Options/Applicative/Complicated.hs
index a11bf51a..e78bd7ab 100644
--- a/src/Options/Applicative/Complicated.hs
+++ b/src/Options/Applicative/Complicated.hs
@@ -152,6 +152,6 @@ hsubparser' commandMetavar m = mkParser d g rdr
 -- | Non-hidden help option.
 helpOption :: Parser (a -> a)
 helpOption =
-    abortOption ShowHelpText $
+    abortOption (ShowHelpText Nothing) $
     long "help" <>
     help "Show this help text"
