--- Makefile	2012-01-15 13:25:28.922775770 +0100
+++ Makefile.new	2012-01-15 13:25:09.185815643 +0100
@@ -142,7 +142,7 @@
 npplayer_LDFLAGS  = $(LDFLAGS)
 npplayer_LDFLAGS += $(libpthread_LDFLAGS)
 npplayer_LIBS     = $(GTK_LIBS) $(GLIB_LIBS) $(CURL_LIBS) $(X_LIBS)
-npplayer_LIBS    += $(libpthread_LIBS) $(libsocket_LIBS)
+npplayer_LIBS    += $(libpthread_LIBS) $(libsocket_LIBS) -ldl
 
 libnoxshm_LIBRARY = libnoxshm.so
 libnoxshm_RAWSRCS = libnoxshm.c
