diff -upr cachy-0.3.0.orig/tests/stores/test_dict_store.py cachy-0.3.0/tests/stores/test_dict_store.py
--- cachy-0.3.0.orig/tests/stores/test_dict_store.py	2019-08-06 22:18:57.993954200 +0300
+++ cachy-0.3.0/tests/stores/test_dict_store.py	2021-12-02 23:24:31.936148644 +0200
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 
 from unittest import TestCase
-from flexmock import flexmock, flexmock_teardown
+from flexmock import flexmock
+from flexmock._api import flexmock_teardown
 
 from cachy.stores import DictStore
 
diff -upr cachy-0.3.0.orig/tests/stores/test_file_store.py cachy-0.3.0/tests/stores/test_file_store.py
--- cachy-0.3.0.orig/tests/stores/test_file_store.py	2019-08-06 22:18:57.994094000 +0300
+++ cachy-0.3.0/tests/stores/test_file_store.py	2021-12-02 23:24:32.446162662 +0200
@@ -7,7 +7,8 @@ import hashlib
 import shutil
 
 from unittest import TestCase
-from flexmock import flexmock, flexmock_teardown
+from flexmock import flexmock
+from flexmock._api import flexmock_teardown
 
 from cachy.serializers import JsonSerializer
 from cachy.stores import FileStore
diff -upr cachy-0.3.0.orig/tests/stores/test_redis_store.py cachy-0.3.0/tests/stores/test_redis_store.py
--- cachy-0.3.0.orig/tests/stores/test_redis_store.py	2019-08-06 22:57:58.550268400 +0300
+++ cachy-0.3.0/tests/stores/test_redis_store.py	2021-12-02 23:24:32.446162662 +0200
@@ -4,7 +4,8 @@ import math
 
 import redis
 from unittest import TestCase
-from flexmock import flexmock, flexmock_teardown
+from flexmock import flexmock
+from flexmock._api import flexmock_teardown
 from fakeredis import FakeServer
 from fakeredis import FakeStrictRedis
 from cachy.stores import RedisStore
diff -upr cachy-0.3.0.orig/tests/test_cache_manager.py cachy-0.3.0/tests/test_cache_manager.py
--- cachy-0.3.0.orig/tests/test_cache_manager.py	2019-08-06 22:18:57.994549300 +0300
+++ cachy-0.3.0/tests/test_cache_manager.py	2021-12-02 23:24:32.446162662 +0200
@@ -3,7 +3,8 @@
 import os
 import tempfile
 from unittest import TestCase
-from flexmock import flexmock, flexmock_teardown
+from flexmock import flexmock
+from flexmock._api import flexmock_teardown
 
 from cachy import CacheManager, Repository
 from cachy.stores import DictStore, FileStore
diff -upr cachy-0.3.0.orig/tests/test_repository.py cachy-0.3.0/tests/test_repository.py
--- cachy-0.3.0.orig/tests/test_repository.py	2019-08-06 22:18:57.994738000 +0300
+++ cachy-0.3.0/tests/test_repository.py	2021-12-02 23:24:32.446162662 +0200
@@ -2,7 +2,8 @@
 
 import datetime
 from unittest import TestCase
-from flexmock import flexmock, flexmock_teardown
+from flexmock import flexmock
+from flexmock._api import flexmock_teardown
 
 from cachy import Repository
 from cachy.contracts.store import Store
diff -upr cachy-0.3.0.orig/tests/test_tagged_cache.py cachy-0.3.0/tests/test_tagged_cache.py
--- cachy-0.3.0.orig/tests/test_tagged_cache.py	2019-08-06 22:18:57.994858300 +0300
+++ cachy-0.3.0/tests/test_tagged_cache.py	2021-12-02 23:24:32.446162662 +0200
@@ -7,7 +7,8 @@ from cachy.stores import DictStore, Redi
 from cachy.tag_set import TagSet
 from cachy.redis_tagged_cache import RedisTaggedCache
 from datetime import datetime, timedelta
-from flexmock import flexmock, flexmock_teardown
+from flexmock import flexmock
+from flexmock._api import flexmock_teardown
 
 
 class TaggedCacheTestCase(TestCase):
