--- def/defines.h.orig	2009-03-09 13:02:26.000000000 +0100
+++ def/defines.h	2009-03-09 13:02:41.000000000 +0100
@@ -27,10 +27,10 @@
 #define DEFINES_H
 
 // OS constants
-#define APP_WIN						true
+#define APP_WIN						false
 #define APP_MAC						false
-#define APP_X11						false
-#define APP_PORTABLE				true //at least one of the 3 OS must be true too!
+#define APP_X11						true
+#define APP_PORTABLE				false //at least one of the 3 OS must be true too!
 
 // Common program constants
 #define APP_ORGANIZATION 			"Tom Thielicke IT Solutions"
