From 118f8134e1d2f7578cba5474b00f3b1cbd1219e3 Mon Sep 17 00:00:00 2001
From: Thomas Dziedzic <gostrc@gmail.com>
Date: Sun, 27 Apr 2014 19:38:59 -0700
Subject: [PATCH] relax network upper bound to 2.6

---
 hslogger.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hslogger.cabal b/hslogger.cabal
index 3e0ca27..74c262a 100644
--- a/hslogger.cabal
+++ b/hslogger.cabal
@@ -42,7 +42,7 @@ Library
         System.Log.Handler.Growl, System.Log.Handler.Log4jXML,
         System.Log.Logger
     Extensions: CPP, ExistentialQuantification
-    Build-Depends: network < 2.5, mtl
+    Build-Depends: network < 2.6, mtl
     if !os(windows)
         Build-Depends: unix
     if flag(small_base)
-- 
1.9.2

