From e797faaa5a11ab47e512cbbff634b5e8cb778abf Mon Sep 17 00:00:00 2001
From: Guido Scholz <gscholz@users.sourceforge.net>
Date: Tue, 6 Nov 2018 21:41:10 +0100
Subject: [PATCH 03/11] Fix compiler warning

---
 src/synthdata.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/synthdata.cpp b/src/synthdata.cpp
index 6ecc16d..f4c17c7 100644
--- a/src/synthdata.cpp
+++ b/src/synthdata.cpp
@@ -729,7 +729,7 @@ void SynthData::call_modules(void)
 
   for (i = 0; i < wavoutModuleList.count(); i++)
     wavoutModuleList.at(i)->generateCycle();
-  
+
   for (i = 0; i < scopeModuleList.count(); i++)
     scopeModuleList.at(i)->generateCycle();
 
@@ -744,7 +744,7 @@ void SynthData::call_modules(void)
   //FIXME: moduleList is member of modularsynth class
   for (i = 0; i < moduleList.count(); i++)
     moduleList.at(i)->cycleReady = false;
-  
+
   for (i = 0; i < poly; i++) {
     noteCounter[i]++;
     if (noteCounter[i] > 1000000000)
@@ -897,10 +897,11 @@ void SynthData::handleMidiEventController(snd_seq_event_t *ev)
         mcctx->setMidiValueRT(
                 (ev->data.control.value << 7) + ev->data.control.value);
 
-    if (ev->data.control.param == MIDI_CTL_ALL_NOTES_OFF)
+    if (ev->data.control.param == MIDI_CTL_ALL_NOTES_OFF){
         for (int i = 0; i < poly; ++i)
             if (noteCounter[i] < 1000000 && channel[i] == ev->data.note.channel)
                 noteCounter[i] = 1000000;
+    }
 
     else if (ev->data.control.param == MIDI_CTL_SUSTAIN) {
         bool sustainFlag = ev->data.control.value > 63;
-- 
2.21.0

