From 5a19a8bc8f0256c143b039d7e4cffbe7696cf1bc Mon Sep 17 00:00:00 2001
From: Guido Scholz <gscholz@users.sourceforge.net>
Date: Tue, 6 Nov 2018 21:54:55 +0100
Subject: [PATCH 05/11] Add git config file for files to be ignored

---
 .gitignore | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)
 create mode 100644 .gitignore

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..604087a
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,32 @@
+*.o
+*.moc.cpp
+*_moc.cpp
+*.qm
+*.gz
+*.xz
+*.bz2
+*.html
+*.swp
+Makefile
+Makefile.in
+configure
+config.h
+config.h.in
+aclocal.m4
+config.guess
+config.sub
+config.log
+config.status
+compile
+depcomp
+install-sh
+missing
+mkinstalldirs
+stamp-h1
+INSTALL
+autom4te.cache/
+src/.deps/
+src/ams
+src/config.h.in~
+tags
+test-driver
-- 
2.21.0

