diff --git i/example_project/settings.py w/example_project/settings.py
index 1fae6bee..8ad39d28 100644
--- i/example_project/settings.py
+++ w/example_project/settings.py
@@ -26,16 +26,10 @@ For the full list of settings and their values, see
 https://docs.djangoproject.com/en/1.9/ref/settings/
 """
 
-# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
-import os
-
 # Compatibility with Bootstrap 3
 from django.contrib.messages import constants as messages
 
 
-BASE_DIR = os.path.dirname(os.path.abspath(__file__))
-
-
 # Quick-start development settings - unsuitable for production
 # See https://docs.djangoproject.com/en/1.9/howto/deployment/checklist/
 
@@ -43,7 +37,7 @@ BASE_DIR = os.path.dirname(os.path.abspath(__file__))
 SECRET_KEY = '$!-7^wl#wiifjbh)5@f7ji%x!vp7s1vzbvwt26hxv$idixq0u0'
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = True
+DEBUG = False
 
 ADMINS = (
     #('Admin', 'webmaster@example.com'),
@@ -85,15 +79,15 @@ INSTALLED_APPS = (
     'allauth.socialaccount.providers.stackexchange',
 
     # Dev only dependencies. Do not include in any production site.
-    'debug_toolbar',
-    'requests_panel',
+    # 'debug_toolbar',
+    # 'requests_panel',
 )
 
 
 MIDDLEWARE = [
     'django.contrib.sessions.middleware.SessionMiddleware',
     # NOTE: Do not include DebugToolbarMiddleware in any production site.
-    'debug_toolbar.middleware.DebugToolbarMiddleware',
+    # 'debug_toolbar.middleware.DebugToolbarMiddleware',
     'django.middleware.common.CommonMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
     'django.middleware.locale.LocaleMiddleware',
@@ -142,7 +136,7 @@ WSGI_APPLICATION = 'wsgi.application'
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'postorius.db'),
+        'NAME': '/var/lib/postorius/data/postorius.db',
     }
 }
 
@@ -190,7 +184,7 @@ USE_TZ = True
 # Don't put anything in this directory yourself; store your static files
 # in apps' "static/" subdirectories and in STATICFILES_DIRS.
 # Example: "/var/www/example.com/static/"
-STATIC_ROOT = os.path.join(BASE_DIR, 'static')
+STATIC_ROOT = '/var/lib/postorius/static'
 
 # URL prefix for static files.
 # Example: "http://example.com/static/", "http://static.example.com/"
@@ -269,7 +263,7 @@ LOGGING = {
             'level': 'INFO',
             #'class': 'logging.handlers.RotatingFileHandler',
             'class': 'logging.handlers.WatchedFileHandler',
-            'filename': os.path.join(BASE_DIR, 'logs', 'postorius.log'),
+            'filename': '/var/log/postorius/postorius.log',
             'formatter': 'verbose',
         },
     },
