diff --git a/cmake/macros/SundialsAddLibrary.cmake b/cmake/macros/SundialsAddLibrary.cmake
index 1d78918..5acd856 100644
--- a/cmake/macros/SundialsAddLibrary.cmake
+++ b/cmake/macros/SundialsAddLibrary.cmake
@@ -122,7 +122,7 @@ macro(sundials_add_library target)
 
   # library types to create
   set(_libtypes "")
-  if(BUILD_STATIC_LIBS AND (NOT sundials_add_library_SHARED_ONLY))
+  if(BUILD_STATIC_LIBS AND sundials_add_library_STATIC_ONLY)
     set(_libtypes "STATIC")
   endif()
 
