diff --git a/src/sage_setup/docbuild/__init__.py b/src/sage_setup/docbuild/__init__.py
index ff67979..83b49e1 100644
--- a/src/sage_setup/docbuild/__init__.py
+++ b/src/sage_setup/docbuild/__init__.py
@@ -332,6 +331,18 @@ class AllBuilder(object):
         for document in refs:
             getattr(get_builder(document), 'inventory')(*args, **kwds)
 
+        # In the inventory build, the directory
+        # SAGE_DOC/inventory/en/reference/_static might be created,
+        # but it is unnecessary in the inventory build -- this
+        # directory should contain javascript and style files
+        # specifically for the html build -- and it can confuse
+        # intersphinx and the html builder when they look for
+        # inventory files. So delete this directory.
+        staticdir = os.path.join(SAGE_DOC, 'inventory', 'en', 'reference', '_static')
+        if os.path.isdir(staticdir):
+            logger.info("Deleting _static directory in the inventory build")
+            shutil.rmtree(staticdir)
+
         logger.warning("Building reference manual, second pass.\n")
         for document in refs:
             getattr(get_builder(document), name)(*args, **kwds)
@@ -526,6 +537,17 @@ class ReferenceBuilder(AllBuilder):
             output_dir = self._output_dir(format, lang)
             L = [(doc, lang, format, kwds) + args for doc in self.get_all_documents(refdir)]
             build_many(build_ref_doc, L)
+            # In the html build, a file -- not a directory --
+            # SAGE_DOC/html/en/reference/_static might be created,
+            # with the contents of Sphinx's graphviz.css file.  This
+            # is supposed to be a directory, and the presence of this
+            # file prevents copying of any css files, etc., to
+            # reference/_static. So delete this file.
+            staticfile = os.path.join(SAGE_DOC, 'html', 'en', 'reference', '_static')
+            if os.path.isfile(staticfile):
+                logger.info("Deleting _static file in the html build")
+                os.unlink(staticfile)
+
             # The html refman must be build at the end to ensure correct
             # merging of indexes and inventories.
             # Sphinx is run here in the current process (not in a
diff --git a/src/sage_setup/docbuild/ext/sage_autodoc.py b/src/sage_setup/docbuild/ext/sage_autodoc.py
index 2c399bad51..96abec3382 100644
--- a/src/sage_setup/docbuild/ext/sage_autodoc.py
+++ b/src/sage_setup/docbuild/ext/sage_autodoc.py
@@ -39,8 +39,9 @@ from docutils.statemachine import ViewList
 import sphinx
 from sphinx.util import rpartition, force_decode
 from sphinx.locale import _
-from sphinx.pycode import ModuleAnalyzer, PycodeError
-from sphinx.application import ExtensionError
+from sphinx.pycode import ModuleAnalyzer
+from sphinx.errors import ExtensionError, PycodeError
+from sphinx.util.logging import getLogger
 from sphinx.util.nodes import nested_parse_with_titles
 from docutils.parsers.rst import Directive
 from sphinx.util.inspect import getargspec, isdescriptor, safe_getmembers, \
