diff -rupN lightdm-1.5.1.orig/data/lightdm.conf lightdm-1.5.1/data/lightdm.conf
--- lightdm-1.5.1.orig/data/lightdm.conf	2013-03-06 03:40:10.000000000 +0100
+++ lightdm-1.5.1/data/lightdm.conf	2013-03-12 22:59:22.494743200 +0100
@@ -24,7 +24,7 @@
 #user-authority-in-system-dir=false
 #guest-account-script=guest-account
 #log-directory=/var/log/lightdm
-#run-directory=/var/run/lightdm
+run-directory=/run/lightdm
 #cache-directory=/var/cache/lightdm
 #xsessions-directory=/usr/share/xsessions
 #remote-sessions-directory=/usr/share/lightdm/remote-sessions
@@ -77,7 +77,7 @@
 #user-session=default
 #allow-guest=true
 #guest-session=UNIMPLEMENTED
-#session-wrapper=lightdm-session
+session-wrapper=/etc/lightdm/xsession
 #display-setup-script=
 #greeter-setup-script=
 #session-setup-script=
diff -rupN lightdm-1.5.1.orig/data/users.conf lightdm-1.5.1/data/users.conf
--- lightdm-1.5.1.orig/data/users.conf	2013-03-06 03:40:10.000000000 +0100
+++ lightdm-1.5.1/data/users.conf	2013-03-12 22:54:29.398854700 +0100
@@ -8,7 +8,7 @@
 # hidden-users = Users that are not shown to the user
 # hidden-shells = Shells that indicate a user cannot login
 #
-[UserAccounts]
-minimum-uid=500
+[UserList]
+minimum-uid=1000
 hidden-users=nobody nobody4 noaccess
-hidden-shells=/bin/false /usr/sbin/nologin
+hidden-shells=/bin/false /sbin/nologin
