--- qsopcast-0.3.8-qt4/src/loadsave.cpp~	2013-02-27 17:01:20.543229693 +0000
+++ qsopcast-0.3.8-qt4/src/loadsave.cpp	2013-02-27 17:01:38.629768649 +0000
@@ -18,6 +18,8 @@
 *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+#include <unistd.h>
+
 #include "mypopupmenu.h"
 #include "loadsave.h"
 #include "mainwindow.h"
--- qsopcast-0.3.8-qt4/src/sopfork.cpp~	2013-02-27 17:03:19.355725555 +0000
+++ qsopcast-0.3.8-qt4/src/sopfork.cpp	2013-02-27 17:03:33.018960848 +0000
@@ -18,6 +18,8 @@
 *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+#include <unistd.h>
+
 #include "sopfork.h"
 #include "header.h"
 #include "playfork.h"
--- qsopcast-0.3.8-qt4/src/record.cpp~	2013-02-27 17:05:04.374982336 +0000
+++ qsopcast-0.3.8-qt4/src/record.cpp	2013-02-27 17:05:15.474903741 +0000
@@ -18,6 +18,8 @@
 *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+#include <unistd.h>
+
 #include "sopfork.h"
 #include "record.h"
 #include "header.h"
