Subject: [PATCH] allow-system-gradle
---
Index: gradle/testing/defaults-tests.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gradle/testing/defaults-tests.gradle b/gradle/testing/defaults-tests.gradle
--- a/gradle/testing/defaults-tests.gradle	(revision 92c5515e2918c22513f7aa527d6c6db943150fea)
+++ b/gradle/testing/defaults-tests.gradle	(date 1681295313069)
@@ -89,7 +89,7 @@
 
     tasks.withType(Test) {
       ext {
-        testOutputsDir = file("${reports.junitXml.destination}/outputs")
+        testOutputsDir = file("${reports.junitXml.outputLocation.get()}/outputs")
       }
 
       // LUCENE-9660: Make it possible to always rerun tests, even if they're incrementally up-to-date.
Index: buildSrc/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/buildSrc/build.gradle b/buildSrc/build.gradle
--- a/buildSrc/build.gradle	(revision 92c5515e2918c22513f7aa527d6c6db943150fea)
+++ b/buildSrc/build.gradle	(date 1681295152900)
@@ -25,7 +25,6 @@
 }
 
 // Make sure the build environment is consistent.
-apply from: file('../gradle/validation/check-environment.gradle')
 
 // Load common buildSrc and script deps.
 apply from: file("scriptDepVersions.gradle")
