From 1a7fa99aac7373d7f1d4355b46656cdf898a1353 Mon Sep 17 00:00:00 2001
From: Simon Byrne <simonbyrne@gmail.com>
Date: Fri, 6 Jan 2023 17:42:59 -0800
Subject: [PATCH] fix printing

---
 base/mpfr.jl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/mpfr.jl b/base/mpfr.jl
index 45a518776989..69b6d104c065 100644
--- a/base/mpfr.jl
+++ b/base/mpfr.jl
@@ -1028,7 +1028,7 @@ function _string(x::BigFloat, fmt::String)::String
     isfinite(x) || return string(Float64(x))
     _prettify_bigfloat(string_mpfr(x, fmt))
 end
-_string(x::BigFloat) = _string(x, "%.Re")
+_string(x::BigFloat) = _string(x, "%Re")
 _string(x::BigFloat, k::Integer) = _string(x, "%.$(k)Re")
 
 string(b::BigFloat) = _string(b)
