diff --git a/makefile b/makefile
index 04e60c2..04e9b4c 100644
--- a/makefile
+++ b/makefile
@@ -149,7 +149,8 @@ ldcclient: githash
 	${LDC} ${CLIENT_SRC} ${LDC_CLIENT_FLAGS}
 
 $(OBJ_DIR)/%.o: $(SERVER_SRC)
-	$(LDC) $*.d $(LDC_SERVER_FLAGS) -od=$(OBJ_DIR) -op -c
+	echo $@
+	$(LDC) $*.d $(LDC_SERVER_FLAGS) -of=$@ -c
 
 ldcserver: githash $(SERVER_OBJS)
 	${LDC} ${SERVER_OBJS} ${LDC_SERVER_FLAGS} -of=bin/dcd-server
