From 746c218c620d7681f6c9d769631ee1ac0d2b5987 Mon Sep 17 00:00:00 2001
From: Casey Bodley <cbodley@redhat.com>
Date: Thu, 6 Jul 2017 16:31:23 -0400
Subject: [PATCH 1/6] rgw: update beast frontend/submodule to v116

Signed-off-by: Casey Bodley <cbodley@redhat.com>
---
 src/rgw/rgw_env.cc           |  2 +-
 5 files changed, 39 insertions(+), 79 deletions(-)

diff --git a/src/rgw/rgw_env.cc b/src/rgw/rgw_env.cc
index 8be133d67f80..f17b7832ba94 100644
--- a/src/rgw/rgw_env.cc
+++ b/src/rgw/rgw_env.cc
@@ -19,7 +19,7 @@ void RGWEnv::init(CephContext *cct)
 
 void RGWEnv::set(const boost::string_ref& name, const boost::string_ref& val)
 {
-  env_map[std::string{name}] = std::string{val};
+  env_map[name.to_string()] = val.to_string();
 }
 
 void RGWEnv::init(CephContext *cct, char **envp)

From b904b575042d6598a9cf74d23beecd4884e097d2 Mon Sep 17 00:00:00 2001
From: Casey Bodley <cbodley@redhat.com>
Date: Mon, 25 Sep 2017 11:13:08 -0400
Subject: [PATCH 5/6] rgw: remove boost::coroutine and context deps

the beast frontend no longer uses stackful coroutines, so these
dependencies aren't necessary

Signed-off-by: Casey Bodley <cbodley@redhat.com>
---
 CMakeLists.txt         | 4 ----
 src/rgw/CMakeLists.txt | 5 -----
 2 files changed, 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d30d00eded8a..6b6a827a2e8f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -543,10 +543,6 @@ if(WITH_MGR)
 	list(APPEND BOOST_COMPONENTS python)
 endif()
 
-if(WITH_RADOSGW_BEAST_FRONTEND)
-	list(APPEND BOOST_COMPONENTS coroutine context)
-endif()
-
 set(Boost_USE_MULTITHREADED ON)
 # require minimally the bundled version
 if(WITH_SYSTEM_BOOST)
diff --git a/src/rgw/CMakeLists.txt b/src/rgw/CMakeLists.txt
index 8aa91b3fc031..cf93257e9a63 100644
--- a/src/rgw/CMakeLists.txt
+++ b/src/rgw/CMakeLists.txt
@@ -173,11 +173,6 @@ endif (WITH_RADOSGW_BEAST_FRONTEND)
 add_library(radosgw_a STATIC ${radosgw_srcs}
   $<TARGET_OBJECTS:civetweb_common_objs>)
 target_link_libraries(radosgw_a rgw_a ${SSL_LIBRARIES})
-if(WITH_RADOSGW_BEAST_FRONTEND)
-  target_link_libraries(radosgw_a
-    Boost::coroutine
-    Boost::context)
-endif()
 
 add_executable(radosgw rgw_main.cc)
 target_link_libraries(radosgw radosgw_a librados

From 18c6235cb1af3268595d879c6b42c21c6dcc735a Mon Sep 17 00:00:00 2001
From: Casey Bodley <cbodley@redhat.com>
Date: Mon, 18 Dec 2017 13:03:35 -0500
Subject: [PATCH 1/5] cmake: update minimum boost version to 1.66

Signed-off-by: Casey Bodley <cbodley@redhat.com>
---
 CMakeLists.txt                 | 4 ++--
 cmake/modules/BuildBoost.cmake | 6 +++---
 make-dist                      | 4 ++--
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 889b6f9fc7d2..fd9761fcfa2b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -555,13 +555,13 @@ if(WITH_SYSTEM_BOOST)
   else()
     set(Boost_USE_STATIC_LIBS ON)
   endif()
-  find_package(Boost 1.61 COMPONENTS ${BOOST_COMPONENTS} REQUIRED)
+  find_package(Boost 1.66 COMPONENTS ${BOOST_COMPONENTS} REQUIRED)
 else()
   set(BOOST_J 1 CACHE STRING
     "max jobs for Boost build") # override w/-DBOOST_J=<n>
   set(Boost_USE_STATIC_LIBS ON)
   include(BuildBoost)
-  build_boost(1.63
+  build_boost(1.66
     COMPONENTS ${BOOST_COMPONENTS} ${BOOST_HEADER_COMPONENTS})
   include_directories(BEFORE SYSTEM ${Boost_INCLUDE_DIRS})
 endif()
diff --git a/cmake/modules/BuildBoost.cmake b/cmake/modules/BuildBoost.cmake
index d70828398ecf..36ac6b925bf1 100644
--- a/cmake/modules/BuildBoost.cmake
+++ b/cmake/modules/BuildBoost.cmake
@@ -79,12 +79,12 @@ function(do_build_boost version)
     message(STATUS "boost already in src")
     set(source_dir
       SOURCE_DIR "${PROJECT_SOURCE_DIR}/src/boost")
-  elseif(version VERSION_GREATER 1.63)
+  elseif(version VERSION_GREATER 1.66)
     message(FATAL_ERROR "Unknown BOOST_REQUESTED_VERSION: ${version}")
   else()
     message(STATUS "boost will be downloaded from sf.net")
-    set(boost_version 1.63.0)
-    set(boost_md5 1c837ecd990bb022d07e7aab32b09847)
+    set(boost_version 1.66.0)
+    set(boost_md5 b2dfbd6c717be4a7bb2d88018eaccf75)
     string(REPLACE "." "_" boost_version_underscore ${boost_version} )
     set(boost_url http://downloads.sourceforge.net/project/boost/boost/${boost_version}/boost_${boost_version_underscore}.tar.bz2)

From 88a75c38287b5a9782bda8925e303f716ad57857 Mon Sep 17 00:00:00 2001
From: Casey Bodley <cbodley@redhat.com>
Date: Sun, 26 Nov 2017 15:46:26 -0500
Subject: [PATCH 4/5] cmake: add WITH_BOOST_CONTEXT option

adds a more specific option for this boost::context dependency, which was
previously only used by the radosgw beast frontend. see
http://tracker.ceph.com/issues/20048 for more background

Signed-off-by: Casey Bodley <cbodley@redhat.com>
---
 CMakeLists.txt                | 12 ++++++++++++
 ceph.spec.in                  |  4 ++--
 debian/rules                  |  6 +++---
 src/include/config-h.in.cmake |  3 +++
 4 files changed, 20 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd9761fcfa2b..2337694180e3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -362,6 +362,11 @@ if(WITH_BLKIN)
   include_directories(SYSTEM src/blkin/blkin-lib)
 endif(WITH_BLKIN)
 
+option(WITH_BOOST_CONTEXT "Enable boost::asio stackful coroutines" ON)
+if(WITH_BOOST_CONTEXT)
+  set(HAVE_BOOST_CONTEXT ON)
+endif()
+
 #option for RGW
 option(WITH_RADOSGW "Rados Gateway is enabled" ON)
 option(WITH_RADOSGW_FCGI_FRONTEND "Rados Gateway's FCGI frontend is enabled" OFF)
@@ -371,6 +376,10 @@ if(WITH_RADOSGW)
   if(WITH_RADOSGW_FCGI_FRONTEND)
     find_package(fcgi REQUIRED)
   endif()
+  if(WITH_RADOSGW_BEAST_FRONTEND AND NOT WITH_BOOST_CONTEXT)
+    message(WARNING "disabling WITH_RADOSGW_BEAST_FRONTEND, which depends on WITH_BOOST_CONTEXT")
+    set(WITH_RADOSGW_BEAST_FRONTEND OFF)
+  endif()
 endif(WITH_RADOSGW)
 
 
@@ -542,6 +551,9 @@ set(BOOST_HEADER_COMPONENTS container)
 if(WITH_MGR)
 	list(APPEND BOOST_COMPONENTS python)
 endif()
+if(WITH_BOOST_CONTEXT)
+  list(APPEND BOOST_COMPONENTS context coroutine)
+endif()
 
 set(Boost_USE_MULTITHREADED ON)
 # require minimally the bundled version
diff --git a/ceph.spec.in b/ceph.spec.in
index 70ab1799eace..b4e10c68e347 100644
--- a/ceph.spec.in
+++ b/ceph.spec.in
@@ -846,9 +846,9 @@ cmake .. \
     -DWITH_OCF=ON \
 %endif
 %ifarch aarch64 armv7hl mips mipsel ppc ppc64 ppc64le %{ix86} x86_64
-    -DWITH_RADOSGW_BEAST_FRONTEND=ON \
+    -DWITH_BOOST_CONTEXT=ON \
 %else
-    -DWITH_RADOSGW_BEAST_FRONTEND=OFF \
+    -DWITH_BOOST_CONTEXT=OFF \
 %endif
     -DBOOST_J=%{_smp_ncpus}
 
diff --git a/src/include/config-h.in.cmake b/src/include/config-h.in.cmake
index 46beb891ed6d..7030a53bd8ca 100644
--- a/src/include/config-h.in.cmake
+++ b/src/include/config-h.in.cmake
@@ -321,4 +321,7 @@
 /* Defined if getentropy() is available */
 #cmakedefine HAVE_GETENTROPY
 
+/* Defined if boost::context is available */
+#cmakedefine HAVE_BOOST_CONTEXT
+
 #endif /* CONFIG_H */
