diff --git i/CMakeLists.txt w/CMakeLists.txt
index 50038d06..7241a0de 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -15,6 +15,7 @@ option(BESPOKE_DEVENDORED_SYSTEM_JUCE "Use system libraries when using system in
 option(BESPOKE_SYSTEM_PYBIND11 "Use a system installation of pybind11" OFF)
 option(BESPOKE_SYSTEM_JSONCPP "Use system-wide installation of jsoncpp" OFF)
 option(BESPOKE_SYSTEM_TUNING_LIBRARY "Use system installation of tuning-library" OFF)
+option(BESPOKE_SYSTEM_ABLETONLINK "Use system installation of ableton-link" OFF)
 option(BESPOKE_USE_ASAN "Build with ASAN" OFF)
 
 # Global CMake options
diff --git i/Source/CMakeLists.txt w/Source/CMakeLists.txt
index 7714ae7f..67256f1d 100644
--- i/Source/CMakeLists.txt
+++ w/Source/CMakeLists.txt
@@ -1,6 +1,10 @@
 include(cmake/lib.cmake)
 include(cmake/versiontools.cmake)
-include(../libs/link/AbletonLinkConfig.cmake)
+if(BESPOKE_SYSTEM_ABLETONLINK)
+    find_package(AbletonLink NAMES AbletonLink ableton-link link REQUIRED)
+else()
+    include(../libs/link/AbletonLinkConfig.cmake)
+endif()
 
 juce_add_gui_app(BespokeSynth
     PRODUCT_NAME BespokeSynth
