--- code/client/cl_curl.h.orig	2008-07-01 11:27:47.000000000 +0300
+++ code/client/cl_curl.h	2008-07-01 11:28:16.000000000 +0300
@@ -34,7 +34,7 @@
 #elif defined(MACOS_X)
 #define DEFAULT_CURL_LIB "libcurl.dylib"
 #else
-#define DEFAULT_CURL_LIB "libcurl.so.3"
+#define DEFAULT_CURL_LIB "libcurl.so"
 #endif
 
 #if USE_LOCAL_HEADERS
--- Makefile.orig	2008-07-01 14:40:01.000000000 +0300
+++ Makefile	2008-07-01 14:53:54.000000000 +0300
@@ -1129,12 +1129,12 @@
 $(B)/ioUrbanTerror.$(ARCH)$(BINEXT): $(Q3OBJ) $(Q3POBJ) $(LIBSDLMAIN)
 	$(echo_cmd) "LD $@"
 	$(Q)$(CC) -o $@ $(Q3OBJ) $(Q3POBJ) $(CLIENT_LDFLAGS) \
-		$(LDFLAGS) $(LIBSDLMAIN)
+		$(LDFLAGS) $(LIBSDLMAIN) -lcurl
 
 $(B)/ioUrbanTerror-smp.$(ARCH)$(BINEXT): $(Q3OBJ) $(Q3POBJ_SMP) $(LIBSDLMAIN)
 	$(echo_cmd) "LD $@"
 	$(Q)$(CC) -o $@ $(Q3OBJ) $(Q3POBJ_SMP) $(CLIENT_LDFLAGS) \
-		$(THREAD_LDFLAGS) $(LDFLAGS) $(LIBSDLMAIN)
+		$(THREAD_LDFLAGS) $(LDFLAGS) $(LIBSDLMAIN) -lcurl
 
 ifneq ($(strip $(LIBSDLMAIN)),)
 ifneq ($(strip $(LIBSDLMAINSRC)),)
