diff -Naur ioUrbanTerrorClientSource.orig/code/q3_ui/ui_mods.c ioUrbanTerrorClientSource/code/q3_ui/ui_mods.c
--- ioUrbanTerrorClientSource.orig/code/q3_ui/ui_mods.c	2007-10-09 02:47:14.000000000 +0300
+++ ioUrbanTerrorClientSource/code/q3_ui/ui_mods.c	2007-12-21 14:44:17.000000000 +0200
@@ -161,7 +161,7 @@
 
 	// always start off with baseq3
 	s_mods.list.numitems = 1;
-	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Quake III Arena";
+	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Urban Terror";
 	s_mods.fs_gameList[0] = "";
 
 	numdirs = trap_FS_GetFileList( "$modlist", "", dirlist, sizeof(dirlist) );

diff -Naur ioUrbanTerrorClientSource.orig/code/unix/unix_shared.c ioUrbanTerrorClientSource/code/unix/unix_shared.c
--- ioUrbanTerrorClientSource.orig/code/unix/unix_shared.c	2007-10-09 02:47:16.000000000 +0300
+++ ioUrbanTerrorClientSource/code/unix/unix_shared.c	2007-12-21 14:44:17.000000000 +0200
@@ -409,7 +409,7 @@
 #ifdef MACOS_X
 		Q_strcat(homePath, sizeof(homePath), "/Library/Application Support/Quake3");
 #else
-		Q_strcat(homePath, sizeof(homePath), "/.q3a");
+		Q_strcat(homePath, sizeof(homePath), "/.urbanterror");
 #endif
 		if (mkdir(homePath, 0777)) {
 			if (errno != EEXIST) 

diff -Naur ioUrbanTerrorServerSource.orig/code/q3_ui/ui_mods.c ioUrbanTerrorServerSource/code/q3_ui/ui_mods.c
--- ioUrbanTerrorServerSource.orig/code/q3_ui/ui_mods.c	2007-11-18 19:53:48.000000000 +0200
+++ ioUrbanTerrorServerSource/code/q3_ui/ui_mods.c	2007-12-21 14:46:30.000000000 +0200
@@ -125,7 +125,7 @@
 
 	// always start off with baseq3
 	s_mods.list.numitems = 1;
-	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Quake III Arena";
+	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Urban Terror";
 	s_mods.fs_gameList[0] = "";
 
 	numdirs = trap_FS_GetFileList( "$modlist", "", dirlist, sizeof(dirlist) );
   
diff -Naur ioUrbanTerrorServerSource.orig/code/sys/sys_unix.c ioUrbanTerrorServerSource/code/sys/sys_unix.c
--- ioUrbanTerrorServerSource.orig/code/sys/sys_unix.c	2007-12-03 02:37:34.000000000 +0200
+++ ioUrbanTerrorServerSource/code/sys/sys_unix.c	2007-12-21 14:47:47.000000000 +0200
@@ -55,7 +55,7 @@
 #ifdef MACOS_X
 			Q_strcat( homePath, sizeof( homePath ), "/Library/Application Support/Quake3" );
 #else
-			Q_strcat( homePath, sizeof( homePath ), "/.q3a" );
+			Q_strcat( homePath, sizeof( homePath ), "/.urbanterror" );
 #endif
 			if( mkdir( homePath, 0777 ) )
 			{
