diff -aur OpenAstroMenaceSVN.pristine/AstroMenaceSource/Core/RendererInterface/RendererInterface.h OpenAstroMenaceSVN.new/AstroMenaceSource/Core/RendererInterface/RendererInterface.h
--- OpenAstroMenaceSVN.pristine/AstroMenaceSource/Core/RendererInterface/RendererInterface.h	2010-02-14 11:39:06.721353373 +0100
+++ OpenAstroMenaceSVN.new/AstroMenaceSource/Core/RendererInterface/RendererInterface.h	2010-02-14 11:46:47.171304764 +0100
@@ -37,6 +37,8 @@
 #include "../Base.h"
 #include "../Math/Math.h"
 #include "../Texture/Texture.h"
+#define PFNGLCLIENTACTIVETEXTUREPROC PFNGLACTIVETEXTUREPROC
+
 
 
 
