--- tracker-0.8.17/configure.ac~	2010-09-02 16:26:20.000000000 +0200
+++ tracker-0.8.17/configure.ac	2010-12-29 11:51:10.848878668 +0100
@@ -1150,6 +1150,11 @@
 	      [enable_poppler_glib=auto])
 
 if test "x$enable_poppler_glib" != "xno" ; then
+   PKG_CHECK_EXISTS([poppler-glib >= 0.16], [have_poppler_0_16_0=yes])
+   if test "x$have_poppler_0_16_0" = "xyes"; then
+	   AC_DEFINE([HAVE_POPPLER_0_16_0], [1], [Define to 1 if you have Poppler version 0.16.0 or higher.])
+   fi
+
    PKG_CHECK_MODULES(POPPLER_GLIB,
    		     [poppler-glib >= $POPPLER_GLIB_REQUIRED],
 		     [have_poppler_glib=yes],
--- tracker-0.8.17/src/tracker-extract/tracker-extract-pdf.c~	2010-09-02 14:51:00.000000000 +0200
+++ tracker-0.8.17/src/tracker-extract/tracker-extract-pdf.c	2010-12-29 11:55:36.165545335 +0100
@@ -243,7 +243,11 @@
 		rect.x1 = rect.y1 = 0;
 		poppler_page_get_size (page, &rect.x2, &rect.y2);
 
+#if defined (HAVE_POPPLER_0_16_0)
+		text = poppler_page_get_selected_text (page, POPPLER_SELECTION_WORD, &rect);
+#else
 		text = poppler_page_get_text (page, POPPLER_SELECTION_WORD, &rect);
+#endif
 		t = tracker_text_normalize (text, n_words - words, &normalized_words);
 
 		words += normalized_words;
