--- quagga-0.99.15/zebra/Makefile.in	2010-01-13 01:34:25.265190407 +0100
+++ quagga-0.99.15/zebra/Makefile.in	2010-01-13 01:35:26.341161580 +0100
@@ -306,8 +306,8 @@
 	connected.h ioctl.h rib.h rt.h zserv.h redistribute.h debug.h rtadv.h \
 	interface.h ipforward.h irdp.h router-id.h kernel_socket.h
 
-zebra_LDADD = $(otherobj) $(LIBCAP) $(LIB_IPV6) ../lib/libzebra.la
-testzebra_LDADD = $(LIBCAP) $(LIB_IPV6) ../lib/libzebra.la
+zebra_LDADD = $(otherobj) ../lib/libzebra.la $(LIBCAP) $(LIB_IPV6)
+testzebra_LDADD = ../lib/libzebra.la $(LIBCAP) $(LIB_IPV6)
 zebra_DEPENDENCIES = $(otherobj)
 EXTRA_DIST = if_ioctl.c if_ioctl_solaris.c if_netlink.c if_proc.c \
         if_sysctl.c ipforward_aix.c ipforward_ews.c ipforward_proc.c \
--- quagga-0.99.15/zebra/Makefile.am	2010-01-13 01:34:31.473182415 +0100
+++ quagga-0.99.15/zebra/Makefile.am	2010-01-13 01:35:51.401207272 +0100
@@ -39,9 +39,9 @@
 	connected.h ioctl.h rib.h rt.h zserv.h redistribute.h debug.h rtadv.h \
 	interface.h ipforward.h irdp.h router-id.h kernel_socket.h
 
-zebra_LDADD = $(otherobj) $(LIBCAP) $(LIB_IPV6) ../lib/libzebra.la
+zebra_LDADD = $(otherobj) ../lib/libzebra.la $(LIBCAP) $(LIB_IPV6)
 
-testzebra_LDADD = $(LIBCAP) $(LIB_IPV6) ../lib/libzebra.la
+testzebra_LDADD = ../lib/libzebra.la $(LIBCAP) $(LIB_IPV6)
 
 zebra_DEPENDENCIES = $(otherobj)
 
