diff -ur quagga-0.99.14/zebra/interface.c quagga-0.99.14-fix-ipremove/zebra/interface.c
--- quagga-0.99.14/zebra/interface.c	2009-07-21 13:11:21.000000000 +0100
+++ quagga-0.99.14-fix-ipremove/zebra/interface.c	2009-08-12 17:04:14.000000000 +0100
@@ -1293,17 +1293,17 @@
       return CMD_WARNING;
     }
 
-#if 0
   /* Redistribute this information. */
   zebra_interface_address_delete_update (ifp, ifc);
 
   /* Remove connected route. */
   connected_down_ipv4 (ifp, ifc);
 
+  if_subnet_delete(ifp, ifc);
+
   /* Free address information. */
   listnode_delete (ifp->connected, ifc);
   connected_free (ifc);
-#endif
 
   return CMD_SUCCESS;
 }
