From 94300c75fc3e113009e68e2ab9db91c31e99e9f4 Mon Sep 17 00:00:00 2001
From: Pablo Neira Ayuso <pablo@netfilter.org>
Date: Mon, 20 Jan 2014 14:02:50 +0000
Subject: build: use libnftnl instead of libnftables in configure.in

Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
---
diff --git a/configure.ac b/configure.ac
index b38295f..9f0d894 100644
--- a/configure.ac
+++ b/configure.ac
@@ -53,7 +53,7 @@ fi
 AC_CHECK_LIB([mnl], [mnl_socket_open], ,
 	     AC_MSG_ERROR([No suitable version of libmnl found]))
 
-AC_CHECK_LIB([nftables], [nft_rule_alloc], ,
+AC_CHECK_LIB([nftnl], [nft_rule_alloc], ,
 	     AC_MSG_ERROR([No suitable version of libnftnl found]))
 
 AC_CHECK_LIB([gmp], [__gmpz_init], ,
--
cgit v0.9.2
