--- src/r8168_n.c	2012-04-08 22:15:51.740091625 +0200
+++ src/r8168_n.c.fix	2012-04-08 22:16:28.053425425 +0200
@@ -2232,16 +2232,6 @@
 	.set_settings		= rtl8168_set_settings,
 	.get_msglevel		= rtl8168_get_msglevel,
 	.set_msglevel		= rtl8168_set_msglevel,
-	.get_rx_csum		= rtl8168_get_rx_csum,
-	.set_rx_csum		= rtl8168_set_rx_csum,
-	.get_tx_csum		= rtl8168_get_tx_csum,
-	.set_tx_csum		= rtl8168_set_tx_csum,
-	.get_sg			= ethtool_op_get_sg,
-	.set_sg			= ethtool_op_set_sg,
-#ifdef NETIF_F_TSO
-	.get_tso		= ethtool_op_get_tso,
-	.set_tso		= ethtool_op_set_tso,
-#endif
 	.get_regs		= rtl8168_get_regs,
 	.get_wol		= rtl8168_get_wol,
 	.set_wol		= rtl8168_set_wol,
