From b96f859abde59decfa9dcf22f90302708f7d2e88 Mon Sep 17 00:00:00 2001
From: Pedro Castro <mail@pedrocastro.org>
Date: Mon, 21 Sep 2009 23:39:19 +0000
Subject: Use libtool to produce shared libraries (fixes #592784, with a patch from Peter Alfredsen)

---
diff --git a/Makefile.am b/Makefile.am
index e8d4ce5..ac4465c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,10 +3,6 @@ SUBDIRS = src data po help
 ACLOCAL_AMFLAGS = -I m4
 
 gstpbdir = $(pkglibdir)
-gstpb_SCRIPTS = build/libgstreamer_playbin.so
-
-$(gstpb_SCRIPTS):
-	cp -f src/libgstreamer_playbin.so build
 
 dist-hook:
 	@if test -d "$(srcdir)/.git"; \
diff --git a/configure.ac b/configure.ac
index d39acc0..e46fca8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6,7 +6,6 @@ GNOME_COMMON_INIT
 GNOME_DOC_INIT
 GTK_DOC_CHECK([1.0])
 
-
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR([configure.ac])
 
@@ -15,6 +14,8 @@ AM_MAINTAINER_MODE
 
 IT_PROG_INTLTOOL([0.35.0])
 
+AC_PROG_LIBTOOL
+
 AC_PROG_INSTALL
 AC_PREFIX_PROGRAM(mono)
 
diff --git a/src/Makefile.am b/src/Makefile.am
index c967d6e..fdb642c 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -8,10 +8,10 @@ ASSEMBLY = $(GS_BUILDDIR)/gnome-subtitles.exe
 ASSEMBLY_CONFIG = $(ASSEMBLY).config
 
 AM_CFLAGS = $(gstreamer_CFLAGS) -Wall -g -fPIC
-noinst_PROGRAMS = libgstreamer_playbin.so
-libgstreamer_playbin_so_SOURCES = $(srcdir)/External/GStreamerPlaybin/main.c
-libgstreamer_playbin_so_LDFLAGS = -shared -Wl,--as-needed,-soname,libgstreamer_playbin.so
-libgstreamer_playbin_so_LDADD = -lgstinterfaces-0.10 $(gstreamer_LIBS)
+gnomesubtitles_LTLIBRARIES = libgstreamer_playbin.la
+libgstreamer_playbin_la_SOURCES = $(srcdir)/External/GStreamerPlaybin/main.c
+libgstreamer_playbin_la_LIBADD = $(gstreamer_LIBS) -lgstinterfaces-0.10
+libgstreamer_playbin_la_LIBTOOLFLAGS = --tag=disable-static 
 
 GS_SOURCES = \
 	$(srcdir)/External/GStreamerPlaybin/*.cs \
--
cgit v0.8.2

