diff --git a/skgbasemodeler/skgdocument.cpp b/skgbasemodeler/skgdocument.cpp
index 5330d6c..88a7932 100644
--- a/skgbasemodeler/skgdocument.cpp
+++ b/skgbasemodeler/skgdocument.cpp
@@ -1386,7 +1386,8 @@ SKGError SKGDocument::load(const QString& iName, const QString& iPassword, bool
                         // Migrate to the last version of SQLCipher
                         IFOKDO(err, SKGServices::executeSqliteOrder(tryOpen, QStringLiteral("PRAGMA cipher_migrate")));
 
-                        // Test the password                        IFOKDO(err, SKGServices::executeSqliteOrder(tryOpen, QStringLiteral("SELECT count(*) FROM sqlite_master")));
+                        // Test the password
+                        IFOKDO(err, SKGServices::executeSqliteOrder(tryOpen, QStringLiteral("SELECT count(*) FROM sqlite_master")));
                         IFKO(err) {
                             SKGTRACEL(10) << "Wrong password in restore mode" << endl;
                             err = SKGError(ERR_ENCRYPTION, i18nc("Error message", "Wrong password"));
