diff -rupN a/IO/vtkFFMPEGWriter.cxx b/IO/vtkFFMPEGWriter.cxx
--- a/IO/vtkFFMPEGWriter.cxx	2013-07-22 18:29:24.625362910 +0000
+++ b/IO/vtkFFMPEGWriter.cxx	2013-07-22 18:39:16.184515101 +0000
@@ -149,7 +149,7 @@ int vtkFFMPEGWriterInternal::Start()
   
   //Set up the codec.
   AVCodecContext *c = this->avStream->codec;
-  c->codec_id = (CodecID)this->avOutputFormat->video_codec;
+  c->codec_id = (AVCodecID)this->avOutputFormat->video_codec;
   c->codec_type = AVMEDIA_TYPE_VIDEO;
   c->width = this->Dim[0];
   c->height = this->Dim[1];
@@ -199,7 +199,7 @@ int vtkFFMPEGWriterInternal::Start()
     vtkGenericWarningMacro (<< "Codec not found." );
     return 0;
     }
-  if (avcodec_open(c, codec) < 0) 
+  if (avcodec_open2(c, codec, NULL) < 0) 
     {
     vtkGenericWarningMacro (<< "Could not open codec.");
     return 0;
