--- VTK/Wrapping/Python/CMakeLists.txt.orig	2013-02-02 12:56:08.067142089 +0100
+++ VTK/Wrapping/Python/CMakeLists.txt	2013-02-02 12:58:16.182794679 +0100
@@ -558,7 +558,7 @@
 
     # Create default python setup arguments if they are not set.
     IF(NOT DEFINED VTK_PYTHON_SETUP_ARGS)
-      SET(VTK_PYTHON_SETUP_ARGS "--prefix=\"${DOLLAR}{CMAKE_INSTALL_PREFIX}\""
+      SET(VTK_PYTHON_SETUP_ARGS "--root=\"${DOLLAR}DESTDIR\" --prefix=\"${DOLLAR}{CMAKE_INSTALL_PREFIX}\""
         CACHE STRING "Arguments passed to \"python setup.py install ...\" during installation.")
       MARK_AS_ADVANCED(VTK_PYTHON_SETUP_ARGS)
     ENDIF(NOT DEFINED VTK_PYTHON_SETUP_ARGS)
