diff -Naur a/lib/PdfSharp/PdfSharp.Pdf.Content/CLexer.cs b/lib/PdfSharp/PdfSharp.Pdf.Content/CLexer.cs
--- a/lib/PdfSharp/PdfSharp.Pdf.Content/CLexer.cs	2018-01-13 01:43:50.475678949 +0100
+++ b/lib/PdfSharp/PdfSharp.Pdf.Content/CLexer.cs	2018-01-13 01:43:49.128974121 +0100
@@ -337,8 +337,7 @@
                     break;
 
                   case Chars.LF:
-                    ch = ScanNextChar();
-                    goto SkipChar;
+                    break;
 
                   default:
                     if (Char.IsDigit(ch))
@@ -436,8 +435,7 @@
                     break;
 
                   case Chars.LF:
-                    ch = ScanNextChar();
-                    goto SkipChar;
+                    break;
 
                   default:
                     if (Char.IsDigit(ch))
diff -Naur a/lib/PdfSharp/PdfSharp.Pdf.IO/Lexer.cs b/lib/PdfSharp/PdfSharp.Pdf.IO/Lexer.cs
--- a/lib/PdfSharp/PdfSharp.Pdf.IO/Lexer.cs	2010-08-27 18:53:58.000000000 +0200
+++ b/lib/PdfSharp/PdfSharp.Pdf.IO/Lexer.cs	2018-01-13 01:42:31.730117288 +0100
@@ -432,8 +432,7 @@
                     break;
 
                   case Chars.LF:
-                    ch = ScanNextChar();
-                    goto SkipChar;
+                    break;
 
                   default:
                     if (Char.IsDigit(ch))
@@ -552,8 +551,7 @@
                     break;
 
                   case Chars.LF:
-                    ch = ScanNextChar();
-                    goto SkipChar;
+                    break;
 
                   default:
                     if (Char.IsDigit(ch))
@@ -663,8 +661,7 @@
                     break;
 
                   case Chars.LF:
-                    ch = ScanNextChar();
-                    goto SkipChar;
+                    break;
 
                   default:
                     if (char.IsDigit(ch))
