diff -rupN a/plugins/gg/commands.c b/plugins/gg/commands.c
--- a/plugins/gg/commands.c	2011-03-17 18:08:11.000000000 +0100
+++ b/plugins/gg/commands.c	2013-07-29 18:45:51.296790942 +0200
@@ -1034,7 +1034,7 @@ static int token_gif_load (char *fname,
 		goto err;
 	}
 	
-	if (!(file = DGifOpenFileHandle(fd))) {
+	if (!(file = DGifOpenFileHandle(fd, NULL))) {
 		snprintf(errbuf, sizeof(errbuf), "DGifOpenFileHandle(): %d", 
 			GifLastError());
 		goto err2;
