diff --git a/drb.gemspec b/drb.gemspec
index c9d7e40..457ee6d 100644
--- a/drb.gemspec
+++ b/drb.gemspec
@@ -1,12 +1,6 @@
-begin
-  require_relative "lib/drb/version"
-rescue LoadError # Fallback to load version file in ruby core repository
-  require_relative "version"
-end
-
 Gem::Specification.new do |spec|
   spec.name          = "drb"
-  spec.version       = DRb::VERSION
+  spec.version       = "2.1.0"
   spec.authors       = ["Masatoshi SEKI"]
   spec.email         = ["seki@ruby-lang.org"]
 
