diff -wbBur gnunet-0.9.0pre2/src/transport/plugin_transport_http.c gnunet-0.9.0pre2.my/src/transport/plugin_transport_http.c
--- gnunet-0.9.0pre2/src/transport/plugin_transport_http.c	2010-12-23 09:37:50.000000000 +0000
+++ gnunet-0.9.0pre2.my/src/transport/plugin_transport_http.c	2011-03-24 07:59:13.000000000 +0000
@@ -1077,7 +1077,7 @@
 
     conn_info = MHD_get_connection_info(mhd_connection, MHD_CONNECTION_INFO_CLIENT_ADDRESS );
     /* Incoming IPv4 connection */
-    if ( AF_INET == conn_info->client_addr->sin_family)
+    if ( AF_INET == ((struct sockaddr_in*)(conn_info->client_addr))->sin_family)
     {
       addrin = conn_info->client_addr;
       inet_ntop(addrin->sin_family, &(addrin->sin_addr),address,INET_ADDRSTRLEN);
@@ -1087,7 +1087,7 @@
       addr_len = sizeof(struct IPv4HttpAddress);
     }
     /* Incoming IPv6 connection */
-    if ( AF_INET6 == conn_info->client_addr->sin_family)
+    if ( AF_INET6 == ((struct sockaddr_in6*)(conn_info->client_addr))->sin6_family)
     {
       addrin6 = (struct sockaddr_in6 *) conn_info->client_addr;
       inet_ntop(addrin6->sin6_family, &(addrin6->sin6_addr),address,INET6_ADDRSTRLEN);
