--- navit-0.5.3/navit/vehicle/gpsd/vehicle_gpsd.c.orig	2018-10-11 11:15:18.019205542 +0000
+++ navit-0.5.3/navit/vehicle/gpsd/vehicle_gpsd.c	2018-10-11 11:15:36.329237820 +0000
@@ -325,7 +325,7 @@
         int read_result;
         /* Read until EOF, in case we are lagging behind.
          * No point in processing old GPS reports. */
-        while((read_result=gps_read(priv->gps))>0);
+        while((read_result=gps_read(priv->gps, NULL, 0))>0);
         if(read_result==-1) {
             dbg(lvl_error,"gps_poll failed");
             vehicle_gpsd_close(priv);
