diff -u -r libdesktop-agnostic-0.3.92/libdesktop-agnostic/wscript libdesktop-agnostic-0.3.92-gir/libdesktop-agnostic/wscript
--- libdesktop-agnostic-0.3.92/libdesktop-agnostic/wscript	2011-03-12 15:14:23.000000000 +0100
+++ libdesktop-agnostic-0.3.92-gir/libdesktop-agnostic/wscript	2011-03-27 18:51:13.170000225 +0200
@@ -67,8 +67,9 @@
 
 
 def create_typelib_task():
-    rule = 'LD_LIBRARY_PATH="${CURRENT_BUILD_DIR}" ${G_IR_COMPILER} ' + \
-           '-o ${TGT} ${SRC}'
+    rule = '${G_IR_COMPILER}' + \
+           ' --includedir "${CURRENT_BUILD_DIR}"' + \
+           ' -o ${TGT} ${SRC}'
     Task.simple_task_type('typelib', rule, after='cc_link')
 
 create_typelib_task()
diff -u -r libdesktop-agnostic-0.3.92/wscript libdesktop-agnostic-0.3.92-gir/wscript
--- libdesktop-agnostic-0.3.92/wscript	2011-03-13 16:09:47.000000000 +0100
+++ libdesktop-agnostic-0.3.92-gir/wscript	2011-03-27 18:51:46.973335650 +0200
@@ -71,7 +71,7 @@
                    help='Enables the library to be built so that it is '
                         'instrumented to measure performance.')
     opt.add_option('--disable-gi', action='store_true',
-                   dest='no_gi', default=True)
+                   dest='no_gi', default=False)
 
 
 def configure(conf):
