--- glob2-0.9.4.4/SConstruct.orig	2019-01-15 14:58:36.130370433 +0100
+++ glob2-0.9.4.4/SConstruct	2019-01-15 15:03:13.040240416 +0100
@@ -13,7 +13,7 @@
 
 
 def establish_options(env):
-    opts = Options('options_cache.py')
+    opts = Variables('options_cache.py')
     opts.Add("CXXFLAGS", "Manually add to the CXXFLAGS", "-g")
     opts.Add("LINKFLAGS", "Manually add to the LINKFLAGS", "-g")
     if isDarwinPlatform:
@@ -22,10 +22,10 @@
 	    opts.Add("INSTALLDIR", "Installation Directory", "/usr/local/share")
     opts.Add("BINDIR", "Binary Installation Directory", "/usr/local/bin")
     opts.Add("DATADIR", "Directory where data will be put, set to the same as INSTALLDIR", "/usr/local/share")
-    opts.Add(BoolOption("release", "Build for release", 0))
-    opts.Add(BoolOption("profile", "Build with profiling on", 0))
-    opts.Add(BoolOption("mingw", "Build with mingw enabled if not auto-detected", 0))
-    opts.Add(BoolOption("server", "Build only the YOG server, excluding the game and any GUI/sound components", 0))
+    opts.Add(BoolVariable("release", "Build for release", 0))
+    opts.Add(BoolVariable("profile", "Build with profiling on", 0))
+    opts.Add(BoolVariable("mingw", "Build with mingw enabled if not auto-detected", 0))
+    opts.Add(BoolVariable("server", "Build only the YOG server, excluding the game and any GUI/sound components", 0))
     opts.Add("font", "Build the game using an alternative font placed in the data/font folder", "sans.ttf")
     Help(opts.GenerateHelpText(env))
     opts.Update(env)
