--- Term-ReadLine-Gnu-1.20/Makefile.PL	2010-05-02 06:37:55.000000000 -0400
+++ Term-ReadLine-Gnu-1.20-patched/Makefile.PL	2011-07-30 16:16:54.000000000 -0400
@@ -58,8 +58,13 @@
     # know why AIX prefers curses.
     # libtermcap.a on HPUX cannot be used for dynamically linked binary.
     # Old Cygwin may require setting false (0).
-    my $PREFER_CURSES = $Config{osname} eq 'aix' || $Config{osname} eq 'hpux'
-			|| $Config{osname} eq 'cygwin';
+    # my $PREFER_CURSES = $Config{osname} eq 'aix' || $Config{osname} eq 'hpux'
+	# 		|| $Config{osname} eq 'cygwin';
+
+    # Hack for ArchLinux by Justin Davis <aurperl@juster.us>
+    # Don't use termcap! readline is linked with ncurses so I know it's
+    # available. termcap causes this module to crash for some people.
+    my $PREFER_CURSES = 1;
     my $TERMCAP_LIB = (! $PREFER_CURSES && &search_lib('-ltermcap'))
 	|| &search_lib('-lncurses')
 	    || &search_lib('-lcurses');
