Index: src/VBox/Additions/linux/lightdm-greeter/vbox-greeter.cpp
===================================================================
--- a/src/VBox/Additions/linux/lightdm-greeter/vbox-greeter.cpp	(revision 57689)
+++ b/src/VBox/Additions/linux/lightdm-greeter/vbox-greeter.cpp	(revision 57690)
@@ -842,7 +842,7 @@
  * @param   enmPhase
  * @param   pfnLog
  */
-static void vboxGreeterLogHeaderFooter(PRTLOGGER pLoggerRelease, RTLOGPHASE enmPhase, PFNRTLOGPHASEMSG pfnLog)
+static DECLCALLBACK(void) vboxGreeterLogHeaderFooter(PRTLOGGER pLoggerRelease, RTLOGPHASE enmPhase, PFNRTLOGPHASEMSG pfnLog)
 {
     /* Some introductory information. */
     static RTTIMESPEC s_TimeSpec;
Index: src/VBox/ExtPacks/VNC/VBoxVNC.cpp
===================================================================
--- a/src/VBox/ExtPacks/VNC/VBoxVNC.cpp	(revision 57657)
+++ b/src/VBox/ExtPacks/VNC/VBoxVNC.cpp	(revision 57658)
@@ -103,8 +103,8 @@
     unsigned char *mScreenBuffer;
     unsigned char *mFrameBuffer;
     uint32_t uClients;
-    static DECLCALLBACK(enum rfbNewClientAction) rfbNewClientEvent(rfbClientPtr cl);
-    static DECLCALLBACK(void) vncMouseEvent(int buttonMask, int x, int y, rfbClientPtr cl);
+    static enum rfbNewClientAction rfbNewClientEvent(rfbClientPtr cl);
+    static void vncMouseEvent(int buttonMask, int x, int y, rfbClientPtr cl);
     static void vncKeyboardEvent(rfbBool down, rfbKeySym keySym, rfbClientPtr cl);
     static void clientGoneHook(rfbClientPtr cl);
 
