diff -wbBur gtkdialog-0.7.20/src/glade_support.c gtkdialog-0.7.20.my/src/glade_support.c
--- gtkdialog-0.7.20/src/glade_support.c	2007-03-16 14:25:49.000000000 +0000
+++ gtkdialog-0.7.20.my/src/glade_support.c	2010-07-08 11:05:46.000000000 +0000
@@ -40,7 +40,7 @@
 typedef struct signal {
 	gchar     *name;
 	GCallback callback;
-} signal;
+} xsignal;
 
 /*
 ** Signal handler callbascks.
@@ -206,7 +206,7 @@
 static gboolean
 find_and_connect_handler(
 		GtkWidget *widget,
-		signal *signals,
+		xsignal *signals,
 		const gchar *signal_name,
 		const gchar *handler_name)
 {
@@ -288,7 +288,7 @@
 		gpointer user_data)
 {
 	gint n;
-	signal entry_signals[] = {
+	xsignal entry_signals[] = {
 		{ "activate",           (GCallback)on_any_entry_almost_any },
 		{ "backspace",          (GCallback)on_any_entry_almost_any },
 		{ "copy-clipboard",     (GCallback)on_any_entry_almost_any },
@@ -319,7 +319,7 @@
 		gpointer user_data)
 {
 	variable *var;
-	signal widget_signals[] = {
+	xsignal widget_signals[] = {
 		{ "accel-closures-changed",    (GCallback)on_any_widget_almost_any },
 		{ "composited-changed",        (GCallback)on_any_widget_almost_any },
 		{ "grab-focus",                (GCallback)on_any_widget_almost_any },
