diff -Nur ipaddr-2.1.10.orig/ipaddr.py ipaddr-2.1.10/ipaddr.py
--- ipaddr-2.1.10.orig/ipaddr.py	2012-01-21 14:03:17.000000000 +1100
+++ ipaddr-2.1.10/ipaddr.py	2012-03-27 18:58:32.792042210 +1100
@@ -1462,7 +1462,7 @@
 
         try:
             # Now, parse the hextets into a 128-bit integer.
-            ip_int = 0L
+            ip_int = 0
             for i in xrange(parts_hi):
                 ip_int <<= 16
                 ip_int |= self._parse_hextet(parts[i])
