diff --git a/src/AIEcho.cpp b/src/AIEcho.cpp
index 77d9f63..0e5f59e 100644
--- a/src/AIEcho.cpp
+++ b/src/AIEcho.cpp
@@ -4387,7 +4387,7 @@ bool enemy_building_iterator::operator!=(const enemy_building_iterator& rhs) con
 {
 	if(is_end && rhs.is_end)
 		return false;
-	return is_end!=rhs.is_end || team!=rhs.team || building_type!=rhs.building_type || level!=rhs.level || construction_site!=rhs.construction_site;
+	return bool {is_end!=rhs.is_end || team!=rhs.team || building_type!=rhs.building_type || level!=rhs.level || construction_site!=rhs.construction_site};
 }
 
 
