--- gecko-dev-042b84af6020b1f2d8029a0dc36ac5955b7f325f.orig/media/libstagefright/moz.build	2020-12-19 15:42:48.971874785 +0000
+++ gecko-dev-042b84af6020b1f2d8029a0dc36ac5955b7f325f/media/libstagefright/moz.build	2020-12-19 15:33:40.415241186 +0000
@@ -162,6 +162,7 @@ elif CONFIG['GNU_CXX']:
     ]
     CXXFLAGS += [
         '-Werror=format',
+        '-Wno-error=format-overflow',
         '-Wno-multichar',
         '-Wno-sign-compare',
         '-Wno-unused',
