diff -ruN a/configure.ac b/configure.ac
--- a/configure.ac	2020-12-28 01:32:57.000000000 +0100
+++ b/configure.ac	2021-09-29 21:12:24.897000721 +0200
@@ -394,9 +394,7 @@
   portmidi=no])
 AS_IF([test x$portmidi = xyes],[
     AS_IF([test x$local_portmidi = xno],[
-        # fall back to local portmidi if not foud
-        AC_CHECK_LIB([portmidi], [Pm_Initialize],
-            [AC_MSG_NOTICE([Using system PortMidi])], [local_portmidi=yes])
+        PKG_CHECK_MODULES([LIBPORTMIDI], [portmidi])
     ])
     AS_IF([test x$local_portmidi = xyes],[
         AS_IF([test -d "$srcdir/portmidi"],[
diff -ruN a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am	2020-12-28 01:32:57.000000000 +0100
+++ b/src/Makefile.am	2021-09-29 21:12:53.977088547 +0200
@@ -221,7 +221,7 @@
 pd_LDADD_core += $(top_builddir)/portmidi/libportmidi.a
 else
 # link the system's portmidi
-pd_LDADD_core += -lportmidi
+pd_LDADD_core += @LIBPORTMIDI_LIBS@
 endif
 
 # need Carbon framework for PM on Mac
