diff -wbBur dangerdeep-0.3.0/src/bspline_test.cpp dangerdeep-0.3.0.my/src/bspline_test.cpp
--- dangerdeep-0.3.0/src/bspline_test.cpp	2007-06-11 15:16:23.000000000 +0000
+++ dangerdeep-0.3.0.my/src/bspline_test.cpp	2010-05-31 09:12:09.000000000 +0000
@@ -1,6 +1,8 @@
 // some test code for the 2d bsplines!
 #include "bspline.h"
 #include <fstream>
+#include <cstdlib>
+
 using namespace std;
 
 double rnd() { return double(rand())/RAND_MAX; }
diff -wbBur dangerdeep-0.3.0/src/convoy.h dangerdeep-0.3.0.my/src/convoy.h
--- dangerdeep-0.3.0/src/convoy.h	2007-06-11 15:16:23.000000000 +0000
+++ dangerdeep-0.3.0.my/src/convoy.h	2010-05-31 09:05:06.000000000 +0000
@@ -26,6 +26,7 @@
 #include "ai.h"
 #include "vector2.h"
 #include <new>
+#include <memory>
 #include <list>
 class ship;
 
diff -wbBur dangerdeep-0.3.0/src/date.cpp dangerdeep-0.3.0.my/src/date.cpp
--- dangerdeep-0.3.0/src/date.cpp	2007-06-11 15:16:23.000000000 +0000
+++ dangerdeep-0.3.0.my/src/date.cpp	2010-05-31 09:06:12.000000000 +0000
@@ -20,6 +20,8 @@
 // date
 // subsim (C)+(W) Markus Petermann and Thorsten Jordan. SEE LICENSE
 
+#include <stdio.h>
+#include <stdlib.h>
 #include <iomanip>
 #include "date.h"
 #include "texts.h"
diff -wbBur dangerdeep-0.3.0/src/filehelper.cpp dangerdeep-0.3.0.my/src/filehelper.cpp
--- dangerdeep-0.3.0/src/filehelper.cpp	2007-06-11 15:16:23.000000000 +0000
+++ dangerdeep-0.3.0.my/src/filehelper.cpp	2010-05-31 09:08:34.000000000 +0000
@@ -23,6 +23,7 @@
 #include "filehelper.h"
 #include "error.h"
 #include <vector>
+#include <stdio.h>
 using namespace std;
 
 #ifdef WIN32
diff -wbBur dangerdeep-0.3.0/src/ocean_wave_generator.h dangerdeep-0.3.0.my/src/ocean_wave_generator.h
--- dangerdeep-0.3.0/src/ocean_wave_generator.h	2007-06-11 15:16:23.000000000 +0000
+++ dangerdeep-0.3.0.my/src/ocean_wave_generator.h	2010-05-31 09:11:24.000000000 +0000
@@ -29,6 +29,7 @@
 #include "environment.h"
 #include <complex>
 #include <vector>
+#include <cstdlib>
 
 // use float fftw (faster) or double (default) ?
 #ifdef WITH_FLOAT_FFTW
diff -wbBur dangerdeep-0.3.0/src/ptrlist.h dangerdeep-0.3.0.my/src/ptrlist.h
--- dangerdeep-0.3.0/src/ptrlist.h	2007-06-11 15:16:23.000000000 +0000
+++ dangerdeep-0.3.0.my/src/ptrlist.h	2010-05-31 09:03:50.000000000 +0000
@@ -25,6 +25,7 @@
 
 #include <list>
 #include <stdexcept>
+#include <memory>
 
 // same as std::list regarding the interface (partly), but handles pointers.
 template <class T>
diff -wbBur dangerdeep-0.3.0/src/ptrvector.h dangerdeep-0.3.0.my/src/ptrvector.h
--- dangerdeep-0.3.0/src/ptrvector.h	2007-06-11 15:16:23.000000000 +0000
+++ dangerdeep-0.3.0.my/src/ptrvector.h	2010-05-31 09:04:21.000000000 +0000
@@ -25,6 +25,7 @@
 
 #include <vector>
 #include <stdexcept>
+#include <memory>
 
 // same as std::vector regarding the interface, but handles pointers.
 template <class T>
diff -wbBur dangerdeep-0.3.0/src/widget.cpp dangerdeep-0.3.0.my/src/widget.cpp
--- dangerdeep-0.3.0/src/widget.cpp	2007-06-11 15:16:23.000000000 +0000
+++ dangerdeep-0.3.0.my/src/widget.cpp	2010-05-31 09:09:40.000000000 +0000
@@ -32,6 +32,8 @@
 #include "datadirs.h"
 #include <set>
 #include <sstream>
+#include <algorithm>
+
 using std::vector;
 using std::list;
 using std::string;
