From 3172dea3fcd07a5311635d6f7c3af691eefe5acd Mon Sep 17 00:00:00 2001
From: Shelikhoo <xiaokangwang@outlook.com>
Date: Tue, 13 Apr 2021 18:15:06 +0100
Subject: [PATCH]  fix not creating test dir in some environment

---
 infra/conf/router_test.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/infra/conf/router_test.go b/infra/conf/router_test.go
index b594f1ca..1b7578d1 100644
--- a/infra/conf/router_test.go
+++ b/infra/conf/router_test.go
@@ -28,9 +28,10 @@ func init() {
 
 	os.Setenv("v2ray.location.asset", tempPath)
 
+	common.Must(os.MkdirAll(tempPath, 0755))
+
 	if _, err := os.Stat(platform.GetAssetLocation("geoip.dat")); err != nil && errors.Is(err, fs.ErrNotExist) {
 		if _, err := os.Stat(geoipPath); err != nil && errors.Is(err, fs.ErrNotExist) {
-			common.Must(os.MkdirAll(tempPath, 0755))
 			geoipBytes, err := common.FetchHTTPContent(geoipURL)
 			common.Must(err)
 			common.Must(filesystem.WriteFile(geoipPath, geoipBytes))
-- 
2.27.0

