diff -wbBur rsyslog-8.4.0/runtime/msg.c rsyslog-8.4.0.my/runtime/msg.c
--- rsyslog-8.4.0/runtime/msg.c	2014-08-18 13:14:06.000000000 +0400
+++ rsyslog-8.4.0.my/runtime/msg.c	2014-08-22 15:58:52.974045109 +0400
@@ -4074,7 +4074,11 @@
 
 			err = tokener->err;
 			if(err != json_tokener_continue)
+#				if HAVE_JSON_TOKENER_ERROR_DESC
+					errMsg = json_tokener_error_desc(err);
+#				else
 				errMsg = json_tokener_errors[err];
+#				endif
 			else
 				errMsg = "Unterminated input";
 		} else if(!json_object_is_type(json, json_type_object))
