diff -Naur ubuntuone-client-gnome-4.2.0.orig/configure.ac ubuntuone-client-gnome-4.2.0/configure.ac
--- ubuntuone-client-gnome-4.2.0.orig/configure.ac	2013-03-26 20:48:40.000000000 +0100
+++ ubuntuone-client-gnome-4.2.0/configure.ac	2013-04-09 22:26:38.135983473 +0200
@@ -5,7 +5,7 @@
 AC_CONFIG_SRCDIR([config.h.in])
 
 AM_INIT_AUTOMAKE([1.10 foreign])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 
diff -Naur ubuntuone-client-gnome-4.2.0.orig/Makefile.am ubuntuone-client-gnome-4.2.0/Makefile.am
--- ubuntuone-client-gnome-4.2.0.orig/Makefile.am	2013-03-21 18:13:58.000000000 +0100
+++ ubuntuone-client-gnome-4.2.0/Makefile.am	2013-04-09 22:22:05.109626589 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = gsd nautilus po
+SUBDIRS = nautilus po
 
 # Shut libtoolize up
 ACLOCAL_AMFLAGS = -I m4
