--- SConstruct.original	2011-07-05 14:23:46.334748007 +0000
+++ SConstruct	2011-07-05 14:26:15.729533840 +0000
@@ -208,6 +208,9 @@
 justClientLib = (COMMAND_LINE_TARGETS == ['mongoclient'])
 
 env = Environment( MSVS_ARCH=msarch , tools = ["default", "gch"], toolpath = '.' )
+
+env.ParseConfig('pkg-config --cflags --libs mozjs185')
+
 if has_option( "cxx" ):
     env["CC"] = get_option( "cxx" )
     env["CXX"] = get_option( "cxx" )
