--- configure.ac.orig	2014-08-17 17:28:46.314141026 -0400
+++ configure.ac	2014-08-17 17:29:21.084084594 -0400
@@ -71,64 +71,7 @@
 
 
 
-dnl Check for WXWidgets
-WX_VER_291=""
 
-AC_PATH_PROG(WX_CONFIG, wx-config, no)
-
-if test x$WX_CONFIG != xno; then
-   WX_VERSION=`wx-config --version`
-
-   WX_MAJOR=${WX_VERSION%%.*};
-   WX_MINOR=${WX_VERSION#*.};
-   WX_MINOR=${WX_MINOR%%.*};
-   WX_MICRO=${WX_VERSION##*.};
-
-   if test $WX_MAJOR -gt 2; then
-      WX_VER_291=yes
-   else
-     if test $WX_MAJOR -eq 2; then
-        if test $WX_MINOR -gt 9; then
-           WX_VER_291=yes
-        else
-           if test $WX_MINOR -eq 9; then
-              if test $WX_MICRO -ge 1; then
-                 WX_VER_291=yes
-              fi
-           fi
-        fi
-     fi
-  fi
-fi
-
-if test "$WX_VER_291" != "yes"; then
-
-dnl Use static version
-echo "installed version is too old,"
-echo "using static source version $reqwx"
-
-dnl Need to reset this, else it will override the value below !
-WX_CONFIG=""
-
-AC_CHECK_PROG(WX_CONFIG, wx-config, yes, no)
-
-if test $WX_CONFIG = no; then
-echo
-echo YOU CAN FIX THIS !
-AC_MSG_ERROR([Please install development libraries for wxWidgets 2.9.1 or higher, and then run this configure script again.])
-fi
-fi
-
-WX_CXXFLAGS="`wx-config --cxxflags`"
-WX_LIBS="-static `wx-config --libs` `wx-config --libs richtext`"
-
-echo "WX_CXXFLAGS is $WX_CXXFLAGS"
-echo "WX_LIBS is $WX_LIBS"
-
-WX_LDADD="$WX_LIBS"
-
-AC_SUBST(WX_CXXFLAGS)
-AC_SUBST(WX_LDADD)
 
 
 
--- MSVC/SettingsGUI/Makefile.am.orig	2014-08-17 20:22:59.680508699 -0400
+++ MSVC/SettingsGUI/Makefile.am	2014-08-17 20:23:46.397099541 -0400
@@ -28,7 +28,6 @@
 	-I /usr/include/libswscale \
 	-I /usr/include/libavutil
 
-bin_PROGRAMS = electricsheep-preferences
 
 electricsheep_preferences_SOURCES = \
 	config.cpp \
